{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "75c4e381",
   "metadata": {},
   "source": [
    "# Import module"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14a33d3e",
   "metadata": {},
   "source": [
    "The link to get [ImageAnalysis3](https://github.com/zhengpuas47/ImageAnalysis3) \n",
    "\n",
    "Or from the Zhuang lab archived [source_tools](https://github.com/ZhuangLab/Chromatin_Analysis_2020_cell/tree/master/sequential_tracing/source)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b411cc21",
   "metadata": {},
   "source": [
    "## ImageAnalysis3 and basic modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "32554ab1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31848\n"
     ]
    }
   ],
   "source": [
    "%run \"C:\\Users\\shiwei\\Documents\\ImageAnalysis3\\required_files\\Startup_py3.py\"\n",
    "sys.path.append(r\"C:\\Users\\shiwei\\Documents\")\n",
    "\n",
    "import ImageAnalysis3 as ia\n",
    "from ImageAnalysis3 import *\n",
    "from ImageAnalysis3.classes import _allowed_kwds\n",
    "\n",
    "import h5py\n",
    "import ast\n",
    "import pandas as pd\n",
    "\n",
    "print(os.getpid())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5f8cd77",
   "metadata": {},
   "source": [
    "## Chromatin_analysis_tools etc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2db5036f",
   "metadata": {},
   "source": [
    "See **functions** in the repository for [AnalysisTool_Chromatin](../../README.md)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9469aba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'atac_to_loci' from 'C:\\\\Users\\\\shiwei\\\\Documents\\\\AnalysisTool_Chromatin\\\\atac_to_loci.py'>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Chromatin_analysis_tools (ATC)\n",
    "# Get path for the py containing functions\n",
    "import os\n",
    "import sys\n",
    "import importlib\n",
    "module_path =r'C:\\Users\\shiwei\\Documents\\AnalysisTool_Chromatin'\n",
    "if module_path not in sys.path:\n",
    "    sys.path.append(module_path)\n",
    "    \n",
    "# import relevant modules\n",
    "import gene_selection \n",
    "importlib.reload(gene_selection)\n",
    "import gene_to_loci\n",
    "importlib.reload(gene_to_loci)\n",
    "import gene_activity\n",
    "importlib.reload(gene_activity)\n",
    "import loci_1d_features\n",
    "importlib.reload(loci_1d_features)  \n",
    "\n",
    "import atac_to_loci\n",
    "importlib.reload(atac_to_loci)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa08c6de",
   "metadata": {},
   "source": [
    "# Define folders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2c8dce53",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use existing analysis folder: L:\\Shiwei\\postanalysis_2024\\v0\\chromosome_scaling\\analysis.\n",
      "Use existing figure folder: L:\\Shiwei\\postanalysis_2024\\v0\\chromosome_scaling\\figures.\n"
     ]
    }
   ],
   "source": [
    "# main folder for postanalysis\n",
    "postanalysis_folder = r'L:\\Shiwei\\postanalysis_2024\\v0'\n",
    "# input files for postanalysis\n",
    "input_folder = os.path.join(postanalysis_folder, 'resources_from_preprocess')\n",
    "\n",
    "# output file to be generated\n",
    "output_main_folder = os.path.join(postanalysis_folder, 'chromosome_scaling')\n",
    "\n",
    "output_analysis_folder = os.path.join(output_main_folder, 'analysis')\n",
    "output_figure_folder = os.path.join(output_main_folder, 'figures')\n",
    "\n",
    "# make new folder if needed\n",
    "make_output_folder = True\n",
    "\n",
    "if make_output_folder and not os.path.exists(output_analysis_folder):\n",
    "    os.makedirs(output_analysis_folder)\n",
    "    print(f'Generating analysis folder: {output_analysis_folder}.')\n",
    "elif os.path.exists(output_analysis_folder):\n",
    "    print(f'Use existing analysis folder: {output_analysis_folder}.')\n",
    "    \n",
    "if make_output_folder and not os.path.exists(output_figure_folder):\n",
    "    os.makedirs(output_figure_folder)\n",
    "    print(f'Generating figure folder: {output_figure_folder}.')\n",
    "elif os.path.exists(output_figure_folder):\n",
    "    print(f'Use existing figure folder: {output_figure_folder}.')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43745cf2",
   "metadata": {},
   "source": [
    "# Plotting parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fb1e3324",
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rc('font', family='serif')\n",
    "plt.rc('font', serif='Arial')\n",
    "\n",
    "from ImageAnalysis3.figure_tools import _double_col_width, _single_col_width, _font_size, _ticklabel_size,_ticklabel_width\n",
    "\n",
    "import seaborn as sns\n",
    "sns.set_context(\"paper\", rc={\"font.size\":_font_size,\"axes.titlesize\":_font_size+1,\"axes.labelsize\":_font_size})  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bc310b7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Other required plotting parameters\n",
    "_dpi = 300\n",
    "_font_size = 7\n",
    "_page_width = 5.5\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1d34726",
   "metadata": {},
   "source": [
    "## cell type color-codess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c33943dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cell labels from RNA-MERFISH and celltype prediction\n",
    "selected_cell_labels = ['L2/3 IT','L4/5 IT','L5 IT','L6 IT','L5 ET','L5/6 NP','L6 CT','L6b',\n",
    "                           'Sst','Pvalb','Lamp5','Sncg','Vip',\n",
    "                           'Astro','Oligo','OPC','Micro','Endo','VLMC','SMC','Peri', \n",
    "                           #'other',\n",
    "                          ]\n",
    "# cell palette from RNA-MERFISH UMAP and stats\n",
    "celltype_palette = {'Astro':'lightcoral', \n",
    "                    'Endo':'skyblue', \n",
    "                    'L2/3 IT':'gold', \n",
    "                    'L4/5 IT':'darkorange', \n",
    "                    'L5 ET':'mediumseagreen', \n",
    "                    'L5 IT':'aqua',\n",
    "                    'L5/6 NP':'darkgreen',\n",
    "                    'L6 CT':'brown',\n",
    "                    'L6 IT':'magenta',\n",
    "                    'L6b':'blue', \n",
    "                    'Lamp5':'orange', \n",
    "                    'Micro':'peachpuff',\n",
    "                    'OPC':'thistle', \n",
    "                    'Oligo':'darkviolet',\n",
    "                    'Peri':'sandybrown',\n",
    "                    'Pvalb':'springgreen',\n",
    "                    'SMC':'rosybrown',\n",
    "                    'Sncg':'darkkhaki',\n",
    "                    'Sst':'steelblue', \n",
    "                    'VLMC':'saddlebrown', \n",
    "                    'Vip':'red',\n",
    "                    'other':'slategray'}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "caad6887",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this is the plotting order noted based on the snRNA transcriptional acitivty if needed\n",
    "sorted_cellplot_order_byRNA = ['Micro', 'Oligo', 'Endo', 'OPC', 'Astro', 'Vip', 'Lamp5',\n",
    "                  'L5/6 NP', 'Sst', 'Sncg', 'Pvalb', 'L4/5 IT', 'L6 CT',\n",
    "                  'L6 IT', 'L6b', 'L2/3 IT', 'L5 IT', 'L5 ET']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b00f58f0",
   "metadata": {},
   "source": [
    "# Load data relevant information"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e667606",
   "metadata": {},
   "source": [
    "## load and format codebook"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b8440c6",
   "metadata": {},
   "source": [
    "[merged codebook](../resources/merged_codebook.csv) as in the repository (merged for all DNA-MERFISH libraries)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c0734aad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>NDB_784</th>\n",
       "      <th>NDB_755</th>\n",
       "      <th>NDB_826</th>\n",
       "      <th>NDB_713</th>\n",
       "      <th>NDB_865</th>\n",
       "      <th>NDB_725</th>\n",
       "      <th>NDB_817</th>\n",
       "      <th>NDB_710</th>\n",
       "      <th>...</th>\n",
       "      <th>NDB_479</th>\n",
       "      <th>NDB_562</th>\n",
       "      <th>NDB_608</th>\n",
       "      <th>NDB_460</th>\n",
       "      <th>NDB_563</th>\n",
       "      <th>NDB_592</th>\n",
       "      <th>NDB_368</th>\n",
       "      <th>NDB_436</th>\n",
       "      <th>NDB_629</th>\n",
       "      <th>NDB_604</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 228 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   name id  NDB_784  NDB_755  NDB_826  NDB_713  NDB_865  \\\n",
       "0     1:3742742-3759944  1      0.0      0.0      0.0      0.0      0.0   \n",
       "1     1:6245958-6258969  2      0.0      0.0      0.0      0.0      0.0   \n",
       "2     1:8740008-8759916  3      0.0      0.0      0.0      0.0      0.0   \n",
       "1016  1:9627926-9637875  1      0.0      0.0      0.0      0.0      0.0   \n",
       "1017  1:9799472-9811359  2      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "      NDB_725  NDB_817  NDB_710  ...  NDB_479  NDB_562  NDB_608  NDB_460  \\\n",
       "0         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "2         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1016      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1017      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "\n",
       "      NDB_563  NDB_592  NDB_368  NDB_436  NDB_629  NDB_604  \n",
       "0         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "2         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1016      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1017      0.0      1.0      0.0      0.0      0.0      0.0  \n",
       "\n",
       "[5 rows x 228 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load codebook \n",
    "codebook_fname = os.path.join(input_folder,'merged_codebook.csv')\n",
    "codebook_df = pd.read_csv (codebook_fname, index_col=0)\n",
    "\n",
    "# sort df by chr and chr_order\n",
    "codebook_df = loci_1d_features.sort_loci_df_by_chr_order (codebook_df)\n",
    "codebook_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3e95ab4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_order</th>\n",
       "      <th>library</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_3742742_3759944</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_6245958_6258969</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_8740008_8759916</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9627926_9637875</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CTP13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CTP13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   name id chr  chr_order library\n",
       "loci_name                                                        \n",
       "chr1_3742742_3759944  1:3742742-3759944  1   1        0.0   CTP11\n",
       "chr1_6245958_6258969  1:6245958-6258969  2   1        1.0   CTP11\n",
       "chr1_8740008_8759916  1:8740008-8759916  3   1        2.0   CTP11\n",
       "chr1_9627926_9637875  1:9627926-9637875  1   1        3.0   CTP13\n",
       "chr1_9799472_9811359  1:9799472-9811359  2   1        4.0   CTP13"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Format the chr loci name by \n",
    "# 1. changing loci name format\n",
    "# 2. extract relevant information such as id, chr, chr_order, and library etc\n",
    "from gene_to_loci import loci_pos_format\n",
    "loci_name_list = list(map(loci_pos_format, codebook_df['name'].tolist()))\n",
    "loci_name_arr = np.array(loci_name_list)\n",
    "\n",
    "# convert to a new dataframe and set loci name as index\n",
    "codebook_df = codebook_df[['name','id','chr','chr_order','library']]\n",
    "codebook_df['loci_name'] = list(loci_name_arr[:,0])\n",
    "codebook_df = codebook_df.set_index ('loci_name')\n",
    "\n",
    "codebook_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d1c18bbd",
   "metadata": {},
   "source": [
    "# Load spatial distance from matrices"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b6c526c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dict_keys(['Oligo', 'L5 IT', 'Micro', 'Peri', 'Endo', 'Astro', 'OPC', 'L6 CT', 'L5 ET', 'L5/6 NP', 'Pvalb', 'L6 IT', 'Lamp5', 'L6b', 'Sst', 'SMC', 'L4/5 IT', 'L2/3 IT', 'Vip', 'Sncg', 'VLMC'])\n"
     ]
    }
   ],
   "source": [
    "# Load class_2_median\n",
    "# class to median distance dict\n",
    "class_2_median_filename = os.path.join(input_folder, 'subclass_2_medianDict.pkl')\n",
    "\n",
    "class_2_medianDict = pickle.load(open(class_2_median_filename, 'rb'))\n",
    "print(class_2_medianDict.keys())"
   ]
  },
  {
   "cell_type": "raw",
   "id": "3f1f423a",
   "metadata": {},
   "source": [
    "class_2_info_filename = os.path.join(input_folder, 'subclass_2_cellInfo.pkl')\n",
    "class_2_infoDict = pickle.load(open(class_2_info_filename, 'rb'))\n",
    "\n",
    "# inspect cell numbers for quality control\n",
    "sel_class_list = []\n",
    "for _cls, _info_list in class_2_infoDict.items():\n",
    "    #print(_cls, len(_info_list))\n",
    "    # include most cell types with abundant number; remove Peri because no snRNAseq data\n",
    "    if len(_info_list) > len(class_2_infoDict['Peri']): \n",
    "        sel_class_list.append(_cls)\n",
    "        \n",
    "np.array(sel_class_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "005cca34",
   "metadata": {},
   "source": [
    "# Summary distances for all celltypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "3307a87b",
   "metadata": {},
   "outputs": [],
   "source": [
    "from ImageAnalysis3.structure_tools import distance\n",
    "sort_by_region=False\n",
    "chr_2_indices, chr_2_orders = distance.Generate_PlotOrder(codebook_df, codebook_df, sort_by_region=False) ## load codebook\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3803ab2c",
   "metadata": {},
   "source": [
    "## load unnormalized distmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "63b07389",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get unnormalized distmap\n",
    "cis_distmap_dict = {}\n",
    "sel_codebook = codebook_df.copy(deep=True)\n",
    "for _class in class_2_medianDict.keys():\n",
    "\n",
    "    _cis_mat, chr_edges, chr_names = distance.assemble_ChrDistDict_2_Matrix(\n",
    "                                            class_2_medianDict[_class], codebook_df, \n",
    "                                            sel_codebook=sel_codebook, \n",
    "                                            use_cis=True,use_trans=False, sort_by_region=sort_by_region,\n",
    "                                  )\n",
    "    \n",
    "    cis_distmap_dict[_class] = _cis_mat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "65440a35",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pickle\n",
    "pickle.dump(cis_distmap_dict,open(os.path.join(output_analysis_folder,'raw_cis_distmap_subclass.pkl'),'wb'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "db801af8",
   "metadata": {},
   "source": [
    "## get all chr-by-pair dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "11f5819b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([(0, 0), (0, 1), (0, 2), ..., (151, 151), (151, 152), (152, 152)],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loc_pair_idx_list_bychr = {}\n",
    "for _chr_key in list(chr_2_orders.keys())[:]:\n",
    "    # get loc name info\n",
    "    loci_list = codebook_df.iloc[chr_2_indices[_chr_key]].index.tolist()\n",
    "    # make loci idx mat from codebook\n",
    "    codebook_idx_mat = np.empty([len(loci_list), len(loci_list)], dtype=object)\n",
    "    for i in range(len(loci_list)):\n",
    "        for j in range(len(loci_list)):\n",
    "            codebook_idx_mat[i,j] = (i,j)\n",
    "    # upper tri for the unique pair idx set        \n",
    "    loc_pair_idx_list = codebook_idx_mat[np.triu_indices(len(codebook_idx_mat),0)] # 0 to include the trans-homolog allele \n",
    "    loc_pair_idx_list_bychr[_chr_key] = loc_pair_idx_list\n",
    "\n",
    "loc_pair_idx_list_bychr['1']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "59831d4a",
   "metadata": {},
   "source": [
    "## summarize unnormalized distmap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e21c3bc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loc_1</th>\n",
       "      <th>loc_2</th>\n",
       "      <th>pairwise_distance</th>\n",
       "      <th>subclass</th>\n",
       "      <th>chr</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_pairs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_6245958_6258969</td>\n",
       "      <td>0.922651</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_8740008_8759916</td>\n",
       "      <td>1.070654</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_9627926_9637875</td>\n",
       "      <td>1.060591</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_9799472_9811359</td>\n",
       "      <td>1.093519</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223475</th>\n",
       "      <td>chrX_167157164_167167452</td>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>0.845943</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223476</th>\n",
       "      <td>chrX_168746045_168757590</td>\n",
       "      <td>chrX_168746045_168757590</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223477</th>\n",
       "      <td>chrX_168746045_168757590</td>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>0.706862</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223478</th>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223479</th>\n",
       "      <td>chrY_1245775_1259992</td>\n",
       "      <td>chrY_1245775_1259992</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>Y</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2223480 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               loc_1                     loc_2  \\\n",
       "loci_pairs                                                       \n",
       "0               chr1_3742742_3759944      chr1_3742742_3759944   \n",
       "1               chr1_3742742_3759944      chr1_6245958_6258969   \n",
       "2               chr1_3742742_3759944      chr1_8740008_8759916   \n",
       "3               chr1_3742742_3759944      chr1_9627926_9637875   \n",
       "4               chr1_3742742_3759944      chr1_9799472_9811359   \n",
       "...                              ...                       ...   \n",
       "2223475     chrX_167157164_167167452  chrX_169963295_170005197   \n",
       "2223476     chrX_168746045_168757590  chrX_168746045_168757590   \n",
       "2223477     chrX_168746045_168757590  chrX_169963295_170005197   \n",
       "2223478     chrX_169963295_170005197  chrX_169963295_170005197   \n",
       "2223479         chrY_1245775_1259992      chrY_1245775_1259992   \n",
       "\n",
       "            pairwise_distance subclass chr  \n",
       "loci_pairs                                  \n",
       "0                    0.000000    Oligo   1  \n",
       "1                    0.922651    Oligo   1  \n",
       "2                    1.070654    Oligo   1  \n",
       "3                    1.060591    Oligo   1  \n",
       "4                    1.093519    Oligo   1  \n",
       "...                       ...      ...  ..  \n",
       "2223475              0.845943     VLMC   X  \n",
       "2223476              0.000000     VLMC   X  \n",
       "2223477              0.706862     VLMC   X  \n",
       "2223478              0.000000     VLMC   X  \n",
       "2223479                   NaN     VLMC   Y  \n",
       "\n",
       "[2223480 rows x 5 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# summarize all unique loci interactions across celltypes            \n",
    "distmap_summary_dict = {'loc_1':[],'loc_2':[], 'pairwise_distance':[],'subclass':[], 'chr':[]}\n",
    "\n",
    "for _class in class_2_medianDict.keys():\n",
    "    _cis_mat = cis_distmap_dict[_class]\n",
    "    # get cist mat for each chr\n",
    "    for _chr_key in list(chr_2_orders.keys())[:]:\n",
    "        _cis_mat_chr = _cis_mat[chr_2_indices[_chr_key],:]\n",
    "        _cis_mat_chr = _cis_mat_chr[:,chr_2_indices[_chr_key]]\n",
    "        _dists = _cis_mat_chr[np.triu_indices(len(_cis_mat_chr),0)] # 0 to include the trans-homolog allele\n",
    "        distmap_summary_dict['pairwise_distance'].extend(_dists)\n",
    "        distmap_summary_dict['subclass'].extend([_class,]*len(_dists))\n",
    "        distmap_summary_dict['chr'].extend([_chr_key,]*len(_dists))\n",
    "        # add loci name saved for each chr\n",
    "        loci_list = codebook_df.iloc[chr_2_indices[_chr_key]].index.tolist()\n",
    "        loc_pair_idx_list = loc_pair_idx_list_bychr[_chr_key] \n",
    "        distmap_summary_dict['loc_1'].extend(loci_list[idx[0]] for idx in loc_pair_idx_list)\n",
    "        distmap_summary_dict['loc_2'].extend(loci_list[idx[1]] for idx in loc_pair_idx_list)\n",
    "    \n",
    "distmap_summary_df = pd.DataFrame(distmap_summary_dict)\n",
    "distmap_summary_df.index.name='loci_pairs'\n",
    "distmap_summary_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "11efbab7",
   "metadata": {},
   "source": [
    "## add genomic distance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "5595315a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\shiwei\\AppData\\Local\\Temp\\3\\ipykernel_31848\\3846640317.py:8: FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`\n",
      "  distmap_summary_df.loc[:, 'genomic_distance']= distmap_summary_df['genomic_distance'].map(lambda x: int(x))\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>loc_1</th>\n",
       "      <th>loc_2</th>\n",
       "      <th>pairwise_distance</th>\n",
       "      <th>subclass</th>\n",
       "      <th>chr</th>\n",
       "      <th>genomic_distance</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_pairs</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_6245958_6258969</td>\n",
       "      <td>0.922651</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "      <td>2503216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_8740008_8759916</td>\n",
       "      <td>1.070654</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "      <td>4997266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_9627926_9637875</td>\n",
       "      <td>1.060591</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "      <td>5885184</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_3742742_3759944</td>\n",
       "      <td>chr1_9799472_9811359</td>\n",
       "      <td>1.093519</td>\n",
       "      <td>Oligo</td>\n",
       "      <td>1</td>\n",
       "      <td>6056730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223475</th>\n",
       "      <td>chrX_167157164_167167452</td>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>0.845943</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "      <td>2806131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223476</th>\n",
       "      <td>chrX_168746045_168757590</td>\n",
       "      <td>chrX_168746045_168757590</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223477</th>\n",
       "      <td>chrX_168746045_168757590</td>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>0.706862</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "      <td>1217250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223478</th>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>chrX_169963295_170005197</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>X</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2223479</th>\n",
       "      <td>chrY_1245775_1259992</td>\n",
       "      <td>chrY_1245775_1259992</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VLMC</td>\n",
       "      <td>Y</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2223480 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                               loc_1                     loc_2  \\\n",
       "loci_pairs                                                       \n",
       "0               chr1_3742742_3759944      chr1_3742742_3759944   \n",
       "1               chr1_3742742_3759944      chr1_6245958_6258969   \n",
       "2               chr1_3742742_3759944      chr1_8740008_8759916   \n",
       "3               chr1_3742742_3759944      chr1_9627926_9637875   \n",
       "4               chr1_3742742_3759944      chr1_9799472_9811359   \n",
       "...                              ...                       ...   \n",
       "2223475     chrX_167157164_167167452  chrX_169963295_170005197   \n",
       "2223476     chrX_168746045_168757590  chrX_168746045_168757590   \n",
       "2223477     chrX_168746045_168757590  chrX_169963295_170005197   \n",
       "2223478     chrX_169963295_170005197  chrX_169963295_170005197   \n",
       "2223479         chrY_1245775_1259992      chrY_1245775_1259992   \n",
       "\n",
       "            pairwise_distance subclass chr  genomic_distance  \n",
       "loci_pairs                                                    \n",
       "0                    0.000000    Oligo   1                 0  \n",
       "1                    0.922651    Oligo   1           2503216  \n",
       "2                    1.070654    Oligo   1           4997266  \n",
       "3                    1.060591    Oligo   1           5885184  \n",
       "4                    1.093519    Oligo   1           6056730  \n",
       "...                       ...      ...  ..               ...  \n",
       "2223475              0.845943     VLMC   X           2806131  \n",
       "2223476              0.000000     VLMC   X                 0  \n",
       "2223477              0.706862     VLMC   X           1217250  \n",
       "2223478              0.000000     VLMC   X                 0  \n",
       "2223479                   NaN     VLMC   Y                 0  \n",
       "\n",
       "[2223480 rows x 6 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for _chr_name, _chr_group in distmap_summary_df.groupby(by='chr'):\n",
    "    _chr_group_temp = _chr_group.copy(deep=True)\n",
    "    _chr_group_temp['loc_1_start'] = _chr_group_temp['loc_1'].map(lambda x: int(x.split('_')[1]))\n",
    "    _chr_group_temp['loc_2_start'] = _chr_group_temp['loc_2'].map(lambda x: int(x.split('_')[1]))\n",
    "    _chr_group_temp['genomic_distance'] = _chr_group_temp['loc_2_start']-_chr_group_temp['loc_1_start']\n",
    "    distmap_summary_df.loc[distmap_summary_df['chr']==_chr_name, 'genomic_distance'] = _chr_group_temp['genomic_distance'] \n",
    "\n",
    "distmap_summary_df.loc[:, 'genomic_distance']= distmap_summary_df['genomic_distance'].map(lambda x: int(x))\n",
    "distmap_summary_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "181472ce",
   "metadata": {},
   "source": [
    "## save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "d9edb0a0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Create analysis output subfolder.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████████| 21/21 [00:16<00:00,  1.29it/s]\n"
     ]
    }
   ],
   "source": [
    "import tqdm\n",
    "output_analysis_subfolder = os.path.join(output_analysis_folder,'Cis_dismap_summary')\n",
    "if not os.path.exists(output_analysis_subfolder):\n",
    "    os.mkdir(output_analysis_subfolder)\n",
    "    print ('Create analysis output subfolder.')\n",
    "\n",
    "for _class in tqdm.tqdm(class_2_medianDict.keys()):\n",
    "    _class_distmap_summary_df = distmap_summary_df[distmap_summary_df['subclass']==_class]\n",
    "    _class_distmap_summary_df.to_csv(os.path.join(output_analysis_subfolder,\n",
    "                                                  f'raw_cis_distmap_subclass_{_class.replace(\"/\",\"_\").replace(\" \",\"_\")}_summary.csv'))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
