{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "75c4e381",
   "metadata": {},
   "source": [
    "# Import module"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14a33d3e",
   "metadata": {},
   "source": [
    "The link to get [ImageAnalysis3](https://github.com/zhengpuas47/ImageAnalysis3) \n",
    "\n",
    "Or from the Zhuang lab archived [source_tools](https://github.com/ZhuangLab/Chromatin_Analysis_2020_cell/tree/master/sequential_tracing/source)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b411cc21",
   "metadata": {},
   "source": [
    "## ImageAnalysis3 and basic modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "32554ab1",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10780\n"
     ]
    }
   ],
   "source": [
    "%run \"C:\\Users\\shiwei\\Documents\\ImageAnalysis3\\required_files\\Startup_py3.py\"\n",
    "sys.path.append(r\"C:\\Users\\shiwei\\Documents\")\n",
    "\n",
    "import ImageAnalysis3 as ia\n",
    "from ImageAnalysis3 import *\n",
    "from ImageAnalysis3.classes import _allowed_kwds\n",
    "\n",
    "import h5py\n",
    "import ast\n",
    "import pandas as pd\n",
    "\n",
    "print(os.getpid())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5f8cd77",
   "metadata": {},
   "source": [
    "## Chromatin_analysis_tools etc"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2db5036f",
   "metadata": {},
   "source": [
    "See **functions** in the repository for [AnalysisTool_Chromatin](../../README.md)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9469aba8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'atac_to_loci' from 'C:\\\\Users\\\\shiwei\\\\Documents\\\\AnalysisTool_Chromatin\\\\atac_to_loci.py'>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Chromatin_analysis_tools (ATC)\n",
    "# Get path for the py containing functions\n",
    "import os\n",
    "import sys\n",
    "import importlib\n",
    "module_path =r'C:\\Users\\shiwei\\Documents\\AnalysisTool_Chromatin'\n",
    "if module_path not in sys.path:\n",
    "    sys.path.append(module_path)\n",
    "    \n",
    "# import relevant modules\n",
    "import gene_selection \n",
    "importlib.reload(gene_selection)\n",
    "import gene_to_loci\n",
    "importlib.reload(gene_to_loci)\n",
    "import gene_activity\n",
    "importlib.reload(gene_activity)\n",
    "import loci_1d_features\n",
    "importlib.reload(loci_1d_features)  \n",
    "\n",
    "import atac_to_loci\n",
    "importlib.reload(atac_to_loci)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa08c6de",
   "metadata": {},
   "source": [
    "# Define folders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "2c8dce53",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use existing analysis folder: L:\\Shiwei\\postanalysis_2024\\v0\\compartment_transcription\\analysis.\n",
      "Use existing figure folder: L:\\Shiwei\\postanalysis_2024\\v0\\compartment_transcription\\figures.\n"
     ]
    }
   ],
   "source": [
    "# main folder for postanalysis\n",
    "postanalysis_folder = r'L:\\Shiwei\\postanalysis_2024\\v0'\n",
    "# input files for postanalysis\n",
    "input_folder = os.path.join(postanalysis_folder, 'resources_from_preprocess')\n",
    "\n",
    "# output file to be generated\n",
    "output_main_folder = os.path.join(postanalysis_folder, 'compartment_transcription')\n",
    "\n",
    "output_analysis_folder = os.path.join(output_main_folder, 'analysis')\n",
    "output_figure_folder = os.path.join(output_main_folder, 'figures')\n",
    "\n",
    "# make new folder if needed\n",
    "make_output_folder = True\n",
    "\n",
    "if make_output_folder and not os.path.exists(output_analysis_folder):\n",
    "    os.makedirs(output_analysis_folder)\n",
    "    print(f'Generating analysis folder: {output_analysis_folder}.')\n",
    "elif os.path.exists(output_analysis_folder):\n",
    "    print(f'Use existing analysis folder: {output_analysis_folder}.')\n",
    "    \n",
    "if make_output_folder and not os.path.exists(output_figure_folder):\n",
    "    os.makedirs(output_figure_folder)\n",
    "    print(f'Generating figure folder: {output_figure_folder}.')\n",
    "elif os.path.exists(output_figure_folder):\n",
    "    print(f'Use existing figure folder: {output_figure_folder}.')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43745cf2",
   "metadata": {},
   "source": [
    "# Plotting parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "fb1e3324",
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rc('font', family='serif')\n",
    "plt.rc('font', serif='Arial')\n",
    "\n",
    "from ImageAnalysis3.figure_tools import _double_col_width, _single_col_width, _font_size, _ticklabel_size,_ticklabel_width\n",
    "\n",
    "import seaborn as sns\n",
    "sns.set_context(\"paper\", rc={\"font.size\":_font_size,\"axes.titlesize\":_font_size+1,\"axes.labelsize\":_font_size})  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "bc310b7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Other required plotting parameters\n",
    "_dpi = 300\n",
    "_font_size = 7\n",
    "_page_width = 5.5\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a1d34726",
   "metadata": {},
   "source": [
    "## cell type color-codes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c33943dd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# cell labels from RNA-MERFISH and celltype prediction\n",
    "selected_cell_labels = ['L2/3 IT','L4/5 IT','L5 IT','L6 IT','L5 ET','L5/6 NP','L6 CT','L6b',\n",
    "                           'Sst','Pvalb','Lamp5','Sncg','Vip',\n",
    "                           'Astro','Oligo','OPC','Micro','Endo','VLMC','SMC','Peri', \n",
    "                           #'other',\n",
    "                          ]\n",
    "# cell palette from RNA-MERFISH UMAP and stats\n",
    "celltype_palette = {'Astro':'lightcoral', \n",
    "                    'Endo':'skyblue', \n",
    "                    'L2/3 IT':'gold', \n",
    "                    'L4/5 IT':'darkorange', \n",
    "                    'L5 ET':'mediumseagreen', \n",
    "                    'L5 IT':'aqua',\n",
    "                    'L5/6 NP':'darkgreen',\n",
    "                    'L6 CT':'brown',\n",
    "                    'L6 IT':'magenta',\n",
    "                    'L6b':'blue', \n",
    "                    'Lamp5':'orange', \n",
    "                    'Micro':'peachpuff',\n",
    "                    'OPC':'thistle', \n",
    "                    'Oligo':'darkviolet',\n",
    "                    'Peri':'sandybrown',\n",
    "                    'Pvalb':'springgreen',\n",
    "                    'SMC':'rosybrown',\n",
    "                    'Sncg':'darkkhaki',\n",
    "                    'Sst':'steelblue', \n",
    "                    'VLMC':'saddlebrown', \n",
    "                    'Vip':'red',\n",
    "                    'other':'slategray'}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "caad6887",
   "metadata": {},
   "outputs": [],
   "source": [
    "# this is the plotting order noted based on the snRNA transcriptional acitivty if needed\n",
    "sorted_cellplot_order_byRNA = ['Micro', 'Oligo', 'Endo', 'OPC', 'Astro', 'Vip', 'Lamp5',\n",
    "                  'L5/6 NP', 'Sst', 'Sncg', 'Pvalb', 'L4/5 IT', 'L6 CT',\n",
    "                  'L6 IT', 'L6b', 'L2/3 IT', 'L5 IT', 'L5 ET']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b00f58f0",
   "metadata": {},
   "source": [
    "# Load data relevant information"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7e667606",
   "metadata": {},
   "source": [
    "## load and format codebook"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b8440c6",
   "metadata": {},
   "source": [
    "[merged codebook](../resources/merged_codebook.csv) as in the repository (merged for all DNA-MERFISH libraries)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c0734aad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>NDB_784</th>\n",
       "      <th>NDB_755</th>\n",
       "      <th>NDB_826</th>\n",
       "      <th>NDB_713</th>\n",
       "      <th>NDB_865</th>\n",
       "      <th>NDB_725</th>\n",
       "      <th>NDB_817</th>\n",
       "      <th>NDB_710</th>\n",
       "      <th>...</th>\n",
       "      <th>NDB_479</th>\n",
       "      <th>NDB_562</th>\n",
       "      <th>NDB_608</th>\n",
       "      <th>NDB_460</th>\n",
       "      <th>NDB_563</th>\n",
       "      <th>NDB_592</th>\n",
       "      <th>NDB_368</th>\n",
       "      <th>NDB_436</th>\n",
       "      <th>NDB_629</th>\n",
       "      <th>NDB_604</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 228 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   name id  NDB_784  NDB_755  NDB_826  NDB_713  NDB_865  \\\n",
       "0     1:3742742-3759944  1      0.0      0.0      0.0      0.0      0.0   \n",
       "1     1:6245958-6258969  2      0.0      0.0      0.0      0.0      0.0   \n",
       "2     1:8740008-8759916  3      0.0      0.0      0.0      0.0      0.0   \n",
       "1016  1:9627926-9637875  1      0.0      0.0      0.0      0.0      0.0   \n",
       "1017  1:9799472-9811359  2      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "      NDB_725  NDB_817  NDB_710  ...  NDB_479  NDB_562  NDB_608  NDB_460  \\\n",
       "0         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "2         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1016      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1017      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "\n",
       "      NDB_563  NDB_592  NDB_368  NDB_436  NDB_629  NDB_604  \n",
       "0         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "2         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1016      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1017      0.0      1.0      0.0      0.0      0.0      0.0  \n",
       "\n",
       "[5 rows x 228 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load codebook \n",
    "codebook_fname = os.path.join(input_folder,'merged_codebook.csv')\n",
    "codebook_df = pd.read_csv (codebook_fname, index_col=0)\n",
    "\n",
    "# sort df by chr and chr_order\n",
    "codebook_df = loci_1d_features.sort_loci_df_by_chr_order (codebook_df)\n",
    "codebook_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "3e95ab4f",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_order</th>\n",
       "      <th>library</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_3742742_3759944</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_6245958_6258969</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_8740008_8759916</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9627926_9637875</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CTP13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CTP13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   name id chr  chr_order library\n",
       "loci_name                                                        \n",
       "chr1_3742742_3759944  1:3742742-3759944  1   1        0.0   CTP11\n",
       "chr1_6245958_6258969  1:6245958-6258969  2   1        1.0   CTP11\n",
       "chr1_8740008_8759916  1:8740008-8759916  3   1        2.0   CTP11\n",
       "chr1_9627926_9637875  1:9627926-9637875  1   1        3.0   CTP13\n",
       "chr1_9799472_9811359  1:9799472-9811359  2   1        4.0   CTP13"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Format the chr loci name by \n",
    "# 1. changing loci name format\n",
    "# 2. extract relevant information such as id, chr, chr_order, and library etc\n",
    "from gene_to_loci import loci_pos_format\n",
    "loci_name_list = list(map(loci_pos_format, codebook_df['name'].tolist()))\n",
    "loci_name_arr = np.array(loci_name_list)\n",
    "\n",
    "# convert to a new dataframe and set loci name as index\n",
    "codebook_df = codebook_df[['name','id','chr','chr_order','library']]\n",
    "codebook_df['loci_name'] = list(loci_name_arr[:,0])\n",
    "codebook_df = codebook_df.set_index ('loci_name')\n",
    "\n",
    "codebook_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6d75011f",
   "metadata": {},
   "source": [
    "# Load pre-summarized AB density medians\n",
    "\n",
    "Data can be generated from notebook\n",
    "\n",
    "[2_2_summarize_ab_density_all_loci_subclassby_ensemble](2_2_summarize_ab_density_all_loci_subclassby_ensemble.ipynb)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d0beba97",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo</th>\n",
       "      <th>OPC</th>\n",
       "      <th>Micro</th>\n",
       "      <th>Astro</th>\n",
       "      <th>Endo</th>\n",
       "      <th>Peri</th>\n",
       "      <th>L2/3 IT</th>\n",
       "      <th>L4/5 IT</th>\n",
       "      <th>L5 IT</th>\n",
       "      <th>L6 IT</th>\n",
       "      <th>L5 ET</th>\n",
       "      <th>L6 CT</th>\n",
       "      <th>L5/6 NP</th>\n",
       "      <th>L6b</th>\n",
       "      <th>Vip</th>\n",
       "      <th>Pvalb</th>\n",
       "      <th>Lamp5</th>\n",
       "      <th>Sst</th>\n",
       "      <th>Sncg</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_3742742_3759944</th>\n",
       "      <td>-0.430767</td>\n",
       "      <td>-0.420269</td>\n",
       "      <td>-0.255699</td>\n",
       "      <td>-0.453668</td>\n",
       "      <td>0.132135</td>\n",
       "      <td>-0.140469</td>\n",
       "      <td>-0.560620</td>\n",
       "      <td>-0.460501</td>\n",
       "      <td>-0.446779</td>\n",
       "      <td>-0.488172</td>\n",
       "      <td>-0.421529</td>\n",
       "      <td>-0.455810</td>\n",
       "      <td>-0.304009</td>\n",
       "      <td>-0.472670</td>\n",
       "      <td>-0.114805</td>\n",
       "      <td>-0.289297</td>\n",
       "      <td>-0.309272</td>\n",
       "      <td>-0.325902</td>\n",
       "      <td>-0.186869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_6245958_6258969</th>\n",
       "      <td>-0.153712</td>\n",
       "      <td>-0.206244</td>\n",
       "      <td>-0.047506</td>\n",
       "      <td>-0.226636</td>\n",
       "      <td>0.387080</td>\n",
       "      <td>0.133823</td>\n",
       "      <td>-0.447946</td>\n",
       "      <td>-0.445162</td>\n",
       "      <td>-0.487868</td>\n",
       "      <td>-0.408086</td>\n",
       "      <td>-0.356980</td>\n",
       "      <td>-0.453940</td>\n",
       "      <td>-0.201882</td>\n",
       "      <td>-0.332559</td>\n",
       "      <td>-0.207538</td>\n",
       "      <td>-0.219555</td>\n",
       "      <td>-0.207729</td>\n",
       "      <td>-0.116463</td>\n",
       "      <td>-0.435847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_8740008_8759916</th>\n",
       "      <td>-0.425727</td>\n",
       "      <td>-0.148089</td>\n",
       "      <td>-0.275335</td>\n",
       "      <td>-0.409980</td>\n",
       "      <td>0.080992</td>\n",
       "      <td>0.116621</td>\n",
       "      <td>-0.400666</td>\n",
       "      <td>-0.314289</td>\n",
       "      <td>-0.418091</td>\n",
       "      <td>-0.412230</td>\n",
       "      <td>-0.292372</td>\n",
       "      <td>-0.487285</td>\n",
       "      <td>-0.505100</td>\n",
       "      <td>-0.498226</td>\n",
       "      <td>-0.032649</td>\n",
       "      <td>-0.221845</td>\n",
       "      <td>-0.257861</td>\n",
       "      <td>-0.206280</td>\n",
       "      <td>-0.333840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9627926_9637875</th>\n",
       "      <td>-0.254296</td>\n",
       "      <td>0.003753</td>\n",
       "      <td>0.080439</td>\n",
       "      <td>-0.209209</td>\n",
       "      <td>0.287116</td>\n",
       "      <td>0.149472</td>\n",
       "      <td>-0.350141</td>\n",
       "      <td>-0.286409</td>\n",
       "      <td>-0.402433</td>\n",
       "      <td>-0.200084</td>\n",
       "      <td>-0.307043</td>\n",
       "      <td>-0.290176</td>\n",
       "      <td>-0.298747</td>\n",
       "      <td>-0.386660</td>\n",
       "      <td>-0.279638</td>\n",
       "      <td>-0.146983</td>\n",
       "      <td>-0.086232</td>\n",
       "      <td>-0.194100</td>\n",
       "      <td>-0.302644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>-0.116000</td>\n",
       "      <td>-0.083814</td>\n",
       "      <td>0.234512</td>\n",
       "      <td>-0.073540</td>\n",
       "      <td>0.492773</td>\n",
       "      <td>0.359415</td>\n",
       "      <td>-0.246629</td>\n",
       "      <td>-0.303253</td>\n",
       "      <td>-0.354788</td>\n",
       "      <td>-0.248334</td>\n",
       "      <td>-0.232831</td>\n",
       "      <td>-0.253817</td>\n",
       "      <td>-0.417956</td>\n",
       "      <td>-0.329905</td>\n",
       "      <td>-0.355471</td>\n",
       "      <td>-0.403508</td>\n",
       "      <td>-0.072171</td>\n",
       "      <td>-0.268272</td>\n",
       "      <td>-0.355233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_166247682_166259932</th>\n",
       "      <td>0.190443</td>\n",
       "      <td>0.086648</td>\n",
       "      <td>0.045090</td>\n",
       "      <td>0.202411</td>\n",
       "      <td>0.315556</td>\n",
       "      <td>0.383854</td>\n",
       "      <td>-0.157234</td>\n",
       "      <td>-0.122200</td>\n",
       "      <td>-0.238115</td>\n",
       "      <td>-0.214688</td>\n",
       "      <td>-0.064618</td>\n",
       "      <td>-0.077107</td>\n",
       "      <td>-0.332221</td>\n",
       "      <td>-0.080294</td>\n",
       "      <td>-0.227362</td>\n",
       "      <td>-0.265453</td>\n",
       "      <td>-0.371283</td>\n",
       "      <td>-0.219211</td>\n",
       "      <td>-0.668820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_167157164_167167452</th>\n",
       "      <td>0.069587</td>\n",
       "      <td>-0.185342</td>\n",
       "      <td>0.435091</td>\n",
       "      <td>0.024904</td>\n",
       "      <td>0.626812</td>\n",
       "      <td>0.500466</td>\n",
       "      <td>-0.203417</td>\n",
       "      <td>-0.119909</td>\n",
       "      <td>-0.157937</td>\n",
       "      <td>-0.201859</td>\n",
       "      <td>-0.364895</td>\n",
       "      <td>-0.162953</td>\n",
       "      <td>0.024588</td>\n",
       "      <td>-0.201043</td>\n",
       "      <td>-0.109136</td>\n",
       "      <td>-0.048074</td>\n",
       "      <td>-0.359983</td>\n",
       "      <td>-0.203461</td>\n",
       "      <td>-0.322641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_168746045_168757590</th>\n",
       "      <td>-0.138497</td>\n",
       "      <td>-0.285900</td>\n",
       "      <td>-0.019425</td>\n",
       "      <td>-0.079555</td>\n",
       "      <td>0.332414</td>\n",
       "      <td>0.382330</td>\n",
       "      <td>-0.297829</td>\n",
       "      <td>-0.237427</td>\n",
       "      <td>-0.256210</td>\n",
       "      <td>-0.317645</td>\n",
       "      <td>-0.250351</td>\n",
       "      <td>-0.389821</td>\n",
       "      <td>0.137434</td>\n",
       "      <td>-0.329275</td>\n",
       "      <td>-0.261027</td>\n",
       "      <td>-0.155011</td>\n",
       "      <td>-0.287720</td>\n",
       "      <td>-0.211167</td>\n",
       "      <td>-0.682236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_169963295_170005197</th>\n",
       "      <td>-0.095872</td>\n",
       "      <td>0.023907</td>\n",
       "      <td>0.074545</td>\n",
       "      <td>0.013206</td>\n",
       "      <td>0.551413</td>\n",
       "      <td>0.311509</td>\n",
       "      <td>-0.235026</td>\n",
       "      <td>-0.220536</td>\n",
       "      <td>-0.293816</td>\n",
       "      <td>-0.380180</td>\n",
       "      <td>-0.200907</td>\n",
       "      <td>-0.360787</td>\n",
       "      <td>-0.075992</td>\n",
       "      <td>-0.169335</td>\n",
       "      <td>-0.296937</td>\n",
       "      <td>-0.053206</td>\n",
       "      <td>-0.307064</td>\n",
       "      <td>-0.229188</td>\n",
       "      <td>-0.193384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY_1245775_1259992</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1982 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Oligo       OPC     Micro     Astro      Endo  \\\n",
       "loci_name                                                                    \n",
       "chr1_3742742_3759944     -0.430767 -0.420269 -0.255699 -0.453668  0.132135   \n",
       "chr1_6245958_6258969     -0.153712 -0.206244 -0.047506 -0.226636  0.387080   \n",
       "chr1_8740008_8759916     -0.425727 -0.148089 -0.275335 -0.409980  0.080992   \n",
       "chr1_9627926_9637875     -0.254296  0.003753  0.080439 -0.209209  0.287116   \n",
       "chr1_9799472_9811359     -0.116000 -0.083814  0.234512 -0.073540  0.492773   \n",
       "...                            ...       ...       ...       ...       ...   \n",
       "chrX_166247682_166259932  0.190443  0.086648  0.045090  0.202411  0.315556   \n",
       "chrX_167157164_167167452  0.069587 -0.185342  0.435091  0.024904  0.626812   \n",
       "chrX_168746045_168757590 -0.138497 -0.285900 -0.019425 -0.079555  0.332414   \n",
       "chrX_169963295_170005197 -0.095872  0.023907  0.074545  0.013206  0.551413   \n",
       "chrY_1245775_1259992           NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "                              Peri   L2/3 IT   L4/5 IT     L5 IT     L6 IT  \\\n",
       "loci_name                                                                    \n",
       "chr1_3742742_3759944     -0.140469 -0.560620 -0.460501 -0.446779 -0.488172   \n",
       "chr1_6245958_6258969      0.133823 -0.447946 -0.445162 -0.487868 -0.408086   \n",
       "chr1_8740008_8759916      0.116621 -0.400666 -0.314289 -0.418091 -0.412230   \n",
       "chr1_9627926_9637875      0.149472 -0.350141 -0.286409 -0.402433 -0.200084   \n",
       "chr1_9799472_9811359      0.359415 -0.246629 -0.303253 -0.354788 -0.248334   \n",
       "...                            ...       ...       ...       ...       ...   \n",
       "chrX_166247682_166259932  0.383854 -0.157234 -0.122200 -0.238115 -0.214688   \n",
       "chrX_167157164_167167452  0.500466 -0.203417 -0.119909 -0.157937 -0.201859   \n",
       "chrX_168746045_168757590  0.382330 -0.297829 -0.237427 -0.256210 -0.317645   \n",
       "chrX_169963295_170005197  0.311509 -0.235026 -0.220536 -0.293816 -0.380180   \n",
       "chrY_1245775_1259992           NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "                             L5 ET     L6 CT   L5/6 NP       L6b       Vip  \\\n",
       "loci_name                                                                    \n",
       "chr1_3742742_3759944     -0.421529 -0.455810 -0.304009 -0.472670 -0.114805   \n",
       "chr1_6245958_6258969     -0.356980 -0.453940 -0.201882 -0.332559 -0.207538   \n",
       "chr1_8740008_8759916     -0.292372 -0.487285 -0.505100 -0.498226 -0.032649   \n",
       "chr1_9627926_9637875     -0.307043 -0.290176 -0.298747 -0.386660 -0.279638   \n",
       "chr1_9799472_9811359     -0.232831 -0.253817 -0.417956 -0.329905 -0.355471   \n",
       "...                            ...       ...       ...       ...       ...   \n",
       "chrX_166247682_166259932 -0.064618 -0.077107 -0.332221 -0.080294 -0.227362   \n",
       "chrX_167157164_167167452 -0.364895 -0.162953  0.024588 -0.201043 -0.109136   \n",
       "chrX_168746045_168757590 -0.250351 -0.389821  0.137434 -0.329275 -0.261027   \n",
       "chrX_169963295_170005197 -0.200907 -0.360787 -0.075992 -0.169335 -0.296937   \n",
       "chrY_1245775_1259992           NaN       NaN       NaN       NaN       NaN   \n",
       "\n",
       "                             Pvalb     Lamp5       Sst      Sncg  \n",
       "loci_name                                                         \n",
       "chr1_3742742_3759944     -0.289297 -0.309272 -0.325902 -0.186869  \n",
       "chr1_6245958_6258969     -0.219555 -0.207729 -0.116463 -0.435847  \n",
       "chr1_8740008_8759916     -0.221845 -0.257861 -0.206280 -0.333840  \n",
       "chr1_9627926_9637875     -0.146983 -0.086232 -0.194100 -0.302644  \n",
       "chr1_9799472_9811359     -0.403508 -0.072171 -0.268272 -0.355233  \n",
       "...                            ...       ...       ...       ...  \n",
       "chrX_166247682_166259932 -0.265453 -0.371283 -0.219211 -0.668820  \n",
       "chrX_167157164_167167452 -0.048074 -0.359983 -0.203461 -0.322641  \n",
       "chrX_168746045_168757590 -0.155011 -0.287720 -0.211167 -0.682236  \n",
       "chrX_169963295_170005197 -0.053206 -0.307064 -0.229188 -0.193384  \n",
       "chrY_1245775_1259992           NaN       NaN       NaN       NaN  \n",
       "\n",
       "[1982 rows x 19 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# the calculated AB desnity ratio for single-cell\n",
    "#compartment_folder = r'\\\\10.245.74.158\\Chromatin_NAS_8\\Analyzed_data\\MouseBrain_PostAnalysis'\n",
    "AB_summary_fname = os.path.join(output_analysis_folder , r'Median_Summary_AB_trans_ratio_notNorm_r0.5_bymajorType_by_pm.csv')\n",
    "\n",
    "AB_ratio_summary = pd.read_csv(AB_summary_fname, index_col=0)\n",
    "\n",
    "source_from_CW = False\n",
    "if source_from_CW:\n",
    "    AB_ratio_summary['chr'] = AB_ratio_summary.index\n",
    "    AB_ratio_summary = AB_ratio_summary.set_index('loci_name')\n",
    "    loci_chr_id_map = {(_chr, _order):v for _chr, _order, v in zip(df_refgen['chr'],\n",
    "                                                                   df_refgen['chr_order'],\n",
    "                                                                   df_refgen['loci_name'],)}\n",
    "    AB_ratio_summary['loci_name'] = AB_ratio_summary.apply(lambda row: loci_chr_id_map[row['chr'], row['hyb']], axis=1)\n",
    "    AB_ratio_summary = pd.pivot(AB_ratio_summary, index='loci_name', columns='subclass', values='AB_density')\n",
    "    \n",
    "AB_ratio_summary"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7d38a7b9",
   "metadata": {},
   "source": [
    "## calculate regression factor between celltypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "b0a71825",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Oligo': (array([0.77329371]), array([0.12521121])),\n",
       " 'OPC': (array([0.71597982]), array([0.10633413])),\n",
       " 'Micro': (array([0.68456728]), array([0.21108297])),\n",
       " 'Astro': (array([0.69342696]), array([0.15908295])),\n",
       " 'Endo': (array([0.44780838]), array([0.49364204])),\n",
       " 'Peri': (array([0.49270282]), array([0.48223804])),\n",
       " 'L2/3 IT': (array([1.]), array([0.])),\n",
       " 'L4/5 IT': (array([0.86112884]), array([-0.00277903])),\n",
       " 'L5 IT': (array([0.77141948]), array([-0.03423645])),\n",
       " 'L6 IT': (array([0.85293924]), array([-0.01805797])),\n",
       " 'L5 ET': (array([0.63679152]), array([-0.04722389])),\n",
       " 'L6 CT': (array([0.89872086]), array([-0.01063253])),\n",
       " 'L5/6 NP': (array([0.58499424]), array([-0.0513388])),\n",
       " 'L6b': (array([0.57913252]), array([-0.04630086])),\n",
       " 'Vip': (array([0.46624995]), array([-0.04151498])),\n",
       " 'Pvalb': (array([0.58214168]), array([-0.03671979])),\n",
       " 'Lamp5': (array([0.49064784]), array([-0.01925425])),\n",
       " 'Sst': (array([0.48761871]), array([-0.02805633])),\n",
       " 'Sncg': (array([0.49812969]), array([-0.04766385]))}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "### get Abratio normalization dict\n",
    "from scipy import stats\n",
    "from sklearn.linear_model import LinearRegression\n",
    "\n",
    "# use common ref to regress which has more number of data\n",
    "regress_each = False\n",
    "fit_intercept=True\n",
    "ab_regression_coef_dict = {}\n",
    "\n",
    "#ref_cls = 'Gluta'\n",
    "# do regression for each class as ref_cls because the downstream marker gene analysis will be similar\n",
    "\n",
    "\n",
    "\n",
    "if not regress_each:\n",
    "    ref_cls = 'L2/3 IT'\n",
    "    for _cls in AB_ratio_summary.columns:\n",
    "        xs = np.array(AB_ratio_summary[ref_cls].tolist())\n",
    "        ys = np.array(AB_ratio_summary[_cls].tolist())\n",
    "        # correlation\n",
    "        xs_good = xs[(~np.isnan(xs)&(~np.isnan(ys)))]\n",
    "        ys_good = ys[(~np.isnan(xs)&(~np.isnan(ys)))]\n",
    "        X=xs_good.reshape(-1, 1)\n",
    "        y=ys_good.reshape(-1, 1)\n",
    "\n",
    "        if len(X) > 0:\n",
    "            reg = LinearRegression(fit_intercept=fit_intercept).fit(X, y)\n",
    "            ab_regression_coef_dict[_cls] = (reg.coef_[0], reg.intercept_)\n",
    "        else:\n",
    "            ab_regression_coef_dict[_cls] = (np.nan, np.nan)\n",
    "\n",
    "    \n",
    "else:\n",
    "    for ref_cls in AB_ratio_summary.columns:\n",
    "        ab_regression_coef_dict[ref_cls] = {}\n",
    "\n",
    "        for _cls in AB_ratio_summary.columns:\n",
    "            xs = np.array(AB_ratio_summary[ref_cls].tolist())\n",
    "            ys = np.array(AB_ratio_summary[_cls].tolist())\n",
    "            # correlation\n",
    "            xs_good = xs[(~np.isnan(xs)&(~np.isnan(ys)))]\n",
    "            ys_good = ys[(~np.isnan(xs)&(~np.isnan(ys)))]\n",
    "            X=xs_good.reshape(-1, 1)\n",
    "            y=ys_good.reshape(-1, 1)\n",
    "\n",
    "            if len(X) > 0:\n",
    "                reg = LinearRegression(fit_intercept=fit_intercept).fit(X, y)\n",
    "                ab_regression_coef_dict[ref_cls][_cls] = (reg.coef_[0], reg.intercept_)\n",
    "            else:\n",
    "                ab_regression_coef_dict[ref_cls][_cls] = (np.nan, np.nan)\n",
    "\n",
    "ab_regression_coef_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f18f6072",
   "metadata": {},
   "source": [
    "# Process all cellgroups for superenhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6ccc4b7",
   "metadata": {},
   "source": [
    "## define shared parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fb45c8d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# the cell groups to be analyzed\n",
    "sorted_group_order = ['L2/3 IT', 'L4/5 IT', 'L5 IT',\n",
    "                       'L6 IT', 'L5 ET', 'L6 CT', 'L5/6 NP', 'L6b', 'Vip', 'Pvalb', 'Lamp5',\n",
    "                       'Sst', ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d8a356b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# re-import functions\n",
    "import loci_1d_features as lf\n",
    "import gene_selection as gs\n",
    "import gene_to_loci as gl\n",
    "\n",
    "re_calculate_AB = False\n",
    "\n",
    "# use SE that is unique to only one cell type \n",
    "unique_SE = True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d45e9ed8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_order</th>\n",
       "      <th>library</th>\n",
       "      <th>cell_type</th>\n",
       "      <th>enhancer_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_3742742_3759944</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>['None']</td>\n",
       "      <td>['None']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_6245958_6258969</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>['None']</td>\n",
       "      <td>['None']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_8740008_8759916</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>['None']</td>\n",
       "      <td>['None']</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9627926_9637875</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CTP13</td>\n",
       "      <td>['L6_CT']</td>\n",
       "      <td>1:9610471-9646458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CTP13</td>\n",
       "      <td>['L2_3_IT']</td>\n",
       "      <td>1:9759317-9835402</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   name id chr  chr_order library  \\\n",
       "loci_name                                                           \n",
       "chr1_3742742_3759944  1:3742742-3759944  1   1        0.0   CTP11   \n",
       "chr1_6245958_6258969  1:6245958-6258969  2   1        1.0   CTP11   \n",
       "chr1_8740008_8759916  1:8740008-8759916  3   1        2.0   CTP11   \n",
       "chr1_9627926_9637875  1:9627926-9637875  1   1        3.0   CTP13   \n",
       "chr1_9799472_9811359  1:9799472-9811359  2   1        4.0   CTP13   \n",
       "\n",
       "                        cell_type      enhancer_name  \n",
       "loci_name                                             \n",
       "chr1_3742742_3759944     ['None']           ['None']  \n",
       "chr1_6245958_6258969     ['None']           ['None']  \n",
       "chr1_8740008_8759916     ['None']           ['None']  \n",
       "chr1_9627926_9637875    ['L6_CT']  1:9610471-9646458  \n",
       "chr1_9799472_9811359  ['L2_3_IT']  1:9759317-9835402  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "celltype_codebook_fname = os.path.join(input_folder,'merged_codebook_w_celltype_w_rank.csv')\n",
    "celltype_codebook_df = pd.read_csv (celltype_codebook_fname, index_col=0)\n",
    "\n",
    "# sort df by chr and chr_order\n",
    "celltype_codebook_df = loci_1d_features.sort_loci_df_by_chr_order (celltype_codebook_df)\n",
    "\n",
    "# Format the chr loci name by \n",
    "# 1. changing loci name format\n",
    "# 2. extract relevant information such as id, chr, chr_order, and library etc\n",
    "from gene_to_loci import loci_pos_format\n",
    "celltype_codebook_df['name'] = celltype_codebook_df.index\n",
    "loci_name_list = list(map(loci_pos_format, celltype_codebook_df['name'].tolist()))\n",
    "loci_name_arr = np.array(loci_name_list)\n",
    "\n",
    "# convert to a new dataframe and set loci name as index\n",
    "celltype_codebook_df = celltype_codebook_df[['name','id','chr','chr_order','library','cell_type','enhancer_name']]\n",
    "celltype_codebook_df['loci_name'] = list(loci_name_arr[:,0])\n",
    "celltype_codebook_df = celltype_codebook_df.set_index ('loci_name')\n",
    "\n",
    "celltype_codebook_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "50d2c8da",
   "metadata": {},
   "source": [
    "## loop to process each cell group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "03b06c57",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Process SE loci for L2/3 IT.\n",
      "L2_3_IT\n",
      "================================================================================\n",
      "Process SE loci for L4/5 IT.\n",
      "L4_5_IT\n",
      "================================================================================\n",
      "Process SE loci for L5 IT.\n",
      "L5_IT\n",
      "================================================================================\n",
      "Process SE loci for L6 IT.\n",
      "L6_IT\n",
      "================================================================================\n",
      "Process SE loci for L5 ET.\n",
      "L5_PT\n",
      "================================================================================\n",
      "Process SE loci for L6 CT.\n",
      "L6_CT\n",
      "================================================================================\n",
      "Process SE loci for L5/6 NP.\n",
      "L5_6_NP\n",
      "================================================================================\n",
      "Process SE loci for L6b.\n",
      "L6b\n",
      "================================================================================\n",
      "Process SE loci for Vip.\n",
      "Vip\n",
      "================================================================================\n",
      "Process SE loci for Pvalb.\n",
      "Pvalb\n",
      "================================================================================\n",
      "Process SE loci for Lamp5.\n",
      "Lamp5\n",
      "================================================================================\n",
      "Process SE loci for Sst.\n",
      "Sst\n",
      "================================================================================\n"
     ]
    }
   ],
   "source": [
    "# compile result as (marker loci of group) by (median of sc for group)\n",
    "compiled_df = pd.DataFrame(columns = sorted_group_order)\n",
    "np.random.seed(42)\n",
    "\n",
    "for _marker_group in sorted_group_order[:]:\n",
    "#for _marker_group in ['L5 ET','Micro']:    \n",
    "    print (f'Process SE loci for {_marker_group}.')  \n",
    "    ##############################################################################\n",
    "    # 1. get specific SE-loci inds\n",
    "    # modify group name consistency\n",
    "    _marker_group_rename = _marker_group.replace('/','_').replace(' ','_')\n",
    "    if _marker_group_rename == 'Micro':\n",
    "        _marker_group_rename_SE = 'MicroPVM'\n",
    "    elif _marker_group_rename == 'L5_ET':\n",
    "        _marker_group_rename_SE = 'L5_PT'\n",
    "    else:\n",
    "        _marker_group_rename_SE = _marker_group_rename\n",
    "\n",
    "    print(_marker_group_rename_SE)\n",
    "\n",
    "    if unique_SE:\n",
    "        im_loci_df = celltype_codebook_df[celltype_codebook_df['cell_type']==f\"['{_marker_group_rename_SE}']\"]\n",
    "    else:\n",
    "        im_loci_df = celltype_codebook_df[celltype_codebook_df['cell_type'].str.contains(_marker_group_rename_SE)]\n",
    "    #sort loci\n",
    "    im_loci_df  = lf.codebook_chr_order_for_loci_dataframe  (im_loci_df, codebook_df,sel_cols =['chr','chr_order','id'], \n",
    "                                               sort_df = True,\n",
    "                                               sort_by_chr= True)\n",
    "    \n",
    "    # add expression change for\n",
    "    im_loci_df['Expression_change']='upregulated'\n",
    "    ##############################################################################\n",
    "    # 2. add radom ctp13 genome loci as control\n",
    "    non_up_loci_df = celltype_codebook_df[~celltype_codebook_df['cell_type'].str.contains(_marker_group_rename_SE)]\n",
    "    non_up_loci_df = non_up_loci_df[~non_up_loci_df['cell_type'].str.contains('None')]\n",
    "    # 2' if add radom ctp11 genome loci as control\n",
    "    #non_up_loci_df = celltype_codebook_df[celltype_codebook_df['cell_type']==\"['None']\"]\n",
    "    for _iter in range(0,100):\n",
    "        random_loci = np.random.choice(non_up_loci_df.index,100, replace=False)\n",
    "    random_loci_df = celltype_codebook_df.loc[random_loci]\n",
    "    random_loci_df['Expression_change']='random_control'\n",
    "    \n",
    "    im_loci_df=pd.concat([im_loci_df,random_loci_df])\n",
    "    #drop barcode cols\n",
    "    keep_cols = [_c for _c in im_loci_df.columns if 'NDB' not in _c]\n",
    "    keep_cols = [_c for _c in keep_cols if 'Stv' not in _c]\n",
    "    im_loci_df=im_loci_df[keep_cols]\n",
    "    \n",
    "    \n",
    "    ##############################################################################\n",
    "\n",
    "    ##############################################################################\n",
    "    # 2. load AB ratio for relevant loci\n",
    "    for _dir in ['upregulated','random_control'][:]:\n",
    "\n",
    "        sel_im_loci_df = im_loci_df[im_loci_df['Expression_change']==_dir]\n",
    "        # initate result df here so marker of each direction can be independently saved\n",
    "        _marker_group_df = pd.DataFrame(columns = sorted_group_order)\n",
    "        # get chr and chr_order\n",
    "        sel_loci_key_list = lf.sorted_loci_keys_for_loci_dataframe(sel_im_loci_df)\n",
    "\n",
    "        # use chr and chr_order to retrieve AB ratio from single cell\n",
    "        for loci_key in sel_loci_key_list:\n",
    "            # append result for each loci from single cell\n",
    "            # function to get the AB ratio for all cell group and append together\n",
    "            if re_calculate_AB:\n",
    "                _loci_group_df = loci_1d_features.sc_compartment_ratio_by_loci_key (AB_ratio_by_group, \n",
    "                                                                               sorted_group_order, \n",
    "                                                                                loci_key, \n",
    "                                                                                report_type = 'median',\n",
    "                                                                                average_ratios_in_cell=True, \n",
    "                                                                                spot_num_th=600)\n",
    "            else:\n",
    "                _loci_name = codebook_df[(codebook_df['chr']==str(loci_key[0])) & (codebook_df['chr_order']==loci_key[1])].index[0]\n",
    "                if 'chrX' not in _loci_name:\n",
    "                    _loci_group_df = pd.DataFrame(AB_ratio_summary.loc[_loci_name]).transpose()\n",
    "                    _loci_group_df = _loci_group_df.reset_index(drop=True)\n",
    "\n",
    "                \n",
    "            if not regress_each:\n",
    "                for _group in sorted_group_order:\n",
    "                    m, b = ab_regression_coef_dict[_group][0],ab_regression_coef_dict[_group][1]\n",
    "                    _loci_group_df.loc[0, _group] = (_loci_group_df.loc[0, _group]-b)/m \n",
    "\n",
    "\n",
    "            # concat loci\n",
    "            _marker_group_df=pd.concat([_marker_group_df,_loci_group_df])\n",
    "\n",
    "\n",
    "        # add loci, expression, group, gene info, etc\n",
    "        _marker_group_df['loci_name'] = sel_im_loci_df.index.tolist()\n",
    "        _marker_group_df['enhancer_name'] = sel_im_loci_df['enhancer_name'].tolist()\n",
    "        _marker_group_df['Marker_group']=_marker_group\n",
    "        _marker_group_df['Expression_change']=_dir\n",
    "\n",
    "        # regression to normalize ABratio using the marker group as ref_cls\n",
    "        if regress_each:\n",
    "            for _group in sorted_group_order:\n",
    "                m, b = ab_regression_coef_dict[_marker_group][_group][0],ab_regression_coef_dict[_marker_group][_group][1]\n",
    "                _marker_group_df.loc[:, _group] = (_marker_group_df.loc[:, _group]-b)/m \n",
    "            # concat loci\n",
    "            _marker_group_df=pd.concat([_marker_group_df,_loci_group_df])\n",
    "\n",
    "        # concat for the marker cell group                  \n",
    "        compiled_df=pd.concat([compiled_df,_marker_group_df])\n",
    "\n",
    "    print('================================================================================')\n",
    "\n",
    "# set loci_name as index \n",
    "compiled_df=compiled_df.set_index('loci_name')    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "db860c98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L2/3 IT</th>\n",
       "      <th>L4/5 IT</th>\n",
       "      <th>L5 IT</th>\n",
       "      <th>L6 IT</th>\n",
       "      <th>L5 ET</th>\n",
       "      <th>L6 CT</th>\n",
       "      <th>L5/6 NP</th>\n",
       "      <th>L6b</th>\n",
       "      <th>Vip</th>\n",
       "      <th>Pvalb</th>\n",
       "      <th>...</th>\n",
       "      <th>Oligo</th>\n",
       "      <th>OPC</th>\n",
       "      <th>Micro</th>\n",
       "      <th>Astro</th>\n",
       "      <th>Endo</th>\n",
       "      <th>Peri</th>\n",
       "      <th>Sncg</th>\n",
       "      <th>enhancer_name</th>\n",
       "      <th>Marker_group</th>\n",
       "      <th>Expression_change</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>-0.246629</td>\n",
       "      <td>-0.348931</td>\n",
       "      <td>-0.415534</td>\n",
       "      <td>-0.269979</td>\n",
       "      <td>-0.291472</td>\n",
       "      <td>-0.270590</td>\n",
       "      <td>-0.626703</td>\n",
       "      <td>-0.489705</td>\n",
       "      <td>-0.673363</td>\n",
       "      <td>-0.630066</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.116000</td>\n",
       "      <td>-0.083814</td>\n",
       "      <td>0.234512</td>\n",
       "      <td>-0.073540</td>\n",
       "      <td>0.492773</td>\n",
       "      <td>0.359415</td>\n",
       "      <td>-0.355233</td>\n",
       "      <td>1:9759317-9835402</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_21522568_21534512</th>\n",
       "      <td>0.157988</td>\n",
       "      <td>0.177744</td>\n",
       "      <td>0.104602</td>\n",
       "      <td>0.134259</td>\n",
       "      <td>-0.301557</td>\n",
       "      <td>-0.356024</td>\n",
       "      <td>-0.052600</td>\n",
       "      <td>-0.521367</td>\n",
       "      <td>-0.094214</td>\n",
       "      <td>-0.230084</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.307273</td>\n",
       "      <td>-0.311462</td>\n",
       "      <td>-0.231718</td>\n",
       "      <td>-0.260091</td>\n",
       "      <td>0.163623</td>\n",
       "      <td>0.212470</td>\n",
       "      <td>-0.053455</td>\n",
       "      <td>1:21485446-21554433</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_62926942_62936078</th>\n",
       "      <td>-0.100317</td>\n",
       "      <td>-0.186968</td>\n",
       "      <td>-0.101095</td>\n",
       "      <td>-0.103074</td>\n",
       "      <td>-0.243412</td>\n",
       "      <td>-0.207737</td>\n",
       "      <td>0.019004</td>\n",
       "      <td>-0.116716</td>\n",
       "      <td>-0.104934</td>\n",
       "      <td>-0.069307</td>\n",
       "      <td>...</td>\n",
       "      <td>0.141556</td>\n",
       "      <td>0.001419</td>\n",
       "      <td>0.224659</td>\n",
       "      <td>0.189864</td>\n",
       "      <td>0.547692</td>\n",
       "      <td>0.661718</td>\n",
       "      <td>0.074894</td>\n",
       "      <td>1:62900963-62948089</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2_135670388_135680581</th>\n",
       "      <td>0.014333</td>\n",
       "      <td>0.050795</td>\n",
       "      <td>-0.122257</td>\n",
       "      <td>-0.000108</td>\n",
       "      <td>-0.186657</td>\n",
       "      <td>-0.017936</td>\n",
       "      <td>0.251213</td>\n",
       "      <td>-0.147799</td>\n",
       "      <td>-0.482289</td>\n",
       "      <td>-0.124177</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034957</td>\n",
       "      <td>0.001804</td>\n",
       "      <td>0.086982</td>\n",
       "      <td>0.188080</td>\n",
       "      <td>0.871917</td>\n",
       "      <td>0.577283</td>\n",
       "      <td>-0.016565</td>\n",
       "      <td>2:135658540-135684141</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3_157181787_157191783</th>\n",
       "      <td>-0.037845</td>\n",
       "      <td>-0.052078</td>\n",
       "      <td>-0.223777</td>\n",
       "      <td>-0.208545</td>\n",
       "      <td>-0.208042</td>\n",
       "      <td>-0.158589</td>\n",
       "      <td>-0.164689</td>\n",
       "      <td>-0.004944</td>\n",
       "      <td>-0.290141</td>\n",
       "      <td>-0.352956</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.112965</td>\n",
       "      <td>-0.003305</td>\n",
       "      <td>-0.247953</td>\n",
       "      <td>0.016224</td>\n",
       "      <td>0.286308</td>\n",
       "      <td>0.259769</td>\n",
       "      <td>-0.129659</td>\n",
       "      <td>3:157149588-157216143</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr4_8587053_8598914</th>\n",
       "      <td>-0.545159</td>\n",
       "      <td>-0.434658</td>\n",
       "      <td>-0.480290</td>\n",
       "      <td>-0.515397</td>\n",
       "      <td>-0.290258</td>\n",
       "      <td>-0.348028</td>\n",
       "      <td>-0.125066</td>\n",
       "      <td>-0.507469</td>\n",
       "      <td>-0.213955</td>\n",
       "      <td>-0.422294</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.107257</td>\n",
       "      <td>-0.127963</td>\n",
       "      <td>0.221188</td>\n",
       "      <td>-0.051064</td>\n",
       "      <td>0.513435</td>\n",
       "      <td>0.414790</td>\n",
       "      <td>-0.245148</td>\n",
       "      <td>4:8556598-8611298</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5_44692530_44701463</th>\n",
       "      <td>-0.082847</td>\n",
       "      <td>-0.078817</td>\n",
       "      <td>-0.239958</td>\n",
       "      <td>-0.089679</td>\n",
       "      <td>-0.157831</td>\n",
       "      <td>-0.173129</td>\n",
       "      <td>-0.054545</td>\n",
       "      <td>-0.223415</td>\n",
       "      <td>-0.776268</td>\n",
       "      <td>-0.468562</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.172726</td>\n",
       "      <td>-0.163259</td>\n",
       "      <td>-0.026733</td>\n",
       "      <td>0.008044</td>\n",
       "      <td>0.419258</td>\n",
       "      <td>0.343656</td>\n",
       "      <td>-0.269089</td>\n",
       "      <td>5:44646554-44742497</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr6_17135040_17145456</th>\n",
       "      <td>-0.475776</td>\n",
       "      <td>-0.641122</td>\n",
       "      <td>-0.664534</td>\n",
       "      <td>-0.397868</td>\n",
       "      <td>-0.604248</td>\n",
       "      <td>-0.598666</td>\n",
       "      <td>-0.673720</td>\n",
       "      <td>-0.620775</td>\n",
       "      <td>-0.372006</td>\n",
       "      <td>-1.304393</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.383223</td>\n",
       "      <td>-0.242896</td>\n",
       "      <td>-0.176918</td>\n",
       "      <td>-0.195448</td>\n",
       "      <td>0.402019</td>\n",
       "      <td>0.229576</td>\n",
       "      <td>-0.119791</td>\n",
       "      <td>6:17102402-17171916</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr6_118622298_118630305</th>\n",
       "      <td>0.504521</td>\n",
       "      <td>0.523629</td>\n",
       "      <td>0.481617</td>\n",
       "      <td>0.548814</td>\n",
       "      <td>0.232272</td>\n",
       "      <td>0.467074</td>\n",
       "      <td>0.388450</td>\n",
       "      <td>0.335871</td>\n",
       "      <td>0.270484</td>\n",
       "      <td>0.288255</td>\n",
       "      <td>...</td>\n",
       "      <td>0.383697</td>\n",
       "      <td>0.371156</td>\n",
       "      <td>0.201901</td>\n",
       "      <td>0.320695</td>\n",
       "      <td>0.551360</td>\n",
       "      <td>0.810908</td>\n",
       "      <td>0.135921</td>\n",
       "      <td>6:118593580-118649770</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr8_8625672_8635363</th>\n",
       "      <td>-0.177647</td>\n",
       "      <td>-0.175592</td>\n",
       "      <td>0.010411</td>\n",
       "      <td>-0.135772</td>\n",
       "      <td>0.029133</td>\n",
       "      <td>-0.184167</td>\n",
       "      <td>-0.197762</td>\n",
       "      <td>-0.208310</td>\n",
       "      <td>-0.156238</td>\n",
       "      <td>-0.290043</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.201971</td>\n",
       "      <td>-0.216499</td>\n",
       "      <td>-0.110555</td>\n",
       "      <td>-0.208502</td>\n",
       "      <td>0.445225</td>\n",
       "      <td>0.298462</td>\n",
       "      <td>-0.382071</td>\n",
       "      <td>8:8600019-8650327</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           L2/3 IT   L4/5 IT     L5 IT     L6 IT     L5 ET  \\\n",
       "loci_name                                                                    \n",
       "chr1_9799472_9811359     -0.246629 -0.348931 -0.415534 -0.269979 -0.291472   \n",
       "chr1_21522568_21534512    0.157988  0.177744  0.104602  0.134259 -0.301557   \n",
       "chr1_62926942_62936078   -0.100317 -0.186968 -0.101095 -0.103074 -0.243412   \n",
       "chr2_135670388_135680581  0.014333  0.050795 -0.122257 -0.000108 -0.186657   \n",
       "chr3_157181787_157191783 -0.037845 -0.052078 -0.223777 -0.208545 -0.208042   \n",
       "chr4_8587053_8598914     -0.545159 -0.434658 -0.480290 -0.515397 -0.290258   \n",
       "chr5_44692530_44701463   -0.082847 -0.078817 -0.239958 -0.089679 -0.157831   \n",
       "chr6_17135040_17145456   -0.475776 -0.641122 -0.664534 -0.397868 -0.604248   \n",
       "chr6_118622298_118630305  0.504521  0.523629  0.481617  0.548814  0.232272   \n",
       "chr8_8625672_8635363     -0.177647 -0.175592  0.010411 -0.135772  0.029133   \n",
       "\n",
       "                             L6 CT   L5/6 NP       L6b       Vip     Pvalb  \\\n",
       "loci_name                                                                    \n",
       "chr1_9799472_9811359     -0.270590 -0.626703 -0.489705 -0.673363 -0.630066   \n",
       "chr1_21522568_21534512   -0.356024 -0.052600 -0.521367 -0.094214 -0.230084   \n",
       "chr1_62926942_62936078   -0.207737  0.019004 -0.116716 -0.104934 -0.069307   \n",
       "chr2_135670388_135680581 -0.017936  0.251213 -0.147799 -0.482289 -0.124177   \n",
       "chr3_157181787_157191783 -0.158589 -0.164689 -0.004944 -0.290141 -0.352956   \n",
       "chr4_8587053_8598914     -0.348028 -0.125066 -0.507469 -0.213955 -0.422294   \n",
       "chr5_44692530_44701463   -0.173129 -0.054545 -0.223415 -0.776268 -0.468562   \n",
       "chr6_17135040_17145456   -0.598666 -0.673720 -0.620775 -0.372006 -1.304393   \n",
       "chr6_118622298_118630305  0.467074  0.388450  0.335871  0.270484  0.288255   \n",
       "chr8_8625672_8635363     -0.184167 -0.197762 -0.208310 -0.156238 -0.290043   \n",
       "\n",
       "                          ...     Oligo       OPC     Micro     Astro  \\\n",
       "loci_name                 ...                                           \n",
       "chr1_9799472_9811359      ... -0.116000 -0.083814  0.234512 -0.073540   \n",
       "chr1_21522568_21534512    ... -0.307273 -0.311462 -0.231718 -0.260091   \n",
       "chr1_62926942_62936078    ...  0.141556  0.001419  0.224659  0.189864   \n",
       "chr2_135670388_135680581  ...  0.034957  0.001804  0.086982  0.188080   \n",
       "chr3_157181787_157191783  ... -0.112965 -0.003305 -0.247953  0.016224   \n",
       "chr4_8587053_8598914      ... -0.107257 -0.127963  0.221188 -0.051064   \n",
       "chr5_44692530_44701463    ... -0.172726 -0.163259 -0.026733  0.008044   \n",
       "chr6_17135040_17145456    ... -0.383223 -0.242896 -0.176918 -0.195448   \n",
       "chr6_118622298_118630305  ...  0.383697  0.371156  0.201901  0.320695   \n",
       "chr8_8625672_8635363      ... -0.201971 -0.216499 -0.110555 -0.208502   \n",
       "\n",
       "                              Endo      Peri      Sncg          enhancer_name  \\\n",
       "loci_name                                                                       \n",
       "chr1_9799472_9811359      0.492773  0.359415 -0.355233      1:9759317-9835402   \n",
       "chr1_21522568_21534512    0.163623  0.212470 -0.053455    1:21485446-21554433   \n",
       "chr1_62926942_62936078    0.547692  0.661718  0.074894    1:62900963-62948089   \n",
       "chr2_135670388_135680581  0.871917  0.577283 -0.016565  2:135658540-135684141   \n",
       "chr3_157181787_157191783  0.286308  0.259769 -0.129659  3:157149588-157216143   \n",
       "chr4_8587053_8598914      0.513435  0.414790 -0.245148      4:8556598-8611298   \n",
       "chr5_44692530_44701463    0.419258  0.343656 -0.269089    5:44646554-44742497   \n",
       "chr6_17135040_17145456    0.402019  0.229576 -0.119791    6:17102402-17171916   \n",
       "chr6_118622298_118630305  0.551360  0.810908  0.135921  6:118593580-118649770   \n",
       "chr8_8625672_8635363      0.445225  0.298462 -0.382071      8:8600019-8650327   \n",
       "\n",
       "                          Marker_group Expression_change  \n",
       "loci_name                                                 \n",
       "chr1_9799472_9811359           L2/3 IT       upregulated  \n",
       "chr1_21522568_21534512         L2/3 IT       upregulated  \n",
       "chr1_62926942_62936078         L2/3 IT       upregulated  \n",
       "chr2_135670388_135680581       L2/3 IT       upregulated  \n",
       "chr3_157181787_157191783       L2/3 IT       upregulated  \n",
       "chr4_8587053_8598914           L2/3 IT       upregulated  \n",
       "chr5_44692530_44701463         L2/3 IT       upregulated  \n",
       "chr6_17135040_17145456         L2/3 IT       upregulated  \n",
       "chr6_118622298_118630305       L2/3 IT       upregulated  \n",
       "chr8_8625672_8635363           L2/3 IT       upregulated  \n",
       "\n",
       "[10 rows x 22 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "compiled_df.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d5977b01",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L2/3 IT</th>\n",
       "      <th>L4/5 IT</th>\n",
       "      <th>L5 IT</th>\n",
       "      <th>L6 IT</th>\n",
       "      <th>L5 ET</th>\n",
       "      <th>L6 CT</th>\n",
       "      <th>L5/6 NP</th>\n",
       "      <th>L6b</th>\n",
       "      <th>Vip</th>\n",
       "      <th>Pvalb</th>\n",
       "      <th>...</th>\n",
       "      <th>Oligo</th>\n",
       "      <th>OPC</th>\n",
       "      <th>Micro</th>\n",
       "      <th>Astro</th>\n",
       "      <th>Endo</th>\n",
       "      <th>Peri</th>\n",
       "      <th>Sncg</th>\n",
       "      <th>enhancer_name</th>\n",
       "      <th>Marker_group</th>\n",
       "      <th>Expression_change</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>-0.246629</td>\n",
       "      <td>-0.348931</td>\n",
       "      <td>-0.415534</td>\n",
       "      <td>-0.269979</td>\n",
       "      <td>-0.291472</td>\n",
       "      <td>-0.270590</td>\n",
       "      <td>-0.626703</td>\n",
       "      <td>-0.489705</td>\n",
       "      <td>-0.673363</td>\n",
       "      <td>-0.630066</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.116000</td>\n",
       "      <td>-0.083814</td>\n",
       "      <td>0.234512</td>\n",
       "      <td>-0.073540</td>\n",
       "      <td>0.492773</td>\n",
       "      <td>0.359415</td>\n",
       "      <td>-0.355233</td>\n",
       "      <td>1:9759317-9835402</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_21522568_21534512</th>\n",
       "      <td>0.157988</td>\n",
       "      <td>0.177744</td>\n",
       "      <td>0.104602</td>\n",
       "      <td>0.134259</td>\n",
       "      <td>-0.301557</td>\n",
       "      <td>-0.356024</td>\n",
       "      <td>-0.052600</td>\n",
       "      <td>-0.521367</td>\n",
       "      <td>-0.094214</td>\n",
       "      <td>-0.230084</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.307273</td>\n",
       "      <td>-0.311462</td>\n",
       "      <td>-0.231718</td>\n",
       "      <td>-0.260091</td>\n",
       "      <td>0.163623</td>\n",
       "      <td>0.212470</td>\n",
       "      <td>-0.053455</td>\n",
       "      <td>1:21485446-21554433</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_62926942_62936078</th>\n",
       "      <td>-0.100317</td>\n",
       "      <td>-0.186968</td>\n",
       "      <td>-0.101095</td>\n",
       "      <td>-0.103074</td>\n",
       "      <td>-0.243412</td>\n",
       "      <td>-0.207737</td>\n",
       "      <td>0.019004</td>\n",
       "      <td>-0.116716</td>\n",
       "      <td>-0.104934</td>\n",
       "      <td>-0.069307</td>\n",
       "      <td>...</td>\n",
       "      <td>0.141556</td>\n",
       "      <td>0.001419</td>\n",
       "      <td>0.224659</td>\n",
       "      <td>0.189864</td>\n",
       "      <td>0.547692</td>\n",
       "      <td>0.661718</td>\n",
       "      <td>0.074894</td>\n",
       "      <td>1:62900963-62948089</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2_135670388_135680581</th>\n",
       "      <td>0.014333</td>\n",
       "      <td>0.050795</td>\n",
       "      <td>-0.122257</td>\n",
       "      <td>-0.000108</td>\n",
       "      <td>-0.186657</td>\n",
       "      <td>-0.017936</td>\n",
       "      <td>0.251213</td>\n",
       "      <td>-0.147799</td>\n",
       "      <td>-0.482289</td>\n",
       "      <td>-0.124177</td>\n",
       "      <td>...</td>\n",
       "      <td>0.034957</td>\n",
       "      <td>0.001804</td>\n",
       "      <td>0.086982</td>\n",
       "      <td>0.188080</td>\n",
       "      <td>0.871917</td>\n",
       "      <td>0.577283</td>\n",
       "      <td>-0.016565</td>\n",
       "      <td>2:135658540-135684141</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3_157181787_157191783</th>\n",
       "      <td>-0.037845</td>\n",
       "      <td>-0.052078</td>\n",
       "      <td>-0.223777</td>\n",
       "      <td>-0.208545</td>\n",
       "      <td>-0.208042</td>\n",
       "      <td>-0.158589</td>\n",
       "      <td>-0.164689</td>\n",
       "      <td>-0.004944</td>\n",
       "      <td>-0.290141</td>\n",
       "      <td>-0.352956</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.112965</td>\n",
       "      <td>-0.003305</td>\n",
       "      <td>-0.247953</td>\n",
       "      <td>0.016224</td>\n",
       "      <td>0.286308</td>\n",
       "      <td>0.259769</td>\n",
       "      <td>-0.129659</td>\n",
       "      <td>3:157149588-157216143</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_21522568_21534512</th>\n",
       "      <td>0.157988</td>\n",
       "      <td>0.177744</td>\n",
       "      <td>0.104602</td>\n",
       "      <td>0.134259</td>\n",
       "      <td>-0.301557</td>\n",
       "      <td>-0.356024</td>\n",
       "      <td>-0.052600</td>\n",
       "      <td>-0.521367</td>\n",
       "      <td>-0.094214</td>\n",
       "      <td>-0.230084</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.307273</td>\n",
       "      <td>-0.311462</td>\n",
       "      <td>-0.231718</td>\n",
       "      <td>-0.260091</td>\n",
       "      <td>0.163623</td>\n",
       "      <td>0.212470</td>\n",
       "      <td>-0.053455</td>\n",
       "      <td>1:21485446-21554433</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5_126260039_126271505</th>\n",
       "      <td>0.651816</td>\n",
       "      <td>0.697825</td>\n",
       "      <td>0.742660</td>\n",
       "      <td>0.739041</td>\n",
       "      <td>0.679804</td>\n",
       "      <td>0.730284</td>\n",
       "      <td>0.557458</td>\n",
       "      <td>0.570463</td>\n",
       "      <td>0.320638</td>\n",
       "      <td>0.750361</td>\n",
       "      <td>...</td>\n",
       "      <td>0.705887</td>\n",
       "      <td>0.529660</td>\n",
       "      <td>0.254934</td>\n",
       "      <td>0.454784</td>\n",
       "      <td>0.302275</td>\n",
       "      <td>0.517443</td>\n",
       "      <td>0.296106</td>\n",
       "      <td>5:126238032-126282172</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12_13114005_13122283</th>\n",
       "      <td>-0.380858</td>\n",
       "      <td>-0.335854</td>\n",
       "      <td>-0.479643</td>\n",
       "      <td>-0.410447</td>\n",
       "      <td>-0.382271</td>\n",
       "      <td>-0.407790</td>\n",
       "      <td>-0.871882</td>\n",
       "      <td>-0.635019</td>\n",
       "      <td>0.030348</td>\n",
       "      <td>-0.198165</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.175269</td>\n",
       "      <td>-0.178913</td>\n",
       "      <td>-0.118585</td>\n",
       "      <td>-0.089060</td>\n",
       "      <td>0.504227</td>\n",
       "      <td>0.389654</td>\n",
       "      <td>-0.156835</td>\n",
       "      <td>12:13079761-13145339</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17_73913910_73922684</th>\n",
       "      <td>0.051063</td>\n",
       "      <td>-0.081972</td>\n",
       "      <td>-0.133891</td>\n",
       "      <td>0.078060</td>\n",
       "      <td>-0.213973</td>\n",
       "      <td>0.008236</td>\n",
       "      <td>-0.058412</td>\n",
       "      <td>-0.117514</td>\n",
       "      <td>0.255262</td>\n",
       "      <td>0.080011</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.078926</td>\n",
       "      <td>-0.005074</td>\n",
       "      <td>0.136457</td>\n",
       "      <td>-0.061259</td>\n",
       "      <td>0.563861</td>\n",
       "      <td>0.429495</td>\n",
       "      <td>0.080657</td>\n",
       "      <td>17:73881483-73942612</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12_113297448_113307996</th>\n",
       "      <td>-0.113798</td>\n",
       "      <td>-0.141071</td>\n",
       "      <td>-0.020921</td>\n",
       "      <td>-0.075515</td>\n",
       "      <td>0.123022</td>\n",
       "      <td>-0.063136</td>\n",
       "      <td>-0.157309</td>\n",
       "      <td>-0.008274</td>\n",
       "      <td>-0.291443</td>\n",
       "      <td>0.081501</td>\n",
       "      <td>...</td>\n",
       "      <td>0.077988</td>\n",
       "      <td>0.016971</td>\n",
       "      <td>0.148314</td>\n",
       "      <td>-0.080520</td>\n",
       "      <td>0.438105</td>\n",
       "      <td>0.392136</td>\n",
       "      <td>-0.149094</td>\n",
       "      <td>12:113275834-113322430</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1569 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            L2/3 IT   L4/5 IT     L5 IT     L6 IT     L5 ET  \\\n",
       "loci_name                                                                     \n",
       "chr1_9799472_9811359      -0.246629 -0.348931 -0.415534 -0.269979 -0.291472   \n",
       "chr1_21522568_21534512     0.157988  0.177744  0.104602  0.134259 -0.301557   \n",
       "chr1_62926942_62936078    -0.100317 -0.186968 -0.101095 -0.103074 -0.243412   \n",
       "chr2_135670388_135680581   0.014333  0.050795 -0.122257 -0.000108 -0.186657   \n",
       "chr3_157181787_157191783  -0.037845 -0.052078 -0.223777 -0.208545 -0.208042   \n",
       "...                             ...       ...       ...       ...       ...   \n",
       "chr1_21522568_21534512     0.157988  0.177744  0.104602  0.134259 -0.301557   \n",
       "chr5_126260039_126271505   0.651816  0.697825  0.742660  0.739041  0.679804   \n",
       "chr12_13114005_13122283   -0.380858 -0.335854 -0.479643 -0.410447 -0.382271   \n",
       "chr17_73913910_73922684    0.051063 -0.081972 -0.133891  0.078060 -0.213973   \n",
       "chr12_113297448_113307996 -0.113798 -0.141071 -0.020921 -0.075515  0.123022   \n",
       "\n",
       "                              L6 CT   L5/6 NP       L6b       Vip     Pvalb  \\\n",
       "loci_name                                                                     \n",
       "chr1_9799472_9811359      -0.270590 -0.626703 -0.489705 -0.673363 -0.630066   \n",
       "chr1_21522568_21534512    -0.356024 -0.052600 -0.521367 -0.094214 -0.230084   \n",
       "chr1_62926942_62936078    -0.207737  0.019004 -0.116716 -0.104934 -0.069307   \n",
       "chr2_135670388_135680581  -0.017936  0.251213 -0.147799 -0.482289 -0.124177   \n",
       "chr3_157181787_157191783  -0.158589 -0.164689 -0.004944 -0.290141 -0.352956   \n",
       "...                             ...       ...       ...       ...       ...   \n",
       "chr1_21522568_21534512    -0.356024 -0.052600 -0.521367 -0.094214 -0.230084   \n",
       "chr5_126260039_126271505   0.730284  0.557458  0.570463  0.320638  0.750361   \n",
       "chr12_13114005_13122283   -0.407790 -0.871882 -0.635019  0.030348 -0.198165   \n",
       "chr17_73913910_73922684    0.008236 -0.058412 -0.117514  0.255262  0.080011   \n",
       "chr12_113297448_113307996 -0.063136 -0.157309 -0.008274 -0.291443  0.081501   \n",
       "\n",
       "                           ...     Oligo       OPC     Micro     Astro  \\\n",
       "loci_name                  ...                                           \n",
       "chr1_9799472_9811359       ... -0.116000 -0.083814  0.234512 -0.073540   \n",
       "chr1_21522568_21534512     ... -0.307273 -0.311462 -0.231718 -0.260091   \n",
       "chr1_62926942_62936078     ...  0.141556  0.001419  0.224659  0.189864   \n",
       "chr2_135670388_135680581   ...  0.034957  0.001804  0.086982  0.188080   \n",
       "chr3_157181787_157191783   ... -0.112965 -0.003305 -0.247953  0.016224   \n",
       "...                        ...       ...       ...       ...       ...   \n",
       "chr1_21522568_21534512     ... -0.307273 -0.311462 -0.231718 -0.260091   \n",
       "chr5_126260039_126271505   ...  0.705887  0.529660  0.254934  0.454784   \n",
       "chr12_13114005_13122283    ... -0.175269 -0.178913 -0.118585 -0.089060   \n",
       "chr17_73913910_73922684    ... -0.078926 -0.005074  0.136457 -0.061259   \n",
       "chr12_113297448_113307996  ...  0.077988  0.016971  0.148314 -0.080520   \n",
       "\n",
       "                               Endo      Peri      Sncg  \\\n",
       "loci_name                                                 \n",
       "chr1_9799472_9811359       0.492773  0.359415 -0.355233   \n",
       "chr1_21522568_21534512     0.163623  0.212470 -0.053455   \n",
       "chr1_62926942_62936078     0.547692  0.661718  0.074894   \n",
       "chr2_135670388_135680581   0.871917  0.577283 -0.016565   \n",
       "chr3_157181787_157191783   0.286308  0.259769 -0.129659   \n",
       "...                             ...       ...       ...   \n",
       "chr1_21522568_21534512     0.163623  0.212470 -0.053455   \n",
       "chr5_126260039_126271505   0.302275  0.517443  0.296106   \n",
       "chr12_13114005_13122283    0.504227  0.389654 -0.156835   \n",
       "chr17_73913910_73922684    0.563861  0.429495  0.080657   \n",
       "chr12_113297448_113307996  0.438105  0.392136 -0.149094   \n",
       "\n",
       "                                    enhancer_name  Marker_group  \\\n",
       "loci_name                                                         \n",
       "chr1_9799472_9811359            1:9759317-9835402       L2/3 IT   \n",
       "chr1_21522568_21534512        1:21485446-21554433       L2/3 IT   \n",
       "chr1_62926942_62936078        1:62900963-62948089       L2/3 IT   \n",
       "chr2_135670388_135680581    2:135658540-135684141       L2/3 IT   \n",
       "chr3_157181787_157191783    3:157149588-157216143       L2/3 IT   \n",
       "...                                           ...           ...   \n",
       "chr1_21522568_21534512        1:21485446-21554433           Sst   \n",
       "chr5_126260039_126271505    5:126238032-126282172           Sst   \n",
       "chr12_13114005_13122283      12:13079761-13145339           Sst   \n",
       "chr17_73913910_73922684      17:73881483-73942612           Sst   \n",
       "chr12_113297448_113307996  12:113275834-113322430           Sst   \n",
       "\n",
       "                          Expression_change  \n",
       "loci_name                                    \n",
       "chr1_9799472_9811359            upregulated  \n",
       "chr1_21522568_21534512          upregulated  \n",
       "chr1_62926942_62936078          upregulated  \n",
       "chr2_135670388_135680581        upregulated  \n",
       "chr3_157181787_157191783        upregulated  \n",
       "...                                     ...  \n",
       "chr1_21522568_21534512       random_control  \n",
       "chr5_126260039_126271505     random_control  \n",
       "chr12_13114005_13122283      random_control  \n",
       "chr17_73913910_73922684      random_control  \n",
       "chr12_113297448_113307996    random_control  \n",
       "\n",
       "[1569 rows x 22 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_groupby_savename = 'subclass'\n",
    "compiled_df_savename = os.path.join(output_analysis_folder,f'{_groupby_savename}_SE_AB_trans_ratio_regressed_heatmap_pm_majorType.csv')\n",
    "compiled_df.to_csv(compiled_df_savename)\n",
    "compiled_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9425029c",
   "metadata": {},
   "source": [
    "## normalize the heatmap to relative foldchange"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "e444b013",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>L2/3 IT</th>\n",
       "      <th>L4/5 IT</th>\n",
       "      <th>L5 IT</th>\n",
       "      <th>L6 IT</th>\n",
       "      <th>L5 ET</th>\n",
       "      <th>L6 CT</th>\n",
       "      <th>L5/6 NP</th>\n",
       "      <th>L6b</th>\n",
       "      <th>Vip</th>\n",
       "      <th>Pvalb</th>\n",
       "      <th>Lamp5</th>\n",
       "      <th>Sst</th>\n",
       "      <th>enhancer_name</th>\n",
       "      <th>Marker_group</th>\n",
       "      <th>Expression_change</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.102301</td>\n",
       "      <td>-0.168905</td>\n",
       "      <td>-0.023350</td>\n",
       "      <td>-0.044843</td>\n",
       "      <td>-0.023960</td>\n",
       "      <td>-0.380073</td>\n",
       "      <td>-0.243076</td>\n",
       "      <td>-0.426734</td>\n",
       "      <td>-0.383437</td>\n",
       "      <td>0.138778</td>\n",
       "      <td>-0.246001</td>\n",
       "      <td>1:9759317-9835402</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_21522568_21534512</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.019756</td>\n",
       "      <td>-0.053386</td>\n",
       "      <td>-0.023729</td>\n",
       "      <td>-0.459545</td>\n",
       "      <td>-0.514012</td>\n",
       "      <td>-0.210588</td>\n",
       "      <td>-0.679355</td>\n",
       "      <td>-0.252202</td>\n",
       "      <td>-0.388072</td>\n",
       "      <td>-0.056369</td>\n",
       "      <td>-0.586402</td>\n",
       "      <td>1:21485446-21554433</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_62926942_62936078</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.086651</td>\n",
       "      <td>-0.000778</td>\n",
       "      <td>-0.002757</td>\n",
       "      <td>-0.143095</td>\n",
       "      <td>-0.107420</td>\n",
       "      <td>0.119321</td>\n",
       "      <td>-0.016398</td>\n",
       "      <td>-0.004617</td>\n",
       "      <td>0.031010</td>\n",
       "      <td>0.277695</td>\n",
       "      <td>-0.205005</td>\n",
       "      <td>1:62900963-62948089</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr2_135670388_135680581</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.036462</td>\n",
       "      <td>-0.136590</td>\n",
       "      <td>-0.014441</td>\n",
       "      <td>-0.200990</td>\n",
       "      <td>-0.032269</td>\n",
       "      <td>0.236881</td>\n",
       "      <td>-0.162132</td>\n",
       "      <td>-0.496622</td>\n",
       "      <td>-0.138510</td>\n",
       "      <td>0.378745</td>\n",
       "      <td>-0.249080</td>\n",
       "      <td>2:135658540-135684141</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr3_157181787_157191783</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.014233</td>\n",
       "      <td>-0.185932</td>\n",
       "      <td>-0.170700</td>\n",
       "      <td>-0.170197</td>\n",
       "      <td>-0.120744</td>\n",
       "      <td>-0.126844</td>\n",
       "      <td>0.032901</td>\n",
       "      <td>-0.252296</td>\n",
       "      <td>-0.315111</td>\n",
       "      <td>-0.177691</td>\n",
       "      <td>0.108569</td>\n",
       "      <td>3:157149588-157216143</td>\n",
       "      <td>L2/3 IT</td>\n",
       "      <td>upregulated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_21522568_21534512</th>\n",
       "      <td>0.586402</td>\n",
       "      <td>0.606159</td>\n",
       "      <td>0.533016</td>\n",
       "      <td>0.562673</td>\n",
       "      <td>0.126857</td>\n",
       "      <td>0.072390</td>\n",
       "      <td>0.375815</td>\n",
       "      <td>-0.092952</td>\n",
       "      <td>0.334201</td>\n",
       "      <td>0.198330</td>\n",
       "      <td>0.530033</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1:21485446-21554433</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr5_126260039_126271505</th>\n",
       "      <td>0.012783</td>\n",
       "      <td>0.058792</td>\n",
       "      <td>0.103627</td>\n",
       "      <td>0.100008</td>\n",
       "      <td>0.040771</td>\n",
       "      <td>0.091252</td>\n",
       "      <td>-0.081575</td>\n",
       "      <td>-0.068570</td>\n",
       "      <td>-0.318395</td>\n",
       "      <td>0.111328</td>\n",
       "      <td>0.197487</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5:126238032-126282172</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12_13114005_13122283</th>\n",
       "      <td>-0.280377</td>\n",
       "      <td>-0.235374</td>\n",
       "      <td>-0.379163</td>\n",
       "      <td>-0.309967</td>\n",
       "      <td>-0.281791</td>\n",
       "      <td>-0.307309</td>\n",
       "      <td>-0.771401</td>\n",
       "      <td>-0.534538</td>\n",
       "      <td>0.130829</td>\n",
       "      <td>-0.097684</td>\n",
       "      <td>-0.339021</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12:13079761-13145339</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr17_73913910_73922684</th>\n",
       "      <td>-0.002295</td>\n",
       "      <td>-0.135330</td>\n",
       "      <td>-0.187249</td>\n",
       "      <td>0.024702</td>\n",
       "      <td>-0.267331</td>\n",
       "      <td>-0.045121</td>\n",
       "      <td>-0.111770</td>\n",
       "      <td>-0.170872</td>\n",
       "      <td>0.201904</td>\n",
       "      <td>0.026653</td>\n",
       "      <td>0.297093</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>17:73881483-73942612</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr12_113297448_113307996</th>\n",
       "      <td>0.010819</td>\n",
       "      <td>-0.016454</td>\n",
       "      <td>0.103696</td>\n",
       "      <td>0.049102</td>\n",
       "      <td>0.247639</td>\n",
       "      <td>0.061481</td>\n",
       "      <td>-0.032692</td>\n",
       "      <td>0.116343</td>\n",
       "      <td>-0.166826</td>\n",
       "      <td>0.206118</td>\n",
       "      <td>-0.183218</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12:113275834-113322430</td>\n",
       "      <td>Sst</td>\n",
       "      <td>random_control</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1569 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                            L2/3 IT   L4/5 IT     L5 IT     L6 IT     L5 ET  \\\n",
       "loci_name                                                                     \n",
       "chr1_9799472_9811359       0.000000 -0.102301 -0.168905 -0.023350 -0.044843   \n",
       "chr1_21522568_21534512     0.000000  0.019756 -0.053386 -0.023729 -0.459545   \n",
       "chr1_62926942_62936078     0.000000 -0.086651 -0.000778 -0.002757 -0.143095   \n",
       "chr2_135670388_135680581   0.000000  0.036462 -0.136590 -0.014441 -0.200990   \n",
       "chr3_157181787_157191783   0.000000 -0.014233 -0.185932 -0.170700 -0.170197   \n",
       "...                             ...       ...       ...       ...       ...   \n",
       "chr1_21522568_21534512     0.586402  0.606159  0.533016  0.562673  0.126857   \n",
       "chr5_126260039_126271505   0.012783  0.058792  0.103627  0.100008  0.040771   \n",
       "chr12_13114005_13122283   -0.280377 -0.235374 -0.379163 -0.309967 -0.281791   \n",
       "chr17_73913910_73922684   -0.002295 -0.135330 -0.187249  0.024702 -0.267331   \n",
       "chr12_113297448_113307996  0.010819 -0.016454  0.103696  0.049102  0.247639   \n",
       "\n",
       "                              L6 CT   L5/6 NP       L6b       Vip     Pvalb  \\\n",
       "loci_name                                                                     \n",
       "chr1_9799472_9811359      -0.023960 -0.380073 -0.243076 -0.426734 -0.383437   \n",
       "chr1_21522568_21534512    -0.514012 -0.210588 -0.679355 -0.252202 -0.388072   \n",
       "chr1_62926942_62936078    -0.107420  0.119321 -0.016398 -0.004617  0.031010   \n",
       "chr2_135670388_135680581  -0.032269  0.236881 -0.162132 -0.496622 -0.138510   \n",
       "chr3_157181787_157191783  -0.120744 -0.126844  0.032901 -0.252296 -0.315111   \n",
       "...                             ...       ...       ...       ...       ...   \n",
       "chr1_21522568_21534512     0.072390  0.375815 -0.092952  0.334201  0.198330   \n",
       "chr5_126260039_126271505   0.091252 -0.081575 -0.068570 -0.318395  0.111328   \n",
       "chr12_13114005_13122283   -0.307309 -0.771401 -0.534538  0.130829 -0.097684   \n",
       "chr17_73913910_73922684   -0.045121 -0.111770 -0.170872  0.201904  0.026653   \n",
       "chr12_113297448_113307996  0.061481 -0.032692  0.116343 -0.166826  0.206118   \n",
       "\n",
       "                              Lamp5       Sst           enhancer_name  \\\n",
       "loci_name                                                               \n",
       "chr1_9799472_9811359       0.138778 -0.246001       1:9759317-9835402   \n",
       "chr1_21522568_21534512    -0.056369 -0.586402     1:21485446-21554433   \n",
       "chr1_62926942_62936078     0.277695 -0.205005     1:62900963-62948089   \n",
       "chr2_135670388_135680581   0.378745 -0.249080   2:135658540-135684141   \n",
       "chr3_157181787_157191783  -0.177691  0.108569   3:157149588-157216143   \n",
       "...                             ...       ...                     ...   \n",
       "chr1_21522568_21534512     0.530033  0.000000     1:21485446-21554433   \n",
       "chr5_126260039_126271505   0.197487  0.000000   5:126238032-126282172   \n",
       "chr12_13114005_13122283   -0.339021  0.000000    12:13079761-13145339   \n",
       "chr17_73913910_73922684    0.297093  0.000000    17:73881483-73942612   \n",
       "chr12_113297448_113307996 -0.183218  0.000000  12:113275834-113322430   \n",
       "\n",
       "                          Marker_group Expression_change  \n",
       "loci_name                                                 \n",
       "chr1_9799472_9811359           L2/3 IT       upregulated  \n",
       "chr1_21522568_21534512         L2/3 IT       upregulated  \n",
       "chr1_62926942_62936078         L2/3 IT       upregulated  \n",
       "chr2_135670388_135680581       L2/3 IT       upregulated  \n",
       "chr3_157181787_157191783       L2/3 IT       upregulated  \n",
       "...                                ...               ...  \n",
       "chr1_21522568_21534512             Sst    random_control  \n",
       "chr5_126260039_126271505           Sst    random_control  \n",
       "chr12_13114005_13122283            Sst    random_control  \n",
       "chr17_73913910_73922684            Sst    random_control  \n",
       "chr12_113297448_113307996          Sst    random_control  \n",
       "\n",
       "[1569 rows x 15 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "norm_compiled_df = pd.DataFrame(columns=sorted_group_order)\n",
    "\n",
    "# simply loop to get the ref value (aka the value from the marker group)\n",
    "ref_value_list = []\n",
    "for _row_df in compiled_df.iloc():\n",
    "    ref_group = _row_df['Marker_group']\n",
    "    ref_value_list.append(_row_df[ref_group])\n",
    "\n",
    "# norm the relevant columns-dervied matrix\n",
    "mat = compiled_df[sorted_group_order].to_numpy()\n",
    "ref_arr = np.array(ref_value_list)[:, np.newaxis]\n",
    "norm_mat = mat-ref_arr\n",
    "\n",
    "# add back relevant info\n",
    "norm_compiled_df[sorted_group_order]=pd.DataFrame(norm_mat)\n",
    "norm_compiled_df.index = compiled_df.index\n",
    "for _col in ['enhancer_name','Marker_group','Expression_change']:\n",
    "    norm_compiled_df[_col]=compiled_df[_col]\n",
    "\n",
    "norm_compiled_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ba903434",
   "metadata": {},
   "outputs": [],
   "source": [
    "norm_compiled_df_savename = os.path.join(output_analysis_folder,f'{_groupby_savename}_SE_AB_trans_ratio_regressed_heatmap_norm_pm_majorType.csv')\n",
    "norm_compiled_df.to_csv(norm_compiled_df_savename)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": false,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
