{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "6b992bdd",
   "metadata": {},
   "source": [
    "# Design CTP-11 mouse 1000 genome DNA-MERFISH library \n",
    "\n",
    "by Pu Zheng\n",
    "\n",
    "2021.10.20\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "39c6c199",
   "metadata": {},
   "source": [
    "# Table of contents\n",
    "\n",
    "\n",
    "> 0. [Minimum required packages and settings](#0)\n",
    ">>\n",
    ">> 0.1: [import required packages](#0.1)\n",
    ">\n",
    "> 1. [Extract region sequences](#1)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3b105423",
   "metadata": {},
   "source": [
    "<a id='0.1'></a>\n",
    "## 0.1 load required packages"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The link to get [ImageAnalysis3](https://github.com/zhengpuas47/ImageAnalysis3) \n",
    "\n",
    "or the Zhuang lab archived [source_tools](https://github.com/ZhuangLab/Chromatin_Analysis_2020_cell/tree/master/sequential_tracing/source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d213e6f1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20272\n"
     ]
    }
   ],
   "source": [
    "%run \"..\\..\\Startup_py3.py\"\n",
    "sys.path.append(r\"..\\..\\..\\..\\Documents\")\n",
    "\n",
    "import ImageAnalysis3 as ia\n",
    "%matplotlib notebook\n",
    "\n",
    "from ImageAnalysis3 import *\n",
    "print(os.getpid())\n",
    "\n",
    "# library design specific tools\n",
    "from ImageAnalysis3.library_tools import LibraryDesigner as ld\n",
    "from ImageAnalysis3.library_tools import LibraryTools as lt\n",
    "# biopython imports\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio.Blast import NCBIXML"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ab33af3",
   "metadata": {},
   "source": [
    "<a id='1'></a>\n",
    "# 1 Extract region sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9f16a3c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Some folders\n",
    "# human genome\n",
    "reference_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Genomes\\mouse\\GRCm38_ensembl'\n",
    "genome_folder = os.path.join(reference_folder, 'Genome')\n",
    "# Library directories\n",
    "pool_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "76d82624",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- library_folder: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\n",
      "-- sequence_folder: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\n",
      "-- report_folder: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\n"
     ]
    }
   ],
   "source": [
    "resolution = 0\n",
    "flanking = 10000\n",
    "# folder for sub-pool\n",
    "library_folder = os.path.join(pool_folder, f'mouse_genome_1000')\n",
    "if not os.path.exists(library_folder):\n",
    "    print(f\"create library folder: {library_folder}\")\n",
    "    os.makedirs(library_folder)\n",
    "# folder for fasta sequences\n",
    "sequence_folder = os.path.join(library_folder, 'sequences')\n",
    "if not os.path.exists(sequence_folder):\n",
    "    print(f\"create sequence folder: {sequence_folder}\")\n",
    "    os.makedirs(sequence_folder)\n",
    "# folder to save result probes\n",
    "report_folder = os.path.join(library_folder, 'reports')\n",
    "if not os.path.exists(report_folder):\n",
    "    print(f\"create report folder: {report_folder}\")\n",
    "    os.makedirs(report_folder)\n",
    "    \n",
    "print(f\"-- library_folder: {library_folder}\")\n",
    "print(f\"-- sequence_folder: {sequence_folder}\")\n",
    "print(f\"-- report_folder: {report_folder}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "29c5c2d4",
   "metadata": {},
   "source": [
    "## 1.1 select regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b3b06fd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from tqdm.notebook import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "4ac82109",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- load sequence: 1, size=195471971\n",
      "-- load sequence: 2, size=182113224\n",
      "-- load sequence: 3, size=160039680\n",
      "-- load sequence: 4, size=156508116\n",
      "-- load sequence: 5, size=151834684\n",
      "-- load sequence: 6, size=149736546\n",
      "-- load sequence: 7, size=145441459\n",
      "-- load sequence: 8, size=129401213\n",
      "-- load sequence: 9, size=124595110\n",
      "-- load sequence: 10, size=130694993\n",
      "-- load sequence: 11, size=122082543\n",
      "-- load sequence: 12, size=120129022\n",
      "-- load sequence: 13, size=120421639\n",
      "-- load sequence: 14, size=124902244\n",
      "-- load sequence: 15, size=104043685\n",
      "-- load sequence: 16, size=98207768\n",
      "-- load sequence: 17, size=94987271\n",
      "-- load sequence: 18, size=90702639\n",
      "-- load sequence: 19, size=61431566\n",
      "-- load sequence: MT, size=16299\n",
      "-- load sequence: X, size=171031299\n",
      "-- load sequence: Y, size=91744698\n",
      "-- load sequence: JH584299.1, size=953012\n",
      "-- load sequence: GL456233.1, size=336933\n",
      "-- load sequence: JH584301.1, size=259875\n",
      "-- load sequence: GL456211.1, size=241735\n",
      "-- load sequence: GL456350.1, size=227966\n",
      "-- load sequence: JH584293.1, size=207968\n",
      "-- load sequence: GL456221.1, size=206961\n",
      "-- load sequence: JH584297.1, size=205776\n",
      "-- load sequence: JH584296.1, size=199368\n",
      "-- load sequence: GL456354.1, size=195993\n",
      "-- load sequence: JH584294.1, size=191905\n",
      "-- load sequence: JH584298.1, size=184189\n",
      "-- load sequence: JH584300.1, size=182347\n",
      "-- load sequence: GL456219.1, size=175968\n",
      "-- load sequence: GL456210.1, size=169725\n",
      "-- load sequence: JH584303.1, size=158099\n",
      "-- load sequence: JH584302.1, size=155838\n",
      "-- load sequence: GL456212.1, size=153618\n",
      "-- load sequence: JH584304.1, size=114452\n",
      "-- load sequence: GL456379.1, size=72385\n",
      "-- load sequence: GL456216.1, size=66673\n",
      "-- load sequence: GL456393.1, size=55711\n",
      "-- load sequence: GL456366.1, size=47073\n",
      "-- load sequence: GL456367.1, size=42057\n",
      "-- load sequence: GL456239.1, size=40056\n",
      "-- load sequence: GL456213.1, size=39340\n",
      "-- load sequence: GL456383.1, size=38659\n",
      "-- load sequence: GL456385.1, size=35240\n",
      "-- load sequence: GL456360.1, size=31704\n",
      "-- load sequence: GL456378.1, size=31602\n",
      "-- load sequence: GL456389.1, size=28772\n",
      "-- load sequence: GL456372.1, size=28664\n",
      "-- load sequence: GL456370.1, size=26764\n",
      "-- load sequence: GL456381.1, size=25871\n",
      "-- load sequence: GL456387.1, size=24685\n",
      "-- load sequence: GL456390.1, size=24668\n",
      "-- load sequence: GL456394.1, size=24323\n",
      "-- load sequence: GL456392.1, size=23629\n",
      "-- load sequence: GL456382.1, size=23158\n",
      "-- load sequence: GL456359.1, size=22974\n",
      "-- load sequence: GL456396.1, size=21240\n",
      "-- load sequence: GL456368.1, size=20208\n",
      "-- load sequence: JH584292.1, size=14945\n",
      "-- load sequence: JH584295.1, size=1976\n",
      "- 66 sequences loaded.\n"
     ]
    }
   ],
   "source": [
    "# prepare sequence reader\n",
    "reload(ia.library_tools.sequences)\n",
    "seq_rd = ia.library_tools.sequences.sequence_reader(genome_folder)\n",
    "seq_rd.load_sequences() # load genomic sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "a9b0dccf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2725521370\n"
     ]
    }
   ],
   "source": [
    "# calculate total length\n",
    "min_chr_len = 1000000\n",
    "chr_len_dict = {_name:len(_record) for _name, _record in seq_rd.ref_seq_dict.items() if len(_record) > min_chr_len}\n",
    "total_len = sum(list(chr_len_dict.values()))\n",
    "print(total_len)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "0dfc0b03",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1099\n",
      "2500000\n"
     ]
    }
   ],
   "source": [
    "# calculate number of regions per chr\n",
    "total_reg_num = 1100\n",
    "chr_2_region_num = {_name: np.round(_len/total_len*total_reg_num).astype(np.int32) for _name, _len in chr_len_dict.items()}\n",
    "\n",
    "genomic_resolution = np.round(total_len / total_reg_num, -5).astype(np.int32)\n",
    "print(np.sum(list(chr_2_region_num.values())))\n",
    "print(genomic_resolution)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 187,
   "id": "5687bb93",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'1': 79,\n",
       " '2': 73,\n",
       " '3': 65,\n",
       " '4': 63,\n",
       " '5': 61,\n",
       " '6': 60,\n",
       " '7': 59,\n",
       " '8': 52,\n",
       " '9': 50,\n",
       " '10': 53,\n",
       " '11': 49,\n",
       " '12': 48,\n",
       " '13': 49,\n",
       " '14': 50,\n",
       " '15': 42,\n",
       " '16': 40,\n",
       " '17': 38,\n",
       " '18': 37,\n",
       " '19': 25,\n",
       " 'X': 69,\n",
       " 'Y': 37}"
      ]
     },
     "execution_count": 187,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr_2_region_num"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "c4a27ae4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# for each chromosome, calculate region center\n",
    "region_size=20000\n",
    "\n",
    "chr_2_region_seq = {}\n",
    "\n",
    "region_dicts = []\n",
    "\n",
    "\n",
    "for _chr, _num in chr_2_region_num.items():\n",
    "    _reg_center = np.array(np.arange(_num)*genomic_resolution + int(genomic_resolution/2), dtype=np.int32)\n",
    "    _reg_starts = _reg_center - int(region_size/2)\n",
    "    _reg_ends = _reg_center + int(region_size/2)\n",
    "    _seqs = [seq_rd.ref_seq_dict[_chr][_s:_e] for _s,_e in zip(_reg_starts, _reg_ends)]\n",
    "    chr_2_region_seq[_chr] = _seqs\n",
    "    \n",
    "    for _s,_e in zip(_reg_starts, _reg_ends):\n",
    "        _reg_dict = {\n",
    "            'Chr':_chr,\n",
    "            'Start':_s,\n",
    "            'End':_e,\n",
    "            'Region':f\"{_chr}:{_s}-{_e}\",\n",
    "            'Strand':'+',\n",
    "        }\n",
    "        region_dicts.append(_reg_dict)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "5821f641",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- searching among 21 references\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 1.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 2.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 3.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 4.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 5.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 6.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 7.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 8.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 9.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 10.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 11.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 12.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 13.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 14.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n",
      "-- a match found in record: 15.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- a match found in record: 15.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 16.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 17.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 18.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: 19.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: X.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n",
      "-- a match found in record: Y.\n"
     ]
    }
   ],
   "source": [
    "seqs = seq_rd.find_sequence_for_region(region_dicts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "466a50b4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- saving sequences into folder: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_2.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_3.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_4.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_5.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_6.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_7.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_8.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_9.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_10.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_11.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_12.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_13.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_14.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_15.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_16.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_17.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_18.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_19.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_20.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_21.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_22.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_23.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_24.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_25.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_26.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_27.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_28.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_29.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_30.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_31.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_32.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_33.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_34.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_35.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_36.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_37.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_38.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_39.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_40.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_41.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_42.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_43.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_44.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_45.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_46.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_47.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_48.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_49.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_50.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_51.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_52.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_53.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_54.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_55.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_56.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_57.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_58.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_59.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_60.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_61.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_62.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_63.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_64.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_65.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_66.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_67.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_68.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_69.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_70.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_71.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_72.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_73.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_74.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_75.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_76.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_77.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_78.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_79.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_80.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_81.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_82.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_83.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_84.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_85.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_86.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_87.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_88.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_89.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_90.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_91.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_92.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_93.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_94.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_95.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_96.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_97.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_98.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_99.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_100.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_101.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_102.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_103.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_104.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_105.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_106.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_107.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_108.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_109.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_110.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_111.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_112.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_113.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_114.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_115.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_116.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_117.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_118.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_119.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_120.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_121.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_122.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_123.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_124.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_125.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_126.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_127.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_128.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_129.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_130.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_131.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_132.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_133.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_134.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_135.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_136.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_137.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_138.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_139.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_140.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_141.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_142.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_143.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_144.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_145.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_146.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_147.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_148.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_149.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_150.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_151.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_152.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_153.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_154.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_155.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_156.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_157.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_158.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_159.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_160.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_161.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_162.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_163.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_164.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_165.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_166.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_167.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_168.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_169.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_170.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_171.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_172.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_173.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_174.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_175.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_176.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_177.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_178.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_179.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_180.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_181.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_182.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_183.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_184.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_185.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_186.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_187.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_188.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_189.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_190.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_191.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_192.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_193.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_194.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_195.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_196.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_197.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_198.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_199.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_200.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_201.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_202.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_203.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_204.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_205.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_206.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_207.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_208.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_209.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_210.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_211.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_212.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_213.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_214.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_215.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_216.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_217.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_218.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_219.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_220.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_221.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_222.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_223.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_224.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_225.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_226.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_227.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_228.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_229.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_230.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_231.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_232.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_233.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_234.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_235.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_236.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_237.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_238.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_239.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_240.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_241.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_242.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_243.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_244.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_245.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_246.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_247.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_248.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_249.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_250.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_251.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_252.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_253.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_254.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_255.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_256.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_257.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_258.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_259.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_260.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_261.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_262.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_263.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_264.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_265.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_266.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_267.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_268.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_269.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_270.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_271.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_272.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_273.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_274.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_275.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_276.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_277.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_278.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_279.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_280.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_281.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_282.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_283.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_284.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_285.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_286.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_287.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_288.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_289.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_290.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_291.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_292.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_293.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_294.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_295.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_296.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_297.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_298.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_299.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_300.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_301.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_302.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_303.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_304.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_305.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_306.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_307.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_308.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_309.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_310.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_311.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_312.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_313.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_314.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_315.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_316.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_317.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_318.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_319.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_320.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_321.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_322.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_323.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_324.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_325.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_326.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_327.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_328.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_329.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_330.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_331.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_332.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_333.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_334.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_335.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_336.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_337.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_338.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_339.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_340.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_341.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_342.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_343.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_344.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_345.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_346.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_347.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_348.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_349.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_350.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_351.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_352.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_353.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_354.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_355.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_356.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_357.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_358.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_359.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_360.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_361.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_362.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_363.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_364.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_365.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_366.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_367.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_368.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_369.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_370.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_371.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_372.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_373.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_374.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_375.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_376.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_377.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_378.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_379.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_380.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_381.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_382.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_383.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_384.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_385.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_386.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_387.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_388.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_389.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_390.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_391.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_392.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_393.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_394.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_395.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_396.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_397.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_398.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_399.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_400.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_401.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_402.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_403.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_404.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_405.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_406.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_407.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_408.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_409.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_410.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_411.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_412.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_413.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_414.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_415.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_416.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_417.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_418.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_419.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_420.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_421.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_422.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_423.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_424.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_425.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_426.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_427.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_428.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_429.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_430.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_431.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_432.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_433.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_434.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_435.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_436.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_437.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_438.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_439.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_440.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_441.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_442.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_443.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_444.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_445.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_446.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_447.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_448.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_449.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_450.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_451.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_452.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_453.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_454.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_455.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_456.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_457.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_458.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_459.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_460.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_461.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_462.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_463.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_464.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_465.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_466.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_467.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_468.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_469.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_470.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_471.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_472.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_473.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_474.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_475.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_476.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_477.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_478.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_479.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_480.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_481.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_482.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_483.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_484.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_485.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_486.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_487.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_488.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_489.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_490.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_491.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_492.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_493.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_494.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_495.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_496.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_497.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_498.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_499.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_500.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_501.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_502.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_503.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_504.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_505.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_506.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_507.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_508.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_509.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_510.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_511.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_512.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_513.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_514.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_515.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_516.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_517.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_518.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_519.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_520.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_521.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_522.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_523.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_524.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_525.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_526.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_527.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_528.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_529.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_530.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_531.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_532.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_533.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_534.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_535.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_536.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_537.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_538.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_539.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_540.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_541.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_542.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_543.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_544.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_545.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_546.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_547.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_548.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_549.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_550.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_551.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_552.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_553.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_554.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_555.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_556.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_557.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_558.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_559.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_560.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_561.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_562.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_563.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_564.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_565.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_566.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_567.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_568.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_569.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_570.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_571.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_572.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_573.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_574.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_575.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_576.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_577.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_578.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_579.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_580.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_581.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_582.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_583.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_584.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_585.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_586.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_587.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_588.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_589.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_590.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_591.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_592.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_593.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_594.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_595.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_596.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_597.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_598.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_599.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_600.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_601.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_602.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_603.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_604.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_605.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_606.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_607.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_608.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_609.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_610.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_611.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_612.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_613.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_614.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_615.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_616.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_617.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_618.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_619.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_620.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_621.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_622.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_623.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_624.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_625.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_626.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_627.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_628.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_629.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_630.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_631.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_632.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_633.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_634.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_635.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_636.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_637.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_638.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_639.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_640.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_641.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_642.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_643.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_644.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_645.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_646.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_647.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_648.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_649.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_650.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_651.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_652.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_653.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_654.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_655.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_656.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_657.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_658.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_659.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_660.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_661.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_662.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_663.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_664.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_665.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_666.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_667.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_668.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_669.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_670.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_671.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_672.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_673.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_674.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_675.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_676.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_677.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_678.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_679.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_680.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_681.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_682.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_683.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_684.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_685.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_686.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_687.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_688.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_689.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_690.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_691.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_692.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_693.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_694.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_695.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_696.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_697.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_698.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_699.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_700.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_701.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_702.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_703.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_704.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_705.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_706.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_707.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_708.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_709.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_710.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_711.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_712.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_713.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_714.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_715.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_716.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_717.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_718.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_719.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_720.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_721.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_722.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_723.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_724.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_725.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_726.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_727.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_728.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_729.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_730.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_731.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_732.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_733.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_734.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_735.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_736.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_737.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_738.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_739.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_740.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_741.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_742.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_743.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_744.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_745.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_746.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_747.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_748.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_749.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_750.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_751.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_752.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_753.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_754.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_755.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_756.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_757.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_758.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_759.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_760.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_761.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_762.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_763.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_764.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_765.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_766.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_767.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_768.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_769.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_770.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_771.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_772.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_773.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_774.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_775.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_776.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_777.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_778.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_779.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_780.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_781.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_782.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_783.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_784.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_785.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_786.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_787.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_788.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_789.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_790.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_791.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_792.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_793.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_794.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_795.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_796.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_797.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_798.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_799.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_800.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_801.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_802.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_803.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_804.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_805.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_806.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_807.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_808.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_809.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_810.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_811.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_812.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_813.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_814.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_815.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_816.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_817.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_818.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_819.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_820.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_821.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_822.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_823.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_824.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_825.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_826.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_827.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_828.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_829.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_830.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_831.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_832.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_833.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_834.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_835.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_836.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_837.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_838.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_839.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_840.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_841.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_842.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_843.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_844.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_845.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_846.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_847.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_848.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_849.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_850.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_851.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_852.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_853.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_854.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_855.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_856.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_857.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_858.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_859.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_860.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_861.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_862.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_863.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_864.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_865.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_866.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_867.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_868.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_869.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_870.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_871.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_872.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_873.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_874.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_875.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_876.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_877.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_878.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_879.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_880.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_881.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_882.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_883.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_884.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_885.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_886.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_887.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_888.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_889.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_890.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_891.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_892.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_893.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_894.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_895.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_896.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_897.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_898.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_899.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_900.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_901.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_902.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_903.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_904.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_905.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_906.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_907.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_908.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_909.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_910.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_911.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_912.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_913.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_914.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_915.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_916.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_917.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_918.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_919.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_920.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_921.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_922.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_923.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_924.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_925.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_926.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_927.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_928.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_929.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_930.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_931.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_932.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_933.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_934.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_935.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_936.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_937.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_938.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_939.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_940.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_941.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_942.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_943.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_944.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_945.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_946.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_947.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_948.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_949.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_950.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_951.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_952.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_953.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_954.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_955.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_956.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_957.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_958.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_959.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_960.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_961.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_962.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_963.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_964.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_965.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_966.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_967.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_968.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_969.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_970.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_971.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_972.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_973.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_974.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_975.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_976.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_977.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_978.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_979.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_980.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_981.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_982.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_983.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_984.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_985.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_986.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_987.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_988.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_989.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_990.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_991.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_992.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_993.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_994.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_995.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_996.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_997.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_998.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_999.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1000.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1001.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1002.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1003.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1004.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1005.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1006.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1007.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1008.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1009.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1010.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1011.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1012.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1013.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1014.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1015.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1016.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1017.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1018.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1019.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1020.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1021.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1022.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1023.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1024.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1025.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1026.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1027.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1028.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1029.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1030.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1031.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1032.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1033.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1034.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1035.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1036.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1037.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1038.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1039.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1040.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1041.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1042.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1043.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1044.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1045.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1046.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1047.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1048.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1049.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1050.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1051.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1052.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1053.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1054.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1055.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1056.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1057.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1058.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1059.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1060.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1061.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1062.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1063.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1064.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1065.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1066.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1067.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1068.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1069.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1070.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1071.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1072.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1073.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1074.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1075.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1076.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1077.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1078.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1079.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1080.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1081.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1082.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1083.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1084.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1085.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1086.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1087.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1088.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1089.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1090.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1091.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1092.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1093.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1094.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1095.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1096.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1097.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1098.fasta\n",
      "-- save to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1099.fasta\n"
     ]
    }
   ],
   "source": [
    "seq_rd.save_sequences(save_folder=sequence_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9252f4d3",
   "metadata": {},
   "source": [
    "<a id='2'></a>\n",
    "# 2. Design probe targeting sequences by probe_designer\n",
    "\n",
    "run probe_designer, remeber to clear the memory usage because each table should take ~32GB"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8e85674f",
   "metadata": {},
   "source": [
    "<a id='2.1'></a>\n",
    "## 2.1 Construct count table with all the 17-mers in the genome\n",
    "\n",
    "Only do this if you don't have pre-built 17-mer\n",
    "\n",
    "However you can do almost the same thing for your own library during quality check\n",
    "\n",
    "This library requires mm10 genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "a634890f",
   "metadata": {},
   "outputs": [],
   "source": [
    "overwrite_table = False"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2e5c8ab3",
   "metadata": {},
   "source": [
    "### construct map for whole genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "b5f98b2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "from ImageAnalysis3 import library_tools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "80be081a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reference table: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Genomes\\mouse\\GRCm38_ensembl\\GRCm38_genome_17w.npy already exist, skip.\n"
     ]
    }
   ],
   "source": [
    "reload(library_tools.design)\n",
    " \n",
    "genome_table_file = os.path.join(reference_folder, 'GRCm38_genome_17w.npy')\n",
    "\n",
    "if not os.path.exists(genome_table_file) or overwrite_table:\n",
    "    # genome\n",
    "    _genome_filenames = [os.path.join(genome_folder, _fl) \n",
    "         for _fl in os.listdir(genome_folder) \n",
    "         if _fl.split(os.extsep)[-1]=='fasta' or _fl.split(os.extsep)[-1]=='fa']\n",
    "    print(len(_genome_filenames))\n",
    "\n",
    "    ct = library_tools.design.countTable(word=17,save_file=genome_table_file, \n",
    "                       sparse=False)\n",
    "    ct.verbose=True\n",
    "\n",
    "    ct.read(_genome_filenames) # read sequences from fasta files\n",
    "\n",
    "    ct.consume_loaded(num_threads=24) # convert sequences into integers\n",
    "\n",
    "    ct.complete(verbose=True)\n",
    "\n",
    "    ct.save()\n",
    "\n",
    "    # clear RAM if contructed countable \n",
    "    del(ct)\n",
    "else:\n",
    "    print(f\"reference table: {genome_table_file} already exist, skip.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1acc14b4",
   "metadata": {},
   "source": [
    "### construct map for transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "1e244aad",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reference table: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Genomes\\mouse\\GRCm38_ensembl\\GRCm38_transcriptome_17w.npy already exist, skip.\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm\n",
    "# transcriptome\n",
    "transcriptome_folder = os.path.join(reference_folder, 'Transcriptome')\n",
    "\n",
    "transcriptome_table_file = os.path.join(reference_folder, 'GRCm38_transcriptome_17w.npy')\n",
    "\n",
    "if not os.path.exists(transcriptome_table_file) or overwrite_table:\n",
    "    # transcriptome\n",
    "    _transcriptome_filenames = [os.path.join(transcriptome_folder, _fl) \n",
    "         for _fl in os.listdir(transcriptome_folder) \n",
    "         if _fl.split(os.extsep)[-1]=='fasta' or _fl.split(os.extsep)[-1]=='fa']\n",
    "    print(len(_transcriptome_filenames))\n",
    "    \n",
    "    ct = library_tools.design.countTable(word=17,save_file=transcriptome_table_file, \n",
    "                       sparse=False)\n",
    "    ct.verbose=True\n",
    "\n",
    "    ct.read(_transcriptome_filenames) # read sequences from fasta files\n",
    "\n",
    "    ct.consume_loaded(num_threads=24) # convert sequences into integers\n",
    "\n",
    "    ct.complete(verbose=True)\n",
    "\n",
    "    ct.save()\n",
    "    \n",
    "    # clear RAM if contructed countable \n",
    "    del(ct)\n",
    "else:\n",
    "    print(f\"reference table: {transcriptome_table_file} already exist, skip.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b477fab",
   "metadata": {},
   "source": [
    "### construct map for repeats from RepBase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "556095da",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "reference table: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Genomes\\mouse\\GRCm38_ensembl\\Repbase_v2603_repeat_17w.npy already exist, skip.\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm\n",
    "# repeat\n",
    "repeat_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Libraries\\Genomes\\Repbase'\n",
    "\n",
    "repeat_table_file = os.path.join(reference_folder, 'Repbase_v2603_repeat_17w.npy')\n",
    "\n",
    "if not os.path.exists(repeat_table_file) or overwrite_table:\n",
    "    # repeat\n",
    "    _repeat_filenames = [os.path.join(repeat_folder, _fl) \n",
    "         for _fl in os.listdir(repeat_folder) \n",
    "         if _fl.split(os.extsep)[-1]=='fasta' or _fl.split(os.extsep)[-1]=='fa']\n",
    "    print(len(_repeat_filenames))\n",
    "    \n",
    "    ct = library_tools.design.countTable(word=17,save_file=repeat_table_file, \n",
    "                       sparse=False)\n",
    "    ct.verbose=True\n",
    "\n",
    "    ct.read(_repeat_filenames) # read sequences from fasta files\n",
    "\n",
    "    ct.consume_loaded(num_threads=24) # convert sequences into integers\n",
    "\n",
    "    ct.complete(verbose=True)\n",
    "\n",
    "    ct.save()\n",
    "    \n",
    "    # clear RAM if contructed countable \n",
    "    del(ct)\n",
    "else:\n",
    "    print(f\"reference table: {repeat_table_file} already exist, skip.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d38ed029",
   "metadata": {},
   "source": [
    "<a id='2.2'></a>\n",
    "## 2.2 Design probes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "8e476dee",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1099 regions loaded to design probes.\n",
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr\n"
     ]
    }
   ],
   "source": [
    "# requires pre_defined genome_folder and library_folder\n",
    "# Indices\n",
    "genome_index = os.path.join(reference_folder, 'GRCm38_genome_17w.npy')\n",
    "transcriptome_index = os.path.join(reference_folder, 'GRCm38_transcriptome_17w.npy') \n",
    "repeat_index = os.path.join(reference_folder, 'Repbase_v2603_repeat_17w.npy')\n",
    "#ref_merfish_index = os.path.join(reference_folder, 'M1_meng_MERFISH_17w.npy') # merfish designed by Meng\n",
    "# get input files \n",
    "input_files = glob.glob(os.path.join(sequence_folder, '*.fasta'))\n",
    "\n",
    "print(f\"{len(input_files)} regions loaded to design probes.\")\n",
    "\n",
    "if not os.path.exists(report_folder):\n",
    "    os.makedirs(report_folder)\n",
    "    \n",
    "# filename to save probe reports\n",
    "save_file = os.path.join(report_folder, f'merged_probes.pbr')\n",
    "print(save_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "722e6b4b",
   "metadata": {},
   "source": [
    "### create pb_designer class"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46911be3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "02721f6d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Probe designer derived from Bogdan Bintu:\n",
      "https://github.com/BogdanBintu/ChromatinImaging/blob/master/LibraryDesign/LibraryDesigner.py\n",
      "by Pu Zheng, 2020.11\n",
      "\n",
      "Major changes:\n",
      "    1. allow design of two strands\n",
      "    2. separate reverse_complement (rev_com) and from two strands (two_stranded) as \n",
      "    two different inputs for map_dic and sequence_dic\n",
      "    3. replace 'local_genome' with 'self_sequences' to be more explicit, and only \n",
      "    exclude the counts for the corresponding self_sequence within each input. \n",
      "\n",
      "Key information:\n",
      "    - number of input_sequence(s): 1099\n",
      "    - save_file location: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr\n",
      "\n",
      "- Loading from savefile: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr.\n",
      "211283\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 146,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reload(library_tools)\n",
    "reload(library_tools.design)\n",
    "\n",
    "pb_designer = library_tools.design.pb_reports_class(\n",
    "    sequence_dic={'file':input_files,\n",
    "                  'rev_com':True, # design two strands\n",
    "                  'two_stranded':True},\n",
    "    map_dic={'genome':{'file':genome_index,'rev_com':False,'two_stranded':True},\n",
    "             'transcriptome':{'file':transcriptome_index,'rev_com':True,'two_stranded':False},\n",
    "             'rep_genome':{'file':repeat_index,'rev_com':False,'two_stranded':True},\n",
    "             'self_sequences':{'file':input_files,'force_list':True,'rev_com':False,'two_stranded':True},\n",
    "             #'ref_merfish':{'file':ref_merfish_index,'rev_com':False,'two_stranded':True},\n",
    "             },\n",
    "        save_file=save_file,\n",
    "    params_dic={'word_size':17,'pb_len':42,'buffer_len':-5,'max_count':2**16-1,\n",
    "                'check_on_go': False, # whether automatically check probes\n",
    "                'auto': False, # whether automatically convert reference maps\n",
    "               },\n",
    "    check_dic={('genome','self_sequences'): 25,\n",
    "               'rep_genome': 0,\n",
    "               'transcriptome': 14,\n",
    "               #'ref_merfish': 14,\n",
    "               'gc':[0.25,0.85],'tm': 37+0.62*50+5,\n",
    "               }\n",
    "    )\n",
    "print(pb_designer)\n",
    "pb_designer.load_from_file(load_probes_only=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b84dae63",
   "metadata": {},
   "source": [
    "### calculate probe reports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "e0c58bbc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- setting attribute: map_genome\n",
      "--- finish map_genome in 90.018s.\n",
      "-- setting attribute: map_transcriptome\n",
      "--- finish map_transcriptome in 89.372s.\n",
      "-- setting attribute: map_rep_genome\n",
      "--- finish map_rep_genome in 97.265s.\n",
      "Time(s): 276.6547210216522\n",
      "- Designing targeting sequence for 1099 regions\n",
      "-- designing region: 1:1240000-1260000_:+_reg_ region: 0, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.206s.\n",
      "in 0.426s.\n",
      "-- designing region: 1:3740000-3760000_:+_reg_ region: 1, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_2.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.593s.\n",
      "in 11.331s.\n",
      "-- designing region: 1:6240000-6260000_:+_reg_ region: 2, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_3.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.255s.\n",
      "in 17.602s.\n",
      "-- designing region: 1:8740000-8760000_:+_reg_ region: 3, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_4.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.663s.\n",
      "in 18.665s.\n",
      "-- designing region: 1:11240000-11260000_:+_reg_ region: 4, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_5.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.345s.\n",
      "in 15.549s.\n",
      "-- designing region: 1:13740000-13760000_:+_reg_ region: 5, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_6.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.256s.\n",
      "in 11.540s.\n",
      "-- designing region: 1:16240000-16260000_:+_reg_ region: 6, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_7.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.182s.\n",
      "in 11.151s.\n",
      "-- designing region: 1:18740000-18760000_:+_reg_ region: 7, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_8.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.514s.\n",
      "-- designing region: 1:21240000-21260000_:+_reg_ region: 8, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_9.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.272s.\n",
      "-- designing region: 1:23740000-23760000_:+_reg_ region: 9, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_10.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.229s.\n",
      "-- designing region: 1:26240000-26260000_:+_reg_ region: 10, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_11.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.171s.\n",
      "in 11.230s.\n",
      "-- designing region: 1:28740000-28760000_:+_reg_ region: 11, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_12.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.634s.\n",
      "-- designing region: 1:31240000-31260000_:+_reg_ region: 12, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_13.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.136s.\n",
      "-- designing region: 1:33740000-33760000_:+_reg_ region: 13, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_14.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 28.731s.\n",
      "-- designing region: 1:36240000-36260000_:+_reg_ region: 14, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_15.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 1.899s.\n",
      "in 18.577s.\n",
      "-- designing region: 1:38740000-38760000_:+_reg_ region: 15, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_16.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 14.931s.\n",
      "-- designing region: 1:41240000-41260000_:+_reg_ region: 16, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_17.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.429s.\n",
      "in 17.894s.\n",
      "-- designing region: 1:43740000-43760000_:+_reg_ region: 17, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_18.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.406s.\n",
      "-- designing region: 1:46240000-46260000_:+_reg_ region: 18, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_19.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.365s.\n",
      "-- designing region: 1:48740000-48760000_:+_reg_ region: 19, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_20.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.174s.\n",
      "in 11.170s.\n",
      "-- designing region: 1:51240000-51260000_:+_reg_ region: 20, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_21.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.398s.\n",
      "-- designing region: 1:53740000-53760000_:+_reg_ region: 21, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_22.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.709s.\n",
      "-- designing region: 1:56240000-56260000_:+_reg_ region: 22, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_23.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.217s.\n",
      "-- designing region: 1:58740000-58760000_:+_reg_ region: 23, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_24.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.388s.\n",
      "-- designing region: 1:61240000-61260000_:+_reg_ region: 24, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_25.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.211s.\n",
      "in 11.230s.\n",
      "-- designing region: 1:63740000-63760000_:+_reg_ region: 25, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_26.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.161s.\n",
      "in 11.334s.\n",
      "-- designing region: 1:66240000-66260000_:+_reg_ region: 26, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_27.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.367s.\n",
      "-- designing region: 1:68740000-68760000_:+_reg_ region: 27, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_28.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.227s.\n",
      "-- designing region: 1:71240000-71260000_:+_reg_ region: 28, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_29.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.204s.\n",
      "in 12.019s.\n",
      "-- designing region: 1:73740000-73760000_:+_reg_ region: 29, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_30.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 11.451s.\n",
      "-- designing region: 1:76240000-76260000_:+_reg_ region: 30, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_31.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.428s.\n",
      "-- designing region: 1:78740000-78760000_:+_reg_ region: 31, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_32.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.260s.\n",
      "-- designing region: 1:81240000-81260000_:+_reg_ region: 32, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_33.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.509s.\n",
      "-- designing region: 1:83740000-83760000_:+_reg_ region: 33, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_34.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.343s.\n",
      "-- designing region: 1:86240000-86260000_:+_reg_ region: 34, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_35.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.194s.\n",
      "in 11.475s.\n",
      "-- designing region: 1:88740000-88760000_:+_reg_ region: 35, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_36.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.440s.\n",
      "-- designing region: 1:91240000-91260000_:+_reg_ region: 36, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_37.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 12.139s.\n",
      "-- designing region: 1:93740000-93760000_:+_reg_ region: 37, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_38.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.398s.\n",
      "-- designing region: 1:96240000-96260000_:+_reg_ region: 38, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_39.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.372s.\n",
      "-- designing region: 1:98740000-98760000_:+_reg_ region: 39, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_40.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.223s.\n",
      "in 11.261s.\n",
      "-- designing region: 1:101240000-101260000_:+_reg_ region: 40, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_41.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.300s.\n",
      "-- designing region: 1:103740000-103760000_:+_reg_ region: 41, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_42.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.436s.\n",
      "-- designing region: 1:106240000-106260000_:+_reg_ region: 42, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_43.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.315s.\n",
      "-- designing region: 1:108740000-108760000_:+_reg_ region: 43, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_44.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.287s.\n",
      "-- designing region: 1:111240000-111260000_:+_reg_ region: 44, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_45.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.399s.\n",
      "-- designing region: 1:113740000-113760000_:+_reg_ region: 45, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_46.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.371s.\n",
      "-- designing region: 1:116240000-116260000_:+_reg_ region: 46, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_47.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.328s.\n",
      "-- designing region: 1:118740000-118760000_:+_reg_ region: 47, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_48.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.225s.\n",
      "in 12.459s.\n",
      "-- designing region: 1:121240000-121260000_:+_reg_ region: 48, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_49.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.360s.\n",
      "-- designing region: 1:123740000-123760000_:+_reg_ region: 49, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_50.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.169s.\n",
      "in 11.381s.\n",
      "-- designing region: 1:126240000-126260000_:+_reg_ region: 50, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_51.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.300s.\n",
      "-- designing region: 1:128740000-128760000_:+_reg_ region: 51, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_52.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.206s.\n",
      "in 11.429s.\n",
      "-- designing region: 1:131240000-131260000_:+_reg_ region: 52, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_53.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.443s.\n",
      "-- designing region: 1:133740000-133760000_:+_reg_ region: 53, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_54.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.192s.\n",
      "in 11.288s.\n",
      "-- designing region: 1:136240000-136260000_:+_reg_ region: 54, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_55.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.415s.\n",
      "-- designing region: 1:138740000-138760000_:+_reg_ region: 55, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_56.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.340s.\n",
      "-- designing region: 1:141240000-141260000_:+_reg_ region: 56, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_57.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.217s.\n",
      "-- designing region: 1:143740000-143760000_:+_reg_ region: 57, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_58.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.456s.\n",
      "-- designing region: 1:146240000-146260000_:+_reg_ region: 58, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_59.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.318s.\n",
      "-- designing region: 1:148740000-148760000_:+_reg_ region: 59, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_60.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.246s.\n",
      "-- designing region: 1:151240000-151260000_:+_reg_ region: 60, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_61.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 12.736s.\n",
      "-- designing region: 1:153740000-153760000_:+_reg_ region: 61, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_62.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.220s.\n",
      "-- designing region: 1:156240000-156260000_:+_reg_ region: 62, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_63.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.485s.\n",
      "-- designing region: 1:158740000-158760000_:+_reg_ region: 63, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_64.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.325s.\n",
      "-- designing region: 1:161240000-161260000_:+_reg_ region: 64, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_65.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.244s.\n",
      "-- designing region: 1:163740000-163760000_:+_reg_ region: 65, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_66.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.465s.\n",
      "-- designing region: 1:166240000-166260000_:+_reg_ region: 66, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_67.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.322s.\n",
      "-- designing region: 1:168740000-168760000_:+_reg_ region: 67, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_68.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.225s.\n",
      "in 11.459s.\n",
      "-- designing region: 1:171240000-171260000_:+_reg_ region: 68, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_69.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.355s.\n",
      "-- designing region: 1:173740000-173760000_:+_reg_ region: 69, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_70.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.312s.\n",
      "-- designing region: 1:176240000-176260000_:+_reg_ region: 70, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_71.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.153s.\n",
      "in 11.389s.\n",
      "-- designing region: 1:178740000-178760000_:+_reg_ region: 71, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_72.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.413s.\n",
      "-- designing region: 1:181240000-181260000_:+_reg_ region: 72, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_73.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.174s.\n",
      "in 11.512s.\n",
      "-- designing region: 1:183740000-183760000_:+_reg_ region: 73, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_74.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.416s.\n",
      "-- designing region: 1:186240000-186260000_:+_reg_ region: 74, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_75.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.205s.\n",
      "in 11.331s.\n",
      "-- designing region: 1:188740000-188760000_:+_reg_ region: 75, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_76.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.439s.\n",
      "-- designing region: 1:191240000-191260000_:+_reg_ region: 76, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_77.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.336s.\n",
      "-- designing region: 1:193740000-193760000_:+_reg_ region: 77, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_78.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.148s.\n",
      "in 12.829s.\n",
      "-- designing region: 1:196240000-195471971_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: 2:1240000-1260000_:+_reg_ region: 79, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_80.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 0.383s.\n",
      "-- designing region: 2:3740000-3760000_:+_reg_ region: 80, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_81.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.560s.\n",
      "in 11.757s.\n",
      "-- designing region: 2:6240000-6260000_:+_reg_ region: 81, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_82.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.174s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.276s.\n",
      "-- designing region: 2:8740000-8760000_:+_reg_ region: 82, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_83.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.364s.\n",
      "-- designing region: 2:11240000-11260000_:+_reg_ region: 83, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_84.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.335s.\n",
      "-- designing region: 2:13740000-13760000_:+_reg_ region: 84, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_85.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.198s.\n",
      "in 11.314s.\n",
      "-- designing region: 2:16240000-16260000_:+_reg_ region: 85, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_86.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 11.358s.\n",
      "-- designing region: 2:18740000-18760000_:+_reg_ region: 86, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_87.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.324s.\n",
      "-- designing region: 2:21240000-21260000_:+_reg_ region: 87, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_88.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.181s.\n",
      "in 11.315s.\n",
      "-- designing region: 2:23740000-23760000_:+_reg_ region: 88, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_89.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.164s.\n",
      "in 11.359s.\n",
      "-- designing region: 2:26240000-26260000_:+_reg_ region: 89, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_90.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.355s.\n",
      "-- designing region: 2:28740000-28760000_:+_reg_ region: 90, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_91.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.289s.\n",
      "-- designing region: 2:31240000-31260000_:+_reg_ region: 91, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_92.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.155s.\n",
      "in 11.375s.\n",
      "-- designing region: 2:33740000-33760000_:+_reg_ region: 92, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_93.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.336s.\n",
      "-- designing region: 2:36240000-36260000_:+_reg_ region: 93, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_94.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.306s.\n",
      "-- designing region: 2:38740000-38760000_:+_reg_ region: 94, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_95.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.143s.\n",
      "in 11.379s.\n",
      "-- designing region: 2:41240000-41260000_:+_reg_ region: 95, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_96.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 15.931s.\n",
      "-- designing region: 2:43740000-43760000_:+_reg_ region: 96, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_97.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.300s.\n",
      "-- designing region: 2:46240000-46260000_:+_reg_ region: 97, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_98.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.215s.\n",
      "in 11.293s.\n",
      "-- designing region: 2:48740000-48760000_:+_reg_ region: 98, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_99.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.344s.\n",
      "-- designing region: 2:51240000-51260000_:+_reg_ region: 99, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_100.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 13.325s.\n",
      "-- designing region: 2:53740000-53760000_:+_reg_ region: 100, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_101.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.213s.\n",
      "in 11.308s.\n",
      "-- designing region: 2:56240000-56260000_:+_reg_ region: 101, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_102.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.268s.\n",
      "-- designing region: 2:58740000-58760000_:+_reg_ region: 102, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_103.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.185s.\n",
      "in 11.247s.\n",
      "-- designing region: 2:61240000-61260000_:+_reg_ region: 103, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_104.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.231s.\n",
      "-- designing region: 2:63740000-63760000_:+_reg_ region: 104, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_105.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.380s.\n",
      "-- designing region: 2:66240000-66260000_:+_reg_ region: 105, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_106.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.249s.\n",
      "-- designing region: 2:68740000-68760000_:+_reg_ region: 106, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_107.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.236s.\n",
      "-- designing region: 2:71240000-71260000_:+_reg_ region: 107, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_108.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.400s.\n",
      "-- designing region: 2:73740000-73760000_:+_reg_ region: 108, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_109.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.284s.\n",
      "-- designing region: 2:76240000-76260000_:+_reg_ region: 109, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_110.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.158s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.240s.\n",
      "-- designing region: 2:78740000-78760000_:+_reg_ region: 110, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_111.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 11.327s.\n",
      "-- designing region: 2:81240000-81260000_:+_reg_ region: 111, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_112.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.308s.\n",
      "-- designing region: 2:83740000-83760000_:+_reg_ region: 112, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_113.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.306s.\n",
      "-- designing region: 2:86240000-86260000_:+_reg_ region: 113, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_114.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.203s.\n",
      "in 11.358s.\n",
      "-- designing region: 2:88740000-88760000_:+_reg_ region: 114, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_115.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.328s.\n",
      "-- designing region: 2:91240000-91260000_:+_reg_ region: 115, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_116.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.334s.\n",
      "-- designing region: 2:93740000-93760000_:+_reg_ region: 116, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_117.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.192s.\n",
      "in 11.377s.\n",
      "-- designing region: 2:96240000-96260000_:+_reg_ region: 117, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_118.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.337s.\n",
      "-- designing region: 2:98740000-98760000_:+_reg_ region: 118, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_119.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.212s.\n",
      "in 11.253s.\n",
      "-- designing region: 2:101240000-101260000_:+_reg_ region: 119, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_120.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.215s.\n",
      "in 11.375s.\n",
      "-- designing region: 2:103740000-103760000_:+_reg_ region: 120, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_121.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.346s.\n",
      "-- designing region: 2:106240000-106260000_:+_reg_ region: 121, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_122.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.210s.\n",
      "in 11.274s.\n",
      "-- designing region: 2:108740000-108760000_:+_reg_ region: 122, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_123.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.216s.\n",
      "in 11.382s.\n",
      "-- designing region: 2:111240000-111260000_:+_reg_ region: 123, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_124.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.332s.\n",
      "-- designing region: 2:113740000-113760000_:+_reg_ region: 124, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_125.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.266s.\n",
      "-- designing region: 2:116240000-116260000_:+_reg_ region: 125, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_126.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 13.844s.\n",
      "-- designing region: 2:118740000-118760000_:+_reg_ region: 126, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_127.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.237s.\n",
      "-- designing region: 2:121240000-121260000_:+_reg_ region: 127, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_128.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.189s.\n",
      "-- designing region: 2:123740000-123760000_:+_reg_ region: 128, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_129.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.383s.\n",
      "-- designing region: 2:126240000-126260000_:+_reg_ region: 129, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_130.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.200s.\n",
      "-- designing region: 2:128740000-128760000_:+_reg_ region: 130, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_131.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 11.193s.\n",
      "-- designing region: 2:131240000-131260000_:+_reg_ region: 131, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_132.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.193s.\n",
      "in 11.369s.\n",
      "-- designing region: 2:133740000-133760000_:+_reg_ region: 132, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_133.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.333s.\n",
      "-- designing region: 2:136240000-136260000_:+_reg_ region: 133, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_134.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.222s.\n",
      "in 11.264s.\n",
      "-- designing region: 2:138740000-138760000_:+_reg_ region: 134, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_135.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.209s.\n",
      "in 11.352s.\n",
      "-- designing region: 2:141240000-141260000_:+_reg_ region: 135, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_136.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.347s.\n",
      "-- designing region: 2:143740000-143760000_:+_reg_ region: 136, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_137.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.310s.\n",
      "-- designing region: 2:146240000-146260000_:+_reg_ region: 137, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_138.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.200s.\n",
      "in 11.376s.\n",
      "-- designing region: 2:148740000-148760000_:+_reg_ region: 138, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_139.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.367s.\n",
      "-- designing region: 2:151240000-151260000_:+_reg_ region: 139, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_140.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.200s.\n",
      "in 11.289s.\n",
      "-- designing region: 2:153740000-153760000_:+_reg_ region: 140, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_141.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.187s.\n",
      "in 11.353s.\n",
      "-- designing region: 2:156240000-156260000_:+_reg_ region: 141, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_142.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.223s.\n",
      "in 11.272s.\n",
      "-- designing region: 2:158740000-158760000_:+_reg_ region: 142, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_143.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.355s.\n",
      "-- designing region: 2:161240000-161260000_:+_reg_ region: 143, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_144.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.194s.\n",
      "in 11.328s.\n",
      "-- designing region: 2:163740000-163760000_:+_reg_ region: 144, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_145.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.363s.\n",
      "-- designing region: 2:166240000-166260000_:+_reg_ region: 145, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_146.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.329s.\n",
      "-- designing region: 2:168740000-168760000_:+_reg_ region: 146, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_147.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.185s.\n",
      "in 11.355s.\n",
      "-- designing region: 2:171240000-171260000_:+_reg_ region: 147, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_148.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.862s.\n",
      "-- designing region: 2:173740000-173760000_:+_reg_ region: 148, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_149.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.223s.\n",
      "in 11.284s.\n",
      "-- designing region: 2:176240000-176260000_:+_reg_ region: 149, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_150.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.291s.\n",
      "-- designing region: 2:178740000-178760000_:+_reg_ region: 150, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_151.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.254s.\n",
      "-- designing region: 2:181240000-181260000_:+_reg_ region: 151, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_152.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.158s.\n",
      "in 11.382s.\n",
      "-- designing region: 3:1240000-1260000_:+_reg_ region: 152, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_153.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.192s.\n",
      "in 0.411s.\n",
      "-- designing region: 3:3740000-3760000_:+_reg_ region: 153, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_154.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.463s.\n",
      "in 11.675s.\n",
      "-- designing region: 3:6240000-6260000_:+_reg_ region: 154, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_155.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.293s.\n",
      "-- designing region: 3:8740000-8760000_:+_reg_ region: 155, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_156.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.146s.\n",
      "in 11.239s.\n",
      "-- designing region: 3:11240000-11260000_:+_reg_ region: 156, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_157.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.152s.\n",
      "in 11.342s.\n",
      "-- designing region: 3:13740000-13760000_:+_reg_ region: 157, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_158.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 14.426s.\n",
      "-- designing region: 3:16240000-16260000_:+_reg_ region: 158, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_159.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.149s.\n",
      "in 11.219s.\n",
      "-- designing region: 3:18740000-18760000_:+_reg_ region: 159, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_160.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.365s.\n",
      "-- designing region: 3:21240000-21260000_:+_reg_ region: 160, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_161.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.331s.\n",
      "-- designing region: 3:23740000-23760000_:+_reg_ region: 161, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_162.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.267s.\n",
      "-- designing region: 3:26240000-26260000_:+_reg_ region: 162, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_163.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.185s.\n",
      "in 11.405s.\n",
      "-- designing region: 3:28740000-28760000_:+_reg_ region: 163, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_164.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.317s.\n",
      "-- designing region: 3:31240000-31260000_:+_reg_ region: 164, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_165.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.197s.\n",
      "in 11.309s.\n",
      "-- designing region: 3:33740000-33760000_:+_reg_ region: 165, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_166.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 11.202s.\n",
      "-- designing region: 3:36240000-36260000_:+_reg_ region: 166, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_167.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.392s.\n",
      "-- designing region: 3:38740000-38760000_:+_reg_ region: 167, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_168.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.365s.\n",
      "-- designing region: 3:41240000-41260000_:+_reg_ region: 168, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_169.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.189s.\n",
      "in 11.397s.\n",
      "-- designing region: 3:43740000-43760000_:+_reg_ region: 169, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_170.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.313s.\n",
      "-- designing region: 3:46240000-46260000_:+_reg_ region: 170, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_171.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.203s.\n",
      "in 11.281s.\n",
      "-- designing region: 3:48740000-48760000_:+_reg_ region: 171, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_172.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 11.395s.\n",
      "-- designing region: 3:51240000-51260000_:+_reg_ region: 172, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_173.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.352s.\n",
      "-- designing region: 3:53740000-53760000_:+_reg_ region: 173, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_174.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.189s.\n",
      "in 11.303s.\n",
      "-- designing region: 3:56240000-56260000_:+_reg_ region: 174, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_175.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.164s.\n",
      "in 11.398s.\n",
      "-- designing region: 3:58740000-58760000_:+_reg_ region: 175, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_176.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.277s.\n",
      "-- designing region: 3:61240000-61260000_:+_reg_ region: 176, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_177.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.208s.\n",
      "in 11.343s.\n",
      "-- designing region: 3:63740000-63760000_:+_reg_ region: 177, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_178.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.152s.\n",
      "in 15.289s.\n",
      "-- designing region: 3:66240000-66260000_:+_reg_ region: 178, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_179.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.803s.\n",
      "in 11.917s.\n",
      "-- designing region: 3:68740000-68760000_:+_reg_ region: 179, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_180.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.402s.\n",
      "-- designing region: 3:71240000-71260000_:+_reg_ region: 180, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_181.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.225s.\n",
      "in 11.289s.\n",
      "-- designing region: 3:73740000-73760000_:+_reg_ region: 181, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_182.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.202s.\n",
      "in 11.351s.\n",
      "-- designing region: 3:76240000-76260000_:+_reg_ region: 182, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_183.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.349s.\n",
      "-- designing region: 3:78740000-78760000_:+_reg_ region: 183, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_184.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.223s.\n",
      "in 11.223s.\n",
      "-- designing region: 3:81240000-81260000_:+_reg_ region: 184, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_185.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.424s.\n",
      "-- designing region: 3:83740000-83760000_:+_reg_ region: 185, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_186.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.361s.\n",
      "-- designing region: 3:86240000-86260000_:+_reg_ region: 186, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_187.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.220s.\n",
      "in 11.236s.\n",
      "-- designing region: 3:88740000-88760000_:+_reg_ region: 187, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_188.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.306s.\n",
      "-- designing region: 3:91240000-91260000_:+_reg_ region: 188, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_189.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.207s.\n",
      "in 11.352s.\n",
      "-- designing region: 3:93740000-93760000_:+_reg_ region: 189, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_190.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.405s.\n",
      "-- designing region: 3:96240000-96260000_:+_reg_ region: 190, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_191.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.279s.\n",
      "-- designing region: 3:98740000-98760000_:+_reg_ region: 191, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_192.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.167s.\n",
      "in 11.336s.\n",
      "-- designing region: 3:101240000-101260000_:+_reg_ region: 192, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_193.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.414s.\n",
      "-- designing region: 3:103740000-103760000_:+_reg_ region: 193, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_194.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.254s.\n",
      "-- designing region: 3:106240000-106260000_:+_reg_ region: 194, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_195.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.297s.\n",
      "-- designing region: 3:108740000-108760000_:+_reg_ region: 195, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_196.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.221s.\n",
      "-- designing region: 3:111240000-111260000_:+_reg_ region: 196, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_197.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.318s.\n",
      "-- designing region: 3:113740000-113760000_:+_reg_ region: 197, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_198.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 15.057s.\n",
      "-- designing region: 3:116240000-116260000_:+_reg_ region: 198, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_199.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.389s.\n",
      "-- designing region: 3:118740000-118760000_:+_reg_ region: 199, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_200.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.226s.\n",
      "-- designing region: 3:121240000-121260000_:+_reg_ region: 200, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_201.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.146s.\n",
      "in 11.324s.\n",
      "-- designing region: 3:123740000-123760000_:+_reg_ region: 201, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_202.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.227s.\n",
      "-- designing region: 3:126240000-126260000_:+_reg_ region: 202, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_203.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.321s.\n",
      "-- designing region: 3:128740000-128760000_:+_reg_ region: 203, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_204.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.210s.\n",
      "in 11.356s.\n",
      "-- designing region: 3:131240000-131260000_:+_reg_ region: 204, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_205.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.280s.\n",
      "-- designing region: 3:133740000-133760000_:+_reg_ region: 205, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_206.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.288s.\n",
      "-- designing region: 3:136240000-136260000_:+_reg_ region: 206, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_207.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.388s.\n",
      "-- designing region: 3:138740000-138760000_:+_reg_ region: 207, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_208.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.295s.\n",
      "-- designing region: 3:141240000-141260000_:+_reg_ region: 208, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_209.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.178s.\n",
      "in 11.321s.\n",
      "-- designing region: 3:143740000-143760000_:+_reg_ region: 209, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_210.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.398s.\n",
      "-- designing region: 3:146240000-146260000_:+_reg_ region: 210, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_211.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.263s.\n",
      "-- designing region: 3:148740000-148760000_:+_reg_ region: 211, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_212.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.187s.\n",
      "in 11.337s.\n",
      "-- designing region: 3:151240000-151260000_:+_reg_ region: 212, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_213.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.396s.\n",
      "-- designing region: 3:153740000-153760000_:+_reg_ region: 213, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_214.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.263s.\n",
      "in 11.317s.\n",
      "-- designing region: 3:156240000-156260000_:+_reg_ region: 214, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_215.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.148s.\n",
      "in 11.287s.\n",
      "-- designing region: 3:158740000-158760000_:+_reg_ region: 215, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_216.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.396s.\n",
      "-- designing region: 3:161240000-160039680_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: 4:1240000-1260000_:+_reg_ region: 217, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_218.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 0.411s.\n",
      "-- designing region: 4:3740000-3760000_:+_reg_ region: 218, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_219.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.515s.\n",
      "in 11.669s.\n",
      "-- designing region: 4:6240000-6260000_:+_reg_ region: 219, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_220.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.230s.\n",
      "-- designing region: 4:8740000-8760000_:+_reg_ region: 220, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_221.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.143s.\n",
      "in 11.272s.\n",
      "-- designing region: 4:11240000-11260000_:+_reg_ region: 221, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_222.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.402s.\n",
      "-- designing region: 4:13740000-13760000_:+_reg_ region: 222, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_223.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.272s.\n",
      "-- designing region: 4:16240000-16260000_:+_reg_ region: 223, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_224.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.331s.\n",
      "-- designing region: 4:18740000-18760000_:+_reg_ region: 224, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_225.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.381s.\n",
      "-- designing region: 4:21240000-21260000_:+_reg_ region: 225, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_226.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.292s.\n",
      "-- designing region: 4:23740000-23760000_:+_reg_ region: 226, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_227.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.340s.\n",
      "-- designing region: 4:26240000-26260000_:+_reg_ region: 227, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_228.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.225s.\n",
      "in 11.221s.\n",
      "-- designing region: 4:28740000-28760000_:+_reg_ region: 228, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_229.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.371s.\n",
      "-- designing region: 4:31240000-31260000_:+_reg_ region: 229, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_230.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.392s.\n",
      "-- designing region: 4:33740000-33760000_:+_reg_ region: 230, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_231.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.228s.\n",
      "-- designing region: 4:36240000-36260000_:+_reg_ region: 231, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_232.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.155s.\n",
      "in 11.189s.\n",
      "-- designing region: 4:38740000-38760000_:+_reg_ region: 232, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_233.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.196s.\n",
      "in 11.362s.\n",
      "-- designing region: 4:41240000-41260000_:+_reg_ region: 233, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_234.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.337s.\n",
      "-- designing region: 4:43740000-43760000_:+_reg_ region: 234, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_235.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.169s.\n",
      "in 11.331s.\n",
      "-- designing region: 4:46240000-46260000_:+_reg_ region: 235, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_236.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.324s.\n",
      "-- designing region: 4:48740000-48760000_:+_reg_ region: 236, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_237.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.329s.\n",
      "-- designing region: 4:51240000-51260000_:+_reg_ region: 237, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_238.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.299s.\n",
      "-- designing region: 4:53740000-53760000_:+_reg_ region: 238, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_239.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.207s.\n",
      "in 11.325s.\n",
      "-- designing region: 4:56240000-56260000_:+_reg_ region: 239, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_240.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.402s.\n",
      "-- designing region: 4:58740000-58760000_:+_reg_ region: 240, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_241.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.463s.\n",
      "-- designing region: 4:61240000-61260000_:+_reg_ region: 241, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_242.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.454s.\n",
      "-- designing region: 4:63740000-63760000_:+_reg_ region: 242, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_243.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.502s.\n",
      "-- designing region: 4:66240000-66260000_:+_reg_ region: 243, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_244.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.461s.\n",
      "-- designing region: 4:68740000-68760000_:+_reg_ region: 244, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_245.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.307s.\n",
      "-- designing region: 4:71240000-71260000_:+_reg_ region: 245, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_246.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.348s.\n",
      "-- designing region: 4:73740000-73760000_:+_reg_ region: 246, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_247.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.449s.\n",
      "-- designing region: 4:76240000-76260000_:+_reg_ region: 247, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_248.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.219s.\n",
      "in 16.074s.\n",
      "-- designing region: 4:78740000-78760000_:+_reg_ region: 248, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_249.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.178s.\n",
      "in 11.271s.\n",
      "-- designing region: 4:81240000-81260000_:+_reg_ region: 249, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_250.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.181s.\n",
      "in 11.450s.\n",
      "-- designing region: 4:83740000-83760000_:+_reg_ region: 250, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_251.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.338s.\n",
      "-- designing region: 4:86240000-86260000_:+_reg_ region: 251, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_252.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.297s.\n",
      "-- designing region: 4:88740000-88760000_:+_reg_ region: 252, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_253.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.487s.\n",
      "-- designing region: 4:91240000-91260000_:+_reg_ region: 253, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_254.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.401s.\n",
      "-- designing region: 4:93740000-93760000_:+_reg_ region: 254, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_255.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.403s.\n",
      "-- designing region: 4:96240000-96260000_:+_reg_ region: 255, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_256.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.354s.\n",
      "-- designing region: 4:98740000-98760000_:+_reg_ region: 256, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_257.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.295s.\n",
      "-- designing region: 4:101240000-101260000_:+_reg_ region: 257, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_258.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 11.314s.\n",
      "-- designing region: 4:103740000-103760000_:+_reg_ region: 258, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_259.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.314s.\n",
      "-- designing region: 4:106240000-106260000_:+_reg_ region: 259, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_260.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.212s.\n",
      "in 11.497s.\n",
      "-- designing region: 4:108740000-108760000_:+_reg_ region: 260, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_261.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.300s.\n",
      "-- designing region: 4:111240000-111260000_:+_reg_ region: 261, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_262.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 19.060s.\n",
      "-- designing region: 4:113740000-113760000_:+_reg_ region: 262, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_263.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.436s.\n",
      "in 11.971s.\n",
      "-- designing region: 4:116240000-116260000_:+_reg_ region: 263, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_264.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.439s.\n",
      "-- designing region: 4:118740000-118760000_:+_reg_ region: 264, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_265.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.476s.\n",
      "-- designing region: 4:121240000-121260000_:+_reg_ region: 265, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_266.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.345s.\n",
      "-- designing region: 4:123740000-123760000_:+_reg_ region: 266, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_267.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.441s.\n",
      "-- designing region: 4:126240000-126260000_:+_reg_ region: 267, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_268.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.198s.\n",
      "in 11.421s.\n",
      "-- designing region: 4:128740000-128760000_:+_reg_ region: 268, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_269.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.176s.\n",
      "in 11.243s.\n",
      "-- designing region: 4:131240000-131260000_:+_reg_ region: 269, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_270.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.426s.\n",
      "-- designing region: 4:133740000-133760000_:+_reg_ region: 270, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_271.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.147s.\n",
      "in 11.343s.\n",
      "-- designing region: 4:136240000-136260000_:+_reg_ region: 271, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_272.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.178s.\n",
      "in 11.447s.\n",
      "-- designing region: 4:138740000-138760000_:+_reg_ region: 272, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_273.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.356s.\n",
      "-- designing region: 4:141240000-141260000_:+_reg_ region: 273, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_274.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.359s.\n",
      "-- designing region: 4:143740000-143760000_:+_reg_ region: 274, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_275.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.383s.\n",
      "-- designing region: 4:146240000-146260000_:+_reg_ region: 275, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_276.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.198s.\n",
      "in 11.401s.\n",
      "-- designing region: 4:148740000-148760000_:+_reg_ region: 276, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_277.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.195s.\n",
      "in 11.466s.\n",
      "-- designing region: 4:151240000-151260000_:+_reg_ region: 277, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_278.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.331s.\n",
      "-- designing region: 4:153740000-153760000_:+_reg_ region: 278, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_279.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.345s.\n",
      "-- designing region: 4:156240000-156260000_:+_reg_ region: 279, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_280.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 8.884s.\n",
      "-- designing region: 5:1240000-1260000_:+_reg_ region: 280, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_281.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.145s.\n",
      "in 0.367s.\n",
      "-- designing region: 5:3740000-3760000_:+_reg_ region: 281, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_282.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.497s.\n",
      "in 11.662s.\n",
      "-- designing region: 5:6240000-6260000_:+_reg_ region: 282, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_283.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.399s.\n",
      "-- designing region: 5:8740000-8760000_:+_reg_ region: 283, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_284.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.460s.\n",
      "-- designing region: 5:11240000-11260000_:+_reg_ region: 284, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_285.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.338s.\n",
      "-- designing region: 5:13740000-13760000_:+_reg_ region: 285, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_286.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.329s.\n",
      "-- designing region: 5:16240000-16260000_:+_reg_ region: 286, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_287.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.487s.\n",
      "-- designing region: 5:18740000-18760000_:+_reg_ region: 287, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_288.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.204s.\n",
      "in 11.406s.\n",
      "-- designing region: 5:21240000-21260000_:+_reg_ region: 288, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_289.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.176s.\n",
      "in 11.430s.\n",
      "-- designing region: 5:23740000-23760000_:+_reg_ region: 289, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_290.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.342s.\n",
      "-- designing region: 5:26240000-26260000_:+_reg_ region: 290, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_291.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.357s.\n",
      "-- designing region: 5:28740000-28760000_:+_reg_ region: 291, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_292.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.490s.\n",
      "-- designing region: 5:31240000-31260000_:+_reg_ region: 292, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_293.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.194s.\n",
      "in 11.394s.\n",
      "-- designing region: 5:33740000-33760000_:+_reg_ region: 293, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_294.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.441s.\n",
      "-- designing region: 5:36240000-36260000_:+_reg_ region: 294, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_295.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.357s.\n",
      "-- designing region: 5:38740000-38760000_:+_reg_ region: 295, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_296.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.297s.\n",
      "-- designing region: 5:41240000-41260000_:+_reg_ region: 296, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_297.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.476s.\n",
      "-- designing region: 5:43740000-43760000_:+_reg_ region: 297, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_298.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.213s.\n",
      "in 11.404s.\n",
      "-- designing region: 5:46240000-46260000_:+_reg_ region: 298, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_299.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.187s.\n",
      "in 11.449s.\n",
      "-- designing region: 5:48740000-48760000_:+_reg_ region: 299, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_300.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.346s.\n",
      "-- designing region: 5:51240000-51260000_:+_reg_ region: 300, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_301.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.413s.\n",
      "-- designing region: 5:53740000-53760000_:+_reg_ region: 301, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_302.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.151s.\n",
      "in 12.358s.\n",
      "-- designing region: 5:56240000-56260000_:+_reg_ region: 302, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_303.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.308s.\n",
      "-- designing region: 5:58740000-58760000_:+_reg_ region: 303, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_304.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.257s.\n",
      "-- designing region: 5:61240000-61260000_:+_reg_ region: 304, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_305.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.182s.\n",
      "in 11.404s.\n",
      "-- designing region: 5:63740000-63760000_:+_reg_ region: 305, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_306.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.456s.\n",
      "-- designing region: 5:66240000-66260000_:+_reg_ region: 306, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_307.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.279s.\n",
      "-- designing region: 5:68740000-68760000_:+_reg_ region: 307, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_308.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.150s.\n",
      "in 11.317s.\n",
      "-- designing region: 5:71240000-71260000_:+_reg_ region: 308, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_309.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.446s.\n",
      "-- designing region: 5:73740000-73760000_:+_reg_ region: 309, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_310.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 17.182s.\n",
      "-- designing region: 5:76240000-76260000_:+_reg_ region: 310, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_311.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.370s.\n",
      "-- designing region: 5:78740000-78760000_:+_reg_ region: 311, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_312.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.225s.\n",
      "in 11.436s.\n",
      "-- designing region: 5:81240000-81260000_:+_reg_ region: 312, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_313.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.251s.\n",
      "-- designing region: 5:83740000-83760000_:+_reg_ region: 313, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_314.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.142s.\n",
      "in 11.338s.\n",
      "-- designing region: 5:86240000-86260000_:+_reg_ region: 314, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_315.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.209s.\n",
      "in 11.197s.\n",
      "-- designing region: 5:88740000-88760000_:+_reg_ region: 315, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_316.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.356s.\n",
      "-- designing region: 5:91240000-91260000_:+_reg_ region: 316, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_317.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.178s.\n",
      "in 11.382s.\n",
      "-- designing region: 5:93740000-93760000_:+_reg_ region: 317, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_318.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.429s.\n",
      "-- designing region: 5:96240000-96260000_:+_reg_ region: 318, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_319.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.272s.\n",
      "-- designing region: 5:98740000-98760000_:+_reg_ region: 319, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_320.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 11.374s.\n",
      "-- designing region: 5:101240000-101260000_:+_reg_ region: 320, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_321.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.440s.\n",
      "-- designing region: 5:103740000-103760000_:+_reg_ region: 321, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_322.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.379s.\n",
      "-- designing region: 5:106240000-106260000_:+_reg_ region: 322, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_323.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.452s.\n",
      "-- designing region: 5:108740000-108760000_:+_reg_ region: 323, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_324.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.306s.\n",
      "-- designing region: 5:111240000-111260000_:+_reg_ region: 324, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_325.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 11.381s.\n",
      "-- designing region: 5:113740000-113760000_:+_reg_ region: 325, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_326.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.223s.\n",
      "in 11.455s.\n",
      "-- designing region: 5:116240000-116260000_:+_reg_ region: 326, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_327.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.283s.\n",
      "-- designing region: 5:118740000-118760000_:+_reg_ region: 327, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_328.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.302s.\n",
      "-- designing region: 5:121240000-121260000_:+_reg_ region: 328, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_329.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.346s.\n",
      "-- designing region: 5:123740000-123760000_:+_reg_ region: 329, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_330.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.205s.\n",
      "in 11.414s.\n",
      "-- designing region: 5:126240000-126260000_:+_reg_ region: 330, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_331.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.336s.\n",
      "-- designing region: 5:128740000-128760000_:+_reg_ region: 331, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_332.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.354s.\n",
      "-- designing region: 5:131240000-131260000_:+_reg_ region: 332, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_333.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.140s.\n",
      "in 11.188s.\n",
      "-- designing region: 5:133740000-133760000_:+_reg_ region: 333, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_334.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.377s.\n",
      "-- designing region: 5:136240000-136260000_:+_reg_ region: 334, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_335.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.169s.\n",
      "in 11.320s.\n",
      "-- designing region: 5:138740000-138760000_:+_reg_ region: 335, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_336.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.208s.\n",
      "in 11.256s.\n",
      "-- designing region: 5:141240000-141260000_:+_reg_ region: 336, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_337.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.381s.\n",
      "-- designing region: 5:143740000-143760000_:+_reg_ region: 337, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_338.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.206s.\n",
      "in 11.337s.\n",
      "-- designing region: 5:146240000-146260000_:+_reg_ region: 338, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_339.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.433s.\n",
      "-- designing region: 5:148740000-148760000_:+_reg_ region: 339, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_340.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.327s.\n",
      "-- designing region: 5:151240000-151260000_:+_reg_ region: 340, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_341.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.312s.\n",
      "-- designing region: 6:1240000-1260000_:+_reg_ region: 341, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_342.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 0.383s.\n",
      "-- designing region: 6:3740000-3760000_:+_reg_ region: 342, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_343.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.492s.\n",
      "in 11.651s.\n",
      "-- designing region: 6:6240000-6260000_:+_reg_ region: 343, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_344.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.161s.\n",
      "in 11.381s.\n",
      "-- designing region: 6:8740000-8760000_:+_reg_ region: 344, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_345.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 17.574s.\n",
      "-- designing region: 6:11240000-11260000_:+_reg_ region: 345, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_346.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.302s.\n",
      "in 13.187s.\n",
      "-- designing region: 6:13740000-13760000_:+_reg_ region: 346, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_347.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.345s.\n",
      "-- designing region: 6:16240000-16260000_:+_reg_ region: 347, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_348.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.214s.\n",
      "in 11.406s.\n",
      "-- designing region: 6:18740000-18760000_:+_reg_ region: 348, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_349.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.218s.\n",
      "in 11.446s.\n",
      "-- designing region: 6:21240000-21260000_:+_reg_ region: 349, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_350.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.304s.\n",
      "-- designing region: 6:23740000-23760000_:+_reg_ region: 350, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_351.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 11.340s.\n",
      "-- designing region: 6:26240000-26260000_:+_reg_ region: 351, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_352.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.167s.\n",
      "in 11.414s.\n",
      "-- designing region: 6:28740000-28760000_:+_reg_ region: 352, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_353.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.346s.\n",
      "-- designing region: 6:31240000-31260000_:+_reg_ region: 353, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_354.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.339s.\n",
      "-- designing region: 6:33740000-33760000_:+_reg_ region: 354, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_355.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.201s.\n",
      "in 11.461s.\n",
      "-- designing region: 6:36240000-36260000_:+_reg_ region: 355, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_356.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.205s.\n",
      "in 11.414s.\n",
      "-- designing region: 6:38740000-38760000_:+_reg_ region: 356, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_357.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.273s.\n",
      "-- designing region: 6:41240000-41260000_:+_reg_ region: 357, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_358.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.411s.\n",
      "-- designing region: 6:43740000-43760000_:+_reg_ region: 358, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_359.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.157s.\n",
      "in 11.330s.\n",
      "-- designing region: 6:46240000-46260000_:+_reg_ region: 359, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_360.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.179s.\n",
      "in 11.438s.\n",
      "-- designing region: 6:48740000-48760000_:+_reg_ region: 360, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_361.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 11.322s.\n",
      "-- designing region: 6:51240000-51260000_:+_reg_ region: 361, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_362.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.347s.\n",
      "-- designing region: 6:53740000-53760000_:+_reg_ region: 362, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_363.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.378s.\n",
      "-- designing region: 6:56240000-56260000_:+_reg_ region: 363, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_364.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.435s.\n",
      "-- designing region: 6:58740000-58760000_:+_reg_ region: 364, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_365.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.218s.\n",
      "in 11.469s.\n",
      "-- designing region: 6:61240000-61260000_:+_reg_ region: 365, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_366.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.347s.\n",
      "-- designing region: 6:63740000-63760000_:+_reg_ region: 366, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_367.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.316s.\n",
      "-- designing region: 6:66240000-66260000_:+_reg_ region: 367, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_368.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.465s.\n",
      "-- designing region: 6:68740000-68760000_:+_reg_ region: 368, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_369.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.429s.\n",
      "-- designing region: 6:71240000-71260000_:+_reg_ region: 369, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_370.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.202s.\n",
      "in 11.453s.\n",
      "-- designing region: 6:73740000-73760000_:+_reg_ region: 370, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_371.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.307s.\n",
      "-- designing region: 6:76240000-76260000_:+_reg_ region: 371, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_372.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.365s.\n",
      "-- designing region: 6:78740000-78760000_:+_reg_ region: 372, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_373.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 11.461s.\n",
      "-- designing region: 6:81240000-81260000_:+_reg_ region: 373, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_374.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.222s.\n",
      "in 11.400s.\n",
      "-- designing region: 6:83740000-83760000_:+_reg_ region: 374, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_375.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.201s.\n",
      "in 11.450s.\n",
      "-- designing region: 6:86240000-86260000_:+_reg_ region: 375, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_376.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.325s.\n",
      "-- designing region: 6:88740000-88760000_:+_reg_ region: 376, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_377.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.360s.\n",
      "-- designing region: 6:91240000-91260000_:+_reg_ region: 377, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_378.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.483s.\n",
      "-- designing region: 6:93740000-93760000_:+_reg_ region: 378, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_379.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.223s.\n",
      "in 11.403s.\n",
      "-- designing region: 6:96240000-96260000_:+_reg_ region: 379, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_380.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.195s.\n",
      "in 11.439s.\n",
      "-- designing region: 6:98740000-98760000_:+_reg_ region: 380, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_381.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.336s.\n",
      "-- designing region: 6:101240000-101260000_:+_reg_ region: 381, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_382.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.349s.\n",
      "-- designing region: 6:103740000-103760000_:+_reg_ region: 382, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_383.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.442s.\n",
      "-- designing region: 6:106240000-106260000_:+_reg_ region: 383, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_384.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 18.715s.\n",
      "-- designing region: 6:108740000-108760000_:+_reg_ region: 384, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_385.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.397s.\n",
      "-- designing region: 6:111240000-111260000_:+_reg_ region: 385, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_386.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.185s.\n",
      "in 11.327s.\n",
      "-- designing region: 6:113740000-113760000_:+_reg_ region: 386, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_387.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.200s.\n",
      "in 11.424s.\n",
      "-- designing region: 6:116240000-116260000_:+_reg_ region: 387, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_388.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.325s.\n",
      "-- designing region: 6:118740000-118760000_:+_reg_ region: 388, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_389.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.251s.\n",
      "in 11.341s.\n",
      "-- designing region: 6:121240000-121260000_:+_reg_ region: 389, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_390.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.461s.\n",
      "-- designing region: 6:123740000-123760000_:+_reg_ region: 390, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_391.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.212s.\n",
      "in 11.407s.\n",
      "-- designing region: 6:126240000-126260000_:+_reg_ region: 391, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_392.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.200s.\n",
      "in 11.267s.\n",
      "-- designing region: 6:128740000-128760000_:+_reg_ region: 392, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_393.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.394s.\n",
      "-- designing region: 6:131240000-131260000_:+_reg_ region: 393, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_394.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.193s.\n",
      "in 11.357s.\n",
      "-- designing region: 6:133740000-133760000_:+_reg_ region: 394, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_395.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.195s.\n",
      "in 11.392s.\n",
      "-- designing region: 6:136240000-136260000_:+_reg_ region: 395, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_396.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.333s.\n",
      "-- designing region: 6:138740000-138760000_:+_reg_ region: 396, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_397.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.348s.\n",
      "-- designing region: 6:141240000-141260000_:+_reg_ region: 397, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_398.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.157s.\n",
      "in 11.155s.\n",
      "-- designing region: 6:143740000-143760000_:+_reg_ region: 398, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_399.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.395s.\n",
      "-- designing region: 6:146240000-146260000_:+_reg_ region: 399, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_400.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 11.357s.\n",
      "-- designing region: 6:148740000-148760000_:+_reg_ region: 400, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_401.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.196s.\n",
      "in 11.243s.\n",
      "-- designing region: 7:1240000-1260000_:+_reg_ region: 401, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_402.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.189s.\n",
      "in 0.415s.\n",
      "-- designing region: 7:3740000-3760000_:+_reg_ region: 402, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_403.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.518s.\n",
      "in 11.607s.\n",
      "-- designing region: 7:6240000-6260000_:+_reg_ region: 403, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_404.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.377s.\n",
      "-- designing region: 7:8740000-8760000_:+_reg_ region: 404, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_405.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.209s.\n",
      "in 11.356s.\n",
      "-- designing region: 7:11240000-11260000_:+_reg_ region: 405, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_406.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.202s.\n",
      "in 11.458s.\n",
      "-- designing region: 7:13740000-13760000_:+_reg_ region: 406, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_407.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.427s.\n",
      "-- designing region: 7:16240000-16260000_:+_reg_ region: 407, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_408.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.182s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.312s.\n",
      "-- designing region: 7:18740000-18760000_:+_reg_ region: 408, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_409.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.486s.\n",
      "-- designing region: 7:21240000-21260000_:+_reg_ region: 409, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_410.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.213s.\n",
      "in 11.295s.\n",
      "-- designing region: 7:23740000-23760000_:+_reg_ region: 410, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_411.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.213s.\n",
      "in 11.420s.\n",
      "-- designing region: 7:26240000-26260000_:+_reg_ region: 411, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_412.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.329s.\n",
      "-- designing region: 7:28740000-28760000_:+_reg_ region: 412, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_413.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.385s.\n",
      "-- designing region: 7:31240000-31260000_:+_reg_ region: 413, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_414.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.396s.\n",
      "-- designing region: 7:33740000-33760000_:+_reg_ region: 414, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_415.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.269s.\n",
      "-- designing region: 7:36240000-36260000_:+_reg_ region: 415, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_416.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.307s.\n",
      "-- designing region: 7:38740000-38760000_:+_reg_ region: 416, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_417.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.391s.\n",
      "-- designing region: 7:41240000-41260000_:+_reg_ region: 417, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_418.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.438s.\n",
      "-- designing region: 7:43740000-43760000_:+_reg_ region: 418, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_419.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.287s.\n",
      "-- designing region: 7:46240000-46260000_:+_reg_ region: 419, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_420.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.428s.\n",
      "-- designing region: 7:48740000-48760000_:+_reg_ region: 420, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_421.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.317s.\n",
      "-- designing region: 7:51240000-51260000_:+_reg_ region: 421, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_422.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.255s.\n",
      "-- designing region: 7:53740000-53760000_:+_reg_ region: 422, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_423.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.406s.\n",
      "-- designing region: 7:56240000-56260000_:+_reg_ region: 423, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_424.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.344s.\n",
      "-- designing region: 7:58740000-58760000_:+_reg_ region: 424, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_425.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.246s.\n",
      "-- designing region: 7:61240000-61260000_:+_reg_ region: 425, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_426.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.158s.\n",
      "in 11.367s.\n",
      "-- designing region: 7:63740000-63760000_:+_reg_ region: 426, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_427.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.365s.\n",
      "-- designing region: 7:66240000-66260000_:+_reg_ region: 427, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_428.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.179s.\n",
      "in 11.269s.\n",
      "-- designing region: 7:68740000-68760000_:+_reg_ region: 428, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_429.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.164s.\n",
      "in 11.385s.\n",
      "-- designing region: 7:71240000-71260000_:+_reg_ region: 429, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_430.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.317s.\n",
      "-- designing region: 7:73740000-73760000_:+_reg_ region: 430, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_431.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.316s.\n",
      "-- designing region: 7:76240000-76260000_:+_reg_ region: 431, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_432.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.146s.\n",
      "in 11.337s.\n",
      "-- designing region: 7:78740000-78760000_:+_reg_ region: 432, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_433.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.366s.\n",
      "-- designing region: 7:81240000-81260000_:+_reg_ region: 433, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_434.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.178s.\n",
      "in 11.209s.\n",
      "-- designing region: 7:83740000-83760000_:+_reg_ region: 434, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_435.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.431s.\n",
      "-- designing region: 7:86240000-86260000_:+_reg_ region: 435, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_436.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.340s.\n",
      "-- designing region: 7:88740000-88760000_:+_reg_ region: 436, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_437.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.187s.\n",
      "in 11.209s.\n",
      "-- designing region: 7:91240000-91260000_:+_reg_ region: 437, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_438.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.212s.\n",
      "in 11.420s.\n",
      "-- designing region: 7:93740000-93760000_:+_reg_ region: 438, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_439.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.369s.\n",
      "-- designing region: 7:96240000-96260000_:+_reg_ region: 439, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_440.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.284s.\n",
      "-- designing region: 7:98740000-98760000_:+_reg_ region: 440, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_441.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.443s.\n",
      "-- designing region: 7:101240000-101260000_:+_reg_ region: 441, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_442.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.311s.\n",
      "-- designing region: 7:103740000-103760000_:+_reg_ region: 442, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_443.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.173s.\n",
      "in 19.009s.\n",
      "-- designing region: 7:106240000-106260000_:+_reg_ region: 443, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_444.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.330s.\n",
      "in 13.077s.\n",
      "-- designing region: 7:108740000-108760000_:+_reg_ region: 444, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_445.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.421s.\n",
      "-- designing region: 7:111240000-111260000_:+_reg_ region: 445, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_446.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.443s.\n",
      "-- designing region: 7:113740000-113760000_:+_reg_ region: 446, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_447.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.192s.\n",
      "in 11.355s.\n",
      "-- designing region: 7:116240000-116260000_:+_reg_ region: 447, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_448.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.254s.\n",
      "in 11.516s.\n",
      "-- designing region: 7:118740000-118760000_:+_reg_ region: 448, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_449.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.383s.\n",
      "-- designing region: 7:121240000-121260000_:+_reg_ region: 449, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_450.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.185s.\n",
      "in 11.335s.\n",
      "-- designing region: 7:123740000-123760000_:+_reg_ region: 450, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_451.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.415s.\n",
      "-- designing region: 7:126240000-126260000_:+_reg_ region: 451, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_452.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.195s.\n",
      "in 11.347s.\n",
      "-- designing region: 7:128740000-128760000_:+_reg_ region: 452, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_453.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.487s.\n",
      "-- designing region: 7:131240000-131260000_:+_reg_ region: 453, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_454.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.331s.\n",
      "-- designing region: 7:133740000-133760000_:+_reg_ region: 454, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_455.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.512s.\n",
      "-- designing region: 7:136240000-136260000_:+_reg_ region: 455, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_456.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.375s.\n",
      "-- designing region: 7:138740000-138760000_:+_reg_ region: 456, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_457.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.319s.\n",
      "-- designing region: 7:141240000-141260000_:+_reg_ region: 457, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_458.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.483s.\n",
      "-- designing region: 7:143740000-143760000_:+_reg_ region: 458, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_459.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.409s.\n",
      "-- designing region: 7:146240000-145441459_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: 8:1240000-1260000_:+_reg_ region: 460, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_461.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.138s.\n",
      "in 0.362s.\n",
      "-- designing region: 8:3740000-3760000_:+_reg_ region: 461, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_462.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.547s.\n",
      "in 11.698s.\n",
      "-- designing region: 8:6240000-6260000_:+_reg_ region: 462, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_463.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.483s.\n",
      "-- designing region: 8:8740000-8760000_:+_reg_ region: 463, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_464.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.324s.\n",
      "-- designing region: 8:11240000-11260000_:+_reg_ region: 464, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_465.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.488s.\n",
      "-- designing region: 8:13740000-13760000_:+_reg_ region: 465, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_466.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.431s.\n",
      "-- designing region: 8:16240000-16260000_:+_reg_ region: 466, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_467.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.153s.\n",
      "in 11.245s.\n",
      "-- designing region: 8:18740000-18760000_:+_reg_ region: 467, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_468.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.164s.\n",
      "in 11.381s.\n",
      "-- designing region: 8:21240000-21260000_:+_reg_ region: 468, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_469.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.362s.\n",
      "-- designing region: 8:23740000-23760000_:+_reg_ region: 469, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_470.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.157s.\n",
      "in 11.307s.\n",
      "-- designing region: 8:26240000-26260000_:+_reg_ region: 470, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_471.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.463s.\n",
      "-- designing region: 8:28740000-28760000_:+_reg_ region: 471, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_472.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.328s.\n",
      "-- designing region: 8:31240000-31260000_:+_reg_ region: 472, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_473.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.497s.\n",
      "-- designing region: 8:33740000-33760000_:+_reg_ region: 473, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_474.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.372s.\n",
      "-- designing region: 8:36240000-36260000_:+_reg_ region: 474, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_475.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.302s.\n",
      "-- designing region: 8:38740000-38760000_:+_reg_ region: 475, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_476.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.392s.\n",
      "-- designing region: 8:41240000-41260000_:+_reg_ region: 476, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_477.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.247s.\n",
      "-- designing region: 8:43740000-43760000_:+_reg_ region: 477, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_478.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.210s.\n",
      "in 11.263s.\n",
      "-- designing region: 8:46240000-46260000_:+_reg_ region: 478, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_479.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.200s.\n",
      "in 20.499s.\n",
      "-- designing region: 8:48740000-48760000_:+_reg_ region: 479, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_480.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.254s.\n",
      "in 11.353s.\n",
      "-- designing region: 8:51240000-51260000_:+_reg_ region: 480, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_481.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.250s.\n",
      "in 11.324s.\n",
      "-- designing region: 8:53740000-53760000_:+_reg_ region: 481, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_482.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.371s.\n",
      "-- designing region: 8:56240000-56260000_:+_reg_ region: 482, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_483.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.252s.\n",
      "in 11.333s.\n",
      "-- designing region: 8:58740000-58760000_:+_reg_ region: 483, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_484.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.298s.\n",
      "-- designing region: 8:61240000-61260000_:+_reg_ region: 484, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_485.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.388s.\n",
      "-- designing region: 8:63740000-63760000_:+_reg_ region: 485, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_486.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.256s.\n",
      "in 11.319s.\n",
      "-- designing region: 8:66240000-66260000_:+_reg_ region: 486, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_487.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.431s.\n",
      "-- designing region: 8:68740000-68760000_:+_reg_ region: 487, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_488.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.278s.\n",
      "-- designing region: 8:71240000-71260000_:+_reg_ region: 488, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_489.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.246s.\n",
      "-- designing region: 8:73740000-73760000_:+_reg_ region: 489, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_490.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.267s.\n",
      "-- designing region: 8:76240000-76260000_:+_reg_ region: 490, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_491.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.352s.\n",
      "-- designing region: 8:78740000-78760000_:+_reg_ region: 491, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_492.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.359s.\n",
      "-- designing region: 8:81240000-81260000_:+_reg_ region: 492, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_493.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.169s.\n",
      "in 11.282s.\n",
      "-- designing region: 8:83740000-83760000_:+_reg_ region: 493, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_494.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.465s.\n",
      "-- designing region: 8:86240000-86260000_:+_reg_ region: 494, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_495.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.339s.\n",
      "-- designing region: 8:88740000-88760000_:+_reg_ region: 495, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_496.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.461s.\n",
      "-- designing region: 8:91240000-91260000_:+_reg_ region: 496, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_497.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.378s.\n",
      "-- designing region: 8:93740000-93760000_:+_reg_ region: 497, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_498.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.313s.\n",
      "-- designing region: 8:96240000-96260000_:+_reg_ region: 498, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_499.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 11.404s.\n",
      "-- designing region: 8:98740000-98760000_:+_reg_ region: 499, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_500.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.344s.\n",
      "-- designing region: 8:101240000-101260000_:+_reg_ region: 500, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_501.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.255s.\n",
      "-- designing region: 8:103740000-103760000_:+_reg_ region: 501, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_502.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.406s.\n",
      "-- designing region: 8:106240000-106260000_:+_reg_ region: 502, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_503.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.352s.\n",
      "-- designing region: 8:108740000-108760000_:+_reg_ region: 503, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_504.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.151s.\n",
      "in 11.258s.\n",
      "-- designing region: 8:111240000-111260000_:+_reg_ region: 504, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_505.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.153s.\n",
      "in 11.366s.\n",
      "-- designing region: 8:113740000-113760000_:+_reg_ region: 505, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_506.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.364s.\n",
      "-- designing region: 8:116240000-116260000_:+_reg_ region: 506, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_507.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.401s.\n",
      "-- designing region: 8:118740000-118760000_:+_reg_ region: 507, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_508.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.374s.\n",
      "-- designing region: 8:121240000-121260000_:+_reg_ region: 508, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_509.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.289s.\n",
      "-- designing region: 8:123740000-123760000_:+_reg_ region: 509, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_510.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.276s.\n",
      "-- designing region: 8:126240000-126260000_:+_reg_ region: 510, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_511.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.511s.\n",
      "-- designing region: 8:128740000-128760000_:+_reg_ region: 511, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_512.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.469s.\n",
      "-- designing region: 9:1240000-1260000_:+_reg_ region: 512, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_513.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.196s.\n",
      "in 0.415s.\n",
      "-- designing region: 9:3740000-3760000_:+_reg_ region: 513, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_514.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.432s.\n",
      "in 11.515s.\n",
      "-- designing region: 9:6240000-6260000_:+_reg_ region: 514, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_515.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.374s.\n",
      "-- designing region: 9:8740000-8760000_:+_reg_ region: 515, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_516.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.340s.\n",
      "-- designing region: 9:11240000-11260000_:+_reg_ region: 516, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_517.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.253s.\n",
      "-- designing region: 9:13740000-13760000_:+_reg_ region: 517, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_518.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.164s.\n",
      "in 11.354s.\n",
      "-- designing region: 9:16240000-16260000_:+_reg_ region: 518, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_519.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.369s.\n",
      "-- designing region: 9:18740000-18760000_:+_reg_ region: 519, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_520.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.265s.\n",
      "-- designing region: 9:21240000-21260000_:+_reg_ region: 520, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_521.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.417s.\n",
      "-- designing region: 9:23740000-23760000_:+_reg_ region: 521, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_522.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.369s.\n",
      "-- designing region: 9:26240000-26260000_:+_reg_ region: 522, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_523.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.301s.\n",
      "-- designing region: 9:28740000-28760000_:+_reg_ region: 523, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_524.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.409s.\n",
      "-- designing region: 9:31240000-31260000_:+_reg_ region: 524, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_525.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.176s.\n",
      "in 11.280s.\n",
      "-- designing region: 9:33740000-33760000_:+_reg_ region: 525, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_526.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.156s.\n",
      "in 17.102s.\n",
      "-- designing region: 9:36240000-36260000_:+_reg_ region: 526, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_527.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.411s.\n",
      "in 15.062s.\n",
      "-- designing region: 9:38740000-38760000_:+_reg_ region: 527, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_528.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.202s.\n",
      "in 11.379s.\n",
      "-- designing region: 9:41240000-41260000_:+_reg_ region: 528, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_529.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.417s.\n",
      "-- designing region: 9:43740000-43760000_:+_reg_ region: 529, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_530.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.179s.\n",
      "in 11.452s.\n",
      "-- designing region: 9:46240000-46260000_:+_reg_ region: 530, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_531.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.509s.\n",
      "-- designing region: 9:48740000-48760000_:+_reg_ region: 531, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_532.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.194s.\n",
      "in 11.355s.\n",
      "-- designing region: 9:51240000-51260000_:+_reg_ region: 532, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_533.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.155s.\n",
      "in 11.383s.\n",
      "-- designing region: 9:53740000-53760000_:+_reg_ region: 533, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_534.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.250s.\n",
      "in 11.368s.\n",
      "-- designing region: 9:56240000-56260000_:+_reg_ region: 534, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_535.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.342s.\n",
      "-- designing region: 9:58740000-58760000_:+_reg_ region: 535, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_536.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.367s.\n",
      "-- designing region: 9:61240000-61260000_:+_reg_ region: 536, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_537.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.350s.\n",
      "-- designing region: 9:63740000-63760000_:+_reg_ region: 537, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_538.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.500s.\n",
      "-- designing region: 9:66240000-66260000_:+_reg_ region: 538, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_539.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.201s.\n",
      "in 11.382s.\n",
      "-- designing region: 9:68740000-68760000_:+_reg_ region: 539, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_540.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.176s.\n",
      "in 11.441s.\n",
      "-- designing region: 9:71240000-71260000_:+_reg_ region: 540, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_541.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.450s.\n",
      "-- designing region: 9:73740000-73760000_:+_reg_ region: 541, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_542.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.218s.\n",
      "in 11.359s.\n",
      "-- designing region: 9:76240000-76260000_:+_reg_ region: 542, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_543.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.433s.\n",
      "-- designing region: 9:78740000-78760000_:+_reg_ region: 543, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_544.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.379s.\n",
      "-- designing region: 9:81240000-81260000_:+_reg_ region: 544, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_545.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.522s.\n",
      "-- designing region: 9:83740000-83760000_:+_reg_ region: 545, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_546.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.413s.\n",
      "-- designing region: 9:86240000-86260000_:+_reg_ region: 546, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_547.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.433s.\n",
      "-- designing region: 9:88740000-88760000_:+_reg_ region: 547, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_548.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.446s.\n",
      "-- designing region: 9:91240000-91260000_:+_reg_ region: 548, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_549.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.392s.\n",
      "-- designing region: 9:93740000-93760000_:+_reg_ region: 549, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_550.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.433s.\n",
      "-- designing region: 9:96240000-96260000_:+_reg_ region: 550, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_551.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.362s.\n",
      "-- designing region: 9:98740000-98760000_:+_reg_ region: 551, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_552.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.326s.\n",
      "-- designing region: 9:101240000-101260000_:+_reg_ region: 552, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_553.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.340s.\n",
      "-- designing region: 9:103740000-103760000_:+_reg_ region: 553, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_554.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.178s.\n",
      "in 11.345s.\n",
      "-- designing region: 9:106240000-106260000_:+_reg_ region: 554, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_555.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.248s.\n",
      "in 11.518s.\n",
      "-- designing region: 9:108740000-108760000_:+_reg_ region: 555, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_556.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.403s.\n",
      "-- designing region: 9:111240000-111260000_:+_reg_ region: 556, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_557.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 11.425s.\n",
      "-- designing region: 9:113740000-113760000_:+_reg_ region: 557, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_558.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.466s.\n",
      "-- designing region: 9:116240000-116260000_:+_reg_ region: 558, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_559.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.335s.\n",
      "-- designing region: 9:118740000-118760000_:+_reg_ region: 559, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_560.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.322s.\n",
      "-- designing region: 9:121240000-121260000_:+_reg_ region: 560, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_561.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.411s.\n",
      "-- designing region: 9:123740000-123760000_:+_reg_ region: 561, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_562.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.187s.\n",
      "in 11.334s.\n",
      "-- designing region: 10:1240000-1260000_:+_reg_ region: 562, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_563.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.193s.\n",
      "in 0.414s.\n",
      "-- designing region: 10:3740000-3760000_:+_reg_ region: 563, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_564.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.464s.\n",
      "in 11.723s.\n",
      "-- designing region: 10:6240000-6260000_:+_reg_ region: 564, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_565.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.320s.\n",
      "-- designing region: 10:8740000-8760000_:+_reg_ region: 565, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_566.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.353s.\n",
      "-- designing region: 10:11240000-11260000_:+_reg_ region: 566, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_567.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.485s.\n",
      "-- designing region: 10:13740000-13760000_:+_reg_ region: 567, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_568.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.390s.\n",
      "-- designing region: 10:16240000-16260000_:+_reg_ region: 568, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_569.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.405s.\n",
      "-- designing region: 10:18740000-18760000_:+_reg_ region: 569, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_570.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.489s.\n",
      "-- designing region: 10:21240000-21260000_:+_reg_ region: 570, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_571.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.410s.\n",
      "-- designing region: 10:23740000-23760000_:+_reg_ region: 571, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_572.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.417s.\n",
      "-- designing region: 10:26240000-26260000_:+_reg_ region: 572, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_573.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.316s.\n",
      "-- designing region: 10:28740000-28760000_:+_reg_ region: 573, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_574.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.253s.\n",
      "in 11.344s.\n",
      "-- designing region: 10:31240000-31260000_:+_reg_ region: 574, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_575.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.440s.\n",
      "-- designing region: 10:33740000-33760000_:+_reg_ region: 575, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_576.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.281s.\n",
      "-- designing region: 10:36240000-36260000_:+_reg_ region: 576, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_577.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.267s.\n",
      "-- designing region: 10:38740000-38760000_:+_reg_ region: 577, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_578.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.442s.\n",
      "-- designing region: 10:41240000-41260000_:+_reg_ region: 578, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_579.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.301s.\n",
      "-- designing region: 10:43740000-43760000_:+_reg_ region: 579, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_580.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.314s.\n",
      "-- designing region: 10:46240000-46260000_:+_reg_ region: 580, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_581.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.443s.\n",
      "-- designing region: 10:48740000-48760000_:+_reg_ region: 581, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_582.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.385s.\n",
      "-- designing region: 10:51240000-51260000_:+_reg_ region: 582, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_583.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.185s.\n",
      "in 11.417s.\n",
      "-- designing region: 10:53740000-53760000_:+_reg_ region: 583, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_584.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.364s.\n",
      "-- designing region: 10:56240000-56260000_:+_reg_ region: 584, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_585.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.150s.\n",
      "in 11.257s.\n",
      "-- designing region: 10:58740000-58760000_:+_reg_ region: 585, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_586.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.373s.\n",
      "-- designing region: 10:61240000-61260000_:+_reg_ region: 586, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_587.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.363s.\n",
      "-- designing region: 10:63740000-63760000_:+_reg_ region: 587, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_588.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.157s.\n",
      "in 11.228s.\n",
      "-- designing region: 10:66240000-66260000_:+_reg_ region: 588, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_589.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.415s.\n",
      "-- designing region: 10:68740000-68760000_:+_reg_ region: 589, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_590.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.284s.\n",
      "-- designing region: 10:71240000-71260000_:+_reg_ region: 590, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_591.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.295s.\n",
      "-- designing region: 10:73740000-73760000_:+_reg_ region: 591, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_592.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.189s.\n",
      "in 11.369s.\n",
      "-- designing region: 10:76240000-76260000_:+_reg_ region: 592, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_593.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.386s.\n",
      "-- designing region: 10:78740000-78760000_:+_reg_ region: 593, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_594.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.164s.\n",
      "in 22.385s.\n",
      "-- designing region: 10:81240000-81260000_:+_reg_ region: 594, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_595.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.443s.\n",
      "-- designing region: 10:83740000-83760000_:+_reg_ region: 595, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_596.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.398s.\n",
      "-- designing region: 10:86240000-86260000_:+_reg_ region: 596, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_597.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.189s.\n",
      "in 11.427s.\n",
      "-- designing region: 10:88740000-88760000_:+_reg_ region: 597, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_598.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.360s.\n",
      "-- designing region: 10:91240000-91260000_:+_reg_ region: 598, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_599.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.152s.\n",
      "in 11.289s.\n",
      "-- designing region: 10:93740000-93760000_:+_reg_ region: 599, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_600.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.155s.\n",
      "in 11.358s.\n",
      "-- designing region: 10:96240000-96260000_:+_reg_ region: 600, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_601.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.314s.\n",
      "-- designing region: 10:98740000-98760000_:+_reg_ region: 601, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_602.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.281s.\n",
      "-- designing region: 10:101240000-101260000_:+_reg_ region: 602, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_603.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.171s.\n",
      "in 11.343s.\n",
      "-- designing region: 10:103740000-103760000_:+_reg_ region: 603, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_604.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.308s.\n",
      "-- designing region: 10:106240000-106260000_:+_reg_ region: 604, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_605.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.208s.\n",
      "in 13.288s.\n",
      "-- designing region: 10:108740000-108760000_:+_reg_ region: 605, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_606.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.207s.\n",
      "in 11.377s.\n",
      "-- designing region: 10:111240000-111260000_:+_reg_ region: 606, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_607.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 16.116s.\n",
      "-- designing region: 10:113740000-113760000_:+_reg_ region: 607, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_608.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 14.480s.\n",
      "-- designing region: 10:116240000-116260000_:+_reg_ region: 608, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_609.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.433s.\n",
      "in 12.579s.\n",
      "-- designing region: 10:118740000-118760000_:+_reg_ region: 609, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_610.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.274s.\n",
      "-- designing region: 10:121240000-121260000_:+_reg_ region: 610, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_611.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.303s.\n",
      "-- designing region: 10:123740000-123760000_:+_reg_ region: 611, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_612.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.250s.\n",
      "in 11.481s.\n",
      "-- designing region: 10:126240000-126260000_:+_reg_ region: 612, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_613.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 11.382s.\n",
      "-- designing region: 10:128740000-128760000_:+_reg_ region: 613, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_614.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.209s.\n",
      "in 11.456s.\n",
      "-- designing region: 10:131240000-130694993_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: 11:1240000-1260000_:+_reg_ region: 615, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_616.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.194s.\n",
      "in 0.413s.\n",
      "-- designing region: 11:3740000-3760000_:+_reg_ region: 616, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_617.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.609s.\n",
      "in 11.806s.\n",
      "-- designing region: 11:6240000-6260000_:+_reg_ region: 617, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_618.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.180s.\n",
      "in 11.417s.\n",
      "-- designing region: 11:8740000-8760000_:+_reg_ region: 618, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_619.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.456s.\n",
      "-- designing region: 11:11240000-11260000_:+_reg_ region: 619, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_620.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.331s.\n",
      "-- designing region: 11:13740000-13760000_:+_reg_ region: 620, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_621.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.439s.\n",
      "-- designing region: 11:16240000-16260000_:+_reg_ region: 621, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_622.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.345s.\n",
      "-- designing region: 11:18740000-18760000_:+_reg_ region: 622, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_623.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.193s.\n",
      "in 11.389s.\n",
      "-- designing region: 11:21240000-21260000_:+_reg_ region: 623, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_624.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.460s.\n",
      "-- designing region: 11:23740000-23760000_:+_reg_ region: 624, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_625.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.221s.\n",
      "-- designing region: 11:26240000-26260000_:+_reg_ region: 625, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_626.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.157s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.342s.\n",
      "-- designing region: 11:28740000-28760000_:+_reg_ region: 626, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_627.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.397s.\n",
      "-- designing region: 11:31240000-31260000_:+_reg_ region: 627, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_628.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.314s.\n",
      "-- designing region: 11:33740000-33760000_:+_reg_ region: 628, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_629.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 11.388s.\n",
      "-- designing region: 11:36240000-36260000_:+_reg_ region: 629, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_630.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.442s.\n",
      "-- designing region: 11:38740000-38760000_:+_reg_ region: 630, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_631.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.222s.\n",
      "in 11.245s.\n",
      "-- designing region: 11:41240000-41260000_:+_reg_ region: 631, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_632.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.250s.\n",
      "in 11.464s.\n",
      "-- designing region: 11:43740000-43760000_:+_reg_ region: 632, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_633.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.403s.\n",
      "-- designing region: 11:46240000-46260000_:+_reg_ region: 633, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_634.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.148s.\n",
      "in 11.386s.\n",
      "-- designing region: 11:48740000-48760000_:+_reg_ region: 634, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_635.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.441s.\n",
      "-- designing region: 11:51240000-51260000_:+_reg_ region: 635, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_636.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.323s.\n",
      "-- designing region: 11:53740000-53760000_:+_reg_ region: 636, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_637.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.478s.\n",
      "-- designing region: 11:56240000-56260000_:+_reg_ region: 637, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_638.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.332s.\n",
      "-- designing region: 11:58740000-58760000_:+_reg_ region: 638, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_639.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 11.363s.\n",
      "-- designing region: 11:61240000-61260000_:+_reg_ region: 639, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_640.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.457s.\n",
      "-- designing region: 11:63740000-63760000_:+_reg_ region: 640, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_641.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.282s.\n",
      "-- designing region: 11:66240000-66260000_:+_reg_ region: 641, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_642.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.164s.\n",
      "in 11.378s.\n",
      "-- designing region: 11:68740000-68760000_:+_reg_ region: 642, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_643.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.432s.\n",
      "-- designing region: 11:71240000-71260000_:+_reg_ region: 643, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_644.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.218s.\n",
      "in 11.285s.\n",
      "-- designing region: 11:73740000-73760000_:+_reg_ region: 644, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_645.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.430s.\n",
      "-- designing region: 11:76240000-76260000_:+_reg_ region: 645, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_646.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.250s.\n",
      "in 11.401s.\n",
      "-- designing region: 11:78740000-78760000_:+_reg_ region: 646, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_647.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.155s.\n",
      "in 11.401s.\n",
      "-- designing region: 11:81240000-81260000_:+_reg_ region: 647, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_648.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.408s.\n",
      "-- designing region: 11:83740000-83760000_:+_reg_ region: 648, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_649.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.289s.\n",
      "-- designing region: 11:86240000-86260000_:+_reg_ region: 649, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_650.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.427s.\n",
      "-- designing region: 11:88740000-88760000_:+_reg_ region: 650, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_651.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.374s.\n",
      "-- designing region: 11:91240000-91260000_:+_reg_ region: 651, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_652.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.204s.\n",
      "in 11.236s.\n",
      "-- designing region: 11:93740000-93760000_:+_reg_ region: 652, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_653.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.488s.\n",
      "-- designing region: 11:96240000-96260000_:+_reg_ region: 653, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_654.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.334s.\n",
      "-- designing region: 11:98740000-98760000_:+_reg_ region: 654, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_655.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.174s.\n",
      "in 11.228s.\n",
      "-- designing region: 11:101240000-101260000_:+_reg_ region: 655, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_656.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.451s.\n",
      "-- designing region: 11:103740000-103760000_:+_reg_ region: 656, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_657.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.366s.\n",
      "-- designing region: 11:106240000-106260000_:+_reg_ region: 657, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_658.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.435s.\n",
      "-- designing region: 11:108740000-108760000_:+_reg_ region: 658, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_659.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.472s.\n",
      "-- designing region: 11:111240000-111260000_:+_reg_ region: 659, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_660.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.225s.\n",
      "in 11.273s.\n",
      "-- designing region: 11:113740000-113760000_:+_reg_ region: 660, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_661.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.446s.\n",
      "-- designing region: 11:116240000-116260000_:+_reg_ region: 661, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_662.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.395s.\n",
      "-- designing region: 11:118740000-118760000_:+_reg_ region: 662, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_663.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.456s.\n",
      "-- designing region: 11:121240000-121260000_:+_reg_ region: 663, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_664.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.457s.\n",
      "-- designing region: 12:1240000-1260000_:+_reg_ region: 664, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_665.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.211s.\n",
      "in 0.436s.\n",
      "-- designing region: 12:3740000-3760000_:+_reg_ region: 665, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_666.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.597s.\n",
      "in 11.796s.\n",
      "-- designing region: 12:6240000-6260000_:+_reg_ region: 666, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_667.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.435s.\n",
      "-- designing region: 12:8740000-8760000_:+_reg_ region: 667, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_668.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.435s.\n",
      "-- designing region: 12:11240000-11260000_:+_reg_ region: 668, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_669.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.408s.\n",
      "-- designing region: 12:13740000-13760000_:+_reg_ region: 669, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_670.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.464s.\n",
      "-- designing region: 12:16240000-16260000_:+_reg_ region: 670, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_671.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.466s.\n",
      "-- designing region: 12:18740000-18760000_:+_reg_ region: 671, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_672.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.291s.\n",
      "-- designing region: 12:21240000-21260000_:+_reg_ region: 672, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_673.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.450s.\n",
      "-- designing region: 12:23740000-23760000_:+_reg_ region: 673, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_674.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.453s.\n",
      "-- designing region: 12:26240000-26260000_:+_reg_ region: 674, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_675.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.443s.\n",
      "-- designing region: 12:28740000-28760000_:+_reg_ region: 675, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_676.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.225s.\n",
      "in 11.446s.\n",
      "-- designing region: 12:31240000-31260000_:+_reg_ region: 676, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_677.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.257s.\n",
      "in 11.469s.\n",
      "-- designing region: 12:33740000-33760000_:+_reg_ region: 677, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_678.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.306s.\n",
      "-- designing region: 12:36240000-36260000_:+_reg_ region: 678, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_679.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.436s.\n",
      "-- designing region: 12:38740000-38760000_:+_reg_ region: 679, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_680.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.438s.\n",
      "-- designing region: 12:41240000-41260000_:+_reg_ region: 680, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_681.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.431s.\n",
      "-- designing region: 12:43740000-43760000_:+_reg_ region: 681, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_682.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.423s.\n",
      "-- designing region: 12:46240000-46260000_:+_reg_ region: 682, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_683.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.451s.\n",
      "-- designing region: 12:48740000-48760000_:+_reg_ region: 683, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_684.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.419s.\n",
      "-- designing region: 12:51240000-51260000_:+_reg_ region: 684, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_685.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.342s.\n",
      "-- designing region: 12:53740000-53760000_:+_reg_ region: 685, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_686.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.441s.\n",
      "-- designing region: 12:56240000-56260000_:+_reg_ region: 686, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_687.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.409s.\n",
      "-- designing region: 12:58740000-58760000_:+_reg_ region: 687, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_688.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.419s.\n",
      "-- designing region: 12:61240000-61260000_:+_reg_ region: 688, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_689.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.427s.\n",
      "-- designing region: 12:63740000-63760000_:+_reg_ region: 689, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_690.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.484s.\n",
      "-- designing region: 12:66240000-66260000_:+_reg_ region: 690, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_691.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.314s.\n",
      "-- designing region: 12:68740000-68760000_:+_reg_ region: 691, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_692.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 17.966s.\n",
      "-- designing region: 12:71240000-71260000_:+_reg_ region: 692, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_693.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.393s.\n",
      "-- designing region: 12:73740000-73760000_:+_reg_ region: 693, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_694.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.406s.\n",
      "-- designing region: 12:76240000-76260000_:+_reg_ region: 694, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_695.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.388s.\n",
      "-- designing region: 12:78740000-78760000_:+_reg_ region: 695, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_696.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.392s.\n",
      "-- designing region: 12:81240000-81260000_:+_reg_ region: 696, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_697.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.404s.\n",
      "-- designing region: 12:83740000-83760000_:+_reg_ region: 697, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_698.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.395s.\n",
      "-- designing region: 12:86240000-86260000_:+_reg_ region: 698, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_699.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.414s.\n",
      "-- designing region: 12:88740000-88760000_:+_reg_ region: 699, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_700.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.405s.\n",
      "-- designing region: 12:91240000-91260000_:+_reg_ region: 700, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_701.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.408s.\n",
      "-- designing region: 12:93740000-93760000_:+_reg_ region: 701, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_702.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.382s.\n",
      "-- designing region: 12:96240000-96260000_:+_reg_ region: 702, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_703.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.390s.\n",
      "-- designing region: 12:98740000-98760000_:+_reg_ region: 703, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_704.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.379s.\n",
      "-- designing region: 12:101240000-101260000_:+_reg_ region: 704, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_705.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.369s.\n",
      "-- designing region: 12:103740000-103760000_:+_reg_ region: 705, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_706.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.386s.\n",
      "-- designing region: 12:106240000-106260000_:+_reg_ region: 706, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_707.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.412s.\n",
      "-- designing region: 12:108740000-108760000_:+_reg_ region: 707, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_708.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.474s.\n",
      "-- designing region: 12:111240000-111260000_:+_reg_ region: 708, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_709.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.251s.\n",
      "in 11.498s.\n",
      "-- designing region: 12:113740000-113760000_:+_reg_ region: 709, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_710.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.254s.\n",
      "in 11.491s.\n",
      "-- designing region: 12:116240000-116260000_:+_reg_ region: 710, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_711.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.373s.\n",
      "-- designing region: 12:118740000-118760000_:+_reg_ region: 711, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_712.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.471s.\n",
      "-- designing region: 13:1240000-1260000_:+_reg_ region: 712, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_713.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.202s.\n",
      "in 0.426s.\n",
      "-- designing region: 13:3740000-3760000_:+_reg_ region: 713, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_714.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.469s.\n",
      "in 11.697s.\n",
      "-- designing region: 13:6240000-6260000_:+_reg_ region: 714, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_715.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.248s.\n",
      "in 11.501s.\n",
      "-- designing region: 13:8740000-8760000_:+_reg_ region: 715, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_716.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.475s.\n",
      "-- designing region: 13:11240000-11260000_:+_reg_ region: 716, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_717.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.484s.\n",
      "-- designing region: 13:13740000-13760000_:+_reg_ region: 717, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_718.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.483s.\n",
      "-- designing region: 13:16240000-16260000_:+_reg_ region: 718, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_719.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.484s.\n",
      "-- designing region: 13:18740000-18760000_:+_reg_ region: 719, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_720.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.477s.\n",
      "-- designing region: 13:21240000-21260000_:+_reg_ region: 720, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_721.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.475s.\n",
      "-- designing region: 13:23740000-23760000_:+_reg_ region: 721, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_722.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.302s.\n",
      "-- designing region: 13:26240000-26260000_:+_reg_ region: 722, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_723.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.503s.\n",
      "-- designing region: 13:28740000-28760000_:+_reg_ region: 723, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_724.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.256s.\n",
      "in 11.482s.\n",
      "-- designing region: 13:31240000-31260000_:+_reg_ region: 724, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_725.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.250s.\n",
      "in 11.498s.\n",
      "-- designing region: 13:33740000-33760000_:+_reg_ region: 725, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_726.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.248s.\n",
      "in 11.485s.\n",
      "-- designing region: 13:36240000-36260000_:+_reg_ region: 726, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_727.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.157s.\n",
      "in 11.403s.\n",
      "-- designing region: 13:38740000-38760000_:+_reg_ region: 727, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_728.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.505s.\n",
      "-- designing region: 13:41240000-41260000_:+_reg_ region: 728, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_729.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.471s.\n",
      "-- designing region: 13:43740000-43760000_:+_reg_ region: 729, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_730.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.250s.\n",
      "in 11.471s.\n",
      "-- designing region: 13:46240000-46260000_:+_reg_ region: 730, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_731.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.495s.\n",
      "-- designing region: 13:48740000-48760000_:+_reg_ region: 731, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_732.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.482s.\n",
      "-- designing region: 13:51240000-51260000_:+_reg_ region: 732, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_733.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.473s.\n",
      "-- designing region: 13:53740000-53760000_:+_reg_ region: 733, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_734.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.470s.\n",
      "-- designing region: 13:56240000-56260000_:+_reg_ region: 734, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_735.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.496s.\n",
      "-- designing region: 13:58740000-58760000_:+_reg_ region: 735, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_736.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.478s.\n",
      "-- designing region: 13:61240000-61260000_:+_reg_ region: 736, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_737.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.471s.\n",
      "-- designing region: 13:63740000-63760000_:+_reg_ region: 737, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_738.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 25.272s.\n",
      "-- designing region: 13:66240000-66260000_:+_reg_ region: 738, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_739.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.456s.\n",
      "-- designing region: 13:68740000-68760000_:+_reg_ region: 739, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_740.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.417s.\n",
      "-- designing region: 13:71240000-71260000_:+_reg_ region: 740, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_741.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.182s.\n",
      "in 11.418s.\n",
      "-- designing region: 13:73740000-73760000_:+_reg_ region: 741, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_742.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.315s.\n",
      "-- designing region: 13:76240000-76260000_:+_reg_ region: 742, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_743.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.319s.\n",
      "-- designing region: 13:78740000-78760000_:+_reg_ region: 743, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_744.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.379s.\n",
      "-- designing region: 13:81240000-81260000_:+_reg_ region: 744, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_745.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.351s.\n",
      "-- designing region: 13:83740000-83760000_:+_reg_ region: 745, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_746.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.280s.\n",
      "-- designing region: 13:86240000-86260000_:+_reg_ region: 746, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_747.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.153s.\n",
      "in 11.386s.\n",
      "-- designing region: 13:88740000-88760000_:+_reg_ region: 747, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_748.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.424s.\n",
      "-- designing region: 13:91240000-91260000_:+_reg_ region: 748, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_749.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.248s.\n",
      "in 11.499s.\n",
      "-- designing region: 13:93740000-93760000_:+_reg_ region: 749, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_750.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.384s.\n",
      "-- designing region: 13:96240000-96260000_:+_reg_ region: 750, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_751.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.181s.\n",
      "in 11.288s.\n",
      "-- designing region: 13:98740000-98760000_:+_reg_ region: 751, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_752.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.397s.\n",
      "-- designing region: 13:101240000-101260000_:+_reg_ region: 752, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_753.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.291s.\n",
      "-- designing region: 13:103740000-103760000_:+_reg_ region: 753, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_754.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.171s.\n",
      "in 11.278s.\n",
      "-- designing region: 13:106240000-106260000_:+_reg_ region: 754, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_755.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.415s.\n",
      "-- designing region: 13:108740000-108760000_:+_reg_ region: 755, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_756.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.355s.\n",
      "-- designing region: 13:111240000-111260000_:+_reg_ region: 756, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_757.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 11.271s.\n",
      "-- designing region: 13:113740000-113760000_:+_reg_ region: 757, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_758.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 11.370s.\n",
      "-- designing region: 13:116240000-116260000_:+_reg_ region: 758, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_759.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.453s.\n",
      "-- designing region: 13:118740000-118760000_:+_reg_ region: 759, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_760.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.441s.\n",
      "-- designing region: 13:121240000-120421639_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: 14:1240000-1260000_:+_reg_ region: 761, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_762.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 0.412s.\n",
      "-- designing region: 14:3740000-3760000_:+_reg_ region: 762, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_763.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.478s.\n",
      "in 11.694s.\n",
      "-- designing region: 14:6240000-6260000_:+_reg_ region: 763, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_764.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.346s.\n",
      "-- designing region: 14:8740000-8760000_:+_reg_ region: 764, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_765.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.330s.\n",
      "-- designing region: 14:11240000-11260000_:+_reg_ region: 765, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_766.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.464s.\n",
      "-- designing region: 14:13740000-13760000_:+_reg_ region: 766, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_767.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.296s.\n",
      "-- designing region: 14:16240000-16260000_:+_reg_ region: 767, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_768.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.466s.\n",
      "-- designing region: 14:18740000-18760000_:+_reg_ region: 768, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_769.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.370s.\n",
      "-- designing region: 14:21240000-21260000_:+_reg_ region: 769, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_770.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.211s.\n",
      "in 11.311s.\n",
      "-- designing region: 14:23740000-23760000_:+_reg_ region: 770, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_771.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.423s.\n",
      "-- designing region: 14:26240000-26260000_:+_reg_ region: 771, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_772.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.352s.\n",
      "-- designing region: 14:28740000-28760000_:+_reg_ region: 772, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_773.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.305s.\n",
      "-- designing region: 14:31240000-31260000_:+_reg_ region: 773, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_774.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 11.489s.\n",
      "-- designing region: 14:33740000-33760000_:+_reg_ region: 774, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_775.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 15.867s.\n",
      "-- designing region: 14:36240000-36260000_:+_reg_ region: 775, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_776.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 14.841s.\n",
      "-- designing region: 14:38740000-38760000_:+_reg_ region: 776, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_777.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.406s.\n",
      "in 11.485s.\n",
      "-- designing region: 14:41240000-41260000_:+_reg_ region: 777, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_778.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.156s.\n",
      "in 11.280s.\n",
      "-- designing region: 14:43740000-43760000_:+_reg_ region: 778, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_779.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.429s.\n",
      "-- designing region: 14:46240000-46260000_:+_reg_ region: 779, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_780.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.349s.\n",
      "-- designing region: 14:48740000-48760000_:+_reg_ region: 780, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_781.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.350s.\n",
      "-- designing region: 14:51240000-51260000_:+_reg_ region: 781, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_782.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.521s.\n",
      "-- designing region: 14:53740000-53760000_:+_reg_ region: 782, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_783.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.208s.\n",
      "in 11.375s.\n",
      "-- designing region: 14:56240000-56260000_:+_reg_ region: 783, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_784.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.498s.\n",
      "-- designing region: 14:58740000-58760000_:+_reg_ region: 784, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_785.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.429s.\n",
      "-- designing region: 14:61240000-61260000_:+_reg_ region: 785, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_786.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 11.435s.\n",
      "-- designing region: 14:63740000-63760000_:+_reg_ region: 786, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_787.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.393s.\n",
      "-- designing region: 14:66240000-66260000_:+_reg_ region: 787, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_788.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.315s.\n",
      "-- designing region: 14:68740000-68760000_:+_reg_ region: 788, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_789.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.500s.\n",
      "-- designing region: 14:71240000-71260000_:+_reg_ region: 789, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_790.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.221s.\n",
      "in 11.404s.\n",
      "-- designing region: 14:73740000-73760000_:+_reg_ region: 790, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_791.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.430s.\n",
      "-- designing region: 14:76240000-76260000_:+_reg_ region: 791, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_792.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.361s.\n",
      "-- designing region: 14:78740000-78760000_:+_reg_ region: 792, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_793.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.248s.\n",
      "in 11.522s.\n",
      "-- designing region: 14:81240000-81260000_:+_reg_ region: 793, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_794.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.252s.\n",
      "in 11.381s.\n",
      "-- designing region: 14:83740000-83760000_:+_reg_ region: 794, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_795.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.280s.\n",
      "-- designing region: 14:86240000-86260000_:+_reg_ region: 795, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_796.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.156s.\n",
      "in 11.423s.\n",
      "-- designing region: 14:88740000-88760000_:+_reg_ region: 796, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_797.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 11.350s.\n",
      "-- designing region: 14:91240000-91260000_:+_reg_ region: 797, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_798.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.315s.\n",
      "-- designing region: 14:93740000-93760000_:+_reg_ region: 798, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_799.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.443s.\n",
      "-- designing region: 14:96240000-96260000_:+_reg_ region: 799, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_800.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.302s.\n",
      "-- designing region: 14:98740000-98760000_:+_reg_ region: 800, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_801.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.494s.\n",
      "-- designing region: 14:101240000-101260000_:+_reg_ region: 801, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_802.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.252s.\n",
      "in 11.412s.\n",
      "-- designing region: 14:103740000-103760000_:+_reg_ region: 802, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_803.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.311s.\n",
      "-- designing region: 14:106240000-106260000_:+_reg_ region: 803, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_804.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.485s.\n",
      "-- designing region: 14:108740000-108760000_:+_reg_ region: 804, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_805.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.305s.\n",
      "-- designing region: 14:111240000-111260000_:+_reg_ region: 805, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_806.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.451s.\n",
      "-- designing region: 14:113740000-113760000_:+_reg_ region: 806, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_807.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.387s.\n",
      "-- designing region: 14:116240000-116260000_:+_reg_ region: 807, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_808.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.315s.\n",
      "-- designing region: 14:118740000-118760000_:+_reg_ region: 808, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_809.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.169s.\n",
      "in 11.437s.\n",
      "-- designing region: 14:121240000-121260000_:+_reg_ region: 809, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_810.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.258s.\n",
      "in 11.400s.\n",
      "-- designing region: 14:123740000-123760000_:+_reg_ region: 810, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_811.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.502s.\n",
      "-- designing region: 15:1240000-1260000_:+_reg_ region: 811, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_812.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.197s.\n",
      "in 0.426s.\n",
      "-- designing region: 15:3740000-3760000_:+_reg_ region: 812, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_813.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.475s.\n",
      "in 11.732s.\n",
      "-- designing region: 15:6240000-6260000_:+_reg_ region: 813, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_814.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.297s.\n",
      "-- designing region: 15:8740000-8760000_:+_reg_ region: 814, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_815.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.309s.\n",
      "-- designing region: 15:11240000-11260000_:+_reg_ region: 815, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_816.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.451s.\n",
      "-- designing region: 15:13740000-13760000_:+_reg_ region: 816, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_817.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.401s.\n",
      "-- designing region: 15:16240000-16260000_:+_reg_ region: 817, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_818.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.431s.\n",
      "-- designing region: 15:18740000-18760000_:+_reg_ region: 818, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_819.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.362s.\n",
      "-- designing region: 15:21240000-21260000_:+_reg_ region: 819, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_820.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.171s.\n",
      "in 11.264s.\n",
      "-- designing region: 15:23740000-23760000_:+_reg_ region: 820, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_821.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.300s.\n",
      "-- designing region: 15:26240000-26260000_:+_reg_ region: 821, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_822.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.368s.\n",
      "-- designing region: 15:28740000-28760000_:+_reg_ region: 822, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_823.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.178s.\n",
      "in 11.292s.\n",
      "-- designing region: 15:31240000-31260000_:+_reg_ region: 823, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_824.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.423s.\n",
      "-- designing region: 15:33740000-33760000_:+_reg_ region: 824, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_825.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.366s.\n",
      "-- designing region: 15:36240000-36260000_:+_reg_ region: 825, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_826.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.318s.\n",
      "-- designing region: 15:38740000-38760000_:+_reg_ region: 826, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_827.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.192s.\n",
      "in 0.415s.\n",
      "-- designing region: 15:41240000-41260000_:+_reg_ region: 827, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_828.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.465s.\n",
      "in 11.568s.\n",
      "-- designing region: 15:43740000-43760000_:+_reg_ region: 828, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_829.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.462s.\n",
      "-- designing region: 15:46240000-46260000_:+_reg_ region: 829, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_830.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.386s.\n",
      "-- designing region: 15:48740000-48760000_:+_reg_ region: 830, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_831.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.212s.\n",
      "in 11.418s.\n",
      "-- designing region: 15:51240000-51260000_:+_reg_ region: 831, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_832.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.375s.\n",
      "-- designing region: 15:53740000-53760000_:+_reg_ region: 832, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_833.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.290s.\n",
      "-- designing region: 15:56240000-56260000_:+_reg_ region: 833, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_834.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.418s.\n",
      "-- designing region: 15:58740000-58760000_:+_reg_ region: 834, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_835.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.352s.\n",
      "-- designing region: 15:61240000-61260000_:+_reg_ region: 835, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_836.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.286s.\n",
      "-- designing region: 15:63740000-63760000_:+_reg_ region: 836, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_837.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.497s.\n",
      "-- designing region: 15:66240000-66260000_:+_reg_ region: 837, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_838.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.395s.\n",
      "-- designing region: 15:68740000-68760000_:+_reg_ region: 838, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_839.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.185s.\n",
      "in 11.421s.\n",
      "-- designing region: 15:71240000-71260000_:+_reg_ region: 839, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_840.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 11.344s.\n",
      "-- designing region: 15:73740000-73760000_:+_reg_ region: 840, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_841.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.304s.\n",
      "-- designing region: 15:76240000-76260000_:+_reg_ region: 841, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_842.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.487s.\n",
      "-- designing region: 15:78740000-78760000_:+_reg_ region: 842, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_843.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.300s.\n",
      "-- designing region: 15:81240000-81260000_:+_reg_ region: 843, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_844.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.452s.\n",
      "-- designing region: 15:83740000-83760000_:+_reg_ region: 844, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_845.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.386s.\n",
      "-- designing region: 15:86240000-86260000_:+_reg_ region: 845, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_846.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.342s.\n",
      "-- designing region: 15:88740000-88760000_:+_reg_ region: 846, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_847.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.156s.\n",
      "in 11.382s.\n",
      "-- designing region: 15:91240000-91260000_:+_reg_ region: 847, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_848.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.462s.\n",
      "-- designing region: 15:93740000-93760000_:+_reg_ region: 848, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_849.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.482s.\n",
      "-- designing region: 15:96240000-96260000_:+_reg_ region: 849, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_850.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.370s.\n",
      "-- designing region: 15:98740000-98760000_:+_reg_ region: 850, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_851.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.343s.\n",
      "-- designing region: 15:101240000-101260000_:+_reg_ region: 851, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_852.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.247s.\n",
      "in 11.483s.\n",
      "-- designing region: 15:103740000-103760000_:+_reg_ region: 852, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_853.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.387s.\n",
      "-- designing region: 16:1240000-1260000_:+_reg_ region: 853, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_854.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 0.390s.\n",
      "-- designing region: 16:3740000-3760000_:+_reg_ region: 854, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_855.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.551s.\n",
      "in 11.642s.\n",
      "-- designing region: 16:6240000-6260000_:+_reg_ region: 855, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_856.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 11.432s.\n",
      "-- designing region: 16:8740000-8760000_:+_reg_ region: 856, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_857.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.382s.\n",
      "-- designing region: 16:11240000-11260000_:+_reg_ region: 857, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_858.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.151s.\n",
      "in 11.309s.\n",
      "-- designing region: 16:13740000-13760000_:+_reg_ region: 858, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_859.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.469s.\n",
      "-- designing region: 16:16240000-16260000_:+_reg_ region: 859, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_860.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 16.799s.\n",
      "-- designing region: 16:18740000-18760000_:+_reg_ region: 860, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_861.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.370s.\n",
      "-- designing region: 16:21240000-21260000_:+_reg_ region: 861, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_862.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.161s.\n",
      "in 11.276s.\n",
      "-- designing region: 16:23740000-23760000_:+_reg_ region: 862, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_863.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.373s.\n",
      "-- designing region: 16:26240000-26260000_:+_reg_ region: 863, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_864.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.337s.\n",
      "-- designing region: 16:28740000-28760000_:+_reg_ region: 864, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_865.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.326s.\n",
      "-- designing region: 16:31240000-31260000_:+_reg_ region: 865, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_866.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.448s.\n",
      "-- designing region: 16:33740000-33760000_:+_reg_ region: 866, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_867.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.249s.\n",
      "in 11.327s.\n",
      "-- designing region: 16:36240000-36260000_:+_reg_ region: 867, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_868.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.244s.\n",
      "-- designing region: 16:38740000-38760000_:+_reg_ region: 868, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_869.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.306s.\n",
      "-- designing region: 16:41240000-41260000_:+_reg_ region: 869, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_870.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.325s.\n",
      "-- designing region: 16:43740000-43760000_:+_reg_ region: 870, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_871.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.258s.\n",
      "-- designing region: 16:46240000-46260000_:+_reg_ region: 871, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_872.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.367s.\n",
      "-- designing region: 16:48740000-48760000_:+_reg_ region: 872, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_873.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.256s.\n",
      "-- designing region: 16:51240000-51260000_:+_reg_ region: 873, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_874.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.362s.\n",
      "-- designing region: 16:53740000-53760000_:+_reg_ region: 874, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_875.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.366s.\n",
      "-- designing region: 16:56240000-56260000_:+_reg_ region: 875, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_876.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.335s.\n",
      "-- designing region: 16:58740000-58760000_:+_reg_ region: 876, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_877.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.200s.\n",
      "in 11.296s.\n",
      "-- designing region: 16:61240000-61260000_:+_reg_ region: 877, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_878.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.196s.\n",
      "in 11.372s.\n",
      "-- designing region: 16:63740000-63760000_:+_reg_ region: 878, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_879.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.314s.\n",
      "-- designing region: 16:66240000-66260000_:+_reg_ region: 879, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_880.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.221s.\n",
      "in 11.337s.\n",
      "-- designing region: 16:68740000-68760000_:+_reg_ region: 880, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_881.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.174s.\n",
      "in 11.383s.\n",
      "-- designing region: 16:71240000-71260000_:+_reg_ region: 881, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_882.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.260s.\n",
      "-- designing region: 16:73740000-73760000_:+_reg_ region: 882, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_883.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.221s.\n",
      "in 11.312s.\n",
      "-- designing region: 16:76240000-76260000_:+_reg_ region: 883, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_884.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.204s.\n",
      "in 11.396s.\n",
      "-- designing region: 16:78740000-78760000_:+_reg_ region: 884, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_885.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.356s.\n",
      "-- designing region: 16:81240000-81260000_:+_reg_ region: 885, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_886.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.196s.\n",
      "in 11.267s.\n",
      "-- designing region: 16:83740000-83760000_:+_reg_ region: 886, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_887.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 11.351s.\n",
      "-- designing region: 16:86240000-86260000_:+_reg_ region: 887, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_888.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.321s.\n",
      "-- designing region: 16:88740000-88760000_:+_reg_ region: 888, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_889.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.224s.\n",
      "in 11.278s.\n",
      "-- designing region: 16:91240000-91260000_:+_reg_ region: 889, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_890.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.216s.\n",
      "in 11.426s.\n",
      "-- designing region: 16:93740000-93760000_:+_reg_ region: 890, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_891.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.359s.\n",
      "-- designing region: 16:96240000-96260000_:+_reg_ region: 891, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_892.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.193s.\n",
      "in 11.261s.\n",
      "-- designing region: 16:98740000-98207768_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: 17:1240000-1260000_:+_reg_ region: 893, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_894.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.151s.\n",
      "in 0.372s.\n",
      "-- designing region: 17:3740000-3760000_:+_reg_ region: 894, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_895.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.554s.\n",
      "in 11.628s.\n",
      "-- designing region: 17:6240000-6260000_:+_reg_ region: 895, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_896.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.399s.\n",
      "-- designing region: 17:8740000-8760000_:+_reg_ region: 896, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_897.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.377s.\n",
      "-- designing region: 17:11240000-11260000_:+_reg_ region: 897, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_898.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.171s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.308s.\n",
      "-- designing region: 17:13740000-13760000_:+_reg_ region: 898, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_899.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.528s.\n",
      "-- designing region: 17:16240000-16260000_:+_reg_ region: 899, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_900.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.393s.\n",
      "-- designing region: 17:18740000-18760000_:+_reg_ region: 900, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_901.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.439s.\n",
      "-- designing region: 17:21240000-21260000_:+_reg_ region: 901, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_902.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.354s.\n",
      "-- designing region: 17:23740000-23760000_:+_reg_ region: 902, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_903.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.154s.\n",
      "in 11.309s.\n",
      "-- designing region: 17:26240000-26260000_:+_reg_ region: 903, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_904.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.248s.\n",
      "in 11.520s.\n",
      "-- designing region: 17:28740000-28760000_:+_reg_ region: 904, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_905.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.442s.\n",
      "-- designing region: 17:31240000-31260000_:+_reg_ region: 905, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_906.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.158s.\n",
      "in 11.415s.\n",
      "-- designing region: 17:33740000-33760000_:+_reg_ region: 906, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_907.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.392s.\n",
      "-- designing region: 17:36240000-36260000_:+_reg_ region: 907, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_908.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.480s.\n",
      "-- designing region: 17:38740000-38760000_:+_reg_ region: 908, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_909.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.382s.\n",
      "-- designing region: 17:41240000-41260000_:+_reg_ region: 909, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_910.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.194s.\n",
      "in 11.307s.\n",
      "-- designing region: 17:43740000-43760000_:+_reg_ region: 910, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_911.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 11.398s.\n",
      "-- designing region: 17:46240000-46260000_:+_reg_ region: 911, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_912.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.413s.\n",
      "-- designing region: 17:48740000-48760000_:+_reg_ region: 912, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_913.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.309s.\n",
      "-- designing region: 17:51240000-51260000_:+_reg_ region: 913, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_914.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.254s.\n",
      "in 11.341s.\n",
      "-- designing region: 17:53740000-53760000_:+_reg_ region: 914, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_915.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.361s.\n",
      "-- designing region: 17:56240000-56260000_:+_reg_ region: 915, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_916.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.496s.\n",
      "-- designing region: 17:58740000-58760000_:+_reg_ region: 916, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_917.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.227s.\n",
      "in 28.523s.\n",
      "-- designing region: 17:61240000-61260000_:+_reg_ region: 917, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_918.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.460s.\n",
      "-- designing region: 17:63740000-63760000_:+_reg_ region: 918, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_919.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.308s.\n",
      "-- designing region: 17:66240000-66260000_:+_reg_ region: 919, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_920.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.380s.\n",
      "-- designing region: 17:68740000-68760000_:+_reg_ region: 920, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_921.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.281s.\n",
      "-- designing region: 17:71240000-71260000_:+_reg_ region: 921, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_922.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.400s.\n",
      "-- designing region: 17:73740000-73760000_:+_reg_ region: 922, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_923.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.321s.\n",
      "-- designing region: 17:76240000-76260000_:+_reg_ region: 923, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_924.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n",
      "in 11.386s.\n",
      "-- designing region: 17:78740000-78760000_:+_reg_ region: 924, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_925.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.330s.\n",
      "-- designing region: 17:81240000-81260000_:+_reg_ region: 925, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_926.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.174s.\n",
      "in 11.288s.\n",
      "-- designing region: 17:83740000-83760000_:+_reg_ region: 926, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_927.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.393s.\n",
      "-- designing region: 17:86240000-86260000_:+_reg_ region: 927, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_928.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.362s.\n",
      "-- designing region: 17:88740000-88760000_:+_reg_ region: 928, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_929.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 11.262s.\n",
      "-- designing region: 17:91240000-91260000_:+_reg_ region: 929, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_930.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.453s.\n",
      "-- designing region: 17:93740000-93760000_:+_reg_ region: 930, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_931.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.425s.\n",
      "-- designing region: 18:1240000-1260000_:+_reg_ region: 931, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_932.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 0.394s.\n",
      "-- designing region: 18:3740000-3760000_:+_reg_ region: 932, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_933.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.559s.\n",
      "in 11.658s.\n",
      "-- designing region: 18:6240000-6260000_:+_reg_ region: 933, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_934.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.408s.\n",
      "-- designing region: 18:8740000-8760000_:+_reg_ region: 934, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_935.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.352s.\n",
      "-- designing region: 18:11240000-11260000_:+_reg_ region: 935, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_936.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 11.244s.\n",
      "-- designing region: 18:13740000-13760000_:+_reg_ region: 936, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_937.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.180s.\n",
      "in 11.419s.\n",
      "-- designing region: 18:16240000-16260000_:+_reg_ region: 937, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_938.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.354s.\n",
      "-- designing region: 18:18740000-18760000_:+_reg_ region: 938, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_939.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.146s.\n",
      "in 11.253s.\n",
      "-- designing region: 18:21240000-21260000_:+_reg_ region: 939, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_940.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.171s.\n",
      "in 11.397s.\n",
      "-- designing region: 18:23740000-23760000_:+_reg_ region: 940, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_941.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.341s.\n",
      "-- designing region: 18:26240000-26260000_:+_reg_ region: 941, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_942.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.155s.\n",
      "in 11.211s.\n",
      "-- designing region: 18:28740000-28760000_:+_reg_ region: 942, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_943.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.191s.\n",
      "in 11.460s.\n",
      "-- designing region: 18:31240000-31260000_:+_reg_ region: 943, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_944.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.325s.\n",
      "-- designing region: 18:33740000-33760000_:+_reg_ region: 944, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_945.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.158s.\n",
      "in 11.257s.\n",
      "-- designing region: 18:36240000-36260000_:+_reg_ region: 945, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_946.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n",
      "in 11.381s.\n",
      "-- designing region: 18:38740000-38760000_:+_reg_ region: 946, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_947.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.356s.\n",
      "-- designing region: 18:41240000-41260000_:+_reg_ region: 947, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_948.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.256s.\n",
      "-- designing region: 18:43740000-43760000_:+_reg_ region: 948, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_949.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.166s.\n",
      "in 11.358s.\n",
      "-- designing region: 18:46240000-46260000_:+_reg_ region: 949, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_950.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.337s.\n",
      "-- designing region: 18:48740000-48760000_:+_reg_ region: 950, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_951.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.182s.\n",
      "in 11.276s.\n",
      "-- designing region: 18:51240000-51260000_:+_reg_ region: 951, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_952.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.180s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.386s.\n",
      "-- designing region: 18:53740000-53760000_:+_reg_ region: 952, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_953.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.254s.\n",
      "in 11.320s.\n",
      "-- designing region: 18:56240000-56260000_:+_reg_ region: 953, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_954.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.198s.\n",
      "in 11.347s.\n",
      "-- designing region: 18:58740000-58760000_:+_reg_ region: 954, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_955.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.456s.\n",
      "-- designing region: 18:61240000-61260000_:+_reg_ region: 955, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_956.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.312s.\n",
      "-- designing region: 18:63740000-63760000_:+_reg_ region: 956, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_957.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.246s.\n",
      "in 11.240s.\n",
      "-- designing region: 18:66240000-66260000_:+_reg_ region: 957, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_958.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.284s.\n",
      "-- designing region: 18:68740000-68760000_:+_reg_ region: 958, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_959.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.387s.\n",
      "-- designing region: 18:71240000-71260000_:+_reg_ region: 959, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_960.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.173s.\n",
      "in 11.277s.\n",
      "-- designing region: 18:73740000-73760000_:+_reg_ region: 960, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_961.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.173s.\n",
      "in 11.409s.\n",
      "-- designing region: 18:76240000-76260000_:+_reg_ region: 961, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_962.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.337s.\n",
      "-- designing region: 18:78740000-78760000_:+_reg_ region: 962, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_963.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.154s.\n",
      "in 11.254s.\n",
      "-- designing region: 18:81240000-81260000_:+_reg_ region: 963, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_964.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.168s.\n",
      "in 11.412s.\n",
      "-- designing region: 18:83740000-83760000_:+_reg_ region: 964, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_965.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.307s.\n",
      "-- designing region: 18:86240000-86260000_:+_reg_ region: 965, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_966.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.180s.\n",
      "in 11.282s.\n",
      "-- designing region: 18:88740000-88760000_:+_reg_ region: 966, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_967.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.167s.\n",
      "in 11.398s.\n",
      "-- designing region: 18:91240000-90702639_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: 19:1240000-1260000_:+_reg_ region: 968, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_969.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 0.412s.\n",
      "-- designing region: 19:3740000-3760000_:+_reg_ region: 969, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_970.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.455s.\n",
      "in 11.614s.\n",
      "-- designing region: 19:6240000-6260000_:+_reg_ region: 970, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_971.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.353s.\n",
      "-- designing region: 19:8740000-8760000_:+_reg_ region: 971, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_972.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.296s.\n",
      "-- designing region: 19:11240000-11260000_:+_reg_ region: 972, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_973.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.238s.\n",
      "in 11.460s.\n",
      "-- designing region: 19:13740000-13760000_:+_reg_ region: 973, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_974.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.269s.\n",
      "-- designing region: 19:16240000-16260000_:+_reg_ region: 974, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_975.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.202s.\n",
      "-- designing region: 19:18740000-18760000_:+_reg_ region: 975, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_976.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.169s.\n",
      "in 11.429s.\n",
      "-- designing region: 19:21240000-21260000_:+_reg_ region: 976, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_977.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 17.057s.\n",
      "-- designing region: 19:23740000-23760000_:+_reg_ region: 977, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_978.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.242s.\n",
      "-- designing region: 19:26240000-26260000_:+_reg_ region: 978, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_979.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.251s.\n",
      "-- designing region: 19:28740000-28760000_:+_reg_ region: 979, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_980.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.163s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.365s.\n",
      "-- designing region: 19:31240000-31260000_:+_reg_ region: 980, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_981.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.316s.\n",
      "-- designing region: 19:33740000-33760000_:+_reg_ region: 981, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_982.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.186s.\n",
      "in 11.288s.\n",
      "-- designing region: 19:36240000-36260000_:+_reg_ region: 982, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_983.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.378s.\n",
      "-- designing region: 19:38740000-38760000_:+_reg_ region: 983, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_984.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.329s.\n",
      "-- designing region: 19:41240000-41260000_:+_reg_ region: 984, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_985.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.283s.\n",
      "-- designing region: 19:43740000-43760000_:+_reg_ region: 985, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_986.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.179s.\n",
      "in 11.375s.\n",
      "-- designing region: 19:46240000-46260000_:+_reg_ region: 986, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_987.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.349s.\n",
      "-- designing region: 19:48740000-48760000_:+_reg_ region: 987, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_988.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.282s.\n",
      "-- designing region: 19:51240000-51260000_:+_reg_ region: 988, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_989.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.172s.\n",
      "in 11.349s.\n",
      "-- designing region: 19:53740000-53760000_:+_reg_ region: 989, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_990.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.337s.\n",
      "-- designing region: 19:56240000-56260000_:+_reg_ region: 990, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_991.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.209s.\n",
      "in 11.322s.\n",
      "-- designing region: 19:58740000-58760000_:+_reg_ region: 991, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_992.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.176s.\n",
      "in 11.372s.\n",
      "-- designing region: 19:61240000-61260000_:+_reg_ region: 992, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_993.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.376s.\n",
      "-- designing region: X:1240000-1260000_:+_reg_ region: 993, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_994.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.140s.\n",
      "in 0.361s.\n",
      "-- designing region: X:3740000-3760000_:+_reg_ region: 994, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_995.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.588s.\n",
      "in 11.727s.\n",
      "-- designing region: X:6240000-6260000_:+_reg_ region: 995, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_996.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.445s.\n",
      "-- designing region: X:8740000-8760000_:+_reg_ region: 996, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_997.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.404s.\n",
      "-- designing region: X:11240000-11260000_:+_reg_ region: 997, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_998.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.206s.\n",
      "in 11.309s.\n",
      "-- designing region: X:13740000-13760000_:+_reg_ region: 998, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_999.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.170s.\n",
      "in 11.353s.\n",
      "-- designing region: X:16240000-16260000_:+_reg_ region: 999, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1000.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.339s.\n",
      "-- designing region: X:18740000-18760000_:+_reg_ region: 1000, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1001.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.187s.\n",
      "in 11.258s.\n",
      "-- designing region: X:21240000-21260000_:+_reg_ region: 1001, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1002.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.192s.\n",
      "in 11.498s.\n",
      "-- designing region: X:23740000-23760000_:+_reg_ region: 1002, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1003.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.248s.\n",
      "-- designing region: X:26240000-26260000_:+_reg_ region: 1003, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1004.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.240s.\n",
      "-- designing region: X:28740000-28760000_:+_reg_ region: 1004, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1005.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.202s.\n",
      "in 11.395s.\n",
      "-- designing region: X:31240000-31260000_:+_reg_ region: 1005, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1006.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.318s.\n",
      "-- designing region: X:33740000-33760000_:+_reg_ region: 1006, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1007.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.215s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 11.315s.\n",
      "-- designing region: X:36240000-36260000_:+_reg_ region: 1007, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1008.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.183s.\n",
      "in 11.355s.\n",
      "-- designing region: X:38740000-38760000_:+_reg_ region: 1008, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1009.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.241s.\n",
      "in 11.457s.\n",
      "-- designing region: X:41240000-41260000_:+_reg_ region: 1009, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1010.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.162s.\n",
      "in 11.207s.\n",
      "-- designing region: X:43740000-43760000_:+_reg_ region: 1010, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1011.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 11.344s.\n",
      "-- designing region: X:46240000-46260000_:+_reg_ region: 1011, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1012.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.320s.\n",
      "-- designing region: X:48740000-48760000_:+_reg_ region: 1012, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1013.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.208s.\n",
      "in 11.299s.\n",
      "-- designing region: X:51240000-51260000_:+_reg_ region: 1013, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1014.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.197s.\n",
      "in 11.328s.\n",
      "-- designing region: X:53740000-53760000_:+_reg_ region: 1014, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1015.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.365s.\n",
      "-- designing region: X:56240000-56260000_:+_reg_ region: 1015, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1016.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.217s.\n",
      "in 11.295s.\n",
      "-- designing region: X:58740000-58760000_:+_reg_ region: 1016, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1017.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.199s.\n",
      "in 11.376s.\n",
      "-- designing region: X:61240000-61260000_:+_reg_ region: 1017, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1018.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.228s.\n",
      "in 11.362s.\n",
      "-- designing region: X:63740000-63760000_:+_reg_ region: 1018, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1019.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.188s.\n",
      "in 11.264s.\n",
      "-- designing region: X:66240000-66260000_:+_reg_ region: 1019, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1020.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.201s.\n",
      "in 11.267s.\n",
      "-- designing region: X:68740000-68760000_:+_reg_ region: 1020, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1021.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.389s.\n",
      "-- designing region: X:71240000-71260000_:+_reg_ region: 1021, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1022.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.314s.\n",
      "-- designing region: X:73740000-73760000_:+_reg_ region: 1022, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1023.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.212s.\n",
      "in 11.367s.\n",
      "-- designing region: X:76240000-76260000_:+_reg_ region: 1023, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1024.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.353s.\n",
      "-- designing region: X:78740000-78760000_:+_reg_ region: 1024, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1025.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.310s.\n",
      "-- designing region: X:81240000-81260000_:+_reg_ region: 1025, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1026.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 11.389s.\n",
      "-- designing region: X:83740000-83760000_:+_reg_ region: 1026, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1027.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.244s.\n",
      "in 11.312s.\n",
      "-- designing region: X:86240000-86260000_:+_reg_ region: 1027, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1028.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.212s.\n",
      "in 11.297s.\n",
      "-- designing region: X:88740000-88760000_:+_reg_ region: 1028, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1029.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.193s.\n",
      "in 11.345s.\n",
      "-- designing region: X:91240000-91260000_:+_reg_ region: 1029, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1030.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.209s.\n",
      "-- designing region: X:93740000-93760000_:+_reg_ region: 1030, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1031.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.271s.\n",
      "-- designing region: X:96240000-96260000_:+_reg_ region: 1031, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1032.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.287s.\n",
      "-- designing region: X:98740000-98760000_:+_reg_ region: 1032, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1033.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.468s.\n",
      "-- designing region: X:101240000-101260000_:+_reg_ region: 1033, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1034.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.214s.\n",
      "-- designing region: X:103740000-103760000_:+_reg_ region: 1034, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1035.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.150s.\n",
      "in 11.199s.\n",
      "-- designing region: X:106240000-106260000_:+_reg_ region: 1035, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1036.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 11.366s.\n",
      "-- designing region: X:108740000-108760000_:+_reg_ region: 1036, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1037.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.308s.\n",
      "-- designing region: X:111240000-111260000_:+_reg_ region: 1037, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1038.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.190s.\n",
      "in 0.410s.\n",
      "-- designing region: X:113740000-113760000_:+_reg_ region: 1038, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1039.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.506s.\n",
      "in 11.632s.\n",
      "-- designing region: X:116240000-116260000_:+_reg_ region: 1039, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1040.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.204s.\n",
      "in 11.279s.\n",
      "-- designing region: X:118740000-118760000_:+_reg_ region: 1040, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1041.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.195s.\n",
      "in 11.360s.\n",
      "-- designing region: X:121240000-121260000_:+_reg_ region: 1041, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1042.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.331s.\n",
      "-- designing region: X:123740000-123760000_:+_reg_ region: 1042, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1043.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.213s.\n",
      "in 11.291s.\n",
      "-- designing region: X:126240000-126260000_:+_reg_ region: 1043, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1044.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.197s.\n",
      "in 11.191s.\n",
      "-- designing region: X:128740000-128760000_:+_reg_ region: 1044, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1045.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.400s.\n",
      "-- designing region: X:131240000-131260000_:+_reg_ region: 1045, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1046.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.226s.\n",
      "-- designing region: X:133740000-133760000_:+_reg_ region: 1046, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1047.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.148s.\n",
      "in 11.171s.\n",
      "-- designing region: X:136240000-136260000_:+_reg_ region: 1047, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1048.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.208s.\n",
      "in 11.371s.\n",
      "-- designing region: X:138740000-138760000_:+_reg_ region: 1048, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1049.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.337s.\n",
      "-- designing region: X:141240000-141260000_:+_reg_ region: 1049, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1050.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.206s.\n",
      "in 11.314s.\n",
      "-- designing region: X:143740000-143760000_:+_reg_ region: 1050, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1051.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.194s.\n",
      "in 11.363s.\n",
      "-- designing region: X:146240000-146260000_:+_reg_ region: 1051, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1052.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.339s.\n",
      "-- designing region: X:148740000-148760000_:+_reg_ region: 1052, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1053.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.223s.\n",
      "in 11.310s.\n",
      "-- designing region: X:151240000-151260000_:+_reg_ region: 1053, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1054.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.196s.\n",
      "in 11.398s.\n",
      "-- designing region: X:153740000-153760000_:+_reg_ region: 1054, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1055.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.237s.\n",
      "in 11.360s.\n",
      "-- designing region: X:156240000-156260000_:+_reg_ region: 1055, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1056.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.177s.\n",
      "in 11.278s.\n",
      "-- designing region: X:158740000-158760000_:+_reg_ region: 1056, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1057.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.167s.\n",
      "in 11.400s.\n",
      "-- designing region: X:161240000-161260000_:+_reg_ region: 1057, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1058.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.229s.\n",
      "in 11.330s.\n",
      "-- designing region: X:163740000-163760000_:+_reg_ region: 1058, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1059.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.156s.\n",
      "in 11.292s.\n",
      "-- designing region: X:166240000-166260000_:+_reg_ region: 1059, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1060.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.161s.\n",
      "in 11.466s.\n",
      "-- designing region: X:168740000-168760000_:+_reg_ region: 1060, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1061.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.358s.\n",
      "-- designing region: X:171240000-171031299_:+_reg_ Too short the sequence, skip.\n",
      "-- designing region: Y:1240000-1260000_:+_reg_ region: 1062, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1063.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 15.752s.\n",
      "-- designing region: Y:3740000-3760000_:+_reg_ region: 1063, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1064.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.226s.\n",
      "in 11.360s.\n",
      "-- designing region: Y:6240000-6260000_:+_reg_ region: 1064, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1065.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.173s.\n",
      "in 11.365s.\n",
      "-- designing region: Y:8740000-8760000_:+_reg_ region: 1065, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1066.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.231s.\n",
      "in 11.342s.\n",
      "-- designing region: Y:11240000-11260000_:+_reg_ region: 1066, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1067.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.200s.\n",
      "in 11.299s.\n",
      "-- designing region: Y:13740000-13760000_:+_reg_ region: 1067, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1068.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.358s.\n",
      "-- designing region: Y:16240000-16260000_:+_reg_ region: 1068, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1069.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.245s.\n",
      "in 11.217s.\n",
      "-- designing region: Y:18740000-18760000_:+_reg_ region: 1069, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1070.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.264s.\n",
      "-- designing region: Y:21240000-21260000_:+_reg_ region: 1070, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1071.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.235s.\n",
      "-- designing region: Y:23740000-23760000_:+_reg_ region: 1071, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1072.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.234s.\n",
      "in 11.447s.\n",
      "-- designing region: Y:26240000-26260000_:+_reg_ region: 1072, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1073.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.232s.\n",
      "in 11.279s.\n",
      "-- designing region: Y:28740000-28760000_:+_reg_ region: 1073, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1074.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.240s.\n",
      "in 11.297s.\n",
      "-- designing region: Y:31240000-31260000_:+_reg_ region: 1074, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1075.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.230s.\n",
      "in 11.403s.\n",
      "-- designing region: Y:33740000-33760000_:+_reg_ region: 1075, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1076.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.239s.\n",
      "in 11.287s.\n",
      "-- designing region: Y:36240000-36260000_:+_reg_ region: 1076, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1077.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.153s.\n",
      "in 11.219s.\n",
      "-- designing region: Y:38740000-38760000_:+_reg_ region: 1077, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1078.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.161s.\n",
      "in 11.357s.\n",
      "-- designing region: Y:41240000-41260000_:+_reg_ region: 1078, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1079.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.338s.\n",
      "-- designing region: Y:43740000-43760000_:+_reg_ region: 1079, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1080.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.279s.\n",
      "-- designing region: Y:46240000-46260000_:+_reg_ region: 1080, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1081.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.172s.\n",
      "in 11.343s.\n",
      "-- designing region: Y:48740000-48760000_:+_reg_ region: 1081, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1082.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.236s.\n",
      "in 11.366s.\n",
      "-- designing region: Y:51240000-51260000_:+_reg_ region: 1082, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1083.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.195s.\n",
      "in 11.285s.\n",
      "-- designing region: Y:53740000-53760000_:+_reg_ region: 1083, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1084.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.180s.\n",
      "in 11.373s.\n",
      "-- designing region: Y:56240000-56260000_:+_reg_ region: 1084, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1085.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.235s.\n",
      "in 11.373s.\n",
      "-- designing region: Y:58740000-58760000_:+_reg_ region: 1085, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1086.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.175s.\n",
      "in 11.279s.\n",
      "-- designing region: Y:61240000-61260000_:+_reg_ region: 1086, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1087.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.158s.\n",
      "in 11.345s.\n",
      "-- designing region: Y:63740000-63760000_:+_reg_ region: 1087, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1088.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.233s.\n",
      "in 11.343s.\n",
      "-- designing region: Y:66240000-66260000_:+_reg_ region: 1088, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1089.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.160s.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "in 0.381s.\n",
      "-- designing region: Y:68740000-68760000_:+_reg_ region: 1089, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1090.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.587s.\n",
      "in 11.627s.\n",
      "-- designing region: Y:71240000-71260000_:+_reg_ region: 1090, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1091.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.184s.\n",
      "in 11.267s.\n",
      "-- designing region: Y:73740000-73760000_:+_reg_ region: 1091, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1092.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.179s.\n",
      "in 11.372s.\n",
      "-- designing region: Y:76240000-76260000_:+_reg_ region: 1092, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1093.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.242s.\n",
      "in 11.312s.\n",
      "-- designing region: Y:78740000-78760000_:+_reg_ region: 1093, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1094.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.204s.\n",
      "in 11.344s.\n",
      "-- designing region: Y:81240000-81260000_:+_reg_ region: 1094, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1095.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.159s.\n",
      "in 11.365s.\n",
      "-- designing region: Y:83740000-83760000_:+_reg_ region: 1095, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1096.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.243s.\n",
      "in 11.352s.\n",
      "-- designing region: Y:86240000-86260000_:+_reg_ region: 1096, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1097.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.165s.\n",
      "in 11.265s.\n",
      "-- designing region: Y:88740000-88760000_:+_reg_ region: 1097, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1098.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.169s.\n",
      "in 11.336s.\n",
      "-- designing region: Y:91240000-91260000_:+_reg_ region: 1098, input file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\sequences\\all_reg_1099.fasta\n",
      "-- setting attribute: map_self_sequences\n",
      "- Mapping no. of seqs: 1\n",
      "--- finish map_self_sequences in 0.195s.\n",
      "in 0.416s.\n",
      "- Save reports into file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr\n"
     ]
    }
   ],
   "source": [
    "reload(library_tools)\n",
    "reload(library_tools.design)\n",
    "pb_designer.computeOTmaps() # load the tables \n",
    "pb_designer.compute_pb_report() # design candidate probes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e23bacce",
   "metadata": {},
   "source": [
    "### check probes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "9bfad542",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- check region:0 1:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.002s, 0 probes kept.\n",
      "-- check region:1 1:3740000-3760000_:+_reg_, 32788 candidate probes\n",
      "--- 7666 probes passed check_dic selection.\n",
      "finish in 0.985s, 163 probes kept.\n",
      "-- check region:2 1:6240000-6260000_:+_reg_, 36676 candidate probes\n",
      "--- 7588 probes passed check_dic selection.\n",
      "finish in 1.059s, 180 probes kept.\n",
      "-- check region:3 1:8740000-8760000_:+_reg_, 38726 candidate probes\n",
      "--- 3130 probes passed check_dic selection.\n",
      "finish in 0.920s, 77 probes kept.\n",
      "-- check region:4 1:11240000-11260000_:+_reg_, 35598 candidate probes\n",
      "--- 10319 probes passed check_dic selection.\n",
      "finish in 1.081s, 205 probes kept.\n",
      "-- check region:5 1:13740000-13760000_:+_reg_, 38308 candidate probes\n",
      "--- 6427 probes passed check_dic selection.\n",
      "finish in 0.988s, 144 probes kept.\n",
      "-- check region:6 1:16240000-16260000_:+_reg_, 39076 candidate probes\n",
      "--- 17280 probes passed check_dic selection.\n",
      "finish in 1.461s, 297 probes kept.\n",
      "-- check region:7 1:18740000-18760000_:+_reg_, 39382 candidate probes\n",
      "--- 9247 probes passed check_dic selection.\n",
      "finish in 1.133s, 210 probes kept.\n",
      "-- check region:8 1:21240000-21260000_:+_reg_, 39698 candidate probes\n",
      "--- 14216 probes passed check_dic selection.\n",
      "finish in 1.341s, 274 probes kept.\n",
      "-- check region:9 1:23740000-23760000_:+_reg_, 39234 candidate probes\n",
      "--- 11720 probes passed check_dic selection.\n",
      "finish in 1.257s, 220 probes kept.\n",
      "-- check region:10 1:26240000-26260000_:+_reg_, 37388 candidate probes\n",
      "--- 7573 probes passed check_dic selection.\n",
      "finish in 1.050s, 158 probes kept.\n",
      "-- check region:11 1:28740000-28760000_:+_reg_, 38748 candidate probes\n",
      "--- 6827 probes passed check_dic selection.\n",
      "finish in 1.005s, 156 probes kept.\n",
      "-- check region:12 1:31240000-31260000_:+_reg_, 36090 candidate probes\n",
      "--- 4137 probes passed check_dic selection.\n",
      "finish in 0.890s, 107 probes kept.\n",
      "-- check region:13 1:33740000-33760000_:+_reg_, 38350 candidate probes\n",
      "--- 15178 probes passed check_dic selection.\n",
      "finish in 1.371s, 291 probes kept.\n",
      "-- check region:14 1:36240000-36260000_:+_reg_, 39206 candidate probes\n",
      "--- 12863 probes passed check_dic selection.\n",
      "finish in 1.305s, 240 probes kept.\n",
      "-- check region:15 1:38740000-38760000_:+_reg_, 39380 candidate probes\n",
      "--- 17811 probes passed check_dic selection.\n",
      "finish in 1.485s, 279 probes kept.\n",
      "-- check region:16 1:41240000-41260000_:+_reg_, 39660 candidate probes\n",
      "--- 14622 probes passed check_dic selection.\n",
      "finish in 1.378s, 248 probes kept.\n",
      "-- check region:17 1:43740000-43760000_:+_reg_, 39578 candidate probes\n",
      "--- 15551 probes passed check_dic selection.\n",
      "finish in 1.398s, 276 probes kept.\n",
      "-- check region:18 1:46240000-46260000_:+_reg_, 39016 candidate probes\n",
      "--- 3653 probes passed check_dic selection.\n",
      "finish in 0.936s, 109 probes kept.\n",
      "-- check region:19 1:48740000-48760000_:+_reg_, 37014 candidate probes\n",
      "--- 4766 probes passed check_dic selection.\n",
      "finish in 0.904s, 111 probes kept.\n",
      "-- check region:20 1:51240000-51260000_:+_reg_, 29186 candidate probes\n",
      "--- 12080 probes passed check_dic selection.\n",
      "finish in 1.099s, 212 probes kept.\n",
      "-- check region:21 1:53740000-53760000_:+_reg_, 37958 candidate probes\n",
      "--- 5809 probes passed check_dic selection.\n",
      "finish in 0.995s, 135 probes kept.\n",
      "-- check region:22 1:56240000-56260000_:+_reg_, 39024 candidate probes\n",
      "--- 12120 probes passed check_dic selection.\n",
      "finish in 1.210s, 229 probes kept.\n",
      "-- check region:23 1:58740000-58760000_:+_reg_, 36130 candidate probes\n",
      "--- 13537 probes passed check_dic selection.\n",
      "finish in 1.317s, 257 probes kept.\n",
      "-- check region:24 1:61240000-61260000_:+_reg_, 39396 candidate probes\n",
      "--- 14248 probes passed check_dic selection.\n",
      "finish in 1.378s, 257 probes kept.\n",
      "-- check region:25 1:63740000-63760000_:+_reg_, 38970 candidate probes\n",
      "--- 11885 probes passed check_dic selection.\n",
      "finish in 1.223s, 249 probes kept.\n",
      "-- check region:26 1:66240000-66260000_:+_reg_, 37774 candidate probes\n",
      "--- 10991 probes passed check_dic selection.\n",
      "finish in 1.212s, 204 probes kept.\n",
      "-- check region:27 1:68740000-68760000_:+_reg_, 39732 candidate probes\n",
      "--- 9224 probes passed check_dic selection.\n",
      "finish in 1.123s, 195 probes kept.\n",
      "-- check region:28 1:71240000-71260000_:+_reg_, 27904 candidate probes\n",
      "--- 9408 probes passed check_dic selection.\n",
      "finish in 0.919s, 198 probes kept.\n",
      "-- check region:29 1:73740000-73760000_:+_reg_, 39684 candidate probes\n",
      "--- 16655 probes passed check_dic selection.\n",
      "finish in 1.462s, 284 probes kept.\n",
      "-- check region:30 1:76240000-76260000_:+_reg_, 39100 candidate probes\n",
      "--- 14516 probes passed check_dic selection.\n",
      "finish in 1.357s, 245 probes kept.\n",
      "-- check region:31 1:78740000-78760000_:+_reg_, 37010 candidate probes\n",
      "--- 7754 probes passed check_dic selection.\n",
      "finish in 1.058s, 151 probes kept.\n",
      "-- check region:32 1:81240000-81260000_:+_reg_, 39644 candidate probes\n",
      "--- 12025 probes passed check_dic selection.\n",
      "finish in 1.273s, 247 probes kept.\n",
      "-- check region:33 1:83740000-83760000_:+_reg_, 39758 candidate probes\n",
      "--- 14789 probes passed check_dic selection.\n",
      "finish in 1.348s, 276 probes kept.\n",
      "-- check region:34 1:86240000-86260000_:+_reg_, 39844 candidate probes\n",
      "--- 17648 probes passed check_dic selection.\n",
      "finish in 1.532s, 300 probes kept.\n",
      "-- check region:35 1:88740000-88760000_:+_reg_, 39870 candidate probes\n",
      "--- 18855 probes passed check_dic selection.\n",
      "finish in 1.557s, 310 probes kept.\n",
      "-- check region:36 1:91240000-91260000_:+_reg_, 38982 candidate probes\n",
      "--- 16297 probes passed check_dic selection.\n",
      "finish in 1.482s, 260 probes kept.\n",
      "-- check region:37 1:93740000-93760000_:+_reg_, 38880 candidate probes\n",
      "--- 13326 probes passed check_dic selection.\n",
      "finish in 1.370s, 244 probes kept.\n",
      "-- check region:38 1:96240000-96260000_:+_reg_, 28178 candidate probes\n",
      "--- 3141 probes passed check_dic selection.\n",
      "finish in 0.712s, 75 probes kept.\n",
      "-- check region:39 1:98740000-98760000_:+_reg_, 39052 candidate probes\n",
      "--- 8815 probes passed check_dic selection.\n",
      "finish in 1.079s, 178 probes kept.\n",
      "-- check region:40 1:101240000-101260000_:+_reg_, 32682 candidate probes\n",
      "--- 6823 probes passed check_dic selection.\n",
      "finish in 0.919s, 142 probes kept.\n",
      "-- check region:41 1:103740000-103760000_:+_reg_, 35206 candidate probes\n",
      "--- 5463 probes passed check_dic selection.\n",
      "finish in 0.903s, 122 probes kept.\n",
      "-- check region:42 1:106240000-106260000_:+_reg_, 37470 candidate probes\n",
      "--- 4603 probes passed check_dic selection.\n",
      "finish in 0.934s, 105 probes kept.\n",
      "-- check region:43 1:108740000-108760000_:+_reg_, 39442 candidate probes\n",
      "--- 6837 probes passed check_dic selection.\n",
      "finish in 1.021s, 173 probes kept.\n",
      "-- check region:44 1:111240000-111260000_:+_reg_, 28830 candidate probes\n",
      "--- 3464 probes passed check_dic selection.\n",
      "finish in 0.736s, 90 probes kept.\n",
      "-- check region:45 1:113740000-113760000_:+_reg_, 26460 candidate probes\n",
      "--- 6427 probes passed check_dic selection.\n",
      "finish in 0.797s, 121 probes kept.\n",
      "-- check region:46 1:116240000-116260000_:+_reg_, 39850 candidate probes\n",
      "--- 11256 probes passed check_dic selection.\n",
      "finish in 1.202s, 241 probes kept.\n",
      "-- check region:47 1:118740000-118760000_:+_reg_, 39140 candidate probes\n",
      "--- 13889 probes passed check_dic selection.\n",
      "finish in 1.329s, 240 probes kept.\n",
      "-- check region:48 1:121240000-121260000_:+_reg_, 38412 candidate probes\n",
      "--- 11360 probes passed check_dic selection.\n",
      "finish in 1.217s, 218 probes kept.\n",
      "-- check region:49 1:123740000-123760000_:+_reg_, 38946 candidate probes\n",
      "--- 10153 probes passed check_dic selection.\n",
      "finish in 1.189s, 225 probes kept.\n",
      "-- check region:50 1:126240000-126260000_:+_reg_, 39000 candidate probes\n",
      "--- 16510 probes passed check_dic selection.\n",
      "finish in 1.394s, 287 probes kept.\n",
      "-- check region:51 1:128740000-128760000_:+_reg_, 38966 candidate probes\n",
      "--- 13327 probes passed check_dic selection.\n",
      "finish in 1.371s, 250 probes kept.\n",
      "-- check region:52 1:131240000-131260000_:+_reg_, 39798 candidate probes\n",
      "--- 20358 probes passed check_dic selection.\n",
      "finish in 1.592s, 329 probes kept.\n",
      "-- check region:53 1:133740000-133760000_:+_reg_, 39332 candidate probes\n",
      "--- 18814 probes passed check_dic selection.\n",
      "finish in 1.554s, 299 probes kept.\n",
      "-- check region:54 1:136240000-136260000_:+_reg_, 39192 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 13081 probes passed check_dic selection.\n",
      "finish in 1.329s, 257 probes kept.\n",
      "-- check region:55 1:138740000-138760000_:+_reg_, 38066 candidate probes\n",
      "--- 10530 probes passed check_dic selection.\n",
      "finish in 1.149s, 206 probes kept.\n",
      "-- check region:56 1:141240000-141260000_:+_reg_, 35830 candidate probes\n",
      "--- 4714 probes passed check_dic selection.\n",
      "finish in 0.921s, 125 probes kept.\n",
      "-- check region:57 1:143740000-143760000_:+_reg_, 39206 candidate probes\n",
      "--- 8759 probes passed check_dic selection.\n",
      "finish in 1.225s, 232 probes kept.\n",
      "-- check region:58 1:146240000-146260000_:+_reg_, 36796 candidate probes\n",
      "--- 5192 probes passed check_dic selection.\n",
      "finish in 0.921s, 112 probes kept.\n",
      "-- check region:59 1:148740000-148760000_:+_reg_, 38712 candidate probes\n",
      "--- 6518 probes passed check_dic selection.\n",
      "finish in 0.996s, 159 probes kept.\n",
      "-- check region:60 1:151240000-151260000_:+_reg_, 39384 candidate probes\n",
      "--- 14182 probes passed check_dic selection.\n",
      "finish in 1.349s, 252 probes kept.\n",
      "-- check region:61 1:153740000-153760000_:+_reg_, 38150 candidate probes\n",
      "--- 15302 probes passed check_dic selection.\n",
      "finish in 1.368s, 281 probes kept.\n",
      "-- check region:62 1:156240000-156260000_:+_reg_, 39052 candidate probes\n",
      "--- 15799 probes passed check_dic selection.\n",
      "finish in 1.386s, 265 probes kept.\n",
      "-- check region:63 1:158740000-158760000_:+_reg_, 38710 candidate probes\n",
      "--- 14704 probes passed check_dic selection.\n",
      "finish in 1.337s, 245 probes kept.\n",
      "-- check region:64 1:161240000-161260000_:+_reg_, 39156 candidate probes\n",
      "--- 16875 probes passed check_dic selection.\n",
      "finish in 1.463s, 303 probes kept.\n",
      "-- check region:65 1:163740000-163760000_:+_reg_, 39050 candidate probes\n",
      "--- 12214 probes passed check_dic selection.\n",
      "finish in 1.247s, 231 probes kept.\n",
      "-- check region:66 1:166240000-166260000_:+_reg_, 39502 candidate probes\n",
      "--- 18286 probes passed check_dic selection.\n",
      "finish in 1.515s, 310 probes kept.\n",
      "-- check region:67 1:168740000-168760000_:+_reg_, 38144 candidate probes\n",
      "--- 10350 probes passed check_dic selection.\n",
      "finish in 1.226s, 197 probes kept.\n",
      "-- check region:68 1:171240000-171260000_:+_reg_, 39462 candidate probes\n",
      "--- 14917 probes passed check_dic selection.\n",
      "finish in 1.406s, 283 probes kept.\n",
      "-- check region:69 1:173740000-173760000_:+_reg_, 39084 candidate probes\n",
      "--- 3824 probes passed check_dic selection.\n",
      "finish in 0.897s, 113 probes kept.\n",
      "-- check region:70 1:176240000-176260000_:+_reg_, 36466 candidate probes\n",
      "--- 11871 probes passed check_dic selection.\n",
      "finish in 1.190s, 209 probes kept.\n",
      "-- check region:71 1:178740000-178760000_:+_reg_, 39446 candidate probes\n",
      "--- 16928 probes passed check_dic selection.\n",
      "finish in 1.463s, 274 probes kept.\n",
      "-- check region:72 1:181240000-181260000_:+_reg_, 39378 candidate probes\n",
      "--- 15815 probes passed check_dic selection.\n",
      "finish in 1.399s, 251 probes kept.\n",
      "-- check region:73 1:183740000-183760000_:+_reg_, 39218 candidate probes\n",
      "--- 16606 probes passed check_dic selection.\n",
      "finish in 1.408s, 282 probes kept.\n",
      "-- check region:74 1:186240000-186260000_:+_reg_, 39706 candidate probes\n",
      "--- 16083 probes passed check_dic selection.\n",
      "finish in 1.436s, 276 probes kept.\n",
      "-- check region:75 1:188740000-188760000_:+_reg_, 38888 candidate probes\n",
      "--- 15643 probes passed check_dic selection.\n",
      "finish in 1.358s, 278 probes kept.\n",
      "-- check region:76 1:191240000-191260000_:+_reg_, 39406 candidate probes\n",
      "--- 16701 probes passed check_dic selection.\n",
      "finish in 1.460s, 265 probes kept.\n",
      "-- check region:77 1:193740000-193760000_:+_reg_, 39388 candidate probes\n",
      "--- 10053 probes passed check_dic selection.\n",
      "finish in 1.169s, 193 probes kept.\n",
      "-- check region:78 1:196240000-195471971_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:79 2:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.003s, 0 probes kept.\n",
      "-- check region:80 2:3740000-3760000_:+_reg_, 39712 candidate probes\n",
      "--- 17940 probes passed check_dic selection.\n",
      "finish in 1.492s, 287 probes kept.\n",
      "-- check region:81 2:6240000-6260000_:+_reg_, 38716 candidate probes\n",
      "--- 13578 probes passed check_dic selection.\n",
      "finish in 1.345s, 240 probes kept.\n",
      "-- check region:82 2:8740000-8760000_:+_reg_, 39804 candidate probes\n",
      "--- 11038 probes passed check_dic selection.\n",
      "finish in 1.198s, 223 probes kept.\n",
      "-- check region:83 2:11240000-11260000_:+_reg_, 37980 candidate probes\n",
      "--- 13209 probes passed check_dic selection.\n",
      "finish in 1.317s, 233 probes kept.\n",
      "-- check region:84 2:13740000-13760000_:+_reg_, 39834 candidate probes\n",
      "--- 15835 probes passed check_dic selection.\n",
      "finish in 1.380s, 290 probes kept.\n",
      "-- check region:85 2:16240000-16260000_:+_reg_, 38908 candidate probes\n",
      "--- 7158 probes passed check_dic selection.\n",
      "finish in 1.037s, 173 probes kept.\n",
      "-- check region:86 2:18740000-18760000_:+_reg_, 38942 candidate probes\n",
      "--- 10278 probes passed check_dic selection.\n",
      "finish in 1.219s, 202 probes kept.\n",
      "-- check region:87 2:21240000-21260000_:+_reg_, 28568 candidate probes\n",
      "--- 8358 probes passed check_dic selection.\n",
      "finish in 0.904s, 154 probes kept.\n",
      "-- check region:88 2:23740000-23760000_:+_reg_, 39194 candidate probes\n",
      "--- 3962 probes passed check_dic selection.\n",
      "finish in 0.946s, 117 probes kept.\n",
      "-- check region:89 2:26240000-26260000_:+_reg_, 39546 candidate probes\n",
      "--- 18274 probes passed check_dic selection.\n",
      "finish in 1.515s, 318 probes kept.\n",
      "-- check region:90 2:28740000-28760000_:+_reg_, 39598 candidate probes\n",
      "--- 20207 probes passed check_dic selection.\n",
      "finish in 1.581s, 315 probes kept.\n",
      "-- check region:91 2:31240000-31260000_:+_reg_, 39668 candidate probes\n",
      "--- 20242 probes passed check_dic selection.\n",
      "finish in 1.621s, 306 probes kept.\n",
      "-- check region:92 2:33740000-33760000_:+_reg_, 39664 candidate probes\n",
      "--- 22488 probes passed check_dic selection.\n",
      "finish in 1.695s, 338 probes kept.\n",
      "-- check region:93 2:36240000-36260000_:+_reg_, 39598 candidate probes\n",
      "--- 15653 probes passed check_dic selection.\n",
      "finish in 1.417s, 284 probes kept.\n",
      "-- check region:94 2:38740000-38760000_:+_reg_, 38342 candidate probes\n",
      "--- 10156 probes passed check_dic selection.\n",
      "finish in 1.134s, 207 probes kept.\n",
      "-- check region:95 2:41240000-41260000_:+_reg_, 39602 candidate probes\n",
      "--- 8865 probes passed check_dic selection.\n",
      "finish in 1.166s, 185 probes kept.\n",
      "-- check region:96 2:43740000-43760000_:+_reg_, 38238 candidate probes\n",
      "--- 11512 probes passed check_dic selection.\n",
      "finish in 1.275s, 207 probes kept.\n",
      "-- check region:97 2:46240000-46260000_:+_reg_, 38772 candidate probes\n",
      "--- 6508 probes passed check_dic selection.\n",
      "finish in 1.019s, 147 probes kept.\n",
      "-- check region:98 2:48740000-48760000_:+_reg_, 39694 candidate probes\n",
      "--- 13117 probes passed check_dic selection.\n",
      "finish in 1.325s, 249 probes kept.\n",
      "-- check region:99 2:51240000-51260000_:+_reg_, 38712 candidate probes\n",
      "--- 10628 probes passed check_dic selection.\n",
      "finish in 1.180s, 192 probes kept.\n",
      "-- check region:100 2:53740000-53760000_:+_reg_, 24096 candidate probes\n",
      "--- 3910 probes passed check_dic selection.\n",
      "finish in 0.657s, 82 probes kept.\n",
      "-- check region:101 2:56240000-56260000_:+_reg_, 36352 candidate probes\n",
      "--- 5386 probes passed check_dic selection.\n",
      "finish in 0.938s, 122 probes kept.\n",
      "-- check region:102 2:58740000-58760000_:+_reg_, 39454 candidate probes\n",
      "--- 15637 probes passed check_dic selection.\n",
      "finish in 1.411s, 267 probes kept.\n",
      "-- check region:103 2:61240000-61260000_:+_reg_, 39752 candidate probes\n",
      "--- 11653 probes passed check_dic selection.\n",
      "finish in 1.246s, 220 probes kept.\n",
      "-- check region:104 2:63740000-63760000_:+_reg_, 39674 candidate probes\n",
      "--- 9343 probes passed check_dic selection.\n",
      "finish in 1.143s, 201 probes kept.\n",
      "-- check region:105 2:66240000-66260000_:+_reg_, 39700 candidate probes\n",
      "--- 15686 probes passed check_dic selection.\n",
      "finish in 1.399s, 271 probes kept.\n",
      "-- check region:106 2:68740000-68760000_:+_reg_, 38094 candidate probes\n",
      "--- 10622 probes passed check_dic selection.\n",
      "finish in 1.180s, 213 probes kept.\n",
      "-- check region:107 2:71240000-71260000_:+_reg_, 39630 candidate probes\n",
      "--- 13025 probes passed check_dic selection.\n",
      "finish in 1.295s, 246 probes kept.\n",
      "-- check region:108 2:73740000-73760000_:+_reg_, 29024 candidate probes\n",
      "--- 7785 probes passed check_dic selection.\n",
      "finish in 0.946s, 152 probes kept.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- check region:109 2:76240000-76260000_:+_reg_, 39216 candidate probes\n",
      "--- 14573 probes passed check_dic selection.\n",
      "finish in 1.349s, 263 probes kept.\n",
      "-- check region:110 2:78740000-78760000_:+_reg_, 39420 candidate probes\n",
      "--- 12941 probes passed check_dic selection.\n",
      "finish in 1.371s, 248 probes kept.\n",
      "-- check region:111 2:81240000-81260000_:+_reg_, 35350 candidate probes\n",
      "--- 4027 probes passed check_dic selection.\n",
      "finish in 0.860s, 97 probes kept.\n",
      "-- check region:112 2:83740000-83760000_:+_reg_, 38732 candidate probes\n",
      "--- 9955 probes passed check_dic selection.\n",
      "finish in 1.211s, 205 probes kept.\n",
      "-- check region:113 2:86240000-86260000_:+_reg_, 36864 candidate probes\n",
      "--- 2954 probes passed check_dic selection.\n",
      "finish in 0.841s, 91 probes kept.\n",
      "-- check region:114 2:88740000-88760000_:+_reg_, 39166 candidate probes\n",
      "--- 5967 probes passed check_dic selection.\n",
      "finish in 1.090s, 163 probes kept.\n",
      "-- check region:115 2:91240000-91260000_:+_reg_, 38974 candidate probes\n",
      "--- 13668 probes passed check_dic selection.\n",
      "finish in 1.310s, 242 probes kept.\n",
      "-- check region:116 2:93740000-93760000_:+_reg_, 39404 candidate probes\n",
      "--- 18088 probes passed check_dic selection.\n",
      "finish in 1.474s, 296 probes kept.\n",
      "-- check region:117 2:96240000-96260000_:+_reg_, 38026 candidate probes\n",
      "--- 8220 probes passed check_dic selection.\n",
      "finish in 1.119s, 175 probes kept.\n",
      "-- check region:118 2:98740000-98760000_:+_reg_, 34718 candidate probes\n",
      "--- 4892 probes passed check_dic selection.\n",
      "finish in 0.872s, 111 probes kept.\n",
      "-- check region:119 2:101240000-101260000_:+_reg_, 36990 candidate probes\n",
      "--- 9948 probes passed check_dic selection.\n",
      "finish in 1.110s, 203 probes kept.\n",
      "-- check region:120 2:103740000-103760000_:+_reg_, 39404 candidate probes\n",
      "--- 15992 probes passed check_dic selection.\n",
      "finish in 1.455s, 279 probes kept.\n",
      "-- check region:121 2:106240000-106260000_:+_reg_, 38134 candidate probes\n",
      "--- 11879 probes passed check_dic selection.\n",
      "finish in 1.192s, 223 probes kept.\n",
      "-- check region:122 2:108740000-108760000_:+_reg_, 39348 candidate probes\n",
      "--- 11638 probes passed check_dic selection.\n",
      "finish in 1.254s, 231 probes kept.\n",
      "-- check region:123 2:111240000-111260000_:+_reg_, 38086 candidate probes\n",
      "--- 3059 probes passed check_dic selection.\n",
      "finish in 0.917s, 102 probes kept.\n",
      "-- check region:124 2:113740000-113760000_:+_reg_, 39350 candidate probes\n",
      "--- 13684 probes passed check_dic selection.\n",
      "finish in 1.300s, 255 probes kept.\n",
      "-- check region:125 2:116240000-116260000_:+_reg_, 39258 candidate probes\n",
      "--- 10740 probes passed check_dic selection.\n",
      "finish in 1.186s, 191 probes kept.\n",
      "-- check region:126 2:118740000-118760000_:+_reg_, 39178 candidate probes\n",
      "--- 16824 probes passed check_dic selection.\n",
      "finish in 1.476s, 318 probes kept.\n",
      "-- check region:127 2:121240000-121260000_:+_reg_, 38286 candidate probes\n",
      "--- 11658 probes passed check_dic selection.\n",
      "finish in 1.264s, 224 probes kept.\n",
      "-- check region:128 2:123740000-123760000_:+_reg_, 32324 candidate probes\n",
      "--- 8028 probes passed check_dic selection.\n",
      "finish in 0.971s, 167 probes kept.\n",
      "-- check region:129 2:126240000-126260000_:+_reg_, 28344 candidate probes\n",
      "--- 7606 probes passed check_dic selection.\n",
      "finish in 0.910s, 147 probes kept.\n",
      "-- check region:130 2:128740000-128760000_:+_reg_, 39260 candidate probes\n",
      "--- 15012 probes passed check_dic selection.\n",
      "finish in 1.356s, 274 probes kept.\n",
      "-- check region:131 2:131240000-131260000_:+_reg_, 39168 candidate probes\n",
      "--- 13633 probes passed check_dic selection.\n",
      "finish in 1.357s, 261 probes kept.\n",
      "-- check region:132 2:133740000-133760000_:+_reg_, 39786 candidate probes\n",
      "--- 13030 probes passed check_dic selection.\n",
      "finish in 1.316s, 248 probes kept.\n",
      "-- check region:133 2:136240000-136260000_:+_reg_, 37954 candidate probes\n",
      "--- 6653 probes passed check_dic selection.\n",
      "finish in 1.023s, 159 probes kept.\n",
      "-- check region:134 2:138740000-138760000_:+_reg_, 39502 candidate probes\n",
      "--- 5860 probes passed check_dic selection.\n",
      "finish in 0.999s, 140 probes kept.\n",
      "-- check region:135 2:141240000-141260000_:+_reg_, 38732 candidate probes\n",
      "--- 12155 probes passed check_dic selection.\n",
      "finish in 1.242s, 232 probes kept.\n",
      "-- check region:136 2:143740000-143760000_:+_reg_, 34628 candidate probes\n",
      "--- 13728 probes passed check_dic selection.\n",
      "finish in 1.236s, 235 probes kept.\n",
      "-- check region:137 2:146240000-146260000_:+_reg_, 39814 candidate probes\n",
      "--- 16992 probes passed check_dic selection.\n",
      "finish in 1.525s, 313 probes kept.\n",
      "-- check region:138 2:148740000-148760000_:+_reg_, 31592 candidate probes\n",
      "--- 11801 probes passed check_dic selection.\n",
      "finish in 1.109s, 199 probes kept.\n",
      "-- check region:139 2:151240000-151260000_:+_reg_, 39358 candidate probes\n",
      "--- 544 probes passed check_dic selection.\n",
      "finish in 0.847s, 22 probes kept.\n",
      "-- check region:140 2:153740000-153760000_:+_reg_, 38924 candidate probes\n",
      "--- 12187 probes passed check_dic selection.\n",
      "finish in 1.292s, 226 probes kept.\n",
      "-- check region:141 2:156240000-156260000_:+_reg_, 39304 candidate probes\n",
      "--- 13007 probes passed check_dic selection.\n",
      "finish in 1.345s, 259 probes kept.\n",
      "-- check region:142 2:158740000-158760000_:+_reg_, 39288 candidate probes\n",
      "--- 17964 probes passed check_dic selection.\n",
      "finish in 1.508s, 308 probes kept.\n",
      "-- check region:143 2:161240000-161260000_:+_reg_, 39556 candidate probes\n",
      "--- 13721 probes passed check_dic selection.\n",
      "finish in 1.301s, 248 probes kept.\n",
      "-- check region:144 2:163740000-163760000_:+_reg_, 39822 candidate probes\n",
      "--- 19874 probes passed check_dic selection.\n",
      "finish in 1.587s, 307 probes kept.\n",
      "-- check region:145 2:166240000-166260000_:+_reg_, 38892 candidate probes\n",
      "--- 19796 probes passed check_dic selection.\n",
      "finish in 1.588s, 302 probes kept.\n",
      "-- check region:146 2:168740000-168760000_:+_reg_, 38791 candidate probes\n",
      "--- 15109 probes passed check_dic selection.\n",
      "finish in 1.337s, 282 probes kept.\n",
      "-- check region:147 2:171240000-171260000_:+_reg_, 36174 candidate probes\n",
      "--- 14960 probes passed check_dic selection.\n",
      "finish in 1.296s, 252 probes kept.\n",
      "-- check region:148 2:173740000-173760000_:+_reg_, 36138 candidate probes\n",
      "--- 15921 probes passed check_dic selection.\n",
      "finish in 1.335s, 268 probes kept.\n",
      "-- check region:149 2:176240000-176260000_:+_reg_, 38662 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.839s, 0 probes kept.\n",
      "-- check region:150 2:178740000-178760000_:+_reg_, 38526 candidate probes\n",
      "--- 15912 probes passed check_dic selection.\n",
      "finish in 1.391s, 258 probes kept.\n",
      "-- check region:151 2:181240000-181260000_:+_reg_, 39384 candidate probes\n",
      "--- 14626 probes passed check_dic selection.\n",
      "finish in 1.386s, 285 probes kept.\n",
      "-- check region:152 3:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:153 3:3740000-3760000_:+_reg_, 39282 candidate probes\n",
      "--- 9229 probes passed check_dic selection.\n",
      "finish in 1.180s, 187 probes kept.\n",
      "-- check region:154 3:6240000-6260000_:+_reg_, 39470 candidate probes\n",
      "--- 6738 probes passed check_dic selection.\n",
      "finish in 1.037s, 159 probes kept.\n",
      "-- check region:155 3:8740000-8760000_:+_reg_, 38774 candidate probes\n",
      "--- 15671 probes passed check_dic selection.\n",
      "finish in 1.403s, 265 probes kept.\n",
      "-- check region:156 3:11240000-11260000_:+_reg_, 38236 candidate probes\n",
      "--- 6286 probes passed check_dic selection.\n",
      "finish in 1.004s, 144 probes kept.\n",
      "-- check region:157 3:13740000-13760000_:+_reg_, 38560 candidate probes\n",
      "--- 9237 probes passed check_dic selection.\n",
      "finish in 1.119s, 199 probes kept.\n",
      "-- check region:158 3:16240000-16260000_:+_reg_, 38848 candidate probes\n",
      "--- 5129 probes passed check_dic selection.\n",
      "finish in 1.007s, 127 probes kept.\n",
      "-- check region:159 3:18740000-18760000_:+_reg_, 39808 candidate probes\n",
      "--- 14765 probes passed check_dic selection.\n",
      "finish in 1.328s, 270 probes kept.\n",
      "-- check region:160 3:21240000-21260000_:+_reg_, 33950 candidate probes\n",
      "--- 5568 probes passed check_dic selection.\n",
      "finish in 0.914s, 141 probes kept.\n",
      "-- check region:161 3:23740000-23760000_:+_reg_, 37914 candidate probes\n",
      "--- 7967 probes passed check_dic selection.\n",
      "finish in 1.067s, 168 probes kept.\n",
      "-- check region:162 3:26240000-26260000_:+_reg_, 39586 candidate probes\n",
      "--- 10781 probes passed check_dic selection.\n",
      "finish in 1.214s, 221 probes kept.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- check region:163 3:28740000-28760000_:+_reg_, 39704 candidate probes\n",
      "--- 15048 probes passed check_dic selection.\n",
      "finish in 1.412s, 257 probes kept.\n",
      "-- check region:164 3:31240000-31260000_:+_reg_, 39876 candidate probes\n",
      "--- 16561 probes passed check_dic selection.\n",
      "finish in 1.434s, 288 probes kept.\n",
      "-- check region:165 3:33740000-33760000_:+_reg_, 39644 candidate probes\n",
      "--- 11699 probes passed check_dic selection.\n",
      "finish in 1.219s, 225 probes kept.\n",
      "-- check region:166 3:36240000-36260000_:+_reg_, 32576 candidate probes\n",
      "--- 4968 probes passed check_dic selection.\n",
      "finish in 0.887s, 103 probes kept.\n",
      "-- check region:167 3:38740000-38760000_:+_reg_, 34030 candidate probes\n",
      "--- 9093 probes passed check_dic selection.\n",
      "finish in 1.095s, 164 probes kept.\n",
      "-- check region:168 3:41240000-41260000_:+_reg_, 33130 candidate probes\n",
      "--- 10690 probes passed check_dic selection.\n",
      "finish in 1.131s, 179 probes kept.\n",
      "-- check region:169 3:43740000-43760000_:+_reg_, 38100 candidate probes\n",
      "--- 5542 probes passed check_dic selection.\n",
      "finish in 1.006s, 140 probes kept.\n",
      "-- check region:170 3:46240000-46260000_:+_reg_, 32648 candidate probes\n",
      "--- 3738 probes passed check_dic selection.\n",
      "finish in 0.821s, 86 probes kept.\n",
      "-- check region:171 3:48740000-48760000_:+_reg_, 39604 candidate probes\n",
      "--- 6076 probes passed check_dic selection.\n",
      "finish in 1.032s, 139 probes kept.\n",
      "-- check region:172 3:51240000-51260000_:+_reg_, 31684 candidate probes\n",
      "--- 8995 probes passed check_dic selection.\n",
      "finish in 1.004s, 222 probes kept.\n",
      "-- check region:173 3:53740000-53760000_:+_reg_, 38206 candidate probes\n",
      "--- 13276 probes passed check_dic selection.\n",
      "finish in 1.280s, 225 probes kept.\n",
      "-- check region:174 3:56240000-56260000_:+_reg_, 27610 candidate probes\n",
      "--- 8603 probes passed check_dic selection.\n",
      "finish in 0.951s, 160 probes kept.\n",
      "-- check region:175 3:58740000-58760000_:+_reg_, 37966 candidate probes\n",
      "--- 11492 probes passed check_dic selection.\n",
      "finish in 1.187s, 210 probes kept.\n",
      "-- check region:176 3:61240000-61260000_:+_reg_, 38004 candidate probes\n",
      "--- 11081 probes passed check_dic selection.\n",
      "finish in 1.210s, 213 probes kept.\n",
      "-- check region:177 3:63740000-63760000_:+_reg_, 39652 candidate probes\n",
      "--- 14506 probes passed check_dic selection.\n",
      "finish in 1.321s, 268 probes kept.\n",
      "-- check region:178 3:66240000-66260000_:+_reg_, 39408 candidate probes\n",
      "--- 15123 probes passed check_dic selection.\n",
      "finish in 1.402s, 261 probes kept.\n",
      "-- check region:179 3:68740000-68760000_:+_reg_, 29536 candidate probes\n",
      "--- 9466 probes passed check_dic selection.\n",
      "finish in 0.963s, 163 probes kept.\n",
      "-- check region:180 3:71240000-71260000_:+_reg_, 39358 candidate probes\n",
      "--- 4845 probes passed check_dic selection.\n",
      "finish in 0.950s, 136 probes kept.\n",
      "-- check region:181 3:73740000-73760000_:+_reg_, 38768 candidate probes\n",
      "--- 7706 probes passed check_dic selection.\n",
      "finish in 1.118s, 181 probes kept.\n",
      "-- check region:182 3:76240000-76260000_:+_reg_, 36470 candidate probes\n",
      "--- 3918 probes passed check_dic selection.\n",
      "finish in 0.886s, 99 probes kept.\n",
      "-- check region:183 3:78740000-78760000_:+_reg_, 22220 candidate probes\n",
      "--- 6131 probes passed check_dic selection.\n",
      "finish in 0.730s, 110 probes kept.\n",
      "-- check region:184 3:81240000-81260000_:+_reg_, 25524 candidate probes\n",
      "--- 6512 probes passed check_dic selection.\n",
      "finish in 0.763s, 126 probes kept.\n",
      "-- check region:185 3:83740000-83760000_:+_reg_, 39770 candidate probes\n",
      "--- 17790 probes passed check_dic selection.\n",
      "finish in 1.526s, 290 probes kept.\n",
      "-- check region:186 3:86240000-86260000_:+_reg_, 36802 candidate probes\n",
      "--- 7644 probes passed check_dic selection.\n",
      "finish in 1.004s, 182 probes kept.\n",
      "-- check region:187 3:88740000-88760000_:+_reg_, 38362 candidate probes\n",
      "--- 9785 probes passed check_dic selection.\n",
      "finish in 1.175s, 216 probes kept.\n",
      "-- check region:188 3:91240000-91260000_:+_reg_, 37420 candidate probes\n",
      "--- 3543 probes passed check_dic selection.\n",
      "finish in 0.890s, 114 probes kept.\n",
      "-- check region:189 3:93740000-93760000_:+_reg_, 30788 candidate probes\n",
      "--- 240 probes passed check_dic selection.\n",
      "finish in 0.645s, 9 probes kept.\n",
      "-- check region:190 3:96240000-96260000_:+_reg_, 38386 candidate probes\n",
      "--- 9685 probes passed check_dic selection.\n",
      "finish in 1.195s, 229 probes kept.\n",
      "-- check region:191 3:98740000-98760000_:+_reg_, 33480 candidate probes\n",
      "--- 6098 probes passed check_dic selection.\n",
      "finish in 0.927s, 142 probes kept.\n",
      "-- check region:192 3:101240000-101260000_:+_reg_, 39564 candidate probes\n",
      "--- 14557 probes passed check_dic selection.\n",
      "finish in 1.397s, 251 probes kept.\n",
      "-- check region:193 3:103740000-103760000_:+_reg_, 39874 candidate probes\n",
      "--- 12559 probes passed check_dic selection.\n",
      "finish in 1.356s, 275 probes kept.\n",
      "-- check region:194 3:106240000-106260000_:+_reg_, 36896 candidate probes\n",
      "--- 634 probes passed check_dic selection.\n",
      "finish in 0.739s, 29 probes kept.\n",
      "-- check region:195 3:108740000-108760000_:+_reg_, 38186 candidate probes\n",
      "--- 8675 probes passed check_dic selection.\n",
      "finish in 1.149s, 186 probes kept.\n",
      "-- check region:196 3:111240000-111260000_:+_reg_, 33726 candidate probes\n",
      "--- 4442 probes passed check_dic selection.\n",
      "finish in 0.859s, 115 probes kept.\n",
      "-- check region:197 3:113740000-113760000_:+_reg_, 29934 candidate probes\n",
      "--- 5467 probes passed check_dic selection.\n",
      "finish in 0.792s, 115 probes kept.\n",
      "-- check region:198 3:116240000-116260000_:+_reg_, 39472 candidate probes\n",
      "--- 15742 probes passed check_dic selection.\n",
      "finish in 1.437s, 289 probes kept.\n",
      "-- check region:199 3:118740000-118760000_:+_reg_, 39472 candidate probes\n",
      "--- 7433 probes passed check_dic selection.\n",
      "finish in 1.114s, 175 probes kept.\n",
      "-- check region:200 3:121240000-121260000_:+_reg_, 38818 candidate probes\n",
      "--- 15068 probes passed check_dic selection.\n",
      "finish in 1.386s, 258 probes kept.\n",
      "-- check region:201 3:123740000-123760000_:+_reg_, 32202 candidate probes\n",
      "--- 8627 probes passed check_dic selection.\n",
      "finish in 0.991s, 169 probes kept.\n",
      "-- check region:202 3:126240000-126260000_:+_reg_, 38164 candidate probes\n",
      "--- 9010 probes passed check_dic selection.\n",
      "finish in 1.099s, 180 probes kept.\n",
      "-- check region:203 3:128740000-128760000_:+_reg_, 39512 candidate probes\n",
      "--- 9190 probes passed check_dic selection.\n",
      "finish in 1.147s, 197 probes kept.\n",
      "-- check region:204 3:131240000-131260000_:+_reg_, 39692 candidate probes\n",
      "--- 18078 probes passed check_dic selection.\n",
      "finish in 1.528s, 314 probes kept.\n",
      "-- check region:205 3:133740000-133760000_:+_reg_, 38715 candidate probes\n",
      "--- 14476 probes passed check_dic selection.\n",
      "finish in 1.349s, 264 probes kept.\n",
      "-- check region:206 3:136240000-136260000_:+_reg_, 39326 candidate probes\n",
      "--- 10603 probes passed check_dic selection.\n",
      "finish in 1.168s, 198 probes kept.\n",
      "-- check region:207 3:138740000-138760000_:+_reg_, 39470 candidate probes\n",
      "--- 18805 probes passed check_dic selection.\n",
      "finish in 1.604s, 294 probes kept.\n",
      "-- check region:208 3:141240000-141260000_:+_reg_, 36888 candidate probes\n",
      "--- 9697 probes passed check_dic selection.\n",
      "finish in 1.098s, 191 probes kept.\n",
      "-- check region:209 3:143740000-143760000_:+_reg_, 39752 candidate probes\n",
      "--- 17096 probes passed check_dic selection.\n",
      "finish in 1.452s, 290 probes kept.\n",
      "-- check region:210 3:146240000-146260000_:+_reg_, 39538 candidate probes\n",
      "--- 19529 probes passed check_dic selection.\n",
      "finish in 1.591s, 295 probes kept.\n",
      "-- check region:211 3:148740000-148760000_:+_reg_, 26818 candidate probes\n",
      "--- 11131 probes passed check_dic selection.\n",
      "finish in 0.988s, 184 probes kept.\n",
      "-- check region:212 3:151240000-151260000_:+_reg_, 38350 candidate probes\n",
      "--- 7404 probes passed check_dic selection.\n",
      "finish in 1.091s, 160 probes kept.\n",
      "-- check region:213 3:153740000-153760000_:+_reg_, 37912 candidate probes\n",
      "--- 14054 probes passed check_dic selection.\n",
      "finish in 1.336s, 241 probes kept.\n",
      "-- check region:214 3:156240000-156260000_:+_reg_, 34190 candidate probes\n",
      "--- 9171 probes passed check_dic selection.\n",
      "finish in 1.023s, 180 probes kept.\n",
      "-- check region:215 3:158740000-158760000_:+_reg_, 31030 candidate probes\n",
      "--- 5638 probes passed check_dic selection.\n",
      "finish in 0.815s, 122 probes kept.\n",
      "-- check region:216 3:161240000-160039680_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.000s, 0 probes kept.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- check region:217 4:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:218 4:3740000-3760000_:+_reg_, 39844 candidate probes\n",
      "--- 2979 probes passed check_dic selection.\n",
      "finish in 0.911s, 89 probes kept.\n",
      "-- check region:219 4:6240000-6260000_:+_reg_, 36372 candidate probes\n",
      "--- 8568 probes passed check_dic selection.\n",
      "finish in 1.033s, 164 probes kept.\n",
      "-- check region:220 4:8740000-8760000_:+_reg_, 38894 candidate probes\n",
      "--- 16768 probes passed check_dic selection.\n",
      "finish in 1.437s, 298 probes kept.\n",
      "-- check region:221 4:11240000-11260000_:+_reg_, 38518 candidate probes\n",
      "--- 10330 probes passed check_dic selection.\n",
      "finish in 1.196s, 203 probes kept.\n",
      "-- check region:222 4:13740000-13760000_:+_reg_, 39876 candidate probes\n",
      "--- 18037 probes passed check_dic selection.\n",
      "finish in 1.520s, 296 probes kept.\n",
      "-- check region:223 4:16240000-16260000_:+_reg_, 35992 candidate probes\n",
      "--- 6550 probes passed check_dic selection.\n",
      "finish in 0.980s, 158 probes kept.\n",
      "-- check region:224 4:18740000-18760000_:+_reg_, 39040 candidate probes\n",
      "--- 3938 probes passed check_dic selection.\n",
      "finish in 0.920s, 104 probes kept.\n",
      "-- check region:225 4:21240000-21260000_:+_reg_, 25992 candidate probes\n",
      "--- 3539 probes passed check_dic selection.\n",
      "finish in 0.716s, 84 probes kept.\n",
      "-- check region:226 4:23740000-23760000_:+_reg_, 39108 candidate probes\n",
      "--- 9294 probes passed check_dic selection.\n",
      "finish in 1.136s, 184 probes kept.\n",
      "-- check region:227 4:26240000-26260000_:+_reg_, 36956 candidate probes\n",
      "--- 4453 probes passed check_dic selection.\n",
      "finish in 0.892s, 119 probes kept.\n",
      "-- check region:228 4:28740000-28760000_:+_reg_, 25472 candidate probes\n",
      "--- 4032 probes passed check_dic selection.\n",
      "finish in 0.705s, 84 probes kept.\n",
      "-- check region:229 4:31240000-31260000_:+_reg_, 39558 candidate probes\n",
      "--- 13829 probes passed check_dic selection.\n",
      "finish in 1.301s, 244 probes kept.\n",
      "-- check region:230 4:33740000-33760000_:+_reg_, 39374 candidate probes\n",
      "--- 15090 probes passed check_dic selection.\n",
      "finish in 1.451s, 264 probes kept.\n",
      "-- check region:231 4:36240000-36260000_:+_reg_, 39064 candidate probes\n",
      "--- 6622 probes passed check_dic selection.\n",
      "finish in 0.999s, 143 probes kept.\n",
      "-- check region:232 4:38740000-38760000_:+_reg_, 30564 candidate probes\n",
      "--- 4229 probes passed check_dic selection.\n",
      "finish in 0.771s, 110 probes kept.\n",
      "-- check region:233 4:41240000-41260000_:+_reg_, 39038 candidate probes\n",
      "--- 9699 probes passed check_dic selection.\n",
      "finish in 1.200s, 223 probes kept.\n",
      "-- check region:234 4:43740000-43760000_:+_reg_, 33322 candidate probes\n",
      "--- 10872 probes passed check_dic selection.\n",
      "finish in 1.130s, 194 probes kept.\n",
      "-- check region:235 4:46240000-46260000_:+_reg_, 31596 candidate probes\n",
      "--- 7978 probes passed check_dic selection.\n",
      "finish in 0.982s, 151 probes kept.\n",
      "-- check region:236 4:48740000-48760000_:+_reg_, 37870 candidate probes\n",
      "--- 11090 probes passed check_dic selection.\n",
      "finish in 1.340s, 220 probes kept.\n",
      "-- check region:237 4:51240000-51260000_:+_reg_, 39496 candidate probes\n",
      "--- 4142 probes passed check_dic selection.\n",
      "finish in 0.992s, 108 probes kept.\n",
      "-- check region:238 4:53740000-53760000_:+_reg_, 39300 candidate probes\n",
      "--- 14942 probes passed check_dic selection.\n",
      "finish in 1.450s, 258 probes kept.\n",
      "-- check region:239 4:56240000-56260000_:+_reg_, 38936 candidate probes\n",
      "--- 15542 probes passed check_dic selection.\n",
      "finish in 1.440s, 257 probes kept.\n",
      "-- check region:240 4:58740000-58760000_:+_reg_, 37042 candidate probes\n",
      "--- 7296 probes passed check_dic selection.\n",
      "finish in 1.089s, 139 probes kept.\n",
      "-- check region:241 4:61240000-61260000_:+_reg_, 39444 candidate probes\n",
      "--- 10 probes passed check_dic selection.\n",
      "finish in 0.826s, 1 probes kept.\n",
      "-- check region:242 4:63740000-63760000_:+_reg_, 39644 candidate probes\n",
      "--- 18996 probes passed check_dic selection.\n",
      "finish in 1.514s, 299 probes kept.\n",
      "-- check region:243 4:66240000-66260000_:+_reg_, 38812 candidate probes\n",
      "--- 13641 probes passed check_dic selection.\n",
      "finish in 1.329s, 251 probes kept.\n",
      "-- check region:244 4:68740000-68760000_:+_reg_, 39530 candidate probes\n",
      "--- 11890 probes passed check_dic selection.\n",
      "finish in 1.278s, 237 probes kept.\n",
      "-- check region:245 4:71240000-71260000_:+_reg_, 39052 candidate probes\n",
      "--- 6963 probes passed check_dic selection.\n",
      "finish in 1.035s, 154 probes kept.\n",
      "-- check region:246 4:73740000-73760000_:+_reg_, 31286 candidate probes\n",
      "--- 6575 probes passed check_dic selection.\n",
      "finish in 0.915s, 151 probes kept.\n",
      "-- check region:247 4:76240000-76260000_:+_reg_, 39580 candidate probes\n",
      "--- 12524 probes passed check_dic selection.\n",
      "finish in 1.276s, 228 probes kept.\n",
      "-- check region:248 4:78740000-78760000_:+_reg_, 37436 candidate probes\n",
      "--- 6138 probes passed check_dic selection.\n",
      "finish in 1.001s, 135 probes kept.\n",
      "-- check region:249 4:81240000-81260000_:+_reg_, 36968 candidate probes\n",
      "--- 11430 probes passed check_dic selection.\n",
      "finish in 1.248s, 211 probes kept.\n",
      "-- check region:250 4:83740000-83760000_:+_reg_, 21032 candidate probes\n",
      "--- 4499 probes passed check_dic selection.\n",
      "finish in 0.629s, 100 probes kept.\n",
      "-- check region:251 4:86240000-86260000_:+_reg_, 31052 candidate probes\n",
      "--- 10951 probes passed check_dic selection.\n",
      "finish in 1.063s, 207 probes kept.\n",
      "-- check region:252 4:88740000-88760000_:+_reg_, 37660 candidate probes\n",
      "--- 6784 probes passed check_dic selection.\n",
      "finish in 1.049s, 137 probes kept.\n",
      "-- check region:253 4:91240000-91260000_:+_reg_, 39572 candidate probes\n",
      "--- 15640 probes passed check_dic selection.\n",
      "finish in 1.406s, 283 probes kept.\n",
      "-- check region:254 4:93740000-93760000_:+_reg_, 36162 candidate probes\n",
      "--- 7160 probes passed check_dic selection.\n",
      "finish in 0.978s, 139 probes kept.\n",
      "-- check region:255 4:96240000-96260000_:+_reg_, 36694 candidate probes\n",
      "--- 5634 probes passed check_dic selection.\n",
      "finish in 0.974s, 119 probes kept.\n",
      "-- check region:256 4:98740000-98760000_:+_reg_, 38514 candidate probes\n",
      "--- 11831 probes passed check_dic selection.\n",
      "finish in 1.257s, 230 probes kept.\n",
      "-- check region:257 4:101240000-101260000_:+_reg_, 38872 candidate probes\n",
      "--- 12243 probes passed check_dic selection.\n",
      "finish in 1.326s, 231 probes kept.\n",
      "-- check region:258 4:103740000-103760000_:+_reg_, 39642 candidate probes\n",
      "--- 18499 probes passed check_dic selection.\n",
      "finish in 1.504s, 300 probes kept.\n",
      "-- check region:259 4:106240000-106260000_:+_reg_, 37368 candidate probes\n",
      "--- 11308 probes passed check_dic selection.\n",
      "finish in 1.203s, 207 probes kept.\n",
      "-- check region:260 4:108740000-108760000_:+_reg_, 39202 candidate probes\n",
      "--- 6290 probes passed check_dic selection.\n",
      "finish in 1.073s, 160 probes kept.\n",
      "-- check region:261 4:111240000-111260000_:+_reg_, 20552 candidate probes\n",
      "--- 7603 probes passed check_dic selection.\n",
      "finish in 0.745s, 137 probes kept.\n",
      "-- check region:262 4:113740000-113760000_:+_reg_, 35026 candidate probes\n",
      "--- 3071 probes passed check_dic selection.\n",
      "finish in 0.815s, 101 probes kept.\n",
      "-- check region:263 4:116240000-116260000_:+_reg_, 38952 candidate probes\n",
      "--- 9846 probes passed check_dic selection.\n",
      "finish in 1.163s, 183 probes kept.\n",
      "-- check region:264 4:118740000-118760000_:+_reg_, 38706 candidate probes\n",
      "--- 8899 probes passed check_dic selection.\n",
      "finish in 1.147s, 186 probes kept.\n",
      "-- check region:265 4:121240000-121260000_:+_reg_, 24184 candidate probes\n",
      "--- 2433 probes passed check_dic selection.\n",
      "finish in 0.613s, 43 probes kept.\n",
      "-- check region:266 4:123740000-123760000_:+_reg_, 38102 candidate probes\n",
      "--- 10964 probes passed check_dic selection.\n",
      "finish in 1.170s, 213 probes kept.\n",
      "-- check region:267 4:126240000-126260000_:+_reg_, 39602 candidate probes\n",
      "--- 20354 probes passed check_dic selection.\n",
      "finish in 1.628s, 309 probes kept.\n",
      "-- check region:268 4:128740000-128760000_:+_reg_, 39610 candidate probes\n",
      "--- 18497 probes passed check_dic selection.\n",
      "finish in 1.530s, 299 probes kept.\n",
      "-- check region:269 4:131240000-131260000_:+_reg_, 39502 candidate probes\n",
      "--- 21334 probes passed check_dic selection.\n",
      "finish in 1.660s, 306 probes kept.\n",
      "-- check region:270 4:133740000-133760000_:+_reg_, 39130 candidate probes\n",
      "--- 17152 probes passed check_dic selection.\n",
      "finish in 1.497s, 292 probes kept.\n",
      "-- check region:271 4:136240000-136260000_:+_reg_, 39182 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 19564 probes passed check_dic selection.\n",
      "finish in 1.566s, 312 probes kept.\n",
      "-- check region:272 4:138740000-138760000_:+_reg_, 39758 candidate probes\n",
      "--- 18493 probes passed check_dic selection.\n",
      "finish in 1.621s, 325 probes kept.\n",
      "-- check region:273 4:141240000-141260000_:+_reg_, 39512 candidate probes\n",
      "--- 17629 probes passed check_dic selection.\n",
      "finish in 1.549s, 300 probes kept.\n",
      "-- check region:274 4:143740000-143760000_:+_reg_, 38780 candidate probes\n",
      "--- 3914 probes passed check_dic selection.\n",
      "finish in 0.905s, 123 probes kept.\n",
      "-- check region:275 4:146240000-146260000_:+_reg_, 38838 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.850s, 0 probes kept.\n",
      "-- check region:276 4:148740000-148760000_:+_reg_, 39033 candidate probes\n",
      "--- 15971 probes passed check_dic selection.\n",
      "finish in 1.462s, 248 probes kept.\n",
      "-- check region:277 4:151240000-151260000_:+_reg_, 39804 candidate probes\n",
      "--- 21256 probes passed check_dic selection.\n",
      "finish in 1.669s, 319 probes kept.\n",
      "-- check region:278 4:153740000-153760000_:+_reg_, 37354 candidate probes\n",
      "--- 10392 probes passed check_dic selection.\n",
      "finish in 1.241s, 158 probes kept.\n",
      "-- check region:279 4:156240000-156260000_:+_reg_, 30032 candidate probes\n",
      "--- 15599 probes passed check_dic selection.\n",
      "finish in 1.238s, 255 probes kept.\n",
      "-- check region:280 5:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:281 5:3740000-3760000_:+_reg_, 37482 candidate probes\n",
      "--- 10223 probes passed check_dic selection.\n",
      "finish in 1.222s, 187 probes kept.\n",
      "-- check region:282 5:6240000-6260000_:+_reg_, 36610 candidate probes\n",
      "--- 6198 probes passed check_dic selection.\n",
      "finish in 0.971s, 141 probes kept.\n",
      "-- check region:283 5:8740000-8760000_:+_reg_, 32178 candidate probes\n",
      "--- 5282 probes passed check_dic selection.\n",
      "finish in 0.869s, 119 probes kept.\n",
      "-- check region:284 5:11240000-11260000_:+_reg_, 39678 candidate probes\n",
      "--- 44 probes passed check_dic selection.\n",
      "finish in 0.802s, 5 probes kept.\n",
      "-- check region:285 5:13740000-13760000_:+_reg_, 34978 candidate probes\n",
      "--- 3819 probes passed check_dic selection.\n",
      "finish in 0.848s, 93 probes kept.\n",
      "-- check region:286 5:16240000-16260000_:+_reg_, 38282 candidate probes\n",
      "--- 9176 probes passed check_dic selection.\n",
      "finish in 1.139s, 196 probes kept.\n",
      "-- check region:287 5:18740000-18760000_:+_reg_, 36440 candidate probes\n",
      "--- 5730 probes passed check_dic selection.\n",
      "finish in 0.969s, 134 probes kept.\n",
      "-- check region:288 5:21240000-21260000_:+_reg_, 35746 candidate probes\n",
      "--- 9028 probes passed check_dic selection.\n",
      "finish in 1.091s, 217 probes kept.\n",
      "-- check region:289 5:23740000-23760000_:+_reg_, 39374 candidate probes\n",
      "--- 13844 probes passed check_dic selection.\n",
      "finish in 1.372s, 252 probes kept.\n",
      "-- check region:290 5:26240000-26260000_:+_reg_, 29238 candidate probes\n",
      "--- 5562 probes passed check_dic selection.\n",
      "finish in 0.817s, 103 probes kept.\n",
      "-- check region:291 5:28740000-28760000_:+_reg_, 38944 candidate probes\n",
      "--- 12921 probes passed check_dic selection.\n",
      "finish in 1.297s, 229 probes kept.\n",
      "-- check region:292 5:31240000-31260000_:+_reg_, 39438 candidate probes\n",
      "--- 16232 probes passed check_dic selection.\n",
      "finish in 1.472s, 316 probes kept.\n",
      "-- check region:293 5:33740000-33760000_:+_reg_, 39020 candidate probes\n",
      "--- 12287 probes passed check_dic selection.\n",
      "finish in 1.265s, 252 probes kept.\n",
      "-- check region:294 5:36240000-36260000_:+_reg_, 28294 candidate probes\n",
      "--- 14282 probes passed check_dic selection.\n",
      "finish in 1.175s, 218 probes kept.\n",
      "-- check region:295 5:38740000-38760000_:+_reg_, 38778 candidate probes\n",
      "--- 14624 probes passed check_dic selection.\n",
      "finish in 1.385s, 257 probes kept.\n",
      "-- check region:296 5:41240000-41260000_:+_reg_, 38782 candidate probes\n",
      "--- 12681 probes passed check_dic selection.\n",
      "finish in 1.253s, 237 probes kept.\n",
      "-- check region:297 5:43740000-43760000_:+_reg_, 39860 candidate probes\n",
      "--- 10082 probes passed check_dic selection.\n",
      "finish in 1.207s, 236 probes kept.\n",
      "-- check region:298 5:46240000-46260000_:+_reg_, 30050 candidate probes\n",
      "--- 7170 probes passed check_dic selection.\n",
      "finish in 0.901s, 151 probes kept.\n",
      "-- check region:299 5:48740000-48760000_:+_reg_, 39692 candidate probes\n",
      "--- 10679 probes passed check_dic selection.\n",
      "finish in 1.236s, 212 probes kept.\n",
      "-- check region:300 5:51240000-51260000_:+_reg_, 38948 candidate probes\n",
      "--- 8839 probes passed check_dic selection.\n",
      "finish in 1.102s, 174 probes kept.\n",
      "-- check region:301 5:53740000-53760000_:+_reg_, 26712 candidate probes\n",
      "--- 6985 probes passed check_dic selection.\n",
      "finish in 0.827s, 119 probes kept.\n",
      "-- check region:302 5:56240000-56260000_:+_reg_, 27790 candidate probes\n",
      "--- 4714 probes passed check_dic selection.\n",
      "finish in 0.711s, 115 probes kept.\n",
      "-- check region:303 5:58740000-58760000_:+_reg_, 39632 candidate probes\n",
      "--- 9146 probes passed check_dic selection.\n",
      "finish in 1.161s, 184 probes kept.\n",
      "-- check region:304 5:61240000-61260000_:+_reg_, 37558 candidate probes\n",
      "--- 5472 probes passed check_dic selection.\n",
      "finish in 0.993s, 130 probes kept.\n",
      "-- check region:305 5:63740000-63760000_:+_reg_, 38692 candidate probes\n",
      "--- 11234 probes passed check_dic selection.\n",
      "finish in 1.224s, 222 probes kept.\n",
      "-- check region:306 5:66240000-66260000_:+_reg_, 37380 candidate probes\n",
      "--- 10594 probes passed check_dic selection.\n",
      "finish in 1.161s, 188 probes kept.\n",
      "-- check region:307 5:68740000-68760000_:+_reg_, 33784 candidate probes\n",
      "--- 7020 probes passed check_dic selection.\n",
      "finish in 0.950s, 149 probes kept.\n",
      "-- check region:308 5:71240000-71260000_:+_reg_, 27524 candidate probes\n",
      "--- 5255 probes passed check_dic selection.\n",
      "finish in 0.780s, 108 probes kept.\n",
      "-- check region:309 5:73740000-73760000_:+_reg_, 39580 candidate probes\n",
      "--- 20793 probes passed check_dic selection.\n",
      "finish in 1.657s, 324 probes kept.\n",
      "-- check region:310 5:76240000-76260000_:+_reg_, 38770 candidate probes\n",
      "--- 9765 probes passed check_dic selection.\n",
      "finish in 1.181s, 205 probes kept.\n",
      "-- check region:311 5:78740000-78760000_:+_reg_, 39662 candidate probes\n",
      "--- 9309 probes passed check_dic selection.\n",
      "finish in 1.159s, 189 probes kept.\n",
      "-- check region:312 5:81240000-81260000_:+_reg_, 39353 candidate probes\n",
      "--- 9774 probes passed check_dic selection.\n",
      "finish in 1.158s, 204 probes kept.\n",
      "-- check region:313 5:83740000-83760000_:+_reg_, 39104 candidate probes\n",
      "--- 5665 probes passed check_dic selection.\n",
      "finish in 1.036s, 135 probes kept.\n",
      "-- check region:314 5:86240000-86260000_:+_reg_, 38228 candidate probes\n",
      "--- 10724 probes passed check_dic selection.\n",
      "finish in 1.240s, 210 probes kept.\n",
      "-- check region:315 5:88740000-88760000_:+_reg_, 39134 candidate probes\n",
      "--- 6479 probes passed check_dic selection.\n",
      "finish in 1.063s, 183 probes kept.\n",
      "-- check region:316 5:91240000-91260000_:+_reg_, 34506 candidate probes\n",
      "--- 9779 probes passed check_dic selection.\n",
      "finish in 1.115s, 192 probes kept.\n",
      "-- check region:317 5:93740000-93760000_:+_reg_, 36388 candidate probes\n",
      "--- 60 probes passed check_dic selection.\n",
      "finish in 0.720s, 7 probes kept.\n",
      "-- check region:318 5:96240000-96260000_:+_reg_, 39258 candidate probes\n",
      "--- 12326 probes passed check_dic selection.\n",
      "finish in 1.284s, 247 probes kept.\n",
      "-- check region:319 5:98740000-98760000_:+_reg_, 37214 candidate probes\n",
      "--- 10935 probes passed check_dic selection.\n",
      "finish in 1.238s, 205 probes kept.\n",
      "-- check region:320 5:101240000-101260000_:+_reg_, 36586 candidate probes\n",
      "--- 11272 probes passed check_dic selection.\n",
      "finish in 1.193s, 200 probes kept.\n",
      "-- check region:321 5:103740000-103760000_:+_reg_, 39028 candidate probes\n",
      "--- 16598 probes passed check_dic selection.\n",
      "finish in 1.462s, 274 probes kept.\n",
      "-- check region:322 5:106240000-106260000_:+_reg_, 38436 candidate probes\n",
      "--- 10981 probes passed check_dic selection.\n",
      "finish in 1.236s, 210 probes kept.\n",
      "-- check region:323 5:108740000-108760000_:+_reg_, 38992 candidate probes\n",
      "--- 14333 probes passed check_dic selection.\n",
      "finish in 1.312s, 242 probes kept.\n",
      "-- check region:324 5:111240000-111260000_:+_reg_, 38670 candidate probes\n",
      "--- 19589 probes passed check_dic selection.\n",
      "finish in 1.562s, 303 probes kept.\n",
      "-- check region:325 5:113740000-113760000_:+_reg_, 39892 candidate probes\n",
      "--- 19501 probes passed check_dic selection.\n",
      "finish in 1.638s, 337 probes kept.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- check region:326 5:116240000-116260000_:+_reg_, 39208 candidate probes\n",
      "--- 14796 probes passed check_dic selection.\n",
      "finish in 1.405s, 237 probes kept.\n",
      "-- check region:327 5:118740000-118760000_:+_reg_, 39518 candidate probes\n",
      "--- 19700 probes passed check_dic selection.\n",
      "finish in 1.596s, 343 probes kept.\n",
      "-- check region:328 5:121240000-121260000_:+_reg_, 39350 candidate probes\n",
      "--- 13800 probes passed check_dic selection.\n",
      "finish in 1.356s, 248 probes kept.\n",
      "-- check region:329 5:123740000-123760000_:+_reg_, 38970 candidate probes\n",
      "--- 11394 probes passed check_dic selection.\n",
      "finish in 1.235s, 237 probes kept.\n",
      "-- check region:330 5:126240000-126260000_:+_reg_, 39528 candidate probes\n",
      "--- 16073 probes passed check_dic selection.\n",
      "finish in 1.416s, 284 probes kept.\n",
      "-- check region:331 5:128740000-128760000_:+_reg_, 39762 candidate probes\n",
      "--- 19955 probes passed check_dic selection.\n",
      "finish in 1.648s, 341 probes kept.\n",
      "-- check region:332 5:131240000-131260000_:+_reg_, 39198 candidate probes\n",
      "--- 11993 probes passed check_dic selection.\n",
      "finish in 1.312s, 235 probes kept.\n",
      "-- check region:333 5:133740000-133760000_:+_reg_, 39732 candidate probes\n",
      "--- 14050 probes passed check_dic selection.\n",
      "finish in 1.391s, 243 probes kept.\n",
      "-- check region:334 5:136240000-136260000_:+_reg_, 38676 candidate probes\n",
      "--- 17576 probes passed check_dic selection.\n",
      "finish in 1.530s, 284 probes kept.\n",
      "-- check region:335 5:138740000-138760000_:+_reg_, 38796 candidate probes\n",
      "--- 19032 probes passed check_dic selection.\n",
      "finish in 1.583s, 318 probes kept.\n",
      "-- check region:336 5:141240000-141260000_:+_reg_, 39768 candidate probes\n",
      "--- 16609 probes passed check_dic selection.\n",
      "finish in 1.435s, 285 probes kept.\n",
      "-- check region:337 5:143740000-143760000_:+_reg_, 39140 candidate probes\n",
      "--- 17586 probes passed check_dic selection.\n",
      "finish in 1.503s, 281 probes kept.\n",
      "-- check region:338 5:146240000-146260000_:+_reg_, 39232 candidate probes\n",
      "--- 14293 probes passed check_dic selection.\n",
      "finish in 1.356s, 259 probes kept.\n",
      "-- check region:339 5:148740000-148760000_:+_reg_, 39746 candidate probes\n",
      "--- 17812 probes passed check_dic selection.\n",
      "finish in 1.537s, 288 probes kept.\n",
      "-- check region:340 5:151240000-151260000_:+_reg_, 36504 candidate probes\n",
      "--- 14305 probes passed check_dic selection.\n",
      "finish in 1.335s, 249 probes kept.\n",
      "-- check region:341 6:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:342 6:3740000-3760000_:+_reg_, 39756 candidate probes\n",
      "--- 14001 probes passed check_dic selection.\n",
      "finish in 1.338s, 258 probes kept.\n",
      "-- check region:343 6:6240000-6260000_:+_reg_, 38276 candidate probes\n",
      "--- 11996 probes passed check_dic selection.\n",
      "finish in 1.257s, 219 probes kept.\n",
      "-- check region:344 6:8740000-8760000_:+_reg_, 39142 candidate probes\n",
      "--- 14891 probes passed check_dic selection.\n",
      "finish in 1.323s, 261 probes kept.\n",
      "-- check region:345 6:11240000-11260000_:+_reg_, 31716 candidate probes\n",
      "--- 8716 probes passed check_dic selection.\n",
      "finish in 1.011s, 150 probes kept.\n",
      "-- check region:346 6:13740000-13760000_:+_reg_, 35670 candidate probes\n",
      "--- 8626 probes passed check_dic selection.\n",
      "finish in 1.066s, 182 probes kept.\n",
      "-- check region:347 6:16240000-16260000_:+_reg_, 33354 candidate probes\n",
      "--- 2021 probes passed check_dic selection.\n",
      "finish in 0.741s, 53 probes kept.\n",
      "-- check region:348 6:18740000-18760000_:+_reg_, 33744 candidate probes\n",
      "--- 6484 probes passed check_dic selection.\n",
      "finish in 0.948s, 139 probes kept.\n",
      "-- check region:349 6:21240000-21260000_:+_reg_, 38798 candidate probes\n",
      "--- 9397 probes passed check_dic selection.\n",
      "finish in 1.152s, 188 probes kept.\n",
      "-- check region:350 6:23740000-23760000_:+_reg_, 39746 candidate probes\n",
      "--- 12842 probes passed check_dic selection.\n",
      "finish in 1.331s, 239 probes kept.\n",
      "-- check region:351 6:26240000-26260000_:+_reg_, 27066 candidate probes\n",
      "--- 2842 probes passed check_dic selection.\n",
      "finish in 0.652s, 69 probes kept.\n",
      "-- check region:352 6:28740000-28760000_:+_reg_, 39736 candidate probes\n",
      "--- 20414 probes passed check_dic selection.\n",
      "finish in 1.624s, 318 probes kept.\n",
      "-- check region:353 6:31240000-31260000_:+_reg_, 39424 candidate probes\n",
      "--- 16343 probes passed check_dic selection.\n",
      "finish in 1.441s, 268 probes kept.\n",
      "-- check region:354 6:33740000-33760000_:+_reg_, 32084 candidate probes\n",
      "--- 9741 probes passed check_dic selection.\n",
      "finish in 1.081s, 175 probes kept.\n",
      "-- check region:355 6:36240000-36260000_:+_reg_, 39258 candidate probes\n",
      "--- 10398 probes passed check_dic selection.\n",
      "finish in 1.206s, 209 probes kept.\n",
      "-- check region:356 6:38740000-38760000_:+_reg_, 39506 candidate probes\n",
      "--- 19124 probes passed check_dic selection.\n",
      "finish in 1.533s, 317 probes kept.\n",
      "-- check region:357 6:41240000-41260000_:+_reg_, 30974 candidate probes\n",
      "--- 6797 probes passed check_dic selection.\n",
      "finish in 0.875s, 138 probes kept.\n",
      "-- check region:358 6:43740000-43760000_:+_reg_, 31220 candidate probes\n",
      "--- 7631 probes passed check_dic selection.\n",
      "finish in 0.957s, 158 probes kept.\n",
      "-- check region:359 6:46240000-46260000_:+_reg_, 39386 candidate probes\n",
      "--- 11327 probes passed check_dic selection.\n",
      "finish in 1.304s, 227 probes kept.\n",
      "-- check region:360 6:48740000-48760000_:+_reg_, 37632 candidate probes\n",
      "--- 9639 probes passed check_dic selection.\n",
      "finish in 1.173s, 221 probes kept.\n",
      "-- check region:361 6:51240000-51260000_:+_reg_, 39752 candidate probes\n",
      "--- 12778 probes passed check_dic selection.\n",
      "finish in 1.305s, 235 probes kept.\n",
      "-- check region:362 6:53740000-53760000_:+_reg_, 38826 candidate probes\n",
      "--- 15195 probes passed check_dic selection.\n",
      "finish in 1.369s, 260 probes kept.\n",
      "-- check region:363 6:56240000-56260000_:+_reg_, 21986 candidate probes\n",
      "--- 6635 probes passed check_dic selection.\n",
      "finish in 0.747s, 116 probes kept.\n",
      "-- check region:364 6:58740000-58760000_:+_reg_, 38662 candidate probes\n",
      "--- 6994 probes passed check_dic selection.\n",
      "finish in 1.041s, 163 probes kept.\n",
      "-- check region:365 6:61240000-61260000_:+_reg_, 37082 candidate probes\n",
      "--- 11683 probes passed check_dic selection.\n",
      "finish in 1.248s, 203 probes kept.\n",
      "-- check region:366 6:63740000-63760000_:+_reg_, 38798 candidate probes\n",
      "--- 8727 probes passed check_dic selection.\n",
      "finish in 1.143s, 195 probes kept.\n",
      "-- check region:367 6:66240000-66260000_:+_reg_, 38568 candidate probes\n",
      "--- 5280 probes passed check_dic selection.\n",
      "finish in 1.007s, 133 probes kept.\n",
      "-- check region:368 6:68740000-68760000_:+_reg_, 38030 candidate probes\n",
      "--- 5194 probes passed check_dic selection.\n",
      "finish in 0.962s, 129 probes kept.\n",
      "-- check region:369 6:71240000-71260000_:+_reg_, 39460 candidate probes\n",
      "--- 11714 probes passed check_dic selection.\n",
      "finish in 1.308s, 226 probes kept.\n",
      "-- check region:370 6:73740000-73760000_:+_reg_, 35272 candidate probes\n",
      "--- 8100 probes passed check_dic selection.\n",
      "finish in 1.040s, 175 probes kept.\n",
      "-- check region:371 6:76240000-76260000_:+_reg_, 38060 candidate probes\n",
      "--- 4553 probes passed check_dic selection.\n",
      "finish in 0.961s, 123 probes kept.\n",
      "-- check region:372 6:78740000-78760000_:+_reg_, 39162 candidate probes\n",
      "--- 9479 probes passed check_dic selection.\n",
      "finish in 1.202s, 210 probes kept.\n",
      "-- check region:373 6:81240000-81260000_:+_reg_, 39344 candidate probes\n",
      "--- 10897 probes passed check_dic selection.\n",
      "finish in 1.201s, 235 probes kept.\n",
      "-- check region:374 6:83740000-83760000_:+_reg_, 39348 candidate probes\n",
      "--- 17941 probes passed check_dic selection.\n",
      "finish in 1.503s, 305 probes kept.\n",
      "-- check region:375 6:86240000-86260000_:+_reg_, 39254 candidate probes\n",
      "--- 17099 probes passed check_dic selection.\n",
      "finish in 1.530s, 285 probes kept.\n",
      "-- check region:376 6:88740000-88760000_:+_reg_, 39582 candidate probes\n",
      "--- 21126 probes passed check_dic selection.\n",
      "finish in 1.621s, 315 probes kept.\n",
      "-- check region:377 6:91240000-91260000_:+_reg_, 39692 candidate probes\n",
      "--- 20317 probes passed check_dic selection.\n",
      "finish in 1.653s, 326 probes kept.\n",
      "-- check region:378 6:93740000-93760000_:+_reg_, 39374 candidate probes\n",
      "--- 15839 probes passed check_dic selection.\n",
      "finish in 1.434s, 270 probes kept.\n",
      "-- check region:379 6:96240000-96260000_:+_reg_, 39904 candidate probes\n",
      "--- 18037 probes passed check_dic selection.\n",
      "finish in 1.486s, 305 probes kept.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- check region:380 6:98740000-98760000_:+_reg_, 39548 candidate probes\n",
      "--- 15119 probes passed check_dic selection.\n",
      "finish in 1.408s, 272 probes kept.\n",
      "-- check region:381 6:101240000-101260000_:+_reg_, 39866 candidate probes\n",
      "--- 20885 probes passed check_dic selection.\n",
      "finish in 1.619s, 327 probes kept.\n",
      "-- check region:382 6:103740000-103760000_:+_reg_, 37184 candidate probes\n",
      "--- 13006 probes passed check_dic selection.\n",
      "finish in 1.336s, 242 probes kept.\n",
      "-- check region:383 6:106240000-106260000_:+_reg_, 39578 candidate probes\n",
      "--- 12097 probes passed check_dic selection.\n",
      "finish in 1.268s, 233 probes kept.\n",
      "-- check region:384 6:108740000-108760000_:+_reg_, 38092 candidate probes\n",
      "--- 11937 probes passed check_dic selection.\n",
      "finish in 1.233s, 212 probes kept.\n",
      "-- check region:385 6:111240000-111260000_:+_reg_, 39596 candidate probes\n",
      "--- 11437 probes passed check_dic selection.\n",
      "finish in 1.286s, 230 probes kept.\n",
      "-- check region:386 6:113740000-113760000_:+_reg_, 39838 candidate probes\n",
      "--- 18407 probes passed check_dic selection.\n",
      "finish in 1.590s, 329 probes kept.\n",
      "-- check region:387 6:116240000-116260000_:+_reg_, 37216 candidate probes\n",
      "--- 14394 probes passed check_dic selection.\n",
      "finish in 1.387s, 269 probes kept.\n",
      "-- check region:388 6:118740000-118760000_:+_reg_, 39074 candidate probes\n",
      "--- 21305 probes passed check_dic selection.\n",
      "finish in 1.621s, 323 probes kept.\n",
      "-- check region:389 6:121240000-121260000_:+_reg_, 39450 candidate probes\n",
      "--- 15248 probes passed check_dic selection.\n",
      "finish in 1.445s, 266 probes kept.\n",
      "-- check region:390 6:123740000-123760000_:+_reg_, 31494 candidate probes\n",
      "--- 3338 probes passed check_dic selection.\n",
      "finish in 0.768s, 103 probes kept.\n",
      "-- check region:391 6:126240000-126260000_:+_reg_, 36372 candidate probes\n",
      "--- 8098 probes passed check_dic selection.\n",
      "finish in 1.052s, 153 probes kept.\n",
      "-- check region:392 6:128740000-128760000_:+_reg_, 37802 candidate probes\n",
      "--- 4234 probes passed check_dic selection.\n",
      "finish in 0.924s, 114 probes kept.\n",
      "-- check region:393 6:131240000-131260000_:+_reg_, 33768 candidate probes\n",
      "--- 4210 probes passed check_dic selection.\n",
      "finish in 0.883s, 93 probes kept.\n",
      "-- check region:394 6:133740000-133760000_:+_reg_, 25998 candidate probes\n",
      "--- 2160 probes passed check_dic selection.\n",
      "finish in 0.647s, 48 probes kept.\n",
      "-- check region:395 6:136240000-136260000_:+_reg_, 39118 candidate probes\n",
      "--- 14467 probes passed check_dic selection.\n",
      "finish in 1.372s, 260 probes kept.\n",
      "-- check region:396 6:138740000-138760000_:+_reg_, 29152 candidate probes\n",
      "--- 7382 probes passed check_dic selection.\n",
      "finish in 0.918s, 127 probes kept.\n",
      "-- check region:397 6:141240000-141260000_:+_reg_, 39544 candidate probes\n",
      "--- 15780 probes passed check_dic selection.\n",
      "finish in 1.440s, 268 probes kept.\n",
      "-- check region:398 6:143740000-143760000_:+_reg_, 39814 candidate probes\n",
      "--- 15647 probes passed check_dic selection.\n",
      "finish in 1.396s, 274 probes kept.\n",
      "-- check region:399 6:146240000-146260000_:+_reg_, 38782 candidate probes\n",
      "--- 14475 probes passed check_dic selection.\n",
      "finish in 1.403s, 252 probes kept.\n",
      "-- check region:400 6:148740000-148760000_:+_reg_, 39414 candidate probes\n",
      "--- 13063 probes passed check_dic selection.\n",
      "finish in 1.339s, 245 probes kept.\n",
      "-- check region:401 7:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:402 7:3740000-3760000_:+_reg_, 31666 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.637s, 0 probes kept.\n",
      "-- check region:403 7:6240000-6260000_:+_reg_, 35268 candidate probes\n",
      "--- 10139 probes passed check_dic selection.\n",
      "finish in 1.129s, 172 probes kept.\n",
      "-- check region:404 7:8740000-8760000_:+_reg_, 39460 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.791s, 0 probes kept.\n",
      "-- check region:405 7:11240000-11260000_:+_reg_, 39370 candidate probes\n",
      "--- 343 probes passed check_dic selection.\n",
      "finish in 0.794s, 20 probes kept.\n",
      "-- check region:406 7:13740000-13760000_:+_reg_, 33890 candidate probes\n",
      "--- 435 probes passed check_dic selection.\n",
      "finish in 0.725s, 22 probes kept.\n",
      "-- check region:407 7:16240000-16260000_:+_reg_, 38179 candidate probes\n",
      "--- 10444 probes passed check_dic selection.\n",
      "finish in 1.222s, 211 probes kept.\n",
      "-- check region:408 7:18740000-18760000_:+_reg_, 38804 candidate probes\n",
      "--- 2591 probes passed check_dic selection.\n",
      "finish in 0.859s, 76 probes kept.\n",
      "-- check region:409 7:21240000-21260000_:+_reg_, 28148 candidate probes\n",
      "--- 14 probes passed check_dic selection.\n",
      "finish in 0.587s, 1 probes kept.\n",
      "-- check region:410 7:23740000-23760000_:+_reg_, 35616 candidate probes\n",
      "--- 472 probes passed check_dic selection.\n",
      "finish in 0.715s, 26 probes kept.\n",
      "-- check region:411 7:26240000-26260000_:+_reg_, 37140 candidate probes\n",
      "--- 5118 probes passed check_dic selection.\n",
      "finish in 0.944s, 126 probes kept.\n",
      "-- check region:412 7:28740000-28760000_:+_reg_, 38530 candidate probes\n",
      "--- 14584 probes passed check_dic selection.\n",
      "finish in 1.376s, 261 probes kept.\n",
      "-- check region:413 7:31240000-31260000_:+_reg_, 36442 candidate probes\n",
      "--- 4310 probes passed check_dic selection.\n",
      "finish in 0.899s, 98 probes kept.\n",
      "-- check region:414 7:33740000-33760000_:+_reg_, 27814 candidate probes\n",
      "--- 3665 probes passed check_dic selection.\n",
      "finish in 0.712s, 96 probes kept.\n",
      "-- check region:415 7:36240000-36260000_:+_reg_, 39384 candidate probes\n",
      "--- 14313 probes passed check_dic selection.\n",
      "finish in 1.337s, 250 probes kept.\n",
      "-- check region:416 7:38740000-38760000_:+_reg_, 37736 candidate probes\n",
      "--- 102 probes passed check_dic selection.\n",
      "finish in 0.755s, 3 probes kept.\n",
      "-- check region:417 7:41240000-41260000_:+_reg_, 36894 candidate probes\n",
      "--- 728 probes passed check_dic selection.\n",
      "finish in 0.768s, 42 probes kept.\n",
      "-- check region:418 7:43740000-43760000_:+_reg_, 37924 candidate probes\n",
      "--- 4539 probes passed check_dic selection.\n",
      "finish in 0.984s, 133 probes kept.\n",
      "-- check region:419 7:46240000-46260000_:+_reg_, 39672 candidate probes\n",
      "--- 19319 probes passed check_dic selection.\n",
      "finish in 1.599s, 309 probes kept.\n",
      "-- check region:420 7:48740000-48760000_:+_reg_, 37576 candidate probes\n",
      "--- 10499 probes passed check_dic selection.\n",
      "finish in 1.246s, 169 probes kept.\n",
      "-- check region:421 7:51240000-51260000_:+_reg_, 38908 candidate probes\n",
      "--- 12148 probes passed check_dic selection.\n",
      "finish in 1.265s, 232 probes kept.\n",
      "-- check region:422 7:53740000-53760000_:+_reg_, 39328 candidate probes\n",
      "--- 11383 probes passed check_dic selection.\n",
      "finish in 1.284s, 221 probes kept.\n",
      "-- check region:423 7:56240000-56260000_:+_reg_, 39352 candidate probes\n",
      "--- 12540 probes passed check_dic selection.\n",
      "finish in 1.341s, 235 probes kept.\n",
      "-- check region:424 7:58740000-58760000_:+_reg_, 39890 candidate probes\n",
      "--- 17925 probes passed check_dic selection.\n",
      "finish in 1.471s, 294 probes kept.\n",
      "-- check region:425 7:61240000-61260000_:+_reg_, 33576 candidate probes\n",
      "--- 2830 probes passed check_dic selection.\n",
      "finish in 0.798s, 75 probes kept.\n",
      "-- check region:426 7:63740000-63760000_:+_reg_, 38998 candidate probes\n",
      "--- 19469 probes passed check_dic selection.\n",
      "finish in 1.559s, 318 probes kept.\n",
      "-- check region:427 7:66240000-66260000_:+_reg_, 39066 candidate probes\n",
      "--- 18358 probes passed check_dic selection.\n",
      "finish in 1.538s, 293 probes kept.\n",
      "-- check region:428 7:68740000-68760000_:+_reg_, 39010 candidate probes\n",
      "--- 13984 probes passed check_dic selection.\n",
      "finish in 1.388s, 268 probes kept.\n",
      "-- check region:429 7:71240000-71260000_:+_reg_, 39086 candidate probes\n",
      "--- 13150 probes passed check_dic selection.\n",
      "finish in 1.284s, 246 probes kept.\n",
      "-- check region:430 7:73740000-73760000_:+_reg_, 39816 candidate probes\n",
      "--- 17313 probes passed check_dic selection.\n",
      "finish in 1.534s, 298 probes kept.\n",
      "-- check region:431 7:76240000-76260000_:+_reg_, 31834 candidate probes\n",
      "--- 4183 probes passed check_dic selection.\n",
      "finish in 0.855s, 80 probes kept.\n",
      "-- check region:432 7:78740000-78760000_:+_reg_, 38124 candidate probes\n",
      "--- 18316 probes passed check_dic selection.\n",
      "finish in 1.483s, 278 probes kept.\n",
      "-- check region:433 7:81240000-81260000_:+_reg_, 38988 candidate probes\n",
      "--- 11240 probes passed check_dic selection.\n",
      "finish in 1.262s, 219 probes kept.\n",
      "-- check region:434 7:83740000-83760000_:+_reg_, 39066 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 15211 probes passed check_dic selection.\n",
      "finish in 1.395s, 243 probes kept.\n",
      "-- check region:435 7:86240000-86260000_:+_reg_, 37798 candidate probes\n",
      "--- 5469 probes passed check_dic selection.\n",
      "finish in 1.029s, 139 probes kept.\n",
      "-- check region:436 7:88740000-88760000_:+_reg_, 39726 candidate probes\n",
      "--- 7516 probes passed check_dic selection.\n",
      "finish in 1.148s, 167 probes kept.\n",
      "-- check region:437 7:91240000-91260000_:+_reg_, 38262 candidate probes\n",
      "--- 11447 probes passed check_dic selection.\n",
      "finish in 1.206s, 212 probes kept.\n",
      "-- check region:438 7:93740000-93760000_:+_reg_, 25968 candidate probes\n",
      "--- 2738 probes passed check_dic selection.\n",
      "finish in 0.642s, 70 probes kept.\n",
      "-- check region:439 7:96240000-96260000_:+_reg_, 39722 candidate probes\n",
      "--- 19058 probes passed check_dic selection.\n",
      "finish in 1.556s, 315 probes kept.\n",
      "-- check region:440 7:98740000-98760000_:+_reg_, 39384 candidate probes\n",
      "--- 21665 probes passed check_dic selection.\n",
      "finish in 1.682s, 327 probes kept.\n",
      "-- check region:441 7:101240000-101260000_:+_reg_, 38202 candidate probes\n",
      "--- 6309 probes passed check_dic selection.\n",
      "finish in 1.021s, 163 probes kept.\n",
      "-- check region:442 7:103740000-103760000_:+_reg_, 39528 candidate probes\n",
      "--- 6543 probes passed check_dic selection.\n",
      "finish in 1.021s, 174 probes kept.\n",
      "-- check region:443 7:106240000-106260000_:+_reg_, 34442 candidate probes\n",
      "--- 4311 probes passed check_dic selection.\n",
      "finish in 0.868s, 98 probes kept.\n",
      "-- check region:444 7:108740000-108760000_:+_reg_, 38610 candidate probes\n",
      "--- 7159 probes passed check_dic selection.\n",
      "finish in 1.075s, 159 probes kept.\n",
      "-- check region:445 7:111240000-111260000_:+_reg_, 38380 candidate probes\n",
      "--- 13810 probes passed check_dic selection.\n",
      "finish in 1.297s, 242 probes kept.\n",
      "-- check region:446 7:113740000-113760000_:+_reg_, 39712 candidate probes\n",
      "--- 18389 probes passed check_dic selection.\n",
      "finish in 1.527s, 304 probes kept.\n",
      "-- check region:447 7:116240000-116260000_:+_reg_, 39456 candidate probes\n",
      "--- 16142 probes passed check_dic selection.\n",
      "finish in 1.442s, 268 probes kept.\n",
      "-- check region:448 7:118740000-118760000_:+_reg_, 38784 candidate probes\n",
      "--- 15517 probes passed check_dic selection.\n",
      "finish in 1.368s, 288 probes kept.\n",
      "-- check region:449 7:121240000-121260000_:+_reg_, 39822 candidate probes\n",
      "--- 16754 probes passed check_dic selection.\n",
      "finish in 1.475s, 272 probes kept.\n",
      "-- check region:450 7:123740000-123760000_:+_reg_, 37772 candidate probes\n",
      "--- 13225 probes passed check_dic selection.\n",
      "finish in 1.280s, 231 probes kept.\n",
      "-- check region:451 7:126240000-126260000_:+_reg_, 38602 candidate probes\n",
      "--- 14094 probes passed check_dic selection.\n",
      "finish in 1.345s, 240 probes kept.\n",
      "-- check region:452 7:128740000-128760000_:+_reg_, 39486 candidate probes\n",
      "--- 10395 probes passed check_dic selection.\n",
      "finish in 1.225s, 230 probes kept.\n",
      "-- check region:453 7:131240000-131260000_:+_reg_, 38772 candidate probes\n",
      "--- 14031 probes passed check_dic selection.\n",
      "finish in 1.337s, 254 probes kept.\n",
      "-- check region:454 7:133740000-133760000_:+_reg_, 39502 candidate probes\n",
      "--- 14844 probes passed check_dic selection.\n",
      "finish in 1.396s, 266 probes kept.\n",
      "-- check region:455 7:136240000-136260000_:+_reg_, 37126 candidate probes\n",
      "--- 10227 probes passed check_dic selection.\n",
      "finish in 1.129s, 190 probes kept.\n",
      "-- check region:456 7:138740000-138760000_:+_reg_, 39304 candidate probes\n",
      "--- 19634 probes passed check_dic selection.\n",
      "finish in 1.651s, 314 probes kept.\n",
      "-- check region:457 7:141240000-141260000_:+_reg_, 38852 candidate probes\n",
      "--- 13318 probes passed check_dic selection.\n",
      "finish in 1.336s, 242 probes kept.\n",
      "-- check region:458 7:143740000-143760000_:+_reg_, 39674 candidate probes\n",
      "--- 18848 probes passed check_dic selection.\n",
      "finish in 1.529s, 292 probes kept.\n",
      "-- check region:459 7:146240000-145441459_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:460 8:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.002s, 0 probes kept.\n",
      "-- check region:461 8:3740000-3760000_:+_reg_, 31622 candidate probes\n",
      "--- 8384 probes passed check_dic selection.\n",
      "finish in 0.974s, 179 probes kept.\n",
      "-- check region:462 8:6240000-6260000_:+_reg_, 38810 candidate probes\n",
      "--- 5790 probes passed check_dic selection.\n",
      "finish in 1.022s, 132 probes kept.\n",
      "-- check region:463 8:8740000-8760000_:+_reg_, 37610 candidate probes\n",
      "--- 15882 probes passed check_dic selection.\n",
      "finish in 1.394s, 269 probes kept.\n",
      "-- check region:464 8:11240000-11260000_:+_reg_, 39858 candidate probes\n",
      "--- 19512 probes passed check_dic selection.\n",
      "finish in 1.617s, 321 probes kept.\n",
      "-- check region:465 8:13740000-13760000_:+_reg_, 39364 candidate probes\n",
      "--- 12755 probes passed check_dic selection.\n",
      "finish in 1.344s, 249 probes kept.\n",
      "-- check region:466 8:16240000-16260000_:+_reg_, 39610 candidate probes\n",
      "--- 17926 probes passed check_dic selection.\n",
      "finish in 1.494s, 295 probes kept.\n",
      "-- check region:467 8:18740000-18760000_:+_reg_, 39920 candidate probes\n",
      "--- 21531 probes passed check_dic selection.\n",
      "finish in 1.645s, 336 probes kept.\n",
      "-- check region:468 8:21240000-21260000_:+_reg_, 34938 candidate probes\n",
      "--- 158 probes passed check_dic selection.\n",
      "finish in 0.683s, 9 probes kept.\n",
      "-- check region:469 8:23740000-23760000_:+_reg_, 39396 candidate probes\n",
      "--- 16157 probes passed check_dic selection.\n",
      "finish in 1.410s, 266 probes kept.\n",
      "-- check region:470 8:26240000-26260000_:+_reg_, 39394 candidate probes\n",
      "--- 13326 probes passed check_dic selection.\n",
      "finish in 1.338s, 241 probes kept.\n",
      "-- check region:471 8:28740000-28760000_:+_reg_, 39458 candidate probes\n",
      "--- 16732 probes passed check_dic selection.\n",
      "finish in 1.495s, 287 probes kept.\n",
      "-- check region:472 8:31240000-31260000_:+_reg_, 38638 candidate probes\n",
      "--- 12306 probes passed check_dic selection.\n",
      "finish in 1.286s, 227 probes kept.\n",
      "-- check region:473 8:33740000-33760000_:+_reg_, 38980 candidate probes\n",
      "--- 7882 probes passed check_dic selection.\n",
      "finish in 1.087s, 182 probes kept.\n",
      "-- check region:474 8:36240000-36260000_:+_reg_, 39306 candidate probes\n",
      "--- 16958 probes passed check_dic selection.\n",
      "finish in 1.487s, 283 probes kept.\n",
      "-- check region:475 8:38740000-38760000_:+_reg_, 36510 candidate probes\n",
      "--- 7494 probes passed check_dic selection.\n",
      "finish in 1.058s, 160 probes kept.\n",
      "-- check region:476 8:41240000-41260000_:+_reg_, 39536 candidate probes\n",
      "--- 10711 probes passed check_dic selection.\n",
      "finish in 1.217s, 220 probes kept.\n",
      "-- check region:477 8:43740000-43760000_:+_reg_, 37863 candidate probes\n",
      "--- 7272 probes passed check_dic selection.\n",
      "finish in 1.041s, 174 probes kept.\n",
      "-- check region:478 8:46240000-46260000_:+_reg_, 38694 candidate probes\n",
      "--- 13732 probes passed check_dic selection.\n",
      "finish in 1.324s, 240 probes kept.\n",
      "-- check region:479 8:48740000-48760000_:+_reg_, 39742 candidate probes\n",
      "--- 18452 probes passed check_dic selection.\n",
      "finish in 1.548s, 286 probes kept.\n",
      "-- check region:480 8:51240000-51260000_:+_reg_, 39552 candidate probes\n",
      "--- 7396 probes passed check_dic selection.\n",
      "finish in 1.068s, 178 probes kept.\n",
      "-- check region:481 8:53740000-53760000_:+_reg_, 38396 candidate probes\n",
      "--- 7825 probes passed check_dic selection.\n",
      "finish in 1.079s, 162 probes kept.\n",
      "-- check region:482 8:56240000-56260000_:+_reg_, 37454 candidate probes\n",
      "--- 4763 probes passed check_dic selection.\n",
      "finish in 1.017s, 113 probes kept.\n",
      "-- check region:483 8:58740000-58760000_:+_reg_, 38962 candidate probes\n",
      "--- 6453 probes passed check_dic selection.\n",
      "finish in 1.050s, 150 probes kept.\n",
      "-- check region:484 8:61240000-61260000_:+_reg_, 39858 candidate probes\n",
      "--- 19051 probes passed check_dic selection.\n",
      "finish in 1.512s, 307 probes kept.\n",
      "-- check region:485 8:63740000-63760000_:+_reg_, 28148 candidate probes\n",
      "--- 5272 probes passed check_dic selection.\n",
      "finish in 0.786s, 118 probes kept.\n",
      "-- check region:486 8:66240000-66260000_:+_reg_, 36540 candidate probes\n",
      "--- 6267 probes passed check_dic selection.\n",
      "finish in 1.040s, 140 probes kept.\n",
      "-- check region:487 8:68740000-68760000_:+_reg_, 32068 candidate probes\n",
      "--- 5397 probes passed check_dic selection.\n",
      "finish in 0.833s, 108 probes kept.\n",
      "-- check region:488 8:71240000-71260000_:+_reg_, 37552 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 8088 probes passed check_dic selection.\n",
      "finish in 1.091s, 160 probes kept.\n",
      "-- check region:489 8:73740000-73760000_:+_reg_, 37590 candidate probes\n",
      "--- 6866 probes passed check_dic selection.\n",
      "finish in 1.013s, 144 probes kept.\n",
      "-- check region:490 8:76240000-76260000_:+_reg_, 39146 candidate probes\n",
      "--- 9140 probes passed check_dic selection.\n",
      "finish in 1.170s, 183 probes kept.\n",
      "-- check region:491 8:78740000-78760000_:+_reg_, 37704 candidate probes\n",
      "--- 16163 probes passed check_dic selection.\n",
      "finish in 1.417s, 282 probes kept.\n",
      "-- check region:492 8:81240000-81260000_:+_reg_, 36426 candidate probes\n",
      "--- 8649 probes passed check_dic selection.\n",
      "finish in 1.130s, 157 probes kept.\n",
      "-- check region:493 8:83740000-83760000_:+_reg_, 38700 candidate probes\n",
      "--- 12718 probes passed check_dic selection.\n",
      "finish in 1.315s, 227 probes kept.\n",
      "-- check region:494 8:86240000-86260000_:+_reg_, 30778 candidate probes\n",
      "--- 11118 probes passed check_dic selection.\n",
      "finish in 1.107s, 182 probes kept.\n",
      "-- check region:495 8:88740000-88760000_:+_reg_, 39456 candidate probes\n",
      "--- 15289 probes passed check_dic selection.\n",
      "finish in 1.367s, 291 probes kept.\n",
      "-- check region:496 8:91240000-91260000_:+_reg_, 39582 candidate probes\n",
      "--- 10242 probes passed check_dic selection.\n",
      "finish in 1.202s, 203 probes kept.\n",
      "-- check region:497 8:93740000-93760000_:+_reg_, 39774 candidate probes\n",
      "--- 15197 probes passed check_dic selection.\n",
      "finish in 1.424s, 259 probes kept.\n",
      "-- check region:498 8:96240000-96260000_:+_reg_, 36594 candidate probes\n",
      "--- 10715 probes passed check_dic selection.\n",
      "finish in 1.180s, 208 probes kept.\n",
      "-- check region:499 8:98740000-98760000_:+_reg_, 39162 candidate probes\n",
      "--- 10548 probes passed check_dic selection.\n",
      "finish in 1.201s, 219 probes kept.\n",
      "-- check region:500 8:101240000-101260000_:+_reg_, 39442 candidate probes\n",
      "--- 6892 probes passed check_dic selection.\n",
      "finish in 1.072s, 169 probes kept.\n",
      "-- check region:501 8:103740000-103760000_:+_reg_, 30168 candidate probes\n",
      "--- 12060 probes passed check_dic selection.\n",
      "finish in 1.170s, 202 probes kept.\n",
      "-- check region:502 8:106240000-106260000_:+_reg_, 39302 candidate probes\n",
      "--- 16345 probes passed check_dic selection.\n",
      "finish in 1.529s, 316 probes kept.\n",
      "-- check region:503 8:108740000-108760000_:+_reg_, 39710 candidate probes\n",
      "--- 22150 probes passed check_dic selection.\n",
      "finish in 1.648s, 321 probes kept.\n",
      "-- check region:504 8:111240000-111260000_:+_reg_, 38524 candidate probes\n",
      "--- 8303 probes passed check_dic selection.\n",
      "finish in 1.112s, 179 probes kept.\n",
      "-- check region:505 8:113740000-113760000_:+_reg_, 39714 candidate probes\n",
      "--- 16862 probes passed check_dic selection.\n",
      "finish in 1.440s, 273 probes kept.\n",
      "-- check region:506 8:116240000-116260000_:+_reg_, 39650 candidate probes\n",
      "--- 17422 probes passed check_dic selection.\n",
      "finish in 1.517s, 279 probes kept.\n",
      "-- check region:507 8:118740000-118760000_:+_reg_, 37484 candidate probes\n",
      "--- 17526 probes passed check_dic selection.\n",
      "finish in 1.477s, 282 probes kept.\n",
      "-- check region:508 8:121240000-121260000_:+_reg_, 38428 candidate probes\n",
      "--- 19565 probes passed check_dic selection.\n",
      "finish in 1.567s, 310 probes kept.\n",
      "-- check region:509 8:123740000-123760000_:+_reg_, 39490 candidate probes\n",
      "--- 15415 probes passed check_dic selection.\n",
      "finish in 1.418s, 261 probes kept.\n",
      "-- check region:510 8:126240000-126260000_:+_reg_, 39490 candidate probes\n",
      "--- 18264 probes passed check_dic selection.\n",
      "finish in 1.529s, 294 probes kept.\n",
      "-- check region:511 8:128740000-128760000_:+_reg_, 25042 candidate probes\n",
      "--- 4304 probes passed check_dic selection.\n",
      "finish in 0.707s, 103 probes kept.\n",
      "-- check region:512 9:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:513 9:3740000-3760000_:+_reg_, 29898 candidate probes\n",
      "--- 1729 probes passed check_dic selection.\n",
      "finish in 0.675s, 53 probes kept.\n",
      "-- check region:514 9:6240000-6260000_:+_reg_, 38658 candidate probes\n",
      "--- 12872 probes passed check_dic selection.\n",
      "finish in 1.292s, 247 probes kept.\n",
      "-- check region:515 9:8740000-8760000_:+_reg_, 37714 candidate probes\n",
      "--- 11508 probes passed check_dic selection.\n",
      "finish in 1.167s, 223 probes kept.\n",
      "-- check region:516 9:11240000-11260000_:+_reg_, 39496 candidate probes\n",
      "--- 5160 probes passed check_dic selection.\n",
      "finish in 0.957s, 120 probes kept.\n",
      "-- check region:517 9:13740000-13760000_:+_reg_, 36648 candidate probes\n",
      "--- 9609 probes passed check_dic selection.\n",
      "finish in 1.127s, 187 probes kept.\n",
      "-- check region:518 9:16240000-16260000_:+_reg_, 39600 candidate probes\n",
      "--- 13482 probes passed check_dic selection.\n",
      "finish in 1.350s, 237 probes kept.\n",
      "-- check region:519 9:18740000-18760000_:+_reg_, 38796 candidate probes\n",
      "--- 8969 probes passed check_dic selection.\n",
      "finish in 1.139s, 216 probes kept.\n",
      "-- check region:520 9:21240000-21260000_:+_reg_, 38856 candidate probes\n",
      "--- 11755 probes passed check_dic selection.\n",
      "finish in 1.246s, 222 probes kept.\n",
      "-- check region:521 9:23740000-23760000_:+_reg_, 31184 candidate probes\n",
      "--- 5707 probes passed check_dic selection.\n",
      "finish in 0.877s, 116 probes kept.\n",
      "-- check region:522 9:26240000-26260000_:+_reg_, 38498 candidate probes\n",
      "--- 12420 probes passed check_dic selection.\n",
      "finish in 1.253s, 219 probes kept.\n",
      "-- check region:523 9:28740000-28760000_:+_reg_, 39516 candidate probes\n",
      "--- 13391 probes passed check_dic selection.\n",
      "finish in 1.309s, 230 probes kept.\n",
      "-- check region:524 9:31240000-31260000_:+_reg_, 38996 candidate probes\n",
      "--- 11796 probes passed check_dic selection.\n",
      "finish in 1.334s, 217 probes kept.\n",
      "-- check region:525 9:33740000-33760000_:+_reg_, 39220 candidate probes\n",
      "--- 8826 probes passed check_dic selection.\n",
      "finish in 1.164s, 180 probes kept.\n",
      "-- check region:526 9:36240000-36260000_:+_reg_, 35016 candidate probes\n",
      "--- 2832 probes passed check_dic selection.\n",
      "finish in 0.871s, 90 probes kept.\n",
      "-- check region:527 9:38740000-38760000_:+_reg_, 39508 candidate probes\n",
      "--- 7050 probes passed check_dic selection.\n",
      "finish in 1.073s, 193 probes kept.\n",
      "-- check region:528 9:41240000-41260000_:+_reg_, 39572 candidate probes\n",
      "--- 13963 probes passed check_dic selection.\n",
      "finish in 1.332s, 256 probes kept.\n",
      "-- check region:529 9:43740000-43760000_:+_reg_, 39714 candidate probes\n",
      "--- 23382 probes passed check_dic selection.\n",
      "finish in 1.742s, 349 probes kept.\n",
      "-- check region:530 9:46240000-46260000_:+_reg_, 39144 candidate probes\n",
      "--- 14814 probes passed check_dic selection.\n",
      "finish in 1.454s, 255 probes kept.\n",
      "-- check region:531 9:48740000-48760000_:+_reg_, 39394 candidate probes\n",
      "--- 21388 probes passed check_dic selection.\n",
      "finish in 1.630s, 330 probes kept.\n",
      "-- check region:532 9:51240000-51260000_:+_reg_, 39436 candidate probes\n",
      "--- 14986 probes passed check_dic selection.\n",
      "finish in 1.364s, 252 probes kept.\n",
      "-- check region:533 9:53740000-53760000_:+_reg_, 38574 candidate probes\n",
      "--- 12248 probes passed check_dic selection.\n",
      "finish in 1.293s, 224 probes kept.\n",
      "-- check region:534 9:56240000-56260000_:+_reg_, 39252 candidate probes\n",
      "--- 13352 probes passed check_dic selection.\n",
      "finish in 1.330s, 264 probes kept.\n",
      "-- check region:535 9:58740000-58760000_:+_reg_, 39492 candidate probes\n",
      "--- 16728 probes passed check_dic selection.\n",
      "finish in 1.436s, 268 probes kept.\n",
      "-- check region:536 9:61240000-61260000_:+_reg_, 39294 candidate probes\n",
      "--- 15557 probes passed check_dic selection.\n",
      "finish in 1.449s, 269 probes kept.\n",
      "-- check region:537 9:63740000-63760000_:+_reg_, 39576 candidate probes\n",
      "--- 21626 probes passed check_dic selection.\n",
      "finish in 1.712s, 331 probes kept.\n",
      "-- check region:538 9:66240000-66260000_:+_reg_, 39032 candidate probes\n",
      "--- 18027 probes passed check_dic selection.\n",
      "finish in 1.498s, 304 probes kept.\n",
      "-- check region:539 9:68740000-68760000_:+_reg_, 39152 candidate probes\n",
      "--- 16007 probes passed check_dic selection.\n",
      "finish in 1.441s, 259 probes kept.\n",
      "-- check region:540 9:71240000-71260000_:+_reg_, 39760 candidate probes\n",
      "--- 14744 probes passed check_dic selection.\n",
      "finish in 1.340s, 252 probes kept.\n",
      "-- check region:541 9:73740000-73760000_:+_reg_, 39640 candidate probes\n",
      "--- 14715 probes passed check_dic selection.\n",
      "finish in 1.381s, 267 probes kept.\n",
      "-- check region:542 9:76240000-76260000_:+_reg_, 39550 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 9583 probes passed check_dic selection.\n",
      "finish in 1.181s, 200 probes kept.\n",
      "-- check region:543 9:78740000-78760000_:+_reg_, 33992 candidate probes\n",
      "--- 7378 probes passed check_dic selection.\n",
      "finish in 1.009s, 140 probes kept.\n",
      "-- check region:544 9:81240000-81260000_:+_reg_, 27570 candidate probes\n",
      "--- 2984 probes passed check_dic selection.\n",
      "finish in 0.692s, 75 probes kept.\n",
      "-- check region:545 9:83740000-83760000_:+_reg_, 33582 candidate probes\n",
      "--- 9735 probes passed check_dic selection.\n",
      "finish in 1.072s, 162 probes kept.\n",
      "-- check region:546 9:86240000-86260000_:+_reg_, 39640 candidate probes\n",
      "--- 12717 probes passed check_dic selection.\n",
      "finish in 1.295s, 234 probes kept.\n",
      "-- check region:547 9:88740000-88760000_:+_reg_, 35892 candidate probes\n",
      "--- 224 probes passed check_dic selection.\n",
      "finish in 0.740s, 14 probes kept.\n",
      "-- check region:548 9:91240000-91260000_:+_reg_, 30778 candidate probes\n",
      "--- 3773 probes passed check_dic selection.\n",
      "finish in 0.827s, 88 probes kept.\n",
      "-- check region:549 9:93740000-93760000_:+_reg_, 39104 candidate probes\n",
      "--- 8406 probes passed check_dic selection.\n",
      "finish in 1.134s, 174 probes kept.\n",
      "-- check region:550 9:96240000-96260000_:+_reg_, 38844 candidate probes\n",
      "--- 8360 probes passed check_dic selection.\n",
      "finish in 1.173s, 181 probes kept.\n",
      "-- check region:551 9:98740000-98760000_:+_reg_, 36820 candidate probes\n",
      "--- 14619 probes passed check_dic selection.\n",
      "finish in 1.419s, 236 probes kept.\n",
      "-- check region:552 9:101240000-101260000_:+_reg_, 39750 candidate probes\n",
      "--- 13072 probes passed check_dic selection.\n",
      "finish in 1.301s, 239 probes kept.\n",
      "-- check region:553 9:103740000-103760000_:+_reg_, 39902 candidate probes\n",
      "--- 19274 probes passed check_dic selection.\n",
      "finish in 1.581s, 314 probes kept.\n",
      "-- check region:554 9:106240000-106260000_:+_reg_, 37576 candidate probes\n",
      "--- 13357 probes passed check_dic selection.\n",
      "finish in 1.335s, 246 probes kept.\n",
      "-- check region:555 9:108740000-108760000_:+_reg_, 38736 candidate probes\n",
      "--- 13357 probes passed check_dic selection.\n",
      "finish in 1.327s, 259 probes kept.\n",
      "-- check region:556 9:111240000-111260000_:+_reg_, 38590 candidate probes\n",
      "--- 14193 probes passed check_dic selection.\n",
      "finish in 1.378s, 267 probes kept.\n",
      "-- check region:557 9:113740000-113760000_:+_reg_, 39462 candidate probes\n",
      "--- 10904 probes passed check_dic selection.\n",
      "finish in 1.212s, 234 probes kept.\n",
      "-- check region:558 9:116240000-116260000_:+_reg_, 39200 candidate probes\n",
      "--- 13281 probes passed check_dic selection.\n",
      "finish in 1.400s, 246 probes kept.\n",
      "-- check region:559 9:118740000-118760000_:+_reg_, 39868 candidate probes\n",
      "--- 18687 probes passed check_dic selection.\n",
      "finish in 1.525s, 296 probes kept.\n",
      "-- check region:560 9:121240000-121260000_:+_reg_, 38804 candidate probes\n",
      "--- 16953 probes passed check_dic selection.\n",
      "finish in 1.443s, 290 probes kept.\n",
      "-- check region:561 9:123740000-123760000_:+_reg_, 24526 candidate probes\n",
      "--- 8560 probes passed check_dic selection.\n",
      "finish in 0.830s, 148 probes kept.\n",
      "-- check region:562 10:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:563 10:3740000-3760000_:+_reg_, 39486 candidate probes\n",
      "--- 17660 probes passed check_dic selection.\n",
      "finish in 1.499s, 296 probes kept.\n",
      "-- check region:564 10:6240000-6260000_:+_reg_, 37970 candidate probes\n",
      "--- 4534 probes passed check_dic selection.\n",
      "finish in 0.943s, 94 probes kept.\n",
      "-- check region:565 10:8740000-8760000_:+_reg_, 39820 candidate probes\n",
      "--- 21086 probes passed check_dic selection.\n",
      "finish in 1.633s, 326 probes kept.\n",
      "-- check region:566 10:11240000-11260000_:+_reg_, 39634 candidate probes\n",
      "--- 7944 probes passed check_dic selection.\n",
      "finish in 1.092s, 184 probes kept.\n",
      "-- check region:567 10:13740000-13760000_:+_reg_, 38178 candidate probes\n",
      "--- 16314 probes passed check_dic selection.\n",
      "finish in 1.419s, 265 probes kept.\n",
      "-- check region:568 10:16240000-16260000_:+_reg_, 39096 candidate probes\n",
      "--- 6674 probes passed check_dic selection.\n",
      "finish in 0.999s, 139 probes kept.\n",
      "-- check region:569 10:18740000-18760000_:+_reg_, 34226 candidate probes\n",
      "--- 9478 probes passed check_dic selection.\n",
      "finish in 1.077s, 166 probes kept.\n",
      "-- check region:570 10:21240000-21260000_:+_reg_, 38818 candidate probes\n",
      "--- 13751 probes passed check_dic selection.\n",
      "finish in 1.330s, 241 probes kept.\n",
      "-- check region:571 10:23740000-23760000_:+_reg_, 39154 candidate probes\n",
      "--- 11582 probes passed check_dic selection.\n",
      "finish in 1.302s, 209 probes kept.\n",
      "-- check region:572 10:26240000-26260000_:+_reg_, 39020 candidate probes\n",
      "--- 13764 probes passed check_dic selection.\n",
      "finish in 1.345s, 246 probes kept.\n",
      "-- check region:573 10:28740000-28760000_:+_reg_, 38314 candidate probes\n",
      "--- 6873 probes passed check_dic selection.\n",
      "finish in 1.018s, 138 probes kept.\n",
      "-- check region:574 10:31240000-31260000_:+_reg_, 39282 candidate probes\n",
      "--- 10030 probes passed check_dic selection.\n",
      "finish in 1.174s, 218 probes kept.\n",
      "-- check region:575 10:33740000-33760000_:+_reg_, 38012 candidate probes\n",
      "--- 2387 probes passed check_dic selection.\n",
      "finish in 0.856s, 73 probes kept.\n",
      "-- check region:576 10:36240000-36260000_:+_reg_, 37988 candidate probes\n",
      "--- 7084 probes passed check_dic selection.\n",
      "finish in 1.066s, 161 probes kept.\n",
      "-- check region:577 10:38740000-38760000_:+_reg_, 29556 candidate probes\n",
      "--- 2526 probes passed check_dic selection.\n",
      "finish in 0.716s, 60 probes kept.\n",
      "-- check region:578 10:41240000-41260000_:+_reg_, 39478 candidate probes\n",
      "--- 16497 probes passed check_dic selection.\n",
      "finish in 1.406s, 284 probes kept.\n",
      "-- check region:579 10:43740000-43760000_:+_reg_, 39054 candidate probes\n",
      "--- 14107 probes passed check_dic selection.\n",
      "finish in 1.362s, 252 probes kept.\n",
      "-- check region:580 10:46240000-46260000_:+_reg_, 39340 candidate probes\n",
      "--- 13458 probes passed check_dic selection.\n",
      "finish in 1.362s, 248 probes kept.\n",
      "-- check region:581 10:48740000-48760000_:+_reg_, 36490 candidate probes\n",
      "--- 4100 probes passed check_dic selection.\n",
      "finish in 0.908s, 94 probes kept.\n",
      "-- check region:582 10:51240000-51260000_:+_reg_, 37050 candidate probes\n",
      "--- 7550 probes passed check_dic selection.\n",
      "finish in 1.073s, 152 probes kept.\n",
      "-- check region:583 10:53740000-53760000_:+_reg_, 39886 candidate probes\n",
      "--- 14969 probes passed check_dic selection.\n",
      "finish in 1.416s, 271 probes kept.\n",
      "-- check region:584 10:56240000-56260000_:+_reg_, 27066 candidate probes\n",
      "--- 5290 probes passed check_dic selection.\n",
      "finish in 0.774s, 118 probes kept.\n",
      "-- check region:585 10:58740000-58760000_:+_reg_, 37074 candidate probes\n",
      "--- 9552 probes passed check_dic selection.\n",
      "finish in 1.129s, 189 probes kept.\n",
      "-- check region:586 10:61240000-61260000_:+_reg_, 39090 candidate probes\n",
      "--- 17984 probes passed check_dic selection.\n",
      "finish in 1.515s, 288 probes kept.\n",
      "-- check region:587 10:63740000-63760000_:+_reg_, 26218 candidate probes\n",
      "--- 3824 probes passed check_dic selection.\n",
      "finish in 0.726s, 78 probes kept.\n",
      "-- check region:588 10:66240000-66260000_:+_reg_, 39614 candidate probes\n",
      "--- 9707 probes passed check_dic selection.\n",
      "finish in 1.140s, 213 probes kept.\n",
      "-- check region:589 10:68740000-68760000_:+_reg_, 37481 candidate probes\n",
      "--- 12082 probes passed check_dic selection.\n",
      "finish in 1.241s, 226 probes kept.\n",
      "-- check region:590 10:71240000-71260000_:+_reg_, 39172 candidate probes\n",
      "--- 12577 probes passed check_dic selection.\n",
      "finish in 1.312s, 235 probes kept.\n",
      "-- check region:591 10:73740000-73760000_:+_reg_, 38642 candidate probes\n",
      "--- 10744 probes passed check_dic selection.\n",
      "finish in 1.182s, 222 probes kept.\n",
      "-- check region:592 10:76240000-76260000_:+_reg_, 39742 candidate probes\n",
      "--- 15324 probes passed check_dic selection.\n",
      "finish in 1.399s, 253 probes kept.\n",
      "-- check region:593 10:78740000-78760000_:+_reg_, 39734 candidate probes\n",
      "--- 13635 probes passed check_dic selection.\n",
      "finish in 1.348s, 251 probes kept.\n",
      "-- check region:594 10:81240000-81260000_:+_reg_, 39764 candidate probes\n",
      "--- 17389 probes passed check_dic selection.\n",
      "finish in 1.559s, 319 probes kept.\n",
      "-- check region:595 10:83740000-83760000_:+_reg_, 39700 candidate probes\n",
      "--- 16571 probes passed check_dic selection.\n",
      "finish in 1.501s, 274 probes kept.\n",
      "-- check region:596 10:86240000-86260000_:+_reg_, 37564 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 18956 probes passed check_dic selection.\n",
      "finish in 1.559s, 304 probes kept.\n",
      "-- check region:597 10:88740000-88760000_:+_reg_, 39392 candidate probes\n",
      "--- 13673 probes passed check_dic selection.\n",
      "finish in 1.345s, 245 probes kept.\n",
      "-- check region:598 10:91240000-91260000_:+_reg_, 30874 candidate probes\n",
      "--- 8792 probes passed check_dic selection.\n",
      "finish in 1.007s, 160 probes kept.\n",
      "-- check region:599 10:93740000-93760000_:+_reg_, 39570 candidate probes\n",
      "--- 15355 probes passed check_dic selection.\n",
      "finish in 1.437s, 258 probes kept.\n",
      "-- check region:600 10:96240000-96260000_:+_reg_, 39760 candidate probes\n",
      "--- 12705 probes passed check_dic selection.\n",
      "finish in 1.261s, 253 probes kept.\n",
      "-- check region:601 10:98740000-98760000_:+_reg_, 39638 candidate probes\n",
      "--- 13951 probes passed check_dic selection.\n",
      "finish in 1.398s, 266 probes kept.\n",
      "-- check region:602 10:101240000-101260000_:+_reg_, 37586 candidate probes\n",
      "--- 7640 probes passed check_dic selection.\n",
      "finish in 1.102s, 162 probes kept.\n",
      "-- check region:603 10:103740000-103760000_:+_reg_, 26710 candidate probes\n",
      "--- 3766 probes passed check_dic selection.\n",
      "finish in 0.682s, 92 probes kept.\n",
      "-- check region:604 10:106240000-106260000_:+_reg_, 39660 candidate probes\n",
      "--- 7884 probes passed check_dic selection.\n",
      "finish in 1.100s, 187 probes kept.\n",
      "-- check region:605 10:108740000-108760000_:+_reg_, 39792 candidate probes\n",
      "--- 10561 probes passed check_dic selection.\n",
      "finish in 1.203s, 212 probes kept.\n",
      "-- check region:606 10:111240000-111260000_:+_reg_, 39546 candidate probes\n",
      "--- 10749 probes passed check_dic selection.\n",
      "finish in 1.210s, 220 probes kept.\n",
      "-- check region:607 10:113740000-113760000_:+_reg_, 39380 candidate probes\n",
      "--- 8263 probes passed check_dic selection.\n",
      "finish in 1.093s, 190 probes kept.\n",
      "-- check region:608 10:116240000-116260000_:+_reg_, 39264 candidate probes\n",
      "--- 15154 probes passed check_dic selection.\n",
      "finish in 1.394s, 256 probes kept.\n",
      "-- check region:609 10:118740000-118760000_:+_reg_, 39652 candidate probes\n",
      "--- 17454 probes passed check_dic selection.\n",
      "finish in 1.547s, 285 probes kept.\n",
      "-- check region:610 10:121240000-121260000_:+_reg_, 39306 candidate probes\n",
      "--- 18733 probes passed check_dic selection.\n",
      "finish in 1.530s, 294 probes kept.\n",
      "-- check region:611 10:123740000-123760000_:+_reg_, 39802 candidate probes\n",
      "--- 15284 probes passed check_dic selection.\n",
      "finish in 1.494s, 281 probes kept.\n",
      "-- check region:612 10:126240000-126260000_:+_reg_, 38656 candidate probes\n",
      "--- 16403 probes passed check_dic selection.\n",
      "finish in 1.390s, 277 probes kept.\n",
      "-- check region:613 10:128740000-128760000_:+_reg_, 39138 candidate probes\n",
      "--- 12658 probes passed check_dic selection.\n",
      "finish in 1.281s, 236 probes kept.\n",
      "-- check region:614 10:131240000-130694993_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:615 11:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:616 11:3740000-3760000_:+_reg_, 38894 candidate probes\n",
      "--- 18491 probes passed check_dic selection.\n",
      "finish in 1.499s, 289 probes kept.\n",
      "-- check region:617 11:6240000-6260000_:+_reg_, 36260 candidate probes\n",
      "--- 8740 probes passed check_dic selection.\n",
      "finish in 1.099s, 158 probes kept.\n",
      "-- check region:618 11:8740000-8760000_:+_reg_, 33660 candidate probes\n",
      "--- 6176 probes passed check_dic selection.\n",
      "finish in 0.940s, 126 probes kept.\n",
      "-- check region:619 11:11240000-11260000_:+_reg_, 39644 candidate probes\n",
      "--- 11245 probes passed check_dic selection.\n",
      "finish in 1.281s, 222 probes kept.\n",
      "-- check region:620 11:13740000-13760000_:+_reg_, 34384 candidate probes\n",
      "--- 4200 probes passed check_dic selection.\n",
      "finish in 0.900s, 103 probes kept.\n",
      "-- check region:621 11:16240000-16260000_:+_reg_, 38428 candidate probes\n",
      "--- 9127 probes passed check_dic selection.\n",
      "finish in 1.146s, 191 probes kept.\n",
      "-- check region:622 11:18740000-18760000_:+_reg_, 39326 candidate probes\n",
      "--- 15310 probes passed check_dic selection.\n",
      "finish in 1.354s, 272 probes kept.\n",
      "-- check region:623 11:21240000-21260000_:+_reg_, 39134 candidate probes\n",
      "--- 8123 probes passed check_dic selection.\n",
      "finish in 1.072s, 177 probes kept.\n",
      "-- check region:624 11:23740000-23760000_:+_reg_, 39484 candidate probes\n",
      "--- 8803 probes passed check_dic selection.\n",
      "finish in 1.192s, 215 probes kept.\n",
      "-- check region:625 11:26240000-26260000_:+_reg_, 38828 candidate probes\n",
      "--- 4576 probes passed check_dic selection.\n",
      "finish in 0.916s, 107 probes kept.\n",
      "-- check region:626 11:28740000-28760000_:+_reg_, 36766 candidate probes\n",
      "--- 8251 probes passed check_dic selection.\n",
      "finish in 1.126s, 175 probes kept.\n",
      "-- check region:627 11:31240000-31260000_:+_reg_, 38422 candidate probes\n",
      "--- 2601 probes passed check_dic selection.\n",
      "finish in 0.844s, 75 probes kept.\n",
      "-- check region:628 11:33740000-33760000_:+_reg_, 39804 candidate probes\n",
      "--- 22203 probes passed check_dic selection.\n",
      "finish in 1.714s, 332 probes kept.\n",
      "-- check region:629 11:36240000-36260000_:+_reg_, 39668 candidate probes\n",
      "--- 21282 probes passed check_dic selection.\n",
      "finish in 1.602s, 315 probes kept.\n",
      "-- check region:630 11:38740000-38760000_:+_reg_, 39860 candidate probes\n",
      "--- 8844 probes passed check_dic selection.\n",
      "finish in 1.115s, 192 probes kept.\n",
      "-- check region:631 11:41240000-41260000_:+_reg_, 39686 candidate probes\n",
      "--- 9171 probes passed check_dic selection.\n",
      "finish in 1.155s, 205 probes kept.\n",
      "-- check region:632 11:43740000-43760000_:+_reg_, 38702 candidate probes\n",
      "--- 10979 probes passed check_dic selection.\n",
      "finish in 1.282s, 213 probes kept.\n",
      "-- check region:633 11:46240000-46260000_:+_reg_, 39652 candidate probes\n",
      "--- 18864 probes passed check_dic selection.\n",
      "finish in 1.570s, 296 probes kept.\n",
      "-- check region:634 11:48740000-48760000_:+_reg_, 35060 candidate probes\n",
      "--- 8932 probes passed check_dic selection.\n",
      "finish in 1.061s, 168 probes kept.\n",
      "-- check region:635 11:51240000-51260000_:+_reg_, 39618 candidate probes\n",
      "--- 11254 probes passed check_dic selection.\n",
      "finish in 1.219s, 234 probes kept.\n",
      "-- check region:636 11:53740000-53760000_:+_reg_, 38922 candidate probes\n",
      "--- 8215 probes passed check_dic selection.\n",
      "finish in 1.145s, 168 probes kept.\n",
      "-- check region:637 11:56240000-56260000_:+_reg_, 37722 candidate probes\n",
      "--- 15552 probes passed check_dic selection.\n",
      "finish in 1.386s, 255 probes kept.\n",
      "-- check region:638 11:58740000-58760000_:+_reg_, 37768 candidate probes\n",
      "--- 7164 probes passed check_dic selection.\n",
      "finish in 1.055s, 159 probes kept.\n",
      "-- check region:639 11:61240000-61260000_:+_reg_, 39584 candidate probes\n",
      "--- 14996 probes passed check_dic selection.\n",
      "finish in 1.392s, 283 probes kept.\n",
      "-- check region:640 11:63740000-63760000_:+_reg_, 39536 candidate probes\n",
      "--- 17152 probes passed check_dic selection.\n",
      "finish in 1.577s, 286 probes kept.\n",
      "-- check region:641 11:66240000-66260000_:+_reg_, 39692 candidate probes\n",
      "--- 17082 probes passed check_dic selection.\n",
      "finish in 1.495s, 263 probes kept.\n",
      "-- check region:642 11:68740000-68760000_:+_reg_, 39810 candidate probes\n",
      "--- 16537 probes passed check_dic selection.\n",
      "finish in 1.499s, 285 probes kept.\n",
      "-- check region:643 11:71240000-71260000_:+_reg_, 39650 candidate probes\n",
      "--- 14239 probes passed check_dic selection.\n",
      "finish in 1.396s, 272 probes kept.\n",
      "-- check region:644 11:73740000-73760000_:+_reg_, 35774 candidate probes\n",
      "--- 4961 probes passed check_dic selection.\n",
      "finish in 0.918s, 131 probes kept.\n",
      "-- check region:645 11:76240000-76260000_:+_reg_, 39770 candidate probes\n",
      "--- 17158 probes passed check_dic selection.\n",
      "finish in 1.517s, 303 probes kept.\n",
      "-- check region:646 11:78740000-78760000_:+_reg_, 39346 candidate probes\n",
      "--- 13522 probes passed check_dic selection.\n",
      "finish in 1.408s, 234 probes kept.\n",
      "-- check region:647 11:81240000-81260000_:+_reg_, 39654 candidate probes\n",
      "--- 17833 probes passed check_dic selection.\n",
      "finish in 1.515s, 285 probes kept.\n",
      "-- check region:648 11:83740000-83760000_:+_reg_, 39718 candidate probes\n",
      "--- 14887 probes passed check_dic selection.\n",
      "finish in 1.409s, 252 probes kept.\n",
      "-- check region:649 11:86240000-86260000_:+_reg_, 38918 candidate probes\n",
      "--- 11369 probes passed check_dic selection.\n",
      "finish in 1.289s, 228 probes kept.\n",
      "-- check region:650 11:88740000-88760000_:+_reg_, 39030 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 15556 probes passed check_dic selection.\n",
      "finish in 1.410s, 260 probes kept.\n",
      "-- check region:651 11:91240000-91260000_:+_reg_, 39698 candidate probes\n",
      "--- 11649 probes passed check_dic selection.\n",
      "finish in 1.248s, 239 probes kept.\n",
      "-- check region:652 11:93740000-93760000_:+_reg_, 33122 candidate probes\n",
      "--- 11289 probes passed check_dic selection.\n",
      "finish in 1.149s, 211 probes kept.\n",
      "-- check region:653 11:96240000-96260000_:+_reg_, 38410 candidate probes\n",
      "--- 14069 probes passed check_dic selection.\n",
      "finish in 1.352s, 233 probes kept.\n",
      "-- check region:654 11:98740000-98760000_:+_reg_, 39544 candidate probes\n",
      "--- 19064 probes passed check_dic selection.\n",
      "finish in 1.546s, 308 probes kept.\n",
      "-- check region:655 11:101240000-101260000_:+_reg_, 38262 candidate probes\n",
      "--- 13093 probes passed check_dic selection.\n",
      "finish in 1.300s, 251 probes kept.\n",
      "-- check region:656 11:103740000-103760000_:+_reg_, 39264 candidate probes\n",
      "--- 21217 probes passed check_dic selection.\n",
      "finish in 1.618s, 320 probes kept.\n",
      "-- check region:657 11:106240000-106260000_:+_reg_, 39510 candidate probes\n",
      "--- 14341 probes passed check_dic selection.\n",
      "finish in 1.341s, 290 probes kept.\n",
      "-- check region:658 11:108740000-108760000_:+_reg_, 39676 candidate probes\n",
      "--- 14602 probes passed check_dic selection.\n",
      "finish in 1.345s, 269 probes kept.\n",
      "-- check region:659 11:111240000-111260000_:+_reg_, 39498 candidate probes\n",
      "--- 14709 probes passed check_dic selection.\n",
      "finish in 1.364s, 257 probes kept.\n",
      "-- check region:660 11:113740000-113760000_:+_reg_, 39298 candidate probes\n",
      "--- 18179 probes passed check_dic selection.\n",
      "finish in 1.520s, 284 probes kept.\n",
      "-- check region:661 11:116240000-116260000_:+_reg_, 39512 candidate probes\n",
      "--- 18169 probes passed check_dic selection.\n",
      "finish in 1.557s, 284 probes kept.\n",
      "-- check region:662 11:118740000-118760000_:+_reg_, 39658 candidate probes\n",
      "--- 20482 probes passed check_dic selection.\n",
      "finish in 1.647s, 322 probes kept.\n",
      "-- check region:663 11:121240000-121260000_:+_reg_, 39256 candidate probes\n",
      "--- 15551 probes passed check_dic selection.\n",
      "finish in 1.432s, 271 probes kept.\n",
      "-- check region:664 12:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.002s, 0 probes kept.\n",
      "-- check region:665 12:3740000-3760000_:+_reg_, 39348 candidate probes\n",
      "--- 14791 probes passed check_dic selection.\n",
      "finish in 1.404s, 270 probes kept.\n",
      "-- check region:666 12:6240000-6260000_:+_reg_, 37164 candidate probes\n",
      "--- 10030 probes passed check_dic selection.\n",
      "finish in 1.111s, 177 probes kept.\n",
      "-- check region:667 12:8740000-8760000_:+_reg_, 39482 candidate probes\n",
      "--- 11582 probes passed check_dic selection.\n",
      "finish in 1.257s, 241 probes kept.\n",
      "-- check region:668 12:11240000-11260000_:+_reg_, 39606 candidate probes\n",
      "--- 7881 probes passed check_dic selection.\n",
      "finish in 1.130s, 196 probes kept.\n",
      "-- check region:669 12:13740000-13760000_:+_reg_, 35642 candidate probes\n",
      "--- 13237 probes passed check_dic selection.\n",
      "finish in 1.247s, 239 probes kept.\n",
      "-- check region:670 12:16240000-16260000_:+_reg_, 39142 candidate probes\n",
      "--- 15645 probes passed check_dic selection.\n",
      "finish in 1.387s, 273 probes kept.\n",
      "-- check region:671 12:18740000-18760000_:+_reg_, 34712 candidate probes\n",
      "--- 38 probes passed check_dic selection.\n",
      "finish in 0.738s, 2 probes kept.\n",
      "-- check region:672 12:21240000-21260000_:+_reg_, 39300 candidate probes\n",
      "--- 17713 probes passed check_dic selection.\n",
      "finish in 1.487s, 280 probes kept.\n",
      "-- check region:673 12:23740000-23760000_:+_reg_, 33990 candidate probes\n",
      "--- 43 probes passed check_dic selection.\n",
      "finish in 0.708s, 7 probes kept.\n",
      "-- check region:674 12:26240000-26260000_:+_reg_, 39358 candidate probes\n",
      "--- 14056 probes passed check_dic selection.\n",
      "finish in 1.340s, 268 probes kept.\n",
      "-- check region:675 12:28740000-28760000_:+_reg_, 37400 candidate probes\n",
      "--- 14879 probes passed check_dic selection.\n",
      "finish in 1.358s, 251 probes kept.\n",
      "-- check region:676 12:31240000-31260000_:+_reg_, 39420 candidate probes\n",
      "--- 14240 probes passed check_dic selection.\n",
      "finish in 1.366s, 246 probes kept.\n",
      "-- check region:677 12:33740000-33760000_:+_reg_, 36390 candidate probes\n",
      "--- 4792 probes passed check_dic selection.\n",
      "finish in 0.928s, 106 probes kept.\n",
      "-- check region:678 12:36240000-36260000_:+_reg_, 39624 candidate probes\n",
      "--- 15160 probes passed check_dic selection.\n",
      "finish in 1.375s, 275 probes kept.\n",
      "-- check region:679 12:38740000-38760000_:+_reg_, 36128 candidate probes\n",
      "--- 9987 probes passed check_dic selection.\n",
      "finish in 1.116s, 186 probes kept.\n",
      "-- check region:680 12:41240000-41260000_:+_reg_, 39186 candidate probes\n",
      "--- 9562 probes passed check_dic selection.\n",
      "finish in 1.173s, 199 probes kept.\n",
      "-- check region:681 12:43740000-43760000_:+_reg_, 25258 candidate probes\n",
      "--- 2916 probes passed check_dic selection.\n",
      "finish in 0.633s, 59 probes kept.\n",
      "-- check region:682 12:46240000-46260000_:+_reg_, 35074 candidate probes\n",
      "--- 6906 probes passed check_dic selection.\n",
      "finish in 0.999s, 138 probes kept.\n",
      "-- check region:683 12:48740000-48760000_:+_reg_, 39454 candidate probes\n",
      "--- 7741 probes passed check_dic selection.\n",
      "finish in 1.094s, 163 probes kept.\n",
      "-- check region:684 12:51240000-51260000_:+_reg_, 39608 candidate probes\n",
      "--- 11999 probes passed check_dic selection.\n",
      "finish in 1.247s, 231 probes kept.\n",
      "-- check region:685 12:53740000-53760000_:+_reg_, 39732 candidate probes\n",
      "--- 17543 probes passed check_dic selection.\n",
      "finish in 1.474s, 282 probes kept.\n",
      "-- check region:686 12:56240000-56260000_:+_reg_, 39314 candidate probes\n",
      "--- 16506 probes passed check_dic selection.\n",
      "finish in 1.417s, 268 probes kept.\n",
      "-- check region:687 12:58740000-58760000_:+_reg_, 39216 candidate probes\n",
      "--- 9982 probes passed check_dic selection.\n",
      "finish in 1.229s, 192 probes kept.\n",
      "-- check region:688 12:61240000-61260000_:+_reg_, 39718 candidate probes\n",
      "--- 7439 probes passed check_dic selection.\n",
      "finish in 1.083s, 172 probes kept.\n",
      "-- check region:689 12:63740000-63760000_:+_reg_, 34176 candidate probes\n",
      "--- 5572 probes passed check_dic selection.\n",
      "finish in 0.914s, 115 probes kept.\n",
      "-- check region:690 12:66240000-66260000_:+_reg_, 30077 candidate probes\n",
      "--- 6549 probes passed check_dic selection.\n",
      "finish in 0.839s, 128 probes kept.\n",
      "-- check region:691 12:68740000-68760000_:+_reg_, 31278 candidate probes\n",
      "--- 7260 probes passed check_dic selection.\n",
      "finish in 0.904s, 140 probes kept.\n",
      "-- check region:692 12:71240000-71260000_:+_reg_, 39236 candidate probes\n",
      "--- 16840 probes passed check_dic selection.\n",
      "finish in 1.464s, 275 probes kept.\n",
      "-- check region:693 12:73740000-73760000_:+_reg_, 39660 candidate probes\n",
      "--- 19569 probes passed check_dic selection.\n",
      "finish in 1.645s, 311 probes kept.\n",
      "-- check region:694 12:76240000-76260000_:+_reg_, 39346 candidate probes\n",
      "--- 9956 probes passed check_dic selection.\n",
      "finish in 1.190s, 194 probes kept.\n",
      "-- check region:695 12:78740000-78760000_:+_reg_, 38372 candidate probes\n",
      "--- 11340 probes passed check_dic selection.\n",
      "finish in 1.259s, 211 probes kept.\n",
      "-- check region:696 12:81240000-81260000_:+_reg_, 37274 candidate probes\n",
      "--- 13552 probes passed check_dic selection.\n",
      "finish in 1.281s, 266 probes kept.\n",
      "-- check region:697 12:83740000-83760000_:+_reg_, 39418 candidate probes\n",
      "--- 13123 probes passed check_dic selection.\n",
      "finish in 1.370s, 242 probes kept.\n",
      "-- check region:698 12:86240000-86260000_:+_reg_, 39368 candidate probes\n",
      "--- 14747 probes passed check_dic selection.\n",
      "finish in 1.386s, 255 probes kept.\n",
      "-- check region:699 12:88740000-88760000_:+_reg_, 39504 candidate probes\n",
      "--- 18118 probes passed check_dic selection.\n",
      "finish in 1.501s, 289 probes kept.\n",
      "-- check region:700 12:91240000-91260000_:+_reg_, 32522 candidate probes\n",
      "--- 9543 probes passed check_dic selection.\n",
      "finish in 1.086s, 167 probes kept.\n",
      "-- check region:701 12:93740000-93760000_:+_reg_, 39144 candidate probes\n",
      "--- 6735 probes passed check_dic selection.\n",
      "finish in 1.095s, 147 probes kept.\n",
      "-- check region:702 12:96240000-96260000_:+_reg_, 36886 candidate probes\n",
      "--- 5283 probes passed check_dic selection.\n",
      "finish in 0.977s, 128 probes kept.\n",
      "-- check region:703 12:98740000-98760000_:+_reg_, 39622 candidate probes\n",
      "--- 12677 probes passed check_dic selection.\n",
      "finish in 1.313s, 246 probes kept.\n",
      "-- check region:704 12:101240000-101260000_:+_reg_, 34694 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 1435 probes passed check_dic selection.\n",
      "finish in 0.745s, 44 probes kept.\n",
      "-- check region:705 12:103740000-103760000_:+_reg_, 38834 candidate probes\n",
      "--- 718 probes passed check_dic selection.\n",
      "finish in 0.838s, 34 probes kept.\n",
      "-- check region:706 12:106240000-106260000_:+_reg_, 39294 candidate probes\n",
      "--- 18184 probes passed check_dic selection.\n",
      "finish in 1.530s, 281 probes kept.\n",
      "-- check region:707 12:108740000-108760000_:+_reg_, 38536 candidate probes\n",
      "--- 13206 probes passed check_dic selection.\n",
      "finish in 1.331s, 222 probes kept.\n",
      "-- check region:708 12:111240000-111260000_:+_reg_, 39498 candidate probes\n",
      "--- 18619 probes passed check_dic selection.\n",
      "finish in 1.523s, 288 probes kept.\n",
      "-- check region:709 12:113740000-113760000_:+_reg_, 38282 candidate probes\n",
      "--- 3387 probes passed check_dic selection.\n",
      "finish in 0.950s, 112 probes kept.\n",
      "-- check region:710 12:116240000-116260000_:+_reg_, 34520 candidate probes\n",
      "--- 11827 probes passed check_dic selection.\n",
      "finish in 1.182s, 227 probes kept.\n",
      "-- check region:711 12:118740000-118760000_:+_reg_, 38546 candidate probes\n",
      "--- 12951 probes passed check_dic selection.\n",
      "finish in 1.303s, 244 probes kept.\n",
      "-- check region:712 13:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:713 13:3740000-3760000_:+_reg_, 39602 candidate probes\n",
      "--- 16063 probes passed check_dic selection.\n",
      "finish in 1.435s, 288 probes kept.\n",
      "-- check region:714 13:6240000-6260000_:+_reg_, 39598 candidate probes\n",
      "--- 14652 probes passed check_dic selection.\n",
      "finish in 1.414s, 268 probes kept.\n",
      "-- check region:715 13:8740000-8760000_:+_reg_, 37950 candidate probes\n",
      "--- 12914 probes passed check_dic selection.\n",
      "finish in 1.327s, 246 probes kept.\n",
      "-- check region:716 13:11240000-11260000_:+_reg_, 27616 candidate probes\n",
      "--- 5316 probes passed check_dic selection.\n",
      "finish in 0.795s, 101 probes kept.\n",
      "-- check region:717 13:13740000-13760000_:+_reg_, 36956 candidate probes\n",
      "--- 12149 probes passed check_dic selection.\n",
      "finish in 1.275s, 253 probes kept.\n",
      "-- check region:718 13:16240000-16260000_:+_reg_, 31052 candidate probes\n",
      "--- 6381 probes passed check_dic selection.\n",
      "finish in 0.892s, 123 probes kept.\n",
      "-- check region:719 13:18740000-18760000_:+_reg_, 39736 candidate probes\n",
      "--- 12191 probes passed check_dic selection.\n",
      "finish in 1.266s, 239 probes kept.\n",
      "-- check region:720 13:21240000-21260000_:+_reg_, 39584 candidate probes\n",
      "--- 16054 probes passed check_dic selection.\n",
      "finish in 1.442s, 268 probes kept.\n",
      "-- check region:721 13:23740000-23760000_:+_reg_, 39076 candidate probes\n",
      "--- 11489 probes passed check_dic selection.\n",
      "finish in 1.283s, 225 probes kept.\n",
      "-- check region:722 13:26240000-26260000_:+_reg_, 28906 candidate probes\n",
      "--- 6561 probes passed check_dic selection.\n",
      "finish in 0.841s, 136 probes kept.\n",
      "-- check region:723 13:28740000-28760000_:+_reg_, 39560 candidate probes\n",
      "--- 14633 probes passed check_dic selection.\n",
      "finish in 1.415s, 265 probes kept.\n",
      "-- check region:724 13:31240000-31260000_:+_reg_, 39054 candidate probes\n",
      "--- 16520 probes passed check_dic selection.\n",
      "finish in 1.450s, 276 probes kept.\n",
      "-- check region:725 13:33740000-33760000_:+_reg_, 32168 candidate probes\n",
      "--- 548 probes passed check_dic selection.\n",
      "finish in 0.657s, 18 probes kept.\n",
      "-- check region:726 13:36240000-36260000_:+_reg_, 37986 candidate probes\n",
      "--- 13635 probes passed check_dic selection.\n",
      "finish in 1.268s, 252 probes kept.\n",
      "-- check region:727 13:38740000-38760000_:+_reg_, 39850 candidate probes\n",
      "--- 15873 probes passed check_dic selection.\n",
      "finish in 1.425s, 279 probes kept.\n",
      "-- check region:728 13:41240000-41260000_:+_reg_, 37182 candidate probes\n",
      "--- 14670 probes passed check_dic selection.\n",
      "finish in 1.412s, 257 probes kept.\n",
      "-- check region:729 13:43740000-43760000_:+_reg_, 39648 candidate probes\n",
      "--- 13069 probes passed check_dic selection.\n",
      "finish in 1.334s, 244 probes kept.\n",
      "-- check region:730 13:46240000-46260000_:+_reg_, 39670 candidate probes\n",
      "--- 17457 probes passed check_dic selection.\n",
      "finish in 1.522s, 295 probes kept.\n",
      "-- check region:731 13:48740000-48760000_:+_reg_, 39752 candidate probes\n",
      "--- 19447 probes passed check_dic selection.\n",
      "finish in 1.585s, 301 probes kept.\n",
      "-- check region:732 13:51240000-51260000_:+_reg_, 39286 candidate probes\n",
      "--- 15820 probes passed check_dic selection.\n",
      "finish in 1.427s, 269 probes kept.\n",
      "-- check region:733 13:53740000-53760000_:+_reg_, 38988 candidate probes\n",
      "--- 17085 probes passed check_dic selection.\n",
      "finish in 1.509s, 281 probes kept.\n",
      "-- check region:734 13:56240000-56260000_:+_reg_, 39720 candidate probes\n",
      "--- 20743 probes passed check_dic selection.\n",
      "finish in 1.635s, 318 probes kept.\n",
      "-- check region:735 13:58740000-58760000_:+_reg_, 39012 candidate probes\n",
      "--- 14426 probes passed check_dic selection.\n",
      "finish in 1.372s, 274 probes kept.\n",
      "-- check region:736 13:61240000-61260000_:+_reg_, 39252 candidate probes\n",
      "--- 7917 probes passed check_dic selection.\n",
      "finish in 1.102s, 173 probes kept.\n",
      "-- check region:737 13:63740000-63760000_:+_reg_, 38940 candidate probes\n",
      "--- 7317 probes passed check_dic selection.\n",
      "finish in 1.089s, 148 probes kept.\n",
      "-- check region:738 13:66240000-66260000_:+_reg_, 38946 candidate probes\n",
      "--- 10 probes passed check_dic selection.\n",
      "finish in 0.780s, 2 probes kept.\n",
      "-- check region:739 13:68740000-68760000_:+_reg_, 39392 candidate probes\n",
      "--- 18193 probes passed check_dic selection.\n",
      "finish in 1.514s, 293 probes kept.\n",
      "-- check region:740 13:71240000-71260000_:+_reg_, 39408 candidate probes\n",
      "--- 14531 probes passed check_dic selection.\n",
      "finish in 1.380s, 262 probes kept.\n",
      "-- check region:741 13:73740000-73760000_:+_reg_, 39108 candidate probes\n",
      "--- 18829 probes passed check_dic selection.\n",
      "finish in 1.554s, 278 probes kept.\n",
      "-- check region:742 13:76240000-76260000_:+_reg_, 38798 candidate probes\n",
      "--- 12517 probes passed check_dic selection.\n",
      "finish in 1.241s, 235 probes kept.\n",
      "-- check region:743 13:78740000-78760000_:+_reg_, 35204 candidate probes\n",
      "--- 4530 probes passed check_dic selection.\n",
      "finish in 0.900s, 91 probes kept.\n",
      "-- check region:744 13:81240000-81260000_:+_reg_, 36896 candidate probes\n",
      "--- 7158 probes passed check_dic selection.\n",
      "finish in 1.060s, 144 probes kept.\n",
      "-- check region:745 13:83740000-83760000_:+_reg_, 39676 candidate probes\n",
      "--- 11287 probes passed check_dic selection.\n",
      "finish in 1.253s, 225 probes kept.\n",
      "-- check region:746 13:86240000-86260000_:+_reg_, 35978 candidate probes\n",
      "--- 6430 probes passed check_dic selection.\n",
      "finish in 0.988s, 149 probes kept.\n",
      "-- check region:747 13:88740000-88760000_:+_reg_, 39616 candidate probes\n",
      "--- 4518 probes passed check_dic selection.\n",
      "finish in 0.942s, 118 probes kept.\n",
      "-- check region:748 13:91240000-91260000_:+_reg_, 33492 candidate probes\n",
      "--- 8030 probes passed check_dic selection.\n",
      "finish in 1.028s, 160 probes kept.\n",
      "-- check region:749 13:93740000-93760000_:+_reg_, 39310 candidate probes\n",
      "--- 12867 probes passed check_dic selection.\n",
      "finish in 1.330s, 224 probes kept.\n",
      "-- check region:750 13:96240000-96260000_:+_reg_, 38912 candidate probes\n",
      "--- 10237 probes passed check_dic selection.\n",
      "finish in 1.170s, 216 probes kept.\n",
      "-- check region:751 13:98740000-98760000_:+_reg_, 38976 candidate probes\n",
      "--- 5832 probes passed check_dic selection.\n",
      "finish in 0.982s, 166 probes kept.\n",
      "-- check region:752 13:101240000-101260000_:+_reg_, 39726 candidate probes\n",
      "--- 14655 probes passed check_dic selection.\n",
      "finish in 1.416s, 264 probes kept.\n",
      "-- check region:753 13:103740000-103760000_:+_reg_, 38896 candidate probes\n",
      "--- 10824 probes passed check_dic selection.\n",
      "finish in 1.242s, 225 probes kept.\n",
      "-- check region:754 13:106240000-106260000_:+_reg_, 39778 candidate probes\n",
      "--- 8766 probes passed check_dic selection.\n",
      "finish in 1.143s, 186 probes kept.\n",
      "-- check region:755 13:108740000-108760000_:+_reg_, 39700 candidate probes\n",
      "--- 14117 probes passed check_dic selection.\n",
      "finish in 1.356s, 252 probes kept.\n",
      "-- check region:756 13:111240000-111260000_:+_reg_, 39190 candidate probes\n",
      "--- 16408 probes passed check_dic selection.\n",
      "finish in 1.498s, 281 probes kept.\n",
      "-- check region:757 13:113740000-113760000_:+_reg_, 39398 candidate probes\n",
      "--- 11650 probes passed check_dic selection.\n",
      "finish in 1.247s, 233 probes kept.\n",
      "-- check region:758 13:116240000-116260000_:+_reg_, 38826 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 9781 probes passed check_dic selection.\n",
      "finish in 1.131s, 197 probes kept.\n",
      "-- check region:759 13:118740000-118760000_:+_reg_, 38814 candidate probes\n",
      "--- 12022 probes passed check_dic selection.\n",
      "finish in 1.251s, 223 probes kept.\n",
      "-- check region:760 13:121240000-120421639_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:761 14:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:762 14:3740000-3760000_:+_reg_, 39738 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.829s, 0 probes kept.\n",
      "-- check region:763 14:6240000-6260000_:+_reg_, 35630 candidate probes\n",
      "--- 12 probes passed check_dic selection.\n",
      "finish in 0.765s, 1 probes kept.\n",
      "-- check region:764 14:8740000-8760000_:+_reg_, 16504 candidate probes\n",
      "--- 1200 probes passed check_dic selection.\n",
      "finish in 0.387s, 31 probes kept.\n",
      "-- check region:765 14:11240000-11260000_:+_reg_, 39312 candidate probes\n",
      "--- 13851 probes passed check_dic selection.\n",
      "finish in 1.382s, 242 probes kept.\n",
      "-- check region:766 14:13740000-13760000_:+_reg_, 38898 candidate probes\n",
      "--- 4226 probes passed check_dic selection.\n",
      "finish in 0.967s, 97 probes kept.\n",
      "-- check region:767 14:16240000-16260000_:+_reg_, 39698 candidate probes\n",
      "--- 12901 probes passed check_dic selection.\n",
      "finish in 1.289s, 272 probes kept.\n",
      "-- check region:768 14:18740000-18760000_:+_reg_, 38590 candidate probes\n",
      "--- 6574 probes passed check_dic selection.\n",
      "finish in 1.033s, 123 probes kept.\n",
      "-- check region:769 14:21240000-21260000_:+_reg_, 39688 candidate probes\n",
      "--- 11552 probes passed check_dic selection.\n",
      "finish in 1.296s, 230 probes kept.\n",
      "-- check region:770 14:23740000-23760000_:+_reg_, 39668 candidate probes\n",
      "--- 19091 probes passed check_dic selection.\n",
      "finish in 1.527s, 313 probes kept.\n",
      "-- check region:771 14:26240000-26260000_:+_reg_, 39204 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.873s, 0 probes kept.\n",
      "-- check region:772 14:28740000-28760000_:+_reg_, 39276 candidate probes\n",
      "--- 13816 probes passed check_dic selection.\n",
      "finish in 1.317s, 238 probes kept.\n",
      "-- check region:773 14:31240000-31260000_:+_reg_, 39392 candidate probes\n",
      "--- 13137 probes passed check_dic selection.\n",
      "finish in 1.340s, 293 probes kept.\n",
      "-- check region:774 14:33740000-33760000_:+_reg_, 39814 candidate probes\n",
      "--- 14698 probes passed check_dic selection.\n",
      "finish in 1.425s, 257 probes kept.\n",
      "-- check region:775 14:36240000-36260000_:+_reg_, 37356 candidate probes\n",
      "--- 9512 probes passed check_dic selection.\n",
      "finish in 1.149s, 185 probes kept.\n",
      "-- check region:776 14:38740000-38760000_:+_reg_, 35160 candidate probes\n",
      "--- 8031 probes passed check_dic selection.\n",
      "finish in 1.004s, 168 probes kept.\n",
      "-- check region:777 14:41240000-41260000_:+_reg_, 27818 candidate probes\n",
      "--- 26 probes passed check_dic selection.\n",
      "finish in 0.571s, 4 probes kept.\n",
      "-- check region:778 14:43740000-43760000_:+_reg_, 37928 candidate probes\n",
      "--- 614 probes passed check_dic selection.\n",
      "finish in 0.802s, 38 probes kept.\n",
      "-- check region:779 14:46240000-46260000_:+_reg_, 39390 candidate probes\n",
      "--- 12949 probes passed check_dic selection.\n",
      "finish in 1.336s, 233 probes kept.\n",
      "-- check region:780 14:48740000-48760000_:+_reg_, 38592 candidate probes\n",
      "--- 13364 probes passed check_dic selection.\n",
      "finish in 1.283s, 237 probes kept.\n",
      "-- check region:781 14:51240000-51260000_:+_reg_, 35302 candidate probes\n",
      "--- 6038 probes passed check_dic selection.\n",
      "finish in 0.981s, 148 probes kept.\n",
      "-- check region:782 14:53740000-53760000_:+_reg_, 35322 candidate probes\n",
      "--- 1233 probes passed check_dic selection.\n",
      "finish in 0.815s, 47 probes kept.\n",
      "-- check region:783 14:56240000-56260000_:+_reg_, 36200 candidate probes\n",
      "--- 2964 probes passed check_dic selection.\n",
      "finish in 0.828s, 72 probes kept.\n",
      "-- check region:784 14:58740000-58760000_:+_reg_, 39354 candidate probes\n",
      "--- 11740 probes passed check_dic selection.\n",
      "finish in 1.274s, 210 probes kept.\n",
      "-- check region:785 14:61240000-61260000_:+_reg_, 39176 candidate probes\n",
      "--- 12081 probes passed check_dic selection.\n",
      "finish in 1.287s, 230 probes kept.\n",
      "-- check region:786 14:63740000-63760000_:+_reg_, 39845 candidate probes\n",
      "--- 19621 probes passed check_dic selection.\n",
      "finish in 1.635s, 319 probes kept.\n",
      "-- check region:787 14:66240000-66260000_:+_reg_, 39266 candidate probes\n",
      "--- 15354 probes passed check_dic selection.\n",
      "finish in 1.358s, 252 probes kept.\n",
      "-- check region:788 14:68740000-68760000_:+_reg_, 27676 candidate probes\n",
      "--- 7711 probes passed check_dic selection.\n",
      "finish in 0.862s, 146 probes kept.\n",
      "-- check region:789 14:71240000-71260000_:+_reg_, 39498 candidate probes\n",
      "--- 17456 probes passed check_dic selection.\n",
      "finish in 1.551s, 291 probes kept.\n",
      "-- check region:790 14:73740000-73760000_:+_reg_, 36626 candidate probes\n",
      "--- 11676 probes passed check_dic selection.\n",
      "finish in 1.221s, 214 probes kept.\n",
      "-- check region:791 14:76240000-76260000_:+_reg_, 39668 candidate probes\n",
      "--- 17642 probes passed check_dic selection.\n",
      "finish in 1.547s, 281 probes kept.\n",
      "-- check region:792 14:78740000-78760000_:+_reg_, 39632 candidate probes\n",
      "--- 10523 probes passed check_dic selection.\n",
      "finish in 1.210s, 213 probes kept.\n",
      "-- check region:793 14:81240000-81260000_:+_reg_, 36640 candidate probes\n",
      "--- 7642 probes passed check_dic selection.\n",
      "finish in 1.016s, 170 probes kept.\n",
      "-- check region:794 14:83740000-83760000_:+_reg_, 38091 candidate probes\n",
      "--- 4475 probes passed check_dic selection.\n",
      "finish in 0.951s, 126 probes kept.\n",
      "-- check region:795 14:86240000-86260000_:+_reg_, 39762 candidate probes\n",
      "--- 12667 probes passed check_dic selection.\n",
      "finish in 1.338s, 232 probes kept.\n",
      "-- check region:796 14:88740000-88760000_:+_reg_, 38090 candidate probes\n",
      "--- 5709 probes passed check_dic selection.\n",
      "finish in 1.026s, 144 probes kept.\n",
      "-- check region:797 14:91240000-91260000_:+_reg_, 39598 candidate probes\n",
      "--- 7965 probes passed check_dic selection.\n",
      "finish in 1.107s, 173 probes kept.\n",
      "-- check region:798 14:93740000-93760000_:+_reg_, 39238 candidate probes\n",
      "--- 6027 probes passed check_dic selection.\n",
      "finish in 0.994s, 137 probes kept.\n",
      "-- check region:799 14:96240000-96260000_:+_reg_, 38804 candidate probes\n",
      "--- 5276 probes passed check_dic selection.\n",
      "finish in 0.950s, 132 probes kept.\n",
      "-- check region:800 14:98740000-98760000_:+_reg_, 37846 candidate probes\n",
      "--- 9250 probes passed check_dic selection.\n",
      "finish in 1.117s, 169 probes kept.\n",
      "-- check region:801 14:101240000-101260000_:+_reg_, 39122 candidate probes\n",
      "--- 10621 probes passed check_dic selection.\n",
      "finish in 1.165s, 200 probes kept.\n",
      "-- check region:802 14:103740000-103760000_:+_reg_, 37620 candidate probes\n",
      "--- 16525 probes passed check_dic selection.\n",
      "finish in 1.420s, 275 probes kept.\n",
      "-- check region:803 14:106240000-106260000_:+_reg_, 39398 candidate probes\n",
      "--- 8451 probes passed check_dic selection.\n",
      "finish in 1.138s, 188 probes kept.\n",
      "-- check region:804 14:108740000-108760000_:+_reg_, 39558 candidate probes\n",
      "--- 7981 probes passed check_dic selection.\n",
      "finish in 1.086s, 176 probes kept.\n",
      "-- check region:805 14:111240000-111260000_:+_reg_, 38598 candidate probes\n",
      "--- 5471 probes passed check_dic selection.\n",
      "finish in 0.997s, 137 probes kept.\n",
      "-- check region:806 14:113740000-113760000_:+_reg_, 38976 candidate probes\n",
      "--- 5858 probes passed check_dic selection.\n",
      "finish in 1.001s, 135 probes kept.\n",
      "-- check region:807 14:116240000-116260000_:+_reg_, 39202 candidate probes\n",
      "--- 10950 probes passed check_dic selection.\n",
      "finish in 1.213s, 199 probes kept.\n",
      "-- check region:808 14:118740000-118760000_:+_reg_, 37100 candidate probes\n",
      "--- 11529 probes passed check_dic selection.\n",
      "finish in 1.226s, 218 probes kept.\n",
      "-- check region:809 14:121240000-121260000_:+_reg_, 39750 candidate probes\n",
      "--- 19453 probes passed check_dic selection.\n",
      "finish in 1.576s, 344 probes kept.\n",
      "-- check region:810 14:123740000-123760000_:+_reg_, 37328 candidate probes\n",
      "--- 8706 probes passed check_dic selection.\n",
      "finish in 1.070s, 178 probes kept.\n",
      "-- check region:811 15:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:812 15:3740000-3760000_:+_reg_, 38166 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 9890 probes passed check_dic selection.\n",
      "finish in 1.176s, 179 probes kept.\n",
      "-- check region:813 15:6240000-6260000_:+_reg_, 37418 candidate probes\n",
      "--- 13890 probes passed check_dic selection.\n",
      "finish in 1.255s, 237 probes kept.\n",
      "-- check region:814 15:8740000-8760000_:+_reg_, 38606 candidate probes\n",
      "--- 10677 probes passed check_dic selection.\n",
      "finish in 1.201s, 186 probes kept.\n",
      "-- check region:815 15:11240000-11260000_:+_reg_, 39832 candidate probes\n",
      "--- 12304 probes passed check_dic selection.\n",
      "finish in 1.355s, 228 probes kept.\n",
      "-- check region:816 15:13740000-13760000_:+_reg_, 38692 candidate probes\n",
      "--- 8394 probes passed check_dic selection.\n",
      "finish in 1.061s, 181 probes kept.\n",
      "-- check region:817 15:16240000-16260000_:+_reg_, 39568 candidate probes\n",
      "--- 7602 probes passed check_dic selection.\n",
      "finish in 1.110s, 183 probes kept.\n",
      "-- check region:818 15:18740000-18760000_:+_reg_, 39588 candidate probes\n",
      "--- 9093 probes passed check_dic selection.\n",
      "finish in 1.160s, 198 probes kept.\n",
      "-- check region:819 15:21240000-21260000_:+_reg_, 39776 candidate probes\n",
      "--- 8322 probes passed check_dic selection.\n",
      "finish in 1.161s, 182 probes kept.\n",
      "-- check region:820 15:23740000-23760000_:+_reg_, 39276 candidate probes\n",
      "--- 4838 probes passed check_dic selection.\n",
      "finish in 0.968s, 115 probes kept.\n",
      "-- check region:821 15:26240000-26260000_:+_reg_, 22116 candidate probes\n",
      "--- 3680 probes passed check_dic selection.\n",
      "finish in 0.608s, 79 probes kept.\n",
      "-- check region:822 15:28740000-28760000_:+_reg_, 39804 candidate probes\n",
      "--- 9586 probes passed check_dic selection.\n",
      "finish in 1.121s, 196 probes kept.\n",
      "-- check region:823 15:31240000-31260000_:+_reg_, 39730 candidate probes\n",
      "--- 23702 probes passed check_dic selection.\n",
      "finish in 1.800s, 341 probes kept.\n",
      "-- check region:824 15:33740000-33760000_:+_reg_, 28890 candidate probes\n",
      "--- 5175 probes passed check_dic selection.\n",
      "finish in 0.847s, 99 probes kept.\n",
      "-- check region:825 15:36240000-36260000_:+_reg_, 39566 candidate probes\n",
      "--- 9277 probes passed check_dic selection.\n",
      "finish in 1.187s, 216 probes kept.\n",
      "-- check region:826 15:38740000-38760000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.002s, 0 probes kept.\n",
      "-- check region:827 15:41240000-41260000_:+_reg_, 28476 candidate probes\n",
      "--- 6466 probes passed check_dic selection.\n",
      "finish in 0.849s, 143 probes kept.\n",
      "-- check region:828 15:43740000-43760000_:+_reg_, 32394 candidate probes\n",
      "--- 5285 probes passed check_dic selection.\n",
      "finish in 0.845s, 106 probes kept.\n",
      "-- check region:829 15:46240000-46260000_:+_reg_, 36916 candidate probes\n",
      "--- 4528 probes passed check_dic selection.\n",
      "finish in 0.936s, 122 probes kept.\n",
      "-- check region:830 15:48740000-48760000_:+_reg_, 39698 candidate probes\n",
      "--- 7094 probes passed check_dic selection.\n",
      "finish in 1.051s, 168 probes kept.\n",
      "-- check region:831 15:51240000-51260000_:+_reg_, 38602 candidate probes\n",
      "--- 11996 probes passed check_dic selection.\n",
      "finish in 1.269s, 227 probes kept.\n",
      "-- check region:832 15:53740000-53760000_:+_reg_, 38854 candidate probes\n",
      "--- 12577 probes passed check_dic selection.\n",
      "finish in 1.280s, 232 probes kept.\n",
      "-- check region:833 15:56240000-56260000_:+_reg_, 32248 candidate probes\n",
      "--- 7511 probes passed check_dic selection.\n",
      "finish in 0.903s, 153 probes kept.\n",
      "-- check region:834 15:58740000-58760000_:+_reg_, 39580 candidate probes\n",
      "--- 18034 probes passed check_dic selection.\n",
      "finish in 1.540s, 297 probes kept.\n",
      "-- check region:835 15:61240000-61260000_:+_reg_, 37962 candidate probes\n",
      "--- 10519 probes passed check_dic selection.\n",
      "finish in 1.187s, 220 probes kept.\n",
      "-- check region:836 15:63740000-63760000_:+_reg_, 39262 candidate probes\n",
      "--- 17446 probes passed check_dic selection.\n",
      "finish in 1.541s, 296 probes kept.\n",
      "-- check region:837 15:66240000-66260000_:+_reg_, 39750 candidate probes\n",
      "--- 19359 probes passed check_dic selection.\n",
      "finish in 1.585s, 312 probes kept.\n",
      "-- check region:838 15:68740000-68760000_:+_reg_, 39440 candidate probes\n",
      "--- 14429 probes passed check_dic selection.\n",
      "finish in 1.385s, 261 probes kept.\n",
      "-- check region:839 15:71240000-71260000_:+_reg_, 39664 candidate probes\n",
      "--- 12388 probes passed check_dic selection.\n",
      "finish in 1.324s, 240 probes kept.\n",
      "-- check region:840 15:73740000-73760000_:+_reg_, 39838 candidate probes\n",
      "--- 18699 probes passed check_dic selection.\n",
      "finish in 1.580s, 333 probes kept.\n",
      "-- check region:841 15:76240000-76260000_:+_reg_, 39796 candidate probes\n",
      "--- 18625 probes passed check_dic selection.\n",
      "finish in 1.535s, 328 probes kept.\n",
      "-- check region:842 15:78740000-78760000_:+_reg_, 39368 candidate probes\n",
      "--- 18376 probes passed check_dic selection.\n",
      "finish in 1.545s, 299 probes kept.\n",
      "-- check region:843 15:81240000-81260000_:+_reg_, 39810 candidate probes\n",
      "--- 14037 probes passed check_dic selection.\n",
      "finish in 1.365s, 248 probes kept.\n",
      "-- check region:844 15:83740000-83760000_:+_reg_, 38292 candidate probes\n",
      "--- 16010 probes passed check_dic selection.\n",
      "finish in 1.372s, 268 probes kept.\n",
      "-- check region:845 15:86240000-86260000_:+_reg_, 38816 candidate probes\n",
      "--- 17552 probes passed check_dic selection.\n",
      "finish in 1.508s, 277 probes kept.\n",
      "-- check region:846 15:88740000-88760000_:+_reg_, 39256 candidate probes\n",
      "--- 14514 probes passed check_dic selection.\n",
      "finish in 1.421s, 237 probes kept.\n",
      "-- check region:847 15:91240000-91260000_:+_reg_, 39920 candidate probes\n",
      "--- 14095 probes passed check_dic selection.\n",
      "finish in 1.344s, 258 probes kept.\n",
      "-- check region:848 15:93740000-93760000_:+_reg_, 38242 candidate probes\n",
      "--- 15431 probes passed check_dic selection.\n",
      "finish in 1.435s, 262 probes kept.\n",
      "-- check region:849 15:96240000-96260000_:+_reg_, 39674 candidate probes\n",
      "--- 10977 probes passed check_dic selection.\n",
      "finish in 1.269s, 227 probes kept.\n",
      "-- check region:850 15:98740000-98760000_:+_reg_, 38552 candidate probes\n",
      "--- 13232 probes passed check_dic selection.\n",
      "finish in 1.305s, 248 probes kept.\n",
      "-- check region:851 15:101240000-101260000_:+_reg_, 39564 candidate probes\n",
      "--- 18203 probes passed check_dic selection.\n",
      "finish in 1.478s, 289 probes kept.\n",
      "-- check region:852 15:103740000-103760000_:+_reg_, 39762 candidate probes\n",
      "--- 4961 probes passed check_dic selection.\n",
      "finish in 0.972s, 148 probes kept.\n",
      "-- check region:853 16:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:854 16:3740000-3760000_:+_reg_, 37810 candidate probes\n",
      "--- 11333 probes passed check_dic selection.\n",
      "finish in 1.235s, 234 probes kept.\n",
      "-- check region:855 16:6240000-6260000_:+_reg_, 35388 candidate probes\n",
      "--- 11232 probes passed check_dic selection.\n",
      "finish in 1.154s, 192 probes kept.\n",
      "-- check region:856 16:8740000-8760000_:+_reg_, 39050 candidate probes\n",
      "--- 12192 probes passed check_dic selection.\n",
      "finish in 1.264s, 237 probes kept.\n",
      "-- check region:857 16:11240000-11260000_:+_reg_, 37926 candidate probes\n",
      "--- 11332 probes passed check_dic selection.\n",
      "finish in 1.205s, 231 probes kept.\n",
      "-- check region:858 16:13740000-13760000_:+_reg_, 37706 candidate probes\n",
      "--- 9057 probes passed check_dic selection.\n",
      "finish in 1.169s, 204 probes kept.\n",
      "-- check region:859 16:16240000-16260000_:+_reg_, 39758 candidate probes\n",
      "--- 12269 probes passed check_dic selection.\n",
      "finish in 1.311s, 234 probes kept.\n",
      "-- check region:860 16:18740000-18760000_:+_reg_, 38602 candidate probes\n",
      "--- 12186 probes passed check_dic selection.\n",
      "finish in 1.329s, 217 probes kept.\n",
      "-- check region:861 16:21240000-21260000_:+_reg_, 39674 candidate probes\n",
      "--- 19218 probes passed check_dic selection.\n",
      "finish in 1.535s, 323 probes kept.\n",
      "-- check region:862 16:23740000-23760000_:+_reg_, 31484 candidate probes\n",
      "--- 5671 probes passed check_dic selection.\n",
      "finish in 0.884s, 101 probes kept.\n",
      "-- check region:863 16:26240000-26260000_:+_reg_, 32204 candidate probes\n",
      "--- 7473 probes passed check_dic selection.\n",
      "finish in 0.946s, 150 probes kept.\n",
      "-- check region:864 16:28740000-28760000_:+_reg_, 39626 candidate probes\n",
      "--- 14952 probes passed check_dic selection.\n",
      "finish in 1.372s, 258 probes kept.\n",
      "-- check region:865 16:31240000-31260000_:+_reg_, 38962 candidate probes\n",
      "--- 11610 probes passed check_dic selection.\n",
      "finish in 1.242s, 259 probes kept.\n",
      "-- check region:866 16:33740000-33760000_:+_reg_, 39602 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 15046 probes passed check_dic selection.\n",
      "finish in 1.420s, 251 probes kept.\n",
      "-- check region:867 16:36240000-36260000_:+_reg_, 39244 candidate probes\n",
      "--- 5525 probes passed check_dic selection.\n",
      "finish in 1.008s, 145 probes kept.\n",
      "-- check region:868 16:38740000-38760000_:+_reg_, 39624 candidate probes\n",
      "--- 17427 probes passed check_dic selection.\n",
      "finish in 1.495s, 294 probes kept.\n",
      "-- check region:869 16:41240000-41260000_:+_reg_, 38866 candidate probes\n",
      "--- 11142 probes passed check_dic selection.\n",
      "finish in 1.248s, 213 probes kept.\n",
      "-- check region:870 16:43740000-43760000_:+_reg_, 39568 candidate probes\n",
      "--- 14845 probes passed check_dic selection.\n",
      "finish in 1.375s, 257 probes kept.\n",
      "-- check region:871 16:46240000-46260000_:+_reg_, 36984 candidate probes\n",
      "--- 9896 probes passed check_dic selection.\n",
      "finish in 1.133s, 184 probes kept.\n",
      "-- check region:872 16:48740000-48760000_:+_reg_, 38402 candidate probes\n",
      "--- 9802 probes passed check_dic selection.\n",
      "finish in 1.160s, 203 probes kept.\n",
      "-- check region:873 16:51240000-51260000_:+_reg_, 27026 candidate probes\n",
      "--- 4516 probes passed check_dic selection.\n",
      "finish in 0.745s, 93 probes kept.\n",
      "-- check region:874 16:53740000-53760000_:+_reg_, 37610 candidate probes\n",
      "--- 5328 probes passed check_dic selection.\n",
      "finish in 1.001s, 116 probes kept.\n",
      "-- check region:875 16:56240000-56260000_:+_reg_, 37058 candidate probes\n",
      "--- 11194 probes passed check_dic selection.\n",
      "finish in 1.163s, 217 probes kept.\n",
      "-- check region:876 16:58740000-58760000_:+_reg_, 39356 candidate probes\n",
      "--- 8727 probes passed check_dic selection.\n",
      "finish in 1.137s, 196 probes kept.\n",
      "-- check region:877 16:61240000-61260000_:+_reg_, 38662 candidate probes\n",
      "--- 2772 probes passed check_dic selection.\n",
      "finish in 0.958s, 76 probes kept.\n",
      "-- check region:878 16:63740000-63760000_:+_reg_, 39134 candidate probes\n",
      "--- 9627 probes passed check_dic selection.\n",
      "finish in 1.199s, 194 probes kept.\n",
      "-- check region:879 16:66240000-66260000_:+_reg_, 34742 candidate probes\n",
      "--- 5819 probes passed check_dic selection.\n",
      "finish in 0.963s, 133 probes kept.\n",
      "-- check region:880 16:68740000-68760000_:+_reg_, 29754 candidate probes\n",
      "--- 3714 probes passed check_dic selection.\n",
      "finish in 0.749s, 88 probes kept.\n",
      "-- check region:881 16:71240000-71260000_:+_reg_, 38538 candidate probes\n",
      "--- 8752 probes passed check_dic selection.\n",
      "finish in 1.133s, 192 probes kept.\n",
      "-- check region:882 16:73740000-73760000_:+_reg_, 39645 candidate probes\n",
      "--- 11027 probes passed check_dic selection.\n",
      "finish in 1.225s, 220 probes kept.\n",
      "-- check region:883 16:76240000-76260000_:+_reg_, 39920 candidate probes\n",
      "--- 15340 probes passed check_dic selection.\n",
      "finish in 1.465s, 277 probes kept.\n",
      "-- check region:884 16:78740000-78760000_:+_reg_, 37696 candidate probes\n",
      "--- 9816 probes passed check_dic selection.\n",
      "finish in 1.130s, 192 probes kept.\n",
      "-- check region:885 16:81240000-81260000_:+_reg_, 39452 candidate probes\n",
      "--- 8986 probes passed check_dic selection.\n",
      "finish in 1.112s, 180 probes kept.\n",
      "-- check region:886 16:83740000-83760000_:+_reg_, 35674 candidate probes\n",
      "--- 6306 probes passed check_dic selection.\n",
      "finish in 0.984s, 145 probes kept.\n",
      "-- check region:887 16:86240000-86260000_:+_reg_, 39876 candidate probes\n",
      "--- 9014 probes passed check_dic selection.\n",
      "finish in 1.169s, 210 probes kept.\n",
      "-- check region:888 16:88740000-88760000_:+_reg_, 39214 candidate probes\n",
      "--- 10792 probes passed check_dic selection.\n",
      "finish in 1.170s, 220 probes kept.\n",
      "-- check region:889 16:91240000-91260000_:+_reg_, 37778 candidate probes\n",
      "--- 13172 probes passed check_dic selection.\n",
      "finish in 1.361s, 194 probes kept.\n",
      "-- check region:890 16:93740000-93760000_:+_reg_, 39496 candidate probes\n",
      "--- 17734 probes passed check_dic selection.\n",
      "finish in 1.506s, 295 probes kept.\n",
      "-- check region:891 16:96240000-96260000_:+_reg_, 38952 candidate probes\n",
      "--- 9519 probes passed check_dic selection.\n",
      "finish in 1.226s, 173 probes kept.\n",
      "-- check region:892 16:98740000-98207768_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:893 17:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:894 17:3740000-3760000_:+_reg_, 39756 candidate probes\n",
      "--- 11324 probes passed check_dic selection.\n",
      "finish in 1.241s, 228 probes kept.\n",
      "-- check region:895 17:6240000-6260000_:+_reg_, 39618 candidate probes\n",
      "--- 324 probes passed check_dic selection.\n",
      "finish in 0.838s, 12 probes kept.\n",
      "-- check region:896 17:8740000-8760000_:+_reg_, 39650 candidate probes\n",
      "--- 20748 probes passed check_dic selection.\n",
      "finish in 1.647s, 316 probes kept.\n",
      "-- check region:897 17:11240000-11260000_:+_reg_, 39822 candidate probes\n",
      "--- 14192 probes passed check_dic selection.\n",
      "finish in 1.358s, 259 probes kept.\n",
      "-- check region:898 17:13740000-13760000_:+_reg_, 39678 candidate probes\n",
      "--- 15934 probes passed check_dic selection.\n",
      "finish in 1.424s, 275 probes kept.\n",
      "-- check region:899 17:16240000-16260000_:+_reg_, 39270 candidate probes\n",
      "--- 7903 probes passed check_dic selection.\n",
      "finish in 1.086s, 179 probes kept.\n",
      "-- check region:900 17:18740000-18760000_:+_reg_, 32992 candidate probes\n",
      "--- 1825 probes passed check_dic selection.\n",
      "finish in 0.736s, 67 probes kept.\n",
      "-- check region:901 17:21240000-21260000_:+_reg_, 34834 candidate probes\n",
      "--- 3015 probes passed check_dic selection.\n",
      "finish in 0.849s, 95 probes kept.\n",
      "-- check region:902 17:23740000-23760000_:+_reg_, 39592 candidate probes\n",
      "--- 16025 probes passed check_dic selection.\n",
      "finish in 1.487s, 288 probes kept.\n",
      "-- check region:903 17:26240000-26260000_:+_reg_, 38942 candidate probes\n",
      "--- 9109 probes passed check_dic selection.\n",
      "finish in 1.142s, 195 probes kept.\n",
      "-- check region:904 17:28740000-28760000_:+_reg_, 38796 candidate probes\n",
      "--- 14402 probes passed check_dic selection.\n",
      "finish in 1.373s, 253 probes kept.\n",
      "-- check region:905 17:31240000-31260000_:+_reg_, 38748 candidate probes\n",
      "--- 12699 probes passed check_dic selection.\n",
      "finish in 1.315s, 237 probes kept.\n",
      "-- check region:906 17:33740000-33760000_:+_reg_, 39256 candidate probes\n",
      "--- 14822 probes passed check_dic selection.\n",
      "finish in 1.364s, 254 probes kept.\n",
      "-- check region:907 17:36240000-36260000_:+_reg_, 39138 candidate probes\n",
      "--- 9458 probes passed check_dic selection.\n",
      "finish in 1.152s, 200 probes kept.\n",
      "-- check region:908 17:38740000-38760000_:+_reg_, 38734 candidate probes\n",
      "--- 2438 probes passed check_dic selection.\n",
      "finish in 0.882s, 87 probes kept.\n",
      "-- check region:909 17:41240000-41260000_:+_reg_, 37976 candidate probes\n",
      "--- 8432 probes passed check_dic selection.\n",
      "finish in 1.065s, 180 probes kept.\n",
      "-- check region:910 17:43740000-43760000_:+_reg_, 39426 candidate probes\n",
      "--- 13333 probes passed check_dic selection.\n",
      "finish in 1.378s, 254 probes kept.\n",
      "-- check region:911 17:46240000-46260000_:+_reg_, 39642 candidate probes\n",
      "--- 16382 probes passed check_dic selection.\n",
      "finish in 1.445s, 331 probes kept.\n",
      "-- check region:912 17:48740000-48760000_:+_reg_, 39682 candidate probes\n",
      "--- 19661 probes passed check_dic selection.\n",
      "finish in 1.652s, 298 probes kept.\n",
      "-- check region:913 17:51240000-51260000_:+_reg_, 39514 candidate probes\n",
      "--- 11183 probes passed check_dic selection.\n",
      "finish in 1.246s, 216 probes kept.\n",
      "-- check region:914 17:53740000-53760000_:+_reg_, 37404 candidate probes\n",
      "--- 2576 probes passed check_dic selection.\n",
      "finish in 0.861s, 72 probes kept.\n",
      "-- check region:915 17:56240000-56260000_:+_reg_, 39310 candidate probes\n",
      "--- 13241 probes passed check_dic selection.\n",
      "finish in 1.335s, 256 probes kept.\n",
      "-- check region:916 17:58740000-58760000_:+_reg_, 37222 candidate probes\n",
      "--- 9588 probes passed check_dic selection.\n",
      "finish in 1.140s, 182 probes kept.\n",
      "-- check region:917 17:61240000-61260000_:+_reg_, 39860 candidate probes\n",
      "--- 7000 probes passed check_dic selection.\n",
      "finish in 1.051s, 159 probes kept.\n",
      "-- check region:918 17:63740000-63760000_:+_reg_, 37144 candidate probes\n",
      "--- 9456 probes passed check_dic selection.\n",
      "finish in 1.162s, 163 probes kept.\n",
      "-- check region:919 17:66240000-66260000_:+_reg_, 39728 candidate probes\n",
      "--- 17798 probes passed check_dic selection.\n",
      "finish in 1.554s, 273 probes kept.\n",
      "-- check region:920 17:68740000-68760000_:+_reg_, 39920 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 18489 probes passed check_dic selection.\n",
      "finish in 1.608s, 287 probes kept.\n",
      "-- check region:921 17:71240000-71260000_:+_reg_, 39780 candidate probes\n",
      "--- 17661 probes passed check_dic selection.\n",
      "finish in 1.535s, 325 probes kept.\n",
      "-- check region:922 17:73740000-73760000_:+_reg_, 39698 candidate probes\n",
      "--- 18372 probes passed check_dic selection.\n",
      "finish in 1.559s, 287 probes kept.\n",
      "-- check region:923 17:76240000-76260000_:+_reg_, 39776 candidate probes\n",
      "--- 6692 probes passed check_dic selection.\n",
      "finish in 1.027s, 160 probes kept.\n",
      "-- check region:924 17:78740000-78760000_:+_reg_, 39434 candidate probes\n",
      "--- 9557 probes passed check_dic selection.\n",
      "finish in 1.208s, 197 probes kept.\n",
      "-- check region:925 17:81240000-81260000_:+_reg_, 29708 candidate probes\n",
      "--- 3502 probes passed check_dic selection.\n",
      "finish in 0.751s, 79 probes kept.\n",
      "-- check region:926 17:83740000-83760000_:+_reg_, 39482 candidate probes\n",
      "--- 14944 probes passed check_dic selection.\n",
      "finish in 1.394s, 255 probes kept.\n",
      "-- check region:927 17:86240000-86260000_:+_reg_, 39312 candidate probes\n",
      "--- 21699 probes passed check_dic selection.\n",
      "finish in 1.665s, 305 probes kept.\n",
      "-- check region:928 17:88740000-88760000_:+_reg_, 39108 candidate probes\n",
      "--- 16435 probes passed check_dic selection.\n",
      "finish in 1.446s, 266 probes kept.\n",
      "-- check region:929 17:91240000-91260000_:+_reg_, 39841 candidate probes\n",
      "--- 5219 probes passed check_dic selection.\n",
      "finish in 1.019s, 125 probes kept.\n",
      "-- check region:930 17:93740000-93760000_:+_reg_, 33280 candidate probes\n",
      "--- 3319 probes passed check_dic selection.\n",
      "finish in 0.795s, 81 probes kept.\n",
      "-- check region:931 18:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:932 18:3740000-3760000_:+_reg_, 38594 candidate probes\n",
      "--- 2838 probes passed check_dic selection.\n",
      "finish in 0.892s, 66 probes kept.\n",
      "-- check region:933 18:6240000-6260000_:+_reg_, 39372 candidate probes\n",
      "--- 9922 probes passed check_dic selection.\n",
      "finish in 1.205s, 186 probes kept.\n",
      "-- check region:934 18:8740000-8760000_:+_reg_, 37538 candidate probes\n",
      "--- 10480 probes passed check_dic selection.\n",
      "finish in 1.159s, 198 probes kept.\n",
      "-- check region:935 18:11240000-11260000_:+_reg_, 37382 candidate probes\n",
      "--- 14306 probes passed check_dic selection.\n",
      "finish in 1.311s, 230 probes kept.\n",
      "-- check region:936 18:13740000-13760000_:+_reg_, 39904 candidate probes\n",
      "--- 17337 probes passed check_dic selection.\n",
      "finish in 1.478s, 293 probes kept.\n",
      "-- check region:937 18:16240000-16260000_:+_reg_, 39494 candidate probes\n",
      "--- 10636 probes passed check_dic selection.\n",
      "finish in 1.233s, 200 probes kept.\n",
      "-- check region:938 18:18740000-18760000_:+_reg_, 39220 candidate probes\n",
      "--- 5975 probes passed check_dic selection.\n",
      "finish in 1.025s, 164 probes kept.\n",
      "-- check region:939 18:21240000-21260000_:+_reg_, 39844 candidate probes\n",
      "--- 16690 probes passed check_dic selection.\n",
      "finish in 1.450s, 292 probes kept.\n",
      "-- check region:940 18:23740000-23760000_:+_reg_, 39588 candidate probes\n",
      "--- 14538 probes passed check_dic selection.\n",
      "finish in 1.358s, 258 probes kept.\n",
      "-- check region:941 18:26240000-26260000_:+_reg_, 38758 candidate probes\n",
      "--- 8848 probes passed check_dic selection.\n",
      "finish in 1.093s, 184 probes kept.\n",
      "-- check region:942 18:28740000-28760000_:+_reg_, 28796 candidate probes\n",
      "--- 6508 probes passed check_dic selection.\n",
      "finish in 0.843s, 136 probes kept.\n",
      "-- check region:943 18:31240000-31260000_:+_reg_, 38502 candidate probes\n",
      "--- 8689 probes passed check_dic selection.\n",
      "finish in 1.095s, 173 probes kept.\n",
      "-- check region:944 18:33740000-33760000_:+_reg_, 39208 candidate probes\n",
      "--- 13964 probes passed check_dic selection.\n",
      "finish in 1.300s, 257 probes kept.\n",
      "-- check region:945 18:36240000-36260000_:+_reg_, 39376 candidate probes\n",
      "--- 9764 probes passed check_dic selection.\n",
      "finish in 1.193s, 216 probes kept.\n",
      "-- check region:946 18:38740000-38760000_:+_reg_, 39258 candidate probes\n",
      "--- 15950 probes passed check_dic selection.\n",
      "finish in 1.423s, 262 probes kept.\n",
      "-- check region:947 18:41240000-41260000_:+_reg_, 39640 candidate probes\n",
      "--- 13017 probes passed check_dic selection.\n",
      "finish in 1.330s, 244 probes kept.\n",
      "-- check region:948 18:43740000-43760000_:+_reg_, 39278 candidate probes\n",
      "--- 9592 probes passed check_dic selection.\n",
      "finish in 1.168s, 187 probes kept.\n",
      "-- check region:949 18:46240000-46260000_:+_reg_, 39690 candidate probes\n",
      "--- 11542 probes passed check_dic selection.\n",
      "finish in 1.305s, 247 probes kept.\n",
      "-- check region:950 18:48740000-48760000_:+_reg_, 36878 candidate probes\n",
      "--- 6597 probes passed check_dic selection.\n",
      "finish in 1.003s, 154 probes kept.\n",
      "-- check region:951 18:51240000-51260000_:+_reg_, 39116 candidate probes\n",
      "--- 11556 probes passed check_dic selection.\n",
      "finish in 1.191s, 212 probes kept.\n",
      "-- check region:952 18:53740000-53760000_:+_reg_, 39816 candidate probes\n",
      "--- 15171 probes passed check_dic selection.\n",
      "finish in 1.463s, 260 probes kept.\n",
      "-- check region:953 18:56240000-56260000_:+_reg_, 39678 candidate probes\n",
      "--- 13852 probes passed check_dic selection.\n",
      "finish in 1.399s, 244 probes kept.\n",
      "-- check region:954 18:58740000-58760000_:+_reg_, 39756 candidate probes\n",
      "--- 9610 probes passed check_dic selection.\n",
      "finish in 1.206s, 190 probes kept.\n",
      "-- check region:955 18:61240000-61260000_:+_reg_, 39682 candidate probes\n",
      "--- 15908 probes passed check_dic selection.\n",
      "finish in 1.388s, 270 probes kept.\n",
      "-- check region:956 18:63740000-63760000_:+_reg_, 39632 candidate probes\n",
      "--- 9205 probes passed check_dic selection.\n",
      "finish in 1.174s, 196 probes kept.\n",
      "-- check region:957 18:66240000-66260000_:+_reg_, 39604 candidate probes\n",
      "--- 13245 probes passed check_dic selection.\n",
      "finish in 1.322s, 238 probes kept.\n",
      "-- check region:958 18:68740000-68760000_:+_reg_, 39646 candidate probes\n",
      "--- 14682 probes passed check_dic selection.\n",
      "finish in 1.392s, 254 probes kept.\n",
      "-- check region:959 18:71240000-71260000_:+_reg_, 37552 candidate probes\n",
      "--- 8410 probes passed check_dic selection.\n",
      "finish in 1.074s, 209 probes kept.\n",
      "-- check region:960 18:73740000-73760000_:+_reg_, 39876 candidate probes\n",
      "--- 16625 probes passed check_dic selection.\n",
      "finish in 1.559s, 288 probes kept.\n",
      "-- check region:961 18:76240000-76260000_:+_reg_, 39418 candidate probes\n",
      "--- 17523 probes passed check_dic selection.\n",
      "finish in 1.474s, 287 probes kept.\n",
      "-- check region:962 18:78740000-78760000_:+_reg_, 39440 candidate probes\n",
      "--- 13741 probes passed check_dic selection.\n",
      "finish in 1.334s, 278 probes kept.\n",
      "-- check region:963 18:81240000-81260000_:+_reg_, 38680 candidate probes\n",
      "--- 18109 probes passed check_dic selection.\n",
      "finish in 1.477s, 282 probes kept.\n",
      "-- check region:964 18:83740000-83760000_:+_reg_, 39298 candidate probes\n",
      "--- 16151 probes passed check_dic selection.\n",
      "finish in 1.442s, 269 probes kept.\n",
      "-- check region:965 18:86240000-86260000_:+_reg_, 35142 candidate probes\n",
      "--- 6480 probes passed check_dic selection.\n",
      "finish in 0.984s, 125 probes kept.\n",
      "-- check region:966 18:88740000-88760000_:+_reg_, 39562 candidate probes\n",
      "--- 10960 probes passed check_dic selection.\n",
      "finish in 1.243s, 213 probes kept.\n",
      "-- check region:967 18:91240000-90702639_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:968 19:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:969 19:3740000-3760000_:+_reg_, 39076 candidate probes\n",
      "--- 9578 probes passed check_dic selection.\n",
      "finish in 1.198s, 182 probes kept.\n",
      "-- check region:970 19:6240000-6260000_:+_reg_, 38952 candidate probes\n",
      "--- 17177 probes passed check_dic selection.\n",
      "finish in 1.520s, 306 probes kept.\n",
      "-- check region:971 19:8740000-8760000_:+_reg_, 39150 candidate probes\n",
      "--- 12615 probes passed check_dic selection.\n",
      "finish in 1.322s, 262 probes kept.\n",
      "-- check region:972 19:11240000-11260000_:+_reg_, 39646 candidate probes\n",
      "--- 10074 probes passed check_dic selection.\n",
      "finish in 1.222s, 235 probes kept.\n",
      "-- check region:973 19:13740000-13760000_:+_reg_, 39372 candidate probes\n",
      "--- 4433 probes passed check_dic selection.\n",
      "finish in 0.956s, 121 probes kept.\n",
      "-- check region:974 19:16240000-16260000_:+_reg_, 39638 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 16292 probes passed check_dic selection.\n",
      "finish in 1.415s, 283 probes kept.\n",
      "-- check region:975 19:18740000-18760000_:+_reg_, 39570 candidate probes\n",
      "--- 15851 probes passed check_dic selection.\n",
      "finish in 1.471s, 276 probes kept.\n",
      "-- check region:976 19:21240000-21260000_:+_reg_, 32768 candidate probes\n",
      "--- 9493 probes passed check_dic selection.\n",
      "finish in 1.090s, 188 probes kept.\n",
      "-- check region:977 19:23740000-23760000_:+_reg_, 35216 candidate probes\n",
      "--- 12692 probes passed check_dic selection.\n",
      "finish in 1.212s, 211 probes kept.\n",
      "-- check region:978 19:26240000-26260000_:+_reg_, 33520 candidate probes\n",
      "--- 9967 probes passed check_dic selection.\n",
      "finish in 1.104s, 190 probes kept.\n",
      "-- check region:979 19:28740000-28760000_:+_reg_, 38430 candidate probes\n",
      "--- 14254 probes passed check_dic selection.\n",
      "finish in 1.395s, 227 probes kept.\n",
      "-- check region:980 19:31240000-31260000_:+_reg_, 39812 candidate probes\n",
      "--- 12474 probes passed check_dic selection.\n",
      "finish in 1.305s, 240 probes kept.\n",
      "-- check region:981 19:33740000-33760000_:+_reg_, 34598 candidate probes\n",
      "--- 677 probes passed check_dic selection.\n",
      "finish in 0.725s, 29 probes kept.\n",
      "-- check region:982 19:36240000-36260000_:+_reg_, 39384 candidate probes\n",
      "--- 14472 probes passed check_dic selection.\n",
      "finish in 1.366s, 251 probes kept.\n",
      "-- check region:983 19:38740000-38760000_:+_reg_, 39362 candidate probes\n",
      "--- 11692 probes passed check_dic selection.\n",
      "finish in 1.272s, 232 probes kept.\n",
      "-- check region:984 19:41240000-41260000_:+_reg_, 39818 candidate probes\n",
      "--- 12959 probes passed check_dic selection.\n",
      "finish in 1.325s, 234 probes kept.\n",
      "-- check region:985 19:43740000-43760000_:+_reg_, 39494 candidate probes\n",
      "--- 12132 probes passed check_dic selection.\n",
      "finish in 1.272s, 241 probes kept.\n",
      "-- check region:986 19:46240000-46260000_:+_reg_, 39398 candidate probes\n",
      "--- 15572 probes passed check_dic selection.\n",
      "finish in 1.478s, 269 probes kept.\n",
      "-- check region:987 19:48740000-48760000_:+_reg_, 37386 candidate probes\n",
      "--- 13965 probes passed check_dic selection.\n",
      "finish in 1.309s, 237 probes kept.\n",
      "-- check region:988 19:51240000-51260000_:+_reg_, 39172 candidate probes\n",
      "--- 6383 probes passed check_dic selection.\n",
      "finish in 1.011s, 139 probes kept.\n",
      "-- check region:989 19:53740000-53760000_:+_reg_, 39614 candidate probes\n",
      "--- 17877 probes passed check_dic selection.\n",
      "finish in 1.536s, 284 probes kept.\n",
      "-- check region:990 19:56240000-56260000_:+_reg_, 37304 candidate probes\n",
      "--- 15824 probes passed check_dic selection.\n",
      "finish in 1.382s, 268 probes kept.\n",
      "-- check region:991 19:58740000-58760000_:+_reg_, 39778 candidate probes\n",
      "--- 16079 probes passed check_dic selection.\n",
      "finish in 1.415s, 265 probes kept.\n",
      "-- check region:992 19:61240000-61260000_:+_reg_, 33396 candidate probes\n",
      "--- 4415 probes passed check_dic selection.\n",
      "finish in 0.901s, 100 probes kept.\n",
      "-- check region:993 X:1240000-1260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:994 X:3740000-3760000_:+_reg_, 39388 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.814s, 0 probes kept.\n",
      "-- check region:995 X:6240000-6260000_:+_reg_, 38216 candidate probes\n",
      "--- 11622 probes passed check_dic selection.\n",
      "finish in 1.235s, 224 probes kept.\n",
      "-- check region:996 X:8740000-8760000_:+_reg_, 34236 candidate probes\n",
      "--- 1030 probes passed check_dic selection.\n",
      "finish in 0.761s, 47 probes kept.\n",
      "-- check region:997 X:11240000-11260000_:+_reg_, 39854 candidate probes\n",
      "--- 19071 probes passed check_dic selection.\n",
      "finish in 1.608s, 305 probes kept.\n",
      "-- check region:998 X:13740000-13760000_:+_reg_, 38158 candidate probes\n",
      "--- 6767 probes passed check_dic selection.\n",
      "finish in 1.021s, 149 probes kept.\n",
      "-- check region:999 X:16240000-16260000_:+_reg_, 39078 candidate probes\n",
      "--- 5214 probes passed check_dic selection.\n",
      "finish in 0.975s, 133 probes kept.\n",
      "-- check region:1000 X:18740000-18760000_:+_reg_, 37248 candidate probes\n",
      "--- 1704 probes passed check_dic selection.\n",
      "finish in 0.881s, 55 probes kept.\n",
      "-- check region:1001 X:21240000-21260000_:+_reg_, 22263 candidate probes\n",
      "--- 2332 probes passed check_dic selection.\n",
      "finish in 0.529s, 69 probes kept.\n",
      "-- check region:1002 X:23740000-23760000_:+_reg_, 38750 candidate probes\n",
      "--- 4266 probes passed check_dic selection.\n",
      "finish in 0.912s, 117 probes kept.\n",
      "-- check region:1003 X:26240000-26260000_:+_reg_, 36808 candidate probes\n",
      "--- 32 probes passed check_dic selection.\n",
      "finish in 0.743s, 2 probes kept.\n",
      "-- check region:1004 X:28740000-28760000_:+_reg_, 38082 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.791s, 0 probes kept.\n",
      "-- check region:1005 X:31240000-31260000_:+_reg_, 39628 candidate probes\n",
      "--- 6 probes passed check_dic selection.\n",
      "finish in 0.853s, 1 probes kept.\n",
      "-- check region:1006 X:33740000-33760000_:+_reg_, 39888 candidate probes\n",
      "--- 24 probes passed check_dic selection.\n",
      "finish in 0.852s, 3 probes kept.\n",
      "-- check region:1007 X:36240000-36260000_:+_reg_, 39501 candidate probes\n",
      "--- 10425 probes passed check_dic selection.\n",
      "finish in 1.175s, 209 probes kept.\n",
      "-- check region:1008 X:38740000-38760000_:+_reg_, 36260 candidate probes\n",
      "--- 7342 probes passed check_dic selection.\n",
      "finish in 1.005s, 136 probes kept.\n",
      "-- check region:1009 X:41240000-41260000_:+_reg_, 38458 candidate probes\n",
      "--- 6829 probes passed check_dic selection.\n",
      "finish in 1.004s, 152 probes kept.\n",
      "-- check region:1010 X:43740000-43760000_:+_reg_, 38930 candidate probes\n",
      "--- 4058 probes passed check_dic selection.\n",
      "finish in 0.949s, 110 probes kept.\n",
      "-- check region:1011 X:46240000-46260000_:+_reg_, 37398 candidate probes\n",
      "--- 11536 probes passed check_dic selection.\n",
      "finish in 1.169s, 232 probes kept.\n",
      "-- check region:1012 X:48740000-48760000_:+_reg_, 24988 candidate probes\n",
      "--- 1240 probes passed check_dic selection.\n",
      "finish in 0.559s, 31 probes kept.\n",
      "-- check region:1013 X:51240000-51260000_:+_reg_, 34320 candidate probes\n",
      "--- 5946 probes passed check_dic selection.\n",
      "finish in 0.939s, 146 probes kept.\n",
      "-- check region:1014 X:53740000-53760000_:+_reg_, 38906 candidate probes\n",
      "--- 11317 probes passed check_dic selection.\n",
      "finish in 1.199s, 211 probes kept.\n",
      "-- check region:1015 X:56240000-56260000_:+_reg_, 35574 candidate probes\n",
      "--- 2319 probes passed check_dic selection.\n",
      "finish in 0.827s, 65 probes kept.\n",
      "-- check region:1016 X:58740000-58760000_:+_reg_, 39644 candidate probes\n",
      "--- 8136 probes passed check_dic selection.\n",
      "finish in 1.100s, 176 probes kept.\n",
      "-- check region:1017 X:61240000-61260000_:+_reg_, 24670 candidate probes\n",
      "--- 5128 probes passed check_dic selection.\n",
      "finish in 0.726s, 117 probes kept.\n",
      "-- check region:1018 X:63740000-63760000_:+_reg_, 38648 candidate probes\n",
      "--- 5888 probes passed check_dic selection.\n",
      "finish in 1.039s, 145 probes kept.\n",
      "-- check region:1019 X:66240000-66260000_:+_reg_, 38318 candidate probes\n",
      "--- 9018 probes passed check_dic selection.\n",
      "finish in 1.088s, 204 probes kept.\n",
      "-- check region:1020 X:68740000-68760000_:+_reg_, 37752 candidate probes\n",
      "--- 5659 probes passed check_dic selection.\n",
      "finish in 0.997s, 133 probes kept.\n",
      "-- check region:1021 X:71240000-71260000_:+_reg_, 38048 candidate probes\n",
      "--- 8593 probes passed check_dic selection.\n",
      "finish in 1.119s, 173 probes kept.\n",
      "-- check region:1022 X:73740000-73760000_:+_reg_, 39014 candidate probes\n",
      "--- 10782 probes passed check_dic selection.\n",
      "finish in 1.210s, 196 probes kept.\n",
      "-- check region:1023 X:76240000-76260000_:+_reg_, 33232 candidate probes\n",
      "--- 5407 probes passed check_dic selection.\n",
      "finish in 0.854s, 112 probes kept.\n",
      "-- check region:1024 X:78740000-78760000_:+_reg_, 39128 candidate probes\n",
      "--- 3284 probes passed check_dic selection.\n",
      "finish in 0.908s, 87 probes kept.\n",
      "-- check region:1025 X:81240000-81260000_:+_reg_, 27232 candidate probes\n",
      "--- 1886 probes passed check_dic selection.\n",
      "finish in 0.618s, 65 probes kept.\n",
      "-- check region:1026 X:83740000-83760000_:+_reg_, 39670 candidate probes\n",
      "--- 4982 probes passed check_dic selection.\n",
      "finish in 0.943s, 130 probes kept.\n",
      "-- check region:1027 X:86240000-86260000_:+_reg_, 39366 candidate probes\n",
      "--- 2647 probes passed check_dic selection.\n",
      "finish in 0.924s, 78 probes kept.\n",
      "-- check region:1028 X:88740000-88760000_:+_reg_, 37680 candidate probes\n",
      "--- 2998 probes passed check_dic selection.\n",
      "finish in 0.861s, 102 probes kept.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- check region:1029 X:91240000-91260000_:+_reg_, 38090 candidate probes\n",
      "--- 4898 probes passed check_dic selection.\n",
      "finish in 1.021s, 127 probes kept.\n",
      "-- check region:1030 X:93740000-93760000_:+_reg_, 39892 candidate probes\n",
      "--- 16706 probes passed check_dic selection.\n",
      "finish in 1.490s, 301 probes kept.\n",
      "-- check region:1031 X:96240000-96260000_:+_reg_, 37504 candidate probes\n",
      "--- 6736 probes passed check_dic selection.\n",
      "finish in 1.048s, 172 probes kept.\n",
      "-- check region:1032 X:98740000-98760000_:+_reg_, 32998 candidate probes\n",
      "--- 8108 probes passed check_dic selection.\n",
      "finish in 0.977s, 163 probes kept.\n",
      "-- check region:1033 X:101240000-101260000_:+_reg_, 39556 candidate probes\n",
      "--- 13419 probes passed check_dic selection.\n",
      "finish in 1.357s, 244 probes kept.\n",
      "-- check region:1034 X:103740000-103760000_:+_reg_, 39046 candidate probes\n",
      "--- 11762 probes passed check_dic selection.\n",
      "finish in 1.274s, 224 probes kept.\n",
      "-- check region:1035 X:106240000-106260000_:+_reg_, 39562 candidate probes\n",
      "--- 4709 probes passed check_dic selection.\n",
      "finish in 1.016s, 126 probes kept.\n",
      "-- check region:1036 X:108740000-108760000_:+_reg_, 39048 candidate probes\n",
      "--- 7803 probes passed check_dic selection.\n",
      "finish in 1.080s, 192 probes kept.\n",
      "-- check region:1037 X:111240000-111260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:1038 X:113740000-113760000_:+_reg_, 38426 candidate probes\n",
      "--- 5868 probes passed check_dic selection.\n",
      "finish in 1.003s, 146 probes kept.\n",
      "-- check region:1039 X:116240000-116260000_:+_reg_, 37128 candidate probes\n",
      "--- 4966 probes passed check_dic selection.\n",
      "finish in 0.919s, 111 probes kept.\n",
      "-- check region:1040 X:118740000-118760000_:+_reg_, 39534 candidate probes\n",
      "--- 2320 probes passed check_dic selection.\n",
      "finish in 0.900s, 62 probes kept.\n",
      "-- check region:1041 X:121240000-121260000_:+_reg_, 39786 candidate probes\n",
      "--- 3347 probes passed check_dic selection.\n",
      "finish in 0.939s, 106 probes kept.\n",
      "-- check region:1042 X:123740000-123760000_:+_reg_, 35476 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.684s, 0 probes kept.\n",
      "-- check region:1043 X:126240000-126260000_:+_reg_, 28014 candidate probes\n",
      "--- 846 probes passed check_dic selection.\n",
      "finish in 0.601s, 33 probes kept.\n",
      "-- check region:1044 X:128740000-128760000_:+_reg_, 39548 candidate probes\n",
      "--- 9173 probes passed check_dic selection.\n",
      "finish in 1.126s, 183 probes kept.\n",
      "-- check region:1045 X:131240000-131260000_:+_reg_, 39472 candidate probes\n",
      "--- 7020 probes passed check_dic selection.\n",
      "finish in 1.082s, 172 probes kept.\n",
      "-- check region:1046 X:133740000-133760000_:+_reg_, 39662 candidate probes\n",
      "--- 11491 probes passed check_dic selection.\n",
      "finish in 1.241s, 221 probes kept.\n",
      "-- check region:1047 X:136240000-136260000_:+_reg_, 39802 candidate probes\n",
      "--- 15083 probes passed check_dic selection.\n",
      "finish in 1.418s, 268 probes kept.\n",
      "-- check region:1048 X:138740000-138760000_:+_reg_, 39840 candidate probes\n",
      "--- 10105 probes passed check_dic selection.\n",
      "finish in 1.213s, 219 probes kept.\n",
      "-- check region:1049 X:141240000-141260000_:+_reg_, 38140 candidate probes\n",
      "--- 8513 probes passed check_dic selection.\n",
      "finish in 1.087s, 181 probes kept.\n",
      "-- check region:1050 X:143740000-143760000_:+_reg_, 35290 candidate probes\n",
      "--- 8955 probes passed check_dic selection.\n",
      "finish in 1.141s, 190 probes kept.\n",
      "-- check region:1051 X:146240000-146260000_:+_reg_, 35618 candidate probes\n",
      "--- 5702 probes passed check_dic selection.\n",
      "finish in 0.945s, 147 probes kept.\n",
      "-- check region:1052 X:148740000-148760000_:+_reg_, 39472 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.862s, 0 probes kept.\n",
      "-- check region:1053 X:151240000-151260000_:+_reg_, 34704 candidate probes\n",
      "--- 3634 probes passed check_dic selection.\n",
      "finish in 0.845s, 96 probes kept.\n",
      "-- check region:1054 X:153740000-153760000_:+_reg_, 39820 candidate probes\n",
      "--- 7684 probes passed check_dic selection.\n",
      "finish in 1.135s, 171 probes kept.\n",
      "-- check region:1055 X:156240000-156260000_:+_reg_, 39580 candidate probes\n",
      "--- 5772 probes passed check_dic selection.\n",
      "finish in 1.030s, 150 probes kept.\n",
      "-- check region:1056 X:158740000-158760000_:+_reg_, 38854 candidate probes\n",
      "--- 9792 probes passed check_dic selection.\n",
      "finish in 1.173s, 213 probes kept.\n",
      "-- check region:1057 X:161240000-161260000_:+_reg_, 39768 candidate probes\n",
      "--- 7186 probes passed check_dic selection.\n",
      "finish in 1.087s, 193 probes kept.\n",
      "-- check region:1058 X:163740000-163760000_:+_reg_, 39614 candidate probes\n",
      "--- 15897 probes passed check_dic selection.\n",
      "finish in 1.443s, 257 probes kept.\n",
      "-- check region:1059 X:166240000-166260000_:+_reg_, 39624 candidate probes\n",
      "--- 11676 probes passed check_dic selection.\n",
      "finish in 1.236s, 221 probes kept.\n",
      "-- check region:1060 X:168740000-168760000_:+_reg_, 38384 candidate probes\n",
      "--- 5082 probes passed check_dic selection.\n",
      "finish in 1.004s, 122 probes kept.\n",
      "-- check region:1061 X:171240000-171031299_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.000s, 0 probes kept.\n",
      "-- check region:1062 Y:1240000-1260000_:+_reg_, 39818 candidate probes\n",
      "--- 7626 probes passed check_dic selection.\n",
      "finish in 1.100s, 168 probes kept.\n",
      "-- check region:1063 Y:3740000-3760000_:+_reg_, 39524 candidate probes\n",
      "--- 4 probes passed check_dic selection.\n",
      "finish in 0.790s, 1 probes kept.\n",
      "-- check region:1064 Y:6240000-6260000_:+_reg_, 28814 candidate probes\n",
      "--- 30 probes passed check_dic selection.\n",
      "finish in 0.622s, 2 probes kept.\n",
      "-- check region:1065 Y:8740000-8760000_:+_reg_, 21056 candidate probes\n",
      "--- 32 probes passed check_dic selection.\n",
      "finish in 0.463s, 3 probes kept.\n",
      "-- check region:1066 Y:11240000-11260000_:+_reg_, 31930 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.673s, 0 probes kept.\n",
      "-- check region:1067 Y:13740000-13760000_:+_reg_, 31314 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.661s, 0 probes kept.\n",
      "-- check region:1068 Y:16240000-16260000_:+_reg_, 32990 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.682s, 0 probes kept.\n",
      "-- check region:1069 Y:18740000-18760000_:+_reg_, 27124 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.558s, 0 probes kept.\n",
      "-- check region:1070 Y:21240000-21260000_:+_reg_, 7384 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.167s, 0 probes kept.\n",
      "-- check region:1071 Y:23740000-23760000_:+_reg_, 33006 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.713s, 0 probes kept.\n",
      "-- check region:1072 Y:26240000-26260000_:+_reg_, 8670 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.175s, 0 probes kept.\n",
      "-- check region:1073 Y:28740000-28760000_:+_reg_, 21984 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.465s, 0 probes kept.\n",
      "-- check region:1074 Y:31240000-31260000_:+_reg_, 30264 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.620s, 0 probes kept.\n",
      "-- check region:1075 Y:33740000-33760000_:+_reg_, 34860 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.689s, 0 probes kept.\n",
      "-- check region:1076 Y:36240000-36260000_:+_reg_, 19980 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.438s, 0 probes kept.\n",
      "-- check region:1077 Y:38740000-38760000_:+_reg_, 10236 candidate probes\n",
      "--- 8 probes passed check_dic selection.\n",
      "finish in 0.202s, 1 probes kept.\n",
      "-- check region:1078 Y:41240000-41260000_:+_reg_, 30282 candidate probes\n",
      "--- 34 probes passed check_dic selection.\n",
      "finish in 0.623s, 2 probes kept.\n",
      "-- check region:1079 Y:43740000-43760000_:+_reg_, 38438 candidate probes\n",
      "--- 26 probes passed check_dic selection.\n",
      "finish in 0.777s, 2 probes kept.\n",
      "-- check region:1080 Y:46240000-46260000_:+_reg_, 35914 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.712s, 0 probes kept.\n",
      "-- check region:1081 Y:48740000-48760000_:+_reg_, 27822 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.560s, 0 probes kept.\n",
      "-- check region:1082 Y:51240000-51260000_:+_reg_, 38588 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.766s, 0 probes kept.\n",
      "-- check region:1083 Y:53740000-53760000_:+_reg_, 34326 candidate probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.739s, 0 probes kept.\n",
      "-- check region:1084 Y:56240000-56260000_:+_reg_, 28804 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.621s, 0 probes kept.\n",
      "-- check region:1085 Y:58740000-58760000_:+_reg_, 39678 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.834s, 0 probes kept.\n",
      "-- check region:1086 Y:61240000-61260000_:+_reg_, 38814 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.792s, 0 probes kept.\n",
      "-- check region:1087 Y:63740000-63760000_:+_reg_, 30110 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.657s, 0 probes kept.\n",
      "-- check region:1088 Y:66240000-66260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.001s, 0 probes kept.\n",
      "-- check region:1089 Y:68740000-68760000_:+_reg_, 18896 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.399s, 0 probes kept.\n",
      "-- check region:1090 Y:71240000-71260000_:+_reg_, 21424 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.440s, 0 probes kept.\n",
      "-- check region:1091 Y:73740000-73760000_:+_reg_, 39870 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.825s, 0 probes kept.\n",
      "-- check region:1092 Y:76240000-76260000_:+_reg_, 39870 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.815s, 0 probes kept.\n",
      "-- check region:1093 Y:78740000-78760000_:+_reg_, 39240 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.783s, 0 probes kept.\n",
      "-- check region:1094 Y:81240000-81260000_:+_reg_, 25810 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.530s, 0 probes kept.\n",
      "-- check region:1095 Y:83740000-83760000_:+_reg_, 39478 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.770s, 0 probes kept.\n",
      "-- check region:1096 Y:86240000-86260000_:+_reg_, 39834 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.806s, 0 probes kept.\n",
      "-- check region:1097 Y:88740000-88760000_:+_reg_, 39790 candidate probes\n",
      "--- 2 probes passed check_dic selection.\n",
      "finish in 0.867s, 1 probes kept.\n",
      "-- check region:1098 Y:91240000-91260000_:+_reg_, 0 candidate probes\n",
      "--- 0 probes passed check_dic selection.\n",
      "finish in 0.002s, 0 probes kept.\n"
     ]
    }
   ],
   "source": [
    "pbs, pb_scores = pb_designer.check_probes()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ae777bea",
   "metadata": {},
   "source": [
    "### save probes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "d6972ee5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Save reports into file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr\n",
      "-- number of probes kept: 211283\n"
     ]
    }
   ],
   "source": [
    "overwrite_savefile = True \n",
    "if not os.path.exists(pb_designer.save_file) or overwrite_savefile:\n",
    "    #pb_designer.plots()\n",
    "    #pb_designer.save_csv()\n",
    "    pb_designer.save_to_file()\n",
    "print(f\"-- number of probes kept: {len(pb_designer.kept_probes)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "24b9d1d4",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Time to release OTmaps: 0.000s. \n"
     ]
    }
   ],
   "source": [
    "# after finish designing, remove all loaded OT maps to freeup memory usage\n",
    "pb_designer.release_OTmaps()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "f7251053",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'name': '1:3740000-3760000_:+_reg__pb_1207',\n",
       " 'reg_index': 1,\n",
       " 'reg_name': '1:3740000-3760000_:+_reg_',\n",
       " 'pb_index': 1207,\n",
       " 'strand': '+',\n",
       " 'gc': 0.40476190476190477,\n",
       " 'tm': 75.15375687037147,\n",
       " 'map_genome': 33,\n",
       " 'map_transcriptome': 0,\n",
       " 'map_rep_genome': 0,\n",
       " 'map_self_sequences': 26}"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "5c5c092c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1:3740000-3760000_:+_reg_'"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "old_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "id": "042d2211",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1:3740000-3760000_strand_+_reg_1'"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "f5619caf",
   "metadata": {},
   "outputs": [],
   "source": [
    "_info['name'] = _info['name'].replace(old_name, new_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "6045d177",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "for _pb, _info in pb_designer.cand_probes.items():\n",
    "    if _info['reg_name'][-1] == '_':\n",
    "        old_name = str(_info['reg_name'])\n",
    "        new_name = _info['reg_name'].replace(':+', 'strand_+') + str(_info['reg_index'])\n",
    "        _info['name'] = _info['name'].replace(old_name, new_name)\n",
    "        _info['reg_name'] = new_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "d8039bf0",
   "metadata": {},
   "outputs": [],
   "source": [
    "for _pb, _info in pb_designer.kept_probes.items():\n",
    "    if _info['reg_name'][-1] == '_':\n",
    "        old_name = str(_info['reg_name'])\n",
    "        new_name = _info['reg_name'].replace(':+', 'strand_+') + str(_info['reg_index'])\n",
    "        _info['name'] = _info['name'].replace(old_name, new_name)\n",
    "        _info['reg_name'] = new_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "44e360e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Save reports into file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr\n",
      "-- number of probes kept: 211283\n"
     ]
    }
   ],
   "source": [
    "overwrite_savefile = True \n",
    "if not os.path.exists(pb_designer.save_file) or overwrite_savefile:\n",
    "    #pb_designer.plots()\n",
    "    #pb_designer.save_csv()\n",
    "    pb_designer.save_to_file()\n",
    "print(f\"-- number of probes kept: {len(pb_designer.kept_probes)}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bcc5f3cd",
   "metadata": {},
   "source": [
    "## 2.3 summarize into a pb_dict\n",
    "\n",
    "start over from here is possible"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "id": "381e6ab7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr\n"
     ]
    }
   ],
   "source": [
    "# filename to save probe reports\n",
    "save_file = os.path.join(report_folder, f'merged_probes.pbr')\n",
    "print(save_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "1a70c0ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Loading from savefile: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\reports\\merged_probes.pbr.\n",
      "211283\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reload(library_tools)\n",
    "reload(library_tools.design)\n",
    "\n",
    "verbose=True\n",
    "cand_prob_nums = []\n",
    "pb_dict = {}\n",
    "# Parse segemented list\n",
    "_pbde = library_tools.design.pb_reports_class()\n",
    "_pbde.load_from_file(save_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c5e09a81",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'np' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-db5a78c969b9>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[0mcand_prob_nums\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[0mpb_dict\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[0mgene_names\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msort\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0m_v\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'gene_name'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'gene_'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0m_s\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0m_v\u001b[0m \u001b[1;32min\u001b[0m \u001b[0m_pbde\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mkept_probes\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0m_gene_name\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mgene_names\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[0m_sel_pbs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m{\u001b[0m\u001b[0m_s\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0m_v\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0m_s\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0m_v\u001b[0m \u001b[1;32min\u001b[0m \u001b[0m_pbde\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mkept_probes\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mif\u001b[0m \u001b[0mint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0m_v\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'gene_name'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'gene_'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m1\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m==\u001b[0m\u001b[0m_gene_name\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mNameError\u001b[0m: name 'np' is not defined"
     ]
    }
   ],
   "source": [
    "cand_prob_nums = []\n",
    "pb_dict = {}\n",
    "gene_names = np.sort(np.unique([int(_v['reg_name'].split('reg_')[1]) for _s, _v in _pbde.kept_probes.items()]))\n",
    "for _gene_name in gene_names:\n",
    "    _sel_pbs = {_s:_v for _s, _v in _pbde.kept_probes.items() if int(_v['reg_name'].split('reg_')[1])==_gene_name}\n",
    "    \n",
    "    if len(_sel_pbs) > 0:\n",
    "        cand_prob_nums.append(len(_sel_pbs))\n",
    "        if verbose:\n",
    "            print(f\"-- appending region:{_gene_name} with {len(_sel_pbs)} probes\")\n",
    "        _splitted_pbde = ld.pb_reports_class()\n",
    "        _splitted_pbde.kept_probes = _sel_pbs\n",
    "        pb_dict[_gene_name] = _splitted_pbde"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "id": "d7cd4048",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Screen probes by genome counts.\n",
      "-- filtering region:1, directly append 163 probes\n",
      "-- filtering region:2, directly append 180 probes\n",
      "-- filtering region:3, directly append 77 probes\n",
      "-- filtering region:4, directly append 205 probes\n",
      "-- filtering region:5, directly append 144 probes\n",
      "-- filtering region:6, directly append 297 probes\n",
      "-- filtering region:7, directly append 210 probes\n",
      "-- filtering region:8, directly append 274 probes\n",
      "-- filtering region:9, directly append 220 probes\n",
      "-- filtering region:10, directly append 158 probes\n",
      "-- filtering region:11, directly append 156 probes\n",
      "-- filtering region:12, directly append 107 probes\n",
      "-- filtering region:13, directly append 291 probes\n",
      "-- filtering region:14, directly append 240 probes\n",
      "-- filtering region:15, directly append 279 probes\n",
      "-- filtering region:16, directly append 248 probes\n",
      "-- filtering region:17, directly append 276 probes\n",
      "-- filtering region:18, directly append 109 probes\n",
      "-- filtering region:19, directly append 111 probes\n",
      "-- filtering region:20, directly append 212 probes\n",
      "-- filtering region:21, directly append 135 probes\n",
      "-- filtering region:22, directly append 229 probes\n",
      "-- filtering region:23, directly append 257 probes\n",
      "-- filtering region:24, directly append 257 probes\n",
      "-- filtering region:25, directly append 249 probes\n",
      "-- filtering region:26, directly append 204 probes\n",
      "-- filtering region:27, directly append 195 probes\n",
      "-- filtering region:28, directly append 198 probes\n",
      "-- filtering region:29, directly append 284 probes\n",
      "-- filtering region:30, directly append 245 probes\n",
      "-- filtering region:31, directly append 151 probes\n",
      "-- filtering region:32, directly append 247 probes\n",
      "-- filtering region:33, directly append 276 probes\n",
      "-- filtering region:34, directly append 300 probes\n",
      "-- filtering region:35, directly append 310 probes\n",
      "-- filtering region:36, directly append 260 probes\n",
      "-- filtering region:37, directly append 244 probes\n",
      "-- filtering region:38, directly append 75 probes\n",
      "-- filtering region:39, directly append 178 probes\n",
      "-- filtering region:40, directly append 142 probes\n",
      "-- filtering region:41, directly append 122 probes\n",
      "-- filtering region:42, directly append 105 probes\n",
      "-- filtering region:43, directly append 173 probes\n",
      "-- filtering region:44, directly append 90 probes\n",
      "-- filtering region:45, directly append 121 probes\n",
      "-- filtering region:46, directly append 241 probes\n",
      "-- filtering region:47, directly append 240 probes\n",
      "-- filtering region:48, directly append 218 probes\n",
      "-- filtering region:49, directly append 225 probes\n",
      "-- filtering region:50, directly append 287 probes\n",
      "-- filtering region:51, directly append 250 probes\n",
      "-- filtering region:52, directly append 329 probes\n",
      "-- filtering region:53, directly append 299 probes\n",
      "-- filtering region:54, directly append 257 probes\n",
      "-- filtering region:55, directly append 206 probes\n",
      "-- filtering region:56, directly append 125 probes\n",
      "-- filtering region:57, directly append 232 probes\n",
      "-- filtering region:58, directly append 112 probes\n",
      "-- filtering region:59, directly append 159 probes\n",
      "-- filtering region:60, directly append 252 probes\n",
      "-- filtering region:61, directly append 281 probes\n",
      "-- filtering region:62, directly append 265 probes\n",
      "-- filtering region:63, directly append 245 probes\n",
      "-- filtering region:64, directly append 303 probes\n",
      "-- filtering region:65, directly append 231 probes\n",
      "-- filtering region:66, directly append 310 probes\n",
      "-- filtering region:67, directly append 197 probes\n",
      "-- filtering region:68, directly append 283 probes\n",
      "-- filtering region:69, directly append 113 probes\n",
      "-- filtering region:70, directly append 209 probes\n",
      "-- filtering region:71, directly append 274 probes\n",
      "-- filtering region:72, directly append 251 probes\n",
      "-- filtering region:73, directly append 282 probes\n",
      "-- filtering region:74, directly append 276 probes\n",
      "-- filtering region:75, directly append 278 probes\n",
      "-- filtering region:76, directly append 265 probes\n",
      "-- filtering region:77, directly append 193 probes\n",
      "-- filtering region:80, directly append 287 probes\n",
      "-- filtering region:81, directly append 240 probes\n",
      "-- filtering region:82, directly append 223 probes\n",
      "-- filtering region:83, directly append 233 probes\n",
      "-- filtering region:84, directly append 290 probes\n",
      "-- filtering region:85, directly append 173 probes\n",
      "-- filtering region:86, directly append 202 probes\n",
      "-- filtering region:87, directly append 154 probes\n",
      "-- filtering region:88, directly append 117 probes\n",
      "-- filtering region:89, directly append 318 probes\n",
      "-- filtering region:90, directly append 315 probes\n",
      "-- filtering region:91, directly append 306 probes\n",
      "-- filtering region:92, directly append 338 probes\n",
      "-- filtering region:93, directly append 284 probes\n",
      "-- filtering region:94, directly append 207 probes\n",
      "-- filtering region:95, directly append 185 probes\n",
      "-- filtering region:96, directly append 207 probes\n",
      "-- filtering region:97, directly append 147 probes\n",
      "-- filtering region:98, directly append 249 probes\n",
      "-- filtering region:99, directly append 192 probes\n",
      "-- filtering region:100, directly append 82 probes\n",
      "-- filtering region:101, directly append 122 probes\n",
      "-- filtering region:102, directly append 267 probes\n",
      "-- filtering region:103, directly append 220 probes\n",
      "-- filtering region:104, directly append 201 probes\n",
      "-- filtering region:105, directly append 271 probes\n",
      "-- filtering region:106, directly append 213 probes\n",
      "-- filtering region:107, directly append 246 probes\n",
      "-- filtering region:108, directly append 152 probes\n",
      "-- filtering region:109, directly append 263 probes\n",
      "-- filtering region:110, directly append 248 probes\n",
      "-- filtering region:111, directly append 97 probes\n",
      "-- filtering region:112, directly append 205 probes\n",
      "-- filtering region:113, directly append 91 probes\n",
      "-- filtering region:114, directly append 163 probes\n",
      "-- filtering region:115, directly append 242 probes\n",
      "-- filtering region:116, directly append 296 probes\n",
      "-- filtering region:117, directly append 175 probes\n",
      "-- filtering region:118, directly append 111 probes\n",
      "-- filtering region:119, directly append 203 probes\n",
      "-- filtering region:120, directly append 279 probes\n",
      "-- filtering region:121, directly append 223 probes\n",
      "-- filtering region:122, directly append 231 probes\n",
      "-- filtering region:123, directly append 102 probes\n",
      "-- filtering region:124, directly append 255 probes\n",
      "-- filtering region:125, directly append 191 probes\n",
      "-- filtering region:126, directly append 318 probes\n",
      "-- filtering region:127, directly append 224 probes\n",
      "-- filtering region:128, directly append 167 probes\n",
      "-- filtering region:129, directly append 147 probes\n",
      "-- filtering region:130, directly append 274 probes\n",
      "-- filtering region:131, directly append 261 probes\n",
      "-- filtering region:132, directly append 248 probes\n",
      "-- filtering region:133, directly append 159 probes\n",
      "-- filtering region:134, directly append 140 probes\n",
      "-- filtering region:135, directly append 232 probes\n",
      "-- filtering region:136, directly append 235 probes\n",
      "-- filtering region:137, directly append 313 probes\n",
      "-- filtering region:138, directly append 199 probes\n",
      "-- filtering region:139, directly append 22 probes\n",
      "-- filtering region:140, directly append 226 probes\n",
      "-- filtering region:141, directly append 259 probes\n",
      "-- filtering region:142, directly append 308 probes\n",
      "-- filtering region:143, directly append 248 probes\n",
      "-- filtering region:144, directly append 307 probes\n",
      "-- filtering region:145, directly append 302 probes\n",
      "-- filtering region:146, directly append 282 probes\n",
      "-- filtering region:147, directly append 252 probes\n",
      "-- filtering region:148, directly append 268 probes\n",
      "-- filtering region:150, directly append 258 probes\n",
      "-- filtering region:151, directly append 285 probes\n",
      "-- filtering region:153, directly append 187 probes\n",
      "-- filtering region:154, directly append 159 probes\n",
      "-- filtering region:155, directly append 265 probes\n",
      "-- filtering region:156, directly append 144 probes\n",
      "-- filtering region:157, directly append 199 probes\n",
      "-- filtering region:158, directly append 127 probes\n",
      "-- filtering region:159, directly append 270 probes\n",
      "-- filtering region:160, directly append 141 probes\n",
      "-- filtering region:161, directly append 168 probes\n",
      "-- filtering region:162, directly append 221 probes\n",
      "-- filtering region:163, directly append 257 probes\n",
      "-- filtering region:164, directly append 288 probes\n",
      "-- filtering region:165, directly append 225 probes\n",
      "-- filtering region:166, directly append 103 probes\n",
      "-- filtering region:167, directly append 164 probes\n",
      "-- filtering region:168, directly append 179 probes\n",
      "-- filtering region:169, directly append 140 probes\n",
      "-- filtering region:170, directly append 86 probes\n",
      "-- filtering region:171, directly append 139 probes\n",
      "-- filtering region:172, directly append 222 probes\n",
      "-- filtering region:173, directly append 225 probes\n",
      "-- filtering region:174, directly append 160 probes\n",
      "-- filtering region:175, directly append 210 probes\n",
      "-- filtering region:176, directly append 213 probes\n",
      "-- filtering region:177, directly append 268 probes\n",
      "-- filtering region:178, directly append 261 probes\n",
      "-- filtering region:179, directly append 163 probes\n",
      "-- filtering region:180, directly append 136 probes\n",
      "-- filtering region:181, directly append 181 probes\n",
      "-- filtering region:182, directly append 99 probes\n",
      "-- filtering region:183, directly append 110 probes\n",
      "-- filtering region:184, directly append 126 probes\n",
      "-- filtering region:185, directly append 290 probes\n",
      "-- filtering region:186, directly append 182 probes\n",
      "-- filtering region:187, directly append 216 probes\n",
      "-- filtering region:188, directly append 114 probes\n",
      "-- filtering region:189, directly append 9 probes\n",
      "-- filtering region:190, directly append 229 probes\n",
      "-- filtering region:191, directly append 142 probes\n",
      "-- filtering region:192, directly append 251 probes\n",
      "-- filtering region:193, directly append 275 probes\n",
      "-- filtering region:194, directly append 29 probes\n",
      "-- filtering region:195, directly append 186 probes\n",
      "-- filtering region:196, directly append 115 probes\n",
      "-- filtering region:197, directly append 115 probes\n",
      "-- filtering region:198, directly append 289 probes\n",
      "-- filtering region:199, directly append 175 probes\n",
      "-- filtering region:200, directly append 258 probes\n",
      "-- filtering region:201, directly append 169 probes\n",
      "-- filtering region:202, directly append 180 probes\n",
      "-- filtering region:203, directly append 197 probes\n",
      "-- filtering region:204, directly append 314 probes\n",
      "-- filtering region:205, directly append 264 probes\n",
      "-- filtering region:206, directly append 198 probes\n",
      "-- filtering region:207, directly append 294 probes\n",
      "-- filtering region:208, directly append 191 probes\n",
      "-- filtering region:209, directly append 290 probes\n",
      "-- filtering region:210, directly append 295 probes\n",
      "-- filtering region:211, directly append 184 probes\n",
      "-- filtering region:212, directly append 160 probes\n",
      "-- filtering region:213, directly append 241 probes\n",
      "-- filtering region:214, directly append 180 probes\n",
      "-- filtering region:215, directly append 122 probes\n",
      "-- filtering region:218, directly append 89 probes\n",
      "-- filtering region:219, directly append 164 probes\n",
      "-- filtering region:220, directly append 298 probes\n",
      "-- filtering region:221, directly append 203 probes\n",
      "-- filtering region:222, directly append 296 probes\n",
      "-- filtering region:223, directly append 158 probes\n",
      "-- filtering region:224, directly append 104 probes\n",
      "-- filtering region:225, directly append 84 probes\n",
      "-- filtering region:226, directly append 184 probes\n",
      "-- filtering region:227, directly append 119 probes\n",
      "-- filtering region:228, directly append 84 probes\n",
      "-- filtering region:229, directly append 244 probes\n",
      "-- filtering region:230, directly append 264 probes\n",
      "-- filtering region:231, directly append 143 probes\n",
      "-- filtering region:232, directly append 110 probes\n",
      "-- filtering region:233, directly append 223 probes\n",
      "-- filtering region:234, directly append 194 probes\n",
      "-- filtering region:235, directly append 151 probes\n",
      "-- filtering region:236, directly append 220 probes\n",
      "-- filtering region:237, directly append 108 probes\n",
      "-- filtering region:238, directly append 258 probes\n",
      "-- filtering region:239, directly append 257 probes\n",
      "-- filtering region:240, directly append 139 probes\n",
      "-- filtering region:241, directly append 1 probes\n",
      "-- filtering region:242, directly append 299 probes\n",
      "-- filtering region:243, directly append 251 probes\n",
      "-- filtering region:244, directly append 237 probes\n",
      "-- filtering region:245, directly append 154 probes\n",
      "-- filtering region:246, directly append 151 probes\n",
      "-- filtering region:247, directly append 228 probes\n",
      "-- filtering region:248, directly append 135 probes\n",
      "-- filtering region:249, directly append 211 probes\n",
      "-- filtering region:250, directly append 100 probes\n",
      "-- filtering region:251, directly append 207 probes\n",
      "-- filtering region:252, directly append 137 probes\n",
      "-- filtering region:253, directly append 283 probes\n",
      "-- filtering region:254, directly append 139 probes\n",
      "-- filtering region:255, directly append 119 probes\n",
      "-- filtering region:256, directly append 230 probes\n",
      "-- filtering region:257, directly append 231 probes\n",
      "-- filtering region:258, directly append 300 probes\n",
      "-- filtering region:259, directly append 207 probes\n",
      "-- filtering region:260, directly append 160 probes\n",
      "-- filtering region:261, directly append 137 probes\n",
      "-- filtering region:262, directly append 101 probes\n",
      "-- filtering region:263, directly append 183 probes\n",
      "-- filtering region:264, directly append 186 probes\n",
      "-- filtering region:265, directly append 43 probes\n",
      "-- filtering region:266, directly append 213 probes\n",
      "-- filtering region:267, directly append 309 probes\n",
      "-- filtering region:268, directly append 299 probes\n",
      "-- filtering region:269, directly append 306 probes\n",
      "-- filtering region:270, directly append 292 probes\n",
      "-- filtering region:271, directly append 312 probes\n",
      "-- filtering region:272, directly append 325 probes\n",
      "-- filtering region:273, directly append 300 probes\n",
      "-- filtering region:274, directly append 123 probes\n",
      "-- filtering region:276, directly append 248 probes\n",
      "-- filtering region:277, directly append 319 probes\n",
      "-- filtering region:278, directly append 158 probes\n",
      "-- filtering region:279, directly append 255 probes\n",
      "-- filtering region:281, directly append 187 probes\n",
      "-- filtering region:282, directly append 141 probes\n",
      "-- filtering region:283, directly append 119 probes\n",
      "-- filtering region:284, directly append 5 probes\n",
      "-- filtering region:285, directly append 93 probes\n",
      "-- filtering region:286, directly append 196 probes\n",
      "-- filtering region:287, directly append 134 probes\n",
      "-- filtering region:288, directly append 217 probes\n",
      "-- filtering region:289, directly append 252 probes\n",
      "-- filtering region:290, directly append 103 probes\n",
      "-- filtering region:291, directly append 229 probes\n",
      "-- filtering region:292, directly append 316 probes\n",
      "-- filtering region:293, directly append 252 probes\n",
      "-- filtering region:294, directly append 218 probes\n",
      "-- filtering region:295, directly append 257 probes\n",
      "-- filtering region:296, directly append 237 probes\n",
      "-- filtering region:297, directly append 236 probes\n",
      "-- filtering region:298, directly append 151 probes\n",
      "-- filtering region:299, directly append 212 probes\n",
      "-- filtering region:300, directly append 174 probes\n",
      "-- filtering region:301, directly append 119 probes\n",
      "-- filtering region:302, directly append 115 probes\n",
      "-- filtering region:303, directly append 184 probes\n",
      "-- filtering region:304, directly append 130 probes\n",
      "-- filtering region:305, directly append 222 probes\n",
      "-- filtering region:306, directly append 188 probes\n",
      "-- filtering region:307, directly append 149 probes\n",
      "-- filtering region:308, directly append 108 probes\n",
      "-- filtering region:309, directly append 324 probes\n",
      "-- filtering region:310, directly append 205 probes\n",
      "-- filtering region:311, directly append 189 probes\n",
      "-- filtering region:312, directly append 204 probes\n",
      "-- filtering region:313, directly append 135 probes\n",
      "-- filtering region:314, directly append 210 probes\n",
      "-- filtering region:315, directly append 183 probes\n",
      "-- filtering region:316, directly append 192 probes\n",
      "-- filtering region:317, directly append 7 probes\n",
      "-- filtering region:318, directly append 247 probes\n",
      "-- filtering region:319, directly append 205 probes\n",
      "-- filtering region:320, directly append 200 probes\n",
      "-- filtering region:321, directly append 274 probes\n",
      "-- filtering region:322, directly append 210 probes\n",
      "-- filtering region:323, directly append 242 probes\n",
      "-- filtering region:324, directly append 303 probes\n",
      "-- filtering region:325, directly append 337 probes\n",
      "-- filtering region:326, directly append 237 probes\n",
      "-- filtering region:327, directly append 343 probes\n",
      "-- filtering region:328, directly append 248 probes\n",
      "-- filtering region:329, directly append 237 probes\n",
      "-- filtering region:330, directly append 284 probes\n",
      "-- filtering region:331, directly append 341 probes\n",
      "-- filtering region:332, directly append 235 probes\n",
      "-- filtering region:333, directly append 243 probes\n",
      "-- filtering region:334, directly append 284 probes\n",
      "-- filtering region:335, directly append 318 probes\n",
      "-- filtering region:336, directly append 285 probes\n",
      "-- filtering region:337, directly append 281 probes\n",
      "-- filtering region:338, directly append 259 probes\n",
      "-- filtering region:339, directly append 288 probes\n",
      "-- filtering region:340, directly append 249 probes\n",
      "-- filtering region:342, directly append 258 probes\n",
      "-- filtering region:343, directly append 219 probes\n",
      "-- filtering region:344, directly append 261 probes\n",
      "-- filtering region:345, directly append 150 probes\n",
      "-- filtering region:346, directly append 182 probes\n",
      "-- filtering region:347, directly append 53 probes\n",
      "-- filtering region:348, directly append 139 probes\n",
      "-- filtering region:349, directly append 188 probes\n",
      "-- filtering region:350, directly append 239 probes\n",
      "-- filtering region:351, directly append 69 probes\n",
      "-- filtering region:352, directly append 318 probes\n",
      "-- filtering region:353, directly append 268 probes\n",
      "-- filtering region:354, directly append 175 probes\n",
      "-- filtering region:355, directly append 209 probes\n",
      "-- filtering region:356, directly append 317 probes\n",
      "-- filtering region:357, directly append 138 probes\n",
      "-- filtering region:358, directly append 158 probes\n",
      "-- filtering region:359, directly append 227 probes\n",
      "-- filtering region:360, directly append 221 probes\n",
      "-- filtering region:361, directly append 235 probes\n",
      "-- filtering region:362, directly append 260 probes\n",
      "-- filtering region:363, directly append 116 probes\n",
      "-- filtering region:364, directly append 163 probes\n",
      "-- filtering region:365, directly append 203 probes\n",
      "-- filtering region:366, directly append 195 probes\n",
      "-- filtering region:367, directly append 133 probes\n",
      "-- filtering region:368, directly append 129 probes\n",
      "-- filtering region:369, directly append 226 probes\n",
      "-- filtering region:370, directly append 175 probes\n",
      "-- filtering region:371, directly append 123 probes\n",
      "-- filtering region:372, directly append 210 probes\n",
      "-- filtering region:373, directly append 235 probes\n",
      "-- filtering region:374, directly append 305 probes\n",
      "-- filtering region:375, directly append 285 probes\n",
      "-- filtering region:376, directly append 315 probes\n",
      "-- filtering region:377, directly append 326 probes\n",
      "-- filtering region:378, directly append 270 probes\n",
      "-- filtering region:379, directly append 305 probes\n",
      "-- filtering region:380, directly append 272 probes\n",
      "-- filtering region:381, directly append 327 probes\n",
      "-- filtering region:382, directly append 242 probes\n",
      "-- filtering region:383, directly append 233 probes\n",
      "-- filtering region:384, directly append 212 probes\n",
      "-- filtering region:385, directly append 230 probes\n",
      "-- filtering region:386, directly append 329 probes\n",
      "-- filtering region:387, directly append 269 probes\n",
      "-- filtering region:388, directly append 323 probes\n",
      "-- filtering region:389, directly append 266 probes\n",
      "-- filtering region:390, directly append 103 probes\n",
      "-- filtering region:391, directly append 153 probes\n",
      "-- filtering region:392, directly append 114 probes\n",
      "-- filtering region:393, directly append 93 probes\n",
      "-- filtering region:394, directly append 48 probes\n",
      "-- filtering region:395, directly append 260 probes\n",
      "-- filtering region:396, directly append 127 probes\n",
      "-- filtering region:397, directly append 268 probes\n",
      "-- filtering region:398, directly append 274 probes\n",
      "-- filtering region:399, directly append 252 probes\n",
      "-- filtering region:400, directly append 245 probes\n",
      "-- filtering region:403, directly append 172 probes\n",
      "-- filtering region:405, directly append 20 probes\n",
      "-- filtering region:406, directly append 22 probes\n",
      "-- filtering region:407, directly append 211 probes\n",
      "-- filtering region:408, directly append 76 probes\n",
      "-- filtering region:409, directly append 1 probes\n",
      "-- filtering region:410, directly append 26 probes\n",
      "-- filtering region:411, directly append 126 probes\n",
      "-- filtering region:412, directly append 261 probes\n",
      "-- filtering region:413, directly append 98 probes\n",
      "-- filtering region:414, directly append 96 probes\n",
      "-- filtering region:415, directly append 250 probes\n",
      "-- filtering region:416, directly append 3 probes\n",
      "-- filtering region:417, directly append 42 probes\n",
      "-- filtering region:418, directly append 133 probes\n",
      "-- filtering region:419, directly append 309 probes\n",
      "-- filtering region:420, directly append 169 probes\n",
      "-- filtering region:421, directly append 232 probes\n",
      "-- filtering region:422, directly append 221 probes\n",
      "-- filtering region:423, directly append 235 probes\n",
      "-- filtering region:424, directly append 294 probes\n",
      "-- filtering region:425, directly append 75 probes\n",
      "-- filtering region:426, directly append 318 probes\n",
      "-- filtering region:427, directly append 293 probes\n",
      "-- filtering region:428, directly append 268 probes\n",
      "-- filtering region:429, directly append 246 probes\n",
      "-- filtering region:430, directly append 298 probes\n",
      "-- filtering region:431, directly append 80 probes\n",
      "-- filtering region:432, directly append 278 probes\n",
      "-- filtering region:433, directly append 219 probes\n",
      "-- filtering region:434, directly append 243 probes\n",
      "-- filtering region:435, directly append 139 probes\n",
      "-- filtering region:436, directly append 167 probes\n",
      "-- filtering region:437, directly append 212 probes\n",
      "-- filtering region:438, directly append 70 probes\n",
      "-- filtering region:439, directly append 315 probes\n",
      "-- filtering region:440, directly append 327 probes\n",
      "-- filtering region:441, directly append 163 probes\n",
      "-- filtering region:442, directly append 174 probes\n",
      "-- filtering region:443, directly append 98 probes\n",
      "-- filtering region:444, directly append 159 probes\n",
      "-- filtering region:445, directly append 242 probes\n",
      "-- filtering region:446, directly append 304 probes\n",
      "-- filtering region:447, directly append 268 probes\n",
      "-- filtering region:448, directly append 288 probes\n",
      "-- filtering region:449, directly append 272 probes\n",
      "-- filtering region:450, directly append 231 probes\n",
      "-- filtering region:451, directly append 240 probes\n",
      "-- filtering region:452, directly append 230 probes\n",
      "-- filtering region:453, directly append 254 probes\n",
      "-- filtering region:454, directly append 266 probes\n",
      "-- filtering region:455, directly append 190 probes\n",
      "-- filtering region:456, directly append 314 probes\n",
      "-- filtering region:457, directly append 242 probes\n",
      "-- filtering region:458, directly append 292 probes\n",
      "-- filtering region:461, directly append 179 probes\n",
      "-- filtering region:462, directly append 132 probes\n",
      "-- filtering region:463, directly append 269 probes\n",
      "-- filtering region:464, directly append 321 probes\n",
      "-- filtering region:465, directly append 249 probes\n",
      "-- filtering region:466, directly append 295 probes\n",
      "-- filtering region:467, directly append 336 probes\n",
      "-- filtering region:468, directly append 9 probes\n",
      "-- filtering region:469, directly append 266 probes\n",
      "-- filtering region:470, directly append 241 probes\n",
      "-- filtering region:471, directly append 287 probes\n",
      "-- filtering region:472, directly append 227 probes\n",
      "-- filtering region:473, directly append 182 probes\n",
      "-- filtering region:474, directly append 283 probes\n",
      "-- filtering region:475, directly append 160 probes\n",
      "-- filtering region:476, directly append 220 probes\n",
      "-- filtering region:477, directly append 174 probes\n",
      "-- filtering region:478, directly append 240 probes\n",
      "-- filtering region:479, directly append 286 probes\n",
      "-- filtering region:480, directly append 178 probes\n",
      "-- filtering region:481, directly append 162 probes\n",
      "-- filtering region:482, directly append 113 probes\n",
      "-- filtering region:483, directly append 150 probes\n",
      "-- filtering region:484, directly append 307 probes\n",
      "-- filtering region:485, directly append 118 probes\n",
      "-- filtering region:486, directly append 140 probes\n",
      "-- filtering region:487, directly append 108 probes\n",
      "-- filtering region:488, directly append 160 probes\n",
      "-- filtering region:489, directly append 144 probes\n",
      "-- filtering region:490, directly append 183 probes\n",
      "-- filtering region:491, directly append 282 probes\n",
      "-- filtering region:492, directly append 157 probes\n",
      "-- filtering region:493, directly append 227 probes\n",
      "-- filtering region:494, directly append 182 probes\n",
      "-- filtering region:495, directly append 291 probes\n",
      "-- filtering region:496, directly append 203 probes\n",
      "-- filtering region:497, directly append 259 probes\n",
      "-- filtering region:498, directly append 208 probes\n",
      "-- filtering region:499, directly append 219 probes\n",
      "-- filtering region:500, directly append 169 probes\n",
      "-- filtering region:501, directly append 202 probes\n",
      "-- filtering region:502, directly append 316 probes\n",
      "-- filtering region:503, directly append 321 probes\n",
      "-- filtering region:504, directly append 179 probes\n",
      "-- filtering region:505, directly append 273 probes\n",
      "-- filtering region:506, directly append 279 probes\n",
      "-- filtering region:507, directly append 282 probes\n",
      "-- filtering region:508, directly append 310 probes\n",
      "-- filtering region:509, directly append 261 probes\n",
      "-- filtering region:510, directly append 294 probes\n",
      "-- filtering region:511, directly append 103 probes\n",
      "-- filtering region:513, directly append 53 probes\n",
      "-- filtering region:514, directly append 247 probes\n",
      "-- filtering region:515, directly append 223 probes\n",
      "-- filtering region:516, directly append 120 probes\n",
      "-- filtering region:517, directly append 187 probes\n",
      "-- filtering region:518, directly append 237 probes\n",
      "-- filtering region:519, directly append 216 probes\n",
      "-- filtering region:520, directly append 222 probes\n",
      "-- filtering region:521, directly append 116 probes\n",
      "-- filtering region:522, directly append 219 probes\n",
      "-- filtering region:523, directly append 230 probes\n",
      "-- filtering region:524, directly append 217 probes\n",
      "-- filtering region:525, directly append 180 probes\n",
      "-- filtering region:526, directly append 90 probes\n",
      "-- filtering region:527, directly append 193 probes\n",
      "-- filtering region:528, directly append 256 probes\n",
      "-- filtering region:529, directly append 349 probes\n",
      "-- filtering region:530, directly append 255 probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- filtering region:531, directly append 330 probes\n",
      "-- filtering region:532, directly append 252 probes\n",
      "-- filtering region:533, directly append 224 probes\n",
      "-- filtering region:534, directly append 264 probes\n",
      "-- filtering region:535, directly append 268 probes\n",
      "-- filtering region:536, directly append 269 probes\n",
      "-- filtering region:537, directly append 331 probes\n",
      "-- filtering region:538, directly append 304 probes\n",
      "-- filtering region:539, directly append 259 probes\n",
      "-- filtering region:540, directly append 252 probes\n",
      "-- filtering region:541, directly append 267 probes\n",
      "-- filtering region:542, directly append 200 probes\n",
      "-- filtering region:543, directly append 140 probes\n",
      "-- filtering region:544, directly append 75 probes\n",
      "-- filtering region:545, directly append 162 probes\n",
      "-- filtering region:546, directly append 234 probes\n",
      "-- filtering region:547, directly append 14 probes\n",
      "-- filtering region:548, directly append 88 probes\n",
      "-- filtering region:549, directly append 174 probes\n",
      "-- filtering region:550, directly append 181 probes\n",
      "-- filtering region:551, directly append 236 probes\n",
      "-- filtering region:552, directly append 239 probes\n",
      "-- filtering region:553, directly append 314 probes\n",
      "-- filtering region:554, directly append 246 probes\n",
      "-- filtering region:555, directly append 259 probes\n",
      "-- filtering region:556, directly append 267 probes\n",
      "-- filtering region:557, directly append 234 probes\n",
      "-- filtering region:558, directly append 246 probes\n",
      "-- filtering region:559, directly append 296 probes\n",
      "-- filtering region:560, directly append 290 probes\n",
      "-- filtering region:561, directly append 148 probes\n",
      "-- filtering region:563, directly append 296 probes\n",
      "-- filtering region:564, directly append 94 probes\n",
      "-- filtering region:565, directly append 326 probes\n",
      "-- filtering region:566, directly append 184 probes\n",
      "-- filtering region:567, directly append 265 probes\n",
      "-- filtering region:568, directly append 139 probes\n",
      "-- filtering region:569, directly append 166 probes\n",
      "-- filtering region:570, directly append 241 probes\n",
      "-- filtering region:571, directly append 209 probes\n",
      "-- filtering region:572, directly append 246 probes\n",
      "-- filtering region:573, directly append 138 probes\n",
      "-- filtering region:574, directly append 218 probes\n",
      "-- filtering region:575, directly append 73 probes\n",
      "-- filtering region:576, directly append 161 probes\n",
      "-- filtering region:577, directly append 60 probes\n",
      "-- filtering region:578, directly append 284 probes\n",
      "-- filtering region:579, directly append 252 probes\n",
      "-- filtering region:580, directly append 248 probes\n",
      "-- filtering region:581, directly append 94 probes\n",
      "-- filtering region:582, directly append 152 probes\n",
      "-- filtering region:583, directly append 271 probes\n",
      "-- filtering region:584, directly append 118 probes\n",
      "-- filtering region:585, directly append 189 probes\n",
      "-- filtering region:586, directly append 288 probes\n",
      "-- filtering region:587, directly append 78 probes\n",
      "-- filtering region:588, directly append 213 probes\n",
      "-- filtering region:589, directly append 226 probes\n",
      "-- filtering region:590, directly append 235 probes\n",
      "-- filtering region:591, directly append 222 probes\n",
      "-- filtering region:592, directly append 253 probes\n",
      "-- filtering region:593, directly append 251 probes\n",
      "-- filtering region:594, directly append 319 probes\n",
      "-- filtering region:595, directly append 274 probes\n",
      "-- filtering region:596, directly append 304 probes\n",
      "-- filtering region:597, directly append 245 probes\n",
      "-- filtering region:598, directly append 160 probes\n",
      "-- filtering region:599, directly append 258 probes\n",
      "-- filtering region:600, directly append 253 probes\n",
      "-- filtering region:601, directly append 266 probes\n",
      "-- filtering region:602, directly append 162 probes\n",
      "-- filtering region:603, directly append 92 probes\n",
      "-- filtering region:604, directly append 187 probes\n",
      "-- filtering region:605, directly append 212 probes\n",
      "-- filtering region:606, directly append 220 probes\n",
      "-- filtering region:607, directly append 190 probes\n",
      "-- filtering region:608, directly append 256 probes\n",
      "-- filtering region:609, directly append 285 probes\n",
      "-- filtering region:610, directly append 294 probes\n",
      "-- filtering region:611, directly append 281 probes\n",
      "-- filtering region:612, directly append 277 probes\n",
      "-- filtering region:613, directly append 236 probes\n",
      "-- filtering region:616, directly append 289 probes\n",
      "-- filtering region:617, directly append 158 probes\n",
      "-- filtering region:618, directly append 126 probes\n",
      "-- filtering region:619, directly append 222 probes\n",
      "-- filtering region:620, directly append 103 probes\n",
      "-- filtering region:621, directly append 191 probes\n",
      "-- filtering region:622, directly append 272 probes\n",
      "-- filtering region:623, directly append 177 probes\n",
      "-- filtering region:624, directly append 215 probes\n",
      "-- filtering region:625, directly append 107 probes\n",
      "-- filtering region:626, directly append 175 probes\n",
      "-- filtering region:627, directly append 75 probes\n",
      "-- filtering region:628, directly append 332 probes\n",
      "-- filtering region:629, directly append 315 probes\n",
      "-- filtering region:630, directly append 192 probes\n",
      "-- filtering region:631, directly append 205 probes\n",
      "-- filtering region:632, directly append 213 probes\n",
      "-- filtering region:633, directly append 296 probes\n",
      "-- filtering region:634, directly append 168 probes\n",
      "-- filtering region:635, directly append 234 probes\n",
      "-- filtering region:636, directly append 168 probes\n",
      "-- filtering region:637, directly append 255 probes\n",
      "-- filtering region:638, directly append 159 probes\n",
      "-- filtering region:639, directly append 283 probes\n",
      "-- filtering region:640, directly append 286 probes\n",
      "-- filtering region:641, directly append 263 probes\n",
      "-- filtering region:642, directly append 285 probes\n",
      "-- filtering region:643, directly append 272 probes\n",
      "-- filtering region:644, directly append 131 probes\n",
      "-- filtering region:645, directly append 303 probes\n",
      "-- filtering region:646, directly append 234 probes\n",
      "-- filtering region:647, directly append 285 probes\n",
      "-- filtering region:648, directly append 252 probes\n",
      "-- filtering region:649, directly append 228 probes\n",
      "-- filtering region:650, directly append 260 probes\n",
      "-- filtering region:651, directly append 239 probes\n",
      "-- filtering region:652, directly append 211 probes\n",
      "-- filtering region:653, directly append 233 probes\n",
      "-- filtering region:654, directly append 308 probes\n",
      "-- filtering region:655, directly append 251 probes\n",
      "-- filtering region:656, directly append 320 probes\n",
      "-- filtering region:657, directly append 290 probes\n",
      "-- filtering region:658, directly append 269 probes\n",
      "-- filtering region:659, directly append 257 probes\n",
      "-- filtering region:660, directly append 284 probes\n",
      "-- filtering region:661, directly append 284 probes\n",
      "-- filtering region:662, directly append 322 probes\n",
      "-- filtering region:663, directly append 271 probes\n",
      "-- filtering region:665, directly append 270 probes\n",
      "-- filtering region:666, directly append 177 probes\n",
      "-- filtering region:667, directly append 241 probes\n",
      "-- filtering region:668, directly append 196 probes\n",
      "-- filtering region:669, directly append 239 probes\n",
      "-- filtering region:670, directly append 273 probes\n",
      "-- filtering region:671, directly append 2 probes\n",
      "-- filtering region:672, directly append 280 probes\n",
      "-- filtering region:673, directly append 7 probes\n",
      "-- filtering region:674, directly append 268 probes\n",
      "-- filtering region:675, directly append 251 probes\n",
      "-- filtering region:676, directly append 246 probes\n",
      "-- filtering region:677, directly append 106 probes\n",
      "-- filtering region:678, directly append 275 probes\n",
      "-- filtering region:679, directly append 186 probes\n",
      "-- filtering region:680, directly append 199 probes\n",
      "-- filtering region:681, directly append 59 probes\n",
      "-- filtering region:682, directly append 138 probes\n",
      "-- filtering region:683, directly append 163 probes\n",
      "-- filtering region:684, directly append 231 probes\n",
      "-- filtering region:685, directly append 282 probes\n",
      "-- filtering region:686, directly append 268 probes\n",
      "-- filtering region:687, directly append 192 probes\n",
      "-- filtering region:688, directly append 172 probes\n",
      "-- filtering region:689, directly append 115 probes\n",
      "-- filtering region:690, directly append 128 probes\n",
      "-- filtering region:691, directly append 140 probes\n",
      "-- filtering region:692, directly append 275 probes\n",
      "-- filtering region:693, directly append 311 probes\n",
      "-- filtering region:694, directly append 194 probes\n",
      "-- filtering region:695, directly append 211 probes\n",
      "-- filtering region:696, directly append 266 probes\n",
      "-- filtering region:697, directly append 242 probes\n",
      "-- filtering region:698, directly append 255 probes\n",
      "-- filtering region:699, directly append 289 probes\n",
      "-- filtering region:700, directly append 167 probes\n",
      "-- filtering region:701, directly append 147 probes\n",
      "-- filtering region:702, directly append 128 probes\n",
      "-- filtering region:703, directly append 246 probes\n",
      "-- filtering region:704, directly append 44 probes\n",
      "-- filtering region:705, directly append 34 probes\n",
      "-- filtering region:706, directly append 281 probes\n",
      "-- filtering region:707, directly append 222 probes\n",
      "-- filtering region:708, directly append 288 probes\n",
      "-- filtering region:709, directly append 112 probes\n",
      "-- filtering region:710, directly append 227 probes\n",
      "-- filtering region:711, directly append 244 probes\n",
      "-- filtering region:713, directly append 288 probes\n",
      "-- filtering region:714, directly append 268 probes\n",
      "-- filtering region:715, directly append 246 probes\n",
      "-- filtering region:716, directly append 101 probes\n",
      "-- filtering region:717, directly append 253 probes\n",
      "-- filtering region:718, directly append 123 probes\n",
      "-- filtering region:719, directly append 239 probes\n",
      "-- filtering region:720, directly append 268 probes\n",
      "-- filtering region:721, directly append 225 probes\n",
      "-- filtering region:722, directly append 136 probes\n",
      "-- filtering region:723, directly append 265 probes\n",
      "-- filtering region:724, directly append 276 probes\n",
      "-- filtering region:725, directly append 18 probes\n",
      "-- filtering region:726, directly append 252 probes\n",
      "-- filtering region:727, directly append 279 probes\n",
      "-- filtering region:728, directly append 257 probes\n",
      "-- filtering region:729, directly append 244 probes\n",
      "-- filtering region:730, directly append 295 probes\n",
      "-- filtering region:731, directly append 301 probes\n",
      "-- filtering region:732, directly append 269 probes\n",
      "-- filtering region:733, directly append 281 probes\n",
      "-- filtering region:734, directly append 318 probes\n",
      "-- filtering region:735, directly append 274 probes\n",
      "-- filtering region:736, directly append 173 probes\n",
      "-- filtering region:737, directly append 148 probes\n",
      "-- filtering region:738, directly append 2 probes\n",
      "-- filtering region:739, directly append 293 probes\n",
      "-- filtering region:740, directly append 262 probes\n",
      "-- filtering region:741, directly append 278 probes\n",
      "-- filtering region:742, directly append 235 probes\n",
      "-- filtering region:743, directly append 91 probes\n",
      "-- filtering region:744, directly append 144 probes\n",
      "-- filtering region:745, directly append 225 probes\n",
      "-- filtering region:746, directly append 149 probes\n",
      "-- filtering region:747, directly append 118 probes\n",
      "-- filtering region:748, directly append 160 probes\n",
      "-- filtering region:749, directly append 224 probes\n",
      "-- filtering region:750, directly append 216 probes\n",
      "-- filtering region:751, directly append 166 probes\n",
      "-- filtering region:752, directly append 264 probes\n",
      "-- filtering region:753, directly append 225 probes\n",
      "-- filtering region:754, directly append 186 probes\n",
      "-- filtering region:755, directly append 252 probes\n",
      "-- filtering region:756, directly append 281 probes\n",
      "-- filtering region:757, directly append 233 probes\n",
      "-- filtering region:758, directly append 197 probes\n",
      "-- filtering region:759, directly append 223 probes\n",
      "-- filtering region:763, directly append 1 probes\n",
      "-- filtering region:764, directly append 31 probes\n",
      "-- filtering region:765, directly append 242 probes\n",
      "-- filtering region:766, directly append 97 probes\n",
      "-- filtering region:767, directly append 272 probes\n",
      "-- filtering region:768, directly append 123 probes\n",
      "-- filtering region:769, directly append 230 probes\n",
      "-- filtering region:770, directly append 313 probes\n",
      "-- filtering region:772, directly append 238 probes\n",
      "-- filtering region:773, directly append 293 probes\n",
      "-- filtering region:774, directly append 257 probes\n",
      "-- filtering region:775, directly append 185 probes\n",
      "-- filtering region:776, directly append 168 probes\n",
      "-- filtering region:777, directly append 4 probes\n",
      "-- filtering region:778, directly append 38 probes\n",
      "-- filtering region:779, directly append 233 probes\n",
      "-- filtering region:780, directly append 237 probes\n",
      "-- filtering region:781, directly append 148 probes\n",
      "-- filtering region:782, directly append 47 probes\n",
      "-- filtering region:783, directly append 72 probes\n",
      "-- filtering region:784, directly append 210 probes\n",
      "-- filtering region:785, directly append 230 probes\n",
      "-- filtering region:786, directly append 319 probes\n",
      "-- filtering region:787, directly append 252 probes\n",
      "-- filtering region:788, directly append 146 probes\n",
      "-- filtering region:789, directly append 291 probes\n",
      "-- filtering region:790, directly append 214 probes\n",
      "-- filtering region:791, directly append 281 probes\n",
      "-- filtering region:792, directly append 213 probes\n",
      "-- filtering region:793, directly append 170 probes\n",
      "-- filtering region:794, directly append 126 probes\n",
      "-- filtering region:795, directly append 232 probes\n",
      "-- filtering region:796, directly append 144 probes\n",
      "-- filtering region:797, directly append 173 probes\n",
      "-- filtering region:798, directly append 137 probes\n",
      "-- filtering region:799, directly append 132 probes\n",
      "-- filtering region:800, directly append 169 probes\n",
      "-- filtering region:801, directly append 200 probes\n",
      "-- filtering region:802, directly append 275 probes\n",
      "-- filtering region:803, directly append 188 probes\n",
      "-- filtering region:804, directly append 176 probes\n",
      "-- filtering region:805, directly append 137 probes\n",
      "-- filtering region:806, directly append 135 probes\n",
      "-- filtering region:807, directly append 199 probes\n",
      "-- filtering region:808, directly append 218 probes\n",
      "-- filtering region:809, directly append 344 probes\n",
      "-- filtering region:810, directly append 178 probes\n",
      "-- filtering region:812, directly append 179 probes\n",
      "-- filtering region:813, directly append 237 probes\n",
      "-- filtering region:814, directly append 186 probes\n",
      "-- filtering region:815, directly append 228 probes\n",
      "-- filtering region:816, directly append 181 probes\n",
      "-- filtering region:817, directly append 183 probes\n",
      "-- filtering region:818, directly append 198 probes\n",
      "-- filtering region:819, directly append 182 probes\n",
      "-- filtering region:820, directly append 115 probes\n",
      "-- filtering region:821, directly append 79 probes\n",
      "-- filtering region:822, directly append 196 probes\n",
      "-- filtering region:823, directly append 341 probes\n",
      "-- filtering region:824, directly append 99 probes\n",
      "-- filtering region:825, directly append 216 probes\n",
      "-- filtering region:827, directly append 143 probes\n",
      "-- filtering region:828, directly append 106 probes\n",
      "-- filtering region:829, directly append 122 probes\n",
      "-- filtering region:830, directly append 168 probes\n",
      "-- filtering region:831, directly append 227 probes\n",
      "-- filtering region:832, directly append 232 probes\n",
      "-- filtering region:833, directly append 153 probes\n",
      "-- filtering region:834, directly append 297 probes\n",
      "-- filtering region:835, directly append 220 probes\n",
      "-- filtering region:836, directly append 296 probes\n",
      "-- filtering region:837, directly append 312 probes\n",
      "-- filtering region:838, directly append 261 probes\n",
      "-- filtering region:839, directly append 240 probes\n",
      "-- filtering region:840, directly append 333 probes\n",
      "-- filtering region:841, directly append 328 probes\n",
      "-- filtering region:842, directly append 299 probes\n",
      "-- filtering region:843, directly append 248 probes\n",
      "-- filtering region:844, directly append 268 probes\n",
      "-- filtering region:845, directly append 277 probes\n",
      "-- filtering region:846, directly append 237 probes\n",
      "-- filtering region:847, directly append 258 probes\n",
      "-- filtering region:848, directly append 262 probes\n",
      "-- filtering region:849, directly append 227 probes\n",
      "-- filtering region:850, directly append 248 probes\n",
      "-- filtering region:851, directly append 289 probes\n",
      "-- filtering region:852, directly append 148 probes\n",
      "-- filtering region:854, directly append 234 probes\n",
      "-- filtering region:855, directly append 192 probes\n",
      "-- filtering region:856, directly append 237 probes\n",
      "-- filtering region:857, directly append 231 probes\n",
      "-- filtering region:858, directly append 204 probes\n",
      "-- filtering region:859, directly append 234 probes\n",
      "-- filtering region:860, directly append 217 probes\n",
      "-- filtering region:861, directly append 323 probes\n",
      "-- filtering region:862, directly append 101 probes\n",
      "-- filtering region:863, directly append 150 probes\n",
      "-- filtering region:864, directly append 258 probes\n",
      "-- filtering region:865, directly append 259 probes\n",
      "-- filtering region:866, directly append 251 probes\n",
      "-- filtering region:867, directly append 145 probes\n",
      "-- filtering region:868, directly append 294 probes\n",
      "-- filtering region:869, directly append 213 probes\n",
      "-- filtering region:870, directly append 257 probes\n",
      "-- filtering region:871, directly append 184 probes\n",
      "-- filtering region:872, directly append 203 probes\n",
      "-- filtering region:873, directly append 93 probes\n",
      "-- filtering region:874, directly append 116 probes\n",
      "-- filtering region:875, directly append 217 probes\n",
      "-- filtering region:876, directly append 196 probes\n",
      "-- filtering region:877, directly append 76 probes\n",
      "-- filtering region:878, directly append 194 probes\n",
      "-- filtering region:879, directly append 133 probes\n",
      "-- filtering region:880, directly append 88 probes\n",
      "-- filtering region:881, directly append 192 probes\n",
      "-- filtering region:882, directly append 220 probes\n",
      "-- filtering region:883, directly append 277 probes\n",
      "-- filtering region:884, directly append 192 probes\n",
      "-- filtering region:885, directly append 180 probes\n",
      "-- filtering region:886, directly append 145 probes\n",
      "-- filtering region:887, directly append 210 probes\n",
      "-- filtering region:888, directly append 220 probes\n",
      "-- filtering region:889, directly append 194 probes\n",
      "-- filtering region:890, directly append 295 probes\n",
      "-- filtering region:891, directly append 173 probes\n",
      "-- filtering region:894, directly append 228 probes\n",
      "-- filtering region:895, directly append 12 probes\n",
      "-- filtering region:896, directly append 316 probes\n",
      "-- filtering region:897, directly append 259 probes\n",
      "-- filtering region:898, directly append 275 probes\n",
      "-- filtering region:899, directly append 179 probes\n",
      "-- filtering region:900, directly append 67 probes\n",
      "-- filtering region:901, directly append 95 probes\n",
      "-- filtering region:902, directly append 288 probes\n",
      "-- filtering region:903, directly append 195 probes\n",
      "-- filtering region:904, directly append 253 probes\n",
      "-- filtering region:905, directly append 237 probes\n",
      "-- filtering region:906, directly append 254 probes\n",
      "-- filtering region:907, directly append 200 probes\n",
      "-- filtering region:908, directly append 87 probes\n",
      "-- filtering region:909, directly append 180 probes\n",
      "-- filtering region:910, directly append 254 probes\n",
      "-- filtering region:911, directly append 331 probes\n",
      "-- filtering region:912, directly append 298 probes\n",
      "-- filtering region:913, directly append 216 probes\n",
      "-- filtering region:914, directly append 72 probes\n",
      "-- filtering region:915, directly append 256 probes\n",
      "-- filtering region:916, directly append 182 probes\n",
      "-- filtering region:917, directly append 159 probes\n",
      "-- filtering region:918, directly append 163 probes\n",
      "-- filtering region:919, directly append 273 probes\n",
      "-- filtering region:920, directly append 287 probes\n",
      "-- filtering region:921, directly append 325 probes\n",
      "-- filtering region:922, directly append 287 probes\n",
      "-- filtering region:923, directly append 160 probes\n",
      "-- filtering region:924, directly append 197 probes\n",
      "-- filtering region:925, directly append 79 probes\n",
      "-- filtering region:926, directly append 255 probes\n",
      "-- filtering region:927, directly append 305 probes\n",
      "-- filtering region:928, directly append 266 probes\n",
      "-- filtering region:929, directly append 125 probes\n",
      "-- filtering region:930, directly append 81 probes\n",
      "-- filtering region:932, directly append 66 probes\n",
      "-- filtering region:933, directly append 186 probes\n",
      "-- filtering region:934, directly append 198 probes\n",
      "-- filtering region:935, directly append 230 probes\n",
      "-- filtering region:936, directly append 293 probes\n",
      "-- filtering region:937, directly append 200 probes\n",
      "-- filtering region:938, directly append 164 probes\n",
      "-- filtering region:939, directly append 292 probes\n",
      "-- filtering region:940, directly append 258 probes\n",
      "-- filtering region:941, directly append 184 probes\n",
      "-- filtering region:942, directly append 136 probes\n",
      "-- filtering region:943, directly append 173 probes\n",
      "-- filtering region:944, directly append 257 probes\n",
      "-- filtering region:945, directly append 216 probes\n",
      "-- filtering region:946, directly append 262 probes\n",
      "-- filtering region:947, directly append 244 probes\n",
      "-- filtering region:948, directly append 187 probes\n",
      "-- filtering region:949, directly append 247 probes\n",
      "-- filtering region:950, directly append 154 probes\n",
      "-- filtering region:951, directly append 212 probes\n",
      "-- filtering region:952, directly append 260 probes\n",
      "-- filtering region:953, directly append 244 probes\n",
      "-- filtering region:954, directly append 190 probes\n",
      "-- filtering region:955, directly append 270 probes\n",
      "-- filtering region:956, directly append 196 probes\n",
      "-- filtering region:957, directly append 238 probes\n",
      "-- filtering region:958, directly append 254 probes\n",
      "-- filtering region:959, directly append 209 probes\n",
      "-- filtering region:960, directly append 288 probes\n",
      "-- filtering region:961, directly append 287 probes\n",
      "-- filtering region:962, directly append 278 probes\n",
      "-- filtering region:963, directly append 282 probes\n",
      "-- filtering region:964, directly append 269 probes\n",
      "-- filtering region:965, directly append 125 probes\n",
      "-- filtering region:966, directly append 213 probes\n",
      "-- filtering region:969, directly append 182 probes\n",
      "-- filtering region:970, directly append 306 probes\n",
      "-- filtering region:971, directly append 262 probes\n",
      "-- filtering region:972, directly append 235 probes\n",
      "-- filtering region:973, directly append 121 probes\n",
      "-- filtering region:974, directly append 283 probes\n",
      "-- filtering region:975, directly append 276 probes\n",
      "-- filtering region:976, directly append 188 probes\n",
      "-- filtering region:977, directly append 211 probes\n",
      "-- filtering region:978, directly append 190 probes\n",
      "-- filtering region:979, directly append 227 probes\n",
      "-- filtering region:980, directly append 240 probes\n",
      "-- filtering region:981, directly append 29 probes\n",
      "-- filtering region:982, directly append 251 probes\n",
      "-- filtering region:983, directly append 232 probes\n",
      "-- filtering region:984, directly append 234 probes\n",
      "-- filtering region:985, directly append 241 probes\n",
      "-- filtering region:986, directly append 269 probes\n",
      "-- filtering region:987, directly append 237 probes\n",
      "-- filtering region:988, directly append 139 probes\n",
      "-- filtering region:989, directly append 284 probes\n",
      "-- filtering region:990, directly append 268 probes\n",
      "-- filtering region:991, directly append 265 probes\n",
      "-- filtering region:992, directly append 100 probes\n",
      "-- filtering region:995, directly append 224 probes\n",
      "-- filtering region:996, directly append 47 probes\n",
      "-- filtering region:997, directly append 305 probes\n",
      "-- filtering region:998, directly append 149 probes\n",
      "-- filtering region:999, directly append 133 probes\n",
      "-- filtering region:1000, directly append 55 probes\n",
      "-- filtering region:1001, directly append 69 probes\n",
      "-- filtering region:1002, directly append 117 probes\n",
      "-- filtering region:1003, directly append 2 probes\n",
      "-- filtering region:1005, directly append 1 probes\n",
      "-- filtering region:1006, directly append 3 probes\n",
      "-- filtering region:1007, directly append 209 probes\n",
      "-- filtering region:1008, directly append 136 probes\n",
      "-- filtering region:1009, directly append 152 probes\n",
      "-- filtering region:1010, directly append 110 probes\n",
      "-- filtering region:1011, directly append 232 probes\n",
      "-- filtering region:1012, directly append 31 probes\n",
      "-- filtering region:1013, directly append 146 probes\n",
      "-- filtering region:1014, directly append 211 probes\n",
      "-- filtering region:1015, directly append 65 probes\n",
      "-- filtering region:1016, directly append 176 probes\n",
      "-- filtering region:1017, directly append 117 probes\n",
      "-- filtering region:1018, directly append 145 probes\n",
      "-- filtering region:1019, directly append 204 probes\n",
      "-- filtering region:1020, directly append 133 probes\n",
      "-- filtering region:1021, directly append 173 probes\n",
      "-- filtering region:1022, directly append 196 probes\n",
      "-- filtering region:1023, directly append 112 probes\n",
      "-- filtering region:1024, directly append 87 probes\n",
      "-- filtering region:1025, directly append 65 probes\n",
      "-- filtering region:1026, directly append 130 probes\n",
      "-- filtering region:1027, directly append 78 probes\n",
      "-- filtering region:1028, directly append 102 probes\n",
      "-- filtering region:1029, directly append 127 probes\n",
      "-- filtering region:1030, directly append 301 probes\n",
      "-- filtering region:1031, directly append 172 probes\n",
      "-- filtering region:1032, directly append 163 probes\n",
      "-- filtering region:1033, directly append 244 probes\n",
      "-- filtering region:1034, directly append 224 probes\n",
      "-- filtering region:1035, directly append 126 probes\n",
      "-- filtering region:1036, directly append 192 probes\n",
      "-- filtering region:1038, directly append 146 probes\n",
      "-- filtering region:1039, directly append 111 probes\n",
      "-- filtering region:1040, directly append 62 probes\n",
      "-- filtering region:1041, directly append 106 probes\n",
      "-- filtering region:1043, directly append 33 probes\n",
      "-- filtering region:1044, directly append 183 probes\n",
      "-- filtering region:1045, directly append 172 probes\n",
      "-- filtering region:1046, directly append 221 probes\n",
      "-- filtering region:1047, directly append 268 probes\n",
      "-- filtering region:1048, directly append 219 probes\n",
      "-- filtering region:1049, directly append 181 probes\n",
      "-- filtering region:1050, directly append 190 probes\n",
      "-- filtering region:1051, directly append 147 probes\n",
      "-- filtering region:1053, directly append 96 probes\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- filtering region:1054, directly append 171 probes\n",
      "-- filtering region:1055, directly append 150 probes\n",
      "-- filtering region:1056, directly append 213 probes\n",
      "-- filtering region:1057, directly append 193 probes\n",
      "-- filtering region:1058, directly append 257 probes\n",
      "-- filtering region:1059, directly append 221 probes\n",
      "-- filtering region:1060, directly append 122 probes\n",
      "-- filtering region:1062, directly append 168 probes\n",
      "-- filtering region:1063, directly append 1 probes\n",
      "-- filtering region:1064, directly append 2 probes\n",
      "-- filtering region:1065, directly append 3 probes\n",
      "-- filtering region:1077, directly append 1 probes\n",
      "-- filtering region:1078, directly append 2 probes\n",
      "-- filtering region:1079, directly append 2 probes\n",
      "-- filtering region:1097, directly append 1 probes\n",
      "region: 139 removed with 22 probes\n",
      "region: 189 removed with 9 probes\n",
      "region: 194 removed with 29 probes\n",
      "region: 241 removed with 1 probes\n",
      "region: 265 removed with 43 probes\n",
      "region: 284 removed with 5 probes\n",
      "region: 317 removed with 7 probes\n",
      "region: 394 removed with 48 probes\n",
      "region: 405 removed with 20 probes\n",
      "region: 406 removed with 22 probes\n",
      "region: 409 removed with 1 probes\n",
      "region: 410 removed with 26 probes\n",
      "region: 416 removed with 3 probes\n",
      "region: 417 removed with 42 probes\n",
      "region: 468 removed with 9 probes\n",
      "region: 547 removed with 14 probes\n",
      "region: 671 removed with 2 probes\n",
      "region: 673 removed with 7 probes\n",
      "region: 704 removed with 44 probes\n",
      "region: 705 removed with 34 probes\n",
      "region: 725 removed with 18 probes\n",
      "region: 738 removed with 2 probes\n",
      "region: 763 removed with 1 probes\n",
      "region: 764 removed with 31 probes\n",
      "region: 777 removed with 4 probes\n",
      "region: 778 removed with 38 probes\n",
      "region: 782 removed with 47 probes\n",
      "region: 895 removed with 12 probes\n",
      "region: 981 removed with 29 probes\n",
      "region: 996 removed with 47 probes\n",
      "region: 1003 removed with 2 probes\n",
      "region: 1005 removed with 1 probes\n",
      "region: 1006 removed with 3 probes\n",
      "region: 1012 removed with 31 probes\n",
      "region: 1043 removed with 33 probes\n",
      "region: 1063 removed with 1 probes\n",
      "region: 1064 removed with 2 probes\n",
      "region: 1065 removed with 3 probes\n",
      "region: 1077 removed with 1 probes\n",
      "region: 1078 removed with 2 probes\n",
      "region: 1079 removed with 2 probes\n",
      "region: 1097 removed with 1 probes\n"
     ]
    }
   ],
   "source": [
    "# filter by genome counts\n",
    "reload(library_tools.quality_check)\n",
    "\n",
    "num_probes_kept = 400 # max num of probes kept for each region\n",
    "filtered_pb_dict = library_tools.quality_check.Screen_probe_by_hit(library_folder, pb_dict, num_probes_kept,\n",
    "                                                                  hit_type='genome')\n",
    "\n",
    "## filter by minimum number of probes in each region\n",
    "min_num_probes = 50\n",
    "removed_regions = []\n",
    "for _rid, _pb_class in filtered_pb_dict.items():\n",
    "    if len(_pb_class.kept_probes) < min_num_probes:\n",
    "        print(f\"region: {_rid} removed with {len(_pb_class.kept_probes)} probes\")\n",
    "        removed_regions.append(_rid)\n",
    "for _rid in removed_regions:\n",
    "    del(filtered_pb_dict[_rid])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "id": "514e3ed4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "988"
      ]
     },
     "execution_count": 186,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(pb_num_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "id": "72ced239",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAApQAAAFKCAYAAACjE8/+AAAAAXNSR0IArs4c6QAAIABJREFUeF7sXQe0FEXTLXKSjCA555weQXIGc0JEUTFg4hNF0Z+gAop84icmxCwqJgwoiiQRJOecowhIzgiS+c+dfb2vtrcn7O7sY4Hqczzydnp6uu/0TN+prrqV5vz58+dJiiAgCAgCgoAgIAgIAoKAIBAlAmmEUEaJnJwmCAgCgoAgIAgIAoKAIGAhIIRSJoIgIAgIAoKAICAICAKCQEwICKGMCT45WRAQBAQBQUAQEAQEAUFACKXMAUFAEBAEBAFBQBAQBASBmBAQQhkTfHKyICAICAKCgCAgCAgCgoAQSpkDgoAgIAgIAoKAICAICAIxISCEMib45GRBQBAQBAQBQUAQEAQEASGUMgcEAUFAEBAEBAFBQBAQBGJCQAhlTPDJyYKAICAICAKCgCAgCAgCQihlDggCgoAgIAgIAoKAICAIxISAEMqY4JOTBQFBQBAQBAQBQUAQEASEUMocEAQEAUFAEBAEBAFBQBCICQEhlDHBJycLAoKAICAICAKCgCAgCAihlDkgCAgCgoAgIAgIAoKAIBATAkIoY4JPThYEBAFBQBAQBAQBQUAQEEIpc0AQEAQEAUFAEBAEBAFBICYEhFDGBJ+cLAgIAoKAICAICAKCgCAghFLmgCAgCAgCgoAgIAgIAoJATAgIoYwJPjlZEBAEBAFBQBAQBAQBQUAIpcwBQUAQEAQEAUFAEBAEBIGYEBBCGRN8crIgIAgIAoKAICAICAKCgBBKmQOCgCAgCAgCgoAgIAgIAjEhIIQyJvjkZEFAEBAEBAFBQBAQBAQBIZQyBwQBQUAQEAQEAUFAEBAEYkJACGVM8MnJgoAgIAgIAoKAICAICAJCKGUOCAKCgCAgCAgCgoAgIAjEhIAQypjgk5MFAUFAEBAEBAFBQBAQBIRQyhwQBAQBQUAQEAQEAUFAEIgJASGUMcEnJwsCgoAgIAgIAoKAICAICKGUOSAICAKCgCAgCAgCgoAgEBMCQihjgk9OFgQEAUFAEBAEBAFBQBAQQilzQBAQBAQBQUAQEAQEAUEgJgSEUMYEn5wsCAgCgoAgIAgIAoKAICCEUuaAICAICAKCgCAgCAgCgkBMCAihjAk+OVkQEAQEAUFAEBAEBAFBQAilzAFBQBAQBAQBQUAQEAQEgZgQEEIZE3xysiAgCAgCgoAgIAgIAoKAEEqZA4KAICAICAKCgCAgCAgCMSEghDIm+FJOPnHiBK1YsYKuvPJKSp8+vU+tSjOCgCAgCAgCgoAgkKgInDlzhvbu3UtVq1alzJkzJ2o3U6VfQih9gnnBggWUlJTkU2vSjCAgCAgCgoAgIAhcLAjMnz+f6tate7F0Ny79FELpE6xbtmyhkiVLEiZVwYIFfWpVmhEEBAFBQBAQBASBREVg586dljHpzz//pBIlSiRqN1OlX0IofYJ5+/btVLRoUdq2bRsVKVLEp1alGUFAEBAEBAFBQBBIVARk7U+5M0IofZqlMql8AlKaEQQEAUFAEBAELhIEZO0XQun7VJVJ5Tuk0qAgIAgIAoKAIJDQCMjaL4TS9wkqk8p3SKVBQUAQEAQEAUEgoRGQtV8Ipe8TVCaV75BKg4KAICAICAKCQEIjIGu/EErfJ6hMKt8hlQYFAUFAEBAEBIGERkDWfiGUvk9QmVS+QyoNCgKCgCAgCAgCCY2ArP1CKH2foDKpfIdUGhQEBAFBQBAQBBIaAVn7hVD6PkFlUvkOqTQoCAgCgoAgIAgkNAKy9guh9H2CyqTyHVJpUBAQBAQBQUAQSGgEZO0XQun7BJVJ5Tuk0qAgIAgIAoKAIJDQCMjaL4TS9wkqk8p3SKVBQUAQEAQEAUEgoRGQtV8Ipe8TVCaV75BKg4KAICAICAKCQEIjIGu/EErfJ6hMKt8hlQYFAUFAEBAEBIGERkDWfiGUvk9QmVS+QyoNCgKCgCAgCAgCCY2ArP2XKKFcv3491apVi4YNG0b33nuvNcqlS5dSjx49aOHChZQ3b156/PHH6emnnw4icO7cORowYAB99NFHdPDgQWrUqBENHz6cypQpE9EklkkVEVxSWRAQBAQBQUAQuOgRkLX/EiSUp0+fpoYNG1rEccSIERah3L9/P1WoUIFuvPFG6tmzJ82dO5ceffRRizB27drVQgFkEn/jnMKFC9MzzzxDmzdvplWrVlHGjBk9T3aZVJ6hkoqCgCAgCAgCgsAlgYCs/ZcgoezTp49FGKdOnRoklIMHD6Z33nmHtmzZQunTp7dGjXqjR4+mtWvX0qlTpyhfvnw0ZMgQevjhh63jhw4dokKFCtEnn3xCnTp18jzhZVJ5hkoqCgKCgCAgCAgClwQCsvZfYoRy+vTpdNNNN1nb28WKFQsSyg4dOlCuXLnoq6++Co548uTJ1Lp1a9q9e7dFNOvVq0fr1q2jcuXKBetg27tatWqW5dJrkUnlFSmpJwgIAoKAICAIXBoIyNp/CRFKWBSrV69OsEZ27tyZ0qRJEySUIIXt27enV155JTji1atXU+XKlWnBggW0detWuuWWW+j48eOUJUuWYJ2OHTtav40dO9bzjJdJ5RkqqSgICAKCgCAgCFwSCMjafwkRyjvuuIPOnz9P33zzjTUqTigRWAOSOXDgwOCI4R9ZunRpmjFjhmWh7NKlC509e5bSpk0brHP33XfTjh07CNZMu3LkyBHCf6rs3LmTkpKSaNu2bVSkSJFL4kGRQQgCgoAgIAgIAoKAPQJCKC8RQjly5Ejq3bs3rVixgnLnzh1GKKtWrUrY9jZZKBcvXmwF39x6661GC+XJkydpzJgxtrOof//+VkCPXoRQyqtHEBAEBAFBQBC4PBAQQnmJEMrmzZvTzJkzKVOmTMERHTt2zPobVsjixYtbUkEgnqqYfCg3btxo1VcFPpTYRkdAj1goL4+XgoxSEBAEBAFBQBCIFAEhlJcIofz777/p33//Dbn/ZcuWtSySt99+O3399df03nvv0aZNmyhdunRWPVg0f/zxRyvKG1bI/Pnz09ChQ+n++++3jkuUd6SPk9QXBAQBQUAQEAQuTwSEUF4ihNI0fbkP5Z49eywdyuuvv97Sl5w/fz498sgjFsm85557rNP79u1L77//viUTVKJECerVq5flW4ltdNGhvDxfEDJqQUAQEAQEAUHACwJCKC8TQolhIpob2XGWLFlCBQsWpKeeeoq6d+8eRAABOdCmhLA5rJ1NmjSxtrpBLiMpMqkiQUvqCgKCgCAgCAgCFz8CsvZfwoTyQk1PmVQXCnm5riAgCAgCgoAgcGEQkLVfCKXvM08mle+QSoOCgCAgCAgCgkBCIyBrvxBK3yeoTCrfIZUGBQFBQBAQBASBhEZA1n4hlL5PUJlUvkMqDQoCgoAgIAgIAgmNgKz9Qih9n6AyqXyHVBoUBAQBQUAQEAQSGgFZ+4VQ+j5BZVL5Dqk0KAgIAoKAICAIJDQCsvYLofR9gsqk8h1SaVAQEAQEAUFAEEhoBGTtF0Lp+wSVSeU7pNKgICAICAKCgCCQ0AjI2i+E0vcJKpPKd0ilQUFAEBAEBAFBIKERkLVfCKXvE1Qmle+QSoOCgCAgCAgCgkBCIyBrvxBK3yeoTCrfIZUGBQFBQBAQBASBhEZA1n4hlL5PUJlUvkMqDQoCgoAgIAgIAgmNgKz9Qih9n6AyqXyHVBoUBAQBQUAQEAQSGgFZ+4VQ+j5BZVL5Dqk0KAgIAoKAICAIJDQCsvYLofR9gsqk8h1SaVAQEAQEAUFAEEhoBGTtF0Lp+wSVSeU7pNKgICAICAKCgCCQ0AjI2i+E0vcJKpPKd0ilQUFAEBAEBAFBIKERkLVfCKXvE1Qmle+QSoOCgCAgCAgCgkBCIyBrvxBK3yeoTCrfIZUGBQFBQBAQBASBhEZA1n4hlL5PUJlUvkMqDQoCgoAgIAgIAgmNgKz9Qih9n6AyqXyHVBoUBAQBQUAQEAQSGgFZ+4VQ+j5BZVL5Dqk0KAgIAoKAICAIJDQCsvYLofR9gsqk8h1SaVAQEAQEAUFAEEhoBGTtF0Lp+wSVSeU7pNKgIHBRIfDZ7C303rRNNPzOWlSzWO6Lqu/SWUFAEIgOAVn7hVBGN3MczpJJ5Tuk0qAgcFEhUOL/frX6mz1TeloxoO1F1XfprCAgCESHgKz9QiijmzlCKH3HTRoUBC4VBBShxHi2/PeaS2VYMg4DAifPnKVM6dMJNoIACaEUQun7YyCTyndIpUGfEVi76wgt3HKQbqtTRBZDn7FFc0Io4wBqAjb59u8b6O0pG+n9LrWpeYX8CdhD6VJqIiBrvxBK3+ebTCrfIZUGfUZAEZ7b6xSljnWLUu3i4ufnJ8RCKP1EM3HbsrvPJ06fpfl/HqCkknkocwaxXibuHfS3Z7L2C6H0d0YRidnbd0SlQb8R4Ash2v6mW32qXyqv35e5bNsTQnl53Hq7+/z410vo52U7qE2lAvTB3XUuDzBklLL2szmQ5vz58+dlTsSOwOX+lXL439NWhGvrSgWolkS4xj6h4tCCTig71ytGL99UNQ5XujybFEJ5edx3u/ss9//yuP/6KC/3tZ/jIYTSp2fgcp9UT3yzhH5ausNCUwISfJpUPjbzv4nraNjUjSEt3lW/GL1046VLKBdvPUh7j56ktpWv8hFJ+6aEUKQKzBf8IkIoL/gtSKgOXO5rvxDKOEzHy31SVXlhIv1z8owQyjjMLT+a1K2TaPPuBsVp4A1V/Gg+4dqAP1uF5yZY/RrVrT7VS4WtfSGUCTcN4tIhIZRxgfWibfRyX/uFUMZh6l7uk6ricxPo39NnhVDGYW750aSJUN7bsAT1v76yH80nXBt7jpygpJd/t/r1ZKty1KNV2bj38WIklKfOnKMTZ85SjswZ4o7PpXKBaAnlW79voCP/nqa+11SkNGnSXCpwXPbjuNzXfiGUcXgELvdJVa7feMLihCJb3nGYYDE2aSKUXa8uQS9cd2kSyl2HT1D9wQFC+VTrcvSflqlPKKeu3UOb9v5D9zcqmZAEAu7z1w2bSet3/0PTejWjgjmzxDjLLo/ToyGUe46eoKRBgfn49h016brqhVIVrIPHTtHnc/6i9lWvonIFsqfqtS/1i13ua78QyjjM8It1Up07d55+W7ObyhfITiXyZYsambJ9x9Hps4H4LiGUUcMYtxNNhBJE57lrK8Xlmv+eOkuT1+ymhqXzUt4rMsXlGk6N7jj0LzX87xSrSq+25emx5mXi3geO8dj/NKJr355pXfON22vQjTULx/36kV6AuwVcytbqSHFxqx8Nodx5+F9qMDgwH7vUL04v3pi6riYPfLaAJq/ZI+9nt5sbxfGLde2PYqiup0hQjitE3ipcrJPqh0Xb6anvlsX8oinV+1c6l6wXIITS25xJzVomQvlg45LU9xp/CeWxk2cIi+eH0/+kUQu3UYm8WemPXs19GSoI0J/7jlGFq7K7Wvy2HzxOjV6Zal33mXbl6dFmqUsoi+TOQtsP/mtdP1GDn5DtpXy/gJ9pxzpFaMit1X25T5d6I9EQSv6Bg4+srx6sn6owXYzuGKkKUAwXu1jX/hiGbHuqEEqfUL1YJ1WXj+fRjA37YiaU8sKKfSJhCxJ+qFkzpo+9Ma0FE6F8qEkp6t2hoi/XmrNpP41Z+jeNXvw3nTobcH1Qxa8PjJuGz6IlWw95svj9tf8YNX31D6sL/9e+Aj3ctLQv43RqhGN8Rab0wSC1O5KK0uCbq7lef9FfB+i7hdvp8ZZlqVCu+G8/cwvltdUK0rDOtVz7KBXsMyI5vQO3HThOjYcEPnBK5ctGU55ulqpQ2vUNH4DZMvn/vknVwV3gi12sa388YBNC6ROqF+ukunfEfPpj3V4hlD7Ng1iaeeyrxTR59W766bGrqWLBHBE3tfLvw7RgywGCvqSeZ9hEKEGyQLb8KKb2/SaU6hr5s2ei+X1bOXZ7455/qNXQaVadPh0qULcmqUsoM2dISydOB4j1bbWL0Ku3uVv/1PiqFs5Jv/ynkR+3xbENTihbVshPH99bN+7XvBQuEI2Fcuv+49Tk1QChLJonC814pkWqQmHqc/+fV9GX8/6iz+5Looal86Vqfy6li12sa3887oEQSp9QvVgnFfet+XNwB9etRDu4xEIZ+0RSGF5dJi99+UDkW2Lq/B4ty9KTrcsFOwTLZ8ne48I6+Giz0vRMu/gTymGda9K11WIPQlDjK5P/Cprcs6kj4Mhb3u6NGVadftdUpAcal4r9Brm0wJ+BjOnSBi21N9UsTK/fXsP1+qn9DMHPteLzgS3v+qXy0DfdGrj2Ua+wZd8x2nXkxGWVcSkaQgmcmv0vYDEvmDMzzend0oj1+BU76aVf19BLN1bxNU+4qc/qt+yZ09OK/m0jvvdyQgCBi3Xtj8f9E0LpE6oX66R6eOQimrBql4XCupfahVm2vMKT2ouh135dLPXOnjtPpfsESF+jMvnoiwfqRdx1dQ9g3Rzfo3HwfETfIwpfL92bl6Gn25aP+DqmE5wslKgf67Y3x6dWsVw0+tGrbfsNf7Ub3plliZqjIPAIAUiRlnW7jtKLY1fT/Y1LUvPy+V1P5xikS5uG0GcURPQistetpPYzxAll9SI5aUx3b1bR02fP0c5DJyxLm/pQuZzSeEZDKDfv/YdavBawmOe7IhMt7Ge2sMdrDph83NW1smZMR6sHtnObnnLcBoGLde2Pxw0VQukTqhfrpMI266/Ld1oorOjfhrJHoUenW8CiJQ8gPhnTp/XpjlxczRw+fpqqD5wU7DT86J5oWZbSpvWuV6cWCAStTHiiSbAt+ElVfmGikVA+0aospU9nxtzr/eDBHXaoRzInTNeF7EnNF3+zmm9W/kr6tGuS7Q3u9MEcmrv5QPD489dWovuiIJSRaqvakeoOVa+i4XfWdp2Q0ZKJr+ZtJaQ+faRZZNv6x0+doUrPB+aFF6uvGsA9n8ynaev3Wr6sT4xaav18T4PiNCCVRfK9zk9X4IkIPo4fz/yTOiUVpQpXObubREMoN+45Sq2GTre6kitrBlr6fBtjt6KdA25jNKlwCKF0Q83b8Yt17fc2ushqCaGMDC/b2hfrpOrxzRIak5wycVG/VlFJvDz7/XIroleVSMiDOmfZtkN0x4dz6Y6kYnGTsvHpVselGe60ry7w2m3V6ZbaRTxfz45QHjp+imoMDJAxvcDK9NuTTSlzhnQhhyat2kXdv15CXRuWoCJ5stK1VQtS7mwZjW3sPnKC6iWLiNt11uucWL3jCN323mxLZmcQyzOO6O7myVuG11cvRG85WPxqDJxEh46fDnal/3WV6N6rI7dQRrq42xHKNpUK0Ad313G9j5FeDw1yOZqvH6xPDUrndb2OqsA/NArlzEyzbbZh9QZVPzOkSxOUCktt2SH4yN70zixqXbkADe3o7k7gBkqL//1Bm/cds6q5zdVoCOX63UepzesBQomArZUDzFvM0cwBt7HheIXnxgd9etX4hFB6Qc69zsW69ruPLPIaQigjx8x4xsU6qZ76dhn9sHi7NaY5vVtQNOLG+kK6+eUOEVnWcO1Gr0wJyqy4vdB9umUJ1QwCapRuoerYQ01LUe/2KVHY70zdSGOX76SP76ljjAK2I5RcVNk0aJNOon5PG5TKS193M/t1cn/FWAnlDcNm0rLth8MW9qXbDtGN78yyfnfT8as+YJJlsVNl4A2V6e4GJSK639FY3e0IZYsK+ekTDwEv0ZCJVTsO0zVvBfQuB99c1fog81qQKhUpU1GcrGZ6e6ZxprZI/p0fzaVZG/eHzROvY3cak9v7JxpCyZ+RTOnT0rqX2hu7Gs0cUA3N3byf+v20kp5uU47aVSkY0r4pNa4QymhnS+h5F+va78/oQ1sRQukTqhfrpOLWRWTLKJ43cnFzfYHZMKg9ZbDZRrWDO2nQZNqT7PPGX+jY6sySMV2YBc2n22ZsBsQNOoYv3ViV4AuXGmX2pn3U+cN5IZfSo7AVznYERR2HSP3EJ1O2vP8+9C9dnSzybRrL/26rTrdqllATabBbaLGQdfpgriNMbou0Ovm6t2fSir/DCSW2WLHVimIKJsL2Z58fV1jR8W9OXk9HTgTyyqNARBokNJKib+N76b8doWxS7kr6/L7QLfpv5m+lOZv3W1ZYWKxQoiETa3YeofZvBoKPBt1Uhe6s532cR0+cpqr9A24WcDVZb0NydNxM44ynSL7pvt3y7mxa9NdB65CXe8PbgMUeFnlulY8E+2gIJSzvHd4K3Ce8Uza93ME4HSPph9N90TGp1n9i8JkQC2UkbwL3uhfr2u8+sshrCKGMHDPjGRfrpMIiDB8slN+ebEJlo0jLpS8wa19sFzEBNBFKRYQK54LMRvOIrZ7R3Fruq5eaGU4mrNxFD3+xKKTLOnFSONvJyqjj5QpcQZOeTImC5hGmJkxiJZRT1+2hriMWOMLtddG/9d3ZtNBAFCCn9MDnC61rPN6iDPVsExpMBP83BNCgcA1I/I2I2bsiJJRcUscrabEjlKaofVW3W5NS1CdZCzQaMoHAobZvBLZSIx0nJ5ROJMcLoXygUUnqF0HWJWToisQ/WO/DzcNn0eKthyImlMo946ocmWnW/7UIfjBGgn2khBLvlCe/XRqUZ3OaT5H0IxJCWevF3+jAsVPWKSCzuN9ioYxmhQg/52Jd+/0ZfWgrQih9QjVRJxVe3L8s30GVC+W0HO/18sKYlfTZnL+sn5EurkrhnMEqyIc8a+M+6lC1oGUltCv6Qgr/IGV18QpvvZcn0+4jgahcRT4Gj1tD70/fbP0WTZter83r8RzQqZWyD9f/duE2eub75SFd1qOwFc5VCuegsf9JieJWJ6njZfNfQb8xWR3uv2XC5NVbq9FtdYqGHIrEQjlx1S56aGQoGdav45VQ8q1MLmPFCeV/WpShpzRCCU29T2dvsS6bJUM6SyBelWgslDxgBe1Aqy9bxnRUp0Qez8+BqphUMg99+1CoJI/J2hwNmeD39sUbKlOXCLb2j5w4TdWSLZQwxG8efI2nR8Y0NzgxdmvkjcnracSsLTSia12qVSy3W3XjcSVyz98XXhp66/cNNPS39VbVBX1b0ZXZA2lBI8E+UkLJpdlUH+2eh0j6oY/X6dy6gyYHVQ/UB78QSi8zxr1Ooq797j33v4YQSp8wTdRJNXLOFnpuzCprlKaX2MBfVtMns/60jo9+tGHIC175ot1Zr1hIgITTiwzHlj3fhnJmzRARsiZC+fK4NfRBMqH87uEGVK5AdsqZJbJ2I+oEEXF/w56ty1lZS+JZYAl78POFhG1jlQtdXU8nTmoBqFwoB/36uHdCafLP5GMacms16hgDofxl2Q76z9dLHGFyIpQIABo2dSMNvKGKtV09NVlon8tY/bZ6t4UTiknuiBNKrgGJ+gOur0z3NIzMh5L7F/KBLX2+NeXKag5OsrNQ1i6em354pGEIPn4Ryg27j1Lr5GCPSMcJP1M84yhp0hD96YFQ6pZbNahIsi6psXsRqLebVJxQRqKfywnl/L4tKX/2zHEnlJF8nMWLUNZ/+XdLLxRlOdQ8MqUPSj6JbFBsb/hEXftjG1V0Z1/0hHLPnj301FNP0YQJE+jff/+lpk2b0quvvkqVKgVyFC9dupR69OhBCxcupLx589Ljjz9OTz/9dBCtc+fO0YABA+ijjz6igwcPUqNGjWj48OFUpkxkuX8TdVK1e2M6rd111JZQctKma8l5fblBP1Fp7uFC0USL8xeeIh+Dfl1NH84IkF1VPry7DrWuVCC62e7hrH3/nKQ6L022akJS54lWKQLhHk6PuAoyVfT9caXxPJBZkFpV1P2oVDAHjWM6k/pxXQJmydaDdNPw2bZ9G3JLNepYN3oLJc8Hb3cRJ0LJ51m7ylcFdVGXvdAm+AEB0tkt2Qr6WPPS1KttqCD7gF9WWVYvFFjbVF55/G2SDUJ09KsT19EttYrQ1WXCs4Tw7WA+pglPNLaVlbEjlNWL5qIxj4XqZvpFKPmWd6TR7LpUlRcr8u3vz6F5f6ZIMilsIsm6pMYOUrPCJtrZ7UFCgBYCtVA2DmpvK32lt/P27xvotWQL5fw+LSl/jsuDUMKHGi5E6v0MeTilTes3ocRHB4quHOF2Ty/W44m69l8IPC96Qlm/fn1KmzYtvfXWW5QtWzZ67rnnaPbs2bRx40aLYFaoUIFuvPFG6tmzJ82dO5ceffRRizB27drVwhtkEn+PGDGCChcuTM888wxt3ryZVq1aRRkzmi0RphuVqJOqweDfaefhwJepacF4ZcJaevePTdZxBA4ggEAnKHbnqnrcPwe/8S9/r5Pa1E8ToXTri9fr2dXjhFIndLG2bTr/k5l/0sBk3z/9uE5o1UKsC5fr96v0ldno96dScgXP//MAdXx/jm33X7mlKt1eNzQ6OBKrCl/c7S7ilVDeUKNQUMaKzyO+rQ69xWfbVSBEYqeBaQ2BN2NXWzqCpmLKlOMW1MGtd7zNyT2bUJn82Y3XsSOUJouyqtu8/JU0IllT0+sHHL84D/aIVMBdl5PyQijtxqjuiZdnRLURSWS53i6E6yE1hhKJz/awKRvof5MCW97z+rSkAheIUNqR4GjmgJf3dZMhU2nrgeNWVah5YKdHaZCCUH7QpQ7BfQLyT7H4toJMQt7r3PnzNK1Xc0dSCXcsuKlg56lR2Ys39WOirv1enkW/61zUhHL//v3UvXt36tevH1WuXNnCZtmyZVSjRg2aP38+TZ48md555x3asmULpU8fiKTs06cPjR49mtauXUunTp2ifPny0ZAhQ+jhhx+2jh86dIgKFSpEn3zyCXXq1Mkz3ok6qSo9P4GOnwp8MZoWjP9NXGdtN6J8dHcdapVs/Ytksan94m+0P9nhW72wIpUfMhHKl8aupo8MJCGe/pR8y9vPTDJ2E+nHJdvpyVHLjIefbFWOerRK2XJXi40uXK4vKKWuzEZTGKGEH+ydH4VGkPMLxkIoEWkLcuZWvBLKjnWK0LcLAzL11dBTAAAgAElEQVRWCMQqmier9W8etARrGO7Tkq2H6MdHG1pb0HZzBef2bl+BHmoaKvrttnDbaXf+/lRTKn1luC+yXXpLXF+/X7xurIRSd2dQARdu98N612n6pKZ7BEFufJA2Lhv40PSTUOa7IiMt7NfaS1fD6nB5qUjeB4lCKFcPbEtZMwbWJF7c5qUTWE7ncp3N6b2aW4RSJVIAoVRrhClAL5IbNHXtHur6aSBAj68npja+XbCNnvkh4Dfu5WMmkn6kZt1EXftTEwN1rYuaUOqA7d6929rO/uOPPyzCeNttt1GuXLnoq6++ClYFyWzdujWhLohmvXr1aN26dVSuXMrWIra9q1WrZlkuvZZEnVRuL6ghE9bS8GQL5fA7a1kBOCjXvDWDVu04Ehy+0wOvWyg5EfCKn4lQ2lmdsH2IbcR4FC7S7Weua7u+8mATvc5TrcvRf5gPZ7SE0i0K26Rf6NVCOWrBVnr2hxWut8IroYS8z8i5gSAxrjowYeVOeviLxdbv8NdTwVoqutjOmo36z7QrT482C3VhcXsuEBGLea0X5BA3Bbfx1JD6OboLAq8bK6GElQ7WOlUiETfnigamRR0WpFLJ6UC/erAeNSydz5ZQmtwQ7CaFwh6R1nP7mHNaO00oWMEgXn/i9DmrmpNfq94OJ5Rze7ekq3JemC1vuz67zUsnXJzObT10Gm3Y8491OuYwrMPKtSdzhrRBLPFBN+TW6q7Ps10FLu/1fpfa1LbyVbZt9fx2KY1e/Ld1PBJCiV0kPEPKuhx1Z306MVHXfp+GF1EzCUMosT2dLl06a5t5zZo19Ouvv1LDhg2t/7yUbt260YcffkiZMmWin3/+mdq0aWORwvbt29Mrr7wSbGL16tWWNXPBggW0detWuuWWW+j48eOUJUuWYJ2OHTtav40dO9b20keOHCH8p8rOnTspKSmJtm3bRkWKeM9u4mVssdRxe0ENHr+G3p8WiKTmMjk6oXB64GsOnEQHWWaSP55uRiXyRaZn2XDw77RD25rnAUMcg3c616JrqoUK98aCET+XR3nrwuKRXAOLnp0PESK6N+89Rs+0LU8ge/d/Fgg20YseZe6ZUObLRlOeTtny5gEtpuuYhL+9EsqflvwdTL/nhI9XQgmCqKzSXHWAE8r7ri4ZDCTrVLco/feWasQVAfR+QOi5e4vQ4Cq354K7PvD2Jj7RhMpfFb7ljfzWZfuG50vHuboLAs+tHiuhXLz1IN3M/GO/eqAeNTT4hJrujU6a9XvE+6mEy+0slJFY81UbkAODdE+kpeuI+cHALZzLo7Xd2oLGLHxnUXBt9AHFbT7wdu3qevldtcP9N7207TYutzFwX/pxjzem3NkyUIPBU6xmecaj22oXoVdvi55Qztywj+76OLAb8u6dtah9soHC1P+eo5bS6CWREUqe3SkaX30vOEZaRwhlCmIJQSinT59u+Tl+//33VjANrIXwi/znn3/o66+/tiyNbgVEEaT03XfftSySM2fOJBDDzp0708CBA4Onwz+ydOnSNGPGDMtC2aVLFzp79qx1PVXuvvtu2rFjh7Vlblf69+9v+V/qJZEJpSmDDQ/K4cEZkRBKPdWdnRXH6R5yp3G1sPFAC36uSebGbX54Pb7j0L/UMFkEXJdC+ffUWYJFDkEcTnqdID/dv1pCPduUC7OM8SjZoR2rW7nTVfSy3kc7QqkLl6vz1D0rmS8bTWWE0qRxya9l8jH0SijHrdhJj34ZsBw6Fa+EElZhZTG/pmpBy7oIsf3xK3bSI8nXQd5oJXWlLCr8w0jvh+464Lb44rhddqFfH29kSXDpxSmfue6CwOdArIRy4ZYDdOt7Kf6xemCd0z3Z/89Jqp0cgIZ6ToRSpVa0I5QmbVC7a6s2iuXJStOfae42dcKO632IJMMXJ5R8JyW1CaXdLk4k/dCBcTqX7zhhhydPtozUeMhUqwm4IZ8/H2gNCQ6w7R1tmbNpv5VCF2VY55p0bbVCtk09OWop/RghoeT+4G92qkE31CgcbVd9O08IZQqUCUEoscUMEvnmm29avouI0t6wYYP1b0RfL1niLEnCZwaitqtWrWptZcMK2aFDB6OFcvHixVbwza233mq0UJ48eZLGjBljO+kuhIUSX3Qb9/5Do7o1cNSF5J3mLxlkwkBGDF74tjLPtBEJodRT3dlZcfh1EUwADl/hqhzWzyZCyaVgQvocod5eJG8OZMhp9ErgRauLNfP+OEkjOb3YebAHtgmrFcllq+Gob9WqdnXhcjU+dbx43qyWQ7wqbqQPAS4IqrCbN+p3Eyl02rLn7XkllCB/r08OBE2gQAJo/aD2xMcAGasvk8X4EaX9Wsfq9N/xa+m9aYHgMr2YgqvcFm67/OR27hZ2cjroSynNYsw1LpuVv9JKr3nw+KmQbENetwDnbd5Pt7MsRd8/3MBRK5Njo1thoQuJFJvKss5Jskp3aUcoTdqgds+dakPHxctzasI5Eheb4X9spCETAhZKvpPiNh/sng1+n0xtcJci3saUp5pSKYMvbiT90PFyOpf7nEKCDYSy5WvTwiBXz5OXe2GqEwnhe+KbJfTT0h1WM16lnyJpP9oxRHqeEMoUxBKCUGbNmpVWrlxJpUqVsghg0aJF6f3336e//vrLitKG5dFUIBn0+++/W5ZIbJerAosmCB9+g1TQyJEjg8dMPpSICIfVUhUQ3OrVq1sBPV5LvCcVd6A3WZTcXt44vmZguzAiykkS3/qMhFDytF64jp0VR/URki1qu2Vhv1aU74pMERHKPh0qULcmoQTI631yq7ftwPHglztSHObOmpEK5cpsCWlzS6zuF+dlwUEdLtWCbUKIlCvfQL1v/9e+AiEABYUHcujC5eo8O0L587Id9LiDTqTJglemzzg6w7V3bPycpq/fS3cnp0R0wtYroeTWR9Uezv11+U567KuAJfT2OkVp1MJt1r9vrlmYht5eg7hagd4P03as28LNXR94eyPvTwoGqPDfdSF0fky3GHNJImhUqhSC/ByvhFJP2alryWLebDvwLxXJnSUsenfv0ZMEwWterq1WkIZ1rmX9xMmb0qK1I5Re/Y29zGOneYQgoVZDA5mBVIE1Hhh7KZxQ8gArt/ng5fk2tWGHl11WMr0NvCvzZssUZgjQx4qPKXxU8WeG1+GEclS3+lYgm8qwxOvdXKswDe1YwwuUxjrcYv767dXpppr27l89vlkSVHQw7Z6ZLrBgywG6Ldkin5qZzJwAiffaH/XNuAAnJgShvPLKK60t6JIlS1oEEBI+IIXz5s2jG264gXbt2mWEBhqTNWvWtEhlixYBX5zTp09bJPT666+nAgUK0HvvvUebNm0KEs7evXvTjz/+aAXtwAqZP39+Gjp0KN1///3W+Yka5c39nXq0LEtPMn1Cp3nDX1ArIGibOVQYnGfK4bIjkRDKqv0n0lGWO/nn7ldblje7wq1NXz5Qz9pCbvTKFNp+MPDhoBZTOwtlPPUh/9p/jJq++kdY1yEG3HTI1BBf0WiyXfAPA1h1YKFVREm/KCfOPJDDjszabSW6+TmaCBc06uBDx4tpvNwJ32keeiWUpjZw7tjlOyw3AhQsesqZHzJDb3aqSXaWINQ3Sdq4EQju+qD3ybT42elW4twSebPSH8xizD8q4MOn9AHdsDZhw33WcBxR7zVZ9hnkDP+/0StI+UDyNkyEkj9/cPGo+PwE65TO9YrRyzdVtQ3K8Spszv0y7eSvnOaRSbEAbgOwQvNx27XBCSX3rXWbD7w9u7qREMrxPRpbeef1wtv44ZEGdMu7c8hOd1ade+bsOSqj+e/qzxsnlAiwypE5A1379syw69+ED7SO1em9aZutLEI31ijkWeMTjXHNW7eIcXzk4mMXxauWKCesQiidnpQLcywhCCUsjLBC5smTh3744QdCgAtI4EMPPWRZLeFHaSr42kXQDXwhEZCTO3duGjRoEE2cONESNM+cOXOQXEJfElJCjzzyiEUy77nnHqvJvn37WtZQbK+XKFGCevXqZbW3YsWKhNKh5BGZ8Fe6o14xKzfsddULOaY55C+oJc+1ptzZQrU1+/20gr6YG8jlzQlMRITyhYl09OSZ4C3SFzX93nFrk4oebTxkimVJ4QsaJ7u8jViCZdweM7u814jKbPHatGA+XN5Pp0VBf7HzDwMQSmiw2WWZ6duhIj3YpJTVPF+I3Qhl0TzIfZ4S7PD9ou309HdmaSK0bSIDFZ+bEJK+0G68XCbECdtYCSXPxnN99ULBhQjz/+07atKrE9fSO1PNW96m8ZkWf7xPxizdYUVxIwpWuT7o48LHBRZkXux0K1EH1sGZz6bcDz4HQDa37A/oA/Li1UKpE3qQkNrFU9JDOhElOz9RdW1udb0jqSgNvrmaLaG8v1FJwgepW+FBFXYpRJ3a+GPdHrrXJm+8F8yguQtrtirqHI6T+si164cfhFJPc6uuxduuXyoPzd0cEJF3Ght/N+jjUn9zQonxQSrIlOwAhBIfaRxjbJHXdUg5ynFavv0QXT8soDpgSpjA63b/ajGNXb7T+mnDoPaUIV2oO5YJfyGUbk/YhT2eEIRy7969lg4kfBpfeOEFK0AH2W9goUSgzlVX2UsPHD58mGB1/OmnnyzrYuPGjS2Lo9KlhB8lsuPAD7NgwYJWu9CuVAUBOdCmhFUUpLZJkybWVjfIZSQl3mZv3bL19fyttO+fU1a0M6KeTUWXMjFFQ/YevYLQFgoPAnEjlPCvuveTBVQiX1b6ZdlOQqo6Vdz8uDg5UDInJkL5/JiV9HlynnE+vrsbFLfS9EVbnvl+mWUVGnFvUnArCfhu2nvMIhIm36LFz7WmVkNjJ5TcKoQPg9L5r6Ae3yw1DoW7NvDtRz1qWF+MdALD9d5MF1IBF/xYlRcmhtxTu0XNbx9KU/+wmPJt+w5Vr6JxKwK7Fmr+vzZpHb09JaCnqhfdFxbHTaRgzNK/g/cCfnkqaEFvz6QhaKdbiXML5sxMc3qnyONw38VYLZQ6ocd2Zr1SeYNd5uPEsVrFcwcX7j1HTlDSy7+H4aXICyd/cDN45VZ7QglXhQEenkmOE6S/PuxSm/YcPUlVCocHOpnupdN880Io9a1hpdupv+/sorDt5o7d73Zb3j89djXVKJrL+kDFrgj+DZF+Xr9h6bw0e9N+CwansZkCwpwslEhgAT9ZU7IDWCRxL176dU0Q/uyZ09OK/m09vW45oTTJkfFGsDMD4wKKyb/fdEEhlJ5uwwWrlBCE0jR6iI5HkqnmgiGYfOF4E0rd906JkTu9bPiWFepx3TWF18MjFwVT3fHc1U6EEu1+PmcLDWY+Oxx/t0hTTg5UXROhfO6nlUFNQt6+nbQFtvcWbDlIL91YxTZoiftv8heekj6C1dBETODr2fb16SEC7tFsefNgD2zTITDhiVFLLSsziB2/rzxlILcW6VHDCht1z3RC+dW8rdTnR2etSD0ASPeLtZtnPION0zMYq4WSz5k2lQrQpNW7g5cDKUAO8LdsCCUsmm/dUTOkeyZCCUsxPnZQpvVqZnR9wLFVA9pStkyhotR2upWoDx9hzB9VOJErkCMT7T5yMgw6J7wQ8T5h1S4rpSTE3R9IznGORpTFX58T6u87kooR5j2KXeCRujbPZ66eOTuCpHws3d7D3CrK/UfxEYoMLR/P+JOqFclJK/4+TH06VKRCybI+ql0nxYJoCKXKsqOPS7f08nH5YaFU7avEE+/dVYvaVSkYQiivLpOXZm10J5SmQKUv7q9HXy/YSv/XroKVHOD6YTNp+fbD1jA+7VrXCnbrbEh2gGcFluOXx6VYcd3yvONDbPr6ffRs+/K049AJQuYsFLyH76pf3HZKPPblYvp1RYBQes12FImPpttc9Ot4vNd+v/qZGu0kDKHENje2reHb+MYbb9C0adOsaG34Q14MJd6Tin/ZIzqYb+/ZvUh1q8nMZ5tTkdyBzCMo+vYul/6wI5T4GkZADRZQu+K2ZcQtQcqiwlODqfHw7Xh+LQQOYJtTpd3TF08nH9Ot+49Tk1cDUdymLf50adOE5CVXbcO62/7N6ZZVOIjff68xQuC0zch989BPRGT3/HaZZRl9uk156vdTSl5vnpuZL+520bHqurq+H4TCQc7dCk91qIvVW/PFMF4u5+PUfqyEks8ZBEtNWbsneDn4Um3ed4ze+n2DbRcUYUYADPwOeSS56hv36UKgB1LImYopM4udPyLO11MM8o8aZIvhc8ptbuG4us+wzl5XrRA9/MWiYDf1oCETAVTjtQs8wvbj8KmbqPxVVwQDxlSwhh2hVHqgbnMMOwNQdEBJKpmHELWLgi31r+cHAq1UqVM8N33/SKgOMXeX0a8VDaFUfuX6uJy2ef0glOq9p9oCFt8+1CCEUDYqk49mbtxnDdNpbE4BYSpL03Vvz7RIOson99ahtGnSGF0H4EJSpVCOMGOB0/XrvPSbNYchVP/uXbWCW+kDrq9M9zS03+l79MtFwZ0Gr4Ry0V8HLL9SFPh63lzrwms+x3vtd3umEul4QhBKRFlD5idnzpyEmwNS+eyzz9L48eNp0qRJnsXNLySw8Z5U3IeSa/XZvWywaB47dSZEkgZWF2j6qQK/OvjXqcIjNflLk3+huuWFRluf3ZdETVlOcP2+8CAR9eLmhFJJSNgRSrSXI3N6euG6ynRL7ZQXiuqz8qszzQdOKHlKPnUu12Tj52MLrMNbMwnblapEY6HkUeQglLAm9vp+uSXjAULJLYn8hXzkxGmq1n+SdWk9alj1x45QfjZ7C73w8yrXx4PLryD6FySJF9N4ebCM0wX8JJTIN4/oclVgWUfu4Dcm2xNK9Zw4ESxOKCHrAp9ZUzEFt9ltH+P87JnS04oBKVuGXJoKKfDgf6kXjhfu/ZQ1e6h5+fyUM2uGIOnAh8XTbcuH6IDC+tSsfH6rObt0kG6EEsoCJgkmbPWr/M96f73qF/KP2Lolcls7Cih31S8W9OV2mnP8w8IJM7u5+P60TSFkCa4sePb0eQFyB5JnKn4QSiVAr9rC9vZXD9YP6UfjsvloxgZ3Qsk/Nk39xf3mOpRIiQghfqXrys/Bxzq2vHnEuHp27DDlePzwSMNgKtYXrqtEXa8uGXYadtumrNtNPyz6O0iYTQokpuvxVK9CKF1f6aleISEIJSK5EekNC2WOHDmsfNzFixene++918pmA2tlopd4E0ouQoyoVTiXq6Iv1mt3HaF2b8wIg0zPQwxfQpU3GZW5kLf+ggUZRSQ3LJNuMjH4Am5RoYDtLRu9eLtllUNRWz9NX51KfyUHJyi/pr4/rgjqDdo1psbO/UXhWP767WbpC07ouCyPneVFXRfuAoiKjJZQgpzBCR1BOCq1IKLVC+XMYuWzhaUK0kTwaVXlRaa3ya3NetSwqq/GUChnZprNfPY+nvknQW/UrXBCWf/l32nXkRMhp5hIodMCz0920plzwx7X5R8h3LcM1wAxh8WFWx1NY+VpHflxtI958cz3y2nO5sAW4+SeTcLkadQ5pqAcO2sfzkFqu7Uvtg9ekn/UwNWB+x+bnmk8byDQymKn8ILrAySfeFDXx/fUoZYVA8+eneVK3UduKXWbGzgOWbHnx5g/TJyeOaRwRBpXZBjaeuBYEFe+5Q13j09nbwnrhj7n+LtDr+zFQvnB9E0h27nz+rS0Uvjpc1D3ReXX8oNQwo8RH0aqLZDHkffXi4pQ8o9N030ELjxTDlIigtSpPNr8HFi9qxTKGRK4hONePwhhIe/y8fxgkw82Lkl9rwkN1rrjg7nB50xVNLmRmMYihNLLk3rh6iQEoQSZBGlElpzs2bNbhBLR3bBUwnKJwJtEL/EmlNyRX7cg6A/7l/P+or4/hm9xTnqyiUVoVPm/H5bTNwtStpmQ0u756yoRzyah4263JczrfXh3HWpdyZ5Q/rBoOz2VHHWML1osLM1enRqMdlURf7DWwf/Pqaixcz8iJ3HeaAnl7P9rYUUvRksouX+eGg/IAHzoIOsCiQ7k7sa/VeE+SNxHTxcuV/XV4qQHgXw4fTMNGpfiZG+HJ7dgc6F5E8lRv/24ZDs9Oco+glzVc5IF8UIo+XWSSuSh+VsCW6Uo8HtFhOjQ31IE0SN5XyAFn9qGVedNeKKx8aMMx5e90IZgWeTFSWYofdo0tPHlDsHq3ErH8yjz9vgzrRMY9TeCs+CHy4O6PuhSm9ok50+2CxRSbTv12YQfLPp2ftNqVwD+1enTpQmJ2H1z8gaL7CPz0WPNy1CHtwIfuwhEWbrtkPVv7GggYl0v+rsNaUtB/E3FRHrgs/m/ieusjCqQJ9OfBfURpc9B9PWdO83Bjn4QSvXRrdqCwP2nXZNCCCXHxInQcf96O1zg/71u91HrMFIiwvWAB96o8zDuyoVzBMXfnZ59/b2Dv+GbeepsqNyY0ho21Ve/mdxITGMRQhnJmy316yYEoYTcD3Qoq1SpEkIoEeUNWaADB1IWj9SHyNsV400ouTM7ZHNU/m30Tt+CU9pzes91wXEe4Y26iJ6GFaf166HCwd4QSKn13l21qV0V+8h8LmODrTLIVMDHT8mnrHupHWVKn87a/vVKKPlXulM+Wk4oeYYYN1ID/1PIbPBt4Ei2vG95d3aYgDW2ahGwgXHC/+jJ1mXp2R9SCCXPXMR99CIllLpUit395CkzeZCU06LiJkmkzlX31HRtN+yBM7dM1SqWixZvDRARFLhqQAblf5OiI5RwLdBdAiDrYtLpw/VMWZL4vDKNkVtoN+39J6gkALKpC8jjfC+EEvJGGLuy9iuyoPInuwXdcH9GL884nhcuucPPQeT9SzdWtfxOMZfH9WhM+PhE4fcXmYZuSA7aqF4kJy1LDhSxuz7HAf2FdWvrgXCZJR0z1V6Xj+eFbBvrhFJlrDHNQZPMmj4ep/uk1+VjBPFvVbEAleozzvq5VcX89NE9dUOwAsmENJzd2FR73B3KhCP6CIWKjXv+sQ5DFWT97qP0psHnGPcRqUVVvnOnZ18dc3t+dV1iU32TG4lpLDx3/Wu3VQ9xefIyh+NRJ95rfzz6HK82E4JQdurUyRrfl19+Sbly5aLly5dbW98333yzJTz+3XffxWv8vrUb70nFfbSgq/f+9M3BvusWObuveP3B1n0UIV58Y43CRjmJSIAafmct6lC1oO0ppv4hr69aKJQ/jU547V6W+J1bcJ0CBLj/Go9sdnspwpIBUgiJE7eXrMmCYUco4b+FQBxsUz/RqlzINhSPQuf33y4HsrouFvS5fVJkapwszhxTWOVUKkwQgz/3HQuB3ESg3SSJVAMmqR2vCxIybpw7R0Grtk5EYLGHtIm+CHqdsyZCCVkXFa2qtwNNUmQa4YVvY5uuyy20G3Yfdf1o80IokTEJbirwwVWF50+2I7mqbf4seMEKsmJ2GCPyHtHZitTz/Nr8eUA0t8o9jvoq8tju+hwHnvzAVB/C63iH8VKq96+kEj6hLZ1Qql0b0/OPXQk9ylz3S42WUMJFAC4EVZP9ooHfB3fXCSGUEGyfmkwoTWL68JMvnDuL5U/Oc7Lr2KCPLf73hxW4hgLFg6VbD9Ens/4Mg7Fd5ausKG/948zrlrfpvugGBhPWJqu/qS1OKN2E073MaT/qxHvt96OPqdVGQhDKHTt2ULNmzWj//v3W9jYiu5F2EULn06dPt/wpE73Ee1JxawMWkQ8YoQQ2XBLIzmqkp2XTdR471ilCkBUxCd5Ggj9f1EznmUgI1+NT/jT6lrypLZM/mMrqYarPrTJYIEHO/zpw3Kg9yc/HdjB027jEi91LVl/E0I6JUGKLGxHAz41ZRRg/fCo5OeCEkvvoFdWEy3Vihm30eX1SZGrUlqPbPfyleyOqWiSgB8gtGuo803ihYcr9Pu2uYfI71Pvt1D9YI5SbBDKHrN55JFgdotog5n4SSpWlxNQnk+XKThBfnc+jWO18nPm1vBBK5HTH2LlV+81ONaztXRRYpHAf9RItocR8fc3GraBlhfxWsNhHMwMkBRZekHwEAXICAVmjzh/Os+pULpTD8q10KqqvPAOLl/qmuYW2PpqxOWSrVwmMuxFKBE5hOz9/9kxBqyKuES2hxLk8Jz0sg8PvrB2CFVcz0F1GuKA9FCj0FJr6XOIuRZgjIKPfsYBMVb9t5QJUtXDKh4HTs6+OmdK08uvrwTlGQvl8GyvgzK1EkonHrS2/jsd77fern6nRTkIQSgz0+PHjVkYcCJCfO3fO2v6+6667LEvlxVDiPancHOjVlgmwsvNr06Uw9NSGkAaBbtjNw2fHBDlf1EwNjVqwNWQRRB1Y1VQQCPIz39eoJH0x968QH09TW+qFzi1E2LZ/8Uaz8Dn3GwOhhBDvxFUpmoZ2A4eMTKcPvBHKsn3H0emz562mVP9MhBKp3xCU0f+X1QSS2KNluZCMNtziwomwrjOpL55Y9Ob3DRDK1TuOBH3W3G4q/+DgPldOiwruEZc6srsGrHoQUx4yYR0hA4jy80N9N+sw6sAaobL9gEit3x3YvkOBRbpY3qxhfl9u41XHIc+Ee8ALgjJu/2CusYlF/VpR3isyhRzbvPcf26hwVOQ+Yl7uiRdCWb5AdkuWhSsD8MhXu+uott226fXBw+fXLvAJASb4KFJJEtS58BW879OFwaYQjKKC+iBps3ZXwK/Prqi+epkj/HnTnwl1TCeUKquXqX1lZUVqQ6SgxDM9vVfzoOyYfj3TzoTXfqsc6rw+SPrvyfJYustIr++WBQkhAovqGQTq+XPLXVgwR35bvZvGrwxPaQxCiaAc/cPByUJZsvevdD7wujMW+Dgj4NB0T9RvJqu/qTEhlF7fahemXsIQSgz/xIkTtH79esqUKZMVlJMhg/sXy4WBLfyq8SaUbg70PB2fXeStykqjej/wl9Uh2x4QtYV1r5PNQuoVa2xR3lTTXh/M5OOJoBRdpgbZOUYtDNWm0/ugXnTcGmPK/KLO0y2UXq1aiJC/88N5IUet7v0AACAASURBVJHPdi/Z8v3G08nkPNhOhBKENkuGdDRw7GpLj/LxFmWDVjj0l/tQ8sVf15nUX9ScUHpd0NAGl0pp/+YMWsOsgKbFGr9B4N4u8pffKzjmY7tRuWrYESa7OfbqrdWC1ltEN2/em7Idj8UYfl92/n1u89ZEKCHrYhJ+Rlt6kAF+s7MGmhbMlX8ftvXP5CTAtAADN3VPQcjurF88RGN0yK3VqGOdotapCHgxbdv/9mQTmvvnAWpQKo8xkt1OOgvR9Ca/O1wLIty5s2YMptJTfQf5h2+0ygs/4t661PXTBdZh/cPAdJ/8JJTQWb313TkhPphqzjtZKPnHvE6qY7FQ8vEiQ80bnWqGfFzBSDB5TUBvVZfV4ZJv2JpvmKztaYchD7LDxxlUE5TGJT8HW+94lvQPB7t3nZ00FW+ze/MylryVaT6r35R8k9uzygkl3gm3Jc91t/PieTzea388++532wlBKJH+8Omnn7ZybCNDDkqWLFksLcp+/fqFCVj7DYIf7cV7Urk50FcsmIPG92hsDcVOGxDZExqVzRcc7ktjVwe3qPAjIvw61i1K93ySIvsQDTbcWRpyR5CRaFwuH/VuX9FqzpS5JW+2jCFZaFAPW/Bc1sjuZQmxW2xJKUuIilY31ed+Y04+Yfq5CFi56yNvhFJlv0AbboQyU/q01hYctgvxJc8DLGBlhbUVBenZmr4aENo2EUr+Ygc5xzYYSiSEkgvSX/v2DFr5d+h2pGlRGTHrTxqgWfdMuMOKcvv7c4KBV5ESShAlFd3L/W1xLfiaId2grp3nde5iS04fA7RU7Z4DUwpTBDm0cQhm4+cs23YoGJhi10c7fLj/KJ55WGd5QNF/b65KnZICfoTzNu+3tbKq511lKuH9wJxUH0T8d7ssUqgDcgv5HT1SG36ee/85SYeOB7Q24Q7T/aslgblpk8ecX9NPQmkisGrOm54TlQiCE0q4pXC9UydC6YVsqbGahOMRtDN5TWD3RJfV6fntUhq9+G/rmG411ecU+shlwJBj+7tF24IaoLw+rKJwe9E1XU3PPj5YcmfNYJtRSrXL9Y3t3kkmq7/p2eAfSUIovb7hUq9eQhDKvn370ieffEIvvviiJRMEgomo7/79+1t5twcMGJB6iER5pXgTSrftKSw0Y7o3snpvl71kRNe6ljiyKi+PWxPii4mvU3zxPchSuUUDB7eScD9I9VIyyRrBl1AtOuqaiNY2+fnwPmFBuFNLIabnboZGJaLkC+bMYukNqhzN2HL2GhkMqw626nYeTtFmtPtqr/rCRDqanNtcOdObtrxBaCGzAUkfSMB0b1EmRIKH61AiQEZlbjERSmj9qYhRnuovEkLJRbERiQviY1rc+W/6FqLdfIEVBf58x0+dtarwqGcvfcQiqHTzEMC0g90HZBSBjp+dpI3bHEYKQ1iJedG3avkxnlFI/e7mF8l9nHlggV3fvBBu+JJCJYH3nUtNzdiwN0QT0A0HdRy+j9Cc1YueUEE/DleM7Qf/DfkZOyfHT54J3i+4cagteshb8efJ1D/g4Obu4zRH3eaWmvOmeiBqcKXg14dME8/I5EQo+TNpGhuX2MHH85Bbq4d8APIUo7oPcs9RS2n0kgChdBLhx3H0MWnQ5GBAIT468F6FBI9e4LoAFQU3Qmln/TaNE0Fz0PxFef239UYrt8nqb2qLX5evM6ou7lWB7JmtNJ6pVeK99qfWOPy4TkIQygIFCtCIESOoQ4cUrTYM7ocffrAIJdIyJnqJ96RyI5TIOvHdw4E0ZXb5lZEhoRXThxw8fk2I/BC+TpHK6rGvFscEt7KS6D5c6uVr8rszLWJYLHkmH1On8GWvvtTVccgqKWsoflMZUDB+iCtHQygnPtGEuo6YH0Jk7Ahl9QGTgtlPlKamiVAiyjxdmjQWEYIlBxp9yOutCqJA724QSF3GpWZ0nUkc58LuEElf2K+1dZ7bgsox5fPj5uGzQqR51MKk3wMlFA0SCz9Q5JY2FUTJK9xxnAepeOnjK7dUDfrdYkufR9vXK5nHyg4T7Za3iVBC1qXbyJSUhnxMyJqUP0fmkGG6+UXypAE8H7Hdg+aFUCKoBdHCXE8Qc+bOesUtyZ7f1+ym+z9L8V/0+lDbpYM0BQO6tYkPpcD8DbgocOkhuyxBvE2kWOXC7W7X059Jt7mlNHNN9eA3jZ0D7m6E9LQ8Z7wToYTvZZm+4227DOvewWTLrQok5P2AP6Py79alqp4ctZR+TCaU+Nh1knpDH1V6RHQGrjR4r5qeVSQNqFsiTxjp03HVjRFO90XNfSe3LdNHmqlNbt3XCaWKHXCSjXObP9Ecj/faH02fLtQ5CUEoIRU0Z84cqlgxsCWqyurVqy2L5dGjzo7bFwo8ft14Tyq+5Wkar0rdhWNwuDZZGXX5BizAPOMOvk5vqlnIk1C1E+bK909/SauXkim3NHQElfVKtQ05pB8Wp6SGNF0TJGbbgVCriN0WCzT/4Aupto6dolb1a8Gd4IHPFlqCwKrYEUqeB1v5PpkIJRbX83TeCiZBgMWjzUuHiFTzXN5casZEKPniFS2hhOCx0jC87b3ZYVtipvEqjUsQ4t96NqUXxqykz+b8FXar9NzY3AnfbdFHY5xQIqKb55L3EtzhNF9NhBJYmFLToR2VXYW36cUvUunAeklf6oVwQ97l2mqFQrb6ISGDAIlPutalfUdP2o7BCQ+dsKu6Xa8uQSNmhWezcWoL/q6wxKngGyc/TD/e45ESSiVxZpqDSpeVEyF929+JUCK9YVkHQslxVoGEvB+Q8JmwKhA4o/sY9vhmCY1ZusM6Nu7xxo6Bd+hjzYGTguQVOx/fL/47bAcCbcEw0aBU3hDSjN9jIZRqx8jJLcT0kWaaD04WSlNQlB9zyq2NeK/9btdPpOMJQSh79OhBe/bsoc8++4wyZkzRd7vvvvsoffr09MEHHyQSZsa+xHtSucmSIKsC/L5QJq3aZbSuQNAWqbVUQQaJYVM3Bv+GYz0Cc7gMSTTAK98//SWNiEpsgeHLWg/kMPlt8RdqJP1wcgL/4+lm1Ox/AV9ECIt7za4CMtDt80VhhPLnZTsI+YH/e3O1oOROnZcmBzPqKMFeO0KJHNQIDAIperR5GcuaqgqX21i36yi1fSMgOK/rTOI3vnjBH3XRc5FbKKFPh/uP0vG9OSHZaEyLCn5TGpfo/4QnmpCuHKDGgsWZS9jY6RTa3WdYvVUWIbst2UjmCK9rIpROljG+fa3agVoAMik5FZUdZc6m/XTHh+YIcnW+UxpUVQfyLkggYAosg3RUnw4VQz5QvOIDazPPCKXOs8u37dQuLHwZ0qUJRuXj2eTvHK990ushzza2N/WPSUV8EOD30MiFYVZ2vR2lSGEilEqjkvuv6/13IpQIRCrXz95CyQmlCiTk/Whf5apgJLa+JczzzkPu67phM22hRB/5rgl0V/GhbtIArVksF8GF5O0pKeuC6dkfPG5NiBay031UPu1OhNL0kWZqk7uLwA0GPv+qCKGM9mny77wLRihbtGgRHMWZM2do5syZVLBgQapbty6lS5fOkg/atm0bIc/3999/79+I49RSvAkl96EzDQEO3B/dU8c6NGHlTnr4i/Bta13OZ+ikdSFfonhJX1etoKWLGEvBCwtyJqaXNLbAsI2rZybBoqOkdtS19ZzNXvvkRCi5pUx3sHdqH3p1D40MJ5RqjJzE1Xt5clCvUlniTIQSfkWnz5yzJDrgD4cc7Xx7D0QH8kkofEtV15nEcb54wYIHiwaKF+ufGjcsol0alLC2S6G5CUsaLyYLJfzJQMrRf2RH0ZUD1Pl6bmzch0K5Mlv3vMoLE11vLTQ5ld4lIuP/PR3wxfSjPHdtpbBc52/cXiPE/YBfh5Nh9bsXnzJoMDYsnY9mbdwX5verjyNbxnS0amA7x3sIYfDWFQsYtSFhpX6mbQVjvmY3zEwBcjgHAUA8VatbO9b8y5vVSsO4ITlLi5421ksbpjp4N4Do/bU/NGuOmqM8aMXpGiqww/ScYFcCgU/c3Uj3I3UilDwdrNs4oSeKecj7AW3KcSsCFkrdgsdTuULuy0nqDX3kft14r4xesj0s6A7XQTpMEEqd9CM96ZAJa+m22kWtwM5ItrwVWXbyMzY9UybMeOpFIZRusyr1j18wQtm1a1fPo4V/ZaKXeBNK7kNnwoLnnh23Yic9+mU4oeQadWhDJ5TIqY2vYlOO10jwV0TIjsyYZFrgQ60yWqhreRE+NvULxAzbyarwfmDLu+VrAbFntyAD3jayDD3yxeIwC6Xpq7jh4N+DvpYqutdEKJEb+cTpc5ZEB7YvsdiqCFhcGwsMFhoUvqXKZYFUH0+eOUvl+02w/oRv1pLn21j/joRQoj62u77uVt/S3Jy7OZRQInvMzA17LZKbNWN6q33lZA9r2S//aUSDfl1NH84Iz8DBo3txHgj6I18usqSigIFb4cEcpnzBbuc7HTcRSi5TpJ9ryqBiCrTJnil9MDgLbSidz+nr9wa1GO36hXNXDGhrHbbT+UMgXvMK+cMCKKz7njcr3d+4VIikkFeMdJcCr+eZ6kEOC4RSpf3D9qcSP4+lXQRhIQDILpsTT7nodB2l9Wp6TpToOXc3wruFuwn5RSiRYKF3h4ohzyve6SoKX7eKd/9qcVCiCfrCt703x3aY6CNXnuh3TUXL75wnB1AnV7cIZV56Z+qmkPawFb5gSyCIxy2qXO8IUvoOvKGKpflrZ8U3PVOmAUHR45Z3A2OFG8ztdVMyI4mFMpYnyp9zLxih9Kf7idNKvAmlm86d0jGzFuvlO0KIiUJJfwBfnbg25MWBBQqC0161Ge3QxwvrgcalbMmMSabF1JYpatTLHVcvZ1WXv2j41iu2Ykzpx0zXAJnCS5xHseJFrb/E9PukFgK7LW9YMaDtB+y7NSkdEhClcER/Vmw/HNzW4rJAqq9+EUq0h3EhZ/KczfuNcPMo+tcmrbO2x7AQIU+z160wfNxwiSS3+wq/3L4/rrSq2ekkurVhd9xEKE0+vep8WGsQac8LX+jU77q2qiKUf6zbQ/eOCGgx2hVs66/oHyCU2DZVOo68PjBvWjZfmL8b6uADBVqwL2rR614wMikueDnPVAcST+nTpQnqhsJa9ensyPwwTe2CUMJCyfVIOWb3jpgfzIPt1HcV+GYilCpVLRet1y2sToTy+KkzVOl5d+s7+qc+gu0+APU5h8DJX5cHglV59iHTWNHHCs+ND3649elQwSKUJlF5WL1hoRz+Ryih5M8cT0LhZV4gI9Cgm6qSUzCa6Zkytc3bEELpBf3UrZMwhBKE7J133rHyeEPQvHLlyvTQQw9RsWKhuVlTFx7vV4s/oTxqFCFWPURkG7J3ILITkcxjk182fAR8CxW/Q7fvvWkpLw5YBLF1bide7BUNvLBAjuxejqYF3NR2tL5yavtItRlKKJsEcURkJTQxvRSQATjCc58tTihhNVs/qD1xAWG0q7Ts7GSDTp4+axECbDWBCPNAkL4dKtKDTUpZ3dO3VHXxeL69BkKwNEoLJa6FcXX+cC7N3mQmlHBb+P2pZla/sA2GxQdSI6MfvdqKtOYWHDtsOUH0gj/kcLxk5PHSll4HxN3NKg/fMhUVi4j1onmyhjRjCrSBv7CyzKEy8ljXKZGHpqzdHZI9xm7uK0IJl4B/kmWoeF3MGfg969Yk1EE2oqbloot8R2DPEYNsUDTYImgufdq0QUtiNH6YputifPv+ORWCL1wh1rwYcBO4/9MFwSwzTv12+vhVWXQ27kl59+IZVeL86lkxvWfwDB07eYYqe3DnwPkqm4zdO1MPNHzsy8VB6+XI+5Mc5aHQl3J9x9Ops4GdgBtqFLJSXvK5qcaAnQZsaXt5hr3OB6TzhcuKk6uH6Zkytb9gy4GgNdYLoYSA+1fztxK2x0vkCygO+F3ivfb73d94tpcQhHLlypXUpEkTypo1KyUlJVk6lAsXLqRjx47RrFmzLHKZ6CXek8pNONkkn6Njhqjm/7QsG/xZ355EpHGLivljfplguxlf3HYvx0gExaO573qmHN4PLrGBFI9Ky83tOj880pCeGGVPKNUWpT5mBAHhRWaXehG+gCAEIGQIxOC+rxnTp6V1L7azhP1NW6rcOsIJJeRYlr3gbcvb5H6AdiHibsqkAZy4iL6SnlKyVbrV2w5XLySOn4tAr+d+Clgo/S5e+sK3/EyL39zN+8MyTMGCyLU8kc4RAux2Kgx8XCB1y5MtlDUGTgrTaEVdzJmkknlDPgpVG60rBVLo2aVK9AvDzBkCIuh2qfewywB1BWTLQbkjqSh9Pd85+5WXvoHgg7DxFJzww94wKCA9B5UL4OxWnN5VyOdeu3ge4gFxXDoJVrs/B18TvIS+W3H0xGmq2n+SWxes44h+f7J1OUcXFf68P/LFomDADs8+ZLrY6oFtrX5AWsytwKjQuOyVxjnldq7dcfje/veWajR17Z5gliS9rldCycX6uYg/2jNteavflI93tGNwOi/ea388+hyvNhOCULZv356yZctGX375pZV2EQVpGJHLG/8fO3ZsvMbvW7vxnlT8pWbqNPTolC6Z3aB0fUY9gAKWJ+j5fTwz3AcuEqBAGKGpaEco0Y/3p22OpMmI6iqfHXUS7wf0JFW0NHd6d7sArEvYot16ICUIgFsolZi4PmYlPWIilIgyB6GENaBO8dyWm8DDX4RqH358Tx1qWbGAcbuILzD/njpr5RtG4WTEzYeSR08rDNCukw+a2t5GffVRAi3IUQ81CPPLtcM1ElF5tHGhCSXGNy85SEkJXvOxzd64LyxVI+7pQiYejXSODcvkowkrd4XdZx0n/lFQd9DksLSkqA+fZ5AWTqpUO/hYKpw7S1i0rts8j+Q4JHcwF5oMmWpLVuAagC1vFTzjJVmBlz5AVQCFb9tygvfwyEVByR2n9pzeVSotI1KQIhUpyoONSwZ9hN0I5ZETp6maR0KJAMEnWnknlHx80NKMNRmFwgjEC1nN/Hw/K11Iu2BRXNv0TJnuG/9wi4RQqh0kL3Mr0jrxXvsj7c+FrJ8QhPKKK66guXPnUpUqVUKwwPY3LJeHDpnFki8kcPq14z2p3DJxYBtD6ZLZ4YLtppdurBo8rEu8wIG+SdkrCTqRsRRlCbUjM35ZKez6qESC1XHejwlPNKZ2bwQWhxYV8tOUtYFcuW4Fi0uv75eFRJVuerkDle4zzjoVlpiZz7YII9EgsHBBMBFKWCVOnDlrvbyTSuSh+xuXtCLJeYG/KT4Wagz8LayLnFByf61ICCXy+iIvMC9oF1mBEDxiKlxEX32UKB1Uu0wYejuRysdAOy9W9QG7ewzXAmQrcipccWBar2ZUPG/o9pkpKw22ZXlgE7YmYf2xC5rj1+duCzzIi9fRLaD8GD6qrsiUPswXzm2eR3JcZYEq23dcmEKDageEMm1aCrqKePnw9dIHbKVfkSlDWL55PJNQKeA+hk7tOb2rvn6wPjUonTckIA7uNOqD241QHj5+mqoPjI+FEpJISvT8vbtqGVU9vOCo1wFRhwQd39aPph1+DvSEQZiHTdlIoxaardNqJ8ftWlxyC9vo2E5XxclCCSv5xpdDE6e4Xcvr8Xiv/V77kQj1EoJQQi4IVsjatWuHYLJgwQJq3bq1EEpNNsY0caAfCE1Ep4KX+eu31whW6ffTCvpibooPIV7+Tcrli3lLCi+PG2oUDqYK1PsUiWUwmodEbbGYXjSQAlHWBkQ02wWe6Nf9plt9QhpJtXWH48ivq3yk4C+H7fSSvQMEUxUVKWoilMi6ceLMOSv9JSxgWKz07Czwe4VV1BTIwAkl99fiwQluFkpISfX4JiU7D/qNdpHHWs/LrMYEp/0vHqhn/ak+ShAkMfL+elZaOi/annDU/9Kj/yqug+AJXbs0mrlhOscLocT4ZmzYZ51uWvymrttDXbVAG5xTqVCOoLVHpfmzC5rjfeOR+u3emG4MoHASdEfAWYb0aXy1NOnYqflXvt94Y+5v1EeaTKTBU8Fs11UvRL+4vKe83FdIG12VM7PlC8iLEoTnOo1O7TlZBlWebx6dzAP5oEyx2WHL+9DxU8YPQVN/8C7o2aa85y3vbp8vpEnJW/qRZhNywgNzCvPWpNTg5b6Y6nhxLVJZidyuMXvTPur84TyrWiSEUr9XbteJ5LgQyhS0EoJQYmt7x44dlt5knjx5rN7t27ePOnbsSMiiM3r06Eju7wWpG+9JtWrHYbrmLXvxWi8kDX5V2B5RBbp+X89PIZTQN8TX6bcLnbPTxAowAglmbTQHfLi1DaFkkC8nPTw99RYnVTyrBKyC87eESuPYXR+RlH1GrwghlJAEwnYkCnyPIDCs8mmrdpTfnIlQwkoH30fIqMAC1vXqkmFbVwhggh+e6WOBE0oEbSg9R+XPydMx2o0L25a6xBTybHf9dIFtlCwsu5/cW9dqEn6NsGgrYf1hUzZ4yo/evPyVNHWd2QJq6iu0TXXtUre54vW4F0KJ8SmCjQ+HsgUCW66qmNIcAiekcFTp95T7wpilf7sKjnMtUbuIZWSh4VHOvD/wI4ZPoZ/EgLePpAPvdQkYACo+N8FWFxQRwbAYqgxTXt5TXu4bfDfLFcgeJs69ckBbyzLLUxM6tedE5D6/L4mQPWzJ1oN00/DZVjM8W5AboUQ2J2TN8lLcgnLQBn/euY+ok2aql2vzOuUKXGHtUvkh7aTa5fJHdv1BPvJSV17h2l3uWhIJodTxc71QBBXivfZH0JULXjUhCCVuSMOGDenAgQNUrlw5Kwhh3bp1lDt3bpo2bRqVKhWIdE3kEu9J5ZbaDXm4f3fZvuXpGYHlM98vCyGPEEOGD6Vd/uw2lQoEv4pjuReQpjBlafDSJhaAnFkzOsqh4It4aLIl9ty58yEkDxbDa98OEHP4oEEo10uBtaLvj6GEkuenRlsgj3ruXpUf20QoEbkJH0qks4PVD4uVnnsZASOILlZ6dLyv619qTwjcQeEBAFhQIaiOcZqkQXgb+MAAGeIEHVuZ93+2wJbwcTLR58cVVqS8IpnD/9hopZJ0KwgAW7fbe0pVU/CQ2zW8Hocqwcvj1jpW158vyEghyloVU3Yq9QEHtwiQe5BLyHKpnMNOF4Swed9rKlnR2gPHrjaSe2z7KtUBWAJ3HD4RbPKeBsUty2CkqRK9YsYtSpWfn0DHTpmF5vGRirL7yEnr/369Q9CWactfJRKAG4fde4yP8bHmpalX2wpGy+CIrnWpefn8xCWhuOwRiDK22FXRt1z3/3OSar8U+OB0K+jH023Kh+1w8PM4oUQa2MlrAkFHTpqpbtfVjyOFKtwyvMqpeWnfS4IK5Wvu1h6PFFcaonb44/fU0KaM99rvhkkiHU8IQnn+/HkronvkyJGEiG/8XbVqVercuTPlzJkzkfCy7Uu8JxXXITR1wovVD1qHY7o3Cp7+1LfLQnJlIxCgZcX8lkaZqfBUYLHclGhzL2OMH99T17KqDvhltW0X4E/6Zqea1nE9ny4nlCAEkOPxUuD/hi1XLqSM1GxtXg+kQ0TfPu2aFJa7V8n7mAglgpNOnj5nbWdjmwmE8r5PF4Z0B9azhX8dCPpL8YOI5MY9Q+EBACCUn91XNygA7DQ+bMOWyJstmI4SdTcOak8PfL7Q1kIJ9wqkaUSBGwDIqMrUBBkqyFG5FfTRJIVjd55X7VJ+Po/IdeqPF0IJcsijhkHgZvduGWx2/IqdYXmzFaFUki3wdWtXpaBFdHS/Vaf+cesor8f1AHVrJXJDw4Jmyqvudm+8HOdBFDwLi34uRPiRXhQSPyiR+C279YNLOam6KpHAs98vt/XX4+1CVxIZq0yuIcqizKVqQNQVpm6EEukrkYZVL6Y0s/i4fKpN+aBPtmnsnFByWSRTYJ0bdnbHEZiJj1s/542X9z0yaZXJH2r1N/WR+ypfW60gDe1YI/hRze+h8u8VQhntTIjuvIQglHXq1KFPPvmEqlWrFt0oEuCseBNKt1zBekQph0Rl7MCiMyVZPxDH9W0hLPJYBO2ixZEHXInpxgI5voJVKrZI2lG6cJ/N3uK4/YkXzbDOtaym9fRnEN++4Z1AzmUQ7GXbD3vqAvKkD/h5FW3edyxYH1aqG5PbwkL57l21gtlqVCUlnGwilCA86N/ncwJbxrB+YKuZFxBKRDaarM88e8bhf09b+XpRYN369L4kx+wZ6hqwvCIYq9ErU4OXheWz20h7Qgkn+9c6Vrfq9/puGX23aDu1rVyA3u9Shz6cvtk1wMUT4FolL7JYerteA3+8EEo9rzz3U8V1TX6RrSrmp4/uqRsUlX6ncy3CM/Ttwm30zPfLPcOAbVdTgBR2FBRR07NKwUcVQSPcR9rzBT1U5DIvPE+0firE3SHKjvmJYjcWD5cMqYJ3WrmrsoftMKhnovfo5Z58wVXOdBOhVBZlHlmMYCc8ryhuhHLP0ROUNOj3sKGBwOofXSC2T7UpF/ZByk/mhPK+TxcEAwr9VEDAGgGLop/zBlZqZaG2u88mNxJTXT3LFD6cMH4Ufg/V7o0QykifrNjqJwShzJcvH82bN49Kly4d22gu4NnxJpRuuYKRGWPl36EO6goOtfCAOEzr1dz6GX5vStdPpbLDl3PbylfZBve4Bf4gku6MB62zUvmyhRAzr7ft18cbUeVCOUP6bjoXltR37wr4d3HfQvzN895CxHfF394IJeruPHyCYHVQBTp1Kg0YLHRIL1jhuYB0jypKlsREKJFxBlHeeHnDp/BuEEotsANkB/6mpgAZvu3II0qR5QX+X7c6pGNT/YMcEkS6672csvCte6kddft8kW1QjhIqRhsqZzJ844bfWZs+mrHZVSTc6/2OtZ7XXO1IgTnYxaqq+4HB1QCLliomv0jlBqCEyREAhWC1b+Zvpf8bvcLz8HhAED+JC5DrH5Qq+pX7SHu+oIeKPFVezYGT6ODxAGHUC+S08NGkrNFeZwjRhwAAIABJREFUdlKcLo/nCWLV0DX87/g1wXSA6hxFdPWAQ7s2TTm0VV1lUeZ+e1yYXY8c1snLniMnKIk9V/ydoGcjw24FZMRU+lRTfzmh7DpiftAlJRrrvR0e8FGvj/SrzLfew3RwrIJnxZTpiZ+k1DDcroX3IAIGeVG4cPyhvYn0sEIo3RD193hCEMpXX32Vfv31V+rVqxeVKVOGsmQJTWt2MWTLiTeh5I7hpingZPXDttOeoycJadCmP9Octuw7FrLFqSyYeEG2q3KVMcsOrukm+eHlxYF2uO9XJNNZbYt8Oe+vYBo+0/nw0/ogOfhIl+7geW+jzRWurgkypkibHaFU2p8mQomIURBK+CDCR69Lg+JhKflAdqZv2GsMYlIEG/3hEaUIWPjygfqWVJFbwdblFZnThwQPwA/NJFOk2uLC8U98s4R+WrqDlFU4UrLk1r9YjnvdCjRZjPTrmj6m+AI/evH2sFSSzcpfablBVOs/0co8o9wf3Oavfm07QonMMPDBRcE2JReih9IBxMYh0+J3qkpcb16fllQgR2br2k5KAojGPnbqTDDtH9fzjObeciLb8f05hAxFvKiPrBfGrPS0bYu5DCUFPZgObeID8dpqhYhvs3J1AjdCuevwCao/ONxCafrQgaUUhFL/IDURJ/zGVRi8BJV5xRo54JNK5ol7YKbeH8i5Vbgqh2s3TWlLTYRSuQMJoXSF1NcKCUEo00KoLLkgIEcV+FLib2TOSfQSb0JpypTCMXEiaQVzZrasa6gz45kWpG+f860zpyjMW2sXcXR0d8p/zPuqBxB4vbfKbwtfz4hQtytqqxHHdcd4yP90+mCudaoX3x6nvnFyCkL51h01wnL3Kk1ME6HEYoYc3MgcgvOxnQb9R15AdpBhQolq82Mqkwd+O3jsFNVMjiiFpfnrbvXp5uTIVKcxQGYFKdm4ALPb9rLaJkS7//l6iSUDo/xWJ6/ebflfJkLBdj7Srrm5aXghlKaPKb4IfrdwG/XStrGVlJKy4EHzE89QpFZcnSwqbPmOAOb85DUpmqod6xQhbBbAXxPR3qfPumdJieSeKV9FnKMTSrXjgWOIVkfAmLp+JIFwpv7M6d2CCuYMGBw6fTAnROcTv6mt0wG/rPIUkIQPIWw3q0A9fk1lUeYkRk/XqoKAdBxAcnYe/pcaDJ4SMgxYjkGyh03dGPL7jTUKWSoPyh3HNHb+AcN1YpHtBylP/SgwOtQtkSfEt96Pdt3agJwbMnC5FZM8l4lQLurXykpFLITSDVF/jycEoUQkt1Np2rSpv6OOQ2vxJpQ80tDUffgq7T2ash3L60B0Gzpw0Jmc9X8twvJC43cl66EHH/B2bq9T1NHR3Wvubae+Ot0a5R81asFWevYHe0KpLENoS/djQraSzh8FdMwgkWHKMuJ1eiCq+/ZkcooFHYFAeu5e6O5BJ85EKEEgsR0ImSb4IN5Vv3hYTl4sFojmNEWjY6sLixAKlygBoQRxVlInTuPBy5hn2fEydhUZi7oqp7CKrHdzzfDSvtc68GFzSicHqae0adKEpUTU2/eyINtp6SlZKNOcVHqdCMyAqwTyCXesW5SUKLVXC7mXbWLdvxnEFQoHSC3KLZlesXWrt/i51hZZRNEJJQ+4gp7moX9PB1MzOomxu10Tx7ll1JRvHrJg0P7U08p6aVuvoyzKPGWgnpQBH9GrBwbyh+vkBe/Uq/+bQihhPS2eJyu9N32TJyUEvT+cUPJMVpFmnXLCAkaHOsXzuGZdiwZPp3P4botTvSlrd4cFLpoIpZonQij9vlPO7SUEoeRdhHRQunTpLprobtX3eBPKhVsOOPrEOZE5+E4i9ZkilLq1k/s02kWUYpz4unbyrUF2j0M2vlT8HmMhAgGKtKhFzC2oQVmG0L5uJYBvobICQox8455/Iu1GsD63dmLL+o1ONcJy9yqJHROhhD/WidPnLCsS/D7vrFec7vo4QHZVeaZdeZq4cpcxeIiPk1ti4XqAvnmxUOJlfObsuTC5IydQlGYe6qgUcCpQZ/vB4yEBPlGD6+FEN4s4Mp0g7d9tLr6kXgglxvfD4nB9VhWMpLaxOclVMl31Xp5sBSXAXw9z/70/NtHRk2fooSalPGUk8SK7ohNeWJnPnD1v+UPHItNldxvsLHOoz59v7ueJY06+3h5uOc3v25LyZw9stZvyzf/c/WqqViQXqRzzXtq0q6MsytzqDsuvrtNrIjT4TX8WVD34k2JOIqVjJAX6sGoHjxNKZNx68/cNkTRlWxdrRK3iuaMSn4eqgAcXeuO1VQIIt0GY9F5N+MNwgrEIoXRD1N/jCUMo4Uf55ptv0s6dO60RlixZkp599ll68MEH/R1xnFqLN6GErxB8huyKk/+ikhRRUie6tdPr1i93SDf1g2+dO8GsLzJeb4kSLXaTXYFFBz6EKNsOHKfGQ1IimKEtpwJfog0OUv0FYbnjw8D2OQjl651qhOXuVf6cJkKJ7TNYKCHThKAP/H1nsvVUXQOEElu2ekYQHOcpELlECbYcYZ3Tg3KQvrBLgxKEr/wHP19EDzQuSb3bV7RkuvQMP073hFsoH/hsgbXVCsvN4JurhUXVe7230dRz+4CB9TBD+rSuxNoLobTLQa0Cb0bO2WKlhlT+yBgPNCS/6daA7FIneokuRzteMjrp7ijYoocrA+YO/r1+91FrDvHdiGgwV+cs79+GcmQOSFbpFkouZ6STfq/vGru+8a12U755pQQxZMLamNNO4iMP7hDQclUpUU3uIHaEct7m/cEdDIyHWxhNuLndD5VWEvU4mfaqZuDWPo5jfkBOzaR763Y+d3Vwq6sfV4QSKWSnrdtr5bxXkmi8rsmlxoS/So8qhDLSOxFb/YQglK+88goNHDiQHn/8cWrQoAGdO3eOZs2aRcOHD6fXX3+dunXrFtsoU+HseBNK/eUUyZBUwA58Kef0bkm6tdPrNhQX9TVdny8kTv1zsyzZnaukIEwBEPwcOP6PeqiB9ZMegAShceXjpyy3kWDJ6/LtcxBKiKkr6R5VT/lzmgglAidOnjlnbS/Bl6tzUrHgdrw6H1GtPy/dYRQBr1UsF41+9GqrKtwdVNYe+MwhDaLyFVVtIUilU3LuW6RqzJYpfXA4bika+bihmfdMuwrWT2pR54E65fqNd43qjBZzfh4nb6b2kH8d2/9Ofmk4D6TdTYzdzt0D29r9rq1oBU29OHY1cXcOK7jhoQbU6JUpwdSDvJ8QpEbqTaSvdNq613OCm8aqW87g04rI2vErdxGsq/ioRGSx12fU7f6sGdiOsmRMZ1XT5w788JAuFEUnGbHuCijfOLRtSg8KFYdaxXKTH4RSYfDunbWCGqPAdczS0BS3doRSxyVWQrlhUHvKkC4Qb3DnR3ODgXoI/Ht/2ma3W+bpONaI6kVy0YRVuzzV55Vica1QluXuXy22gkJBaiHLphdowSJLEC8m/JVQuhDKiG9jTCckBKEsWrQovfzyy9SlS5eQwUCbcvDgwbRhgz/m/JiQcjk53oSSa6FFOg5lFcBiMrdPSysykls7sfDp0ZImSR2ex9bUB6/WD6/R4Po11JaPW+o6brnDlnaroSk+uu93qR20NkSKo14fQR/KoghL1dCO1cOioy2R4PuSLLx1P0iQABBKLFCIIu6UVDSYp1ZdC4QSBHrT3hT9S3WMv3S5rygI5Wddk8LIKfp3c60ixmFHQigfaVaaYNVDue292ZZ0i4pmx29V+0+koyfOxAqv6/luPrCIws+cIZ0x4II37oVQgvw7pftU7XEypeZhs1enhqTsVHWVj56bdqeXyGjdtw++uydPn7UyW4EMl8iXzQrcgIwPl75yBdmmghKOxmF97kAcW81XPcIcsjQ8OUCk1+e+m1w6R7XzwyMNrQxY0HmEyL4fBfqhj3212GrKpMWrLIe6DiI+rEzER/0WyTOHc1Secvz7jg/m0pzNgfS1kB/zK1Ui1ogqhXMGs/BEgp8XH3ooUMDNRy/QB4Zhw40AmjJSmQglAubK5c8eEr2vE/pIxuZUN95rv1/9TI12EoJQXnHFFbR48WIr7SIvIJLVq1en48cDX7uJXOI9qWZv2hdGNrzioZz/IR80v28rSyibW69MsiTICDFr0/4QH8MHG5d0zA3MF1OnvkXra6NeCPALe/zrJbaX4JY7bPWpbDY4wZS72iuOej1kz+nycSAqG4TytduqByOteV1EL+JFihSKvGCbElve+CJHlCesh7pVEQ73EA6HD6xe4Bv3c3LmI655h+hf5NrWI8aVDIppvJEsbpw8Xj9sppVGE35cT7YOPL9OQtfRYm06D0QL2UXsXEEQBQ8tuvZvznC8rBdC6eY/rC7AyRSIDQhOi9f+MObcVlHEblHfpg8+fUB69DGIDwhlwB2hmCVeD5KFIBk7zchI7g1fnPW547StHa1kmOob993k2WLUcXxE1CmRhwaPW+PJP9XLmJEVSr1vdIF7nK8C7zgOPIuWukasFkqlrYj2eIS7286RlzGqOlgjENT0x7q9kZxm1XWbW/gI3X3khJUZTC8qlakboZywchc9/MWikNNNhBJb6OWvyh4iFC+EMuJbGvEJCUEo77nnHsqePTsNGzYsZAADBgywUjF+9913EQ8stU+IO6HcuC/M4uR1jMopH9tx8EHSyakuOYJ28ZJC4Q+/29ZKrD6JbuNRLwRTVhJ+rkoxiewcX8z9y9rqUwUR15C68aMgbeG9yULkIJRw4q+VLN2jtw9/IJUtRB2DPxZSL8JfCf+GJUlFjas60KYbtWBbMAqft4sAh7H/aWz9hBe1EidHYAi29vWsO7DOQrjeVCIilE1KUatKBaz0gYrogpQ92qyM1XRqWShV0IudDyi2P7Et3jo5PabdPYcVWBea1uvqhM2uLW41VRbk1kOnGTNDqcw5H8/80zE3fVKJPDR/S6jeon593b8Z8l+I3p+6bi/hGD72kK/cixXJNDZ9O9OJUDoFASkJs2ifP55uFFufPB0m2oQ/YfcWZei1SescP34juf4bt9egJ0YttU6BvJfKoc3bwO4J90M27YTESihX9G9D2ZP9Vm9/f05QSszNtz2SsWKNwByGC0ekxc36PeimKrRpzzFjnnDlquBOKHfSw18ErMWqqJ0rfi58afEhz3U9hVBGekcjr58QhLJnz56Wv2TFihUJEkEZMmSgBQsW0IwZM+iGG26gXLlyBUeGbfBELPEmlLM27gsL2PCKg8pZjQd+Yb9WxDM/oA09Cwh+Q0Qq/Lo4oYTv3PA/7LeRok2p6HUc6oUwbsVOevTL0JcKb0MRrZuHz6LFmlUQVqEe3wQWh1jLiHvrBkkbrMAf3l2HGjKZEN6+KYAEVklsecPPDdbKjnWKhlnbnmxVjr6a/5cxdRlemNBvQ+EiyrAAIw2i7muEgKTm5fMbhx0JoTRZqp+7tpI1Z1CccjvHijk/X8ny4DdT/7GogMi3eM1ZlsyL7IrXRZtb59SHTbs3pluBHXp5767aViKBEbP+dMxNj61zuBU4FZ5jGvVgSYOg+IwN+wjHiuTOaqXEjNZ/GZJdXKTbiVBih0B/7lTfo5UMU+fzYCAlv6TjgtSjx075p12MnYenvltmXQYZrUDS9QIXAC6ObnrP6IRm4qpdEbnfLHu+DeXMGgiE4qLuXt0xvDx7CKyEW4LbfDO15ZZiETgist20Pa9cFdwI5fgVO4P+rKoPpjSLsFTj/chl3IRQepkBsdVJCELZvHkgHaCXMnVqSsSul/qpVSfehJJna4h0TCotG14WC/u1ppkb9oXI05i2TEAaIEbMCeXjLcrQW1NCBXl5X2KN4HQbl3ohTFgZ/pXKz61UMAeN6X61MS8u/Ah7fhtYHGItPMAHbTlFr5u2g+A3CWHziat2E/wpb6tTNEziBoTy8zlbaL9BZgl4T3iiCa3Yfpjen74pmOEIhBJb+/qXPHw+ry6TzzjsSAilyWcL1gfIHqFUfn6Crwu63X1CXmjIQKGY+g+/LEjY8Ch/U1uwAg/9bb3jdNAJm11lLouDDCz3NSpJ17w1wxilj/kDS++ns/6k/r+str2+nlbRVFF/hqEuAEIJS1PXq0tY0bsv/bom6ikPMlN9YCBXPIoToXTqb7SSYeq6SukBfz/yxSLrYyzeZcit1YK51+2yFm0c1D5Eesv0njERGlOkut14uP+o8l1GXbeEE5HgA9H1Inmy0rJtoe45XtpwS1gB14GVfx+mD6aHBxCpJA1uhNJkTFAfGfxcKHBULpwjRHVDCKWXuxhbnYQglLENITHOjjehnL5+b5hPnNeRK6d+vMzxUtLbMuWChUYe/Ps+m/NX8DJuuZG9CjV77bdeT70Q3L7sQbQgMt72jelhl+KLQ7T9UOdFEuBjIpSI7IaFEtt2sDJgYdClfoD5iFlbwrbL0QdYhH/r2TSMTCEQYtgdKYEEqr/I7IMsGKYSCaE0BWcpzT60XfG5CcGUgLFi7HQ+F7A39R+Ro7CI6dlK9Dbd5jXqe/VTg1XytY41aMXfh+i6aoUofbq0pPxM9evCwv3/7F0HtBTFtj0iAoIkQXLOGclIzgJmxRwx6zOiooAgooCKYgIjKj7MAUQlSs6Sc04CkiQjOfjX7rln7umaqu7qmWm4vn9rrbee3KnU1dVdu0/Yu3mFfM4HQ4/hy4yXagMoYR2G65yL5JPFB0ChXOdTr1/NoNXvPuHQlmpK8QLKeCnDeH4yjpBJ9f3mnujvr15XNSqkYKJwQgZ22W6pSTjI4FeVkxIFlJIySbJGlM+fXcsCEc914z1VIOf5gTkyMZZfUiay5Rdu2afNSOd3kx+gBA0WJ0jx9bHlVrZFfDsSS6WEbDqgjGdHBGuTDiiDrZexdtiAUqdhajt1JkbGy2JBjzak9qVa2tAv5MiOHD/pApRPtS5HbwhLjhpXhcN00Zb9ttMKXI9fCDrqCNkZAGW/DtXpigHTYsaw1Xe2mZykE/GrrwOUCDWAhZITJwAoVf1tJLsAKIAMWS1IABn/VDOtdU7ncuPA90QBJSxeALmyIDYVyQkoFbqP0mZy+q1R0N9B1fTJXXWcZjpAicD8fDkyU93esXrKciw/yzvq+jEccH81iuWiYSlUTvy3qwdOd9Sp1ALrKqyszGFpun647rxIsH95pBH9vOhPY8wgYp+RvfuihxXUb+0Rv1e1p52F0stFH6/Lnecn6YqYYsZv7on+Dut7t2FLnW5M8ayrXm5L5Z8fHR1K955JFFDO7tqS8qXop1/z3vSYJL9ErxPtEZqD0Kh4BB9Ykc00D5wz8zbtpfc1YVOg10LymR+ghMyrGgO/oHtryp0tk6stPtYQy1vr5XHR6aQDymTsEO8+0gFlktY4bECp0zC1nTq7afCyWNijDal9Qa6s/TvuTFiQV4P65b/CQqkmL6ixSl6xU7Zz9arHLwSdWoJsh6Dy1zpUJxzkapGHQ6JzknQifn3pACXi3AAoEZN1a71idF2tIjEk3I+1LEsfTl7nWDLVgiSoCU/rAaXO5caydLq5BrFQ6qx1iB+FbCfKmeKhRILEoDtrO2OaACWsJqxxbrpH2OsDJ3pTzNhSs8Ca+MNDDVxD6ThIUQE8piBwRuLY8z9FAEvQwvyfXsowiH3OnyMLvfCz2QrqNy5czVVeGKM9nNW19wKUiZBfY3BJnYPMazA+hF1evLJydO1MWuSYl0wAUd8zz19Wke5tXCpmqjq1H9P13F6/OCHhDqEU+FjWiR0kuhawIOfKminKIxqkP78MftCnzd6wW/ussYSpH6DUMXyAmzR31kyuGNaPbq/lcFnW7ZP6MZkOKIPczfjqpgPK+NYtplXogHLlzpisXdups/sLCQrIkpR6qCA9xgtKxkehX8jrQUZxyKxUl3eXdhWo76iV0WFVYmmb5AHbOevq8QvBD1yDPBmubZ30YK+rKnu6F4PMT9KJ+LXTxY4hzg1AcTLCGS4p7iiaqPrbsJ4hEeqkRtMMVDCTn2muBVM61/5vTzahsvmza6eqynF6XY8unpCtbWhXrtsoR6Ul7MKyhxhHBygBoHHIScuabk6wxvtxFt7fpJQ29kvtT5Lq828ygULW50OUZRvjWS/sj05tyjsckzrLDz/LcP17udX9xl724qXGBIcggDJeyjCeHyyBmTNGCNWf+GYB/aSQjPtdRzy/Awxy/KnJWgxXfKUeqYBbTVCc2aUFFcx5fszwkqDcdm74kH1z3Oq4rIh+Y4AFIFumjLT9wFG/qjG/+wlF4AMK3JnvauLwEfN4Sek8vhZKHQcx5DhznZ/J+ZDlAu/RxcVyucJd0gFl4FsauMFZA5Q///wztWnThrJkieiy/ttL2IDSzyrntX5wDY5fudOhDFnS81Ji+SrE2m3oexlBNUVmw6EvWMb2HDpGX8zaFO1avljxRzUeykbRI5H7zC8EADCoZJgKVEFeu66aVvsc4DmRWDI5pqQT8bsuHaCEhQ0WSmTiwtJ0dY3CMVZVAHsASlZSgYUHkn1I4gBYmtq5hRZM6Vxu459qSqUvukA7VZBNN399kt9lOL/DUiI/NPA3dlnhv8t0HRkFwCqxtdUAlpUgjff+bbWc2jpAiQz4EnmyUcUeqa5IXdc2YDGI7vZX90VkP7lIEmr5d05E+Or3TdR12BLLq3ZXY8vX62NW0YCJ+oQ5hE3kzZ6ZusdpBcWIKmDyiqEM88NSqsV0+nYhDV3wZ1zrFqSRlOY0WeHwoa6qZMkxkCVfIGfsWXfLx7NoxrpgFD2g6Ppm9ua4rIh+1w0jAYQndEmAfm1L5MmqJfDndoiTxLvuHY3uOKRiG5TOGxeg/L1rS+dsk4AeH/ugypJeqnRA6XcHE//9rAFKkJmvXLmSihQpQqVKlXJogvLkyZP4FZ2lHsIGlDoNU9tLhSUMihl4WSx58VJitQFYC9b3vUyrv4xEBShqSECpgjGVCgc0LtPW7rKdVuB6/ELwy3iHK/iV66ppCa9VUBx4EqKBpBOJpx8A/aMnT0Uzca++uHCMTCAA5cCJa4kNlACU3S+v6OhGo0htcjmHl6+uEuNGnQT3eN5s2qmqEpVe16Oj0GGlC7Qr1WVEdL6JWqS85gHyblhrUHSAcswTTRzJQZksoetPTWjR1bGxYqIdwksgeymLya3JutPfzN5Ezw2ND1AiYeTGOsWo/9hVRgYGZLHnuSBTNA4wnr2qunS9AKUNb2Y8c0AbqWf91HeL6Mf5W+LtyrqdpJUyZamDjq22iNdTO5fxj/I308eG1+TwgfDNnE1aKjHrizJURBhThnPOoYOamG2/vsEisuvv48Zq2O9TVu9yrKtq4fAP+RzLe831f1rwZ5QTlP8G6y9kZGXSGEJ+QPX0zz+pI6UDSr87mPjvZw1Q5suXjx544AFq1aoVgTZo2LBhlDt3bu0VNWnSJPErDbmHsAGlXyKK1+XBkgN6jQsyZyTEQnGWNBRV1vZp71i/Sncd6eoChxDk/CSglLFEqKzGBcrM0jCWm18Ifpyc+FLue201uvnjWTHTUN32icwz0YxxZCkfPnbKIa2GBezyaoViEolA1Dxw0troixGAsscVleKKuZvauTkVvTCr9pI37T5MTfrZUXIh3vPL31Mt1+gQUmcVCuRw+n7zt9X09vg1hCD9bfuPeupUJ7L+rFCCPnSAEi5+hD9IwmndeDYZ3H4crNyvzDznv+k0p/EbkmmqFslJ387ZFM0iDroe+KhB7C2vua49Yp8RYxavFRR9phVAKSUfn/l+kaMiFXYBgMN+RjHFgALUeLEJwC2bL3ushVIq3theBz5usGeSoXikjomkKZwHupht2/mZ6iFJbuLKna7ETq77xT31qHrRnK7wFIQB9bqqiqs7yNCqtG/Tn2tBWc871xUrDQ9V5x8Xu9qmA8pE76B/+7MGKF999VXq2rUrQeXinHPOcf5fV/DbqVPJI6n1X5L4aoQNKP2ocrxmzfqz+Ppc1qstMY8jXo6re7dz1l49dHEIbd13xAUcXrq6isttBs4y6RqBWsyElTvjW0CLVvxC8JOhhKsDLt9bBv0e06uNKorFVJwqiWaMI8N3/+HjTmY8YuHaVC4QozsN3WwZGwd3VM8rKscFDqTShnqNm/cc9uVr5DY6GcIJTzWlUinu9BOnTjvSbch4rt9nvDb+03aNveqBGP6tm2o4VXSAclwnAMrs2t9kvzoXvjougL3JpSzrysxz/vvdg+donwtOkvpu7uYoz2HQdWE9cLgRTVya2PN4VuO1gmJOiF38ZdE2gmsdFFGNyqbymaprH6aFklVRMKdnf1hM387dHHTJAte3+ZiY8kxzzw8yWDCRPa0WqXhjOzGwLHw3Z3MoXK+QiAWYNBzHnlNEwhCH5ugqQo4ShhGdKhXec9C0V2PFVRD447wtUZJ5HgMfypnPy+Bic9AlX8q9Y7vWNvXCPvtt5pBW6pw1QIkF2L9/P+3bt49KlixJs2fPposuuki7LsWLRwiTdWXPnj0OMP3111/pwIEDVK1aNXrllVeoUaNGTvWFCxfS448/TnPnznVc6o899hg9/fTT0a5Onz5NkHgcNGgQ7d2712kH1Z4yZSIycrYl7E2l0zC1ndtVFxei4Qu3Emh+VrzUlpgcFuAEKgMo6qGAOJ0te48Q4ru49LmmqgvIqFJbyPJVpdBs52hTj18uqha52hZxTpgr62zL31XqI5txTXXU9QjaF9yjfx085qio4NCHhffyd91UR2rsHu7ZS1dVDmzRgmX2gaaljVP8c98RamhQ+VEb3Vy3KH09232Q40AtlifW+hlmgs61NQpT/xsv1u5f/JFjRv0y2G1kFRF6oEsmUNcGzwAy3mW59/O5Wrk+HLDl8men7+dujuEstN1LrAc+YMIaen2snpwdex70Sc/+GJ9bHXNhNRLdvM4koJQAo8vQxTH70HbdgtSzibEd16kptepvVmRCJnIeDaA0JWx5zQ/7FYBSl6gX5Lp0deG1iqdfeMGQLHn0hDmt+6qMAAAgAElEQVQZDx+do5dtp9dGp0rh+s1XBZQ/zNviSL7KgnfPeRnPcVmIVW8a6utc6H7j2/we9tlvM4e0UuesAkpehMmTJ1PDhg0pY8aMgdcFiT07d+50dMABSAcOHOiAw/nz5zv/rlChAl199dUEecdZs2bRww8/7ADGjh07OmMBTOLfn332GRUuXJg6d+5M69evp2XLllGmTJms5xP2prpq4PS41AtwATh4EbyOr8+VL7Uj1sLmf6OOeig8166Co9P89exUQCkJftEmX/bMtPPgsega6SQcrRfQoiK/XGZv2KONj+QuQBWDL1TW2ZZd25BYW0zFqaJabG3bcT2QJEODe/2uQ4TYTmQ5XvaOG1CqdDUAlIiP7PyD253jNzYOPLh/TWXb/iO+BODcFprjqmXI5E4Pk0IIvJ2wlun2L/7GVlM/QGkjXYckNV0ygbqeMlGIfzNJBDLg1Vld/O4n/86A8r1Ja40HNVy2hXOfH3jPyDnIZBh1bjGAsuSFhGfUpmTOGLGI2RYJMODClx+8tn0ErWcTEoEEsHZvu6nX5DjMlaiOfcMHM3112tU2/D4Peh1h1of1D7RSXpKXeEeMWLKNXhFMIX5zgjs71/nnOTGSKLqPL8SGwzoqFbEgBfuSQuTvtYf95uH1e9hnfyJzO9Nt0wSgxEXDwti7d29avHixo+VdqVIleuaZZ+iaa64xrsnatWupbNmyNH36dGrQIML9BvdtuXLl6Oabb6bzzz/fAZgbN26MglVYM4cOHeokBB0/fpzy5s1Lr732Gj344INOe1hMCxUqRNAMv+mmm6zvR9ibynQo+rkZcAEszYWX96qX2xGTwyJeZnmvts41qv0jk3jDLjegVGMGQZYs6SVAQRRmkDwfJvP+2EPXvT/TeG8gAdb7mqpamiVbS5PNjdd9Bdu04zpwDcIyiP+Bzqh28Qtj+EDVwwyAsu81VWPcPn7jSpe0ri6AbT3B2ebXn/r7jOdaOGosain//KhAgCHIuJCrBN+obv/ib5yE5Acor69VxDcWDxKYumQCdb4yUYh/Mym68Px0cWG268CE8qA9Mh3UcNcjGUu17NiOgXpe1p1EAKVfIoc6Rwkon/9piSvGO8j1BKlrExIBVaYrB8Ty3vI4C3u0dvgd1SIlFG3nBHaIcSt22FY/I/VwXyr1GE2HPTTUkY0N2p8+I1Op52wmx4IcqKsLD8G7DaFxkqVCFysvOUxtxrWtE/bZbzuPtFAvTQDK4cOH07XXXuuARyTgIGZyypQp9Msvvzjg78orr9SuFcDfjBkzqGXLlpQ5c2p8CtzVbdu2dSyNuXLloq+++irafty4cdS6dWvasWOHAzTr1atHq1atckAoF7i94TqH5dK2hL2pTIfieeee42huexW2wLCLm7m8OElHdyDDYrbur0MuCyWsQfJQUt0jNi9e2/XU1ePDxI8zEUAXVrx7/zs3phs1JjGR+SRKQQQi+E17DjuZkbD+ViuSK8bKgcB0SS6Pe4i6T34bTI988jPNqHgefYY31gAJWH6KMl5rZaJFCVOGEa53JF/p9i/+xtfsByhtLD62oRLQZwdliSxQ9sBHnFrYqqvLXLXdlwNuqeEkc4H8XnLEyvbY82UuuiDwR4jsQybDqHNLBFD6Uc2oY0lA+cLwpS4lL9s1Q70grl1diIc61o8PNYhRuZJ1QCsEHmC1dHh/Bs39Y2+QqRNL6QZqFHJl3Be/Zx1u/2EL/oxLU57jH3UJbIiVRmnVP1VqVxcrL1WWkrkcYZ/9yZxr2H2lCUBZs2ZNB0x2797ddb29evVyLJeIr7Qt33//Pd1www0Enstu3bpRu3btCAlAXJYvX06VK1d2aIo2bdpE1113HR0+fNixZnJBe/wNY5sK4jXxPy7btm2junXr0ubNmx0qpGQWXdIM9w+AcdzHZcQxYgCfa3q3Jz7AmEYIfamHAsDSmp0HXTFKOvUVeZ02sUaJrAsfJpCx06ngcN/5c2Sml66qQvcPmRcznC2foM08E6UggjQY+B+hSAROS5Amq/rjakY1rMywFD/+zUKbKUbreHFQohIoorxoT/wGM9Gi+Fkt/Pr1+h37GnGsuv2LvzFg8wOUHGPsNZZtMhfI6d9Mievk/kwE3Ex2nQigfO/WmtS+akFPQInnsmLB7IE/QuR6eCU0JAIoqxTOQUv/TH2P+u0HCSh7/ryMBs9wS4D6teffEe7jFe8n+7HxvHxzf3266aNYVgnuB1roObL4A0ob4YVKBXPQ8m32a2a7JonUw33xk1zFGvwwd0tcPMBsIddRbCEW+fQ//1Dbt1JDDnShTSDnZ9d5Iteqtk0HlKkrkiYAJcDckiVLYhJh1qxZQ9WrV3fAnU2B6xuWSVgsf/rpJ6e/W265hQBMucBqWbp0aZo6dapjobz99tsdi2iGDBmide644w7aunUrwZppKj179nTiL9USBqA8ffofl6yUHNPmxchWLqYJYhcbiMkX97xUeyD3vKISrdrhBpR+RN6SXsPmfgWpg7hIxNOgLN6yz9O9BFUQJK48+MX8mCFsJfRs5pYoBREOhrV//e18EEDZoXS+C6jNm6lf2ZiDah0BoISlWNWz9Zuv38t099/HXLq3fv2pv5uyWCv3GB1KNirGh0UccawoOtBoCygvr1aQfl28zfOSbQGljOvkDjt9t5CGzo8l4GYqmUQAJfZNu6oF6aMp64yuROiQ4+PliW+DfYTIBfGiXFHXPogFLagYgpxHr1+W06fTNwTdqk59EGHjQ86mwOrsJ/GoMl6o/ZoYFlRZTpu4bD+JQ5trSnYd3Be/8BZYCGFhhChD0MJJYYjp76JwtoKy7OSpf1wJjTpWBi+Wi6DzkfXTAWXqaqQJQAmA9+6771L79u1d93XEiBF03333OeDOr8BtDvBYv359x7IIkFq1alWnT52FEkk7AJcdOnTQWiiPHTtG6NNUzqSFUscTyfNS9bRVsnHU4zg8xFviS4+TAFjbW3cg40t57c6/Xe5WJAB4WcaQyKOL40LGeJDMPnXNYbkDPQrzHC79c39MNrRsg7gs8Jc9/GUsoATlxmfT47NqqPOSChp++1P3u5Rn++TO2o6yQ2sFUIJ6Z8GmfdHmAJSwgOmuTTcGDjpoS5c0EJpzm72HjvtqXntd4/zurQmkz2pBoP7fcZAk26wnJCBfTOGp0wFKfIDgQ8TPQtm+agEauWS755C29xoJS692iLjhuXT+YRF9NzeWL5HXLBFA+cFttahtlQKegBLPP/ZRUKu2vIawAGXQeEA5j5d/XU6DpsUHKE0E5bpNwMIQNnvSVMf0QXfte9Npvni+dYIEap+6d3wic0tGW9wXvwS8tb3b0VezN8UlAXpf45L0ZOty9NOCrTGUaaDfgtSr9FrphAhMYQeJXn86oExdwTQBKOHqHjJkCL3//vtOtjfKtGnTnIzsq666it5++23Pe44Mb1ADIQ7ziy++iMZTAkyCKgh9c9HFUCK5B6CWC2IoYRlFQo9tCXNTgdfPpPah6mlDJQZZw7KwEggr43CmHF6qONRQ1EMXL7Y2lfO74uoQr/XIVwuMS2Jy1yRiyUMm+exurVxjLtu6PyYbWlYAiOp5ZWWtFU+NSbS9v7p6tlYr0xgg/gY1EwqIfQvkzOyKA9K1g0UalmKd9VVXHxagb+6/xPcy9x8+EaPn7ttIVFjUow3lzBrr0qvac4y1JSjIeKiLj4MXrqis3b/4IycK+QFKG8Bgu4fB0dn32ogbnouJ3obXLBFA+eHttejSyt6AEspGdUvmoce+Nj+7fmtvCyjxMQd5z98ts7zBJRpEj1vOo8/IFVb66rprgxcDlF02JRmCDZCuzJoplsXkmvemuz4YdfyJNnM8m3VaVcxHg+6sQ34UYQib+OL3TXFLgD7QtJTz0d1t2FLX5YIwHRK2MlFT54kyJUYlunZhnv2Jzu1Mt08TgPLo0aNORjXiHpGthYK4QQDC7777jrJm1at7oB5AKIAn+CXffPNNl+safJQffPABrVu3js4991yn3y5dujiqPMjyhhUSij39+/ene+65x/k9LWZ542Ep/7xejxiB3vuPnIjuG1D3QJrwgHDnyNhGvJDBYQYVAQCveQZAidg0xKg99+Ni+mZOhHMQ8VpeljGTckwisYZIsJnVtaXruVix7YAnRQeyAgEodRYZncpLvA8d1IRMZNI2fUqlIejcQs3Ei8sOfQJQvnNTDW18qG5MZJJ/96A/oDxw9IRLusxm/rKOKUasWs8xrr0YtF+v+vhQAj0Iig40cqKQH6DEgThuhTchf7f2Fan3yBW+05dueK7cbdiSGGUh/AbVKiTGcZKcb+eaCuC8dLgvp6w3zg9hEw3L5PX8GPQb2xZQInP92R8XWwNKZqDwG59/l/PoO2oFfTh5vVVT1XsCJoit+49atQ3iwjd1aMowhlUNMeFcEuW2tbqgJFdiS3vZbiM9E0Rx7778/Y+4JUCrF83lMJaomvSf3VWHzs90riuGVUf1ZPKiJLoc6YAydQXTBKDk6QDkIZYSYBLu6ooVK3re69WrV1OVKlXo8ssvj8nIhssbgBE8lMgSB78kknseeughB2TeeeedTt9I3Pnwww8dmqASJUo4VEWIrcQ80goP5dETp6hCdz2gVOUPkdENyxlczMwVKDOb8VBzphyIyRH7pjuQoQJzU91iJKk5PritpqP4AfedLvbJZMF84YpK9GIccTOYl4yd5M2wesfBmFhDuVEAsnteWUmbhGCTsYm+8F3jpxaRaMyoTKgC7QjARYs3zOTImBcA5YCba2oz2HUPC1ydwx6OWP29CtzScE/HWxgcqe2rvzjW9cETb/+6dvhQ6to+8o7QgUbQlOTPkcXX5d28/EU0cdVfnlOz/SiSbnjusMfwpa7Qkej77qW2lOW8cxMClAiVaFnRG1CCFql5hXzWYRK6hbAFlKhnq/4C2i14G3ThAKabIefx6uiVLhUprxuo3j9YusCwYFMuLprLBfps2qh1TIBS5ReGdVuNEYxnPG4DKVCoVoVZ+J6U6TrSkxQd9XQxkLZzw8fxFdULUvfhy2KawMNz2yepymhIjkNGuSymOG/b8U310gFl6sqkKUAZ9Mb26dPHAYS6AsA4ePBgJ5sb1ssFCxZQwYIF6amnnqJHHnkk2gQJOeCmBLH5kSNHHNoiuLoBLoOUMDfVkeOnqGIPPaBUedzY5YZYRnDToUjuRbgdYHHES0u6k9UDGVqoN9QpSvIwhHsN7p+Z63dTnRIXxgCQT++qTXcPjqXqQYKM7iVgs75wC097NpKMw2XtzoOermEE3EOe8ClFUQHtbTgHUa9W8dw0z4fOA3KJ70xYa3MZvnXGPNGEADAll5quEdSOYCnuOHiOb5+oYOvyPnz8JFXqET+gNFFyXNxrLO07nGpBt5q0ZSW4wLq0MwNKTnrxs1BCAnPKau9DV0eUrJumdMPz7y/+skwbt8tEy4lYKPHMtajgDyghi/qQJqbYcqkpCKC00afGe+Tzu+u6PCA2c5Hz6DdmJQ2cGHnH+RWV4qv0RdkcWjSbghhueEUSKSaloSsHTKPFW/Y7XYM8v2DOLEkFlGGLTWDefE9Kdx3pKb0ojRnxrGXDMnmc8I4eGkCpO3tUBhSTnno8c5Ftwjz7E53bmW7/rwaUZ3qxvMYLc1MdOnaSKhusR2osEOKlXr66qpMcw4BSWtLAJ/f1nE2O20G6k9VDF5nEcC9Irjd2r/E6qG2+vq8+3fxxLHVGIm4cWBKmdG7uWvp1f/1NLT0seYgr7X5FJa0yiA3nIAZDEHiNYrk9rTr/aV7a+kDz26sTn25GCPZo9vokz6oAlO/fVlOrAqRriPjAjxQpQF09Lyu439zxu8kCU/Ol32jPoeM2XQSuA43lzm0rOO3eGreaBkxY67KQzOnWivB8+AHKRmXy0rS1uzzHt7WySzc8d2hKHmFux0RiKD/rWIeal89ndHnD6j3mySa0ZMt+evCLWBotnqNfOEuyASUsZ4M71qWg8olyHm+MXWUlh4lrVO9fhQLZHclTm6KLS7dpJ+uYVFokoIRLduyy7QlprqvzsslQD3otXB+sIdg3bSoXcP5UqssIOm2gRGbC/0RkRuFJaFY+H73wc6yF0m//Yn4mrtx4r5/bhXn2Jzq3M90+HVAmacXD3FRe7khwLu44kBpcjqQTZDhLQClj/ZDljUw7xKHga3hml0h8onrognPy2ppFSHK9sXuNl0xtM/ThBnTtezNiVhTWTsRsxlNAfDzpGTegBHejlyUPme89rqik1S624RzEPAHCkVVYp/c4Y/A+LGS2MVx+144EEiRfNe3nDyhhKb7jUztuVh2NjW4uXnG6fnPH7yYLTK2XfqPdIQFKUIM8fWn56PRUSz7rJ/sBShvqGtBo2dCd4EOk22WRuE4ufUeuoA+nuGP9EFKxoe9lTpVEAOXgjnWcQ3bQ1PUxhNFIusO+glt9zLLt9ICGl5Xn6HcgJxtQwmL66V11nIzdIPKJch6IX7aRw8Q1qvcvCP9lkHhL07NiUhq64t1ptOTPiIUSSVqjl21LSHNdHV/n+rV5nm3qqF6Jkl1GaMOEwEyBD07kRyQiM4qP4wal82ifQz+OZFyPSc3L5lq96oR59ic6tzPdPh1QJmnFw9xUXgkTAIXbRHC5DlDKbGR8KYMcFi5oGZ+oHrrIJL66RmEXoGRriAlQItvu8nfdWtSoqyrsBFlyWAcmPN3M1WTL3sPU6NWJxm5gxYOLEoeVWmw4B9EGSRj3NSnlkH2D9DvsAo60vYdOUJN+5uvCHHBtg+6sTbcOSo0X8pqbzgWrq+/FJGBz7aYDs/bLvzlKQGEUuAjxsSSL3McchO8HKBGbNXujt/Y04v16/rLMN64WxPldUuI6eV6vjV5J701yu2ZZZAB1EnF5//fuugSXvQ5QypCW35bvoPs0ylE8xzMNKFtWyEef3FXHFaNts0ckoIRV+q1xa3ybgd0CSXrSVRokLtKPY9J3AkRkUhq6/N2pUWJ3JLaNXrI97o9v3TxsSNlt5o8619Ys7OJTVT8yTIBS8j8OW7AlboJ9vLsRiqSLx7fJjp/2bHMqktuc4Gu7Dmq9MM/+eOd0ttqlA8okrXyYmwpZ3Ehu0BWAQmhBc9EBSpnhCEsSAqPhNpDxieqhC87Jqy52A0rEPCH2iYvaBmopdTV60H6E6F63oNRF2WjCU25A6ScTiMSV5y+rRM//5KaXwDjtqhSgUUu9OQdRDy+oW+sVPyOAEtaSXx9tTJv3HKbGr/kDyk/uqk23fGwHKNnS6rfNT546TWW6jfKrZvzdpKTiZeGNe7CUhlDDeKKVGVAyTYgfoLSJlwWgRCykyaXH1wL+Ozxvsuhcs9ijK19q51RLBFAOuacuNS6rB5QypGX8ih10z+ex8c08z4G31KT/fBXL28q/B7FQ3vzRLCfO2quAfxIfRqaEJVNbOQ9YJ21YFmCZwvXLOG5Yumas854jz0Hl+o1n35qej8vemUrLtkbiM8FVOWLJNm2oTjxjog307oMkPXmNo4YNqHvC9JzJeonsdYQrVS6ck176NTgxOq6LhQ7iXUtTuzDP/mTPNez+0gSg3LBhAz377LO0dOlSAoWQWkBAntZLmJtq+dYD1P6dVFkpuRaqaoIOUMoMx1Uvt3VcTPjKk/GJ6svg3Ztr0BXVC7liKPnw4vFjAGW3ljRhxc6YGCD0FVTZhcfQWSi9ADbaIRgb16wL3rbdR2yhbdB3vDW9iG3faj0kCvW7vjr5WV7RDhZKuDlv9JB5k/2z29dvbl5qTH5t8bsJcNTtPY52WvL92Ywj60Bf+9GWZV3N5J5knkc/QFm9SE5alJIYgc50CRvgWMUzA5EBryLjOrnem7+tprfHuy1piG1EZjxKIofsl/fWcyiBdBZKuGpnpIS0TFi5Q5swx3P0e0aTDSgdqqM7arveLzb3X85jwIQ19PrY1Z7NyuW/gMY+2ZS+mPWH6wPThnuUOw6i+22ajGn92r89NSqjCBfyr4u30jM/xBcepBvbltXCZu3xUSXjF+MBlFAcipcPFQwmZfJdEJcWOK5v8jPNqHiebDaXGqhOmGd/oImkgcppAlCCSBxa2NDQzpIlS8yyvPDCC2lgqbynEOam8joQEWO4cXcq/QXz4MkYSpnhiK9gZHnjK694nqw0OSU+UR2DNYKlBYEPL14JtQ2SICat2hnzQvRzp3mtrA5QJppAYrOZPrq9lhNs3rTfRPpDrK9N26B18NLve201x9Lc8JUJns2zZjrXyY69/oOZvsPYxk+iIy+9eN+BPABlvT7jXDG+Nn3Z1kEox3+al3FVl3uSuTH9pDrVLF4ob6gfcGAqQAylH6BU4zoxOZ0lTVJ2JQIov7q3HjUwAErpgZi4aid1/MzMDPDOzTU8D/pkA8pLK+enD2+v7Qqpsbnvch7vTVrrq8AFADKuU9MY/sNkuoKDzlvWb/f21GgGOT72f1m0jZ7WsFPYjKGrAy7hIDGqXuOospDxAMpfFm2N27gAYwmMIC+P8OeD1V0HEh/9FMPiWecwz/545nM226QJQJktWzZHW7tmzZpncy0SGjvMTeUFKNUMRF2Wt/yybFw2r+O2xkOJhwsPGYo6Bjgn21Yp6Ipx+uq+etSgdN7oOqltwPMF+pVO3y1yrSWSSLwSAjwBpcblnag1zeZGf3t/fapXKo9DNA4JyjALXvrIhN+67wg1sACUQ+6pR9AA9iu28ZPcj58lz2s8E+Co32c8bT9gRyDtdz3q75BDBMeqLPIaZOyW17UBdMh7DIk2NcRkxGONHP14P0AJKqlObVIThTC3gRPXUr8xq1zzhFV0+CONnL8lAijBrHBJ6TxaC6X0QOBD7y4DoESCECzyXtKMyQaUCD15/7ZaFFSPW84DLBY6qVe50Bwyo/If6oivg+4/1H+6TTlauHk/jVuxw7O5af3avjUlmm2O+PafF27V0p3FMze0gYFhyKw/4m3uaqfGKcYDKEcs3uYZWuE1UajfgFfWRmBA18/4p5o6Kk7JLmGe/cmea9j9pQlACSD51ltvORyQ/9YS1qbyi21TD0MdoFT1YVn1Q8YnqgcuS7pJYvNv7q9P9Uvlid4itQ3cq6BfUQ+mQXfUtibiVu+/zkKJOomAH5s9NvbJJlQuf3ZHkSdRDjq/8diqvG3/Ebqkr7+FEpbiazTZ9Oo4dzcs6WS725ZE1tR0YIYZMgA5xAeamgGl1E/2ujaV5BrtmKYLPK9IKkMmtR/PHtZZF7OqAz4MqBIFlPzho3N5S4YEfOiZmAEAKJEl++S37g9BuW+8ACW7beEaXtunPd3y8Szf+ETop793ay3HU/JJAD1uOY+PpqyjPiNXem5v/mhGIuJzQ1OT9CQ3r+3zodZDHCbiQD+ZuoHe+M3b9W4DKJHYho8L9YM83vmhXbKAM/qC2IVcw3gA5cgl2+Im2MfHI4Q8/O45Et5OnIoNTRnXqQmVyZc9keXUtg3r7E/6RM9Ah2kCUM6aNctRsAEBecmSJV3yiViDfwPQDGtT+UniIUZo9Y5UC5oOUKrqC6wMwu4gHUBjzkkpGwd5QBCac1EPaWTVzli3K0biDdnhXu42r31+tgAlk+BeNWCaK74ujGeS1VW27z9K9fuO9xwCCQJf31/fsZb5FR0nolebMAAlXPgyacxvzkF+R5zsvY1LuZrIa5D6yV7XpjIlgOKElanaVi5AH9xeyxnDBlA+2aocPd7KHdepk0WU9yYRC+V3D1xCdUte6IAyNVlBPjuQY739EzPVlB/tihegxL79fOZGghsZ7xQbQMm8hL1HLKePp26wvu1yHjoQrXbEYT2sDsa/22qzmyYGGqI7G5RwqHB0Fmi1nWn9Ln1zCq3aEeHDROLOTwv/jDsLWjdXfFR+Ot1+fb1uhEr/dqYBJeKTc5x/nq9VGglvR0+cjrkUiEeUL5AOKK0ftjgqpglA+cknnziA8uTJkzGXgAcWajZpvYQFKP2sVipBL7Sqe1/jJjY38UCWzXcB/dapqbO06oHLChySJ+7Hhy6hWsXNgHJB99b0+4bd9OAX7mxRJPN4HWaegFLj8tbNN9n7g3kVO7w/g+b6KOYkOja7pnceOKrNkpf9A1B++8AlWnomdR7/64AS1FC4RlnkPpZk616AUqWFgaUI4BGFLWn4by/iZp6DLlFIB/Y6ty1PDzeLxH8mAih/ePASql1CDyiRXDQ+hSFh+tpdnlRTftReXoBS3Xc2gBIUMANuqUk6jk6v50nOQ7eualuOI/1u7mZX9nQiYgsYA16f2+oXd4azcb2b1q/Nm5OjBgHUSYSTVLdu4EUNAti91r5fh2qu+Ph4AOWoJdviVmwCgwJkdSG56VWQ8AbuZrWMerwxIV462SWssz/Z8zwT/aUJQFm0aFFHjxsWyqxZY3miihePPLhpuYS1qRDbhTg+U1ETCjgeT+rcvnF9dW1cTvn82R0VDRT1wGXOSUg0Iv4IBcTlNYvljk5FVT0ATcvsDXvofoVAGbGXtjQ36nWeLQslvyxtKFAS3ZcM/PzokDAOAOX3DzYwZv3LuSQDUNYreSH9vsGboxFjmg7MRq9OoC17U2mtEl0r2R5WorsaJg4ooax0UBxAsBSV7BIBlGxJw3/bAEpdotDnMzbGqHt0bV+B7m8ScdcnAij5I+/TaRuol0KnwhnOGAOeA69nUAUL6n1KNqCEggsSgWTyoM3ekPMYMnOjr6Qrc+2q7yrQonnFjPrNBV4fyNyi6CzQtuvXuv9kWpMSo41ri3cv4OMhwznnOPHxg0QIQTLFF1T6tzMNKO9vUopyZT3PNxELdXRyr4iDrlwop9+tDfx7WGd/4ImkgQZpAlAiKWfZsmWB9bPTwPpFpxDWpvLLUFUVHxhQyhe16eUJ6+boJ/SAkjknpTTaT/9pSCAElkWlaZn7x54YvjvEednS3KQ1QHnnp7Npso/Oc6L7EC/Kru0rOoo84G30KnwWUDMAACAASURBVACUPz7cgNq+paeRkm2TEUN5Sak8vpyCXoCyyWsTadOeVBYC1IXVb+QSfy5Qv3UFlc8dl5Qw7kep3uNloYSSx7GTqS4yHJRcH9RZoNSxBZTS8sgTQ1IElKlkke76eEEE+uOPPB2glM/3zHW7tbKoPKfXOlTz5D8MAihvHTSLpq91czyquspQrHr7phqOtel9hfTd677LeXw3Z7MvCTirgakKLQNuqRETmuO33+Tv8PrcUKeo8yfd2tsCSpn0lwig5PVcuf2A692AuMMg6+u1BipXaTyAcvTSbTEeLNt1R1JO7myZYhLcZHtwZSIEQSemcGPtovRqh2q2w1nXC+vst55AGqqYJgBlhw4dqH379nT33XenoaUJNpWwNpXfQaBy6OkApYljrlLBHDTy8cbOhaoHLnNOPvfjYodmCOXnRxpStSIegPKFNjTvjz0xfHewolz3vj/NjW7Fz4aFEpnwyKRGuffzub4ZnMF2SmxtWBG6tKvoKPJAmcerAFAC2Ld+c4rvsMkAlDY615iICXDoaJdgnfMLrPe9OMXtyPXlPpb6yV6AEkoqkl4S18KWVag/VSkcsWrYxFCC1ByuOVlA26KqNoHK6+4Ud30igHLYww0czfnPpm+IURCB9wJuPpTf1+/2/KjzS5xLFFCqVmBIAr5548XUb8xKGjjRrSLkde/lPGzWjcndh87f4kp2SYTKDPOD1+e6WkWcqeos0Oo1mNZPBZTx8jTyh8/qHQepjXg3gMZqwMS1No+Tbx1QqUnv05kGlAgNAt2WypggJ45z65nvFxuZJYLsY98FSakQ1tlvO35aqpcmAOWAAQOoS5cu1K5dOypfvjydd955rjXq0aNHWloz7VzC2lTjlu/wzJBWJcTghoE7RlooTS/PyoVy0IjH9ICS+e2e/WExfTs3Aijl4ao7wMH7N2/jXuo42M13BwB09UD/JBLdwp5pQIm5Yl3OOzeDM52Hv5yXFGua1wZmMuzdfx+jWj6AEvFBmKNXGASPlQxACXA9dc0u3+fP9KJu1m+iiycVHUmifdlxUKsR4uDwASWLBI5SDjJIwhGuBbrgew4fd+RJudgASunK5nZqQgj+Lt31NsDIdAOG/6chVS+aSwso5fONUJQbPjR/1EEa8KnvF9GwBX9qhwpyEOsslGqcKlRP+t94saN0Y6vHjYnJeYxeup0e/GKe595kK60amwhaNAgkPPtjrDyr72anCM0SpGlRdBZotQ/T+rV8YxKt++uQUx114uVpvKxqQRp4a01au/Mgteqf+rEJ1gGVVN/m+nR1PrurjuvdHh+g9L9npvkhYx37yCujHucWiOFNiYBB9rHtOoV19tuOn5bqpQlAicxuU0FSzv9npRy/w6ZmsVw0f9O+6PIxoJSupA9uq6V98VYtnJN+eTTChaceuMxv1/mHRVHpLhA+VyrkDmpWef+QwKJmdOPQu+pfAiiRWAS3CpfHv1lAwxduDfWZZTLsPYeOU82XfvMcC4AS96z565N855QMQAmdaFDO+BXTi7rF65No/a7IgclFlXDjv0OX2kRtoxv/1euq0o11zIBS6icHBZS68WwApS7z/Id5W2LIqkGUfnuKu97vGfdae/Ya6CyU8vmeu3EPdfAgw+f71+m7hVG9Zmm5DXIQ6wClmkmPjPA3bqjuuGP9kizk9ct5+JG1Z8+SkX58qIFD/6WuMdOi9R21gj6cHFyJDfGfiANF0Vmg1XtmfD7emETrBaCEUs4jXy3we9xifmc2gnV//U0t30iNuYfWvY08pc2AX9xTj277JFXyNR5AOWbZ9rg5iUGvdlH2zJ7XA2o7eNWk2Idp/9hcs02ddECZukppAlDa3LS0XiesTaUS8qrrUKdEbpqzcW/0z6y6IgGlyZ0lyZXVA5f57Z75fhF9P2+L0//oJxpThQJmQAkpuXl/7CXEHcryyyON6IoB0+K6hWfaQslyfTxZqFYAEIRZHmtZlvDi33f4OF3cyx9QIri8aT9/QJkMYvNm5S+iSasSAJTiwOQ1ROyjThZTPbD81hxxfzfUjsSxcZH7WOony79DvvLICTNzhOnwtwGU0pXNcxq2YEsMFYzMEk4EULLXQAcoYbnExxwKnksvMnwdoDw3wzlRIvcggPK2Qb87fLSyqFyfrOJk4y42AQK/cCAQ1CMrGEVdY6ZFCxrDyXNBPCEStlB0Fmh179p8cKFOvICSpSw37Drk+thEkpiXi9j0jMHiOWXNX3TwaGq2tBoLHw+gHLtse0zSpt9zzr+DwSRf9iz05jgz5ydotBD3z1Zf2/tgOwddvbDO/kTmdLbapgPKJK18WJtq4sqdMS5kOWVw0MGdxYUBpXR5m3gg4S6H+xRFBZTMOSkBFZN9mw5wEELj4FKtTDj0Ln/3XwIoxSGE65S0SUnaKjHdMHfh/sMnqHqvsZ7DIBYNca+NX5voO51kAMoWFfLRhJU7fceycelxJyrRPv8d7qpbBqVaQPwGlXFsXFfuYzkn+fcLs2UiWINNJRFAqcs81wFGmSUcFqCsUSwXDXs48nzP37SXrvUgw+drfuq7RfTj/MgHVDIBpUpvdkPtIvRah+qk0vn43XN5b/yuCR+4sOijqK5kpkV7fcyquGIM4fVpW6WA07eaQa67BtOegqcBIBAFdeJVkmlVMR8NurMO/bH7kOtjE2pSQSzAPHe8OzpfWoHuHzI3GvICa6/8KIkHUP62fAfd99+5frdZ+zvONijlvDVujbE94vW7DVsaVR9SKwb5MLKdZFhnv+34aalemgCUGTJkcAhiTeX/Mw8l1sSLrqR+qQtp1npvQGlyJcoDRwWUTEciDxid0oBKJD3/j30utwjmD4vaZe/EByilPKTcH0FcmEEeONZ/5jZeCiNB+vWqy9yFiOlSJf/UdgCUo55oTI1ePTOAEgfVuBXxA0qddKVKtM/XCHfVTR/Nsl7WN2+sTtfUiCRGcJFWRHl4cLb5TXWKOgB558FjxnESAZS6zHOd1Um66xMBlByGsnzrgRgqqVrFczsuX5SFm/d5xjHzNfcYvpT+OzMi1ZdMQKmG5uA+vHJdtcAxg/LeLNu63/O9InlI1XvAtGj9x66idyYET1phCyfWSU340W0s056SMcaoE0RJRt4ffPh9elcd2rznsOtjM14Cd6Ycu/2T36OAUvU0nWlAiSztAjmzODGh1YrkpMVb9scsNZLUnv9pKS3bekD7fKcDSuvXa1wV0wSgHDx4sAtQnjhxgtasWUP4e//+/enWW2+N6+LOZKMwv1K8AGXDMnlcFB38opYWSkj13aqx/MgDRwVoTBEkY6p0Wqiy3YpebR1LiDoWDr327/jT3OjuFxMTq78FIQAOAmil/jOPacMzl8heY+5CP1UkjHGmLZRwpcGq4FdML2pJ3Mx9mIj2g9JLgQ7rqosjiRFcynUbRcdPRSiA5JyQQQ8XKa4HcZ1b95v1xRMBlC9dXcXRT5ZFR5UC3sfrU9z1iQBKGYYyY+0uhzOUkzAQDgPOUhQ/+jG+ZiSGId65bokLCRnHJ1PS34McxDqXt5rcxZ6U8St2xNCMee01OQ8/jt61vdtRxpTkOhWoMS3aW+NWe1q8THNhCyd+t7l/pvWTLAioE4T4O2umc+nw8UjoBvgncU1b9h52fWyaEuDQBlnbB46ejInvxW8gRO92WSWSgBJ7TdKVxQMo/ZJMve49wiQK5czifAAgnGPR5tTcAW6HmGJQdC3SgE31neD3TrP9Pcyz33YOaaVemgCUpsX46quvaMiQITRq1Ki0sl7GeYS5qdTYLemyU1/UzLUlY4NMxOLywFEBJcdmdfp2IQ1Nyfyc+HQzgsVQFlWZBIBSJVBWX0RBbiZoIuY+30rbxMZKCdqbL+6tZ0VKjUGkm4wHVeOSgszfpi7cUuCLO3j0BFXtmbZc3m0q5aexCQBKKS3Ha2FSZQkKKGUcG/ddofuoqOya6RBv/NoE2rxHT7Yus3fVe2cTQ9n7mip0az03oNTFjUl3vQ0gMe0jVU5OhsgAFH734CVOU+jRQ5feVORa/fPPP84HfpmuI+MClBKE8HhIGhm9LJV7lOnN/AjX1fnKeargSa0rY2hVoMa0aMgwjydpZXDHOo6+O4pN3KNpL0qeVtQJwtMIXeu9h084c2Cqs637jlCDVyZElwIxvSrhPX7kkAsTnRRz48qPg3GdmrrYJc40oLy2ZmGHdeHdCWsdPmRY3dWCc+uFn5c5oVe6EuTDyObdjTphnv22c0gr9dI0oPzjjz+oUqVKdOiQO0s0rSyenEeYm0o9yGBZ5AcGX6aSeFsHKE2uRMRfIogZRQVnbPmQgHLKM82pWB63kpEKKBds2hdDoIxD79K3/HkTdfcVmZpLel6qveU2gBJZynD5l+02kk6c+sd360j9Z668afdhatLP38Xs27mocN6550Tnw1Qzh46dpMovjHFqqUTQ3BTrAYuvTQwlaDZ6XlnZelq69QwDUMJV/eS3i2Lmhb3oRW2jNgD1S9sqkcQILpV7jKZDKVYb0+GhyzxHe5m5q1s0G0ApYyO5D50VTrrrEwGUvz3ZhMrmT9UnxruAk+IkoPSz5unWKpmAEjRB/GGKdWGJWD9XvHof5Dz9hABkXZViCAlgjcrmtdLh1u0Fbo/fbKyKNh83EUBpT6sDjs3tByKW9gal89BX99Un6KrX7zs+OuUXr6wco9KEH1+5tirdVLcYzdm4h67XZP8zN66U0cT7X74H4wGUQS3Scu2vvrgQFcmd1Yl5RbgWzhq14N3Y85dlrrwCWScdUFofB3FVTNOA8pVXXqGBAwfS5s0RHsS0XMIElPLFjjWQiTjNy19EE0UWrg5Qmiw/kNWDLjSKCiY4XvLJbxdGuemmdm5ORS80A0ookyzYtDeGQBmHng0Rt+7+Zjo3A63u3U57620AJbKUB3esS+WfH+VSQzHtJbjtz890rutncJo1FF/9ydiHkuiZqWYOHz9JlXpEACUIzBkYyfEAKEc82tgK4CYbUOJesDvZ66CXv+nukUm5KSiglHFsPGbVnmOimammw0PK3cm54sOrfqk8xttrAyh1VEaTVu2kuz5zc7NKd30igFKNa562Zlc0hll6INTYOpv7l0xAeVv9YvTFrIiEKwrCAhAesGr7wUAfm/Ke+oWIyLoqXQ3CgBqWyUvvTVrrK+XHc86Y4ZyoxRZenwal8zo/2VDhmPailCZFHZu+eD5FLzw/amlHLP03919COw8cpbp9UgEl1lhVaUJ73qcQotCJTsBjAs+JlJ6d2aWFk0D08q8rCIpQbKHl+Zjex/LaJ6zcESN8Yfs+BU0TrhlE+GpMLvcB40WvX5fFKDXx7+mA0na146uXJgAleChlUg5cLgcOHKB9+/ZR79696bnnnovv6s5gqzABpWpdk3J4LSvko/EiC5ezJ2VWo4lYHP18fX99Z5XUl8Gkp5tRibzZSPIwTn+uhYvoWW0HZRJ8NapWJhx6H0xe70u/I4PM5a0zvQRsACXW55O76lClHqOj8UZe20IG8nM99as/GdsKrnzE9aEw1QzItCv2GO38DXQnSNJRCwAlLAU1fPgq0S4ZgPLSyvlpzLJIDKWqdhLvPTIRmINZQGctMa23jGPjOhf3GhvV8TXtm7ZvTdFmgXIimmk89cNOV09HZTR1zV90+yduKi2oV0HdBCURQKnGNUsXcu3iuemHlKScHQeOUj0BNNS569ZKvneCHMQ6lzdcqB9NSeV7vOOS4tTrqioxSSR+z5acx/GTp6nc8+ZwKFlXzS5m4YYPJ6+jvqNW+g3r/C4/8phWDX/3igtsV6UAPdWmHJXJl2pFloPhQ5VJuDHfILQ6JfJkjfItsjVatdqC/F9VacL4vE9NmfL/aV6anrm0At300cxo0uecbq0cHkhTsQGUfqwlXjcCNE3FL8xK701aR9JLJ9vAePHSiBVG7twg+9hqU6S7vF3LlCYAZc+ePWOyvDNlykQNGzakJk0iWtNpvZxJQCnl8NSkietrFaF+11cngHI8eMXzZKUy+S7Qaj8joefLe/WAkq2Rj329wAnOR8EXasGcqcoh+FvJLiPonxRPMoLg4cJSCZQRe4M5+B3IJqtcIoCyVcX8NOjO2iQtV157SQco/Vxrpv7wRc1rp9ZBstGWvZE4PqaaOXriFFXoHgGUeS/IpNWj5RAAHbehOkYyACUOxFFLI7FvXnQ7Qe7R+7fWpIe+nB+zbEEBpYxj485qvfQb7U6hBDLN6fJ3p9LSP2OzQFl1xnQ//fYv2iE+FMkDsiBZRqVDgnpV+6oRd30igHLCU02p1EUXRIeTMXEy6c6PND9sQPloizIOiTkn+fDetJEblWsp54l3XMkuI42Ps6yrgj62MAZJuJP7/4cHL6HaJS50xvYCSZ/cWZtaVsxvnKMKKIPQ6oCjl0UD+F6raltwbT83NFYNiJPC4FG6RkMnxWILMA4wLZ3K0atelBWgXLUzRvjC9mxvX7UAFbswG30weR2p/MvcBz6w+oxY4TKymPaP7bh+9cI8+/3GTmu/pwlAmdYWJZ75hLmpVAulVC+RFiTMmwGlvAZV35V/k5rV6stgVpeWDkWDBJS/d23p8IDJIucGqTsE/6uck3zoyQxc3Rrj6xfgTS1BwIraFuvz4e21SVquvO7vqpfbUuaMbpe332Fs6u+JVmWNGaQA2IhrQ2HVFAkoZXyU7F/GlNbtPc6T/iYooNQpj0hAmT9HZtpxQE+3E+QeQaXkgSGxknlBAaWMY+M1ghY6W35Nc7pywDQt5cjQhxtQzWK5jdvDxuXd/4bqdG1NN6CctX53DB2SjP9MJKSCPQk8aamIA7fg0BQeyr+PnaQqKfG5ugsMG1A+3aack0xx7GQkA5/3pgzzsHnv2sbtoS9ZV43dY0D5ybQN9NKvy22GJqn2wxrqaCjjVtWOOJvcNMCQmRup+/Blzsfa/O6tPa2dah+lL8oWJfBmTuG9h467vBewRHb+YXHM8Pzhg0xpnYrZYy3KUKc25emGD2bS7I0RWjrwDGdL4fXUXU/YgBLvcnjNoGwEiyzPS84Fz0OfkSuiiYTyHQwPGM6oZJcwz/5kzzXs/tIMoJw+fTpNmzaNjh8/7ljXZPn/rOWNdVCBmIyblAc+6uoA5ZodB7UxjF6Akt0bj369wOGKQ5ndraWjVCCLzKqF1B3oRAEWZGYwZ4f7JcaUzXeBk723euffLkqIIGBFfWCwPu/fVsuRNPQis+Z2cNuzjjf/zYYfUvegegHKKoVzRK1krJpy7OQpKv98xEIpLZiy7xxZMtLilCSlS/qOp20e9DdBAaWMv+MxZUiFaU7q4S3nqztkTMpNsPqwdRv7SHkNxCyxjGPjH+v1GRcFvaZ9c8W702jJn7Ecdiw3anrp2gBKXZa4TvYQlC1tKkeIsVH+89V8h9Q6aJn8TDMqnieVeUG6MCWglHtLN0bYgBJxwsimZpoblgU9ffofKtXVbGVU56rOc9v+I/Tn3iMOWTZnPHMbWVe1IvK91ikMScufay8LFzNLXuJ33XPD7ThW03Rfcf2zNux2FMgAKoMkrciPUrZQquIIYBOARrtamGVgyZb9WhUzaIA/2bocXf/BjKgSG2LkkSxoKjaAUhdPbLvn4Y3DvflwynpXHoFsD88aKPNGLIk8S90vrxT9YEAi5Jre6YDSdr3jqZcmACWSb7p27Uq5c+emHDnc0n7/37W8cVNVQCnJphFXIg8iljSTm2HtzoPUqn9sljVzl6Gu+jJY2KM15cqaiR75aj79mnLQzXu+FeW5wB1DA6sHrB8oJqk7PvRsAOVvnZo6fZkUT1wv+OdG+O55rA/oZer0Hqe1fqodSO46/s3PumOaBF7KzAmo1pEuG84MPnHqNJXtFokJAz0TK2jIthJQyoB+3RyCAkqdaxbZozPW7Xa6Nx20+C0I6P/srjpa9ScJKKWOtGl9kcSDBDVZGvQdH+WYNM3psnemaomPdUpQsm8bQKnjxtTFqQFUt6qU6godMGENvT7WLClnWgM1UU5anKRwQRD3MI9l4vT0e+h0MZSwwr82ZlU0YereRiXp+csrxTzr3Lfpg8J0T2v0GhsIUHIC1n9nbozKgOLdWblQDmpXpaArU5rnJAEc06rhNy/qIxlr6bdu+F1NWslyXgZHlrXPyNg4T1goaxTL7ZChw7oOUKomKpkS4NiSvvTP/VoVM1bvgjIOM4rI97vuWsIGlPi4LZ3vAicWFwml4FxVy7RnmztJVhxqhHCinr9ELNAAwwDFyS7pFsrUFU0TgLJIkSJ09913U69evZJ9r89Yf2FuKgSeIwAdBZmtkPpiC6Aap6cHlH+7+MN4UWDp/KxjXe1LnQm+peVkQffWlDtbJteaQtmFk0fky16+XJhuyC8GDS9sxFui2ABK08tQThCJD0iAqN9nfJRiw2tTwCUC14gs0hUdZEN5AUrJHwqi7xvqFKWTp05TmRRACWvtmhSXuBxTAkpJiqybV2BAuW5XDIeopOconz87rdpxULsEQQAl3ICq3js6lYDSlKAlB9cl0ahZs7rJtn97Ki3f5o6hhL73ipfaet5eG0Apk224M51bUU0oihdQ4gAFlQoX+UxIaVX1mVIvVHf/4gWUkF6FwpQs2ON9R62Igj4mzjbNK2iCXrWeYxySblm8LJQM9IbM+iOaBc2ZzSbwLff/qMcbU8WCEeMHYgxNdFdQKoL10LZISyo8HA82LU24p2pcOvordVE2Gt+pqRNGkOW8SJiO+vGL/Qgvk1qYtsqkNgQQ+1jLsnTte9Npfgo9j19Ciw2g9AoP8FsjnFc4Iz6euoFUhThuizj/fqNXRSmqoFzVY/gy52eA85UvpQNKv3VO5Pc0ASizZs1KixcvpjJlyiRyLWe17ZkClLAcdhm6JJooAW6unxZGXNIo19UsQm/cUN21Fuv/+ptavDE5Zn04A1r3UufklIe/nEcjl0SSMnRB2aYkCPlyYStKsgGlOm9cO+sQ88Vifd66qYZD+8PZlF4bBW77DAqg9MsmNfWHFzKIk3UFyULjVkSypzlA/tTpfwigBaVSwRwxoAd/l4BS6gDrxggKKHWxflKDWbrp1fGCAErEPt72SaxmNwAiU5jYAEpWc5JzUZVHdOsCgm/E+srCFhmvvWEDKHVk67oPHzWhKF5AqTIvSAlGqIkg0YiLFytC2IASljLEKu76O6KhzsTZuncP/iYpeuQ9Me2zqi+MoYMpnhKu7wKUSjIIA8qvft8UzYLmzGbTnOSzIPk/TdQ76Mcv0UvdbzKOGXGnj7Qo61gIpYY2t9HJ0qoxqUj+eliTAMehGSbCex77mvemR/kekwEoE5GyRe4APrQR9yo9J3INZ3dtSa+PXUXfzY3o0SOcCFKMKEj6XNbL+6PR6/k3/Rbm2R/PfM5mmzQBKNu0aUMdO3akm2+++WyuRUJjh7mppIUSoK7bT0uibmiVMFgHKE1KL3CdD7qzjnPdMlsb/+ZYQgkoVZ1r1JOJISYLJVtR/A5kaaGEy6L3iOWOokOLCuYsSXlI6gjUsT79b7yYvNRR5I3XuXWCxnlxf7BC3T14rnZfXV6tYPQesvtJjmPiWZOA0kTQzQMGBZQ61QxYQdb/FREWMJEJ47cggNKk3CQBpQlQyMWEtnDVIjld6yvXxDQnHW0Qy196vQT89i/aIoO9XUr2NvelO7RBto8Dkku8gFJlXpC8jtWL5KThjzSKjnE2AeUHt9WiF35eGo1v5Rg9TE43L0n8L++J6Z7qaMG8ACWHS3w7ZxM9+2MkC5ozm01zkh95kq7JlCmNfiD7WrmQe4967TEZYwiux4eblXG4fXWZ2KANmvRMc1d3qjcFyV8PfhHLqMChGSYeUH4ekLDDEofJAJR/7D5ETftNiuushVenbL7s9On0DaRKDnOHiP1HrO7XsyOcpzg/YIBBAe3Zkhf1IhlxTSilUZhnfyLzOhtt0wSgHDRoEHXu3NkBlRUqVKDMmd1xenfcccfZWJtAY4a5qSQpN0AdiGqHp1glVasc5Kn633Cxa+4bdx2iZq/HPsQIcoYLHUUFlAysHvpiXtQaqpMlNMWsyUNixnMtqFCu833lDxETNP6pZoHWnccBABnzZBNqqVhimZezWb+JUc42rwGCACOvfiDBd0vdYkZaE4Qm/DAv8hXN1gLpajO9MMFPueiFNk67Fm9MioI93VzuvKQ4vXhVFev11KlmwO32+YyNTuxstcI5XWonNge9DiyYiPbhHmRLjAQUJvc/VDEqFXLHXLfqPzmaPW+6lzo5yOfaVXDci17FBlACOLWtkppsg/50LAtqsoYElCD9vvLiQlacnMzGwPOW8dJpCVCCPgeuR/YSMPcq5q3bIyYSfdM9lcmBvBayrpoMAkaBOpCmnLs5mgXNmc2mOUn9aJld76WTrkpj+j2Mcp5d2lWgB5qWdgCdLhO72IVZaUpnN6BUk6+Q/HW/hlGBVaFMCZssB3vVgGlRXexkAEpcP6QqB0xYq+WC9VofWCXLF8hOn03fSJI6T7ZBpjz02f878w/nzwi16PxjJMtdvjv97kOQ38M8+4PMIy3UTROAMkMGc+YYknJOnTqVFtbKcw5hbioJKAHqevy0NHqwAzCxeR8TxKGIw1EW01chU+qgbqkuI+i0SK7nl8eDQ+ZFNXh1soTS8meyULIVRR1DXdBEACVcpIi/hBtYlpvrFnO+Um30dtEuWYCS+zFZhTAv/oqWkn9cXyWs52sKAiifal2OHm1Z1vrZ0bnukCUJYJzx3HPo2R8WJwVQylhJOTkJKCWggKUGgfZq0R3WEiya7qVOKYfVirwWywZQqtnb6E8ne6hmqA+cuJb6jYlcI9NsTV+7i24dFBsaIOcIF18+QeUlw1uqFclJPydioRSx235gQs4J8bFSDha/fdaxDvUYvjSq7CK1zLWAMmOGaNw49y09Kup90lGSeVkomUfyx3lbolnQCFFB7CCKbk6w2vE9cNUNqgAAIABJREFUkrGrXrHcKvG838MoYwx5T5oysaEaM7VzC1eXMrEPPwDI3/N5rJeEY31Nkpz8gcW8mIVyZqEZXVp6Tt8mhpI7kPvdb034d8RNIvFo8IyNjnb51DW7YprCg/fW+NUO6ERBONEzKbRJ0D5f0CPyMZ7MEubZn8x5nom+0gSgPBMXGvYYYW4qCSjBBfbCz8ui1q2b6hSlb+akSlNyMo28XpMWNVPqoK4JUD4wZG5UKUVH+i3j+EyAkq0oYQJKZIXCaqC6U1jiDdY//LZpz2HPrZBsQDl66TatywnWw89TvqKhHHN5tYhqCr+U21YuEAXycsISULZ8Y1KUh07WQSzY4i376aPba8fISHpdvC5WSwItqZqk9hNk3ZCReq2GTFkqOtkASlV2EHOS8ZGmOUkrJl/HC1dUoo4NS3ruDb8YYDRWs7fxN13IiSrzKA9YptnSxbSqE1SpvOTHY9XCOemXRxNweScRUMLF3334Uvpjd+T5k7KZOiCSOWMGl1QqwhtgnTLR1ujujReg5IQuKRAAa/wTrcyAEhy1kKLNmCEDwWXM6m6mOERcp0rr5HcOyRhD3pMmwFo41/mEGFpZZBw2/g4g31GR/cTf+Z2z7q+/Y7w6+L1r+wp0f5OIxR7XB8qw7FnO85x+2IAS3JPwSHgBSpx/4DtlVSaEE3X6LkKbBLGIuc+39rsFgX8P8+wPPJmz3CAdUCbpBoS5qaQ7B1rTiEViq+Qt9YoRAstRTHFgJh1fKA+8d2stp61qfeGX8f3/nRvNKNfxkElQYwKUbEVR3erq0idioURfSP5p/NpEV7cyjlDGg5puuw0wsonvM60FjwvalEHTNjj/lHF3/FIG2EfSzolTbk7WXFnPo4UpX9k6YIT+gliT5DroYrUkoPTiSrRZNx7LJAUKyibOcgdwYGYDk4WSgZe8BnDQQUnDax10QJzJ5b1eBzaAUicHqXv+2OXK40lAye5UXUyrOr+5z7ciyHhykWMhierXRxtHfwsaQyk/ZIPsKZ2FEolYT32/MBpDKSmfbACl3/i6d4uXy5sBpVQpkolZujmZ5mASjsDC6+RqvfaYlOlEhvIdl5QgmWgl2+qshmqGOoA8su7Vwsljpvh6G4u92mfYgBJGgwsyZXSSr0B5p1rBMR8YXAZMXOuoMqFI2iQIZyDGMtklzLM/2XMNu790QJmkFQ5zU0lACSvhi78sj7pLb6tfjL6YFQGUHPeiXtKWvYep0atuoIU6zNGI/zYBShAGw+2BouNoNLkY5cuFrShnA1De06ikQ26LYkMebQOMsmY611cX3A9QgqKEX3oy7o7XDWB/8qq/6NBxd7iHBJQ6160XkPLb6pDNvHrgdFc1ebDIeFq1L5t14zYghb5ygHucIffUpYal80ZJrqWFygQomY5KzgVJCbBO1C+VJ4ajkuvpkpmgeYyPM69iAyhhEWpePp+rG50Sjkp5JAElW7V0Ma3q/FRuWDkWOBVHPBY/oGR35zU1CtObN7rjsr3WSQco4eK//7/zopy1MlwhGYDSDwCqMZRM5wPRBqbVkSEi3B+oZu5rXIrAyoAYSl0xWflQV6cu5rV2kiSdBQ9MFlCoac3qGuuGlmuBWF1d2ARLf5rCofDOxLszSAkLUMLSrgoRNCt/EU1a5aamwlxxPuJZgpUSRdImQenr967pgDLIPQ1aNx1QBl0xQ/0wAWXF7qPpyIkIsIDbBfQbDCKl65SDuNUpmqTdmKMR9dXDkgHCvZ/PjdLb6Ch1JKefCUSxFcXLQoI5IKN4QpxJOWgPawDogWR5oEkp6tK+ovMnKSNpuu02wAigbt/hE547xw9QSkohnfsPgHLmut0xZM0SULZ5czKt3hGRb5TFz5pjmrguuUACSmmtth1Td8/hvrxiwDRXFzj4LimVJxCgDGr94QF1dEvMBep1U20ApU5qb/v+ozFE2VK6D2NKQMlAWaewo85P5YaVYyEreeTj8QNKjAUZyzzZMkXduzavSx2gVBOxZDKRbo8AyB09EeHeRfHb036AUuU/ZJlNkIIzrQ5T5WA87o8lEb2u25T0iDY6MQivvmTcLHS4b6pbjEyZ2CaLm1wLKALd/PGsmCHZK2LyXsmkKZt7LtfM5t0QhNUA4gWsJx59hstfRBM1gBJeNHyovzkuIhIgaZMgnTnTJw7U9lplvTDP/njmczbbpAPKJK1+mJtKAkrQ+bz86/Jo/B1cuogpQTG5KbbuO0INFKCF+lddXIjevqmG01ZVsUkFlHNo3IqdTh0dpY7URTaBKH6phgEo+ZBHkstjLcvQJX3dgPLhZqWpc9tIktIT3yxwcXbqbr0NoDRpjsv+/AAlrCFv/BZ56SFwvmXFCDUSr9FlVQsSLFQ7FW3zMAGlLlZL7ql7P0/dC3A3ywQom3Xj9YHKiKr3/hUAZek80ax4CShMFko1w9n2UdZl/MskEVM/NoASltbGZVPpgNAX9Omh1CSLyk8oASXztpr4B2U/KjfszoNHqW7v8U6VZABK2zWV9e76bHaM9UjVaZcJfkgG6zp0KeXOdh7NWh9RPwHRPH9E49/JBpQM6GWMswwZ4ufQZAWU12sCZagDRgbEPdsWqVYFHe4bahclUyY2gP687rExgfI9a2JUYC1509yhMHOXT0yxek1BLJRBAKXOvW1KWoQXDdKMnDwlaZN0Mae298WrXphnfzLmdyb7SAeUSVrtMDeVBJR4YHqPXBHNYoMmLni5UExuCp2FBPWZ9Bv/rVqfdIBS91I3Ed/KlwtbUcIAlHghTlu7ywHHB4+epHp9IocpF0kF0um7hTR0/p+ed9wGGOHF5EeS7gcoJUiSblIJKBdt2Udb9h5xzVdmKurob2wOX9MC6ABlt/YV6b4mpZwmEijg46Jkl1QNZpt1Qx+wGBTPk5Uue8dtoWRtZb5+CShMgJL15oM+wk1emxiTnKWTTFT7tQGUOu1maMhDS14WKd2Hv+sApU6yUZ2TClh2/32Mar0cAa9QcoGiC5dBU9fTyyNWaJfLD7AFWWMdoJQZ/OhL93EqZRDVsBK/+QW1UDIp/thl26O0Osz7iPkhFhfzGXJPPV+lG5MHCP0gpi9b5ozWyydlHPkjx5SJbcpalmthYlRgQGmaO8dvWk+ciKRxQbbT3bsggFKXoIgPbiToMHDk8bCvEPLSd1REqlLSJhXVZMUHuT5T3TDP/mTM70z2kQ4ok7TaYW4qSdoLtzMkzCA/hQIJM/5vU6bqjgNHY4AW2jLpN/5bPfT4JXDP4Dk0fmXEQql7MVz/wQyas3FvzO/ypca64L6AMm82mvB0MB5Keft2HjhKdRVAKQPtn/5+UTQ73nTbbYCRSWfb9BLVXTfCE/ilJ92kXBf0T2OXb4/hmZSHiI6g23SfbLa5LvhfAkqp0Yx1ktdls26YA6xvkIdDNrYsnPXMfUpAgdjgV0fHahmDcw4uyaBFR3IvM+1N/dnQBql0QOhr/+ETVL3XWFe3KuG1BJRMSaOLaVXnprI67Dt8nC7uFQGvUHYZ/UQTV5MgVqSg68r1dYBSTcTS7RfwnYLBAuWCzBmj8ZY2ezoooATFmCOksHwH3fvfCK2OykUKCp7zzjVT2vH1mj7Y8TtClDJnjMgi2hSEubCLmj9yTEpnUuRA9i3XwsSowIBy2/4jMV4d9PXS1VUIDBlBCtbh46nrHbW29u+kPt+6e/3u+DVRD43fGKoaHOrz2YWz8MPJ650ukLSzoe9lJD+cJG2Sjgjeb2yb38M8+23GT0t10gFlku5GmJtKAkp8gb0yemX0IUKMIEz8KC9eWZnubFAi5op0QMt5KBUSdB1A6PjZ7Gisiu7FcNNHM6NuKpNVjq0oYQNKnWtRurHAo/jt3FSKJd2ttwFGXprW3KefhRKuZLYWIQO2Udm8TtPPpm9w4oVev766Q/K9crtbO9sEKPEy/SclIdzPmmPa8iu3H6C2b7mBnqQPueXjWTRj3W6nOcYAP94bv62i/zQv45BE64p6zwGWACjVcVgKj+tLQGEClDopUJvHWSfDqVO4UfuyAZQqHRD6OHj0BFXt6QaUUgsadSSg5NhQE6G1nJcqNnDg6AmqljJWWgKUaiJWEEBpYz0OAihLiQ9XqZ1tQ26v218yzED9XRd37rVHJaBkrkhTjCaeEdx/tci10CXAoT4DSpOxgROCbJ4nv+c+UUB5Y+2iMe9t5hd+e9yaaLwk2DfW9mnvvEORuOq8TzvWoUe+nO8kNyZ6Taa1CPPsj3f9z1a7dECZpJUPc1NJQImHE9YazhCGJYtpUkzUJ6YXHjgsX7muWnQFdIBSWht0L4bbBv3uuJwZZHBnsi9QHZ2f6VwtWbBcfvmij+e2SHcft5eHRJehi+nr2YkDSl3WoTpfP0AJazK/9HRWLfSncyGZAKXMio4XUOqC/yWgvPHDmfT7hkiMm+0Y6kEP1SQAyjZvTnEtGdPocH3IpLE2swlQ6pSbbPaNDlBy1qtXextAKelwuK8jx09RxR6jXV2PfbIJlcufPfo3HaD0UmDhhqrYwKFjJ6nyC2Ocn/HhA/UoWc6EhVJ+hPLYatysH6CE7jIzHNjsNT9AKfkdP7y9Fl1aOaJmJLO/TSwZfntK997hNjZzl/3LpBzekyYeYVjxl2u0qeVa6OKVMR4DStPZYMN64LUuIIx/e/wa58MYSTVqCWKhlEwm3A/Twb03aW1U9ADctat7t3OUvdjSDdokhH7gYxksEhkynON3OwP/HubZH3gyZ7lBOqBM0g0Ic1NV7jHa9XJ9fcwqh2sLBSTWAydGOLdMX2A6yx3qM+k3L4EOUMqMTd3L0fS77It1wcO2UOpi1WRSSbdhS+jLFM5O0223sVDWKp6bkDDhVUyAEi+9DrWLOIc9v/R0Vi303eH9GTRXGUcCSmkxzJ4loxNDihL0EOPr0AX/S0BpCm/wWgf1nkM1CaCnVX83oGTlkiigFNdjApT8oRL0EZZyodyWrUFefUlACUqVT1J4RGUbvg75N1UOD7+ppOzyYGQLpZcCC/evroHUck5LgBIWWRnmoNujg6dvoJ4pliVpobbZz+o+U6/dBCgl76OMoQyyp2SYgdrOZu6yjZwn04mZEmfwEbnq5XYxU5Vroa47V2aLvOls4AzzIOsQpO4749c4mts2pWPDEtGcAa7PinDSvY1EvpUvtXNiXyHziSK9PzZjxVMnzLM/nvmczTbpgDJJqx/mplIB5Zu/rXa+/lCQdPJOCueW6avS9AWNhJ4eV0Q4GlHiAZRMvqxmQ8q+OAA/bECpe7HLuFJIv7HGq+m22wBKSIBxNqpNP3zdcEsvf7Et4cUHYPv8T0ud5irJNfcpASP/TdKYINufydEBNPemUBkFPcS4b6kDzX+TVFRQCRm2IJLUZDuGes+RmQ0LJUjZZWFexiCAUke0b/M41+8znrYfOOqqauNWlYASsaBISNv193HlOhrEJHGcPHU6StjOlVlekf8dL6BU1atABl/u+VFOt+XyX0Bjn2zqmt/ZslCCd/LSt1I/InT7R7oqpYXaZq/J64I7FOwORS/MGr12E6CUFkGTMITfntKFNHAbm7nL/qULnlWXTIkz0Ltf07t9zPTkWqjrzpUZUJrOBhsaLb918fo9CKCElwlJUrI83rIsPdm6nMsaCav2sl5tacisP6h7yrvV5P1JZO5q2zDP/mTO80z0lQ4ok7TKYW6qKi+MiQao4wUlH0YJKE1flTrLHS5bcjTi3zpACZUFvIy9QATUFvJlz+zKZtT15QcokewCOpp4y/4jJ6j6i+5YNRkG0PPnZVGKJTlG7eK5HUsg5MWmPeuWMuN6cu5Nyl0UXRPTXHUWSg4aRxvoeHcZusRprpJcc586Pj8JKCXhcdl8F9CanRFOyqCHGI+nI2iWgBKchACVIHnWxerq1kK95yB6hoWyxRtuQMm8gFwfCQcHUiyuJgvluj7tCRruQUu9PuOiii3c9s0bq9M1NYp4diWzvAEoDx0/SS8MX+bs+wkpiWucPSw7UtVL8Bur4XA9HaA0KaTgkk+nxMuqSR9Seg974rdOaQNQwiKLDyh8iElBBblOn07bQL1+jcS+yftvs5917xvZt87yh99lzGK8gFIX0sBj28xdzlMmCX12Vx1qXiEfmRJnsPfxDKhFrgXWXfUGoD4Dyr2HjlMNhYEAvzNlUdBny7a+jH30avPqdVXp0LFT0X3BdTk8ASpxXYdF3qPw0izpeSl9+fsf1G1Y5GOdY7Nt5xVPvTDP/njmczbbpAPKJK1+mJtKBZQy3urRFmWiqgCmr0rTS0NyNGIZdC/lDyevi2YjB3k5ng1AqbMUSKttr1+WRymWWlTIR4hRQ5hAjWK5HTWgRmXyUom82bQ7Ql4PABVkEb2KH6D8bu5m6vzDYqcLBlNqf1KliH9TiZbBVYlM1N4jlmuz7YNsb50Mm4ks37ZfFVBCNQkHhOSwRF/MC8j1wd2HDwQUE6DUUc/YzOuF4UujPK5cPygPJWdio738OEASRLUisYoq6jqoKj86QGlSSJHSnxxOwtchwauTxXwWAOXdg+dEATbPCxbZPNkyO/HWUDnRUemECSila1sqU0l5S0lsbrOPuA6ywct2i1iF1RLknYm2Y5ZtpweGzHO6QfwfPl5NiTOoo+tf7jWsu/rxhnYcn6ljIMDviH3sUMv7AyvIGql1bQElaH+27T8aDQ/ifph4/fu5m+mZlPco3hlIAJUgUxeCksi8dW3DPPuTPdew+0sHlEla4TA3lQookZDDNCoylqv/DdXp2pqxLwFTjI/kaMQysGrEA01LUZd2EXUZuNAGz9hA1Yvkonql8liv1tkAlHClYq1kkSBbuogRlI7A/HNgNrQo0t0JFZuRS7Z7ttIBSliW1ve9zGmHoPWnvl/k/LcJUOqkIk1kxqZse4tLi1bRZZPGm/nKnapACtyRh4+fpKb9JrmmxpY9ri/ViEyAMuhhzQPCQgoxgNxZM0WtGzYWGWmhlCo9ElCq/JKmdZCA1DngRXIB921SSIGrkzXedVZaXsPSF2Wj8Yry1JlweesApWqR1e1LfEA0emUC5bkgE+U4/zxavGW/U83mPvtZKE2AUqoRSenFIM+NzgLN7W3mLseSyj3MaWpi6TCtjVwLyHiqz5qz326tSe2rFnQ+2lSvDn43nSVB1sWrri2ghNt/x8GjUYsj98mGgp8W/ElPfLvQ+TN/bH8zexM9l+L9Mb1bk3Ud6CfMsz+Z8zwTff1PAcqXX36Zxo0bR5MmpR5WCxcupMcff5zmzp1LefLkoccee4yefvrp6NqePn2aXnzxRRo0aBDt3buXGjVqRO+99x6VKVMm0PqHuamqvjAmmvGKF9THU9Y75OYoN9ctGs1cNknRmV4anVqXI0gAyoIv1pxZ7ZUdTItkAyjvbVSS8ufIEr2WRF3eACqVergBpbQ8yRhKHQG11w0v120UHT8VkYIDifrwhVs994cfoJQvQtNXtIxZ5MFMgFLliAy0eVMq67JJ48185fFVAAMZzsPHTlGTfm5tebbscX0ZE5psQMlzk0kvNkkIElAiW71QrvOdruIBlLK9c8BrAOXqHQdjsuFRF4ldvBd1tDRpEVCy+o/fvgTYh8X9xo9m0oJN+5zqNqAsXkApyeN170O/+Zr2Of/dZu5yjJ8XbXUkYlE4Wc+UOGNaG7kWWPfGr0WeNRkqwYDSFP/51o0X09U1CttefuB6b41bTW+Ni+QBeBXwSELmlj++uS6/139dvJUe+SqyXnkvyEx4v3w3ZzN1/jHi/dGFoPiNGfT3MM/+oHM52/X/ZwDlW2+9RZ06daImTZpEAeXu3bupQoUKdPXVVzu/zZo1ix5++GEHMHbs2NFZe4BJ/Puzzz6jwoULU+fOnWn9+vW0bNkyypTJnjQ5zE2lAsohMzdS95QsNiiMvDZmJTUrl48eb+UGh7y5JDed3HDxxgzZbFobQMkvW66bKKCUGa48R5lsIaUXVWJpv2uq0H1UVFv4+lpF6Pt5W1xNIKU3fsWOaIKUH6DUHRzqHHS8mSZAKV+iQQ8xHleXTZpsQAkZzsPHT0UPOR4b+t5Vi+R0aSgj9hclLEApQ0FGP9GYKhTI4bkNJKBEtnrBnBFAKS0ipn2liyXFxxQXHaA0Se5Jiiid25/HKnVRNpqQRiyUQXXXTQpcphsUL6CU5PFSBMHvfaD+nizLr/zQ5GS9oLREci74cGHZXcRcIsYWhQGlpJmS12STpBZ0jWR9E6DEuFAHYwUcxJHifQFvjSwDb6npxOJK6UzE8c/u1oqkG1yVOE1kzqa2YZ79Ycw3zD7/9YDyzz//pHvvvZemTp1KRYsWpfz580cBZd++fWngwIG0ceNGypgxIn/VtWtXGjp0KK1cuZKOHz9OefPmpddee40efPBB5/d9+/ZRoUKF6NNPP6WbbrrJeu3D3FQqoEQQeLu3p1C+HFmcoGM/t63pKzTR+DivxYkHUCaqZKADlFIBRfLjqRYivxstww6QRQqrFJdMGTMQMo4RHjBiyVaqXfxCV4Ypr4V0eUvXFnSsG5SJEJvLgkxFZCzKYgKUp0//QyOWbHPUUcoKfkO/65K/b9l7mBq96rYcJhtQQoYTySzqOOwq5rXCde4OGVDi2uHyRKhEs/L5fJdKAkqpIx4PoEQsab7s3oDSJLknZSm9Yuh0vK7JAj5eiyVDcrieBOC+C03kZNCHZ6GsSW2rFHSmAXL+KwZEZECfaFWWnmhVzmZ6MXV061qv5IX07QOXBOpPhsJArhIUZaYYeHTsF0OJfVq/b0SOVgJKBmSmhKJ3bq5BV1YvFGjuQSpLphLZDt6anQeP0cNfRgDk4I51HBB8z+cRNSMunAEvk5gK5sxCM7u0dNTQoIqGkm6hDHJXEq/7rweUv/zyC3399dcEd3evXr0c8Mgu7/bt21OuXLnoq6++iq4UXOKtW7emHTt2OHXr1atHq1atonLlUl8kcHtXq1bNsVzallABZc8xMRyDABAI//MDk5i/LrYQf5ccjbbXaVsvrQBKqYAC5Rnmj1QJof2uC3FGnCRya71iLj5LCRR1/egA5eil2+nBL9zB92pbmUTEv5kApd/8bX7fuu9I1JrB9ZMNKCHDCYsDW014HLbs8VohHipsC6XNmsg6ElAiW50tjBJQ2sZQwlKb54LM0e51Fkpd1j0a+JF+e1n8zwSgxIcVxBYkz+Dsri2dD2DbkmxAOW3NLrrtk9+d4ZnUG/8twx6YisZ2jrKeuq4QP/j0rjp0UfbUe2zTr0zWY+uaKXEG/fkBSny41O0dAZQymYsBpe4jHHVtpEhtrsdUxwQokZy3dd/RqEUSiUkAwrcOitw7LswvOXHVTur42Rznz4VznU+whEtQzsl+iczVr22YZ7/f2Gnt9389oJQLetddd7kAJUBhu3bt6NVXX41WW758OVWuXJnmzJlDmzZtouuuu44OHz5M558fcV+h3HDDDc7ffv31V+v7FeamqtZzTJRCJR53powtlBrJJqlG64v2qBgPoKxTIjd9/2CDuIcHyC7VdaSrvTw8WvefHKXWCZohXPOl36IAR0e063VfdIBy7LLtdH9KNie+wnUWMnCvsQoSX5RJrz3uRRMNdfQkiQLKB4fMo9HLUhOYIJd45MSpqNWEhx/5WGOqVCjHGXV5B12zst1GRpNhJECKB1DCUptb6JDrAKUu6x5z9iP95v2ms/i3eXMyrd7xNyEB5Q1BLB3Pe8Vv/aTWPJKxgoCrMAGl/MiUmfSIJ0ccZTxFBZRD7qlLjcteFLgrXTyuKWQJnfsBSny41Hp5nDMPmczFgFFHuo+6NspRgS9ONMDHBujv1IIPso27D0XjIgEcwdvb4YOZrqocDiCTrRhQDluwhZ78NmKhZCtvInP1axvm2e83dlr7/X8aUCKx5pZbbnEsl1wQH1m6dGnHRQ4L5e23306nTp2iDBkyROvccccdtHXrVifBx1QOHDhA+B+Xbdu2Ud26dWnz5s1UpEhy6RakdSyeF790a+CljiBvlLC0TdF3EED50ZR1NHT+nwRJtOJ59LQ9tg8OSIARh8QxOBJQ1u09znGnmF7EXmPU6T0uum5S7pLb2ABKVnJAmz92H4pmX6oUMtynfOkiaQWWvbolLgxFPgxj6uhJ4lUP4WuABeSSvuOjpOug9cDf6vWJWE24sLa1zkKpIzaO5x7a7iFTvfLPj6JjJyOJWdJl/e2cTfTsjxEuPFsLpapDLgElZ4CbNJz9VJF4DYvnyUqTn2nuuhyAE1jl6pXMQ2AusNm/8a6bBJTzu7d2snBti0w8tHnn+cVQSgJzCShlJr3KemE7V/V9h3/Hq9AiLaZsxTZ5mEzPgFwLeAQu7vWbcykymYsBpYnyyEbbPsj6qHVNxOZ4D8Ayz4k2CAdCxv/l70bCErhwzLXkEWVAOXzhn/T4N5HM73TaoETuUvC2/9OAsmrVqgS3t85COX/+fCf5pkOHDloL5bFjx2j48OHGFe3Zs6eT0KOWtAgopVsjf47MUVJnm8zW4Fsq0iIIoIx3DFM7E0XI498siGZn2xxSsn+proJYKzVD0as/SGUOmraeBnesS/UF9dKElTvon3+IWlbMr72UARPW0OtjI/Jktlmyiayljp4kGYlbnb5dSENTFHaW9IxYKNkNx/OFokf5Atmj+0ZmeacVQFmpx2gH1KNIi1s8gBLrkD1LKpuCjDtkQGnScJYcnV4WKh2glPvDD4QlspfQ9tI3p9CqHQedblQA7dc3gA5cl9WL5nK0mP2K37WYAKVUhwKn71NtyvsNpf1dtVCq0ppBOkWiyQWZz6NGZSNx1Tr2Cu7Pz0KJDzimBUKsN8IRUFhqVBLhyzlKrs4gc7etCzd+y/6T6dTp09GPTbT97ckmzp5hQImk04uyZ4ohZx/7ZBMqlz87gYf3+hTrJQtTyITH7x64RKslbjtPm3rpFsrUVfqfBpQAk6AKGjJkSPSKdTGUa9eudazibwKpAAAgAElEQVSWXBBDWb16dSehx1T+TRZKCSilhbJfh2p0fe2iNs9M4Dq6F3yL1yfR+l2Hon0FBXW2kzABSsTkvTthjcO/VqfEhbbdOfUavjKBYP1E0WUd+10LXuR4oQcpsNz2GRmRHDNZMYP051dXR0+SDEDZZejiKLXV0hcvJVjMYfGVhbOsed/4AUrEqP3yaCO/S0rq7zIxC/QkoClBiQdQLnvxUhe5tw5QmjSc/WQ2eQ2LXZiVpnR2WyjlgviBsEQXj93r6GdxzzaUQwDoRPtW2/tdi0y+kXJ8Mk71keZl6OlLEweUPa+oRHc1LJm0SzTFOWIAP0CJda/WM6IeJtkBGFCaODRBKN6mcoGkXYOuI3w0/LH7sEuGFUppy7bujwJKUCfB8sjUR9wPvw8l7VPRC8+nqZ1b0C+LttKjKdRL6Uo5od7CmM7/pwHlK6+8Qh988AGtW7eOzj33XOfiu3TpQsOGDXOyvGGFzJcvH/Xv35/uuece5/e0mOV9ca+xDheX6QXit2VknIzMng2Ta0z3gl//198u1QY/EOZ3XabfTYAy3v7QrslrE2nTnsNOF1BpYIk47jOMaxk8fQP1/CUiRQfFEyifhFl09CTJAJRS8hJACgckx3XhepDUBIsfklR430hic9VCCWttvhyZKXPGyDN9poqMZZZJNZKyydblrWpw6wClLuse1yoTlrwABR+wpvXxA2GJrquMWVYBdKJ9q+1trgX7EO9CkGJzMqMMK/hP89L0zKUV4pqazfhxdUzkzLn886O1zf0AJSzhVVMAJUJujp6IWChl0o0uUYuzqOOds207NawDzzaonBgQAlCWvuiCmA9QToqD2tmVA6Y7w/EHlHxu1OQ323kFqZduoUxdrf9pQLlz506Hh/LKK690+CVnz55NDz30kAMy77zzTmcVunXrRh9++KFDE1SiRAl65plnnNjKJUuWpBkeykQBpYyTkQc1f6UGeXhs65pesGG+eHlubkCZShFiO3ddPWld7X1NlRjlhjAApZQQ4xjDRK7Br61O8z0ZgBKqTgBMKHDBwc2GJCcUJI68eePFjvyl8+/nRjj/L/cp6K36jopYalHCWGu/tcHvMjFLJtVIQMmxXWp/6qGtanDPWLuLbknJZGVrHizisIyrBQo46/6KWPq9AIWXNr1c67DWtFX/yQTqIxQVQNusd5A68b5XZFiBKkV7Jsa3GQPejXLP20s7yrXAulfoHgGjkm5Kvvt1gBKE4qZQHJs529ZRwzpALzVn414XuXvlQjmioJj75RAKqXcvLfIAmsgOr1wop+1U4q6XDihTl+5/GlDiMpHNDXWcBQsWUMGCBempp56iRx55JLoCSMgBNyWIzY8cOeIQo8PVDXAZpIS5qRIFlCdPnaYyKVqz2TNnjKruhBl4/b8GKOXhCJUGVbkhDJAD0uFmr08iJGH89mRT5wUZZtFJdCYDUMLyCT1hxPSBCgX0S5woAK47cN5x4X0j4wTTCqCs/fI42vV3JKkLyQ65skaSTOIBlGt7t6OM57pDIEDInDd7Zmqewomp0jiByPm6moXp3QlrPTkaeQ2h6PT2Talrq+6deEGY7R5s8cYkWp8CfFUAbduHbb14r0Vasx5qVtoJZ4mnxDu+zVhwSyPLmSnPZBs/CyWUlJj5QjJ8+AFKEIo3r+DPzWozf686qhUeyW6z1u+JAkq4rGsWzx2jlc4fKFJNyi/EI9G5mtqHefaHNeew+v2fApRhLZJNv2Fuqhq9xkYDl+MBLjLwWiblyFgim2sMUsf0gmX+Mehhv3drrSBdWtcNw0IpEwzgLuKgcZ5UPPfF5oLgHgaQhBxd2EUn0ZkMQIl54zpwDbgWyat3TY3CjoXy3wAo6/UZF01ok0kmkgLKlIyhWoF0konq/d2+/6iLXunTu2pTiwr5fUm/wc332/Id1PnS8lHQq9s7YYIgjCet+joAncz93HfkCvpwynq6tmZh6n9D6n7yG0OCdrA3ILwinhL2WuIdDiUuphrzeu+oc+F/S/5SSVyus1CaqMziWRuvNipVGdgA8P7mLG2OgVTnyLRvMgbWLwkt2XPn/sI8+8Oac1j9pgPKJK1smJsqUUAp+RnbVi7gJAOc/ucf6n9DdSti9HiWyPSCxdf28m0HnAy9sEBSGIBSxjMivuje/7qVG8IClPGsfbxtdHx3T7cpR4+00Et6JmOc62oWoTduqP6vAJQN+o6nrfuPOnOF6x5WVBQ8X4j5ypn1PCc+zw+84XcbHlSVxondkFcPnO7EmaEksu/CBkHNX59E4NK0vd549xPfA7xXoBSlWn69+pVr/EDTUtSlXcW4phH2WmJS8r0WD6CU/KV+gPLzu+tS03LBeTSDLp66xxHuMXHlziig5CxtFVDyvpf0a4kqrQWdezqgjF2xdEAZ7y5S2oUJKGXsVjwHiMzka1elAL1/WziWQbkkZ+IFa7p1YQBKWAhgiapSOKdDvHv7J7Ndw8dzX5K09ZLWjY7vLgxAKaVAoYve7/pYQJkWYygbvTrB0RlGCZq1LJ8H1c1vuoE7Dx510St9fEdtal0pP/1bAGWzfhNp4+5IIltafT7kGj/QpBR1af/vAZQm0GeyULaqmJ/Grdjh3A+p1a2zUMZLzB70ZaQyS0DBDNZ1tlD6AUoZZ5wOKIOufvLrpwPKJK1pWgaUuER+afx/AJRwr0Ir+uTp0wTN7qyZIjruySqS+8zLUpCs8c5UP4jZrPzCGNdwUFR5tGVyLZRynBtrF6VXO1SLjpmWk3IkQFJ5JP3uUTwfWOphy1QuVw2cTov+BRbKpv0mOrQwaRlQSmaD+5uUoq7/EkApBRvUvafutRnrdtGizfupepGc0cQvP0D55b31qGGZCA9mmEVNBESsLWRpGVCyIo7JQiktnOmAMsw7Zdd3OqC0WyffWv8WQBlm7KJcpHgOUN9FDlABXIf/0D9JB5OYguoavqtBCep5ZeUAs0ubVaWiEs8wDEApx7m5bjHqe22qmzgtA0qZZAI+TbgQbUs8z4NK4wQlqUsrF6B/C6CUVFtp1UK599BxqpHCOHBf45LU7bJKtrfUVS+e+xt0IFs6NNNcZm/YQ//X3plAR1FsffwaBAkgBJKgMQZICKgIhCAkLALysW+yBMLiA2V5EFDZwxEF2RHxY3tCJOwKCgESQECfbLIqa0Ah7Pu+PFafIGHxnVuhh85kkumZqZ7uSf51jkeS6b5d/aubqv9U3aobHZ+WwtCeoNQzvl793tYbATm2eOXvFzULSt4kx5vluAT75Sc+x9LdRc+x393v4urzIChdJfjkfj2d6o1Ra+n6n6niSc52zEon06RcAE17p6Kkt87cjLIruvYr/jS3c4Tuz3P3A/jQaV4i5v/Cg3wcittyd121Ps/WAcp6CEr1c/5RpRiNbuEZgtKVcxWdERzWglLJXtJ86lb67fxtl/oDvtmZOmn1Jb6uxvgNdO5GWoiAs/2WI89z5lr1RrSubwbT0KaeIigzPw5NhqDkDDVVS/o6g9She6y/nLOfcOrcvglpqRPtzVCqN/iF+OWnDRCUDvGXfTEEpSSiHiMoywfQtA76C0peruNdiY3KBVg2L0hCDTM6EbB1gLIeglL9nHerFqcRzcta3kgZDIOKeFvECO9u7jLv6SYoo8SJOjc1x3o5EkrhjHhTz74wIOWYL08RlOqYU6PazN6fijpuuEv1YPq0GQQlM3NHhhl+jnWYjbWgVHJxZ7bkrW4/CEp73q7/5xCUkhhDUEoCCTOGEVAfgK9Uon+90tRbcgyl+qBm63ABZeAoG1iQur0ZQpfv/EW8WSJ48A8WLkaJk8ZTtogTCrgcGtmQvPNoz9QjQ1DGvVNRpA3tv3gfJSVfcHnmL3jwapFLXq8ZRHW6UqPazN4fkzpPdufqJWhYM+dCV5RTIPQMf9FrybvRlC106NIdqlyisDhUXD0zaI+fq59bh9mwnyzbe576JfwmTNsTlOrVDghKV1vD9fshKF1nKCzoKShnbz1Fo1YdFLmDOYewM0UZ0JqWD6CpbpihdKaOuMdYAuoD8JWa9KtbmvrUlbspRy1c+9YtRX3rlra8uFpQrvqwRobf6yV+tJB/e+pW+v3JUrOjmV+cEZTWm3J4ZYEPN+e4v4+X7ReZTFq/8bKWqtu8puTHP4isRXoxVR+zZFZBqRYkropBPqSb804raR2dbphMblRvBszqnMjMfE19vzqGkr/g3bybSt/uOEv/Wn9MPD2xZ1V6o3gR2a+QwZ71qoi1oFTqoX4n9bur+5IQ//y0YQBiKHVvtCweAEEpib6egpI7/e0nr9NrAQVFHl9nivIH+UXr8tSmUpAzJnBPNiegPq9UT0HJtgcn7acDF27Twu5V0m1uUfzU+rgiZwSZ7OZSH9fjqKCMGLOOrv5xn2qU8qP5XSM1Vc362CB1/mVNBuxcNDjpd1q485xugrLqZ+vp0pNzO80qKLOaLZfBWKYN/vvk829ZRM3rHJFp5iwtgnJy2wrUIjwwXfXiN52wpDhN7FmN3iielg5Vz2L9JZb9JCn5PPVfnDZDaUtQqn1JfSQepyRdD0GpZ3PZtQ1BaReRtgv0FJTaapD1Vclnb9LRy39QdKUg8tI5hZ+M+sKG+wmoO2e9BWVmb8e5efecvUl8nFCeZ59mBzKDoGwVt42Sz6YdKH50dKN09bPXWpyR5aeUy8SZgZSUjfbuUZ/Xydeq0+XZu1fL5xy/tnTPeSFyQ/wLaLnFoWuqjF0vQha4mFVQqgWNdTyvQy9roosz+1vZffqGSOHIxZagnL/9DA1dfkB8vqxXNQovpr+gtP4Sm1FQpgnbrP7+lc8gKI13QghKSW1gdkEp6TVhJpsTsA5+12PJ2xmEZhCUbab/YokxOzamkW6ZntR81PGS6uwmzjB09z3qVJVmFZT8JarssJ/oz9RHtKZfTZHBy9NLZn8r6njRPUPqkm+B59K9qq3YRXewsK5v4p7zNGCJMkMJQemONpD1DAhKSSQhKCWBhBlDCVgLysGNXqUetUoaWid+uHKmoSsbJ1x9ibbxv9KOUzeEGb1zUyt13XHyOrWdsV38qI57c/Vd3HG/sszPzzKroOS68UwwHx/0cuF87sCi+zOy+vLFB4nzrGzRgnkz1INn0HvM3yN+/02XCKrphtSL/KysBGVSr2pUsZi2GcrQogVoXf9auvO1fgDG/qdEICgluR+cShJImDGUgLpzf+WF52nZ+9UcOh5Hr8rzQMibCt56xZ+ee1b77mqZ9Xln1nbadvy6MHlibONMY9hkPjOzw6hlPkMvW5XHrCPeWGR2QanX+xtl19nZ/K3H/kP/mL1DVFs5osod76BFUIYMXk1P9o9l+HKi3F+qaAFaC0HpjibL9BkQlJLwQ1BKAgkzhhJQOueI4CLEeXRRnhLoOHsHbTn2H/ELzujhjlhk9c5cW3FvZm4fzmDCZ2lCULq3lab9fJz+f80RmhSdceNNVjU5ce2/VGfCJnHJou5VqEqI/geb87OsBSXH9Q58suStzFC+MuRHuv/wsU1fgqB0r39l9TQISkltAUEpCSTMGEoAgjJz/O/N3Ukbj1wTF5z6rLFux8Ooa6DeSDGpbRi1DHf+mCB3O5aMDF/urnN2eR5noCmYN7fDrzN90wm6c+8BxTZ4xS3+bUtQqvNzKylOv91xhj5ZdoAqBPnQ8verp3svCEqHm1m3GyAoJaGFoJQEEmYMJaB0zpHBRSgBM5Tp2qLrvF20/vBVt8647Tlzg6K+StuZOzE6jFpV9BxBWXHUWuJQBS5mjqE09A8OD7e5g5tT2+bO5UUvFkqL9eTNUykX71BJ/wIZEgoofVZ4MR9a1iu92HQHXoz9TylDUEryODiVJJAwYygBpXOuElKEFnXHkre6Mf75zW5ae/CKWwUSH/fVKu4X8cwJbcIoyoWDzN3tWOEj19DNuw/cysvd74jnuU5A6XO8c+eiQ6MaOmxw9KqDtGzvBUroUYVCi7p/lz7GfghKh53W3g1wKnuE8LknEFA696ohvuLQcZSnBGLm76F/p1x2q0DiTCKRY9fT3fuPaNeQulTI2/FlTKPasNOcnbT5aFqIAGYojWoF8z9X6XP8CuSh3UPqOVVhnsHUK0ORvQph7IegtOcjDn8Op3IYGW4wIQGlc69W0pe++ycEpbqJ3v8umVb/fsntAonzHT98/JiedyImzkgX41g4ThnbvEIg1SvzgpFVwbNNTEDpc4r75qNNsbVNXFPbVcPYD0Ep3WnhVNKRwqABBCAoM4fee+Fe+v63i24XlAa4AR4JAm4joPQ5YUE+tMJqw43bKuHCgzD2Q1C64D74liIdHgyahoDSuVcP9aVvu2GGUt0w/RL2iVgtLljCNY3LoiIeToBzyq/67RIl9qrmkZmKICghKKX/CcKppCOFQQMINJi0mY5c+YOWxlSlSiWKGFAD8z7y8OU71HDyFgr08aZtH/2feSuKmoGAhxF48OixW1KZ6oEFYz8EpXS/glNJRwqDBhDgTSDX/5tKL/l4G/B08z/y8u2/yCdfbsqb25hsPeYnhBqCQM4igLEfglK6x8OppCOFQRAAARAAARAwNQGM/RCU0h0UTiUdKQyCAAiAAAiAgKkJYOyHoJTuoHAq6UhhEARAAARAAARMTQBjPwSldAeFU0lHCoMgAAIgAAIgYGoCGPshKKU7KJxKOlIYBAEQAAEQAAFTE8DYD0Ep3UHhVNKRwiAIgAAIgAAImJoAxn4ISukOCqeSjhQGQQAEQAAEQMDUBDD2Q1BKd1A4lXSkMAgCIAACIAACpiaAsR+CUrqDwqmkI4VBEAABEAABEDA1AYz9EJTSHRROJR0pDIIACIAACICAqQlg7IeglO6gcCrpSGEQBEAABEAABExNAGM/BKV0B4VTSUcKgyAAAiAAAiBgagIY+yEopTsonEo6UhgEARAAARAAAVMTwNgPQSndQeFU0pHCIAiAAAiAAAiYmgDGfghK6Q4Kp5KOFAZBAARAAARAwNQEMPZDUEp3UDiVdKQwCAIgAAIgAAKmJoCxH4JSuoPCqaQjhUEQAAEQAAEQMDUBjP0QlNIdFE4lHSkMggAIgAAIgICpCWDsh6CU7qBwKulIYRAEQAAEQAAETE0AYz8EpXQHhVNJRwqDIAACIAACIGBqAhj7ISilOyicSjpSGAQBEAABEAABUxPA2A9BKd1B4VTSkcIgCIAACIAACJiaAMZ+CErpDgqnko4UBkEABEAABEDA1AQw9kNQpnPQx48f04gRI2jWrFl08+ZNevPNNykuLo5CQ0M1OzKcSjMqXAgCIAACIAAC2YIAxn4IynSOzGKSBeTcuXMpMDCQBg0aRCdPnqSUlBTKkyePJqeHU2nChItAAARAAARAINsQwNgPQWkhkJqaSn5+fjR+/HiKiYkRv7916xa99NJLNGfOHGrXrp0mx4dTacKEi0AABEAABEAg2xDA2A9BaSGwc+dOioyMpCNHjlDp0qUtv+dl7/Lly4uZSy0FTqWFEq4BARAAARAAgexDAGM/BKWFQFJSEkVFRdHdu3fJ29vb8vvo6Gjxu1WrVmnyfDiVJky4CARAAARAAASyDQGM/RCUFgILFiygjh070qNHj8jLy8vy+06dOtHFixdp3bp1Nh3/zp07xP8p5dy5c1StWjXiGc+AgIBs88eCFwEBEAABEAABELBN4NKlSxQREUGnTp2iEiVK5GhMz/z9999/52QCiYmJ1Lp1a5szlPfv36cVK1bYxDN8+HCxMxwFBEAABEAABEAgZxPgyaTKlSvnaAg5XlAqMZTHjx+nkiVLWpyBYyjDwsJo2rRpmmYo//rrL+JZyuDgYHr22WelOpXyDQizn1Kx2jUG7nYR6XIBuOuC1a5RcLeLSLcLwF43tFkalsH94cOHdO3aNSpXrhzlzZvXmBcxyVNzvKDkWciiRYvSxIkTqWvXrqJZnNnlrWd7IkZDT7qZ2wZ3cDeGgDFPhb8bw52fCvbGsAd3udxzvKBknJ988gnFx8eLY4I4BiI2NpZOnz5N+/fv13wOpdxmSW8NTq8nXQhKY+iCO7ibjYBx9UEfbwx7cJfLHYKSSGzI+fjjj8XB5vfu3aOaNWuKpW6zBNjC6eU6vVZr4K6VlNzrwF0uT63WwF0rKfnXgb18plosgrsWStqvgaDUzsqwK3k3OS/J9+/fnwoWLGhYPXLag8HdmBYHd3A3hoBxT4XPG8Me3OVyh6CUyxPWQAAEQAAEQAAEQCDHEYCgzHFNjhcGARAAARAAARAAAbkEICjl8oQ1EAABEAABEAABEMhxBCAoc1yT44VBAARAAARAAARAQC4BCEq5PKVae/z4scjGM2vWLLp58ybxYetxcXEUGhoq9Tk5zdiNGzfErn7O085B2eXLl6dx48YJvlz27dtHffr0od27d5Ovry/17t2bBg4caMGEdnHdY44ePUoVK1akqVOn0nvvvQfuriPN0sI333wjfPzkyZMigQNn+mrTpg2468z9wYMHgvX8+fNFH16hQgX6/PPPRZpe9DX6wB89erRImbxx40bLA2T06fZs6PM2nmUVgtLE7cVikgUkH2cUGBhIgwYNEgNCSkqKKc7HNDG6LKtWv359unr1qhAz/v7+4ogoFu3Jycni51dffZVatGghdtVv376devXqJdqhc+fOwi7axbWW50GWB1QW7OzbLCivX78O7q5hzfTuBQsWCMYTJkygJk2a0MKFC4XI2bp1K5UuXRrcdeLOZocNG0YzZsygr7/+mkJCQoSYXLx4MR06dIiee+45sJfMfvLkyaLf5qP/FEEpo2/RYkPyq3ikOQhKkzZbamoq+fn50fjx4ykmJkbU0mwZfEyKLstqcYrNUqVK0bZt2yyzBJzOngfW9u3bk7e3txCYfLC9kkKTZzOTkpLo8OHDhHZxvdWZJwv1n3/+2SIoP/vsM3B3HW0GC+zbLGRat25NX3zxheXzBg0a0FtvvSV+hr/rAP6JSZ6RrFOnjhDzXHhFpFChQrR06VLiWXqwl8P+woUL1K1bN9qyZQsFBQXRCy+8YBGUMvoWezbkvIXnW4GgNGkbKjnGjxw5IsSOUnhZlpdoecYMxXECLMp/+eUX0cnzDIFSOIygYcOGYgbYx8eHvvvuO8tnvHxSr149unLlihCakZGRhHZxnD3fsXnzZmrZsqUIKyhWrJhFUDZu3BjcnUOa5V38Jei1114Ts+/h4eEZrgV3HaCrTHI/w4dn//TTT0LoTJo0iYYMGSJWmT788EP4vCT8K1euFDPvvNw9cuRI0U8rM5QyfNyeDU7fjEIEQWlSL+AZsaioKLp7966YNVNKdHS0+B3H/6HIIbBkyRJirt9//71Iw9moUSOxNKWUgwcP0uuvv067du2is2fPol2cxM5iPiwsjPjbfocOHeiZZ56xCEr+kgTuToLN4jb26ebNm4uZm1GjRtHevXspODhYiJpmzZqJL6fgLp+7YvHAgQPUtm1b4j4kV65c5OXlJWYn3377bbDXCTuHd6gFpQwft2ejUqVKOr2NZ5mFoDRpe3HcU8eOHUVaSO6ElNKpUye6ePGiCDpGcZ0AL33zzCTPJCxfvlxseGKxw99ylaJsZOBBmTsqtItz3DmkgJdgFy1aJAyoBSW4O8fU3l1KP8IikuP5WNAnJibSmDFjaO3atdSjRw/4uz2ILnzOX1anTJlCsbGxIg5+5syZlJCQIGbqW7VqBfYusM3sVmtBKaNvsWdD2dCpw+t4lEkISpM2F3f6HPdka4by/v37tGLFCpPW3HOqxQxZPFapUkXM+PJMcLly5YiXN2zNUPKyIYtLtIvjbcy7XAcPHkz79++nwoULZxCU4O44Uy138AYQniHjEJmePXtabuHNOVx4xh3+roWk49ecOXNGxGuvX7+eatSoYTHA/+b4eI7nBnvHudq7w1pQyuhb7NmwFU5ir57Z8XMISpO2qhJDyZ0OH/OhFP4mxLMMHMyN4jwB3uHNRwPxLAHP4ijxlNzB81FBLICUYiuGEu3iGPvatWuLXcXquNU///xT/Mz+Xbx4cXB3DKmmqzleuHr16mLJWz2LwidG8JeoEiVKgLsmko5fpITS3Lt3j/LmzWsxwLOVP/zwA3zecaSa7rAWlDL6dHs2EEOZ1jQQlJpc1P0X8SwkO+nEiROpa9euogLY5S2nHb766itxFBCfL8lB8uqQAj6rb/r06XTixAkR88SFZ9aWLVsmdnmjXZxrA96FyQOruvDsDc8E8wwaB9SDu3Nss7qLmfOO108//TTdWaocN5k7d25x0gG4y+fOFvkkg6pVq9KOHTsoIiLC8hAOsWGBySsjYC+fvbWglNGn27Mh/y080yIEpYnbjTeIxMfH05w5c8RMAn+z5Rg+XjbMkyePiWtu3qrxUR1ly5alpk2bZtgpz0veLBj5HEoOmudZHJ4p5qVC7vjfffdd8WJoFzntq46h5HNBwV0OV2srvPOVjx/jvoSFDcewssDkpdgyZcqAuz7YiRMg8NFM165dE33Nyy+/THzAPG9K49l6Ps4JPi8fvrWglNG3aLEh/008zyIEpYnbjDfk8Jl9fPgzzzTwYa281M3iEsU5AmPHjhWC0FZhwThv3jyxm5tnL3lHbEBAAA0YMIA++OADyy1oF+fYW9+lFpT8GbjL4WrLCq90cJgHzxTzMUJ8OD/v/gZ3/ZizZc6OwzvqObyA/82xeNwH1apVC+x1Qm8tKGX5uL3+SafX8SizEJQe1VyoLAiAAAiAAAiAAAiYjwAEpfnaBDUCARAAARAAARAAAY8iAEHpUc2FyoIACIAACIAACICA+QhAUJqvTVAjEAABEAABEAABEPAoAhCUHtVcqCwIgAAIgAAIgAAImI8ABKX52gQ1AgEQMCkBPmGBd5EOHz5clxpu3LiR+BD4U6dOZXqag9510OXFYBQEQCDbE4CgzPZNjBcEARCQRYDPFOTzSgsUKCDLZDo7qampdCvy0FMAAAPmSURBVOPGDfL397ccrG/9IAhKXdDDKAiAgIsEIChdBIjbQQAEQMCdBCAo3UkbzwIBENBKAIJSKylcBwIg4DEE+NB0PsCec7Jz9qNNmzZRcHAwDR06VORuv337tsiYNHLkSKpfv77lvdasWSNSbR48eFBkMuFD7Tn1qbIEbS3mVq9eTaNGjaIDBw5QwYIFqX379jRmzBhL7maux4wZM2jx4sUiO0qRIkXEIfn8DFvFesmb68mH7K9YsUJkx+JEB5MnT9Z12d1jGhkVBQEQMBUBCEpTNQcqAwIgIIMACzk/Pz/68ccf6eHDhyJvcocOHSglJUUIMk6Dt3LlSvroo49EnvYmTZrQvn37qHLlytSvXz8hIvlnTrvJGU5sCcrly5dTVFSUiKeMjo6mY8eOies5vWFiYqJ4Da5HoUKFRJYa/j0LXE6FuHnzZqpRo0aGV7UWlA0aNKCzZ88KUfr8888LgbthwwYaNmyYbnGcMvjDBgiAQM4jAEGZ89ocbwwC2Z4ACzkWhpxykMvx48epVKlSIr1jpUqVLO/P6TbPnDlDLOT435zr/ddff7V8/uWXX4oZQluCMjIykoKCgmjp0qWW61mkch54nuHkFIdcjz59+ggRqxSepYyNjbU5S6kWlEpe+XXr1lGdOnXE7VeuXKHixYsLIazXxqBs7xx4QRAAAV0IQFDqghVGQQAEjCTAQi4uLk7MGHJZsmSJmEXMnz9/umo9ePCAfHx8hFDjPMv16tWziFC+kGcpw8PDbQrKfPnyibzMffv2tdjkJWq2t2jRImrbtq0QlNOnT6cePXpYruFl886dO4tZRuuiFpS7d++mNm3aiOV5Xk5XSpkyZcS7QFAa6WF4NgiAgDUBCEr4BAiAQLYjwEJu7ty5ItaQS0JCArVr146Sk5PF0rG65MqVS8RXsnCsVatWutnErAQl7/YeN26cmIFUCu/Q9vX1FUverVq1EoJSXQ++LqtNNbYEJS+5s0hVSlhYGLVs2RKCMtt5LV4IBDybAASlZ7cfag8CIGCDgLWQ49hJ3oSzatUqES+pFN644+XlJTbWsPg8fPgwbd++3fJ5fHw8xcTE2Jyh5JjIYsWKpVvy5s0zLVq0SLfk7aygfPToEYWGhqar861btygwMFAsmWOGEq4PAiBgJgIQlGZqDdQFBEBACgFbM4PNmjUTS9jTpk0T4jIpKYkGDRpEs2fPFkvQvFO7QoUKNHDgQOrSpYsQhd27dyc+e/L06dMidlE9u8jL6LysPWLECPF/jr/kJXa2wbGUXFyZoeRn8dI2x33OnDmTXnzxRRF3yRuNhgwZAkEpxVNgBARAQBaB/wFJqX5WR53H3gAAAABJRU5ErkJggg==\" width=\"599.9999869953505\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pb_num_dict = {_k:len(_v.kept_probes) for _k,_v in filtered_pb_dict.items()}\n",
    "\n",
    "pb_nums = np.array(list(pb_num_dict.values()))\n",
    "plt.figure(figsize=(6,3), dpi=100)\n",
    "plt.plot(pb_nums)\n",
    "#plt.plot(np.ones(len(pb_nums))*20)\n",
    "plt.ylabel(f\"num of probes\", labelpad=1)\n",
    "plt.xlabel(f\"region id\", labelpad=1)\n",
    "plt.ylim([0, num_probes_kept*1.1])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "54167bca",
   "metadata": {},
   "outputs": [],
   "source": [
    "filtered_reg_names = {}\n",
    "for _reg, _pbde in filtered_pb_dict.items():\n",
    "    if len(_pbde.kept_probes) > 0:\n",
    "        filtered_reg_names[_reg] = list(_pbde.kept_probes.values())[0]['reg_name']\n",
    "    \n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "id": "dd982543",
   "metadata": {},
   "outputs": [],
   "source": [
    "chr_2_region_num = {}\n",
    "for _reg, _info in filtered_reg_names.items():\n",
    "    _chr = _info.split('_')[0].split(':')[0]\n",
    "    if _chr not in chr_2_region_num:\n",
    "        chr_2_region_num[_chr] = 1\n",
    "    else:\n",
    "        chr_2_region_num[_chr] += 1\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "id": "c90e35f5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "id": "3a9d87f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "chr_2_region_names = {}\n",
    "for _reg, _info in filtered_reg_names.items():\n",
    "    _chr = _info.split('_')[0].split(':')[0]\n",
    "    if _chr not in chr_2_region_names:\n",
    "        chr_2_region_names[_chr] = [_reg]\n",
    "    else:\n",
    "        chr_2_region_names[_chr].append(_reg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "42b3a3cb",
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "pickle.dump(chr_2_region_names, open(os.path.join(library_folder, 'chr_2_region_names.pkl'), 'wb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "678e67ba",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bd009a6e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "79b6ce50",
   "metadata": {},
   "source": [
    "# 3. Load and asssemble probes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "973463cd",
   "metadata": {},
   "source": [
    "## 3.1 design readout mapping dict (region_to_readout)\n",
    "\n",
    "Prepare files below using the notebook\n",
    "\n",
    "[1_library_design_ctp11](1_library_design_ctp11.ipynb)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f23efd57",
   "metadata": {},
   "source": [
    "## 3.2 loading components"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6ae1aaf5",
   "metadata": {},
   "source": [
    "### primers\n",
    "pick primer set 2,9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4758090d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Picked primer: ID: W1A03_primer_2\n",
      "Name: W1A03_primer_2\n",
      "Description: W1A03_primer_2\n",
      "Number of features: 0\n",
      "Seq('CCCGCAATGGCTGACAACCG')\n",
      "- Picked primer: ID: W1A10_primer_9\n",
      "Name: W1A10_primer_9\n",
      "Description: W1A10_primer_9\n",
      "Number of features: 0\n",
      "Seq('TAATACGACTCACTATAGGGATTGCCGCATGGTTTCCG')\n"
     ]
    }
   ],
   "source": [
    "from ImageAnalysis3 import library_tools\n",
    "primer_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Primers'\n",
    "primers = library_tools.assemble.load_primers([2,9], _primer_folder=primer_folder)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce11e3c2",
   "metadata": {},
   "source": [
    "### readouts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "92f9cee7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# summarize total readout usage\n",
    "region_2_readout_dict = pickle.load(open(os.path.join(library_folder, 'region_2_readout.pkl'), 'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3a8ff3a6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{1: ['c36', 'c62', 'c77'],\n",
       " 2: ['c17', 'c42', 'c86'],\n",
       " 3: ['c34', 'c72', 'c82'],\n",
       " 4: ['c57', 'c58', 'c94'],\n",
       " 5: ['c41', 'c78', 'c93'],\n",
       " 6: ['c61', 'c65', 'c80'],\n",
       " 7: ['c66', 'c74', 'c83'],\n",
       " 8: ['c23', 'c44', 'c54'],\n",
       " 9: ['c4', 'c45', 'c90'],\n",
       " 10: ['c51', 'c64', 'c75'],\n",
       " 11: ['c12', 'c67', 'c69'],\n",
       " 12: ['c15', 'c43', 'c68'],\n",
       " 13: ['c30', 'c31', 'c62'],\n",
       " 14: ['c10', 'c22', 'c47'],\n",
       " 15: ['c5', 'c6', 'c13'],\n",
       " 16: ['c3', 'c35', 'c49'],\n",
       " 17: ['c39', 'c84', 'c88'],\n",
       " 18: ['c8', 'c20', 'c56'],\n",
       " 19: ['c24', 'c32', 'c55'],\n",
       " 20: ['c18', 'c29', 'c73'],\n",
       " 21: ['c53', 'c89', 'c91'],\n",
       " 22: ['c7', 'c14', 'c63'],\n",
       " 23: ['c0', 'c37', 'c81'],\n",
       " 24: ['c70', 'c95', 'c97'],\n",
       " 25: ['c26', 'c28', 'c60'],\n",
       " 26: ['c11', 'c68', 'c74'],\n",
       " 27: ['c21', 'c56', 'c85'],\n",
       " 28: ['c54', 'c83', 'c87'],\n",
       " 29: ['c2', 'c76', 'c79'],\n",
       " 30: ['c48', 'c51', 'c96'],\n",
       " 31: ['c8', 'c44', 'c65'],\n",
       " 32: ['c1', 'c19', 'c53'],\n",
       " 33: ['c5', 'c27', 'c33'],\n",
       " 34: ['c67', 'c92', 'c98'],\n",
       " 35: ['c10', 'c52', 'c58'],\n",
       " 36: ['c12', 'c34', 'c42'],\n",
       " 37: ['c4', 'c38', 'c77'],\n",
       " 38: ['c40', 'c59', 'c90'],\n",
       " 39: ['c29', 'c37', 'c57'],\n",
       " 40: ['c78', 'c86', 'c94'],\n",
       " 41: ['c17', 'c35', 'c80'],\n",
       " 42: ['c72', 'c84', 'c96'],\n",
       " 43: ['c9', 'c13', 'c88'],\n",
       " 44: ['c47', 'c48', 'c64'],\n",
       " 45: ['c16', 'c50', 'c71'],\n",
       " 46: ['c2', 'c25', 'c30'],\n",
       " 47: ['c11', 'c62', 'c65'],\n",
       " 48: ['c36', 'c73', 'c93'],\n",
       " 49: ['c43', 'c55', 'c75'],\n",
       " 50: ['c45', 'c46', 'c89'],\n",
       " 51: ['c14', 'c19', 'c26'],\n",
       " 52: ['c50', 'c59', 'c92'],\n",
       " 53: ['c39', 'c87', 'c96'],\n",
       " 54: ['c3', 'c25', 'c40'],\n",
       " 55: ['c20', 'c49', 'c79'],\n",
       " 56: ['c7', 'c27', 'c61'],\n",
       " 57: ['c15', 'c32', 'c48'],\n",
       " 58: ['c41', 'c81', 'c85'],\n",
       " 59: ['c21', 'c60', 'c83'],\n",
       " 60: ['c69', 'c70', 'c82'],\n",
       " 61: ['c24', 'c33', 'c58'],\n",
       " 62: ['c9', 'c18', 'c42'],\n",
       " 63: ['c28', 'c31', 'c68'],\n",
       " 64: ['c23', 'c36', 'c63'],\n",
       " 65: ['c44', 'c66', 'c97'],\n",
       " 66: ['c0', 'c80', 'c95'],\n",
       " 67: ['c22', 'c71', 'c77'],\n",
       " 68: ['c1', 'c38', 'c84'],\n",
       " 69: ['c6', 'c16', 'c47'],\n",
       " 70: ['c12', 'c87', 'c91'],\n",
       " 71: ['c17', 'c34', 'c52'],\n",
       " 72: ['c8', 'c54', 'c98'],\n",
       " 73: ['c4', 'c35', 'c46'],\n",
       " 74: ['c11', 'c76', 'c78'],\n",
       " 75: ['c64', 'c72', 'c74'],\n",
       " 76: ['c27', 'c41', 'c86'],\n",
       " 77: ['c13', 'c82', 'c94'],\n",
       " 80: ['c23', 'c73', 'c91'],\n",
       " 81: ['c10', 'c33', 'c40'],\n",
       " 82: ['c7', 'c22', 'c30'],\n",
       " 83: ['c15', 'c24', 'c88'],\n",
       " 84: ['c3', 'c14', 'c20'],\n",
       " 85: ['c19', 'c28', 'c37'],\n",
       " 86: ['c1', 'c57', 'c69'],\n",
       " 87: ['c16', 'c18', 'c31'],\n",
       " 88: ['c43', 'c76', 'c93'],\n",
       " 89: ['c21', 'c26', 'c63'],\n",
       " 90: ['c44', 'c47', 'c83'],\n",
       " 91: ['c34', 'c55', 'c74'],\n",
       " 92: ['c35', 'c53', 'c58'],\n",
       " 93: ['c51', 'c86', 'c98'],\n",
       " 94: ['c22', 'c82', 'c87'],\n",
       " 95: ['c56', 'c62', 'c66'],\n",
       " 96: ['c8', 'c79', 'c81'],\n",
       " 97: ['c45', 'c61', 'c68'],\n",
       " 98: ['c0', 'c29', 'c50'],\n",
       " 99: ['c4', 'c16', 'c80'],\n",
       " 100: ['c71', 'c85', 'c92'],\n",
       " 101: ['c6', 'c49', 'c95'],\n",
       " 102: ['c25', 'c38', 'c60'],\n",
       " 103: ['c11', 'c77', 'c93'],\n",
       " 104: ['c17', 'c32', 'c75'],\n",
       " 105: ['c42', 'c58', 'c70'],\n",
       " 106: ['c21', 'c64', 'c82'],\n",
       " 107: ['c9', 'c12', 'c54'],\n",
       " 108: ['c67', 'c72', 'c86'],\n",
       " 109: ['c5', 'c78', 'c84'],\n",
       " 110: ['c36', 'c39', 'c80'],\n",
       " 111: ['c2', 'c43', 'c59'],\n",
       " 112: ['c40', 'c74', 'c97'],\n",
       " 113: ['c20', 'c47', 'c89'],\n",
       " 114: ['c13', 'c48', 'c62'],\n",
       " 115: ['c44', 'c63', 'c94'],\n",
       " 116: ['c52', 'c90', 'c93'],\n",
       " 117: ['c34', 'c65', 'c96'],\n",
       " 118: ['c8', 'c35', 'c68'],\n",
       " 119: ['c4', 'c27', 'c87'],\n",
       " 120: ['c37', 'c46', 'c83'],\n",
       " 121: ['c32', 'c41', 'c95'],\n",
       " 122: ['c14', 'c15', 'c77'],\n",
       " 123: ['c1', 'c33', 'c88'],\n",
       " 124: ['c3', 'c67', 'c81'],\n",
       " 125: ['c7', 'c23', 'c51'],\n",
       " 126: ['c31', 'c45', 'c71'],\n",
       " 127: ['c9', 'c38', 'c56'],\n",
       " 128: ['c19', 'c48', 'c78'],\n",
       " 129: ['c13', 'c57', 'c73'],\n",
       " 130: ['c24', 'c63', 'c91'],\n",
       " 131: ['c2', 'c20', 'c40'],\n",
       " 132: ['c53', 'c61', 'c85'],\n",
       " 133: ['c29', 'c70', 'c79'],\n",
       " 134: ['c66', 'c69', 'c98'],\n",
       " 135: ['c39', 'c49', 'c59'],\n",
       " 136: ['c6', 'c28', 'c75'],\n",
       " 137: ['c46', 'c76', 'c97'],\n",
       " 138: ['c12', 'c26', 'c89'],\n",
       " 140: ['c17', 'c41', 'c84'],\n",
       " 141: ['c27', 'c36', 'c72'],\n",
       " 142: ['c42', 'c94', 'c96'],\n",
       " 143: ['c54', 'c64', 'c65'],\n",
       " 144: ['c10', 'c16', 'c25'],\n",
       " 145: ['c18', 'c22', 'c92'],\n",
       " 146: ['c5', 'c19', 'c90'],\n",
       " 147: ['c52', 'c55', 'c73'],\n",
       " 148: ['c30', 'c60', 'c91'],\n",
       " 150: ['c0', 'c11', 'c57'],\n",
       " 151: ['c2', 'c24', 'c50'],\n",
       " 153: ['c41', 'c42', 'c48'],\n",
       " 154: ['c2', 'c62', 'c63'],\n",
       " 155: ['c8', 'c26', 'c85'],\n",
       " 156: ['c56', 'c70', 'c72'],\n",
       " 157: ['c69', 'c79', 'c89'],\n",
       " 158: ['c9', 'c59', 'c67'],\n",
       " 159: ['c5', 'c50', 'c81'],\n",
       " 160: ['c3', 'c6', 'c31'],\n",
       " 161: ['c0', 'c53', 'c55'],\n",
       " 162: ['c12', 'c68', 'c96'],\n",
       " 163: ['c57', 'c74', 'c78'],\n",
       " 164: ['c40', 'c65', 'c77'],\n",
       " 165: ['c13', 'c24', 'c52'],\n",
       " 166: ['c30', 'c35', 'c84'],\n",
       " 167: ['c25', 'c44', 'c85'],\n",
       " 168: ['c22', 'c60', 'c70'],\n",
       " 169: ['c15', 'c17', 'c33'],\n",
       " 170: ['c51', 'c56', 'c71'],\n",
       " 171: ['c88', 'c89', 'c92'],\n",
       " 172: ['c29', 'c45', 'c49'],\n",
       " 173: ['c30', 'c97', 'c98'],\n",
       " 174: ['c18', 'c26', 'c32'],\n",
       " 175: ['c7', 'c43', 'c79'],\n",
       " 176: ['c5', 'c21', 'c76'],\n",
       " 177: ['c0', 'c6', 'c61'],\n",
       " 178: ['c9', 'c53', 'c81'],\n",
       " 179: ['c38', 'c46', 'c95'],\n",
       " 180: ['c66', 'c75', 'c90'],\n",
       " 181: ['c28', 'c59', 'c98'],\n",
       " 182: ['c1', 'c3', 'c10'],\n",
       " 183: ['c14', 'c23', 'c58'],\n",
       " 184: ['c36', 'c37', 'c86'],\n",
       " 185: ['c2', 'c73', 'c92'],\n",
       " 186: ['c16', 'c19', 'c33'],\n",
       " 187: ['c11', 'c48', 'c60'],\n",
       " 188: ['c39', 'c52', 'c95'],\n",
       " 190: ['c4', 'c82', 'c83'],\n",
       " 191: ['c47', 'c62', 'c91'],\n",
       " 192: ['c42', 'c54', 'c93'],\n",
       " 193: ['c41', 'c50', 'c64'],\n",
       " 195: ['c27', 'c34', 'c63'],\n",
       " 196: ['c20', 'c72', 'c94'],\n",
       " 197: ['c8', 'c80', 'c87'],\n",
       " 198: ['c37', 'c67', 'c97'],\n",
       " 199: ['c23', 'c31', 'c55'],\n",
       " 200: ['c14', 'c39', 'c69'],\n",
       " 201: ['c1', 'c71', 'c76'],\n",
       " 202: ['c51', 'c61', 'c90'],\n",
       " 203: ['c7', 'c29', 'c32'],\n",
       " 204: ['c10', 'c21', 'c75'],\n",
       " 205: ['c15', 'c28', 'c49'],\n",
       " 206: ['c18', 'c25', 'c46'],\n",
       " 207: ['c38', 'c43', 'c45'],\n",
       " 208: ['c17', 'c57', 'c66'],\n",
       " 209: ['c12', 'c22', 'c88'],\n",
       " 210: ['c16', 'c78', 'c91'],\n",
       " 211: ['c77', 'c86', 'c96'],\n",
       " 212: ['c35', 'c40', 'c83'],\n",
       " 213: ['c68', 'c84', 'c93'],\n",
       " 214: ['c4', 'c24', 'c65'],\n",
       " 215: ['c13', 'c44', 'c74'],\n",
       " 218: ['c7', 'c52', 'c81'],\n",
       " 219: ['c11', 'c79', 'c98'],\n",
       " 220: ['c13', 'c33', 'c65'],\n",
       " 221: ['c12', 'c17', 'c30'],\n",
       " 222: ['c5', 'c55', 'c67'],\n",
       " 223: ['c54', 'c90', 'c94'],\n",
       " 224: ['c36', 'c43', 'c71'],\n",
       " 225: ['c38', 'c58', 'c88'],\n",
       " 226: ['c34', 'c37', 'c50'],\n",
       " 227: ['c20', 'c76', 'c80'],\n",
       " 228: ['c29', 'c53', 'c92'],\n",
       " 229: ['c15', 'c26', 'c97'],\n",
       " 230: ['c19', 'c49', 'c70'],\n",
       " 231: ['c9', 'c47', 'c75'],\n",
       " 232: ['c32', 'c61', 'c66'],\n",
       " 233: ['c10', 'c28', 'c82'],\n",
       " 234: ['c45', 'c64', 'c98'],\n",
       " 235: ['c0', 'c1', 'c56'],\n",
       " 236: ['c21', 'c59', 'c79'],\n",
       " 237: ['c52', 'c60', 'c69'],\n",
       " 238: ['c14', 'c87', 'c95'],\n",
       " 239: ['c3', 'c18', 'c33'],\n",
       " 240: ['c7', 'c25', 'c31'],\n",
       " 242: ['c27', 'c39', 'c51'],\n",
       " 243: ['c40', 'c49', 'c96'],\n",
       " 244: ['c50', 'c63', 'c83'],\n",
       " 245: ['c16', 'c46', 'c48'],\n",
       " 246: ['c2', 'c23', 'c93'],\n",
       " 247: ['c22', 'c73', 'c85'],\n",
       " 248: ['c72', 'c81', 'c89'],\n",
       " 249: ['c6', 'c11', 'c35'],\n",
       " 250: ['c26', 'c68', 'c78'],\n",
       " 251: ['c42', 'c44', 'c91'],\n",
       " 252: ['c4', 'c74', 'c86'],\n",
       " 253: ['c57', 'c77', 'c84'],\n",
       " 254: ['c8', 'c24', 'c41'],\n",
       " 255: ['c20', 'c62', 'c92'],\n",
       " 256: ['c15', 'c30', 'c34'],\n",
       " 257: ['c10', 'c56', 'c69'],\n",
       " 258: ['c39', 'c67', 'c70'],\n",
       " 259: ['c19', 'c45', 'c66'],\n",
       " 260: ['c0', 'c18', 'c88'],\n",
       " 261: ['c14', 'c43', 'c64'],\n",
       " 262: ['c27', 'c28', 'c38'],\n",
       " 263: ['c58', 'c60', 'c95'],\n",
       " 264: ['c47', 'c76', 'c82'],\n",
       " 266: ['c46', 'c51', 'c94'],\n",
       " 267: ['c21', 'c37', 'c73'],\n",
       " 268: ['c6', 'c71', 'c80'],\n",
       " 269: ['c59', 'c61', 'c75'],\n",
       " 270: ['c5', 'c23', 'c29'],\n",
       " 271: ['c3', 'c53', 'c97'],\n",
       " 272: ['c31', 'c32', 'c90'],\n",
       " 273: ['c13', 'c25', 'c54'],\n",
       " 274: ['c1', 'c89', 'c93'],\n",
       " 276: ['c9', 'c55', 'c62'],\n",
       " 277: ['c17', 'c36', 'c87'],\n",
       " 278: ['c12', 'c40', 'c85'],\n",
       " 279: ['c16', 'c65', 'c83'],\n",
       " 281: ['c37', 'c53', 'c76'],\n",
       " 282: ['c9', 'c60', 'c72'],\n",
       " 283: ['c30', 'c42', 'c63'],\n",
       " 285: ['c23', 'c27', 'c85'],\n",
       " 286: ['c3', 'c38', 'c64'],\n",
       " 287: ['c34', 'c47', 'c56'],\n",
       " 288: ['c55', 'c90', 'c98'],\n",
       " 289: ['c74', 'c84', 'c95'],\n",
       " 290: ['c32', 'c86', 'c88'],\n",
       " 291: ['c6', 'c33', 'c51'],\n",
       " 292: ['c41', 'c52', 'c68'],\n",
       " 293: ['c2', 'c70', 'c81'],\n",
       " 294: ['c26', 'c43', 'c91'],\n",
       " 295: ['c22', 'c58', 'c69'],\n",
       " 296: ['c15', 'c46', 'c63'],\n",
       " 297: ['c11', 'c30', 'c61'],\n",
       " 298: ['c48', 'c54', 'c57'],\n",
       " 299: ['c10', 'c12', 'c66'],\n",
       " 300: ['c25', 'c80', 'c83'],\n",
       " 301: ['c50', 'c84', 'c90'],\n",
       " 302: ['c9', 'c36', 'c65'],\n",
       " 303: ['c18', 'c49', 'c76'],\n",
       " 304: ['c3', 'c86', 'c89'],\n",
       " 305: ['c60', 'c79', 'c87'],\n",
       " 306: ['c32', 'c35', 'c38'],\n",
       " 307: ['c5', 'c24', 'c72'],\n",
       " 308: ['c0', 'c19', 'c92'],\n",
       " 309: ['c1', 'c59', 'c82'],\n",
       " 310: ['c8', 'c51', 'c91'],\n",
       " 311: ['c20', 'c73', 'c97'],\n",
       " 312: ['c27', 'c71', 'c94'],\n",
       " 313: ['c21', 'c25', 'c45'],\n",
       " 314: ['c31', 'c42', 'c75'],\n",
       " 315: ['c28', 'c50', 'c78'],\n",
       " 316: ['c16', 'c40', 'c55'],\n",
       " 318: ['c62', 'c93', 'c96'],\n",
       " 319: ['c14', 'c44', 'c67'],\n",
       " 320: ['c4', 'c29', 'c89'],\n",
       " 321: ['c13', 'c17', 'c39'],\n",
       " 322: ['c7', 'c53', 'c66'],\n",
       " 323: ['c10', 'c35', 'c37'],\n",
       " 324: ['c8', 'c69', 'c77'],\n",
       " 325: ['c21', 'c88', 'c98'],\n",
       " 326: ['c18', 'c74', 'c85'],\n",
       " 327: ['c26', 'c41', 'c61'],\n",
       " 328: ['c19', 'c20', 'c43'],\n",
       " 329: ['c33', 'c54', 'c78'],\n",
       " 330: ['c6', 'c22', 'c68'],\n",
       " 331: ['c2', 'c57', 'c82'],\n",
       " 332: ['c34', 'c87', 'c92'],\n",
       " 333: ['c11', 'c71', 'c75'],\n",
       " 334: ['c1', 'c23', 'c81'],\n",
       " 335: ['c15', 'c36', 'c56'],\n",
       " 336: ['c44', 'c48', 'c95'],\n",
       " 337: ['c46', 'c47', 'c70'],\n",
       " 338: ['c5', 'c64', 'c73'],\n",
       " 339: ['c7', 'c58', 'c96'],\n",
       " 340: ['c52', 'c77', 'c80'],\n",
       " 342: ['c30', 'c88', 'c90'],\n",
       " 343: ['c13', 'c22', 'c42'],\n",
       " 344: ['c4', 'c39', 'c60'],\n",
       " 345: ['c16', 'c28', 'c94'],\n",
       " 346: ['c24', 'c27', 'c97'],\n",
       " 347: ['c12', 'c64', 'c83'],\n",
       " 348: ['c44', 'c55', 'c58'],\n",
       " 349: ['c3', 'c15', 'c71'],\n",
       " 350: ['c46', 'c54', 'c85'],\n",
       " 351: ['c35', 'c61', 'c63'],\n",
       " 352: ['c26', 'c47', 'c95'],\n",
       " 353: ['c6', 'c88', 'c91'],\n",
       " 354: ['c5', 'c52', 'c57'],\n",
       " 355: ['c13', 'c68', 'c87'],\n",
       " 356: ['c0', 'c2', 'c31'],\n",
       " 357: ['c23', 'c67', 'c94'],\n",
       " 358: ['c29', 'c59', 'c60'],\n",
       " 359: ['c14', 'c21', 'c24'],\n",
       " 360: ['c28', 'c39', 'c79'],\n",
       " 361: ['c34', 'c45', 'c84'],\n",
       " 362: ['c1', 'c80', 'c86'],\n",
       " 363: ['c10', 'c38', 'c41'],\n",
       " 364: ['c50', 'c56', 'c77'],\n",
       " 365: ['c32', 'c42', 'c78'],\n",
       " 366: ['c55', 'c65', 'c95'],\n",
       " 367: ['c8', 'c33', 'c48'],\n",
       " 368: ['c25', 'c66', 'c72'],\n",
       " 369: ['c18', 'c20', 'c37'],\n",
       " 370: ['c4', 'c17', 'c49'],\n",
       " 371: ['c27', 'c53', 'c93'],\n",
       " 372: ['c7', 'c73', 'c89'],\n",
       " 373: ['c9', 'c74', 'c92'],\n",
       " 374: ['c11', 'c82', 'c90'],\n",
       " 375: ['c36', 'c70', 'c76'],\n",
       " 376: ['c16', 'c51', 'c69'],\n",
       " 377: ['c19', 'c81', 'c96'],\n",
       " 378: ['c18', 'c43', 'c62'],\n",
       " 379: ['c22', 'c45', 'c97'],\n",
       " 380: ['c0', 'c65', 'c98'],\n",
       " 381: ['c14', 'c30', 'c75'],\n",
       " 382: ['c29', 'c40', 'c67'],\n",
       " 383: ['c15', 'c31', 'c59'],\n",
       " 384: ['c52', 'c71', 'c79'],\n",
       " 385: ['c1', 'c49', 'c61'],\n",
       " 386: ['c7', 'c57', 'c64'],\n",
       " 387: ['c23', 'c34', 'c38'],\n",
       " 388: ['c26', 'c50', 'c54'],\n",
       " 389: ['c21', 'c72', 'c80'],\n",
       " 390: ['c46', 'c92', 'c96'],\n",
       " 391: ['c44', 'c77', 'c98'],\n",
       " 392: ['c2', 'c5', 'c8'],\n",
       " 393: ['c25', 'c58', 'c68'],\n",
       " 395: ['c3', 'c63', 'c85'],\n",
       " 396: ['c35', 'c75', 'c78'],\n",
       " 397: ['c6', 'c37', 'c40'],\n",
       " 398: ['c33', 'c62', 'c87'],\n",
       " 399: ['c41', 'c83', 'c91'],\n",
       " 400: ['c12', 'c36', 'c47'],\n",
       " 995: ['c34', 'c71', 'c98'],\n",
       " 997: ['c6', 'c14', 'c62'],\n",
       " 998: ['c0', 'c86', 'c97'],\n",
       " 999: ['c2', 'c13', 'c29'],\n",
       " 1000: ['c49', 'c75', 'c81'],\n",
       " 1001: ['c9', 'c41', 'c79'],\n",
       " 1002: ['c28', 'c42', 'c53'],\n",
       " 1007: ['c4', 'c66', 'c73'],\n",
       " 1008: ['c19', 'c32', 'c56'],\n",
       " 1009: ['c48', 'c67', 'c82'],\n",
       " 1010: ['c20', 'c45', 'c69'],\n",
       " 1011: ['c31', 'c70', 'c93'],\n",
       " 1013: ['c10', 'c84', 'c94'],\n",
       " 1014: ['c74', 'c76', 'c89'],\n",
       " 1015: ['c3', 'c43', 'c51'],\n",
       " 1016: ['c17', 'c40', 'c58'],\n",
       " 1017: ['c54', 'c59', 'c63'],\n",
       " 1018: ['c33', 'c50', 'c55'],\n",
       " 1019: ['c85', 'c90', 'c91'],\n",
       " 1020: ['c8', 'c25', 'c64'],\n",
       " 1021: ['c27', 'c68', 'c92'],\n",
       " 1022: ['c30', 'c37', 'c95'],\n",
       " 1023: ['c6', 'c57', 'c60'],\n",
       " 1024: ['c21', 'c44', 'c71'],\n",
       " 1025: ['c19', 'c24', 'c84'],\n",
       " 1026: ['c16', 'c66', 'c93'],\n",
       " 1027: ['c5', 'c34', 'c36'],\n",
       " 1028: ['c11', 'c22', 'c39'],\n",
       " 1029: ['c28', 'c35', 'c77'],\n",
       " 1030: ['c47', 'c61', 'c78'],\n",
       " 1031: ['c38', 'c52', 'c65'],\n",
       " 1032: ['c46', 'c62', 'c88'],\n",
       " 1033: ['c12', 'c15', 'c80'],\n",
       " 1034: ['c1', 'c72', 'c98'],\n",
       " 1035: ['c23', 'c26', 'c83'],\n",
       " 1036: ['c7', 'c18', 'c87'],\n",
       " 1038: ['c14', 'c17', 'c97'],\n",
       " 1039: ['c24', 'c43', 'c86'],\n",
       " 1040: ['c0', 'c51', 'c59'],\n",
       " 1041: ['c16', 'c53', 'c96'],\n",
       " 1044: ['c29', 'c76', 'c94'],\n",
       " 1045: ['c9', 'c11', 'c32'],\n",
       " 1046: ['c20', 'c48', 'c74'],\n",
       " 1047: ['c45', 'c67', 'c79'],\n",
       " 1048: ['c49', 'c64', 'c89'],\n",
       " 1049: ['c10', 'c44', 'c73'],\n",
       " 1050: ['c4', 'c40', 'c70'],\n",
       " 1051: ['c31', 'c82', 'c92'],\n",
       " 1053: ['c69', 'c81', 'c90'],\n",
       " 1054: ['c39', 'c56', 'c75'],\n",
       " 1055: ['c12', 'c25', 'c41'],\n",
       " 1056: ['c13', 'c47', 'c55'],\n",
       " 1057: ['c2', 'c85', 'c96'],\n",
       " 1058: ['c27', 'c54', 'c95'],\n",
       " 1059: ['c5', 'c42', 'c65'],\n",
       " 1060: ['c30', 'c33', 'c57'],\n",
       " 563: ['c32', 'c65', 'c68'],\n",
       " 564: ['c21', 'c69', 'c74'],\n",
       " 565: ['c36', 'c46', 'c79'],\n",
       " 566: ['c35', 'c52', 'c89'],\n",
       " 567: ['c17', 'c31', 'c76'],\n",
       " 568: ['c18', 'c66', 'c82'],\n",
       " 569: ['c8', 'c11', 'c50'],\n",
       " 570: ['c48', 'c58', 'c83'],\n",
       " 571: ['c38', 'c53', 'c72'],\n",
       " 572: ['c56', 'c92', 'c93'],\n",
       " 573: ['c3', 'c9', 'c19'],\n",
       " 574: ['c22', 'c43', 'c81'],\n",
       " 575: ['c7', 'c59', 'c94'],\n",
       " 576: ['c4', 'c78', 'c88'],\n",
       " 577: ['c10', 'c70', 'c77'],\n",
       " 578: ['c24', 'c49', 'c73'],\n",
       " 579: ['c0', 'c45', 'c63'],\n",
       " 580: ['c23', 'c37', 'c61'],\n",
       " 581: ['c39', 'c86', 'c91'],\n",
       " 582: ['c1', 'c14', 'c28'],\n",
       " 583: ['c26', 'c80', 'c84'],\n",
       " 584: ['c6', 'c65', 'c85'],\n",
       " 585: ['c12', 'c33', 'c53'],\n",
       " 586: ['c38', 'c42', 'c98'],\n",
       " 587: ['c25', 'c27', 'c50'],\n",
       " 588: ['c49', 'c57', 'c62'],\n",
       " 589: ['c29', 'c30', 'c96'],\n",
       " 590: ['c15', 'c20', 'c60'],\n",
       " 591: ['c51', 'c87', 'c97'],\n",
       " 592: ['c5', 'c47', 'c71'],\n",
       " 593: ['c34', 'c41', 'c75'],\n",
       " 594: ['c16', 'c90', 'c95'],\n",
       " 595: ['c2', 'c54', 'c55'],\n",
       " 596: ['c13', 'c64', 'c67'],\n",
       " 597: ['c40', 'c44', 'c60'],\n",
       " 598: ['c17', 'c72', 'c83'],\n",
       " 599: ['c10', 'c18', 'c78'],\n",
       " 600: ['c21', 'c22', 'c84'],\n",
       " 601: ['c11', 'c52', 'c56'],\n",
       " 602: ['c0', 'c36', 'c89'],\n",
       " 603: ['c28', 'c74', 'c93'],\n",
       " 604: ['c29', 'c46', 'c80'],\n",
       " 605: ['c39', 'c68', 'c81'],\n",
       " 606: ['c9', 'c58', 'c82'],\n",
       " 607: ['c3', 'c66', 'c79'],\n",
       " 608: ['c35', 'c69', 'c97'],\n",
       " 609: ['c31', 'c48', 'c88'],\n",
       " 610: ['c8', 'c61', 'c94'],\n",
       " 611: ['c14', 'c32', 'c45'],\n",
       " 612: ['c23', 'c76', 'c86'],\n",
       " 613: ['c24', 'c59', 'c70'],\n",
       " 461: ['c14', 'c33', 'c66'],\n",
       " 462: ['c7', 'c15', 'c70'],\n",
       " 463: ['c43', 'c63', 'c80'],\n",
       " 464: ['c1', 'c67', 'c96'],\n",
       " 465: ['c0', 'c52', 'c75'],\n",
       " 466: ['c49', 'c50', 'c72'],\n",
       " 467: ['c37', 'c65', 'c87'],\n",
       " 469: ['c60', 'c76', 'c90'],\n",
       " 470: ['c45', 'c51', 'c92'],\n",
       " 471: ['c2', 'c9', 'c26'],\n",
       " 472: ['c17', 'c46', 'c61'],\n",
       " 473: ['c31', 'c34', 'c79'],\n",
       " 474: ['c6', 'c89', 'c96'],\n",
       " 475: ['c12', 'c55', 'c70'],\n",
       " 476: ['c3', 'c57', 'c80'],\n",
       " 477: ['c18', 'c53', 'c84'],\n",
       " 478: ['c28', 'c69', 'c88'],\n",
       " 479: ['c29', 'c44', 'c56'],\n",
       " 480: ['c4', 'c5', 'c15'],\n",
       " 481: ['c32', 'c63', 'c73'],\n",
       " 482: ['c1', 'c11', 'c20'],\n",
       " 483: ['c19', 'c91', 'c98'],\n",
       " 484: ['c36', 'c58', 'c85'],\n",
       " 485: ['c17', 'c74', 'c94'],\n",
       " 486: ['c24', 'c42', 'c62'],\n",
       " 487: ['c22', 'c33', 'c95'],\n",
       " 488: ['c21', 'c35', 'c41'],\n",
       " 489: ['c27', 'c64', 'c66'],\n",
       " 490: ['c30', 'c40', 'c81'],\n",
       " 491: ['c8', 'c14', 'c71'],\n",
       " 492: ['c77', 'c91', 'c92'],\n",
       " 493: ['c13', 'c23', 'c90'],\n",
       " 494: ['c10', 'c48', 'c86'],\n",
       " 495: ['c25', 'c47', 'c93'],\n",
       " 496: ['c16', 'c54', 'c68'],\n",
       " 497: ['c38', 'c78', 'c82'],\n",
       " 498: ['c59', 'c83', 'c97'],\n",
       " 499: ['c39', 'c73', 'c94'],\n",
       " 500: ['c43', 'c77', 'c87'],\n",
       " 501: ['c7', 'c26', 'c67'],\n",
       " 502: ['c37', 'c51', 'c74'],\n",
       " 503: ['c0', 'c4', 'c20'],\n",
       " 504: ['c19', 'c44', 'c50'],\n",
       " 505: ['c36', 'c42', 'c49'],\n",
       " 506: ['c18', 'c56', 'c60'],\n",
       " 507: ['c40', 'c69', 'c75'],\n",
       " 508: ['c28', 'c52', 'c72'],\n",
       " 509: ['c6', 'c55', 'c79'],\n",
       " 510: ['c21', 'c31', 'c53'],\n",
       " 511: ['c2', 'c3', 'c46'],\n",
       " 403: ['c13', 'c85', 'c95'],\n",
       " 407: ['c9', 'c23', 'c71'],\n",
       " 408: ['c10', 'c61', 'c67'],\n",
       " 411: ['c25', 'c57', 'c63'],\n",
       " 412: ['c1', 'c5', 'c7'],\n",
       " 413: ['c16', 'c41', 'c58'],\n",
       " 414: ['c4', 'c54', 'c81'],\n",
       " 415: ['c22', 'c91', 'c94'],\n",
       " 418: ['c26', 'c35', 'c42'],\n",
       " 419: ['c32', 'c37', 'c82'],\n",
       " 420: ['c15', 'c39', 'c93'],\n",
       " 421: ['c19', 'c23', 'c59'],\n",
       " 422: ['c8', 'c30', 'c83'],\n",
       " 423: ['c64', 'c78', 'c95'],\n",
       " 424: ['c24', 'c51', 'c85'],\n",
       " 425: ['c12', 'c27', 'c77'],\n",
       " 426: ['c31', 'c36', 'c84'],\n",
       " 427: ['c40', 'c43', 'c72'],\n",
       " 428: ['c2', 'c44', 'c45'],\n",
       " 429: ['c0', 'c21', 'c48'],\n",
       " 430: ['c50', 'c52', 'c97'],\n",
       " 431: ['c1', 'c16', 'c87'],\n",
       " 432: ['c70', 'c74', 'c88'],\n",
       " 433: ['c34', 'c35', 'c57'],\n",
       " 434: ['c14', 'c89', 'c98'],\n",
       " 435: ['c17', 'c62', 'c73'],\n",
       " 436: ['c33', 'c47', 'c68'],\n",
       " 437: ['c20', 'c55', 'c71'],\n",
       " 438: ['c29', 'c38', 'c86'],\n",
       " 439: ['c49', 'c60', 'c66'],\n",
       " 440: ['c6', 'c69', 'c92'],\n",
       " 441: ['c18', 'c79', 'c96'],\n",
       " 442: ['c46', 'c56', 'c90'],\n",
       " 443: ['c3', 'c28', 'c65'],\n",
       " 444: ['c11', 'c53', 'c80'],\n",
       " 445: ['c13', 'c76', 'c84'],\n",
       " 446: ['c68', 'c75', 'c91'],\n",
       " 447: ['c9', 'c45', 'c87'],\n",
       " 448: ['c34', 'c43', 'c48'],\n",
       " 449: ['c25', 'c26', 'c81'],\n",
       " 450: ['c37', 'c58', 'c63'],\n",
       " 451: ['c19', 'c27', 'c88'],\n",
       " 452: ['c5', 'c10', 'c54'],\n",
       " 453: ['c8', 'c38', 'c89'],\n",
       " 454: ['c12', 'c93', 'c97'],\n",
       " 455: ['c4', 'c61', 'c64'],\n",
       " 456: ['c22', 'c41', 'c51'],\n",
       " 457: ['c7', 'c39', 'c62'],\n",
       " 458: ['c32', 'c67', 'c83'],\n",
       " 513: ['c7', 'c54', 'c84'],\n",
       " 514: ['c9', 'c24', 'c77'],\n",
       " 515: ['c20', 'c29', 'c98'],\n",
       " 516: ['c30', 'c47', 'c73'],\n",
       " 517: ['c48', 'c66', 'c85'],\n",
       " 518: ['c50', 'c87', 'c89'],\n",
       " 519: ['c3', 'c5', 'c44'],\n",
       " 520: ['c37', 'c62', 'c71'],\n",
       " 521: ['c32', 'c80', 'c97'],\n",
       " 522: ['c26', 'c64', 'c94'],\n",
       " 523: ['c23', 'c43', 'c78'],\n",
       " 524: ['c15', 'c42', 'c82'],\n",
       " 525: ['c11', 'c70', 'c86'],\n",
       " 526: ['c59', 'c76', 'c96'],\n",
       " 527: ['c8', 'c22', 'c27'],\n",
       " 528: ['c0', 'c10', 'c49'],\n",
       " 529: ['c25', 'c39', 'c65'],\n",
       " 530: ['c63', 'c69', 'c84'],\n",
       " 531: ['c40', 'c73', 'c98'],\n",
       " 532: ['c6', 'c17', 'c54'],\n",
       " 533: ['c57', 'c75', 'c88'],\n",
       " 534: ['c16', 'c37', 'c38'],\n",
       " 535: ['c51', 'c68', 'c83'],\n",
       " 536: ['c19', 'c36', 'c60'],\n",
       " 537: ['c46', 'c74', 'c81'],\n",
       " 538: ['c1', 'c31', 'c52'],\n",
       " 539: ['c4', 'c56', 'c91'],\n",
       " 540: ['c2', 'c14', 'c72'],\n",
       " 541: ['c13', 'c34', 'c53'],\n",
       " 542: ['c58', 'c79', 'c93'],\n",
       " 543: ['c21', 'c28', 'c55'],\n",
       " 544: ['c33', 'c41', 'c92'],\n",
       " 545: ['c35', 'c67', 'c95'],\n",
       " 546: ['c7', 'c45', 'c78'],\n",
       " 548: ['c40', 'c61', 'c76'],\n",
       " 549: ['c38', 'c87', 'c90'],\n",
       " 550: ['c12', 'c16', 'c44'],\n",
       " 551: ['c5', 'c18', 'c30'],\n",
       " 552: ['c11', 'c47', 'c59'],\n",
       " 553: ['c20', 'c77', 'c82'],\n",
       " 554: ['c15', 'c19', 'c86'],\n",
       " 555: ['c22', 'c24', 'c29'],\n",
       " 556: ['c31', 'c39', 'c57'],\n",
       " 557: ['c0', 'c72', 'c85'],\n",
       " 558: ['c4', 'c9', 'c48'],\n",
       " 559: ['c43', 'c50', 'c94'],\n",
       " 560: ['c32', 'c62', 'c74'],\n",
       " 561: ['c64', 'c68', 'c80'],\n",
       " 616: ['c23', 'c49', 'c69'],\n",
       " 617: ['c73', 'c79', 'c86'],\n",
       " 618: ['c6', 'c10', 'c81'],\n",
       " 619: ['c12', 'c35', 'c92'],\n",
       " 620: ['c53', 'c64', 'c70'],\n",
       " 621: ['c51', 'c80', 'c82'],\n",
       " 622: ['c15', 'c76', 'c85'],\n",
       " 623: ['c29', 'c36', 'c75'],\n",
       " 624: ['c28', 'c58', 'c71'],\n",
       " 625: ['c17', 'c18', 'c98'],\n",
       " 626: ['c13', 'c20', 'c26'],\n",
       " 627: ['c21', 'c42', 'c52'],\n",
       " 628: ['c25', 'c49', 'c90'],\n",
       " 629: ['c11', 'c33', 'c45'],\n",
       " 630: ['c8', 'c55', 'c63'],\n",
       " 631: ['c31', 'c43', 'c89'],\n",
       " 632: ['c7', 'c47', 'c72'],\n",
       " 633: ['c50', 'c67', 'c68'],\n",
       " 634: ['c16', 'c27', 'c32'],\n",
       " 635: ['c9', 'c61', 'c84'],\n",
       " 636: ['c23', 'c74', 'c77'],\n",
       " 637: ['c1', 'c41', 'c44'],\n",
       " 638: ['c18', 'c65', 'c81'],\n",
       " 639: ['c15', 'c29', 'c78'],\n",
       " 640: ['c56', 'c96', 'c97'],\n",
       " 641: ['c3', 'c34', 'c60'],\n",
       " 642: ['c30', 'c52', 'c76'],\n",
       " 643: ['c59', 'c91', 'c95'],\n",
       " 644: ['c24', 'c66', 'c89'],\n",
       " 645: ['c5', 'c48', 'c94'],\n",
       " 646: ['c40', 'c54', 'c62'],\n",
       " 647: ['c2', 'c19', 'c22'],\n",
       " 648: ['c0', 'c38', 'c93'],\n",
       " 649: ['c14', 'c83', 'c88'],\n",
       " 650: ['c4', 'c57', 'c97'],\n",
       " 651: ['c27', 'c46', 'c65'],\n",
       " 652: ['c37', 'c47', 'c69'],\n",
       " 653: ['c10', 'c42', 'c87'],\n",
       " 654: ['c12', 'c39', 'c82'],\n",
       " 655: ['c30', 'c67', 'c78'],\n",
       " 656: ['c51', 'c63', 'c70'],\n",
       " 657: ['c33', 'c79', 'c90'],\n",
       " 658: ['c58', 'c86', 'c92'],\n",
       " 659: ['c28', 'c61', 'c73'],\n",
       " 660: ['c35', 'c45', 'c60'],\n",
       " 661: ['c17', 'c20', 'c91'],\n",
       " 662: ['c6', 'c8', 'c53'],\n",
       " 663: ['c13', 'c36', 'c41'],\n",
       " 713: ['c39', 'c41', 'c98'],\n",
       " 714: ['c21', 'c50', 'c96'],\n",
       " 715: ['c7', 'c33', 'c35'],\n",
       " 716: ['c1', 'c24', 'c26'],\n",
       " 717: ['c2', 'c15', 'c95'],\n",
       " 718: ['c59', 'c84', 'c86'],\n",
       " 719: ['c3', 'c56', 'c94'],\n",
       " 720: ['c5', 'c14', 'c46'],\n",
       " 721: ['c4', 'c63', 'c93'],\n",
       " 722: ['c37', 'c54', 'c88'],\n",
       " 723: ['c27', 'c57', 'c83'],\n",
       " 724: ['c19', 'c42', 'c79'],\n",
       " 726: ['c22', 'c31', 'c74'],\n",
       " 727: ['c30', 'c50', 'c51'],\n",
       " 728: ['c10', 'c39', 'c64'],\n",
       " 729: ['c29', 'c43', 'c85'],\n",
       " 730: ['c6', 'c67', 'c87'],\n",
       " 731: ['c41', 'c45', 'c82'],\n",
       " 732: ['c36', 'c40', 'c53'],\n",
       " 733: ['c60', 'c92', 'c97'],\n",
       " 734: ['c1', 'c55', 'c66'],\n",
       " 735: ['c3', 'c24', 'c75'],\n",
       " 736: ['c25', 'c35', 'c56'],\n",
       " 737: ['c26', 'c46', 'c77'],\n",
       " 739: ['c13', 'c81', 'c91'],\n",
       " 740: ['c49', 'c58', 'c65'],\n",
       " 741: ['c16', 'c52', 'c62'],\n",
       " 742: ['c11', 'c67', 'c84'],\n",
       " 743: ['c0', 'c28', 'c70'],\n",
       " 744: ['c7', 'c40', 'c71'],\n",
       " 745: ['c14', 'c34', 'c85'],\n",
       " 746: ['c38', 'c48', 'c73'],\n",
       " 747: ['c12', 'c20', 'c23'],\n",
       " 748: ['c18', 'c44', 'c61'],\n",
       " 749: ['c8', 'c17', 'c78'],\n",
       " 750: ['c69', 'c72', 'c76'],\n",
       " 751: ['c47', 'c80', 'c90'],\n",
       " 752: ['c9', 'c68', 'c89'],\n",
       " 753: ['c32', 'c93', 'c98'],\n",
       " 754: ['c2', 'c11', 'c21'],\n",
       " 755: ['c75', 'c77', 'c83'],\n",
       " 756: ['c25', 'c59', 'c71'],\n",
       " 757: ['c55', 'c88', 'c96'],\n",
       " 758: ['c34', 'c66', 'c95'],\n",
       " 759: ['c33', 'c42', 'c74'],\n",
       " 665: ['c1', 'c34', 'c51'],\n",
       " 666: ['c5', 'c38', 'c75'],\n",
       " 667: ['c33', 'c80', 'c81'],\n",
       " 668: ['c25', 'c61', 'c62'],\n",
       " 669: ['c6', 'c24', 'c30'],\n",
       " 670: ['c21', 'c29', 'c95'],\n",
       " 672: ['c40', 'c42', 'c46'],\n",
       " 674: ['c41', 'c74', 'c96'],\n",
       " 675: ['c52', 'c59', 'c66'],\n",
       " 676: ['c27', 'c49', 'c67'],\n",
       " 677: ['c0', 'c47', 'c77'],\n",
       " 678: ['c10', 'c43', 'c92'],\n",
       " 679: ['c32', 'c58', 'c91'],\n",
       " 680: ['c15', 'c57', 'c65'],\n",
       " 681: ['c11', 'c14', 'c73'],\n",
       " 682: ['c19', 'c76', 'c87'],\n",
       " 683: ['c3', 'c83', 'c98'],\n",
       " 684: ['c2', 'c34', 'c69'],\n",
       " 685: ['c18', 'c80', 'c93'],\n",
       " 686: ['c25', 'c29', 'c82'],\n",
       " 687: ['c1', 'c17', 'c45'],\n",
       " 688: ['c31', 'c35', 'c85'],\n",
       " 689: ['c16', 'c64', 'c84'],\n",
       " 690: ['c47', 'c53', 'c54'],\n",
       " 691: ['c6', 'c26', 'c27'],\n",
       " 692: ['c10', 'c71', 'c91'],\n",
       " 693: ['c7', 'c28', 'c90'],\n",
       " 694: ['c4', 'c12', 'c37'],\n",
       " 695: ['c56', 'c63', 'c68'],\n",
       " 696: ['c55', 'c89', 'c97'],\n",
       " 697: ['c8', 'c60', 'c86'],\n",
       " 698: ['c9', 'c70', 'c94'],\n",
       " 699: ['c44', 'c72', 'c88'],\n",
       " 700: ['c22', 'c36', 'c48'],\n",
       " 701: ['c13', 'c78', 'c79'],\n",
       " 702: ['c3', 'c39', 'c50'],\n",
       " 703: ['c23', 'c46', 'c82'],\n",
       " 706: ['c20', 'c38', 'c66'],\n",
       " 707: ['c5', 'c53', 'c62'],\n",
       " 708: ['c26', 'c71', 'c96'],\n",
       " 709: ['c54', 'c73', 'c75'],\n",
       " 710: ['c51', 'c81', 'c95'],\n",
       " 711: ['c19', 'c30', 'c65'],\n",
       " 765: ['c23', 'c72', 'c79'],\n",
       " 766: ['c8', 'c29', 'c93'],\n",
       " 767: ['c45', 'c57', 'c81'],\n",
       " 768: ['c48', 'c49', 'c77'],\n",
       " 769: ['c64', 'c76', 'c92'],\n",
       " 770: ['c32', 'c69', 'c87'],\n",
       " 772: ['c42', 'c43', 'c67'],\n",
       " 773: ['c13', 'c27', 'c35'],\n",
       " 774: ['c11', 'c38', 'c91'],\n",
       " 775: ['c41', 'c56', 'c80'],\n",
       " 776: ['c0', 'c24', 'c25'],\n",
       " 779: ['c60', 'c61', 'c89'],\n",
       " 780: ['c2', 'c88', 'c94'],\n",
       " 781: ['c20', 'c28', 'c36'],\n",
       " 783: ['c15', 'c37', 'c44'],\n",
       " 784: ['c18', 'c70', 'c90'],\n",
       " 785: ['c16', 'c59', 'c72'],\n",
       " 786: ['c17', 'c68', 'c79'],\n",
       " 787: ['c6', 'c50', 'c73'],\n",
       " 788: ['c31', 'c49', 'c87'],\n",
       " 789: ['c48', 'c76', 'c98'],\n",
       " 790: ['c30', 'c53', 'c74'],\n",
       " 791: ['c19', 'c34', 'c54'],\n",
       " 792: ['c7', 'c75', 'c85'],\n",
       " 793: ['c5', 'c95', 'c96'],\n",
       " 794: ['c15', 'c45', 'c58'],\n",
       " 795: ['c8', 'c88', 'c97'],\n",
       " 796: ['c40', 'c47', 'c66'],\n",
       " 797: ['c71', 'c82', 'c84'],\n",
       " 798: ['c10', 'c51', 'c62'],\n",
       " 799: ['c1', 'c65', 'c78'],\n",
       " 800: ['c9', 'c39', 'c63'],\n",
       " 801: ['c4', 'c22', 'c52'],\n",
       " 802: ['c14', 'c55', 'c86'],\n",
       " 803: ['c28', 'c33', 'c46'],\n",
       " 804: ['c3', 'c26', 'c90'],\n",
       " 805: ['c31', 'c56', 'c83'],\n",
       " 806: ['c13', 'c21', 'c58'],\n",
       " 807: ['c12', 'c43', 'c98'],\n",
       " 808: ['c11', 'c23', 'c97'],\n",
       " 809: ['c57', 'c61', 'c92'],\n",
       " 810: ['c24', 'c69', 'c93'],\n",
       " 812: ['c44', 'c52', 'c86'],\n",
       " 813: ['c22', 'c32', 'c72'],\n",
       " 814: ['c35', 'c36', 'c91'],\n",
       " 815: ['c2', 'c33', 'c71'],\n",
       " 816: ['c6', 'c38', 'c63'],\n",
       " 817: ['c40', 'c45', 'c80'],\n",
       " 818: ['c24', 'c56', 'c81'],\n",
       " 819: ['c34', 'c58', 'c61'],\n",
       " 820: ['c20', 'c27', 'c96'],\n",
       " 821: ['c0', 'c9', 'c17'],\n",
       " 822: ['c16', 'c21', 'c77'],\n",
       " 823: ['c18', 'c55', 'c83'],\n",
       " 824: ['c14', 'c70', 'c78'],\n",
       " 825: ['c39', 'c43', 'c54'],\n",
       " 827: ['c49', 'c51', 'c53'],\n",
       " 828: ['c3', 'c62', 'c82'],\n",
       " 829: ['c7', 'c50', 'c69'],\n",
       " 830: ['c31', 'c65', 'c97'],\n",
       " 831: ['c28', 'c48', 'c84'],\n",
       " 832: ['c46', 'c57', 'c87'],\n",
       " 833: ['c37', 'c80', 'c94'],\n",
       " 834: ['c1', 'c15', 'c90'],\n",
       " 835: ['c17', 'c44', 'c89'],\n",
       " 836: ['c20', 'c22', 'c93'],\n",
       " 837: ['c10', 'c30', 'c85'],\n",
       " 838: ['c47', 'c79', 'c88'],\n",
       " 839: ['c18', 'c64', 'c86'],\n",
       " 840: ['c8', 'c66', 'c76'],\n",
       " 841: ['c26', 'c29', 'c74'],\n",
       " 842: ['c23', 'c25', 'c98'],\n",
       " 843: ['c4', 'c19', 'c41'],\n",
       " 844: ['c42', 'c59', 'c68'],\n",
       " 845: ['c5', 'c12', 'c60'],\n",
       " 846: ['c16', 'c75', 'c92'],\n",
       " 847: ['c56', 'c67', 'c73'],\n",
       " 848: ['c11', 'c36', 'c94'],\n",
       " 849: ['c13', 'c63', 'c89'],\n",
       " 850: ['c61', 'c93', 'c95'],\n",
       " 851: ['c2', 'c37', 'c64'],\n",
       " 852: ['c32', 'c52', 'c70'],\n",
       " 854: ['c12', 'c21', 'c90'],\n",
       " 855: ['c22', 'c46', 'c55'],\n",
       " 856: ['c13', 'c18', 'c51'],\n",
       " 857: ['c57', 'c71', 'c95'],\n",
       " 858: ['c19', 'c39', 'c58'],\n",
       " 859: ['c31', 'c33', 'c73'],\n",
       " 860: ['c25', 'c34', 'c70'],\n",
       " 861: ['c27', 'c42', 'c69'],\n",
       " 862: ['c4', 'c6', 'c94'],\n",
       " 863: ['c1', 'c29', 'c83'],\n",
       " 864: ['c5', 'c9', 'c16'],\n",
       " 865: ['c0', 'c60', 'c74'],\n",
       " 866: ['c14', 'c53', 'c59'],\n",
       " 867: ['c48', 'c61', 'c81'],\n",
       " 868: ['c2', 'c75', 'c80'],\n",
       " 869: ['c7', 'c17', 'c93'],\n",
       " 870: ['c54', 'c82', 'c91'],\n",
       " 871: ['c45', 'c85', 'c86'],\n",
       " 872: ['c19', 'c68', 'c72'],\n",
       " 873: ['c11', 'c37', 'c41'],\n",
       " 874: ['c24', 'c36', 'c67'],\n",
       " 875: ['c26', 'c38', 'c76'],\n",
       " 876: ['c63', 'c87', 'c98'],\n",
       " 877: ['c15', 'c52', 'c64'],\n",
       " 878: ['c8', 'c47', 'c96'],\n",
       " 879: ['c44', 'c49', 'c92'],\n",
       " 880: ['c20', 'c30', 'c32'],\n",
       " 881: ['c28', 'c43', 'c97'],\n",
       " 882: ['c65', 'c79', 'c84'],\n",
       " 883: ['c35', 'c50', 'c62'],\n",
       " 884: ['c10', 'c23', 'c89'],\n",
       " 885: ['c66', 'c77', 'c78'],\n",
       " 886: ['c42', 'c56', 'c88'],\n",
       " 887: ['c3', 'c55', 'c68'],\n",
       " 888: ['c9', 'c34', 'c40'],\n",
       " 889: ['c4', 'c18', 'c21'],\n",
       " 890: ['c12', 'c14', 'c31'],\n",
       " 891: ['c0', 'c71', 'c83'],\n",
       " 894: ['c34', 'c67', 'c88'],\n",
       " 896: ['c42', 'c64', 'c85'],\n",
       " 897: ['c38', 'c39', 'c71'],\n",
       " 898: ['c9', 'c22', 'c28'],\n",
       " 899: ['c3', 'c48', 'c52'],\n",
       " 900: ['c33', 'c83', 'c86'],\n",
       " 901: ['c8', 'c43', 'c74'],\n",
       " 902: ['c49', 'c78', 'c97'],\n",
       " 903: ['c2', 'c60', 'c77'],\n",
       " 904: ['c18', 'c68', 'c69'],\n",
       " 905: ['c7', 'c12', 'c56'],\n",
       " 906: ['c10', 'c63', 'c65'],\n",
       " 907: ['c5', 'c79', 'c91'],\n",
       " 908: ['c51', 'c58', 'c73'],\n",
       " 909: ['c16', 'c24', 'c98'],\n",
       " 910: ['c27', 'c29', 'c81'],\n",
       " 911: ['c17', 'c50', 'c82'],\n",
       " 912: ['c1', 'c32', 'c40'],\n",
       " 913: ['c4', 'c47', 'c92'],\n",
       " 914: ['c15', 'c35', 'c96'],\n",
       " 915: ['c26', 'c30', 'c66'],\n",
       " 916: ['c45', 'c62', 'c76'],\n",
       " 917: ['c89', 'c94', 'c95'],\n",
       " 918: ['c20', 'c21', 'c46'],\n",
       " 919: ['c0', 'c23', 'c87'],\n",
       " 920: ['c25', 'c37', 'c84'],\n",
       " 921: ['c41', 'c55', 'c57'],\n",
       " 922: ['c11', 'c13', 'c72'],\n",
       " 923: ['c6', 'c44', 'c59'],\n",
       " 924: ['c14', 'c52', 'c54'],\n",
       " 925: ['c8', 'c70', 'c75'],\n",
       " 926: ['c36', 'c83', 'c90'],\n",
       " 927: ['c19', 'c64', 'c93'],\n",
       " 928: ['c9', 'c31', 'c80'],\n",
       " 929: ['c39', 'c53', 'c78'],\n",
       " 930: ['c61', 'c77', 'c88'],\n",
       " 932: ['c65', 'c72', 'c73'],\n",
       " 933: ['c25', 'c51', 'c55'],\n",
       " 934: ['c0', 'c43', 'c44'],\n",
       " 935: ['c77', 'c94', 'c97'],\n",
       " 936: ['c28', 'c30', 'c89'],\n",
       " 937: ['c7', 'c21', 'c38'],\n",
       " 938: ['c3', 'c58', 'c84'],\n",
       " 939: ['c40', 'c88', 'c93'],\n",
       " 940: ['c29', 'c31', 'c61'],\n",
       " 941: ['c20', 'c64', 'c81'],\n",
       " 942: ['c6', 'c39', 'c46'],\n",
       " 943: ['c52', 'c78', 'c92'],\n",
       " 944: ['c16', 'c42', 'c57'],\n",
       " 945: ['c33', 'c36', 'c69'],\n",
       " 946: ['c17', 'c24', 'c37'],\n",
       " 947: ['c67', 'c74', 'c91'],\n",
       " 948: ['c5', 'c68', 'c86'],\n",
       " 949: ['c22', 'c63', 'c66'],\n",
       " 950: ['c15', 'c23', 'c50'],\n",
       " 951: ['c11', 'c85', 'c87'],\n",
       " 952: ['c2', 'c4', 'c32'],\n",
       " 953: ['c41', 'c49', 'c54'],\n",
       " 954: ['c27', 'c48', 'c75'],\n",
       " 955: ['c12', 'c62', 'c95'],\n",
       " 956: ['c10', 'c53', 'c79'],\n",
       " 957: ['c35', 'c70', 'c98'],\n",
       " 958: ['c14', 'c60', 'c96'],\n",
       " 959: ['c13', 'c45', 'c59'],\n",
       " 960: ['c0', 'c26', 'c82'],\n",
       " 961: ['c1', 'c47', 'c94'],\n",
       " 962: ['c55', 'c56', 'c76'],\n",
       " 963: ['c65', 'c71', 'c90'],\n",
       " 964: ['c19', 'c38', 'c80'],\n",
       " 965: ['c8', 'c18', 'c58'],\n",
       " 966: ['c28', 'c34', 'c83'],\n",
       " 969: ['c24', 'c64', 'c87'],\n",
       " 970: ['c54', 'c58', 'c67'],\n",
       " 971: ['c3', 'c4', 'c69'],\n",
       " 972: ['c26', 'c44', 'c62'],\n",
       " 973: ['c12', 'c49', 'c86'],\n",
       " 974: ['c17', 'c43', 'c96'],\n",
       " 975: ['c1', 'c9', 'c73'],\n",
       " 976: ['c14', 'c16', 'c29'],\n",
       " 977: ['c13', 'c37', 'c75'],\n",
       " 978: ['c63', 'c72', 'c78'],\n",
       " 979: ['c10', 'c20', 'c90'],\n",
       " 980: ['c32', 'c51', 'c79'],\n",
       " 982: ['c33', 'c84', 'c85'],\n",
       " 983: ['c57', 'c70', 'c91'],\n",
       " 984: ['c21', 'c81', 'c92'],\n",
       " 985: ['c11', 'c15', 'c27'],\n",
       " 986: ['c40', 'c48', 'c50'],\n",
       " 987: ['c41', 'c77', 'c89'],\n",
       " 988: ['c5', 'c35', 'c74'],\n",
       " 989: ['c36', 'c68', 'c95'],\n",
       " 990: ['c23', 'c45', 'c53'],\n",
       " 991: ['c30', 'c46', 'c59'],\n",
       " 992: ['c60', 'c82', 'c98'],\n",
       " 1062: ['c7', 'c41', 'c76']}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "region_2_readout_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "63528cd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# collect types\n",
    "readout_usage_by_type = {}\n",
    "for _region, _infos in region_2_readout_dict.items():\n",
    "    for _info in _infos:\n",
    "        if _info[0] not in readout_usage_by_type:\n",
    "            readout_usage_by_type[_info[0]] = []\n",
    "        if int(_info[1:]) not in readout_usage_by_type[_info[0]]:\n",
    "            readout_usage_by_type[_info[0]].append(int(_info[1:]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "3aea1f39",
   "metadata": {},
   "outputs": [],
   "source": [
    "readout_usage_dict = {_type:len(_v) for _type,_v in readout_usage_by_type.items()}\n",
    "readout_dict ={}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f1a9f83c",
   "metadata": {},
   "outputs": [],
   "source": [
    "reload(library_tools.assemble)\n",
    "overwrite_readout_usage = True\n",
    "readout_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Readouts'\n",
    "readout_start = 0\n",
    "num_color = 2\n",
    "comb_readouts = library_tools.assemble.load_readouts(readout_usage_dict['c'], \n",
    "                                                       _type='snr_after_630', \n",
    "                                                       _readout_folder=readout_folder, \n",
    "                                                       _num_colors=2, # 750, 647\n",
    "                                                       _start_id=readout_start,\n",
    "                                                       )\n",
    "readout_dict['c'] = comb_readouts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "7520548c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[SeqRecord(seq=Seq('AACCACGGGCTAGAAATATCCGAGTGGCAT'), id='NDB_784', name='NDB_784', description='NDB_784', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GCCTAAGACGTGCGACATGTTATGCTAGGC'), id='NDB_755', name='NDB_755', description='NDB_755', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GCCGTTTCGTATATGTCGATGTCCTTAGAC'), id='NDB_826', name='NDB_826', description='NDB_826', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TACAAATACCGGTGGTTGTCACTTTCCGAA'), id='NDB_713', name='NDB_713', description='NDB_713', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCGTGACGAGATCCCGGTCGGATATGTGA'), id='NDB_865', name='NDB_865', description='NDB_865', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GAGCTTATGACTACCACGGCATCGATTTAG'), id='NDB_725', name='NDB_725', description='NDB_725', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGCGAAGATCGAACTCGTTGACTAAGCATT'), id='NDB_817', name='NDB_817', description='NDB_817', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTGGTAAACCTCAACTACTAAGCCGCGGTA'), id='NDB_710', name='NDB_710', description='NDB_710', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCAGTAGTCATAGGCGCTTTAGTAGGTACC'), id='NDB_652', name='NDB_652', description='NDB_652', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GCGCAAAGTTCTTATCGCAAGGTGGCCTTA'), id='NDB_971', name='NDB_971', description='NDB_971', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCGATAACGGCGTATTGCAGGATCCTTATG'), id='NDB_718', name='NDB_718', description='NDB_718', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TTGTTCTATACGATCAAGGTGTGCCGCGTC'), id='NDB_743', name='NDB_743', description='NDB_743', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AGCCTGTCTGGGTCACGTACGAGACATTGG'), id='NDB_847', name='NDB_847', description='NDB_847', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AGTAAAGTATACCGCGTCCGCAGAGCAAAC'), id='NDB_683', name='NDB_683', description='NDB_683', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTATTGATCGCGAACGACCTGTTACCTTGC'), id='NDB_643', name='NDB_643', description='NDB_643', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ATACTAAATGTTGCACGCAGCGATTGACGC'), id='NDB_815', name='NDB_815', description='NDB_815', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCAATGTAACGTTAGATGGCCGCTCGGGTT'), id='NDB_760', name='NDB_760', description='NDB_760', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTTCAGAGTTATCCGGCGCGCGAGACTAAG'), id='NDB_965', name='NDB_965', description='NDB_965', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CACGTAGCTCGGTGTCGGTTAAATATTCCT'), id='NDB_790', name='NDB_790', description='NDB_790', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CAGTATCATGGTCGTCCGCAGTAATTTGTC'), id='NDB_917', name='NDB_917', description='NDB_917', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GACCTGACGAAATGATGCCTTGCGGTTCTT'), id='NDB_778', name='NDB_778', description='NDB_778', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GGGATGGTAGCCACACGCTTCGATGACTTA'), id='NDB_728', name='NDB_728', description='NDB_728', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACGTAAAGCACGTCTGGATCGTCTATCAAT'), id='NDB_844', name='NDB_844', description='NDB_844', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AATGACAACGAAATAGCGTGCTGCCTACCT'), id='NDB_974', name='NDB_974', description='NDB_974', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CTCGAGACTGCTACCAAGTTACGAGGCATA'), id='NDB_961', name='NDB_961', description='NDB_961', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACGGTTACGGTGCTTATCAACTCGACAGCG'), id='NDB_746', name='NDB_746', description='NDB_746', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GAGATCTCCGAATTTCTCGCATGTCGTGTA'), id='NDB_661', name='NDB_661', description='NDB_661', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACTCAACTTGATGTTCGGCAATAAGCGCGT'), id='NDB_875', name='NDB_875', description='NDB_875', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CACATATGTCGACAATCGCTGATCAGACGT'), id='NDB_901', name='NDB_901', description='NDB_901', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGACTTCTGGTCTGATTAGCTGCGCACGTT'), id='NDB_773', name='NDB_773', description='NDB_773', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AAAGACTAGTGTAAGCTGCAATCCACCCGT'), id='NDB_868', name='NDB_868', description='NDB_868', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CTCTAGGTAGGGATCCGTATAACGTACGTC'), id='NDB_812', name='NDB_812', description='NDB_812', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCCTAACTAGGTGGAAGCGTTAGGACCGGA'), id='NDB_1027', name='NDB_1027', description='NDB_1027', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GGATTAGCACGCTCGCCTTATTACCTGTTA'), id='NDB_716', name='NDB_716', description='NDB_716', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCGCTACGGTTAATTCGATGTCGAAGCGTC'), id='NDB_754', name='NDB_754', description='NDB_754', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ATCTATTAGGTTCGTAACCGGCGTCGTCTA'), id='NDB_860', name='NDB_860', description='NDB_860', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GCGATGTGTCGCTCGCAATCCTAATGTATA'), id='NDB_856', name='NDB_856', description='NDB_856', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCTGAGTGGGCTTAACGTATTAGAATCCGC'), id='NDB_680', name='NDB_680', description='NDB_680', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACGCACTTCCGAGGCGGATTGAGATTCGGT'), id='NDB_634', name='NDB_634', description='NDB_634', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GATTACGTAACACGCTAATCGATCGACCTG'), id='NDB_737', name='NDB_737', description='NDB_737', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TTGCTCTTATGCCGTCGTCACGTGCGAGTA'), id='NDB_715', name='NDB_715', description='NDB_715', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACCTAGGGACGGTCTGTTACTATCGGTCAC'), id='NDB_698', name='NDB_698', description='NDB_698', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTAAACTGAGCCGCCTCGGTTACTTAGCTT'), id='NDB_883', name='NDB_883', description='NDB_883', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGTCAAGGCGTGCTCGCACAGGATTCTTAA'), id='NDB_863', name='NDB_863', description='NDB_863', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GCGGTTAACTGAATCCACGCGGTATTTCTT'), id='NDB_1033', name='NDB_1033', description='NDB_1033', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGCTACTTACCTGAGCGCAGTTCGATGTTA'), id='NDB_878', name='NDB_878', description='NDB_878', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTTATTCGATATTTCCTACGCTGCATCGTG'), id='NDB_631', name='NDB_631', description='NDB_631', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CTTGAGTGTCGATGCCTCTTCGATAGATTC'), id='NDB_758', name='NDB_758', description='NDB_758', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACACAGTTCGCGTGCTATGTAGTCGCATAT'), id='NDB_1075', name='NDB_1075', description='NDB_1075', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TGAGTTGTGTCGACAACCTTCATACGTGGC'), id='NDB_653', name='NDB_653', description='NDB_653', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TACCATTAAACGAGAATTGCGAGCTGTCCA'), id='NDB_1060', name='NDB_1060', description='NDB_1060', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCCAAGTATAAGGGCGATGTAACGGCGCA'), id='NDB_905', name='NDB_905', description='NDB_905', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCCTATGCGTAGTTCGGGCTTGCCAATACT'), id='NDB_805', name='NDB_805', description='NDB_805', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCGTTCGTTAAGTACATCGCTTGCTGCAAA'), id='NDB_866', name='NDB_866', description='NDB_866', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGCGTATCACGGAGTACAACGAGATCGTTA'), id='NDB_730', name='NDB_730', description='NDB_730', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GAACTAATTAGGTATCTCGACGGCGTGCAC'), id='NDB_887', name='NDB_887', description='NDB_887', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TAATAGACCGCTAGGCCTAGTATACTTCTC'), id='NDB_880', name='NDB_880', description='NDB_880', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CAACACACTAATTTGCGTCCAGGGTGTATC'), id='NDB_752', name='NDB_752', description='NDB_752', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCATACACGAGTTATTTGATCGAGACGCTG'), id='NDB_832', name='NDB_832', description='NDB_832', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AGCTATTAGGATCGAGCCCTCGGTAGTCCT'), id='NDB_896', name='NDB_896', description='NDB_896', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CACTTGTAGGTCAACTCGAACGTATGCCTA'), id='NDB_835', name='NDB_835', description='NDB_835', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ATAATCCTGAGTCTGCGATACCTCCTTGTT'), id='NDB_989', name='NDB_989', description='NDB_989', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GCGGACTTCCATCGTGGCCGGCAATAAGTA'), id='NDB_946', name='NDB_946', description='NDB_946', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AATAACATAGCCCGGTATTCCACGAGAACT'), id='NDB_641', name='NDB_641', description='NDB_641', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGTTAGTTAAGAGCTGACCGCAACGCTATC'), id='NDB_721', name='NDB_721', description='NDB_721', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TATATAGGTCAAGATGCGTCTTTCGCCTCG'), id='NDB_899', name='NDB_899', description='NDB_899', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACTCTCGAGTATATCGGTCTAGTGGATACC'), id='NDB_853', name='NDB_853', description='NDB_853', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCTCTATCGTGTGAAAGGTGCCGCTATTGC'), id='NDB_941', name='NDB_941', description='NDB_941', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTGAACGCTAGTATTAGGCACTCATCCGAC'), id='NDB_838', name='NDB_838', description='NDB_838', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACAAATCTGACTTGCCGCACGGGTTATTTG'), id='NDB_647', name='NDB_647', description='NDB_647', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACTCTCGCTTAAGATTTAGCGACGCTCTAA'), id='NDB_994', name='NDB_994', description='NDB_994', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CTCAAGGACTGGTTGCCGATACCTTTCTTG'), id='NDB_881', name='NDB_881', description='NDB_881', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TAGCTTACCATTCGTGTCCAAACGTCAGTT'), id='NDB_1066', name='NDB_1066', description='NDB_1066', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GGCGATTTAGCGTATCTCCTGATTCCGATT'), id='NDB_665', name='NDB_665', description='NDB_665', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GCTATGAAGCGGCGACCCGGTTCTTATTTA'), id='NDB_637', name='NDB_637', description='NDB_637', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CTTCAGCGTTGGGTCGCACGTGTATTATCT'), id='NDB_734', name='NDB_734', description='NDB_734', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AGTCAGTGAACTATCGGCCTTAATACCGGT'), id='NDB_706', name='NDB_706', description='NDB_706', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ACCTACGTAAGACCTGGTTCGTGGATGCGA'), id='NDB_935', name='NDB_935', description='NDB_935', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTGAAAGAATACCGGTTCACCCTCGTGAGA'), id='NDB_889', name='NDB_889', description='NDB_889', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GACTTCCGTCGGTAATCGATTAGATTCTCC'), id='NDB_788', name='NDB_788', description='NDB_788', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CTTGAAGTAGTCACGGCCGTACCTACTGTT'), id='NDB_862', name='NDB_862', description='NDB_862', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGACAATTGATAAGTGCCCGGGTCCGTGAA'), id='NDB_779', name='NDB_779', description='NDB_779', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TGTAACTAACGGTATACTTAATCGCTGGCC'), id='NDB_694', name='NDB_694', description='NDB_694', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCGATATCAGCTTGTTATCGCTTTACGTGA'), id='NDB_977', name='NDB_977', description='NDB_977', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGCAATTACGTCCGGTTGCCAGGCATAAGT'), id='NDB_751', name='NDB_751', description='NDB_751', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('AACTATGATCGGCTATCCTCAGCGACATTT'), id='NDB_836', name='NDB_836', description='NDB_836', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCAGTATTATCGCTGTTTCGCCAATCAAGT'), id='NDB_742', name='NDB_742', description='NDB_742', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CGCCTTAGGTCATGCGATCATTGATCCAGG'), id='NDB_929', name='NDB_929', description='NDB_929', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GGACTAGTTATCACGCTTCGGCAGAGAATC'), id='NDB_958', name='NDB_958', description='NDB_958', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('ATGGTCTATCTCTCGCCAAAGACGATCAAT'), id='NDB_1028', name='NDB_1028', description='NDB_1028', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GTGAACTGGTACCGACCAGAGCGATCACTG'), id='NDB_925', name='NDB_925', description='NDB_925', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CAGTTCGATTGACACGACTATTGTTTACGC'), id='NDB_824', name='NDB_824', description='NDB_824', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCCGTAAAGGATAATGCTCCTTCGGCTATG'), id='NDB_712', name='NDB_712', description='NDB_712', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TGGGTTACTGGGTCGTCTATACCAACCTTA'), id='NDB_818', name='NDB_818', description='NDB_818', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GACCTATCATTCGGGCTAGCAACTGCGATG'), id='NDB_940', name='NDB_940', description='NDB_940', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TCAGAGCGAATTAGCGGCTTATTCCTCTAC'), id='NDB_872', name='NDB_872', description='NDB_872', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('CCCGTGAGTATTCGCGCGCATTTGGGAAGT'), id='NDB_934', name='NDB_934', description='NDB_934', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('TACGTAGATTGCTGCGACGACTCAACGCTT'), id='NDB_926', name='NDB_926', description='NDB_926', dbxrefs=[]),\n",
       " SeqRecord(seq=Seq('GATCTGTTTAGCGCTCCGGAGTTCACCAGT'), id='NDB_910', name='NDB_910', description='NDB_910', dbxrefs=[])]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "readout_dict['c']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "id": "f6c80620",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "save\n"
     ]
    }
   ],
   "source": [
    "# save\n",
    "readout_usage_file = os.path.join(library_folder, 'readout_usage.pkl')\n",
    "if not os.path.isfile(readout_usage_file) or overwrite_readout_usage:\n",
    "    print('save')\n",
    "    pickle.dump(readout_dict, open(readout_usage_file, 'wb'))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f466634e",
   "metadata": {},
   "source": [
    "## 3.3 assemble probes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "5165339c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Assemble probes by given target sequences, readouts and primers.\n",
      "-- included readout types: ['c']\n",
      "--- assemblying 163 probes in region: 1\n",
      "--- assemblying 180 probes in region: 2\n",
      "--- assemblying 77 probes in region: 3\n",
      "--- assemblying 205 probes in region: 4\n",
      "--- assemblying 144 probes in region: 5\n",
      "--- assemblying 297 probes in region: 6\n",
      "--- assemblying 210 probes in region: 7\n",
      "--- assemblying 274 probes in region: 8\n",
      "--- assemblying 220 probes in region: 9\n",
      "--- assemblying 158 probes in region: 10\n",
      "--- assemblying 156 probes in region: 11\n",
      "--- assemblying 107 probes in region: 12\n",
      "--- assemblying 291 probes in region: 13\n",
      "--- assemblying 240 probes in region: 14\n",
      "--- assemblying 279 probes in region: 15\n",
      "--- assemblying 248 probes in region: 16\n",
      "--- assemblying 276 probes in region: 17\n",
      "--- assemblying 109 probes in region: 18\n",
      "--- assemblying 111 probes in region: 19\n",
      "--- assemblying 212 probes in region: 20\n",
      "--- assemblying 135 probes in region: 21\n",
      "--- assemblying 229 probes in region: 22\n",
      "--- assemblying 257 probes in region: 23\n",
      "--- assemblying 257 probes in region: 24\n",
      "--- assemblying 249 probes in region: 25\n",
      "--- assemblying 204 probes in region: 26\n",
      "--- assemblying 195 probes in region: 27\n",
      "--- assemblying 198 probes in region: 28\n",
      "--- assemblying 284 probes in region: 29\n",
      "--- assemblying 245 probes in region: 30\n",
      "--- assemblying 151 probes in region: 31\n",
      "--- assemblying 247 probes in region: 32\n",
      "--- assemblying 276 probes in region: 33\n",
      "--- assemblying 300 probes in region: 34\n",
      "--- assemblying 310 probes in region: 35\n",
      "--- assemblying 260 probes in region: 36\n",
      "--- assemblying 244 probes in region: 37\n",
      "--- assemblying 75 probes in region: 38\n",
      "--- assemblying 178 probes in region: 39\n",
      "--- assemblying 142 probes in region: 40\n",
      "--- assemblying 122 probes in region: 41\n",
      "--- assemblying 105 probes in region: 42\n",
      "--- assemblying 173 probes in region: 43\n",
      "--- assemblying 90 probes in region: 44\n",
      "--- assemblying 121 probes in region: 45\n",
      "--- assemblying 241 probes in region: 46\n",
      "--- assemblying 240 probes in region: 47\n",
      "--- assemblying 218 probes in region: 48\n",
      "--- assemblying 225 probes in region: 49\n",
      "--- assemblying 287 probes in region: 50\n",
      "--- assemblying 250 probes in region: 51\n",
      "--- assemblying 329 probes in region: 52\n",
      "--- assemblying 299 probes in region: 53\n",
      "--- assemblying 257 probes in region: 54\n",
      "--- assemblying 206 probes in region: 55\n",
      "--- assemblying 125 probes in region: 56\n",
      "--- assemblying 232 probes in region: 57\n",
      "--- assemblying 112 probes in region: 58\n",
      "--- assemblying 159 probes in region: 59\n",
      "--- assemblying 252 probes in region: 60\n",
      "--- assemblying 281 probes in region: 61\n",
      "--- assemblying 265 probes in region: 62\n",
      "--- assemblying 245 probes in region: 63\n",
      "--- assemblying 303 probes in region: 64\n",
      "--- assemblying 231 probes in region: 65\n",
      "--- assemblying 310 probes in region: 66\n",
      "--- assemblying 197 probes in region: 67\n",
      "--- assemblying 283 probes in region: 68\n",
      "--- assemblying 113 probes in region: 69\n",
      "--- assemblying 209 probes in region: 70\n",
      "--- assemblying 274 probes in region: 71\n",
      "--- assemblying 251 probes in region: 72\n",
      "--- assemblying 282 probes in region: 73\n",
      "--- assemblying 276 probes in region: 74\n",
      "--- assemblying 278 probes in region: 75\n",
      "--- assemblying 265 probes in region: 76\n",
      "--- assemblying 193 probes in region: 77\n",
      "--- assemblying 287 probes in region: 80\n",
      "--- assemblying 240 probes in region: 81\n",
      "--- assemblying 223 probes in region: 82\n",
      "--- assemblying 233 probes in region: 83\n",
      "--- assemblying 290 probes in region: 84\n",
      "--- assemblying 173 probes in region: 85\n",
      "--- assemblying 202 probes in region: 86\n",
      "--- assemblying 154 probes in region: 87\n",
      "--- assemblying 117 probes in region: 88\n",
      "--- assemblying 318 probes in region: 89\n",
      "--- assemblying 315 probes in region: 90\n",
      "--- assemblying 306 probes in region: 91\n",
      "--- assemblying 338 probes in region: 92\n",
      "--- assemblying 284 probes in region: 93\n",
      "--- assemblying 207 probes in region: 94\n",
      "--- assemblying 185 probes in region: 95\n",
      "--- assemblying 207 probes in region: 96\n",
      "--- assemblying 147 probes in region: 97\n",
      "--- assemblying 249 probes in region: 98\n",
      "--- assemblying 192 probes in region: 99\n",
      "--- assemblying 82 probes in region: 100\n",
      "--- assemblying 122 probes in region: 101\n",
      "--- assemblying 267 probes in region: 102\n",
      "--- assemblying 220 probes in region: 103\n",
      "--- assemblying 201 probes in region: 104\n",
      "--- assemblying 271 probes in region: 105\n",
      "--- assemblying 213 probes in region: 106\n",
      "--- assemblying 246 probes in region: 107\n",
      "--- assemblying 152 probes in region: 108\n",
      "--- assemblying 263 probes in region: 109\n",
      "--- assemblying 248 probes in region: 110\n",
      "--- assemblying 97 probes in region: 111\n",
      "--- assemblying 205 probes in region: 112\n",
      "--- assemblying 91 probes in region: 113\n",
      "--- assemblying 163 probes in region: 114\n",
      "--- assemblying 242 probes in region: 115\n",
      "--- assemblying 296 probes in region: 116\n",
      "--- assemblying 175 probes in region: 117\n",
      "--- assemblying 111 probes in region: 118\n",
      "--- assemblying 203 probes in region: 119\n",
      "--- assemblying 279 probes in region: 120\n",
      "--- assemblying 223 probes in region: 121\n",
      "--- assemblying 231 probes in region: 122\n",
      "--- assemblying 102 probes in region: 123\n",
      "--- assemblying 255 probes in region: 124\n",
      "--- assemblying 191 probes in region: 125\n",
      "--- assemblying 318 probes in region: 126\n",
      "--- assemblying 224 probes in region: 127\n",
      "--- assemblying 167 probes in region: 128\n",
      "--- assemblying 147 probes in region: 129\n",
      "--- assemblying 274 probes in region: 130\n",
      "--- assemblying 261 probes in region: 131\n",
      "--- assemblying 248 probes in region: 132\n",
      "--- assemblying 159 probes in region: 133\n",
      "--- assemblying 140 probes in region: 134\n",
      "--- assemblying 232 probes in region: 135\n",
      "--- assemblying 235 probes in region: 136\n",
      "--- assemblying 313 probes in region: 137\n",
      "--- assemblying 199 probes in region: 138\n",
      "--- assemblying 226 probes in region: 140\n",
      "--- assemblying 259 probes in region: 141\n",
      "--- assemblying 308 probes in region: 142\n",
      "--- assemblying 248 probes in region: 143\n",
      "--- assemblying 307 probes in region: 144\n",
      "--- assemblying 302 probes in region: 145\n",
      "--- assemblying 282 probes in region: 146\n",
      "--- assemblying 252 probes in region: 147\n",
      "--- assemblying 268 probes in region: 148\n",
      "--- assemblying 258 probes in region: 150\n",
      "--- assemblying 285 probes in region: 151\n",
      "--- assemblying 187 probes in region: 153\n",
      "--- assemblying 159 probes in region: 154\n",
      "--- assemblying 265 probes in region: 155\n",
      "--- assemblying 144 probes in region: 156\n",
      "--- assemblying 199 probes in region: 157\n",
      "--- assemblying 127 probes in region: 158\n",
      "--- assemblying 270 probes in region: 159\n",
      "--- assemblying 141 probes in region: 160\n",
      "--- assemblying 168 probes in region: 161\n",
      "--- assemblying 221 probes in region: 162\n",
      "--- assemblying 257 probes in region: 163\n",
      "--- assemblying 288 probes in region: 164\n",
      "--- assemblying 225 probes in region: 165\n",
      "--- assemblying 103 probes in region: 166\n",
      "--- assemblying 164 probes in region: 167\n",
      "--- assemblying 179 probes in region: 168\n",
      "--- assemblying 140 probes in region: 169\n",
      "--- assemblying 86 probes in region: 170\n",
      "--- assemblying 139 probes in region: 171\n",
      "--- assemblying 222 probes in region: 172\n",
      "--- assemblying 225 probes in region: 173\n",
      "--- assemblying 160 probes in region: 174\n",
      "--- assemblying 210 probes in region: 175\n",
      "--- assemblying 213 probes in region: 176\n",
      "--- assemblying 268 probes in region: 177\n",
      "--- assemblying 261 probes in region: 178\n",
      "--- assemblying 163 probes in region: 179\n",
      "--- assemblying 136 probes in region: 180\n",
      "--- assemblying 181 probes in region: 181\n",
      "--- assemblying 99 probes in region: 182\n",
      "--- assemblying 110 probes in region: 183\n",
      "--- assemblying 126 probes in region: 184\n",
      "--- assemblying 290 probes in region: 185\n",
      "--- assemblying 182 probes in region: 186\n",
      "--- assemblying 216 probes in region: 187\n",
      "--- assemblying 114 probes in region: 188\n",
      "--- assemblying 229 probes in region: 190\n",
      "--- assemblying 142 probes in region: 191\n",
      "--- assemblying 251 probes in region: 192\n",
      "--- assemblying 275 probes in region: 193\n",
      "--- assemblying 186 probes in region: 195\n",
      "--- assemblying 115 probes in region: 196\n",
      "--- assemblying 115 probes in region: 197\n",
      "--- assemblying 289 probes in region: 198\n",
      "--- assemblying 175 probes in region: 199\n",
      "--- assemblying 258 probes in region: 200\n",
      "--- assemblying 169 probes in region: 201\n",
      "--- assemblying 180 probes in region: 202\n",
      "--- assemblying 197 probes in region: 203\n",
      "--- assemblying 314 probes in region: 204\n",
      "--- assemblying 264 probes in region: 205\n",
      "--- assemblying 198 probes in region: 206\n",
      "--- assemblying 294 probes in region: 207\n",
      "--- assemblying 191 probes in region: 208\n",
      "--- assemblying 290 probes in region: 209\n",
      "--- assemblying 295 probes in region: 210\n",
      "--- assemblying 184 probes in region: 211\n",
      "--- assemblying 160 probes in region: 212\n",
      "--- assemblying 241 probes in region: 213\n",
      "--- assemblying 180 probes in region: 214\n",
      "--- assemblying 122 probes in region: 215\n",
      "--- assemblying 89 probes in region: 218\n",
      "--- assemblying 164 probes in region: 219\n",
      "--- assemblying 298 probes in region: 220\n",
      "--- assemblying 203 probes in region: 221\n",
      "--- assemblying 296 probes in region: 222\n",
      "--- assemblying 158 probes in region: 223\n",
      "--- assemblying 104 probes in region: 224\n",
      "--- assemblying 84 probes in region: 225\n",
      "--- assemblying 184 probes in region: 226\n",
      "--- assemblying 119 probes in region: 227\n",
      "--- assemblying 84 probes in region: 228\n",
      "--- assemblying 244 probes in region: 229\n",
      "--- assemblying 264 probes in region: 230\n",
      "--- assemblying 143 probes in region: 231\n",
      "--- assemblying 110 probes in region: 232\n",
      "--- assemblying 223 probes in region: 233\n",
      "--- assemblying 194 probes in region: 234\n",
      "--- assemblying 151 probes in region: 235\n",
      "--- assemblying 220 probes in region: 236\n",
      "--- assemblying 108 probes in region: 237\n",
      "--- assemblying 258 probes in region: 238\n",
      "--- assemblying 257 probes in region: 239\n",
      "--- assemblying 139 probes in region: 240\n",
      "--- assemblying 299 probes in region: 242\n",
      "--- assemblying 251 probes in region: 243\n",
      "--- assemblying 237 probes in region: 244\n",
      "--- assemblying 154 probes in region: 245\n",
      "--- assemblying 151 probes in region: 246\n",
      "--- assemblying 228 probes in region: 247\n",
      "--- assemblying 135 probes in region: 248\n",
      "--- assemblying 211 probes in region: 249\n",
      "--- assemblying 100 probes in region: 250\n",
      "--- assemblying 207 probes in region: 251\n",
      "--- assemblying 137 probes in region: 252\n",
      "--- assemblying 283 probes in region: 253\n",
      "--- assemblying 139 probes in region: 254\n",
      "--- assemblying 119 probes in region: 255\n",
      "--- assemblying 230 probes in region: 256\n",
      "--- assemblying 231 probes in region: 257\n",
      "--- assemblying 300 probes in region: 258\n",
      "--- assemblying 207 probes in region: 259\n",
      "--- assemblying 160 probes in region: 260\n",
      "--- assemblying 137 probes in region: 261\n",
      "--- assemblying 101 probes in region: 262\n",
      "--- assemblying 183 probes in region: 263\n",
      "--- assemblying 186 probes in region: 264\n",
      "--- assemblying 213 probes in region: 266\n",
      "--- assemblying 309 probes in region: 267\n",
      "--- assemblying 299 probes in region: 268\n",
      "--- assemblying 306 probes in region: 269\n",
      "--- assemblying 292 probes in region: 270\n",
      "--- assemblying 312 probes in region: 271\n",
      "--- assemblying 325 probes in region: 272\n",
      "--- assemblying 300 probes in region: 273\n",
      "--- assemblying 123 probes in region: 274\n",
      "--- assemblying 248 probes in region: 276\n",
      "--- assemblying 319 probes in region: 277\n",
      "--- assemblying 158 probes in region: 278\n",
      "--- assemblying 255 probes in region: 279\n",
      "--- assemblying 187 probes in region: 281\n",
      "--- assemblying 141 probes in region: 282\n",
      "--- assemblying 119 probes in region: 283\n",
      "--- assemblying 93 probes in region: 285\n",
      "--- assemblying 196 probes in region: 286\n",
      "--- assemblying 134 probes in region: 287\n",
      "--- assemblying 217 probes in region: 288\n",
      "--- assemblying 252 probes in region: 289\n",
      "--- assemblying 103 probes in region: 290\n",
      "--- assemblying 229 probes in region: 291\n",
      "--- assemblying 316 probes in region: 292\n",
      "--- assemblying 252 probes in region: 293\n",
      "--- assemblying 218 probes in region: 294\n",
      "--- assemblying 257 probes in region: 295\n",
      "--- assemblying 237 probes in region: 296\n",
      "--- assemblying 236 probes in region: 297\n",
      "--- assemblying 151 probes in region: 298\n",
      "--- assemblying 212 probes in region: 299\n",
      "--- assemblying 174 probes in region: 300\n",
      "--- assemblying 119 probes in region: 301\n",
      "--- assemblying 115 probes in region: 302\n",
      "--- assemblying 184 probes in region: 303\n",
      "--- assemblying 130 probes in region: 304\n",
      "--- assemblying 222 probes in region: 305\n",
      "--- assemblying 188 probes in region: 306\n",
      "--- assemblying 149 probes in region: 307\n",
      "--- assemblying 108 probes in region: 308\n",
      "--- assemblying 324 probes in region: 309\n",
      "--- assemblying 205 probes in region: 310\n",
      "--- assemblying 189 probes in region: 311\n",
      "--- assemblying 204 probes in region: 312\n",
      "--- assemblying 135 probes in region: 313\n",
      "--- assemblying 210 probes in region: 314\n",
      "--- assemblying 183 probes in region: 315\n",
      "--- assemblying 192 probes in region: 316\n",
      "--- assemblying 247 probes in region: 318\n",
      "--- assemblying 205 probes in region: 319\n",
      "--- assemblying 200 probes in region: 320\n",
      "--- assemblying 274 probes in region: 321\n",
      "--- assemblying 210 probes in region: 322\n",
      "--- assemblying 242 probes in region: 323\n",
      "--- assemblying 303 probes in region: 324\n",
      "--- assemblying 337 probes in region: 325\n",
      "--- assemblying 237 probes in region: 326\n",
      "--- assemblying 343 probes in region: 327\n",
      "--- assemblying 248 probes in region: 328\n",
      "--- assemblying 237 probes in region: 329\n",
      "--- assemblying 284 probes in region: 330\n",
      "--- assemblying 341 probes in region: 331\n",
      "--- assemblying 235 probes in region: 332\n",
      "--- assemblying 243 probes in region: 333\n",
      "--- assemblying 284 probes in region: 334\n",
      "--- assemblying 318 probes in region: 335\n",
      "--- assemblying 285 probes in region: 336\n",
      "--- assemblying 281 probes in region: 337\n",
      "--- assemblying 259 probes in region: 338\n",
      "--- assemblying 288 probes in region: 339\n",
      "--- assemblying 249 probes in region: 340\n",
      "--- assemblying 258 probes in region: 342\n",
      "--- assemblying 219 probes in region: 343\n",
      "--- assemblying 261 probes in region: 344\n",
      "--- assemblying 150 probes in region: 345\n",
      "--- assemblying 182 probes in region: 346\n",
      "--- assemblying 53 probes in region: 347\n",
      "--- assemblying 139 probes in region: 348\n",
      "--- assemblying 188 probes in region: 349\n",
      "--- assemblying 239 probes in region: 350\n",
      "--- assemblying 69 probes in region: 351\n",
      "--- assemblying 318 probes in region: 352\n",
      "--- assemblying 268 probes in region: 353\n",
      "--- assemblying 175 probes in region: 354\n",
      "--- assemblying 209 probes in region: 355\n",
      "--- assemblying 317 probes in region: 356\n",
      "--- assemblying 138 probes in region: 357\n",
      "--- assemblying 158 probes in region: 358\n",
      "--- assemblying 227 probes in region: 359\n",
      "--- assemblying 221 probes in region: 360\n",
      "--- assemblying 235 probes in region: 361\n",
      "--- assemblying 260 probes in region: 362\n",
      "--- assemblying 116 probes in region: 363\n",
      "--- assemblying 163 probes in region: 364\n",
      "--- assemblying 203 probes in region: 365\n",
      "--- assemblying 195 probes in region: 366\n",
      "--- assemblying 133 probes in region: 367\n",
      "--- assemblying 129 probes in region: 368\n",
      "--- assemblying 226 probes in region: 369\n",
      "--- assemblying 175 probes in region: 370\n",
      "--- assemblying 123 probes in region: 371\n",
      "--- assemblying 210 probes in region: 372\n",
      "--- assemblying 235 probes in region: 373\n",
      "--- assemblying 305 probes in region: 374\n",
      "--- assemblying 285 probes in region: 375\n",
      "--- assemblying 315 probes in region: 376\n",
      "--- assemblying 326 probes in region: 377\n",
      "--- assemblying 270 probes in region: 378\n",
      "--- assemblying 305 probes in region: 379\n",
      "--- assemblying 272 probes in region: 380\n",
      "--- assemblying 327 probes in region: 381\n",
      "--- assemblying 242 probes in region: 382\n",
      "--- assemblying 233 probes in region: 383\n",
      "--- assemblying 212 probes in region: 384\n",
      "--- assemblying 230 probes in region: 385\n",
      "--- assemblying 329 probes in region: 386\n",
      "--- assemblying 269 probes in region: 387\n",
      "--- assemblying 323 probes in region: 388\n",
      "--- assemblying 266 probes in region: 389\n",
      "--- assemblying 103 probes in region: 390\n",
      "--- assemblying 153 probes in region: 391\n",
      "--- assemblying 114 probes in region: 392\n",
      "--- assemblying 93 probes in region: 393\n",
      "--- assemblying 260 probes in region: 395\n",
      "--- assemblying 127 probes in region: 396\n",
      "--- assemblying 268 probes in region: 397\n",
      "--- assemblying 274 probes in region: 398\n",
      "--- assemblying 252 probes in region: 399\n",
      "--- assemblying 245 probes in region: 400\n",
      "--- assemblying 172 probes in region: 403\n",
      "--- assemblying 211 probes in region: 407\n",
      "--- assemblying 76 probes in region: 408\n",
      "--- assemblying 126 probes in region: 411\n",
      "--- assemblying 261 probes in region: 412\n",
      "--- assemblying 98 probes in region: 413\n",
      "--- assemblying 96 probes in region: 414\n",
      "--- assemblying 250 probes in region: 415\n",
      "--- assemblying 133 probes in region: 418\n",
      "--- assemblying 309 probes in region: 419\n",
      "--- assemblying 169 probes in region: 420\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- assemblying 232 probes in region: 421\n",
      "--- assemblying 221 probes in region: 422\n",
      "--- assemblying 235 probes in region: 423\n",
      "--- assemblying 294 probes in region: 424\n",
      "--- assemblying 75 probes in region: 425\n",
      "--- assemblying 318 probes in region: 426\n",
      "--- assemblying 293 probes in region: 427\n",
      "--- assemblying 268 probes in region: 428\n",
      "--- assemblying 246 probes in region: 429\n",
      "--- assemblying 298 probes in region: 430\n",
      "--- assemblying 80 probes in region: 431\n",
      "--- assemblying 278 probes in region: 432\n",
      "--- assemblying 219 probes in region: 433\n",
      "--- assemblying 243 probes in region: 434\n",
      "--- assemblying 139 probes in region: 435\n",
      "--- assemblying 167 probes in region: 436\n",
      "--- assemblying 212 probes in region: 437\n",
      "--- assemblying 70 probes in region: 438\n",
      "--- assemblying 315 probes in region: 439\n",
      "--- assemblying 327 probes in region: 440\n",
      "--- assemblying 163 probes in region: 441\n",
      "--- assemblying 174 probes in region: 442\n",
      "--- assemblying 98 probes in region: 443\n",
      "--- assemblying 159 probes in region: 444\n",
      "--- assemblying 242 probes in region: 445\n",
      "--- assemblying 304 probes in region: 446\n",
      "--- assemblying 268 probes in region: 447\n",
      "--- assemblying 288 probes in region: 448\n",
      "--- assemblying 272 probes in region: 449\n",
      "--- assemblying 231 probes in region: 450\n",
      "--- assemblying 240 probes in region: 451\n",
      "--- assemblying 230 probes in region: 452\n",
      "--- assemblying 254 probes in region: 453\n",
      "--- assemblying 266 probes in region: 454\n",
      "--- assemblying 190 probes in region: 455\n",
      "--- assemblying 314 probes in region: 456\n",
      "--- assemblying 242 probes in region: 457\n",
      "--- assemblying 292 probes in region: 458\n",
      "--- assemblying 179 probes in region: 461\n",
      "--- assemblying 132 probes in region: 462\n",
      "--- assemblying 269 probes in region: 463\n",
      "--- assemblying 321 probes in region: 464\n",
      "--- assemblying 249 probes in region: 465\n",
      "--- assemblying 295 probes in region: 466\n",
      "--- assemblying 336 probes in region: 467\n",
      "--- assemblying 266 probes in region: 469\n",
      "--- assemblying 241 probes in region: 470\n",
      "--- assemblying 287 probes in region: 471\n",
      "--- assemblying 227 probes in region: 472\n",
      "--- assemblying 182 probes in region: 473\n",
      "--- assemblying 283 probes in region: 474\n",
      "--- assemblying 160 probes in region: 475\n",
      "--- assemblying 220 probes in region: 476\n",
      "--- assemblying 174 probes in region: 477\n",
      "--- assemblying 240 probes in region: 478\n",
      "--- assemblying 286 probes in region: 479\n",
      "--- assemblying 178 probes in region: 480\n",
      "--- assemblying 162 probes in region: 481\n",
      "--- assemblying 113 probes in region: 482\n",
      "--- assemblying 150 probes in region: 483\n",
      "--- assemblying 307 probes in region: 484\n",
      "--- assemblying 118 probes in region: 485\n",
      "--- assemblying 140 probes in region: 486\n",
      "--- assemblying 108 probes in region: 487\n",
      "--- assemblying 160 probes in region: 488\n",
      "--- assemblying 144 probes in region: 489\n",
      "--- assemblying 183 probes in region: 490\n",
      "--- assemblying 282 probes in region: 491\n",
      "--- assemblying 157 probes in region: 492\n",
      "--- assemblying 227 probes in region: 493\n",
      "--- assemblying 182 probes in region: 494\n",
      "--- assemblying 291 probes in region: 495\n",
      "--- assemblying 203 probes in region: 496\n",
      "--- assemblying 259 probes in region: 497\n",
      "--- assemblying 208 probes in region: 498\n",
      "--- assemblying 219 probes in region: 499\n",
      "--- assemblying 169 probes in region: 500\n",
      "--- assemblying 202 probes in region: 501\n",
      "--- assemblying 316 probes in region: 502\n",
      "--- assemblying 321 probes in region: 503\n",
      "--- assemblying 179 probes in region: 504\n",
      "--- assemblying 273 probes in region: 505\n",
      "--- assemblying 279 probes in region: 506\n",
      "--- assemblying 282 probes in region: 507\n",
      "--- assemblying 310 probes in region: 508\n",
      "--- assemblying 261 probes in region: 509\n",
      "--- assemblying 294 probes in region: 510\n",
      "--- assemblying 103 probes in region: 511\n",
      "--- assemblying 53 probes in region: 513\n",
      "--- assemblying 247 probes in region: 514\n",
      "--- assemblying 223 probes in region: 515\n",
      "--- assemblying 120 probes in region: 516\n",
      "--- assemblying 187 probes in region: 517\n",
      "--- assemblying 237 probes in region: 518\n",
      "--- assemblying 216 probes in region: 519\n",
      "--- assemblying 222 probes in region: 520\n",
      "--- assemblying 116 probes in region: 521\n",
      "--- assemblying 219 probes in region: 522\n",
      "--- assemblying 230 probes in region: 523\n",
      "--- assemblying 217 probes in region: 524\n",
      "--- assemblying 180 probes in region: 525\n",
      "--- assemblying 90 probes in region: 526\n",
      "--- assemblying 193 probes in region: 527\n",
      "--- assemblying 256 probes in region: 528\n",
      "--- assemblying 349 probes in region: 529\n",
      "--- assemblying 255 probes in region: 530\n",
      "--- assemblying 330 probes in region: 531\n",
      "--- assemblying 252 probes in region: 532\n",
      "--- assemblying 224 probes in region: 533\n",
      "--- assemblying 264 probes in region: 534\n",
      "--- assemblying 268 probes in region: 535\n",
      "--- assemblying 269 probes in region: 536\n",
      "--- assemblying 331 probes in region: 537\n",
      "--- assemblying 304 probes in region: 538\n",
      "--- assemblying 259 probes in region: 539\n",
      "--- assemblying 252 probes in region: 540\n",
      "--- assemblying 267 probes in region: 541\n",
      "--- assemblying 200 probes in region: 542\n",
      "--- assemblying 140 probes in region: 543\n",
      "--- assemblying 75 probes in region: 544\n",
      "--- assemblying 162 probes in region: 545\n",
      "--- assemblying 234 probes in region: 546\n",
      "--- assemblying 88 probes in region: 548\n",
      "--- assemblying 174 probes in region: 549\n",
      "--- assemblying 181 probes in region: 550\n",
      "--- assemblying 236 probes in region: 551\n",
      "--- assemblying 239 probes in region: 552\n",
      "--- assemblying 314 probes in region: 553\n",
      "--- assemblying 246 probes in region: 554\n",
      "--- assemblying 259 probes in region: 555\n",
      "--- assemblying 267 probes in region: 556\n",
      "--- assemblying 234 probes in region: 557\n",
      "--- assemblying 246 probes in region: 558\n",
      "--- assemblying 296 probes in region: 559\n",
      "--- assemblying 290 probes in region: 560\n",
      "--- assemblying 148 probes in region: 561\n",
      "--- assemblying 296 probes in region: 563\n",
      "--- assemblying 94 probes in region: 564\n",
      "--- assemblying 326 probes in region: 565\n",
      "--- assemblying 184 probes in region: 566\n",
      "--- assemblying 265 probes in region: 567\n",
      "--- assemblying 139 probes in region: 568\n",
      "--- assemblying 166 probes in region: 569\n",
      "--- assemblying 241 probes in region: 570\n",
      "--- assemblying 209 probes in region: 571\n",
      "--- assemblying 246 probes in region: 572\n",
      "--- assemblying 138 probes in region: 573\n",
      "--- assemblying 218 probes in region: 574\n",
      "--- assemblying 73 probes in region: 575\n",
      "--- assemblying 161 probes in region: 576\n",
      "--- assemblying 60 probes in region: 577\n",
      "--- assemblying 284 probes in region: 578\n",
      "--- assemblying 252 probes in region: 579\n",
      "--- assemblying 248 probes in region: 580\n",
      "--- assemblying 94 probes in region: 581\n",
      "--- assemblying 152 probes in region: 582\n",
      "--- assemblying 271 probes in region: 583\n",
      "--- assemblying 118 probes in region: 584\n",
      "--- assemblying 189 probes in region: 585\n",
      "--- assemblying 288 probes in region: 586\n",
      "--- assemblying 78 probes in region: 587\n",
      "--- assemblying 213 probes in region: 588\n",
      "--- assemblying 226 probes in region: 589\n",
      "--- assemblying 235 probes in region: 590\n",
      "--- assemblying 222 probes in region: 591\n",
      "--- assemblying 253 probes in region: 592\n",
      "--- assemblying 251 probes in region: 593\n",
      "--- assemblying 319 probes in region: 594\n",
      "--- assemblying 274 probes in region: 595\n",
      "--- assemblying 304 probes in region: 596\n",
      "--- assemblying 245 probes in region: 597\n",
      "--- assemblying 160 probes in region: 598\n",
      "--- assemblying 258 probes in region: 599\n",
      "--- assemblying 253 probes in region: 600\n",
      "--- assemblying 266 probes in region: 601\n",
      "--- assemblying 162 probes in region: 602\n",
      "--- assemblying 92 probes in region: 603\n",
      "--- assemblying 187 probes in region: 604\n",
      "--- assemblying 212 probes in region: 605\n",
      "--- assemblying 220 probes in region: 606\n",
      "--- assemblying 190 probes in region: 607\n",
      "--- assemblying 256 probes in region: 608\n",
      "--- assemblying 285 probes in region: 609\n",
      "--- assemblying 294 probes in region: 610\n",
      "--- assemblying 281 probes in region: 611\n",
      "--- assemblying 277 probes in region: 612\n",
      "--- assemblying 236 probes in region: 613\n",
      "--- assemblying 289 probes in region: 616\n",
      "--- assemblying 158 probes in region: 617\n",
      "--- assemblying 126 probes in region: 618\n",
      "--- assemblying 222 probes in region: 619\n",
      "--- assemblying 103 probes in region: 620\n",
      "--- assemblying 191 probes in region: 621\n",
      "--- assemblying 272 probes in region: 622\n",
      "--- assemblying 177 probes in region: 623\n",
      "--- assemblying 215 probes in region: 624\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- assemblying 107 probes in region: 625\n",
      "--- assemblying 175 probes in region: 626\n",
      "--- assemblying 75 probes in region: 627\n",
      "--- assemblying 332 probes in region: 628\n",
      "--- assemblying 315 probes in region: 629\n",
      "--- assemblying 192 probes in region: 630\n",
      "--- assemblying 205 probes in region: 631\n",
      "--- assemblying 213 probes in region: 632\n",
      "--- assemblying 296 probes in region: 633\n",
      "--- assemblying 168 probes in region: 634\n",
      "--- assemblying 234 probes in region: 635\n",
      "--- assemblying 168 probes in region: 636\n",
      "--- assemblying 255 probes in region: 637\n",
      "--- assemblying 159 probes in region: 638\n",
      "--- assemblying 283 probes in region: 639\n",
      "--- assemblying 286 probes in region: 640\n",
      "--- assemblying 263 probes in region: 641\n",
      "--- assemblying 285 probes in region: 642\n",
      "--- assemblying 272 probes in region: 643\n",
      "--- assemblying 131 probes in region: 644\n",
      "--- assemblying 303 probes in region: 645\n",
      "--- assemblying 234 probes in region: 646\n",
      "--- assemblying 285 probes in region: 647\n",
      "--- assemblying 252 probes in region: 648\n",
      "--- assemblying 228 probes in region: 649\n",
      "--- assemblying 260 probes in region: 650\n",
      "--- assemblying 239 probes in region: 651\n",
      "--- assemblying 211 probes in region: 652\n",
      "--- assemblying 233 probes in region: 653\n",
      "--- assemblying 308 probes in region: 654\n",
      "--- assemblying 251 probes in region: 655\n",
      "--- assemblying 320 probes in region: 656\n",
      "--- assemblying 290 probes in region: 657\n",
      "--- assemblying 269 probes in region: 658\n",
      "--- assemblying 257 probes in region: 659\n",
      "--- assemblying 284 probes in region: 660\n",
      "--- assemblying 284 probes in region: 661\n",
      "--- assemblying 322 probes in region: 662\n",
      "--- assemblying 271 probes in region: 663\n",
      "--- assemblying 270 probes in region: 665\n",
      "--- assemblying 177 probes in region: 666\n",
      "--- assemblying 241 probes in region: 667\n",
      "--- assemblying 196 probes in region: 668\n",
      "--- assemblying 239 probes in region: 669\n",
      "--- assemblying 273 probes in region: 670\n",
      "--- assemblying 280 probes in region: 672\n",
      "--- assemblying 268 probes in region: 674\n",
      "--- assemblying 251 probes in region: 675\n",
      "--- assemblying 246 probes in region: 676\n",
      "--- assemblying 106 probes in region: 677\n",
      "--- assemblying 275 probes in region: 678\n",
      "--- assemblying 186 probes in region: 679\n",
      "--- assemblying 199 probes in region: 680\n",
      "--- assemblying 59 probes in region: 681\n",
      "--- assemblying 138 probes in region: 682\n",
      "--- assemblying 163 probes in region: 683\n",
      "--- assemblying 231 probes in region: 684\n",
      "--- assemblying 282 probes in region: 685\n",
      "--- assemblying 268 probes in region: 686\n",
      "--- assemblying 192 probes in region: 687\n",
      "--- assemblying 172 probes in region: 688\n",
      "--- assemblying 115 probes in region: 689\n",
      "--- assemblying 128 probes in region: 690\n",
      "--- assemblying 140 probes in region: 691\n",
      "--- assemblying 275 probes in region: 692\n",
      "--- assemblying 311 probes in region: 693\n",
      "--- assemblying 194 probes in region: 694\n",
      "--- assemblying 211 probes in region: 695\n",
      "--- assemblying 266 probes in region: 696\n",
      "--- assemblying 242 probes in region: 697\n",
      "--- assemblying 255 probes in region: 698\n",
      "--- assemblying 289 probes in region: 699\n",
      "--- assemblying 167 probes in region: 700\n",
      "--- assemblying 147 probes in region: 701\n",
      "--- assemblying 128 probes in region: 702\n",
      "--- assemblying 246 probes in region: 703\n",
      "--- assemblying 281 probes in region: 706\n",
      "--- assemblying 222 probes in region: 707\n",
      "--- assemblying 288 probes in region: 708\n",
      "--- assemblying 112 probes in region: 709\n",
      "--- assemblying 227 probes in region: 710\n",
      "--- assemblying 244 probes in region: 711\n",
      "--- assemblying 288 probes in region: 713\n",
      "--- assemblying 268 probes in region: 714\n",
      "--- assemblying 246 probes in region: 715\n",
      "--- assemblying 101 probes in region: 716\n",
      "--- assemblying 253 probes in region: 717\n",
      "--- assemblying 123 probes in region: 718\n",
      "--- assemblying 239 probes in region: 719\n",
      "--- assemblying 268 probes in region: 720\n",
      "--- assemblying 225 probes in region: 721\n",
      "--- assemblying 136 probes in region: 722\n",
      "--- assemblying 265 probes in region: 723\n",
      "--- assemblying 276 probes in region: 724\n",
      "--- assemblying 252 probes in region: 726\n",
      "--- assemblying 279 probes in region: 727\n",
      "--- assemblying 257 probes in region: 728\n",
      "--- assemblying 244 probes in region: 729\n",
      "--- assemblying 295 probes in region: 730\n",
      "--- assemblying 301 probes in region: 731\n",
      "--- assemblying 269 probes in region: 732\n",
      "--- assemblying 281 probes in region: 733\n",
      "--- assemblying 318 probes in region: 734\n",
      "--- assemblying 274 probes in region: 735\n",
      "--- assemblying 173 probes in region: 736\n",
      "--- assemblying 148 probes in region: 737\n",
      "--- assemblying 293 probes in region: 739\n",
      "--- assemblying 262 probes in region: 740\n",
      "--- assemblying 278 probes in region: 741\n",
      "--- assemblying 235 probes in region: 742\n",
      "--- assemblying 91 probes in region: 743\n",
      "--- assemblying 144 probes in region: 744\n",
      "--- assemblying 225 probes in region: 745\n",
      "--- assemblying 149 probes in region: 746\n",
      "--- assemblying 118 probes in region: 747\n",
      "--- assemblying 160 probes in region: 748\n",
      "--- assemblying 224 probes in region: 749\n",
      "--- assemblying 216 probes in region: 750\n",
      "--- assemblying 166 probes in region: 751\n",
      "--- assemblying 264 probes in region: 752\n",
      "--- assemblying 225 probes in region: 753\n",
      "--- assemblying 186 probes in region: 754\n",
      "--- assemblying 252 probes in region: 755\n",
      "--- assemblying 281 probes in region: 756\n",
      "--- assemblying 233 probes in region: 757\n",
      "--- assemblying 197 probes in region: 758\n",
      "--- assemblying 223 probes in region: 759\n",
      "--- assemblying 242 probes in region: 765\n",
      "--- assemblying 97 probes in region: 766\n",
      "--- assemblying 272 probes in region: 767\n",
      "--- assemblying 123 probes in region: 768\n",
      "--- assemblying 230 probes in region: 769\n",
      "--- assemblying 313 probes in region: 770\n",
      "--- assemblying 238 probes in region: 772\n",
      "--- assemblying 293 probes in region: 773\n",
      "--- assemblying 257 probes in region: 774\n",
      "--- assemblying 185 probes in region: 775\n",
      "--- assemblying 168 probes in region: 776\n",
      "--- assemblying 233 probes in region: 779\n",
      "--- assemblying 237 probes in region: 780\n",
      "--- assemblying 148 probes in region: 781\n",
      "--- assemblying 72 probes in region: 783\n",
      "--- assemblying 210 probes in region: 784\n",
      "--- assemblying 230 probes in region: 785\n",
      "--- assemblying 319 probes in region: 786\n",
      "--- assemblying 252 probes in region: 787\n",
      "--- assemblying 146 probes in region: 788\n",
      "--- assemblying 291 probes in region: 789\n",
      "--- assemblying 214 probes in region: 790\n",
      "--- assemblying 281 probes in region: 791\n",
      "--- assemblying 213 probes in region: 792\n",
      "--- assemblying 170 probes in region: 793\n",
      "--- assemblying 126 probes in region: 794\n",
      "--- assemblying 232 probes in region: 795\n",
      "--- assemblying 144 probes in region: 796\n",
      "--- assemblying 173 probes in region: 797\n",
      "--- assemblying 137 probes in region: 798\n",
      "--- assemblying 132 probes in region: 799\n",
      "--- assemblying 169 probes in region: 800\n",
      "--- assemblying 200 probes in region: 801\n",
      "--- assemblying 275 probes in region: 802\n",
      "--- assemblying 188 probes in region: 803\n",
      "--- assemblying 176 probes in region: 804\n",
      "--- assemblying 137 probes in region: 805\n",
      "--- assemblying 135 probes in region: 806\n",
      "--- assemblying 199 probes in region: 807\n",
      "--- assemblying 218 probes in region: 808\n",
      "--- assemblying 344 probes in region: 809\n",
      "--- assemblying 178 probes in region: 810\n",
      "--- assemblying 179 probes in region: 812\n",
      "--- assemblying 237 probes in region: 813\n",
      "--- assemblying 186 probes in region: 814\n",
      "--- assemblying 228 probes in region: 815\n",
      "--- assemblying 181 probes in region: 816\n",
      "--- assemblying 183 probes in region: 817\n",
      "--- assemblying 198 probes in region: 818\n",
      "--- assemblying 182 probes in region: 819\n",
      "--- assemblying 115 probes in region: 820\n",
      "--- assemblying 79 probes in region: 821\n",
      "--- assemblying 196 probes in region: 822\n",
      "--- assemblying 341 probes in region: 823\n",
      "--- assemblying 99 probes in region: 824\n",
      "--- assemblying 216 probes in region: 825\n",
      "--- assemblying 143 probes in region: 827\n",
      "--- assemblying 106 probes in region: 828\n",
      "--- assemblying 122 probes in region: 829\n",
      "--- assemblying 168 probes in region: 830\n",
      "--- assemblying 227 probes in region: 831\n",
      "--- assemblying 232 probes in region: 832\n",
      "--- assemblying 153 probes in region: 833\n",
      "--- assemblying 297 probes in region: 834\n",
      "--- assemblying 220 probes in region: 835\n",
      "--- assemblying 296 probes in region: 836\n",
      "--- assemblying 312 probes in region: 837\n",
      "--- assemblying 261 probes in region: 838\n",
      "--- assemblying 240 probes in region: 839\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- assemblying 333 probes in region: 840\n",
      "--- assemblying 328 probes in region: 841\n",
      "--- assemblying 299 probes in region: 842\n",
      "--- assemblying 248 probes in region: 843\n",
      "--- assemblying 268 probes in region: 844\n",
      "--- assemblying 277 probes in region: 845\n",
      "--- assemblying 237 probes in region: 846\n",
      "--- assemblying 258 probes in region: 847\n",
      "--- assemblying 262 probes in region: 848\n",
      "--- assemblying 227 probes in region: 849\n",
      "--- assemblying 248 probes in region: 850\n",
      "--- assemblying 289 probes in region: 851\n",
      "--- assemblying 148 probes in region: 852\n",
      "--- assemblying 234 probes in region: 854\n",
      "--- assemblying 192 probes in region: 855\n",
      "--- assemblying 237 probes in region: 856\n",
      "--- assemblying 231 probes in region: 857\n",
      "--- assemblying 204 probes in region: 858\n",
      "--- assemblying 234 probes in region: 859\n",
      "--- assemblying 217 probes in region: 860\n",
      "--- assemblying 323 probes in region: 861\n",
      "--- assemblying 101 probes in region: 862\n",
      "--- assemblying 150 probes in region: 863\n",
      "--- assemblying 258 probes in region: 864\n",
      "--- assemblying 259 probes in region: 865\n",
      "--- assemblying 251 probes in region: 866\n",
      "--- assemblying 145 probes in region: 867\n",
      "--- assemblying 294 probes in region: 868\n",
      "--- assemblying 213 probes in region: 869\n",
      "--- assemblying 257 probes in region: 870\n",
      "--- assemblying 184 probes in region: 871\n",
      "--- assemblying 203 probes in region: 872\n",
      "--- assemblying 93 probes in region: 873\n",
      "--- assemblying 116 probes in region: 874\n",
      "--- assemblying 217 probes in region: 875\n",
      "--- assemblying 196 probes in region: 876\n",
      "--- assemblying 76 probes in region: 877\n",
      "--- assemblying 194 probes in region: 878\n",
      "--- assemblying 133 probes in region: 879\n",
      "--- assemblying 88 probes in region: 880\n",
      "--- assemblying 192 probes in region: 881\n",
      "--- assemblying 220 probes in region: 882\n",
      "--- assemblying 277 probes in region: 883\n",
      "--- assemblying 192 probes in region: 884\n",
      "--- assemblying 180 probes in region: 885\n",
      "--- assemblying 145 probes in region: 886\n",
      "--- assemblying 210 probes in region: 887\n",
      "--- assemblying 220 probes in region: 888\n",
      "--- assemblying 194 probes in region: 889\n",
      "--- assemblying 295 probes in region: 890\n",
      "--- assemblying 173 probes in region: 891\n",
      "--- assemblying 228 probes in region: 894\n",
      "--- assemblying 316 probes in region: 896\n",
      "--- assemblying 259 probes in region: 897\n",
      "--- assemblying 275 probes in region: 898\n",
      "--- assemblying 179 probes in region: 899\n",
      "--- assemblying 67 probes in region: 900\n",
      "--- assemblying 95 probes in region: 901\n",
      "--- assemblying 288 probes in region: 902\n",
      "--- assemblying 195 probes in region: 903\n",
      "--- assemblying 253 probes in region: 904\n",
      "--- assemblying 237 probes in region: 905\n",
      "--- assemblying 254 probes in region: 906\n",
      "--- assemblying 200 probes in region: 907\n",
      "--- assemblying 87 probes in region: 908\n",
      "--- assemblying 180 probes in region: 909\n",
      "--- assemblying 254 probes in region: 910\n",
      "--- assemblying 331 probes in region: 911\n",
      "--- assemblying 298 probes in region: 912\n",
      "--- assemblying 216 probes in region: 913\n",
      "--- assemblying 72 probes in region: 914\n",
      "--- assemblying 256 probes in region: 915\n",
      "--- assemblying 182 probes in region: 916\n",
      "--- assemblying 159 probes in region: 917\n",
      "--- assemblying 163 probes in region: 918\n",
      "--- assemblying 273 probes in region: 919\n",
      "--- assemblying 287 probes in region: 920\n",
      "--- assemblying 325 probes in region: 921\n",
      "--- assemblying 287 probes in region: 922\n",
      "--- assemblying 160 probes in region: 923\n",
      "--- assemblying 197 probes in region: 924\n",
      "--- assemblying 79 probes in region: 925\n",
      "--- assemblying 255 probes in region: 926\n",
      "--- assemblying 305 probes in region: 927\n",
      "--- assemblying 266 probes in region: 928\n",
      "--- assemblying 125 probes in region: 929\n",
      "--- assemblying 81 probes in region: 930\n",
      "--- assemblying 66 probes in region: 932\n",
      "--- assemblying 186 probes in region: 933\n",
      "--- assemblying 198 probes in region: 934\n",
      "--- assemblying 230 probes in region: 935\n",
      "--- assemblying 293 probes in region: 936\n",
      "--- assemblying 200 probes in region: 937\n",
      "--- assemblying 164 probes in region: 938\n",
      "--- assemblying 292 probes in region: 939\n",
      "--- assemblying 258 probes in region: 940\n",
      "--- assemblying 184 probes in region: 941\n",
      "--- assemblying 136 probes in region: 942\n",
      "--- assemblying 173 probes in region: 943\n",
      "--- assemblying 257 probes in region: 944\n",
      "--- assemblying 216 probes in region: 945\n",
      "--- assemblying 262 probes in region: 946\n",
      "--- assemblying 244 probes in region: 947\n",
      "--- assemblying 187 probes in region: 948\n",
      "--- assemblying 247 probes in region: 949\n",
      "--- assemblying 154 probes in region: 950\n",
      "--- assemblying 212 probes in region: 951\n",
      "--- assemblying 260 probes in region: 952\n",
      "--- assemblying 244 probes in region: 953\n",
      "--- assemblying 190 probes in region: 954\n",
      "--- assemblying 270 probes in region: 955\n",
      "--- assemblying 196 probes in region: 956\n",
      "--- assemblying 238 probes in region: 957\n",
      "--- assemblying 254 probes in region: 958\n",
      "--- assemblying 209 probes in region: 959\n",
      "--- assemblying 288 probes in region: 960\n",
      "--- assemblying 287 probes in region: 961\n",
      "--- assemblying 278 probes in region: 962\n",
      "--- assemblying 282 probes in region: 963\n",
      "--- assemblying 269 probes in region: 964\n",
      "--- assemblying 125 probes in region: 965\n",
      "--- assemblying 213 probes in region: 966\n",
      "--- assemblying 182 probes in region: 969\n",
      "--- assemblying 306 probes in region: 970\n",
      "--- assemblying 262 probes in region: 971\n",
      "--- assemblying 235 probes in region: 972\n",
      "--- assemblying 121 probes in region: 973\n",
      "--- assemblying 283 probes in region: 974\n",
      "--- assemblying 276 probes in region: 975\n",
      "--- assemblying 188 probes in region: 976\n",
      "--- assemblying 211 probes in region: 977\n",
      "--- assemblying 190 probes in region: 978\n",
      "--- assemblying 227 probes in region: 979\n",
      "--- assemblying 240 probes in region: 980\n",
      "--- assemblying 251 probes in region: 982\n",
      "--- assemblying 232 probes in region: 983\n",
      "--- assemblying 234 probes in region: 984\n",
      "--- assemblying 241 probes in region: 985\n",
      "--- assemblying 269 probes in region: 986\n",
      "--- assemblying 237 probes in region: 987\n",
      "--- assemblying 139 probes in region: 988\n",
      "--- assemblying 284 probes in region: 989\n",
      "--- assemblying 268 probes in region: 990\n",
      "--- assemblying 265 probes in region: 991\n",
      "--- assemblying 100 probes in region: 992\n",
      "--- assemblying 224 probes in region: 995\n",
      "--- assemblying 305 probes in region: 997\n",
      "--- assemblying 149 probes in region: 998\n",
      "--- assemblying 133 probes in region: 999\n",
      "--- assemblying 55 probes in region: 1000\n",
      "--- assemblying 69 probes in region: 1001\n",
      "--- assemblying 117 probes in region: 1002\n",
      "--- assemblying 209 probes in region: 1007\n",
      "--- assemblying 136 probes in region: 1008\n",
      "--- assemblying 152 probes in region: 1009\n",
      "--- assemblying 110 probes in region: 1010\n",
      "--- assemblying 232 probes in region: 1011\n",
      "--- assemblying 146 probes in region: 1013\n",
      "--- assemblying 211 probes in region: 1014\n",
      "--- assemblying 65 probes in region: 1015\n",
      "--- assemblying 176 probes in region: 1016\n",
      "--- assemblying 117 probes in region: 1017\n",
      "--- assemblying 145 probes in region: 1018\n",
      "--- assemblying 204 probes in region: 1019\n",
      "--- assemblying 133 probes in region: 1020\n",
      "--- assemblying 173 probes in region: 1021\n",
      "--- assemblying 196 probes in region: 1022\n",
      "--- assemblying 112 probes in region: 1023\n",
      "--- assemblying 87 probes in region: 1024\n",
      "--- assemblying 65 probes in region: 1025\n",
      "--- assemblying 130 probes in region: 1026\n",
      "--- assemblying 78 probes in region: 1027\n",
      "--- assemblying 102 probes in region: 1028\n",
      "--- assemblying 127 probes in region: 1029\n",
      "--- assemblying 301 probes in region: 1030\n",
      "--- assemblying 172 probes in region: 1031\n",
      "--- assemblying 163 probes in region: 1032\n",
      "--- assemblying 244 probes in region: 1033\n",
      "--- assemblying 224 probes in region: 1034\n",
      "--- assemblying 126 probes in region: 1035\n",
      "--- assemblying 192 probes in region: 1036\n",
      "--- assemblying 146 probes in region: 1038\n",
      "--- assemblying 111 probes in region: 1039\n",
      "--- assemblying 62 probes in region: 1040\n",
      "--- assemblying 106 probes in region: 1041\n",
      "--- assemblying 183 probes in region: 1044\n",
      "--- assemblying 172 probes in region: 1045\n",
      "--- assemblying 221 probes in region: 1046\n",
      "--- assemblying 268 probes in region: 1047\n",
      "--- assemblying 219 probes in region: 1048\n",
      "--- assemblying 181 probes in region: 1049\n",
      "--- assemblying 190 probes in region: 1050\n",
      "--- assemblying 147 probes in region: 1051\n",
      "--- assemblying 96 probes in region: 1053\n",
      "--- assemblying 171 probes in region: 1054\n",
      "--- assemblying 150 probes in region: 1055\n",
      "--- assemblying 213 probes in region: 1056\n",
      "--- assemblying 193 probes in region: 1057\n",
      "--- assemblying 257 probes in region: 1058\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- assemblying 221 probes in region: 1059\n",
      "--- assemblying 122 probes in region: 1060\n",
      "--- assemblying 168 probes in region: 1062\n",
      "-- 210584 probes assembled in total.\n",
      "-- saving 210584 probes into file:\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\candidate_probes.fasta\n",
      "-- saving readout_summary into file:\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\readout_summary.pkl\n"
     ]
    }
   ],
   "source": [
    "reload(library_tools.assemble)\n",
    "from ImageAnalysis3.library_tools.assemble import Assemble_probes\n",
    "\n",
    "cand_probes, readout_summary = Assemble_probes(library_folder, filtered_pb_dict, \n",
    "                                               region_2_readout_dict, readout_dict, primers, \n",
    "                                               primer_len=20, \n",
    "                                               num_readout_per_probe=3, \n",
    "                                               unique_readout_per_probe=False,\n",
    "                                               rc_targets=False, overwrite=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "25a09e19",
   "metadata": {},
   "source": [
    "# 4.Check quality"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "70b60bb2",
   "metadata": {},
   "source": [
    "## 4.1 Load probes\n",
    "\n",
    "Checking procedures could restart from here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "21b21f7a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total probe loaded: 210584\n"
     ]
    }
   ],
   "source": [
    "# folder for this library\n",
    "# candidate full-length probe filename\n",
    "candidate_basename = 'candidate_probes.fasta' # old version\n",
    "# load full probes\n",
    "full_records = []\n",
    "with open(os.path.join(library_folder, candidate_basename), 'r') as handle:\n",
    "    for record in SeqIO.parse(handle, \"fasta\"):\n",
    "        full_records.append(record)\n",
    "\n",
    "print(f\"Total probe loaded: {len(full_records)}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c27e7e91",
   "metadata": {},
   "outputs": [],
   "source": [
    "num_readout_per_probe = 3\n",
    "primer_len=20\n",
    "target_len=42\n",
    "min_num_probes = 40"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2da410b1",
   "metadata": {},
   "source": [
    "### 4.1.1 check primer usage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2058ae11",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Checking primer usage, total probes: 210584\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "import ImageAnalysis3.library_tools.quality_check as check\n",
    "reload(check)\n",
    "primer_check = check._check_primer_usage(full_records, primers[0], primers[1], primer_len=primer_len)\n",
    "print(primer_check)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30ce1025",
   "metadata": {},
   "source": [
    "### 4.1.2 check region probe number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "01c5f87b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gene: 1 -> 163\n",
      "gene: 2 -> 180\n",
      "gene: 3 -> 77\n",
      "gene: 4 -> 205\n",
      "gene: 5 -> 144\n",
      "gene: 6 -> 297\n",
      "gene: 7 -> 210\n",
      "gene: 8 -> 274\n",
      "gene: 9 -> 220\n",
      "gene: 10 -> 158\n",
      "gene: 11 -> 156\n",
      "gene: 12 -> 107\n",
      "gene: 13 -> 291\n",
      "gene: 14 -> 240\n",
      "gene: 15 -> 279\n",
      "gene: 16 -> 248\n",
      "gene: 17 -> 276\n",
      "gene: 18 -> 109\n",
      "gene: 19 -> 111\n",
      "gene: 20 -> 212\n",
      "gene: 21 -> 135\n",
      "gene: 22 -> 229\n",
      "gene: 23 -> 257\n",
      "gene: 24 -> 257\n",
      "gene: 25 -> 249\n",
      "gene: 26 -> 204\n",
      "gene: 27 -> 195\n",
      "gene: 28 -> 198\n",
      "gene: 29 -> 284\n",
      "gene: 30 -> 245\n",
      "gene: 31 -> 151\n",
      "gene: 32 -> 247\n",
      "gene: 33 -> 276\n",
      "gene: 34 -> 300\n",
      "gene: 35 -> 310\n",
      "gene: 36 -> 260\n",
      "gene: 37 -> 244\n",
      "gene: 38 -> 75\n",
      "gene: 39 -> 178\n",
      "gene: 40 -> 142\n",
      "gene: 41 -> 122\n",
      "gene: 42 -> 105\n",
      "gene: 43 -> 173\n",
      "gene: 44 -> 90\n",
      "gene: 45 -> 121\n",
      "gene: 46 -> 241\n",
      "gene: 47 -> 240\n",
      "gene: 48 -> 218\n",
      "gene: 49 -> 225\n",
      "gene: 50 -> 287\n",
      "gene: 51 -> 250\n",
      "gene: 52 -> 329\n",
      "gene: 53 -> 299\n",
      "gene: 54 -> 257\n",
      "gene: 55 -> 206\n",
      "gene: 56 -> 125\n",
      "gene: 57 -> 232\n",
      "gene: 58 -> 112\n",
      "gene: 59 -> 159\n",
      "gene: 60 -> 252\n",
      "gene: 61 -> 281\n",
      "gene: 62 -> 265\n",
      "gene: 63 -> 245\n",
      "gene: 64 -> 303\n",
      "gene: 65 -> 231\n",
      "gene: 66 -> 310\n",
      "gene: 67 -> 197\n",
      "gene: 68 -> 283\n",
      "gene: 69 -> 113\n",
      "gene: 70 -> 209\n",
      "gene: 71 -> 274\n",
      "gene: 72 -> 251\n",
      "gene: 73 -> 282\n",
      "gene: 74 -> 276\n",
      "gene: 75 -> 278\n",
      "gene: 76 -> 265\n",
      "gene: 77 -> 193\n",
      "gene: 80 -> 287\n",
      "gene: 81 -> 240\n",
      "gene: 82 -> 223\n",
      "gene: 83 -> 233\n",
      "gene: 84 -> 290\n",
      "gene: 85 -> 173\n",
      "gene: 86 -> 202\n",
      "gene: 87 -> 154\n",
      "gene: 88 -> 117\n",
      "gene: 89 -> 318\n",
      "gene: 90 -> 315\n",
      "gene: 91 -> 306\n",
      "gene: 92 -> 338\n",
      "gene: 93 -> 284\n",
      "gene: 94 -> 207\n",
      "gene: 95 -> 185\n",
      "gene: 96 -> 207\n",
      "gene: 97 -> 147\n",
      "gene: 98 -> 249\n",
      "gene: 99 -> 192\n",
      "gene: 100 -> 82\n",
      "gene: 101 -> 122\n",
      "gene: 102 -> 267\n",
      "gene: 103 -> 220\n",
      "gene: 104 -> 201\n",
      "gene: 105 -> 271\n",
      "gene: 106 -> 213\n",
      "gene: 107 -> 246\n",
      "gene: 108 -> 152\n",
      "gene: 109 -> 263\n",
      "gene: 110 -> 248\n",
      "gene: 111 -> 97\n",
      "gene: 112 -> 205\n",
      "gene: 113 -> 91\n",
      "gene: 114 -> 163\n",
      "gene: 115 -> 242\n",
      "gene: 116 -> 296\n",
      "gene: 117 -> 175\n",
      "gene: 118 -> 111\n",
      "gene: 119 -> 203\n",
      "gene: 120 -> 279\n",
      "gene: 121 -> 223\n",
      "gene: 122 -> 231\n",
      "gene: 123 -> 102\n",
      "gene: 124 -> 255\n",
      "gene: 125 -> 191\n",
      "gene: 126 -> 318\n",
      "gene: 127 -> 224\n",
      "gene: 128 -> 167\n",
      "gene: 129 -> 147\n",
      "gene: 130 -> 274\n",
      "gene: 131 -> 261\n",
      "gene: 132 -> 248\n",
      "gene: 133 -> 159\n",
      "gene: 134 -> 140\n",
      "gene: 135 -> 232\n",
      "gene: 136 -> 235\n",
      "gene: 137 -> 313\n",
      "gene: 138 -> 199\n",
      "gene: 140 -> 226\n",
      "gene: 141 -> 259\n",
      "gene: 142 -> 308\n",
      "gene: 143 -> 248\n",
      "gene: 144 -> 307\n",
      "gene: 145 -> 302\n",
      "gene: 146 -> 282\n",
      "gene: 147 -> 252\n",
      "gene: 148 -> 268\n",
      "gene: 150 -> 258\n",
      "gene: 151 -> 285\n",
      "gene: 153 -> 187\n",
      "gene: 154 -> 159\n",
      "gene: 155 -> 265\n",
      "gene: 156 -> 144\n",
      "gene: 157 -> 199\n",
      "gene: 158 -> 127\n",
      "gene: 159 -> 270\n",
      "gene: 160 -> 141\n",
      "gene: 161 -> 168\n",
      "gene: 162 -> 221\n",
      "gene: 163 -> 257\n",
      "gene: 164 -> 288\n",
      "gene: 165 -> 225\n",
      "gene: 166 -> 103\n",
      "gene: 167 -> 164\n",
      "gene: 168 -> 179\n",
      "gene: 169 -> 140\n",
      "gene: 170 -> 86\n",
      "gene: 171 -> 139\n",
      "gene: 172 -> 222\n",
      "gene: 173 -> 225\n",
      "gene: 174 -> 160\n",
      "gene: 175 -> 210\n",
      "gene: 176 -> 213\n",
      "gene: 177 -> 268\n",
      "gene: 178 -> 261\n",
      "gene: 179 -> 163\n",
      "gene: 180 -> 136\n",
      "gene: 181 -> 181\n",
      "gene: 182 -> 99\n",
      "gene: 183 -> 110\n",
      "gene: 184 -> 126\n",
      "gene: 185 -> 290\n",
      "gene: 186 -> 182\n",
      "gene: 187 -> 216\n",
      "gene: 188 -> 114\n",
      "gene: 190 -> 229\n",
      "gene: 191 -> 142\n",
      "gene: 192 -> 251\n",
      "gene: 193 -> 275\n",
      "gene: 195 -> 186\n",
      "gene: 196 -> 115\n",
      "gene: 197 -> 115\n",
      "gene: 198 -> 289\n",
      "gene: 199 -> 175\n",
      "gene: 200 -> 258\n",
      "gene: 201 -> 169\n",
      "gene: 202 -> 180\n",
      "gene: 203 -> 197\n",
      "gene: 204 -> 314\n",
      "gene: 205 -> 264\n",
      "gene: 206 -> 198\n",
      "gene: 207 -> 294\n",
      "gene: 208 -> 191\n",
      "gene: 209 -> 290\n",
      "gene: 210 -> 295\n",
      "gene: 211 -> 184\n",
      "gene: 212 -> 160\n",
      "gene: 213 -> 241\n",
      "gene: 214 -> 180\n",
      "gene: 215 -> 122\n",
      "gene: 218 -> 89\n",
      "gene: 219 -> 164\n",
      "gene: 220 -> 298\n",
      "gene: 221 -> 203\n",
      "gene: 222 -> 296\n",
      "gene: 223 -> 158\n",
      "gene: 224 -> 104\n",
      "gene: 225 -> 84\n",
      "gene: 226 -> 184\n",
      "gene: 227 -> 119\n",
      "gene: 228 -> 84\n",
      "gene: 229 -> 244\n",
      "gene: 230 -> 264\n",
      "gene: 231 -> 143\n",
      "gene: 232 -> 110\n",
      "gene: 233 -> 223\n",
      "gene: 234 -> 194\n",
      "gene: 235 -> 151\n",
      "gene: 236 -> 220\n",
      "gene: 237 -> 108\n",
      "gene: 238 -> 258\n",
      "gene: 239 -> 257\n",
      "gene: 240 -> 139\n",
      "gene: 242 -> 299\n",
      "gene: 243 -> 251\n",
      "gene: 244 -> 237\n",
      "gene: 245 -> 154\n",
      "gene: 246 -> 151\n",
      "gene: 247 -> 228\n",
      "gene: 248 -> 135\n",
      "gene: 249 -> 211\n",
      "gene: 250 -> 100\n",
      "gene: 251 -> 207\n",
      "gene: 252 -> 137\n",
      "gene: 253 -> 283\n",
      "gene: 254 -> 139\n",
      "gene: 255 -> 119\n",
      "gene: 256 -> 230\n",
      "gene: 257 -> 231\n",
      "gene: 258 -> 300\n",
      "gene: 259 -> 207\n",
      "gene: 260 -> 160\n",
      "gene: 261 -> 137\n",
      "gene: 262 -> 101\n",
      "gene: 263 -> 183\n",
      "gene: 264 -> 186\n",
      "gene: 266 -> 213\n",
      "gene: 267 -> 309\n",
      "gene: 268 -> 299\n",
      "gene: 269 -> 306\n",
      "gene: 270 -> 292\n",
      "gene: 271 -> 312\n",
      "gene: 272 -> 325\n",
      "gene: 273 -> 300\n",
      "gene: 274 -> 123\n",
      "gene: 276 -> 248\n",
      "gene: 277 -> 319\n",
      "gene: 278 -> 158\n",
      "gene: 279 -> 255\n",
      "gene: 281 -> 187\n",
      "gene: 282 -> 141\n",
      "gene: 283 -> 119\n",
      "gene: 285 -> 93\n",
      "gene: 286 -> 196\n",
      "gene: 287 -> 134\n",
      "gene: 288 -> 217\n",
      "gene: 289 -> 252\n",
      "gene: 290 -> 103\n",
      "gene: 291 -> 229\n",
      "gene: 292 -> 316\n",
      "gene: 293 -> 252\n",
      "gene: 294 -> 218\n",
      "gene: 295 -> 257\n",
      "gene: 296 -> 237\n",
      "gene: 297 -> 236\n",
      "gene: 298 -> 151\n",
      "gene: 299 -> 212\n",
      "gene: 300 -> 174\n",
      "gene: 301 -> 119\n",
      "gene: 302 -> 115\n",
      "gene: 303 -> 184\n",
      "gene: 304 -> 130\n",
      "gene: 305 -> 222\n",
      "gene: 306 -> 188\n",
      "gene: 307 -> 149\n",
      "gene: 308 -> 108\n",
      "gene: 309 -> 324\n",
      "gene: 310 -> 205\n",
      "gene: 311 -> 189\n",
      "gene: 312 -> 204\n",
      "gene: 313 -> 135\n",
      "gene: 314 -> 210\n",
      "gene: 315 -> 183\n",
      "gene: 316 -> 192\n",
      "gene: 318 -> 247\n",
      "gene: 319 -> 205\n",
      "gene: 320 -> 200\n",
      "gene: 321 -> 274\n",
      "gene: 322 -> 210\n",
      "gene: 323 -> 242\n",
      "gene: 324 -> 303\n",
      "gene: 325 -> 337\n",
      "gene: 326 -> 237\n",
      "gene: 327 -> 343\n",
      "gene: 328 -> 248\n",
      "gene: 329 -> 237\n",
      "gene: 330 -> 284\n",
      "gene: 331 -> 341\n",
      "gene: 332 -> 235\n",
      "gene: 333 -> 243\n",
      "gene: 334 -> 284\n",
      "gene: 335 -> 318\n",
      "gene: 336 -> 285\n",
      "gene: 337 -> 281\n",
      "gene: 338 -> 259\n",
      "gene: 339 -> 288\n",
      "gene: 340 -> 249\n",
      "gene: 342 -> 258\n",
      "gene: 343 -> 219\n",
      "gene: 344 -> 261\n",
      "gene: 345 -> 150\n",
      "gene: 346 -> 182\n",
      "gene: 347 -> 53\n",
      "gene: 348 -> 139\n",
      "gene: 349 -> 188\n",
      "gene: 350 -> 239\n",
      "gene: 351 -> 69\n",
      "gene: 352 -> 318\n",
      "gene: 353 -> 268\n",
      "gene: 354 -> 175\n",
      "gene: 355 -> 209\n",
      "gene: 356 -> 317\n",
      "gene: 357 -> 138\n",
      "gene: 358 -> 158\n",
      "gene: 359 -> 227\n",
      "gene: 360 -> 221\n",
      "gene: 361 -> 235\n",
      "gene: 362 -> 260\n",
      "gene: 363 -> 116\n",
      "gene: 364 -> 163\n",
      "gene: 365 -> 203\n",
      "gene: 366 -> 195\n",
      "gene: 367 -> 133\n",
      "gene: 368 -> 129\n",
      "gene: 369 -> 226\n",
      "gene: 370 -> 175\n",
      "gene: 371 -> 123\n",
      "gene: 372 -> 210\n",
      "gene: 373 -> 235\n",
      "gene: 374 -> 305\n",
      "gene: 375 -> 285\n",
      "gene: 376 -> 315\n",
      "gene: 377 -> 326\n",
      "gene: 378 -> 270\n",
      "gene: 379 -> 305\n",
      "gene: 380 -> 272\n",
      "gene: 381 -> 327\n",
      "gene: 382 -> 242\n",
      "gene: 383 -> 233\n",
      "gene: 384 -> 212\n",
      "gene: 385 -> 230\n",
      "gene: 386 -> 329\n",
      "gene: 387 -> 269\n",
      "gene: 388 -> 323\n",
      "gene: 389 -> 266\n",
      "gene: 390 -> 103\n",
      "gene: 391 -> 153\n",
      "gene: 392 -> 114\n",
      "gene: 393 -> 93\n",
      "gene: 395 -> 260\n",
      "gene: 396 -> 127\n",
      "gene: 397 -> 268\n",
      "gene: 398 -> 274\n",
      "gene: 399 -> 252\n",
      "gene: 400 -> 245\n",
      "gene: 403 -> 172\n",
      "gene: 407 -> 211\n",
      "gene: 408 -> 76\n",
      "gene: 411 -> 126\n",
      "gene: 412 -> 261\n",
      "gene: 413 -> 98\n",
      "gene: 414 -> 96\n",
      "gene: 415 -> 250\n",
      "gene: 418 -> 133\n",
      "gene: 419 -> 309\n",
      "gene: 420 -> 169\n",
      "gene: 421 -> 232\n",
      "gene: 422 -> 221\n",
      "gene: 423 -> 235\n",
      "gene: 424 -> 294\n",
      "gene: 425 -> 75\n",
      "gene: 426 -> 318\n",
      "gene: 427 -> 293\n",
      "gene: 428 -> 268\n",
      "gene: 429 -> 246\n",
      "gene: 430 -> 298\n",
      "gene: 431 -> 80\n",
      "gene: 432 -> 278\n",
      "gene: 433 -> 219\n",
      "gene: 434 -> 243\n",
      "gene: 435 -> 139\n",
      "gene: 436 -> 167\n",
      "gene: 437 -> 212\n",
      "gene: 438 -> 70\n",
      "gene: 439 -> 315\n",
      "gene: 440 -> 327\n",
      "gene: 441 -> 163\n",
      "gene: 442 -> 174\n",
      "gene: 443 -> 98\n",
      "gene: 444 -> 159\n",
      "gene: 445 -> 242\n",
      "gene: 446 -> 304\n",
      "gene: 447 -> 268\n",
      "gene: 448 -> 288\n",
      "gene: 449 -> 272\n",
      "gene: 450 -> 231\n",
      "gene: 451 -> 240\n",
      "gene: 452 -> 230\n",
      "gene: 453 -> 254\n",
      "gene: 454 -> 266\n",
      "gene: 455 -> 190\n",
      "gene: 456 -> 314\n",
      "gene: 457 -> 242\n",
      "gene: 458 -> 292\n",
      "gene: 461 -> 179\n",
      "gene: 462 -> 132\n",
      "gene: 463 -> 269\n",
      "gene: 464 -> 321\n",
      "gene: 465 -> 249\n",
      "gene: 466 -> 295\n",
      "gene: 467 -> 336\n",
      "gene: 469 -> 266\n",
      "gene: 470 -> 241\n",
      "gene: 471 -> 287\n",
      "gene: 472 -> 227\n",
      "gene: 473 -> 182\n",
      "gene: 474 -> 283\n",
      "gene: 475 -> 160\n",
      "gene: 476 -> 220\n",
      "gene: 477 -> 174\n",
      "gene: 478 -> 240\n",
      "gene: 479 -> 286\n",
      "gene: 480 -> 178\n",
      "gene: 481 -> 162\n",
      "gene: 482 -> 113\n",
      "gene: 483 -> 150\n",
      "gene: 484 -> 307\n",
      "gene: 485 -> 118\n",
      "gene: 486 -> 140\n",
      "gene: 487 -> 108\n",
      "gene: 488 -> 160\n",
      "gene: 489 -> 144\n",
      "gene: 490 -> 183\n",
      "gene: 491 -> 282\n",
      "gene: 492 -> 157\n",
      "gene: 493 -> 227\n",
      "gene: 494 -> 182\n",
      "gene: 495 -> 291\n",
      "gene: 496 -> 203\n",
      "gene: 497 -> 259\n",
      "gene: 498 -> 208\n",
      "gene: 499 -> 219\n",
      "gene: 500 -> 169\n",
      "gene: 501 -> 202\n",
      "gene: 502 -> 316\n",
      "gene: 503 -> 321\n",
      "gene: 504 -> 179\n",
      "gene: 505 -> 273\n",
      "gene: 506 -> 279\n",
      "gene: 507 -> 282\n",
      "gene: 508 -> 310\n",
      "gene: 509 -> 261\n",
      "gene: 510 -> 294\n",
      "gene: 511 -> 103\n",
      "gene: 513 -> 53\n",
      "gene: 514 -> 247\n",
      "gene: 515 -> 223\n",
      "gene: 516 -> 120\n",
      "gene: 517 -> 187\n",
      "gene: 518 -> 237\n",
      "gene: 519 -> 216\n",
      "gene: 520 -> 222\n",
      "gene: 521 -> 116\n",
      "gene: 522 -> 219\n",
      "gene: 523 -> 230\n",
      "gene: 524 -> 217\n",
      "gene: 525 -> 180\n",
      "gene: 526 -> 90\n",
      "gene: 527 -> 193\n",
      "gene: 528 -> 256\n",
      "gene: 529 -> 349\n",
      "gene: 530 -> 255\n",
      "gene: 531 -> 330\n",
      "gene: 532 -> 252\n",
      "gene: 533 -> 224\n",
      "gene: 534 -> 264\n",
      "gene: 535 -> 268\n",
      "gene: 536 -> 269\n",
      "gene: 537 -> 331\n",
      "gene: 538 -> 304\n",
      "gene: 539 -> 259\n",
      "gene: 540 -> 252\n",
      "gene: 541 -> 267\n",
      "gene: 542 -> 200\n",
      "gene: 543 -> 140\n",
      "gene: 544 -> 75\n",
      "gene: 545 -> 162\n",
      "gene: 546 -> 234\n",
      "gene: 548 -> 88\n",
      "gene: 549 -> 174\n",
      "gene: 550 -> 181\n",
      "gene: 551 -> 236\n",
      "gene: 552 -> 239\n",
      "gene: 553 -> 314\n",
      "gene: 554 -> 246\n",
      "gene: 555 -> 259\n",
      "gene: 556 -> 267\n",
      "gene: 557 -> 234\n",
      "gene: 558 -> 246\n",
      "gene: 559 -> 296\n",
      "gene: 560 -> 290\n",
      "gene: 561 -> 148\n",
      "gene: 563 -> 296\n",
      "gene: 564 -> 94\n",
      "gene: 565 -> 326\n",
      "gene: 566 -> 184\n",
      "gene: 567 -> 265\n",
      "gene: 568 -> 139\n",
      "gene: 569 -> 166\n",
      "gene: 570 -> 241\n",
      "gene: 571 -> 209\n",
      "gene: 572 -> 246\n",
      "gene: 573 -> 138\n",
      "gene: 574 -> 218\n",
      "gene: 575 -> 73\n",
      "gene: 576 -> 161\n",
      "gene: 577 -> 60\n",
      "gene: 578 -> 284\n",
      "gene: 579 -> 252\n",
      "gene: 580 -> 248\n",
      "gene: 581 -> 94\n",
      "gene: 582 -> 152\n",
      "gene: 583 -> 271\n",
      "gene: 584 -> 118\n",
      "gene: 585 -> 189\n",
      "gene: 586 -> 288\n",
      "gene: 587 -> 78\n",
      "gene: 588 -> 213\n",
      "gene: 589 -> 226\n",
      "gene: 590 -> 235\n",
      "gene: 591 -> 222\n",
      "gene: 592 -> 253\n",
      "gene: 593 -> 251\n",
      "gene: 594 -> 319\n",
      "gene: 595 -> 274\n",
      "gene: 596 -> 304\n",
      "gene: 597 -> 245\n",
      "gene: 598 -> 160\n",
      "gene: 599 -> 258\n",
      "gene: 600 -> 253\n",
      "gene: 601 -> 266\n",
      "gene: 602 -> 162\n",
      "gene: 603 -> 92\n",
      "gene: 604 -> 187\n",
      "gene: 605 -> 212\n",
      "gene: 606 -> 220\n",
      "gene: 607 -> 190\n",
      "gene: 608 -> 256\n",
      "gene: 609 -> 285\n",
      "gene: 610 -> 294\n",
      "gene: 611 -> 281\n",
      "gene: 612 -> 277\n",
      "gene: 613 -> 236\n",
      "gene: 616 -> 289\n",
      "gene: 617 -> 158\n",
      "gene: 618 -> 126\n",
      "gene: 619 -> 222\n",
      "gene: 620 -> 103\n",
      "gene: 621 -> 191\n",
      "gene: 622 -> 272\n",
      "gene: 623 -> 177\n",
      "gene: 624 -> 215\n",
      "gene: 625 -> 107\n",
      "gene: 626 -> 175\n",
      "gene: 627 -> 75\n",
      "gene: 628 -> 332\n",
      "gene: 629 -> 315\n",
      "gene: 630 -> 192\n",
      "gene: 631 -> 205\n",
      "gene: 632 -> 213\n",
      "gene: 633 -> 296\n",
      "gene: 634 -> 168\n",
      "gene: 635 -> 234\n",
      "gene: 636 -> 168\n",
      "gene: 637 -> 255\n",
      "gene: 638 -> 159\n",
      "gene: 639 -> 283\n",
      "gene: 640 -> 286\n",
      "gene: 641 -> 263\n",
      "gene: 642 -> 285\n",
      "gene: 643 -> 272\n",
      "gene: 644 -> 131\n",
      "gene: 645 -> 303\n",
      "gene: 646 -> 234\n",
      "gene: 647 -> 285\n",
      "gene: 648 -> 252\n",
      "gene: 649 -> 228\n",
      "gene: 650 -> 260\n",
      "gene: 651 -> 239\n",
      "gene: 652 -> 211\n",
      "gene: 653 -> 233\n",
      "gene: 654 -> 308\n",
      "gene: 655 -> 251\n",
      "gene: 656 -> 320\n",
      "gene: 657 -> 290\n",
      "gene: 658 -> 269\n",
      "gene: 659 -> 257\n",
      "gene: 660 -> 284\n",
      "gene: 661 -> 284\n",
      "gene: 662 -> 322\n",
      "gene: 663 -> 271\n",
      "gene: 665 -> 270\n",
      "gene: 666 -> 177\n",
      "gene: 667 -> 241\n",
      "gene: 668 -> 196\n",
      "gene: 669 -> 239\n",
      "gene: 670 -> 273\n",
      "gene: 672 -> 280\n",
      "gene: 674 -> 268\n",
      "gene: 675 -> 251\n",
      "gene: 676 -> 246\n",
      "gene: 677 -> 106\n",
      "gene: 678 -> 275\n",
      "gene: 679 -> 186\n",
      "gene: 680 -> 199\n",
      "gene: 681 -> 59\n",
      "gene: 682 -> 138\n",
      "gene: 683 -> 163\n",
      "gene: 684 -> 231\n",
      "gene: 685 -> 282\n",
      "gene: 686 -> 268\n",
      "gene: 687 -> 192\n",
      "gene: 688 -> 172\n",
      "gene: 689 -> 115\n",
      "gene: 690 -> 128\n",
      "gene: 691 -> 140\n",
      "gene: 692 -> 275\n",
      "gene: 693 -> 311\n",
      "gene: 694 -> 194\n",
      "gene: 695 -> 211\n",
      "gene: 696 -> 266\n",
      "gene: 697 -> 242\n",
      "gene: 698 -> 255\n",
      "gene: 699 -> 289\n",
      "gene: 700 -> 167\n",
      "gene: 701 -> 147\n",
      "gene: 702 -> 128\n",
      "gene: 703 -> 246\n",
      "gene: 706 -> 281\n",
      "gene: 707 -> 222\n",
      "gene: 708 -> 288\n",
      "gene: 709 -> 112\n",
      "gene: 710 -> 227\n",
      "gene: 711 -> 244\n",
      "gene: 713 -> 288\n",
      "gene: 714 -> 268\n",
      "gene: 715 -> 246\n",
      "gene: 716 -> 101\n",
      "gene: 717 -> 253\n",
      "gene: 718 -> 123\n",
      "gene: 719 -> 239\n",
      "gene: 720 -> 268\n",
      "gene: 721 -> 225\n",
      "gene: 722 -> 136\n",
      "gene: 723 -> 265\n",
      "gene: 724 -> 276\n",
      "gene: 726 -> 252\n",
      "gene: 727 -> 279\n",
      "gene: 728 -> 257\n",
      "gene: 729 -> 244\n",
      "gene: 730 -> 295\n",
      "gene: 731 -> 301\n",
      "gene: 732 -> 269\n",
      "gene: 733 -> 281\n",
      "gene: 734 -> 318\n",
      "gene: 735 -> 274\n",
      "gene: 736 -> 173\n",
      "gene: 737 -> 148\n",
      "gene: 739 -> 293\n",
      "gene: 740 -> 262\n",
      "gene: 741 -> 278\n",
      "gene: 742 -> 235\n",
      "gene: 743 -> 91\n",
      "gene: 744 -> 144\n",
      "gene: 745 -> 225\n",
      "gene: 746 -> 149\n",
      "gene: 747 -> 118\n",
      "gene: 748 -> 160\n",
      "gene: 749 -> 224\n",
      "gene: 750 -> 216\n",
      "gene: 751 -> 166\n",
      "gene: 752 -> 264\n",
      "gene: 753 -> 225\n",
      "gene: 754 -> 186\n",
      "gene: 755 -> 252\n",
      "gene: 756 -> 281\n",
      "gene: 757 -> 233\n",
      "gene: 758 -> 197\n",
      "gene: 759 -> 223\n",
      "gene: 765 -> 242\n",
      "gene: 766 -> 97\n",
      "gene: 767 -> 272\n",
      "gene: 768 -> 123\n",
      "gene: 769 -> 230\n",
      "gene: 770 -> 313\n",
      "gene: 772 -> 238\n",
      "gene: 773 -> 293\n",
      "gene: 774 -> 257\n",
      "gene: 775 -> 185\n",
      "gene: 776 -> 168\n",
      "gene: 779 -> 233\n",
      "gene: 780 -> 237\n",
      "gene: 781 -> 148\n",
      "gene: 783 -> 72\n",
      "gene: 784 -> 210\n",
      "gene: 785 -> 230\n",
      "gene: 786 -> 319\n",
      "gene: 787 -> 252\n",
      "gene: 788 -> 146\n",
      "gene: 789 -> 291\n",
      "gene: 790 -> 214\n",
      "gene: 791 -> 281\n",
      "gene: 792 -> 213\n",
      "gene: 793 -> 170\n",
      "gene: 794 -> 126\n",
      "gene: 795 -> 232\n",
      "gene: 796 -> 144\n",
      "gene: 797 -> 173\n",
      "gene: 798 -> 137\n",
      "gene: 799 -> 132\n",
      "gene: 800 -> 169\n",
      "gene: 801 -> 200\n",
      "gene: 802 -> 275\n",
      "gene: 803 -> 188\n",
      "gene: 804 -> 176\n",
      "gene: 805 -> 137\n",
      "gene: 806 -> 135\n",
      "gene: 807 -> 199\n",
      "gene: 808 -> 218\n",
      "gene: 809 -> 344\n",
      "gene: 810 -> 178\n",
      "gene: 812 -> 179\n",
      "gene: 813 -> 237\n",
      "gene: 814 -> 186\n",
      "gene: 815 -> 228\n",
      "gene: 816 -> 181\n",
      "gene: 817 -> 183\n",
      "gene: 818 -> 198\n",
      "gene: 819 -> 182\n",
      "gene: 820 -> 115\n",
      "gene: 821 -> 79\n",
      "gene: 822 -> 196\n",
      "gene: 823 -> 341\n",
      "gene: 824 -> 99\n",
      "gene: 825 -> 216\n",
      "gene: 827 -> 143\n",
      "gene: 828 -> 106\n",
      "gene: 829 -> 122\n",
      "gene: 830 -> 168\n",
      "gene: 831 -> 227\n",
      "gene: 832 -> 232\n",
      "gene: 833 -> 153\n",
      "gene: 834 -> 297\n",
      "gene: 835 -> 220\n",
      "gene: 836 -> 296\n",
      "gene: 837 -> 312\n",
      "gene: 838 -> 261\n",
      "gene: 839 -> 240\n",
      "gene: 840 -> 333\n",
      "gene: 841 -> 328\n",
      "gene: 842 -> 299\n",
      "gene: 843 -> 248\n",
      "gene: 844 -> 268\n",
      "gene: 845 -> 277\n",
      "gene: 846 -> 237\n",
      "gene: 847 -> 258\n",
      "gene: 848 -> 262\n",
      "gene: 849 -> 227\n",
      "gene: 850 -> 248\n",
      "gene: 851 -> 289\n",
      "gene: 852 -> 148\n",
      "gene: 854 -> 234\n",
      "gene: 855 -> 192\n",
      "gene: 856 -> 237\n",
      "gene: 857 -> 231\n",
      "gene: 858 -> 204\n",
      "gene: 859 -> 234\n",
      "gene: 860 -> 217\n",
      "gene: 861 -> 323\n",
      "gene: 862 -> 101\n",
      "gene: 863 -> 150\n",
      "gene: 864 -> 258\n",
      "gene: 865 -> 259\n",
      "gene: 866 -> 251\n",
      "gene: 867 -> 145\n",
      "gene: 868 -> 294\n",
      "gene: 869 -> 213\n",
      "gene: 870 -> 257\n",
      "gene: 871 -> 184\n",
      "gene: 872 -> 203\n",
      "gene: 873 -> 93\n",
      "gene: 874 -> 116\n",
      "gene: 875 -> 217\n",
      "gene: 876 -> 196\n",
      "gene: 877 -> 76\n",
      "gene: 878 -> 194\n",
      "gene: 879 -> 133\n",
      "gene: 880 -> 88\n",
      "gene: 881 -> 192\n",
      "gene: 882 -> 220\n",
      "gene: 883 -> 277\n",
      "gene: 884 -> 192\n",
      "gene: 885 -> 180\n",
      "gene: 886 -> 145\n",
      "gene: 887 -> 210\n",
      "gene: 888 -> 220\n",
      "gene: 889 -> 194\n",
      "gene: 890 -> 295\n",
      "gene: 891 -> 173\n",
      "gene: 894 -> 228\n",
      "gene: 896 -> 316\n",
      "gene: 897 -> 259\n",
      "gene: 898 -> 275\n",
      "gene: 899 -> 179\n",
      "gene: 900 -> 67\n",
      "gene: 901 -> 95\n",
      "gene: 902 -> 288\n",
      "gene: 903 -> 195\n",
      "gene: 904 -> 253\n",
      "gene: 905 -> 237\n",
      "gene: 906 -> 254\n",
      "gene: 907 -> 200\n",
      "gene: 908 -> 87\n",
      "gene: 909 -> 180\n",
      "gene: 910 -> 254\n",
      "gene: 911 -> 331\n",
      "gene: 912 -> 298\n",
      "gene: 913 -> 216\n",
      "gene: 914 -> 72\n",
      "gene: 915 -> 256\n",
      "gene: 916 -> 182\n",
      "gene: 917 -> 159\n",
      "gene: 918 -> 163\n",
      "gene: 919 -> 273\n",
      "gene: 920 -> 287\n",
      "gene: 921 -> 325\n",
      "gene: 922 -> 287\n",
      "gene: 923 -> 160\n",
      "gene: 924 -> 197\n",
      "gene: 925 -> 79\n",
      "gene: 926 -> 255\n",
      "gene: 927 -> 305\n",
      "gene: 928 -> 266\n",
      "gene: 929 -> 125\n",
      "gene: 930 -> 81\n",
      "gene: 932 -> 66\n",
      "gene: 933 -> 186\n",
      "gene: 934 -> 198\n",
      "gene: 935 -> 230\n",
      "gene: 936 -> 293\n",
      "gene: 937 -> 200\n",
      "gene: 938 -> 164\n",
      "gene: 939 -> 292\n",
      "gene: 940 -> 258\n",
      "gene: 941 -> 184\n",
      "gene: 942 -> 136\n",
      "gene: 943 -> 173\n",
      "gene: 944 -> 257\n",
      "gene: 945 -> 216\n",
      "gene: 946 -> 262\n",
      "gene: 947 -> 244\n",
      "gene: 948 -> 187\n",
      "gene: 949 -> 247\n",
      "gene: 950 -> 154\n",
      "gene: 951 -> 212\n",
      "gene: 952 -> 260\n",
      "gene: 953 -> 244\n",
      "gene: 954 -> 190\n",
      "gene: 955 -> 270\n",
      "gene: 956 -> 196\n",
      "gene: 957 -> 238\n",
      "gene: 958 -> 254\n",
      "gene: 959 -> 209\n",
      "gene: 960 -> 288\n",
      "gene: 961 -> 287\n",
      "gene: 962 -> 278\n",
      "gene: 963 -> 282\n",
      "gene: 964 -> 269\n",
      "gene: 965 -> 125\n",
      "gene: 966 -> 213\n",
      "gene: 969 -> 182\n",
      "gene: 970 -> 306\n",
      "gene: 971 -> 262\n",
      "gene: 972 -> 235\n",
      "gene: 973 -> 121\n",
      "gene: 974 -> 283\n",
      "gene: 975 -> 276\n",
      "gene: 976 -> 188\n",
      "gene: 977 -> 211\n",
      "gene: 978 -> 190\n",
      "gene: 979 -> 227\n",
      "gene: 980 -> 240\n",
      "gene: 982 -> 251\n",
      "gene: 983 -> 232\n",
      "gene: 984 -> 234\n",
      "gene: 985 -> 241\n",
      "gene: 986 -> 269\n",
      "gene: 987 -> 237\n",
      "gene: 988 -> 139\n",
      "gene: 989 -> 284\n",
      "gene: 990 -> 268\n",
      "gene: 991 -> 265\n",
      "gene: 992 -> 100\n",
      "gene: 995 -> 224\n",
      "gene: 997 -> 305\n",
      "gene: 998 -> 149\n",
      "gene: 999 -> 133\n",
      "gene: 1000 -> 55\n",
      "gene: 1001 -> 69\n",
      "gene: 1002 -> 117\n",
      "gene: 1007 -> 209\n",
      "gene: 1008 -> 136\n",
      "gene: 1009 -> 152\n",
      "gene: 1010 -> 110\n",
      "gene: 1011 -> 232\n",
      "gene: 1013 -> 146\n",
      "gene: 1014 -> 211\n",
      "gene: 1015 -> 65\n",
      "gene: 1016 -> 176\n",
      "gene: 1017 -> 117\n",
      "gene: 1018 -> 145\n",
      "gene: 1019 -> 204\n",
      "gene: 1020 -> 133\n",
      "gene: 1021 -> 173\n",
      "gene: 1022 -> 196\n",
      "gene: 1023 -> 112\n",
      "gene: 1024 -> 87\n",
      "gene: 1025 -> 65\n",
      "gene: 1026 -> 130\n",
      "gene: 1027 -> 78\n",
      "gene: 1028 -> 102\n",
      "gene: 1029 -> 127\n",
      "gene: 1030 -> 301\n",
      "gene: 1031 -> 172\n",
      "gene: 1032 -> 163\n",
      "gene: 1033 -> 244\n",
      "gene: 1034 -> 224\n",
      "gene: 1035 -> 126\n",
      "gene: 1036 -> 192\n",
      "gene: 1038 -> 146\n",
      "gene: 1039 -> 111\n",
      "gene: 1040 -> 62\n",
      "gene: 1041 -> 106\n",
      "gene: 1044 -> 183\n",
      "gene: 1045 -> 172\n",
      "gene: 1046 -> 221\n",
      "gene: 1047 -> 268\n",
      "gene: 1048 -> 219\n",
      "gene: 1049 -> 181\n",
      "gene: 1050 -> 190\n",
      "gene: 1051 -> 147\n",
      "gene: 1053 -> 96\n",
      "gene: 1054 -> 171\n",
      "gene: 1055 -> 150\n",
      "gene: 1056 -> 213\n",
      "gene: 1057 -> 193\n",
      "gene: 1058 -> 257\n",
      "gene: 1059 -> 221\n",
      "gene: 1060 -> 122\n",
      "gene: 1062 -> 168\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "reload(check)\n",
    "reg_size_dic, len_check = check._check_region_size(full_records, \n",
    "                                                   min_size=min_num_probes)\n",
    "print(len_check)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "30326bc1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAsAAAAIQCAYAAACPEdjAAAAAAXNSR0IArs4c6QAAIABJREFUeF7t3X+QldV9x/EvsC1aMw0OCrMNthjQIRJQLEFLMalxUMGU0kRRmTHIxImBUpgBZAS1AeIP0In5MYIYbZIpFDOmgr+amhFbpxiHWRggccIQK2STrWHAMSHM+Dt1O+dJ95ZlL3ufe85zzvece947kz/inud8z/M63/vczz5792FAd3d3t/CFAAIIIIAAAggggEAmAgMIwJnsNKeJAAIIIIAAAgggUAgQgGkEBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAGYHkAAAQQQQAABBBDISoAAnNV2c7IIIIAAAggggAACBGB6AAEEEEAAAQQQQCArAQJwVtvNySKAAAIIIIAAAggQgOkBBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAGYHkAAAQQQQAABBBDISoAAnNV2c7IIIIAAAggggAACBGB6AAEEEEAAAQQQQCArAQJwVtvNySKAAAIIIIAAAggQgOkBBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAGYHkAAAQQQQAABBBDISoAAnNV2c7IIIIAAAggggAACBGB6AAEEEEAAAQQQQCArAQJwVtvNySKAAAIIIIAAAggQgOkBBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAGYHkAAAQQQQAABBBDISoAAnNV2c7IIIIAAAggggAACBGB6AAEEEEAAAQQQQCArAQJwVtvNySKAAAIIIIAAAggQgOkBBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAGYHkAAAQQQQAABBBDISoAAnNV2c7IIIIAAAggggAACBGB6AAEEEEAAAQQQQCArAQJwVtvNySKAAAIIIIAAAggQgOkBBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAGYHkAAAQQQQAABBBDISoAAnNV2c7IIIIAAAggggAACBGB6AAEEEEAAAQQQQCArAQJwVtvNySKAAAIIIIAAAggQgOkBBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAGYHkAAAQQQQAABBBDISoAAnNV2c7IIIIAAAggggAACBGB6AAEEEEAAAQQQQCArAQJwVtvNySKAAAIIIIAAAggQgOkBBBBAAAEEEEAAgawECMBZbTcniwACCCCAAAIIIEAApgcQQAABBBBAAAEEshIgAGe13ZwsAggggAACCCCAAAHYYw+888478vLLL8uZZ54pbW1tHisxNQIIIIAAAggg0Fvgd7/7nbz++usybtw4OeWUU+A5ToAA7LEddu7cKZMmTfJYgakRQAABBBBAAIH+BTo6OuQTn/gETATgMD3Q2dkpZ599tpjGa29vD1OUKggggAACCCCAgIgcOnSouBH385//XEaOHIkJAThMD/z3f/+3nHXWWdLV1SUjRowIU5QqCCCAAAIIIICAiJBDTt4GfATC40uExvOIy9QIIIAAAggg0K8AOYQArPISofFU2CmKAAIIIIAAAtwB7rcHuAPs8SVCAPaIy9QIIIAAAgggwB1gyx4gAFvClTmMAFxGiTEIIIAAAggg4EOAHHJyVQKwj477vzlpPI+4TI0AAggggAAC3AG27AECsCVcmcMIwGWUGIMAAggggAACPgTIIdwB9tFXDeek8RoSMQABBBBAAAEEPAmQQwjAnlqr/2lpPBV2iiKAAAIIIIAAT4Hotwf4CITHlwgB2CMuUyOAAAIIIIBAvwLkEO4Aq7xEaDwVdooigAACCCCAAHeAuQOs9SogAGvJUxcBBBBAAAEEyCHcAVZ5FdB4KuwURQABBBBAAAHuAHMHWOtVQADWkqcuAggggAACCJBDuAOs8iqg8VTYKYoAAggggAAC3AHmDrDWq4AArCVPXQQQQAABBBAgh3AHWOVVQOOpsFMUAQQQQAABBLgDzB1grVcBAVhLnroIIIAAAgggQA7hDrDKq4DGU2GnKAIIIBC1wMhb/9VqfZ1rrrI6joPyFSCHEIBVup/GU2GnKAIIIBC1AAE46u1pqcWRQwjAKg1N46mwUxQBBBCIWoAAHPX2tNTiyCEEYJWGpvFU2CmKAAIIRC1AAI56e1pqceQQArBKQ9N4KuwURQABBKIWIABHvT0ttThyCAFYpaFpPBV2iiKAAAJRCxCAo96ellocOYQArNLQNJ4KO0URQACBqAUIwFFvT0stjhxCAFZpaBpPhZ2iCCCAQNQCBOCot6elFkcOIQCrNDSNp8JOUQQQQCBqAQJw1NvTUosjhxCAVRqaxlNhpygCCCAQtQABOOrtaanFkUMIwCoNTeOpsFMUAQQQiFqAABz19rTU4sghBGCVhqbxVNgpigACCEQtQACOentaanHkEAKwSkPTeCrsFEUAAQSiFiAAR709LbU4cggBWKWhaTwVdooigAACUQsQgKPenpZaHDmEAKzS0DSeCjtFEUAAgagFCMBRb09LLY4cQgBWaWgaT4WdoggggEDUAgTgqLenpRZHDiEAqzQ0jafCTlEEEEAgagECcNTb01KLI4cQgFUamsZTYacoAgggELUAATjq7WmpxZFDCMAqDU3jqbBTFAEEEIhagAAc9fa01OLIIQRglYam8VTYKYoAAghELUAAjnp7Wmpx5BACsEpD03gq7BRFAAEEohYgAEe9PS21OHIIAViloWk8FXaKIoAAAlELEICj3p6WWhw5hACs0tA0ngo7RRFAAIGoBQjAUW9PSy2OHEIAVmloGk+FnaIIIIBA1AIE4Ki3p6UWRw4hAKs0NI2nwk5RBBBAIGoBAnDU29NSiyOHEIBVGprGU2GnKAIIIBC1AAE46u1pqcWRQwjAKg1N46mwUxQBBBCIWoAAHPX2tNTiyCEEYJWGpvFU2CmKAAIIRC1AAI56e1pqceQQArBKQ9N4KuwURQABBKIWIABHvT0ttThyCAG4rsA//dM/yZo1a+TgwYMyatQoWblypVxzzTXF2L1798qiRYtk165dMnToUFm4cKEsXbq0qRcGjdcUF4MRQACBoAK2QdQssnPNVdZrta3rUtN6sRyYtAA5hADcR2DTpk1y4403yle/+lW56qqr5NFHHy0C8IsvvijnnnuujBkzRmbOnCmLFy+WHTt2yPz582X9+vUyd+7c0i8GGq80FQMRQACB4AK2QZQAHHyrKGgpQA4hAPcS6O7ulo9+9KNy9dVXy3333Vf73hVXXCF/9Vd/Vfz/devWSWdnp7S1tRX/f8WKFbJlyxbZv39/6Tak8UpTMRABBBAILkAADk5OwcAC5BACcC8BE2I/9rGPye7du2XChAl9dKZPny5DhgyRzZs31763bds2mTp1qhw+fFiGDRtWqoVpvFJMDEIAAQRUBAjAKuwUDShADiEA9xJ46qmn5G/+5m9k+/bt8pWvfEX27NkjZ599ttx+++3y13/91zJ+/HiZNm2arF27tnbcvn37ZOzYsbJz506ZOHFiqfal8UoxMQgBBBBQESAAq7BTNKAAOYQA3EvAfP73hhtuKELvl7/8ZTn//PPl8ccfl7vuukuee+45ufnmm2X27NmyevXq2nE9fyhnQvOUKVPqih47dkzM/3q+Dh06JJMmTZKuri4ZMWJEwJanFAIIIIBAIwECcCMhvp+6AAGYANxL4LHHHpNrr722+KO2efPm1b5n/hjOfP3yl78U8zGIeneAT/axCXOc+SO6VatW9dEmAKd+CWH9CCDQigIE4FbcVc7peAECMAG4l8BLL70kf/mXf1l8BOL4u7nLli2TZ555RkaOHFk8+mzjxo2148p8Bpg7wFx4EEAAgXQECMDp7BUrtRMgABOAewm8/fbbMnz4cPmHf/iHXs/2NZ/7/YM/+AOZPHmybNiwQQ4cOCCDBg0qjl2+fLls3bqVp0DYvQY5CgEEEIhOgAAc3ZawoIoFCMAE4D4Cd955p9x7773y0EMPFZ/T/d73vlcE4ueff17OO++84jnAM2bMEHNXuKOjo/iohAnFc+bMKd2eNF5pKgYigAACwQUIwMHJKRhYgBxCAK4rcP/998sDDzwgr732WvFYNPP5XfN0CPNlnvZg/vU384SI9vZ2WbJkiSxYsKCp1qXxmuJiMAIIIBBUgAAclJtiCgLkEAKwQtuJ0Hgq7BRFAAEESgkQgEsxMShhAXIIAVilfWk8FXaKIoAAAqUECMClmBiUsAA5hACs0r40ngo7RRFAAIFSAgTgUkwMSliAHEIAVmlfGk+FnaIIIIBAKQECcCkmBiUsQA4hAKu0L42nwk5RBBBAoJQAAbgUE4MSFiCHEIBV2pfGU2GnKAIIIFBKgABciolBCQuQQwjAKu1L46mwUxQBBBAoJUAALsXEoIQFyCEEYJX2pfFU2CmKAAIIlBJwCcClClQ8qHPNVRXPyHStLkAOIQCr9DiNp8JOUQQQQKCUAAG4FBODEhYghxCAVdqXxlNhpygCCCBQSoAAXIqJQQkLkEMIwCrtS+OpsFMUAQQQKCVAAC7FxKCEBcghBGCV9qXxVNgpigACCJQSIACXYmJQwgLkEAKwSvvSeCrsFEUAAQRKCRCASzExKGEBcggBWKV9aTwVdooigAACpQRSC8ClTuokg3iChIteuseSQwjAKt1L46mwUxQBBBAoJUAALsXEoIQFyCEEYJX2pfFU2CmKAAIIlBIgAJdiYlDCAuQQArBK+9J4KuwURQABBEoJEIBLMTEoYQFyCAFYpX1pPBV2iiKAAAKlBAjApZgYlLAAOYQArNK+NJ4KO0URQACBUgIE4FJMDEpYgBxCAFZpXxpPhZ2iCCCAQCkBAnApJgYlLEAOIQCrtC+Np8JOUQQQQKCUAAG4FBODEhYghxCAVdqXxlNhpygCCCBQSoAAXIqJQQkLkEMIwCrtS+OpsFMUAQQQKCVAAC7FxKCEBcghBGCV9qXxVNgpigACCJQSIACXYmJQwgLkEAKwSvvSeCrsFEUAAQRKCRCASzExKGEBcggBWKV9aTwVdooigAACpQQIwKWYGJSwADmEAKzSvjSeCjtFEUAAgVICBOBSTAxKWIAcQgBWaV8aT4WdoggggEApAQJwKSYGJSxADiEAq7QvjafCTlEEEECglAABuBQTgxIWIIcQgFXal8ZTYacoAgggUEqAAFyKiUEJC5BDCMAq7UvjqbBTFAEEECglQAAuxcSghAXIIQRglfal8VTYKYoAAgiUEiAAl2JiUMIC5BACsEr70ngq7BRFAAEESgkQgEsxMShhAXIIAVilfWk8FXaKIoAAAqUECMClmBiUsAA5hACs0r40ngo7RRFAAIFSAgTgUkwMSliAHEIAVmlfGk+FnaIIIIBAKQECcCkmBiUsQA4hAKu0L42nwk5RBBBAoJQAAbgxk4tR55qrGhdghFcBcggB2GuDnWxyGk+FnaIIIIBAKQGXcFeqQESDbMOoi5FtzYjYkl8KOYQArNLENJ4KO0URQACBUgIu4a5UgYgG2YZRFyPbmhGxJb8UcggBWKWJaTwVdooigAACpQRcwl2pAhENsg2jLka2NSNiS34p5BACsEoT03gq7BRFAAEESgm4hLtSBSIaZBtGXYxsa0bElvxSyCEEYJUmpvFU2CmKAAIIlBJwCXelCkQ0yDaMuhjZ1oyILfmlkEMIwCpNTOOpsFMUAQQyE3AJablQ2YZRF1vbmrnsSYjzJIcQgEP0WZ8aNJ4KO0URQCAzAZeQlguVbRh1sbWtmcuehDhPcggBOESfEYBVlCmKAAK5C7iEtFzsbMOoi61tzVz2JMR5EoAJwCH6jACsokxRBBDIXcAlpOViZxtGXWxta+ayJyHOkwBMAA7RZwRgFWWKIoBA7gIuIS0XO9sw6mJrWzOXPQlxngRgAnCIPiMAqyhTFAEEchdwCWm52NmGURdb25q57EmI8yQAE4D7CPziF7+QkSNH9vnvDz/8sNx0002yd+9eWbRokezatUuGDh0qCxculKVLlzbVrzReU1wMRgABBKwEXEKaVcEED7INoy62tjUT5I12yeQQAnAfgaefflpmzZolBw8elAEDBtS+/+EPf1jeeustGTNmjMycOVMWL14sO3bskPnz58v69etl7ty5pRudxitNxUAEEEDAWsAlpFkXTexA2zDqYmtbMzHaqJdLDiEA9xG4++675bHHHivu9J74dc8998i6deuks7NT2traim+vWLFCtmzZIvv37y/d7DReaSoGIoAAAtYCLiHNumhiB9qGURdb25qJ0Ua9XHIIAbiPwHXXXScDBw6UzZs39/ne9OnTZciQIb2+t23bNpk6daocPnxYhg0bVqrhabxSTAxCAAEEnARcQppT4YQOtg2jLra2NRNijX6p5BACcB+BsWPHSnt7u7z77rvyyiuvyDnnnCN33HGHXHHFFTJ+/HiZNm2arF27tnbcvn37xByzc+dOmThxYqmmp/FKMTEIAQQQcBJwCWlOhRM62DaMutja1kyINfqlkkMIwL0E3nvvPTnttNNkwoQJct9998mHPvQh2bRpk3zjG9+Q5557Tm6++WaZPXu2rF69unac+azwqFGjZPv27TJlypS6oseOHRPzv56vQ4cOyaRJk6Srq0tGjBgR/QuFBSKAAAIpCriEtBTP12bNtmHUxda2ps35cUx9AQIwAbiPwJtvvll8vnfw4MG171155ZXFH8SZhjEfg6h3B3j37t1FcK73tXLlSlm1alWfbxGAuTQhgAAC/gRcQpq/VcU1s20YdbG1rRmXXNqrIQATgEt18C233CLPPvusnHXWWcWjzzZu3Fg7rsxngLkDXIqZQQgggEClAi4hrdKFRDyZbRh1sbWtGTFjcksjABOAewn85Cc/kcmTJxdh9/iPM1x66aUyfPhwueCCC2TDhg1y4MABGTRoUHHs8uXLZevWrTwFIrmXPwtGAIFWF3AJaa1u03N+tmHUxda2Zi57EuI8CcAE4F4CH3zwQRGAzccgHnzwQTnjjDPkW9/6VvGcX/NHbiYEm+cAz5gxQ5YtWyYdHR0yb968IhTPmTOndM/SeKWpGIgAAghYC7iENOuiiR1oG0ZdbG1rJkYb9XLJIQTgPgKvv/663HrrrfKDH/xAjh49KhdeeKGsWbNGLrnkkmKsCcLmX3/bs2dP8bSIJUuWyIIFC5pqdBqvKS4GI4AAAlYCLiHNqmCCB9mGURdb25oJ8ka7ZHIIAVilOWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoBAggIuQSvB0w2+ZNsw6rIvtjWD47RwQXIIAVilvWk8FXaKIoCAiKQWXFzWy4Y3FrANoy77Yluz8dkwoqwAOYQAXLZXKh1H41XKyWQIINCEQGrBxWW9TbBkO9Q2jLrsi23NbDfJw4mTQwjAHtqq8ZQ0XmMjRiCAgB+B1IKLy3r9CLbWrLZh1GVfbGu2lrzu2ZBDCMAqHUjjqbBTFAEE+AgEPXCCgG0YJQCn3UrkEAKwSgfTeCrsFEUAAQIwPUAApgdEhBxCAFZ5IdB4KuwURQABAjA9QACmBwjA/fbAgO7u7m66xI8AAdiPK7MigEBjgdR+de2y3sYajOAjEHn2ADmEO8AqnU/jqbBTFAEEuANMD3AHmB7gDjB3gLVeBQRgLXnqIoCAyx1V27uFLuou63Wpm8uxtnvqsi+2NXPZkxDnSQ7hDnCIPutTg8ZTYacoAghwB5ge4A4wPcAdYO4Aa70KCMBa8tRFAIHU7ty5rJfdbixgezfWZV9sazY+G0aUFSCHcAe4bK9UOo7Gq5STyRBAoAmB1IKLy3qbYMl2qG0YddkX25rZbpKHEyeHEIA9tFXjKWm8xkaMQAABPwKpBReX9foRbK1ZbcOoy77Y1mwted2zIYcQgFU6kMZTYacoAgjwGWB64AQB2zBKAE67lcghBGCVDqbxVNgpigACBGB6gABMD/BHcP32AP8QhseXCAHYIy5TI4BAvwKp3blzWS+t0FiAO8CNjVpxBDmEO8AqfU3jqbBTFAEEuANMD3AHmB7gDjB3gLVeBQRgLXnqIoCAyx1V27uFLuou63Wpm8uxtnvqsi+2NXPZkxDnSQ7hDnCIPutTg8ZTYacoAgg43gF2AbQNPS5By2W9uRyrsS+2NXPZkxDnSQ4hAIfoMwKwijJFEUCgnoBWoLQNPVrrzaV7NPbFtmYuexLiPAnABOAQfUYAVlGmKAIIEIDpgUYCtmHU5QcT25qNzoXvlxcgABOAy3dLhSNpvAoxmQoBBJoScAkuTRU6YbBt6NFar8u5pnSsxr7Y1kzJNfa1kkMIwCo9SuOpsFMUAQT4DDA9EMEPJgRg/TYkhxCAVbqQxlNhpygCCBCA6QECMD3AY9D67QH+IQyPLxECsEdcpkYAgX4FtD5SYHvXT2u9tJE/Adte8Lei/GYmh3AHWKXraTwVdooigAB3gOmBCAQIwPqbQA4hAKt0IY2nwk5RBBAgANMDEQgQgPU3gRxCAFbpQhpPhZ2iCCBAAKYHIhAgAOtvAjmEAKzShTSeCjtFEUCAAEwPRCBAANbfBHIIAVilC2k8FXaKIoAAAZgeiECAAKy/CeQQArBKF9J4KuwURQABAjA9EIEAAVh/E8ghBGCVLqTxVNgpigACBGB6IAIBArD+JpBDCMAqXUjjqbBTFAEECMD0QAQCBGD9TSCHEIBVupDGU2GnKAIIEIDpgQgECMD6m0AOIQCrdCGNp8JOUQQQIADTAxEIEID1N4EcQgBW6UIaT4WdogggQACmByIQIADrbwI5hADcbxe+8sorcuGFF8oDDzwgN954YzF27969smjRItm1a5cMHTpUFi5cKEuXLm2qm2m8prgYjAACFQqMvPVfK5yt/FS2oUdrveXPjJHNCtj2QrN1GH9yAXIIAfikAu+//75Mnjy5CLrf+c53igD8xhtvyJgxY2TmzJmyePFi2bFjh8yfP1/Wr18vc+fOLf1ao/FKUzEQAQQqFtAKlLahR2u9FbMz3XECtr0AYnUC5BAC8EkFVqxYUQTc//iP/6gF4HvuuUfWrVsnnZ2d0tbWVhxrxm3ZskX2799fujNpvNJUDEQAgYoFtAKlbejRWm/F7ExHAI6qB8ghBOC6Av/5n/8pf/u3f1t83OFP//RPawF4+vTpMmTIENm8eXPtuG3btsnUqVPl8OHDMmzYsFINTuOVYmIQAgh4ENAKlARgD5uZ6JS2vZDo6Ua5bHIIAbiPwNGjR+X8888Xc7d39uzZMmDAgFoAHj9+vEybNk3Wrl1bO27fvn0yduxY2blzp0ycOLGu6LFjx8T8r+fr0KFDMmnSJOnq6pIRI0ZE+eJgUQgg0JoCBODW3NeUzooArL9bBGACcB+B66+/Xrq7u+V73/te8b3jA/Do0aOLULx69eracQcPHpRRo0bJ9u3bZcqUKXVFV65cKatWrerzPQKw/kWAFSCQmwABOLcdj+98CcD6e0IAJgD3Eti4caMsX75cXn75ZTn99NP7BOBx48aJ+RhEvTvAu3fvlgkTJnAHWP91zQoQQKAfAQIw7aEtQADW3gERAjABuJfApZdeKi+++KIMHjy49t/ffPPN4v+bu7x/9md/Vjz6zATlni8+A6z/QmYFCCBQXoAAXN6KkX4ECMB+XJuZlQBMAO4l8Nprr8nbb7/d67+dc845xR3fa6+9Vh599FHZsGGDHDhwQAYNGlSMM3eMt27dylMgmnnlMRYBBNQECMBq9BT+PwECsH4rEIAJwA278PjPAB85cqR4DvCMGTNk2bJl0tHRIfPmzStC8Zw5cxrO1TOAxitNxUAEEKhYgABcMSjTNS1AAG6arPIDyCEE4IZNdXwANoPN0x7Mv/62Z88eaW9vlyVLlsiCBQsaznP8ABqvKS4GI4BAhQIE4AoxmcpKgABsxVbpQeQQAnClDVV2MhqvrBTjEECgagECcNWizBdKgOBcnTQ5hABcXTc1MRON1wQWQxFAoFIBAnClnEwWUIAAXB02OYQAXF03NTETjdcEFkMRQKBSAQJwpZxMFlCAAFwdNjmEAFxdNzUxE43XBBZDEUCgUgECcKWcTBZQgABcHTY5hABcXTc1MRON1wQWQxFAoFIBAnClnEwWUIAAXB02OYQAXF03NTETjdcEFkMRQKBSAQJwpZxMFlCAAFwdNjmEAFxdNzUxE43XBBZDEWhhAZcwahsGXGq6bEVq63Wj/qUxAAAgAElEQVQ5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBlzBqGwZcaroAprZel3PlWD8Ctj3kZzVpz0oOIQCrdDCNp8JOUQSiE3AJo7ZhwKWmC2Bq63U5V471I2DbQ35Wk/as5BACsEoH03gq7BRFIDoBrTCqAWEbXnIy0tiXlGra9lBK5xhqreQQAnCoXutVh8ZTYacoAtEJ5BTubMNLTkbRNWhkC7LtochOI4rlkEMIwCqNSOOpsFMUgegEcgp3tuElJ6PoGjSyBdn2UGSnEcVyyCEEYJVGpPFU2CmKQHQCOYU72/CSk1F0DRrZgmx7KLLTiGI55BACsEoj0ngq7BRFIDqBnMKdbXjJySi6Bo1sQbY9FNlpRLEccggBWKURaTwVdooiEJ1ATuHONrzkZBRdg0a2INseiuw0olgOOYQArNKINJ4KO0UzENAKS7ZvzFrrzaAVOMUWFLB9nbUghfMpkUMIwM5NZDMBjWejxjEINBbQCpS2b8xa620syQgE4hOwfZ3Fdyb6KyKHEIBVupDGU2GnaAYCWoHS9o1Za70ZtAKn2IICtq+zFqRwPiVyCAHYuYlsJqDxbNQ4BoHGAlqB0vaNWWu9jSUZgUB8Aravs/jORH9F5BACsEoX0ngq7BTNQEArUNq+MWutN4NW4BRbUMD2ddaCFM6nRA4hADs3kc0ENJ6NGscg0FhAK1DavjFrrbexJCMQiE/A9nUW35nor4gcQgBW6UIaT4WdohkIaAVK2zdmrfVm0AqcYgsK2L7OWpDC+ZTIIQRg5yaymYDGs1HjGAQaC2gFSts3Zq31NpZkBALxCdi+zuI7E/0VkUMIwCpdSOOpsFM0AwGtQGn7xqy13gxagVNsQQHb11kLUjifEjmEAOzcRDYT0Hg2ahyDQGMBrUBp+8astd7GkoxAID4B29dZfGeivyJyCAFYpQtpPBV2imYgoBUobd+YtdabQStwii0oYPs6a0EK51MihxCAnZvIZgIaz0aNYxBoLKAVKG3fmLXW21iSEQjEJ2D7OovvTPRXRA4hAPcROHLkiCxZskSeffZZefvtt+VTn/qU3HfffXLeeecVY/fu3SuLFi2SXbt2ydChQ2XhwoWydOnSprqZxmuKi8GJCriEO9s3OpeaLsyprdflXDkWAS0B29eZ1npjrksOIQD3Ebj44otl4MCB8s1vflNOO+00ueOOO+Sll16SV199tQjEY8aMkZkzZ8rixYtlx44dMn/+fFm/fr3MnTu3dK/TeKWpGJiwgEsYtX2jc6npQp3ael3OlWMR0BKwfZ1prTfmuuQQAnAvgTfeeEMWLFggt99+u4wdO7b43o9//GO54IILpKOjQ7Zt2ybr1q2Tzs5OaWtrK76/YsUK2bJli+zfv790r9N4pakYmLCASxi1faNzqelCndp6Xc6VYxHQErB9nWmtN+a65BACcL/9efjw4eLjDS+88EIRcK+55hoZMmSIbN68uXacCcVTp04VM3bYsGGl+p3GK8XEoMQFXMKo7RudS00X7tTW63KuHIuAloDt60xrvTHXJYcQgE8q8MUvflEefvhhGTx4sDz11FNy+eWXy/jx42XatGmydu3a2nH79u0r7hbv3LlTJk6cWKrfabxSTAxKXMAljNq+0bnUdOFObb0u58qxCGgJ2L7OtNYbc11yCAH4pAIm2JrP/D744IPFHd8XX3xRZs2aJbNnz5bVq1fXjjt48KCMGjVKtm/fLlOmTKk737Fjx8T8r+fr0KFDMmnSJOnq6pIRI0bE/BphbQhYC7iEUds3Opea1icqIqmt1+VcORYBLQHb15nWemOuSwAmADfszw8++EDGjRsnF110UXGXd/r06XXvAO/evVsmTJhQd76VK1fKqlWr+nyPANyQnwEJC7iEUds3OpeaLtSprdflXDkWAS0B29eZ1npjrksAJgD3EjCPQHv++eeLO72DBg2qfc989tfcwTX/zTz6bOPGjbXvlfkMMHeAY74MsDZfAi5h1PaNzqWmi0Nq63U5V45FQEvA9nWmtd6Y6xKACcC9BMwzfs1dXBOCP/3pTxffe//994tHn82YMUOGDx8uGzZskAMHDtQC8vLly2Xr1q08BSLmVzprUxFwCaO2b3QuNV2QUluvy7lyLAJaAravM631xlyXAEwA7iXQ3d1d/JGbecyZ+QO4008/Xe666y754Q9/WPwDGKecckotDC9btqx4NNq8efOKUDxnzpzSvU7jlaZiYMICLmHU9o3OpaYLdWrrdTlXjkVAS8D2daa13pjrkkMIwH0Efvvb34q5q/vEE0/I0aNH5ZJLLpH777+/9lxg8zlg86+/7dmzR9rb24t/Nc48O7iZLxqvGS3GpirgEkZt3+hcaro4p7Zel3PlWAS0BGxfZ1rrjbkuOYQArNKfNJ4KO0UDC7iEUds3OpeaLjyprdflXDkWAS0B29eZ1npjrksOIQCr9CeNp8JO0cACLmHU9o3OpWZgHsohgECTArbXhSbLZDGcHEIAVml0Gk+FnaKBBVzCqO0bnUvNwDyUQwCBJgVsrwtNlsliODmEAKzS6DSeCjtFAwu4hFHbNzqXmoF5KIcAAk0K2F4XmiyTxXByCAFYpdFpPBV2igYWcAmjtm90LjUD81AOAQSaFLC9LjRZJovh5BACsEqj03gq7BQNLOASRm3f6FxqBuahHAIINClge11oskwWw8khBGCVRqfxVNgpGljAJYzavtG51AzMQzkEEGhSwPa60GSZLIaTQwjAKo1O46mwUzSwgEsYtX2jc6kZmIdyCCDQpIDtdcGUcbk2uNRt8hSDDSeHEICDNdvxhWg8FXaKBhbQeMNxqRmYh3IIINCkgEsQdbk2uNRt8hSDDSeHEICDNRsBWIWaoooCGm84LjUVqSiNAAIlBFyCqMu1waVuidNSGUIAJgDTeCoCFM1BQOMNx6VmDnvCOSKQsoBLEHW5NrjUjdWbAEwAVulNGk+FnaKBBTTecFxqBuahHAIINCngEkRdrg0udZs8xWDDySEE4GDNdnwhGk+FnaKBBTTecFxqBuahHAIINCngEkRdrg0udZs8xWDDySEE4GDNRgBWoaaoooDGG45LTUUqSiOAQAkBlyDqcm1wqVvitFSGEIAJwDSeigBFcxDQeMNxqZnDnnCOCKQs4BJEXa4NLnVj9SYAE4BVepPGU2GnaGABjTccl5qBeSiHAAJNCrgEUZdrg0vdJk8x2HByCAE4WLMdX4jGU2GnaGABjTccl5qBeSiHAAJNCrgEUZdrg0vdJk8x2HByCAE4WLMRgFWoKaoooPGG41JTkYrSCCBQQsAliLpcG1zqljgtlSEEYAIwjaciQNEcBFzecHLw4RwRQKA5AZcg6nI9cqnb3BmGG00AJgCH67bjKtF4KuwUDSzg8oYTeKmUQwCBBARcgqjL9cilbqys5BACsEpv0ngq7BQNLODyhhN4qZRDAIEEBFyCqMv1yKVurKzkEAKwSm/SeCrsFA0s4PKGE3iplEMAgQQEXIKoy/XIpW6srOQQArBKb9J4KuwUDSzg8oYTeKmUQwCBBARcgqjL9cilbqys5BACsEpv0ngq7BQNLODyhhN4qZRDAIEEBFyCqMv1yKVurKzkEAKwSm/SeCrsFA0s4PKGE3iplEMAAQROKkAAzqs5BnR3d3fndcrhzpYAHM6aSnoCBGA9eyojgEB1AgTg6ixTmIkA7HGXCMAecZk6GgECcDRbwUIQQMBBgADsgJfgoQRgj5tGAPaIy9TRCBCAo9kKFoIAAg4CBGAHvAQPJQB73DQCsEdcpo5GgAAczVawEAQQcBAgADvgJXgoAdjjphGAPeIydTQCBOBotoKFIICAgwAB2AEvwUMJwB43jQDsEZepoxEgAEezFSwEAQQcBAjADngJHkoA9rhpBGCPuEwdjQABOJqtYCEIIOAgQAB2wEvwUAKwx00jAHvEZepoBAjA0WwFC0EAAQcBArADXoKHEoA9bhoB2CMuU0cjQACOZitYCAIIOAgQgB3wEjyUAOxx0wjAHnGZOhoBAnA0W8FCEEDAQYAA7ICX4KEEYI+bRgD2iMvU0QgQgKPZChaCAAIOAgRgB7wEDyUAe9w0ArBHXKauXIAgWzkpEyKAQEICBOCENquCpRKAK0A82RQEYI+4TF25AAG4clImRACBhAQIwAltVgVLJQBXgEgA9ojI1MEECMDBqCmEAAIRChCAI9wUj0siAHvE5Q6wR1ymrlyAAFw5KRMigEBCAgTghDargqUSgCtA5A6wR0SmDiZAAA5GTSEEEIhQgAAc4aZ4XBIB2CMud4A94jJ15QIE4MpJmRABBBISIAAntFkVLJUAXAEid4A9IjJ1MAECcDBqCiGAQIQCBOAIN8XjkgjAHnG5A+wRl6krFyAAV07KhAggkJAAATihzapgqQTgChC5A+wRkamDCRCAg1FTCAEEIhQgAEe4KR6XRAD2iMsdYI+4TF25AAG4clImRACBhAQIwAltVgVLJQBXgMgdYI+ITB1MgAAcjJpCCCAQoQABOMJN8bgkArBHXO4Ae8Rl6soFCMCVkzIhAggkJEAATmizKlhqtgH417/+taxYsUKeeeYZOXbsmIwfP17WrFkjU6ZMKVj37t0rixYtkl27dsnQoUNl4cKFsnTp0qbICcBNcTFYWYAArLwBlEcAAVUBArAqf/Di2Qbgyy+/XI4cOSIPPPCAnHnmmbJu3Tp55JFHZPfu3cX/HzNmjMycOVMWL14sO3bskPnz58v69etl7ty5pTeJAFyaioERCBCAI9gEloAAAmoCBGA1epXCWQbgV199Vc455xz50Y9+JJMnTy7gu7u75dxzz5Xrr79eTj311CIQd3Z2SltbW/F9c7d4y5Ytsn///tIbRQAuTcXACAQIwBFsAktAAAE1AQKwGr1K4SwD8NGjR+Wll16Syy67TAYPHlyDHz16tFx55ZVy8OBBGTJkiGzevLn2vW3btsnUqVPl8OHDMmzYsFKbRQAuxcSgSAQIwJFsBMtAAAEVAQKwCrta0SwDcD3t73//+zJr1ix56qmn5LbbbpNp06bJ2rVra0P37dsnY8eOlZ07d8rEiRNLbRgBuBQTgyIRIABHshEsAwEEVAQIwCrsakUJwCLFRyHMnV9zR/iJJ54Qcyd49uzZsnr16trGmLvCo0aNku3bt9f+UO7EXTN/TGf+1/N16NAhmTRpknR1dcmIESPUNpnCCJQRIACXUWIMAgi0qgABuFV3tv55ZR+An3zyySLsXnzxxcUTIcznf8eNGyfTp0+vewfY/JHchAkT6mquXLlSVq1a1ed7BOC8XlSpni0BONWdY90IIFCFAAG4CsV05sg6AJsnQJhHnX32s5+VTZs21T4PbMKvefTZxo0baztZ5jPA3AFOp/FZaV8BAjBdgQACOQsQgPPa/WwD8IMPPlg82sw83/drX/uaDBw4sLbz5nnAGzZskAMHDsigQYOK/758+XLZunUrT4HI6/WR1dkSgLPabk4WAQROECAA59USWQbgV155RT7+8Y/LZz7zmeLZvsd/mY9AvPvuu8VzgGfMmCHLli2Tjo4OmTdvXhGK58yZU7pD+CO40lQMjECAABzBJrAEBBBQEyAAq9GrFM4yAN99993Fkx7qfZmA+93vfrd42oO5O7xnzx5pb2+XJUuWyIIFC5raJAJwU1wMVhYgACtvAOURQEBVgACsyh+8eJYBOJQyATiUNHV6BAix9AICCCBgJ0AAtnNL9SgCsMedIwB7xGXqugIEYBoDAQQQsBMgANu5pXoUAdjjzhGAPeIyNQGYHkAAAQQqFCAAV4iZwFQEYI+bRAD2iMvUBGB6AAEEEKhQgABcIWYCUxGAPW4SAdgjLlMTgOkBBBBAoEIBAnCFmAlMRQD2uEkEYI+4TE0ApgcQQACBCgUIwBViJjAVAdjjJhGAPeIyNQGYHkAAAQQqFCAAV4iZwFQEYI+bRAD2iMvUBGB6AAEEEKhQgABcIWYCUxGAPW4SAdgjLlMTgOkBBBBAoEIBAnCFmAlMRQD2uEkEYI+4TE0ApgcQQACBCgUIwBViJjAVAdjjJhGAPeIyNQGYHkAAAQQqFCAAV4iZwFQEYI+bRAD2iMvUBGB6AAEEEKhQgABcIWYCUxGAPW4SAdgjLlMTgOkBBBBAoEIBAnCFmAlMRQD2uEkEYI+4TE0ApgcQQACBCgUIwBViJjAVAdjjJhGAPeIyNQGYHkAAAQQqFCAAV4iZwFQEYI+bRAD2iMvUBGB6AAEEEKhQgABcIWYCUxGAPW4SAdgjLlMTgOkBBBBAoEIBAnCFmAlMRQD2uEkEYI+4TE0ApgcQQACBCgUIwBViJjAVAdjjJhGAPeIyNQGYHkAAAQQiEIg5OJNDTt4gBGCPLx4azyMuUxOA6QEEEEAgAgECcASbYLEEArAFWtlDCMBlpRhXlcDIW/+1qqmYBwEEEECghAABuARShEMIwB43hQDsEZepuQNMDyCAAAIRCBCAI9gEiyUQgC3Qyh5CAC4rxbiqBLgDXJUk8yCAAALlBAjA5ZxiG0UA9rgjBGCPuEzNHWB6AAEEEIhAgAAcwSZYLIEAbIFW9hACcFmp1hzncjfW9oLqUrM1d4GzQgABBPwK2F6v/a7q97OTQ06uTAD22IE0nkfcBKZ2CaO2F1SXmgmQskQEEEAgOgHb63WIEyGHEIBD9FmfGjSeCns0RV3CqO0F1aVmNHAsBAEEEEhIwPZ6HeIUySEE4BB9RgBWUY63qEsYtb2gutSMV5KVIYAAAvEK2F6vQ5wRAZgAHKLPCMAqyvEWdQmjthdUl5rxSrIyBBBAIF4B2+t1iDMiABOAQ/QZAVhFOd6iLmHU9oLqUjNeSVaGAAIIxCtge70OcUYEYAJwiD4jAKsox1vUJYzaXlBdasYrycoQQACBeAVsr9chzogATAAO0WcEYBXleIu6hFHbC6pLzXglWRkCCCAQr4Dt9TrEGRGACcAh+owArKIcb1GXMGp7QXWpGa8kK0MAAQTiFbC9Xoc4IwIwAThEnxGAVZTjLeoSRm0vqC4145VkZQgggEC8ArbX6xBnRAAmAIfoMwKwinK8RV3CqO0F1aVmvJKsDAEEEIhXwPZ6HeKMCMAE4BB9RgBWUY63qEsYtb2gutSMV5KVIYAAAvEK2F6vQ5wRAZgAHKLPCMAqyvEWdQmjthdUl5rxSrIyBBBAIF4B2+t1iDMiABOAQ/QZAVhFOd6iLmHU9oLqUjNeSVaGAAIIxCtge70OcUYEYAJwiD4jAKsox1vUJYzaXlBdasYrycoQQACBeAVsr9chzogATAAO0WcEYBXleIu6hFHbC6pLzXglWRkCCCAQr4Dt9TrEGRGACcAh+owArKIcb1GXMGp7QXWpGa8kK0MAAQTiFbC9Xoc4IwIwAThEnxGAVZTjLeoSRm0vqC4145VkZQgggEC8ArbX6xBnRAAmAIfoMwKwinK8RV3CqO0F1aVmvJKsDAEEEIhXwPZ6HeKMCMAE4BB9RgBWUY63qEsYtb2gutSMV5KVIYAAAvEK2F6vQ5wRAZgAHKLPCMAqyvEWdQmjthdUl5rxSrIyBBBAIF4B2+t1iDMiABOAQ/QZAVhFOd6iLmHU9oLqUjNeSVaGAAIIxCtge70OcUYEYAJwiD4jAKsox1vUJYzaXlBdasYrycoQQACBeAVsr9chzogATAAO0WcEYBXleIsSRuPdG1aGAAIIVCVAAK5KMuw8A7q7u7vDloyv2p133inbtm2TF154oba4vXv3yqJFi2TXrl0ydOhQWbhwoSxdurSpxfOTV1NcLTeYANxyW8oJIYAAAn0ECMBpNkX2AfjrX/+6LF68WD75yU/WAvAbb7whY8aMkZkzZxbf27Fjh8yfP1/Wr18vc+fOLb3TBODSVC05kADcktvKSSGAAAK9BAjAaTZEtgH4tddek5tuukm2b98uZ511lgwfPrwWgO+55x5Zt26ddHZ2SltbW7GzK1askC1btsj+/ftL7zQBuDRVSw4kALfktnJSCCCAAAG4BXog2wD89NNPy6OPPirm4w+rV68uwm7PRyCmT58uQ4YMkc2bN9e22HxEYurUqXL48GEZNmxYqa0nAJdiatlBBOCW3VpODAEEEKgJcAc4zWbINgAfv1033nhjrwA8fvx4mTZtmqxdu7Y2bN++fTJ27FjZuXOnTJw4se5uHzt2TMz/er4OHTokkyZNkq6uLhkxYkSaHcKqrQUIwNZ0HIgAAggkI0AATmarei2UACwiJwbg0aNHy+zZs4s7wz1fBw8elFGjRhUfmZgyZUrd3V65cqWsWrWqz/cIwGm+OFxXTQB2FeR4BBBAIH4BAnD8e1RvhQTgOgF43LhxYj4GUe8O8O7du2XChAncAU6z34OumgAclJtiCCCAgIoAAViF3bkoAbhOADbh1zz6bOPGjTVgPgPs3GvZTUAAzm7LOWEEEMhQgACc5qYTgOsE4DVr1siGDRvkwIEDMmjQoGJnly9fLlu3buUpEGn2ucqqCcAq7BRFAAEEggoQgINyV1aMAFwnAB85cqR4DvCMGTNk2bJl0tHRIfPmzStC8Zw5c0rj8xSI0lQtOZAA3JLbykkhgAACvQQIwGk2BAG4TgA2W2me9mD+9bc9e/ZIe3u7LFmyRBYsWNDULhOAm+JqucEE4JbbUk4IAQQQ6CNAAE6zKQjAHveNAOwRN4GpCcAJbBJLRAABBBwFCMCOgEqHE4A9whOAPeImMDUBOIFNYokIIICAowAB2BFQ6XACsEd4ArBH3ASmJgAnsEksEQEEEHAUIAA7AiodTgD2CE8A9oibwNQE4AQ2iSUigAACjgIEYEdApcMJwB7hCcAecROYmgCcwCaxRAQQQMBRgADsCKh0OAHYIzwB2CNuAlMTgBPYJJaIAAIIOAoQgB0BlQ4nAHuEJwB7xA00NSE2EDRlEEAAgQwFfIdncsjJm4oA7PEFR+N5xA00NQE4EDRlEEAAgQwFCMB6m04A9mhPAPaIG2hqAnAgaMoggAACGQoQgPU2nQDs0Z4A7BE30NQE4EDQlEEAAQQyFCAA6206AdijPQHYI26gqQnAgaApgwACCGQoQADW23QCsEd7ArBH3EBTE4ADQVMGAQQQyFCAAKy36QRgj/YEYI+4gaYmAAeCpgwCCCCQoQABWG/TCcAe7QnAHnEDTU0ADgRNGQQQQCBDAQKw3qYTgD3aE4A94gaamgAcCJoyCCCAQIYCBGC9TScAe7QnAHvEDTQ1ATgQNGUQQACBDAUIwHqbTgD2aE8A9ogbaGoCcCBoyiCAAAIZChCA9TadAOzRngDsETfQ1ATgQNCUQQABBDIUIADrbToB2KM9AdgjbqCpCcCBoCmDAAIIZChAANbbdAKwR3sCsEfcQFMTgANBUwYBBBDIUIAArLfpBGCP9gRgj7iBpiYAB4KmDAIIIJChAAFYb9MJwB7tCcAecQNNTQAOBE0ZBBBAIEMBArDephOAPdoTgD3iBpqaABwImjIIIIBAhgIEYL1NJwB7tCcAe8QNNDUBOBA0ZRBAAIEMBQjAeptOAPZoTwD2iBtoagJwIGjKIIAAAhkKEID1Np0A7NGeAOwRN9DUBOBA0JRBAAEEMhQgAOttOgHYoz0B2CNuoKkJwIGgKYMAAghkKEAA1tt0ArBHewKwR9xAUxOAA0FTBgEEEMhQgACst+kEYI/2BGCPuIGmJgAHgqYMAgggkKEAAVhv0wnAHu0JwB5xm5yaINskGMMRQAABBLwLEIC9E5+0AAHYoz0B2CNuk1MTgJsEYzgCCCCAgHcBArB3YgKwBjEBWEO9fk0CcDx7wUoQQAABBH4vQADW6wTuAHu0JwB7xG1yagJwk2AMRwABBBDwLkAA9k7MHejWkCsAAA++SURBVGANYgKwhjp3gONRZyUIIIAAAv0JEID1+oM7wB7tCcAecZucmjvATYIxHAEEEEDAuwAB2Dsxd4A1iAnAGurcAY5HnZUggAACCHAHOM4e4A6wx30hAHvEbXJq7gA3CcZwBBBAAAHvAtwB9k7MHWANYgKwhjp3gONRZyUIIIAAAtwBjrMHuAPscV8IwB5xm5yaO8BNgjEcAQQQQMC7AHeAvRNzB1iDmACsoc4d4HjUWQkCCCCAAHeA4+wB7gB73BcCsEfcJqfmDnCTYAxHAAEEEPAuwB1g78TcAdYgJgBrqHMHOB51VoIAAgggwB3gOHuAO8Ae94UA7BG3yam5A9wkGMMRQAABBLwLcAfYOzF3gDWICcAa6twBjkedlSCAAAIIcAc4zh7gDrDHfSEAe8RtcmruADcJxnAEEEAAAe8C3AH2TswdYA1iArCGOneA41FnJQgggAAC3AGOswe4A+xxXwjAHnGbnJo7wE2CMRwBBBBAwLsAd4C9E3MHuFniDz74QFatWiWPPPKI/OY3v5EpU6bI+vXrZfTo0aWnIgCXpvI+kADsnZgCCCCAAAJNChCAmwSrcDh3gE+CacKvCbzf+c535CMf+YgsW7ZMDh48KD/96U/lD//wD0ttAQG4FFOQQQTgIMwUQQABBBBoQoAA3ARWxUMJwHVA33vvPTnjjDPk3nvvlS996UvFiKNHj8qf/MmfyLe//W257rrrSm0DAbgUU5BBBOAgzBRBAAEEEGhCgADcBFbFQwnAdUA7Ojrkoosukp/97Gdy7rnn1kaYj0GMHz++uDNc5osAXEYpzBgCcBhnqiCAAAIIlBcgAJe3qnokAbiO6JYtW+Rzn/ucvPXWW3LqqafWRsyaNav4b88880ypfSAAl2IKMogAHISZIggggAACTQgQgJvAqngoAbgO6KZNm+SGG26Q//mf/5GBAwfWRnz+85+XX/3qV7Jt27a623Ds2DEx/+v56urqksmTJ4u5o9ze3l7x1jFdMwIX3/18M8MZiwACCCCAgHeBHSsu81rj0KFDMmnSJPn5z38uI0eO9FortckJwHV27PHHH5err7667h3gd999V5588sm6+7xy5criyRF8IYAAAggggAACsQiYG3Gf+MQnYllOFOsgANfZhp7PAL/66qsyatSo2gjzGeDzzz9f1q1bV3fzTrwD/M4774i5C3z22WdLW1ublw3v+emOu8wn58WocethhFFjgcYj6COMGgv0P4IeaizYjNHvfvc7ef3112XcuHFyyimnNJ48oxEE4Dqbbe7yDhs2TO6//375whe+UIyweQpEiD7ic8aNlTHCqLFA4xH0EUaNBRqPoI/6N8KHHmosUM0IAvBJHG+77TZ56KGHiseemc/N3HLLLdLZ2Skvv/xy6ecAV7NFXCxcHbmgNhbECKPGAo1H0EcYNRbgPQ0jV4FqjicAn8TR/AHcihUrin8I4+2335ZPfvKTxUcfYvsQOW84jV8IGGHUWKDxCPoIo8YCjUfQRwTgxl2CkatRmeMJwGWUIh5jPndsPqqxePFi+eM//uOIV6q3NIwa22OEUWOBxiPoI4waC/Q/gh5qLIhRY6MyIwjAZZQYgwACCCCAAAIIINAyAgTgltlKTgQBBBBAAAEEEECgjAABuIwSYxBAAAEEEEAAAQRaRoAA3DJbyYkggAACCCCAAAIIlBEgAJdRYgwCCCCAAAIIIIBAywgQgBPdyg8++KD4Z5cfeeQR+c1vfiPmX6lbv369jB49OtEzcl/2L37xi7qPqXv44Yflpptukr1798qiRYtk165dMnToUFm4cKEsXbrUvXAiM9x5552ybds2eeGFF2orbmSSW5/VM5o7d65897vf7bXLH/nIR8Q8zsp8tbrRr3/96+KRkM8884yYvz4fP368rFmzprjmmC96SKSRUe49ZPrkyJEjsmTJEnn22WeLR4t+6lOfkvvuu0/OO+88+uj/ri6NjOijat9sCcDVegabzYRfE3jNc4rNm/GyZcvk4MGD8tOf/jSqf6gjGIiIPP300zJr1qzCYcCAAbXSH/7wh+Wtt96SMWPGyMyZM4tHxu3YsUPmz59fGJqLSqt/ff3rXy/O2zzPuicAv/HGGw1NcuqzekamL/78z/9crrzySvn7v//7WpsMGjRIzjzzzOL/t7rR5ZdfXoSXBx54oDhn8zx084P37t27i//f6HXV6j6mB/ozMj6595Axuvjii2XgwIHyzW9+U0477TS544475KWXXpJXX321CMT0Uf9Gf/RHf0QfVfxGTQCuGDTEdO+9956cccYZcu+998qXvvSlomSs/1RzCI+eGnfffbc89thjxR2pE7/uueee4o3b/Gt+bW1txbfNXa0tW7bI/v37Qy4zaK3XXnutuPu9fft2Oeuss2T48OG1ANzIJJc+68/I/IM4H/rQh+TRRx8tfng68avVjUw4Oeecc+RHP/qRTJ48uTj97u5uOffcc+X666+XU089td/XVav7GI9GRl/+8pez7iFjZH7YXrBggdx+++0yduzYoo9+/OMfywUXXCAdHR3Fb6b6uz7n0EeNjC688MLs+6jqN08CcNWiAeYzF4yLLrpIfvaznxVvRD1f5leS5teT5q5mjl/XXXddcYdh8+bNfU5/+vTpMmTIkF7fMxfdqVOnyuHDh2XYsGEtSWbuipvwZn61v3r16uIHgJ47wI1MzNgc+qw/I/PD0cc+9rE+r7WeZmn116L5wdrcpbvssstk8ODBtdeI+aiVuStuftvS3+sqhx5qZGSCX849VO/Caq655uNn5lpkXmPXXHNN9n10otOJRl1dXfRRxe/SBOCKQUNMZ+5afu5znyt+rW/uwPR8mV//m/9mPquX45e5s9De3i7vvvuuvPLKK8WdK/NrtiuuuKL4wWDatGmydu3aGs2+ffuKuxE7d+6UiRMntjzZjTfe2CsANzL55S9/mV2fnWj0/e9/X6699triNy3/9m//VvyAZX5wMD9QmI/W5PhaNCbmWvPUU0/Jbbfd1u/rKsceMheS443eeecdeui4q+sXv/hFMX+XYX6gMj1kPj7Ctaj32089I65F1b9FE4CrN/U+46ZNm+SGG24Q8+tZ84bc8/X5z39efvWrXxW/Tsrty/yKzHyubMKECcUfVphfWxunb3zjG/Lcc8/JzTffLLNnzy7ugvZ8mbtXo0aNKj4e0PMHPa3sdmK4M3fx+jMxd+9y67MTjcyvr++66y4xH68xwfe//uu/ijtX5uMk//7v/178RiEnI/NRCHPn19wRfuKJJ4o/uqWHel81TjSih3r7mBsP5jO/Dz74YPH6efHFF4sfqOij/3eqZ/Tkk09yLar4DZoAXDFoiOkef/xxufrqq+veATZ3P80LJcevN998s/h87/G/qjVv1uYP4sxf7JsAU+8OsPljHhOcW/3rxHA3bty4fk3MDwi59dmJRubzrr/97W+LX8/2fJk/oPyLv/iL4g8pTV/lYmSuKyakmD9mMr9lMr99ood6XzXqGdFD9a+s5ukppn/Mx6zMb+H6uz7neC0yascb/eM//iPXoorfpAnAFYOGmK7nc4fmjy/MHcyeL3MX8/zzzy/+mICv3wvccsstxWN3zB078+izjRs31mhy+Azw8X1wYrgzbzj9mfR8fjOnPjvRqN7ryPygZX7D8C//8i9FX5k38FY3Mk+AMI8Q/OxnP1v8ZqXnh0x66P875GRG9NDvH4H2/PPPF3d6zRNUer7MZ3/No/XMf8v9WtTI6Ic//GGfVsrxWlRltiEAV6kZaC5zl9f80db9998vX/jCF4qquT8F4ic/+UnxV+om7B7/cYZLL720ePKB+WvjDRs2yIEDB2oX4OXLl8vWrVtb+ikQ/QVg8yzX/kxy7LMTA7C542n+Ovv4Nx/zhzumr8yvKT/60Y+2/GvR/KraPDLQPDf7a1/7Wq+PXdFDv3+F9WdED/3+WdHmt2wmBH/6058uzN5///3i0WczZswortG5X4saGZk/isv9WlR1xCIAVy0aaD7zxycPPfSQfPvb3y7+8Qdzp9PcsXv55ZezfA6w+VWRCcDmJ2LzZmQeE/etb32reCKG+fWaucD2XGzNM5PNXfR58+YVF905c+YE2jXdMieGO3PHoZFJbn12opH5Vb95g/7KV74i5ikj5o8r/+7v/q74CMQ///M/FxvaykbmfD/+8Y/LZz7zmT5PlzEfgTA/JOXeQ42MzN8Y5NxD5jViPgZi/gjZvEeZP4A7/fTTi8+zmh8sTfA75ZRTsu+jRkbmJk/ufVT1OygBuGrRQPOZP4Azz7E1/xCG+YMC8w8cmI8+mDCc69frr78ut956q/zgBz8o7oib5yaaO1SXXHJJQWKCsLmLtWfPnuJpEeZfJTKPKMrlq96v9xuZ5NZn9YzMZ+7Nm7V5XJP5LLC5o2eeAmHetM1XKxuZP/4zAb/el/nB0fwLebn3UBmjnHuop3fMZ+nNb93MH0+a67O5LpvfYvY8Fzj3PjJOjYzoo2rfrQnA1XoyGwIIIIAAAggggEDkAgTgyDeI5SGAAAIIIIAAAghUK0AArtaT2RBAAAEEEEAAAQQiFyAAR75BLA8BBBBAAAEEEECgWgECcLWezIYAAggggAACCCAQuQABOPINYnkIIIAAAggggAAC1QoQgKv1ZDYEEEAAAQQQQACByAUIwJFvEMtDAAEEEEAAAQQQqFaAAFytJ7MhgAACCCCAAAIIRC5AAI58g1geAggggAACCCCAQLUCBOBqPZkNAQQQQAABBBBAIHIBAnDkG8TyEEAAAQQQQAABBKoVIABX68lsCCCAAAIIIIAAApELEIAj3yCWhwACCCCAAAIIIFCtAAG4Wk9mQwABBBBAAAEEEIhcgAAc+QaxPAQQQAABBBBAAIFqBQjA1XoyGwIIIIAAAggggEDkAgTgyDeI5SGAAAIIIIAAAghUK0AArtaT2RBAAAEEEEAAAQQiFyAAR75BLA8BBBBAAAEEEECgWgECcLWezIYAAggggAACCCAQuQABOPINYnkIIIAAAggggAAC1QoQgKv1ZDYEEEAAAQQQQACByAUIwJFvEMtDAAEEEEAAAQQQqFaAAFytJ7MhgAACCCCAAAIIRC5AAI58g1geAggggAACCCCAQLUCBOBqPZkNAQQQQAABBBBAIHIBAnDkG8TyEEAAAQQQQAABBKoVIABX68lsCCCAAAIIIIAAApELEIAj3yCWhwACCCCAAAIIIFCtAAG4Wk9mQwABBBBAAAEEEIhcgAAc+QaxPAQQQAABBBBAAIFqBQjA1XoyGwIIIIAAAggggEDkAgTgyDeI5SGAAAIIIIAAAghUK0AArtaT2RBAAAEEEEAAAQQiFyAAR75BLA8BBBBAAAEEEECgWoH/BanyJ78jdK9NAAAAAElFTkSuQmCC\" width=\"639.9999861283738\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure()\n",
    "plt.hist(list(reg_size_dic.values()), bins=np.arange(0,350,10))\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "511ed38f",
   "metadata": {},
   "source": [
    "### 4.1.3 check readout usage for each gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "acece126",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "num_readout_per_probe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f46dbd8e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 -> ['NDB_856', 'NDB_935', 'NDB_946']\n",
      "2 -> ['NDB_742', 'NDB_883', 'NDB_965']\n",
      "3 -> ['NDB_1066', 'NDB_694', 'NDB_754']\n",
      "4 -> ['NDB_752', 'NDB_832', 'NDB_940']\n",
      "5 -> ['NDB_698', 'NDB_818', 'NDB_889']\n",
      "6 -> ['NDB_862', 'NDB_899', 'NDB_989']\n",
      "7 -> ['NDB_637', 'NDB_853', 'NDB_977']\n",
      "8 -> ['NDB_1033', 'NDB_730', 'NDB_974']\n",
      "9 -> ['NDB_865', 'NDB_878', 'NDB_925']\n",
      "10 -> ['NDB_721', 'NDB_734', 'NDB_905']\n",
      "11 -> ['NDB_647', 'NDB_847', 'NDB_941']\n",
      "12 -> ['NDB_815', 'NDB_838', 'NDB_863']\n",
      "13 -> ['NDB_812', 'NDB_868', 'NDB_946']\n",
      "14 -> ['NDB_718', 'NDB_758', 'NDB_844']\n",
      "15 -> ['NDB_683', 'NDB_725', 'NDB_817']\n",
      "16 -> ['NDB_653', 'NDB_713', 'NDB_860']\n",
      "17 -> ['NDB_737', 'NDB_751', 'NDB_958']\n",
      "18 -> ['NDB_652', 'NDB_778', 'NDB_880']\n",
      "19 -> ['NDB_1027', 'NDB_887', 'NDB_961']\n",
      "20 -> ['NDB_665', 'NDB_773', 'NDB_790']\n",
      "21 -> ['NDB_1028', 'NDB_824', 'NDB_866']\n",
      "22 -> ['NDB_641', 'NDB_643', 'NDB_710']\n",
      "23 -> ['NDB_680', 'NDB_779', 'NDB_784']\n",
      "24 -> ['NDB_872', 'NDB_926', 'NDB_994']\n",
      "25 -> ['NDB_661', 'NDB_835', 'NDB_901']\n",
      "26 -> ['NDB_637', 'NDB_743', 'NDB_838']\n",
      "27 -> ['NDB_728', 'NDB_836', 'NDB_880']\n",
      "28 -> ['NDB_730', 'NDB_929', 'NDB_977']\n",
      "29 -> ['NDB_706', 'NDB_788', 'NDB_826']\n",
      "30 -> ['NDB_1075', 'NDB_905', 'NDB_934']\n",
      "31 -> ['NDB_1033', 'NDB_652', 'NDB_899']\n",
      "32 -> ['NDB_755', 'NDB_866', 'NDB_917']\n",
      "33 -> ['NDB_716', 'NDB_725', 'NDB_875']\n",
      "34 -> ['NDB_712', 'NDB_910', 'NDB_941']\n",
      "35 -> ['NDB_718', 'NDB_805', 'NDB_832']\n",
      "36 -> ['NDB_754', 'NDB_847', 'NDB_883']\n",
      "37 -> ['NDB_634', 'NDB_865', 'NDB_935']\n",
      "38 -> ['NDB_715', 'NDB_896', 'NDB_925']\n",
      "39 -> ['NDB_680', 'NDB_752', 'NDB_773']\n",
      "40 -> ['NDB_742', 'NDB_889', 'NDB_940']\n",
      "41 -> ['NDB_860', 'NDB_862', 'NDB_965']\n",
      "42 -> ['NDB_1066', 'NDB_751', 'NDB_934']\n",
      "43 -> ['NDB_683', 'NDB_958', 'NDB_971']\n",
      "44 -> ['NDB_1075', 'NDB_721', 'NDB_758']\n",
      "45 -> ['NDB_1060', 'NDB_760', 'NDB_881']\n",
      "46 -> ['NDB_746', 'NDB_826', 'NDB_868']\n",
      "47 -> ['NDB_743', 'NDB_899', 'NDB_946']\n",
      "48 -> ['NDB_665', 'NDB_818', 'NDB_856']\n",
      "49 -> ['NDB_734', 'NDB_863', 'NDB_887']\n",
      "50 -> ['NDB_1028', 'NDB_631', 'NDB_878']\n",
      "51 -> ['NDB_643', 'NDB_661', 'NDB_917']\n",
      "52 -> ['NDB_1060', 'NDB_712', 'NDB_896']\n",
      "53 -> ['NDB_737', 'NDB_929', 'NDB_934']\n",
      "54 -> ['NDB_713', 'NDB_715', 'NDB_746']\n",
      "55 -> ['NDB_653', 'NDB_778', 'NDB_788']\n",
      "56 -> ['NDB_710', 'NDB_875', 'NDB_989']\n",
      "57 -> ['NDB_1027', 'NDB_1075', 'NDB_815']\n",
      "58 -> ['NDB_698', 'NDB_779', 'NDB_836']\n",
      "59 -> ['NDB_728', 'NDB_835', 'NDB_977']\n",
      "60 -> ['NDB_647', 'NDB_694', 'NDB_994']\n",
      "61 -> ['NDB_716', 'NDB_832', 'NDB_961']\n",
      "62 -> ['NDB_790', 'NDB_883', 'NDB_971']\n",
      "63 -> ['NDB_812', 'NDB_838', 'NDB_901']\n",
      "64 -> ['NDB_641', 'NDB_856', 'NDB_974']\n",
      "65 -> ['NDB_1033', 'NDB_853', 'NDB_926']\n",
      "66 -> ['NDB_784', 'NDB_862', 'NDB_872']\n",
      "67 -> ['NDB_844', 'NDB_881', 'NDB_935']\n",
      "68 -> ['NDB_634', 'NDB_751', 'NDB_755']\n",
      "69 -> ['NDB_758', 'NDB_760', 'NDB_817']\n",
      "70 -> ['NDB_824', 'NDB_847', 'NDB_929']\n",
      "71 -> ['NDB_754', 'NDB_805', 'NDB_965']\n",
      "72 -> ['NDB_652', 'NDB_730', 'NDB_910']\n",
      "73 -> ['NDB_631', 'NDB_860', 'NDB_865']\n",
      "74 -> ['NDB_706', 'NDB_743', 'NDB_889']\n",
      "75 -> ['NDB_1066', 'NDB_637', 'NDB_721']\n",
      "76 -> ['NDB_698', 'NDB_742', 'NDB_875']\n",
      "77 -> ['NDB_683', 'NDB_694', 'NDB_940']\n",
      "80 -> ['NDB_665', 'NDB_824', 'NDB_974']\n",
      "81 -> ['NDB_715', 'NDB_716', 'NDB_718']\n",
      "82 -> ['NDB_710', 'NDB_844', 'NDB_868']\n",
      "83 -> ['NDB_815', 'NDB_958', 'NDB_961']\n",
      "84 -> ['NDB_643', 'NDB_713', 'NDB_778']\n",
      "85 -> ['NDB_680', 'NDB_901', 'NDB_917']\n",
      "86 -> ['NDB_647', 'NDB_752', 'NDB_755']\n",
      "87 -> ['NDB_760', 'NDB_790', 'NDB_812']\n",
      "88 -> ['NDB_706', 'NDB_818', 'NDB_863']\n",
      "89 -> ['NDB_641', 'NDB_661', 'NDB_728']\n",
      "90 -> ['NDB_1033', 'NDB_758', 'NDB_977']\n",
      "91 -> ['NDB_637', 'NDB_754', 'NDB_887']\n",
      "92 -> ['NDB_832', 'NDB_860', 'NDB_866']\n",
      "93 -> ['NDB_742', 'NDB_905', 'NDB_910']\n",
      "94 -> ['NDB_694', 'NDB_844', 'NDB_929']\n",
      "95 -> ['NDB_853', 'NDB_880', 'NDB_946']\n",
      "96 -> ['NDB_652', 'NDB_779', 'NDB_788']\n",
      "97 -> ['NDB_838', 'NDB_878', 'NDB_989']\n",
      "98 -> ['NDB_1060', 'NDB_773', 'NDB_784']\n",
      "99 -> ['NDB_760', 'NDB_862', 'NDB_865']\n",
      "100 -> ['NDB_712', 'NDB_836', 'NDB_881']\n",
      "101 -> ['NDB_653', 'NDB_817', 'NDB_872']\n",
      "102 -> ['NDB_634', 'NDB_746', 'NDB_835']\n",
      "103 -> ['NDB_743', 'NDB_818', 'NDB_935']\n",
      "104 -> ['NDB_1027', 'NDB_734', 'NDB_965']\n",
      "105 -> ['NDB_832', 'NDB_883', 'NDB_994']\n",
      "106 -> ['NDB_694', 'NDB_721', 'NDB_728']\n",
      "107 -> ['NDB_730', 'NDB_847', 'NDB_971']\n",
      "108 -> ['NDB_1066', 'NDB_742', 'NDB_941']\n",
      "109 -> ['NDB_725', 'NDB_751', 'NDB_889']\n",
      "110 -> ['NDB_737', 'NDB_856', 'NDB_862']\n",
      "111 -> ['NDB_826', 'NDB_863', 'NDB_896']\n",
      "112 -> ['NDB_637', 'NDB_715', 'NDB_926']\n",
      "113 -> ['NDB_1028', 'NDB_758', 'NDB_778']\n",
      "114 -> ['NDB_1075', 'NDB_683', 'NDB_946']\n",
      "115 -> ['NDB_1033', 'NDB_641', 'NDB_940']\n",
      "116 -> ['NDB_805', 'NDB_818', 'NDB_925']\n",
      "117 -> ['NDB_754', 'NDB_899', 'NDB_934']\n",
      "118 -> ['NDB_652', 'NDB_838', 'NDB_860']\n",
      "119 -> ['NDB_865', 'NDB_875', 'NDB_929']\n",
      "120 -> ['NDB_631', 'NDB_680', 'NDB_977']\n",
      "121 -> ['NDB_1027', 'NDB_698', 'NDB_872']\n",
      "122 -> ['NDB_643', 'NDB_815', 'NDB_935']\n",
      "123 -> ['NDB_716', 'NDB_755', 'NDB_958']\n",
      "124 -> ['NDB_713', 'NDB_779', 'NDB_941']\n",
      "125 -> ['NDB_710', 'NDB_905', 'NDB_974']\n",
      "126 -> ['NDB_812', 'NDB_878', 'NDB_881']\n",
      "127 -> ['NDB_634', 'NDB_880', 'NDB_971']\n",
      "128 -> ['NDB_1075', 'NDB_889', 'NDB_917']\n",
      "129 -> ['NDB_665', 'NDB_683', 'NDB_752']\n",
      "130 -> ['NDB_641', 'NDB_824', 'NDB_961']\n",
      "131 -> ['NDB_715', 'NDB_778', 'NDB_826']\n",
      "132 -> ['NDB_836', 'NDB_866', 'NDB_989']\n",
      "133 -> ['NDB_773', 'NDB_788', 'NDB_994']\n",
      "134 -> ['NDB_647', 'NDB_853', 'NDB_910']\n",
      "135 -> ['NDB_653', 'NDB_737', 'NDB_896']\n",
      "136 -> ['NDB_734', 'NDB_817', 'NDB_901']\n",
      "137 -> ['NDB_631', 'NDB_706', 'NDB_926']\n",
      "138 -> ['NDB_1028', 'NDB_661', 'NDB_847']\n",
      "140 -> ['NDB_698', 'NDB_751', 'NDB_965']\n",
      "141 -> ['NDB_1066', 'NDB_856', 'NDB_875']\n",
      "142 -> ['NDB_883', 'NDB_934', 'NDB_940']\n",
      "143 -> ['NDB_721', 'NDB_730', 'NDB_899']\n",
      "144 -> ['NDB_718', 'NDB_746', 'NDB_760']\n",
      "145 -> ['NDB_712', 'NDB_790', 'NDB_844']\n",
      "146 -> ['NDB_725', 'NDB_917', 'NDB_925']\n",
      "147 -> ['NDB_665', 'NDB_805', 'NDB_887']\n",
      "148 -> ['NDB_824', 'NDB_835', 'NDB_868']\n",
      "150 -> ['NDB_743', 'NDB_752', 'NDB_784']\n",
      "151 -> ['NDB_1060', 'NDB_826', 'NDB_961']\n",
      "153 -> ['NDB_1075', 'NDB_698', 'NDB_883']\n",
      "154 -> ['NDB_641', 'NDB_826', 'NDB_946']\n",
      "155 -> ['NDB_652', 'NDB_661', 'NDB_836']\n",
      "156 -> ['NDB_1066', 'NDB_880', 'NDB_994']\n",
      "157 -> ['NDB_1028', 'NDB_647', 'NDB_788']\n",
      "158 -> ['NDB_896', 'NDB_941', 'NDB_971']\n",
      "159 -> ['NDB_1060', 'NDB_725', 'NDB_779']\n",
      "160 -> ['NDB_713', 'NDB_812', 'NDB_817']\n",
      "161 -> ['NDB_784', 'NDB_866', 'NDB_887']\n",
      "162 -> ['NDB_838', 'NDB_847', 'NDB_934']\n",
      "163 -> ['NDB_637', 'NDB_752', 'NDB_889']\n",
      "164 -> ['NDB_715', 'NDB_899', 'NDB_935']\n",
      "165 -> ['NDB_683', 'NDB_805', 'NDB_961']\n",
      "166 -> ['NDB_751', 'NDB_860', 'NDB_868']\n",
      "167 -> ['NDB_1033', 'NDB_746', 'NDB_836']\n",
      "168 -> ['NDB_835', 'NDB_844', 'NDB_994']\n",
      "169 -> ['NDB_716', 'NDB_815', 'NDB_965']\n",
      "170 -> ['NDB_880', 'NDB_881', 'NDB_905']\n",
      "171 -> ['NDB_1028', 'NDB_712', 'NDB_958']\n",
      "172 -> ['NDB_653', 'NDB_773', 'NDB_878']\n",
      "173 -> ['NDB_868', 'NDB_910', 'NDB_926']\n",
      "174 -> ['NDB_1027', 'NDB_661', 'NDB_790']\n",
      "175 -> ['NDB_710', 'NDB_788', 'NDB_863']\n",
      "176 -> ['NDB_706', 'NDB_725', 'NDB_728']\n",
      "177 -> ['NDB_784', 'NDB_817', 'NDB_989']\n",
      "178 -> ['NDB_779', 'NDB_866', 'NDB_971']\n",
      "179 -> ['NDB_631', 'NDB_634', 'NDB_872']\n",
      "180 -> ['NDB_734', 'NDB_853', 'NDB_925']\n",
      "181 -> ['NDB_896', 'NDB_901', 'NDB_910']\n",
      "182 -> ['NDB_713', 'NDB_718', 'NDB_755']\n",
      "183 -> ['NDB_643', 'NDB_832', 'NDB_974']\n",
      "184 -> ['NDB_680', 'NDB_742', 'NDB_856']\n",
      "185 -> ['NDB_665', 'NDB_712', 'NDB_826']\n",
      "186 -> ['NDB_716', 'NDB_760', 'NDB_917']\n",
      "187 -> ['NDB_1075', 'NDB_743', 'NDB_835']\n",
      "188 -> ['NDB_737', 'NDB_805', 'NDB_872']\n",
      "190 -> ['NDB_694', 'NDB_865', 'NDB_977']\n",
      "191 -> ['NDB_758', 'NDB_824', 'NDB_946']\n",
      "192 -> ['NDB_730', 'NDB_818', 'NDB_883']\n",
      "193 -> ['NDB_1060', 'NDB_698', 'NDB_721']\n",
      "195 -> ['NDB_641', 'NDB_754', 'NDB_875']\n",
      "196 -> ['NDB_1066', 'NDB_778', 'NDB_940']\n",
      "197 -> ['NDB_652', 'NDB_862', 'NDB_929']\n",
      "198 -> ['NDB_680', 'NDB_926', 'NDB_941']\n",
      "199 -> ['NDB_812', 'NDB_887', 'NDB_974']\n",
      "200 -> ['NDB_643', 'NDB_647', 'NDB_737']\n",
      "201 -> ['NDB_706', 'NDB_755', 'NDB_881']\n",
      "202 -> ['NDB_905', 'NDB_925', 'NDB_989']\n",
      "203 -> ['NDB_1027', 'NDB_710', 'NDB_773']\n",
      "204 -> ['NDB_718', 'NDB_728', 'NDB_734']\n",
      "205 -> ['NDB_653', 'NDB_815', 'NDB_901']\n",
      "206 -> ['NDB_631', 'NDB_746', 'NDB_790']\n",
      "207 -> ['NDB_634', 'NDB_863', 'NDB_878']\n",
      "208 -> ['NDB_752', 'NDB_853', 'NDB_965']\n",
      "209 -> ['NDB_844', 'NDB_847', 'NDB_958']\n",
      "210 -> ['NDB_760', 'NDB_824', 'NDB_889']\n",
      "211 -> ['NDB_742', 'NDB_934', 'NDB_935']\n",
      "212 -> ['NDB_715', 'NDB_860', 'NDB_977']\n",
      "213 -> ['NDB_751', 'NDB_818', 'NDB_838']\n",
      "214 -> ['NDB_865', 'NDB_899', 'NDB_961']\n",
      "215 -> ['NDB_1033', 'NDB_637', 'NDB_683']\n",
      "218 -> ['NDB_710', 'NDB_779', 'NDB_805']\n",
      "219 -> ['NDB_743', 'NDB_788', 'NDB_910']\n",
      "220 -> ['NDB_683', 'NDB_716', 'NDB_899']\n",
      "221 -> ['NDB_847', 'NDB_868', 'NDB_965']\n",
      "222 -> ['NDB_725', 'NDB_887', 'NDB_941']\n",
      "223 -> ['NDB_730', 'NDB_925', 'NDB_940']\n",
      "224 -> ['NDB_856', 'NDB_863', 'NDB_881']\n",
      "225 -> ['NDB_634', 'NDB_832', 'NDB_958']\n",
      "226 -> ['NDB_1060', 'NDB_680', 'NDB_754']\n",
      "227 -> ['NDB_706', 'NDB_778', 'NDB_862']\n",
      "228 -> ['NDB_712', 'NDB_773', 'NDB_866']\n",
      "229 -> ['NDB_661', 'NDB_815', 'NDB_926']\n",
      "230 -> ['NDB_653', 'NDB_917', 'NDB_994']\n",
      "231 -> ['NDB_734', 'NDB_758', 'NDB_971']\n",
      "232 -> ['NDB_1027', 'NDB_853', 'NDB_989']\n",
      "233 -> ['NDB_694', 'NDB_718', 'NDB_901']\n",
      "234 -> ['NDB_721', 'NDB_878', 'NDB_910']\n",
      "235 -> ['NDB_755', 'NDB_784', 'NDB_880']\n",
      "236 -> ['NDB_728', 'NDB_788', 'NDB_896']\n",
      "237 -> ['NDB_647', 'NDB_805', 'NDB_835']\n",
      "238 -> ['NDB_643', 'NDB_872', 'NDB_929']\n",
      "239 -> ['NDB_713', 'NDB_716', 'NDB_790']\n",
      "240 -> ['NDB_710', 'NDB_746', 'NDB_812']\n",
      "242 -> ['NDB_737', 'NDB_875', 'NDB_905']\n",
      "243 -> ['NDB_653', 'NDB_715', 'NDB_934']\n",
      "244 -> ['NDB_1060', 'NDB_641', 'NDB_977']\n",
      "245 -> ['NDB_1075', 'NDB_631', 'NDB_760']\n",
      "246 -> ['NDB_818', 'NDB_826', 'NDB_974']\n",
      "247 -> ['NDB_665', 'NDB_836', 'NDB_844']\n",
      "248 -> ['NDB_1028', 'NDB_1066', 'NDB_779']\n",
      "249 -> ['NDB_743', 'NDB_817', 'NDB_860']\n",
      "250 -> ['NDB_661', 'NDB_838', 'NDB_889']\n",
      "251 -> ['NDB_1033', 'NDB_824', 'NDB_883']\n",
      "252 -> ['NDB_637', 'NDB_742', 'NDB_865']\n",
      "253 -> ['NDB_751', 'NDB_752', 'NDB_935']\n",
      "254 -> ['NDB_652', 'NDB_698', 'NDB_961']\n",
      "255 -> ['NDB_712', 'NDB_778', 'NDB_946']\n",
      "256 -> ['NDB_754', 'NDB_815', 'NDB_868']\n",
      "257 -> ['NDB_647', 'NDB_718', 'NDB_880']\n",
      "258 -> ['NDB_737', 'NDB_941', 'NDB_994']\n",
      "259 -> ['NDB_853', 'NDB_878', 'NDB_917']\n",
      "260 -> ['NDB_784', 'NDB_790', 'NDB_958']\n",
      "261 -> ['NDB_643', 'NDB_721', 'NDB_863']\n",
      "262 -> ['NDB_634', 'NDB_875', 'NDB_901']\n",
      "263 -> ['NDB_832', 'NDB_835', 'NDB_872']\n",
      "264 -> ['NDB_694', 'NDB_706', 'NDB_758']\n",
      "266 -> ['NDB_631', 'NDB_905', 'NDB_940']\n",
      "267 -> ['NDB_665', 'NDB_680', 'NDB_728']\n",
      "268 -> ['NDB_817', 'NDB_862', 'NDB_881']\n",
      "269 -> ['NDB_734', 'NDB_896', 'NDB_989']\n",
      "270 -> ['NDB_725', 'NDB_773', 'NDB_974']\n",
      "271 -> ['NDB_713', 'NDB_866', 'NDB_926']\n",
      "272 -> ['NDB_1027', 'NDB_812', 'NDB_925']\n",
      "273 -> ['NDB_683', 'NDB_730', 'NDB_746']\n",
      "274 -> ['NDB_1028', 'NDB_755', 'NDB_818']\n",
      "276 -> ['NDB_887', 'NDB_946', 'NDB_971']\n",
      "277 -> ['NDB_856', 'NDB_929', 'NDB_965']\n",
      "278 -> ['NDB_715', 'NDB_836', 'NDB_847']\n",
      "279 -> ['NDB_760', 'NDB_899', 'NDB_977']\n",
      "281 -> ['NDB_680', 'NDB_706', 'NDB_866']\n",
      "282 -> ['NDB_1066', 'NDB_835', 'NDB_971']\n",
      "283 -> ['NDB_641', 'NDB_868', 'NDB_883']\n",
      "285 -> ['NDB_836', 'NDB_875', 'NDB_974']\n",
      "286 -> ['NDB_634', 'NDB_713', 'NDB_721']\n",
      "287 -> ['NDB_754', 'NDB_758', 'NDB_880']\n",
      "288 -> ['NDB_887', 'NDB_910', 'NDB_925']\n",
      "289 -> ['NDB_637', 'NDB_751', 'NDB_872']\n",
      "290 -> ['NDB_1027', 'NDB_742', 'NDB_958']\n",
      "291 -> ['NDB_716', 'NDB_817', 'NDB_905']\n",
      "292 -> ['NDB_698', 'NDB_805', 'NDB_838']\n",
      "293 -> ['NDB_779', 'NDB_826', 'NDB_994']\n",
      "294 -> ['NDB_661', 'NDB_824', 'NDB_863']\n",
      "295 -> ['NDB_647', 'NDB_832', 'NDB_844']\n",
      "296 -> ['NDB_631', 'NDB_641', 'NDB_815']\n",
      "297 -> ['NDB_743', 'NDB_868', 'NDB_989']\n",
      "298 -> ['NDB_1075', 'NDB_730', 'NDB_752']\n",
      "299 -> ['NDB_718', 'NDB_847', 'NDB_853']\n",
      "300 -> ['NDB_746', 'NDB_862', 'NDB_977']\n",
      "301 -> ['NDB_1060', 'NDB_751', 'NDB_925']\n",
      "302 -> ['NDB_856', 'NDB_899', 'NDB_971']\n",
      "303 -> ['NDB_653', 'NDB_706', 'NDB_790']\n",
      "304 -> ['NDB_1028', 'NDB_713', 'NDB_742']\n",
      "305 -> ['NDB_788', 'NDB_835', 'NDB_929']\n",
      "306 -> ['NDB_1027', 'NDB_634', 'NDB_860']\n",
      "307 -> ['NDB_1066', 'NDB_725', 'NDB_961']\n",
      "308 -> ['NDB_712', 'NDB_784', 'NDB_917']\n",
      "309 -> ['NDB_694', 'NDB_755', 'NDB_896']\n",
      "310 -> ['NDB_652', 'NDB_824', 'NDB_905']\n",
      "311 -> ['NDB_665', 'NDB_778', 'NDB_926']\n",
      "312 -> ['NDB_875', 'NDB_881', 'NDB_940']\n",
      "313 -> ['NDB_728', 'NDB_746', 'NDB_878']\n",
      "314 -> ['NDB_734', 'NDB_812', 'NDB_883']\n",
      "315 -> ['NDB_1060', 'NDB_889', 'NDB_901']\n",
      "316 -> ['NDB_715', 'NDB_760', 'NDB_887']\n",
      "318 -> ['NDB_818', 'NDB_934', 'NDB_946']\n",
      "319 -> ['NDB_1033', 'NDB_643', 'NDB_941']\n",
      "320 -> ['NDB_1028', 'NDB_773', 'NDB_865']\n",
      "321 -> ['NDB_683', 'NDB_737', 'NDB_965']\n",
      "322 -> ['NDB_710', 'NDB_853', 'NDB_866']\n",
      "323 -> ['NDB_680', 'NDB_718', 'NDB_860']\n",
      "324 -> ['NDB_647', 'NDB_652', 'NDB_935']\n",
      "325 -> ['NDB_728', 'NDB_910', 'NDB_958']\n",
      "326 -> ['NDB_637', 'NDB_790', 'NDB_836']\n",
      "327 -> ['NDB_661', 'NDB_698', 'NDB_989']\n",
      "328 -> ['NDB_778', 'NDB_863', 'NDB_917']\n",
      "329 -> ['NDB_716', 'NDB_730', 'NDB_889']\n",
      "330 -> ['NDB_817', 'NDB_838', 'NDB_844']\n",
      "331 -> ['NDB_694', 'NDB_752', 'NDB_826']\n",
      "332 -> ['NDB_712', 'NDB_754', 'NDB_929']\n",
      "333 -> ['NDB_734', 'NDB_743', 'NDB_881']\n",
      "334 -> ['NDB_755', 'NDB_779', 'NDB_974']\n",
      "335 -> ['NDB_815', 'NDB_856', 'NDB_880']\n",
      "336 -> ['NDB_1033', 'NDB_1075', 'NDB_872']\n",
      "337 -> ['NDB_631', 'NDB_758', 'NDB_994']\n",
      "338 -> ['NDB_665', 'NDB_721', 'NDB_725']\n",
      "339 -> ['NDB_710', 'NDB_832', 'NDB_934']\n",
      "340 -> ['NDB_805', 'NDB_862', 'NDB_935']\n",
      "342 -> ['NDB_868', 'NDB_925', 'NDB_958']\n",
      "343 -> ['NDB_683', 'NDB_844', 'NDB_883']\n",
      "344 -> ['NDB_737', 'NDB_835', 'NDB_865']\n",
      "345 -> ['NDB_760', 'NDB_901', 'NDB_940']\n",
      "346 -> ['NDB_875', 'NDB_926', 'NDB_961']\n",
      "347 -> ['NDB_721', 'NDB_847', 'NDB_977']\n",
      "348 -> ['NDB_1033', 'NDB_832', 'NDB_887']\n",
      "349 -> ['NDB_713', 'NDB_815', 'NDB_881']\n",
      "350 -> ['NDB_631', 'NDB_730', 'NDB_836']\n",
      "351 -> ['NDB_641', 'NDB_860', 'NDB_989']\n",
      "352 -> ['NDB_661', 'NDB_758', 'NDB_872']\n",
      "353 -> ['NDB_817', 'NDB_824', 'NDB_958']\n",
      "354 -> ['NDB_725', 'NDB_752', 'NDB_805']\n",
      "355 -> ['NDB_683', 'NDB_838', 'NDB_929']\n",
      "356 -> ['NDB_784', 'NDB_812', 'NDB_826']\n",
      "357 -> ['NDB_940', 'NDB_941', 'NDB_974']\n",
      "358 -> ['NDB_773', 'NDB_835', 'NDB_896']\n",
      "359 -> ['NDB_643', 'NDB_728', 'NDB_961']\n",
      "360 -> ['NDB_737', 'NDB_788', 'NDB_901']\n",
      "361 -> ['NDB_751', 'NDB_754', 'NDB_878']\n",
      "362 -> ['NDB_742', 'NDB_755', 'NDB_862']\n",
      "363 -> ['NDB_634', 'NDB_698', 'NDB_718']\n",
      "364 -> ['NDB_1060', 'NDB_880', 'NDB_935']\n",
      "365 -> ['NDB_1027', 'NDB_883', 'NDB_889']\n",
      "366 -> ['NDB_872', 'NDB_887', 'NDB_899']\n",
      "367 -> ['NDB_1075', 'NDB_652', 'NDB_716']\n",
      "368 -> ['NDB_1066', 'NDB_746', 'NDB_853']\n",
      "369 -> ['NDB_680', 'NDB_778', 'NDB_790']\n",
      "370 -> ['NDB_653', 'NDB_865', 'NDB_965']\n",
      "371 -> ['NDB_818', 'NDB_866', 'NDB_875']\n",
      "372 -> ['NDB_1028', 'NDB_665', 'NDB_710']\n",
      "373 -> ['NDB_637', 'NDB_712', 'NDB_971']\n",
      "374 -> ['NDB_694', 'NDB_743', 'NDB_925']\n",
      "375 -> ['NDB_706', 'NDB_856', 'NDB_994']\n",
      "376 -> ['NDB_647', 'NDB_760', 'NDB_905']\n",
      "377 -> ['NDB_779', 'NDB_917', 'NDB_934']\n",
      "378 -> ['NDB_790', 'NDB_863', 'NDB_946']\n",
      "379 -> ['NDB_844', 'NDB_878', 'NDB_926']\n",
      "380 -> ['NDB_784', 'NDB_899', 'NDB_910']\n",
      "381 -> ['NDB_643', 'NDB_734', 'NDB_868']\n",
      "382 -> ['NDB_715', 'NDB_773', 'NDB_941']\n",
      "383 -> ['NDB_812', 'NDB_815', 'NDB_896']\n",
      "384 -> ['NDB_788', 'NDB_805', 'NDB_881']\n",
      "385 -> ['NDB_653', 'NDB_755', 'NDB_989']\n",
      "386 -> ['NDB_710', 'NDB_721', 'NDB_752']\n",
      "387 -> ['NDB_634', 'NDB_754', 'NDB_974']\n",
      "388 -> ['NDB_1060', 'NDB_661', 'NDB_730']\n",
      "389 -> ['NDB_1066', 'NDB_728', 'NDB_862']\n",
      "390 -> ['NDB_631', 'NDB_712', 'NDB_934']\n",
      "391 -> ['NDB_1033', 'NDB_910', 'NDB_935']\n",
      "392 -> ['NDB_652', 'NDB_725', 'NDB_826']\n",
      "393 -> ['NDB_746', 'NDB_832', 'NDB_838']\n",
      "395 -> ['NDB_641', 'NDB_713', 'NDB_836']\n",
      "396 -> ['NDB_734', 'NDB_860', 'NDB_889']\n",
      "397 -> ['NDB_680', 'NDB_715', 'NDB_817']\n",
      "398 -> ['NDB_716', 'NDB_929', 'NDB_946']\n",
      "399 -> ['NDB_698', 'NDB_824', 'NDB_977']\n",
      "400 -> ['NDB_758', 'NDB_847', 'NDB_856']\n",
      "403 -> ['NDB_683', 'NDB_836', 'NDB_872']\n",
      "407 -> ['NDB_881', 'NDB_971', 'NDB_974']\n",
      "408 -> ['NDB_718', 'NDB_941', 'NDB_989']\n",
      "411 -> ['NDB_641', 'NDB_746', 'NDB_752']\n",
      "412 -> ['NDB_710', 'NDB_725', 'NDB_755']\n",
      "413 -> ['NDB_698', 'NDB_760', 'NDB_832']\n",
      "414 -> ['NDB_730', 'NDB_779', 'NDB_865']\n",
      "415 -> ['NDB_824', 'NDB_844', 'NDB_940']\n",
      "418 -> ['NDB_661', 'NDB_860', 'NDB_883']\n",
      "419 -> ['NDB_1027', 'NDB_680', 'NDB_694']\n",
      "420 -> ['NDB_737', 'NDB_815', 'NDB_818']\n",
      "421 -> ['NDB_896', 'NDB_917', 'NDB_974']\n",
      "422 -> ['NDB_652', 'NDB_868', 'NDB_977']\n",
      "423 -> ['NDB_721', 'NDB_872', 'NDB_889']\n",
      "424 -> ['NDB_836', 'NDB_905', 'NDB_961']\n",
      "425 -> ['NDB_847', 'NDB_875', 'NDB_935']\n",
      "426 -> ['NDB_751', 'NDB_812', 'NDB_856']\n",
      "427 -> ['NDB_1066', 'NDB_715', 'NDB_863']\n",
      "428 -> ['NDB_1033', 'NDB_826', 'NDB_878']\n",
      "429 -> ['NDB_1075', 'NDB_728', 'NDB_784']\n",
      "430 -> ['NDB_1060', 'NDB_805', 'NDB_926']\n",
      "431 -> ['NDB_755', 'NDB_760', 'NDB_929']\n",
      "432 -> ['NDB_637', 'NDB_958', 'NDB_994']\n",
      "433 -> ['NDB_752', 'NDB_754', 'NDB_860']\n",
      "434 -> ['NDB_1028', 'NDB_643', 'NDB_910']\n",
      "435 -> ['NDB_665', 'NDB_946', 'NDB_965']\n",
      "436 -> ['NDB_716', 'NDB_758', 'NDB_838']\n",
      "437 -> ['NDB_778', 'NDB_881', 'NDB_887']\n",
      "438 -> ['NDB_634', 'NDB_742', 'NDB_773']\n",
      "439 -> ['NDB_653', 'NDB_835', 'NDB_853']\n",
      "440 -> ['NDB_647', 'NDB_712', 'NDB_817']\n",
      "441 -> ['NDB_788', 'NDB_790', 'NDB_934']\n",
      "442 -> ['NDB_631', 'NDB_880', 'NDB_925']\n",
      "443 -> ['NDB_713', 'NDB_899', 'NDB_901']\n",
      "444 -> ['NDB_743', 'NDB_862', 'NDB_866']\n",
      "445 -> ['NDB_683', 'NDB_706', 'NDB_751']\n",
      "446 -> ['NDB_734', 'NDB_824', 'NDB_838']\n",
      "447 -> ['NDB_878', 'NDB_929', 'NDB_971']\n",
      "448 -> ['NDB_1075', 'NDB_754', 'NDB_863']\n",
      "449 -> ['NDB_661', 'NDB_746', 'NDB_779']\n",
      "450 -> ['NDB_641', 'NDB_680', 'NDB_832']\n",
      "451 -> ['NDB_875', 'NDB_917', 'NDB_958']\n",
      "452 -> ['NDB_718', 'NDB_725', 'NDB_730']\n",
      "453 -> ['NDB_1028', 'NDB_634', 'NDB_652']\n",
      "454 -> ['NDB_818', 'NDB_847', 'NDB_926']\n",
      "455 -> ['NDB_721', 'NDB_865', 'NDB_989']\n",
      "456 -> ['NDB_698', 'NDB_844', 'NDB_905']\n",
      "457 -> ['NDB_710', 'NDB_737', 'NDB_946']\n",
      "458 -> ['NDB_1027', 'NDB_941', 'NDB_977']\n",
      "461 -> ['NDB_643', 'NDB_716', 'NDB_853']\n",
      "462 -> ['NDB_710', 'NDB_815', 'NDB_994']\n",
      "463 -> ['NDB_641', 'NDB_862', 'NDB_863']\n",
      "464 -> ['NDB_755', 'NDB_934', 'NDB_941']\n",
      "465 -> ['NDB_734', 'NDB_784', 'NDB_805']\n",
      "466 -> ['NDB_1060', 'NDB_1066', 'NDB_653']\n",
      "467 -> ['NDB_680', 'NDB_899', 'NDB_929']\n",
      "469 -> ['NDB_706', 'NDB_835', 'NDB_925']\n",
      "470 -> ['NDB_712', 'NDB_878', 'NDB_905']\n",
      "471 -> ['NDB_661', 'NDB_826', 'NDB_971']\n",
      "472 -> ['NDB_631', 'NDB_965', 'NDB_989']\n",
      "473 -> ['NDB_754', 'NDB_788', 'NDB_812']\n",
      "474 -> ['NDB_1028', 'NDB_817', 'NDB_934']\n",
      "475 -> ['NDB_847', 'NDB_887', 'NDB_994']\n",
      "476 -> ['NDB_713', 'NDB_752', 'NDB_862']\n",
      "477 -> ['NDB_751', 'NDB_790', 'NDB_866']\n",
      "478 -> ['NDB_647', 'NDB_901', 'NDB_958']\n",
      "479 -> ['NDB_1033', 'NDB_773', 'NDB_880']\n",
      "480 -> ['NDB_725', 'NDB_815', 'NDB_865']\n",
      "481 -> ['NDB_1027', 'NDB_641', 'NDB_665']\n",
      "482 -> ['NDB_743', 'NDB_755', 'NDB_778']\n",
      "483 -> ['NDB_824', 'NDB_910', 'NDB_917']\n",
      "484 -> ['NDB_832', 'NDB_836', 'NDB_856']\n",
      "485 -> ['NDB_637', 'NDB_940', 'NDB_965']\n",
      "486 -> ['NDB_883', 'NDB_946', 'NDB_961']\n",
      "487 -> ['NDB_716', 'NDB_844', 'NDB_872']\n",
      "488 -> ['NDB_698', 'NDB_728', 'NDB_860']\n",
      "489 -> ['NDB_721', 'NDB_853', 'NDB_875']\n",
      "490 -> ['NDB_715', 'NDB_779', 'NDB_868']\n",
      "491 -> ['NDB_643', 'NDB_652', 'NDB_881']\n",
      "492 -> ['NDB_712', 'NDB_824', 'NDB_935']\n",
      "493 -> ['NDB_683', 'NDB_925', 'NDB_974']\n",
      "494 -> ['NDB_1075', 'NDB_718', 'NDB_742']\n",
      "495 -> ['NDB_746', 'NDB_758', 'NDB_818']\n",
      "496 -> ['NDB_730', 'NDB_760', 'NDB_838']\n",
      "497 -> ['NDB_634', 'NDB_694', 'NDB_889']\n",
      "498 -> ['NDB_896', 'NDB_926', 'NDB_977']\n",
      "499 -> ['NDB_665', 'NDB_737', 'NDB_940']\n",
      "500 -> ['NDB_863', 'NDB_929', 'NDB_935']\n",
      "501 -> ['NDB_661', 'NDB_710', 'NDB_941']\n",
      "502 -> ['NDB_637', 'NDB_680', 'NDB_905']\n",
      "503 -> ['NDB_778', 'NDB_784', 'NDB_865']\n",
      "504 -> ['NDB_1033', 'NDB_1060', 'NDB_917']\n",
      "505 -> ['NDB_653', 'NDB_856', 'NDB_883']\n",
      "506 -> ['NDB_790', 'NDB_835', 'NDB_880']\n",
      "507 -> ['NDB_647', 'NDB_715', 'NDB_734']\n",
      "508 -> ['NDB_1066', 'NDB_805', 'NDB_901']\n",
      "509 -> ['NDB_788', 'NDB_817', 'NDB_887']\n",
      "510 -> ['NDB_728', 'NDB_812', 'NDB_866']\n",
      "511 -> ['NDB_631', 'NDB_713', 'NDB_826']\n",
      "513 -> ['NDB_710', 'NDB_730', 'NDB_751']\n",
      "514 -> ['NDB_935', 'NDB_961', 'NDB_971']\n",
      "515 -> ['NDB_773', 'NDB_778', 'NDB_910']\n",
      "516 -> ['NDB_665', 'NDB_758', 'NDB_868']\n",
      "517 -> ['NDB_1075', 'NDB_836', 'NDB_853']\n",
      "518 -> ['NDB_1028', 'NDB_1060', 'NDB_929']\n",
      "519 -> ['NDB_1033', 'NDB_713', 'NDB_725']\n",
      "520 -> ['NDB_680', 'NDB_881', 'NDB_946']\n",
      "521 -> ['NDB_1027', 'NDB_862', 'NDB_926']\n",
      "522 -> ['NDB_661', 'NDB_721', 'NDB_940']\n",
      "523 -> ['NDB_863', 'NDB_889', 'NDB_974']\n",
      "524 -> ['NDB_694', 'NDB_815', 'NDB_883']\n",
      "525 -> ['NDB_742', 'NDB_743', 'NDB_994']\n",
      "526 -> ['NDB_706', 'NDB_896', 'NDB_934']\n",
      "527 -> ['NDB_652', 'NDB_844', 'NDB_875']\n",
      "528 -> ['NDB_653', 'NDB_718', 'NDB_784']\n",
      "529 -> ['NDB_737', 'NDB_746', 'NDB_899']\n",
      "530 -> ['NDB_641', 'NDB_647', 'NDB_751']\n",
      "531 -> ['NDB_665', 'NDB_715', 'NDB_910']\n",
      "532 -> ['NDB_730', 'NDB_817', 'NDB_965']\n",
      "533 -> ['NDB_734', 'NDB_752', 'NDB_958']\n",
      "534 -> ['NDB_634', 'NDB_680', 'NDB_760']\n",
      "535 -> ['NDB_838', 'NDB_905', 'NDB_977']\n",
      "536 -> ['NDB_835', 'NDB_856', 'NDB_917']\n",
      "537 -> ['NDB_631', 'NDB_637', 'NDB_779']\n",
      "538 -> ['NDB_755', 'NDB_805', 'NDB_812']\n",
      "539 -> ['NDB_824', 'NDB_865', 'NDB_880']\n",
      "540 -> ['NDB_1066', 'NDB_643', 'NDB_826']\n",
      "541 -> ['NDB_683', 'NDB_754', 'NDB_866']\n",
      "542 -> ['NDB_788', 'NDB_818', 'NDB_832']\n",
      "543 -> ['NDB_728', 'NDB_887', 'NDB_901']\n",
      "544 -> ['NDB_698', 'NDB_712', 'NDB_716']\n",
      "545 -> ['NDB_860', 'NDB_872', 'NDB_941']\n",
      "546 -> ['NDB_710', 'NDB_878', 'NDB_889']\n",
      "548 -> ['NDB_706', 'NDB_715', 'NDB_989']\n",
      "549 -> ['NDB_634', 'NDB_925', 'NDB_929']\n",
      "550 -> ['NDB_1033', 'NDB_760', 'NDB_847']\n",
      "551 -> ['NDB_725', 'NDB_790', 'NDB_868']\n",
      "552 -> ['NDB_743', 'NDB_758', 'NDB_896']\n",
      "553 -> ['NDB_694', 'NDB_778', 'NDB_935']\n",
      "554 -> ['NDB_742', 'NDB_815', 'NDB_917']\n",
      "555 -> ['NDB_773', 'NDB_844', 'NDB_961']\n",
      "556 -> ['NDB_737', 'NDB_752', 'NDB_812']\n",
      "557 -> ['NDB_1066', 'NDB_784', 'NDB_836']\n",
      "558 -> ['NDB_1075', 'NDB_865', 'NDB_971']\n",
      "559 -> ['NDB_1060', 'NDB_863', 'NDB_940']\n",
      "560 -> ['NDB_1027', 'NDB_637', 'NDB_946']\n",
      "561 -> ['NDB_721', 'NDB_838', 'NDB_862']\n",
      "563 -> ['NDB_1027', 'NDB_838', 'NDB_899']\n",
      "564 -> ['NDB_637', 'NDB_647', 'NDB_728']\n",
      "565 -> ['NDB_631', 'NDB_788', 'NDB_856']\n",
      "566 -> ['NDB_1028', 'NDB_805', 'NDB_860']\n",
      "567 -> ['NDB_706', 'NDB_812', 'NDB_965']\n",
      "568 -> ['NDB_694', 'NDB_790', 'NDB_853']\n",
      "569 -> ['NDB_1060', 'NDB_652', 'NDB_743']\n",
      "570 -> ['NDB_1075', 'NDB_832', 'NDB_977']\n",
      "571 -> ['NDB_1066', 'NDB_634', 'NDB_866']\n",
      "572 -> ['NDB_712', 'NDB_818', 'NDB_880']\n",
      "573 -> ['NDB_713', 'NDB_917', 'NDB_971']\n",
      "574 -> ['NDB_779', 'NDB_844', 'NDB_863']\n",
      "575 -> ['NDB_710', 'NDB_896', 'NDB_940']\n",
      "576 -> ['NDB_865', 'NDB_889', 'NDB_958']\n",
      "577 -> ['NDB_718', 'NDB_935', 'NDB_994']\n",
      "578 -> ['NDB_653', 'NDB_665', 'NDB_961']\n",
      "579 -> ['NDB_641', 'NDB_784', 'NDB_878']\n",
      "580 -> ['NDB_680', 'NDB_974', 'NDB_989']\n",
      "581 -> ['NDB_737', 'NDB_742', 'NDB_824']\n",
      "582 -> ['NDB_643', 'NDB_755', 'NDB_901']\n",
      "583 -> ['NDB_661', 'NDB_751', 'NDB_862']\n",
      "584 -> ['NDB_817', 'NDB_836', 'NDB_899']\n",
      "585 -> ['NDB_716', 'NDB_847', 'NDB_866']\n",
      "586 -> ['NDB_634', 'NDB_883', 'NDB_910']\n",
      "587 -> ['NDB_1060', 'NDB_746', 'NDB_875']\n",
      "588 -> ['NDB_653', 'NDB_752', 'NDB_946']\n",
      "589 -> ['NDB_773', 'NDB_868', 'NDB_934']\n",
      "590 -> ['NDB_778', 'NDB_815', 'NDB_835']\n",
      "591 -> ['NDB_905', 'NDB_926', 'NDB_929']\n",
      "592 -> ['NDB_725', 'NDB_758', 'NDB_881']\n",
      "593 -> ['NDB_698', 'NDB_734', 'NDB_754']\n",
      "594 -> ['NDB_760', 'NDB_872', 'NDB_925']\n",
      "595 -> ['NDB_730', 'NDB_826', 'NDB_887']\n",
      "596 -> ['NDB_683', 'NDB_721', 'NDB_941']\n",
      "597 -> ['NDB_1033', 'NDB_715', 'NDB_835']\n",
      "598 -> ['NDB_1066', 'NDB_965', 'NDB_977']\n",
      "599 -> ['NDB_718', 'NDB_790', 'NDB_889']\n",
      "600 -> ['NDB_728', 'NDB_751', 'NDB_844']\n",
      "601 -> ['NDB_743', 'NDB_805', 'NDB_880']\n",
      "602 -> ['NDB_1028', 'NDB_784', 'NDB_856']\n",
      "603 -> ['NDB_637', 'NDB_818', 'NDB_901']\n",
      "604 -> ['NDB_631', 'NDB_773', 'NDB_862']\n",
      "605 -> ['NDB_737', 'NDB_779', 'NDB_838']\n",
      "606 -> ['NDB_694', 'NDB_832', 'NDB_971']\n",
      "607 -> ['NDB_713', 'NDB_788', 'NDB_853']\n",
      "608 -> ['NDB_647', 'NDB_860', 'NDB_926']\n",
      "609 -> ['NDB_1075', 'NDB_812', 'NDB_958']\n",
      "610 -> ['NDB_652', 'NDB_940', 'NDB_989']\n",
      "611 -> ['NDB_1027', 'NDB_643', 'NDB_878']\n",
      "612 -> ['NDB_706', 'NDB_742', 'NDB_974']\n",
      "613 -> ['NDB_896', 'NDB_961', 'NDB_994']\n",
      "616 -> ['NDB_647', 'NDB_653', 'NDB_974']\n",
      "617 -> ['NDB_665', 'NDB_742', 'NDB_788']\n",
      "618 -> ['NDB_718', 'NDB_779', 'NDB_817']\n",
      "619 -> ['NDB_712', 'NDB_847', 'NDB_860']\n",
      "620 -> ['NDB_721', 'NDB_866', 'NDB_994']\n",
      "621 -> ['NDB_694', 'NDB_862', 'NDB_905']\n",
      "622 -> ['NDB_706', 'NDB_815', 'NDB_836']\n",
      "623 -> ['NDB_734', 'NDB_773', 'NDB_856']\n",
      "624 -> ['NDB_832', 'NDB_881', 'NDB_901']\n",
      "625 -> ['NDB_790', 'NDB_910', 'NDB_965']\n",
      "626 -> ['NDB_661', 'NDB_683', 'NDB_778']\n",
      "627 -> ['NDB_728', 'NDB_805', 'NDB_883']\n",
      "628 -> ['NDB_653', 'NDB_746', 'NDB_925']\n",
      "629 -> ['NDB_716', 'NDB_743', 'NDB_878']\n",
      "630 -> ['NDB_641', 'NDB_652', 'NDB_887']\n",
      "631 -> ['NDB_1028', 'NDB_812', 'NDB_863']\n",
      "632 -> ['NDB_1066', 'NDB_710', 'NDB_758']\n",
      "633 -> ['NDB_1060', 'NDB_838', 'NDB_941']\n",
      "634 -> ['NDB_1027', 'NDB_760', 'NDB_875']\n",
      "635 -> ['NDB_751', 'NDB_971', 'NDB_989']\n",
      "636 -> ['NDB_637', 'NDB_935', 'NDB_974']\n",
      "637 -> ['NDB_1033', 'NDB_698', 'NDB_755']\n",
      "638 -> ['NDB_779', 'NDB_790', 'NDB_899']\n",
      "639 -> ['NDB_773', 'NDB_815', 'NDB_889']\n",
      "640 -> ['NDB_880', 'NDB_926', 'NDB_934']\n",
      "641 -> ['NDB_713', 'NDB_754', 'NDB_835']\n",
      "642 -> ['NDB_706', 'NDB_805', 'NDB_868']\n",
      "643 -> ['NDB_824', 'NDB_872', 'NDB_896']\n",
      "644 -> ['NDB_1028', 'NDB_853', 'NDB_961']\n",
      "645 -> ['NDB_1075', 'NDB_725', 'NDB_940']\n",
      "646 -> ['NDB_715', 'NDB_730', 'NDB_946']\n",
      "647 -> ['NDB_826', 'NDB_844', 'NDB_917']\n",
      "648 -> ['NDB_634', 'NDB_784', 'NDB_818']\n",
      "649 -> ['NDB_643', 'NDB_958', 'NDB_977']\n",
      "650 -> ['NDB_752', 'NDB_865', 'NDB_926']\n",
      "651 -> ['NDB_631', 'NDB_875', 'NDB_899']\n",
      "652 -> ['NDB_647', 'NDB_680', 'NDB_758']\n",
      "653 -> ['NDB_718', 'NDB_883', 'NDB_929']\n",
      "654 -> ['NDB_694', 'NDB_737', 'NDB_847']\n",
      "655 -> ['NDB_868', 'NDB_889', 'NDB_941']\n",
      "656 -> ['NDB_641', 'NDB_905', 'NDB_994']\n",
      "657 -> ['NDB_716', 'NDB_788', 'NDB_925']\n",
      "658 -> ['NDB_712', 'NDB_742', 'NDB_832']\n",
      "659 -> ['NDB_665', 'NDB_901', 'NDB_989']\n",
      "660 -> ['NDB_835', 'NDB_860', 'NDB_878']\n",
      "661 -> ['NDB_778', 'NDB_824', 'NDB_965']\n",
      "662 -> ['NDB_652', 'NDB_817', 'NDB_866']\n",
      "663 -> ['NDB_683', 'NDB_698', 'NDB_856']\n",
      "665 -> ['NDB_754', 'NDB_755', 'NDB_905']\n",
      "666 -> ['NDB_634', 'NDB_725', 'NDB_734']\n",
      "667 -> ['NDB_716', 'NDB_779', 'NDB_862']\n",
      "668 -> ['NDB_746', 'NDB_946', 'NDB_989']\n",
      "669 -> ['NDB_817', 'NDB_868', 'NDB_961']\n",
      "670 -> ['NDB_728', 'NDB_773', 'NDB_872']\n",
      "672 -> ['NDB_631', 'NDB_715', 'NDB_883']\n",
      "674 -> ['NDB_637', 'NDB_698', 'NDB_934']\n",
      "675 -> ['NDB_805', 'NDB_853', 'NDB_896']\n",
      "676 -> ['NDB_653', 'NDB_875', 'NDB_941']\n",
      "677 -> ['NDB_758', 'NDB_784', 'NDB_935']\n",
      "678 -> ['NDB_712', 'NDB_718', 'NDB_863']\n",
      "679 -> ['NDB_1027', 'NDB_824', 'NDB_832']\n",
      "680 -> ['NDB_752', 'NDB_815', 'NDB_899']\n",
      "681 -> ['NDB_643', 'NDB_665', 'NDB_743']\n",
      "682 -> ['NDB_706', 'NDB_917', 'NDB_929']\n",
      "683 -> ['NDB_713', 'NDB_910', 'NDB_977']\n",
      "684 -> ['NDB_647', 'NDB_754', 'NDB_826']\n",
      "685 -> ['NDB_790', 'NDB_818', 'NDB_862']\n",
      "686 -> ['NDB_694', 'NDB_746', 'NDB_773']\n",
      "687 -> ['NDB_755', 'NDB_878', 'NDB_965']\n",
      "688 -> ['NDB_812', 'NDB_836', 'NDB_860']\n",
      "689 -> ['NDB_721', 'NDB_751', 'NDB_760']\n",
      "690 -> ['NDB_730', 'NDB_758', 'NDB_866']\n",
      "691 -> ['NDB_661', 'NDB_817', 'NDB_875']\n",
      "692 -> ['NDB_718', 'NDB_824', 'NDB_881']\n",
      "693 -> ['NDB_710', 'NDB_901', 'NDB_925']\n",
      "694 -> ['NDB_680', 'NDB_847', 'NDB_865']\n",
      "695 -> ['NDB_641', 'NDB_838', 'NDB_880']\n",
      "696 -> ['NDB_1028', 'NDB_887', 'NDB_926']\n",
      "697 -> ['NDB_652', 'NDB_742', 'NDB_835']\n",
      "698 -> ['NDB_940', 'NDB_971', 'NDB_994']\n",
      "699 -> ['NDB_1033', 'NDB_1066', 'NDB_958']\n",
      "700 -> ['NDB_1075', 'NDB_844', 'NDB_856']\n",
      "701 -> ['NDB_683', 'NDB_788', 'NDB_889']\n",
      "702 -> ['NDB_1060', 'NDB_713', 'NDB_737']\n",
      "703 -> ['NDB_631', 'NDB_694', 'NDB_974']\n",
      "706 -> ['NDB_634', 'NDB_778', 'NDB_853']\n",
      "707 -> ['NDB_725', 'NDB_866', 'NDB_946']\n",
      "708 -> ['NDB_661', 'NDB_881', 'NDB_934']\n",
      "709 -> ['NDB_665', 'NDB_730', 'NDB_734']\n",
      "710 -> ['NDB_779', 'NDB_872', 'NDB_905']\n",
      "711 -> ['NDB_868', 'NDB_899', 'NDB_917']\n",
      "713 -> ['NDB_698', 'NDB_737', 'NDB_910']\n",
      "714 -> ['NDB_1060', 'NDB_728', 'NDB_934']\n",
      "715 -> ['NDB_710', 'NDB_716', 'NDB_860']\n",
      "716 -> ['NDB_661', 'NDB_755', 'NDB_961']\n",
      "717 -> ['NDB_815', 'NDB_826', 'NDB_872']\n",
      "718 -> ['NDB_742', 'NDB_751', 'NDB_896']\n",
      "719 -> ['NDB_713', 'NDB_880', 'NDB_940']\n",
      "720 -> ['NDB_631', 'NDB_643', 'NDB_725']\n",
      "721 -> ['NDB_641', 'NDB_818', 'NDB_865']\n",
      "722 -> ['NDB_680', 'NDB_730', 'NDB_958']\n",
      "723 -> ['NDB_752', 'NDB_875', 'NDB_977']\n",
      "724 -> ['NDB_788', 'NDB_883', 'NDB_917']\n",
      "726 -> ['NDB_637', 'NDB_812', 'NDB_844']\n",
      "727 -> ['NDB_1060', 'NDB_868', 'NDB_905']\n",
      "728 -> ['NDB_718', 'NDB_721', 'NDB_737']\n",
      "729 -> ['NDB_773', 'NDB_836', 'NDB_863']\n",
      "730 -> ['NDB_817', 'NDB_929', 'NDB_941']\n",
      "731 -> ['NDB_694', 'NDB_698', 'NDB_878']\n",
      "732 -> ['NDB_715', 'NDB_856', 'NDB_866']\n",
      "733 -> ['NDB_712', 'NDB_835', 'NDB_926']\n",
      "734 -> ['NDB_755', 'NDB_853', 'NDB_887']\n",
      "735 -> ['NDB_713', 'NDB_734', 'NDB_961']\n",
      "736 -> ['NDB_746', 'NDB_860', 'NDB_880']\n",
      "737 -> ['NDB_631', 'NDB_661', 'NDB_935']\n",
      "739 -> ['NDB_683', 'NDB_779', 'NDB_824']\n",
      "740 -> ['NDB_653', 'NDB_832', 'NDB_899']\n",
      "741 -> ['NDB_760', 'NDB_805', 'NDB_946']\n",
      "742 -> ['NDB_743', 'NDB_751', 'NDB_941']\n",
      "743 -> ['NDB_784', 'NDB_901', 'NDB_994']\n",
      "744 -> ['NDB_710', 'NDB_715', 'NDB_881']\n",
      "745 -> ['NDB_643', 'NDB_754', 'NDB_836']\n",
      "746 -> ['NDB_1075', 'NDB_634', 'NDB_665']\n",
      "747 -> ['NDB_778', 'NDB_847', 'NDB_974']\n",
      "748 -> ['NDB_1033', 'NDB_790', 'NDB_989']\n",
      "749 -> ['NDB_652', 'NDB_889', 'NDB_965']\n",
      "750 -> ['NDB_1066', 'NDB_647', 'NDB_706']\n",
      "751 -> ['NDB_758', 'NDB_862', 'NDB_925']\n",
      "752 -> ['NDB_1028', 'NDB_838', 'NDB_971']\n",
      "753 -> ['NDB_1027', 'NDB_818', 'NDB_910']\n",
      "754 -> ['NDB_728', 'NDB_743', 'NDB_826']\n",
      "755 -> ['NDB_734', 'NDB_935', 'NDB_977']\n",
      "756 -> ['NDB_746', 'NDB_881', 'NDB_896']\n",
      "757 -> ['NDB_887', 'NDB_934', 'NDB_958']\n",
      "758 -> ['NDB_754', 'NDB_853', 'NDB_872']\n",
      "759 -> ['NDB_637', 'NDB_716', 'NDB_883']\n",
      "765 -> ['NDB_1066', 'NDB_788', 'NDB_974']\n",
      "766 -> ['NDB_652', 'NDB_773', 'NDB_818']\n",
      "767 -> ['NDB_752', 'NDB_779', 'NDB_878']\n",
      "768 -> ['NDB_1075', 'NDB_653', 'NDB_935']\n",
      "769 -> ['NDB_706', 'NDB_712', 'NDB_721']\n",
      "770 -> ['NDB_1027', 'NDB_647', 'NDB_929']\n",
      "772 -> ['NDB_863', 'NDB_883', 'NDB_941']\n",
      "773 -> ['NDB_683', 'NDB_860', 'NDB_875']\n",
      "774 -> ['NDB_634', 'NDB_743', 'NDB_824']\n",
      "775 -> ['NDB_698', 'NDB_862', 'NDB_880']\n",
      "776 -> ['NDB_746', 'NDB_784', 'NDB_961']\n",
      "779 -> ['NDB_1028', 'NDB_835', 'NDB_989']\n",
      "780 -> ['NDB_826', 'NDB_940', 'NDB_958']\n",
      "781 -> ['NDB_778', 'NDB_856', 'NDB_901']\n",
      "783 -> ['NDB_1033', 'NDB_680', 'NDB_815']\n",
      "784 -> ['NDB_790', 'NDB_925', 'NDB_994']\n",
      "785 -> ['NDB_1066', 'NDB_760', 'NDB_896']\n",
      "786 -> ['NDB_788', 'NDB_838', 'NDB_965']\n",
      "787 -> ['NDB_1060', 'NDB_665', 'NDB_817']\n",
      "788 -> ['NDB_653', 'NDB_812', 'NDB_929']\n",
      "789 -> ['NDB_1075', 'NDB_706', 'NDB_910']\n",
      "790 -> ['NDB_637', 'NDB_866', 'NDB_868']\n",
      "791 -> ['NDB_730', 'NDB_754', 'NDB_917']\n",
      "792 -> ['NDB_710', 'NDB_734', 'NDB_836']\n",
      "793 -> ['NDB_725', 'NDB_872', 'NDB_934']\n",
      "794 -> ['NDB_815', 'NDB_832', 'NDB_878']\n",
      "795 -> ['NDB_652', 'NDB_926', 'NDB_958']\n",
      "796 -> ['NDB_715', 'NDB_758', 'NDB_853']\n",
      "797 -> ['NDB_694', 'NDB_751', 'NDB_881']\n",
      "798 -> ['NDB_718', 'NDB_905', 'NDB_946']\n",
      "799 -> ['NDB_755', 'NDB_889', 'NDB_899']\n",
      "800 -> ['NDB_641', 'NDB_737', 'NDB_971']\n",
      "801 -> ['NDB_805', 'NDB_844', 'NDB_865']\n",
      "802 -> ['NDB_643', 'NDB_742', 'NDB_887']\n",
      "803 -> ['NDB_631', 'NDB_716', 'NDB_901']\n",
      "804 -> ['NDB_661', 'NDB_713', 'NDB_925']\n",
      "805 -> ['NDB_812', 'NDB_880', 'NDB_977']\n",
      "806 -> ['NDB_683', 'NDB_728', 'NDB_832']\n",
      "807 -> ['NDB_847', 'NDB_863', 'NDB_910']\n",
      "808 -> ['NDB_743', 'NDB_926', 'NDB_974']\n",
      "809 -> ['NDB_712', 'NDB_752', 'NDB_989']\n",
      "810 -> ['NDB_647', 'NDB_818', 'NDB_961']\n",
      "812 -> ['NDB_1033', 'NDB_742', 'NDB_805']\n",
      "813 -> ['NDB_1027', 'NDB_1066', 'NDB_844']\n",
      "814 -> ['NDB_824', 'NDB_856', 'NDB_860']\n",
      "815 -> ['NDB_716', 'NDB_826', 'NDB_881']\n",
      "816 -> ['NDB_634', 'NDB_641', 'NDB_817']\n",
      "817 -> ['NDB_715', 'NDB_862', 'NDB_878']\n",
      "818 -> ['NDB_779', 'NDB_880', 'NDB_961']\n",
      "819 -> ['NDB_754', 'NDB_832', 'NDB_989']\n",
      "820 -> ['NDB_778', 'NDB_875', 'NDB_934']\n",
      "821 -> ['NDB_784', 'NDB_965', 'NDB_971']\n",
      "822 -> ['NDB_728', 'NDB_760', 'NDB_935']\n",
      "823 -> ['NDB_790', 'NDB_887', 'NDB_977']\n",
      "824 -> ['NDB_643', 'NDB_889', 'NDB_994']\n",
      "825 -> ['NDB_730', 'NDB_737', 'NDB_863']\n",
      "827 -> ['NDB_653', 'NDB_866', 'NDB_905']\n",
      "828 -> ['NDB_694', 'NDB_713', 'NDB_946']\n",
      "829 -> ['NDB_1060', 'NDB_647', 'NDB_710']\n",
      "830 -> ['NDB_812', 'NDB_899', 'NDB_926']\n",
      "831 -> ['NDB_1075', 'NDB_751', 'NDB_901']\n",
      "832 -> ['NDB_631', 'NDB_752', 'NDB_929']\n",
      "833 -> ['NDB_680', 'NDB_862', 'NDB_940']\n",
      "834 -> ['NDB_755', 'NDB_815', 'NDB_925']\n",
      "835 -> ['NDB_1028', 'NDB_1033', 'NDB_965']\n",
      "836 -> ['NDB_778', 'NDB_818', 'NDB_844']\n",
      "837 -> ['NDB_718', 'NDB_836', 'NDB_868']\n",
      "838 -> ['NDB_758', 'NDB_788', 'NDB_958']\n",
      "839 -> ['NDB_721', 'NDB_742', 'NDB_790']\n",
      "840 -> ['NDB_652', 'NDB_706', 'NDB_853']\n",
      "841 -> ['NDB_637', 'NDB_661', 'NDB_773']\n",
      "842 -> ['NDB_746', 'NDB_910', 'NDB_974']\n",
      "843 -> ['NDB_698', 'NDB_865', 'NDB_917']\n",
      "844 -> ['NDB_838', 'NDB_883', 'NDB_896']\n",
      "845 -> ['NDB_725', 'NDB_835', 'NDB_847']\n",
      "846 -> ['NDB_712', 'NDB_734', 'NDB_760']\n",
      "847 -> ['NDB_665', 'NDB_880', 'NDB_941']\n",
      "848 -> ['NDB_743', 'NDB_856', 'NDB_940']\n",
      "849 -> ['NDB_1028', 'NDB_641', 'NDB_683']\n",
      "850 -> ['NDB_818', 'NDB_872', 'NDB_989']\n",
      "851 -> ['NDB_680', 'NDB_721', 'NDB_826']\n",
      "852 -> ['NDB_1027', 'NDB_805', 'NDB_994']\n",
      "854 -> ['NDB_728', 'NDB_847', 'NDB_925']\n",
      "855 -> ['NDB_631', 'NDB_844', 'NDB_887']\n",
      "856 -> ['NDB_683', 'NDB_790', 'NDB_905']\n",
      "857 -> ['NDB_752', 'NDB_872', 'NDB_881']\n",
      "858 -> ['NDB_737', 'NDB_832', 'NDB_917']\n",
      "859 -> ['NDB_665', 'NDB_716', 'NDB_812']\n",
      "860 -> ['NDB_746', 'NDB_754', 'NDB_994']\n",
      "861 -> ['NDB_647', 'NDB_875', 'NDB_883']\n",
      "862 -> ['NDB_817', 'NDB_865', 'NDB_940']\n",
      "863 -> ['NDB_755', 'NDB_773', 'NDB_977']\n",
      "864 -> ['NDB_725', 'NDB_760', 'NDB_971']\n",
      "865 -> ['NDB_637', 'NDB_784', 'NDB_835']\n",
      "866 -> ['NDB_643', 'NDB_866', 'NDB_896']\n",
      "867 -> ['NDB_1075', 'NDB_779', 'NDB_989']\n",
      "868 -> ['NDB_734', 'NDB_826', 'NDB_862']\n",
      "869 -> ['NDB_710', 'NDB_818', 'NDB_965']\n",
      "870 -> ['NDB_694', 'NDB_730', 'NDB_824']\n",
      "871 -> ['NDB_742', 'NDB_836', 'NDB_878']\n",
      "872 -> ['NDB_1066', 'NDB_838', 'NDB_917']\n",
      "873 -> ['NDB_680', 'NDB_698', 'NDB_743']\n",
      "874 -> ['NDB_856', 'NDB_941', 'NDB_961']\n",
      "875 -> ['NDB_634', 'NDB_661', 'NDB_706']\n",
      "876 -> ['NDB_641', 'NDB_910', 'NDB_929']\n",
      "877 -> ['NDB_721', 'NDB_805', 'NDB_815']\n",
      "878 -> ['NDB_652', 'NDB_758', 'NDB_934']\n",
      "879 -> ['NDB_1033', 'NDB_653', 'NDB_712']\n",
      "880 -> ['NDB_1027', 'NDB_778', 'NDB_868']\n",
      "881 -> ['NDB_863', 'NDB_901', 'NDB_926']\n",
      "882 -> ['NDB_751', 'NDB_788', 'NDB_899']\n",
      "883 -> ['NDB_1060', 'NDB_860', 'NDB_946']\n",
      "884 -> ['NDB_1028', 'NDB_718', 'NDB_974']\n",
      "885 -> ['NDB_853', 'NDB_889', 'NDB_935']\n",
      "886 -> ['NDB_880', 'NDB_883', 'NDB_958']\n",
      "887 -> ['NDB_713', 'NDB_838', 'NDB_887']\n",
      "888 -> ['NDB_715', 'NDB_754', 'NDB_971']\n",
      "889 -> ['NDB_728', 'NDB_790', 'NDB_865']\n",
      "890 -> ['NDB_643', 'NDB_812', 'NDB_847']\n",
      "891 -> ['NDB_784', 'NDB_881', 'NDB_977']\n",
      "894 -> ['NDB_754', 'NDB_941', 'NDB_958']\n",
      "896 -> ['NDB_721', 'NDB_836', 'NDB_883']\n",
      "897 -> ['NDB_634', 'NDB_737', 'NDB_881']\n",
      "898 -> ['NDB_844', 'NDB_901', 'NDB_971']\n",
      "899 -> ['NDB_1075', 'NDB_713', 'NDB_805']\n",
      "900 -> ['NDB_716', 'NDB_742', 'NDB_977']\n",
      "901 -> ['NDB_637', 'NDB_652', 'NDB_863']\n",
      "902 -> ['NDB_653', 'NDB_889', 'NDB_926']\n",
      "903 -> ['NDB_826', 'NDB_835', 'NDB_935']\n",
      "904 -> ['NDB_647', 'NDB_790', 'NDB_838']\n",
      "905 -> ['NDB_710', 'NDB_847', 'NDB_880']\n",
      "906 -> ['NDB_641', 'NDB_718', 'NDB_899']\n",
      "907 -> ['NDB_725', 'NDB_788', 'NDB_824']\n",
      "908 -> ['NDB_665', 'NDB_832', 'NDB_905']\n",
      "909 -> ['NDB_760', 'NDB_910', 'NDB_961']\n",
      "910 -> ['NDB_773', 'NDB_779', 'NDB_875']\n",
      "911 -> ['NDB_1060', 'NDB_694', 'NDB_965']\n",
      "912 -> ['NDB_1027', 'NDB_715', 'NDB_755']\n",
      "913 -> ['NDB_712', 'NDB_758', 'NDB_865']\n",
      "914 -> ['NDB_815', 'NDB_860', 'NDB_934']\n",
      "915 -> ['NDB_661', 'NDB_853', 'NDB_868']\n",
      "916 -> ['NDB_706', 'NDB_878', 'NDB_946']\n",
      "917 -> ['NDB_1028', 'NDB_872', 'NDB_940']\n",
      "918 -> ['NDB_631', 'NDB_728', 'NDB_778']\n",
      "919 -> ['NDB_784', 'NDB_929', 'NDB_974']\n",
      "920 -> ['NDB_680', 'NDB_746', 'NDB_751']\n",
      "921 -> ['NDB_698', 'NDB_752', 'NDB_887']\n",
      "922 -> ['NDB_1066', 'NDB_683', 'NDB_743']\n",
      "923 -> ['NDB_1033', 'NDB_817', 'NDB_896']\n",
      "924 -> ['NDB_643', 'NDB_730', 'NDB_805']\n",
      "925 -> ['NDB_652', 'NDB_734', 'NDB_994']\n",
      "926 -> ['NDB_856', 'NDB_925', 'NDB_977']\n",
      "927 -> ['NDB_721', 'NDB_818', 'NDB_917']\n",
      "928 -> ['NDB_812', 'NDB_862', 'NDB_971']\n",
      "929 -> ['NDB_737', 'NDB_866', 'NDB_889']\n",
      "930 -> ['NDB_935', 'NDB_958', 'NDB_989']\n",
      "932 -> ['NDB_1066', 'NDB_665', 'NDB_899']\n",
      "933 -> ['NDB_746', 'NDB_887', 'NDB_905']\n",
      "934 -> ['NDB_1033', 'NDB_784', 'NDB_863']\n",
      "935 -> ['NDB_926', 'NDB_935', 'NDB_940']\n",
      "936 -> ['NDB_1028', 'NDB_868', 'NDB_901']\n",
      "937 -> ['NDB_634', 'NDB_710', 'NDB_728']\n",
      "938 -> ['NDB_713', 'NDB_751', 'NDB_832']\n",
      "939 -> ['NDB_715', 'NDB_818', 'NDB_958']\n",
      "940 -> ['NDB_773', 'NDB_812', 'NDB_989']\n",
      "941 -> ['NDB_721', 'NDB_778', 'NDB_779']\n",
      "942 -> ['NDB_631', 'NDB_737', 'NDB_817']\n",
      "943 -> ['NDB_712', 'NDB_805', 'NDB_889']\n",
      "944 -> ['NDB_752', 'NDB_760', 'NDB_883']\n",
      "945 -> ['NDB_647', 'NDB_716', 'NDB_856']\n",
      "946 -> ['NDB_680', 'NDB_961', 'NDB_965']\n",
      "947 -> ['NDB_637', 'NDB_824', 'NDB_941']\n",
      "948 -> ['NDB_725', 'NDB_742', 'NDB_838']\n",
      "949 -> ['NDB_641', 'NDB_844', 'NDB_853']\n",
      "950 -> ['NDB_1060', 'NDB_815', 'NDB_974']\n",
      "951 -> ['NDB_743', 'NDB_836', 'NDB_929']\n",
      "952 -> ['NDB_1027', 'NDB_826', 'NDB_865']\n",
      "953 -> ['NDB_653', 'NDB_698', 'NDB_730']\n",
      "954 -> ['NDB_1075', 'NDB_734', 'NDB_875']\n",
      "955 -> ['NDB_847', 'NDB_872', 'NDB_946']\n",
      "956 -> ['NDB_718', 'NDB_788', 'NDB_866']\n",
      "957 -> ['NDB_860', 'NDB_910', 'NDB_994']\n",
      "958 -> ['NDB_643', 'NDB_835', 'NDB_934']\n",
      "959 -> ['NDB_683', 'NDB_878', 'NDB_896']\n",
      "960 -> ['NDB_661', 'NDB_694', 'NDB_784']\n",
      "961 -> ['NDB_755', 'NDB_758', 'NDB_940']\n",
      "962 -> ['NDB_706', 'NDB_880', 'NDB_887']\n",
      "963 -> ['NDB_881', 'NDB_899', 'NDB_925']\n",
      "964 -> ['NDB_634', 'NDB_862', 'NDB_917']\n",
      "965 -> ['NDB_652', 'NDB_790', 'NDB_832']\n",
      "966 -> ['NDB_754', 'NDB_901', 'NDB_977']\n",
      "969 -> ['NDB_721', 'NDB_929', 'NDB_961']\n",
      "970 -> ['NDB_730', 'NDB_832', 'NDB_941']\n",
      "971 -> ['NDB_647', 'NDB_713', 'NDB_865']\n",
      "972 -> ['NDB_1033', 'NDB_661', 'NDB_946']\n",
      "973 -> ['NDB_653', 'NDB_742', 'NDB_847']\n",
      "974 -> ['NDB_863', 'NDB_934', 'NDB_965']\n",
      "975 -> ['NDB_665', 'NDB_755', 'NDB_971']\n",
      "976 -> ['NDB_643', 'NDB_760', 'NDB_773']\n",
      "977 -> ['NDB_680', 'NDB_683', 'NDB_734']\n",
      "978 -> ['NDB_1066', 'NDB_641', 'NDB_889']\n",
      "979 -> ['NDB_718', 'NDB_778', 'NDB_925']\n",
      "980 -> ['NDB_1027', 'NDB_788', 'NDB_905']\n",
      "982 -> ['NDB_716', 'NDB_751', 'NDB_836']\n",
      "983 -> ['NDB_752', 'NDB_824', 'NDB_994']\n",
      "984 -> ['NDB_712', 'NDB_728', 'NDB_779']\n",
      "985 -> ['NDB_743', 'NDB_815', 'NDB_875']\n",
      "986 -> ['NDB_1060', 'NDB_1075', 'NDB_715']\n",
      "987 -> ['NDB_1028', 'NDB_698', 'NDB_935']\n",
      "988 -> ['NDB_637', 'NDB_725', 'NDB_860']\n",
      "989 -> ['NDB_838', 'NDB_856', 'NDB_872']\n",
      "990 -> ['NDB_866', 'NDB_878', 'NDB_974']\n",
      "991 -> ['NDB_631', 'NDB_868', 'NDB_896']\n",
      "992 -> ['NDB_694', 'NDB_835', 'NDB_910']\n",
      "995 -> ['NDB_754', 'NDB_881', 'NDB_910']\n",
      "997 -> ['NDB_643', 'NDB_817', 'NDB_946']\n",
      "998 -> ['NDB_742', 'NDB_784', 'NDB_926']\n",
      "999 -> ['NDB_683', 'NDB_773', 'NDB_826']\n",
      "1000 -> ['NDB_653', 'NDB_734', 'NDB_779']\n",
      "1001 -> ['NDB_698', 'NDB_788', 'NDB_971']\n",
      "1002 -> ['NDB_866', 'NDB_883', 'NDB_901']\n",
      "1007 -> ['NDB_665', 'NDB_853', 'NDB_865']\n",
      "1008 -> ['NDB_1027', 'NDB_880', 'NDB_917']\n",
      "1009 -> ['NDB_1075', 'NDB_694', 'NDB_941']\n",
      "1010 -> ['NDB_647', 'NDB_778', 'NDB_878']\n",
      "1011 -> ['NDB_812', 'NDB_818', 'NDB_994']\n",
      "1013 -> ['NDB_718', 'NDB_751', 'NDB_940']\n",
      "1014 -> ['NDB_1028', 'NDB_637', 'NDB_706']\n",
      "1015 -> ['NDB_713', 'NDB_863', 'NDB_905']\n",
      "1016 -> ['NDB_715', 'NDB_832', 'NDB_965']\n",
      "1017 -> ['NDB_641', 'NDB_730', 'NDB_896']\n",
      "1018 -> ['NDB_1060', 'NDB_716', 'NDB_887']\n",
      "1019 -> ['NDB_824', 'NDB_836', 'NDB_925']\n",
      "1020 -> ['NDB_652', 'NDB_721', 'NDB_746']\n",
      "1021 -> ['NDB_712', 'NDB_838', 'NDB_875']\n",
      "1022 -> ['NDB_680', 'NDB_868', 'NDB_872']\n",
      "1023 -> ['NDB_752', 'NDB_817', 'NDB_835']\n",
      "1024 -> ['NDB_1033', 'NDB_728', 'NDB_881']\n",
      "1025 -> ['NDB_751', 'NDB_917', 'NDB_961']\n",
      "1026 -> ['NDB_760', 'NDB_818', 'NDB_853']\n",
      "1027 -> ['NDB_725', 'NDB_754', 'NDB_856']\n",
      "1028 -> ['NDB_737', 'NDB_743', 'NDB_844']\n",
      "1029 -> ['NDB_860', 'NDB_901', 'NDB_935']\n",
      "1030 -> ['NDB_758', 'NDB_889', 'NDB_989']\n",
      "1031 -> ['NDB_634', 'NDB_805', 'NDB_899']\n",
      "1032 -> ['NDB_631', 'NDB_946', 'NDB_958']\n",
      "1033 -> ['NDB_815', 'NDB_847', 'NDB_862']\n",
      "1034 -> ['NDB_1066', 'NDB_755', 'NDB_910']\n",
      "1035 -> ['NDB_661', 'NDB_974', 'NDB_977']\n",
      "1036 -> ['NDB_710', 'NDB_790', 'NDB_929']\n",
      "1038 -> ['NDB_643', 'NDB_926', 'NDB_965']\n",
      "1039 -> ['NDB_742', 'NDB_863', 'NDB_961']\n",
      "1040 -> ['NDB_784', 'NDB_896', 'NDB_905']\n",
      "1041 -> ['NDB_760', 'NDB_866', 'NDB_934']\n",
      "1044 -> ['NDB_706', 'NDB_773', 'NDB_940']\n",
      "1045 -> ['NDB_1027', 'NDB_743', 'NDB_971']\n",
      "1046 -> ['NDB_1075', 'NDB_637', 'NDB_778']\n",
      "1047 -> ['NDB_788', 'NDB_878', 'NDB_941']\n",
      "1048 -> ['NDB_1028', 'NDB_653', 'NDB_721']\n",
      "1049 -> ['NDB_1033', 'NDB_665', 'NDB_718']\n",
      "1050 -> ['NDB_715', 'NDB_865', 'NDB_994']\n",
      "1051 -> ['NDB_694', 'NDB_712', 'NDB_812']\n",
      "1053 -> ['NDB_647', 'NDB_779', 'NDB_925']\n",
      "1054 -> ['NDB_734', 'NDB_737', 'NDB_880']\n",
      "1055 -> ['NDB_698', 'NDB_746', 'NDB_847']\n",
      "1056 -> ['NDB_683', 'NDB_758', 'NDB_887']\n",
      "1057 -> ['NDB_826', 'NDB_836', 'NDB_934']\n",
      "1058 -> ['NDB_730', 'NDB_872', 'NDB_875']\n",
      "1059 -> ['NDB_725', 'NDB_883', 'NDB_899']\n",
      "1060 -> ['NDB_716', 'NDB_752', 'NDB_868']\n",
      "1062 -> ['NDB_698', 'NDB_706', 'NDB_710']\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "reload(check)\n",
    "check_readout_dict = pickle.load(open(os.path.join(library_folder, 'readout_usage.pkl'), 'rb'))\n",
    "\n",
    "reg_readout_dic, reg2readout_check = check._check_region_to_readouts(full_records, check_readout_dict, \n",
    "                                                                     total_readout_num=num_readout_per_probe)\n",
    "print(reg2readout_check)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e64afd0",
   "metadata": {},
   "source": [
    "### 4.1.4 check readout usage in the library"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "dbfae27e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- extracting probes from >region:1, region:2, region:3, region:4, region:5, region:6, region:7, region:8, region:9, region:10, region:11, region:12, region:13, region:14, region:15, region:16, region:17, region:18, region:19, region:20, region:21, region:22, region:23, region:24, region:25, region:26, region:27, region:28, region:29, region:30, region:31, region:32, region:33, region:34, region:35, region:36, region:37, region:38, region:39, region:40, region:41, region:42, region:43, region:44, region:45, region:46, region:47, region:48, region:49, region:50, region:51, region:52, region:53, region:54, region:55, region:56, region:57, region:58, region:59, region:60, region:61, region:62, region:63, region:64, region:65, region:66, region:67, region:68, region:69, region:70, region:71, region:72, region:73, region:74, region:75, region:76, region:77, region:80, region:81, region:82, region:83, region:84, region:85, region:86, region:87, region:88, region:89, region:90, region:91, region:92, region:93, region:94, region:95, region:96, region:97, region:98, region:99, region:100, region:101, region:102, region:103, region:104, region:105, region:106, region:107, region:108, region:109, region:110, region:111, region:112, region:113, region:114, region:115, region:116, region:117, region:118, region:119, region:120, region:121, region:122, region:123, region:124, region:125, region:126, region:127, region:128, region:129, region:130, region:131, region:132, region:133, region:134, region:135, region:136, region:137, region:138, region:140, region:141, region:142, region:143, region:144, region:145, region:146, region:147, region:148, region:150, region:151, region:153, region:154, region:155, region:156, region:157, region:158, region:159, region:160, region:161, region:162, region:163, region:164, region:165, region:166, region:167, region:168, region:169, region:170, region:171, region:172, region:173, region:174, region:175, region:176, region:177, region:178, region:179, region:180, region:181, region:182, region:183, region:184, region:185, region:186, region:187, region:188, region:190, region:191, region:192, region:193, region:195, region:196, region:197, region:198, region:199, region:200, region:201, region:202, region:203, region:204, region:205, region:206, region:207, region:208, region:209, region:210, region:211, region:212, region:213, region:214, region:215, region:218, region:219, region:220, region:221, region:222, region:223, region:224, region:225, region:226, region:227, region:228, region:229, region:230, region:231, region:232, region:233, region:234, region:235, region:236, region:237, region:238, region:239, region:240, region:242, region:243, region:244, region:245, region:246, region:247, region:248, region:249, region:250, region:251, region:252, region:253, region:254, region:255, region:256, region:257, region:258, region:259, region:260, region:261, region:262, region:263, region:264, region:266, region:267, region:268, region:269, region:270, region:271, region:272, region:273, region:274, region:276, region:277, region:278, region:279, region:281, region:282, region:283, region:285, region:286, region:287, region:288, region:289, region:290, region:291, region:292, region:293, region:294, region:295, region:296, region:297, region:298, region:299, region:300, region:301, region:302, region:303, region:304, region:305, region:306, region:307, region:308, region:309, region:310, region:311, region:312, region:313, region:314, region:315, region:316, region:318, region:319, region:320, region:321, region:322, region:323, region:324, region:325, region:326, region:327, region:328, region:329, region:330, region:331, region:332, region:333, region:334, region:335, region:336, region:337, region:338, region:339, region:340, region:342, region:343, region:344, region:345, region:346, region:347, region:348, region:349, region:350, region:351, region:352, region:353, region:354, region:355, region:356, region:357, region:358, region:359, region:360, region:361, region:362, region:363, region:364, region:365, region:366, region:367, region:368, region:369, region:370, region:371, region:372, region:373, region:374, region:375, region:376, region:377, region:378, region:379, region:380, region:381, region:382, region:383, region:384, region:385, region:386, region:387, region:388, region:389, region:390, region:391, region:392, region:393, region:395, region:396, region:397, region:398, region:399, region:400, region:403, region:407, region:408, region:411, region:412, region:413, region:414, region:415, region:418, region:419, region:420, region:421, region:422, region:423, region:424, region:425, region:426, region:427, region:428, region:429, region:430, region:431, region:432, region:433, region:434, region:435, region:436, region:437, region:438, region:439, region:440, region:441, region:442, region:443, region:444, region:445, region:446, region:447, region:448, region:449, region:450, region:451, region:452, region:453, region:454, region:455, region:456, region:457, region:458, region:461, region:462, region:463, region:464, region:465, region:466, region:467, region:469, region:470, region:471, region:472, region:473, region:474, region:475, region:476, region:477, region:478, region:479, region:480, region:481, region:482, region:483, region:484, region:485, region:486, region:487, region:488, region:489, region:490, region:491, region:492, region:493, region:494, region:495, region:496, region:497, region:498, region:499, region:500, region:501, region:502, region:503, region:504, region:505, region:506, region:507, region:508, region:509, region:510, region:511, region:513, region:514, region:515, region:516, region:517, region:518, region:519, region:520, region:521, region:522, region:523, region:524, region:525, region:526, region:527, region:528, region:529, region:530, region:531, region:532, region:533, region:534, region:535, region:536, region:537, region:538, region:539, region:540, region:541, region:542, region:543, region:544, region:545, region:546, region:548, region:549, region:550, region:551, region:552, region:553, region:554, region:555, region:556, region:557, region:558, region:559, region:560, region:561, region:563, region:564, region:565, region:566, region:567, region:568, region:569, region:570, region:571, region:572, region:573, region:574, region:575, region:576, region:577, region:578, region:579, region:580, region:581, region:582, region:583, region:584, region:585, region:586, region:587, region:588, region:589, region:590, region:591, region:592, region:593, region:594, region:595, region:596, region:597, region:598, region:599, region:600, region:601, region:602, region:603, region:604, region:605, region:606, region:607, region:608, region:609, region:610, region:611, region:612, region:613, region:616, region:617, region:618, region:619, region:620, region:621, region:622, region:623, region:624, region:625, region:626, region:627, region:628, region:629, region:630, region:631, region:632, region:633, region:634, region:635, region:636, region:637, region:638, region:639, region:640, region:641, region:642, region:643, region:644, region:645, region:646, region:647, region:648, region:649, region:650, region:651, region:652, region:653, region:654, region:655, region:656, region:657, region:658, region:659, region:660, region:661, region:662, region:663, region:665, region:666, region:667, region:668, region:669, region:670, region:672, region:674, region:675, region:676, region:677, region:678, region:679, region:680, region:681, region:682, region:683, region:684, region:685, region:686, region:687, region:688, region:689, region:690, region:691, region:692, region:693, region:694, region:695, region:696, region:697, region:698, region:699, region:700, region:701, region:702, region:703, region:706, region:707, region:708, region:709, region:710, region:711, region:713, region:714, region:715, region:716, region:717, region:718, region:719, region:720, region:721, region:722, region:723, region:724, region:726, region:727, region:728, region:729, region:730, region:731, region:732"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ", region:733, region:734, region:735, region:736, region:737, region:739, region:740, region:741, region:742, region:743, region:744, region:745, region:746, region:747, region:748, region:749, region:750, region:751, region:752, region:753, region:754, region:755, region:756, region:757, region:758, region:759, region:765, region:766, region:767, region:768, region:769, region:770, region:772, region:773, region:774, region:775, region:776, region:779, region:780, region:781, region:783, region:784, region:785, region:786, region:787, region:788, region:789, region:790, region:791, region:792, region:793, region:794, region:795, region:796, region:797, region:798, region:799, region:800, region:801, region:802, region:803, region:804, region:805, region:806, region:807, region:808, region:809, region:810, region:812, region:813, region:814, region:815, region:816, region:817, region:818, region:819, region:820, region:821, region:822, region:823, region:824, region:825, region:827, region:828, region:829, region:830, region:831, region:832, region:833, region:834, region:835, region:836, region:837, region:838, region:839, region:840, region:841, region:842, region:843, region:844, region:845, region:846, region:847, region:848, region:849, region:850, region:851, region:852, region:854, region:855, region:856, region:857, region:858, region:859, region:860, region:861, region:862, region:863, region:864, region:865, region:866, region:867, region:868, region:869, region:870, region:871, region:872, region:873, region:874, region:875, region:876, region:877, region:878, region:879, region:880, region:881, region:882, region:883, region:884, region:885, region:886, region:887, region:888, region:889, region:890, region:891, region:894, region:896, region:897, region:898, region:899, region:900, region:901, region:902, region:903, region:904, region:905, region:906, region:907, region:908, region:909, region:910, region:911, region:912, region:913, region:914, region:915, region:916, region:917, region:918, region:919, region:920, region:921, region:922, region:923, region:924, region:925, region:926, region:927, region:928, region:929, region:930, region:932, region:933, region:934, region:935, region:936, region:937, region:938, region:939, region:940, region:941, region:942, region:943, region:944, region:945, region:946, region:947, region:948, region:949, region:950, region:951, region:952, region:953, region:954, region:955, region:956, region:957, region:958, region:959, region:960, region:961, region:962, region:963, region:964, region:965, region:966, region:969, region:970, region:971, region:972, region:973, region:974, region:975, region:976, region:977, region:978, region:979, region:980, region:982, region:983, region:984, region:985, region:986, region:987, region:988, region:989, region:990, region:991, region:992, region:995, region:997, region:998, region:999, region:1000, region:1001, region:1002, region:1007, region:1008, region:1009, region:1010, region:1011, region:1013, region:1014, region:1015, region:1016, region:1017, region:1018, region:1019, region:1020, region:1021, region:1022, region:1023, region:1024, region:1025, region:1026, region:1027, region:1028, region:1029, region:1030, region:1031, region:1032, region:1033, region:1034, region:1035, region:1036, region:1038, region:1039, region:1040, region:1041, region:1044, region:1045, region:1046, region:1047, region:1048, region:1049, region:1050, region:1051, region:1053, region:1054, region:1055, region:1056, region:1057, region:1058, region:1059, region:1060, region:1062,  > Done.\n",
      "-- inverting region_to_readout dict\n",
      "NDB_1027 -> ['1008', '104', '1045', '121', '174', '19', '203', '232', '272', '290', '306', '365', '419', '458', '481', '521', '560', '563', '57', '611', '634', '679', '753', '770', '813', '852', '880', '912', '952', '980']\n",
      "NDB_1028 -> ['1014', '1048', '113', '138', '157', '171', '21', '248', '274', '304', '320', '372', '434', '453', '474', '50', '518', '566', '602', '631', '644', '696', '752', '779', '835', '849', '884', '917', '936', '987']\n",
      "NDB_1033 -> ['1024', '1049', '115', '167', '215', '251', '31', '319', '336', '348', '391', '428', '479', '504', '519', '550', '597', '637', '65', '699', '748', '783', '8', '812', '835', '879', '90', '923', '934', '972']\n",
      "NDB_1060 -> ['1018', '151', '159', '193', '226', '244', '301', '315', '364', '388', '430', '45', '466', '504', '518', '52', '559', '569', '587', '633', '702', '714', '727', '787', '829', '883', '911', '950', '98', '986']\n",
      "NDB_1066 -> ['1034', '108', '141', '156', '196', '248', '282', '3', '307', '368', '389', '42', '427', '466', '508', '540', '557', '571', '598', '632', '699', '75', '750', '765', '785', '813', '872', '922', '932', '978']\n",
      "NDB_1075 -> ['1009', '1046', '114', '128', '153', '187', '245', '298', '30', '336', '367', '429', '44', '448', '494', '517', '558', '57', '570', '609', '645', '700', '746', '768', '789', '831', '867', '899', '954', '986']\n",
      "NDB_631 -> ['1032', '120', '137', '179', '206', '245', '266', '296', '337', '350', '390', '442', '472', '50', '511', '537', '565', '604', '651', '672', '703', '720', '73', '737', '803', '832', '855', '918', '942', '991']\n",
      "NDB_634 -> ['102', '1031', '127', '179', '207', '225', '262', '286', '306', '363', '37', '387', '438', '453', '497', '534', '549', '571', '586', '648', '666', '68', '706', '746', '774', '816', '875', '897', '937', '964']\n",
      "NDB_637 -> ['1014', '1046', '112', '163', '215', '252', '26', '289', '326', '373', '432', '485', '502', '537', '560', '564', '603', '636', '674', '7', '726', '75', '759', '790', '841', '865', '901', '91', '947', '988']\n",
      "NDB_641 -> ['1017', '115', '130', '154', '195', '22', '244', '283', '296', '351', '395', '411', '450', '463', '481', '530', '579', '630', '64', '656', '695', '721', '800', '816', '849', '876', '89', '906', '949', '978']\n",
      "NDB_643 -> ['1038', '122', '183', '200', '22', '238', '261', '319', '359', '381', '434', '461', '491', '51', '540', '582', '611', '649', '681', '720', '745', '802', '824', '84', '866', '890', '924', '958', '976', '997']\n",
      "NDB_647 -> ['1010', '1053', '11', '134', '157', '200', '237', '257', '295', '324', '376', '440', '478', '507', '530', '564', '60', '608', '616', '652', '684', '750', '770', '810', '829', '86', '861', '904', '945', '971']\n",
      "NDB_652 -> ['1020', '118', '155', '18', '197', '254', '31', '310', '324', '367', '392', '422', '453', '491', '527', '569', '610', '630', '662', '697', '72', '749', '766', '795', '840', '878', '901', '925', '96', '965']\n",
      "NDB_653 -> ['1000', '101', '1048', '135', '16', '172', '205', '230', '243', '303', '370', '385', '439', '466', '505', '528', '55', '578', '588', '616', '628', '676', '740', '768', '788', '827', '879', '902', '953', '973']\n",
      "NDB_661 -> ['1035', '138', '155', '174', '229', '25', '250', '294', '327', '352', '388', '418', '449', '471', '501', '51', '522', '583', '626', '691', '708', '716', '737', '804', '841', '875', '89', '915', '960', '972']\n",
      "NDB_665 -> ['1007', '1049', '129', '147', '185', '20', '247', '267', '311', '338', '372', '435', '48', '481', '499', '516', '531', '578', '617', '659', '681', '709', '746', '787', '80', '847', '859', '908', '932', '975']\n",
      "NDB_680 -> ['1022', '120', '184', '198', '226', '23', '267', '281', '323', '369', '39', '397', '419', '450', '467', '502', '520', '534', '580', '652', '694', '722', '783', '833', '85', '851', '873', '920', '946', '977']\n",
      "NDB_683 -> ['1056', '114', '129', '15', '165', '215', '220', '273', '321', '343', '355', '403', '43', '445', '493', '541', '596', '626', '663', '701', '739', '77', '773', '806', '849', '856', '922', '959', '977', '999']\n",
      "NDB_694 -> ['1009', '1051', '106', '190', '233', '264', '3', '309', '331', '374', '419', '497', '524', '553', '568', '60', '606', '621', '654', '686', '703', '731', '77', '797', '828', '870', '911', '94', '960', '992']\n",
      "NDB_698 -> ['1001', '1055', '1062', '121', '140', '153', '193', '254', '292', '327', '363', '399', '413', '456', '488', '5', '544', '58', '593', '637', '663', '674', '713', '731', '76', '775', '843', '873', '921', '953', '987']\n",
      "NDB_706 -> ['1014', '1044', '1062', '137', '176', '201', '227', '264', '281', '29', '303', '375', '445', '469', '526', '548', '567', '612', '622', '642', '682', '74', '750', '769', '789', '840', '875', '88', '916', '962']\n",
      "NDB_710 -> ['1036', '1062', '125', '175', '203', '218', '22', '240', '322', '339', '372', '386', '412', '457', '462', '501', '513', '546', '56', '575', '632', '693', '715', '744', '792', '82', '829', '869', '905', '937']\n",
      "NDB_712 -> ['100', '1021', '1051', '145', '171', '185', '228', '255', '308', '332', '34', '373', '390', '440', '470', '492', '52', '544', '572', '619', '658', '678', '733', '769', '809', '846', '879', '913', '943', '984']\n",
      "NDB_713 -> ['1015', '124', '16', '160', '182', '239', '271', '286', '304', '349', '395', '443', '476', '511', '519', '54', '573', '607', '641', '683', '702', '719', '735', '804', '828', '84', '887', '899', '938', '971']\n",
      "NDB_715 -> ['1016', '1050', '112', '131', '164', '212', '243', '278', '316', '38', '382', '397', '427', '490', '507', '531', '54', '548', '597', '646', '672', '732', '744', '796', '81', '817', '888', '912', '939', '986']\n",
      "NDB_716 -> ['1018', '1060', '123', '169', '186', '220', '239', '291', '329', '33', '367', '398', '436', '461', '487', '544', '585', '61', '629', '657', '667', '715', '759', '803', '81', '815', '859', '900', '945', '982']\n",
      "NDB_718 -> ['1013', '1049', '14', '144', '182', '204', '233', '257', '299', '323', '35', '363', '408', '452', '494', '528', '577', '599', '618', '653', '678', '692', '728', '798', '81', '837', '884', '906', '956', '979']\n",
      "NDB_721 -> ['10', '1020', '1048', '106', '143', '193', '234', '261', '286', '338', '347', '386', '423', '44', '455', '489', '522', '561', '596', '620', '689', '728', '75', '769', '839', '851', '877', '896', '927', '941', '969']\n",
      "NDB_725 -> ['1027', '1059', '109', '146', '15', '159', '176', '222', '270', '307', '33', '338', '354', '392', '412', '452', '480', '519', '551', '592', '645', '666', '707', '720', '793', '845', '864', '907', '948', '988']\n",
      "NDB_728 -> ['1024', '106', '176', '204', '236', '267', '27', '313', '325', '359', '389', '429', '488', '510', '543', '564', '59', '600', '627', '670', '714', '754', '806', '822', '854', '889', '89', '918', '937', '984']\n",
      "NDB_730 -> ['1017', '1058', '107', '143', '192', '223', '273', '28', '298', '329', '350', '388', '414', '452', '496', '513', '532', '595', '646', '690', '709', '72', '722', '791', '8', '825', '870', '924', '953', '970']\n",
      "NDB_734 -> ['10', '1000', '104', '1054', '136', '180', '204', '231', '269', '314', '333', '381', '396', '446', '465', '49', '507', '533', '593', '623', '666', '709', '735', '755', '792', '846', '868', '925', '954', '977']\n",
      "NDB_737 -> ['1028', '1054', '110', '135', '17', '188', '200', '242', '258', '321', '344', '360', '420', '457', '499', '529', '53', '556', '581', '605', '654', '702', '713', '728', '800', '825', '858', '897', '929', '942']\n",
      "NDB_742 -> ['1039', '108', '184', '2', '211', '252', '290', '304', '362', '40', '438', '494', '525', '554', '581', '612', '617', '658', '697', '718', '76', '802', '812', '839', '871', '900', '93', '948', '973', '998']\n",
      "NDB_743 -> ['1028', '103', '1045', '150', '187', '219', '249', '26', '297', '333', '374', '444', '47', '482', '525', '552', '569', '601', '629', '681', '74', '742', '754', '774', '808', '848', '873', '922', '951', '985']\n",
      "NDB_746 -> ['102', '1020', '1055', '144', '167', '206', '240', '273', '300', '313', '368', '393', '411', '449', '46', '495', '529', '54', '587', '628', '668', '686', '736', '756', '776', '842', '860', '920', '933']\n",
      "NDB_751 -> ['1013', '1025', '109', '140', '166', '17', '213', '253', '289', '301', '361', '42', '426', '445', '477', '513', '530', '583', '600', '635', '68', '689', '718', '742', '797', '831', '882', '920', '938', '982']\n",
      "NDB_752 -> ['1023', '1060', '129', '150', '163', '208', '253', '298', '331', '354', '386', '39', '4', '411', '433', '476', '533', '556', '588', '650', '680', '723', '767', '809', '832', '857', '86', '921', '944', '983']\n",
      "NDB_754 -> ['1027', '117', '195', '226', '256', '287', '3', '332', '36', '361', '387', '433', '448', '473', '541', '593', '641', '665', '684', '71', '745', '758', '791', '819', '860', '888', '894', '91', '966', '995']\n",
      "NDB_755 -> ['1034', '123', '182', '201', '235', '274', '309', '32', '334', '362', '385', '412', '431', '464', '482', '538', '582', '637', '665', '68', '687', '716', '734', '799', '834', '86', '863', '912', '961', '975']\n",
      "NDB_758 -> ['1030', '1056', '113', '14', '191', '231', '264', '287', '337', '352', '400', '436', '44', '495', '516', '552', '592', '632', '652', '677', '69', '690', '751', '796', '838', '878', '90', '913', '961']\n",
      "NDB_760 -> ['1026', '1041', '144', '186', '210', '245', '279', '316', '345', '376', '413', '431', '45', '496', '534', '550', '594', '634', '689', '69', '741', '785', '822', '846', '864', '87', '909', '944', '976', '99']\n",
      "NDB_773 -> ['1044', '133', '172', '20', '203', '228', '270', '320', '358', '382', '39', '438', '479', '515', '555', '589', '604', '623', '639', '670', '686', '729', '766', '841', '863', '910', '940', '976', '98', '999']\n",
      "NDB_778 -> ['1010', '1046', '113', '131', '18', '196', '227', '255', '311', '328', '369', '437', '482', '503', '515', '55', '553', '590', '626', '661', '706', '747', '781', '820', '836', '84', '880', '918', '941', '979']\n",
      "NDB_779 -> ['1000', '1053', '124', '159', '178', '218', '23', '248', '293', '334', '377', '414', '449', '490', '537', '574', '58', '605', '618', '638', '667', '710', '739', '767', '818', '867', '910', '941', '96', '984']\n",
      "NDB_784 -> ['1040', '150', '161', '177', '23', '235', '260', '308', '356', '380', '429', '465', '503', '528', '557', '579', '602', '648', '66', '677', '743', '776', '821', '865', '891', '919', '934', '960', '98', '998']\n",
      "NDB_788 -> ['1001', '1047', '133', '157', '175', '219', '236', '29', '305', '360', '384', '441', '473', '509', '542', '55', '565', '607', '617', '657', '701', '724', '765', '786', '838', '882', '907', '956', '96', '980']\n",
      "NDB_790 -> ['1036', '145', '174', '20', '206', '239', '260', '303', '326', '369', '378', '441', '477', '506', '551', '568', '599', '62', '625', '638', '685', '748', '784', '823', '839', '856', '87', '889', '904', '965']\n",
      "NDB_805 -> ['1031', '116', '147', '165', '188', '218', '237', '292', '340', '35', '354', '384', '430', '465', '508', '538', '566', '601', '627', '642', '675', '71', '741', '801', '812', '852', '877', '899', '924', '943']\n",
      "NDB_812 -> ['1011', '1051', '126', '13', '160', '199', '240', '272', '314', '356', '383', '426', '473', '510', '538', '556', '567', '609', '63', '631', '688', '726', '788', '805', '830', '859', '87', '890', '928', '940']\n",
      "NDB_815 -> ['1033', '12', '122', '169', '205', '229', '256', '296', '335', '349', '383', '420', '462', '480', '524', '554', '57', '590', '622', '639', '680', '717', '783', '794', '83', '834', '877', '914', '950', '985']\n",
      "NDB_817 -> ['101', '1023', '136', '15', '160', '177', '249', '268', '291', '330', '353', '397', '440', '474', '509', '532', '584', '618', '662', '669', '69', '691', '730', '787', '816', '862', '923', '942', '997']\n",
      "NDB_818 -> ['1011', '1026', '103', '116', '192', '213', '246', '274', '318', '371', '420', '454', '48', '495', '5', '542', '572', '603', '648', '685', '721', '753', '766', '810', '836', '850', '869', '88', '927', '939']\n",
      "NDB_824 -> ['1019', '130', '148', '191', '21', '210', '251', '294', '310', '353', '399', '415', '446', '483', '492', '539', '581', '643', '661', '679', '692', '70', '739', '774', '80', '814', '870', '907', '947', '983']\n",
      "NDB_826 -> ['1057', '111', '131', '151', '154', '185', '246', '29', '293', '331', '356', '392', '428', '46', '471', '511', '540', '595', '647', '684', '717', '754', '780', '815', '851', '868', '903', '952', '999']\n",
      "NDB_832 -> ['1016', '105', '183', '225', '263', '295', '339', '348', '35', '393', '4', '413', '450', '484', '542', '570', '606', '61', '624', '658', '679', '740', '794', '806', '819', '858', '908', '92', '938', '965', '970']\n",
      "NDB_835 -> ['102', '1023', '148', '168', '187', '237', '25', '263', '282', '305', '344', '358', '439', '469', '506', '536', '59', '590', '597', '641', '660', '697', '733', '779', '845', '865', '903', '958', '992']\n",
      "NDB_836 -> ['100', '1019', '1057', '132', '155', '167', '247', '27', '278', '285', '326', '350', '395', '403', '424', '484', '517', '557', '58', '584', '622', '688', '729', '745', '792', '837', '871', '896', '951', '982']\n",
      "NDB_838 -> ['1021', '118', '12', '162', '213', '250', '26', '292', '330', '355', '393', '436', '446', '496', '535', '561', '563', '605', '63', '633', '695', '752', '786', '844', '872', '887', '904', '948', '97', '989']\n",
      "NDB_844 -> ['1028', '14', '145', '168', '209', '247', '295', '330', '343', '379', '415', '456', '487', '527', '555', '574', '600', '647', '67', '700', '726', '801', '813', '82', '836', '855', '898', '94', '949']\n",
      "NDB_847 -> ['1033', '1055', '107', '11', '138', '162', '209', '221', '278', '299', '347', '36', '400', '425', '454', '475', '550', '585', '619', '654', '694', '70', '747', '807', '845', '854', '890', '905', '955', '973']\n",
      "NDB_853 -> ['1007', '1026', '134', '180', '208', '232', '259', '299', '322', '368', '439', '461', '489', '517', '568', '607', '644', '65', '675', '7', '706', '734', '758', '796', '840', '885', '915', '949', '95']\n",
      "NDB_856 -> ['1', '1027', '110', '141', '184', '224', '277', '302', '335', '375', '400', '426', '48', '484', '505', '536', '565', '602', '623', '64', '663', '700', '732', '781', '814', '848', '874', '926', '945', '989']\n",
      "NDB_860 -> ['1029', '118', '16', '166', '212', '249', '306', '323', '351', '396', '41', '418', '433', '488', '545', '566', '608', '619', '660', '688', '715', '73', '736', '773', '814', '883', '914', '92', '957', '988']\n",
      "NDB_862 -> ['1033', '110', '197', '227', '268', '300', '340', '362', '389', '41', '444', '463', '476', '521', '561', '583', '6', '604', '621', '66', '667', '685', '751', '775', '817', '833', '868', '928', '964', '99']\n",
      "NDB_863 -> ['1015', '1039', '111', '12', '175', '207', '224', '261', '294', '328', '378', '427', '448', '463', '49', '500', '523', '559', '574', '631', '678', '729', '772', '807', '825', '88', '881', '901', '934', '974']\n",
      "NDB_865 -> ['1007', '1050', '119', '190', '214', '252', '320', '344', '37', '370', '414', '455', '480', '503', '539', '558', '576', '650', '694', '721', '73', '801', '843', '862', '889', '9', '913', '952', '971', '99']\n",
      "NDB_866 -> ['1002', '1041', '132', '161', '178', '21', '228', '271', '281', '32', '322', '371', '444', '477', '510', '541', '571', '585', '620', '662', '690', '707', '732', '790', '827', '866', '92', '929', '956', '990']\n",
      "NDB_868 -> ['1022', '1060', '13', '148', '166', '173', '221', '256', '283', '297', '342', '381', '422', '46', '490', '516', '551', '589', '642', '655', '669', '711', '727', '790', '82', '837', '880', '915', '936', '991']\n",
      "NDB_872 -> ['101', '1022', '1058', '121', '179', '188', '238', '24', '263', '289', '336', '352', '366', '403', '423', '487', '545', '594', '643', '66', '670', '710', '717', '758', '793', '850', '857', '917', '955', '989']\n",
      "NDB_875 -> ['1021', '1058', '119', '141', '195', '242', '262', '285', '312', '33', '346', '371', '425', '451', '489', '527', '56', '587', '634', '651', '676', '691', '723', '76', '773', '820', '861', '910', '954', '985']\n",
      "NDB_878 -> ['1010', '1047', '126', '172', '207', '234', '259', '313', '361', '379', '428', '447', '470', '50', '546', '579', '611', '629', '660', '687', '731', '767', '794', '817', '871', '9', '916', '959', '97', '990']\n",
      "NDB_880 -> ['1008', '1054', '127', '156', '170', '18', '235', '257', '27', '287', '335', '364', '442', '479', '506', '539', '572', '601', '640', '695', '719', '736', '775', '805', '818', '847', '886', '905', '95', '962']\n",
      "NDB_881 -> ['100', '1024', '126', '170', '201', '224', '268', '312', '333', '349', '384', '407', '437', '45', '491', '520', '592', '624', '67', '692', '708', '744', '756', '797', '815', '857', '891', '897', '963', '995']\n",
      "NDB_883 -> ['1002', '105', '1059', '142', '153', '192', '2', '251', '283', '314', '343', '36', '365', '418', '486', '505', '524', '586', '62', '627', '653', '672', '724', '759', '772', '844', '861', '886', '896', '944']\n",
      "NDB_887 -> ['1018', '1056', '147', '161', '19', '199', '222', '276', '288', '316', '348', '366', '437', '475', '49', '509', '543', '595', '630', '696', '734', '757', '802', '823', '855', '887', '91', '921', '933', '962']\n",
      "NDB_889 -> ['1030', '109', '128', '163', '210', '250', '315', '329', '365', '396', '40', '423', '497', '5', '523', '546', '576', '599', '639', '655', '701', '74', '749', '799', '824', '885', '902', '929', '943', '978']\n",
      "NDB_896 -> ['1017', '1040', '111', '135', '158', '181', '236', '269', '309', '358', '38', '383', '421', '498', '52', '526', '552', '575', '613', '643', '675', '718', '756', '785', '844', '866', '923', '959', '991']\n",
      "NDB_899 -> ['1031', '1059', '117', '143', '164', '214', '220', '279', '302', '31', '366', '380', '443', '467', '47', '529', '563', '584', '6', '638', '651', '680', '711', '740', '799', '830', '882', '906', '932', '963']\n",
      "NDB_901 -> ['1002', '1029', '136', '181', '205', '233', '25', '262', '315', '345', '360', '443', '478', '508', '543', '582', '603', '624', '63', '659', '693', '743', '781', '803', '831', '85', '881', '898', '936', '966']\n",
      "NDB_905 -> ['10', '1015', '1040', '125', '170', '202', '242', '266', '291', '30', '310', '376', '424', '456', '470', '502', '535', '591', '621', '656', '665', '710', '727', '798', '827', '856', '908', '93', '933', '980']\n",
      "NDB_910 -> ['1034', '134', '173', '181', '219', '234', '288', '325', '34', '380', '391', '434', '483', '515', '531', '586', '625', '683', '713', '72', '753', '789', '807', '842', '876', '909', '93', '957', '992', '995']\n",
      "NDB_917 -> ['1008', '1025', '128', '146', '186', '230', '259', '308', '32', '328', '377', '421', '451', '483', '504', '51', '536', '554', '573', '647', '682', '711', '724', '791', '843', '85', '858', '872', '927', '964']\n",
      "NDB_925 -> ['1019', '1053', '116', '146', '180', '202', '223', '272', '288', '301', '342', '374', '38', '442', '469', '493', '549', '594', '628', '657', '693', '751', '784', '804', '834', '854', '9', '926', '963', '979']\n",
      "NDB_926 -> ['1038', '112', '137', '173', '198', '229', '24', '271', '311', '346', '379', '430', '454', '498', '521', '591', '608', '640', '65', '650', '696', '733', '795', '808', '830', '881', '902', '935', '998']\n",
      "NDB_929 -> ['1036', '119', '197', '238', '277', '28', '305', '332', '355', '398', '431', '447', '467', '500', '518', '53', '549', '591', '653', '682', '70', '730', '770', '788', '832', '876', '919', '94', '951', '969']\n",
      "NDB_934 -> ['1041', '1057', '117', '142', '162', '211', '243', '30', '318', '339', '377', '390', '42', '441', '464', '474', '526', '53', '589', '640', '674', '708', '714', '757', '793', '820', '878', '914', '958', '974']\n",
      "NDB_935 -> ['1', '1029', '103', '122', '164', '211', '253', '324', '340', '364', '37', '391', '425', '492', '500', '514', '553', '577', '636', '67', '677', '737', '755', '768', '822', '885', '903', '930', '935', '987']\n",
      "NDB_940 -> ['1013', '1044', '115', '142', '196', '223', '266', '312', '345', '357', '4', '40', '415', '485', '499', '522', '559', '575', '610', '645', '698', '719', '77', '780', '833', '848', '862', '917', '935', '961']\n",
      "NDB_941 -> ['1009', '1047', '108', '11', '124', '158', '198', '222', '258', '319', '34', '357', '382', '408', '458', '464', '501', '545', '596', '633', '655', '676', '730', '742', '772', '847', '874', '894', '947', '970']\n",
      "NDB_946 -> ['1', '1032', '114', '13', '154', '191', '255', '276', '318', '378', '398', '435', '457', '47', '486', '520', '560', '588', '646', '668', '707', '741', '798', '828', '883', '916', '95', '955', '972', '997']\n",
      "NDB_958 -> ['1032', '123', '17', '171', '209', '225', '260', '290', '325', '342', '353', '43', '432', '451', '478', '533', '576', '609', '649', '699', '722', '757', '780', '795', '83', '838', '886', '894', '930', '939']\n",
      "NDB_961 -> ['1025', '1039', '130', '151', '165', '19', '214', '254', '307', '346', '359', '424', '486', '514', '555', '578', '61', '613', '644', '669', '716', '735', '776', '810', '818', '83', '874', '909', '946', '969']\n",
      "NDB_965 -> ['1016', '1038', '104', '140', '169', '2', '208', '221', '277', '321', '370', '41', '435', '472', '485', '532', '567', '598', '625', '661', '687', '71', '749', '786', '821', '835', '869', '911', '946', '974']\n",
      "NDB_971 -> ['1001', '1045', '107', '127', '158', '178', '231', '276', '282', '302', '373', '407', '43', '447', '471', '514', '558', '573', '606', '62', '635', '698', '752', '800', '821', '864', '888', '898', '928', '975']\n",
      "NDB_974 -> ['1035', '125', '183', '199', '246', '270', '285', '334', '357', '387', '407', '421', '493', '523', '580', '612', '616', '636', '64', '703', '747', '765', '8', '80', '808', '842', '884', '919', '950', '990']\n",
      "NDB_977 -> ['1035', '120', '190', '212', '244', '279', '28', '300', '347', '399', '422', '458', '498', '535', '570', '59', '598', '649', '683', '7', '723', '755', '805', '823', '863', '891', '90', '900', '926', '966']\n",
      "NDB_989 -> ['1030', '132', '177', '202', '232', '269', '297', '327', '351', '385', '408', '455', '472', '548', '56', '580', '6', '610', '635', '659', '668', '748', '779', '809', '819', '850', '867', '930', '940', '97']\n",
      "NDB_994 -> ['1011', '105', '1050', '133', '156', '168', '230', '24', '258', '293', '337', '375', '432', '462', '475', '525', '577', '60', '613', '620', '656', '698', '743', '784', '824', '852', '860', '925', '957', '983']\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "reload(check)\n",
    "readout_reg_dic, readout2reg_check = check._check_readout_to_region(reg_readout_dic, full_records, \n",
    "                                                                    check_readout_dict, primer_len=primer_len, \n",
    "                                                                    target_len=target_len,\n",
    "                                                                    probe_readout_num=num_readout_per_probe)\n",
    "\n",
    "print(readout2reg_check)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7445b10e",
   "metadata": {},
   "source": [
    "### 4.1.5 check bewteen probes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "781de8ea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- saving internal 17-mer map to file:\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\probe_table_17.npz\n"
     ]
    }
   ],
   "source": [
    "# construct internal map\n",
    "reload(check)\n",
    "int_map = check._construct_internal_map(full_records, library_folder, save=False) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "8d1a7cd9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "True\n"
     ]
    }
   ],
   "source": [
    "# internal check probes\n",
    "readout_count_dic, readout_count_check = check._check_readout_in_probes(readout_reg_dic, reg_size_dic, int_map, readout_dict)\n",
    "print(readout_count_check)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "fabaf942",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- Sequence:loc_1:86240000-86260000_gene_34_pb_4_pos_274_strand_+_readouts_[NDB_941_c,NDB_712_c,NDB_910_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:85, dumped\n",
      "--- Sequence:loc_1:86240000-86260000_gene_34_pb_5_pos_315_strand_+_readouts_[NDB_941_c,NDB_712_c,NDB_910_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:81, dumped\n",
      "--- Sequence:loc_1:86240000-86260000_gene_34_pb_6_pos_352_strand_-_readouts_[NDB_941_c,NDB_712_c,NDB_910_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:94, dumped\n",
      "--- Sequence:loc_1:86240000-86260000_gene_34_pb_7_pos_389_strand_+_readouts_[NDB_941_c,NDB_712_c,NDB_910_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:81, dumped\n",
      "--- Sequence:loc_1:86240000-86260000_gene_34_pb_8_pos_426_strand_+_readouts_[NDB_941_c,NDB_712_c,NDB_910_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:85, dumped\n",
      "--- Sequence:loc_1:86240000-86260000_gene_34_pb_9_pos_464_strand_-_readouts_[NDB_941_c,NDB_712_c,NDB_910_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:88, dumped\n",
      "--- Sequence:loc_2:33740000-33760000_gene_92_pb_80_pos_5271_strand_+_readouts_[NDB_860_c,NDB_866_c,NDB_832_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:86, dumped\n",
      "--- Sequence:loc_2:33740000-33760000_gene_92_pb_82_pos_5358_strand_-_readouts_[NDB_860_c,NDB_866_c,NDB_832_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:76, dumped\n",
      "--- Sequence:loc_2:33740000-33760000_gene_92_pb_83_pos_5409_strand_+_readouts_[NDB_860_c,NDB_866_c,NDB_832_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:86, dumped\n",
      "--- Sequence:loc_2:163740000-163760000_gene_144_pb_300_pos_19151_strand_-_readouts_[NDB_718_c,NDB_760_c,NDB_746_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:55, dumped\n",
      "--- Sequence:loc_2:163740000-163760000_gene_144_pb_301_pos_19188_strand_-_readouts_[NDB_718_c,NDB_760_c,NDB_746_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:58, dumped\n",
      "--- Sequence:loc_2:168740000-168760000_gene_146_pb_94_pos_7451_strand_+_readouts_[NDB_725_c,NDB_917_c,NDB_925_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:69, dumped\n",
      "--- Sequence:loc_2:168740000-168760000_gene_146_pb_95_pos_7654_strand_+_readouts_[NDB_725_c,NDB_917_c,NDB_925_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:69, dumped\n",
      "--- Sequence:loc_2:168740000-168760000_gene_146_pb_96_pos_7856_strand_+_readouts_[NDB_725_c,NDB_917_c,NDB_925_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:62, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_126_pos_9009_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:57, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_127_pos_9048_strand_-_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_129_pos_10723_strand_-_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:93, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_130_pos_10790_strand_-_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:134, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_131_pos_10873_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:124, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_132_pos_10959_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:130, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_133_pos_11019_strand_-_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:105, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_134_pos_11109_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:126, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_135_pos_11165_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:72, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_136_pos_11202_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:63, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_137_pos_11245_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:75, dumped\n",
      "--- Sequence:loc_2:171240000-171260000_gene_147_pb_138_pos_11286_strand_+_readouts_[NDB_805_c,NDB_887_c,NDB_665_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:105, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_178_pos_12141_strand_-_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:56, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_179_pos_12850_strand_-_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:77, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_180_pos_13321_strand_+_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_182_pos_13458_strand_-_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:82, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_184_pos_13644_strand_-_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_189_pos_14160_strand_+_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:68, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_190_pos_14226_strand_+_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:68, dumped\n",
      "--- Sequence:loc_2:173740000-173760000_gene_148_pb_193_pos_14851_strand_-_readouts_[NDB_868_c,NDB_835_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:61, dumped\n",
      "--- Sequence:loc_2:178740000-178760000_gene_150_pb_33_pos_3059_strand_-_readouts_[NDB_784_c,NDB_743_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:220, dumped\n",
      "--- Sequence:loc_2:178740000-178760000_gene_150_pb_82_pos_6292_strand_+_readouts_[NDB_784_c,NDB_743_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:75, dumped\n",
      "--- Sequence:loc_2:178740000-178760000_gene_150_pb_84_pos_6390_strand_+_readouts_[NDB_784_c,NDB_743_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:63, dumped\n",
      "--- Sequence:loc_2:178740000-178760000_gene_150_pb_85_pos_6462_strand_+_readouts_[NDB_784_c,NDB_743_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:72, dumped\n",
      "--- Sequence:loc_3:3740000-3760000_gene_153_pb_138_pos_14423_strand_+_readouts_[NDB_698_c,NDB_883_c,NDB_1075_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:66, dumped\n",
      "--- Sequence:loc_3:3740000-3760000_gene_153_pb_139_pos_14502_strand_-_readouts_[NDB_698_c,NDB_883_c,NDB_1075_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:60, dumped\n",
      "--- Sequence:loc_3:8740000-8760000_gene_155_pb_76_pos_5672_strand_+_readouts_[NDB_652_c,NDB_661_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:62, dumped\n",
      "--- Sequence:loc_3:8740000-8760000_gene_155_pb_77_pos_5879_strand_-_readouts_[NDB_652_c,NDB_661_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:76, dumped\n",
      "--- Sequence:loc_3:8740000-8760000_gene_155_pb_78_pos_5953_strand_-_readouts_[NDB_652_c,NDB_661_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:60, dumped\n",
      "--- Sequence:loc_3:8740000-8760000_gene_155_pb_80_pos_6064_strand_+_readouts_[NDB_652_c,NDB_661_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:87, dumped\n",
      "--- Sequence:loc_3:8740000-8760000_gene_155_pb_81_pos_6104_strand_-_readouts_[NDB_652_c,NDB_661_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_3:8740000-8760000_gene_155_pb_82_pos_6181_strand_-_readouts_[NDB_652_c,NDB_661_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:65, dumped\n",
      "--- Sequence:loc_3:8740000-8760000_gene_155_pb_83_pos_6236_strand_-_readouts_[NDB_652_c,NDB_661_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:74, dumped\n",
      "--- Sequence:loc_3:41240000-41260000_gene_168_pb_97_pos_7786_strand_+_readouts_[NDB_844_c,NDB_835_c,NDB_994_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_3:41240000-41260000_gene_168_pb_117_pos_9262_strand_-_readouts_[NDB_844_c,NDB_835_c,NDB_994_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- Sequence:loc_3:96240000-96260000_gene_190_pb_1_pos_503_strand_+_readouts_[NDB_865_c,NDB_694_c,NDB_977_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_3:96240000-96260000_gene_190_pb_66_pos_5288_strand_-_readouts_[NDB_865_c,NDB_694_c,NDB_977_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_3:146240000-146260000_gene_210_pb_215_pos_15152_strand_+_readouts_[NDB_760_c,NDB_889_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:61, dumped\n",
      "--- Sequence:loc_4:53740000-53760000_gene_238_pb_250_pos_19614_strand_+_readouts_[NDB_643_c,NDB_929_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:58, dumped\n",
      "--- Sequence:loc_4:53740000-53760000_gene_238_pb_251_pos_19682_strand_-_readouts_[NDB_643_c,NDB_929_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:57, dumped\n",
      "--- Sequence:loc_4:91240000-91260000_gene_253_pb_78_pos_5217_strand_-_readouts_[NDB_752_c,NDB_935_c,NDB_751_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:55, dumped\n",
      "--- Sequence:loc_4:123740000-123760000_gene_266_pb_167_pos_13624_strand_+_readouts_[NDB_631_c,NDB_905_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:94, dumped\n",
      "--- Sequence:loc_4:123740000-123760000_gene_266_pb_168_pos_13801_strand_-_readouts_[NDB_631_c,NDB_905_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:97, dumped\n",
      "--- Sequence:loc_4:123740000-123760000_gene_266_pb_169_pos_13901_strand_+_readouts_[NDB_631_c,NDB_905_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:87, dumped\n",
      "--- Sequence:loc_4:123740000-123760000_gene_266_pb_170_pos_13959_strand_+_readouts_[NDB_631_c,NDB_905_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:94, dumped\n",
      "--- Sequence:loc_4:123740000-123760000_gene_266_pb_172_pos_14087_strand_+_readouts_[NDB_631_c,NDB_905_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:109, dumped\n",
      "--- Sequence:loc_4:123740000-123760000_gene_266_pb_173_pos_14153_strand_-_readouts_[NDB_631_c,NDB_905_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:104, dumped\n",
      "--- Sequence:loc_4:153740000-153760000_gene_278_pb_64_pos_12854_strand_+_readouts_[NDB_847_c,NDB_715_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:64, dumped\n",
      "--- Sequence:loc_4:153740000-153760000_gene_278_pb_67_pos_12971_strand_-_readouts_[NDB_847_c,NDB_715_c,NDB_836_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:57, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_11_pos_1096_strand_-_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:62, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_13_pos_1225_strand_+_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:67, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_14_pos_1268_strand_+_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:73, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_17_pos_1391_strand_-_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:77, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_18_pos_1431_strand_+_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:68, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_20_pos_1513_strand_-_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:75, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_21_pos_1557_strand_+_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_5:16240000-16260000_gene_286_pb_22_pos_1600_strand_+_readouts_[NDB_713_c,NDB_634_c,NDB_721_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:76, dumped\n",
      "--- Sequence:loc_5:73740000-73760000_gene_309_pb_66_pos_3900_strand_-_readouts_[NDB_755_c,NDB_896_c,NDB_694_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:55, dumped\n",
      "--- Sequence:loc_5:73740000-73760000_gene_309_pb_68_pos_4005_strand_-_readouts_[NDB_755_c,NDB_896_c,NDB_694_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:61, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_134_pos_12115_strand_-_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:77, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_135_pos_12153_strand_+_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:82, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_136_pos_12203_strand_-_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:94, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_137_pos_12253_strand_-_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:72, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_138_pos_12303_strand_+_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:91, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_139_pos_12353_strand_+_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:113, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_141_pos_12453_strand_-_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:109, dumped\n",
      "--- Sequence:loc_5:96240000-96260000_gene_318_pb_142_pos_12503_strand_-_readouts_[NDB_946_c,NDB_818_c,NDB_934_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:54, dumped\n",
      "--- Sequence:loc_5:118740000-118760000_gene_327_pb_135_pos_7983_strand_-_readouts_[NDB_661_c,NDB_698_c,NDB_989_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:56, dumped\n",
      "--- Sequence:loc_5:118740000-118760000_gene_327_pb_136_pos_8049_strand_+_readouts_[NDB_661_c,NDB_698_c,NDB_989_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:74, dumped\n",
      "--- Sequence:loc_5:118740000-118760000_gene_327_pb_138_pos_8155_strand_+_readouts_[NDB_661_c,NDB_698_c,NDB_989_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_5:118740000-118760000_gene_327_pb_139_pos_8219_strand_+_readouts_[NDB_661_c,NDB_698_c,NDB_989_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:71, dumped\n",
      "--- Sequence:loc_6:38740000-38760000_gene_356_pb_72_pos_5114_strand_+_readouts_[NDB_784_c,NDB_826_c,NDB_812_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_6:38740000-38760000_gene_356_pb_95_pos_6505_strand_+_readouts_[NDB_784_c,NDB_826_c,NDB_812_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_143_pos_11028_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:106, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_144_pos_11178_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:117, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_145_pos_11281_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:103, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_146_pos_11435_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:109, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_147_pos_11603_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:97, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_148_pos_11811_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:146, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_149_pos_11856_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:115, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_150_pos_11902_strand_-_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:105, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_151_pos_11939_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:141, dumped\n",
      "--- Sequence:loc_6:116240000-116260000_gene_387_pb_152_pos_11997_strand_+_readouts_[NDB_974_c,NDB_754_c,NDB_634_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:71, dumped\n",
      "--- Sequence:loc_6:146240000-146260000_gene_399_pb_138_pos_11084_strand_+_readouts_[NDB_698_c,NDB_977_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:68, dumped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- Sequence:loc_6:146240000-146260000_gene_399_pb_139_pos_11236_strand_+_readouts_[NDB_698_c,NDB_977_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:65, dumped\n",
      "--- Sequence:loc_6:146240000-146260000_gene_399_pb_140_pos_11303_strand_+_readouts_[NDB_698_c,NDB_977_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:73, dumped\n",
      "--- Sequence:loc_6:146240000-146260000_gene_399_pb_141_pos_11410_strand_+_readouts_[NDB_698_c,NDB_977_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:84, dumped\n",
      "--- Sequence:loc_6:146240000-146260000_gene_399_pb_143_pos_11755_strand_+_readouts_[NDB_698_c,NDB_977_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:83, dumped\n",
      "--- Sequence:loc_6:146240000-146260000_gene_399_pb_144_pos_11860_strand_+_readouts_[NDB_698_c,NDB_977_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_7:98740000-98760000_gene_440_pb_224_pos_14017_strand_+_readouts_[NDB_817_c,NDB_647_c,NDB_712_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_7:98740000-98760000_gene_440_pb_226_pos_14114_strand_-_readouts_[NDB_817_c,NDB_647_c,NDB_712_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:53, dumped\n",
      "--- Sequence:loc_7:118740000-118760000_gene_448_pb_213_pos_14177_strand_+_readouts_[NDB_754_c,NDB_863_c,NDB_1075_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:73, dumped\n",
      "--- Sequence:loc_7:118740000-118760000_gene_448_pb_214_pos_14408_strand_+_readouts_[NDB_754_c,NDB_863_c,NDB_1075_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:70, dumped\n",
      "--- Sequence:loc_7:118740000-118760000_gene_448_pb_215_pos_14459_strand_+_readouts_[NDB_754_c,NDB_863_c,NDB_1075_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:69, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_54_pos_4534_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:120, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_55_pos_4606_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:108, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_56_pos_4750_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:84, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_57_pos_4787_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:88, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_58_pos_5146_strand_+_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:82, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_59_pos_5183_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:95, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_60_pos_5246_strand_+_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:79, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_62_pos_5432_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:112, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_63_pos_5470_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:96, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_117_pos_9377_strand_+_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:60, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_118_pos_9520_strand_-_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_8:8740000-8760000_gene_463_pb_119_pos_9584_strand_+_readouts_[NDB_863_c,NDB_641_c,NDB_862_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_8:121240000-121260000_gene_508_pb_113_pos_7995_strand_-_readouts_[NDB_901_c,NDB_805_c,NDB_1066_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_9:106240000-106260000_gene_554_pb_32_pos_2615_strand_+_readouts_[NDB_815_c,NDB_917_c,NDB_742_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:66, dumped\n",
      "--- Sequence:loc_9:106240000-106260000_gene_554_pb_33_pos_2654_strand_+_readouts_[NDB_815_c,NDB_917_c,NDB_742_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:68, dumped\n",
      "--- Sequence:loc_9:106240000-106260000_gene_554_pb_34_pos_2697_strand_+_readouts_[NDB_815_c,NDB_917_c,NDB_742_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:58, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_175_pos_13645_strand_+_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:95, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_176_pos_13760_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:121, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_177_pos_13817_strand_+_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:119, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_178_pos_13854_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:106, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_179_pos_13898_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:152, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_180_pos_13994_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:104, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_181_pos_14047_strand_+_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:99, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_182_pos_14173_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:82, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_183_pos_14211_strand_+_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:139, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_184_pos_14254_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:83, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_185_pos_14295_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:123, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_186_pos_14338_strand_+_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:145, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_187_pos_14379_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:129, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_189_pos_14461_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:112, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_190_pos_14498_strand_-_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:124, dumped\n",
      "--- Sequence:loc_9:111240000-111260000_gene_556_pb_191_pos_14535_strand_+_readouts_[NDB_812_c,NDB_737_c,NDB_752_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:93, dumped\n",
      "--- Sequence:loc_9:121240000-121260000_gene_560_pb_80_pos_6689_strand_-_readouts_[NDB_1027_c,NDB_946_c,NDB_637_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:78, dumped\n",
      "--- Sequence:loc_9:121240000-121260000_gene_560_pb_81_pos_6734_strand_-_readouts_[NDB_1027_c,NDB_946_c,NDB_637_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:77, dumped\n",
      "--- Sequence:loc_9:121240000-121260000_gene_560_pb_82_pos_6773_strand_+_readouts_[NDB_1027_c,NDB_946_c,NDB_637_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:78, dumped\n",
      "--- Sequence:loc_9:121240000-121260000_gene_560_pb_83_pos_6810_strand_-_readouts_[NDB_1027_c,NDB_946_c,NDB_637_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:74, dumped\n",
      "--- Sequence:loc_10:58740000-58760000_gene_585_pb_109_pos_14216_strand_+_readouts_[NDB_847_c,NDB_716_c,NDB_866_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_10:101240000-101260000_gene_602_pb_84_pos_12000_strand_+_readouts_[NDB_784_c,NDB_856_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_10:101240000-101260000_gene_602_pb_99_pos_13471_strand_+_readouts_[NDB_784_c,NDB_856_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_10:121240000-121260000_gene_610_pb_84_pos_5443_strand_+_readouts_[NDB_652_c,NDB_989_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:66, dumped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- Sequence:loc_12:71240000-71260000_gene_692_pb_27_pos_2198_strand_+_readouts_[NDB_718_c,NDB_881_c,NDB_824_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:54, dumped\n",
      "--- Sequence:loc_12:106240000-106260000_gene_706_pb_168_pos_11458_strand_+_readouts_[NDB_778_c,NDB_634_c,NDB_853_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:60, dumped\n",
      "--- Sequence:loc_12:106240000-106260000_gene_706_pb_169_pos_11495_strand_+_readouts_[NDB_778_c,NDB_634_c,NDB_853_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:58, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_123_pos_8464_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:83, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_124_pos_8532_strand_+_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:71, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_127_pos_9493_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:99, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_128_pos_9750_strand_+_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:69, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_129_pos_10008_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:82, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_130_pos_10146_strand_+_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:68, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_131_pos_10255_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:70, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_132_pos_10891_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:86, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_133_pos_10977_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:83, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_135_pos_11553_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:56, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_137_pos_11686_strand_+_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:58, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_138_pos_11913_strand_-_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:100, dumped\n",
      "--- Sequence:loc_12:116240000-116260000_gene_710_pb_139_pos_11991_strand_+_readouts_[NDB_905_c,NDB_779_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:68, dumped\n",
      "--- Sequence:loc_13:8740000-8760000_gene_715_pb_204_pos_16755_strand_+_readouts_[NDB_710_c,NDB_716_c,NDB_860_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:145, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_159_pos_12947_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:111, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_160_pos_13136_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:156, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_161_pos_13219_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:318, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_162_pos_13285_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:150, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_163_pos_13326_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:196, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_164_pos_13382_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:335, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_165_pos_13419_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:300, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_166_pos_13482_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:252, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_167_pos_13519_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:84, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_168_pos_13650_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:295, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_169_pos_13718_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:73, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_170_pos_13759_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:62, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_171_pos_13824_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:314, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_172_pos_13893_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:198, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_173_pos_13931_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:215, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_174_pos_14264_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:155, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_175_pos_14647_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:307, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_176_pos_14707_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:167, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_177_pos_14776_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:142, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_178_pos_14910_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:76, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_179_pos_14947_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:312, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_180_pos_14984_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:270, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_181_pos_15047_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:323, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_182_pos_15084_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:335, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_183_pos_15123_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:191, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_184_pos_15171_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:163, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_185_pos_15216_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:190, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_186_pos_15254_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:140, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_188_pos_15389_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:322, dumped\n",
      "--- Sequence:loc_13:13740000-13760000_gene_717_pb_190_pos_15527_strand_+_readouts_[NDB_826_c,NDB_815_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:137, dumped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- Sequence:loc_15:43740000-43760000_gene_828_pb_98_pos_18814_strand_+_readouts_[NDB_713_c,NDB_946_c,NDB_694_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_15:43740000-43760000_gene_828_pb_99_pos_18901_strand_+_readouts_[NDB_713_c,NDB_946_c,NDB_694_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:64, dumped\n",
      "--- Sequence:loc_15:43740000-43760000_gene_828_pb_100_pos_18992_strand_+_readouts_[NDB_713_c,NDB_946_c,NDB_694_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:66, dumped\n",
      "--- Sequence:loc_15:78740000-78760000_gene_842_pb_178_pos_13175_strand_+_readouts_[NDB_974_c,NDB_746_c,NDB_910_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:62, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_86_pos_5353_strand_+_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_87_pos_5391_strand_-_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:104, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_88_pos_5450_strand_-_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:99, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_89_pos_5513_strand_+_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:114, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_90_pos_5568_strand_-_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:88, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_91_pos_5605_strand_+_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:94, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_93_pos_5935_strand_+_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:99, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_94_pos_6001_strand_+_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:81, dumped\n",
      "--- Sequence:loc_15:86240000-86260000_gene_845_pb_95_pos_6099_strand_+_readouts_[NDB_725_c,NDB_847_c,NDB_835_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:76, dumped\n",
      "--- Sequence:loc_15:98740000-98760000_gene_850_pb_66_pos_3998_strand_+_readouts_[NDB_989_c,NDB_818_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:54, dumped\n",
      "--- Sequence:loc_15:98740000-98760000_gene_850_pb_67_pos_4057_strand_+_readouts_[NDB_989_c,NDB_818_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_16:11240000-11260000_gene_857_pb_98_pos_8592_strand_+_readouts_[NDB_752_c,NDB_881_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:56, dumped\n",
      "--- Sequence:loc_16:11240000-11260000_gene_857_pb_99_pos_8827_strand_-_readouts_[NDB_752_c,NDB_881_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_16:11240000-11260000_gene_857_pb_100_pos_9307_strand_+_readouts_[NDB_752_c,NDB_881_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:81, dumped\n",
      "--- Sequence:loc_16:11240000-11260000_gene_857_pb_101_pos_9422_strand_+_readouts_[NDB_752_c,NDB_881_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:70, dumped\n",
      "--- Sequence:loc_16:11240000-11260000_gene_857_pb_102_pos_9472_strand_-_readouts_[NDB_752_c,NDB_881_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:72, dumped\n",
      "--- Sequence:loc_16:13740000-13760000_gene_858_pb_158_pos_15854_strand_+_readouts_[NDB_917_c,NDB_737_c,NDB_832_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:67, dumped\n",
      "--- Sequence:loc_16:13740000-13760000_gene_858_pb_161_pos_16025_strand_+_readouts_[NDB_917_c,NDB_737_c,NDB_832_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:64, dumped\n",
      "--- Sequence:loc_16:13740000-13760000_gene_858_pb_163_pos_16109_strand_+_readouts_[NDB_917_c,NDB_737_c,NDB_832_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_16:13740000-13760000_gene_858_pb_165_pos_16206_strand_+_readouts_[NDB_917_c,NDB_737_c,NDB_832_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_16:88740000-88760000_gene_888_pb_134_pos_11300_strand_-_readouts_[NDB_971_c,NDB_754_c,NDB_715_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_16:88740000-88760000_gene_888_pb_208_pos_19187_strand_+_readouts_[NDB_971_c,NDB_754_c,NDB_715_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_17:73740000-73760000_gene_922_pb_107_pos_7656_strand_-_readouts_[NDB_743_c,NDB_683_c,NDB_1066_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:54, dumped\n",
      "--- Sequence:loc_17:73740000-73760000_gene_922_pb_108_pos_7720_strand_-_readouts_[NDB_743_c,NDB_683_c,NDB_1066_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:56, dumped\n",
      "--- Sequence:loc_17:83740000-83760000_gene_926_pb_216_pos_16726_strand_-_readouts_[NDB_856_c,NDB_977_c,NDB_925_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_17:86240000-86260000_gene_927_pb_275_pos_17691_strand_+_readouts_[NDB_917_c,NDB_721_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:54, dumped\n",
      "--- Sequence:loc_17:86240000-86260000_gene_927_pb_279_pos_17953_strand_+_readouts_[NDB_917_c,NDB_721_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:56, dumped\n",
      "--- Sequence:loc_18:76240000-76260000_gene_961_pb_67_pos_4020_strand_+_readouts_[NDB_755_c,NDB_758_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:55, dumped\n",
      "--- Sequence:loc_18:76240000-76260000_gene_961_pb_68_pos_4248_strand_+_readouts_[NDB_755_c,NDB_758_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:73, dumped\n",
      "--- Sequence:loc_18:76240000-76260000_gene_961_pb_69_pos_4286_strand_-_readouts_[NDB_755_c,NDB_758_c,NDB_940_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:70, dumped\n",
      "--- Sequence:loc_19:41240000-41260000_gene_984_pb_26_pos_1925_strand_-_readouts_[NDB_728_c,NDB_779_c,NDB_712_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:55, dumped\n",
      "--- Sequence:loc_19:53740000-53760000_gene_989_pb_280_pos_19589_strand_+_readouts_[NDB_856_c,NDB_838_c,NDB_872_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:444, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_163_pos_13259_strand_-_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:55, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_164_pos_13304_strand_-_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:60, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_165_pos_13632_strand_-_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_169_pos_14267_strand_+_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_174_pos_14658_strand_-_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_188_pos_16280_strand_+_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:56, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_189_pos_16337_strand_+_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:55, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_190_pos_16415_strand_+_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_193_pos_16658_strand_+_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_199_pos_17295_strand_+_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_226_pos_19411_strand_-_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_X:46240000-46260000_gene_1011_pb_228_pos_19494_strand_-_readouts_[NDB_812_c,NDB_994_c,NDB_818_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_X:53740000-53760000_gene_1014_pb_154_pos_16522_strand_-_readouts_[NDB_637_c,NDB_706_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:62, dumped\n",
      "--- Sequence:loc_X:53740000-53760000_gene_1014_pb_156_pos_16597_strand_+_readouts_[NDB_637_c,NDB_706_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:59, dumped\n",
      "--- Sequence:loc_X:53740000-53760000_gene_1014_pb_163_pos_16947_strand_+_readouts_[NDB_637_c,NDB_706_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:57, dumped\n",
      "--- Sequence:loc_X:53740000-53760000_gene_1014_pb_166_pos_17061_strand_+_readouts_[NDB_637_c,NDB_706_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n",
      "--- Sequence:loc_X:53740000-53760000_gene_1014_pb_168_pos_17180_strand_-_readouts_[NDB_637_c,NDB_706_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:51, dumped\n",
      "--- Sequence:loc_X:53740000-53760000_gene_1014_pb_171_pos_17338_strand_+_readouts_[NDB_637_c,NDB_706_c,NDB_1028_c]_primers_[W1A03_primer_2,W1A10_primer_9] got hits:52, dumped\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- total probes removed by internal screening: 249\n"
     ]
    }
   ],
   "source": [
    "# check between probes\n",
    "kept_records, removed_count = check._check_between_probes(full_records, int_map, \n",
    "                                                          primer_len=primer_len, \n",
    "                                                          target_len=target_len)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "96f2083c",
   "metadata": {},
   "outputs": [],
   "source": [
    "del(int_map)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ef0c0c1c",
   "metadata": {},
   "source": [
    "### 4.1.6 finish checking, save"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "2baa5b6c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save kept records\n",
    "with open(os.path.join(library_folder, 'filtered_full_probes.fasta'), 'w') as output_handle:\n",
    "    SeqIO.write(kept_records, output_handle, \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c40e93f",
   "metadata": {},
   "source": [
    "## 4.2 BLAST screening of probes\n",
    "\n",
    "create the local blast database by:\n",
    "```bash\n",
    "makeblastdb -in C:\\Users\\puzheng\\Documents\\Tools\\lowercase_masked\\Mus_musculus.GRCm38.dna_sm.merged.fa -dbtype nucl -parse_seqids -out GRCm38_ensembl -title \"Mouse chromosomes soft masked\" -blastdb_version 4![image.png](attachment:image.png)\n",
    "```\n",
    "\n",
    "The outputs are:\n",
    "```bash\n",
    "Building a new DB, current time: 03/30/2021 21:49:18\n",
    "New DB name:   C:\\Users\\puzheng\\Documents\\Tools\\GRCm38_ensembl\n",
    "New DB title:  Mouse chromosomes soft masked\n",
    "Sequence type: Nucleotide\n",
    "Keep MBits: T\n",
    "Maximum file size: 1000000000B\n",
    "Adding sequences from FASTA; added 66 sequences in 42.2668 seconds.\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "0face513",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Blast probe against \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Genomes\\mouse\\GRCm38_ensembl\\blast_database\\GRCm38_ensembl\n",
      "-- number of probes loaded: 210335\n",
      "-- processing region: 1 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1.fasta\n",
      "--- *blasting region: 1\n",
      "--- total time for blast 1: 11.298415660858154\n",
      "-- processing region: 10 with 158 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_10.fasta\n",
      "--- *blasting region: 10\n",
      "--- total time for blast 10: 7.147331714630127\n",
      "-- processing region: 100 with 82 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_100.fasta\n",
      "--- *blasting region: 100\n",
      "--- total time for blast 100: 5.612882614135742\n",
      "-- processing region: 1000 with 55 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1000.fasta\n",
      "--- *blasting region: 1000\n",
      "--- total time for blast 1000: 5.73490309715271\n",
      "-- processing region: 1001 with 69 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1001.fasta\n",
      "--- *blasting region: 1001\n",
      "--- total time for blast 1001: 3.444300889968872\n",
      "-- processing region: 1002 with 117 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1002.fasta\n",
      "--- *blasting region: 1002\n",
      "--- total time for blast 1002: 7.721790790557861\n",
      "-- processing region: 1007 with 209 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1007.fasta\n",
      "--- *blasting region: 1007\n",
      "--- total time for blast 1007: 9.54343557357788\n",
      "-- processing region: 1008 with 136 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1008.fasta\n",
      "--- *blasting region: 1008\n",
      "--- total time for blast 1008: 5.749286413192749\n",
      "-- processing region: 1009 with 152 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1009.fasta\n",
      "--- *blasting region: 1009\n",
      "--- total time for blast 1009: 5.520097017288208\n",
      "-- processing region: 101 with 122 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_101.fasta\n",
      "--- *blasting region: 101\n",
      "--- total time for blast 101: 6.799291610717773\n",
      "-- processing region: 1010 with 110 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1010.fasta\n",
      "--- *blasting region: 1010\n",
      "--- total time for blast 1010: 11.204709768295288\n",
      "-- processing region: 1011 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1011.fasta\n",
      "--- *blasting region: 1011\n",
      "--- total time for blast 1011: 15.453700304031372\n",
      "-- processing region: 1013 with 146 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1013.fasta\n",
      "--- *blasting region: 1013\n",
      "--- total time for blast 1013: 9.72873044013977\n",
      "-- processing region: 1014 with 205 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1014.fasta\n",
      "--- *blasting region: 1014\n",
      "--- total time for blast 1014: 7.599663257598877\n",
      "-- processing region: 1015 with 65 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1015.fasta\n",
      "--- *blasting region: 1015\n",
      "--- total time for blast 1015: 6.57349157333374\n",
      "-- processing region: 1016 with 176 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1016.fasta\n",
      "--- *blasting region: 1016\n",
      "--- total time for blast 1016: 8.908676624298096\n",
      "-- processing region: 1017 with 117 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1017.fasta\n",
      "--- *blasting region: 1017\n",
      "--- total time for blast 1017: 5.363956689834595\n",
      "-- processing region: 1018 with 145 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1018.fasta\n",
      "--- *blasting region: 1018\n",
      "--- total time for blast 1018: 7.900723457336426\n",
      "-- processing region: 1019 with 204 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1019.fasta\n",
      "--- *blasting region: 1019\n",
      "--- total time for blast 1019: 10.617606401443481\n",
      "-- processing region: 102 with 267 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_102.fasta\n",
      "--- *blasting region: 102\n",
      "--- total time for blast 102: 10.332889080047607\n",
      "-- processing region: 1020 with 133 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1020.fasta\n",
      "--- *blasting region: 1020\n",
      "--- total time for blast 1020: 5.729698419570923\n",
      "-- processing region: 1021 with 173 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1021.fasta\n",
      "--- *blasting region: 1021\n",
      "--- total time for blast 1021: 7.0718114376068115\n",
      "-- processing region: 1022 with 196 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1022.fasta\n",
      "--- *blasting region: 1022\n",
      "--- total time for blast 1022: 9.670461654663086\n",
      "-- processing region: 1023 with 112 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1023.fasta\n",
      "--- *blasting region: 1023\n",
      "--- total time for blast 1023: 6.403309345245361\n",
      "-- processing region: 1024 with 87 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1024.fasta\n",
      "--- *blasting region: 1024\n",
      "--- total time for blast 1024: 15.543578147888184\n",
      "-- processing region: 1025 with 65 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1025.fasta\n",
      "--- *blasting region: 1025\n",
      "--- total time for blast 1025: 7.430537223815918\n",
      "-- processing region: 1026 with 130 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1026.fasta\n",
      "--- *blasting region: 1026\n",
      "--- total time for blast 1026: 8.06849980354309\n",
      "-- processing region: 1027 with 78 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1027.fasta\n",
      "--- *blasting region: 1027\n",
      "--- total time for blast 1027: 5.523866653442383\n",
      "-- processing region: 1028 with 102 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1028.fasta\n",
      "--- *blasting region: 1028\n",
      "--- total time for blast 1028: 7.035099267959595\n",
      "-- processing region: 1029 with 127 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1029.fasta\n",
      "--- *blasting region: 1029\n",
      "--- total time for blast 1029: 8.685464143753052\n",
      "-- processing region: 103 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_103.fasta\n",
      "--- *blasting region: 103\n",
      "--- total time for blast 103: 7.349891901016235\n",
      "-- processing region: 1030 with 301 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1030.fasta\n",
      "--- *blasting region: 1030\n",
      "--- total time for blast 1030: 13.000713586807251\n",
      "-- processing region: 1031 with 172 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1031.fasta\n",
      "--- *blasting region: 1031\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 1031: 8.029948949813843\n",
      "-- processing region: 1032 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1032.fasta\n",
      "--- *blasting region: 1032\n",
      "--- total time for blast 1032: 9.001835584640503\n",
      "-- processing region: 1033 with 244 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1033.fasta\n",
      "--- *blasting region: 1033\n",
      "--- total time for blast 1033: 7.4063403606414795\n",
      "-- processing region: 1034 with 224 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1034.fasta\n",
      "--- *blasting region: 1034\n",
      "--- total time for blast 1034: 8.648493766784668\n",
      "-- processing region: 1035 with 126 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1035.fasta\n",
      "--- *blasting region: 1035\n",
      "--- total time for blast 1035: 5.661897659301758\n",
      "-- processing region: 1036 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1036.fasta\n",
      "--- *blasting region: 1036\n",
      "--- total time for blast 1036: 10.246686935424805\n",
      "-- processing region: 1038 with 146 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1038.fasta\n",
      "--- *blasting region: 1038\n",
      "--- total time for blast 1038: 7.12569785118103\n",
      "-- processing region: 1039 with 111 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1039.fasta\n",
      "--- *blasting region: 1039\n",
      "--- total time for blast 1039: 5.636317014694214\n",
      "-- processing region: 104 with 201 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_104.fasta\n",
      "--- *blasting region: 104\n",
      "--- total time for blast 104: 8.036515235900879\n",
      "-- processing region: 1040 with 62 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1040.fasta\n",
      "--- *blasting region: 1040\n",
      "--- total time for blast 1040: 15.77501630783081\n",
      "-- processing region: 1041 with 106 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1041.fasta\n",
      "--- *blasting region: 1041\n",
      "--- total time for blast 1041: 15.240931749343872\n",
      "-- processing region: 1044 with 183 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1044.fasta\n",
      "--- *blasting region: 1044\n",
      "--- total time for blast 1044: 15.918424367904663\n",
      "-- processing region: 1045 with 172 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1045.fasta\n",
      "--- *blasting region: 1045\n",
      "--- total time for blast 1045: 8.65943431854248\n",
      "-- processing region: 1046 with 221 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1046.fasta\n",
      "--- *blasting region: 1046\n",
      "--- total time for blast 1046: 19.03213334083557\n",
      "-- processing region: 1047 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1047.fasta\n",
      "--- *blasting region: 1047\n",
      "--- total time for blast 1047: 10.28264832496643\n",
      "-- processing region: 1048 with 219 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1048.fasta\n",
      "--- *blasting region: 1048\n",
      "--- total time for blast 1048: 9.704658031463623\n",
      "-- processing region: 1049 with 181 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1049.fasta\n",
      "--- *blasting region: 1049\n",
      "--- total time for blast 1049: 11.783076524734497\n",
      "-- processing region: 105 with 271 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_105.fasta\n",
      "--- *blasting region: 105\n",
      "--- total time for blast 105: 9.628975868225098\n",
      "-- processing region: 1050 with 190 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1050.fasta\n",
      "--- *blasting region: 1050\n",
      "--- total time for blast 1050: 6.925901889801025\n",
      "-- processing region: 1051 with 147 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1051.fasta\n",
      "--- *blasting region: 1051\n",
      "--- total time for blast 1051: 8.711516618728638\n",
      "-- processing region: 1053 with 96 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1053.fasta\n",
      "--- *blasting region: 1053\n",
      "--- total time for blast 1053: 5.474930286407471\n",
      "-- processing region: 1054 with 171 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1054.fasta\n",
      "--- *blasting region: 1054\n",
      "--- total time for blast 1054: 10.482383966445923\n",
      "-- processing region: 1055 with 150 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1055.fasta\n",
      "--- *blasting region: 1055\n",
      "--- total time for blast 1055: 9.32740044593811\n",
      "-- processing region: 1056 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1056.fasta\n",
      "--- *blasting region: 1056\n",
      "--- total time for blast 1056: 8.206006526947021\n",
      "-- processing region: 1057 with 193 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1057.fasta\n",
      "--- *blasting region: 1057\n",
      "--- total time for blast 1057: 12.34124755859375\n",
      "-- processing region: 1058 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1058.fasta\n",
      "--- *blasting region: 1058\n",
      "--- total time for blast 1058: 10.148186683654785\n",
      "-- processing region: 1059 with 221 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1059.fasta\n",
      "--- *blasting region: 1059\n",
      "--- total time for blast 1059: 8.994704484939575\n",
      "-- processing region: 106 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_106.fasta\n",
      "--- *blasting region: 106\n",
      "--- total time for blast 106: 12.120802164077759\n",
      "-- processing region: 1060 with 122 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1060.fasta\n",
      "--- *blasting region: 1060\n",
      "--- total time for blast 1060: 24.98292088508606\n",
      "-- processing region: 1062 with 168 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_1062.fasta\n",
      "--- *blasting region: 1062\n",
      "--- total time for blast 1062: 8.48129916191101\n",
      "-- processing region: 107 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_107.fasta\n",
      "--- *blasting region: 107\n",
      "--- total time for blast 107: 8.968770503997803\n",
      "-- processing region: 108 with 152 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_108.fasta\n",
      "--- *blasting region: 108\n",
      "--- total time for blast 108: 6.616482734680176\n",
      "-- processing region: 109 with 263 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_109.fasta\n",
      "--- *blasting region: 109\n",
      "--- total time for blast 109: 10.418758392333984\n",
      "-- processing region: 11 with 156 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_11.fasta\n",
      "--- *blasting region: 11\n",
      "--- total time for blast 11: 9.79556131362915\n",
      "-- processing region: 110 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_110.fasta\n",
      "--- *blasting region: 110\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 110: 9.135655403137207\n",
      "-- processing region: 111 with 97 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_111.fasta\n",
      "--- *blasting region: 111\n",
      "--- total time for blast 111: 5.679330587387085\n",
      "-- processing region: 112 with 205 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_112.fasta\n",
      "--- *blasting region: 112\n",
      "--- total time for blast 112: 9.724242448806763\n",
      "-- processing region: 113 with 91 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_113.fasta\n",
      "--- *blasting region: 113\n",
      "--- total time for blast 113: 6.779293537139893\n",
      "-- processing region: 114 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_114.fasta\n",
      "--- *blasting region: 114\n",
      "--- total time for blast 114: 8.572869062423706\n",
      "-- processing region: 115 with 242 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_115.fasta\n",
      "--- *blasting region: 115\n",
      "--- total time for blast 115: 9.180439949035645\n",
      "-- processing region: 116 with 296 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_116.fasta\n",
      "--- *blasting region: 116\n",
      "--- total time for blast 116: 11.376476526260376\n",
      "-- processing region: 117 with 175 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_117.fasta\n",
      "--- *blasting region: 117\n",
      "--- total time for blast 117: 7.897277355194092\n",
      "-- processing region: 118 with 111 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_118.fasta\n",
      "--- *blasting region: 118\n",
      "--- total time for blast 118: 12.30662202835083\n",
      "-- processing region: 119 with 203 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_119.fasta\n",
      "--- *blasting region: 119\n",
      "--- total time for blast 119: 7.286406755447388\n",
      "-- processing region: 12 with 107 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_12.fasta\n",
      "--- *blasting region: 12\n",
      "--- total time for blast 12: 7.196396827697754\n",
      "-- processing region: 120 with 279 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_120.fasta\n",
      "--- *blasting region: 120\n",
      "--- total time for blast 120: 10.26378870010376\n",
      "-- processing region: 121 with 223 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_121.fasta\n",
      "--- *blasting region: 121\n",
      "--- total time for blast 121: 8.214184522628784\n",
      "-- processing region: 122 with 231 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_122.fasta\n",
      "--- *blasting region: 122\n",
      "--- total time for blast 122: 7.721517562866211\n",
      "-- processing region: 123 with 102 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_123.fasta\n",
      "--- *blasting region: 123\n",
      "--- total time for blast 123: 12.95364785194397\n",
      "-- processing region: 124 with 255 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_124.fasta\n",
      "--- *blasting region: 124\n",
      "--- total time for blast 124: 9.69922399520874\n",
      "-- processing region: 125 with 191 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_125.fasta\n",
      "--- *blasting region: 125\n",
      "--- total time for blast 125: 9.581136465072632\n",
      "-- processing region: 126 with 318 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_126.fasta\n",
      "--- *blasting region: 126\n",
      "--- total time for blast 126: 11.170319080352783\n",
      "-- processing region: 127 with 224 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_127.fasta\n",
      "--- *blasting region: 127\n",
      "--- total time for blast 127: 9.36866044998169\n",
      "-- processing region: 128 with 167 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_128.fasta\n",
      "--- *blasting region: 128\n",
      "--- total time for blast 128: 15.056520700454712\n",
      "-- processing region: 129 with 147 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_129.fasta\n",
      "--- *blasting region: 129\n",
      "--- total time for blast 129: 7.921657562255859\n",
      "-- processing region: 13 with 291 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_13.fasta\n",
      "--- *blasting region: 13\n",
      "--- total time for blast 13: 9.142621517181396\n",
      "-- processing region: 130 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_130.fasta\n",
      "--- *blasting region: 130\n",
      "--- total time for blast 130: 11.293930768966675\n",
      "-- processing region: 131 with 261 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_131.fasta\n",
      "--- *blasting region: 131\n",
      "--- total time for blast 131: 10.621145248413086\n",
      "-- processing region: 132 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_132.fasta\n",
      "--- *blasting region: 132\n",
      "--- total time for blast 132: 12.22451400756836\n",
      "-- processing region: 133 with 159 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_133.fasta\n",
      "--- *blasting region: 133\n",
      "--- total time for blast 133: 10.892491817474365\n",
      "-- processing region: 134 with 140 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_134.fasta\n",
      "--- *blasting region: 134\n",
      "--- total time for blast 134: 14.78148603439331\n",
      "-- processing region: 135 with 232 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_135.fasta\n",
      "--- *blasting region: 135\n",
      "--- total time for blast 135: 11.31692099571228\n",
      "-- processing region: 136 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_136.fasta\n",
      "--- *blasting region: 136\n",
      "--- total time for blast 136: 7.853988170623779\n",
      "-- processing region: 137 with 313 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_137.fasta\n",
      "--- *blasting region: 137\n",
      "--- total time for blast 137: 10.756011247634888\n",
      "-- processing region: 138 with 199 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_138.fasta\n",
      "--- *blasting region: 138\n",
      "--- total time for blast 138: 7.260330677032471\n",
      "-- processing region: 14 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_14.fasta\n",
      "--- *blasting region: 14\n",
      "--- total time for blast 14: 11.815080881118774\n",
      "-- processing region: 140 with 226 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_140.fasta\n",
      "--- *blasting region: 140\n",
      "--- total time for blast 140: 8.259139060974121\n",
      "-- processing region: 141 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_141.fasta\n",
      "--- *blasting region: 141\n",
      "--- total time for blast 141: 12.17206597328186\n",
      "-- processing region: 142 with 308 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_142.fasta\n",
      "--- *blasting region: 142\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 142: 11.665688753128052\n",
      "-- processing region: 143 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_143.fasta\n",
      "--- *blasting region: 143\n",
      "--- total time for blast 143: 8.781205415725708\n",
      "-- processing region: 144 with 305 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_144.fasta\n",
      "--- *blasting region: 144\n",
      "--- total time for blast 144: 13.12364912033081\n",
      "-- processing region: 145 with 302 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_145.fasta\n",
      "--- *blasting region: 145\n",
      "--- total time for blast 145: 11.611449718475342\n",
      "-- processing region: 146 with 279 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_146.fasta\n",
      "--- *blasting region: 146\n",
      "--- total time for blast 146: 13.41443419456482\n",
      "-- processing region: 147 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_147.fasta\n",
      "--- *blasting region: 147\n",
      "--- total time for blast 147: 9.079896211624146\n",
      "-- processing region: 148 with 260 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_148.fasta\n",
      "--- *blasting region: 148\n",
      "--- total time for blast 148: 9.009891748428345\n",
      "-- processing region: 15 with 279 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_15.fasta\n",
      "--- *blasting region: 15\n",
      "--- total time for blast 15: 10.892026901245117\n",
      "-- processing region: 150 with 254 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_150.fasta\n",
      "--- *blasting region: 150\n",
      "--- total time for blast 150: 11.0001962184906\n",
      "-- processing region: 151 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_151.fasta\n",
      "--- *blasting region: 151\n",
      "--- total time for blast 151: 10.779025554656982\n",
      "-- processing region: 153 with 185 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_153.fasta\n",
      "--- *blasting region: 153\n",
      "--- total time for blast 153: 8.5141019821167\n",
      "-- processing region: 154 with 159 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_154.fasta\n",
      "--- *blasting region: 154\n",
      "--- total time for blast 154: 9.861964702606201\n",
      "-- processing region: 155 with 258 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_155.fasta\n",
      "--- *blasting region: 155\n",
      "--- total time for blast 155: 11.343749046325684\n",
      "-- processing region: 156 with 144 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_156.fasta\n",
      "--- *blasting region: 156\n",
      "--- total time for blast 156: 13.015495777130127\n",
      "-- processing region: 157 with 199 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_157.fasta\n",
      "--- *blasting region: 157\n",
      "--- total time for blast 157: 8.474657773971558\n",
      "-- processing region: 158 with 127 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_158.fasta\n",
      "--- *blasting region: 158\n",
      "--- total time for blast 158: 11.523953914642334\n",
      "-- processing region: 159 with 270 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_159.fasta\n",
      "--- *blasting region: 159\n",
      "--- total time for blast 159: 9.749795198440552\n",
      "-- processing region: 16 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_16.fasta\n",
      "--- *blasting region: 16\n",
      "--- total time for blast 16: 9.18277621269226\n",
      "-- processing region: 160 with 141 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_160.fasta\n",
      "--- *blasting region: 160\n",
      "--- total time for blast 160: 12.557095527648926\n",
      "-- processing region: 161 with 168 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_161.fasta\n",
      "--- *blasting region: 161\n",
      "--- total time for blast 161: 10.03776216506958\n",
      "-- processing region: 162 with 221 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_162.fasta\n",
      "--- *blasting region: 162\n",
      "--- total time for blast 162: 8.166081666946411\n",
      "-- processing region: 163 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_163.fasta\n",
      "--- *blasting region: 163\n",
      "--- total time for blast 163: 13.7727792263031\n",
      "-- processing region: 164 with 288 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_164.fasta\n",
      "--- *blasting region: 164\n",
      "--- total time for blast 164: 8.79178786277771\n",
      "-- processing region: 165 with 225 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_165.fasta\n",
      "--- *blasting region: 165\n",
      "--- total time for blast 165: 9.595146656036377\n",
      "-- processing region: 166 with 103 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_166.fasta\n",
      "--- *blasting region: 166\n",
      "--- total time for blast 166: 5.4295361042022705\n",
      "-- processing region: 167 with 164 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_167.fasta\n",
      "--- *blasting region: 167\n",
      "--- total time for blast 167: 8.87955117225647\n",
      "-- processing region: 168 with 177 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_168.fasta\n",
      "--- *blasting region: 168\n",
      "--- total time for blast 168: 6.169431447982788\n",
      "-- processing region: 169 with 140 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_169.fasta\n",
      "--- *blasting region: 169\n",
      "--- total time for blast 169: 8.689609289169312\n",
      "-- processing region: 17 with 276 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_17.fasta\n",
      "--- *blasting region: 17\n",
      "--- total time for blast 17: 13.935815334320068\n",
      "-- processing region: 170 with 86 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_170.fasta\n",
      "--- *blasting region: 170\n",
      "--- total time for blast 170: 11.92388105392456\n",
      "-- processing region: 171 with 139 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_171.fasta\n",
      "--- *blasting region: 171\n",
      "--- total time for blast 171: 6.724575757980347\n",
      "-- processing region: 172 with 222 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_172.fasta\n",
      "--- *blasting region: 172\n",
      "--- total time for blast 172: 8.904950618743896\n",
      "-- processing region: 173 with 225 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_173.fasta\n",
      "--- *blasting region: 173\n",
      "--- total time for blast 173: 8.052997589111328\n",
      "-- processing region: 174 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_174.fasta\n",
      "--- *blasting region: 174\n",
      "--- total time for blast 174: 7.3334619998931885\n",
      "-- processing region: 175 with 210 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_175.fasta\n",
      "--- *blasting region: 175\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 175: 11.393676996231079\n",
      "-- processing region: 176 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_176.fasta\n",
      "--- *blasting region: 176\n",
      "--- total time for blast 176: 8.82885456085205\n",
      "-- processing region: 177 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_177.fasta\n",
      "--- *blasting region: 177\n",
      "--- total time for blast 177: 13.503256797790527\n",
      "-- processing region: 178 with 261 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_178.fasta\n",
      "--- *blasting region: 178\n",
      "--- total time for blast 178: 11.63425064086914\n",
      "-- processing region: 179 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_179.fasta\n",
      "--- *blasting region: 179\n",
      "--- total time for blast 179: 7.740273952484131\n",
      "-- processing region: 18 with 109 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_18.fasta\n",
      "--- *blasting region: 18\n",
      "--- total time for blast 18: 8.798553466796875\n",
      "-- processing region: 180 with 136 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_180.fasta\n",
      "--- *blasting region: 180\n",
      "--- total time for blast 180: 12.22584080696106\n",
      "-- processing region: 181 with 181 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_181.fasta\n",
      "--- *blasting region: 181\n",
      "--- total time for blast 181: 10.692665576934814\n",
      "-- processing region: 182 with 99 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_182.fasta\n",
      "--- *blasting region: 182\n",
      "--- total time for blast 182: 14.451773166656494\n",
      "-- processing region: 183 with 110 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_183.fasta\n",
      "--- *blasting region: 183\n",
      "--- total time for blast 183: 5.297971963882446\n",
      "-- processing region: 184 with 126 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_184.fasta\n",
      "--- *blasting region: 184\n",
      "--- total time for blast 184: 4.707038640975952\n",
      "-- processing region: 185 with 290 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_185.fasta\n",
      "--- *blasting region: 185\n",
      "--- total time for blast 185: 11.302133560180664\n",
      "-- processing region: 186 with 182 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_186.fasta\n",
      "--- *blasting region: 186\n",
      "--- total time for blast 186: 9.312089681625366\n",
      "-- processing region: 187 with 216 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_187.fasta\n",
      "--- *blasting region: 187\n",
      "--- total time for blast 187: 8.866119623184204\n",
      "-- processing region: 188 with 114 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_188.fasta\n",
      "--- *blasting region: 188\n",
      "--- total time for blast 188: 6.75403094291687\n",
      "-- processing region: 19 with 111 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_19.fasta\n",
      "--- *blasting region: 19\n",
      "--- total time for blast 19: 5.367564916610718\n",
      "-- processing region: 190 with 227 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_190.fasta\n",
      "--- *blasting region: 190\n",
      "--- total time for blast 190: 14.566322565078735\n",
      "-- processing region: 191 with 142 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_191.fasta\n",
      "--- *blasting region: 191\n",
      "--- total time for blast 191: 6.585642099380493\n",
      "-- processing region: 192 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_192.fasta\n",
      "--- *blasting region: 192\n",
      "--- total time for blast 192: 8.721983671188354\n",
      "-- processing region: 193 with 275 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_193.fasta\n",
      "--- *blasting region: 193\n",
      "--- total time for blast 193: 9.96670126914978\n",
      "-- processing region: 195 with 186 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_195.fasta\n",
      "--- *blasting region: 195\n",
      "--- total time for blast 195: 7.97894287109375\n",
      "-- processing region: 196 with 115 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_196.fasta\n",
      "--- *blasting region: 196\n",
      "--- total time for blast 196: 14.643419981002808\n",
      "-- processing region: 197 with 115 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_197.fasta\n",
      "--- *blasting region: 197\n",
      "--- total time for blast 197: 6.1818602085113525\n",
      "-- processing region: 198 with 289 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_198.fasta\n",
      "--- *blasting region: 198\n",
      "--- total time for blast 198: 9.931609630584717\n",
      "-- processing region: 199 with 175 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_199.fasta\n",
      "--- *blasting region: 199\n",
      "--- total time for blast 199: 11.273254156112671\n",
      "-- processing region: 2 with 180 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_2.fasta\n",
      "--- *blasting region: 2\n",
      "--- total time for blast 2: 7.762182950973511\n",
      "-- processing region: 20 with 212 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_20.fasta\n",
      "--- *blasting region: 20\n",
      "--- total time for blast 20: 7.716747283935547\n",
      "-- processing region: 200 with 258 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_200.fasta\n",
      "--- *blasting region: 200\n",
      "--- total time for blast 200: 8.316734313964844\n",
      "-- processing region: 201 with 169 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_201.fasta\n",
      "--- *blasting region: 201\n",
      "--- total time for blast 201: 8.173081398010254\n",
      "-- processing region: 202 with 180 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_202.fasta\n",
      "--- *blasting region: 202\n",
      "--- total time for blast 202: 19.79953670501709\n",
      "-- processing region: 203 with 197 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_203.fasta\n",
      "--- *blasting region: 203\n",
      "--- total time for blast 203: 9.735998153686523\n",
      "-- processing region: 204 with 314 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_204.fasta\n",
      "--- *blasting region: 204\n",
      "--- total time for blast 204: 13.79518175125122\n",
      "-- processing region: 205 with 264 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_205.fasta\n",
      "--- *blasting region: 205\n",
      "--- total time for blast 205: 9.168240308761597\n",
      "-- processing region: 206 with 198 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_206.fasta\n",
      "--- *blasting region: 206\n",
      "--- total time for blast 206: 9.62582278251648\n",
      "-- processing region: 207 with 294 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_207.fasta\n",
      "--- *blasting region: 207\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 207: 11.939893007278442\n",
      "-- processing region: 208 with 191 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_208.fasta\n",
      "--- *blasting region: 208\n",
      "--- total time for blast 208: 8.098031759262085\n",
      "-- processing region: 209 with 290 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_209.fasta\n",
      "--- *blasting region: 209\n",
      "--- total time for blast 209: 10.910980463027954\n",
      "-- processing region: 21 with 135 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_21.fasta\n",
      "--- *blasting region: 21\n",
      "--- total time for blast 21: 9.001704692840576\n",
      "-- processing region: 210 with 294 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_210.fasta\n",
      "--- *blasting region: 210\n",
      "--- total time for blast 210: 9.48147964477539\n",
      "-- processing region: 211 with 184 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_211.fasta\n",
      "--- *blasting region: 211\n",
      "--- total time for blast 211: 8.259151458740234\n",
      "-- processing region: 212 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_212.fasta\n",
      "--- *blasting region: 212\n",
      "--- total time for blast 212: 11.015671968460083\n",
      "-- processing region: 213 with 241 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_213.fasta\n",
      "--- *blasting region: 213\n",
      "--- total time for blast 213: 10.429347276687622\n",
      "-- processing region: 214 with 180 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_214.fasta\n",
      "--- *blasting region: 214\n",
      "--- total time for blast 214: 9.629170894622803\n",
      "-- processing region: 215 with 122 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_215.fasta\n",
      "--- *blasting region: 215\n",
      "--- total time for blast 215: 6.919268369674683\n",
      "-- processing region: 218 with 89 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_218.fasta\n",
      "--- *blasting region: 218\n",
      "--- total time for blast 218: 4.546386003494263\n",
      "-- processing region: 219 with 164 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_219.fasta\n",
      "--- *blasting region: 219\n",
      "--- total time for blast 219: 9.69650411605835\n",
      "-- processing region: 22 with 229 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_22.fasta\n",
      "--- *blasting region: 22\n",
      "--- total time for blast 22: 9.386967897415161\n",
      "-- processing region: 220 with 298 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_220.fasta\n",
      "--- *blasting region: 220\n",
      "--- total time for blast 220: 10.66829800605774\n",
      "-- processing region: 221 with 203 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_221.fasta\n",
      "--- *blasting region: 221\n",
      "--- total time for blast 221: 8.26215386390686\n",
      "-- processing region: 222 with 296 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_222.fasta\n",
      "--- *blasting region: 222\n",
      "--- total time for blast 222: 8.949665784835815\n",
      "-- processing region: 223 with 158 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_223.fasta\n",
      "--- *blasting region: 223\n",
      "--- total time for blast 223: 9.23788595199585\n",
      "-- processing region: 224 with 104 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_224.fasta\n",
      "--- *blasting region: 224\n",
      "--- total time for blast 224: 13.538081645965576\n",
      "-- processing region: 225 with 84 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_225.fasta\n",
      "--- *blasting region: 225\n",
      "--- total time for blast 225: 16.08891272544861\n",
      "-- processing region: 226 with 184 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_226.fasta\n",
      "--- *blasting region: 226\n",
      "--- total time for blast 226: 7.377473831176758\n",
      "-- processing region: 227 with 119 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_227.fasta\n",
      "--- *blasting region: 227\n",
      "--- total time for blast 227: 7.856329679489136\n",
      "-- processing region: 228 with 84 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_228.fasta\n",
      "--- *blasting region: 228\n",
      "--- total time for blast 228: 5.290311336517334\n",
      "-- processing region: 229 with 244 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_229.fasta\n",
      "--- *blasting region: 229\n",
      "--- total time for blast 229: 8.55683708190918\n",
      "-- processing region: 23 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_23.fasta\n",
      "--- *blasting region: 23\n",
      "--- total time for blast 23: 10.132548809051514\n",
      "-- processing region: 230 with 264 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_230.fasta\n",
      "--- *blasting region: 230\n",
      "--- total time for blast 230: 8.291733741760254\n",
      "-- processing region: 231 with 143 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_231.fasta\n",
      "--- *blasting region: 231\n",
      "--- total time for blast 231: 7.348968029022217\n",
      "-- processing region: 232 with 110 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_232.fasta\n",
      "--- *blasting region: 232\n",
      "--- total time for blast 232: 10.25075387954712\n",
      "-- processing region: 233 with 223 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_233.fasta\n",
      "--- *blasting region: 233\n",
      "--- total time for blast 233: 9.380985736846924\n",
      "-- processing region: 234 with 194 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_234.fasta\n",
      "--- *blasting region: 234\n",
      "--- total time for blast 234: 9.281599998474121\n",
      "-- processing region: 235 with 151 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_235.fasta\n",
      "--- *blasting region: 235\n",
      "--- total time for blast 235: 8.660450458526611\n",
      "-- processing region: 236 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_236.fasta\n",
      "--- *blasting region: 236\n",
      "--- total time for blast 236: 8.91947603225708\n",
      "-- processing region: 237 with 108 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_237.fasta\n",
      "--- *blasting region: 237\n",
      "--- total time for blast 237: 20.708423614501953\n",
      "-- processing region: 238 with 256 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_238.fasta\n",
      "--- *blasting region: 238\n",
      "--- total time for blast 238: 9.556719303131104\n",
      "-- processing region: 239 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_239.fasta\n",
      "--- *blasting region: 239\n",
      "--- total time for blast 239: 12.88259482383728\n",
      "-- processing region: 24 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_24.fasta\n",
      "--- *blasting region: 24\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 24: 13.324691772460938\n",
      "-- processing region: 240 with 139 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_240.fasta\n",
      "--- *blasting region: 240\n",
      "--- total time for blast 240: 7.360535144805908\n",
      "-- processing region: 242 with 299 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_242.fasta\n",
      "--- *blasting region: 242\n",
      "--- total time for blast 242: 11.148815393447876\n",
      "-- processing region: 243 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_243.fasta\n",
      "--- *blasting region: 243\n",
      "--- total time for blast 243: 8.821013450622559\n",
      "-- processing region: 244 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_244.fasta\n",
      "--- *blasting region: 244\n",
      "--- total time for blast 244: 8.885164976119995\n",
      "-- processing region: 245 with 154 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_245.fasta\n",
      "--- *blasting region: 245\n",
      "--- total time for blast 245: 8.268465042114258\n",
      "-- processing region: 246 with 151 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_246.fasta\n",
      "--- *blasting region: 246\n",
      "--- total time for blast 246: 8.824506998062134\n",
      "-- processing region: 247 with 228 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_247.fasta\n",
      "--- *blasting region: 247\n",
      "--- total time for blast 247: 8.763963222503662\n",
      "-- processing region: 248 with 135 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_248.fasta\n",
      "--- *blasting region: 248\n",
      "--- total time for blast 248: 10.599894046783447\n",
      "-- processing region: 249 with 211 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_249.fasta\n",
      "--- *blasting region: 249\n",
      "--- total time for blast 249: 7.33185076713562\n",
      "-- processing region: 25 with 249 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_25.fasta\n",
      "--- *blasting region: 25\n",
      "--- total time for blast 25: 10.153562307357788\n",
      "-- processing region: 250 with 100 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_250.fasta\n",
      "--- *blasting region: 250\n",
      "--- total time for blast 250: 5.449126482009888\n",
      "-- processing region: 251 with 207 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_251.fasta\n",
      "--- *blasting region: 251\n",
      "--- total time for blast 251: 9.566157341003418\n",
      "-- processing region: 252 with 137 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_252.fasta\n",
      "--- *blasting region: 252\n",
      "--- total time for blast 252: 7.460524320602417\n",
      "-- processing region: 253 with 282 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_253.fasta\n",
      "--- *blasting region: 253\n",
      "--- total time for blast 253: 10.006811380386353\n",
      "-- processing region: 254 with 139 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_254.fasta\n",
      "--- *blasting region: 254\n",
      "--- total time for blast 254: 11.527582168579102\n",
      "-- processing region: 255 with 119 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_255.fasta\n",
      "--- *blasting region: 255\n",
      "--- total time for blast 255: 10.365235567092896\n",
      "-- processing region: 256 with 230 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_256.fasta\n",
      "--- *blasting region: 256\n",
      "--- total time for blast 256: 9.293922424316406\n",
      "-- processing region: 257 with 231 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_257.fasta\n",
      "--- *blasting region: 257\n",
      "--- total time for blast 257: 20.09063220024109\n",
      "-- processing region: 258 with 300 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_258.fasta\n",
      "--- *blasting region: 258\n",
      "--- total time for blast 258: 10.53184723854065\n",
      "-- processing region: 259 with 207 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_259.fasta\n",
      "--- *blasting region: 259\n",
      "--- total time for blast 259: 7.850844144821167\n",
      "-- processing region: 26 with 204 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_26.fasta\n",
      "--- *blasting region: 26\n",
      "--- total time for blast 26: 8.176047086715698\n",
      "-- processing region: 260 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_260.fasta\n",
      "--- *blasting region: 260\n",
      "--- total time for blast 260: 11.512574672698975\n",
      "-- processing region: 261 with 137 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_261.fasta\n",
      "--- *blasting region: 261\n",
      "--- total time for blast 261: 5.759337902069092\n",
      "-- processing region: 262 with 101 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_262.fasta\n",
      "--- *blasting region: 262\n",
      "--- total time for blast 262: 5.255596399307251\n",
      "-- processing region: 263 with 183 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_263.fasta\n",
      "--- *blasting region: 263\n",
      "--- total time for blast 263: 8.110872268676758\n",
      "-- processing region: 264 with 186 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_264.fasta\n",
      "--- *blasting region: 264\n",
      "--- total time for blast 264: 7.724753141403198\n",
      "-- processing region: 266 with 207 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_266.fasta\n",
      "--- *blasting region: 266\n",
      "--- total time for blast 266: 8.277440071105957\n",
      "-- processing region: 267 with 309 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_267.fasta\n",
      "--- *blasting region: 267\n",
      "--- total time for blast 267: 11.010189771652222\n",
      "-- processing region: 268 with 299 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_268.fasta\n",
      "--- *blasting region: 268\n",
      "--- total time for blast 268: 13.976001501083374\n",
      "-- processing region: 269 with 306 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_269.fasta\n",
      "--- *blasting region: 269\n",
      "--- total time for blast 269: 12.28100037574768\n",
      "-- processing region: 27 with 195 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_27.fasta\n",
      "--- *blasting region: 27\n",
      "--- total time for blast 27: 10.775025844573975\n",
      "-- processing region: 270 with 292 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_270.fasta\n",
      "--- *blasting region: 270\n",
      "--- total time for blast 270: 11.30746841430664\n",
      "-- processing region: 271 with 312 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_271.fasta\n",
      "--- *blasting region: 271\n",
      "--- total time for blast 271: 14.010851621627808\n",
      "-- processing region: 272 with 325 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_272.fasta\n",
      "--- *blasting region: 272\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 272: 10.807867050170898\n",
      "-- processing region: 273 with 300 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_273.fasta\n",
      "--- *blasting region: 273\n",
      "--- total time for blast 273: 9.96724796295166\n",
      "-- processing region: 274 with 123 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_274.fasta\n",
      "--- *blasting region: 274\n",
      "--- total time for blast 274: 12.881633758544922\n",
      "-- processing region: 276 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_276.fasta\n",
      "--- *blasting region: 276\n",
      "--- total time for blast 276: 9.810691356658936\n",
      "-- processing region: 277 with 319 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_277.fasta\n",
      "--- *blasting region: 277\n",
      "--- total time for blast 277: 12.691731929779053\n",
      "-- processing region: 278 with 156 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_278.fasta\n",
      "--- *blasting region: 278\n",
      "--- total time for blast 278: 5.47007417678833\n",
      "-- processing region: 279 with 255 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_279.fasta\n",
      "--- *blasting region: 279\n",
      "--- total time for blast 279: 7.100456714630127\n",
      "-- processing region: 28 with 198 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_28.fasta\n",
      "--- *blasting region: 28\n",
      "--- total time for blast 28: 12.187077045440674\n",
      "-- processing region: 281 with 187 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_281.fasta\n",
      "--- *blasting region: 281\n",
      "--- total time for blast 281: 8.75852346420288\n",
      "-- processing region: 282 with 141 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_282.fasta\n",
      "--- *blasting region: 282\n",
      "--- total time for blast 282: 8.221123218536377\n",
      "-- processing region: 283 with 119 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_283.fasta\n",
      "--- *blasting region: 283\n",
      "--- total time for blast 283: 6.548451662063599\n",
      "-- processing region: 285 with 93 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_285.fasta\n",
      "--- *blasting region: 285\n",
      "--- total time for blast 285: 6.7750701904296875\n",
      "-- processing region: 286 with 188 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_286.fasta\n",
      "--- *blasting region: 286\n",
      "--- total time for blast 286: 17.185800075531006\n",
      "-- processing region: 287 with 134 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_287.fasta\n",
      "--- *blasting region: 287\n",
      "--- total time for blast 287: 10.943434238433838\n",
      "-- processing region: 288 with 217 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_288.fasta\n",
      "--- *blasting region: 288\n",
      "--- total time for blast 288: 8.379223585128784\n",
      "-- processing region: 289 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_289.fasta\n",
      "--- *blasting region: 289\n",
      "--- total time for blast 289: 10.09807276725769\n",
      "-- processing region: 29 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_29.fasta\n",
      "--- *blasting region: 29\n",
      "--- total time for blast 29: 10.444915294647217\n",
      "-- processing region: 290 with 103 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_290.fasta\n",
      "--- *blasting region: 290\n",
      "--- total time for blast 290: 5.685251712799072\n",
      "-- processing region: 291 with 229 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_291.fasta\n",
      "--- *blasting region: 291\n",
      "--- total time for blast 291: 9.628589630126953\n",
      "-- processing region: 292 with 316 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_292.fasta\n",
      "--- *blasting region: 292\n",
      "--- total time for blast 292: 10.24602222442627\n",
      "-- processing region: 293 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_293.fasta\n",
      "--- *blasting region: 293\n",
      "--- total time for blast 293: 10.290968179702759\n",
      "-- processing region: 294 with 218 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_294.fasta\n",
      "--- *blasting region: 294\n",
      "--- total time for blast 294: 10.168878078460693\n",
      "-- processing region: 295 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_295.fasta\n",
      "--- *blasting region: 295\n",
      "--- total time for blast 295: 10.906805276870728\n",
      "-- processing region: 296 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_296.fasta\n",
      "--- *blasting region: 296\n",
      "--- total time for blast 296: 10.577689409255981\n",
      "-- processing region: 297 with 236 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_297.fasta\n",
      "--- *blasting region: 297\n",
      "--- total time for blast 297: 8.68364143371582\n",
      "-- processing region: 298 with 151 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_298.fasta\n",
      "--- *blasting region: 298\n",
      "--- total time for blast 298: 7.724764347076416\n",
      "-- processing region: 299 with 212 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_299.fasta\n",
      "--- *blasting region: 299\n",
      "--- total time for blast 299: 13.105026483535767\n",
      "-- processing region: 3 with 77 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_3.fasta\n",
      "--- *blasting region: 3\n",
      "--- total time for blast 3: 15.241503477096558\n",
      "-- processing region: 30 with 245 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_30.fasta\n",
      "--- *blasting region: 30\n",
      "--- total time for blast 30: 8.887916088104248\n",
      "-- processing region: 300 with 174 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_300.fasta\n",
      "--- *blasting region: 300\n",
      "--- total time for blast 300: 13.011074304580688\n",
      "-- processing region: 301 with 119 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_301.fasta\n",
      "--- *blasting region: 301\n",
      "--- total time for blast 301: 5.616628646850586\n",
      "-- processing region: 302 with 115 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_302.fasta\n",
      "--- *blasting region: 302\n",
      "--- total time for blast 302: 6.891028881072998\n",
      "-- processing region: 303 with 184 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_303.fasta\n",
      "--- *blasting region: 303\n",
      "--- total time for blast 303: 7.173912763595581\n",
      "-- processing region: 304 with 130 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_304.fasta\n",
      "--- *blasting region: 304\n",
      "--- total time for blast 304: 9.274750232696533\n",
      "-- processing region: 305 with 222 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_305.fasta\n",
      "--- *blasting region: 305\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 305: 16.951761484146118\n",
      "-- processing region: 306 with 188 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_306.fasta\n",
      "--- *blasting region: 306\n",
      "--- total time for blast 306: 8.279724359512329\n",
      "-- processing region: 307 with 149 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_307.fasta\n",
      "--- *blasting region: 307\n",
      "--- total time for blast 307: 8.8944730758667\n",
      "-- processing region: 308 with 108 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_308.fasta\n",
      "--- *blasting region: 308\n",
      "--- total time for blast 308: 6.976195812225342\n",
      "-- processing region: 309 with 322 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_309.fasta\n",
      "--- *blasting region: 309\n",
      "--- total time for blast 309: 11.466540336608887\n",
      "-- processing region: 31 with 151 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_31.fasta\n",
      "--- *blasting region: 31\n",
      "--- total time for blast 31: 13.114034175872803\n",
      "-- processing region: 310 with 205 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_310.fasta\n",
      "--- *blasting region: 310\n",
      "--- total time for blast 310: 7.621358394622803\n",
      "-- processing region: 311 with 189 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_311.fasta\n",
      "--- *blasting region: 311\n",
      "--- total time for blast 311: 9.178832054138184\n",
      "-- processing region: 312 with 204 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_312.fasta\n",
      "--- *blasting region: 312\n",
      "--- total time for blast 312: 9.260154962539673\n",
      "-- processing region: 313 with 135 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_313.fasta\n",
      "--- *blasting region: 313\n",
      "--- total time for blast 313: 10.988182067871094\n",
      "-- processing region: 314 with 210 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_314.fasta\n",
      "--- *blasting region: 314\n",
      "--- total time for blast 314: 8.928902864456177\n",
      "-- processing region: 315 with 183 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_315.fasta\n",
      "--- *blasting region: 315\n",
      "--- total time for blast 315: 10.698307991027832\n",
      "-- processing region: 316 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_316.fasta\n",
      "--- *blasting region: 316\n",
      "--- total time for blast 316: 7.230385780334473\n",
      "-- processing region: 318 with 239 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_318.fasta\n",
      "--- *blasting region: 318\n",
      "--- total time for blast 318: 9.628296852111816\n",
      "-- processing region: 319 with 205 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_319.fasta\n",
      "--- *blasting region: 319\n",
      "--- total time for blast 319: 10.103896141052246\n",
      "-- processing region: 32 with 247 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_32.fasta\n",
      "--- *blasting region: 32\n",
      "--- total time for blast 32: 10.088324069976807\n",
      "-- processing region: 320 with 200 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_320.fasta\n",
      "--- *blasting region: 320\n",
      "--- total time for blast 320: 13.461782693862915\n",
      "-- processing region: 321 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_321.fasta\n",
      "--- *blasting region: 321\n",
      "--- total time for blast 321: 11.190022230148315\n",
      "-- processing region: 322 with 210 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_322.fasta\n",
      "--- *blasting region: 322\n",
      "--- total time for blast 322: 10.366721391677856\n",
      "-- processing region: 323 with 242 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_323.fasta\n",
      "--- *blasting region: 323\n",
      "--- total time for blast 323: 11.284285545349121\n",
      "-- processing region: 324 with 303 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_324.fasta\n",
      "--- *blasting region: 324\n",
      "--- total time for blast 324: 11.569869041442871\n",
      "-- processing region: 325 with 337 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_325.fasta\n",
      "--- *blasting region: 325\n",
      "--- total time for blast 325: 12.39237380027771\n",
      "-- processing region: 326 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_326.fasta\n",
      "--- *blasting region: 326\n",
      "--- total time for blast 326: 13.380067586898804\n",
      "-- processing region: 327 with 339 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_327.fasta\n",
      "--- *blasting region: 327\n",
      "--- total time for blast 327: 12.898899793624878\n",
      "-- processing region: 328 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_328.fasta\n",
      "--- *blasting region: 328\n",
      "--- total time for blast 328: 12.227107286453247\n",
      "-- processing region: 329 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_329.fasta\n",
      "--- *blasting region: 329\n",
      "--- total time for blast 329: 10.048370361328125\n",
      "-- processing region: 33 with 276 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_33.fasta\n",
      "--- *blasting region: 33\n",
      "--- total time for blast 33: 10.399320363998413\n",
      "-- processing region: 330 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_330.fasta\n",
      "--- *blasting region: 330\n",
      "--- total time for blast 330: 11.630343914031982\n",
      "-- processing region: 331 with 341 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_331.fasta\n",
      "--- *blasting region: 331\n",
      "--- total time for blast 331: 11.78801703453064\n",
      "-- processing region: 332 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_332.fasta\n",
      "--- *blasting region: 332\n",
      "--- total time for blast 332: 10.964892387390137\n",
      "-- processing region: 333 with 243 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_333.fasta\n",
      "--- *blasting region: 333\n",
      "--- total time for blast 333: 10.538295984268188\n",
      "-- processing region: 334 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_334.fasta\n",
      "--- *blasting region: 334\n",
      "--- total time for blast 334: 11.106563329696655\n",
      "-- processing region: 335 with 318 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_335.fasta\n",
      "--- *blasting region: 335\n",
      "--- total time for blast 335: 12.13804030418396\n",
      "-- processing region: 336 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_336.fasta\n",
      "--- *blasting region: 336\n",
      "--- total time for blast 336: 9.949410915374756\n",
      "-- processing region: 337 with 281 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_337.fasta\n",
      "--- *blasting region: 337\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 337: 12.539339542388916\n",
      "-- processing region: 338 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_338.fasta\n",
      "--- *blasting region: 338\n",
      "--- total time for blast 338: 10.421015739440918\n",
      "-- processing region: 339 with 288 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_339.fasta\n",
      "--- *blasting region: 339\n",
      "--- total time for blast 339: 12.66845989227295\n",
      "-- processing region: 34 with 294 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_34.fasta\n",
      "--- *blasting region: 34\n",
      "--- total time for blast 34: 12.026940107345581\n",
      "-- processing region: 340 with 249 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_340.fasta\n",
      "--- *blasting region: 340\n",
      "--- total time for blast 340: 8.271959781646729\n",
      "-- processing region: 342 with 258 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_342.fasta\n",
      "--- *blasting region: 342\n",
      "--- total time for blast 342: 10.828064918518066\n",
      "-- processing region: 343 with 219 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_343.fasta\n",
      "--- *blasting region: 343\n",
      "--- total time for blast 343: 9.576937437057495\n",
      "-- processing region: 344 with 261 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_344.fasta\n",
      "--- *blasting region: 344\n",
      "--- total time for blast 344: 10.082764863967896\n",
      "-- processing region: 345 with 150 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_345.fasta\n",
      "--- *blasting region: 345\n",
      "--- total time for blast 345: 5.844919681549072\n",
      "-- processing region: 346 with 182 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_346.fasta\n",
      "--- *blasting region: 346\n",
      "--- total time for blast 346: 7.205366611480713\n",
      "-- processing region: 347 with 53 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_347.fasta\n",
      "--- *blasting region: 347\n",
      "--- total time for blast 347: 4.376259088516235\n",
      "-- processing region: 348 with 139 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_348.fasta\n",
      "--- *blasting region: 348\n",
      "--- total time for blast 348: 6.933403968811035\n",
      "-- processing region: 349 with 188 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_349.fasta\n",
      "--- *blasting region: 349\n",
      "--- total time for blast 349: 7.968369245529175\n",
      "-- processing region: 35 with 310 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_35.fasta\n",
      "--- *blasting region: 35\n",
      "--- total time for blast 35: 15.112489700317383\n",
      "-- processing region: 350 with 239 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_350.fasta\n",
      "--- *blasting region: 350\n",
      "--- total time for blast 350: 11.646078109741211\n",
      "-- processing region: 351 with 69 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_351.fasta\n",
      "--- *blasting region: 351\n",
      "--- total time for blast 351: 13.02008056640625\n",
      "-- processing region: 352 with 318 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_352.fasta\n",
      "--- *blasting region: 352\n",
      "--- total time for blast 352: 12.540340423583984\n",
      "-- processing region: 353 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_353.fasta\n",
      "--- *blasting region: 353\n",
      "--- total time for blast 353: 10.580818176269531\n",
      "-- processing region: 354 with 175 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_354.fasta\n",
      "--- *blasting region: 354\n",
      "--- total time for blast 354: 8.82399582862854\n",
      "-- processing region: 355 with 209 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_355.fasta\n",
      "--- *blasting region: 355\n",
      "--- total time for blast 355: 7.690728425979614\n",
      "-- processing region: 356 with 315 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_356.fasta\n",
      "--- *blasting region: 356\n",
      "--- total time for blast 356: 12.685582399368286\n",
      "-- processing region: 357 with 138 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_357.fasta\n",
      "--- *blasting region: 357\n",
      "--- total time for blast 357: 9.0733962059021\n",
      "-- processing region: 358 with 158 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_358.fasta\n",
      "--- *blasting region: 358\n",
      "--- total time for blast 358: 9.862334251403809\n",
      "-- processing region: 359 with 227 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_359.fasta\n",
      "--- *blasting region: 359\n",
      "--- total time for blast 359: 9.367177486419678\n",
      "-- processing region: 36 with 260 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_36.fasta\n",
      "--- *blasting region: 36\n",
      "--- total time for blast 36: 9.413800716400146\n",
      "-- processing region: 360 with 221 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_360.fasta\n",
      "--- *blasting region: 360\n",
      "--- total time for blast 360: 8.794355154037476\n",
      "-- processing region: 361 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_361.fasta\n",
      "--- *blasting region: 361\n",
      "--- total time for blast 361: 10.000448226928711\n",
      "-- processing region: 362 with 260 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_362.fasta\n",
      "--- *blasting region: 362\n",
      "--- total time for blast 362: 10.144002437591553\n",
      "-- processing region: 363 with 116 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_363.fasta\n",
      "--- *blasting region: 363\n",
      "--- total time for blast 363: 6.555882930755615\n",
      "-- processing region: 364 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_364.fasta\n",
      "--- *blasting region: 364\n",
      "--- total time for blast 364: 11.616651773452759\n",
      "-- processing region: 365 with 203 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_365.fasta\n",
      "--- *blasting region: 365\n",
      "--- total time for blast 365: 9.444368600845337\n",
      "-- processing region: 366 with 195 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_366.fasta\n",
      "--- *blasting region: 366\n",
      "--- total time for blast 366: 9.5514395236969\n",
      "-- processing region: 367 with 133 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_367.fasta\n",
      "--- *blasting region: 367\n",
      "--- total time for blast 367: 8.348630666732788\n",
      "-- processing region: 368 with 129 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_368.fasta\n",
      "--- *blasting region: 368\n",
      "--- total time for blast 368: 18.278522491455078\n",
      "-- processing region: 369 with 226 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_369.fasta\n",
      "--- *blasting region: 369\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 369: 10.837334156036377\n",
      "-- processing region: 37 with 244 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_37.fasta\n",
      "--- *blasting region: 37\n",
      "--- total time for blast 37: 7.918634414672852\n",
      "-- processing region: 370 with 175 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_370.fasta\n",
      "--- *blasting region: 370\n",
      "--- total time for blast 370: 7.235055685043335\n",
      "-- processing region: 371 with 123 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_371.fasta\n",
      "--- *blasting region: 371\n",
      "--- total time for blast 371: 15.231513261795044\n",
      "-- processing region: 372 with 210 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_372.fasta\n",
      "--- *blasting region: 372\n",
      "--- total time for blast 372: 9.948737144470215\n",
      "-- processing region: 373 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_373.fasta\n",
      "--- *blasting region: 373\n",
      "--- total time for blast 373: 13.729187488555908\n",
      "-- processing region: 374 with 305 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_374.fasta\n",
      "--- *blasting region: 374\n",
      "--- total time for blast 374: 9.850968599319458\n",
      "-- processing region: 375 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_375.fasta\n",
      "--- *blasting region: 375\n",
      "--- total time for blast 375: 9.682873964309692\n",
      "-- processing region: 376 with 315 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_376.fasta\n",
      "--- *blasting region: 376\n",
      "--- total time for blast 376: 10.890871524810791\n",
      "-- processing region: 377 with 326 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_377.fasta\n",
      "--- *blasting region: 377\n",
      "--- total time for blast 377: 11.1571364402771\n",
      "-- processing region: 378 with 270 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_378.fasta\n",
      "--- *blasting region: 378\n",
      "--- total time for blast 378: 11.277103185653687\n",
      "-- processing region: 379 with 305 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_379.fasta\n",
      "--- *blasting region: 379\n",
      "--- total time for blast 379: 12.527259588241577\n",
      "-- processing region: 38 with 75 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_38.fasta\n",
      "--- *blasting region: 38\n",
      "--- total time for blast 38: 4.587416887283325\n",
      "-- processing region: 380 with 272 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_380.fasta\n",
      "--- *blasting region: 380\n",
      "--- total time for blast 380: 11.219204664230347\n",
      "-- processing region: 381 with 327 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_381.fasta\n",
      "--- *blasting region: 381\n",
      "--- total time for blast 381: 10.054054975509644\n",
      "-- processing region: 382 with 242 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_382.fasta\n",
      "--- *blasting region: 382\n",
      "--- total time for blast 382: 9.566913843154907\n",
      "-- processing region: 383 with 233 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_383.fasta\n",
      "--- *blasting region: 383\n",
      "--- total time for blast 383: 12.445606470108032\n",
      "-- processing region: 384 with 212 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_384.fasta\n",
      "--- *blasting region: 384\n",
      "--- total time for blast 384: 9.920887231826782\n",
      "-- processing region: 385 with 230 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_385.fasta\n",
      "--- *blasting region: 385\n",
      "--- total time for blast 385: 8.454825401306152\n",
      "-- processing region: 386 with 329 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_386.fasta\n",
      "--- *blasting region: 386\n",
      "--- total time for blast 386: 12.438779354095459\n",
      "-- processing region: 387 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_387.fasta\n",
      "--- *blasting region: 387\n",
      "--- total time for blast 387: 8.9796884059906\n",
      "-- processing region: 388 with 323 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_388.fasta\n",
      "--- *blasting region: 388\n",
      "--- total time for blast 388: 13.065733194351196\n",
      "-- processing region: 389 with 266 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_389.fasta\n",
      "--- *blasting region: 389\n",
      "--- total time for blast 389: 10.050807476043701\n",
      "-- processing region: 39 with 178 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_39.fasta\n",
      "--- *blasting region: 39\n",
      "--- total time for blast 39: 17.909301042556763\n",
      "-- processing region: 390 with 103 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_390.fasta\n",
      "--- *blasting region: 390\n",
      "--- total time for blast 390: 6.317704439163208\n",
      "-- processing region: 391 with 153 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_391.fasta\n",
      "--- *blasting region: 391\n",
      "--- total time for blast 391: 10.805264472961426\n",
      "-- processing region: 392 with 114 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_392.fasta\n",
      "--- *blasting region: 392\n",
      "--- total time for blast 392: 6.9488365650177\n",
      "-- processing region: 393 with 93 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_393.fasta\n",
      "--- *blasting region: 393\n",
      "--- total time for blast 393: 5.73352313041687\n",
      "-- processing region: 395 with 260 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_395.fasta\n",
      "--- *blasting region: 395\n",
      "--- total time for blast 395: 11.032216787338257\n",
      "-- processing region: 396 with 127 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_396.fasta\n",
      "--- *blasting region: 396\n",
      "--- total time for blast 396: 5.7718892097473145\n",
      "-- processing region: 397 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_397.fasta\n",
      "--- *blasting region: 397\n",
      "--- total time for blast 397: 9.021718740463257\n",
      "-- processing region: 398 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_398.fasta\n",
      "--- *blasting region: 398\n",
      "--- total time for blast 398: 10.76201581954956\n",
      "-- processing region: 399 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_399.fasta\n",
      "--- *blasting region: 399\n",
      "--- total time for blast 399: 8.689499139785767\n",
      "-- processing region: 4 with 205 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_4.fasta\n",
      "--- *blasting region: 4\n",
      "--- total time for blast 4: 8.969643354415894\n",
      "-- processing region: 40 with 142 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_40.fasta\n",
      "--- *blasting region: 40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 40: 6.200113296508789\n",
      "-- processing region: 400 with 245 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_400.fasta\n",
      "--- *blasting region: 400\n",
      "--- total time for blast 400: 10.87867283821106\n",
      "-- processing region: 403 with 172 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_403.fasta\n",
      "--- *blasting region: 403\n",
      "--- total time for blast 403: 14.512754917144775\n",
      "-- processing region: 407 with 211 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_407.fasta\n",
      "--- *blasting region: 407\n",
      "--- total time for blast 407: 11.060325384140015\n",
      "-- processing region: 408 with 76 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_408.fasta\n",
      "--- *blasting region: 408\n",
      "--- total time for blast 408: 5.8292388916015625\n",
      "-- processing region: 41 with 122 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_41.fasta\n",
      "--- *blasting region: 41\n",
      "--- total time for blast 41: 4.989842176437378\n",
      "-- processing region: 411 with 126 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_411.fasta\n",
      "--- *blasting region: 411\n",
      "--- total time for blast 411: 14.132601022720337\n",
      "-- processing region: 412 with 261 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_412.fasta\n",
      "--- *blasting region: 412\n",
      "--- total time for blast 412: 9.48176646232605\n",
      "-- processing region: 413 with 98 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_413.fasta\n",
      "--- *blasting region: 413\n",
      "--- total time for blast 413: 5.470394849777222\n",
      "-- processing region: 414 with 96 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_414.fasta\n",
      "--- *blasting region: 414\n",
      "--- total time for blast 414: 10.914368629455566\n",
      "-- processing region: 415 with 250 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_415.fasta\n",
      "--- *blasting region: 415\n",
      "--- total time for blast 415: 7.948883295059204\n",
      "-- processing region: 418 with 133 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_418.fasta\n",
      "--- *blasting region: 418\n",
      "--- total time for blast 418: 8.025904655456543\n",
      "-- processing region: 419 with 309 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_419.fasta\n",
      "--- *blasting region: 419\n",
      "--- total time for blast 419: 15.953132390975952\n",
      "-- processing region: 42 with 105 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_42.fasta\n",
      "--- *blasting region: 42\n",
      "--- total time for blast 42: 13.629152059555054\n",
      "-- processing region: 420 with 169 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_420.fasta\n",
      "--- *blasting region: 420\n",
      "--- total time for blast 420: 11.260031461715698\n",
      "-- processing region: 421 with 232 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_421.fasta\n",
      "--- *blasting region: 421\n",
      "--- total time for blast 421: 9.485877990722656\n",
      "-- processing region: 422 with 221 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_422.fasta\n",
      "--- *blasting region: 422\n",
      "--- total time for blast 422: 8.132996082305908\n",
      "-- processing region: 423 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_423.fasta\n",
      "--- *blasting region: 423\n",
      "--- total time for blast 423: 13.983924150466919\n",
      "-- processing region: 424 with 294 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_424.fasta\n",
      "--- *blasting region: 424\n",
      "--- total time for blast 424: 11.263123273849487\n",
      "-- processing region: 425 with 75 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_425.fasta\n",
      "--- *blasting region: 425\n",
      "--- total time for blast 425: 17.00366449356079\n",
      "-- processing region: 426 with 318 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_426.fasta\n",
      "--- *blasting region: 426\n",
      "--- total time for blast 426: 13.1096670627594\n",
      "-- processing region: 427 with 293 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_427.fasta\n",
      "--- *blasting region: 427\n",
      "--- total time for blast 427: 11.81827449798584\n",
      "-- processing region: 428 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_428.fasta\n",
      "--- *blasting region: 428\n",
      "--- total time for blast 428: 15.771390438079834\n",
      "-- processing region: 429 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_429.fasta\n",
      "--- *blasting region: 429\n",
      "--- total time for blast 429: 11.860841035842896\n",
      "-- processing region: 43 with 173 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_43.fasta\n",
      "--- *blasting region: 43\n",
      "--- total time for blast 43: 9.652188539505005\n",
      "-- processing region: 430 with 298 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_430.fasta\n",
      "--- *blasting region: 430\n",
      "--- total time for blast 430: 10.01664113998413\n",
      "-- processing region: 431 with 80 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_431.fasta\n",
      "--- *blasting region: 431\n",
      "--- total time for blast 431: 7.8733811378479\n",
      "-- processing region: 432 with 278 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_432.fasta\n",
      "--- *blasting region: 432\n",
      "--- total time for blast 432: 14.315802097320557\n",
      "-- processing region: 433 with 219 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_433.fasta\n",
      "--- *blasting region: 433\n",
      "--- total time for blast 433: 9.185904741287231\n",
      "-- processing region: 434 with 243 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_434.fasta\n",
      "--- *blasting region: 434\n",
      "--- total time for blast 434: 8.773533821105957\n",
      "-- processing region: 435 with 139 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_435.fasta\n",
      "--- *blasting region: 435\n",
      "--- total time for blast 435: 7.6382896900177\n",
      "-- processing region: 436 with 167 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_436.fasta\n",
      "--- *blasting region: 436\n",
      "--- total time for blast 436: 8.075427293777466\n",
      "-- processing region: 437 with 212 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_437.fasta\n",
      "--- *blasting region: 437\n",
      "--- total time for blast 437: 9.342723369598389\n",
      "-- processing region: 438 with 70 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_438.fasta\n",
      "--- *blasting region: 438\n",
      "--- total time for blast 438: 4.768190145492554\n",
      "-- processing region: 439 with 315 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_439.fasta\n",
      "--- *blasting region: 439\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 439: 11.30353331565857\n",
      "-- processing region: 44 with 90 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_44.fasta\n",
      "--- *blasting region: 44\n",
      "--- total time for blast 44: 7.447296142578125\n",
      "-- processing region: 440 with 325 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_440.fasta\n",
      "--- *blasting region: 440\n",
      "--- total time for blast 440: 13.252742767333984\n",
      "-- processing region: 441 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_441.fasta\n",
      "--- *blasting region: 441\n",
      "--- total time for blast 441: 9.0118408203125\n",
      "-- processing region: 442 with 174 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_442.fasta\n",
      "--- *blasting region: 442\n",
      "--- total time for blast 442: 9.120792865753174\n",
      "-- processing region: 443 with 98 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_443.fasta\n",
      "--- *blasting region: 443\n",
      "--- total time for blast 443: 8.428858518600464\n",
      "-- processing region: 444 with 159 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_444.fasta\n",
      "--- *blasting region: 444\n",
      "--- total time for blast 444: 9.801913976669312\n",
      "-- processing region: 445 with 242 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_445.fasta\n",
      "--- *blasting region: 445\n",
      "--- total time for blast 445: 9.58370852470398\n",
      "-- processing region: 446 with 304 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_446.fasta\n",
      "--- *blasting region: 446\n",
      "--- total time for blast 446: 11.969120264053345\n",
      "-- processing region: 447 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_447.fasta\n",
      "--- *blasting region: 447\n",
      "--- total time for blast 447: 10.899454593658447\n",
      "-- processing region: 448 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_448.fasta\n",
      "--- *blasting region: 448\n",
      "--- total time for blast 448: 12.819238662719727\n",
      "-- processing region: 449 with 272 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_449.fasta\n",
      "--- *blasting region: 449\n",
      "--- total time for blast 449: 15.970926761627197\n",
      "-- processing region: 45 with 121 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_45.fasta\n",
      "--- *blasting region: 45\n",
      "--- total time for blast 45: 6.578891754150391\n",
      "-- processing region: 450 with 231 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_450.fasta\n",
      "--- *blasting region: 450\n",
      "--- total time for blast 450: 9.662840843200684\n",
      "-- processing region: 451 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_451.fasta\n",
      "--- *blasting region: 451\n",
      "--- total time for blast 451: 7.361502408981323\n",
      "-- processing region: 452 with 230 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_452.fasta\n",
      "--- *blasting region: 452\n",
      "--- total time for blast 452: 12.003890991210938\n",
      "-- processing region: 453 with 254 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_453.fasta\n",
      "--- *blasting region: 453\n",
      "--- total time for blast 453: 8.341074228286743\n",
      "-- processing region: 454 with 266 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_454.fasta\n",
      "--- *blasting region: 454\n",
      "--- total time for blast 454: 9.131421566009521\n",
      "-- processing region: 455 with 190 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_455.fasta\n",
      "--- *blasting region: 455\n",
      "--- total time for blast 455: 9.531076192855835\n",
      "-- processing region: 456 with 314 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_456.fasta\n",
      "--- *blasting region: 456\n",
      "--- total time for blast 456: 9.291729927062988\n",
      "-- processing region: 457 with 242 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_457.fasta\n",
      "--- *blasting region: 457\n",
      "--- total time for blast 457: 9.755215883255005\n",
      "-- processing region: 458 with 292 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_458.fasta\n",
      "--- *blasting region: 458\n",
      "--- total time for blast 458: 10.577030181884766\n",
      "-- processing region: 46 with 241 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_46.fasta\n",
      "--- *blasting region: 46\n",
      "--- total time for blast 46: 9.242169618606567\n",
      "-- processing region: 461 with 179 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_461.fasta\n",
      "--- *blasting region: 461\n",
      "--- total time for blast 461: 8.158882856369019\n",
      "-- processing region: 462 with 132 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_462.fasta\n",
      "--- *blasting region: 462\n",
      "--- total time for blast 462: 11.301937580108643\n",
      "-- processing region: 463 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_463.fasta\n",
      "--- *blasting region: 463\n",
      "--- total time for blast 463: 11.101319551467896\n",
      "-- processing region: 464 with 321 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_464.fasta\n",
      "--- *blasting region: 464\n",
      "--- total time for blast 464: 12.350575685501099\n",
      "-- processing region: 465 with 249 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_465.fasta\n",
      "--- *blasting region: 465\n",
      "--- total time for blast 465: 13.315264701843262\n",
      "-- processing region: 466 with 295 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_466.fasta\n",
      "--- *blasting region: 466\n",
      "--- total time for blast 466: 10.649528980255127\n",
      "-- processing region: 467 with 336 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_467.fasta\n",
      "--- *blasting region: 467\n",
      "--- total time for blast 467: 11.662954330444336\n",
      "-- processing region: 469 with 266 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_469.fasta\n",
      "--- *blasting region: 469\n",
      "--- total time for blast 469: 10.063539505004883\n",
      "-- processing region: 47 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_47.fasta\n",
      "--- *blasting region: 47\n",
      "--- total time for blast 47: 8.53135371208191\n",
      "-- processing region: 470 with 241 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_470.fasta\n",
      "--- *blasting region: 470\n",
      "--- total time for blast 470: 8.209804773330688\n",
      "-- processing region: 471 with 287 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_471.fasta\n",
      "--- *blasting region: 471\n",
      "--- total time for blast 471: 13.388777732849121\n",
      "-- processing region: 472 with 227 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_472.fasta\n",
      "--- *blasting region: 472\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 472: 14.67583703994751\n",
      "-- processing region: 473 with 182 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_473.fasta\n",
      "--- *blasting region: 473\n",
      "--- total time for blast 473: 7.254590272903442\n",
      "-- processing region: 474 with 283 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_474.fasta\n",
      "--- *blasting region: 474\n",
      "--- total time for blast 474: 10.699969053268433\n",
      "-- processing region: 475 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_475.fasta\n",
      "--- *blasting region: 475\n",
      "--- total time for blast 475: 7.8618550300598145\n",
      "-- processing region: 476 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_476.fasta\n",
      "--- *blasting region: 476\n",
      "--- total time for blast 476: 10.810384750366211\n",
      "-- processing region: 477 with 174 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_477.fasta\n",
      "--- *blasting region: 477\n",
      "--- total time for blast 477: 14.6318039894104\n",
      "-- processing region: 478 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_478.fasta\n",
      "--- *blasting region: 478\n",
      "--- total time for blast 478: 12.825653314590454\n",
      "-- processing region: 479 with 286 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_479.fasta\n",
      "--- *blasting region: 479\n",
      "--- total time for blast 479: 16.357632637023926\n",
      "-- processing region: 48 with 218 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_48.fasta\n",
      "--- *blasting region: 48\n",
      "--- total time for blast 48: 18.42829418182373\n",
      "-- processing region: 480 with 178 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_480.fasta\n",
      "--- *blasting region: 480\n",
      "--- total time for blast 480: 8.4482581615448\n",
      "-- processing region: 481 with 162 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_481.fasta\n",
      "--- *blasting region: 481\n",
      "--- total time for blast 481: 7.8418357372283936\n",
      "-- processing region: 482 with 113 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_482.fasta\n",
      "--- *blasting region: 482\n",
      "--- total time for blast 482: 9.445035457611084\n",
      "-- processing region: 483 with 150 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_483.fasta\n",
      "--- *blasting region: 483\n",
      "--- total time for blast 483: 9.289918899536133\n",
      "-- processing region: 484 with 307 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_484.fasta\n",
      "--- *blasting region: 484\n",
      "--- total time for blast 484: 12.135795593261719\n",
      "-- processing region: 485 with 118 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_485.fasta\n",
      "--- *blasting region: 485\n",
      "--- total time for blast 485: 7.662782669067383\n",
      "-- processing region: 486 with 140 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_486.fasta\n",
      "--- *blasting region: 486\n",
      "--- total time for blast 486: 8.695767641067505\n",
      "-- processing region: 487 with 108 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_487.fasta\n",
      "--- *blasting region: 487\n",
      "--- total time for blast 487: 5.193654298782349\n",
      "-- processing region: 488 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_488.fasta\n",
      "--- *blasting region: 488\n",
      "--- total time for blast 488: 9.619175434112549\n",
      "-- processing region: 489 with 144 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_489.fasta\n",
      "--- *blasting region: 489\n",
      "--- total time for blast 489: 6.223614931106567\n",
      "-- processing region: 49 with 225 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_49.fasta\n",
      "--- *blasting region: 49\n",
      "--- total time for blast 49: 9.368150234222412\n",
      "-- processing region: 490 with 183 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_490.fasta\n",
      "--- *blasting region: 490\n",
      "--- total time for blast 490: 8.664482116699219\n",
      "-- processing region: 491 with 282 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_491.fasta\n",
      "--- *blasting region: 491\n",
      "--- total time for blast 491: 11.228819608688354\n",
      "-- processing region: 492 with 157 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_492.fasta\n",
      "--- *blasting region: 492\n",
      "--- total time for blast 492: 6.005410671234131\n",
      "-- processing region: 493 with 227 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_493.fasta\n",
      "--- *blasting region: 493\n",
      "--- total time for blast 493: 9.14786410331726\n",
      "-- processing region: 494 with 182 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_494.fasta\n",
      "--- *blasting region: 494\n",
      "--- total time for blast 494: 9.185790300369263\n",
      "-- processing region: 495 with 291 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_495.fasta\n",
      "--- *blasting region: 495\n",
      "--- total time for blast 495: 11.916141510009766\n",
      "-- processing region: 496 with 203 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_496.fasta\n",
      "--- *blasting region: 496\n",
      "--- total time for blast 496: 10.515120029449463\n",
      "-- processing region: 497 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_497.fasta\n",
      "--- *blasting region: 497\n",
      "--- total time for blast 497: 9.336450338363647\n",
      "-- processing region: 498 with 208 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_498.fasta\n",
      "--- *blasting region: 498\n",
      "--- total time for blast 498: 8.35424542427063\n",
      "-- processing region: 499 with 219 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_499.fasta\n",
      "--- *blasting region: 499\n",
      "--- total time for blast 499: 20.4680392742157\n",
      "-- processing region: 5 with 144 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_5.fasta\n",
      "--- *blasting region: 5\n",
      "--- total time for blast 5: 7.6494362354278564\n",
      "-- processing region: 50 with 287 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_50.fasta\n",
      "--- *blasting region: 50\n",
      "--- total time for blast 50: 10.922984600067139\n",
      "-- processing region: 500 with 169 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_500.fasta\n",
      "--- *blasting region: 500\n",
      "--- total time for blast 500: 12.948328256607056\n",
      "-- processing region: 501 with 202 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_501.fasta\n",
      "--- *blasting region: 501\n",
      "--- total time for blast 501: 10.045149087905884\n",
      "-- processing region: 502 with 316 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_502.fasta\n",
      "--- *blasting region: 502\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 502: 10.461658716201782\n",
      "-- processing region: 503 with 321 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_503.fasta\n",
      "--- *blasting region: 503\n",
      "--- total time for blast 503: 14.87893009185791\n",
      "-- processing region: 504 with 179 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_504.fasta\n",
      "--- *blasting region: 504\n",
      "--- total time for blast 504: 10.839454889297485\n",
      "-- processing region: 505 with 273 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_505.fasta\n",
      "--- *blasting region: 505\n",
      "--- total time for blast 505: 10.567364692687988\n",
      "-- processing region: 506 with 279 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_506.fasta\n",
      "--- *blasting region: 506\n",
      "--- total time for blast 506: 12.304046869277954\n",
      "-- processing region: 507 with 282 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_507.fasta\n",
      "--- *blasting region: 507\n",
      "--- total time for blast 507: 10.403301000595093\n",
      "-- processing region: 508 with 309 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_508.fasta\n",
      "--- *blasting region: 508\n",
      "--- total time for blast 508: 11.483283281326294\n",
      "-- processing region: 509 with 261 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_509.fasta\n",
      "--- *blasting region: 509\n",
      "--- total time for blast 509: 9.446155071258545\n",
      "-- processing region: 51 with 250 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_51.fasta\n",
      "--- *blasting region: 51\n",
      "--- total time for blast 51: 9.580124855041504\n",
      "-- processing region: 510 with 294 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_510.fasta\n",
      "--- *blasting region: 510\n",
      "--- total time for blast 510: 14.326168775558472\n",
      "-- processing region: 511 with 103 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_511.fasta\n",
      "--- *blasting region: 511\n",
      "--- total time for blast 511: 7.161057710647583\n",
      "-- processing region: 513 with 53 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_513.fasta\n",
      "--- *blasting region: 513\n",
      "--- total time for blast 513: 6.397764444351196\n",
      "-- processing region: 514 with 247 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_514.fasta\n",
      "--- *blasting region: 514\n",
      "--- total time for blast 514: 14.32791805267334\n",
      "-- processing region: 515 with 223 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_515.fasta\n",
      "--- *blasting region: 515\n",
      "--- total time for blast 515: 11.533801078796387\n",
      "-- processing region: 516 with 120 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_516.fasta\n",
      "--- *blasting region: 516\n",
      "--- total time for blast 516: 8.448292255401611\n",
      "-- processing region: 517 with 187 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_517.fasta\n",
      "--- *blasting region: 517\n",
      "--- total time for blast 517: 7.383498907089233\n",
      "-- processing region: 518 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_518.fasta\n",
      "--- *blasting region: 518\n",
      "--- total time for blast 518: 9.101851224899292\n",
      "-- processing region: 519 with 216 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_519.fasta\n",
      "--- *blasting region: 519\n",
      "--- total time for blast 519: 9.64715027809143\n",
      "-- processing region: 52 with 329 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_52.fasta\n",
      "--- *blasting region: 52\n",
      "--- total time for blast 52: 10.47980523109436\n",
      "-- processing region: 520 with 222 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_520.fasta\n",
      "--- *blasting region: 520\n",
      "--- total time for blast 520: 9.52450704574585\n",
      "-- processing region: 521 with 116 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_521.fasta\n",
      "--- *blasting region: 521\n",
      "--- total time for blast 521: 7.055363178253174\n",
      "-- processing region: 522 with 219 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_522.fasta\n",
      "--- *blasting region: 522\n",
      "--- total time for blast 522: 11.989439249038696\n",
      "-- processing region: 523 with 230 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_523.fasta\n",
      "--- *blasting region: 523\n",
      "--- total time for blast 523: 13.910094976425171\n",
      "-- processing region: 524 with 217 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_524.fasta\n",
      "--- *blasting region: 524\n",
      "--- total time for blast 524: 11.076137065887451\n",
      "-- processing region: 525 with 180 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_525.fasta\n",
      "--- *blasting region: 525\n",
      "--- total time for blast 525: 8.206091165542603\n",
      "-- processing region: 526 with 90 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_526.fasta\n",
      "--- *blasting region: 526\n",
      "--- total time for blast 526: 12.994367361068726\n",
      "-- processing region: 527 with 193 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_527.fasta\n",
      "--- *blasting region: 527\n",
      "--- total time for blast 527: 9.410340785980225\n",
      "-- processing region: 528 with 256 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_528.fasta\n",
      "--- *blasting region: 528\n",
      "--- total time for blast 528: 13.245858907699585\n",
      "-- processing region: 529 with 349 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_529.fasta\n",
      "--- *blasting region: 529\n",
      "--- total time for blast 529: 13.549930810928345\n",
      "-- processing region: 53 with 299 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_53.fasta\n",
      "--- *blasting region: 53\n",
      "--- total time for blast 53: 12.203523397445679\n",
      "-- processing region: 530 with 255 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_530.fasta\n",
      "--- *blasting region: 530\n",
      "--- total time for blast 530: 10.30967903137207\n",
      "-- processing region: 531 with 330 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_531.fasta\n",
      "--- *blasting region: 531\n",
      "--- total time for blast 531: 10.686623573303223\n",
      "-- processing region: 532 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_532.fasta\n",
      "--- *blasting region: 532\n",
      "--- total time for blast 532: 10.169899702072144\n",
      "-- processing region: 533 with 224 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_533.fasta\n",
      "--- *blasting region: 533\n",
      "--- total time for blast 533: 10.162569284439087\n",
      "-- processing region: 534 with 264 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_534.fasta\n",
      "--- *blasting region: 534\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 534: 9.574130773544312\n",
      "-- processing region: 535 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_535.fasta\n",
      "--- *blasting region: 535\n",
      "--- total time for blast 535: 9.3654146194458\n",
      "-- processing region: 536 with 269 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_536.fasta\n",
      "--- *blasting region: 536\n",
      "--- total time for blast 536: 10.946852207183838\n",
      "-- processing region: 537 with 331 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_537.fasta\n",
      "--- *blasting region: 537\n",
      "--- total time for blast 537: 13.60220742225647\n",
      "-- processing region: 538 with 304 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_538.fasta\n",
      "--- *blasting region: 538\n",
      "--- total time for blast 538: 10.67740273475647\n",
      "-- processing region: 539 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_539.fasta\n",
      "--- *blasting region: 539\n",
      "--- total time for blast 539: 11.521791219711304\n",
      "-- processing region: 54 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_54.fasta\n",
      "--- *blasting region: 54\n",
      "--- total time for blast 54: 13.216846704483032\n",
      "-- processing region: 540 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_540.fasta\n",
      "--- *blasting region: 540\n",
      "--- total time for blast 540: 10.504961729049683\n",
      "-- processing region: 541 with 267 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_541.fasta\n",
      "--- *blasting region: 541\n",
      "--- total time for blast 541: 8.462023496627808\n",
      "-- processing region: 542 with 200 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_542.fasta\n",
      "--- *blasting region: 542\n",
      "--- total time for blast 542: 11.711766481399536\n",
      "-- processing region: 543 with 140 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_543.fasta\n",
      "--- *blasting region: 543\n",
      "--- total time for blast 543: 5.670223236083984\n",
      "-- processing region: 544 with 75 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_544.fasta\n",
      "--- *blasting region: 544\n",
      "--- total time for blast 544: 4.2394726276397705\n",
      "-- processing region: 545 with 162 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_545.fasta\n",
      "--- *blasting region: 545\n",
      "--- total time for blast 545: 5.901395320892334\n",
      "-- processing region: 546 with 234 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_546.fasta\n",
      "--- *blasting region: 546\n",
      "--- total time for blast 546: 11.251106262207031\n",
      "-- processing region: 548 with 88 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_548.fasta\n",
      "--- *blasting region: 548\n",
      "--- total time for blast 548: 10.215608596801758\n",
      "-- processing region: 549 with 174 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_549.fasta\n",
      "--- *blasting region: 549\n",
      "--- total time for blast 549: 10.31521201133728\n",
      "-- processing region: 55 with 206 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_55.fasta\n",
      "--- *blasting region: 55\n",
      "--- total time for blast 55: 8.982012271881104\n",
      "-- processing region: 550 with 181 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_550.fasta\n",
      "--- *blasting region: 550\n",
      "--- total time for blast 550: 9.098942041397095\n",
      "-- processing region: 551 with 236 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_551.fasta\n",
      "--- *blasting region: 551\n",
      "--- total time for blast 551: 10.883005380630493\n",
      "-- processing region: 552 with 239 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_552.fasta\n",
      "--- *blasting region: 552\n",
      "--- total time for blast 552: 8.518893480300903\n",
      "-- processing region: 553 with 314 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_553.fasta\n",
      "--- *blasting region: 553\n",
      "--- total time for blast 553: 11.944926977157593\n",
      "-- processing region: 554 with 243 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_554.fasta\n",
      "--- *blasting region: 554\n",
      "--- total time for blast 554: 10.07448148727417\n",
      "-- processing region: 555 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_555.fasta\n",
      "--- *blasting region: 555\n",
      "--- total time for blast 555: 8.803510189056396\n",
      "-- processing region: 556 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_556.fasta\n",
      "--- *blasting region: 556\n",
      "--- total time for blast 556: 8.196812629699707\n",
      "-- processing region: 557 with 234 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_557.fasta\n",
      "--- *blasting region: 557\n",
      "--- total time for blast 557: 12.48213791847229\n",
      "-- processing region: 558 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_558.fasta\n",
      "--- *blasting region: 558\n",
      "--- total time for blast 558: 9.173990488052368\n",
      "-- processing region: 559 with 296 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_559.fasta\n",
      "--- *blasting region: 559\n",
      "--- total time for blast 559: 12.045259475708008\n",
      "-- processing region: 56 with 125 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_56.fasta\n",
      "--- *blasting region: 56\n",
      "--- total time for blast 56: 11.891514301300049\n",
      "-- processing region: 560 with 286 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_560.fasta\n",
      "--- *blasting region: 560\n",
      "--- total time for blast 560: 13.334940910339355\n",
      "-- processing region: 561 with 148 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_561.fasta\n",
      "--- *blasting region: 561\n",
      "--- total time for blast 561: 6.58890700340271\n",
      "-- processing region: 563 with 296 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_563.fasta\n",
      "--- *blasting region: 563\n",
      "--- total time for blast 563: 9.71723747253418\n",
      "-- processing region: 564 with 94 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_564.fasta\n",
      "--- *blasting region: 564\n",
      "--- total time for blast 564: 17.60212779045105\n",
      "-- processing region: 565 with 326 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_565.fasta\n",
      "--- *blasting region: 565\n",
      "--- total time for blast 565: 11.655975580215454\n",
      "-- processing region: 566 with 184 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_566.fasta\n",
      "--- *blasting region: 566\n",
      "--- total time for blast 566: 10.664147853851318\n",
      "-- processing region: 567 with 265 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_567.fasta\n",
      "--- *blasting region: 567\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 567: 7.759770631790161\n",
      "-- processing region: 568 with 139 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_568.fasta\n",
      "--- *blasting region: 568\n",
      "--- total time for blast 568: 13.28598403930664\n",
      "-- processing region: 569 with 166 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_569.fasta\n",
      "--- *blasting region: 569\n",
      "--- total time for blast 569: 5.676033973693848\n",
      "-- processing region: 57 with 232 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_57.fasta\n",
      "--- *blasting region: 57\n",
      "--- total time for blast 57: 7.7214062213897705\n",
      "-- processing region: 570 with 241 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_570.fasta\n",
      "--- *blasting region: 570\n",
      "--- total time for blast 570: 9.928735733032227\n",
      "-- processing region: 571 with 209 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_571.fasta\n",
      "--- *blasting region: 571\n",
      "--- total time for blast 571: 9.04651403427124\n",
      "-- processing region: 572 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_572.fasta\n",
      "--- *blasting region: 572\n",
      "--- total time for blast 572: 10.475569248199463\n",
      "-- processing region: 573 with 138 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_573.fasta\n",
      "--- *blasting region: 573\n",
      "--- total time for blast 573: 7.9629316329956055\n",
      "-- processing region: 574 with 218 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_574.fasta\n",
      "--- *blasting region: 574\n",
      "--- total time for blast 574: 9.855873107910156\n",
      "-- processing region: 575 with 73 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_575.fasta\n",
      "--- *blasting region: 575\n",
      "--- total time for blast 575: 17.42411994934082\n",
      "-- processing region: 576 with 161 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_576.fasta\n",
      "--- *blasting region: 576\n",
      "--- total time for blast 576: 9.033487796783447\n",
      "-- processing region: 577 with 60 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_577.fasta\n",
      "--- *blasting region: 577\n",
      "--- total time for blast 577: 7.186837673187256\n",
      "-- processing region: 578 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_578.fasta\n",
      "--- *blasting region: 578\n",
      "--- total time for blast 578: 10.312029600143433\n",
      "-- processing region: 579 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_579.fasta\n",
      "--- *blasting region: 579\n",
      "--- total time for blast 579: 10.255358457565308\n",
      "-- processing region: 58 with 112 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_58.fasta\n",
      "--- *blasting region: 58\n",
      "--- total time for blast 58: 12.034814834594727\n",
      "-- processing region: 580 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_580.fasta\n",
      "--- *blasting region: 580\n",
      "--- total time for blast 580: 11.119450092315674\n",
      "-- processing region: 581 with 94 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_581.fasta\n",
      "--- *blasting region: 581\n",
      "--- total time for blast 581: 6.747358322143555\n",
      "-- processing region: 582 with 152 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_582.fasta\n",
      "--- *blasting region: 582\n",
      "--- total time for blast 582: 6.308698892593384\n",
      "-- processing region: 583 with 271 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_583.fasta\n",
      "--- *blasting region: 583\n",
      "--- total time for blast 583: 11.834460020065308\n",
      "-- processing region: 584 with 118 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_584.fasta\n",
      "--- *blasting region: 584\n",
      "--- total time for blast 584: 5.004251480102539\n",
      "-- processing region: 585 with 188 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_585.fasta\n",
      "--- *blasting region: 585\n",
      "--- total time for blast 585: 9.812974452972412\n",
      "-- processing region: 586 with 288 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_586.fasta\n",
      "--- *blasting region: 586\n",
      "--- total time for blast 586: 12.031499147415161\n",
      "-- processing region: 587 with 78 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_587.fasta\n",
      "--- *blasting region: 587\n",
      "--- total time for blast 587: 5.984801292419434\n",
      "-- processing region: 588 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_588.fasta\n",
      "--- *blasting region: 588\n",
      "--- total time for blast 588: 9.585639476776123\n",
      "-- processing region: 589 with 226 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_589.fasta\n",
      "--- *blasting region: 589\n",
      "--- total time for blast 589: 17.39197063446045\n",
      "-- processing region: 59 with 159 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_59.fasta\n",
      "--- *blasting region: 59\n",
      "--- total time for blast 59: 13.192432880401611\n",
      "-- processing region: 590 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_590.fasta\n",
      "--- *blasting region: 590\n",
      "--- total time for blast 590: 10.54087519645691\n",
      "-- processing region: 591 with 222 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_591.fasta\n",
      "--- *blasting region: 591\n",
      "--- total time for blast 591: 10.348310708999634\n",
      "-- processing region: 592 with 253 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_592.fasta\n",
      "--- *blasting region: 592\n",
      "--- total time for blast 592: 10.024466037750244\n",
      "-- processing region: 593 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_593.fasta\n",
      "--- *blasting region: 593\n",
      "--- total time for blast 593: 9.376984119415283\n",
      "-- processing region: 594 with 319 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_594.fasta\n",
      "--- *blasting region: 594\n",
      "--- total time for blast 594: 11.184330224990845\n",
      "-- processing region: 595 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_595.fasta\n",
      "--- *blasting region: 595\n",
      "--- total time for blast 595: 11.783552646636963\n",
      "-- processing region: 596 with 304 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_596.fasta\n",
      "--- *blasting region: 596\n",
      "--- total time for blast 596: 9.474716663360596\n",
      "-- processing region: 597 with 245 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_597.fasta\n",
      "--- *blasting region: 597\n",
      "--- total time for blast 597: 10.874600887298584\n",
      "-- processing region: 598 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_598.fasta\n",
      "--- *blasting region: 598\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 598: 6.194962024688721\n",
      "-- processing region: 599 with 258 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_599.fasta\n",
      "--- *blasting region: 599\n",
      "--- total time for blast 599: 15.639772176742554\n",
      "-- processing region: 6 with 297 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_6.fasta\n",
      "--- *blasting region: 6\n",
      "--- total time for blast 6: 10.005953550338745\n",
      "-- processing region: 60 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_60.fasta\n",
      "--- *blasting region: 60\n",
      "--- total time for blast 60: 9.559448719024658\n",
      "-- processing region: 600 with 253 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_600.fasta\n",
      "--- *blasting region: 600\n",
      "--- total time for blast 600: 14.044826030731201\n",
      "-- processing region: 601 with 266 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_601.fasta\n",
      "--- *blasting region: 601\n",
      "--- total time for blast 601: 10.241201400756836\n",
      "-- processing region: 602 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_602.fasta\n",
      "--- *blasting region: 602\n",
      "--- total time for blast 602: 8.777356386184692\n",
      "-- processing region: 603 with 92 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_603.fasta\n",
      "--- *blasting region: 603\n",
      "--- total time for blast 603: 7.418757438659668\n",
      "-- processing region: 604 with 187 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_604.fasta\n",
      "--- *blasting region: 604\n",
      "--- total time for blast 604: 9.271591901779175\n",
      "-- processing region: 605 with 212 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_605.fasta\n",
      "--- *blasting region: 605\n",
      "--- total time for blast 605: 12.27627420425415\n",
      "-- processing region: 606 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_606.fasta\n",
      "--- *blasting region: 606\n",
      "--- total time for blast 606: 9.838433265686035\n",
      "-- processing region: 607 with 190 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_607.fasta\n",
      "--- *blasting region: 607\n",
      "--- total time for blast 607: 8.923835515975952\n",
      "-- processing region: 608 with 256 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_608.fasta\n",
      "--- *blasting region: 608\n",
      "--- total time for blast 608: 9.893943548202515\n",
      "-- processing region: 609 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_609.fasta\n",
      "--- *blasting region: 609\n",
      "--- total time for blast 609: 10.990487813949585\n",
      "-- processing region: 61 with 281 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_61.fasta\n",
      "--- *blasting region: 61\n",
      "--- total time for blast 61: 11.16423749923706\n",
      "-- processing region: 610 with 293 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_610.fasta\n",
      "--- *blasting region: 610\n",
      "--- total time for blast 610: 12.710309267044067\n",
      "-- processing region: 611 with 281 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_611.fasta\n",
      "--- *blasting region: 611\n",
      "--- total time for blast 611: 8.833579301834106\n",
      "-- processing region: 612 with 277 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_612.fasta\n",
      "--- *blasting region: 612\n",
      "--- total time for blast 612: 9.864375114440918\n",
      "-- processing region: 613 with 236 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_613.fasta\n",
      "--- *blasting region: 613\n",
      "--- total time for blast 613: 18.81522011756897\n",
      "-- processing region: 616 with 289 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_616.fasta\n",
      "--- *blasting region: 616\n",
      "--- total time for blast 616: 11.490652322769165\n",
      "-- processing region: 617 with 158 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_617.fasta\n",
      "--- *blasting region: 617\n",
      "--- total time for blast 617: 6.857108116149902\n",
      "-- processing region: 618 with 126 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_618.fasta\n",
      "--- *blasting region: 618\n",
      "--- total time for blast 618: 7.8735902309417725\n",
      "-- processing region: 619 with 222 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_619.fasta\n",
      "--- *blasting region: 619\n",
      "--- total time for blast 619: 9.61227297782898\n",
      "-- processing region: 62 with 265 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_62.fasta\n",
      "--- *blasting region: 62\n",
      "--- total time for blast 62: 12.749919176101685\n",
      "-- processing region: 620 with 103 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_620.fasta\n",
      "--- *blasting region: 620\n",
      "--- total time for blast 620: 5.778303623199463\n",
      "-- processing region: 621 with 191 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_621.fasta\n",
      "--- *blasting region: 621\n",
      "--- total time for blast 621: 12.876387596130371\n",
      "-- processing region: 622 with 272 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_622.fasta\n",
      "--- *blasting region: 622\n",
      "--- total time for blast 622: 9.527109622955322\n",
      "-- processing region: 623 with 177 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_623.fasta\n",
      "--- *blasting region: 623\n",
      "--- total time for blast 623: 10.610362768173218\n",
      "-- processing region: 624 with 215 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_624.fasta\n",
      "--- *blasting region: 624\n",
      "--- total time for blast 624: 11.461774349212646\n",
      "-- processing region: 625 with 107 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_625.fasta\n",
      "--- *blasting region: 625\n",
      "--- total time for blast 625: 8.463835000991821\n",
      "-- processing region: 626 with 175 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_626.fasta\n",
      "--- *blasting region: 626\n",
      "--- total time for blast 626: 7.821399927139282\n",
      "-- processing region: 627 with 75 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_627.fasta\n",
      "--- *blasting region: 627\n",
      "--- total time for blast 627: 9.736211061477661\n",
      "-- processing region: 628 with 332 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_628.fasta\n",
      "--- *blasting region: 628\n",
      "--- total time for blast 628: 11.81419825553894\n",
      "-- processing region: 629 with 315 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_629.fasta\n",
      "--- *blasting region: 629\n",
      "--- total time for blast 629: 10.335283041000366\n",
      "-- processing region: 63 with 245 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_63.fasta\n",
      "--- *blasting region: 63\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 63: 8.62207317352295\n",
      "-- processing region: 630 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_630.fasta\n",
      "--- *blasting region: 630\n",
      "--- total time for blast 630: 11.045226812362671\n",
      "-- processing region: 631 with 205 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_631.fasta\n",
      "--- *blasting region: 631\n",
      "--- total time for blast 631: 9.808301210403442\n",
      "-- processing region: 632 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_632.fasta\n",
      "--- *blasting region: 632\n",
      "--- total time for blast 632: 9.774118423461914\n",
      "-- processing region: 633 with 296 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_633.fasta\n",
      "--- *blasting region: 633\n",
      "--- total time for blast 633: 8.837615966796875\n",
      "-- processing region: 634 with 168 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_634.fasta\n",
      "--- *blasting region: 634\n",
      "--- total time for blast 634: 6.06351637840271\n",
      "-- processing region: 635 with 234 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_635.fasta\n",
      "--- *blasting region: 635\n",
      "--- total time for blast 635: 11.450812816619873\n",
      "-- processing region: 636 with 168 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_636.fasta\n",
      "--- *blasting region: 636\n",
      "--- total time for blast 636: 10.198566913604736\n",
      "-- processing region: 637 with 255 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_637.fasta\n",
      "--- *blasting region: 637\n",
      "--- total time for blast 637: 10.177637577056885\n",
      "-- processing region: 638 with 159 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_638.fasta\n",
      "--- *blasting region: 638\n",
      "--- total time for blast 638: 5.6997528076171875\n",
      "-- processing region: 639 with 283 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_639.fasta\n",
      "--- *blasting region: 639\n",
      "--- total time for blast 639: 12.616602182388306\n",
      "-- processing region: 64 with 303 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_64.fasta\n",
      "--- *blasting region: 64\n",
      "--- total time for blast 64: 13.002670049667358\n",
      "-- processing region: 640 with 286 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_640.fasta\n",
      "--- *blasting region: 640\n",
      "--- total time for blast 640: 13.873749017715454\n",
      "-- processing region: 641 with 263 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_641.fasta\n",
      "--- *blasting region: 641\n",
      "--- total time for blast 641: 11.880834341049194\n",
      "-- processing region: 642 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_642.fasta\n",
      "--- *blasting region: 642\n",
      "--- total time for blast 642: 11.385950565338135\n",
      "-- processing region: 643 with 272 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_643.fasta\n",
      "--- *blasting region: 643\n",
      "--- total time for blast 643: 9.956500768661499\n",
      "-- processing region: 644 with 131 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_644.fasta\n",
      "--- *blasting region: 644\n",
      "--- total time for blast 644: 8.104036092758179\n",
      "-- processing region: 645 with 303 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_645.fasta\n",
      "--- *blasting region: 645\n",
      "--- total time for blast 645: 10.79582166671753\n",
      "-- processing region: 646 with 234 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_646.fasta\n",
      "--- *blasting region: 646\n",
      "--- total time for blast 646: 7.761284351348877\n",
      "-- processing region: 647 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_647.fasta\n",
      "--- *blasting region: 647\n",
      "--- total time for blast 647: 12.251063346862793\n",
      "-- processing region: 648 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_648.fasta\n",
      "--- *blasting region: 648\n",
      "--- total time for blast 648: 10.625855922698975\n",
      "-- processing region: 649 with 228 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_649.fasta\n",
      "--- *blasting region: 649\n",
      "--- total time for blast 649: 7.751773834228516\n",
      "-- processing region: 65 with 231 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_65.fasta\n",
      "--- *blasting region: 65\n",
      "--- total time for blast 65: 9.664236545562744\n",
      "-- processing region: 650 with 260 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_650.fasta\n",
      "--- *blasting region: 650\n",
      "--- total time for blast 650: 10.834014892578125\n",
      "-- processing region: 651 with 239 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_651.fasta\n",
      "--- *blasting region: 651\n",
      "--- total time for blast 651: 9.913459300994873\n",
      "-- processing region: 652 with 211 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_652.fasta\n",
      "--- *blasting region: 652\n",
      "--- total time for blast 652: 10.244610786437988\n",
      "-- processing region: 653 with 233 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_653.fasta\n",
      "--- *blasting region: 653\n",
      "--- total time for blast 653: 12.572260856628418\n",
      "-- processing region: 654 with 308 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_654.fasta\n",
      "--- *blasting region: 654\n",
      "--- total time for blast 654: 9.05098009109497\n",
      "-- processing region: 655 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_655.fasta\n",
      "--- *blasting region: 655\n",
      "--- total time for blast 655: 8.995465755462646\n",
      "-- processing region: 656 with 320 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_656.fasta\n",
      "--- *blasting region: 656\n",
      "--- total time for blast 656: 11.745334386825562\n",
      "-- processing region: 657 with 290 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_657.fasta\n",
      "--- *blasting region: 657\n",
      "--- total time for blast 657: 12.236741304397583\n",
      "-- processing region: 658 with 269 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_658.fasta\n",
      "--- *blasting region: 658\n",
      "--- total time for blast 658: 12.712936401367188\n",
      "-- processing region: 659 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_659.fasta\n",
      "--- *blasting region: 659\n",
      "--- total time for blast 659: 17.31121850013733\n",
      "-- processing region: 66 with 310 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_66.fasta\n",
      "--- *blasting region: 66\n",
      "--- total time for blast 66: 13.94550347328186\n",
      "-- processing region: 660 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_660.fasta\n",
      "--- *blasting region: 660\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 660: 8.869733572006226\n",
      "-- processing region: 661 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_661.fasta\n",
      "--- *blasting region: 661\n",
      "--- total time for blast 661: 12.934610843658447\n",
      "-- processing region: 662 with 322 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_662.fasta\n",
      "--- *blasting region: 662\n",
      "--- total time for blast 662: 12.18977165222168\n",
      "-- processing region: 663 with 271 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_663.fasta\n",
      "--- *blasting region: 663\n",
      "--- total time for blast 663: 11.293832063674927\n",
      "-- processing region: 665 with 270 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_665.fasta\n",
      "--- *blasting region: 665\n",
      "--- total time for blast 665: 8.602141618728638\n",
      "-- processing region: 666 with 177 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_666.fasta\n",
      "--- *blasting region: 666\n",
      "--- total time for blast 666: 7.3945348262786865\n",
      "-- processing region: 667 with 241 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_667.fasta\n",
      "--- *blasting region: 667\n",
      "--- total time for blast 667: 9.047753095626831\n",
      "-- processing region: 668 with 196 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_668.fasta\n",
      "--- *blasting region: 668\n",
      "--- total time for blast 668: 10.102031707763672\n",
      "-- processing region: 669 with 239 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_669.fasta\n",
      "--- *blasting region: 669\n",
      "--- total time for blast 669: 9.525931596755981\n",
      "-- processing region: 67 with 197 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_67.fasta\n",
      "--- *blasting region: 67\n",
      "--- total time for blast 67: 9.286636352539062\n",
      "-- processing region: 670 with 273 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_670.fasta\n",
      "--- *blasting region: 670\n",
      "--- total time for blast 670: 20.371794939041138\n",
      "-- processing region: 672 with 280 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_672.fasta\n",
      "--- *blasting region: 672\n",
      "--- total time for blast 672: 11.764763116836548\n",
      "-- processing region: 674 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_674.fasta\n",
      "--- *blasting region: 674\n",
      "--- total time for blast 674: 13.765698194503784\n",
      "-- processing region: 675 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_675.fasta\n",
      "--- *blasting region: 675\n",
      "--- total time for blast 675: 10.329694271087646\n",
      "-- processing region: 676 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_676.fasta\n",
      "--- *blasting region: 676\n",
      "--- total time for blast 676: 8.422631740570068\n",
      "-- processing region: 677 with 106 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_677.fasta\n",
      "--- *blasting region: 677\n",
      "--- total time for blast 677: 6.167243480682373\n",
      "-- processing region: 678 with 275 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_678.fasta\n",
      "--- *blasting region: 678\n",
      "--- total time for blast 678: 14.767825603485107\n",
      "-- processing region: 679 with 186 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_679.fasta\n",
      "--- *blasting region: 679\n",
      "--- total time for blast 679: 7.548038959503174\n",
      "-- processing region: 68 with 283 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_68.fasta\n",
      "--- *blasting region: 68\n",
      "--- total time for blast 68: 10.316683769226074\n",
      "-- processing region: 680 with 199 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_680.fasta\n",
      "--- *blasting region: 680\n",
      "--- total time for blast 680: 8.466012239456177\n",
      "-- processing region: 681 with 59 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_681.fasta\n",
      "--- *blasting region: 681\n",
      "--- total time for blast 681: 5.582187175750732\n",
      "-- processing region: 682 with 138 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_682.fasta\n",
      "--- *blasting region: 682\n",
      "--- total time for blast 682: 11.810394525527954\n",
      "-- processing region: 683 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_683.fasta\n",
      "--- *blasting region: 683\n",
      "--- total time for blast 683: 10.726013898849487\n",
      "-- processing region: 684 with 231 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_684.fasta\n",
      "--- *blasting region: 684\n",
      "--- total time for blast 684: 9.304261922836304\n",
      "-- processing region: 685 with 282 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_685.fasta\n",
      "--- *blasting region: 685\n",
      "--- total time for blast 685: 11.65713906288147\n",
      "-- processing region: 686 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_686.fasta\n",
      "--- *blasting region: 686\n",
      "--- total time for blast 686: 11.043228387832642\n",
      "-- processing region: 687 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_687.fasta\n",
      "--- *blasting region: 687\n",
      "--- total time for blast 687: 8.728926658630371\n",
      "-- processing region: 688 with 172 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_688.fasta\n",
      "--- *blasting region: 688\n",
      "--- total time for blast 688: 13.202331781387329\n",
      "-- processing region: 689 with 115 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_689.fasta\n",
      "--- *blasting region: 689\n",
      "--- total time for blast 689: 6.519532203674316\n",
      "-- processing region: 69 with 113 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_69.fasta\n",
      "--- *blasting region: 69\n",
      "--- total time for blast 69: 10.693283081054688\n",
      "-- processing region: 690 with 128 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_690.fasta\n",
      "--- *blasting region: 690\n",
      "--- total time for blast 690: 8.394349336624146\n",
      "-- processing region: 691 with 140 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_691.fasta\n",
      "--- *blasting region: 691\n",
      "--- total time for blast 691: 8.048995733261108\n",
      "-- processing region: 692 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_692.fasta\n",
      "--- *blasting region: 692\n",
      "--- total time for blast 692: 12.898229837417603\n",
      "-- processing region: 693 with 311 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_693.fasta\n",
      "--- *blasting region: 693\n",
      "--- total time for blast 693: 11.30678939819336\n",
      "-- processing region: 694 with 194 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_694.fasta\n",
      "--- *blasting region: 694\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 694: 8.245142936706543\n",
      "-- processing region: 695 with 211 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_695.fasta\n",
      "--- *blasting region: 695\n",
      "--- total time for blast 695: 9.841764211654663\n",
      "-- processing region: 696 with 266 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_696.fasta\n",
      "--- *blasting region: 696\n",
      "--- total time for blast 696: 9.457723617553711\n",
      "-- processing region: 697 with 242 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_697.fasta\n",
      "--- *blasting region: 697\n",
      "--- total time for blast 697: 9.101778745651245\n",
      "-- processing region: 698 with 255 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_698.fasta\n",
      "--- *blasting region: 698\n",
      "--- total time for blast 698: 9.57881498336792\n",
      "-- processing region: 699 with 289 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_699.fasta\n",
      "--- *blasting region: 699\n",
      "--- total time for blast 699: 13.141788005828857\n",
      "-- processing region: 7 with 210 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_7.fasta\n",
      "--- *blasting region: 7\n",
      "--- total time for blast 7: 11.29218053817749\n",
      "-- processing region: 70 with 209 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_70.fasta\n",
      "--- *blasting region: 70\n",
      "--- total time for blast 70: 8.623758316040039\n",
      "-- processing region: 700 with 167 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_700.fasta\n",
      "--- *blasting region: 700\n",
      "--- total time for blast 700: 7.791799545288086\n",
      "-- processing region: 701 with 147 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_701.fasta\n",
      "--- *blasting region: 701\n",
      "--- total time for blast 701: 8.373236656188965\n",
      "-- processing region: 702 with 128 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_702.fasta\n",
      "--- *blasting region: 702\n",
      "--- total time for blast 702: 7.2303853034973145\n",
      "-- processing region: 703 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_703.fasta\n",
      "--- *blasting region: 703\n",
      "--- total time for blast 703: 8.72521162033081\n",
      "-- processing region: 706 with 279 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_706.fasta\n",
      "--- *blasting region: 706\n",
      "--- total time for blast 706: 10.977068424224854\n",
      "-- processing region: 707 with 222 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_707.fasta\n",
      "--- *blasting region: 707\n",
      "--- total time for blast 707: 10.521835088729858\n",
      "-- processing region: 708 with 288 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_708.fasta\n",
      "--- *blasting region: 708\n",
      "--- total time for blast 708: 12.929791688919067\n",
      "-- processing region: 709 with 112 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_709.fasta\n",
      "--- *blasting region: 709\n",
      "--- total time for blast 709: 11.589232683181763\n",
      "-- processing region: 71 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_71.fasta\n",
      "--- *blasting region: 71\n",
      "--- total time for blast 71: 8.544364213943481\n",
      "-- processing region: 710 with 214 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_710.fasta\n",
      "--- *blasting region: 710\n",
      "--- total time for blast 710: 8.106709480285645\n",
      "-- processing region: 711 with 244 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_711.fasta\n",
      "--- *blasting region: 711\n",
      "--- total time for blast 711: 7.779181957244873\n",
      "-- processing region: 713 with 288 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_713.fasta\n",
      "--- *blasting region: 713\n",
      "--- total time for blast 713: 10.72453761100769\n",
      "-- processing region: 714 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_714.fasta\n",
      "--- *blasting region: 714\n",
      "--- total time for blast 714: 10.854057550430298\n",
      "-- processing region: 715 with 245 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_715.fasta\n",
      "--- *blasting region: 715\n",
      "--- total time for blast 715: 10.411468744277954\n",
      "-- processing region: 716 with 101 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_716.fasta\n",
      "--- *blasting region: 716\n",
      "--- total time for blast 716: 9.393499374389648\n",
      "-- processing region: 717 with 223 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_717.fasta\n",
      "--- *blasting region: 717\n",
      "--- total time for blast 717: 8.309442281723022\n",
      "-- processing region: 718 with 123 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_718.fasta\n",
      "--- *blasting region: 718\n",
      "--- total time for blast 718: 5.665219306945801\n",
      "-- processing region: 719 with 239 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_719.fasta\n",
      "--- *blasting region: 719\n",
      "--- total time for blast 719: 12.65081787109375\n",
      "-- processing region: 72 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_72.fasta\n",
      "--- *blasting region: 72\n",
      "--- total time for blast 72: 9.414396524429321\n",
      "-- processing region: 720 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_720.fasta\n",
      "--- *blasting region: 720\n",
      "--- total time for blast 720: 9.527446508407593\n",
      "-- processing region: 721 with 225 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_721.fasta\n",
      "--- *blasting region: 721\n",
      "--- total time for blast 721: 9.024595022201538\n",
      "-- processing region: 722 with 136 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_722.fasta\n",
      "--- *blasting region: 722\n",
      "--- total time for blast 722: 5.761290788650513\n",
      "-- processing region: 723 with 265 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_723.fasta\n",
      "--- *blasting region: 723\n",
      "--- total time for blast 723: 10.340336561203003\n",
      "-- processing region: 724 with 276 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_724.fasta\n",
      "--- *blasting region: 724\n",
      "--- total time for blast 724: 10.791040420532227\n",
      "-- processing region: 726 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_726.fasta\n",
      "--- *blasting region: 726\n",
      "--- total time for blast 726: 8.311585903167725\n",
      "-- processing region: 727 with 279 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_727.fasta\n",
      "--- *blasting region: 727\n",
      "--- total time for blast 727: 10.89900827407837\n",
      "-- processing region: 728 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_728.fasta\n",
      "--- *blasting region: 728\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 728: 12.205148220062256\n",
      "-- processing region: 729 with 244 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_729.fasta\n",
      "--- *blasting region: 729\n",
      "--- total time for blast 729: 22.33016562461853\n",
      "-- processing region: 73 with 282 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_73.fasta\n",
      "--- *blasting region: 73\n",
      "--- total time for blast 73: 9.320429563522339\n",
      "-- processing region: 730 with 295 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_730.fasta\n",
      "--- *blasting region: 730\n",
      "--- total time for blast 730: 11.802908182144165\n",
      "-- processing region: 731 with 301 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_731.fasta\n",
      "--- *blasting region: 731\n",
      "--- total time for blast 731: 11.463462591171265\n",
      "-- processing region: 732 with 269 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_732.fasta\n",
      "--- *blasting region: 732\n",
      "--- total time for blast 732: 9.924652338027954\n",
      "-- processing region: 733 with 281 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_733.fasta\n",
      "--- *blasting region: 733\n",
      "--- total time for blast 733: 10.35732626914978\n",
      "-- processing region: 734 with 318 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_734.fasta\n",
      "--- *blasting region: 734\n",
      "--- total time for blast 734: 8.605348587036133\n",
      "-- processing region: 735 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_735.fasta\n",
      "--- *blasting region: 735\n",
      "--- total time for blast 735: 11.26425814628601\n",
      "-- processing region: 736 with 173 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_736.fasta\n",
      "--- *blasting region: 736\n",
      "--- total time for blast 736: 7.850847482681274\n",
      "-- processing region: 737 with 148 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_737.fasta\n",
      "--- *blasting region: 737\n",
      "--- total time for blast 737: 8.368232727050781\n",
      "-- processing region: 739 with 293 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_739.fasta\n",
      "--- *blasting region: 739\n",
      "--- total time for blast 739: 9.268903255462646\n",
      "-- processing region: 74 with 276 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_74.fasta\n",
      "--- *blasting region: 74\n",
      "--- total time for blast 74: 8.172115564346313\n",
      "-- processing region: 740 with 262 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_740.fasta\n",
      "--- *blasting region: 740\n",
      "--- total time for blast 740: 9.36099123954773\n",
      "-- processing region: 741 with 278 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_741.fasta\n",
      "--- *blasting region: 741\n",
      "--- total time for blast 741: 9.814507007598877\n",
      "-- processing region: 742 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_742.fasta\n",
      "--- *blasting region: 742\n",
      "--- total time for blast 742: 8.75726056098938\n",
      "-- processing region: 743 with 91 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_743.fasta\n",
      "--- *blasting region: 743\n",
      "--- total time for blast 743: 6.263665437698364\n",
      "-- processing region: 744 with 144 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_744.fasta\n",
      "--- *blasting region: 744\n",
      "--- total time for blast 744: 11.350460290908813\n",
      "-- processing region: 745 with 225 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_745.fasta\n",
      "--- *blasting region: 745\n",
      "--- total time for blast 745: 8.131048440933228\n",
      "-- processing region: 746 with 149 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_746.fasta\n",
      "--- *blasting region: 746\n",
      "--- total time for blast 746: 13.90530776977539\n",
      "-- processing region: 747 with 118 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_747.fasta\n",
      "--- *blasting region: 747\n",
      "--- total time for blast 747: 19.882632732391357\n",
      "-- processing region: 748 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_748.fasta\n",
      "--- *blasting region: 748\n",
      "--- total time for blast 748: 10.128000497817993\n",
      "-- processing region: 749 with 224 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_749.fasta\n",
      "--- *blasting region: 749\n",
      "--- total time for blast 749: 9.080754518508911\n",
      "-- processing region: 75 with 278 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_75.fasta\n",
      "--- *blasting region: 75\n",
      "--- total time for blast 75: 12.775014400482178\n",
      "-- processing region: 750 with 216 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_750.fasta\n",
      "--- *blasting region: 750\n",
      "--- total time for blast 750: 10.626471996307373\n",
      "-- processing region: 751 with 166 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_751.fasta\n",
      "--- *blasting region: 751\n",
      "--- total time for blast 751: 8.500980138778687\n",
      "-- processing region: 752 with 264 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_752.fasta\n",
      "--- *blasting region: 752\n",
      "--- total time for blast 752: 9.416529893875122\n",
      "-- processing region: 753 with 225 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_753.fasta\n",
      "--- *blasting region: 753\n",
      "--- total time for blast 753: 8.437283515930176\n",
      "-- processing region: 754 with 186 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_754.fasta\n",
      "--- *blasting region: 754\n",
      "--- total time for blast 754: 8.632895708084106\n",
      "-- processing region: 755 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_755.fasta\n",
      "--- *blasting region: 755\n",
      "--- total time for blast 755: 9.126797914505005\n",
      "-- processing region: 756 with 281 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_756.fasta\n",
      "--- *blasting region: 756\n",
      "--- total time for blast 756: 10.655698776245117\n",
      "-- processing region: 757 with 233 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_757.fasta\n",
      "--- *blasting region: 757\n",
      "--- total time for blast 757: 11.030214548110962\n",
      "-- processing region: 758 with 197 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_758.fasta\n",
      "--- *blasting region: 758\n",
      "--- total time for blast 758: 11.519588708877563\n",
      "-- processing region: 759 with 223 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_759.fasta\n",
      "--- *blasting region: 759\n",
      "--- total time for blast 759: 8.992689371109009\n",
      "-- processing region: 76 with 265 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_76.fasta\n",
      "--- *blasting region: 76\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 76: 10.114533424377441\n",
      "-- processing region: 765 with 242 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_765.fasta\n",
      "--- *blasting region: 765\n",
      "--- total time for blast 765: 11.065551996231079\n",
      "-- processing region: 766 with 97 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_766.fasta\n",
      "--- *blasting region: 766\n",
      "--- total time for blast 766: 7.359191656112671\n",
      "-- processing region: 767 with 272 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_767.fasta\n",
      "--- *blasting region: 767\n",
      "--- total time for blast 767: 9.542107105255127\n",
      "-- processing region: 768 with 123 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_768.fasta\n",
      "--- *blasting region: 768\n",
      "--- total time for blast 768: 5.474076986312866\n",
      "-- processing region: 769 with 230 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_769.fasta\n",
      "--- *blasting region: 769\n",
      "--- total time for blast 769: 8.568110704421997\n",
      "-- processing region: 77 with 193 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_77.fasta\n",
      "--- *blasting region: 77\n",
      "--- total time for blast 77: 7.54339075088501\n",
      "-- processing region: 770 with 313 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_770.fasta\n",
      "--- *blasting region: 770\n",
      "--- total time for blast 770: 12.8189857006073\n",
      "-- processing region: 772 with 238 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_772.fasta\n",
      "--- *blasting region: 772\n",
      "--- total time for blast 772: 11.053122282028198\n",
      "-- processing region: 773 with 293 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_773.fasta\n",
      "--- *blasting region: 773\n",
      "--- total time for blast 773: 7.567636013031006\n",
      "-- processing region: 774 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_774.fasta\n",
      "--- *blasting region: 774\n",
      "--- total time for blast 774: 8.18717336654663\n",
      "-- processing region: 775 with 185 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_775.fasta\n",
      "--- *blasting region: 775\n",
      "--- total time for blast 775: 10.508753776550293\n",
      "-- processing region: 776 with 168 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_776.fasta\n",
      "--- *blasting region: 776\n",
      "--- total time for blast 776: 8.827574968338013\n",
      "-- processing region: 779 with 233 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_779.fasta\n",
      "--- *blasting region: 779\n",
      "--- total time for blast 779: 9.88883090019226\n",
      "-- processing region: 780 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_780.fasta\n",
      "--- *blasting region: 780\n",
      "--- total time for blast 780: 10.541487216949463\n",
      "-- processing region: 781 with 148 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_781.fasta\n",
      "--- *blasting region: 781\n",
      "--- total time for blast 781: 7.42424201965332\n",
      "-- processing region: 783 with 72 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_783.fasta\n",
      "--- *blasting region: 783\n",
      "--- total time for blast 783: 4.820590257644653\n",
      "-- processing region: 784 with 210 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_784.fasta\n",
      "--- *blasting region: 784\n",
      "--- total time for blast 784: 9.46371078491211\n",
      "-- processing region: 785 with 230 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_785.fasta\n",
      "--- *blasting region: 785\n",
      "--- total time for blast 785: 9.886800765991211\n",
      "-- processing region: 786 with 319 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_786.fasta\n",
      "--- *blasting region: 786\n",
      "--- total time for blast 786: 9.952414989471436\n",
      "-- processing region: 787 with 252 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_787.fasta\n",
      "--- *blasting region: 787\n",
      "--- total time for blast 787: 10.077855825424194\n",
      "-- processing region: 788 with 146 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_788.fasta\n",
      "--- *blasting region: 788\n",
      "--- total time for blast 788: 5.659215211868286\n",
      "-- processing region: 789 with 291 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_789.fasta\n",
      "--- *blasting region: 789\n",
      "--- total time for blast 789: 8.14540982246399\n",
      "-- processing region: 790 with 214 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_790.fasta\n",
      "--- *blasting region: 790\n",
      "--- total time for blast 790: 9.778286695480347\n",
      "-- processing region: 791 with 281 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_791.fasta\n",
      "--- *blasting region: 791\n",
      "--- total time for blast 791: 8.145062923431396\n",
      "-- processing region: 792 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_792.fasta\n",
      "--- *blasting region: 792\n",
      "--- total time for blast 792: 15.03066635131836\n",
      "-- processing region: 793 with 170 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_793.fasta\n",
      "--- *blasting region: 793\n",
      "--- total time for blast 793: 10.688356161117554\n",
      "-- processing region: 794 with 126 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_794.fasta\n",
      "--- *blasting region: 794\n",
      "--- total time for blast 794: 11.570621728897095\n",
      "-- processing region: 795 with 232 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_795.fasta\n",
      "--- *blasting region: 795\n",
      "--- total time for blast 795: 8.927510976791382\n",
      "-- processing region: 796 with 144 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_796.fasta\n",
      "--- *blasting region: 796\n",
      "--- total time for blast 796: 6.020914077758789\n",
      "-- processing region: 797 with 173 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_797.fasta\n",
      "--- *blasting region: 797\n",
      "--- total time for blast 797: 10.749006032943726\n",
      "-- processing region: 798 with 137 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_798.fasta\n",
      "--- *blasting region: 798\n",
      "--- total time for blast 798: 8.266058683395386\n",
      "-- processing region: 799 with 132 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_799.fasta\n",
      "--- *blasting region: 799\n",
      "--- total time for blast 799: 15.079763889312744\n",
      "-- processing region: 8 with 274 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_8.fasta\n",
      "--- *blasting region: 8\n",
      "--- total time for blast 8: 13.324287176132202\n",
      "-- processing region: 80 with 287 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_80.fasta\n",
      "--- *blasting region: 80\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 80: 10.49958848953247\n",
      "-- processing region: 800 with 169 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_800.fasta\n",
      "--- *blasting region: 800\n",
      "--- total time for blast 800: 7.0898706912994385\n",
      "-- processing region: 801 with 200 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_801.fasta\n",
      "--- *blasting region: 801\n",
      "--- total time for blast 801: 10.283659219741821\n",
      "-- processing region: 802 with 275 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_802.fasta\n",
      "--- *blasting region: 802\n",
      "--- total time for blast 802: 8.330924272537231\n",
      "-- processing region: 803 with 188 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_803.fasta\n",
      "--- *blasting region: 803\n",
      "--- total time for blast 803: 10.394797325134277\n",
      "-- processing region: 804 with 176 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_804.fasta\n",
      "--- *blasting region: 804\n",
      "--- total time for blast 804: 9.764726400375366\n",
      "-- processing region: 805 with 137 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_805.fasta\n",
      "--- *blasting region: 805\n",
      "--- total time for blast 805: 7.536639928817749\n",
      "-- processing region: 806 with 135 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_806.fasta\n",
      "--- *blasting region: 806\n",
      "--- total time for blast 806: 14.237587213516235\n",
      "-- processing region: 807 with 199 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_807.fasta\n",
      "--- *blasting region: 807\n",
      "--- total time for blast 807: 10.036539554595947\n",
      "-- processing region: 808 with 218 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_808.fasta\n",
      "--- *blasting region: 808\n",
      "--- total time for blast 808: 9.161823987960815\n",
      "-- processing region: 809 with 344 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_809.fasta\n",
      "--- *blasting region: 809\n",
      "--- total time for blast 809: 13.670360565185547\n",
      "-- processing region: 81 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_81.fasta\n",
      "--- *blasting region: 81\n",
      "--- total time for blast 81: 12.505646467208862\n",
      "-- processing region: 810 with 178 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_810.fasta\n",
      "--- *blasting region: 810\n",
      "--- total time for blast 810: 9.240257501602173\n",
      "-- processing region: 812 with 179 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_812.fasta\n",
      "--- *blasting region: 812\n",
      "--- total time for blast 812: 7.879250764846802\n",
      "-- processing region: 813 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_813.fasta\n",
      "--- *blasting region: 813\n",
      "--- total time for blast 813: 9.693508625030518\n",
      "-- processing region: 814 with 186 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_814.fasta\n",
      "--- *blasting region: 814\n",
      "--- total time for blast 814: 8.090025424957275\n",
      "-- processing region: 815 with 228 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_815.fasta\n",
      "--- *blasting region: 815\n",
      "--- total time for blast 815: 12.939265489578247\n",
      "-- processing region: 816 with 181 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_816.fasta\n",
      "--- *blasting region: 816\n",
      "--- total time for blast 816: 10.092516899108887\n",
      "-- processing region: 817 with 183 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_817.fasta\n",
      "--- *blasting region: 817\n",
      "--- total time for blast 817: 9.661837100982666\n",
      "-- processing region: 818 with 198 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_818.fasta\n",
      "--- *blasting region: 818\n",
      "--- total time for blast 818: 15.887382984161377\n",
      "-- processing region: 819 with 182 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_819.fasta\n",
      "--- *blasting region: 819\n",
      "--- total time for blast 819: 8.681108713150024\n",
      "-- processing region: 82 with 223 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_82.fasta\n",
      "--- *blasting region: 82\n",
      "--- total time for blast 82: 9.717220067977905\n",
      "-- processing region: 820 with 115 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_820.fasta\n",
      "--- *blasting region: 820\n",
      "--- total time for blast 820: 10.426429271697998\n",
      "-- processing region: 821 with 79 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_821.fasta\n",
      "--- *blasting region: 821\n",
      "--- total time for blast 821: 6.39976692199707\n",
      "-- processing region: 822 with 196 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_822.fasta\n",
      "--- *blasting region: 822\n",
      "--- total time for blast 822: 10.066518545150757\n",
      "-- processing region: 823 with 341 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_823.fasta\n",
      "--- *blasting region: 823\n",
      "--- total time for blast 823: 11.345845460891724\n",
      "-- processing region: 824 with 99 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_824.fasta\n",
      "--- *blasting region: 824\n",
      "--- total time for blast 824: 8.242491006851196\n",
      "-- processing region: 825 with 216 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_825.fasta\n",
      "--- *blasting region: 825\n",
      "--- total time for blast 825: 10.848796129226685\n",
      "-- processing region: 827 with 143 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_827.fasta\n",
      "--- *blasting region: 827\n",
      "--- total time for blast 827: 6.229640245437622\n",
      "-- processing region: 828 with 103 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_828.fasta\n",
      "--- *blasting region: 828\n",
      "--- total time for blast 828: 4.988129138946533\n",
      "-- processing region: 829 with 122 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_829.fasta\n",
      "--- *blasting region: 829\n",
      "--- total time for blast 829: 7.533558130264282\n",
      "-- processing region: 83 with 233 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_83.fasta\n",
      "--- *blasting region: 83\n",
      "--- total time for blast 83: 8.732577800750732\n",
      "-- processing region: 830 with 168 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_830.fasta\n",
      "--- *blasting region: 830\n",
      "--- total time for blast 830: 8.07601523399353\n",
      "-- processing region: 831 with 227 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_831.fasta\n",
      "--- *blasting region: 831\n",
      "--- total time for blast 831: 8.862838983535767\n",
      "-- processing region: 832 with 232 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_832.fasta\n",
      "--- *blasting region: 832\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 832: 10.601896524429321\n",
      "-- processing region: 833 with 153 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_833.fasta\n",
      "--- *blasting region: 833\n",
      "--- total time for blast 833: 12.150683164596558\n",
      "-- processing region: 834 with 297 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_834.fasta\n",
      "--- *blasting region: 834\n",
      "--- total time for blast 834: 9.690229177474976\n",
      "-- processing region: 835 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_835.fasta\n",
      "--- *blasting region: 835\n",
      "--- total time for blast 835: 14.07128381729126\n",
      "-- processing region: 836 with 296 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_836.fasta\n",
      "--- *blasting region: 836\n",
      "--- total time for blast 836: 17.3085675239563\n",
      "-- processing region: 837 with 312 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_837.fasta\n",
      "--- *blasting region: 837\n",
      "--- total time for blast 837: 22.389033794403076\n",
      "-- processing region: 838 with 261 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_838.fasta\n",
      "--- *blasting region: 838\n",
      "--- total time for blast 838: 11.406495809555054\n",
      "-- processing region: 839 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_839.fasta\n",
      "--- *blasting region: 839\n",
      "--- total time for blast 839: 9.38882040977478\n",
      "-- processing region: 84 with 290 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_84.fasta\n",
      "--- *blasting region: 84\n",
      "--- total time for blast 84: 14.188207149505615\n",
      "-- processing region: 840 with 333 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_840.fasta\n",
      "--- *blasting region: 840\n",
      "--- total time for blast 840: 10.128736734390259\n",
      "-- processing region: 841 with 328 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_841.fasta\n",
      "--- *blasting region: 841\n",
      "--- total time for blast 841: 11.801852703094482\n",
      "-- processing region: 842 with 298 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_842.fasta\n",
      "--- *blasting region: 842\n",
      "--- total time for blast 842: 10.769277572631836\n",
      "-- processing region: 843 with 248 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_843.fasta\n",
      "--- *blasting region: 843\n",
      "--- total time for blast 843: 9.900880336761475\n",
      "-- processing region: 844 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_844.fasta\n",
      "--- *blasting region: 844\n",
      "--- total time for blast 844: 9.681206226348877\n",
      "-- processing region: 845 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_845.fasta\n",
      "--- *blasting region: 845\n",
      "--- total time for blast 845: 9.403817176818848\n",
      "-- processing region: 846 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_846.fasta\n",
      "--- *blasting region: 846\n",
      "--- total time for blast 846: 9.124220609664917\n",
      "-- processing region: 847 with 258 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_847.fasta\n",
      "--- *blasting region: 847\n",
      "--- total time for blast 847: 9.858624458312988\n",
      "-- processing region: 848 with 262 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_848.fasta\n",
      "--- *blasting region: 848\n",
      "--- total time for blast 848: 11.013202428817749\n",
      "-- processing region: 849 with 227 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_849.fasta\n",
      "--- *blasting region: 849\n",
      "--- total time for blast 849: 8.281167984008789\n",
      "-- processing region: 85 with 173 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_85.fasta\n",
      "--- *blasting region: 85\n",
      "--- total time for blast 85: 7.292561769485474\n",
      "-- processing region: 850 with 246 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_850.fasta\n",
      "--- *blasting region: 850\n",
      "--- total time for blast 850: 10.443804740905762\n",
      "-- processing region: 851 with 289 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_851.fasta\n",
      "--- *blasting region: 851\n",
      "--- total time for blast 851: 10.301011323928833\n",
      "-- processing region: 852 with 148 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_852.fasta\n",
      "--- *blasting region: 852\n",
      "--- total time for blast 852: 7.921090841293335\n",
      "-- processing region: 854 with 234 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_854.fasta\n",
      "--- *blasting region: 854\n",
      "--- total time for blast 854: 16.416653871536255\n",
      "-- processing region: 855 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_855.fasta\n",
      "--- *blasting region: 855\n",
      "--- total time for blast 855: 10.635056257247925\n",
      "-- processing region: 856 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_856.fasta\n",
      "--- *blasting region: 856\n",
      "--- total time for blast 856: 12.876470804214478\n",
      "-- processing region: 857 with 226 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_857.fasta\n",
      "--- *blasting region: 857\n",
      "--- total time for blast 857: 11.038415431976318\n",
      "-- processing region: 858 with 200 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_858.fasta\n",
      "--- *blasting region: 858\n",
      "--- total time for blast 858: 7.932796001434326\n",
      "-- processing region: 859 with 234 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_859.fasta\n",
      "--- *blasting region: 859\n",
      "--- total time for blast 859: 11.391401290893555\n",
      "-- processing region: 86 with 202 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_86.fasta\n",
      "--- *blasting region: 86\n",
      "--- total time for blast 86: 11.20004677772522\n",
      "-- processing region: 860 with 217 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_860.fasta\n",
      "--- *blasting region: 860\n",
      "--- total time for blast 860: 9.64051604270935\n",
      "-- processing region: 861 with 323 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_861.fasta\n",
      "--- *blasting region: 861\n",
      "--- total time for blast 861: 13.254199981689453\n",
      "-- processing region: 862 with 101 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_862.fasta\n",
      "--- *blasting region: 862\n",
      "--- total time for blast 862: 9.809550285339355\n",
      "-- processing region: 863 with 150 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_863.fasta\n",
      "--- *blasting region: 863\n",
      "--- total time for blast 863: 6.900139570236206\n",
      "-- processing region: 864 with 258 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_864.fasta\n",
      "--- *blasting region: 864\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 864: 9.790291547775269\n",
      "-- processing region: 865 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_865.fasta\n",
      "--- *blasting region: 865\n",
      "--- total time for blast 865: 12.275944709777832\n",
      "-- processing region: 866 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_866.fasta\n",
      "--- *blasting region: 866\n",
      "--- total time for blast 866: 10.985858678817749\n",
      "-- processing region: 867 with 145 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_867.fasta\n",
      "--- *blasting region: 867\n",
      "--- total time for blast 867: 8.443289518356323\n",
      "-- processing region: 868 with 294 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_868.fasta\n",
      "--- *blasting region: 868\n",
      "--- total time for blast 868: 11.729735612869263\n",
      "-- processing region: 869 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_869.fasta\n",
      "--- *blasting region: 869\n",
      "--- total time for blast 869: 8.777537822723389\n",
      "-- processing region: 87 with 154 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_87.fasta\n",
      "--- *blasting region: 87\n",
      "--- total time for blast 87: 7.725754261016846\n",
      "-- processing region: 870 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_870.fasta\n",
      "--- *blasting region: 870\n",
      "--- total time for blast 870: 10.187500715255737\n",
      "-- processing region: 871 with 184 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_871.fasta\n",
      "--- *blasting region: 871\n",
      "--- total time for blast 871: 12.87756896018982\n",
      "-- processing region: 872 with 203 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_872.fasta\n",
      "--- *blasting region: 872\n",
      "--- total time for blast 872: 7.044479608535767\n",
      "-- processing region: 873 with 93 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_873.fasta\n",
      "--- *blasting region: 873\n",
      "--- total time for blast 873: 5.895390748977661\n",
      "-- processing region: 874 with 116 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_874.fasta\n",
      "--- *blasting region: 874\n",
      "--- total time for blast 874: 6.27228045463562\n",
      "-- processing region: 875 with 217 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_875.fasta\n",
      "--- *blasting region: 875\n",
      "--- total time for blast 875: 9.7041916847229\n",
      "-- processing region: 876 with 196 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_876.fasta\n",
      "--- *blasting region: 876\n",
      "--- total time for blast 876: 9.076205492019653\n",
      "-- processing region: 877 with 76 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_877.fasta\n",
      "--- *blasting region: 877\n",
      "--- total time for blast 877: 13.360950469970703\n",
      "-- processing region: 878 with 194 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_878.fasta\n",
      "--- *blasting region: 878\n",
      "--- total time for blast 878: 9.036731243133545\n",
      "-- processing region: 879 with 133 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_879.fasta\n",
      "--- *blasting region: 879\n",
      "--- total time for blast 879: 12.793406009674072\n",
      "-- processing region: 88 with 117 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_88.fasta\n",
      "--- *blasting region: 88\n",
      "--- total time for blast 88: 9.569820642471313\n",
      "-- processing region: 880 with 88 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_880.fasta\n",
      "--- *blasting region: 880\n",
      "--- total time for blast 880: 9.537710428237915\n",
      "-- processing region: 881 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_881.fasta\n",
      "--- *blasting region: 881\n",
      "--- total time for blast 881: 11.408527851104736\n",
      "-- processing region: 882 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_882.fasta\n",
      "--- *blasting region: 882\n",
      "--- total time for blast 882: 10.537338733673096\n",
      "-- processing region: 883 with 277 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_883.fasta\n",
      "--- *blasting region: 883\n",
      "--- total time for blast 883: 10.560184001922607\n",
      "-- processing region: 884 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_884.fasta\n",
      "--- *blasting region: 884\n",
      "--- total time for blast 884: 10.09841251373291\n",
      "-- processing region: 885 with 180 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_885.fasta\n",
      "--- *blasting region: 885\n",
      "--- total time for blast 885: 6.781578302383423\n",
      "-- processing region: 886 with 145 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_886.fasta\n",
      "--- *blasting region: 886\n",
      "--- total time for blast 886: 8.352566003799438\n",
      "-- processing region: 887 with 210 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_887.fasta\n",
      "--- *blasting region: 887\n",
      "--- total time for blast 887: 9.092681407928467\n",
      "-- processing region: 888 with 218 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_888.fasta\n",
      "--- *blasting region: 888\n",
      "--- total time for blast 888: 7.2894957065582275\n",
      "-- processing region: 889 with 194 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_889.fasta\n",
      "--- *blasting region: 889\n",
      "--- total time for blast 889: 8.145066499710083\n",
      "-- processing region: 89 with 318 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_89.fasta\n",
      "--- *blasting region: 89\n",
      "--- total time for blast 89: 11.691761493682861\n",
      "-- processing region: 890 with 295 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_890.fasta\n",
      "--- *blasting region: 890\n",
      "--- total time for blast 890: 11.030215501785278\n",
      "-- processing region: 891 with 173 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_891.fasta\n",
      "--- *blasting region: 891\n",
      "--- total time for blast 891: 11.20238471031189\n",
      "-- processing region: 894 with 228 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_894.fasta\n",
      "--- *blasting region: 894\n",
      "--- total time for blast 894: 10.63488245010376\n",
      "-- processing region: 896 with 316 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_896.fasta\n",
      "--- *blasting region: 896\n",
      "--- total time for blast 896: 11.082359790802002\n",
      "-- processing region: 897 with 259 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_897.fasta\n",
      "--- *blasting region: 897\n",
      "--- total time for blast 897: 14.546476364135742\n",
      "-- processing region: 898 with 275 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_898.fasta\n",
      "--- *blasting region: 898\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 898: 19.10613775253296\n",
      "-- processing region: 899 with 179 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_899.fasta\n",
      "--- *blasting region: 899\n",
      "--- total time for blast 899: 9.518476963043213\n",
      "-- processing region: 9 with 220 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_9.fasta\n",
      "--- *blasting region: 9\n",
      "--- total time for blast 9: 8.588090896606445\n",
      "-- processing region: 90 with 315 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_90.fasta\n",
      "--- *blasting region: 90\n",
      "--- total time for blast 90: 11.671586036682129\n",
      "-- processing region: 900 with 67 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_900.fasta\n",
      "--- *blasting region: 900\n",
      "--- total time for blast 900: 8.433280944824219\n",
      "-- processing region: 901 with 95 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_901.fasta\n",
      "--- *blasting region: 901\n",
      "--- total time for blast 901: 6.4271399974823\n",
      "-- processing region: 902 with 288 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_902.fasta\n",
      "--- *blasting region: 902\n",
      "--- total time for blast 902: 9.11553430557251\n",
      "-- processing region: 903 with 195 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_903.fasta\n",
      "--- *blasting region: 903\n",
      "--- total time for blast 903: 10.595080852508545\n",
      "-- processing region: 904 with 253 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_904.fasta\n",
      "--- *blasting region: 904\n",
      "--- total time for blast 904: 8.74819803237915\n",
      "-- processing region: 905 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_905.fasta\n",
      "--- *blasting region: 905\n",
      "--- total time for blast 905: 8.873608827590942\n",
      "-- processing region: 906 with 254 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_906.fasta\n",
      "--- *blasting region: 906\n",
      "--- total time for blast 906: 12.700733661651611\n",
      "-- processing region: 907 with 200 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_907.fasta\n",
      "--- *blasting region: 907\n",
      "--- total time for blast 907: 21.36509656906128\n",
      "-- processing region: 908 with 87 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_908.fasta\n",
      "--- *blasting region: 908\n",
      "--- total time for blast 908: 7.230755567550659\n",
      "-- processing region: 909 with 180 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_909.fasta\n",
      "--- *blasting region: 909\n",
      "--- total time for blast 909: 8.331204652786255\n",
      "-- processing region: 91 with 306 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_91.fasta\n",
      "--- *blasting region: 91\n",
      "--- total time for blast 91: 10.290665864944458\n",
      "-- processing region: 910 with 254 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_910.fasta\n",
      "--- *blasting region: 910\n",
      "--- total time for blast 910: 9.159936666488647\n",
      "-- processing region: 911 with 331 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_911.fasta\n",
      "--- *blasting region: 911\n",
      "--- total time for blast 911: 9.969401836395264\n",
      "-- processing region: 912 with 298 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_912.fasta\n",
      "--- *blasting region: 912\n",
      "--- total time for blast 912: 9.15693211555481\n",
      "-- processing region: 913 with 216 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_913.fasta\n",
      "--- *blasting region: 913\n",
      "--- total time for blast 913: 8.517343759536743\n",
      "-- processing region: 914 with 72 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_914.fasta\n",
      "--- *blasting region: 914\n",
      "--- total time for blast 914: 4.7675511837005615\n",
      "-- processing region: 915 with 256 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_915.fasta\n",
      "--- *blasting region: 915\n",
      "--- total time for blast 915: 9.218335628509521\n",
      "-- processing region: 916 with 182 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_916.fasta\n",
      "--- *blasting region: 916\n",
      "--- total time for blast 916: 16.95062494277954\n",
      "-- processing region: 917 with 159 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_917.fasta\n",
      "--- *blasting region: 917\n",
      "--- total time for blast 917: 10.467345476150513\n",
      "-- processing region: 918 with 163 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_918.fasta\n",
      "--- *blasting region: 918\n",
      "--- total time for blast 918: 8.782257795333862\n",
      "-- processing region: 919 with 273 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_919.fasta\n",
      "--- *blasting region: 919\n",
      "--- total time for blast 919: 14.091220140457153\n",
      "-- processing region: 92 with 335 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_92.fasta\n",
      "--- *blasting region: 92\n",
      "--- total time for blast 92: 13.330928325653076\n",
      "-- processing region: 920 with 287 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_920.fasta\n",
      "--- *blasting region: 920\n",
      "--- total time for blast 920: 11.59700894355774\n",
      "-- processing region: 921 with 325 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_921.fasta\n",
      "--- *blasting region: 921\n",
      "--- total time for blast 921: 10.827773809432983\n",
      "-- processing region: 922 with 285 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_922.fasta\n",
      "--- *blasting region: 922\n",
      "--- total time for blast 922: 17.08163857460022\n",
      "-- processing region: 923 with 160 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_923.fasta\n",
      "--- *blasting region: 923\n",
      "--- total time for blast 923: 13.641270160675049\n",
      "-- processing region: 924 with 197 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_924.fasta\n",
      "--- *blasting region: 924\n",
      "--- total time for blast 924: 7.188707590103149\n",
      "-- processing region: 925 with 79 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_925.fasta\n",
      "--- *blasting region: 925\n",
      "--- total time for blast 925: 5.89397120475769\n",
      "-- processing region: 926 with 254 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_926.fasta\n",
      "--- *blasting region: 926\n",
      "--- total time for blast 926: 9.813573837280273\n",
      "-- processing region: 927 with 303 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_927.fasta\n",
      "--- *blasting region: 927\n",
      "--- total time for blast 927: 8.736507654190063\n",
      "-- processing region: 928 with 266 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_928.fasta\n",
      "--- *blasting region: 928\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 928: 7.837761163711548\n",
      "-- processing region: 929 with 125 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_929.fasta\n",
      "--- *blasting region: 929\n",
      "--- total time for blast 929: 12.119728088378906\n",
      "-- processing region: 93 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_93.fasta\n",
      "--- *blasting region: 93\n",
      "--- total time for blast 93: 11.370543956756592\n",
      "-- processing region: 930 with 81 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_930.fasta\n",
      "--- *blasting region: 930\n",
      "--- total time for blast 930: 16.169583797454834\n",
      "-- processing region: 932 with 66 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_932.fasta\n",
      "--- *blasting region: 932\n",
      "--- total time for blast 932: 12.99522089958191\n",
      "-- processing region: 933 with 186 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_933.fasta\n",
      "--- *blasting region: 933\n",
      "--- total time for blast 933: 8.278115272521973\n",
      "-- processing region: 934 with 198 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_934.fasta\n",
      "--- *blasting region: 934\n",
      "--- total time for blast 934: 12.772512912750244\n",
      "-- processing region: 935 with 230 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_935.fasta\n",
      "--- *blasting region: 935\n",
      "--- total time for blast 935: 9.592683792114258\n",
      "-- processing region: 936 with 293 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_936.fasta\n",
      "--- *blasting region: 936\n",
      "--- total time for blast 936: 10.436262369155884\n",
      "-- processing region: 937 with 200 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_937.fasta\n",
      "--- *blasting region: 937\n",
      "--- total time for blast 937: 9.207470893859863\n",
      "-- processing region: 938 with 164 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_938.fasta\n",
      "--- *blasting region: 938\n",
      "--- total time for blast 938: 16.285316467285156\n",
      "-- processing region: 939 with 292 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_939.fasta\n",
      "--- *blasting region: 939\n",
      "--- total time for blast 939: 14.635601997375488\n",
      "-- processing region: 94 with 207 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_94.fasta\n",
      "--- *blasting region: 94\n",
      "--- total time for blast 94: 8.096778392791748\n",
      "-- processing region: 940 with 258 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_940.fasta\n",
      "--- *blasting region: 940\n",
      "--- total time for blast 940: 10.62495470046997\n",
      "-- processing region: 941 with 184 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_941.fasta\n",
      "--- *blasting region: 941\n",
      "--- total time for blast 941: 13.066048383712769\n",
      "-- processing region: 942 with 136 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_942.fasta\n",
      "--- *blasting region: 942\n",
      "--- total time for blast 942: 10.210608005523682\n",
      "-- processing region: 943 with 173 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_943.fasta\n",
      "--- *blasting region: 943\n",
      "--- total time for blast 943: 7.45086407661438\n",
      "-- processing region: 944 with 257 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_944.fasta\n",
      "--- *blasting region: 944\n",
      "--- total time for blast 944: 11.594271898269653\n",
      "-- processing region: 945 with 216 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_945.fasta\n",
      "--- *blasting region: 945\n",
      "--- total time for blast 945: 11.303930759429932\n",
      "-- processing region: 946 with 262 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_946.fasta\n",
      "--- *blasting region: 946\n",
      "--- total time for blast 946: 11.481302261352539\n",
      "-- processing region: 947 with 244 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_947.fasta\n",
      "--- *blasting region: 947\n",
      "--- total time for blast 947: 13.293322086334229\n",
      "-- processing region: 948 with 187 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_948.fasta\n",
      "--- *blasting region: 948\n",
      "--- total time for blast 948: 10.695626735687256\n",
      "-- processing region: 949 with 247 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_949.fasta\n",
      "--- *blasting region: 949\n",
      "--- total time for blast 949: 8.885176181793213\n",
      "-- processing region: 95 with 185 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_95.fasta\n",
      "--- *blasting region: 95\n",
      "--- total time for blast 95: 8.688806295394897\n",
      "-- processing region: 950 with 154 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_950.fasta\n",
      "--- *blasting region: 950\n",
      "--- total time for blast 950: 9.529747009277344\n",
      "-- processing region: 951 with 212 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_951.fasta\n",
      "--- *blasting region: 951\n",
      "--- total time for blast 951: 9.135878562927246\n",
      "-- processing region: 952 with 260 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_952.fasta\n",
      "--- *blasting region: 952\n",
      "--- total time for blast 952: 10.92445707321167\n",
      "-- processing region: 953 with 244 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_953.fasta\n",
      "--- *blasting region: 953\n",
      "--- total time for blast 953: 8.511919260025024\n",
      "-- processing region: 954 with 190 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_954.fasta\n",
      "--- *blasting region: 954\n",
      "--- total time for blast 954: 8.717352628707886\n",
      "-- processing region: 955 with 270 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_955.fasta\n",
      "--- *blasting region: 955\n",
      "--- total time for blast 955: 13.079073429107666\n",
      "-- processing region: 956 with 196 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_956.fasta\n",
      "--- *blasting region: 956\n",
      "--- total time for blast 956: 12.214359998703003\n",
      "-- processing region: 957 with 238 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_957.fasta\n",
      "--- *blasting region: 957\n",
      "--- total time for blast 957: 19.788414239883423\n",
      "-- processing region: 958 with 254 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_958.fasta\n",
      "--- *blasting region: 958\n",
      "--- total time for blast 958: 9.225471496582031\n",
      "-- processing region: 959 with 209 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_959.fasta\n",
      "--- *blasting region: 959\n",
      "--- total time for blast 959: 11.959296226501465\n",
      "-- processing region: 96 with 207 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_96.fasta\n",
      "--- *blasting region: 96\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 96: 9.930009365081787\n",
      "-- processing region: 960 with 288 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_960.fasta\n",
      "--- *blasting region: 960\n",
      "--- total time for blast 960: 18.17253541946411\n",
      "-- processing region: 961 with 284 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_961.fasta\n",
      "--- *blasting region: 961\n",
      "--- total time for blast 961: 10.132984399795532\n",
      "-- processing region: 962 with 278 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_962.fasta\n",
      "--- *blasting region: 962\n",
      "--- total time for blast 962: 10.54274845123291\n",
      "-- processing region: 963 with 282 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_963.fasta\n",
      "--- *blasting region: 963\n",
      "--- total time for blast 963: 17.584444284439087\n",
      "-- processing region: 964 with 269 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_964.fasta\n",
      "--- *blasting region: 964\n",
      "--- total time for blast 964: 9.268792629241943\n",
      "-- processing region: 965 with 125 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_965.fasta\n",
      "--- *blasting region: 965\n",
      "--- total time for blast 965: 6.7950615882873535\n",
      "-- processing region: 966 with 213 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_966.fasta\n",
      "--- *blasting region: 966\n",
      "--- total time for blast 966: 8.593532085418701\n",
      "-- processing region: 969 with 182 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_969.fasta\n",
      "--- *blasting region: 969\n",
      "--- total time for blast 969: 7.7745232582092285\n",
      "-- processing region: 97 with 147 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_97.fasta\n",
      "--- *blasting region: 97\n",
      "--- total time for blast 97: 16.924112796783447\n",
      "-- processing region: 970 with 306 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_970.fasta\n",
      "--- *blasting region: 970\n",
      "--- total time for blast 970: 9.492056131362915\n",
      "-- processing region: 971 with 262 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_971.fasta\n",
      "--- *blasting region: 971\n",
      "--- total time for blast 971: 10.715452909469604\n",
      "-- processing region: 972 with 235 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_972.fasta\n",
      "--- *blasting region: 972\n",
      "--- total time for blast 972: 13.258067607879639\n",
      "-- processing region: 973 with 121 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_973.fasta\n",
      "--- *blasting region: 973\n",
      "--- total time for blast 973: 6.147374868392944\n",
      "-- processing region: 974 with 283 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_974.fasta\n",
      "--- *blasting region: 974\n",
      "--- total time for blast 974: 13.122426748275757\n",
      "-- processing region: 975 with 276 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_975.fasta\n",
      "--- *blasting region: 975\n",
      "--- total time for blast 975: 11.58277440071106\n",
      "-- processing region: 976 with 188 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_976.fasta\n",
      "--- *blasting region: 976\n",
      "--- total time for blast 976: 7.029419898986816\n",
      "-- processing region: 977 with 211 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_977.fasta\n",
      "--- *blasting region: 977\n",
      "--- total time for blast 977: 6.535887718200684\n",
      "-- processing region: 978 with 190 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_978.fasta\n",
      "--- *blasting region: 978\n",
      "--- total time for blast 978: 8.24744176864624\n",
      "-- processing region: 979 with 227 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_979.fasta\n",
      "--- *blasting region: 979\n",
      "--- total time for blast 979: 11.311447620391846\n",
      "-- processing region: 98 with 249 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_98.fasta\n",
      "--- *blasting region: 98\n",
      "--- total time for blast 98: 11.005082368850708\n",
      "-- processing region: 980 with 240 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_980.fasta\n",
      "--- *blasting region: 980\n",
      "--- total time for blast 980: 9.08109974861145\n",
      "-- processing region: 982 with 251 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_982.fasta\n",
      "--- *blasting region: 982\n",
      "--- total time for blast 982: 11.072519779205322\n",
      "-- processing region: 983 with 232 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_983.fasta\n",
      "--- *blasting region: 983\n",
      "--- total time for blast 983: 9.697543859481812\n",
      "-- processing region: 984 with 233 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_984.fasta\n",
      "--- *blasting region: 984\n",
      "--- total time for blast 984: 8.844383239746094\n",
      "-- processing region: 985 with 241 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_985.fasta\n",
      "--- *blasting region: 985\n",
      "--- total time for blast 985: 7.14833927154541\n",
      "-- processing region: 986 with 269 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_986.fasta\n",
      "--- *blasting region: 986\n",
      "--- total time for blast 986: 10.046466827392578\n",
      "-- processing region: 987 with 237 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_987.fasta\n",
      "--- *blasting region: 987\n",
      "--- total time for blast 987: 7.904274225234985\n",
      "-- processing region: 988 with 139 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_988.fasta\n",
      "--- *blasting region: 988\n",
      "--- total time for blast 988: 7.0801355838775635\n",
      "-- processing region: 989 with 283 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_989.fasta\n",
      "--- *blasting region: 989\n",
      "--- total time for blast 989: 9.043438196182251\n",
      "-- processing region: 99 with 192 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_99.fasta\n",
      "--- *blasting region: 99\n",
      "--- total time for blast 99: 7.029100179672241\n",
      "-- processing region: 990 with 268 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_990.fasta\n",
      "--- *blasting region: 990\n",
      "--- total time for blast 990: 10.541622877120972\n",
      "-- processing region: 991 with 265 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_991.fasta\n",
      "--- *blasting region: 991\n",
      "--- total time for blast 991: 9.710232019424438\n",
      "-- processing region: 992 with 100 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_992.fasta\n",
      "--- *blasting region: 992\n",
      "--- total time for blast 992: 5.07978367805481\n",
      "-- processing region: 995 with 224 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_995.fasta\n",
      "--- *blasting region: 995\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- total time for blast 995: 9.972090005874634\n",
      "-- processing region: 997 with 305 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_997.fasta\n",
      "--- *blasting region: 997\n",
      "--- total time for blast 997: 11.520379066467285\n",
      "-- processing region: 998 with 149 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_998.fasta\n",
      "--- *blasting region: 998\n",
      "--- total time for blast 998: 9.132471323013306\n",
      "-- processing region: 999 with 133 probes\n",
      "--- *writing file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\blast\\probe_gene_999.fasta\n",
      "--- *blasting region: 999\n",
      "--- total time for blast 999: 13.170165538787842\n"
     ]
    }
   ],
   "source": [
    "blast_database = os.path.join(reference_folder, 'blast_database', 'GRCm38_ensembl')\n",
    "\n",
    "check.Blast_probes(kept_records, library_folder, num_threads=32, \n",
    "                   blast_db=blast_database, blast_evalue=100, overwrite=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7c6a9506",
   "metadata": {},
   "source": [
    "### 4.2.1 screen probe by blast results\n",
    "\n",
    "This step has 2 different cutoffs:\n",
    "\n",
    "hard_cutoff: whether a probe has any ~40mer hit somewhere else, if so just dump the probe\n",
    "\n",
    "soft_cutoff: whether a probe has a lot of ~20mer off-target hits, if exceed certain amount then dump the probe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "14cec736",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Number of probes loaded: 210335\n",
      "- Number of regions in this library: 988\n",
      "-- checking probes in region: 1\n",
      "--- gene=1, id=67 removed by soft count = 586\n",
      "--- gene=1, id=78 removed by soft count = 500\n",
      "--- gene=1, id=83 removed by soft count = 107\n",
      "--- gene=1, id=92 removed by soft count = 153\n",
      "--- gene=1, id=93 removed by soft count = 63\n",
      "--- gene=1, id=94 removed by soft count = 159\n",
      "--- gene=1, id=113 removed by soft count = 70\n",
      "--- gene=1, id=122 removed by soft count = 56\n",
      "--- gene=1, id=133 removed by soft count = 105\n",
      "--- gene=1, id=136 removed by soft count = 87\n",
      "--- gene=1, id=137 removed by soft count = 117\n",
      "--- gene=1, id=142 removed by soft count = 86\n",
      "--- gene=1, id=144 removed by soft count = 58\n",
      "--- gene=1, id=146 removed by soft count = 63\n",
      "--- gene=1, id=147 removed by soft count = 61\n",
      "--- gene=1, id=148 removed by soft count = 75\n",
      "--- gene=1, id=154 removed by soft count = 65\n",
      "--- gene=1, id=157 removed by soft count = 77\n",
      "--- gene=1, id=158 removed by soft count = 59\n",
      "--- gene=1, id=160 removed by soft count = 51\n",
      "--- number of probes: 163 , kept by blast: 143 , if remove dups: 143\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 2\n",
      "--- gene=2, id=23 removed by soft count = 201\n",
      "--- gene=2, id=64 removed by soft count = 82\n",
      "--- number of probes: 180 , kept by blast: 178 , if remove dups: 177\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 3\n",
      "--- gene=3, id=52 removed by soft count = 1015\n",
      "--- gene=3, id=53 removed by soft count = 8524\n",
      "--- gene=3, id=54 removed by soft count = 820\n",
      "--- gene=3, id=55 removed by soft count = 200\n",
      "--- gene=3, id=56 removed by soft count = 19119\n",
      "--- gene=3, id=57 removed by soft count = 2471\n",
      "--- gene=3, id=58 removed by soft count = 190\n",
      "--- gene=3, id=69 removed by soft count = 401\n",
      "--- gene=3, id=76 removed by soft count = 74\n",
      "--- number of probes: 77 , kept by blast: 68 , if remove dups: 68\n",
      "-- number of probes kept for this region: 68\n",
      "-- checking probes in region: 4\n",
      "--- gene=4, id=92 removed by soft count = 95\n",
      "--- gene=4, id=93 removed by soft count = 236\n",
      "--- number of probes: 205 , kept by blast: 203 , if remove dups: 203\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 5\n",
      "--- gene=5, id=1 removed by soft count = 62\n",
      "--- gene=5, id=2 removed by soft count = 566\n",
      "--- gene=5, id=48 removed by soft count = 97\n",
      "--- gene=5, id=50 removed by soft count = 101\n",
      "--- gene=5, id=55 removed by soft count = 136\n",
      "--- gene=5, id=64 removed by soft count = 68\n",
      "--- gene=5, id=67 removed by soft count = 374\n",
      "--- number of probes: 144 , kept by blast: 137 , if remove dups: 137\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 6\n",
      "--- gene=6, id=14 removed by soft count = 57\n",
      "--- gene=6, id=192 removed by soft count = 198\n",
      "--- gene=6, id=288 removed by soft count = 125\n",
      "--- number of probes: 297 , kept by blast: 294 , if remove dups: 294\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 7\n",
      "--- gene=7, id=12 removed by soft count = 237\n",
      "--- gene=7, id=16 removed by soft count = 69\n",
      "--- gene=7, id=26 removed by soft count = 153\n",
      "--- gene=7, id=27 removed by soft count = 152\n",
      "--- gene=7, id=81 removed by soft count = 108\n",
      "--- gene=7, id=83 removed by soft count = 51\n",
      "--- gene=7, id=85 removed by soft count = 83\n",
      "--- gene=7, id=89 removed by soft count = 92\n",
      "--- number of probes: 210 , kept by blast: 202 , if remove dups: 202\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 8\n",
      "--- gene=8, id=0 removed by soft count = 98\n",
      "--- gene=8, id=1 removed by soft count = 74\n",
      "--- gene=8, id=86 removed by soft count = 65\n",
      "--- gene=8, id=115 removed by soft count = 81\n",
      "--- gene=8, id=171 removed by soft count = 70\n",
      "--- gene=8, id=230 removed by soft count = 184\n",
      "--- gene=8, id=233 removed by soft count = 94\n",
      "--- gene=8, id=241 removed by soft count = 148\n",
      "--- number of probes: 274 , kept by blast: 266 , if remove dups: 266\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 9\n",
      "--- gene=9, id=3 removed by soft count = 54\n",
      "--- gene=9, id=161 removed by soft count = 59\n",
      "--- gene=9, id=177 removed by soft count = 56\n",
      "--- gene=9, id=195 removed by soft count = 102\n",
      "--- number of probes: 220 , kept by blast: 216 , if remove dups: 216\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 10\n",
      "--- gene=10, id=88 removed by soft count = 131\n",
      "--- gene=10, id=90 removed by soft count = 61\n",
      "--- gene=10, id=93 removed by soft count = 109\n",
      "--- gene=10, id=105 removed by soft count = 823\n",
      "--- gene=10, id=106 removed by soft count = 61\n",
      "--- gene=10, id=108 removed by soft count = 179\n",
      "--- gene=10, id=109 removed by soft count = 173\n",
      "--- gene=10, id=145 removed by soft count = 71\n",
      "--- gene=10, id=146 removed by soft count = 144\n",
      "--- gene=10, id=157 removed by soft count = 503\n",
      "--- number of probes: 158 , kept by blast: 148 , if remove dups: 148\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 11\n",
      "--- gene=11, id=1 removed by soft count = 125\n",
      "--- gene=11, id=2 removed by soft count = 290\n",
      "--- gene=11, id=11 removed by soft count = 51\n",
      "--- gene=11, id=26 removed by soft count = 70\n",
      "--- gene=11, id=56 removed by soft count = 78\n",
      "--- gene=11, id=98 removed by soft count = 60\n",
      "--- gene=11, id=103 removed by soft count = 53\n",
      "--- gene=11, id=105 removed by soft count = 132\n",
      "--- gene=11, id=107 removed by soft count = 3152\n",
      "--- number of probes: 156 , kept by blast: 147 , if remove dups: 147\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 12\n",
      "--- gene=12, id=6 removed by soft count = 70\n",
      "--- gene=12, id=27 removed by soft count = 209\n",
      "--- gene=12, id=28 removed by soft count = 111\n",
      "--- gene=12, id=34 removed by soft count = 421\n",
      "--- gene=12, id=42 removed by soft count = 61\n",
      "--- gene=12, id=47 removed by soft count = 218\n",
      "--- gene=12, id=48 removed by soft count = 211\n",
      "--- gene=12, id=51 removed by soft count = 137\n",
      "--- gene=12, id=53 removed by soft count = 525\n",
      "--- gene=12, id=75 removed by soft count = 168\n",
      "--- gene=12, id=80 removed by soft count = 148\n",
      "--- gene=12, id=81 removed by soft count = 86\n",
      "--- gene=12, id=95 removed by soft count = 59\n",
      "--- gene=12, id=97 removed by soft count = 56\n",
      "--- gene=12, id=101 removed by soft count = 108\n",
      "--- number of probes: 107 , kept by blast: 92 , if remove dups: 92\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 92\n",
      "-- checking probes in region: 13\n",
      "--- gene=13, id=56 removed by soft count = 90\n",
      "--- number of probes: 291 , kept by blast: 290 , if remove dups: 290\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 14\n",
      "--- gene=14, id=16 removed by soft count = 65\n",
      "--- gene=14, id=81 removed by soft count = 374\n",
      "--- gene=14, id=168 removed by soft count = 126\n",
      "--- gene=14, id=205 removed by soft count = 135\n",
      "--- gene=14, id=229 removed by soft count = 60\n",
      "--- gene=14, id=232 removed by soft count = 77\n",
      "--- gene=14, id=233 removed by soft count = 547\n",
      "--- gene=14, id=234 removed by soft count = 228\n",
      "--- number of probes: 240 , kept by blast: 232 , if remove dups: 232\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 15\n",
      "--- gene=15, id=30 removed by soft count = 94\n",
      "--- gene=15, id=194 removed by soft count = 96\n",
      "--- gene=15, id=208 removed by soft count = 60\n",
      "--- number of probes: 279 , kept by blast: 276 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 16\n",
      "--- gene=16, id=100 removed by soft count = 131\n",
      "--- gene=16, id=101 removed by soft count = 142\n",
      "--- gene=16, id=102 removed by soft count = 96\n",
      "--- gene=16, id=105 removed by soft count = 78\n",
      "--- gene=16, id=107 removed by soft count = 189\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=16, id=144 removed by soft count = 75\n",
      "--- gene=16, id=154 removed by soft count = 77\n",
      "--- gene=16, id=183 removed by soft count = 78\n",
      "--- gene=16, id=247 removed by soft count = 165\n",
      "--- number of probes: 248 , kept by blast: 239 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 17\n",
      "--- gene=17, id=56 removed by soft count = 3644\n",
      "--- gene=17, id=57 removed by soft count = 236\n",
      "--- gene=17, id=235 removed by soft count = 54\n",
      "--- number of probes: 276 , kept by blast: 273 , if remove dups: 273\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 18\n",
      "--- gene=18, id=45 removed by soft count = 195\n",
      "--- gene=18, id=50 removed by soft count = 65\n",
      "--- gene=18, id=99 removed by soft count = 1071\n",
      "--- gene=18, id=100 removed by soft count = 254\n",
      "--- gene=18, id=101 removed by soft count = 87\n",
      "--- gene=18, id=102 removed by soft count = 740\n",
      "--- gene=18, id=103 removed by soft count = 226\n",
      "--- gene=18, id=107 removed by soft count = 121\n",
      "--- gene=18, id=108 removed by soft count = 455\n",
      "--- number of probes: 109 , kept by blast: 100 , if remove dups: 99\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 99\n",
      "-- checking probes in region: 19\n",
      "--- gene=19, id=3 removed by soft count = 262\n",
      "--- gene=19, id=4 removed by soft count = 124\n",
      "--- gene=19, id=5 removed by soft count = 512\n",
      "--- gene=19, id=6 removed by soft count = 85\n",
      "--- number of probes: 111 , kept by blast: 107 , if remove dups: 107\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 107\n",
      "-- checking probes in region: 20\n",
      "--- number of probes: 212 , kept by blast: 212 , if remove dups: 212\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 21\n",
      "--- gene=21, id=58 removed by soft count = 1516\n",
      "--- gene=21, id=60 removed by soft count = 71\n",
      "--- gene=21, id=61 removed by soft count = 168\n",
      "--- gene=21, id=62 removed by soft count = 373\n",
      "--- gene=21, id=63 removed by soft count = 200\n",
      "--- gene=21, id=66 removed by soft count = 166\n",
      "--- gene=21, id=68 removed by soft count = 6326\n",
      "--- number of probes: 135 , kept by blast: 128 , if remove dups: 128\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 22\n",
      "--- number of probes: 229 , kept by blast: 229 , if remove dups: 229\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 23\n",
      "--- gene=23, id=199 removed by soft count = 84\n",
      "--- gene=23, id=201 removed by soft count = 78\n",
      "--- gene=23, id=229 removed by soft count = 149\n",
      "--- number of probes: 257 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 24\n",
      "--- gene=24, id=244 removed by soft count = 94\n",
      "--- gene=24, id=245 removed by soft count = 115\n",
      "--- gene=24, id=246 removed by soft count = 564\n",
      "--- number of probes: 257 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 25\n",
      "--- gene=25, id=18 removed by soft count = 65\n",
      "--- gene=25, id=241 removed by soft count = 55\n",
      "--- number of probes: 249 , kept by blast: 247 , if remove dups: 247\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 26\n",
      "--- gene=26, id=6 removed by soft count = 888\n",
      "--- number of probes: 204 , kept by blast: 203 , if remove dups: 203\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 27\n",
      "--- gene=27, id=52 removed by soft count = 475\n",
      "--- gene=27, id=56 removed by soft count = 242\n",
      "--- gene=27, id=112 removed by soft count = 51\n",
      "--- gene=27, id=142 removed by soft count = 83\n",
      "--- gene=27, id=143 removed by soft count = 380\n",
      "--- gene=27, id=174 removed by soft count = 343\n",
      "--- number of probes: 195 , kept by blast: 189 , if remove dups: 189\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 28\n",
      "--- gene=28, id=18 removed by soft count = 296\n",
      "--- gene=28, id=19 removed by soft count = 170\n",
      "--- gene=28, id=137 removed by soft count = 11218\n",
      "--- number of probes: 198 , kept by blast: 195 , if remove dups: 195\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 29\n",
      "--- gene=29, id=11 removed by soft count = 61\n",
      "--- gene=29, id=90 removed by soft count = 55\n",
      "--- number of probes: 284 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 30\n",
      "--- gene=30, id=166 removed by soft count = 54\n",
      "--- gene=30, id=167 removed by soft count = 309\n",
      "--- number of probes: 245 , kept by blast: 243 , if remove dups: 243\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 31\n",
      "--- gene=31, id=102 removed by soft count = 55\n",
      "--- gene=31, id=115 removed by soft count = 169\n",
      "--- gene=31, id=135 removed by soft count = 1874\n",
      "--- gene=31, id=136 removed by soft count = 1524\n",
      "--- gene=31, id=137 removed by soft count = 824\n",
      "--- gene=31, id=138 removed by soft count = 51\n",
      "--- gene=31, id=140 removed by soft count = 99\n",
      "--- number of probes: 151 , kept by blast: 144 , if remove dups: 144\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 32\n",
      "--- gene=32, id=48 removed by soft count = 775\n",
      "--- gene=32, id=103 removed by soft count = 175\n",
      "--- gene=32, id=134 removed by soft count = 182\n",
      "--- gene=32, id=135 removed by soft count = 56\n",
      "--- gene=32, id=137 removed by soft count = 778\n",
      "--- gene=32, id=138 removed by soft count = 504\n",
      "--- gene=32, id=139 removed by soft count = 52\n",
      "--- gene=32, id=140 removed by soft count = 193\n",
      "--- number of probes: 247 , kept by blast: 239 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 33\n",
      "--- gene=33, id=144 removed by soft count = 59\n",
      "--- gene=33, id=261 removed by soft count = 273\n",
      "--- number of probes: 276 , kept by blast: 274 , if remove dups: 274\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 34\n",
      "--- gene=34, id=218 removed by soft count = 450\n",
      "--- number of probes: 294 , kept by blast: 293 , if remove dups: 293\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 35\n",
      "--- gene=35, id=122 removed by soft count = 59\n",
      "--- gene=35, id=164 removed by soft count = 71\n",
      "--- gene=35, id=207 removed by soft count = 68\n",
      "--- gene=35, id=233 removed by soft count = 98\n",
      "--- gene=35, id=234 removed by soft count = 61\n",
      "--- gene=35, id=236 removed by soft count = 58\n",
      "--- gene=35, id=237 removed by soft count = 151\n",
      "--- gene=35, id=238 removed by soft count = 216\n",
      "--- gene=35, id=271 removed by soft count = 158\n",
      "--- gene=35, id=289 removed by soft count = 2269\n",
      "--- number of probes: 310 , kept by blast: 300 , if remove dups: 300\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 36\n",
      "--- gene=36, id=13 removed by soft count = 115\n",
      "--- gene=36, id=14 removed by soft count = 165\n",
      "--- gene=36, id=63 removed by soft count = 76\n",
      "--- gene=36, id=64 removed by soft count = 580\n",
      "--- gene=36, id=81 removed by soft count = 551\n",
      "--- gene=36, id=82 removed by soft count = 2863\n",
      "--- number of probes: 260 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 37\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=37, id=20 removed by soft count = 124\n",
      "--- gene=37, id=228 removed by soft count = 52\n",
      "--- gene=37, id=230 removed by soft count = 203\n",
      "--- gene=37, id=231 removed by soft count = 88\n",
      "--- number of probes: 244 , kept by blast: 240 , if remove dups: 240\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 38\n",
      "--- gene=38, id=13 removed by soft count = 65\n",
      "--- gene=38, id=34 removed by soft count = 78\n",
      "--- gene=38, id=55 removed by soft count = 155\n",
      "--- gene=38, id=56 removed by soft count = 62\n",
      "--- gene=38, id=63 removed by soft count = 54\n",
      "--- gene=38, id=68 removed by soft count = 70\n",
      "--- number of probes: 75 , kept by blast: 69 , if remove dups: 69\n",
      "-- number of probes kept for this region: 69\n",
      "-- checking probes in region: 39\n",
      "--- gene=39, id=62 removed by soft count = 67\n",
      "--- gene=39, id=77 removed by soft count = 76\n",
      "--- gene=39, id=101 removed by soft count = 118\n",
      "--- gene=39, id=123 removed by soft count = 214\n",
      "--- gene=39, id=125 removed by soft count = 218\n",
      "--- gene=39, id=128 removed by soft count = 245\n",
      "--- gene=39, id=132 removed by soft count = 397\n",
      "--- gene=39, id=133 removed by soft count = 65\n",
      "--- gene=39, id=134 removed by soft count = 9383\n",
      "--- gene=39, id=137 removed by soft count = 58\n",
      "--- gene=39, id=138 removed by soft count = 224\n",
      "--- gene=39, id=140 removed by soft count = 96\n",
      "--- number of probes: 178 , kept by blast: 166 , if remove dups: 166\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 40\n",
      "--- gene=40, id=1 removed by soft count = 92\n",
      "--- gene=40, id=2 removed by soft count = 216\n",
      "--- gene=40, id=4 removed by soft count = 161\n",
      "--- gene=40, id=6 removed by soft count = 75\n",
      "--- gene=40, id=7 removed by soft count = 107\n",
      "--- gene=40, id=28 removed by soft count = 172\n",
      "--- number of probes: 142 , kept by blast: 136 , if remove dups: 136\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 41\n",
      "--- gene=41, id=0 removed by soft count = 57\n",
      "--- gene=41, id=1 removed by soft count = 280\n",
      "--- gene=41, id=2 removed by soft count = 154\n",
      "--- gene=41, id=4 removed by soft count = 201\n",
      "--- gene=41, id=5 removed by soft count = 84\n",
      "--- gene=41, id=13 removed by soft count = 166\n",
      "--- gene=41, id=15 removed by soft count = 111\n",
      "--- gene=41, id=16 removed by soft count = 73\n",
      "--- gene=41, id=100 removed by soft count = 84\n",
      "--- number of probes: 122 , kept by blast: 113 , if remove dups: 113\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 113\n",
      "-- checking probes in region: 42\n",
      "--- gene=42, id=43 removed by soft count = 281\n",
      "--- gene=42, id=59 removed by soft count = 59\n",
      "--- gene=42, id=60 removed by soft count = 54\n",
      "--- gene=42, id=61 removed by soft count = 89\n",
      "--- gene=42, id=62 removed by soft count = 65\n",
      "--- gene=42, id=63 removed by soft count = 387\n",
      "--- gene=42, id=64 removed by soft count = 439\n",
      "--- gene=42, id=65 removed by soft count = 2625\n",
      "--- gene=42, id=70 removed by soft count = 340\n",
      "--- gene=42, id=92 removed by soft count = 1158\n",
      "--- gene=42, id=93 removed by soft count = 80\n",
      "--- gene=42, id=94 removed by soft count = 300\n",
      "--- gene=42, id=95 removed by soft count = 8979\n",
      "--- gene=42, id=96 removed by soft count = 52\n",
      "--- gene=42, id=98 removed by soft count = 754\n",
      "--- gene=42, id=99 removed by soft count = 153\n",
      "--- number of probes: 105 , kept by blast: 89 , if remove dups: 89\n",
      "-- number of probes kept for this region: 89\n",
      "-- checking probes in region: 43\n",
      "--- gene=43, id=3 removed by soft count = 582\n",
      "--- gene=43, id=8 removed by soft count = 175\n",
      "--- gene=43, id=89 removed by soft count = 93\n",
      "--- gene=43, id=90 removed by soft count = 95\n",
      "--- gene=43, id=126 removed by soft count = 112\n",
      "--- gene=43, id=169 removed by soft count = 69\n",
      "--- gene=43, id=171 removed by soft count = 183\n",
      "--- number of probes: 173 , kept by blast: 166 , if remove dups: 166\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 44\n",
      "--- gene=44, id=0 removed by soft count = 1283\n",
      "--- gene=44, id=3 removed by soft count = 238\n",
      "--- gene=44, id=17 removed by soft count = 166\n",
      "--- gene=44, id=42 removed by soft count = 513\n",
      "--- gene=44, id=60 removed by soft count = 440\n",
      "--- gene=44, id=63 removed by soft count = 74\n",
      "--- gene=44, id=65 removed by soft count = 230\n",
      "--- gene=44, id=71 removed by soft count = 106\n",
      "--- gene=44, id=72 removed by soft count = 340\n",
      "--- gene=44, id=89 removed by soft count = 791\n",
      "--- number of probes: 90 , kept by blast: 80 , if remove dups: 80\n",
      "-- number of probes kept for this region: 80\n",
      "-- checking probes in region: 45\n",
      "--- gene=45, id=68 removed by soft count = 51\n",
      "--- gene=45, id=69 removed by soft count = 60\n",
      "--- gene=45, id=75 removed by soft count = 69\n",
      "--- gene=45, id=76 removed by soft count = 962\n",
      "--- gene=45, id=77 removed by soft count = 541\n",
      "--- number of probes: 121 , kept by blast: 116 , if remove dups: 116\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 116\n",
      "-- checking probes in region: 46\n",
      "--- gene=46, id=7 removed by soft count = 57\n",
      "--- gene=46, id=26 removed by soft count = 132\n",
      "--- gene=46, id=77 removed by soft count = 59\n",
      "--- gene=46, id=187 removed by soft count = 66\n",
      "--- gene=46, id=204 removed by soft count = 88\n",
      "--- gene=46, id=205 removed by soft count = 91\n",
      "--- number of probes: 241 , kept by blast: 235 , if remove dups: 235\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 47\n",
      "--- gene=47, id=35 removed by soft count = 103\n",
      "--- gene=47, id=36 removed by soft count = 146\n",
      "--- gene=47, id=160 removed by soft count = 84\n",
      "--- number of probes: 240 , kept by blast: 237 , if remove dups: 237\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 48\n",
      "--- gene=48, id=108 removed by soft count = 458\n",
      "--- gene=48, id=185 removed by soft count = 122\n",
      "--- gene=48, id=196 removed by soft count = 1891\n",
      "--- gene=48, id=197 removed by soft count = 105\n",
      "--- gene=48, id=203 removed by soft count = 23196\n",
      "--- gene=48, id=204 removed by soft count = 100\n",
      "--- gene=48, id=205 removed by soft count = 2084\n",
      "--- number of probes: 218 , kept by blast: 211 , if remove dups: 211\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 49\n",
      "--- gene=49, id=57 removed by soft count = 129\n",
      "--- gene=49, id=81 removed by soft count = 92\n",
      "--- gene=49, id=83 removed by soft count = 131\n",
      "--- number of probes: 225 , kept by blast: 222 , if remove dups: 222\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 50\n",
      "--- gene=50, id=227 removed by soft count = 159\n",
      "--- number of probes: 287 , kept by blast: 286 , if remove dups: 286\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 51\n",
      "--- gene=51, id=15 removed by soft count = 79\n",
      "--- gene=51, id=16 removed by soft count = 137\n",
      "--- gene=51, id=17 removed by soft count = 88\n",
      "--- gene=51, id=87 removed by soft count = 125\n",
      "--- gene=51, id=89 removed by soft count = 61\n",
      "--- gene=51, id=90 removed by soft count = 870\n",
      "--- gene=51, id=96 removed by soft count = 165\n",
      "--- gene=51, id=133 removed by soft count = 74\n",
      "--- number of probes: 250 , kept by blast: 242 , if remove dups: 242\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 52\n",
      "--- gene=52, id=162 removed by soft count = 446\n",
      "--- gene=52, id=261 removed by soft count = 485\n",
      "--- number of probes: 329 , kept by blast: 327 , if remove dups: 327\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 53\n",
      "--- gene=53, id=72 removed by soft count = 52\n",
      "--- number of probes: 299 , kept by blast: 298 , if remove dups: 298\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 54\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=54, id=1 removed by soft count = 154\n",
      "--- gene=54, id=3 removed by soft count = 382\n",
      "--- gene=54, id=58 removed by soft count = 84\n",
      "--- gene=54, id=60 removed by soft count = 128\n",
      "--- gene=54, id=74 removed by soft count = 181\n",
      "--- gene=54, id=127 removed by soft count = 506\n",
      "--- gene=54, id=151 removed by soft count = 58\n",
      "--- gene=54, id=171 removed by soft count = 352\n",
      "--- gene=54, id=180 removed by soft count = 63\n",
      "--- gene=54, id=209 removed by soft count = 1405\n",
      "--- number of probes: 257 , kept by blast: 247 , if remove dups: 247\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 55\n",
      "--- gene=55, id=127 removed by soft count = 168\n",
      "--- number of probes: 206 , kept by blast: 205 , if remove dups: 205\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 56\n",
      "--- gene=56, id=12 removed by soft count = 81\n",
      "--- gene=56, id=13 removed by soft count = 1333\n",
      "--- gene=56, id=32 removed by soft count = 91\n",
      "--- gene=56, id=37 removed by soft count = 741\n",
      "--- gene=56, id=46 removed by soft count = 554\n",
      "--- gene=56, id=47 removed by soft count = 190\n",
      "--- gene=56, id=49 removed by soft count = 62\n",
      "--- gene=56, id=59 removed by soft count = 890\n",
      "--- gene=56, id=60 removed by soft count = 237\n",
      "--- gene=56, id=61 removed by soft count = 118\n",
      "--- gene=56, id=64 removed by soft count = 258\n",
      "--- gene=56, id=65 removed by soft count = 56\n",
      "--- gene=56, id=66 removed by soft count = 11458\n",
      "--- gene=56, id=94 removed by soft count = 363\n",
      "--- gene=56, id=124 removed by soft count = 240\n",
      "--- number of probes: 125 , kept by blast: 110 , if remove dups: 110\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 110\n",
      "-- checking probes in region: 57\n",
      "--- number of probes: 232 , kept by blast: 232 , if remove dups: 232\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 58\n",
      "--- gene=58, id=0 removed by soft count = 21591\n",
      "--- gene=58, id=37 removed by soft count = 121\n",
      "--- gene=58, id=72 removed by soft count = 271\n",
      "--- number of probes: 112 , kept by blast: 109 , if remove dups: 109\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 109\n",
      "-- checking probes in region: 59\n",
      "--- gene=59, id=58 removed by soft count = 6386\n",
      "--- gene=59, id=59 removed by soft count = 81\n",
      "--- gene=59, id=60 removed by soft count = 3647\n",
      "--- gene=59, id=62 removed by soft count = 97\n",
      "--- gene=59, id=65 removed by soft count = 348\n",
      "--- gene=59, id=98 removed by soft count = 118\n",
      "--- gene=59, id=109 removed by soft count = 242\n",
      "--- number of probes: 159 , kept by blast: 152 , if remove dups: 152\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 60\n",
      "--- gene=60, id=3 removed by soft count = 101\n",
      "--- gene=60, id=88 removed by soft count = 59\n",
      "--- gene=60, id=251 removed by soft count = 54\n",
      "--- number of probes: 252 , kept by blast: 249 , if remove dups: 249\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 61\n",
      "--- gene=61, id=152 removed by soft count = 73\n",
      "--- number of probes: 281 , kept by blast: 280 , if remove dups: 280\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 62\n",
      "--- gene=62, id=84 removed by soft count = 97\n",
      "--- gene=62, id=188 removed by soft count = 115\n",
      "--- gene=62, id=189 removed by soft count = 68\n",
      "--- gene=62, id=191 removed by soft count = 66\n",
      "--- gene=62, id=247 removed by soft count = 54\n",
      "--- gene=62, id=249 removed by soft count = 99\n",
      "--- number of probes: 265 , kept by blast: 259 , if remove dups: 259\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 63\n",
      "--- gene=63, id=10 removed by soft count = 725\n",
      "--- gene=63, id=80 removed by soft count = 52\n",
      "--- number of probes: 245 , kept by blast: 243 , if remove dups: 243\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 64\n",
      "--- gene=64, id=217 removed by soft count = 151\n",
      "--- gene=64, id=239 removed by soft count = 192\n",
      "--- number of probes: 303 , kept by blast: 301 , if remove dups: 300\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 65\n",
      "--- gene=65, id=32 removed by soft count = 82\n",
      "--- gene=65, id=36 removed by soft count = 72\n",
      "--- gene=65, id=139 removed by soft count = 215\n",
      "--- number of probes: 231 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 66\n",
      "--- number of probes: 310 , kept by blast: 310 , if remove dups: 310\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 67\n",
      "--- gene=67, id=84 removed by soft count = 672\n",
      "--- gene=67, id=86 removed by soft count = 67\n",
      "--- gene=67, id=89 removed by soft count = 528\n",
      "--- number of probes: 197 , kept by blast: 194 , if remove dups: 194\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 68\n",
      "--- gene=68, id=28 removed by soft count = 150\n",
      "--- gene=68, id=29 removed by soft count = 132\n",
      "--- gene=68, id=30 removed by soft count = 390\n",
      "--- gene=68, id=45 removed by soft count = 59\n",
      "--- gene=68, id=135 removed by soft count = 77\n",
      "--- gene=68, id=212 removed by soft count = 177\n",
      "--- gene=68, id=214 removed by soft count = 199\n",
      "--- number of probes: 283 , kept by blast: 276 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 69\n",
      "--- gene=69, id=35 removed by soft count = 1093\n",
      "--- gene=69, id=37 removed by soft count = 6648\n",
      "--- gene=69, id=38 removed by soft count = 813\n",
      "--- gene=69, id=57 removed by soft count = 142\n",
      "--- gene=69, id=67 removed by soft count = 66\n",
      "--- number of probes: 113 , kept by blast: 108 , if remove dups: 108\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 108\n",
      "-- checking probes in region: 70\n",
      "--- number of probes: 209 , kept by blast: 209 , if remove dups: 209\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 71\n",
      "--- gene=71, id=200 removed by soft count = 98\n",
      "--- gene=71, id=228 removed by soft count = 54\n",
      "--- gene=71, id=229 removed by soft count = 96\n",
      "--- number of probes: 274 , kept by blast: 271 , if remove dups: 271\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 72\n",
      "--- gene=72, id=0 removed by soft count = 69\n",
      "--- gene=72, id=1 removed by soft count = 530\n",
      "--- gene=72, id=2 removed by soft count = 125\n",
      "--- gene=72, id=125 removed by soft count = 208\n",
      "--- gene=72, id=126 removed by soft count = 73\n",
      "--- number of probes: 251 , kept by blast: 246 , if remove dups: 246\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 73\n",
      "--- number of probes: 282 , kept by blast: 282 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 74\n",
      "--- gene=74, id=51 removed by soft count = 85\n",
      "--- gene=74, id=55 removed by soft count = 274\n",
      "--- gene=74, id=226 removed by soft count = 68\n",
      "--- number of probes: 276 , kept by blast: 273 , if remove dups: 273\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 75\n",
      "--- number of probes: 278 , kept by blast: 278 , if remove dups: 278\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 76\n",
      "--- gene=76, id=95 removed by soft count = 399\n",
      "--- gene=76, id=97 removed by soft count = 315\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=76, id=261 removed by soft count = 261\n",
      "--- number of probes: 265 , kept by blast: 262 , if remove dups: 262\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 77\n",
      "--- gene=77, id=18 removed by soft count = 81\n",
      "--- gene=77, id=180 removed by soft count = 145\n",
      "--- gene=77, id=190 removed by soft count = 55\n",
      "--- number of probes: 193 , kept by blast: 190 , if remove dups: 190\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 80\n",
      "--- number of probes: 287 , kept by blast: 287 , if remove dups: 287\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 81\n",
      "--- gene=81, id=2 removed by soft count = 99\n",
      "--- gene=81, id=42 removed by soft count = 212\n",
      "--- gene=81, id=43 removed by soft count = 106\n",
      "--- gene=81, id=44 removed by soft count = 124\n",
      "--- gene=81, id=147 removed by soft count = 76\n",
      "--- gene=81, id=174 removed by soft count = 81\n",
      "--- gene=81, id=232 removed by soft count = 413\n",
      "--- number of probes: 240 , kept by blast: 233 , if remove dups: 233\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 82\n",
      "--- gene=82, id=79 removed by soft count = 51\n",
      "--- number of probes: 223 , kept by blast: 222 , if remove dups: 222\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 83\n",
      "--- gene=83, id=3 removed by soft count = 102\n",
      "--- gene=83, id=6 removed by soft count = 147\n",
      "--- number of probes: 233 , kept by blast: 231 , if remove dups: 231\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 84\n",
      "--- gene=84, id=50 removed by soft count = 722\n",
      "--- gene=84, id=101 removed by soft count = 282\n",
      "--- gene=84, id=102 removed by soft count = 52\n",
      "--- gene=84, id=177 removed by soft count = 56\n",
      "--- gene=84, id=210 removed by soft count = 94\n",
      "--- number of probes: 290 , kept by blast: 285 , if remove dups: 285\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 85\n",
      "--- gene=85, id=12 removed by soft count = 136\n",
      "--- gene=85, id=156 removed by soft count = 212\n",
      "--- gene=85, id=172 removed by soft count = 1574\n",
      "--- number of probes: 173 , kept by blast: 170 , if remove dups: 169\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 86\n",
      "--- gene=86, id=90 removed by soft count = 111\n",
      "--- number of probes: 202 , kept by blast: 201 , if remove dups: 201\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 87\n",
      "--- gene=87, id=19 removed by soft count = 60\n",
      "--- gene=87, id=38 removed by soft count = 280\n",
      "--- number of probes: 154 , kept by blast: 152 , if remove dups: 152\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 88\n",
      "--- gene=88, id=2 removed by soft count = 67\n",
      "--- gene=88, id=6 removed by soft count = 244\n",
      "--- gene=88, id=35 removed by soft count = 1841\n",
      "--- gene=88, id=36 removed by soft count = 62\n",
      "--- gene=88, id=60 removed by soft count = 177\n",
      "--- gene=88, id=66 removed by soft count = 176\n",
      "--- gene=88, id=69 removed by soft count = 198\n",
      "--- gene=88, id=70 removed by soft count = 96\n",
      "--- gene=88, id=77 removed by soft count = 72\n",
      "--- gene=88, id=78 removed by soft count = 246\n",
      "--- gene=88, id=81 removed by soft count = 2249\n",
      "--- gene=88, id=86 removed by soft count = 93\n",
      "--- gene=88, id=90 removed by soft count = 56\n",
      "--- number of probes: 117 , kept by blast: 104 , if remove dups: 104\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 104\n",
      "-- checking probes in region: 89\n",
      "--- gene=89, id=28 removed by soft count = 222\n",
      "--- gene=89, id=43 removed by soft count = 152\n",
      "--- gene=89, id=45 removed by soft count = 613\n",
      "--- gene=89, id=46 removed by soft count = 402\n",
      "--- number of probes: 318 , kept by blast: 314 , if remove dups: 314\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 90\n",
      "--- gene=90, id=302 removed by soft count = 98\n",
      "--- number of probes: 315 , kept by blast: 314 , if remove dups: 314\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 91\n",
      "--- number of probes: 306 , kept by blast: 306 , if remove dups: 306\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 92\n",
      "--- gene=92, id=42 removed by soft count = 684\n",
      "--- number of probes: 335 , kept by blast: 334 , if remove dups: 334\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 93\n",
      "--- gene=93, id=25 removed by soft count = 61\n",
      "--- gene=93, id=26 removed by soft count = 125\n",
      "--- gene=93, id=176 removed by soft count = 58\n",
      "--- number of probes: 284 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 94\n",
      "--- gene=94, id=27 removed by soft count = 71\n",
      "--- gene=94, id=202 removed by soft count = 153\n",
      "--- number of probes: 207 , kept by blast: 205 , if remove dups: 205\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 95\n",
      "--- gene=95, id=0 removed by soft count = 220\n",
      "--- gene=95, id=109 removed by soft count = 103\n",
      "--- gene=95, id=113 removed by soft count = 238\n",
      "--- gene=95, id=147 removed by soft count = 141\n",
      "--- gene=95, id=149 removed by soft count = 388\n",
      "--- number of probes: 185 , kept by blast: 180 , if remove dups: 180\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 96\n",
      "--- gene=96, id=10 removed by soft count = 51\n",
      "--- gene=96, id=54 removed by soft count = 2032\n",
      "--- gene=96, id=63 removed by soft count = 58\n",
      "--- gene=96, id=64 removed by soft count = 92\n",
      "--- number of probes: 207 , kept by blast: 203 , if remove dups: 203\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 97\n",
      "--- gene=97, id=91 removed by soft count = 74\n",
      "--- gene=97, id=99 removed by soft count = 186\n",
      "--- gene=97, id=113 removed by soft count = 242\n",
      "--- gene=97, id=115 removed by soft count = 3130\n",
      "--- gene=97, id=116 removed by soft count = 1156\n",
      "--- gene=97, id=117 removed by soft count = 61\n",
      "--- gene=97, id=119 removed by soft count = 27642\n",
      "--- gene=97, id=120 removed by soft count = 1136\n",
      "--- gene=97, id=121 removed by soft count = 54\n",
      "--- gene=97, id=123 removed by soft count = 147\n",
      "--- gene=97, id=124 removed by soft count = 79\n",
      "--- gene=97, id=131 removed by soft count = 99\n",
      "--- gene=97, id=132 removed by soft count = 59\n",
      "--- gene=97, id=135 removed by soft count = 130\n",
      "--- gene=97, id=136 removed by soft count = 306\n",
      "--- gene=97, id=138 removed by soft count = 173\n",
      "--- number of probes: 147 , kept by blast: 131 , if remove dups: 131\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 98\n",
      "--- gene=98, id=74 removed by soft count = 710\n",
      "--- number of probes: 249 , kept by blast: 248 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 99\n",
      "--- gene=99, id=164 removed by soft count = 129\n",
      "--- gene=99, id=190 removed by soft count = 138\n",
      "--- gene=99, id=191 removed by soft count = 64\n",
      "--- number of probes: 192 , kept by blast: 189 , if remove dups: 189\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 100\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=100, id=2 removed by soft count = 336\n",
      "--- gene=100, id=3 removed by soft count = 72\n",
      "--- gene=100, id=15 removed by soft count = 82\n",
      "--- gene=100, id=26 removed by soft count = 83\n",
      "--- gene=100, id=28 removed by soft count = 108\n",
      "--- gene=100, id=71 removed by soft count = 106\n",
      "--- number of probes: 82 , kept by blast: 76 , if remove dups: 76\n",
      "-- number of probes kept for this region: 76\n",
      "-- checking probes in region: 101\n",
      "--- gene=101, id=9 removed by soft count = 77\n",
      "--- gene=101, id=14 removed by soft count = 468\n",
      "--- gene=101, id=16 removed by soft count = 1158\n",
      "--- gene=101, id=17 removed by soft count = 153\n",
      "--- gene=101, id=22 removed by soft count = 201\n",
      "--- number of probes: 122 , kept by blast: 117 , if remove dups: 117\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 117\n",
      "-- checking probes in region: 102\n",
      "--- gene=102, id=11 removed by soft count = 89\n",
      "--- gene=102, id=12 removed by soft count = 77\n",
      "--- gene=102, id=13 removed by soft count = 90\n",
      "--- gene=102, id=14 removed by soft count = 95\n",
      "--- gene=102, id=148 removed by soft count = 118\n",
      "--- number of probes: 267 , kept by blast: 262 , if remove dups: 262\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 103\n",
      "--- gene=103, id=111 removed by soft count = 220\n",
      "--- gene=103, id=180 removed by soft count = 82\n",
      "--- gene=103, id=181 removed by soft count = 191\n",
      "--- gene=103, id=183 removed by soft count = 60\n",
      "--- number of probes: 220 , kept by blast: 216 , if remove dups: 216\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 104\n",
      "--- gene=104, id=6 removed by soft count = 246\n",
      "--- gene=104, id=70 removed by soft count = 1385\n",
      "--- gene=104, id=71 removed by hard count = 0\n",
      "--- gene=104, id=72 removed by soft count = 92\n",
      "--- gene=104, id=97 removed by soft count = 105\n",
      "--- number of probes: 201 , kept by blast: 196 , if remove dups: 196\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 105\n",
      "--- gene=105, id=164 removed by soft count = 152\n",
      "--- gene=105, id=169 removed by soft count = 73\n",
      "--- number of probes: 271 , kept by blast: 269 , if remove dups: 269\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 106\n",
      "--- gene=106, id=9 removed by soft count = 233\n",
      "--- gene=106, id=120 removed by soft count = 80\n",
      "--- gene=106, id=121 removed by soft count = 53\n",
      "--- gene=106, id=207 removed by soft count = 292\n",
      "--- gene=106, id=211 removed by soft count = 351\n",
      "--- gene=106, id=212 removed by soft count = 10201\n",
      "--- number of probes: 213 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 107\n",
      "--- gene=107, id=43 removed by soft count = 110\n",
      "--- gene=107, id=162 removed by soft count = 140\n",
      "--- number of probes: 246 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 108\n",
      "--- gene=108, id=119 removed by soft count = 56\n",
      "--- gene=108, id=120 removed by soft count = 109\n",
      "--- gene=108, id=121 removed by soft count = 103\n",
      "--- gene=108, id=122 removed by soft count = 144\n",
      "--- gene=108, id=142 removed by soft count = 72\n",
      "--- gene=108, id=151 removed by soft count = 66\n",
      "--- number of probes: 152 , kept by blast: 146 , if remove dups: 146\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 109\n",
      "--- gene=109, id=27 removed by soft count = 56\n",
      "--- gene=109, id=36 removed by soft count = 53\n",
      "--- number of probes: 263 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 110\n",
      "--- gene=110, id=5 removed by soft count = 52\n",
      "--- gene=110, id=14 removed by soft count = 92\n",
      "--- gene=110, id=36 removed by soft count = 1155\n",
      "--- gene=110, id=243 removed by soft count = 57\n",
      "--- number of probes: 248 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 111\n",
      "--- gene=111, id=95 removed by soft count = 54\n",
      "--- number of probes: 97 , kept by blast: 96 , if remove dups: 96\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 96\n",
      "-- checking probes in region: 112\n",
      "--- gene=112, id=70 removed by soft count = 67\n",
      "--- gene=112, id=100 removed by soft count = 113\n",
      "--- gene=112, id=101 removed by soft count = 129\n",
      "--- gene=112, id=114 removed by soft count = 385\n",
      "--- gene=112, id=116 removed by soft count = 66\n",
      "--- gene=112, id=132 removed by soft count = 99\n",
      "--- number of probes: 205 , kept by blast: 199 , if remove dups: 199\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 113\n",
      "--- gene=113, id=29 removed by soft count = 386\n",
      "--- gene=113, id=33 removed by soft count = 85\n",
      "--- gene=113, id=38 removed by soft count = 53\n",
      "--- gene=113, id=42 removed by soft count = 180\n",
      "--- gene=113, id=48 removed by soft count = 315\n",
      "--- gene=113, id=49 removed by soft count = 100\n",
      "--- gene=113, id=51 removed by soft count = 855\n",
      "--- gene=113, id=52 removed by soft count = 161\n",
      "--- gene=113, id=53 removed by soft count = 224\n",
      "--- gene=113, id=54 removed by soft count = 249\n",
      "--- number of probes: 91 , kept by blast: 81 , if remove dups: 81\n",
      "-- number of probes kept for this region: 81\n",
      "-- checking probes in region: 114\n",
      "--- gene=114, id=11 removed by soft count = 52\n",
      "--- gene=114, id=27 removed by soft count = 245\n",
      "--- gene=114, id=29 removed by soft count = 63\n",
      "--- gene=114, id=33 removed by soft count = 1032\n",
      "--- gene=114, id=35 removed by soft count = 143\n",
      "--- gene=114, id=36 removed by soft count = 3132\n",
      "--- gene=114, id=49 removed by soft count = 102\n",
      "--- number of probes: 163 , kept by blast: 156 , if remove dups: 155\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 115\n",
      "--- number of probes: 242 , kept by blast: 242 , if remove dups: 242\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 116\n",
      "--- gene=116, id=147 removed by soft count = 78\n",
      "--- gene=116, id=148 removed by soft count = 169\n",
      "--- gene=116, id=149 removed by soft count = 161\n",
      "--- gene=116, id=195 removed by soft count = 62\n",
      "--- number of probes: 296 , kept by blast: 292 , if remove dups: 292\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 117\n",
      "--- gene=117, id=101 removed by soft count = 59\n",
      "--- gene=117, id=166 removed by soft count = 315\n",
      "--- gene=117, id=167 removed by soft count = 255\n",
      "--- gene=117, id=168 removed by soft count = 104\n",
      "--- gene=117, id=169 removed by soft count = 311\n",
      "--- gene=117, id=171 removed by soft count = 988\n",
      "--- gene=117, id=172 removed by soft count = 75\n",
      "--- number of probes: 175 , kept by blast: 168 , if remove dups: 168\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 118\n",
      "--- gene=118, id=95 removed by soft count = 545\n",
      "--- gene=118, id=96 removed by soft count = 570\n",
      "--- gene=118, id=97 removed by soft count = 372\n",
      "--- gene=118, id=100 removed by soft count = 129\n",
      "--- gene=118, id=103 removed by soft count = 414\n",
      "--- gene=118, id=105 removed by soft count = 249\n",
      "--- gene=118, id=109 removed by soft count = 1122\n",
      "--- gene=118, id=110 removed by soft count = 9969\n",
      "--- number of probes: 111 , kept by blast: 103 , if remove dups: 103\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 103\n",
      "-- checking probes in region: 119\n",
      "--- gene=119, id=1 removed by soft count = 130\n",
      "--- gene=119, id=2 removed by soft count = 61\n",
      "--- gene=119, id=3 removed by soft count = 75\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=119, id=145 removed by soft count = 54\n",
      "--- gene=119, id=202 removed by soft count = 94\n",
      "--- number of probes: 203 , kept by blast: 198 , if remove dups: 198\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 120\n",
      "--- gene=120, id=2 removed by soft count = 107\n",
      "--- gene=120, id=15 removed by soft count = 52\n",
      "--- gene=120, id=83 removed by soft count = 198\n",
      "--- gene=120, id=97 removed by soft count = 270\n",
      "--- gene=120, id=129 removed by soft count = 273\n",
      "--- number of probes: 279 , kept by blast: 274 , if remove dups: 274\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 121\n",
      "--- number of probes: 223 , kept by blast: 223 , if remove dups: 223\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 122\n",
      "--- gene=122, id=83 removed by soft count = 99\n",
      "--- gene=122, id=104 removed by soft count = 81\n",
      "--- gene=122, id=118 removed by soft count = 108\n",
      "--- gene=122, id=119 removed by soft count = 121\n",
      "--- gene=122, id=186 removed by soft count = 97\n",
      "--- gene=122, id=188 removed by soft count = 53\n",
      "--- gene=122, id=193 removed by soft count = 143\n",
      "--- gene=122, id=194 removed by soft count = 134\n",
      "--- number of probes: 231 , kept by blast: 223 , if remove dups: 223\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 123\n",
      "--- gene=123, id=1 removed by soft count = 339\n",
      "--- gene=123, id=2 removed by soft count = 473\n",
      "--- gene=123, id=3 removed by soft count = 1901\n",
      "--- gene=123, id=4 removed by soft count = 890\n",
      "--- gene=123, id=6 removed by soft count = 3891\n",
      "--- gene=123, id=7 removed by soft count = 55\n",
      "--- gene=123, id=8 removed by soft count = 672\n",
      "--- gene=123, id=10 removed by soft count = 125\n",
      "--- gene=123, id=11 removed by soft count = 132\n",
      "--- gene=123, id=12 removed by soft count = 465\n",
      "--- gene=123, id=14 removed by soft count = 68\n",
      "--- gene=123, id=15 removed by soft count = 55\n",
      "--- gene=123, id=16 removed by soft count = 83\n",
      "--- gene=123, id=19 removed by soft count = 1772\n",
      "--- number of probes: 102 , kept by blast: 88 , if remove dups: 88\n",
      "-- number of probes kept for this region: 88\n",
      "-- checking probes in region: 124\n",
      "--- number of probes: 255 , kept by blast: 255 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 125\n",
      "--- gene=125, id=9 removed by soft count = 382\n",
      "--- gene=125, id=10 removed by soft count = 249\n",
      "--- gene=125, id=11 removed by soft count = 933\n",
      "--- gene=125, id=12 removed by soft count = 1767\n",
      "--- number of probes: 191 , kept by blast: 187 , if remove dups: 187\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 126\n",
      "--- gene=126, id=21 removed by soft count = 102\n",
      "--- number of probes: 318 , kept by blast: 317 , if remove dups: 317\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 127\n",
      "--- gene=127, id=96 removed by soft count = 92\n",
      "--- gene=127, id=170 removed by soft count = 117\n",
      "--- gene=127, id=175 removed by soft count = 53\n",
      "--- gene=127, id=191 removed by soft count = 64\n",
      "--- gene=127, id=203 removed by soft count = 767\n",
      "--- number of probes: 224 , kept by blast: 219 , if remove dups: 219\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 128\n",
      "--- gene=128, id=96 removed by soft count = 697\n",
      "--- gene=128, id=127 removed by soft count = 59\n",
      "--- number of probes: 167 , kept by blast: 165 , if remove dups: 165\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 129\n",
      "--- gene=129, id=27 removed by soft count = 70\n",
      "--- gene=129, id=44 removed by soft count = 110\n",
      "--- gene=129, id=74 removed by soft count = 97\n",
      "--- gene=129, id=80 removed by soft count = 52\n",
      "--- gene=129, id=83 removed by soft count = 80\n",
      "--- gene=129, id=86 removed by soft count = 86\n",
      "--- gene=129, id=124 removed by soft count = 3180\n",
      "--- number of probes: 147 , kept by blast: 140 , if remove dups: 140\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 130\n",
      "--- gene=130, id=235 removed by soft count = 108\n",
      "--- gene=130, id=237 removed by soft count = 172\n",
      "--- number of probes: 274 , kept by blast: 272 , if remove dups: 272\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 131\n",
      "--- gene=131, id=191 removed by soft count = 97\n",
      "--- gene=131, id=225 removed by soft count = 62\n",
      "--- number of probes: 261 , kept by blast: 259 , if remove dups: 259\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 132\n",
      "--- gene=132, id=80 removed by soft count = 200\n",
      "--- gene=132, id=243 removed by soft count = 59\n",
      "--- number of probes: 248 , kept by blast: 246 , if remove dups: 246\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 133\n",
      "--- gene=133, id=8 removed by soft count = 127\n",
      "--- gene=133, id=23 removed by soft count = 203\n",
      "--- gene=133, id=24 removed by soft count = 2196\n",
      "--- gene=133, id=149 removed by soft count = 559\n",
      "--- number of probes: 159 , kept by blast: 155 , if remove dups: 155\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 134\n",
      "--- gene=134, id=36 removed by soft count = 56\n",
      "--- gene=134, id=39 removed by soft count = 114\n",
      "--- gene=134, id=40 removed by soft count = 105\n",
      "--- gene=134, id=49 removed by soft count = 55\n",
      "--- gene=134, id=52 removed by soft count = 125\n",
      "--- gene=134, id=54 removed by soft count = 94\n",
      "--- gene=134, id=55 removed by soft count = 69\n",
      "--- gene=134, id=56 removed by soft count = 437\n",
      "--- gene=134, id=63 removed by soft count = 148\n",
      "--- gene=134, id=77 removed by soft count = 71\n",
      "--- gene=134, id=81 removed by soft count = 181\n",
      "--- gene=134, id=102 removed by soft count = 149\n",
      "--- gene=134, id=106 removed by soft count = 82\n",
      "--- gene=134, id=135 removed by soft count = 312\n",
      "--- gene=134, id=137 removed by soft count = 705\n",
      "--- number of probes: 140 , kept by blast: 125 , if remove dups: 125\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 135\n",
      "--- gene=135, id=108 removed by soft count = 4584\n",
      "--- gene=135, id=231 removed by soft count = 803\n",
      "--- number of probes: 232 , kept by blast: 230 , if remove dups: 230\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 136\n",
      "--- gene=136, id=141 removed by soft count = 105\n",
      "--- number of probes: 235 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 137\n",
      "--- number of probes: 313 , kept by blast: 313 , if remove dups: 313\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 138\n",
      "--- gene=138, id=6 removed by soft count = 65\n",
      "--- gene=138, id=8 removed by soft count = 55\n",
      "--- gene=138, id=184 removed by hard count = 0\n",
      "--- gene=138, id=197 removed by soft count = 63\n",
      "--- number of probes: 199 , kept by blast: 195 , if remove dups: 195\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 140\n",
      "--- gene=140, id=195 removed by soft count = 554\n",
      "--- number of probes: 226 , kept by blast: 225 , if remove dups: 225\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 141\n",
      "--- gene=141, id=28 removed by soft count = 293\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=141, id=61 removed by soft count = 89\n",
      "--- gene=141, id=103 removed by soft count = 60\n",
      "--- gene=141, id=122 removed by soft count = 112\n",
      "--- gene=141, id=246 removed by soft count = 84\n",
      "--- gene=141, id=257 removed by soft count = 91\n",
      "--- number of probes: 259 , kept by blast: 253 , if remove dups: 253\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 142\n",
      "--- gene=142, id=123 removed by soft count = 104\n",
      "--- gene=142, id=188 removed by soft count = 225\n",
      "--- gene=142, id=191 removed by soft count = 87\n",
      "--- gene=142, id=214 removed by soft count = 66\n",
      "--- gene=142, id=299 removed by soft count = 168\n",
      "--- number of probes: 308 , kept by blast: 303 , if remove dups: 301\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 143\n",
      "--- gene=143, id=3 removed by soft count = 63\n",
      "--- gene=143, id=138 removed by soft count = 125\n",
      "--- gene=143, id=157 removed by soft count = 507\n",
      "--- gene=143, id=162 removed by soft count = 133\n",
      "--- number of probes: 248 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 144\n",
      "--- gene=144, id=208 removed by soft count = 74\n",
      "--- gene=144, id=263 removed by soft count = 54\n",
      "--- gene=144, id=301 removed by soft count = 55\n",
      "--- number of probes: 305 , kept by blast: 302 , if remove dups: 302\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 145\n",
      "--- gene=145, id=64 removed by soft count = 81\n",
      "--- gene=145, id=218 removed by soft count = 524\n",
      "--- number of probes: 302 , kept by blast: 300 , if remove dups: 300\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 146\n",
      "--- gene=146, id=72 removed by soft count = 3346\n",
      "--- gene=146, id=158 removed by soft count = 54\n",
      "--- number of probes: 279 , kept by blast: 277 , if remove dups: 277\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 147\n",
      "--- gene=147, id=13 removed by soft count = 197\n",
      "--- gene=147, id=14 removed by soft count = 146\n",
      "--- gene=147, id=31 removed by soft count = 139\n",
      "--- gene=147, id=43 removed by soft count = 140\n",
      "--- number of probes: 240 , kept by blast: 236 , if remove dups: 235\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 148\n",
      "--- gene=148, id=92 removed by hard count = 4\n",
      "--- gene=148, id=178 removed by soft count = 92\n",
      "--- gene=148, id=179 removed by soft count = 92\n",
      "--- gene=148, id=180 removed by soft count = 86\n",
      "--- gene=148, id=181 removed by soft count = 94\n",
      "--- gene=148, id=182 removed by soft count = 91\n",
      "--- gene=148, id=183 removed by soft count = 79\n",
      "--- gene=148, id=184 removed by hard count = 8\n",
      "--- gene=148, id=185 removed by soft count = 95\n",
      "--- gene=148, id=186 removed by soft count = 81\n",
      "--- gene=148, id=187 removed by soft count = 83\n",
      "--- gene=148, id=188 removed by soft count = 84\n",
      "--- number of probes: 260 , kept by blast: 248 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 150\n",
      "--- gene=150, id=85 removed by soft count = 1329\n",
      "--- gene=150, id=98 removed by soft count = 128\n",
      "--- number of probes: 254 , kept by blast: 252 , if remove dups: 252\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 151\n",
      "--- gene=151, id=49 removed by soft count = 79\n",
      "--- gene=151, id=50 removed by soft count = 90\n",
      "--- gene=151, id=54 removed by soft count = 87\n",
      "--- gene=151, id=55 removed by soft count = 74\n",
      "--- gene=151, id=107 removed by soft count = 91\n",
      "--- gene=151, id=108 removed by soft count = 161\n",
      "--- gene=151, id=117 removed by soft count = 72\n",
      "--- number of probes: 285 , kept by blast: 278 , if remove dups: 278\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 153\n",
      "--- gene=153, id=66 removed by soft count = 109\n",
      "--- gene=153, id=173 removed by soft count = 128\n",
      "--- number of probes: 185 , kept by blast: 183 , if remove dups: 183\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 154\n",
      "--- gene=154, id=0 removed by soft count = 89\n",
      "--- gene=154, id=21 removed by soft count = 645\n",
      "--- gene=154, id=27 removed by soft count = 190\n",
      "--- gene=154, id=28 removed by soft count = 70\n",
      "--- gene=154, id=29 removed by soft count = 51\n",
      "--- gene=154, id=32 removed by soft count = 89\n",
      "--- gene=154, id=33 removed by soft count = 56\n",
      "--- gene=154, id=34 removed by soft count = 352\n",
      "--- gene=154, id=35 removed by soft count = 66\n",
      "--- gene=154, id=36 removed by soft count = 221\n",
      "--- gene=154, id=38 removed by soft count = 106\n",
      "--- gene=154, id=42 removed by soft count = 671\n",
      "--- gene=154, id=69 removed by soft count = 60\n",
      "--- gene=154, id=72 removed by soft count = 148\n",
      "--- gene=154, id=75 removed by soft count = 205\n",
      "--- number of probes: 159 , kept by blast: 144 , if remove dups: 144\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 155\n",
      "--- gene=155, id=44 removed by soft count = 92\n",
      "--- gene=155, id=80 removed by soft count = 199\n",
      "--- gene=155, id=103 removed by soft count = 69\n",
      "--- gene=155, id=235 removed by soft count = 79\n",
      "--- gene=155, id=237 removed by soft count = 412\n",
      "--- gene=155, id=238 removed by soft count = 96\n",
      "--- number of probes: 258 , kept by blast: 252 , if remove dups: 252\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 156\n",
      "--- gene=156, id=66 removed by soft count = 116\n",
      "--- gene=156, id=67 removed by soft count = 85\n",
      "--- gene=156, id=68 removed by soft count = 17207\n",
      "--- gene=156, id=74 removed by soft count = 687\n",
      "--- gene=156, id=76 removed by soft count = 54\n",
      "--- gene=156, id=80 removed by soft count = 62\n",
      "--- gene=156, id=83 removed by soft count = 93\n",
      "--- gene=156, id=84 removed by soft count = 97\n",
      "--- gene=156, id=143 removed by soft count = 171\n",
      "--- number of probes: 144 , kept by blast: 135 , if remove dups: 135\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 157\n",
      "--- gene=157, id=35 removed by soft count = 114\n",
      "--- gene=157, id=144 removed by soft count = 90\n",
      "--- number of probes: 199 , kept by blast: 197 , if remove dups: 197\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 158\n",
      "--- gene=158, id=21 removed by soft count = 55\n",
      "--- gene=158, id=23 removed by soft count = 99\n",
      "--- gene=158, id=24 removed by soft count = 3871\n",
      "--- gene=158, id=25 removed by soft count = 162\n",
      "--- gene=158, id=29 removed by soft count = 335\n",
      "--- gene=158, id=32 removed by soft count = 139\n",
      "--- gene=158, id=33 removed by soft count = 351\n",
      "--- gene=158, id=58 removed by soft count = 80\n",
      "--- gene=158, id=70 removed by soft count = 61\n",
      "--- gene=158, id=74 removed by soft count = 250\n",
      "--- gene=158, id=78 removed by soft count = 88\n",
      "--- gene=158, id=80 removed by soft count = 213\n",
      "--- gene=158, id=81 removed by soft count = 799\n",
      "--- gene=158, id=97 removed by soft count = 133\n",
      "--- gene=158, id=98 removed by soft count = 437\n",
      "--- gene=158, id=99 removed by soft count = 207\n",
      "--- gene=158, id=100 removed by soft count = 70\n",
      "--- gene=158, id=101 removed by soft count = 118\n",
      "--- gene=158, id=102 removed by soft count = 123\n",
      "--- gene=158, id=103 removed by soft count = 191\n",
      "--- gene=158, id=105 removed by soft count = 89\n",
      "--- gene=158, id=108 removed by soft count = 184\n",
      "--- gene=158, id=109 removed by soft count = 91\n",
      "--- gene=158, id=110 removed by soft count = 65\n",
      "--- gene=158, id=111 removed by soft count = 108\n",
      "--- gene=158, id=112 removed by soft count = 79\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=158, id=113 removed by soft count = 6359\n",
      "--- number of probes: 127 , kept by blast: 100 , if remove dups: 100\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 100\n",
      "-- checking probes in region: 159\n",
      "--- number of probes: 270 , kept by blast: 270 , if remove dups: 270\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 160\n",
      "--- gene=160, id=8 removed by soft count = 142\n",
      "--- gene=160, id=90 removed by soft count = 204\n",
      "--- gene=160, id=91 removed by soft count = 56\n",
      "--- gene=160, id=94 removed by soft count = 288\n",
      "--- gene=160, id=97 removed by soft count = 980\n",
      "--- gene=160, id=110 removed by soft count = 189\n",
      "--- gene=160, id=114 removed by soft count = 151\n",
      "--- gene=160, id=117 removed by soft count = 73\n",
      "--- gene=160, id=118 removed by soft count = 56\n",
      "--- gene=160, id=133 removed by soft count = 60\n",
      "--- gene=160, id=136 removed by soft count = 72\n",
      "--- gene=160, id=137 removed by soft count = 942\n",
      "--- gene=160, id=138 removed by soft count = 3979\n",
      "--- gene=160, id=139 removed by soft count = 144\n",
      "--- number of probes: 141 , kept by blast: 127 , if remove dups: 127\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 161\n",
      "--- gene=161, id=1 removed by soft count = 56\n",
      "--- gene=161, id=44 removed by soft count = 106\n",
      "--- gene=161, id=45 removed by soft count = 52\n",
      "--- gene=161, id=60 removed by soft count = 149\n",
      "--- gene=161, id=111 removed by soft count = 93\n",
      "--- number of probes: 168 , kept by blast: 163 , if remove dups: 163\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 162\n",
      "--- number of probes: 221 , kept by blast: 221 , if remove dups: 221\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 163\n",
      "--- gene=163, id=23 removed by soft count = 53\n",
      "--- gene=163, id=27 removed by soft count = 98\n",
      "--- gene=163, id=29 removed by soft count = 146\n",
      "--- gene=163, id=184 removed by soft count = 410\n",
      "--- gene=163, id=235 removed by soft count = 74\n",
      "--- number of probes: 257 , kept by blast: 252 , if remove dups: 252\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 164\n",
      "--- gene=164, id=31 removed by soft count = 90\n",
      "--- gene=164, id=36 removed by soft count = 101\n",
      "--- gene=164, id=87 removed by soft count = 70\n",
      "--- gene=164, id=235 removed by soft count = 84\n",
      "--- gene=164, id=237 removed by soft count = 60\n",
      "--- gene=164, id=267 removed by soft count = 76\n",
      "--- number of probes: 288 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 165\n",
      "--- gene=165, id=105 removed by soft count = 364\n",
      "--- gene=165, id=106 removed by soft count = 496\n",
      "--- gene=165, id=141 removed by soft count = 62\n",
      "--- gene=165, id=211 removed by soft count = 176\n",
      "--- gene=165, id=212 removed by soft count = 111\n",
      "--- gene=165, id=216 removed by soft count = 279\n",
      "--- gene=165, id=221 removed by soft count = 64\n",
      "--- gene=165, id=224 removed by soft count = 104\n",
      "--- number of probes: 225 , kept by blast: 217 , if remove dups: 217\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 166\n",
      "--- gene=166, id=0 removed by soft count = 52\n",
      "--- gene=166, id=34 removed by soft count = 54\n",
      "--- gene=166, id=35 removed by soft count = 90\n",
      "--- gene=166, id=90 removed by soft count = 624\n",
      "--- number of probes: 103 , kept by blast: 99 , if remove dups: 99\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 99\n",
      "-- checking probes in region: 167\n",
      "--- number of probes: 164 , kept by blast: 164 , if remove dups: 164\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 168\n",
      "--- gene=168, id=68 removed by soft count = 73\n",
      "--- gene=168, id=89 removed by soft count = 57\n",
      "--- gene=168, id=172 removed by soft count = 59\n",
      "--- number of probes: 177 , kept by blast: 174 , if remove dups: 174\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 169\n",
      "--- gene=169, id=27 removed by soft count = 87\n",
      "--- gene=169, id=73 removed by soft count = 248\n",
      "--- gene=169, id=75 removed by soft count = 777\n",
      "--- gene=169, id=110 removed by soft count = 126\n",
      "--- gene=169, id=111 removed by soft count = 53\n",
      "--- gene=169, id=116 removed by soft count = 347\n",
      "--- gene=169, id=121 removed by soft count = 61\n",
      "--- gene=169, id=136 removed by soft count = 2056\n",
      "--- gene=169, id=137 removed by soft count = 57\n",
      "--- number of probes: 140 , kept by blast: 131 , if remove dups: 131\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 170\n",
      "--- gene=170, id=0 removed by soft count = 18330\n",
      "--- gene=170, id=1 removed by soft count = 474\n",
      "--- number of probes: 86 , kept by blast: 84 , if remove dups: 84\n",
      "-- number of probes kept for this region: 84\n",
      "-- checking probes in region: 171\n",
      "--- gene=171, id=2 removed by soft count = 67\n",
      "--- gene=171, id=6 removed by soft count = 483\n",
      "--- gene=171, id=7 removed by soft count = 364\n",
      "--- gene=171, id=9 removed by soft count = 67\n",
      "--- gene=171, id=13 removed by soft count = 138\n",
      "--- gene=171, id=17 removed by soft count = 106\n",
      "--- gene=171, id=122 removed by soft count = 72\n",
      "--- gene=171, id=123 removed by soft count = 65\n",
      "--- number of probes: 139 , kept by blast: 131 , if remove dups: 131\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 172\n",
      "--- gene=172, id=54 removed by soft count = 61\n",
      "--- gene=172, id=215 removed by soft count = 66\n",
      "--- number of probes: 222 , kept by blast: 220 , if remove dups: 220\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 173\n",
      "--- gene=173, id=37 removed by soft count = 58\n",
      "--- gene=173, id=165 removed by soft count = 58\n",
      "--- number of probes: 225 , kept by blast: 223 , if remove dups: 223\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 174\n",
      "--- gene=174, id=62 removed by soft count = 60\n",
      "--- gene=174, id=108 removed by soft count = 222\n",
      "--- gene=174, id=109 removed by soft count = 520\n",
      "--- gene=174, id=137 removed by soft count = 57\n",
      "--- number of probes: 160 , kept by blast: 156 , if remove dups: 156\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 175\n",
      "--- gene=175, id=0 removed by soft count = 72\n",
      "--- gene=175, id=1 removed by soft count = 89\n",
      "--- gene=175, id=2 removed by soft count = 135\n",
      "--- gene=175, id=30 removed by soft count = 195\n",
      "--- gene=175, id=180 removed by soft count = 62\n",
      "--- gene=175, id=193 removed by soft count = 369\n",
      "--- gene=175, id=207 removed by soft count = 86\n",
      "--- number of probes: 210 , kept by blast: 203 , if remove dups: 203\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 176\n",
      "--- gene=176, id=0 removed by soft count = 77\n",
      "--- gene=176, id=125 removed by soft count = 91\n",
      "--- gene=176, id=130 removed by soft count = 75\n",
      "--- gene=176, id=131 removed by soft count = 113\n",
      "--- gene=176, id=133 removed by soft count = 61\n",
      "--- gene=176, id=134 removed by soft count = 53\n",
      "--- number of probes: 213 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 177\n",
      "--- gene=177, id=161 removed by soft count = 54\n",
      "--- number of probes: 268 , kept by blast: 267 , if remove dups: 267\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 178\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- number of probes: 261 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 179\n",
      "--- gene=179, id=79 removed by soft count = 118\n",
      "--- gene=179, id=89 removed by soft count = 91\n",
      "--- gene=179, id=90 removed by soft count = 58\n",
      "--- gene=179, id=91 removed by soft count = 74\n",
      "--- gene=179, id=92 removed by soft count = 138\n",
      "--- gene=179, id=113 removed by soft count = 83\n",
      "--- number of probes: 163 , kept by blast: 157 , if remove dups: 157\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 180\n",
      "--- gene=180, id=0 removed by soft count = 65\n",
      "--- gene=180, id=1 removed by soft count = 1167\n",
      "--- gene=180, id=2 removed by soft count = 84\n",
      "--- gene=180, id=3 removed by soft count = 157\n",
      "--- gene=180, id=5 removed by soft count = 820\n",
      "--- gene=180, id=7 removed by soft count = 643\n",
      "--- gene=180, id=9 removed by soft count = 6065\n",
      "--- gene=180, id=37 removed by soft count = 1492\n",
      "--- gene=180, id=68 removed by soft count = 55\n",
      "--- gene=180, id=69 removed by soft count = 97\n",
      "--- gene=180, id=75 removed by soft count = 220\n",
      "--- gene=180, id=77 removed by soft count = 72\n",
      "--- gene=180, id=79 removed by soft count = 281\n",
      "--- gene=180, id=81 removed by soft count = 61\n",
      "--- gene=180, id=120 removed by soft count = 63\n",
      "--- gene=180, id=123 removed by soft count = 163\n",
      "--- gene=180, id=134 removed by soft count = 76\n",
      "--- number of probes: 136 , kept by blast: 119 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 181\n",
      "--- gene=181, id=4 removed by soft count = 298\n",
      "--- gene=181, id=29 removed by soft count = 448\n",
      "--- gene=181, id=52 removed by soft count = 203\n",
      "--- gene=181, id=53 removed by soft count = 138\n",
      "--- gene=181, id=87 removed by soft count = 194\n",
      "--- gene=181, id=101 removed by soft count = 52\n",
      "--- gene=181, id=102 removed by soft count = 1753\n",
      "--- gene=181, id=140 removed by soft count = 306\n",
      "--- gene=181, id=173 removed by soft count = 137\n",
      "--- number of probes: 181 , kept by blast: 172 , if remove dups: 172\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 182\n",
      "--- gene=182, id=25 removed by soft count = 628\n",
      "--- gene=182, id=26 removed by soft count = 409\n",
      "--- gene=182, id=27 removed by soft count = 172\n",
      "--- gene=182, id=30 removed by soft count = 54\n",
      "--- gene=182, id=61 removed by soft count = 475\n",
      "--- gene=182, id=67 removed by soft count = 656\n",
      "--- gene=182, id=68 removed by soft count = 59\n",
      "--- gene=182, id=74 removed by soft count = 183\n",
      "--- gene=182, id=90 removed by soft count = 24615\n",
      "--- gene=182, id=91 removed by soft count = 238\n",
      "--- gene=182, id=92 removed by soft count = 142\n",
      "--- gene=182, id=93 removed by soft count = 240\n",
      "--- gene=182, id=94 removed by soft count = 114\n",
      "--- number of probes: 99 , kept by blast: 86 , if remove dups: 86\n",
      "-- number of probes kept for this region: 86\n",
      "-- checking probes in region: 183\n",
      "--- gene=183, id=34 removed by soft count = 219\n",
      "--- gene=183, id=35 removed by soft count = 312\n",
      "--- number of probes: 110 , kept by blast: 108 , if remove dups: 108\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 108\n",
      "-- checking probes in region: 184\n",
      "--- gene=184, id=117 removed by soft count = 201\n",
      "--- number of probes: 126 , kept by blast: 125 , if remove dups: 125\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 185\n",
      "--- gene=185, id=63 removed by soft count = 243\n",
      "--- gene=185, id=81 removed by soft count = 175\n",
      "--- gene=185, id=86 removed by soft count = 396\n",
      "--- gene=185, id=147 removed by soft count = 80\n",
      "--- number of probes: 290 , kept by blast: 286 , if remove dups: 286\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 186\n",
      "--- gene=186, id=45 removed by soft count = 148\n",
      "--- gene=186, id=71 removed by soft count = 105\n",
      "--- number of probes: 182 , kept by blast: 180 , if remove dups: 180\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 187\n",
      "--- gene=187, id=4 removed by soft count = 142\n",
      "--- gene=187, id=5 removed by soft count = 174\n",
      "--- gene=187, id=9 removed by soft count = 69\n",
      "--- gene=187, id=10 removed by soft count = 206\n",
      "--- gene=187, id=109 removed by soft count = 74\n",
      "--- gene=187, id=142 removed by soft count = 63\n",
      "--- gene=187, id=188 removed by soft count = 146\n",
      "--- number of probes: 216 , kept by blast: 209 , if remove dups: 209\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 188\n",
      "--- gene=188, id=7 removed by soft count = 506\n",
      "--- gene=188, id=36 removed by soft count = 63\n",
      "--- gene=188, id=40 removed by soft count = 56\n",
      "--- gene=188, id=41 removed by soft count = 77\n",
      "--- gene=188, id=42 removed by soft count = 184\n",
      "--- gene=188, id=56 removed by soft count = 247\n",
      "--- gene=188, id=73 removed by soft count = 85\n",
      "--- gene=188, id=74 removed by soft count = 387\n",
      "--- gene=188, id=75 removed by soft count = 237\n",
      "--- gene=188, id=76 removed by soft count = 359\n",
      "--- gene=188, id=85 removed by soft count = 100\n",
      "--- gene=188, id=86 removed by soft count = 53\n",
      "--- gene=188, id=87 removed by soft count = 226\n",
      "--- gene=188, id=88 removed by soft count = 97\n",
      "--- gene=188, id=90 removed by soft count = 212\n",
      "--- gene=188, id=91 removed by soft count = 154\n",
      "--- gene=188, id=92 removed by soft count = 64\n",
      "--- gene=188, id=93 removed by soft count = 172\n",
      "--- gene=188, id=94 removed by soft count = 177\n",
      "--- gene=188, id=95 removed by soft count = 168\n",
      "--- gene=188, id=96 removed by soft count = 176\n",
      "--- gene=188, id=97 removed by soft count = 87\n",
      "--- gene=188, id=98 removed by soft count = 202\n",
      "--- gene=188, id=100 removed by soft count = 121\n",
      "--- gene=188, id=101 removed by soft count = 152\n",
      "--- gene=188, id=102 removed by soft count = 241\n",
      "--- number of probes: 114 , kept by blast: 88 , if remove dups: 88\n",
      "-- number of probes kept for this region: 88\n",
      "-- checking probes in region: 190\n",
      "--- gene=190, id=22 removed by soft count = 73\n",
      "--- gene=190, id=143 removed by soft count = 70\n",
      "--- gene=190, id=144 removed by soft count = 94\n",
      "--- gene=190, id=146 removed by soft count = 115\n",
      "--- gene=190, id=148 removed by soft count = 119\n",
      "--- gene=190, id=153 removed by soft count = 80\n",
      "--- gene=190, id=168 removed by soft count = 74\n",
      "--- gene=190, id=174 removed by soft count = 633\n",
      "--- gene=190, id=190 removed by soft count = 222\n",
      "--- gene=190, id=207 removed by soft count = 85\n",
      "--- gene=190, id=211 removed by soft count = 72\n",
      "--- gene=190, id=220 removed by soft count = 407\n",
      "--- gene=190, id=223 removed by soft count = 426\n",
      "--- number of probes: 227 , kept by blast: 214 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 191\n",
      "--- gene=191, id=46 removed by soft count = 1462\n",
      "--- number of probes: 142 , kept by blast: 141 , if remove dups: 139\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 192\n",
      "--- gene=192, id=145 removed by soft count = 163\n",
      "--- gene=192, id=178 removed by soft count = 60\n",
      "--- gene=192, id=211 removed by soft count = 62\n",
      "--- gene=192, id=215 removed by soft count = 70\n",
      "--- gene=192, id=224 removed by soft count = 258\n",
      "--- number of probes: 251 , kept by blast: 246 , if remove dups: 246\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 193\n",
      "--- gene=193, id=95 removed by soft count = 57\n",
      "--- number of probes: 275 , kept by blast: 274 , if remove dups: 274\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 195\n",
      "--- gene=195, id=14 removed by soft count = 161\n",
      "--- gene=195, id=60 removed by soft count = 155\n",
      "--- gene=195, id=61 removed by soft count = 84\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=195, id=62 removed by soft count = 572\n",
      "--- number of probes: 186 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 196\n",
      "--- gene=196, id=7 removed by soft count = 108\n",
      "--- gene=196, id=8 removed by soft count = 504\n",
      "--- gene=196, id=9 removed by soft count = 226\n",
      "--- gene=196, id=10 removed by soft count = 80\n",
      "--- gene=196, id=12 removed by soft count = 274\n",
      "--- gene=196, id=14 removed by soft count = 1032\n",
      "--- gene=196, id=15 removed by soft count = 62\n",
      "--- gene=196, id=16 removed by soft count = 1287\n",
      "--- gene=196, id=18 removed by soft count = 2812\n",
      "--- gene=196, id=19 removed by soft count = 85\n",
      "--- gene=196, id=21 removed by soft count = 79\n",
      "--- gene=196, id=30 removed by soft count = 77\n",
      "--- gene=196, id=47 removed by soft count = 70\n",
      "--- gene=196, id=48 removed by soft count = 153\n",
      "--- gene=196, id=52 removed by soft count = 138\n",
      "--- gene=196, id=60 removed by soft count = 162\n",
      "--- gene=196, id=68 removed by soft count = 133\n",
      "--- number of probes: 115 , kept by blast: 98 , if remove dups: 98\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 98\n",
      "-- checking probes in region: 197\n",
      "--- gene=197, id=30 removed by soft count = 174\n",
      "--- gene=197, id=31 removed by soft count = 642\n",
      "--- gene=197, id=66 removed by soft count = 109\n",
      "--- number of probes: 115 , kept by blast: 112 , if remove dups: 112\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 112\n",
      "-- checking probes in region: 198\n",
      "--- number of probes: 289 , kept by blast: 289 , if remove dups: 289\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 199\n",
      "--- gene=199, id=12 removed by soft count = 119\n",
      "--- gene=199, id=17 removed by soft count = 401\n",
      "--- gene=199, id=18 removed by soft count = 284\n",
      "--- gene=199, id=37 removed by soft count = 74\n",
      "--- gene=199, id=38 removed by soft count = 119\n",
      "--- gene=199, id=39 removed by soft count = 52\n",
      "--- gene=199, id=40 removed by soft count = 114\n",
      "--- gene=199, id=41 removed by soft count = 893\n",
      "--- gene=199, id=43 removed by soft count = 3324\n",
      "--- gene=199, id=44 removed by soft count = 480\n",
      "--- gene=199, id=45 removed by soft count = 413\n",
      "--- gene=199, id=47 removed by soft count = 54\n",
      "--- gene=199, id=87 removed by soft count = 311\n",
      "--- gene=199, id=88 removed by soft count = 65\n",
      "--- number of probes: 175 , kept by blast: 161 , if remove dups: 161\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 200\n",
      "--- gene=200, id=165 removed by soft count = 74\n",
      "--- number of probes: 258 , kept by blast: 257 , if remove dups: 257\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 201\n",
      "--- gene=201, id=41 removed by soft count = 52\n",
      "--- gene=201, id=84 removed by soft count = 70\n",
      "--- gene=201, id=103 removed by soft count = 105\n",
      "--- gene=201, id=159 removed by soft count = 84\n",
      "--- number of probes: 169 , kept by blast: 165 , if remove dups: 165\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 202\n",
      "--- gene=202, id=33 removed by soft count = 73\n",
      "--- gene=202, id=102 removed by soft count = 122\n",
      "--- gene=202, id=103 removed by soft count = 88\n",
      "--- gene=202, id=174 removed by soft count = 41528\n",
      "--- gene=202, id=176 removed by soft count = 76\n",
      "--- gene=202, id=177 removed by soft count = 92\n",
      "--- gene=202, id=179 removed by soft count = 103\n",
      "--- number of probes: 180 , kept by blast: 173 , if remove dups: 173\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 203\n",
      "--- gene=203, id=56 removed by soft count = 96\n",
      "--- gene=203, id=57 removed by soft count = 111\n",
      "--- gene=203, id=115 removed by soft count = 69\n",
      "--- gene=203, id=116 removed by soft count = 190\n",
      "--- gene=203, id=117 removed by soft count = 255\n",
      "--- gene=203, id=123 removed by soft count = 58\n",
      "--- gene=203, id=180 removed by soft count = 116\n",
      "--- number of probes: 197 , kept by blast: 190 , if remove dups: 190\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 204\n",
      "--- gene=204, id=132 removed by soft count = 93\n",
      "--- gene=204, id=137 removed by soft count = 114\n",
      "--- gene=204, id=139 removed by soft count = 52\n",
      "--- gene=204, id=155 removed by soft count = 63\n",
      "--- gene=204, id=234 removed by soft count = 152\n",
      "--- number of probes: 314 , kept by blast: 309 , if remove dups: 309\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 205\n",
      "--- gene=205, id=2 removed by soft count = 75\n",
      "--- gene=205, id=26 removed by soft count = 62\n",
      "--- gene=205, id=247 removed by soft count = 81\n",
      "--- gene=205, id=249 removed by soft count = 123\n",
      "--- number of probes: 264 , kept by blast: 260 , if remove dups: 259\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 206\n",
      "--- gene=206, id=179 removed by soft count = 370\n",
      "--- gene=206, id=180 removed by soft count = 194\n",
      "--- number of probes: 198 , kept by blast: 196 , if remove dups: 196\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 207\n",
      "--- gene=207, id=117 removed by soft count = 146\n",
      "--- gene=207, id=165 removed by soft count = 171\n",
      "--- gene=207, id=221 removed by soft count = 75\n",
      "--- number of probes: 294 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 208\n",
      "--- gene=208, id=110 removed by soft count = 1619\n",
      "--- gene=208, id=115 removed by soft count = 172\n",
      "--- gene=208, id=120 removed by soft count = 74\n",
      "--- gene=208, id=152 removed by soft count = 78\n",
      "--- gene=208, id=153 removed by soft count = 191\n",
      "--- number of probes: 191 , kept by blast: 186 , if remove dups: 186\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 209\n",
      "--- gene=209, id=174 removed by soft count = 173\n",
      "--- number of probes: 290 , kept by blast: 289 , if remove dups: 289\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 210\n",
      "--- gene=210, id=84 removed by soft count = 57\n",
      "--- number of probes: 294 , kept by blast: 293 , if remove dups: 293\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 211\n",
      "--- gene=211, id=22 removed by soft count = 360\n",
      "--- gene=211, id=24 removed by soft count = 87\n",
      "--- gene=211, id=40 removed by soft count = 88\n",
      "--- gene=211, id=86 removed by soft count = 69\n",
      "--- gene=211, id=93 removed by soft count = 89\n",
      "--- number of probes: 184 , kept by blast: 179 , if remove dups: 179\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 212\n",
      "--- gene=212, id=9 removed by soft count = 120\n",
      "--- gene=212, id=11 removed by soft count = 255\n",
      "--- gene=212, id=14 removed by soft count = 136\n",
      "--- gene=212, id=16 removed by soft count = 1782\n",
      "--- gene=212, id=20 removed by soft count = 174\n",
      "--- gene=212, id=22 removed by soft count = 132\n",
      "--- gene=212, id=24 removed by soft count = 65\n",
      "--- gene=212, id=25 removed by soft count = 522\n",
      "--- gene=212, id=33 removed by soft count = 657\n",
      "--- gene=212, id=34 removed by soft count = 117\n",
      "--- gene=212, id=42 removed by soft count = 80\n",
      "--- gene=212, id=45 removed by soft count = 136\n",
      "--- gene=212, id=46 removed by soft count = 79\n",
      "--- gene=212, id=65 removed by soft count = 287\n",
      "--- gene=212, id=119 removed by soft count = 202\n",
      "--- number of probes: 160 , kept by blast: 145 , if remove dups: 145\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 213\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=213, id=55 removed by soft count = 92\n",
      "--- gene=213, id=106 removed by soft count = 165\n",
      "--- gene=213, id=139 removed by soft count = 87\n",
      "--- gene=213, id=149 removed by soft count = 123\n",
      "--- gene=213, id=150 removed by soft count = 478\n",
      "--- number of probes: 241 , kept by blast: 236 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 214\n",
      "--- gene=214, id=5 removed by soft count = 126\n",
      "--- gene=214, id=11 removed by soft count = 98\n",
      "--- gene=214, id=19 removed by soft count = 69\n",
      "--- gene=214, id=20 removed by soft count = 71\n",
      "--- gene=214, id=67 removed by soft count = 783\n",
      "--- gene=214, id=83 removed by soft count = 566\n",
      "--- gene=214, id=87 removed by soft count = 89\n",
      "--- gene=214, id=137 removed by soft count = 68\n",
      "--- gene=214, id=169 removed by soft count = 78\n",
      "--- gene=214, id=170 removed by soft count = 65\n",
      "--- number of probes: 180 , kept by blast: 170 , if remove dups: 170\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 215\n",
      "--- gene=215, id=87 removed by soft count = 112\n",
      "--- gene=215, id=106 removed by soft count = 619\n",
      "--- number of probes: 122 , kept by blast: 120 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 218\n",
      "--- gene=218, id=22 removed by soft count = 100\n",
      "--- gene=218, id=78 removed by soft count = 66\n",
      "--- number of probes: 89 , kept by blast: 87 , if remove dups: 85\n",
      "-- number of probes kept for this region: 87\n",
      "-- checking probes in region: 219\n",
      "--- gene=219, id=5 removed by soft count = 56\n",
      "--- gene=219, id=121 removed by soft count = 542\n",
      "--- gene=219, id=122 removed by soft count = 620\n",
      "--- gene=219, id=123 removed by soft count = 2771\n",
      "--- gene=219, id=124 removed by soft count = 80\n",
      "--- gene=219, id=157 removed by soft count = 55\n",
      "--- gene=219, id=162 removed by soft count = 53\n",
      "--- number of probes: 164 , kept by blast: 157 , if remove dups: 157\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 220\n",
      "--- gene=220, id=35 removed by soft count = 440\n",
      "--- gene=220, id=37 removed by soft count = 163\n",
      "--- number of probes: 298 , kept by blast: 296 , if remove dups: 296\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 221\n",
      "--- gene=221, id=92 removed by soft count = 184\n",
      "--- gene=221, id=99 removed by soft count = 55\n",
      "--- gene=221, id=175 removed by soft count = 181\n",
      "--- gene=221, id=198 removed by soft count = 119\n",
      "--- gene=221, id=199 removed by soft count = 211\n",
      "--- gene=221, id=201 removed by soft count = 245\n",
      "--- number of probes: 203 , kept by blast: 197 , if remove dups: 197\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 222\n",
      "--- number of probes: 296 , kept by blast: 296 , if remove dups: 296\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 223\n",
      "--- gene=223, id=19 removed by soft count = 84\n",
      "--- gene=223, id=22 removed by soft count = 105\n",
      "--- gene=223, id=23 removed by soft count = 56\n",
      "--- gene=223, id=27 removed by soft count = 342\n",
      "--- gene=223, id=28 removed by soft count = 543\n",
      "--- gene=223, id=31 removed by soft count = 84\n",
      "--- gene=223, id=51 removed by soft count = 210\n",
      "--- gene=223, id=52 removed by soft count = 166\n",
      "--- gene=223, id=53 removed by soft count = 142\n",
      "--- gene=223, id=60 removed by soft count = 53\n",
      "--- gene=223, id=152 removed by soft count = 113\n",
      "--- number of probes: 158 , kept by blast: 147 , if remove dups: 147\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 224\n",
      "--- gene=224, id=0 removed by soft count = 514\n",
      "--- gene=224, id=20 removed by soft count = 55\n",
      "--- gene=224, id=32 removed by soft count = 1058\n",
      "--- gene=224, id=39 removed by soft count = 296\n",
      "--- gene=224, id=40 removed by soft count = 259\n",
      "--- gene=224, id=48 removed by soft count = 145\n",
      "--- gene=224, id=49 removed by soft count = 534\n",
      "--- gene=224, id=50 removed by soft count = 14799\n",
      "--- gene=224, id=85 removed by soft count = 148\n",
      "--- gene=224, id=86 removed by soft count = 335\n",
      "--- gene=224, id=87 removed by soft count = 1952\n",
      "--- number of probes: 104 , kept by blast: 93 , if remove dups: 93\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 93\n",
      "-- checking probes in region: 225\n",
      "--- gene=225, id=0 removed by soft count = 47519\n",
      "--- gene=225, id=15 removed by soft count = 158\n",
      "--- gene=225, id=36 removed by soft count = 63\n",
      "--- gene=225, id=57 removed by soft count = 105\n",
      "--- gene=225, id=64 removed by soft count = 132\n",
      "--- gene=225, id=65 removed by soft count = 68\n",
      "--- gene=225, id=66 removed by soft count = 63\n",
      "--- number of probes: 84 , kept by blast: 77 , if remove dups: 77\n",
      "-- number of probes kept for this region: 77\n",
      "-- checking probes in region: 226\n",
      "--- gene=226, id=183 removed by soft count = 649\n",
      "--- number of probes: 184 , kept by blast: 183 , if remove dups: 183\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 227\n",
      "--- gene=227, id=7 removed by soft count = 172\n",
      "--- gene=227, id=9 removed by soft count = 89\n",
      "--- gene=227, id=10 removed by soft count = 189\n",
      "--- gene=227, id=18 removed by soft count = 76\n",
      "--- gene=227, id=21 removed by soft count = 121\n",
      "--- gene=227, id=22 removed by soft count = 110\n",
      "--- gene=227, id=23 removed by soft count = 400\n",
      "--- gene=227, id=84 removed by soft count = 51\n",
      "--- gene=227, id=87 removed by soft count = 187\n",
      "--- gene=227, id=94 removed by soft count = 927\n",
      "--- gene=227, id=103 removed by soft count = 77\n",
      "--- gene=227, id=111 removed by soft count = 113\n",
      "--- gene=227, id=115 removed by soft count = 379\n",
      "--- gene=227, id=116 removed by soft count = 91\n",
      "--- gene=227, id=118 removed by soft count = 54\n",
      "--- number of probes: 119 , kept by blast: 104 , if remove dups: 104\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 104\n",
      "-- checking probes in region: 228\n",
      "--- gene=228, id=5 removed by soft count = 105\n",
      "--- gene=228, id=8 removed by soft count = 294\n",
      "--- number of probes: 84 , kept by blast: 82 , if remove dups: 82\n",
      "-- number of probes kept for this region: 82\n",
      "-- checking probes in region: 229\n",
      "--- gene=229, id=61 removed by soft count = 131\n",
      "--- gene=229, id=153 removed by soft count = 314\n",
      "--- number of probes: 244 , kept by blast: 242 , if remove dups: 242\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 230\n",
      "--- gene=230, id=4 removed by soft count = 61\n",
      "--- gene=230, id=35 removed by soft count = 58\n",
      "--- gene=230, id=175 removed by soft count = 93\n",
      "--- number of probes: 264 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 231\n",
      "--- gene=231, id=95 removed by soft count = 234\n",
      "--- gene=231, id=97 removed by soft count = 64\n",
      "--- gene=231, id=100 removed by soft count = 54\n",
      "--- number of probes: 143 , kept by blast: 140 , if remove dups: 140\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 232\n",
      "--- gene=232, id=12 removed by soft count = 51\n",
      "--- gene=232, id=14 removed by soft count = 135\n",
      "--- gene=232, id=90 removed by soft count = 260\n",
      "--- gene=232, id=92 removed by soft count = 120\n",
      "--- gene=232, id=108 removed by soft count = 76\n",
      "--- number of probes: 110 , kept by blast: 105 , if remove dups: 105\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 105\n",
      "-- checking probes in region: 233\n",
      "--- gene=233, id=2 removed by soft count = 223\n",
      "--- gene=233, id=21 removed by soft count = 120\n",
      "--- gene=233, id=24 removed by soft count = 261\n",
      "--- number of probes: 223 , kept by blast: 220 , if remove dups: 220\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 234\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=234, id=11 removed by soft count = 438\n",
      "--- gene=234, id=111 removed by soft count = 81\n",
      "--- gene=234, id=134 removed by soft count = 143\n",
      "--- gene=234, id=155 removed by soft count = 97\n",
      "--- number of probes: 194 , kept by blast: 190 , if remove dups: 190\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 235\n",
      "--- gene=235, id=40 removed by soft count = 65\n",
      "--- gene=235, id=48 removed by soft count = 88\n",
      "--- gene=235, id=121 removed by soft count = 125\n",
      "--- gene=235, id=122 removed by soft count = 146\n",
      "--- number of probes: 151 , kept by blast: 147 , if remove dups: 147\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 236\n",
      "--- gene=236, id=38 removed by soft count = 110\n",
      "--- gene=236, id=49 removed by soft count = 78\n",
      "--- gene=236, id=50 removed by soft count = 295\n",
      "--- number of probes: 220 , kept by blast: 217 , if remove dups: 217\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 237\n",
      "--- gene=237, id=5 removed by soft count = 52\n",
      "--- gene=237, id=34 removed by soft count = 20399\n",
      "--- gene=237, id=35 removed by soft count = 883\n",
      "--- gene=237, id=37 removed by soft count = 55\n",
      "--- gene=237, id=39 removed by soft count = 8435\n",
      "--- gene=237, id=77 removed by soft count = 55\n",
      "--- gene=237, id=95 removed by soft count = 94\n",
      "--- gene=237, id=102 removed by soft count = 121\n",
      "--- gene=237, id=103 removed by soft count = 111\n",
      "--- gene=237, id=105 removed by soft count = 1073\n",
      "--- gene=237, id=106 removed by soft count = 4241\n",
      "--- gene=237, id=107 removed by soft count = 1193\n",
      "--- number of probes: 108 , kept by blast: 96 , if remove dups: 96\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 96\n",
      "-- checking probes in region: 238\n",
      "--- gene=238, id=142 removed by soft count = 79\n",
      "--- gene=238, id=146 removed by soft count = 54\n",
      "--- number of probes: 256 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 239\n",
      "--- gene=239, id=38 removed by soft count = 296\n",
      "--- number of probes: 257 , kept by blast: 256 , if remove dups: 256\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 240\n",
      "--- gene=240, id=119 removed by soft count = 526\n",
      "--- gene=240, id=120 removed by soft count = 88\n",
      "--- gene=240, id=133 removed by soft count = 81\n",
      "--- gene=240, id=135 removed by soft count = 530\n",
      "--- gene=240, id=136 removed by soft count = 2111\n",
      "--- gene=240, id=138 removed by soft count = 352\n",
      "--- number of probes: 139 , kept by blast: 133 , if remove dups: 133\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 242\n",
      "--- gene=242, id=51 removed by soft count = 198\n",
      "--- gene=242, id=102 removed by soft count = 817\n",
      "--- gene=242, id=172 removed by soft count = 142\n",
      "--- gene=242, id=240 removed by soft count = 51\n",
      "--- gene=242, id=250 removed by soft count = 53\n",
      "--- number of probes: 299 , kept by blast: 294 , if remove dups: 294\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 243\n",
      "--- gene=243, id=117 removed by soft count = 99\n",
      "--- gene=243, id=149 removed by soft count = 85\n",
      "--- gene=243, id=166 removed by soft count = 55\n",
      "--- number of probes: 251 , kept by blast: 248 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 244\n",
      "--- gene=244, id=23 removed by soft count = 86\n",
      "--- gene=244, id=102 removed by soft count = 259\n",
      "--- gene=244, id=153 removed by soft count = 62\n",
      "--- number of probes: 237 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 245\n",
      "--- gene=245, id=3 removed by soft count = 90\n",
      "--- gene=245, id=17 removed by soft count = 1291\n",
      "--- gene=245, id=18 removed by soft count = 97\n",
      "--- gene=245, id=68 removed by soft count = 105\n",
      "--- gene=245, id=84 removed by soft count = 59\n",
      "--- gene=245, id=104 removed by soft count = 60\n",
      "--- gene=245, id=106 removed by soft count = 179\n",
      "--- gene=245, id=107 removed by soft count = 139\n",
      "--- gene=245, id=111 removed by soft count = 150\n",
      "--- number of probes: 154 , kept by blast: 145 , if remove dups: 145\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 246\n",
      "--- gene=246, id=71 removed by soft count = 124\n",
      "--- gene=246, id=72 removed by soft count = 164\n",
      "--- gene=246, id=76 removed by soft count = 131\n",
      "--- gene=246, id=77 removed by soft count = 101\n",
      "--- gene=246, id=79 removed by soft count = 184\n",
      "--- gene=246, id=113 removed by soft count = 124\n",
      "--- gene=246, id=115 removed by soft count = 91\n",
      "--- number of probes: 151 , kept by blast: 144 , if remove dups: 144\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 247\n",
      "--- number of probes: 228 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 248\n",
      "--- gene=248, id=38 removed by soft count = 69\n",
      "--- gene=248, id=55 removed by soft count = 113\n",
      "--- gene=248, id=56 removed by soft count = 193\n",
      "--- gene=248, id=57 removed by soft count = 9921\n",
      "--- gene=248, id=87 removed by soft count = 74\n",
      "--- gene=248, id=89 removed by soft count = 56\n",
      "--- gene=248, id=90 removed by soft count = 131\n",
      "--- gene=248, id=91 removed by soft count = 1599\n",
      "--- gene=248, id=92 removed by soft count = 271\n",
      "--- gene=248, id=93 removed by soft count = 122\n",
      "--- gene=248, id=118 removed by soft count = 95\n",
      "--- gene=248, id=119 removed by soft count = 249\n",
      "--- number of probes: 135 , kept by blast: 123 , if remove dups: 123\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 249\n",
      "--- gene=249, id=24 removed by soft count = 109\n",
      "--- gene=249, id=25 removed by soft count = 68\n",
      "--- gene=249, id=26 removed by soft count = 52\n",
      "--- number of probes: 211 , kept by blast: 208 , if remove dups: 208\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 250\n",
      "--- gene=250, id=71 removed by soft count = 58\n",
      "--- gene=250, id=75 removed by soft count = 162\n",
      "--- gene=250, id=78 removed by soft count = 66\n",
      "--- number of probes: 100 , kept by blast: 97 , if remove dups: 97\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 97\n",
      "-- checking probes in region: 251\n",
      "--- gene=251, id=192 removed by soft count = 555\n",
      "--- number of probes: 207 , kept by blast: 206 , if remove dups: 206\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 252\n",
      "--- gene=252, id=30 removed by soft count = 98\n",
      "--- gene=252, id=66 removed by soft count = 203\n",
      "--- gene=252, id=67 removed by soft count = 61\n",
      "--- gene=252, id=75 removed by soft count = 82\n",
      "--- gene=252, id=77 removed by soft count = 72\n",
      "--- gene=252, id=79 removed by soft count = 114\n",
      "--- gene=252, id=80 removed by soft count = 157\n",
      "--- gene=252, id=89 removed by soft count = 66\n",
      "--- gene=252, id=114 removed by soft count = 54\n",
      "--- gene=252, id=115 removed by soft count = 594\n",
      "--- number of probes: 137 , kept by blast: 127 , if remove dups: 127\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 253\n",
      "--- gene=253, id=78 removed by hard count = 4\n",
      "--- number of probes: 282 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 254\n",
      "--- gene=254, id=24 removed by soft count = 79\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=254, id=27 removed by soft count = 3291\n",
      "--- gene=254, id=51 removed by soft count = 51\n",
      "--- gene=254, id=52 removed by soft count = 103\n",
      "--- gene=254, id=53 removed by soft count = 471\n",
      "--- gene=254, id=96 removed by soft count = 6458\n",
      "--- number of probes: 139 , kept by blast: 133 , if remove dups: 133\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 255\n",
      "--- gene=255, id=0 removed by soft count = 358\n",
      "--- gene=255, id=2 removed by soft count = 72\n",
      "--- gene=255, id=3 removed by soft count = 92\n",
      "--- gene=255, id=4 removed by soft count = 141\n",
      "--- gene=255, id=18 removed by soft count = 890\n",
      "--- gene=255, id=31 removed by soft count = 121\n",
      "--- gene=255, id=117 removed by soft count = 320\n",
      "--- number of probes: 119 , kept by blast: 112 , if remove dups: 112\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 112\n",
      "-- checking probes in region: 256\n",
      "--- gene=256, id=38 removed by soft count = 91\n",
      "--- gene=256, id=76 removed by soft count = 156\n",
      "--- gene=256, id=109 removed by soft count = 201\n",
      "--- gene=256, id=110 removed by soft count = 506\n",
      "--- gene=256, id=111 removed by soft count = 352\n",
      "--- gene=256, id=112 removed by soft count = 75\n",
      "--- gene=256, id=113 removed by soft count = 353\n",
      "--- gene=256, id=114 removed by soft count = 297\n",
      "--- gene=256, id=128 removed by soft count = 154\n",
      "--- gene=256, id=129 removed by soft count = 148\n",
      "--- gene=256, id=131 removed by soft count = 179\n",
      "--- gene=256, id=134 removed by soft count = 273\n",
      "--- gene=256, id=135 removed by soft count = 266\n",
      "--- number of probes: 230 , kept by blast: 217 , if remove dups: 217\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 257\n",
      "--- gene=257, id=8 removed by soft count = 60\n",
      "--- gene=257, id=9 removed by soft count = 80\n",
      "--- gene=257, id=14 removed by soft count = 107\n",
      "--- gene=257, id=112 removed by soft count = 92\n",
      "--- gene=257, id=123 removed by soft count = 63\n",
      "--- gene=257, id=192 removed by soft count = 138\n",
      "--- number of probes: 231 , kept by blast: 225 , if remove dups: 225\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 258\n",
      "--- gene=258, id=39 removed by soft count = 87\n",
      "--- gene=258, id=75 removed by soft count = 113\n",
      "--- number of probes: 300 , kept by blast: 298 , if remove dups: 298\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 259\n",
      "--- gene=259, id=79 removed by soft count = 54\n",
      "--- gene=259, id=80 removed by soft count = 101\n",
      "--- gene=259, id=128 removed by soft count = 72\n",
      "--- gene=259, id=202 removed by soft count = 103\n",
      "--- gene=259, id=203 removed by soft count = 557\n",
      "--- number of probes: 207 , kept by blast: 202 , if remove dups: 202\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 260\n",
      "--- gene=260, id=29 removed by soft count = 65\n",
      "--- gene=260, id=35 removed by soft count = 64\n",
      "--- gene=260, id=65 removed by soft count = 139\n",
      "--- gene=260, id=66 removed by soft count = 977\n",
      "--- gene=260, id=69 removed by soft count = 61\n",
      "--- gene=260, id=81 removed by soft count = 1034\n",
      "--- gene=260, id=110 removed by soft count = 64\n",
      "--- gene=260, id=125 removed by soft count = 373\n",
      "--- number of probes: 160 , kept by blast: 152 , if remove dups: 152\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 261\n",
      "--- number of probes: 137 , kept by blast: 137 , if remove dups: 137\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 262\n",
      "--- gene=262, id=0 removed by soft count = 57\n",
      "--- gene=262, id=1 removed by soft count = 426\n",
      "--- gene=262, id=41 removed by soft count = 52\n",
      "--- gene=262, id=44 removed by soft count = 152\n",
      "--- gene=262, id=46 removed by soft count = 176\n",
      "--- gene=262, id=47 removed by soft count = 55\n",
      "--- gene=262, id=48 removed by soft count = 54\n",
      "--- gene=262, id=49 removed by soft count = 280\n",
      "--- gene=262, id=59 removed by soft count = 72\n",
      "--- gene=262, id=60 removed by soft count = 183\n",
      "--- gene=262, id=63 removed by soft count = 74\n",
      "--- gene=262, id=93 removed by soft count = 62\n",
      "--- gene=262, id=94 removed by soft count = 62\n",
      "--- number of probes: 101 , kept by blast: 88 , if remove dups: 87\n",
      "-- number of probes kept for this region: 88\n",
      "-- checking probes in region: 263\n",
      "--- gene=263, id=155 removed by soft count = 54\n",
      "--- number of probes: 183 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 264\n",
      "--- gene=264, id=25 removed by soft count = 149\n",
      "--- gene=264, id=28 removed by soft count = 147\n",
      "--- gene=264, id=29 removed by soft count = 112\n",
      "--- gene=264, id=32 removed by soft count = 53\n",
      "--- gene=264, id=152 removed by soft count = 56\n",
      "--- gene=264, id=182 removed by soft count = 58\n",
      "--- gene=264, id=185 removed by soft count = 283\n",
      "--- number of probes: 186 , kept by blast: 179 , if remove dups: 179\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 266\n",
      "--- gene=266, id=167 removed by hard count = 5\n",
      "--- number of probes: 207 , kept by blast: 206 , if remove dups: 206\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 267\n",
      "--- gene=267, id=112 removed by soft count = 102\n",
      "--- gene=267, id=278 removed by soft count = 97\n",
      "--- number of probes: 309 , kept by blast: 307 , if remove dups: 307\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 268\n",
      "--- gene=268, id=3 removed by soft count = 150\n",
      "--- gene=268, id=4 removed by soft count = 90\n",
      "--- gene=268, id=138 removed by soft count = 2560\n",
      "--- gene=268, id=196 removed by hard count = 3\n",
      "--- number of probes: 299 , kept by blast: 295 , if remove dups: 295\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 269\n",
      "--- gene=269, id=147 removed by soft count = 214\n",
      "--- number of probes: 306 , kept by blast: 305 , if remove dups: 305\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 270\n",
      "--- gene=270, id=54 removed by soft count = 97\n",
      "--- gene=270, id=272 removed by soft count = 251\n",
      "--- number of probes: 292 , kept by blast: 290 , if remove dups: 290\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 271\n",
      "--- gene=271, id=211 removed by soft count = 65\n",
      "--- gene=271, id=220 removed by soft count = 919\n",
      "--- number of probes: 312 , kept by blast: 310 , if remove dups: 310\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 272\n",
      "--- number of probes: 325 , kept by blast: 325 , if remove dups: 325\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 273\n",
      "--- gene=273, id=0 removed by soft count = 94\n",
      "--- gene=273, id=2 removed by soft count = 70\n",
      "--- gene=273, id=201 removed by soft count = 98\n",
      "--- gene=273, id=204 removed by soft count = 401\n",
      "--- number of probes: 300 , kept by blast: 296 , if remove dups: 296\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 274\n",
      "--- gene=274, id=0 removed by soft count = 51\n",
      "--- gene=274, id=9 removed by soft count = 393\n",
      "--- gene=274, id=10 removed by soft count = 2912\n",
      "--- gene=274, id=13 removed by soft count = 59\n",
      "--- gene=274, id=39 removed by soft count = 67\n",
      "--- gene=274, id=43 removed by soft count = 1610\n",
      "--- gene=274, id=114 removed by soft count = 55\n",
      "--- gene=274, id=116 removed by soft count = 4489\n",
      "--- gene=274, id=117 removed by soft count = 226\n",
      "--- gene=274, id=119 removed by soft count = 206\n",
      "--- number of probes: 123 , kept by blast: 113 , if remove dups: 113\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 113\n",
      "-- checking probes in region: 276\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=276, id=29 removed by soft count = 100\n",
      "--- gene=276, id=224 removed by soft count = 65\n",
      "--- number of probes: 248 , kept by blast: 246 , if remove dups: 246\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 277\n",
      "--- gene=277, id=104 removed by soft count = 1959\n",
      "--- gene=277, id=105 removed by soft count = 72\n",
      "--- gene=277, id=205 removed by soft count = 67\n",
      "--- gene=277, id=288 removed by soft count = 55\n",
      "--- number of probes: 319 , kept by blast: 315 , if remove dups: 315\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 278\n",
      "--- gene=278, id=11 removed by soft count = 125\n",
      "--- gene=278, id=12 removed by soft count = 143\n",
      "--- gene=278, id=13 removed by soft count = 90\n",
      "--- number of probes: 156 , kept by blast: 153 , if remove dups: 153\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 279\n",
      "--- number of probes: 255 , kept by blast: 255 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 281\n",
      "--- number of probes: 187 , kept by blast: 187 , if remove dups: 187\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 282\n",
      "--- gene=282, id=14 removed by soft count = 51\n",
      "--- gene=282, id=15 removed by soft count = 249\n",
      "--- gene=282, id=16 removed by soft count = 53\n",
      "--- gene=282, id=45 removed by soft count = 157\n",
      "--- gene=282, id=79 removed by soft count = 138\n",
      "--- gene=282, id=86 removed by soft count = 79\n",
      "--- gene=282, id=133 removed by soft count = 72\n",
      "--- number of probes: 141 , kept by blast: 134 , if remove dups: 134\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 283\n",
      "--- gene=283, id=1 removed by soft count = 207\n",
      "--- gene=283, id=118 removed by soft count = 987\n",
      "--- number of probes: 119 , kept by blast: 117 , if remove dups: 117\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 117\n",
      "-- checking probes in region: 285\n",
      "--- gene=285, id=45 removed by soft count = 1754\n",
      "--- gene=285, id=47 removed by soft count = 172\n",
      "--- gene=285, id=48 removed by soft count = 54\n",
      "--- gene=285, id=61 removed by soft count = 134\n",
      "--- gene=285, id=62 removed by soft count = 93\n",
      "--- gene=285, id=68 removed by soft count = 143\n",
      "--- gene=285, id=71 removed by soft count = 389\n",
      "--- number of probes: 93 , kept by blast: 86 , if remove dups: 86\n",
      "-- number of probes kept for this region: 86\n",
      "-- checking probes in region: 286\n",
      "--- gene=286, id=58 removed by soft count = 61\n",
      "--- gene=286, id=59 removed by soft count = 60\n",
      "--- gene=286, id=83 removed by soft count = 129\n",
      "--- gene=286, id=84 removed by soft count = 1103\n",
      "--- gene=286, id=85 removed by soft count = 458\n",
      "--- gene=286, id=165 removed by soft count = 5544\n",
      "--- number of probes: 188 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 287\n",
      "--- gene=287, id=0 removed by soft count = 65\n",
      "--- gene=287, id=1 removed by soft count = 1007\n",
      "--- gene=287, id=2 removed by soft count = 101\n",
      "--- gene=287, id=3 removed by soft count = 148\n",
      "--- gene=287, id=20 removed by soft count = 140\n",
      "--- gene=287, id=22 removed by soft count = 618\n",
      "--- gene=287, id=23 removed by soft count = 160\n",
      "--- gene=287, id=24 removed by soft count = 1277\n",
      "--- gene=287, id=25 removed by soft count = 590\n",
      "--- gene=287, id=27 removed by soft count = 117\n",
      "--- gene=287, id=32 removed by soft count = 57\n",
      "--- gene=287, id=78 removed by soft count = 98\n",
      "--- gene=287, id=81 removed by soft count = 572\n",
      "--- gene=287, id=82 removed by soft count = 228\n",
      "--- gene=287, id=85 removed by soft count = 196\n",
      "--- gene=287, id=88 removed by soft count = 216\n",
      "--- gene=287, id=96 removed by soft count = 127\n",
      "--- gene=287, id=97 removed by soft count = 259\n",
      "--- gene=287, id=98 removed by soft count = 773\n",
      "--- gene=287, id=99 removed by soft count = 343\n",
      "--- gene=287, id=115 removed by soft count = 94\n",
      "--- number of probes: 134 , kept by blast: 113 , if remove dups: 112\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 112\n",
      "-- checking probes in region: 288\n",
      "--- gene=288, id=11 removed by soft count = 92\n",
      "--- gene=288, id=36 removed by soft count = 132\n",
      "--- gene=288, id=187 removed by soft count = 55\n",
      "--- number of probes: 217 , kept by blast: 214 , if remove dups: 214\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 289\n",
      "--- gene=289, id=36 removed by soft count = 103\n",
      "--- gene=289, id=83 removed by soft count = 67\n",
      "--- gene=289, id=229 removed by soft count = 141\n",
      "--- number of probes: 252 , kept by blast: 249 , if remove dups: 249\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 290\n",
      "--- gene=290, id=28 removed by soft count = 1344\n",
      "--- gene=290, id=91 removed by soft count = 187\n",
      "--- gene=290, id=99 removed by soft count = 72\n",
      "--- number of probes: 103 , kept by blast: 100 , if remove dups: 100\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 100\n",
      "-- checking probes in region: 291\n",
      "--- gene=291, id=47 removed by soft count = 139\n",
      "--- gene=291, id=88 removed by soft count = 361\n",
      "--- gene=291, id=159 removed by soft count = 74\n",
      "--- number of probes: 229 , kept by blast: 226 , if remove dups: 226\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 292\n",
      "--- gene=292, id=296 removed by soft count = 51\n",
      "--- gene=292, id=297 removed by soft count = 68\n",
      "--- number of probes: 316 , kept by blast: 314 , if remove dups: 314\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 293\n",
      "--- gene=293, id=60 removed by soft count = 203\n",
      "--- gene=293, id=189 removed by soft count = 125\n",
      "--- gene=293, id=205 removed by soft count = 145\n",
      "--- number of probes: 252 , kept by blast: 249 , if remove dups: 249\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 294\n",
      "--- gene=294, id=81 removed by soft count = 57\n",
      "--- gene=294, id=170 removed by soft count = 268\n",
      "--- gene=294, id=173 removed by soft count = 94\n",
      "--- number of probes: 218 , kept by blast: 215 , if remove dups: 215\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 295\n",
      "--- gene=295, id=120 removed by soft count = 250\n",
      "--- gene=295, id=152 removed by soft count = 85\n",
      "--- number of probes: 257 , kept by blast: 255 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 296\n",
      "--- gene=296, id=150 removed by soft count = 102\n",
      "--- gene=296, id=152 removed by soft count = 142\n",
      "--- gene=296, id=225 removed by soft count = 6939\n",
      "--- number of probes: 237 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 297\n",
      "--- gene=297, id=18 removed by soft count = 54\n",
      "--- gene=297, id=29 removed by soft count = 81\n",
      "--- gene=297, id=33 removed by soft count = 127\n",
      "--- gene=297, id=135 removed by soft count = 54\n",
      "--- number of probes: 236 , kept by blast: 232 , if remove dups: 232\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 298\n",
      "--- gene=298, id=42 removed by soft count = 2081\n",
      "--- gene=298, id=93 removed by soft count = 57\n",
      "--- gene=298, id=133 removed by soft count = 152\n",
      "--- gene=298, id=137 removed by soft count = 175\n",
      "--- gene=298, id=138 removed by soft count = 124\n",
      "--- number of probes: 151 , kept by blast: 146 , if remove dups: 146\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 299\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=299, id=110 removed by soft count = 58\n",
      "--- gene=299, id=117 removed by soft count = 63\n",
      "--- gene=299, id=118 removed by soft count = 82\n",
      "--- gene=299, id=189 removed by soft count = 273\n",
      "--- gene=299, id=190 removed by soft count = 858\n",
      "--- number of probes: 212 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 300\n",
      "--- gene=300, id=54 removed by soft count = 373\n",
      "--- gene=300, id=56 removed by soft count = 812\n",
      "--- gene=300, id=60 removed by soft count = 79\n",
      "--- gene=300, id=159 removed by soft count = 1442\n",
      "--- gene=300, id=160 removed by soft count = 204\n",
      "--- gene=300, id=161 removed by soft count = 96\n",
      "--- gene=300, id=173 removed by soft count = 67\n",
      "--- number of probes: 174 , kept by blast: 167 , if remove dups: 167\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 301\n",
      "--- gene=301, id=1 removed by soft count = 681\n",
      "--- gene=301, id=2 removed by soft count = 217\n",
      "--- number of probes: 119 , kept by blast: 117 , if remove dups: 117\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 117\n",
      "-- checking probes in region: 302\n",
      "--- gene=302, id=18 removed by soft count = 178\n",
      "--- gene=302, id=46 removed by soft count = 145\n",
      "--- gene=302, id=55 removed by soft count = 118\n",
      "--- gene=302, id=108 removed by soft count = 136\n",
      "--- gene=302, id=110 removed by soft count = 261\n",
      "--- gene=302, id=111 removed by soft count = 100\n",
      "--- number of probes: 115 , kept by blast: 109 , if remove dups: 109\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 109\n",
      "-- checking probes in region: 303\n",
      "--- gene=303, id=34 removed by soft count = 88\n",
      "--- gene=303, id=103 removed by soft count = 171\n",
      "--- number of probes: 184 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 304\n",
      "--- gene=304, id=22 removed by soft count = 110\n",
      "--- gene=304, id=23 removed by soft count = 95\n",
      "--- gene=304, id=70 removed by soft count = 333\n",
      "--- gene=304, id=76 removed by soft count = 298\n",
      "--- gene=304, id=83 removed by soft count = 4665\n",
      "--- number of probes: 130 , kept by blast: 125 , if remove dups: 125\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 305\n",
      "--- gene=305, id=124 removed by soft count = 115\n",
      "--- gene=305, id=138 removed by soft count = 55\n",
      "--- gene=305, id=139 removed by soft count = 51\n",
      "--- gene=305, id=190 removed by soft count = 7012\n",
      "--- number of probes: 222 , kept by blast: 218 , if remove dups: 218\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 306\n",
      "--- gene=306, id=37 removed by soft count = 118\n",
      "--- gene=306, id=80 removed by soft count = 123\n",
      "--- gene=306, id=97 removed by soft count = 75\n",
      "--- gene=306, id=125 removed by soft count = 1202\n",
      "--- gene=306, id=126 removed by soft count = 366\n",
      "--- gene=306, id=156 removed by soft count = 130\n",
      "--- number of probes: 188 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 307\n",
      "--- gene=307, id=53 removed by soft count = 104\n",
      "--- gene=307, id=108 removed by soft count = 273\n",
      "--- gene=307, id=110 removed by soft count = 175\n",
      "--- gene=307, id=111 removed by soft count = 645\n",
      "--- gene=307, id=112 removed by soft count = 458\n",
      "--- gene=307, id=113 removed by soft count = 116\n",
      "--- gene=307, id=114 removed by soft count = 1247\n",
      "--- gene=307, id=146 removed by soft count = 303\n",
      "--- gene=307, id=147 removed by soft count = 226\n",
      "--- number of probes: 149 , kept by blast: 140 , if remove dups: 140\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 308\n",
      "--- gene=308, id=97 removed by soft count = 1825\n",
      "--- number of probes: 108 , kept by blast: 107 , if remove dups: 107\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 107\n",
      "-- checking probes in region: 309\n",
      "--- gene=309, id=250 removed by soft count = 171\n",
      "--- number of probes: 322 , kept by blast: 321 , if remove dups: 321\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 310\n",
      "--- gene=310, id=83 removed by soft count = 794\n",
      "--- gene=310, id=134 removed by soft count = 56\n",
      "--- number of probes: 205 , kept by blast: 203 , if remove dups: 203\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 311\n",
      "--- gene=311, id=188 removed by soft count = 674\n",
      "--- number of probes: 189 , kept by blast: 188 , if remove dups: 188\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 312\n",
      "--- number of probes: 204 , kept by blast: 204 , if remove dups: 204\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 313\n",
      "--- gene=313, id=10 removed by soft count = 60\n",
      "--- gene=313, id=11 removed by soft count = 88\n",
      "--- gene=313, id=15 removed by soft count = 823\n",
      "--- gene=313, id=16 removed by soft count = 14307\n",
      "--- gene=313, id=69 removed by soft count = 128\n",
      "--- gene=313, id=96 removed by soft count = 68\n",
      "--- gene=313, id=100 removed by soft count = 388\n",
      "--- gene=313, id=101 removed by soft count = 295\n",
      "--- gene=313, id=127 removed by soft count = 94\n",
      "--- gene=313, id=133 removed by soft count = 182\n",
      "--- number of probes: 135 , kept by blast: 125 , if remove dups: 125\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 314\n",
      "--- gene=314, id=147 removed by soft count = 67\n",
      "--- number of probes: 210 , kept by blast: 209 , if remove dups: 209\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 315\n",
      "--- gene=315, id=62 removed by soft count = 3198\n",
      "--- number of probes: 183 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 316\n",
      "--- number of probes: 192 , kept by blast: 192 , if remove dups: 192\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 318\n",
      "--- gene=318, id=4 removed by soft count = 77\n",
      "--- number of probes: 239 , kept by blast: 238 , if remove dups: 238\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 319\n",
      "--- gene=319, id=92 removed by soft count = 58\n",
      "--- gene=319, id=94 removed by soft count = 275\n",
      "--- gene=319, id=110 removed by soft count = 78\n",
      "--- number of probes: 205 , kept by blast: 202 , if remove dups: 202\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 320\n",
      "--- gene=320, id=199 removed by soft count = 9634\n",
      "--- number of probes: 200 , kept by blast: 199 , if remove dups: 199\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 321\n",
      "--- gene=321, id=41 removed by soft count = 568\n",
      "--- gene=321, id=98 removed by soft count = 3633\n",
      "--- gene=321, id=145 removed by soft count = 77\n",
      "--- number of probes: 274 , kept by blast: 271 , if remove dups: 271\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 322\n",
      "--- gene=322, id=16 removed by soft count = 51\n",
      "--- gene=322, id=23 removed by soft count = 52\n",
      "--- gene=322, id=88 removed by soft count = 93\n",
      "--- gene=322, id=91 removed by soft count = 55\n",
      "--- gene=322, id=165 removed by soft count = 253\n",
      "--- gene=322, id=166 removed by soft count = 74\n",
      "--- gene=322, id=190 removed by soft count = 96\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- number of probes: 210 , kept by blast: 203 , if remove dups: 203\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 323\n",
      "--- gene=323, id=5 removed by soft count = 69\n",
      "--- gene=323, id=6 removed by soft count = 178\n",
      "--- gene=323, id=15 removed by soft count = 68\n",
      "--- gene=323, id=18 removed by soft count = 112\n",
      "--- gene=323, id=56 removed by soft count = 146\n",
      "--- gene=323, id=127 removed by soft count = 978\n",
      "--- gene=323, id=231 removed by soft count = 99\n",
      "--- number of probes: 242 , kept by blast: 235 , if remove dups: 235\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 324\n",
      "--- gene=324, id=115 removed by hard count = 3\n",
      "--- gene=324, id=143 removed by soft count = 61\n",
      "--- gene=324, id=301 removed by soft count = 89\n",
      "--- gene=324, id=302 removed by soft count = 72\n",
      "--- number of probes: 303 , kept by blast: 299 , if remove dups: 299\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 325\n",
      "--- gene=325, id=227 removed by soft count = 79\n",
      "--- number of probes: 337 , kept by blast: 336 , if remove dups: 336\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 326\n",
      "--- gene=326, id=114 removed by soft count = 5121\n",
      "--- gene=326, id=115 removed by soft count = 880\n",
      "--- gene=326, id=137 removed by soft count = 53\n",
      "--- gene=326, id=167 removed by hard count = 3\n",
      "--- gene=326, id=198 removed by soft count = 71\n",
      "--- gene=326, id=199 removed by soft count = 310\n",
      "--- number of probes: 237 , kept by blast: 231 , if remove dups: 231\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 327\n",
      "--- number of probes: 339 , kept by blast: 339 , if remove dups: 339\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 328\n",
      "--- number of probes: 248 , kept by blast: 248 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 329\n",
      "--- gene=329, id=129 removed by soft count = 79\n",
      "--- gene=329, id=135 removed by soft count = 72\n",
      "--- gene=329, id=137 removed by soft count = 259\n",
      "--- gene=329, id=147 removed by soft count = 329\n",
      "--- gene=329, id=221 removed by soft count = 194\n",
      "--- number of probes: 237 , kept by blast: 232 , if remove dups: 232\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 330\n",
      "--- gene=330, id=17 removed by soft count = 74\n",
      "--- gene=330, id=147 removed by soft count = 425\n",
      "--- gene=330, id=195 removed by soft count = 51\n",
      "--- number of probes: 284 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 331\n",
      "--- number of probes: 341 , kept by blast: 341 , if remove dups: 341\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 332\n",
      "--- gene=332, id=11 removed by soft count = 284\n",
      "--- gene=332, id=58 removed by soft count = 52\n",
      "--- gene=332, id=132 removed by soft count = 632\n",
      "--- gene=332, id=150 removed by soft count = 64\n",
      "--- number of probes: 235 , kept by blast: 231 , if remove dups: 231\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 333\n",
      "--- gene=333, id=2 removed by soft count = 203\n",
      "--- gene=333, id=40 removed by soft count = 365\n",
      "--- gene=333, id=44 removed by soft count = 208\n",
      "--- gene=333, id=53 removed by soft count = 51\n",
      "--- gene=333, id=204 removed by soft count = 76\n",
      "--- gene=333, id=218 removed by soft count = 90\n",
      "--- number of probes: 243 , kept by blast: 237 , if remove dups: 237\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 334\n",
      "--- number of probes: 284 , kept by blast: 284 , if remove dups: 284\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 335\n",
      "--- number of probes: 318 , kept by blast: 318 , if remove dups: 318\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 336\n",
      "--- number of probes: 285 , kept by blast: 285 , if remove dups: 285\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 337\n",
      "--- gene=337, id=5 removed by soft count = 535\n",
      "--- gene=337, id=6 removed by soft count = 152\n",
      "--- gene=337, id=7 removed by soft count = 1692\n",
      "--- gene=337, id=223 removed by soft count = 137\n",
      "--- number of probes: 281 , kept by blast: 277 , if remove dups: 277\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 338\n",
      "--- gene=338, id=181 removed by soft count = 88\n",
      "--- gene=338, id=182 removed by soft count = 98\n",
      "--- gene=338, id=183 removed by soft count = 224\n",
      "--- gene=338, id=184 removed by soft count = 303\n",
      "--- gene=338, id=186 removed by soft count = 1263\n",
      "--- number of probes: 259 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 339\n",
      "--- gene=339, id=111 removed by soft count = 153\n",
      "--- gene=339, id=116 removed by soft count = 70\n",
      "--- gene=339, id=167 removed by soft count = 789\n",
      "--- gene=339, id=177 removed by soft count = 134\n",
      "--- gene=339, id=198 removed by soft count = 148\n",
      "--- number of probes: 288 , kept by blast: 283 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 340\n",
      "--- gene=340, id=159 removed by soft count = 298\n",
      "--- gene=340, id=160 removed by soft count = 93\n",
      "--- number of probes: 249 , kept by blast: 247 , if remove dups: 247\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 342\n",
      "--- number of probes: 258 , kept by blast: 258 , if remove dups: 258\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 343\n",
      "--- gene=343, id=142 removed by soft count = 113\n",
      "--- gene=343, id=143 removed by soft count = 69\n",
      "--- gene=343, id=144 removed by soft count = 425\n",
      "--- number of probes: 219 , kept by blast: 216 , if remove dups: 216\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 344\n",
      "--- gene=344, id=9 removed by soft count = 77\n",
      "--- gene=344, id=102 removed by soft count = 68\n",
      "--- gene=344, id=104 removed by soft count = 55\n",
      "--- gene=344, id=203 removed by soft count = 190\n",
      "--- number of probes: 261 , kept by blast: 257 , if remove dups: 257\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 345\n",
      "--- gene=345, id=110 removed by soft count = 119\n",
      "--- gene=345, id=113 removed by soft count = 275\n",
      "--- gene=345, id=114 removed by soft count = 597\n",
      "--- gene=345, id=116 removed by soft count = 75\n",
      "--- gene=345, id=124 removed by soft count = 131\n",
      "--- number of probes: 150 , kept by blast: 145 , if remove dups: 144\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 346\n",
      "--- gene=346, id=15 removed by soft count = 186\n",
      "--- gene=346, id=16 removed by soft count = 91\n",
      "--- gene=346, id=79 removed by soft count = 354\n",
      "--- gene=346, id=80 removed by soft count = 246\n",
      "--- gene=346, id=81 removed by soft count = 698\n",
      "--- gene=346, id=99 removed by soft count = 112\n",
      "--- gene=346, id=157 removed by soft count = 52\n",
      "--- number of probes: 182 , kept by blast: 175 , if remove dups: 175\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 347\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=347, id=9 removed by soft count = 81\n",
      "--- gene=347, id=10 removed by soft count = 914\n",
      "--- gene=347, id=14 removed by soft count = 631\n",
      "--- gene=347, id=50 removed by soft count = 177\n",
      "--- number of probes: 53 , kept by blast: 49 , if remove dups: 49\n",
      "-- number of probes kept for this region: 49\n",
      "-- checking probes in region: 348\n",
      "--- gene=348, id=0 removed by soft count = 332\n",
      "--- gene=348, id=14 removed by soft count = 55\n",
      "--- gene=348, id=74 removed by soft count = 95\n",
      "--- gene=348, id=75 removed by soft count = 152\n",
      "--- gene=348, id=80 removed by soft count = 62\n",
      "--- gene=348, id=84 removed by soft count = 707\n",
      "--- number of probes: 139 , kept by blast: 133 , if remove dups: 132\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 349\n",
      "--- number of probes: 188 , kept by blast: 188 , if remove dups: 187\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 350\n",
      "--- gene=350, id=13 removed by soft count = 58\n",
      "--- gene=350, id=114 removed by soft count = 60\n",
      "--- gene=350, id=237 removed by soft count = 181\n",
      "--- gene=350, id=238 removed by soft count = 1528\n",
      "--- number of probes: 239 , kept by blast: 235 , if remove dups: 235\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 351\n",
      "--- gene=351, id=54 removed by soft count = 510\n",
      "--- gene=351, id=58 removed by soft count = 53\n",
      "--- gene=351, id=64 removed by soft count = 20222\n",
      "--- number of probes: 69 , kept by blast: 66 , if remove dups: 66\n",
      "-- number of probes kept for this region: 66\n",
      "-- checking probes in region: 352\n",
      "--- number of probes: 318 , kept by blast: 318 , if remove dups: 318\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 353\n",
      "--- gene=353, id=253 removed by soft count = 107\n",
      "--- gene=353, id=256 removed by soft count = 198\n",
      "--- number of probes: 268 , kept by blast: 266 , if remove dups: 264\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 354\n",
      "--- gene=354, id=43 removed by soft count = 243\n",
      "--- gene=354, id=118 removed by soft count = 2134\n",
      "--- gene=354, id=151 removed by soft count = 67\n",
      "--- number of probes: 175 , kept by blast: 172 , if remove dups: 172\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 355\n",
      "--- gene=355, id=128 removed by soft count = 1129\n",
      "--- gene=355, id=170 removed by soft count = 158\n",
      "--- number of probes: 209 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 356\n",
      "--- gene=356, id=13 removed by soft count = 54\n",
      "--- gene=356, id=245 removed by soft count = 104\n",
      "--- number of probes: 315 , kept by blast: 313 , if remove dups: 312\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 357\n",
      "--- gene=357, id=1 removed by soft count = 2322\n",
      "--- gene=357, id=4 removed by soft count = 271\n",
      "--- gene=357, id=19 removed by soft count = 61\n",
      "--- gene=357, id=43 removed by soft count = 1222\n",
      "--- gene=357, id=44 removed by soft count = 1456\n",
      "--- number of probes: 138 , kept by blast: 133 , if remove dups: 133\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 358\n",
      "--- gene=358, id=40 removed by soft count = 128\n",
      "--- gene=358, id=83 removed by soft count = 2326\n",
      "--- gene=358, id=94 removed by soft count = 252\n",
      "--- gene=358, id=121 removed by soft count = 62\n",
      "--- gene=358, id=149 removed by soft count = 127\n",
      "--- number of probes: 158 , kept by blast: 153 , if remove dups: 153\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 359\n",
      "--- gene=359, id=126 removed by soft count = 57\n",
      "--- gene=359, id=144 removed by soft count = 150\n",
      "--- gene=359, id=161 removed by soft count = 58\n",
      "--- gene=359, id=171 removed by soft count = 983\n",
      "--- gene=359, id=172 removed by soft count = 715\n",
      "--- gene=359, id=173 removed by soft count = 187\n",
      "--- gene=359, id=175 removed by soft count = 234\n",
      "--- gene=359, id=188 removed by soft count = 116\n",
      "--- number of probes: 227 , kept by blast: 219 , if remove dups: 219\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 360\n",
      "--- gene=360, id=209 removed by soft count = 66\n",
      "--- gene=360, id=212 removed by soft count = 857\n",
      "--- gene=360, id=214 removed by soft count = 72\n",
      "--- gene=360, id=215 removed by soft count = 381\n",
      "--- gene=360, id=216 removed by soft count = 183\n",
      "--- number of probes: 221 , kept by blast: 216 , if remove dups: 215\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 361\n",
      "--- gene=361, id=25 removed by soft count = 105\n",
      "--- gene=361, id=67 removed by soft count = 524\n",
      "--- gene=361, id=69 removed by soft count = 85\n",
      "--- gene=361, id=70 removed by soft count = 65\n",
      "--- gene=361, id=71 removed by soft count = 606\n",
      "--- gene=361, id=105 removed by soft count = 67\n",
      "--- number of probes: 235 , kept by blast: 229 , if remove dups: 229\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 362\n",
      "--- gene=362, id=89 removed by soft count = 73\n",
      "--- gene=362, id=106 removed by soft count = 360\n",
      "--- number of probes: 260 , kept by blast: 258 , if remove dups: 258\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 363\n",
      "--- number of probes: 116 , kept by blast: 116 , if remove dups: 116\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 116\n",
      "-- checking probes in region: 364\n",
      "--- gene=364, id=29 removed by soft count = 124\n",
      "--- gene=364, id=30 removed by soft count = 158\n",
      "--- gene=364, id=33 removed by soft count = 650\n",
      "--- gene=364, id=39 removed by soft count = 310\n",
      "--- gene=364, id=42 removed by soft count = 156\n",
      "--- gene=364, id=46 removed by soft count = 2244\n",
      "--- gene=364, id=47 removed by soft count = 164\n",
      "--- gene=364, id=48 removed by soft count = 88\n",
      "--- gene=364, id=49 removed by soft count = 118\n",
      "--- gene=364, id=78 removed by soft count = 89\n",
      "--- gene=364, id=84 removed by soft count = 91\n",
      "--- gene=364, id=85 removed by soft count = 121\n",
      "--- gene=364, id=139 removed by soft count = 208\n",
      "--- number of probes: 163 , kept by blast: 150 , if remove dups: 150\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 365\n",
      "--- gene=365, id=130 removed by soft count = 64\n",
      "--- number of probes: 203 , kept by blast: 202 , if remove dups: 202\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 366\n",
      "--- gene=366, id=0 removed by soft count = 350\n",
      "--- gene=366, id=1 removed by soft count = 1184\n",
      "--- gene=366, id=30 removed by soft count = 56\n",
      "--- gene=366, id=31 removed by soft count = 1125\n",
      "--- gene=366, id=62 removed by soft count = 134\n",
      "--- gene=366, id=153 removed by soft count = 54\n",
      "--- gene=366, id=156 removed by soft count = 173\n",
      "--- gene=366, id=190 removed by soft count = 135\n",
      "--- number of probes: 195 , kept by blast: 187 , if remove dups: 187\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 367\n",
      "--- gene=367, id=0 removed by soft count = 190\n",
      "--- gene=367, id=7 removed by soft count = 152\n",
      "--- gene=367, id=8 removed by soft count = 56\n",
      "--- gene=367, id=12 removed by soft count = 73\n",
      "--- gene=367, id=15 removed by soft count = 250\n",
      "--- gene=367, id=16 removed by soft count = 141\n",
      "--- gene=367, id=18 removed by soft count = 143\n",
      "--- gene=367, id=19 removed by soft count = 250\n",
      "--- gene=367, id=20 removed by soft count = 203\n",
      "--- gene=367, id=21 removed by soft count = 180\n",
      "--- gene=367, id=23 removed by soft count = 59\n",
      "--- gene=367, id=25 removed by soft count = 89\n",
      "--- gene=367, id=26 removed by soft count = 91\n",
      "--- gene=367, id=27 removed by soft count = 182\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=367, id=28 removed by soft count = 331\n",
      "--- gene=367, id=29 removed by soft count = 2693\n",
      "--- gene=367, id=31 removed by soft count = 73\n",
      "--- gene=367, id=33 removed by soft count = 572\n",
      "--- gene=367, id=34 removed by soft count = 122\n",
      "--- gene=367, id=35 removed by soft count = 75\n",
      "--- gene=367, id=36 removed by soft count = 88\n",
      "--- gene=367, id=37 removed by soft count = 126\n",
      "--- gene=367, id=38 removed by soft count = 105\n",
      "--- gene=367, id=39 removed by soft count = 161\n",
      "--- gene=367, id=78 removed by soft count = 262\n",
      "--- gene=367, id=95 removed by soft count = 63\n",
      "--- gene=367, id=98 removed by soft count = 92\n",
      "--- gene=367, id=128 removed by soft count = 53\n",
      "--- number of probes: 133 , kept by blast: 105 , if remove dups: 105\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 105\n",
      "-- checking probes in region: 368\n",
      "--- gene=368, id=2 removed by soft count = 64\n",
      "--- gene=368, id=9 removed by soft count = 81\n",
      "--- gene=368, id=12 removed by soft count = 69\n",
      "--- gene=368, id=20 removed by soft count = 120\n",
      "--- gene=368, id=21 removed by soft count = 55\n",
      "--- gene=368, id=22 removed by soft count = 2052\n",
      "--- gene=368, id=23 removed by soft count = 848\n",
      "--- gene=368, id=24 removed by soft count = 30392\n",
      "--- gene=368, id=25 removed by soft count = 7420\n",
      "--- gene=368, id=26 removed by soft count = 86\n",
      "--- gene=368, id=52 removed by soft count = 92\n",
      "--- number of probes: 129 , kept by blast: 118 , if remove dups: 118\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 118\n",
      "-- checking probes in region: 369\n",
      "--- gene=369, id=54 removed by soft count = 75\n",
      "--- gene=369, id=79 removed by soft count = 61\n",
      "--- gene=369, id=114 removed by soft count = 220\n",
      "--- gene=369, id=120 removed by soft count = 460\n",
      "--- gene=369, id=200 removed by soft count = 129\n",
      "--- number of probes: 226 , kept by blast: 221 , if remove dups: 221\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 370\n",
      "--- gene=370, id=107 removed by soft count = 80\n",
      "--- number of probes: 175 , kept by blast: 174 , if remove dups: 174\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 371\n",
      "--- gene=371, id=21 removed by soft count = 65\n",
      "--- gene=371, id=22 removed by soft count = 135\n",
      "--- gene=371, id=23 removed by soft count = 940\n",
      "--- gene=371, id=24 removed by soft count = 92\n",
      "--- gene=371, id=25 removed by soft count = 156\n",
      "--- gene=371, id=32 removed by soft count = 77\n",
      "--- gene=371, id=95 removed by soft count = 2265\n",
      "--- gene=371, id=101 removed by soft count = 89\n",
      "--- gene=371, id=108 removed by soft count = 12793\n",
      "--- gene=371, id=110 removed by soft count = 1975\n",
      "--- gene=371, id=111 removed by soft count = 408\n",
      "--- gene=371, id=114 removed by soft count = 217\n",
      "--- gene=371, id=117 removed by soft count = 91\n",
      "--- gene=371, id=119 removed by soft count = 251\n",
      "--- gene=371, id=120 removed by soft count = 178\n",
      "--- number of probes: 123 , kept by blast: 108 , if remove dups: 108\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 108\n",
      "-- checking probes in region: 372\n",
      "--- gene=372, id=17 removed by soft count = 80\n",
      "--- gene=372, id=43 removed by soft count = 70\n",
      "--- gene=372, id=73 removed by soft count = 130\n",
      "--- gene=372, id=74 removed by soft count = 90\n",
      "--- gene=372, id=76 removed by soft count = 65\n",
      "--- gene=372, id=77 removed by soft count = 173\n",
      "--- gene=372, id=187 removed by soft count = 155\n",
      "--- number of probes: 210 , kept by blast: 203 , if remove dups: 203\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 373\n",
      "--- gene=373, id=28 removed by soft count = 51\n",
      "--- gene=373, id=30 removed by soft count = 146\n",
      "--- gene=373, id=32 removed by soft count = 397\n",
      "--- gene=373, id=45 removed by soft count = 54\n",
      "--- gene=373, id=46 removed by soft count = 64\n",
      "--- gene=373, id=47 removed by soft count = 78\n",
      "--- gene=373, id=48 removed by soft count = 631\n",
      "--- gene=373, id=50 removed by soft count = 52\n",
      "--- gene=373, id=66 removed by soft count = 56\n",
      "--- gene=373, id=104 removed by soft count = 64\n",
      "--- gene=373, id=127 removed by soft count = 91\n",
      "--- gene=373, id=168 removed by soft count = 190\n",
      "--- gene=373, id=169 removed by soft count = 57\n",
      "--- gene=373, id=232 removed by soft count = 75\n",
      "--- gene=373, id=233 removed by soft count = 150\n",
      "--- gene=373, id=234 removed by soft count = 104\n",
      "--- number of probes: 235 , kept by blast: 219 , if remove dups: 219\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 374\n",
      "--- gene=374, id=300 removed by soft count = 57\n",
      "--- number of probes: 305 , kept by blast: 304 , if remove dups: 304\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 375\n",
      "--- gene=375, id=42 removed by soft count = 239\n",
      "--- number of probes: 285 , kept by blast: 284 , if remove dups: 284\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 376\n",
      "--- gene=376, id=28 removed by soft count = 55\n",
      "--- gene=376, id=109 removed by soft count = 80\n",
      "--- gene=376, id=195 removed by soft count = 51\n",
      "--- gene=376, id=256 removed by soft count = 79\n",
      "--- gene=376, id=306 removed by soft count = 265\n",
      "--- number of probes: 315 , kept by blast: 310 , if remove dups: 310\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 377\n",
      "--- gene=377, id=56 removed by soft count = 135\n",
      "--- number of probes: 326 , kept by blast: 325 , if remove dups: 325\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 378\n",
      "--- gene=378, id=241 removed by soft count = 203\n",
      "--- number of probes: 270 , kept by blast: 269 , if remove dups: 269\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 379\n",
      "--- gene=379, id=46 removed by soft count = 73\n",
      "--- gene=379, id=74 removed by soft count = 2403\n",
      "--- gene=379, id=75 removed by soft count = 308\n",
      "--- number of probes: 305 , kept by blast: 302 , if remove dups: 302\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 380\n",
      "--- gene=380, id=233 removed by soft count = 120\n",
      "--- gene=380, id=234 removed by soft count = 89\n",
      "--- gene=380, id=235 removed by soft count = 308\n",
      "--- number of probes: 272 , kept by blast: 269 , if remove dups: 269\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 381\n",
      "--- gene=381, id=84 removed by soft count = 72\n",
      "--- number of probes: 327 , kept by blast: 326 , if remove dups: 326\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 382\n",
      "--- gene=382, id=157 removed by soft count = 742\n",
      "--- number of probes: 242 , kept by blast: 241 , if remove dups: 241\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 383\n",
      "--- gene=383, id=92 removed by soft count = 2924\n",
      "--- gene=383, id=119 removed by soft count = 265\n",
      "--- gene=383, id=189 removed by soft count = 10686\n",
      "--- number of probes: 233 , kept by blast: 230 , if remove dups: 230\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 384\n",
      "--- gene=384, id=34 removed by soft count = 105\n",
      "--- gene=384, id=82 removed by soft count = 156\n",
      "--- gene=384, id=83 removed by soft count = 62\n",
      "--- gene=384, id=85 removed by soft count = 71\n",
      "--- gene=384, id=147 removed by soft count = 73\n",
      "--- gene=384, id=168 removed by soft count = 282\n",
      "--- number of probes: 212 , kept by blast: 206 , if remove dups: 206\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 385\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=385, id=24 removed by soft count = 96\n",
      "--- number of probes: 230 , kept by blast: 229 , if remove dups: 229\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 386\n",
      "--- gene=386, id=46 removed by soft count = 138\n",
      "--- number of probes: 329 , kept by blast: 328 , if remove dups: 328\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 387\n",
      "--- gene=387, id=6 removed by soft count = 61\n",
      "--- gene=387, id=137 removed by soft count = 52\n",
      "--- gene=387, id=257 removed by hard count = 0\n",
      "--- number of probes: 259 , kept by blast: 256 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 388\n",
      "--- gene=388, id=319 removed by soft count = 277\n",
      "--- number of probes: 323 , kept by blast: 322 , if remove dups: 322\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 389\n",
      "--- gene=389, id=264 removed by soft count = 76\n",
      "--- number of probes: 266 , kept by blast: 265 , if remove dups: 265\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 390\n",
      "--- gene=390, id=73 removed by soft count = 65\n",
      "--- gene=390, id=99 removed by soft count = 63\n",
      "--- gene=390, id=102 removed by soft count = 92\n",
      "--- number of probes: 103 , kept by blast: 100 , if remove dups: 100\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 100\n",
      "-- checking probes in region: 391\n",
      "--- gene=391, id=103 removed by soft count = 77\n",
      "--- gene=391, id=105 removed by soft count = 189\n",
      "--- gene=391, id=106 removed by soft count = 124\n",
      "--- gene=391, id=127 removed by soft count = 311\n",
      "--- gene=391, id=128 removed by soft count = 1281\n",
      "--- gene=391, id=130 removed by soft count = 63\n",
      "--- gene=391, id=132 removed by soft count = 80\n",
      "--- gene=391, id=138 removed by soft count = 870\n",
      "--- gene=391, id=139 removed by soft count = 210\n",
      "--- gene=391, id=141 removed by soft count = 222\n",
      "--- gene=391, id=144 removed by soft count = 352\n",
      "--- gene=391, id=145 removed by soft count = 2342\n",
      "--- gene=391, id=147 removed by soft count = 541\n",
      "--- number of probes: 153 , kept by blast: 140 , if remove dups: 140\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 392\n",
      "--- gene=392, id=3 removed by soft count = 54\n",
      "--- gene=392, id=7 removed by soft count = 64\n",
      "--- gene=392, id=9 removed by soft count = 62\n",
      "--- gene=392, id=10 removed by soft count = 98\n",
      "--- gene=392, id=12 removed by soft count = 82\n",
      "--- gene=392, id=13 removed by soft count = 100\n",
      "--- gene=392, id=14 removed by soft count = 64\n",
      "--- gene=392, id=18 removed by soft count = 51\n",
      "--- gene=392, id=21 removed by soft count = 126\n",
      "--- gene=392, id=23 removed by soft count = 66\n",
      "--- gene=392, id=24 removed by soft count = 83\n",
      "--- gene=392, id=25 removed by soft count = 90\n",
      "--- gene=392, id=26 removed by soft count = 57\n",
      "--- gene=392, id=29 removed by soft count = 84\n",
      "--- gene=392, id=30 removed by soft count = 119\n",
      "--- gene=392, id=33 removed by soft count = 177\n",
      "--- gene=392, id=34 removed by soft count = 66\n",
      "--- gene=392, id=46 removed by soft count = 139\n",
      "--- gene=392, id=56 removed by soft count = 381\n",
      "--- gene=392, id=66 removed by soft count = 72\n",
      "--- gene=392, id=67 removed by soft count = 119\n",
      "--- gene=392, id=100 removed by soft count = 62\n",
      "--- gene=392, id=102 removed by soft count = 70\n",
      "--- gene=392, id=111 removed by soft count = 622\n",
      "--- gene=392, id=112 removed by soft count = 359\n",
      "--- number of probes: 114 , kept by blast: 89 , if remove dups: 88\n",
      "-- number of probes kept for this region: 89\n",
      "-- checking probes in region: 393\n",
      "--- gene=393, id=67 removed by soft count = 163\n",
      "--- gene=393, id=70 removed by soft count = 312\n",
      "--- number of probes: 93 , kept by blast: 91 , if remove dups: 91\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 91\n",
      "-- checking probes in region: 395\n",
      "--- number of probes: 260 , kept by blast: 260 , if remove dups: 260\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 396\n",
      "--- gene=396, id=124 removed by soft count = 70\n",
      "--- number of probes: 127 , kept by blast: 126 , if remove dups: 126\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 397\n",
      "--- gene=397, id=119 removed by hard count = 0\n",
      "--- number of probes: 268 , kept by blast: 267 , if remove dups: 267\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 398\n",
      "--- gene=398, id=11 removed by soft count = 69\n",
      "--- gene=398, id=18 removed by soft count = 175\n",
      "--- gene=398, id=37 removed by soft count = 70\n",
      "--- gene=398, id=38 removed by soft count = 306\n",
      "--- number of probes: 274 , kept by blast: 270 , if remove dups: 270\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 399\n",
      "--- number of probes: 246 , kept by blast: 246 , if remove dups: 246\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 400\n",
      "--- gene=400, id=162 removed by soft count = 82\n",
      "--- gene=400, id=165 removed by soft count = 87\n",
      "--- gene=400, id=166 removed by soft count = 90\n",
      "--- gene=400, id=167 removed by soft count = 72\n",
      "--- gene=400, id=216 removed by soft count = 54\n",
      "--- gene=400, id=239 removed by soft count = 108\n",
      "--- gene=400, id=241 removed by soft count = 361\n",
      "--- number of probes: 245 , kept by blast: 238 , if remove dups: 238\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 403\n",
      "--- gene=403, id=17 removed by soft count = 653\n",
      "--- gene=403, id=18 removed by soft count = 532\n",
      "--- gene=403, id=21 removed by soft count = 152\n",
      "--- gene=403, id=77 removed by soft count = 193\n",
      "--- gene=403, id=79 removed by soft count = 71\n",
      "--- gene=403, id=109 removed by soft count = 9340\n",
      "--- gene=403, id=111 removed by soft count = 1470\n",
      "--- gene=403, id=132 removed by soft count = 585\n",
      "--- number of probes: 172 , kept by blast: 164 , if remove dups: 164\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 407\n",
      "--- gene=407, id=45 removed by soft count = 99\n",
      "--- gene=407, id=61 removed by soft count = 82\n",
      "--- gene=407, id=62 removed by soft count = 130\n",
      "--- gene=407, id=127 removed by soft count = 471\n",
      "--- number of probes: 211 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 408\n",
      "--- gene=408, id=24 removed by soft count = 185\n",
      "--- gene=408, id=25 removed by soft count = 159\n",
      "--- gene=408, id=32 removed by soft count = 174\n",
      "--- gene=408, id=33 removed by soft count = 129\n",
      "--- gene=408, id=34 removed by soft count = 83\n",
      "--- gene=408, id=56 removed by soft count = 309\n",
      "--- gene=408, id=57 removed by soft count = 159\n",
      "--- gene=408, id=58 removed by soft count = 87\n",
      "--- gene=408, id=61 removed by soft count = 304\n",
      "--- gene=408, id=62 removed by soft count = 104\n",
      "--- gene=408, id=63 removed by soft count = 128\n",
      "--- number of probes: 76 , kept by blast: 65 , if remove dups: 65\n",
      "-- number of probes kept for this region: 65\n",
      "-- checking probes in region: 411\n",
      "--- gene=411, id=0 removed by soft count = 541\n",
      "--- gene=411, id=1 removed by soft count = 31963\n",
      "--- number of probes: 126 , kept by blast: 124 , if remove dups: 123\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 412\n",
      "--- gene=412, id=149 removed by hard count = 6\n",
      "--- gene=412, id=166 removed by soft count = 2794\n",
      "--- number of probes: 261 , kept by blast: 259 , if remove dups: 259\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 413\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=413, id=0 removed by soft count = 268\n",
      "--- gene=413, id=2 removed by soft count = 114\n",
      "--- gene=413, id=9 removed by soft count = 173\n",
      "--- gene=413, id=17 removed by soft count = 583\n",
      "--- gene=413, id=18 removed by soft count = 330\n",
      "--- gene=413, id=66 removed by soft count = 58\n",
      "--- gene=413, id=71 removed by soft count = 53\n",
      "--- number of probes: 98 , kept by blast: 91 , if remove dups: 91\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 91\n",
      "-- checking probes in region: 414\n",
      "--- gene=414, id=79 removed by soft count = 16270\n",
      "--- gene=414, id=80 removed by soft count = 78\n",
      "--- number of probes: 96 , kept by blast: 94 , if remove dups: 94\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 94\n",
      "-- checking probes in region: 415\n",
      "--- gene=415, id=5 removed by soft count = 342\n",
      "--- gene=415, id=76 removed by soft count = 114\n",
      "--- number of probes: 250 , kept by blast: 248 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 418\n",
      "--- gene=418, id=7 removed by soft count = 53\n",
      "--- gene=418, id=8 removed by soft count = 54\n",
      "--- gene=418, id=39 removed by soft count = 64\n",
      "--- gene=418, id=81 removed by soft count = 224\n",
      "--- gene=418, id=85 removed by soft count = 191\n",
      "--- gene=418, id=86 removed by soft count = 380\n",
      "--- gene=418, id=88 removed by soft count = 136\n",
      "--- gene=418, id=90 removed by soft count = 181\n",
      "--- gene=418, id=93 removed by soft count = 265\n",
      "--- gene=418, id=95 removed by soft count = 298\n",
      "--- gene=418, id=96 removed by soft count = 234\n",
      "--- gene=418, id=113 removed by soft count = 73\n",
      "--- gene=418, id=119 removed by soft count = 103\n",
      "--- gene=418, id=132 removed by soft count = 52\n",
      "--- number of probes: 133 , kept by blast: 119 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 419\n",
      "--- gene=419, id=253 removed by soft count = 125\n",
      "--- gene=419, id=256 removed by soft count = 864\n",
      "--- number of probes: 309 , kept by blast: 307 , if remove dups: 307\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 420\n",
      "--- gene=420, id=0 removed by soft count = 139\n",
      "--- gene=420, id=1 removed by soft count = 9996\n",
      "--- gene=420, id=3 removed by soft count = 245\n",
      "--- gene=420, id=6 removed by soft count = 592\n",
      "--- gene=420, id=50 removed by soft count = 615\n",
      "--- gene=420, id=52 removed by soft count = 383\n",
      "--- gene=420, id=54 removed by soft count = 244\n",
      "--- number of probes: 169 , kept by blast: 162 , if remove dups: 162\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 421\n",
      "--- gene=421, id=10 removed by soft count = 315\n",
      "--- gene=421, id=103 removed by soft count = 114\n",
      "--- number of probes: 232 , kept by blast: 230 , if remove dups: 230\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 422\n",
      "--- gene=422, id=127 removed by soft count = 118\n",
      "--- gene=422, id=128 removed by soft count = 96\n",
      "--- gene=422, id=219 removed by soft count = 136\n",
      "--- number of probes: 221 , kept by blast: 218 , if remove dups: 218\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 423\n",
      "--- gene=423, id=65 removed by soft count = 59\n",
      "--- gene=423, id=66 removed by soft count = 75\n",
      "--- gene=423, id=109 removed by soft count = 156\n",
      "--- gene=423, id=110 removed by soft count = 256\n",
      "--- gene=423, id=113 removed by soft count = 51\n",
      "--- gene=423, id=114 removed by soft count = 171\n",
      "--- gene=423, id=122 removed by soft count = 3260\n",
      "--- gene=423, id=124 removed by soft count = 63\n",
      "--- gene=423, id=127 removed by soft count = 136\n",
      "--- gene=423, id=128 removed by soft count = 98\n",
      "--- gene=423, id=176 removed by soft count = 254\n",
      "--- gene=423, id=225 removed by soft count = 67\n",
      "--- gene=423, id=229 removed by soft count = 51\n",
      "--- number of probes: 235 , kept by blast: 222 , if remove dups: 222\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 424\n",
      "--- gene=424, id=124 removed by soft count = 207\n",
      "--- gene=424, id=128 removed by soft count = 80\n",
      "--- gene=424, id=139 removed by soft count = 189\n",
      "--- number of probes: 294 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 425\n",
      "--- gene=425, id=0 removed by soft count = 215\n",
      "--- gene=425, id=1 removed by soft count = 7283\n",
      "--- gene=425, id=3 removed by soft count = 62\n",
      "--- gene=425, id=6 removed by soft count = 74\n",
      "--- gene=425, id=9 removed by soft count = 2771\n",
      "--- gene=425, id=51 removed by soft count = 272\n",
      "--- gene=425, id=54 removed by soft count = 77\n",
      "--- gene=425, id=57 removed by soft count = 97\n",
      "--- gene=425, id=64 removed by soft count = 445\n",
      "--- gene=425, id=65 removed by soft count = 293\n",
      "--- gene=425, id=67 removed by soft count = 74\n",
      "--- gene=425, id=68 removed by soft count = 2898\n",
      "--- gene=425, id=69 removed by soft count = 262\n",
      "--- gene=425, id=70 removed by soft count = 845\n",
      "--- gene=425, id=71 removed by soft count = 736\n",
      "--- gene=425, id=74 removed by soft count = 87\n",
      "--- number of probes: 75 , kept by blast: 59 , if remove dups: 59\n",
      "-- number of probes kept for this region: 59\n",
      "-- checking probes in region: 426\n",
      "--- gene=426, id=43 removed by soft count = 237\n",
      "--- gene=426, id=44 removed by soft count = 84\n",
      "--- gene=426, id=62 removed by soft count = 276\n",
      "--- gene=426, id=70 removed by soft count = 2865\n",
      "--- number of probes: 318 , kept by blast: 314 , if remove dups: 314\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 427\n",
      "--- gene=427, id=2 removed by soft count = 88\n",
      "--- gene=427, id=3 removed by soft count = 66\n",
      "--- gene=427, id=6 removed by soft count = 52\n",
      "--- gene=427, id=164 removed by soft count = 79\n",
      "--- number of probes: 293 , kept by blast: 289 , if remove dups: 289\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 428\n",
      "--- gene=428, id=198 removed by soft count = 145\n",
      "--- gene=428, id=200 removed by soft count = 107\n",
      "--- gene=428, id=201 removed by soft count = 64\n",
      "--- gene=428, id=203 removed by soft count = 56\n",
      "--- gene=428, id=204 removed by soft count = 86\n",
      "--- gene=428, id=212 removed by soft count = 986\n",
      "--- gene=428, id=244 removed by soft count = 64\n",
      "--- gene=428, id=245 removed by soft count = 137\n",
      "--- number of probes: 268 , kept by blast: 260 , if remove dups: 260\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 429\n",
      "--- gene=429, id=43 removed by soft count = 136\n",
      "--- gene=429, id=44 removed by soft count = 170\n",
      "--- gene=429, id=63 removed by soft count = 61\n",
      "--- number of probes: 246 , kept by blast: 243 , if remove dups: 243\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 430\n",
      "--- gene=430, id=195 removed by soft count = 70\n",
      "--- gene=430, id=242 removed by soft count = 128\n",
      "--- gene=430, id=257 removed by soft count = 68\n",
      "--- number of probes: 298 , kept by blast: 295 , if remove dups: 295\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 431\n",
      "--- gene=431, id=4 removed by soft count = 351\n",
      "--- gene=431, id=5 removed by soft count = 550\n",
      "--- gene=431, id=6 removed by soft count = 1840\n",
      "--- gene=431, id=7 removed by soft count = 810\n",
      "--- gene=431, id=9 removed by soft count = 106\n",
      "--- number of probes: 80 , kept by blast: 75 , if remove dups: 75\n",
      "-- number of probes kept for this region: 75\n",
      "-- checking probes in region: 432\n",
      "--- gene=432, id=112 removed by soft count = 989\n",
      "--- gene=432, id=114 removed by soft count = 68\n",
      "--- number of probes: 278 , kept by blast: 276 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 433\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=433, id=100 removed by soft count = 315\n",
      "--- gene=433, id=101 removed by soft count = 237\n",
      "--- gene=433, id=103 removed by soft count = 475\n",
      "--- gene=433, id=182 removed by soft count = 51\n",
      "--- gene=433, id=184 removed by soft count = 99\n",
      "--- gene=433, id=185 removed by soft count = 84\n",
      "--- number of probes: 219 , kept by blast: 213 , if remove dups: 213\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 434\n",
      "--- gene=434, id=119 removed by soft count = 57\n",
      "--- gene=434, id=205 removed by soft count = 84\n",
      "--- gene=434, id=242 removed by soft count = 89\n",
      "--- number of probes: 243 , kept by blast: 240 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 435\n",
      "--- gene=435, id=115 removed by soft count = 432\n",
      "--- gene=435, id=118 removed by soft count = 61\n",
      "--- gene=435, id=119 removed by soft count = 581\n",
      "--- gene=435, id=120 removed by soft count = 222\n",
      "--- gene=435, id=122 removed by soft count = 310\n",
      "--- gene=435, id=126 removed by soft count = 101\n",
      "--- gene=435, id=131 removed by soft count = 166\n",
      "--- gene=435, id=132 removed by soft count = 183\n",
      "--- gene=435, id=133 removed by soft count = 120\n",
      "--- gene=435, id=136 removed by soft count = 85\n",
      "--- number of probes: 139 , kept by blast: 129 , if remove dups: 129\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 436\n",
      "--- gene=436, id=49 removed by soft count = 81\n",
      "--- gene=436, id=51 removed by soft count = 485\n",
      "--- gene=436, id=59 removed by soft count = 74\n",
      "--- gene=436, id=63 removed by soft count = 127\n",
      "--- gene=436, id=64 removed by soft count = 155\n",
      "--- gene=436, id=67 removed by soft count = 67\n",
      "--- gene=436, id=71 removed by soft count = 135\n",
      "--- gene=436, id=77 removed by soft count = 118\n",
      "--- gene=436, id=78 removed by soft count = 55\n",
      "--- number of probes: 167 , kept by blast: 158 , if remove dups: 158\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 437\n",
      "--- gene=437, id=139 removed by soft count = 145\n",
      "--- number of probes: 212 , kept by blast: 211 , if remove dups: 211\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 438\n",
      "--- gene=438, id=18 removed by soft count = 740\n",
      "--- gene=438, id=19 removed by soft count = 67\n",
      "--- gene=438, id=20 removed by soft count = 206\n",
      "--- gene=438, id=29 removed by soft count = 192\n",
      "--- number of probes: 70 , kept by blast: 66 , if remove dups: 66\n",
      "-- number of probes kept for this region: 66\n",
      "-- checking probes in region: 439\n",
      "--- gene=439, id=57 removed by soft count = 61\n",
      "--- gene=439, id=69 removed by soft count = 112\n",
      "--- gene=439, id=91 removed by soft count = 75\n",
      "--- number of probes: 315 , kept by blast: 312 , if remove dups: 312\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 440\n",
      "--- gene=440, id=29 removed by soft count = 180\n",
      "--- gene=440, id=140 removed by soft count = 67\n",
      "--- gene=440, id=249 removed by soft count = 74\n",
      "--- gene=440, id=252 removed by soft count = 131\n",
      "--- number of probes: 325 , kept by blast: 321 , if remove dups: 320\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 441\n",
      "--- gene=441, id=49 removed by soft count = 68\n",
      "--- gene=441, id=61 removed by soft count = 59\n",
      "--- gene=441, id=80 removed by soft count = 56\n",
      "--- gene=441, id=82 removed by soft count = 109\n",
      "--- gene=441, id=85 removed by soft count = 58\n",
      "--- gene=441, id=149 removed by soft count = 183\n",
      "--- number of probes: 163 , kept by blast: 157 , if remove dups: 157\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 442\n",
      "--- gene=442, id=82 removed by soft count = 178\n",
      "--- gene=442, id=83 removed by soft count = 189\n",
      "--- gene=442, id=85 removed by soft count = 91\n",
      "--- number of probes: 174 , kept by blast: 171 , if remove dups: 171\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 443\n",
      "--- gene=443, id=4 removed by soft count = 409\n",
      "--- gene=443, id=18 removed by soft count = 2313\n",
      "--- gene=443, id=28 removed by soft count = 186\n",
      "--- gene=443, id=42 removed by soft count = 224\n",
      "--- gene=443, id=46 removed by soft count = 589\n",
      "--- gene=443, id=51 removed by soft count = 85\n",
      "--- gene=443, id=52 removed by soft count = 96\n",
      "--- gene=443, id=70 removed by soft count = 2951\n",
      "--- number of probes: 98 , kept by blast: 90 , if remove dups: 90\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 90\n",
      "-- checking probes in region: 444\n",
      "--- gene=444, id=8 removed by soft count = 115\n",
      "--- gene=444, id=10 removed by soft count = 109\n",
      "--- gene=444, id=11 removed by soft count = 134\n",
      "--- gene=444, id=13 removed by soft count = 80\n",
      "--- gene=444, id=14 removed by soft count = 264\n",
      "--- gene=444, id=15 removed by soft count = 782\n",
      "--- gene=444, id=16 removed by soft count = 126\n",
      "--- gene=444, id=18 removed by soft count = 1740\n",
      "--- gene=444, id=22 removed by soft count = 246\n",
      "--- gene=444, id=23 removed by soft count = 75\n",
      "--- gene=444, id=24 removed by soft count = 58\n",
      "--- gene=444, id=28 removed by soft count = 122\n",
      "--- gene=444, id=30 removed by soft count = 60\n",
      "--- gene=444, id=31 removed by soft count = 69\n",
      "--- gene=444, id=32 removed by soft count = 177\n",
      "--- gene=444, id=37 removed by soft count = 74\n",
      "--- gene=444, id=103 removed by soft count = 76\n",
      "--- number of probes: 159 , kept by blast: 142 , if remove dups: 142\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 445\n",
      "--- gene=445, id=2 removed by soft count = 69\n",
      "--- gene=445, id=3 removed by soft count = 119\n",
      "--- gene=445, id=23 removed by soft count = 254\n",
      "--- gene=445, id=54 removed by soft count = 75\n",
      "--- gene=445, id=61 removed by soft count = 54\n",
      "--- gene=445, id=146 removed by soft count = 86\n",
      "--- gene=445, id=147 removed by soft count = 289\n",
      "--- gene=445, id=228 removed by soft count = 1707\n",
      "--- gene=445, id=229 removed by soft count = 128\n",
      "--- number of probes: 242 , kept by blast: 233 , if remove dups: 233\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 446\n",
      "--- gene=446, id=231 removed by soft count = 147\n",
      "--- number of probes: 304 , kept by blast: 303 , if remove dups: 303\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 447\n",
      "--- gene=447, id=34 removed by soft count = 62\n",
      "--- gene=447, id=35 removed by soft count = 80\n",
      "--- gene=447, id=150 removed by soft count = 205\n",
      "--- gene=447, id=167 removed by soft count = 52\n",
      "--- gene=447, id=172 removed by soft count = 57\n",
      "--- gene=447, id=173 removed by soft count = 166\n",
      "--- gene=447, id=189 removed by soft count = 134\n",
      "--- gene=447, id=245 removed by soft count = 54\n",
      "--- number of probes: 268 , kept by blast: 260 , if remove dups: 260\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 448\n",
      "--- gene=448, id=7 removed by soft count = 625\n",
      "--- gene=448, id=48 removed by soft count = 338\n",
      "--- gene=448, id=58 removed by soft count = 105\n",
      "--- gene=448, id=83 removed by soft count = 91\n",
      "--- gene=448, id=84 removed by soft count = 56\n",
      "--- gene=448, id=106 removed by soft count = 209\n",
      "--- gene=448, id=158 removed by soft count = 60\n",
      "--- gene=448, id=256 removed by soft count = 89\n",
      "--- number of probes: 285 , kept by blast: 277 , if remove dups: 277\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 449\n",
      "--- gene=449, id=26 removed by soft count = 190\n",
      "--- gene=449, id=31 removed by soft count = 514\n",
      "--- gene=449, id=33 removed by soft count = 244\n",
      "--- gene=449, id=36 removed by soft count = 309\n",
      "--- gene=449, id=37 removed by soft count = 56\n",
      "--- gene=449, id=39 removed by soft count = 78\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=449, id=40 removed by soft count = 159\n",
      "--- gene=449, id=41 removed by soft count = 351\n",
      "--- gene=449, id=44 removed by soft count = 1818\n",
      "--- number of probes: 272 , kept by blast: 263 , if remove dups: 263\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 450\n",
      "--- gene=450, id=42 removed by soft count = 453\n",
      "--- gene=450, id=97 removed by soft count = 52\n",
      "--- number of probes: 231 , kept by blast: 229 , if remove dups: 229\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 451\n",
      "--- number of probes: 240 , kept by blast: 240 , if remove dups: 240\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 452\n",
      "--- gene=452, id=202 removed by soft count = 2055\n",
      "--- gene=452, id=216 removed by soft count = 310\n",
      "--- number of probes: 230 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 453\n",
      "--- gene=453, id=49 removed by soft count = 107\n",
      "--- gene=453, id=220 removed by soft count = 63\n",
      "--- number of probes: 254 , kept by blast: 252 , if remove dups: 252\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 454\n",
      "--- gene=454, id=87 removed by soft count = 130\n",
      "--- gene=454, id=161 removed by soft count = 81\n",
      "--- gene=454, id=196 removed by soft count = 105\n",
      "--- gene=454, id=197 removed by soft count = 71\n",
      "--- gene=454, id=198 removed by soft count = 155\n",
      "--- gene=454, id=233 removed by soft count = 116\n",
      "--- gene=454, id=234 removed by soft count = 145\n",
      "--- number of probes: 266 , kept by blast: 259 , if remove dups: 259\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 455\n",
      "--- gene=455, id=0 removed by soft count = 287\n",
      "--- gene=455, id=18 removed by soft count = 53\n",
      "--- gene=455, id=19 removed by soft count = 386\n",
      "--- gene=455, id=165 removed by soft count = 174\n",
      "--- gene=455, id=168 removed by soft count = 139\n",
      "--- gene=455, id=169 removed by soft count = 157\n",
      "--- gene=455, id=170 removed by soft count = 63\n",
      "--- gene=455, id=180 removed by soft count = 260\n",
      "--- number of probes: 190 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 456\n",
      "--- number of probes: 314 , kept by blast: 314 , if remove dups: 314\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 457\n",
      "--- gene=457, id=58 removed by soft count = 96\n",
      "--- gene=457, id=146 removed by soft count = 80\n",
      "--- gene=457, id=153 removed by soft count = 69\n",
      "--- number of probes: 242 , kept by blast: 239 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 458\n",
      "--- number of probes: 292 , kept by blast: 292 , if remove dups: 292\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 461\n",
      "--- gene=461, id=8 removed by soft count = 59\n",
      "--- gene=461, id=54 removed by soft count = 53\n",
      "--- gene=461, id=61 removed by soft count = 151\n",
      "--- gene=461, id=85 removed by soft count = 56\n",
      "--- gene=461, id=142 removed by soft count = 131\n",
      "--- gene=461, id=144 removed by soft count = 337\n",
      "--- gene=461, id=145 removed by soft count = 193\n",
      "--- number of probes: 179 , kept by blast: 172 , if remove dups: 172\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 462\n",
      "--- gene=462, id=27 removed by soft count = 4795\n",
      "--- gene=462, id=28 removed by soft count = 3637\n",
      "--- gene=462, id=29 removed by soft count = 53\n",
      "--- gene=462, id=30 removed by soft count = 145\n",
      "--- gene=462, id=32 removed by soft count = 112\n",
      "--- gene=462, id=81 removed by soft count = 150\n",
      "--- gene=462, id=82 removed by soft count = 170\n",
      "--- gene=462, id=83 removed by soft count = 1133\n",
      "--- gene=462, id=123 removed by soft count = 397\n",
      "--- number of probes: 132 , kept by blast: 123 , if remove dups: 123\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 463\n",
      "--- gene=463, id=29 removed by soft count = 116\n",
      "--- gene=463, id=30 removed by soft count = 208\n",
      "--- gene=463, id=31 removed by soft count = 169\n",
      "--- gene=463, id=140 removed by soft count = 88\n",
      "--- gene=463, id=141 removed by soft count = 220\n",
      "--- number of probes: 257 , kept by blast: 252 , if remove dups: 251\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 464\n",
      "--- gene=464, id=48 removed by soft count = 57\n",
      "--- number of probes: 321 , kept by blast: 320 , if remove dups: 320\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 465\n",
      "--- gene=465, id=62 removed by soft count = 138\n",
      "--- gene=465, id=63 removed by soft count = 59\n",
      "--- gene=465, id=78 removed by soft count = 175\n",
      "--- gene=465, id=97 removed by soft count = 2006\n",
      "--- gene=465, id=99 removed by soft count = 359\n",
      "--- gene=465, id=101 removed by soft count = 94\n",
      "--- gene=465, id=104 removed by soft count = 2160\n",
      "--- gene=465, id=107 removed by soft count = 88\n",
      "--- gene=465, id=161 removed by soft count = 55\n",
      "--- number of probes: 249 , kept by blast: 240 , if remove dups: 240\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 466\n",
      "--- gene=466, id=26 removed by soft count = 188\n",
      "--- gene=466, id=27 removed by soft count = 133\n",
      "--- gene=466, id=224 removed by soft count = 79\n",
      "--- number of probes: 295 , kept by blast: 292 , if remove dups: 292\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 467\n",
      "--- gene=467, id=227 removed by soft count = 833\n",
      "--- gene=467, id=228 removed by soft count = 157\n",
      "--- number of probes: 336 , kept by blast: 334 , if remove dups: 334\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 469\n",
      "--- gene=469, id=184 removed by soft count = 64\n",
      "--- gene=469, id=185 removed by soft count = 55\n",
      "--- gene=469, id=212 removed by soft count = 1055\n",
      "--- gene=469, id=215 removed by soft count = 92\n",
      "--- gene=469, id=218 removed by soft count = 98\n",
      "--- gene=469, id=230 removed by soft count = 184\n",
      "--- gene=469, id=231 removed by soft count = 87\n",
      "--- number of probes: 266 , kept by blast: 259 , if remove dups: 259\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 470\n",
      "--- gene=470, id=100 removed by soft count = 101\n",
      "--- gene=470, id=170 removed by soft count = 121\n",
      "--- gene=470, id=171 removed by soft count = 188\n",
      "--- number of probes: 241 , kept by blast: 238 , if remove dups: 238\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 471\n",
      "--- number of probes: 287 , kept by blast: 287 , if remove dups: 287\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 472\n",
      "--- gene=472, id=3 removed by soft count = 441\n",
      "--- gene=472, id=4 removed by soft count = 322\n",
      "--- gene=472, id=17 removed by soft count = 172\n",
      "--- gene=472, id=18 removed by soft count = 4794\n",
      "--- gene=472, id=20 removed by soft count = 148\n",
      "--- gene=472, id=78 removed by soft count = 162\n",
      "--- gene=472, id=195 removed by soft count = 114\n",
      "--- gene=472, id=213 removed by soft count = 93\n",
      "--- number of probes: 227 , kept by blast: 219 , if remove dups: 219\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 473\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=473, id=95 removed by soft count = 74\n",
      "--- gene=473, id=137 removed by soft count = 94\n",
      "--- number of probes: 182 , kept by blast: 180 , if remove dups: 180\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 474\n",
      "--- gene=474, id=243 removed by soft count = 58\n",
      "--- number of probes: 283 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 475\n",
      "--- gene=475, id=34 removed by soft count = 87\n",
      "--- gene=475, id=35 removed by soft count = 430\n",
      "--- gene=475, id=36 removed by soft count = 86\n",
      "--- gene=475, id=41 removed by soft count = 550\n",
      "--- number of probes: 160 , kept by blast: 156 , if remove dups: 156\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 476\n",
      "--- gene=476, id=44 removed by soft count = 308\n",
      "--- gene=476, id=61 removed by soft count = 68\n",
      "--- gene=476, id=62 removed by soft count = 344\n",
      "--- gene=476, id=100 removed by soft count = 821\n",
      "--- gene=476, id=141 removed by soft count = 63\n",
      "--- gene=476, id=219 removed by soft count = 95\n",
      "--- number of probes: 220 , kept by blast: 214 , if remove dups: 214\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 477\n",
      "--- gene=477, id=35 removed by soft count = 18416\n",
      "--- gene=477, id=43 removed by soft count = 68\n",
      "--- gene=477, id=52 removed by soft count = 109\n",
      "--- gene=477, id=58 removed by soft count = 53\n",
      "--- gene=477, id=59 removed by soft count = 330\n",
      "--- gene=477, id=66 removed by soft count = 664\n",
      "--- gene=477, id=165 removed by soft count = 331\n",
      "--- number of probes: 174 , kept by blast: 167 , if remove dups: 167\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 478\n",
      "--- gene=478, id=105 removed by soft count = 62\n",
      "--- gene=478, id=116 removed by soft count = 732\n",
      "--- gene=478, id=117 removed by soft count = 360\n",
      "--- gene=478, id=185 removed by soft count = 288\n",
      "--- gene=478, id=189 removed by soft count = 74\n",
      "--- number of probes: 240 , kept by blast: 235 , if remove dups: 235\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 479\n",
      "--- gene=479, id=59 removed by soft count = 120\n",
      "--- gene=479, id=60 removed by soft count = 1897\n",
      "--- gene=479, id=159 removed by soft count = 245\n",
      "--- number of probes: 286 , kept by blast: 283 , if remove dups: 283\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 480\n",
      "--- gene=480, id=11 removed by soft count = 57\n",
      "--- gene=480, id=55 removed by soft count = 246\n",
      "--- gene=480, id=56 removed by soft count = 235\n",
      "--- gene=480, id=58 removed by soft count = 123\n",
      "--- gene=480, id=87 removed by soft count = 319\n",
      "--- gene=480, id=88 removed by soft count = 296\n",
      "--- gene=480, id=93 removed by soft count = 72\n",
      "--- gene=480, id=94 removed by soft count = 178\n",
      "--- gene=480, id=95 removed by soft count = 108\n",
      "--- gene=480, id=104 removed by soft count = 70\n",
      "--- gene=480, id=106 removed by soft count = 52\n",
      "--- number of probes: 178 , kept by blast: 167 , if remove dups: 167\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 481\n",
      "--- gene=481, id=122 removed by soft count = 71\n",
      "--- gene=481, id=123 removed by soft count = 119\n",
      "--- gene=481, id=125 removed by soft count = 80\n",
      "--- gene=481, id=130 removed by soft count = 226\n",
      "--- gene=481, id=161 removed by soft count = 57\n",
      "--- number of probes: 162 , kept by blast: 157 , if remove dups: 157\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 482\n",
      "--- gene=482, id=32 removed by soft count = 107\n",
      "--- gene=482, id=33 removed by soft count = 178\n",
      "--- gene=482, id=35 removed by soft count = 176\n",
      "--- gene=482, id=36 removed by soft count = 73\n",
      "--- gene=482, id=38 removed by soft count = 103\n",
      "--- gene=482, id=40 removed by soft count = 271\n",
      "--- gene=482, id=41 removed by soft count = 116\n",
      "--- gene=482, id=42 removed by soft count = 422\n",
      "--- gene=482, id=43 removed by soft count = 266\n",
      "--- gene=482, id=44 removed by soft count = 181\n",
      "--- gene=482, id=45 removed by soft count = 90\n",
      "--- gene=482, id=46 removed by soft count = 351\n",
      "--- gene=482, id=67 removed by soft count = 171\n",
      "--- gene=482, id=68 removed by soft count = 51\n",
      "--- gene=482, id=75 removed by soft count = 172\n",
      "--- gene=482, id=77 removed by soft count = 272\n",
      "--- gene=482, id=98 removed by soft count = 62\n",
      "--- number of probes: 113 , kept by blast: 96 , if remove dups: 96\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 96\n",
      "-- checking probes in region: 483\n",
      "--- gene=483, id=17 removed by soft count = 241\n",
      "--- gene=483, id=19 removed by soft count = 1326\n",
      "--- gene=483, id=36 removed by soft count = 72\n",
      "--- gene=483, id=40 removed by soft count = 4090\n",
      "--- gene=483, id=84 removed by soft count = 554\n",
      "--- number of probes: 150 , kept by blast: 145 , if remove dups: 145\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 484\n",
      "--- gene=484, id=121 removed by soft count = 151\n",
      "--- gene=484, id=195 removed by soft count = 120\n",
      "--- gene=484, id=276 removed by soft count = 167\n",
      "--- number of probes: 307 , kept by blast: 304 , if remove dups: 304\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 485\n",
      "--- gene=485, id=12 removed by soft count = 842\n",
      "--- gene=485, id=38 removed by soft count = 472\n",
      "--- gene=485, id=48 removed by soft count = 52\n",
      "--- gene=485, id=110 removed by soft count = 99\n",
      "--- gene=485, id=111 removed by soft count = 259\n",
      "--- gene=485, id=115 removed by soft count = 64\n",
      "--- gene=485, id=116 removed by soft count = 142\n",
      "--- gene=485, id=117 removed by soft count = 83\n",
      "--- number of probes: 118 , kept by blast: 110 , if remove dups: 110\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 110\n",
      "-- checking probes in region: 486\n",
      "--- gene=486, id=0 removed by soft count = 692\n",
      "--- gene=486, id=1 removed by soft count = 159\n",
      "--- gene=486, id=2 removed by soft count = 311\n",
      "--- gene=486, id=3 removed by soft count = 358\n",
      "--- gene=486, id=4 removed by soft count = 175\n",
      "--- gene=486, id=6 removed by soft count = 220\n",
      "--- gene=486, id=18 removed by soft count = 154\n",
      "--- gene=486, id=20 removed by soft count = 85\n",
      "--- gene=486, id=21 removed by soft count = 226\n",
      "--- gene=486, id=57 removed by soft count = 182\n",
      "--- gene=486, id=120 removed by soft count = 80\n",
      "--- gene=486, id=121 removed by soft count = 87\n",
      "--- number of probes: 140 , kept by blast: 128 , if remove dups: 128\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 487\n",
      "--- gene=487, id=33 removed by soft count = 59\n",
      "--- gene=487, id=64 removed by soft count = 67\n",
      "--- gene=487, id=105 removed by soft count = 96\n",
      "--- number of probes: 108 , kept by blast: 105 , if remove dups: 105\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 105\n",
      "-- checking probes in region: 488\n",
      "--- gene=488, id=15 removed by soft count = 94\n",
      "--- gene=488, id=47 removed by soft count = 2402\n",
      "--- gene=488, id=48 removed by soft count = 67\n",
      "--- gene=488, id=49 removed by soft count = 731\n",
      "--- gene=488, id=51 removed by soft count = 249\n",
      "--- gene=488, id=52 removed by soft count = 1434\n",
      "--- gene=488, id=58 removed by soft count = 819\n",
      "--- gene=488, id=59 removed by soft count = 77\n",
      "--- gene=488, id=60 removed by soft count = 221\n",
      "--- number of probes: 160 , kept by blast: 151 , if remove dups: 151\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 489\n",
      "--- gene=489, id=0 removed by soft count = 888\n",
      "--- gene=489, id=1 removed by soft count = 191\n",
      "--- gene=489, id=140 removed by soft count = 101\n",
      "--- gene=489, id=142 removed by soft count = 144\n",
      "--- number of probes: 144 , kept by blast: 140 , if remove dups: 139\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 490\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=490, id=0 removed by soft count = 119\n",
      "--- gene=490, id=52 removed by soft count = 72\n",
      "--- gene=490, id=70 removed by soft count = 744\n",
      "--- gene=490, id=71 removed by soft count = 568\n",
      "--- gene=490, id=178 removed by soft count = 333\n",
      "--- gene=490, id=179 removed by soft count = 427\n",
      "--- gene=490, id=180 removed by soft count = 87\n",
      "--- number of probes: 183 , kept by blast: 176 , if remove dups: 176\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 491\n",
      "--- gene=491, id=65 removed by soft count = 173\n",
      "--- gene=491, id=81 removed by soft count = 157\n",
      "--- gene=491, id=84 removed by soft count = 58\n",
      "--- gene=491, id=264 removed by soft count = 99\n",
      "--- gene=491, id=265 removed by soft count = 595\n",
      "--- number of probes: 282 , kept by blast: 277 , if remove dups: 277\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 492\n",
      "--- gene=492, id=105 removed by soft count = 69\n",
      "--- gene=492, id=117 removed by soft count = 235\n",
      "--- gene=492, id=125 removed by soft count = 74\n",
      "--- gene=492, id=134 removed by soft count = 102\n",
      "--- number of probes: 157 , kept by blast: 153 , if remove dups: 153\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 493\n",
      "--- gene=493, id=52 removed by soft count = 55\n",
      "--- gene=493, id=53 removed by soft count = 125\n",
      "--- gene=493, id=182 removed by soft count = 77\n",
      "--- number of probes: 227 , kept by blast: 224 , if remove dups: 224\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 494\n",
      "--- gene=494, id=37 removed by soft count = 62\n",
      "--- gene=494, id=38 removed by soft count = 61\n",
      "--- gene=494, id=40 removed by soft count = 64\n",
      "--- gene=494, id=176 removed by soft count = 63\n",
      "--- number of probes: 182 , kept by blast: 178 , if remove dups: 178\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 495\n",
      "--- number of probes: 291 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 496\n",
      "--- gene=496, id=15 removed by soft count = 57\n",
      "--- gene=496, id=16 removed by soft count = 51\n",
      "--- gene=496, id=49 removed by soft count = 171\n",
      "--- gene=496, id=50 removed by soft count = 81\n",
      "--- gene=496, id=109 removed by soft count = 142\n",
      "--- gene=496, id=164 removed by soft count = 288\n",
      "--- gene=496, id=171 removed by soft count = 58\n",
      "--- gene=496, id=175 removed by soft count = 172\n",
      "--- gene=496, id=176 removed by soft count = 51\n",
      "--- gene=496, id=179 removed by soft count = 66\n",
      "--- gene=496, id=180 removed by soft count = 89\n",
      "--- gene=496, id=181 removed by soft count = 98\n",
      "--- gene=496, id=182 removed by soft count = 4128\n",
      "--- gene=496, id=195 removed by soft count = 62\n",
      "--- number of probes: 203 , kept by blast: 189 , if remove dups: 189\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 497\n",
      "--- gene=497, id=68 removed by soft count = 60\n",
      "--- number of probes: 259 , kept by blast: 258 , if remove dups: 258\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 498\n",
      "--- gene=498, id=102 removed by soft count = 255\n",
      "--- gene=498, id=103 removed by soft count = 63\n",
      "--- gene=498, id=113 removed by soft count = 62\n",
      "--- number of probes: 208 , kept by blast: 205 , if remove dups: 205\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 499\n",
      "--- gene=499, id=38 removed by soft count = 65\n",
      "--- gene=499, id=54 removed by soft count = 101\n",
      "--- gene=499, id=55 removed by soft count = 1886\n",
      "--- gene=499, id=66 removed by soft count = 78\n",
      "--- gene=499, id=81 removed by soft count = 21335\n",
      "--- gene=499, id=82 removed by soft count = 1139\n",
      "--- gene=499, id=83 removed by soft count = 1792\n",
      "--- gene=499, id=210 removed by soft count = 60\n",
      "--- gene=499, id=213 removed by soft count = 53\n",
      "--- gene=499, id=216 removed by soft count = 111\n",
      "--- gene=499, id=217 removed by soft count = 126\n",
      "--- number of probes: 219 , kept by blast: 208 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 500\n",
      "--- gene=500, id=0 removed by soft count = 7767\n",
      "--- gene=500, id=124 removed by soft count = 54\n",
      "--- gene=500, id=125 removed by soft count = 233\n",
      "--- gene=500, id=126 removed by soft count = 139\n",
      "--- gene=500, id=138 removed by soft count = 114\n",
      "--- gene=500, id=143 removed by soft count = 3393\n",
      "--- gene=500, id=144 removed by soft count = 67\n",
      "--- gene=500, id=147 removed by soft count = 92\n",
      "--- number of probes: 169 , kept by blast: 161 , if remove dups: 161\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 501\n",
      "--- gene=501, id=180 removed by soft count = 78\n",
      "--- gene=501, id=181 removed by soft count = 2584\n",
      "--- number of probes: 202 , kept by blast: 200 , if remove dups: 200\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 502\n",
      "--- gene=502, id=127 removed by soft count = 63\n",
      "--- number of probes: 316 , kept by blast: 315 , if remove dups: 315\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 503\n",
      "--- gene=503, id=76 removed by soft count = 287\n",
      "--- gene=503, id=279 removed by soft count = 85\n",
      "--- number of probes: 321 , kept by blast: 319 , if remove dups: 319\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 504\n",
      "--- gene=504, id=1 removed by soft count = 52\n",
      "--- gene=504, id=36 removed by soft count = 188\n",
      "--- gene=504, id=88 removed by soft count = 66\n",
      "--- gene=504, id=177 removed by soft count = 64\n",
      "--- number of probes: 179 , kept by blast: 175 , if remove dups: 175\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 505\n",
      "--- gene=505, id=86 removed by soft count = 51\n",
      "--- gene=505, id=258 removed by soft count = 147\n",
      "--- gene=505, id=259 removed by soft count = 133\n",
      "--- number of probes: 273 , kept by blast: 270 , if remove dups: 270\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 506\n",
      "--- gene=506, id=164 removed by soft count = 54\n",
      "--- gene=506, id=165 removed by soft count = 87\n",
      "--- gene=506, id=206 removed by soft count = 64\n",
      "--- number of probes: 279 , kept by blast: 276 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 507\n",
      "--- gene=507, id=146 removed by soft count = 58\n",
      "--- number of probes: 282 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 508\n",
      "--- gene=508, id=20 removed by soft count = 91\n",
      "--- gene=508, id=88 removed by soft count = 103\n",
      "--- number of probes: 309 , kept by blast: 307 , if remove dups: 306\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 509\n",
      "--- gene=509, id=232 removed by soft count = 68\n",
      "--- number of probes: 261 , kept by blast: 260 , if remove dups: 260\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 510\n",
      "--- gene=510, id=14 removed by soft count = 65\n",
      "--- gene=510, id=15 removed by soft count = 60\n",
      "--- gene=510, id=24 removed by soft count = 11828\n",
      "--- gene=510, id=56 removed by soft count = 73\n",
      "--- number of probes: 294 , kept by blast: 290 , if remove dups: 290\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 511\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=511, id=4 removed by soft count = 210\n",
      "--- gene=511, id=5 removed by soft count = 107\n",
      "--- gene=511, id=8 removed by soft count = 1101\n",
      "--- gene=511, id=9 removed by soft count = 1019\n",
      "--- gene=511, id=12 removed by soft count = 131\n",
      "--- gene=511, id=14 removed by soft count = 183\n",
      "--- gene=511, id=22 removed by soft count = 70\n",
      "--- number of probes: 103 , kept by blast: 96 , if remove dups: 96\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 96\n",
      "-- checking probes in region: 513\n",
      "--- gene=513, id=0 removed by soft count = 1006\n",
      "--- gene=513, id=2 removed by soft count = 67\n",
      "--- gene=513, id=8 removed by soft count = 104\n",
      "--- gene=513, id=22 removed by soft count = 898\n",
      "--- gene=513, id=24 removed by soft count = 179\n",
      "--- gene=513, id=25 removed by soft count = 616\n",
      "--- gene=513, id=28 removed by soft count = 74\n",
      "--- gene=513, id=31 removed by soft count = 124\n",
      "--- gene=513, id=32 removed by soft count = 2844\n",
      "--- gene=513, id=34 removed by soft count = 232\n",
      "--- gene=513, id=35 removed by soft count = 122\n",
      "--- gene=513, id=38 removed by soft count = 147\n",
      "--- number of probes: 53 , kept by blast: 41 , if remove dups: 41\n",
      "-- number of probes kept for this region: 41\n",
      "-- checking probes in region: 514\n",
      "--- gene=514, id=95 removed by soft count = 53\n",
      "--- gene=514, id=96 removed by soft count = 10382\n",
      "--- gene=514, id=187 removed by soft count = 437\n",
      "--- number of probes: 247 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 515\n",
      "--- gene=515, id=94 removed by soft count = 58\n",
      "--- gene=515, id=99 removed by soft count = 85\n",
      "--- gene=515, id=100 removed by soft count = 227\n",
      "--- gene=515, id=103 removed by soft count = 57\n",
      "--- gene=515, id=220 removed by soft count = 596\n",
      "--- number of probes: 223 , kept by blast: 218 , if remove dups: 218\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 516\n",
      "--- gene=516, id=96 removed by soft count = 120\n",
      "--- gene=516, id=97 removed by soft count = 983\n",
      "--- gene=516, id=98 removed by soft count = 176\n",
      "--- gene=516, id=99 removed by soft count = 54\n",
      "--- gene=516, id=100 removed by soft count = 123\n",
      "--- gene=516, id=102 removed by soft count = 106\n",
      "--- gene=516, id=103 removed by soft count = 113\n",
      "--- gene=516, id=106 removed by soft count = 61\n",
      "--- gene=516, id=109 removed by soft count = 137\n",
      "--- gene=516, id=110 removed by soft count = 1114\n",
      "--- gene=516, id=113 removed by soft count = 89\n",
      "--- gene=516, id=115 removed by soft count = 136\n",
      "--- gene=516, id=116 removed by soft count = 66\n",
      "--- gene=516, id=117 removed by soft count = 181\n",
      "--- number of probes: 120 , kept by blast: 106 , if remove dups: 106\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 106\n",
      "-- checking probes in region: 517\n",
      "--- gene=517, id=38 removed by soft count = 68\n",
      "--- gene=517, id=40 removed by soft count = 668\n",
      "--- gene=517, id=41 removed by soft count = 54\n",
      "--- gene=517, id=62 removed by soft count = 213\n",
      "--- gene=517, id=63 removed by soft count = 63\n",
      "--- gene=517, id=65 removed by soft count = 174\n",
      "--- number of probes: 187 , kept by blast: 181 , if remove dups: 181\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 518\n",
      "--- gene=518, id=116 removed by soft count = 130\n",
      "--- gene=518, id=117 removed by soft count = 348\n",
      "--- gene=518, id=236 removed by soft count = 352\n",
      "--- number of probes: 237 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 519\n",
      "--- gene=519, id=38 removed by soft count = 140\n",
      "--- gene=519, id=118 removed by soft count = 95\n",
      "--- gene=519, id=179 removed by soft count = 82\n",
      "--- number of probes: 216 , kept by blast: 213 , if remove dups: 213\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 520\n",
      "--- gene=520, id=117 removed by soft count = 66\n",
      "--- gene=520, id=169 removed by soft count = 92\n",
      "--- gene=520, id=188 removed by soft count = 226\n",
      "--- gene=520, id=192 removed by soft count = 89\n",
      "--- gene=520, id=193 removed by soft count = 363\n",
      "--- gene=520, id=196 removed by soft count = 158\n",
      "--- gene=520, id=208 removed by soft count = 104\n",
      "--- number of probes: 222 , kept by blast: 215 , if remove dups: 215\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 521\n",
      "--- gene=521, id=8 removed by soft count = 251\n",
      "--- gene=521, id=9 removed by soft count = 114\n",
      "--- gene=521, id=46 removed by soft count = 204\n",
      "--- gene=521, id=103 removed by soft count = 489\n",
      "--- gene=521, id=104 removed by soft count = 374\n",
      "--- gene=521, id=105 removed by soft count = 176\n",
      "--- gene=521, id=107 removed by soft count = 57\n",
      "--- gene=521, id=108 removed by soft count = 171\n",
      "--- gene=521, id=112 removed by soft count = 298\n",
      "--- gene=521, id=113 removed by soft count = 190\n",
      "--- number of probes: 116 , kept by blast: 106 , if remove dups: 106\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 106\n",
      "-- checking probes in region: 522\n",
      "--- gene=522, id=67 removed by soft count = 110\n",
      "--- gene=522, id=104 removed by soft count = 103\n",
      "--- gene=522, id=182 removed by soft count = 179\n",
      "--- gene=522, id=191 removed by soft count = 318\n",
      "--- gene=522, id=192 removed by soft count = 222\n",
      "--- number of probes: 219 , kept by blast: 214 , if remove dups: 214\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 523\n",
      "--- gene=523, id=11 removed by soft count = 8378\n",
      "--- number of probes: 230 , kept by blast: 229 , if remove dups: 229\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 524\n",
      "--- gene=524, id=0 removed by soft count = 1149\n",
      "--- gene=524, id=11 removed by soft count = 171\n",
      "--- gene=524, id=199 removed by soft count = 194\n",
      "--- number of probes: 217 , kept by blast: 214 , if remove dups: 214\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 525\n",
      "--- gene=525, id=43 removed by soft count = 53\n",
      "--- gene=525, id=44 removed by soft count = 178\n",
      "--- gene=525, id=115 removed by soft count = 98\n",
      "--- gene=525, id=116 removed by soft count = 1050\n",
      "--- gene=525, id=117 removed by soft count = 228\n",
      "--- gene=525, id=119 removed by soft count = 240\n",
      "--- number of probes: 180 , kept by blast: 174 , if remove dups: 174\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 526\n",
      "--- gene=526, id=0 removed by soft count = 672\n",
      "--- gene=526, id=1 removed by soft count = 224\n",
      "--- gene=526, id=2 removed by soft count = 93\n",
      "--- gene=526, id=3 removed by soft count = 81\n",
      "--- gene=526, id=9 removed by soft count = 52\n",
      "--- gene=526, id=37 removed by soft count = 1129\n",
      "--- gene=526, id=38 removed by soft count = 289\n",
      "--- gene=526, id=46 removed by soft count = 422\n",
      "--- gene=526, id=47 removed by soft count = 1942\n",
      "--- gene=526, id=48 removed by soft count = 489\n",
      "--- gene=526, id=49 removed by soft count = 71\n",
      "--- gene=526, id=50 removed by soft count = 209\n",
      "--- gene=526, id=51 removed by soft count = 17693\n",
      "--- gene=526, id=53 removed by soft count = 143\n",
      "--- gene=526, id=54 removed by soft count = 100\n",
      "--- number of probes: 90 , kept by blast: 75 , if remove dups: 75\n",
      "-- number of probes kept for this region: 75\n",
      "-- checking probes in region: 527\n",
      "--- gene=527, id=23 removed by soft count = 514\n",
      "--- gene=527, id=24 removed by soft count = 59\n",
      "--- gene=527, id=28 removed by soft count = 121\n",
      "--- gene=527, id=96 removed by soft count = 633\n",
      "--- gene=527, id=97 removed by soft count = 379\n",
      "--- gene=527, id=98 removed by soft count = 85\n",
      "--- gene=527, id=99 removed by soft count = 69\n",
      "--- gene=527, id=102 removed by soft count = 158\n",
      "--- number of probes: 193 , kept by blast: 185 , if remove dups: 185\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 528\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=528, id=1 removed by soft count = 56\n",
      "--- gene=528, id=2 removed by soft count = 103\n",
      "--- gene=528, id=56 removed by soft count = 55\n",
      "--- gene=528, id=112 removed by soft count = 51\n",
      "--- gene=528, id=113 removed by soft count = 58\n",
      "--- gene=528, id=159 removed by soft count = 70\n",
      "--- gene=528, id=252 removed by soft count = 156\n",
      "--- number of probes: 256 , kept by blast: 249 , if remove dups: 249\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 529\n",
      "--- gene=529, id=9 removed by soft count = 73\n",
      "--- gene=529, id=13 removed by soft count = 68\n",
      "--- gene=529, id=19 removed by soft count = 100\n",
      "--- gene=529, id=292 removed by soft count = 53\n",
      "--- number of probes: 349 , kept by blast: 345 , if remove dups: 345\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 530\n",
      "--- gene=530, id=144 removed by soft count = 115\n",
      "--- gene=530, id=200 removed by soft count = 192\n",
      "--- gene=530, id=250 removed by soft count = 64\n",
      "--- number of probes: 255 , kept by blast: 252 , if remove dups: 252\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 531\n",
      "--- gene=531, id=278 removed by soft count = 735\n",
      "--- number of probes: 330 , kept by blast: 329 , if remove dups: 329\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 532\n",
      "--- gene=532, id=39 removed by soft count = 264\n",
      "--- gene=532, id=40 removed by soft count = 223\n",
      "--- gene=532, id=44 removed by soft count = 59\n",
      "--- gene=532, id=45 removed by soft count = 69\n",
      "--- number of probes: 252 , kept by blast: 248 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 533\n",
      "--- gene=533, id=91 removed by soft count = 67\n",
      "--- gene=533, id=125 removed by soft count = 59\n",
      "--- gene=533, id=130 removed by soft count = 228\n",
      "--- gene=533, id=131 removed by soft count = 76\n",
      "--- gene=533, id=133 removed by soft count = 71\n",
      "--- gene=533, id=163 removed by soft count = 139\n",
      "--- number of probes: 224 , kept by blast: 218 , if remove dups: 218\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 534\n",
      "--- gene=534, id=141 removed by soft count = 445\n",
      "--- gene=534, id=142 removed by soft count = 185\n",
      "--- gene=534, id=143 removed by soft count = 459\n",
      "--- number of probes: 264 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 535\n",
      "--- gene=535, id=166 removed by soft count = 171\n",
      "--- gene=535, id=232 removed by hard count = 3\n",
      "--- gene=535, id=233 removed by soft count = 51\n",
      "--- number of probes: 268 , kept by blast: 265 , if remove dups: 265\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 536\n",
      "--- gene=536, id=71 removed by soft count = 58\n",
      "--- gene=536, id=188 removed by soft count = 80\n",
      "--- gene=536, id=189 removed by soft count = 81\n",
      "--- number of probes: 269 , kept by blast: 266 , if remove dups: 266\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 537\n",
      "--- gene=537, id=21 removed by soft count = 90\n",
      "--- number of probes: 331 , kept by blast: 330 , if remove dups: 330\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 538\n",
      "--- gene=538, id=77 removed by soft count = 347\n",
      "--- gene=538, id=78 removed by soft count = 52\n",
      "--- gene=538, id=301 removed by soft count = 63\n",
      "--- gene=538, id=302 removed by soft count = 59\n",
      "--- gene=538, id=303 removed by hard count = 3\n",
      "--- number of probes: 304 , kept by blast: 299 , if remove dups: 299\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 539\n",
      "--- gene=539, id=36 removed by soft count = 53\n",
      "--- gene=539, id=66 removed by soft count = 70\n",
      "--- gene=539, id=67 removed by soft count = 52\n",
      "--- gene=539, id=101 removed by soft count = 87\n",
      "--- gene=539, id=250 removed by soft count = 262\n",
      "--- number of probes: 259 , kept by blast: 254 , if remove dups: 253\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 540\n",
      "--- number of probes: 252 , kept by blast: 252 , if remove dups: 252\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 541\n",
      "--- gene=541, id=119 removed by soft count = 147\n",
      "--- gene=541, id=142 removed by soft count = 97\n",
      "--- number of probes: 267 , kept by blast: 265 , if remove dups: 265\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 542\n",
      "--- gene=542, id=65 removed by soft count = 152\n",
      "--- gene=542, id=99 removed by soft count = 123\n",
      "--- gene=542, id=112 removed by soft count = 575\n",
      "--- gene=542, id=129 removed by soft count = 143\n",
      "--- number of probes: 200 , kept by blast: 196 , if remove dups: 196\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 543\n",
      "--- gene=543, id=12 removed by soft count = 57\n",
      "--- gene=543, id=137 removed by soft count = 73\n",
      "--- gene=543, id=138 removed by soft count = 198\n",
      "--- number of probes: 140 , kept by blast: 137 , if remove dups: 137\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 544\n",
      "--- gene=544, id=20 removed by soft count = 99\n",
      "--- gene=544, id=21 removed by soft count = 188\n",
      "--- gene=544, id=22 removed by soft count = 130\n",
      "--- number of probes: 75 , kept by blast: 72 , if remove dups: 72\n",
      "-- number of probes kept for this region: 72\n",
      "-- checking probes in region: 545\n",
      "--- gene=545, id=39 removed by soft count = 69\n",
      "--- gene=545, id=106 removed by soft count = 56\n",
      "--- number of probes: 162 , kept by blast: 160 , if remove dups: 160\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 546\n",
      "--- gene=546, id=85 removed by soft count = 57\n",
      "--- gene=546, id=90 removed by soft count = 78\n",
      "--- gene=546, id=152 removed by soft count = 52\n",
      "--- gene=546, id=156 removed by soft count = 153\n",
      "--- gene=546, id=159 removed by soft count = 3569\n",
      "--- gene=546, id=173 removed by soft count = 144\n",
      "--- number of probes: 234 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 548\n",
      "--- gene=548, id=6 removed by soft count = 57\n",
      "--- gene=548, id=26 removed by soft count = 15130\n",
      "--- gene=548, id=61 removed by soft count = 419\n",
      "--- number of probes: 88 , kept by blast: 85 , if remove dups: 85\n",
      "-- number of probes kept for this region: 85\n",
      "-- checking probes in region: 549\n",
      "--- gene=549, id=37 removed by soft count = 138\n",
      "--- gene=549, id=38 removed by soft count = 54\n",
      "--- gene=549, id=75 removed by soft count = 88\n",
      "--- gene=549, id=90 removed by soft count = 162\n",
      "--- gene=549, id=91 removed by soft count = 254\n",
      "--- gene=549, id=92 removed by soft count = 628\n",
      "--- gene=549, id=172 removed by soft count = 444\n",
      "--- gene=549, id=173 removed by soft count = 6851\n",
      "--- number of probes: 174 , kept by blast: 166 , if remove dups: 166\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 550\n",
      "--- gene=550, id=3 removed by soft count = 55\n",
      "--- gene=550, id=4 removed by soft count = 72\n",
      "--- gene=550, id=11 removed by soft count = 759\n",
      "--- gene=550, id=18 removed by soft count = 2775\n",
      "--- gene=550, id=23 removed by soft count = 223\n",
      "--- gene=550, id=106 removed by soft count = 120\n",
      "--- gene=550, id=125 removed by soft count = 57\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- number of probes: 181 , kept by blast: 174 , if remove dups: 174\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 551\n",
      "--- gene=551, id=16 removed by soft count = 116\n",
      "--- gene=551, id=39 removed by soft count = 138\n",
      "--- gene=551, id=52 removed by soft count = 69\n",
      "--- gene=551, id=81 removed by soft count = 307\n",
      "--- gene=551, id=138 removed by soft count = 137\n",
      "--- gene=551, id=188 removed by soft count = 128\n",
      "--- gene=551, id=189 removed by soft count = 63\n",
      "--- gene=551, id=196 removed by soft count = 140\n",
      "--- number of probes: 236 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 552\n",
      "--- gene=552, id=54 removed by soft count = 52\n",
      "--- gene=552, id=232 removed by soft count = 99\n",
      "--- number of probes: 239 , kept by blast: 237 , if remove dups: 237\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 553\n",
      "--- gene=553, id=217 removed by soft count = 132\n",
      "--- gene=553, id=218 removed by soft count = 500\n",
      "--- gene=553, id=219 removed by soft count = 51\n",
      "--- gene=553, id=232 removed by soft count = 166\n",
      "--- number of probes: 314 , kept by blast: 310 , if remove dups: 310\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 554\n",
      "--- gene=554, id=7 removed by soft count = 85\n",
      "--- gene=554, id=22 removed by soft count = 185\n",
      "--- gene=554, id=66 removed by soft count = 80\n",
      "--- gene=554, id=111 removed by soft count = 95\n",
      "--- gene=554, id=113 removed by soft count = 73\n",
      "--- gene=554, id=115 removed by soft count = 135\n",
      "--- gene=554, id=168 removed by soft count = 67\n",
      "--- gene=554, id=172 removed by soft count = 216\n",
      "--- gene=554, id=176 removed by soft count = 87\n",
      "--- gene=554, id=178 removed by soft count = 130\n",
      "--- number of probes: 243 , kept by blast: 233 , if remove dups: 233\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 555\n",
      "--- gene=555, id=57 removed by soft count = 58\n",
      "--- number of probes: 259 , kept by blast: 258 , if remove dups: 258\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 556\n",
      "--- gene=556, id=45 removed by soft count = 64\n",
      "--- gene=556, id=52 removed by soft count = 65\n",
      "--- number of probes: 251 , kept by blast: 249 , if remove dups: 249\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 557\n",
      "--- gene=557, id=1 removed by soft count = 90\n",
      "--- gene=557, id=68 removed by soft count = 72\n",
      "--- gene=557, id=83 removed by soft count = 127\n",
      "--- gene=557, id=115 removed by soft count = 96\n",
      "--- gene=557, id=117 removed by soft count = 100\n",
      "--- gene=557, id=159 removed by soft count = 60\n",
      "--- gene=557, id=187 removed by soft count = 245\n",
      "--- gene=557, id=210 removed by soft count = 55\n",
      "--- gene=557, id=212 removed by soft count = 66\n",
      "--- gene=557, id=216 removed by soft count = 780\n",
      "--- number of probes: 234 , kept by blast: 224 , if remove dups: 223\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 558\n",
      "--- gene=558, id=144 removed by soft count = 66\n",
      "--- gene=558, id=150 removed by soft count = 58\n",
      "--- number of probes: 246 , kept by blast: 244 , if remove dups: 243\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 559\n",
      "--- gene=559, id=84 removed by soft count = 96\n",
      "--- gene=559, id=189 removed by soft count = 155\n",
      "--- gene=559, id=191 removed by soft count = 274\n",
      "--- gene=559, id=227 removed by soft count = 79\n",
      "--- gene=559, id=229 removed by soft count = 55\n",
      "--- number of probes: 296 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 560\n",
      "--- gene=560, id=7 removed by soft count = 2674\n",
      "--- gene=560, id=55 removed by soft count = 884\n",
      "--- gene=560, id=76 removed by soft count = 70\n",
      "--- number of probes: 286 , kept by blast: 283 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 561\n",
      "--- gene=561, id=124 removed by soft count = 72\n",
      "--- number of probes: 148 , kept by blast: 147 , if remove dups: 147\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 563\n",
      "--- gene=563, id=66 removed by soft count = 78\n",
      "--- number of probes: 296 , kept by blast: 295 , if remove dups: 295\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 564\n",
      "--- gene=564, id=49 removed by soft count = 67\n",
      "--- gene=564, id=58 removed by soft count = 253\n",
      "--- gene=564, id=67 removed by soft count = 345\n",
      "--- gene=564, id=83 removed by soft count = 181\n",
      "--- gene=564, id=84 removed by soft count = 433\n",
      "--- gene=564, id=86 removed by soft count = 174\n",
      "--- gene=564, id=87 removed by soft count = 1686\n",
      "--- gene=564, id=88 removed by soft count = 3380\n",
      "--- gene=564, id=89 removed by soft count = 753\n",
      "--- gene=564, id=90 removed by soft count = 15059\n",
      "--- gene=564, id=91 removed by soft count = 4423\n",
      "--- gene=564, id=93 removed by soft count = 242\n",
      "--- number of probes: 94 , kept by blast: 82 , if remove dups: 82\n",
      "-- number of probes kept for this region: 82\n",
      "-- checking probes in region: 565\n",
      "--- gene=565, id=63 removed by soft count = 181\n",
      "--- number of probes: 326 , kept by blast: 325 , if remove dups: 325\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 566\n",
      "--- gene=566, id=0 removed by soft count = 3891\n",
      "--- gene=566, id=15 removed by soft count = 134\n",
      "--- gene=566, id=43 removed by soft count = 280\n",
      "--- gene=566, id=59 removed by soft count = 68\n",
      "--- gene=566, id=63 removed by soft count = 118\n",
      "--- gene=566, id=66 removed by soft count = 202\n",
      "--- gene=566, id=70 removed by soft count = 662\n",
      "--- gene=566, id=71 removed by soft count = 104\n",
      "--- gene=566, id=72 removed by soft count = 90\n",
      "--- gene=566, id=136 removed by soft count = 227\n",
      "--- gene=566, id=138 removed by soft count = 61\n",
      "--- gene=566, id=140 removed by soft count = 129\n",
      "--- gene=566, id=141 removed by soft count = 70\n",
      "--- gene=566, id=146 removed by soft count = 363\n",
      "--- gene=566, id=147 removed by soft count = 94\n",
      "--- gene=566, id=148 removed by soft count = 105\n",
      "--- number of probes: 184 , kept by blast: 168 , if remove dups: 168\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 567\n",
      "--- gene=567, id=116 removed by soft count = 56\n",
      "--- gene=567, id=187 removed by soft count = 56\n",
      "--- number of probes: 265 , kept by blast: 263 , if remove dups: 263\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 568\n",
      "--- gene=568, id=0 removed by soft count = 389\n",
      "--- gene=568, id=1 removed by soft count = 205\n",
      "--- gene=568, id=2 removed by soft count = 405\n",
      "--- gene=568, id=3 removed by soft count = 19190\n",
      "--- gene=568, id=4 removed by soft count = 201\n",
      "--- gene=568, id=5 removed by soft count = 1064\n",
      "--- gene=568, id=6 removed by soft count = 310\n",
      "--- gene=568, id=7 removed by soft count = 86\n",
      "--- gene=568, id=83 removed by soft count = 57\n",
      "--- gene=568, id=99 removed by soft count = 170\n",
      "--- number of probes: 139 , kept by blast: 129 , if remove dups: 129\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 569\n",
      "--- gene=569, id=6 removed by soft count = 56\n",
      "--- number of probes: 166 , kept by blast: 165 , if remove dups: 165\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 570\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=570, id=114 removed by soft count = 96\n",
      "--- number of probes: 241 , kept by blast: 240 , if remove dups: 240\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 571\n",
      "--- gene=571, id=23 removed by soft count = 112\n",
      "--- gene=571, id=112 removed by soft count = 58\n",
      "--- number of probes: 209 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 572\n",
      "--- gene=572, id=226 removed by soft count = 102\n",
      "--- gene=572, id=228 removed by soft count = 53\n",
      "--- number of probes: 246 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 573\n",
      "--- gene=573, id=39 removed by soft count = 177\n",
      "--- gene=573, id=40 removed by soft count = 193\n",
      "--- gene=573, id=41 removed by soft count = 354\n",
      "--- gene=573, id=42 removed by soft count = 185\n",
      "--- gene=573, id=43 removed by soft count = 495\n",
      "--- gene=573, id=45 removed by soft count = 309\n",
      "--- gene=573, id=46 removed by soft count = 202\n",
      "--- gene=573, id=47 removed by soft count = 60\n",
      "--- gene=573, id=64 removed by soft count = 77\n",
      "--- number of probes: 138 , kept by blast: 129 , if remove dups: 129\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 574\n",
      "--- gene=574, id=0 removed by soft count = 277\n",
      "--- gene=574, id=1 removed by soft count = 208\n",
      "--- gene=574, id=5 removed by soft count = 158\n",
      "--- gene=574, id=48 removed by soft count = 175\n",
      "--- gene=574, id=126 removed by soft count = 58\n",
      "--- gene=574, id=156 removed by soft count = 231\n",
      "--- number of probes: 218 , kept by blast: 212 , if remove dups: 211\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 575\n",
      "--- gene=575, id=0 removed by soft count = 331\n",
      "--- gene=575, id=1 removed by soft count = 48636\n",
      "--- gene=575, id=2 removed by soft count = 6146\n",
      "--- gene=575, id=3 removed by soft count = 368\n",
      "--- gene=575, id=4 removed by soft count = 209\n",
      "--- gene=575, id=5 removed by soft count = 101\n",
      "--- gene=575, id=6 removed by soft count = 51\n",
      "--- gene=575, id=7 removed by soft count = 104\n",
      "--- gene=575, id=9 removed by soft count = 54\n",
      "--- gene=575, id=63 removed by soft count = 106\n",
      "--- gene=575, id=64 removed by soft count = 67\n",
      "--- gene=575, id=65 removed by soft count = 72\n",
      "--- gene=575, id=68 removed by soft count = 351\n",
      "--- gene=575, id=69 removed by soft count = 232\n",
      "--- gene=575, id=70 removed by soft count = 179\n",
      "--- number of probes: 73 , kept by blast: 58 , if remove dups: 58\n",
      "-- number of probes kept for this region: 58\n",
      "-- checking probes in region: 576\n",
      "--- gene=576, id=7 removed by soft count = 240\n",
      "--- gene=576, id=100 removed by soft count = 79\n",
      "--- gene=576, id=112 removed by soft count = 342\n",
      "--- gene=576, id=113 removed by soft count = 88\n",
      "--- gene=576, id=159 removed by soft count = 1393\n",
      "--- gene=576, id=160 removed by soft count = 141\n",
      "--- number of probes: 161 , kept by blast: 155 , if remove dups: 155\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 577\n",
      "--- gene=577, id=24 removed by soft count = 116\n",
      "--- gene=577, id=32 removed by soft count = 52\n",
      "--- gene=577, id=34 removed by soft count = 184\n",
      "--- gene=577, id=35 removed by soft count = 192\n",
      "--- gene=577, id=36 removed by soft count = 356\n",
      "--- gene=577, id=40 removed by soft count = 144\n",
      "--- gene=577, id=46 removed by soft count = 86\n",
      "--- gene=577, id=47 removed by soft count = 182\n",
      "--- gene=577, id=51 removed by soft count = 76\n",
      "--- gene=577, id=55 removed by soft count = 62\n",
      "--- gene=577, id=57 removed by soft count = 508\n",
      "--- gene=577, id=58 removed by soft count = 180\n",
      "--- gene=577, id=59 removed by soft count = 6544\n",
      "--- number of probes: 60 , kept by blast: 47 , if remove dups: 47\n",
      "-- number of probes kept for this region: 47\n",
      "-- checking probes in region: 578\n",
      "--- gene=578, id=44 removed by soft count = 98\n",
      "--- gene=578, id=69 removed by soft count = 158\n",
      "--- gene=578, id=230 removed by soft count = 61\n",
      "--- number of probes: 284 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 579\n",
      "--- gene=579, id=237 removed by soft count = 74\n",
      "--- number of probes: 252 , kept by blast: 251 , if remove dups: 251\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 580\n",
      "--- gene=580, id=138 removed by soft count = 72\n",
      "--- gene=580, id=241 removed by soft count = 373\n",
      "--- number of probes: 248 , kept by blast: 246 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 581\n",
      "--- gene=581, id=12 removed by soft count = 112\n",
      "--- gene=581, id=13 removed by soft count = 113\n",
      "--- gene=581, id=14 removed by soft count = 72\n",
      "--- gene=581, id=17 removed by soft count = 85\n",
      "--- gene=581, id=21 removed by soft count = 73\n",
      "--- gene=581, id=23 removed by soft count = 95\n",
      "--- gene=581, id=29 removed by soft count = 107\n",
      "--- gene=581, id=30 removed by soft count = 159\n",
      "--- gene=581, id=37 removed by soft count = 187\n",
      "--- gene=581, id=40 removed by soft count = 56\n",
      "--- gene=581, id=43 removed by soft count = 274\n",
      "--- gene=581, id=44 removed by soft count = 1022\n",
      "--- gene=581, id=46 removed by soft count = 69\n",
      "--- gene=581, id=79 removed by soft count = 333\n",
      "--- gene=581, id=80 removed by soft count = 76\n",
      "--- gene=581, id=83 removed by soft count = 969\n",
      "--- number of probes: 94 , kept by blast: 78 , if remove dups: 78\n",
      "-- number of probes kept for this region: 78\n",
      "-- checking probes in region: 582\n",
      "--- gene=582, id=2 removed by soft count = 120\n",
      "--- gene=582, id=85 removed by soft count = 57\n",
      "--- number of probes: 152 , kept by blast: 150 , if remove dups: 150\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 583\n",
      "--- gene=583, id=8 removed by soft count = 59\n",
      "--- gene=583, id=48 removed by soft count = 120\n",
      "--- gene=583, id=49 removed by soft count = 69\n",
      "--- gene=583, id=52 removed by soft count = 53\n",
      "--- gene=583, id=55 removed by soft count = 87\n",
      "--- gene=583, id=244 removed by soft count = 92\n",
      "--- gene=583, id=246 removed by soft count = 117\n",
      "--- gene=583, id=247 removed by soft count = 226\n",
      "--- gene=583, id=249 removed by soft count = 186\n",
      "--- number of probes: 271 , kept by blast: 262 , if remove dups: 262\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 584\n",
      "--- gene=584, id=37 removed by soft count = 96\n",
      "--- number of probes: 118 , kept by blast: 117 , if remove dups: 117\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 117\n",
      "-- checking probes in region: 585\n",
      "--- gene=585, id=6 removed by soft count = 61\n",
      "--- gene=585, id=8 removed by soft count = 63\n",
      "--- gene=585, id=10 removed by soft count = 189\n",
      "--- gene=585, id=69 removed by soft count = 325\n",
      "--- gene=585, id=78 removed by soft count = 124\n",
      "--- gene=585, id=81 removed by soft count = 694\n",
      "--- gene=585, id=95 removed by soft count = 154\n",
      "--- gene=585, id=97 removed by soft count = 174\n",
      "--- gene=585, id=100 removed by soft count = 152\n",
      "--- gene=585, id=101 removed by soft count = 89\n",
      "--- gene=585, id=104 removed by soft count = 89\n",
      "--- gene=585, id=106 removed by soft count = 315\n",
      "--- gene=585, id=107 removed by hard count = 3\n",
      "--- gene=585, id=108 removed by hard count = 3\n",
      "--- gene=585, id=109 removed by hard count = 3\n",
      "--- gene=585, id=111 removed by soft count = 410\n",
      "--- gene=585, id=127 removed by soft count = 107\n",
      "--- number of probes: 188 , kept by blast: 171 , if remove dups: 169\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 586\n",
      "--- gene=586, id=66 removed by soft count = 65\n",
      "--- gene=586, id=198 removed by soft count = 59\n",
      "--- gene=586, id=199 removed by soft count = 79\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=586, id=200 removed by soft count = 95\n",
      "--- gene=586, id=281 removed by soft count = 109\n",
      "--- number of probes: 288 , kept by blast: 283 , if remove dups: 283\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 587\n",
      "--- gene=587, id=0 removed by soft count = 3531\n",
      "--- number of probes: 78 , kept by blast: 77 , if remove dups: 77\n",
      "-- number of probes kept for this region: 77\n",
      "-- checking probes in region: 588\n",
      "--- gene=588, id=32 removed by soft count = 59\n",
      "--- gene=588, id=92 removed by soft count = 70\n",
      "--- gene=588, id=128 removed by soft count = 100\n",
      "--- gene=588, id=129 removed by soft count = 317\n",
      "--- gene=588, id=134 removed by soft count = 139\n",
      "--- gene=588, id=153 removed by soft count = 195\n",
      "--- gene=588, id=159 removed by soft count = 103\n",
      "--- gene=588, id=178 removed by soft count = 81\n",
      "--- number of probes: 213 , kept by blast: 205 , if remove dups: 205\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 589\n",
      "--- gene=589, id=19 removed by soft count = 499\n",
      "--- gene=589, id=45 removed by soft count = 163\n",
      "--- gene=589, id=136 removed by soft count = 230\n",
      "--- gene=589, id=158 removed by soft count = 79\n",
      "--- number of probes: 226 , kept by blast: 222 , if remove dups: 222\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 590\n",
      "--- gene=590, id=121 removed by soft count = 127\n",
      "--- number of probes: 235 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 591\n",
      "--- gene=591, id=64 removed by soft count = 225\n",
      "--- gene=591, id=66 removed by soft count = 161\n",
      "--- gene=591, id=68 removed by soft count = 107\n",
      "--- gene=591, id=70 removed by soft count = 150\n",
      "--- gene=591, id=76 removed by soft count = 102\n",
      "--- gene=591, id=145 removed by soft count = 98\n",
      "--- gene=591, id=146 removed by soft count = 162\n",
      "--- gene=591, id=147 removed by soft count = 736\n",
      "--- gene=591, id=204 removed by soft count = 88\n",
      "--- number of probes: 222 , kept by blast: 213 , if remove dups: 213\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 592\n",
      "--- gene=592, id=29 removed by soft count = 54\n",
      "--- gene=592, id=50 removed by soft count = 52\n",
      "--- gene=592, id=146 removed by soft count = 82\n",
      "--- number of probes: 253 , kept by blast: 250 , if remove dups: 250\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 593\n",
      "--- gene=593, id=14 removed by soft count = 118\n",
      "--- gene=593, id=35 removed by soft count = 104\n",
      "--- gene=593, id=212 removed by soft count = 182\n",
      "--- gene=593, id=231 removed by soft count = 52\n",
      "--- gene=593, id=233 removed by soft count = 147\n",
      "--- gene=593, id=237 removed by soft count = 92\n",
      "--- gene=593, id=244 removed by soft count = 64\n",
      "--- number of probes: 251 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 594\n",
      "--- gene=594, id=45 removed by soft count = 150\n",
      "--- number of probes: 319 , kept by blast: 318 , if remove dups: 318\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 595\n",
      "--- gene=595, id=34 removed by soft count = 2631\n",
      "--- gene=595, id=35 removed by soft count = 113\n",
      "--- gene=595, id=168 removed by soft count = 1629\n",
      "--- gene=595, id=183 removed by soft count = 177\n",
      "--- number of probes: 274 , kept by blast: 270 , if remove dups: 270\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 596\n",
      "--- gene=596, id=99 removed by soft count = 108\n",
      "--- gene=596, id=302 removed by soft count = 114\n",
      "--- number of probes: 304 , kept by blast: 302 , if remove dups: 302\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 597\n",
      "--- gene=597, id=62 removed by soft count = 424\n",
      "--- gene=597, id=97 removed by soft count = 233\n",
      "--- gene=597, id=185 removed by soft count = 55\n",
      "--- gene=597, id=235 removed by soft count = 93\n",
      "--- gene=597, id=237 removed by soft count = 108\n",
      "--- number of probes: 245 , kept by blast: 240 , if remove dups: 240\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 598\n",
      "--- gene=598, id=134 removed by soft count = 252\n",
      "--- number of probes: 160 , kept by blast: 159 , if remove dups: 159\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 599\n",
      "--- gene=599, id=41 removed by soft count = 109\n",
      "--- gene=599, id=59 removed by soft count = 82\n",
      "--- gene=599, id=145 removed by soft count = 1263\n",
      "--- gene=599, id=154 removed by soft count = 64\n",
      "--- number of probes: 258 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 600\n",
      "--- gene=600, id=140 removed by soft count = 3163\n",
      "--- gene=600, id=219 removed by soft count = 56\n",
      "--- gene=600, id=225 removed by soft count = 64\n",
      "--- number of probes: 253 , kept by blast: 250 , if remove dups: 250\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 601\n",
      "--- gene=601, id=34 removed by soft count = 84\n",
      "--- gene=601, id=146 removed by soft count = 83\n",
      "--- gene=601, id=231 removed by soft count = 389\n",
      "--- gene=601, id=232 removed by soft count = 381\n",
      "--- number of probes: 266 , kept by blast: 262 , if remove dups: 262\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 602\n",
      "--- gene=602, id=41 removed by soft count = 56\n",
      "--- gene=602, id=84 removed by soft count = 73\n",
      "--- gene=602, id=88 removed by soft count = 65\n",
      "--- gene=602, id=99 removed by soft count = 53\n",
      "--- number of probes: 160 , kept by blast: 156 , if remove dups: 156\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 603\n",
      "--- gene=603, id=0 removed by soft count = 88\n",
      "--- gene=603, id=1 removed by soft count = 318\n",
      "--- gene=603, id=2 removed by soft count = 123\n",
      "--- gene=603, id=13 removed by soft count = 84\n",
      "--- gene=603, id=35 removed by soft count = 65\n",
      "--- gene=603, id=68 removed by soft count = 136\n",
      "--- gene=603, id=71 removed by soft count = 110\n",
      "--- gene=603, id=72 removed by soft count = 79\n",
      "--- gene=603, id=75 removed by soft count = 834\n",
      "--- gene=603, id=78 removed by soft count = 109\n",
      "--- gene=603, id=79 removed by soft count = 107\n",
      "--- gene=603, id=82 removed by soft count = 351\n",
      "--- gene=603, id=87 removed by soft count = 71\n",
      "--- gene=603, id=89 removed by soft count = 74\n",
      "--- number of probes: 92 , kept by blast: 78 , if remove dups: 78\n",
      "-- number of probes kept for this region: 78\n",
      "-- checking probes in region: 604\n",
      "--- gene=604, id=51 removed by soft count = 67\n",
      "--- gene=604, id=58 removed by soft count = 384\n",
      "--- gene=604, id=61 removed by soft count = 65\n",
      "--- gene=604, id=66 removed by soft count = 72\n",
      "--- gene=604, id=71 removed by soft count = 84\n",
      "--- gene=604, id=99 removed by soft count = 108\n",
      "--- gene=604, id=101 removed by soft count = 79\n",
      "--- gene=604, id=154 removed by soft count = 60\n",
      "--- gene=604, id=156 removed by soft count = 85\n",
      "--- gene=604, id=157 removed by soft count = 92\n",
      "--- gene=604, id=160 removed by soft count = 162\n",
      "--- gene=604, id=166 removed by soft count = 57\n",
      "--- gene=604, id=179 removed by soft count = 101\n",
      "--- gene=604, id=180 removed by soft count = 159\n",
      "--- number of probes: 187 , kept by blast: 173 , if remove dups: 173\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 605\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=605, id=106 removed by soft count = 146\n",
      "--- gene=605, id=201 removed by soft count = 533\n",
      "--- number of probes: 212 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 606\n",
      "--- gene=606, id=10 removed by soft count = 377\n",
      "--- gene=606, id=12 removed by soft count = 56\n",
      "--- gene=606, id=15 removed by soft count = 94\n",
      "--- gene=606, id=162 removed by soft count = 113\n",
      "--- gene=606, id=172 removed by soft count = 205\n",
      "--- gene=606, id=173 removed by soft count = 983\n",
      "--- gene=606, id=174 removed by soft count = 544\n",
      "--- gene=606, id=183 removed by soft count = 206\n",
      "--- gene=606, id=185 removed by soft count = 54\n",
      "--- gene=606, id=217 removed by soft count = 68\n",
      "--- number of probes: 220 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 607\n",
      "--- gene=607, id=171 removed by soft count = 405\n",
      "--- gene=607, id=176 removed by soft count = 439\n",
      "--- number of probes: 190 , kept by blast: 188 , if remove dups: 188\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 608\n",
      "--- number of probes: 256 , kept by blast: 256 , if remove dups: 256\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 609\n",
      "--- gene=609, id=65 removed by soft count = 69\n",
      "--- gene=609, id=68 removed by soft count = 137\n",
      "--- gene=609, id=69 removed by soft count = 75\n",
      "--- gene=609, id=78 removed by soft count = 141\n",
      "--- gene=609, id=231 removed by hard count = 0\n",
      "--- gene=609, id=261 removed by soft count = 859\n",
      "--- gene=609, id=272 removed by soft count = 53\n",
      "--- number of probes: 285 , kept by blast: 278 , if remove dups: 278\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 610\n",
      "--- gene=610, id=111 removed by soft count = 283\n",
      "--- gene=610, id=170 removed by soft count = 788\n",
      "--- number of probes: 293 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 611\n",
      "--- gene=611, id=150 removed by soft count = 318\n",
      "--- number of probes: 281 , kept by blast: 280 , if remove dups: 280\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 612\n",
      "--- gene=612, id=90 removed by soft count = 59\n",
      "--- number of probes: 277 , kept by blast: 276 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 613\n",
      "--- gene=613, id=195 removed by soft count = 59\n",
      "--- gene=613, id=233 removed by soft count = 106\n",
      "--- number of probes: 236 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 616\n",
      "--- number of probes: 289 , kept by blast: 289 , if remove dups: 289\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 617\n",
      "--- gene=617, id=22 removed by soft count = 134\n",
      "--- gene=617, id=24 removed by soft count = 61\n",
      "--- gene=617, id=25 removed by soft count = 923\n",
      "--- gene=617, id=109 removed by soft count = 51\n",
      "--- number of probes: 158 , kept by blast: 154 , if remove dups: 154\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 618\n",
      "--- gene=618, id=0 removed by soft count = 273\n",
      "--- gene=618, id=1 removed by soft count = 133\n",
      "--- gene=618, id=46 removed by soft count = 162\n",
      "--- gene=618, id=76 removed by soft count = 107\n",
      "--- gene=618, id=96 removed by soft count = 455\n",
      "--- gene=618, id=106 removed by soft count = 65\n",
      "--- gene=618, id=118 removed by soft count = 231\n",
      "--- number of probes: 126 , kept by blast: 119 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 619\n",
      "--- gene=619, id=144 removed by soft count = 58\n",
      "--- gene=619, id=185 removed by soft count = 251\n",
      "--- gene=619, id=187 removed by soft count = 698\n",
      "--- gene=619, id=191 removed by soft count = 798\n",
      "--- gene=619, id=192 removed by soft count = 842\n",
      "--- gene=619, id=194 removed by soft count = 727\n",
      "--- gene=619, id=197 removed by soft count = 820\n",
      "--- gene=619, id=200 removed by soft count = 260\n",
      "--- gene=619, id=201 removed by soft count = 125\n",
      "--- gene=619, id=220 removed by soft count = 237\n",
      "--- gene=619, id=221 removed by soft count = 114\n",
      "--- number of probes: 222 , kept by blast: 211 , if remove dups: 211\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 620\n",
      "--- gene=620, id=71 removed by soft count = 1213\n",
      "--- number of probes: 103 , kept by blast: 102 , if remove dups: 102\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 102\n",
      "-- checking probes in region: 621\n",
      "--- gene=621, id=5 removed by soft count = 638\n",
      "--- gene=621, id=6 removed by soft count = 74\n",
      "--- gene=621, id=9 removed by soft count = 264\n",
      "--- gene=621, id=18 removed by soft count = 83\n",
      "--- gene=621, id=19 removed by soft count = 329\n",
      "--- gene=621, id=25 removed by soft count = 72\n",
      "--- gene=621, id=31 removed by soft count = 111\n",
      "--- gene=621, id=32 removed by soft count = 524\n",
      "--- gene=621, id=33 removed by soft count = 63\n",
      "--- gene=621, id=34 removed by soft count = 73\n",
      "--- gene=621, id=35 removed by soft count = 3322\n",
      "--- gene=621, id=37 removed by soft count = 373\n",
      "--- number of probes: 191 , kept by blast: 179 , if remove dups: 179\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 622\n",
      "--- gene=622, id=4 removed by soft count = 132\n",
      "--- gene=622, id=7 removed by soft count = 79\n",
      "--- gene=622, id=9 removed by soft count = 86\n",
      "--- gene=622, id=15 removed by soft count = 72\n",
      "--- number of probes: 272 , kept by blast: 268 , if remove dups: 268\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 623\n",
      "--- gene=623, id=29 removed by soft count = 127\n",
      "--- gene=623, id=30 removed by soft count = 55\n",
      "--- gene=623, id=49 removed by soft count = 70\n",
      "--- gene=623, id=90 removed by soft count = 176\n",
      "--- gene=623, id=91 removed by soft count = 172\n",
      "--- gene=623, id=98 removed by soft count = 104\n",
      "--- gene=623, id=99 removed by soft count = 74\n",
      "--- gene=623, id=114 removed by soft count = 3221\n",
      "--- gene=623, id=166 removed by soft count = 111\n",
      "--- number of probes: 177 , kept by blast: 168 , if remove dups: 168\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 624\n",
      "--- gene=624, id=76 removed by soft count = 72\n",
      "--- gene=624, id=77 removed by soft count = 268\n",
      "--- gene=624, id=78 removed by soft count = 633\n",
      "--- gene=624, id=94 removed by soft count = 711\n",
      "--- gene=624, id=189 removed by soft count = 153\n",
      "--- number of probes: 215 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 625\n",
      "--- gene=625, id=20 removed by soft count = 831\n",
      "--- gene=625, id=25 removed by soft count = 143\n",
      "--- gene=625, id=26 removed by soft count = 749\n",
      "--- gene=625, id=27 removed by soft count = 107\n",
      "--- gene=625, id=28 removed by soft count = 204\n",
      "--- gene=625, id=29 removed by soft count = 86\n",
      "--- gene=625, id=32 removed by soft count = 479\n",
      "--- gene=625, id=33 removed by soft count = 766\n",
      "--- gene=625, id=89 removed by soft count = 484\n",
      "--- gene=625, id=94 removed by soft count = 125\n",
      "--- gene=625, id=96 removed by soft count = 66\n",
      "--- gene=625, id=98 removed by soft count = 52\n",
      "--- number of probes: 107 , kept by blast: 95 , if remove dups: 95\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 95\n",
      "-- checking probes in region: 626\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- number of probes: 175 , kept by blast: 175 , if remove dups: 175\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 627\n",
      "--- gene=627, id=10 removed by soft count = 498\n",
      "--- gene=627, id=12 removed by soft count = 10436\n",
      "--- gene=627, id=13 removed by soft count = 82\n",
      "--- gene=627, id=14 removed by soft count = 428\n",
      "--- gene=627, id=15 removed by soft count = 120\n",
      "--- gene=627, id=41 removed by soft count = 323\n",
      "--- gene=627, id=43 removed by soft count = 55\n",
      "--- gene=627, id=45 removed by soft count = 74\n",
      "--- gene=627, id=48 removed by soft count = 389\n",
      "--- gene=627, id=49 removed by soft count = 133\n",
      "--- gene=627, id=51 removed by soft count = 700\n",
      "--- gene=627, id=53 removed by soft count = 70\n",
      "--- gene=627, id=72 removed by soft count = 101\n",
      "--- gene=627, id=73 removed by soft count = 331\n",
      "--- number of probes: 75 , kept by blast: 61 , if remove dups: 61\n",
      "-- number of probes kept for this region: 61\n",
      "-- checking probes in region: 628\n",
      "--- number of probes: 332 , kept by blast: 332 , if remove dups: 332\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 629\n",
      "--- number of probes: 315 , kept by blast: 315 , if remove dups: 315\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 630\n",
      "--- gene=630, id=7 removed by soft count = 1485\n",
      "--- gene=630, id=9 removed by soft count = 136\n",
      "--- gene=630, id=98 removed by soft count = 648\n",
      "--- gene=630, id=99 removed by soft count = 1750\n",
      "--- number of probes: 192 , kept by blast: 188 , if remove dups: 188\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 631\n",
      "--- gene=631, id=57 removed by soft count = 868\n",
      "--- gene=631, id=119 removed by soft count = 66\n",
      "--- gene=631, id=152 removed by soft count = 181\n",
      "--- gene=631, id=154 removed by soft count = 57\n",
      "--- number of probes: 205 , kept by blast: 201 , if remove dups: 201\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 632\n",
      "--- gene=632, id=40 removed by soft count = 51\n",
      "--- gene=632, id=64 removed by soft count = 83\n",
      "--- gene=632, id=112 removed by soft count = 266\n",
      "--- gene=632, id=113 removed by soft count = 158\n",
      "--- gene=632, id=137 removed by soft count = 67\n",
      "--- gene=632, id=147 removed by soft count = 209\n",
      "--- gene=632, id=160 removed by soft count = 56\n",
      "--- number of probes: 213 , kept by blast: 206 , if remove dups: 206\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 633\n",
      "--- gene=633, id=134 removed by soft count = 103\n",
      "--- gene=633, id=137 removed by soft count = 207\n",
      "--- gene=633, id=246 removed by soft count = 83\n",
      "--- number of probes: 296 , kept by blast: 293 , if remove dups: 293\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 634\n",
      "--- gene=634, id=141 removed by soft count = 91\n",
      "--- number of probes: 168 , kept by blast: 167 , if remove dups: 167\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 635\n",
      "--- gene=635, id=38 removed by soft count = 58\n",
      "--- gene=635, id=57 removed by soft count = 746\n",
      "--- gene=635, id=58 removed by soft count = 80\n",
      "--- gene=635, id=60 removed by soft count = 130\n",
      "--- gene=635, id=127 removed by soft count = 69\n",
      "--- gene=635, id=128 removed by soft count = 156\n",
      "--- gene=635, id=131 removed by soft count = 68\n",
      "--- gene=635, id=137 removed by soft count = 70\n",
      "--- number of probes: 234 , kept by blast: 226 , if remove dups: 226\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 636\n",
      "--- gene=636, id=2 removed by soft count = 385\n",
      "--- gene=636, id=3 removed by soft count = 111\n",
      "--- gene=636, id=5 removed by soft count = 76\n",
      "--- gene=636, id=8 removed by soft count = 67\n",
      "--- gene=636, id=14 removed by soft count = 205\n",
      "--- gene=636, id=16 removed by soft count = 329\n",
      "--- gene=636, id=17 removed by soft count = 58\n",
      "--- gene=636, id=22 removed by soft count = 147\n",
      "--- gene=636, id=35 removed by soft count = 103\n",
      "--- gene=636, id=47 removed by soft count = 62\n",
      "--- gene=636, id=83 removed by soft count = 75\n",
      "--- gene=636, id=85 removed by soft count = 147\n",
      "--- gene=636, id=86 removed by soft count = 117\n",
      "--- gene=636, id=126 removed by soft count = 194\n",
      "--- number of probes: 168 , kept by blast: 154 , if remove dups: 154\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 637\n",
      "--- gene=637, id=115 removed by soft count = 100\n",
      "--- number of probes: 255 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 638\n",
      "--- gene=638, id=16 removed by soft count = 82\n",
      "--- gene=638, id=80 removed by soft count = 129\n",
      "--- number of probes: 159 , kept by blast: 157 , if remove dups: 157\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 639\n",
      "--- gene=639, id=10 removed by soft count = 177\n",
      "--- gene=639, id=216 removed by soft count = 164\n",
      "--- gene=639, id=217 removed by soft count = 108\n",
      "--- gene=639, id=218 removed by soft count = 2061\n",
      "--- gene=639, id=222 removed by soft count = 220\n",
      "--- gene=639, id=223 removed by soft count = 2022\n",
      "--- gene=639, id=263 removed by soft count = 91\n",
      "--- gene=639, id=264 removed by soft count = 882\n",
      "--- number of probes: 283 , kept by blast: 275 , if remove dups: 275\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 640\n",
      "--- gene=640, id=29 removed by soft count = 137\n",
      "--- gene=640, id=73 removed by soft count = 1175\n",
      "--- gene=640, id=90 removed by soft count = 113\n",
      "--- gene=640, id=285 removed by soft count = 61\n",
      "--- number of probes: 286 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 641\n",
      "--- gene=641, id=34 removed by soft count = 4319\n",
      "--- gene=641, id=35 removed by soft count = 86\n",
      "--- gene=641, id=37 removed by soft count = 130\n",
      "--- gene=641, id=38 removed by soft count = 148\n",
      "--- gene=641, id=86 removed by soft count = 139\n",
      "--- gene=641, id=87 removed by soft count = 182\n",
      "--- gene=641, id=242 removed by soft count = 54\n",
      "--- number of probes: 263 , kept by blast: 256 , if remove dups: 256\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 642\n",
      "--- gene=642, id=10 removed by soft count = 105\n",
      "--- gene=642, id=45 removed by soft count = 223\n",
      "--- gene=642, id=68 removed by soft count = 197\n",
      "--- gene=642, id=255 removed by soft count = 117\n",
      "--- number of probes: 285 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 643\n",
      "--- gene=643, id=22 removed by soft count = 893\n",
      "--- gene=643, id=129 removed by soft count = 94\n",
      "--- gene=643, id=152 removed by soft count = 105\n",
      "--- number of probes: 272 , kept by blast: 269 , if remove dups: 269\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 644\n",
      "--- gene=644, id=10 removed by soft count = 85\n",
      "--- gene=644, id=24 removed by soft count = 142\n",
      "--- gene=644, id=57 removed by soft count = 65\n",
      "--- gene=644, id=59 removed by soft count = 83\n",
      "--- gene=644, id=60 removed by soft count = 482\n",
      "--- gene=644, id=61 removed by soft count = 861\n",
      "--- gene=644, id=62 removed by soft count = 85\n",
      "--- gene=644, id=75 removed by soft count = 511\n",
      "--- gene=644, id=122 removed by soft count = 412\n",
      "--- number of probes: 131 , kept by blast: 122 , if remove dups: 122\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 645\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=645, id=18 removed by soft count = 58\n",
      "--- gene=645, id=95 removed by soft count = 80\n",
      "--- number of probes: 303 , kept by blast: 301 , if remove dups: 301\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 646\n",
      "--- gene=646, id=5 removed by soft count = 65\n",
      "--- gene=646, id=10 removed by soft count = 52\n",
      "--- gene=646, id=11 removed by soft count = 146\n",
      "--- gene=646, id=83 removed by soft count = 52\n",
      "--- gene=646, id=120 removed by soft count = 151\n",
      "--- gene=646, id=184 removed by soft count = 52\n",
      "--- number of probes: 234 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 647\n",
      "--- gene=647, id=135 removed by soft count = 393\n",
      "--- gene=647, id=156 removed by soft count = 106\n",
      "--- gene=647, id=184 removed by soft count = 82\n",
      "--- gene=647, id=284 removed by soft count = 63\n",
      "--- number of probes: 285 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 648\n",
      "--- gene=648, id=49 removed by soft count = 478\n",
      "--- gene=648, id=144 removed by soft count = 98\n",
      "--- gene=648, id=146 removed by soft count = 149\n",
      "--- gene=648, id=147 removed by soft count = 61\n",
      "--- gene=648, id=148 removed by soft count = 89\n",
      "--- gene=648, id=219 removed by soft count = 78\n",
      "--- number of probes: 252 , kept by blast: 246 , if remove dups: 245\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 649\n",
      "--- gene=649, id=140 removed by hard count = 0\n",
      "--- number of probes: 228 , kept by blast: 227 , if remove dups: 227\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 650\n",
      "--- gene=650, id=52 removed by soft count = 312\n",
      "--- gene=650, id=86 removed by soft count = 72\n",
      "--- gene=650, id=234 removed by soft count = 90\n",
      "--- number of probes: 260 , kept by blast: 257 , if remove dups: 257\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 651\n",
      "--- gene=651, id=22 removed by soft count = 110\n",
      "--- gene=651, id=25 removed by soft count = 160\n",
      "--- gene=651, id=228 removed by soft count = 76\n",
      "--- gene=651, id=235 removed by soft count = 1170\n",
      "--- gene=651, id=236 removed by soft count = 93\n",
      "--- gene=651, id=237 removed by soft count = 973\n",
      "--- number of probes: 239 , kept by blast: 233 , if remove dups: 233\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 652\n",
      "--- gene=652, id=7 removed by soft count = 274\n",
      "--- gene=652, id=28 removed by soft count = 105\n",
      "--- gene=652, id=29 removed by soft count = 129\n",
      "--- gene=652, id=114 removed by soft count = 121\n",
      "--- gene=652, id=161 removed by soft count = 429\n",
      "--- gene=652, id=195 removed by soft count = 71\n",
      "--- number of probes: 211 , kept by blast: 205 , if remove dups: 205\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 653\n",
      "--- gene=653, id=48 removed by soft count = 87\n",
      "--- gene=653, id=57 removed by soft count = 90\n",
      "--- number of probes: 233 , kept by blast: 231 , if remove dups: 231\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 654\n",
      "--- gene=654, id=213 removed by hard count = 0\n",
      "--- number of probes: 308 , kept by blast: 307 , if remove dups: 307\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 655\n",
      "--- gene=655, id=6 removed by soft count = 75\n",
      "--- number of probes: 251 , kept by blast: 250 , if remove dups: 250\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 656\n",
      "--- gene=656, id=141 removed by soft count = 107\n",
      "--- gene=656, id=263 removed by soft count = 414\n",
      "--- number of probes: 320 , kept by blast: 318 , if remove dups: 318\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 657\n",
      "--- gene=657, id=3 removed by soft count = 311\n",
      "--- gene=657, id=7 removed by soft count = 58\n",
      "--- gene=657, id=271 removed by soft count = 61\n",
      "--- gene=657, id=273 removed by soft count = 175\n",
      "--- number of probes: 290 , kept by blast: 286 , if remove dups: 286\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 658\n",
      "--- gene=658, id=18 removed by soft count = 59\n",
      "--- gene=658, id=25 removed by soft count = 54\n",
      "--- gene=658, id=26 removed by soft count = 178\n",
      "--- gene=658, id=27 removed by soft count = 65\n",
      "--- gene=658, id=223 removed by soft count = 788\n",
      "--- number of probes: 269 , kept by blast: 264 , if remove dups: 264\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 659\n",
      "--- gene=659, id=58 removed by soft count = 15218\n",
      "--- number of probes: 257 , kept by blast: 256 , if remove dups: 256\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 660\n",
      "--- gene=660, id=199 removed by soft count = 148\n",
      "--- gene=660, id=254 removed by soft count = 97\n",
      "--- number of probes: 284 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 661\n",
      "--- gene=661, id=14 removed by soft count = 117\n",
      "--- gene=661, id=201 removed by soft count = 119\n",
      "--- gene=661, id=251 removed by soft count = 3222\n",
      "--- number of probes: 284 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 662\n",
      "--- gene=662, id=177 removed by soft count = 64\n",
      "--- number of probes: 322 , kept by blast: 321 , if remove dups: 321\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 663\n",
      "--- gene=663, id=1 removed by soft count = 347\n",
      "--- gene=663, id=37 removed by soft count = 204\n",
      "--- gene=663, id=66 removed by soft count = 212\n",
      "--- gene=663, id=96 removed by soft count = 296\n",
      "--- gene=663, id=222 removed by soft count = 240\n",
      "--- gene=663, id=246 removed by soft count = 114\n",
      "--- number of probes: 271 , kept by blast: 265 , if remove dups: 264\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 665\n",
      "--- number of probes: 270 , kept by blast: 270 , if remove dups: 270\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 666\n",
      "--- gene=666, id=0 removed by soft count = 713\n",
      "--- gene=666, id=46 removed by soft count = 270\n",
      "--- gene=666, id=52 removed by soft count = 155\n",
      "--- number of probes: 177 , kept by blast: 174 , if remove dups: 174\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 667\n",
      "--- gene=667, id=143 removed by soft count = 122\n",
      "--- gene=667, id=224 removed by soft count = 56\n",
      "--- number of probes: 241 , kept by blast: 239 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 668\n",
      "--- gene=668, id=60 removed by soft count = 817\n",
      "--- gene=668, id=67 removed by soft count = 58\n",
      "--- gene=668, id=100 removed by soft count = 52\n",
      "--- gene=668, id=115 removed by soft count = 60\n",
      "--- gene=668, id=175 removed by soft count = 108\n",
      "--- number of probes: 196 , kept by blast: 191 , if remove dups: 191\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 669\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=669, id=119 removed by soft count = 62\n",
      "--- gene=669, id=187 removed by soft count = 177\n",
      "--- gene=669, id=188 removed by soft count = 253\n",
      "--- gene=669, id=189 removed by soft count = 294\n",
      "--- gene=669, id=190 removed by soft count = 211\n",
      "--- gene=669, id=191 removed by soft count = 85\n",
      "--- number of probes: 239 , kept by blast: 233 , if remove dups: 233\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 670\n",
      "--- gene=670, id=81 removed by soft count = 90\n",
      "--- gene=670, id=223 removed by soft count = 66\n",
      "--- gene=670, id=232 removed by soft count = 87\n",
      "--- number of probes: 273 , kept by blast: 270 , if remove dups: 269\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 672\n",
      "--- gene=672, id=13 removed by soft count = 67\n",
      "--- gene=672, id=74 removed by soft count = 75\n",
      "--- gene=672, id=128 removed by soft count = 383\n",
      "--- gene=672, id=129 removed by soft count = 424\n",
      "--- gene=672, id=153 removed by soft count = 88\n",
      "--- number of probes: 280 , kept by blast: 275 , if remove dups: 275\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 674\n",
      "--- gene=674, id=141 removed by soft count = 69\n",
      "--- gene=674, id=192 removed by soft count = 215\n",
      "--- gene=674, id=195 removed by soft count = 110\n",
      "--- number of probes: 268 , kept by blast: 265 , if remove dups: 265\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 675\n",
      "--- gene=675, id=50 removed by soft count = 1111\n",
      "--- gene=675, id=51 removed by soft count = 149\n",
      "--- gene=675, id=193 removed by soft count = 314\n",
      "--- gene=675, id=204 removed by soft count = 66\n",
      "--- number of probes: 251 , kept by blast: 247 , if remove dups: 247\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 676\n",
      "--- gene=676, id=31 removed by soft count = 94\n",
      "--- gene=676, id=58 removed by soft count = 82\n",
      "--- gene=676, id=152 removed by soft count = 84\n",
      "--- number of probes: 246 , kept by blast: 243 , if remove dups: 243\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 677\n",
      "--- gene=677, id=6 removed by soft count = 78\n",
      "--- gene=677, id=77 removed by soft count = 219\n",
      "--- gene=677, id=91 removed by soft count = 62\n",
      "--- gene=677, id=92 removed by soft count = 60\n",
      "--- number of probes: 106 , kept by blast: 102 , if remove dups: 102\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 102\n",
      "-- checking probes in region: 678\n",
      "--- gene=678, id=273 removed by soft count = 84\n",
      "--- number of probes: 275 , kept by blast: 274 , if remove dups: 274\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 679\n",
      "--- gene=679, id=73 removed by soft count = 52\n",
      "--- number of probes: 186 , kept by blast: 185 , if remove dups: 185\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 680\n",
      "--- gene=680, id=97 removed by soft count = 124\n",
      "--- gene=680, id=188 removed by soft count = 523\n",
      "--- number of probes: 199 , kept by blast: 197 , if remove dups: 197\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 681\n",
      "--- gene=681, id=15 removed by soft count = 98\n",
      "--- gene=681, id=16 removed by soft count = 175\n",
      "--- gene=681, id=17 removed by soft count = 255\n",
      "--- gene=681, id=19 removed by soft count = 3077\n",
      "--- gene=681, id=20 removed by soft count = 1166\n",
      "--- gene=681, id=53 removed by soft count = 113\n",
      "--- number of probes: 59 , kept by blast: 53 , if remove dups: 53\n",
      "-- number of probes kept for this region: 53\n",
      "-- checking probes in region: 682\n",
      "--- gene=682, id=100 removed by soft count = 11532\n",
      "--- number of probes: 138 , kept by blast: 137 , if remove dups: 137\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 683\n",
      "--- gene=683, id=0 removed by soft count = 659\n",
      "--- gene=683, id=1 removed by soft count = 6480\n",
      "--- gene=683, id=3 removed by soft count = 257\n",
      "--- gene=683, id=69 removed by soft count = 143\n",
      "--- gene=683, id=140 removed by soft count = 590\n",
      "--- gene=683, id=144 removed by soft count = 151\n",
      "--- gene=683, id=145 removed by soft count = 70\n",
      "--- number of probes: 163 , kept by blast: 156 , if remove dups: 156\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 684\n",
      "--- number of probes: 231 , kept by blast: 231 , if remove dups: 231\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 685\n",
      "--- gene=685, id=40 removed by soft count = 154\n",
      "--- gene=685, id=246 removed by soft count = 85\n",
      "--- gene=685, id=247 removed by soft count = 92\n",
      "--- gene=685, id=248 removed by soft count = 245\n",
      "--- number of probes: 282 , kept by blast: 278 , if remove dups: 278\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 686\n",
      "--- gene=686, id=98 removed by soft count = 218\n",
      "--- gene=686, id=145 removed by soft count = 246\n",
      "--- number of probes: 268 , kept by blast: 266 , if remove dups: 266\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 687\n",
      "--- gene=687, id=62 removed by soft count = 311\n",
      "--- gene=687, id=84 removed by soft count = 457\n",
      "--- gene=687, id=162 removed by soft count = 53\n",
      "--- gene=687, id=169 removed by soft count = 599\n",
      "--- gene=687, id=171 removed by soft count = 152\n",
      "--- number of probes: 192 , kept by blast: 187 , if remove dups: 187\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 688\n",
      "--- gene=688, id=19 removed by soft count = 686\n",
      "--- gene=688, id=21 removed by soft count = 110\n",
      "--- gene=688, id=22 removed by soft count = 65\n",
      "--- gene=688, id=24 removed by soft count = 69\n",
      "--- gene=688, id=25 removed by soft count = 85\n",
      "--- gene=688, id=54 removed by soft count = 153\n",
      "--- gene=688, id=60 removed by soft count = 79\n",
      "--- gene=688, id=61 removed by soft count = 62\n",
      "--- gene=688, id=62 removed by soft count = 136\n",
      "--- gene=688, id=86 removed by soft count = 161\n",
      "--- gene=688, id=87 removed by soft count = 94\n",
      "--- gene=688, id=120 removed by soft count = 13281\n",
      "--- gene=688, id=124 removed by soft count = 126\n",
      "--- gene=688, id=125 removed by soft count = 59\n",
      "--- gene=688, id=137 removed by soft count = 93\n",
      "--- gene=688, id=141 removed by soft count = 107\n",
      "--- number of probes: 172 , kept by blast: 156 , if remove dups: 156\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 689\n",
      "--- gene=689, id=1 removed by soft count = 258\n",
      "--- gene=689, id=3 removed by soft count = 712\n",
      "--- gene=689, id=4 removed by soft count = 1053\n",
      "--- gene=689, id=20 removed by soft count = 74\n",
      "--- gene=689, id=37 removed by soft count = 90\n",
      "--- number of probes: 115 , kept by blast: 110 , if remove dups: 110\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 110\n",
      "-- checking probes in region: 690\n",
      "--- gene=690, id=55 removed by soft count = 510\n",
      "--- gene=690, id=57 removed by soft count = 55\n",
      "--- gene=690, id=61 removed by soft count = 626\n",
      "--- gene=690, id=63 removed by soft count = 2104\n",
      "--- gene=690, id=64 removed by soft count = 87\n",
      "--- gene=690, id=65 removed by soft count = 66\n",
      "--- gene=690, id=67 removed by soft count = 448\n",
      "--- gene=690, id=84 removed by soft count = 94\n",
      "--- gene=690, id=93 removed by soft count = 63\n",
      "--- number of probes: 128 , kept by blast: 119 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 691\n",
      "--- gene=691, id=0 removed by soft count = 437\n",
      "--- gene=691, id=1 removed by soft count = 276\n",
      "--- gene=691, id=2 removed by soft count = 89\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=691, id=3 removed by soft count = 146\n",
      "--- gene=691, id=7 removed by soft count = 112\n",
      "--- gene=691, id=8 removed by soft count = 188\n",
      "--- gene=691, id=9 removed by soft count = 53\n",
      "--- number of probes: 140 , kept by blast: 133 , if remove dups: 133\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 692\n",
      "--- gene=692, id=84 removed by soft count = 74\n",
      "--- gene=692, id=86 removed by soft count = 68\n",
      "--- gene=692, id=87 removed by soft count = 219\n",
      "--- gene=692, id=237 removed by soft count = 138\n",
      "--- number of probes: 274 , kept by blast: 270 , if remove dups: 270\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 693\n",
      "--- gene=693, id=173 removed by soft count = 58\n",
      "--- number of probes: 311 , kept by blast: 310 , if remove dups: 310\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 694\n",
      "--- gene=694, id=119 removed by soft count = 54\n",
      "--- gene=694, id=183 removed by soft count = 101\n",
      "--- gene=694, id=192 removed by soft count = 86\n",
      "--- number of probes: 194 , kept by blast: 191 , if remove dups: 191\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 695\n",
      "--- gene=695, id=33 removed by soft count = 159\n",
      "--- gene=695, id=179 removed by soft count = 113\n",
      "--- gene=695, id=181 removed by soft count = 1219\n",
      "--- gene=695, id=207 removed by soft count = 69\n",
      "--- number of probes: 211 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 696\n",
      "--- gene=696, id=106 removed by soft count = 52\n",
      "--- gene=696, id=168 removed by soft count = 69\n",
      "--- gene=696, id=169 removed by soft count = 75\n",
      "--- gene=696, id=211 removed by soft count = 182\n",
      "--- number of probes: 266 , kept by blast: 262 , if remove dups: 262\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 697\n",
      "--- gene=697, id=103 removed by soft count = 51\n",
      "--- gene=697, id=124 removed by soft count = 164\n",
      "--- gene=697, id=182 removed by soft count = 106\n",
      "--- gene=697, id=216 removed by soft count = 126\n",
      "--- number of probes: 242 , kept by blast: 238 , if remove dups: 238\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 698\n",
      "--- gene=698, id=3 removed by soft count = 752\n",
      "--- gene=698, id=4 removed by soft count = 340\n",
      "--- gene=698, id=171 removed by soft count = 134\n",
      "--- gene=698, id=221 removed by soft count = 54\n",
      "--- gene=698, id=222 removed by soft count = 63\n",
      "--- number of probes: 255 , kept by blast: 250 , if remove dups: 250\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 699\n",
      "--- gene=699, id=139 removed by soft count = 57\n",
      "--- number of probes: 289 , kept by blast: 288 , if remove dups: 288\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 700\n",
      "--- gene=700, id=75 removed by soft count = 75\n",
      "--- number of probes: 167 , kept by blast: 166 , if remove dups: 166\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 701\n",
      "--- gene=701, id=27 removed by soft count = 262\n",
      "--- gene=701, id=28 removed by soft count = 184\n",
      "--- gene=701, id=92 removed by soft count = 175\n",
      "--- gene=701, id=99 removed by soft count = 65\n",
      "--- gene=701, id=102 removed by soft count = 61\n",
      "--- gene=701, id=104 removed by soft count = 122\n",
      "--- gene=701, id=108 removed by soft count = 3893\n",
      "--- gene=701, id=111 removed by soft count = 140\n",
      "--- number of probes: 147 , kept by blast: 139 , if remove dups: 139\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 702\n",
      "--- gene=702, id=37 removed by soft count = 78\n",
      "--- gene=702, id=39 removed by soft count = 595\n",
      "--- gene=702, id=40 removed by soft count = 70\n",
      "--- gene=702, id=41 removed by soft count = 341\n",
      "--- gene=702, id=42 removed by soft count = 60\n",
      "--- gene=702, id=43 removed by soft count = 89\n",
      "--- gene=702, id=77 removed by soft count = 552\n",
      "--- gene=702, id=87 removed by soft count = 544\n",
      "--- gene=702, id=108 removed by soft count = 58\n",
      "--- number of probes: 128 , kept by blast: 119 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 703\n",
      "--- gene=703, id=51 removed by soft count = 51\n",
      "--- number of probes: 246 , kept by blast: 245 , if remove dups: 245\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 706\n",
      "--- gene=706, id=215 removed by soft count = 659\n",
      "--- gene=706, id=228 removed by soft count = 88\n",
      "--- number of probes: 279 , kept by blast: 277 , if remove dups: 277\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 707\n",
      "--- gene=707, id=27 removed by soft count = 817\n",
      "--- gene=707, id=28 removed by soft count = 284\n",
      "--- gene=707, id=29 removed by soft count = 94\n",
      "--- gene=707, id=30 removed by soft count = 103\n",
      "--- gene=707, id=36 removed by soft count = 231\n",
      "--- gene=707, id=37 removed by soft count = 80\n",
      "--- gene=707, id=38 removed by soft count = 161\n",
      "--- gene=707, id=149 removed by soft count = 128\n",
      "--- gene=707, id=181 removed by soft count = 80\n",
      "--- gene=707, id=201 removed by soft count = 338\n",
      "--- number of probes: 222 , kept by blast: 212 , if remove dups: 212\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 708\n",
      "--- gene=708, id=212 removed by soft count = 70\n",
      "--- number of probes: 288 , kept by blast: 287 , if remove dups: 287\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 709\n",
      "--- gene=709, id=12 removed by soft count = 1528\n",
      "--- gene=709, id=16 removed by soft count = 77\n",
      "--- gene=709, id=19 removed by soft count = 319\n",
      "--- gene=709, id=20 removed by soft count = 55\n",
      "--- gene=709, id=29 removed by soft count = 54\n",
      "--- gene=709, id=53 removed by soft count = 503\n",
      "--- gene=709, id=65 removed by soft count = 801\n",
      "--- gene=709, id=66 removed by soft count = 133\n",
      "--- gene=709, id=108 removed by soft count = 2735\n",
      "--- gene=709, id=109 removed by soft count = 1893\n",
      "--- gene=709, id=110 removed by soft count = 1555\n",
      "--- gene=709, id=111 removed by soft count = 63\n",
      "--- number of probes: 112 , kept by blast: 100 , if remove dups: 100\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 100\n",
      "-- checking probes in region: 710\n",
      "--- gene=710, id=123 removed by soft count = 94\n",
      "--- gene=710, id=125 removed by soft count = 95\n",
      "--- gene=710, id=126 removed by hard count = 4\n",
      "--- gene=710, id=127 removed by soft count = 112\n",
      "--- number of probes: 214 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 711\n",
      "--- gene=711, id=26 removed by soft count = 59\n",
      "--- number of probes: 244 , kept by blast: 243 , if remove dups: 243\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 713\n",
      "--- gene=713, id=87 removed by soft count = 139\n",
      "--- gene=713, id=136 removed by soft count = 56\n",
      "--- gene=713, id=168 removed by soft count = 129\n",
      "--- gene=713, id=169 removed by soft count = 64\n",
      "--- gene=713, id=170 removed by soft count = 694\n",
      "--- gene=713, id=181 removed by soft count = 99\n",
      "--- number of probes: 288 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 714\n",
      "--- gene=714, id=59 removed by soft count = 68\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- number of probes: 268 , kept by blast: 267 , if remove dups: 267\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 715\n",
      "--- gene=715, id=60 removed by soft count = 216\n",
      "--- gene=715, id=69 removed by soft count = 97\n",
      "--- gene=715, id=70 removed by soft count = 91\n",
      "--- gene=715, id=177 removed by soft count = 51\n",
      "--- gene=715, id=178 removed by soft count = 79\n",
      "--- gene=715, id=233 removed by soft count = 61\n",
      "--- number of probes: 245 , kept by blast: 239 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 716\n",
      "--- gene=716, id=5 removed by soft count = 125\n",
      "--- gene=716, id=7 removed by soft count = 59\n",
      "--- gene=716, id=21 removed by soft count = 60\n",
      "--- gene=716, id=71 removed by soft count = 416\n",
      "--- gene=716, id=74 removed by soft count = 2238\n",
      "--- gene=716, id=76 removed by soft count = 375\n",
      "--- gene=716, id=95 removed by soft count = 160\n",
      "--- gene=716, id=96 removed by soft count = 121\n",
      "--- gene=716, id=97 removed by soft count = 149\n",
      "--- gene=716, id=99 removed by soft count = 202\n",
      "--- gene=716, id=100 removed by soft count = 125\n",
      "--- number of probes: 101 , kept by blast: 90 , if remove dups: 90\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 90\n",
      "-- checking probes in region: 717\n",
      "--- gene=717, id=116 removed by soft count = 61\n",
      "--- gene=717, id=119 removed by soft count = 172\n",
      "--- gene=717, id=120 removed by soft count = 63\n",
      "--- gene=717, id=125 removed by soft count = 204\n",
      "--- gene=717, id=159 removed by soft count = 79\n",
      "--- gene=717, id=160 removed by hard count = 4\n",
      "--- number of probes: 223 , kept by blast: 217 , if remove dups: 217\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 718\n",
      "--- gene=718, id=62 removed by soft count = 158\n",
      "--- gene=718, id=71 removed by soft count = 99\n",
      "--- gene=718, id=73 removed by soft count = 67\n",
      "--- number of probes: 123 , kept by blast: 120 , if remove dups: 120\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 719\n",
      "--- gene=719, id=107 removed by soft count = 100\n",
      "--- gene=719, id=129 removed by soft count = 54\n",
      "--- gene=719, id=150 removed by soft count = 157\n",
      "--- number of probes: 239 , kept by blast: 236 , if remove dups: 236\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 720\n",
      "--- gene=720, id=240 removed by soft count = 54\n",
      "--- number of probes: 268 , kept by blast: 267 , if remove dups: 267\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 721\n",
      "--- gene=721, id=104 removed by soft count = 73\n",
      "--- gene=721, id=176 removed by soft count = 140\n",
      "--- gene=721, id=177 removed by soft count = 1191\n",
      "--- number of probes: 225 , kept by blast: 222 , if remove dups: 222\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 722\n",
      "--- gene=722, id=133 removed by soft count = 55\n",
      "--- number of probes: 136 , kept by blast: 135 , if remove dups: 135\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 723\n",
      "--- gene=723, id=42 removed by soft count = 329\n",
      "--- number of probes: 265 , kept by blast: 264 , if remove dups: 264\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 724\n",
      "--- gene=724, id=29 removed by soft count = 52\n",
      "--- gene=724, id=31 removed by soft count = 446\n",
      "--- gene=724, id=32 removed by soft count = 510\n",
      "--- gene=724, id=165 removed by soft count = 55\n",
      "--- number of probes: 276 , kept by blast: 272 , if remove dups: 271\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 726\n",
      "--- gene=726, id=167 removed by soft count = 142\n",
      "--- gene=726, id=168 removed by soft count = 69\n",
      "--- gene=726, id=170 removed by soft count = 103\n",
      "--- gene=726, id=171 removed by soft count = 67\n",
      "--- gene=726, id=172 removed by soft count = 353\n",
      "--- number of probes: 252 , kept by blast: 247 , if remove dups: 247\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 727\n",
      "--- gene=727, id=44 removed by soft count = 137\n",
      "--- number of probes: 279 , kept by blast: 278 , if remove dups: 278\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 728\n",
      "--- gene=728, id=31 removed by soft count = 78\n",
      "--- gene=728, id=41 removed by soft count = 652\n",
      "--- gene=728, id=232 removed by soft count = 78\n",
      "--- gene=728, id=256 removed by soft count = 250\n",
      "--- number of probes: 257 , kept by blast: 253 , if remove dups: 253\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 729\n",
      "--- gene=729, id=45 removed by soft count = 73\n",
      "--- gene=729, id=48 removed by soft count = 54\n",
      "--- gene=729, id=49 removed by soft count = 170\n",
      "--- gene=729, id=53 removed by soft count = 98\n",
      "--- gene=729, id=56 removed by soft count = 148\n",
      "--- gene=729, id=69 removed by soft count = 25143\n",
      "--- gene=729, id=76 removed by soft count = 241\n",
      "--- gene=729, id=86 removed by soft count = 186\n",
      "--- gene=729, id=87 removed by soft count = 88\n",
      "--- gene=729, id=88 removed by soft count = 53\n",
      "--- gene=729, id=90 removed by soft count = 86\n",
      "--- gene=729, id=92 removed by soft count = 96\n",
      "--- gene=729, id=95 removed by soft count = 102\n",
      "--- gene=729, id=96 removed by soft count = 131\n",
      "--- gene=729, id=100 removed by soft count = 797\n",
      "--- gene=729, id=233 removed by soft count = 1438\n",
      "--- number of probes: 244 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 730\n",
      "--- gene=730, id=35 removed by soft count = 105\n",
      "--- gene=730, id=107 removed by soft count = 70\n",
      "--- gene=730, id=144 removed by soft count = 1052\n",
      "--- gene=730, id=226 removed by soft count = 67\n",
      "--- number of probes: 295 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 731\n",
      "--- gene=731, id=78 removed by soft count = 62\n",
      "--- gene=731, id=153 removed by soft count = 52\n",
      "--- number of probes: 301 , kept by blast: 299 , if remove dups: 299\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 732\n",
      "--- gene=732, id=62 removed by soft count = 70\n",
      "--- gene=732, id=64 removed by soft count = 51\n",
      "--- number of probes: 269 , kept by blast: 267 , if remove dups: 267\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 733\n",
      "--- gene=733, id=86 removed by soft count = 64\n",
      "--- gene=733, id=123 removed by soft count = 74\n",
      "--- gene=733, id=164 removed by soft count = 79\n",
      "--- gene=733, id=263 removed by soft count = 131\n",
      "--- gene=733, id=265 removed by soft count = 141\n",
      "--- number of probes: 281 , kept by blast: 276 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 734\n",
      "--- number of probes: 318 , kept by blast: 318 , if remove dups: 318\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 735\n",
      "--- gene=735, id=18 removed by soft count = 67\n",
      "--- gene=735, id=52 removed by soft count = 131\n",
      "--- gene=735, id=262 removed by soft count = 86\n",
      "--- number of probes: 274 , kept by blast: 271 , if remove dups: 271\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 736\n",
      "--- gene=736, id=37 removed by soft count = 259\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=736, id=90 removed by soft count = 51\n",
      "--- gene=736, id=116 removed by soft count = 135\n",
      "--- number of probes: 173 , kept by blast: 170 , if remove dups: 170\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 737\n",
      "--- gene=737, id=41 removed by soft count = 115\n",
      "--- gene=737, id=45 removed by soft count = 52\n",
      "--- gene=737, id=46 removed by soft count = 151\n",
      "--- gene=737, id=86 removed by soft count = 162\n",
      "--- gene=737, id=122 removed by soft count = 906\n",
      "--- number of probes: 148 , kept by blast: 143 , if remove dups: 143\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 739\n",
      "--- number of probes: 293 , kept by blast: 293 , if remove dups: 293\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 740\n",
      "--- gene=740, id=230 removed by soft count = 70\n",
      "--- number of probes: 262 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 741\n",
      "--- gene=741, id=3 removed by hard count = 3\n",
      "--- gene=741, id=157 removed by soft count = 243\n",
      "--- gene=741, id=158 removed by soft count = 316\n",
      "--- number of probes: 278 , kept by blast: 275 , if remove dups: 274\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 742\n",
      "--- gene=742, id=79 removed by soft count = 163\n",
      "--- gene=742, id=231 removed by soft count = 927\n",
      "--- gene=742, id=232 removed by soft count = 131\n",
      "--- number of probes: 235 , kept by blast: 232 , if remove dups: 232\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 743\n",
      "--- gene=743, id=0 removed by soft count = 173\n",
      "--- gene=743, id=51 removed by soft count = 113\n",
      "--- gene=743, id=55 removed by soft count = 97\n",
      "--- gene=743, id=79 removed by soft count = 164\n",
      "--- gene=743, id=80 removed by soft count = 55\n",
      "--- gene=743, id=82 removed by soft count = 75\n",
      "--- gene=743, id=84 removed by soft count = 368\n",
      "--- gene=743, id=86 removed by soft count = 83\n",
      "--- gene=743, id=87 removed by soft count = 120\n",
      "--- gene=743, id=88 removed by soft count = 390\n",
      "--- number of probes: 91 , kept by blast: 81 , if remove dups: 81\n",
      "-- number of probes kept for this region: 81\n",
      "-- checking probes in region: 744\n",
      "--- gene=744, id=66 removed by soft count = 11282\n",
      "--- gene=744, id=67 removed by soft count = 3290\n",
      "--- gene=744, id=68 removed by soft count = 193\n",
      "--- gene=744, id=103 removed by soft count = 71\n",
      "--- number of probes: 144 , kept by blast: 140 , if remove dups: 140\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 745\n",
      "--- gene=745, id=46 removed by soft count = 280\n",
      "--- number of probes: 225 , kept by blast: 224 , if remove dups: 224\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 746\n",
      "--- gene=746, id=0 removed by soft count = 779\n",
      "--- gene=746, id=1 removed by soft count = 535\n",
      "--- gene=746, id=5 removed by soft count = 125\n",
      "--- gene=746, id=6 removed by soft count = 129\n",
      "--- gene=746, id=46 removed by soft count = 67\n",
      "--- gene=746, id=148 removed by soft count = 12047\n",
      "--- number of probes: 149 , kept by blast: 143 , if remove dups: 143\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 747\n",
      "--- gene=747, id=6 removed by soft count = 136\n",
      "--- gene=747, id=25 removed by soft count = 347\n",
      "--- gene=747, id=26 removed by soft count = 5015\n",
      "--- gene=747, id=27 removed by soft count = 266\n",
      "--- gene=747, id=28 removed by soft count = 21988\n",
      "--- gene=747, id=29 removed by soft count = 11490\n",
      "--- gene=747, id=30 removed by soft count = 199\n",
      "--- gene=747, id=31 removed by soft count = 1359\n",
      "--- gene=747, id=32 removed by soft count = 751\n",
      "--- gene=747, id=34 removed by soft count = 99\n",
      "--- gene=747, id=115 removed by soft count = 75\n",
      "--- gene=747, id=116 removed by soft count = 84\n",
      "--- gene=747, id=117 removed by soft count = 64\n",
      "--- number of probes: 118 , kept by blast: 105 , if remove dups: 105\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 105\n",
      "-- checking probes in region: 748\n",
      "--- gene=748, id=11 removed by soft count = 53\n",
      "--- gene=748, id=12 removed by soft count = 354\n",
      "--- gene=748, id=14 removed by soft count = 89\n",
      "--- gene=748, id=20 removed by soft count = 62\n",
      "--- gene=748, id=21 removed by soft count = 68\n",
      "--- gene=748, id=36 removed by soft count = 359\n",
      "--- gene=748, id=46 removed by soft count = 64\n",
      "--- gene=748, id=110 removed by soft count = 379\n",
      "--- number of probes: 160 , kept by blast: 152 , if remove dups: 152\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 749\n",
      "--- gene=749, id=13 removed by soft count = 53\n",
      "--- gene=749, id=157 removed by soft count = 1312\n",
      "--- gene=749, id=188 removed by soft count = 101\n",
      "--- number of probes: 224 , kept by blast: 221 , if remove dups: 221\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 750\n",
      "--- gene=750, id=10 removed by soft count = 169\n",
      "--- gene=750, id=104 removed by soft count = 351\n",
      "--- gene=750, id=107 removed by soft count = 223\n",
      "--- gene=750, id=117 removed by soft count = 105\n",
      "--- gene=750, id=123 removed by soft count = 58\n",
      "--- gene=750, id=185 removed by soft count = 223\n",
      "--- number of probes: 216 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 751\n",
      "--- gene=751, id=1 removed by soft count = 56\n",
      "--- gene=751, id=57 removed by soft count = 66\n",
      "--- gene=751, id=58 removed by soft count = 110\n",
      "--- gene=751, id=127 removed by soft count = 63\n",
      "--- number of probes: 166 , kept by blast: 162 , if remove dups: 162\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 752\n",
      "--- number of probes: 264 , kept by blast: 264 , if remove dups: 264\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 753\n",
      "--- number of probes: 225 , kept by blast: 225 , if remove dups: 225\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 754\n",
      "--- gene=754, id=0 removed by soft count = 202\n",
      "--- gene=754, id=4 removed by soft count = 62\n",
      "--- gene=754, id=34 removed by soft count = 494\n",
      "--- gene=754, id=35 removed by soft count = 299\n",
      "--- gene=754, id=65 removed by soft count = 384\n",
      "--- gene=754, id=67 removed by soft count = 307\n",
      "--- gene=754, id=69 removed by soft count = 128\n",
      "--- gene=754, id=123 removed by soft count = 68\n",
      "--- gene=754, id=176 removed by soft count = 62\n",
      "--- number of probes: 186 , kept by blast: 177 , if remove dups: 177\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 755\n",
      "--- gene=755, id=0 removed by soft count = 909\n",
      "--- gene=755, id=91 removed by soft count = 165\n",
      "--- number of probes: 252 , kept by blast: 250 , if remove dups: 250\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 756\n",
      "--- number of probes: 281 , kept by blast: 281 , if remove dups: 281\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 757\n",
      "--- gene=757, id=22 removed by soft count = 212\n",
      "--- gene=757, id=41 removed by soft count = 60\n",
      "--- gene=757, id=78 removed by soft count = 396\n",
      "--- gene=757, id=103 removed by soft count = 1202\n",
      "--- gene=757, id=108 removed by soft count = 64\n",
      "--- number of probes: 233 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 758\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=758, id=0 removed by soft count = 383\n",
      "--- gene=758, id=74 removed by soft count = 132\n",
      "--- gene=758, id=75 removed by soft count = 870\n",
      "--- gene=758, id=77 removed by soft count = 165\n",
      "--- gene=758, id=126 removed by soft count = 275\n",
      "--- number of probes: 197 , kept by blast: 192 , if remove dups: 192\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 759\n",
      "--- number of probes: 223 , kept by blast: 223 , if remove dups: 223\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 765\n",
      "--- gene=765, id=81 removed by soft count = 561\n",
      "--- number of probes: 242 , kept by blast: 241 , if remove dups: 241\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 766\n",
      "--- gene=766, id=66 removed by soft count = 341\n",
      "--- gene=766, id=67 removed by soft count = 1465\n",
      "--- gene=766, id=68 removed by soft count = 1206\n",
      "--- gene=766, id=69 removed by soft count = 154\n",
      "--- gene=766, id=70 removed by soft count = 88\n",
      "--- gene=766, id=71 removed by soft count = 88\n",
      "--- gene=766, id=75 removed by soft count = 52\n",
      "--- gene=766, id=76 removed by soft count = 90\n",
      "--- gene=766, id=77 removed by soft count = 148\n",
      "--- gene=766, id=83 removed by soft count = 186\n",
      "--- gene=766, id=86 removed by soft count = 56\n",
      "--- gene=766, id=87 removed by soft count = 125\n",
      "--- number of probes: 97 , kept by blast: 85 , if remove dups: 85\n",
      "-- number of probes kept for this region: 85\n",
      "-- checking probes in region: 767\n",
      "--- gene=767, id=260 removed by soft count = 51\n",
      "--- number of probes: 272 , kept by blast: 271 , if remove dups: 271\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 768\n",
      "--- gene=768, id=13 removed by soft count = 61\n",
      "--- gene=768, id=14 removed by soft count = 100\n",
      "--- gene=768, id=36 removed by soft count = 156\n",
      "--- gene=768, id=37 removed by soft count = 116\n",
      "--- gene=768, id=43 removed by soft count = 66\n",
      "--- gene=768, id=46 removed by soft count = 78\n",
      "--- gene=768, id=48 removed by soft count = 63\n",
      "--- gene=768, id=50 removed by soft count = 66\n",
      "--- gene=768, id=51 removed by soft count = 103\n",
      "--- number of probes: 123 , kept by blast: 114 , if remove dups: 114\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 114\n",
      "-- checking probes in region: 769\n",
      "--- gene=769, id=80 removed by soft count = 66\n",
      "--- gene=769, id=84 removed by soft count = 79\n",
      "--- number of probes: 230 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 770\n",
      "--- number of probes: 313 , kept by blast: 313 , if remove dups: 313\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 772\n",
      "--- gene=772, id=61 removed by soft count = 89\n",
      "--- gene=772, id=222 removed by soft count = 61\n",
      "--- gene=772, id=236 removed by soft count = 87\n",
      "--- gene=772, id=237 removed by soft count = 97\n",
      "--- number of probes: 238 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 773\n",
      "--- number of probes: 293 , kept by blast: 293 , if remove dups: 293\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 774\n",
      "--- gene=774, id=39 removed by soft count = 220\n",
      "--- gene=774, id=40 removed by soft count = 700\n",
      "--- gene=774, id=231 removed by soft count = 110\n",
      "--- number of probes: 257 , kept by blast: 254 , if remove dups: 254\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 775\n",
      "--- gene=775, id=112 removed by soft count = 83\n",
      "--- gene=775, id=118 removed by soft count = 98\n",
      "--- gene=775, id=119 removed by soft count = 256\n",
      "--- gene=775, id=166 removed by soft count = 231\n",
      "--- number of probes: 185 , kept by blast: 181 , if remove dups: 181\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 776\n",
      "--- gene=776, id=65 removed by soft count = 131\n",
      "--- number of probes: 168 , kept by blast: 167 , if remove dups: 167\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 779\n",
      "--- gene=779, id=49 removed by soft count = 60\n",
      "--- number of probes: 233 , kept by blast: 232 , if remove dups: 232\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 780\n",
      "--- gene=780, id=198 removed by soft count = 157\n",
      "--- gene=780, id=235 removed by soft count = 156\n",
      "--- gene=780, id=236 removed by soft count = 135\n",
      "--- number of probes: 237 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 781\n",
      "--- gene=781, id=49 removed by soft count = 67\n",
      "--- gene=781, id=53 removed by soft count = 73\n",
      "--- gene=781, id=54 removed by soft count = 58\n",
      "--- gene=781, id=55 removed by soft count = 258\n",
      "--- gene=781, id=56 removed by soft count = 137\n",
      "--- gene=781, id=57 removed by soft count = 79\n",
      "--- gene=781, id=61 removed by soft count = 65\n",
      "--- gene=781, id=63 removed by soft count = 180\n",
      "--- gene=781, id=64 removed by soft count = 508\n",
      "--- gene=781, id=65 removed by soft count = 112\n",
      "--- gene=781, id=66 removed by soft count = 99\n",
      "--- gene=781, id=115 removed by soft count = 152\n",
      "--- gene=781, id=116 removed by soft count = 57\n",
      "--- gene=781, id=118 removed by soft count = 126\n",
      "--- number of probes: 148 , kept by blast: 134 , if remove dups: 134\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 783\n",
      "--- gene=783, id=11 removed by soft count = 111\n",
      "--- gene=783, id=19 removed by soft count = 74\n",
      "--- gene=783, id=20 removed by soft count = 132\n",
      "--- gene=783, id=24 removed by soft count = 221\n",
      "--- gene=783, id=29 removed by soft count = 182\n",
      "--- gene=783, id=30 removed by soft count = 241\n",
      "--- gene=783, id=31 removed by soft count = 274\n",
      "--- gene=783, id=32 removed by soft count = 583\n",
      "--- gene=783, id=33 removed by soft count = 60\n",
      "--- number of probes: 72 , kept by blast: 63 , if remove dups: 63\n",
      "-- number of probes kept for this region: 63\n",
      "-- checking probes in region: 784\n",
      "--- gene=784, id=68 removed by soft count = 285\n",
      "--- gene=784, id=74 removed by soft count = 533\n",
      "--- gene=784, id=165 removed by soft count = 56\n",
      "--- number of probes: 210 , kept by blast: 207 , if remove dups: 207\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 785\n",
      "--- gene=785, id=52 removed by soft count = 91\n",
      "--- gene=785, id=65 removed by soft count = 106\n",
      "--- gene=785, id=86 removed by soft count = 113\n",
      "--- gene=785, id=89 removed by soft count = 126\n",
      "--- gene=785, id=90 removed by soft count = 63\n",
      "--- gene=785, id=96 removed by soft count = 56\n",
      "--- gene=785, id=98 removed by soft count = 98\n",
      "--- gene=785, id=100 removed by soft count = 233\n",
      "--- gene=785, id=104 removed by soft count = 284\n",
      "--- number of probes: 230 , kept by blast: 221 , if remove dups: 221\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 786\n",
      "--- gene=786, id=199 removed by soft count = 114\n",
      "--- gene=786, id=213 removed by soft count = 126\n",
      "--- number of probes: 319 , kept by blast: 317 , if remove dups: 317\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 787\n",
      "--- gene=787, id=83 removed by soft count = 316\n",
      "--- number of probes: 252 , kept by blast: 251 , if remove dups: 251\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 788\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=788, id=106 removed by soft count = 82\n",
      "--- gene=788, id=110 removed by soft count = 65\n",
      "--- number of probes: 146 , kept by blast: 144 , if remove dups: 144\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 789\n",
      "--- number of probes: 291 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 790\n",
      "--- gene=790, id=80 removed by soft count = 84\n",
      "--- gene=790, id=124 removed by soft count = 72\n",
      "--- gene=790, id=154 removed by soft count = 85\n",
      "--- number of probes: 214 , kept by blast: 211 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 791\n",
      "--- gene=791, id=25 removed by soft count = 118\n",
      "--- gene=791, id=77 removed by soft count = 56\n",
      "--- number of probes: 281 , kept by blast: 279 , if remove dups: 278\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 792\n",
      "--- gene=792, id=4 removed by soft count = 60\n",
      "--- gene=792, id=16 removed by soft count = 887\n",
      "--- gene=792, id=17 removed by soft count = 159\n",
      "--- gene=792, id=42 removed by soft count = 68\n",
      "--- gene=792, id=43 removed by soft count = 441\n",
      "--- gene=792, id=44 removed by soft count = 280\n",
      "--- gene=792, id=159 removed by soft count = 121\n",
      "--- gene=792, id=162 removed by soft count = 220\n",
      "--- gene=792, id=171 removed by soft count = 86\n",
      "--- gene=792, id=176 removed by soft count = 52\n",
      "--- gene=792, id=178 removed by soft count = 244\n",
      "--- gene=792, id=179 removed by soft count = 57\n",
      "--- number of probes: 213 , kept by blast: 201 , if remove dups: 201\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 793\n",
      "--- gene=793, id=53 removed by soft count = 51\n",
      "--- gene=793, id=54 removed by soft count = 191\n",
      "--- gene=793, id=55 removed by soft count = 878\n",
      "--- gene=793, id=59 removed by soft count = 371\n",
      "--- gene=793, id=61 removed by soft count = 77\n",
      "--- gene=793, id=62 removed by soft count = 110\n",
      "--- gene=793, id=141 removed by soft count = 217\n",
      "--- number of probes: 170 , kept by blast: 163 , if remove dups: 163\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 794\n",
      "--- gene=794, id=31 removed by soft count = 63\n",
      "--- gene=794, id=35 removed by soft count = 158\n",
      "--- gene=794, id=37 removed by soft count = 16563\n",
      "--- gene=794, id=85 removed by soft count = 53\n",
      "--- gene=794, id=87 removed by soft count = 77\n",
      "--- gene=794, id=90 removed by soft count = 86\n",
      "--- gene=794, id=91 removed by soft count = 68\n",
      "--- number of probes: 126 , kept by blast: 119 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 795\n",
      "--- gene=795, id=3 removed by soft count = 93\n",
      "--- gene=795, id=50 removed by soft count = 102\n",
      "--- gene=795, id=52 removed by soft count = 243\n",
      "--- gene=795, id=53 removed by soft count = 278\n",
      "--- number of probes: 232 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 796\n",
      "--- gene=796, id=32 removed by soft count = 214\n",
      "--- gene=796, id=77 removed by soft count = 59\n",
      "--- gene=796, id=119 removed by soft count = 68\n",
      "--- number of probes: 144 , kept by blast: 141 , if remove dups: 141\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 797\n",
      "--- gene=797, id=1 removed by soft count = 276\n",
      "--- gene=797, id=2 removed by soft count = 69\n",
      "--- gene=797, id=3 removed by soft count = 69\n",
      "--- gene=797, id=54 removed by soft count = 126\n",
      "--- gene=797, id=63 removed by soft count = 3734\n",
      "--- gene=797, id=79 removed by soft count = 103\n",
      "--- gene=797, id=80 removed by soft count = 278\n",
      "--- gene=797, id=171 removed by soft count = 151\n",
      "--- gene=797, id=172 removed by soft count = 557\n",
      "--- number of probes: 173 , kept by blast: 164 , if remove dups: 164\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 798\n",
      "--- gene=798, id=17 removed by soft count = 1339\n",
      "--- gene=798, id=18 removed by soft count = 64\n",
      "--- gene=798, id=76 removed by soft count = 2115\n",
      "--- gene=798, id=98 removed by soft count = 58\n",
      "--- gene=798, id=110 removed by soft count = 86\n",
      "--- number of probes: 137 , kept by blast: 132 , if remove dups: 132\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 799\n",
      "--- gene=799, id=16 removed by soft count = 65\n",
      "--- gene=799, id=18 removed by soft count = 60\n",
      "--- gene=799, id=19 removed by soft count = 6946\n",
      "--- gene=799, id=26 removed by soft count = 208\n",
      "--- gene=799, id=34 removed by soft count = 11702\n",
      "--- gene=799, id=55 removed by soft count = 1365\n",
      "--- gene=799, id=56 removed by soft count = 119\n",
      "--- gene=799, id=99 removed by soft count = 196\n",
      "--- gene=799, id=101 removed by soft count = 232\n",
      "--- gene=799, id=103 removed by soft count = 130\n",
      "--- gene=799, id=104 removed by soft count = 116\n",
      "--- gene=799, id=107 removed by soft count = 242\n",
      "--- gene=799, id=108 removed by soft count = 73\n",
      "--- number of probes: 132 , kept by blast: 119 , if remove dups: 119\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 119\n",
      "-- checking probes in region: 800\n",
      "--- gene=800, id=45 removed by soft count = 55\n",
      "--- gene=800, id=122 removed by soft count = 259\n",
      "--- gene=800, id=123 removed by soft count = 92\n",
      "--- number of probes: 169 , kept by blast: 166 , if remove dups: 166\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 801\n",
      "--- gene=801, id=15 removed by soft count = 1188\n",
      "--- gene=801, id=50 removed by soft count = 136\n",
      "--- gene=801, id=142 removed by soft count = 83\n",
      "--- gene=801, id=143 removed by soft count = 1149\n",
      "--- gene=801, id=145 removed by soft count = 1048\n",
      "--- gene=801, id=146 removed by soft count = 175\n",
      "--- gene=801, id=184 removed by soft count = 110\n",
      "--- gene=801, id=185 removed by soft count = 139\n",
      "--- number of probes: 200 , kept by blast: 192 , if remove dups: 192\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 802\n",
      "--- gene=802, id=70 removed by hard count = 0\n",
      "--- gene=802, id=231 removed by soft count = 204\n",
      "--- number of probes: 275 , kept by blast: 273 , if remove dups: 273\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 803\n",
      "--- gene=803, id=38 removed by soft count = 73\n",
      "--- gene=803, id=96 removed by soft count = 800\n",
      "--- gene=803, id=97 removed by soft count = 59\n",
      "--- gene=803, id=98 removed by soft count = 438\n",
      "--- gene=803, id=99 removed by soft count = 764\n",
      "--- gene=803, id=100 removed by soft count = 837\n",
      "--- gene=803, id=102 removed by soft count = 775\n",
      "--- gene=803, id=103 removed by soft count = 808\n",
      "--- gene=803, id=104 removed by soft count = 573\n",
      "--- gene=803, id=105 removed by soft count = 828\n",
      "--- gene=803, id=106 removed by soft count = 721\n",
      "--- gene=803, id=107 removed by soft count = 75\n",
      "--- gene=803, id=110 removed by soft count = 55\n",
      "--- gene=803, id=111 removed by soft count = 399\n",
      "--- gene=803, id=112 removed by soft count = 802\n",
      "--- gene=803, id=118 removed by soft count = 52\n",
      "--- gene=803, id=146 removed by soft count = 168\n",
      "--- gene=803, id=148 removed by soft count = 111\n",
      "--- number of probes: 188 , kept by blast: 170 , if remove dups: 170\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 804\n",
      "--- gene=804, id=13 removed by soft count = 151\n",
      "--- gene=804, id=14 removed by soft count = 81\n",
      "--- gene=804, id=20 removed by soft count = 318\n",
      "--- gene=804, id=22 removed by soft count = 263\n",
      "--- gene=804, id=23 removed by soft count = 302\n",
      "--- gene=804, id=27 removed by soft count = 233\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=804, id=76 removed by soft count = 54\n",
      "--- gene=804, id=96 removed by soft count = 536\n",
      "--- gene=804, id=102 removed by soft count = 94\n",
      "--- gene=804, id=107 removed by soft count = 184\n",
      "--- gene=804, id=112 removed by soft count = 279\n",
      "--- gene=804, id=158 removed by soft count = 250\n",
      "--- number of probes: 176 , kept by blast: 164 , if remove dups: 164\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 805\n",
      "--- gene=805, id=23 removed by soft count = 171\n",
      "--- gene=805, id=85 removed by soft count = 75\n",
      "--- gene=805, id=86 removed by soft count = 155\n",
      "--- gene=805, id=98 removed by soft count = 114\n",
      "--- gene=805, id=99 removed by soft count = 2059\n",
      "--- gene=805, id=105 removed by soft count = 52\n",
      "--- gene=805, id=106 removed by soft count = 195\n",
      "--- number of probes: 137 , kept by blast: 130 , if remove dups: 130\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 806\n",
      "--- gene=806, id=37 removed by soft count = 10087\n",
      "--- gene=806, id=40 removed by soft count = 54\n",
      "--- gene=806, id=93 removed by soft count = 77\n",
      "--- gene=806, id=94 removed by soft count = 19912\n",
      "--- gene=806, id=97 removed by soft count = 302\n",
      "--- gene=806, id=98 removed by soft count = 151\n",
      "--- gene=806, id=100 removed by soft count = 199\n",
      "--- gene=806, id=102 removed by soft count = 134\n",
      "--- gene=806, id=132 removed by soft count = 239\n",
      "--- gene=806, id=133 removed by soft count = 405\n",
      "--- number of probes: 135 , kept by blast: 125 , if remove dups: 125\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 807\n",
      "--- gene=807, id=124 removed by soft count = 345\n",
      "--- gene=807, id=125 removed by soft count = 759\n",
      "--- number of probes: 199 , kept by blast: 197 , if remove dups: 197\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 808\n",
      "--- gene=808, id=2 removed by soft count = 52\n",
      "--- gene=808, id=6 removed by soft count = 52\n",
      "--- gene=808, id=7 removed by soft count = 59\n",
      "--- gene=808, id=8 removed by soft count = 59\n",
      "--- gene=808, id=10 removed by soft count = 54\n",
      "--- gene=808, id=12 removed by soft count = 392\n",
      "--- gene=808, id=110 removed by soft count = 225\n",
      "--- gene=808, id=182 removed by soft count = 65\n",
      "--- gene=808, id=212 removed by soft count = 238\n",
      "--- number of probes: 218 , kept by blast: 209 , if remove dups: 209\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 809\n",
      "--- gene=809, id=326 removed by soft count = 51\n",
      "--- number of probes: 344 , kept by blast: 343 , if remove dups: 343\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 810\n",
      "--- number of probes: 178 , kept by blast: 178 , if remove dups: 178\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 812\n",
      "--- gene=812, id=40 removed by soft count = 135\n",
      "--- gene=812, id=41 removed by soft count = 307\n",
      "--- gene=812, id=42 removed by soft count = 274\n",
      "--- gene=812, id=43 removed by soft count = 118\n",
      "--- gene=812, id=44 removed by soft count = 194\n",
      "--- gene=812, id=45 removed by soft count = 166\n",
      "--- gene=812, id=153 removed by soft count = 51\n",
      "--- number of probes: 179 , kept by blast: 172 , if remove dups: 172\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 813\n",
      "--- gene=813, id=152 removed by soft count = 52\n",
      "--- number of probes: 237 , kept by blast: 236 , if remove dups: 236\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 814\n",
      "--- gene=814, id=116 removed by soft count = 89\n",
      "--- gene=814, id=141 removed by soft count = 94\n",
      "--- number of probes: 186 , kept by blast: 184 , if remove dups: 184\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 815\n",
      "--- gene=815, id=82 removed by soft count = 615\n",
      "--- gene=815, id=83 removed by soft count = 173\n",
      "--- gene=815, id=84 removed by soft count = 461\n",
      "--- gene=815, id=85 removed by soft count = 170\n",
      "--- gene=815, id=181 removed by soft count = 1040\n",
      "--- number of probes: 228 , kept by blast: 223 , if remove dups: 223\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 816\n",
      "--- gene=816, id=11 removed by soft count = 61\n",
      "--- gene=816, id=152 removed by soft count = 1295\n",
      "--- number of probes: 181 , kept by blast: 179 , if remove dups: 179\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 817\n",
      "--- gene=817, id=2 removed by soft count = 252\n",
      "--- gene=817, id=45 removed by soft count = 127\n",
      "--- gene=817, id=46 removed by soft count = 962\n",
      "--- gene=817, id=48 removed by soft count = 84\n",
      "--- gene=817, id=50 removed by soft count = 414\n",
      "--- gene=817, id=130 removed by soft count = 92\n",
      "--- gene=817, id=148 removed by soft count = 155\n",
      "--- gene=817, id=150 removed by soft count = 637\n",
      "--- gene=817, id=158 removed by soft count = 109\n",
      "--- number of probes: 183 , kept by blast: 174 , if remove dups: 174\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 818\n",
      "--- gene=818, id=72 removed by soft count = 293\n",
      "--- gene=818, id=155 removed by soft count = 4472\n",
      "--- gene=818, id=173 removed by soft count = 68\n",
      "--- gene=818, id=174 removed by soft count = 96\n",
      "--- gene=818, id=177 removed by soft count = 161\n",
      "--- gene=818, id=178 removed by soft count = 537\n",
      "--- gene=818, id=180 removed by soft count = 58\n",
      "--- gene=818, id=182 removed by soft count = 132\n",
      "--- gene=818, id=183 removed by soft count = 264\n",
      "--- gene=818, id=184 removed by soft count = 684\n",
      "--- number of probes: 198 , kept by blast: 188 , if remove dups: 188\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 819\n",
      "--- gene=819, id=36 removed by soft count = 92\n",
      "--- gene=819, id=54 removed by soft count = 722\n",
      "--- gene=819, id=169 removed by soft count = 95\n",
      "--- number of probes: 182 , kept by blast: 179 , if remove dups: 179\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 820\n",
      "--- gene=820, id=0 removed by soft count = 744\n",
      "--- gene=820, id=1 removed by soft count = 678\n",
      "--- gene=820, id=2 removed by soft count = 749\n",
      "--- gene=820, id=3 removed by soft count = 399\n",
      "--- gene=820, id=4 removed by soft count = 84\n",
      "--- gene=820, id=8 removed by soft count = 108\n",
      "--- gene=820, id=10 removed by soft count = 86\n",
      "--- gene=820, id=12 removed by soft count = 136\n",
      "--- gene=820, id=14 removed by soft count = 65\n",
      "--- gene=820, id=22 removed by soft count = 101\n",
      "--- gene=820, id=47 removed by soft count = 500\n",
      "--- gene=820, id=48 removed by soft count = 239\n",
      "--- gene=820, id=49 removed by soft count = 258\n",
      "--- gene=820, id=50 removed by soft count = 163\n",
      "--- gene=820, id=51 removed by soft count = 53\n",
      "--- gene=820, id=54 removed by soft count = 669\n",
      "--- gene=820, id=55 removed by soft count = 57\n",
      "--- gene=820, id=57 removed by soft count = 140\n",
      "--- gene=820, id=67 removed by soft count = 51\n",
      "--- gene=820, id=97 removed by soft count = 52\n",
      "--- gene=820, id=100 removed by soft count = 73\n",
      "--- number of probes: 115 , kept by blast: 94 , if remove dups: 94\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 94\n",
      "-- checking probes in region: 821\n",
      "--- gene=821, id=44 removed by soft count = 247\n",
      "--- gene=821, id=47 removed by soft count = 76\n",
      "--- gene=821, id=48 removed by soft count = 103\n",
      "--- gene=821, id=62 removed by soft count = 1559\n",
      "--- number of probes: 79 , kept by blast: 75 , if remove dups: 75\n",
      "-- number of probes kept for this region: 75\n",
      "-- checking probes in region: 822\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=822, id=27 removed by soft count = 51\n",
      "--- gene=822, id=32 removed by soft count = 92\n",
      "--- gene=822, id=34 removed by soft count = 91\n",
      "--- gene=822, id=72 removed by soft count = 130\n",
      "--- gene=822, id=173 removed by soft count = 71\n",
      "--- gene=822, id=174 removed by soft count = 131\n",
      "--- gene=822, id=177 removed by soft count = 209\n",
      "--- gene=822, id=178 removed by soft count = 279\n",
      "--- gene=822, id=180 removed by soft count = 297\n",
      "--- gene=822, id=181 removed by soft count = 212\n",
      "--- gene=822, id=182 removed by soft count = 64\n",
      "--- gene=822, id=186 removed by soft count = 244\n",
      "--- gene=822, id=188 removed by soft count = 469\n",
      "--- gene=822, id=189 removed by soft count = 93\n",
      "--- gene=822, id=190 removed by soft count = 65\n",
      "--- number of probes: 196 , kept by blast: 181 , if remove dups: 181\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 823\n",
      "--- number of probes: 341 , kept by blast: 341 , if remove dups: 339\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 824\n",
      "--- gene=824, id=0 removed by soft count = 5981\n",
      "--- gene=824, id=1 removed by soft count = 2228\n",
      "--- gene=824, id=3 removed by soft count = 64\n",
      "--- gene=824, id=65 removed by soft count = 391\n",
      "--- gene=824, id=66 removed by soft count = 875\n",
      "--- gene=824, id=76 removed by soft count = 55\n",
      "--- number of probes: 99 , kept by blast: 93 , if remove dups: 93\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 93\n",
      "-- checking probes in region: 825\n",
      "--- gene=825, id=200 removed by soft count = 575\n",
      "--- gene=825, id=204 removed by soft count = 85\n",
      "--- gene=825, id=206 removed by soft count = 128\n",
      "--- gene=825, id=209 removed by soft count = 61\n",
      "--- gene=825, id=210 removed by soft count = 283\n",
      "--- number of probes: 216 , kept by blast: 211 , if remove dups: 211\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 827\n",
      "--- gene=827, id=8 removed by soft count = 82\n",
      "--- gene=827, id=11 removed by soft count = 99\n",
      "--- gene=827, id=21 removed by soft count = 124\n",
      "--- gene=827, id=31 removed by soft count = 60\n",
      "--- gene=827, id=39 removed by soft count = 55\n",
      "--- gene=827, id=58 removed by soft count = 179\n",
      "--- gene=827, id=60 removed by soft count = 145\n",
      "--- number of probes: 143 , kept by blast: 136 , if remove dups: 136\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 828\n",
      "--- number of probes: 103 , kept by blast: 103 , if remove dups: 103\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 103\n",
      "-- checking probes in region: 829\n",
      "--- gene=829, id=29 removed by soft count = 53\n",
      "--- gene=829, id=39 removed by soft count = 158\n",
      "--- gene=829, id=65 removed by soft count = 75\n",
      "--- gene=829, id=69 removed by soft count = 131\n",
      "--- gene=829, id=74 removed by soft count = 586\n",
      "--- gene=829, id=75 removed by soft count = 430\n",
      "--- number of probes: 122 , kept by blast: 116 , if remove dups: 116\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 116\n",
      "-- checking probes in region: 830\n",
      "--- gene=830, id=5 removed by soft count = 211\n",
      "--- gene=830, id=6 removed by soft count = 56\n",
      "--- gene=830, id=11 removed by soft count = 60\n",
      "--- gene=830, id=21 removed by soft count = 1333\n",
      "--- gene=830, id=22 removed by soft count = 70\n",
      "--- gene=830, id=47 removed by soft count = 152\n",
      "--- gene=830, id=48 removed by soft count = 376\n",
      "--- gene=830, id=104 removed by soft count = 59\n",
      "--- number of probes: 168 , kept by blast: 160 , if remove dups: 159\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 831\n",
      "--- gene=831, id=16 removed by soft count = 220\n",
      "--- number of probes: 227 , kept by blast: 226 , if remove dups: 226\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 832\n",
      "--- gene=832, id=42 removed by soft count = 253\n",
      "--- gene=832, id=43 removed by soft count = 73\n",
      "--- number of probes: 232 , kept by blast: 230 , if remove dups: 230\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 833\n",
      "--- gene=833, id=5 removed by soft count = 708\n",
      "--- gene=833, id=6 removed by soft count = 2086\n",
      "--- gene=833, id=61 removed by soft count = 1551\n",
      "--- gene=833, id=62 removed by soft count = 188\n",
      "--- gene=833, id=63 removed by soft count = 439\n",
      "--- gene=833, id=106 removed by soft count = 99\n",
      "--- gene=833, id=108 removed by soft count = 104\n",
      "--- gene=833, id=109 removed by soft count = 111\n",
      "--- number of probes: 153 , kept by blast: 145 , if remove dups: 145\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 834\n",
      "--- gene=834, id=126 removed by soft count = 104\n",
      "--- gene=834, id=129 removed by soft count = 75\n",
      "--- gene=834, id=152 removed by soft count = 91\n",
      "--- gene=834, id=189 removed by soft count = 52\n",
      "--- gene=834, id=195 removed by soft count = 60\n",
      "--- gene=834, id=233 removed by soft count = 105\n",
      "--- gene=834, id=245 removed by soft count = 140\n",
      "--- number of probes: 297 , kept by blast: 290 , if remove dups: 290\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 835\n",
      "--- gene=835, id=107 removed by soft count = 4357\n",
      "--- number of probes: 220 , kept by blast: 219 , if remove dups: 219\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 836\n",
      "--- gene=836, id=28 removed by soft count = 52\n",
      "--- gene=836, id=46 removed by soft count = 388\n",
      "--- gene=836, id=50 removed by soft count = 77\n",
      "--- gene=836, id=86 removed by soft count = 174\n",
      "--- gene=836, id=166 removed by soft count = 236\n",
      "--- number of probes: 296 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 837\n",
      "--- gene=837, id=123 removed by soft count = 25278\n",
      "--- gene=837, id=153 removed by soft count = 75\n",
      "--- gene=837, id=186 removed by soft count = 59\n",
      "--- gene=837, id=187 removed by soft count = 118\n",
      "--- gene=837, id=234 removed by soft count = 65\n",
      "--- number of probes: 312 , kept by blast: 307 , if remove dups: 307\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 838\n",
      "--- number of probes: 261 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 839\n",
      "--- gene=839, id=33 removed by soft count = 52\n",
      "--- gene=839, id=34 removed by soft count = 57\n",
      "--- gene=839, id=36 removed by soft count = 54\n",
      "--- gene=839, id=62 removed by soft count = 67\n",
      "--- gene=839, id=63 removed by soft count = 203\n",
      "--- gene=839, id=105 removed by soft count = 51\n",
      "--- number of probes: 240 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 840\n",
      "--- gene=840, id=91 removed by soft count = 176\n",
      "--- number of probes: 333 , kept by blast: 332 , if remove dups: 332\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 841\n",
      "--- number of probes: 328 , kept by blast: 328 , if remove dups: 328\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 842\n",
      "--- gene=842, id=86 removed by soft count = 75\n",
      "--- gene=842, id=108 removed by soft count = 112\n",
      "--- gene=842, id=112 removed by soft count = 457\n",
      "--- gene=842, id=140 removed by soft count = 81\n",
      "--- number of probes: 298 , kept by blast: 294 , if remove dups: 294\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 843\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=843, id=78 removed by soft count = 114\n",
      "--- number of probes: 248 , kept by blast: 247 , if remove dups: 247\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 844\n",
      "--- gene=844, id=60 removed by soft count = 51\n",
      "--- gene=844, id=63 removed by soft count = 73\n",
      "--- gene=844, id=177 removed by soft count = 60\n",
      "--- gene=844, id=184 removed by soft count = 87\n",
      "--- number of probes: 268 , kept by blast: 264 , if remove dups: 264\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 845\n",
      "--- gene=845, id=86 removed by hard count = 4\n",
      "--- number of probes: 268 , kept by blast: 267 , if remove dups: 267\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 846\n",
      "--- gene=846, id=36 removed by soft count = 78\n",
      "--- gene=846, id=47 removed by soft count = 157\n",
      "--- gene=846, id=48 removed by soft count = 70\n",
      "--- gene=846, id=64 removed by soft count = 198\n",
      "--- number of probes: 237 , kept by blast: 233 , if remove dups: 233\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 847\n",
      "--- gene=847, id=102 removed by soft count = 199\n",
      "--- number of probes: 258 , kept by blast: 257 , if remove dups: 257\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 848\n",
      "--- gene=848, id=113 removed by soft count = 114\n",
      "--- gene=848, id=181 removed by soft count = 96\n",
      "--- gene=848, id=182 removed by soft count = 557\n",
      "--- gene=848, id=193 removed by soft count = 185\n",
      "--- number of probes: 262 , kept by blast: 258 , if remove dups: 258\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 849\n",
      "--- gene=849, id=57 removed by soft count = 103\n",
      "--- gene=849, id=58 removed by soft count = 183\n",
      "--- gene=849, id=78 removed by soft count = 65\n",
      "--- gene=849, id=80 removed by soft count = 55\n",
      "--- gene=849, id=81 removed by soft count = 60\n",
      "--- gene=849, id=84 removed by soft count = 84\n",
      "--- gene=849, id=86 removed by soft count = 75\n",
      "--- gene=849, id=146 removed by soft count = 127\n",
      "--- gene=849, id=147 removed by soft count = 167\n",
      "--- gene=849, id=148 removed by soft count = 93\n",
      "--- gene=849, id=201 removed by soft count = 55\n",
      "--- gene=849, id=202 removed by soft count = 140\n",
      "--- number of probes: 227 , kept by blast: 215 , if remove dups: 215\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 850\n",
      "--- gene=850, id=65 removed by hard count = 3\n",
      "--- gene=850, id=67 removed by soft count = 79\n",
      "--- gene=850, id=146 removed by soft count = 56\n",
      "--- gene=850, id=182 removed by soft count = 52\n",
      "--- gene=850, id=192 removed by soft count = 90\n",
      "--- gene=850, id=210 removed by soft count = 66\n",
      "--- gene=850, id=230 removed by soft count = 548\n",
      "--- number of probes: 246 , kept by blast: 239 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 851\n",
      "--- gene=851, id=5 removed by soft count = 160\n",
      "--- gene=851, id=139 removed by soft count = 62\n",
      "--- gene=851, id=182 removed by soft count = 112\n",
      "--- gene=851, id=185 removed by soft count = 113\n",
      "--- number of probes: 289 , kept by blast: 285 , if remove dups: 285\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 852\n",
      "--- gene=852, id=10 removed by soft count = 70\n",
      "--- gene=852, id=12 removed by soft count = 66\n",
      "--- gene=852, id=25 removed by soft count = 99\n",
      "--- gene=852, id=38 removed by soft count = 224\n",
      "--- gene=852, id=39 removed by soft count = 884\n",
      "--- gene=852, id=41 removed by soft count = 153\n",
      "--- gene=852, id=59 removed by soft count = 95\n",
      "--- gene=852, id=66 removed by soft count = 92\n",
      "--- gene=852, id=146 removed by soft count = 65\n",
      "--- number of probes: 148 , kept by blast: 139 , if remove dups: 139\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 854\n",
      "--- gene=854, id=35 removed by soft count = 52\n",
      "--- gene=854, id=36 removed by soft count = 73\n",
      "--- gene=854, id=173 removed by soft count = 86\n",
      "--- gene=854, id=190 removed by soft count = 29212\n",
      "--- gene=854, id=192 removed by soft count = 682\n",
      "--- gene=854, id=223 removed by soft count = 62\n",
      "--- gene=854, id=224 removed by soft count = 52\n",
      "--- gene=854, id=225 removed by soft count = 447\n",
      "--- number of probes: 234 , kept by blast: 226 , if remove dups: 226\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 855\n",
      "--- gene=855, id=8 removed by soft count = 77\n",
      "--- number of probes: 192 , kept by blast: 191 , if remove dups: 191\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 856\n",
      "--- gene=856, id=1 removed by soft count = 70\n",
      "--- gene=856, id=23 removed by soft count = 65\n",
      "--- gene=856, id=30 removed by soft count = 82\n",
      "--- gene=856, id=49 removed by soft count = 123\n",
      "--- gene=856, id=65 removed by soft count = 56\n",
      "--- gene=856, id=92 removed by soft count = 149\n",
      "--- gene=856, id=93 removed by soft count = 146\n",
      "--- gene=856, id=159 removed by soft count = 217\n",
      "--- gene=856, id=169 removed by soft count = 60\n",
      "--- number of probes: 237 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 857\n",
      "--- gene=857, id=104 removed by soft count = 120\n",
      "--- gene=857, id=200 removed by soft count = 181\n",
      "--- number of probes: 226 , kept by blast: 224 , if remove dups: 224\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 858\n",
      "--- gene=858, id=0 removed by soft count = 83\n",
      "--- gene=858, id=42 removed by soft count = 90\n",
      "--- gene=858, id=94 removed by soft count = 150\n",
      "--- gene=858, id=95 removed by soft count = 282\n",
      "--- gene=858, id=98 removed by soft count = 113\n",
      "--- gene=858, id=101 removed by soft count = 183\n",
      "--- gene=858, id=161 removed by hard count = 3\n",
      "--- gene=858, id=190 removed by soft count = 98\n",
      "--- number of probes: 200 , kept by blast: 192 , if remove dups: 190\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 859\n",
      "--- gene=859, id=0 removed by soft count = 141\n",
      "--- gene=859, id=1 removed by soft count = 631\n",
      "--- gene=859, id=2 removed by soft count = 250\n",
      "--- gene=859, id=36 removed by soft count = 69\n",
      "--- gene=859, id=224 removed by soft count = 137\n",
      "--- number of probes: 234 , kept by blast: 229 , if remove dups: 229\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 860\n",
      "--- gene=860, id=12 removed by soft count = 88\n",
      "--- gene=860, id=46 removed by soft count = 59\n",
      "--- gene=860, id=155 removed by soft count = 315\n",
      "--- gene=860, id=164 removed by soft count = 149\n",
      "--- gene=860, id=190 removed by soft count = 61\n",
      "--- gene=860, id=191 removed by soft count = 281\n",
      "--- gene=860, id=205 removed by soft count = 70\n",
      "--- number of probes: 217 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 861\n",
      "--- gene=861, id=195 removed by soft count = 657\n",
      "--- gene=861, id=235 removed by soft count = 94\n",
      "--- gene=861, id=287 removed by soft count = 69\n",
      "--- number of probes: 323 , kept by blast: 320 , if remove dups: 320\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 862\n",
      "--- gene=862, id=49 removed by soft count = 113\n",
      "--- gene=862, id=50 removed by soft count = 10065\n",
      "--- gene=862, id=51 removed by soft count = 69\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- number of probes: 101 , kept by blast: 98 , if remove dups: 98\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 98\n",
      "-- checking probes in region: 863\n",
      "--- gene=863, id=0 removed by soft count = 802\n",
      "--- gene=863, id=2 removed by soft count = 247\n",
      "--- gene=863, id=4 removed by soft count = 60\n",
      "--- gene=863, id=29 removed by soft count = 281\n",
      "--- gene=863, id=32 removed by soft count = 138\n",
      "--- gene=863, id=33 removed by soft count = 225\n",
      "--- number of probes: 150 , kept by blast: 144 , if remove dups: 144\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 864\n",
      "--- gene=864, id=135 removed by soft count = 104\n",
      "--- gene=864, id=231 removed by soft count = 351\n",
      "--- number of probes: 258 , kept by blast: 256 , if remove dups: 256\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 865\n",
      "--- gene=865, id=91 removed by soft count = 229\n",
      "--- gene=865, id=189 removed by soft count = 58\n",
      "--- gene=865, id=219 removed by soft count = 65\n",
      "--- gene=865, id=220 removed by soft count = 57\n",
      "--- number of probes: 259 , kept by blast: 255 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 866\n",
      "--- gene=866, id=139 removed by soft count = 335\n",
      "--- gene=866, id=237 removed by soft count = 55\n",
      "--- gene=866, id=242 removed by soft count = 237\n",
      "--- number of probes: 251 , kept by blast: 248 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 867\n",
      "--- gene=867, id=13 removed by soft count = 434\n",
      "--- gene=867, id=29 removed by soft count = 300\n",
      "--- gene=867, id=32 removed by soft count = 252\n",
      "--- gene=867, id=64 removed by soft count = 207\n",
      "--- gene=867, id=65 removed by soft count = 66\n",
      "--- gene=867, id=78 removed by soft count = 414\n",
      "--- number of probes: 145 , kept by blast: 139 , if remove dups: 139\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 868\n",
      "--- gene=868, id=68 removed by soft count = 82\n",
      "--- gene=868, id=220 removed by soft count = 60\n",
      "--- number of probes: 294 , kept by blast: 292 , if remove dups: 292\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 869\n",
      "--- gene=869, id=0 removed by soft count = 80\n",
      "--- number of probes: 213 , kept by blast: 212 , if remove dups: 212\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 870\n",
      "--- gene=870, id=12 removed by soft count = 70\n",
      "--- gene=870, id=72 removed by soft count = 108\n",
      "--- gene=870, id=73 removed by soft count = 74\n",
      "--- gene=870, id=144 removed by soft count = 213\n",
      "--- gene=870, id=205 removed by soft count = 70\n",
      "--- gene=870, id=214 removed by soft count = 64\n",
      "--- number of probes: 257 , kept by blast: 251 , if remove dups: 251\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 871\n",
      "--- gene=871, id=0 removed by soft count = 148\n",
      "--- gene=871, id=1 removed by soft count = 959\n",
      "--- gene=871, id=2 removed by soft count = 509\n",
      "--- gene=871, id=3 removed by soft count = 64\n",
      "--- gene=871, id=11 removed by soft count = 337\n",
      "--- gene=871, id=12 removed by soft count = 481\n",
      "--- gene=871, id=129 removed by soft count = 8063\n",
      "--- number of probes: 184 , kept by blast: 177 , if remove dups: 177\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 872\n",
      "--- gene=872, id=65 removed by soft count = 112\n",
      "--- gene=872, id=124 removed by soft count = 87\n",
      "--- number of probes: 203 , kept by blast: 201 , if remove dups: 201\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 873\n",
      "--- gene=873, id=3 removed by soft count = 110\n",
      "--- gene=873, id=4 removed by soft count = 59\n",
      "--- gene=873, id=5 removed by soft count = 73\n",
      "--- gene=873, id=6 removed by soft count = 1172\n",
      "--- gene=873, id=11 removed by soft count = 165\n",
      "--- gene=873, id=12 removed by soft count = 68\n",
      "--- gene=873, id=27 removed by soft count = 1286\n",
      "--- gene=873, id=28 removed by soft count = 450\n",
      "--- gene=873, id=29 removed by soft count = 1114\n",
      "--- number of probes: 93 , kept by blast: 84 , if remove dups: 84\n",
      "-- number of probes kept for this region: 84\n",
      "-- checking probes in region: 874\n",
      "--- gene=874, id=54 removed by soft count = 266\n",
      "--- gene=874, id=55 removed by soft count = 82\n",
      "--- gene=874, id=62 removed by soft count = 252\n",
      "--- gene=874, id=71 removed by soft count = 53\n",
      "--- gene=874, id=105 removed by soft count = 56\n",
      "--- number of probes: 116 , kept by blast: 111 , if remove dups: 111\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 111\n",
      "-- checking probes in region: 875\n",
      "--- gene=875, id=103 removed by soft count = 77\n",
      "--- gene=875, id=136 removed by soft count = 150\n",
      "--- gene=875, id=202 removed by soft count = 1087\n",
      "--- number of probes: 217 , kept by blast: 214 , if remove dups: 214\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 876\n",
      "--- gene=876, id=74 removed by soft count = 57\n",
      "--- gene=876, id=125 removed by soft count = 76\n",
      "--- gene=876, id=128 removed by soft count = 61\n",
      "--- gene=876, id=136 removed by soft count = 88\n",
      "--- gene=876, id=142 removed by soft count = 54\n",
      "--- gene=876, id=143 removed by soft count = 274\n",
      "--- gene=876, id=146 removed by soft count = 121\n",
      "--- gene=876, id=156 removed by soft count = 51\n",
      "--- gene=876, id=161 removed by soft count = 142\n",
      "--- gene=876, id=162 removed by soft count = 99\n",
      "--- gene=876, id=163 removed by soft count = 88\n",
      "--- gene=876, id=164 removed by soft count = 55\n",
      "--- number of probes: 196 , kept by blast: 184 , if remove dups: 184\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 877\n",
      "--- gene=877, id=23 removed by soft count = 8723\n",
      "--- gene=877, id=44 removed by soft count = 65\n",
      "--- gene=877, id=45 removed by soft count = 125\n",
      "--- gene=877, id=55 removed by soft count = 3287\n",
      "--- gene=877, id=67 removed by soft count = 225\n",
      "--- gene=877, id=68 removed by soft count = 313\n",
      "--- gene=877, id=69 removed by soft count = 290\n",
      "--- number of probes: 76 , kept by blast: 69 , if remove dups: 69\n",
      "-- number of probes kept for this region: 69\n",
      "-- checking probes in region: 878\n",
      "--- gene=878, id=109 removed by soft count = 203\n",
      "--- gene=878, id=110 removed by soft count = 192\n",
      "--- number of probes: 194 , kept by blast: 192 , if remove dups: 192\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 879\n",
      "--- gene=879, id=14 removed by soft count = 51\n",
      "--- gene=879, id=15 removed by soft count = 69\n",
      "--- gene=879, id=17 removed by soft count = 77\n",
      "--- gene=879, id=71 removed by soft count = 7132\n",
      "--- gene=879, id=72 removed by soft count = 180\n",
      "--- gene=879, id=73 removed by soft count = 102\n",
      "--- gene=879, id=78 removed by soft count = 673\n",
      "--- gene=879, id=92 removed by soft count = 78\n",
      "--- gene=879, id=98 removed by soft count = 1606\n",
      "--- gene=879, id=99 removed by soft count = 240\n",
      "--- number of probes: 133 , kept by blast: 123 , if remove dups: 123\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 880\n",
      "--- gene=880, id=0 removed by soft count = 10159\n",
      "--- gene=880, id=1 removed by soft count = 3481\n",
      "--- gene=880, id=15 removed by soft count = 151\n",
      "--- gene=880, id=78 removed by soft count = 369\n",
      "--- number of probes: 88 , kept by blast: 84 , if remove dups: 84\n",
      "-- number of probes kept for this region: 84\n",
      "-- checking probes in region: 881\n",
      "--- gene=881, id=2 removed by soft count = 191\n",
      "--- gene=881, id=51 removed by soft count = 437\n",
      "--- gene=881, id=52 removed by hard count = 4\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=881, id=54 removed by soft count = 972\n",
      "--- gene=881, id=56 removed by soft count = 252\n",
      "--- gene=881, id=59 removed by soft count = 126\n",
      "--- gene=881, id=62 removed by soft count = 68\n",
      "--- gene=881, id=71 removed by soft count = 1448\n",
      "--- gene=881, id=93 removed by soft count = 582\n",
      "--- gene=881, id=107 removed by soft count = 113\n",
      "--- gene=881, id=164 removed by soft count = 116\n",
      "--- number of probes: 192 , kept by blast: 181 , if remove dups: 181\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 882\n",
      "--- gene=882, id=37 removed by soft count = 131\n",
      "--- gene=882, id=120 removed by soft count = 142\n",
      "--- gene=882, id=182 removed by soft count = 192\n",
      "--- gene=882, id=187 removed by soft count = 150\n",
      "--- gene=882, id=188 removed by soft count = 56\n",
      "--- gene=882, id=189 removed by soft count = 75\n",
      "--- gene=882, id=190 removed by soft count = 87\n",
      "--- gene=882, id=191 removed by soft count = 92\n",
      "--- number of probes: 220 , kept by blast: 212 , if remove dups: 212\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 883\n",
      "--- gene=883, id=165 removed by soft count = 64\n",
      "--- gene=883, id=166 removed by soft count = 103\n",
      "--- gene=883, id=167 removed by soft count = 624\n",
      "--- gene=883, id=176 removed by soft count = 65\n",
      "--- gene=883, id=266 removed by soft count = 205\n",
      "--- number of probes: 277 , kept by blast: 272 , if remove dups: 272\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 884\n",
      "--- gene=884, id=53 removed by soft count = 145\n",
      "--- gene=884, id=186 removed by soft count = 67\n",
      "--- gene=884, id=190 removed by soft count = 85\n",
      "--- number of probes: 192 , kept by blast: 189 , if remove dups: 189\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 885\n",
      "--- gene=885, id=166 removed by soft count = 225\n",
      "--- number of probes: 180 , kept by blast: 179 , if remove dups: 179\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 886\n",
      "--- gene=886, id=3 removed by soft count = 227\n",
      "--- gene=886, id=25 removed by soft count = 276\n",
      "--- gene=886, id=26 removed by soft count = 64\n",
      "--- gene=886, id=37 removed by soft count = 70\n",
      "--- gene=886, id=39 removed by soft count = 63\n",
      "--- gene=886, id=42 removed by soft count = 134\n",
      "--- gene=886, id=43 removed by soft count = 114\n",
      "--- gene=886, id=70 removed by soft count = 52\n",
      "--- gene=886, id=74 removed by soft count = 66\n",
      "--- gene=886, id=99 removed by soft count = 143\n",
      "--- gene=886, id=100 removed by soft count = 227\n",
      "--- gene=886, id=101 removed by soft count = 211\n",
      "--- gene=886, id=121 removed by soft count = 110\n",
      "--- number of probes: 145 , kept by blast: 132 , if remove dups: 132\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 887\n",
      "--- gene=887, id=0 removed by soft count = 134\n",
      "--- gene=887, id=1 removed by soft count = 657\n",
      "--- gene=887, id=4 removed by soft count = 73\n",
      "--- gene=887, id=74 removed by soft count = 80\n",
      "--- number of probes: 210 , kept by blast: 206 , if remove dups: 206\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 888\n",
      "--- gene=888, id=41 removed by soft count = 211\n",
      "--- number of probes: 218 , kept by blast: 217 , if remove dups: 214\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 889\n",
      "--- number of probes: 194 , kept by blast: 194 , if remove dups: 194\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 890\n",
      "--- gene=890, id=10 removed by soft count = 187\n",
      "--- gene=890, id=13 removed by soft count = 492\n",
      "--- gene=890, id=163 removed by soft count = 121\n",
      "--- gene=890, id=280 removed by soft count = 351\n",
      "--- number of probes: 295 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 891\n",
      "--- gene=891, id=6 removed by soft count = 107\n",
      "--- gene=891, id=21 removed by soft count = 2243\n",
      "--- gene=891, id=43 removed by soft count = 192\n",
      "--- gene=891, id=45 removed by soft count = 103\n",
      "--- gene=891, id=117 removed by soft count = 213\n",
      "--- gene=891, id=119 removed by soft count = 198\n",
      "--- gene=891, id=120 removed by soft count = 52\n",
      "--- number of probes: 173 , kept by blast: 166 , if remove dups: 166\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 894\n",
      "--- gene=894, id=43 removed by soft count = 82\n",
      "--- gene=894, id=87 removed by soft count = 419\n",
      "--- gene=894, id=88 removed by soft count = 96\n",
      "--- gene=894, id=101 removed by soft count = 161\n",
      "--- gene=894, id=151 removed by soft count = 523\n",
      "--- gene=894, id=163 removed by soft count = 117\n",
      "--- gene=894, id=164 removed by soft count = 120\n",
      "--- gene=894, id=165 removed by soft count = 52\n",
      "--- gene=894, id=187 removed by soft count = 96\n",
      "--- number of probes: 228 , kept by blast: 219 , if remove dups: 219\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 896\n",
      "--- gene=896, id=79 removed by soft count = 162\n",
      "--- number of probes: 316 , kept by blast: 315 , if remove dups: 314\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 897\n",
      "--- gene=897, id=88 removed by soft count = 564\n",
      "--- gene=897, id=99 removed by soft count = 87\n",
      "--- gene=897, id=185 removed by soft count = 69\n",
      "--- number of probes: 259 , kept by blast: 256 , if remove dups: 256\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 898\n",
      "--- gene=898, id=128 removed by soft count = 14374\n",
      "--- gene=898, id=169 removed by soft count = 122\n",
      "--- gene=898, id=208 removed by soft count = 443\n",
      "--- gene=898, id=243 removed by soft count = 82\n",
      "--- number of probes: 275 , kept by blast: 271 , if remove dups: 271\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 899\n",
      "--- gene=899, id=6 removed by soft count = 355\n",
      "--- gene=899, id=38 removed by soft count = 91\n",
      "--- gene=899, id=77 removed by soft count = 54\n",
      "--- gene=899, id=83 removed by soft count = 136\n",
      "--- gene=899, id=148 removed by soft count = 66\n",
      "--- gene=899, id=158 removed by soft count = 71\n",
      "--- gene=899, id=159 removed by soft count = 54\n",
      "--- number of probes: 179 , kept by blast: 172 , if remove dups: 172\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 900\n",
      "--- gene=900, id=3 removed by soft count = 56\n",
      "--- gene=900, id=11 removed by soft count = 144\n",
      "--- gene=900, id=12 removed by soft count = 351\n",
      "--- gene=900, id=20 removed by soft count = 290\n",
      "--- gene=900, id=21 removed by soft count = 324\n",
      "--- gene=900, id=24 removed by soft count = 233\n",
      "--- gene=900, id=41 removed by soft count = 81\n",
      "--- gene=900, id=48 removed by soft count = 3406\n",
      "--- number of probes: 67 , kept by blast: 59 , if remove dups: 59\n",
      "-- number of probes kept for this region: 59\n",
      "-- checking probes in region: 901\n",
      "--- gene=901, id=5 removed by soft count = 74\n",
      "--- gene=901, id=28 removed by soft count = 302\n",
      "--- gene=901, id=29 removed by soft count = 173\n",
      "--- gene=901, id=30 removed by soft count = 118\n",
      "--- gene=901, id=35 removed by soft count = 78\n",
      "--- gene=901, id=38 removed by soft count = 77\n",
      "--- gene=901, id=42 removed by soft count = 119\n",
      "--- number of probes: 95 , kept by blast: 88 , if remove dups: 88\n",
      "-- number of probes kept for this region: 88\n",
      "-- checking probes in region: 902\n",
      "--- number of probes: 288 , kept by blast: 288 , if remove dups: 288\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 903\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=903, id=70 removed by soft count = 654\n",
      "--- gene=903, id=71 removed by soft count = 344\n",
      "--- gene=903, id=72 removed by soft count = 336\n",
      "--- gene=903, id=74 removed by soft count = 181\n",
      "--- gene=903, id=85 removed by soft count = 137\n",
      "--- gene=903, id=94 removed by soft count = 261\n",
      "--- gene=903, id=118 removed by soft count = 53\n",
      "--- gene=903, id=192 removed by soft count = 132\n",
      "--- gene=903, id=193 removed by soft count = 2157\n",
      "--- number of probes: 195 , kept by blast: 186 , if remove dups: 186\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 904\n",
      "--- gene=904, id=196 removed by soft count = 63\n",
      "--- number of probes: 253 , kept by blast: 252 , if remove dups: 252\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 905\n",
      "--- gene=905, id=58 removed by soft count = 183\n",
      "--- gene=905, id=89 removed by soft count = 65\n",
      "--- gene=905, id=111 removed by soft count = 189\n",
      "--- gene=905, id=196 removed by soft count = 79\n",
      "--- gene=905, id=197 removed by soft count = 51\n",
      "--- number of probes: 237 , kept by blast: 232 , if remove dups: 232\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 906\n",
      "--- gene=906, id=0 removed by soft count = 203\n",
      "--- gene=906, id=10 removed by soft count = 106\n",
      "--- gene=906, id=26 removed by soft count = 65\n",
      "--- gene=906, id=239 removed by soft count = 64\n",
      "--- number of probes: 254 , kept by blast: 250 , if remove dups: 250\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 907\n",
      "--- gene=907, id=35 removed by soft count = 65\n",
      "--- gene=907, id=51 removed by soft count = 135\n",
      "--- gene=907, id=52 removed by soft count = 89\n",
      "--- gene=907, id=54 removed by soft count = 53\n",
      "--- gene=907, id=56 removed by soft count = 204\n",
      "--- gene=907, id=92 removed by soft count = 104\n",
      "--- gene=907, id=114 removed by soft count = 1253\n",
      "--- gene=907, id=150 removed by soft count = 9337\n",
      "--- number of probes: 200 , kept by blast: 192 , if remove dups: 192\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 908\n",
      "--- gene=908, id=9 removed by soft count = 2446\n",
      "--- gene=908, id=24 removed by soft count = 132\n",
      "--- gene=908, id=25 removed by soft count = 52\n",
      "--- gene=908, id=26 removed by soft count = 1692\n",
      "--- gene=908, id=83 removed by soft count = 269\n",
      "--- number of probes: 87 , kept by blast: 82 , if remove dups: 81\n",
      "-- number of probes kept for this region: 82\n",
      "-- checking probes in region: 909\n",
      "--- gene=909, id=1 removed by soft count = 215\n",
      "--- gene=909, id=3 removed by soft count = 461\n",
      "--- gene=909, id=20 removed by soft count = 121\n",
      "--- gene=909, id=34 removed by soft count = 111\n",
      "--- gene=909, id=72 removed by soft count = 172\n",
      "--- gene=909, id=87 removed by soft count = 58\n",
      "--- gene=909, id=94 removed by soft count = 142\n",
      "--- gene=909, id=97 removed by soft count = 87\n",
      "--- gene=909, id=98 removed by soft count = 164\n",
      "--- gene=909, id=106 removed by soft count = 119\n",
      "--- number of probes: 180 , kept by blast: 170 , if remove dups: 170\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 910\n",
      "--- gene=910, id=4 removed by soft count = 52\n",
      "--- gene=910, id=9 removed by soft count = 71\n",
      "--- gene=910, id=14 removed by soft count = 95\n",
      "--- number of probes: 254 , kept by blast: 251 , if remove dups: 251\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 911\n",
      "--- number of probes: 331 , kept by blast: 331 , if remove dups: 329\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 912\n",
      "--- gene=912, id=58 removed by soft count = 67\n",
      "--- gene=912, id=140 removed by soft count = 281\n",
      "--- number of probes: 298 , kept by blast: 296 , if remove dups: 296\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 913\n",
      "--- gene=913, id=92 removed by soft count = 74\n",
      "--- gene=913, id=103 removed by soft count = 92\n",
      "--- gene=913, id=109 removed by soft count = 72\n",
      "--- gene=913, id=147 removed by soft count = 72\n",
      "--- number of probes: 216 , kept by blast: 212 , if remove dups: 212\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 914\n",
      "--- gene=914, id=13 removed by soft count = 381\n",
      "--- gene=914, id=18 removed by soft count = 395\n",
      "--- gene=914, id=45 removed by soft count = 62\n",
      "--- gene=914, id=51 removed by soft count = 504\n",
      "--- gene=914, id=52 removed by soft count = 231\n",
      "--- number of probes: 72 , kept by blast: 67 , if remove dups: 65\n",
      "-- number of probes kept for this region: 67\n",
      "-- checking probes in region: 915\n",
      "--- gene=915, id=90 removed by soft count = 105\n",
      "--- number of probes: 256 , kept by blast: 255 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 916\n",
      "--- gene=916, id=22 removed by soft count = 7569\n",
      "--- gene=916, id=23 removed by soft count = 19944\n",
      "--- gene=916, id=116 removed by soft count = 118\n",
      "--- gene=916, id=168 removed by soft count = 108\n",
      "--- gene=916, id=171 removed by soft count = 108\n",
      "--- gene=916, id=172 removed by soft count = 61\n",
      "--- number of probes: 182 , kept by blast: 176 , if remove dups: 176\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 917\n",
      "--- gene=917, id=30 removed by soft count = 114\n",
      "--- gene=917, id=71 removed by soft count = 751\n",
      "--- gene=917, id=77 removed by soft count = 54\n",
      "--- gene=917, id=82 removed by soft count = 59\n",
      "--- gene=917, id=83 removed by soft count = 237\n",
      "--- gene=917, id=84 removed by soft count = 714\n",
      "--- gene=917, id=85 removed by soft count = 289\n",
      "--- gene=917, id=134 removed by soft count = 983\n",
      "--- gene=917, id=135 removed by soft count = 1978\n",
      "--- number of probes: 159 , kept by blast: 150 , if remove dups: 150\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 918\n",
      "--- gene=918, id=66 removed by soft count = 115\n",
      "--- gene=918, id=67 removed by soft count = 647\n",
      "--- gene=918, id=70 removed by soft count = 147\n",
      "--- gene=918, id=76 removed by soft count = 299\n",
      "--- gene=918, id=77 removed by soft count = 54\n",
      "--- gene=918, id=78 removed by soft count = 159\n",
      "--- gene=918, id=79 removed by soft count = 87\n",
      "--- number of probes: 163 , kept by blast: 156 , if remove dups: 156\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 919\n",
      "--- gene=919, id=23 removed by soft count = 503\n",
      "--- gene=919, id=84 removed by soft count = 73\n",
      "--- gene=919, id=125 removed by soft count = 111\n",
      "--- gene=919, id=128 removed by soft count = 74\n",
      "--- gene=919, id=157 removed by soft count = 68\n",
      "--- gene=919, id=237 removed by soft count = 1239\n",
      "--- number of probes: 273 , kept by blast: 267 , if remove dups: 267\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 920\n",
      "--- gene=920, id=242 removed by soft count = 238\n",
      "--- gene=920, id=251 removed by soft count = 114\n",
      "--- gene=920, id=253 removed by soft count = 108\n",
      "--- gene=920, id=255 removed by soft count = 181\n",
      "--- number of probes: 287 , kept by blast: 283 , if remove dups: 283\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 921\n",
      "--- number of probes: 325 , kept by blast: 325 , if remove dups: 325\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 922\n",
      "--- gene=922, id=103 removed by soft count = 386\n",
      "--- gene=922, id=144 removed by soft count = 5979\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=922, id=151 removed by soft count = 120\n",
      "--- gene=922, id=152 removed by soft count = 232\n",
      "--- gene=922, id=180 removed by soft count = 272\n",
      "--- gene=922, id=181 removed by soft count = 429\n",
      "--- number of probes: 285 , kept by blast: 279 , if remove dups: 279\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 923\n",
      "--- gene=923, id=16 removed by soft count = 95\n",
      "--- gene=923, id=28 removed by soft count = 76\n",
      "--- gene=923, id=39 removed by soft count = 167\n",
      "--- gene=923, id=87 removed by soft count = 246\n",
      "--- gene=923, id=92 removed by soft count = 212\n",
      "--- gene=923, id=94 removed by soft count = 495\n",
      "--- gene=923, id=95 removed by soft count = 1221\n",
      "--- gene=923, id=101 removed by soft count = 110\n",
      "--- gene=923, id=103 removed by soft count = 67\n",
      "--- gene=923, id=113 removed by soft count = 60\n",
      "--- number of probes: 160 , kept by blast: 150 , if remove dups: 150\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 924\n",
      "--- gene=924, id=43 removed by soft count = 276\n",
      "--- gene=924, id=52 removed by soft count = 245\n",
      "--- gene=924, id=54 removed by soft count = 111\n",
      "--- gene=924, id=125 removed by soft count = 57\n",
      "--- number of probes: 197 , kept by blast: 193 , if remove dups: 193\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 925\n",
      "--- gene=925, id=15 removed by soft count = 103\n",
      "--- gene=925, id=19 removed by soft count = 269\n",
      "--- gene=925, id=20 removed by soft count = 296\n",
      "--- gene=925, id=21 removed by soft count = 75\n",
      "--- gene=925, id=24 removed by soft count = 102\n",
      "--- gene=925, id=27 removed by soft count = 63\n",
      "--- gene=925, id=54 removed by soft count = 87\n",
      "--- gene=925, id=56 removed by soft count = 337\n",
      "--- gene=925, id=57 removed by soft count = 1509\n",
      "--- number of probes: 79 , kept by blast: 70 , if remove dups: 70\n",
      "-- number of probes kept for this region: 70\n",
      "-- checking probes in region: 926\n",
      "--- gene=926, id=77 removed by soft count = 58\n",
      "--- gene=926, id=80 removed by soft count = 104\n",
      "--- gene=926, id=83 removed by soft count = 253\n",
      "--- gene=926, id=124 removed by soft count = 55\n",
      "--- gene=926, id=137 removed by soft count = 87\n",
      "--- gene=926, id=141 removed by soft count = 102\n",
      "--- gene=926, id=163 removed by soft count = 222\n",
      "--- gene=926, id=216 removed by hard count = 3\n",
      "--- gene=926, id=221 removed by soft count = 181\n",
      "--- gene=926, id=253 removed by soft count = 147\n",
      "--- number of probes: 254 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 927\n",
      "--- number of probes: 303 , kept by blast: 303 , if remove dups: 303\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 928\n",
      "--- number of probes: 266 , kept by blast: 266 , if remove dups: 266\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 929\n",
      "--- gene=929, id=11 removed by soft count = 167\n",
      "--- gene=929, id=14 removed by soft count = 615\n",
      "--- gene=929, id=15 removed by soft count = 66\n",
      "--- gene=929, id=76 removed by soft count = 82\n",
      "--- gene=929, id=78 removed by soft count = 317\n",
      "--- gene=929, id=79 removed by soft count = 468\n",
      "--- gene=929, id=80 removed by soft count = 350\n",
      "--- gene=929, id=81 removed by soft count = 291\n",
      "--- gene=929, id=82 removed by soft count = 78\n",
      "--- gene=929, id=85 removed by soft count = 1436\n",
      "--- gene=929, id=86 removed by soft count = 266\n",
      "--- gene=929, id=87 removed by soft count = 95\n",
      "--- gene=929, id=89 removed by soft count = 348\n",
      "--- gene=929, id=90 removed by soft count = 96\n",
      "--- gene=929, id=99 removed by soft count = 420\n",
      "--- gene=929, id=119 removed by soft count = 255\n",
      "--- gene=929, id=120 removed by soft count = 175\n",
      "--- gene=929, id=122 removed by soft count = 238\n",
      "--- gene=929, id=123 removed by soft count = 816\n",
      "--- number of probes: 125 , kept by blast: 106 , if remove dups: 106\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 106\n",
      "-- checking probes in region: 930\n",
      "--- gene=930, id=33 removed by soft count = 165\n",
      "--- gene=930, id=35 removed by soft count = 370\n",
      "--- gene=930, id=37 removed by soft count = 315\n",
      "--- gene=930, id=40 removed by soft count = 133\n",
      "--- gene=930, id=61 removed by soft count = 588\n",
      "--- gene=930, id=62 removed by soft count = 301\n",
      "--- gene=930, id=63 removed by soft count = 620\n",
      "--- gene=930, id=64 removed by soft count = 41743\n",
      "--- number of probes: 81 , kept by blast: 73 , if remove dups: 73\n",
      "-- number of probes kept for this region: 73\n",
      "-- checking probes in region: 932\n",
      "--- gene=932, id=39 removed by soft count = 237\n",
      "--- gene=932, id=40 removed by soft count = 854\n",
      "--- gene=932, id=43 removed by soft count = 143\n",
      "--- gene=932, id=44 removed by soft count = 2794\n",
      "--- gene=932, id=45 removed by soft count = 899\n",
      "--- gene=932, id=47 removed by soft count = 101\n",
      "--- gene=932, id=54 removed by soft count = 4143\n",
      "--- gene=932, id=55 removed by soft count = 160\n",
      "--- gene=932, id=56 removed by soft count = 8895\n",
      "--- gene=932, id=57 removed by soft count = 993\n",
      "--- gene=932, id=60 removed by soft count = 137\n",
      "--- gene=932, id=61 removed by soft count = 154\n",
      "--- gene=932, id=63 removed by soft count = 189\n",
      "--- gene=932, id=64 removed by soft count = 100\n",
      "--- number of probes: 66 , kept by blast: 52 , if remove dups: 52\n",
      "-- number of probes kept for this region: 52\n",
      "-- checking probes in region: 933\n",
      "--- gene=933, id=56 removed by soft count = 334\n",
      "--- gene=933, id=75 removed by soft count = 82\n",
      "--- gene=933, id=135 removed by soft count = 107\n",
      "--- gene=933, id=149 removed by soft count = 331\n",
      "--- number of probes: 186 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 934\n",
      "--- gene=934, id=15 removed by soft count = 1822\n",
      "--- gene=934, id=22 removed by soft count = 886\n",
      "--- gene=934, id=24 removed by soft count = 281\n",
      "--- number of probes: 198 , kept by blast: 195 , if remove dups: 194\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 935\n",
      "--- gene=935, id=94 removed by soft count = 176\n",
      "--- gene=935, id=112 removed by soft count = 1322\n",
      "--- number of probes: 230 , kept by blast: 228 , if remove dups: 228\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 936\n",
      "--- gene=936, id=143 removed by soft count = 54\n",
      "--- number of probes: 293 , kept by blast: 292 , if remove dups: 292\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 937\n",
      "--- gene=937, id=0 removed by soft count = 338\n",
      "--- gene=937, id=1 removed by soft count = 217\n",
      "--- gene=937, id=2 removed by soft count = 204\n",
      "--- gene=937, id=31 removed by soft count = 72\n",
      "--- gene=937, id=67 removed by soft count = 152\n",
      "--- gene=937, id=141 removed by soft count = 97\n",
      "--- number of probes: 200 , kept by blast: 194 , if remove dups: 194\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 938\n",
      "--- gene=938, id=7 removed by soft count = 263\n",
      "--- gene=938, id=14 removed by soft count = 376\n",
      "--- gene=938, id=21 removed by soft count = 103\n",
      "--- gene=938, id=22 removed by soft count = 177\n",
      "--- gene=938, id=23 removed by soft count = 560\n",
      "--- gene=938, id=28 removed by soft count = 159\n",
      "--- gene=938, id=29 removed by soft count = 142\n",
      "--- gene=938, id=30 removed by soft count = 436\n",
      "--- gene=938, id=31 removed by soft count = 121\n",
      "--- gene=938, id=35 removed by soft count = 72\n",
      "--- gene=938, id=43 removed by soft count = 654\n",
      "--- gene=938, id=44 removed by soft count = 906\n",
      "--- gene=938, id=47 removed by soft count = 417\n",
      "--- gene=938, id=53 removed by soft count = 86\n",
      "--- gene=938, id=54 removed by soft count = 282\n",
      "--- gene=938, id=55 removed by soft count = 110\n",
      "--- gene=938, id=60 removed by soft count = 190\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=938, id=61 removed by soft count = 1197\n",
      "--- gene=938, id=62 removed by soft count = 266\n",
      "--- gene=938, id=63 removed by soft count = 752\n",
      "--- gene=938, id=68 removed by soft count = 53\n",
      "--- gene=938, id=69 removed by soft count = 141\n",
      "--- gene=938, id=70 removed by soft count = 71\n",
      "--- gene=938, id=72 removed by soft count = 172\n",
      "--- gene=938, id=74 removed by soft count = 353\n",
      "--- gene=938, id=75 removed by soft count = 116\n",
      "--- gene=938, id=79 removed by soft count = 348\n",
      "--- gene=938, id=80 removed by soft count = 333\n",
      "--- gene=938, id=85 removed by soft count = 63\n",
      "--- gene=938, id=110 removed by soft count = 94\n",
      "--- number of probes: 164 , kept by blast: 134 , if remove dups: 134\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 939\n",
      "--- gene=939, id=16 removed by soft count = 67\n",
      "--- number of probes: 292 , kept by blast: 291 , if remove dups: 291\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 940\n",
      "--- gene=940, id=35 removed by soft count = 667\n",
      "--- gene=940, id=40 removed by soft count = 120\n",
      "--- number of probes: 258 , kept by blast: 256 , if remove dups: 256\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 941\n",
      "--- gene=941, id=38 removed by soft count = 63\n",
      "--- gene=941, id=44 removed by soft count = 133\n",
      "--- gene=941, id=46 removed by soft count = 53\n",
      "--- gene=941, id=49 removed by soft count = 519\n",
      "--- gene=941, id=51 removed by soft count = 99\n",
      "--- gene=941, id=96 removed by soft count = 103\n",
      "--- gene=941, id=97 removed by soft count = 2695\n",
      "--- gene=941, id=99 removed by soft count = 363\n",
      "--- gene=941, id=169 removed by soft count = 51\n",
      "--- gene=941, id=175 removed by soft count = 326\n",
      "--- gene=941, id=176 removed by soft count = 2355\n",
      "--- gene=941, id=177 removed by soft count = 782\n",
      "--- number of probes: 184 , kept by blast: 172 , if remove dups: 172\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 942\n",
      "--- gene=942, id=82 removed by soft count = 149\n",
      "--- gene=942, id=94 removed by soft count = 314\n",
      "--- gene=942, id=97 removed by soft count = 1644\n",
      "--- gene=942, id=119 removed by soft count = 69\n",
      "--- gene=942, id=120 removed by soft count = 79\n",
      "--- number of probes: 136 , kept by blast: 131 , if remove dups: 131\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 943\n",
      "--- gene=943, id=45 removed by soft count = 82\n",
      "--- gene=943, id=127 removed by soft count = 159\n",
      "--- number of probes: 173 , kept by blast: 171 , if remove dups: 171\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 944\n",
      "--- gene=944, id=8 removed by soft count = 86\n",
      "--- gene=944, id=86 removed by soft count = 93\n",
      "--- number of probes: 257 , kept by blast: 255 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 945\n",
      "--- gene=945, id=0 removed by soft count = 784\n",
      "--- gene=945, id=2 removed by soft count = 810\n",
      "--- gene=945, id=3 removed by soft count = 67\n",
      "--- gene=945, id=10 removed by soft count = 80\n",
      "--- gene=945, id=49 removed by soft count = 56\n",
      "--- gene=945, id=51 removed by soft count = 484\n",
      "--- gene=945, id=54 removed by soft count = 192\n",
      "--- gene=945, id=56 removed by soft count = 72\n",
      "--- gene=945, id=106 removed by soft count = 82\n",
      "--- gene=945, id=149 removed by soft count = 250\n",
      "--- number of probes: 216 , kept by blast: 206 , if remove dups: 206\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 946\n",
      "--- gene=946, id=15 removed by soft count = 1146\n",
      "--- gene=946, id=56 removed by soft count = 815\n",
      "--- gene=946, id=57 removed by soft count = 797\n",
      "--- gene=946, id=58 removed by soft count = 700\n",
      "--- gene=946, id=61 removed by soft count = 717\n",
      "--- gene=946, id=72 removed by soft count = 774\n",
      "--- gene=946, id=74 removed by soft count = 691\n",
      "--- gene=946, id=75 removed by soft count = 671\n",
      "--- gene=946, id=76 removed by soft count = 102\n",
      "--- gene=946, id=77 removed by soft count = 55\n",
      "--- gene=946, id=83 removed by soft count = 76\n",
      "--- gene=946, id=84 removed by soft count = 96\n",
      "--- gene=946, id=91 removed by soft count = 73\n",
      "--- gene=946, id=92 removed by soft count = 138\n",
      "--- gene=946, id=99 removed by soft count = 97\n",
      "--- gene=946, id=218 removed by soft count = 544\n",
      "--- gene=946, id=256 removed by soft count = 234\n",
      "--- gene=946, id=261 removed by soft count = 139\n",
      "--- number of probes: 262 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 947\n",
      "--- gene=947, id=225 removed by soft count = 506\n",
      "--- number of probes: 244 , kept by blast: 243 , if remove dups: 243\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 948\n",
      "--- gene=948, id=40 removed by soft count = 386\n",
      "--- gene=948, id=139 removed by soft count = 3112\n",
      "--- gene=948, id=155 removed by soft count = 821\n",
      "--- gene=948, id=164 removed by soft count = 268\n",
      "--- gene=948, id=165 removed by soft count = 93\n",
      "--- number of probes: 187 , kept by blast: 182 , if remove dups: 182\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 949\n",
      "--- gene=949, id=55 removed by soft count = 110\n",
      "--- gene=949, id=217 removed by soft count = 192\n",
      "--- number of probes: 247 , kept by blast: 245 , if remove dups: 245\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 950\n",
      "--- gene=950, id=39 removed by soft count = 162\n",
      "--- gene=950, id=40 removed by soft count = 1119\n",
      "--- gene=950, id=41 removed by soft count = 122\n",
      "--- gene=950, id=42 removed by soft count = 536\n",
      "--- gene=950, id=50 removed by soft count = 284\n",
      "--- gene=950, id=52 removed by soft count = 78\n",
      "--- gene=950, id=53 removed by soft count = 67\n",
      "--- gene=950, id=57 removed by soft count = 316\n",
      "--- number of probes: 154 , kept by blast: 146 , if remove dups: 146\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 951\n",
      "--- gene=951, id=58 removed by soft count = 101\n",
      "--- gene=951, id=211 removed by soft count = 239\n",
      "--- number of probes: 212 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 952\n",
      "--- gene=952, id=98 removed by soft count = 125\n",
      "--- gene=952, id=125 removed by soft count = 279\n",
      "--- gene=952, id=146 removed by soft count = 122\n",
      "--- gene=952, id=229 removed by soft count = 108\n",
      "--- gene=952, id=243 removed by soft count = 63\n",
      "--- number of probes: 260 , kept by blast: 255 , if remove dups: 255\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 953\n",
      "--- number of probes: 244 , kept by blast: 244 , if remove dups: 244\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 954\n",
      "--- gene=954, id=33 removed by soft count = 71\n",
      "--- gene=954, id=47 removed by soft count = 81\n",
      "--- gene=954, id=54 removed by soft count = 256\n",
      "--- gene=954, id=118 removed by soft count = 70\n",
      "--- gene=954, id=119 removed by soft count = 57\n",
      "--- gene=954, id=163 removed by soft count = 1117\n",
      "--- number of probes: 190 , kept by blast: 184 , if remove dups: 184\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 955\n",
      "--- gene=955, id=121 removed by soft count = 580\n",
      "--- number of probes: 270 , kept by blast: 269 , if remove dups: 269\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 956\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=956, id=51 removed by soft count = 67\n",
      "--- gene=956, id=112 removed by soft count = 197\n",
      "--- gene=956, id=193 removed by soft count = 178\n",
      "--- number of probes: 196 , kept by blast: 193 , if remove dups: 193\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 957\n",
      "--- gene=957, id=37 removed by soft count = 31899\n",
      "--- gene=957, id=86 removed by soft count = 78\n",
      "--- gene=957, id=99 removed by soft count = 139\n",
      "--- number of probes: 238 , kept by blast: 235 , if remove dups: 235\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 958\n",
      "--- gene=958, id=10 removed by soft count = 165\n",
      "--- gene=958, id=150 removed by soft count = 154\n",
      "--- gene=958, id=151 removed by soft count = 132\n",
      "--- gene=958, id=192 removed by soft count = 222\n",
      "--- gene=958, id=194 removed by soft count = 85\n",
      "--- number of probes: 254 , kept by blast: 249 , if remove dups: 248\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 959\n",
      "--- gene=959, id=10 removed by soft count = 65\n",
      "--- gene=959, id=33 removed by soft count = 202\n",
      "--- gene=959, id=34 removed by soft count = 64\n",
      "--- gene=959, id=39 removed by soft count = 405\n",
      "--- gene=959, id=56 removed by soft count = 101\n",
      "--- gene=959, id=100 removed by soft count = 177\n",
      "--- gene=959, id=104 removed by soft count = 151\n",
      "--- gene=959, id=106 removed by soft count = 178\n",
      "--- gene=959, id=108 removed by soft count = 72\n",
      "--- gene=959, id=112 removed by soft count = 168\n",
      "--- gene=959, id=129 removed by soft count = 298\n",
      "--- number of probes: 209 , kept by blast: 198 , if remove dups: 197\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 960\n",
      "--- gene=960, id=8 removed by soft count = 201\n",
      "--- gene=960, id=9 removed by soft count = 62\n",
      "--- gene=960, id=10 removed by soft count = 70\n",
      "--- gene=960, id=33 removed by soft count = 241\n",
      "--- gene=960, id=122 removed by soft count = 80\n",
      "--- gene=960, id=180 removed by soft count = 1550\n",
      "--- gene=960, id=257 removed by soft count = 68\n",
      "--- gene=960, id=259 removed by soft count = 62\n",
      "--- number of probes: 288 , kept by blast: 280 , if remove dups: 280\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 961\n",
      "--- gene=961, id=236 removed by soft count = 59\n",
      "--- number of probes: 284 , kept by blast: 283 , if remove dups: 283\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 962\n",
      "--- gene=962, id=208 removed by soft count = 162\n",
      "--- number of probes: 278 , kept by blast: 277 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 963\n",
      "--- gene=963, id=48 removed by soft count = 466\n",
      "--- gene=963, id=108 removed by soft count = 271\n",
      "--- gene=963, id=110 removed by soft count = 54\n",
      "--- gene=963, id=239 removed by soft count = 5474\n",
      "--- gene=963, id=240 removed by soft count = 1101\n",
      "--- gene=963, id=254 removed by soft count = 6867\n",
      "--- number of probes: 282 , kept by blast: 276 , if remove dups: 276\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 964\n",
      "--- gene=964, id=88 removed by soft count = 53\n",
      "--- gene=964, id=92 removed by soft count = 105\n",
      "--- gene=964, id=93 removed by soft count = 587\n",
      "--- gene=964, id=95 removed by soft count = 124\n",
      "--- gene=964, id=96 removed by soft count = 76\n",
      "--- gene=964, id=100 removed by soft count = 117\n",
      "--- gene=964, id=224 removed by soft count = 122\n",
      "--- gene=964, id=252 removed by soft count = 144\n",
      "--- number of probes: 269 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 965\n",
      "--- gene=965, id=22 removed by soft count = 67\n",
      "--- gene=965, id=69 removed by soft count = 705\n",
      "--- number of probes: 125 , kept by blast: 123 , if remove dups: 121\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 966\n",
      "--- gene=966, id=6 removed by soft count = 1299\n",
      "--- gene=966, id=9 removed by soft count = 71\n",
      "--- gene=966, id=12 removed by soft count = 548\n",
      "--- gene=966, id=38 removed by soft count = 59\n",
      "--- gene=966, id=40 removed by soft count = 64\n",
      "--- gene=966, id=147 removed by soft count = 61\n",
      "--- gene=966, id=151 removed by soft count = 299\n",
      "--- gene=966, id=153 removed by soft count = 253\n",
      "--- gene=966, id=154 removed by soft count = 94\n",
      "--- gene=966, id=155 removed by soft count = 52\n",
      "--- number of probes: 213 , kept by blast: 203 , if remove dups: 202\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 969\n",
      "--- gene=969, id=113 removed by soft count = 143\n",
      "--- gene=969, id=134 removed by soft count = 90\n",
      "--- gene=969, id=170 removed by soft count = 125\n",
      "--- number of probes: 182 , kept by blast: 179 , if remove dups: 179\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 970\n",
      "--- number of probes: 306 , kept by blast: 306 , if remove dups: 306\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 971\n",
      "--- gene=971, id=168 removed by soft count = 106\n",
      "--- gene=971, id=169 removed by soft count = 327\n",
      "--- number of probes: 262 , kept by blast: 260 , if remove dups: 260\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 972\n",
      "--- gene=972, id=7 removed by soft count = 1546\n",
      "--- gene=972, id=15 removed by soft count = 120\n",
      "--- gene=972, id=61 removed by soft count = 77\n",
      "--- gene=972, id=62 removed by soft count = 81\n",
      "--- gene=972, id=63 removed by soft count = 73\n",
      "--- gene=972, id=79 removed by soft count = 78\n",
      "--- gene=972, id=130 removed by soft count = 326\n",
      "--- gene=972, id=131 removed by soft count = 128\n",
      "--- number of probes: 235 , kept by blast: 227 , if remove dups: 227\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 973\n",
      "--- gene=973, id=35 removed by soft count = 81\n",
      "--- gene=973, id=103 removed by soft count = 212\n",
      "--- gene=973, id=104 removed by soft count = 1092\n",
      "--- gene=973, id=106 removed by soft count = 67\n",
      "--- gene=973, id=109 removed by soft count = 276\n",
      "--- gene=973, id=112 removed by soft count = 57\n",
      "--- gene=973, id=115 removed by soft count = 53\n",
      "--- number of probes: 121 , kept by blast: 114 , if remove dups: 114\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 114\n",
      "-- checking probes in region: 974\n",
      "--- gene=974, id=65 removed by soft count = 82\n",
      "--- number of probes: 283 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 975\n",
      "--- gene=975, id=1 removed by soft count = 58\n",
      "--- gene=975, id=21 removed by soft count = 128\n",
      "--- gene=975, id=22 removed by soft count = 81\n",
      "--- gene=975, id=193 removed by soft count = 146\n",
      "--- number of probes: 276 , kept by blast: 272 , if remove dups: 272\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 976\n",
      "--- gene=976, id=110 removed by soft count = 81\n",
      "--- gene=976, id=112 removed by soft count = 64\n",
      "--- gene=976, id=168 removed by soft count = 72\n",
      "--- gene=976, id=170 removed by soft count = 71\n",
      "--- number of probes: 188 , kept by blast: 184 , if remove dups: 184\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 977\n",
      "--- gene=977, id=6 removed by soft count = 351\n",
      "--- number of probes: 211 , kept by blast: 210 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 978\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=978, id=37 removed by soft count = 182\n",
      "--- gene=978, id=89 removed by soft count = 71\n",
      "--- gene=978, id=96 removed by soft count = 51\n",
      "--- gene=978, id=138 removed by soft count = 170\n",
      "--- number of probes: 190 , kept by blast: 186 , if remove dups: 186\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 979\n",
      "--- gene=979, id=192 removed by soft count = 150\n",
      "--- gene=979, id=193 removed by soft count = 70\n",
      "--- number of probes: 227 , kept by blast: 225 , if remove dups: 224\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 980\n",
      "--- gene=980, id=96 removed by soft count = 190\n",
      "--- gene=980, id=98 removed by soft count = 200\n",
      "--- gene=980, id=125 removed by soft count = 171\n",
      "--- gene=980, id=126 removed by soft count = 84\n",
      "--- gene=980, id=127 removed by soft count = 67\n",
      "--- gene=980, id=195 removed by soft count = 116\n",
      "--- number of probes: 240 , kept by blast: 234 , if remove dups: 234\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 982\n",
      "--- gene=982, id=23 removed by soft count = 55\n",
      "--- gene=982, id=79 removed by soft count = 66\n",
      "--- gene=982, id=94 removed by soft count = 111\n",
      "--- gene=982, id=130 removed by soft count = 83\n",
      "--- gene=982, id=131 removed by soft count = 209\n",
      "--- gene=982, id=141 removed by soft count = 758\n",
      "--- gene=982, id=144 removed by soft count = 274\n",
      "--- gene=982, id=154 removed by soft count = 678\n",
      "--- gene=982, id=197 removed by soft count = 129\n",
      "--- number of probes: 251 , kept by blast: 242 , if remove dups: 242\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 983\n",
      "--- gene=983, id=17 removed by soft count = 51\n",
      "--- gene=983, id=102 removed by soft count = 68\n",
      "--- gene=983, id=217 removed by soft count = 184\n",
      "--- number of probes: 232 , kept by blast: 229 , if remove dups: 229\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 984\n",
      "--- gene=984, id=72 removed by soft count = 252\n",
      "--- gene=984, id=218 removed by soft count = 110\n",
      "--- number of probes: 233 , kept by blast: 231 , if remove dups: 231\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 985\n",
      "--- gene=985, id=44 removed by soft count = 202\n",
      "--- gene=985, id=77 removed by soft count = 128\n",
      "--- gene=985, id=103 removed by soft count = 69\n",
      "--- gene=985, id=106 removed by soft count = 53\n",
      "--- number of probes: 241 , kept by blast: 237 , if remove dups: 237\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 986\n",
      "--- gene=986, id=8 removed by soft count = 528\n",
      "--- gene=986, id=17 removed by soft count = 64\n",
      "--- gene=986, id=46 removed by soft count = 92\n",
      "--- gene=986, id=48 removed by soft count = 62\n",
      "--- number of probes: 269 , kept by blast: 265 , if remove dups: 263\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 987\n",
      "--- number of probes: 237 , kept by blast: 237 , if remove dups: 237\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 988\n",
      "--- gene=988, id=21 removed by soft count = 68\n",
      "--- gene=988, id=55 removed by soft count = 153\n",
      "--- gene=988, id=73 removed by soft count = 727\n",
      "--- gene=988, id=74 removed by soft count = 1468\n",
      "--- gene=988, id=75 removed by soft count = 968\n",
      "--- gene=988, id=78 removed by soft count = 275\n",
      "--- gene=988, id=123 removed by soft count = 153\n",
      "--- number of probes: 139 , kept by blast: 132 , if remove dups: 132\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 989\n",
      "--- gene=989, id=259 removed by soft count = 313\n",
      "--- number of probes: 283 , kept by blast: 282 , if remove dups: 282\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 990\n",
      "--- gene=990, id=180 removed by soft count = 150\n",
      "--- gene=990, id=181 removed by soft count = 138\n",
      "--- gene=990, id=184 removed by soft count = 62\n",
      "--- number of probes: 268 , kept by blast: 265 , if remove dups: 265\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 991\n",
      "--- gene=991, id=226 removed by soft count = 85\n",
      "--- gene=991, id=227 removed by soft count = 61\n",
      "--- gene=991, id=232 removed by soft count = 670\n",
      "--- gene=991, id=236 removed by soft count = 336\n",
      "--- number of probes: 265 , kept by blast: 261 , if remove dups: 261\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 992\n",
      "--- gene=992, id=31 removed by soft count = 81\n",
      "--- gene=992, id=32 removed by soft count = 70\n",
      "--- gene=992, id=33 removed by soft count = 244\n",
      "--- gene=992, id=38 removed by soft count = 151\n",
      "--- number of probes: 100 , kept by blast: 96 , if remove dups: 96\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 96\n",
      "-- checking probes in region: 995\n",
      "--- gene=995, id=31 removed by soft count = 122\n",
      "--- gene=995, id=32 removed by soft count = 103\n",
      "--- gene=995, id=194 removed by soft count = 70\n",
      "--- gene=995, id=219 removed by soft count = 55\n",
      "--- number of probes: 224 , kept by blast: 220 , if remove dups: 220\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 997\n",
      "--- gene=997, id=20 removed by soft count = 99\n",
      "--- gene=997, id=117 removed by soft count = 194\n",
      "--- gene=997, id=121 removed by soft count = 68\n",
      "--- gene=997, id=185 removed by soft count = 68\n",
      "--- number of probes: 305 , kept by blast: 301 , if remove dups: 301\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 998\n",
      "--- gene=998, id=28 removed by soft count = 73\n",
      "--- gene=998, id=31 removed by soft count = 119\n",
      "--- gene=998, id=40 removed by soft count = 94\n",
      "--- gene=998, id=50 removed by soft count = 1324\n",
      "--- gene=998, id=55 removed by soft count = 52\n",
      "--- number of probes: 149 , kept by blast: 144 , if remove dups: 144\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 999\n",
      "--- gene=999, id=36 removed by soft count = 78\n",
      "--- gene=999, id=52 removed by soft count = 200\n",
      "--- gene=999, id=53 removed by soft count = 82\n",
      "--- gene=999, id=54 removed by soft count = 749\n",
      "--- gene=999, id=55 removed by soft count = 80\n",
      "--- gene=999, id=56 removed by soft count = 1222\n",
      "--- gene=999, id=58 removed by soft count = 152\n",
      "--- gene=999, id=59 removed by soft count = 917\n",
      "--- gene=999, id=61 removed by soft count = 1038\n",
      "--- gene=999, id=63 removed by soft count = 374\n",
      "--- gene=999, id=65 removed by soft count = 66\n",
      "--- gene=999, id=66 removed by soft count = 2767\n",
      "--- gene=999, id=68 removed by soft count = 113\n",
      "--- gene=999, id=70 removed by soft count = 175\n",
      "--- gene=999, id=77 removed by soft count = 103\n",
      "--- gene=999, id=81 removed by soft count = 70\n",
      "--- gene=999, id=82 removed by soft count = 78\n",
      "--- gene=999, id=86 removed by soft count = 227\n",
      "--- gene=999, id=130 removed by soft count = 289\n",
      "--- gene=999, id=132 removed by soft count = 1128\n",
      "--- number of probes: 133 , kept by blast: 113 , if remove dups: 113\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 113\n",
      "-- checking probes in region: 1000\n",
      "--- gene=1000, id=20 removed by soft count = 67\n",
      "--- gene=1000, id=21 removed by soft count = 764\n",
      "--- gene=1000, id=22 removed by soft count = 455\n",
      "--- gene=1000, id=24 removed by soft count = 433\n",
      "--- gene=1000, id=26 removed by soft count = 1670\n",
      "--- gene=1000, id=27 removed by soft count = 71\n",
      "--- gene=1000, id=28 removed by soft count = 226\n",
      "--- gene=1000, id=31 removed by soft count = 110\n",
      "--- number of probes: 55 , kept by blast: 47 , if remove dups: 47\n",
      "-- number of probes kept for this region: 47\n",
      "-- checking probes in region: 1001\n",
      "--- number of probes: 69 , kept by blast: 69 , if remove dups: 69\n",
      "-- number of probes kept for this region: 69\n",
      "-- checking probes in region: 1002\n",
      "--- gene=1002, id=13 removed by soft count = 52\n",
      "--- gene=1002, id=28 removed by soft count = 56\n",
      "--- gene=1002, id=29 removed by soft count = 3585\n",
      "--- number of probes: 117 , kept by blast: 114 , if remove dups: 114\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 114\n",
      "-- checking probes in region: 1007\n",
      "--- gene=1007, id=183 removed by soft count = 76\n",
      "--- gene=1007, id=203 removed by soft count = 134\n",
      "--- gene=1007, id=204 removed by soft count = 57\n",
      "--- number of probes: 209 , kept by blast: 206 , if remove dups: 206\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1008\n",
      "--- gene=1008, id=6 removed by soft count = 59\n",
      "--- gene=1008, id=10 removed by soft count = 243\n",
      "--- gene=1008, id=15 removed by soft count = 71\n",
      "--- gene=1008, id=38 removed by soft count = 354\n",
      "--- gene=1008, id=107 removed by soft count = 224\n",
      "--- gene=1008, id=116 removed by soft count = 234\n",
      "--- number of probes: 136 , kept by blast: 130 , if remove dups: 130\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1009\n",
      "--- gene=1009, id=74 removed by soft count = 57\n",
      "--- number of probes: 152 , kept by blast: 151 , if remove dups: 151\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1010\n",
      "--- gene=1010, id=48 removed by soft count = 433\n",
      "--- gene=1010, id=49 removed by soft count = 387\n",
      "--- gene=1010, id=50 removed by soft count = 374\n",
      "--- gene=1010, id=51 removed by soft count = 2177\n",
      "--- gene=1010, id=63 removed by soft count = 1862\n",
      "--- gene=1010, id=76 removed by soft count = 763\n",
      "--- gene=1010, id=78 removed by soft count = 137\n",
      "--- gene=1010, id=85 removed by soft count = 1008\n",
      "--- gene=1010, id=86 removed by soft count = 107\n",
      "--- gene=1010, id=87 removed by soft count = 589\n",
      "--- number of probes: 110 , kept by blast: 100 , if remove dups: 100\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 100\n",
      "-- checking probes in region: 1011\n",
      "--- gene=1011, id=10 removed by soft count = 182\n",
      "--- gene=1011, id=16 removed by soft count = 65\n",
      "--- gene=1011, id=20 removed by soft count = 13935\n",
      "--- gene=1011, id=22 removed by soft count = 195\n",
      "--- gene=1011, id=23 removed by soft count = 61\n",
      "--- gene=1011, id=25 removed by soft count = 377\n",
      "--- gene=1011, id=44 removed by soft count = 84\n",
      "--- gene=1011, id=73 removed by soft count = 68\n",
      "--- gene=1011, id=75 removed by soft count = 64\n",
      "--- gene=1011, id=76 removed by soft count = 66\n",
      "--- gene=1011, id=82 removed by soft count = 59\n",
      "--- gene=1011, id=86 removed by soft count = 69\n",
      "--- gene=1011, id=91 removed by soft count = 65\n",
      "--- gene=1011, id=107 removed by soft count = 53\n",
      "--- gene=1011, id=111 removed by soft count = 58\n",
      "--- gene=1011, id=113 removed by soft count = 99\n",
      "--- gene=1011, id=118 removed by soft count = 249\n",
      "--- gene=1011, id=140 removed by soft count = 82\n",
      "--- gene=1011, id=214 removed by hard count = 3\n",
      "--- gene=1011, id=216 removed by hard count = 3\n",
      "--- number of probes: 220 , kept by blast: 200 , if remove dups: 193\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1013\n",
      "--- gene=1013, id=17 removed by soft count = 101\n",
      "--- gene=1013, id=38 removed by soft count = 56\n",
      "--- gene=1013, id=40 removed by soft count = 136\n",
      "--- gene=1013, id=55 removed by soft count = 59\n",
      "--- gene=1013, id=57 removed by soft count = 155\n",
      "--- gene=1013, id=60 removed by soft count = 193\n",
      "--- gene=1013, id=61 removed by soft count = 76\n",
      "--- gene=1013, id=99 removed by soft count = 156\n",
      "--- gene=1013, id=103 removed by soft count = 650\n",
      "--- gene=1013, id=107 removed by soft count = 70\n",
      "--- number of probes: 146 , kept by blast: 136 , if remove dups: 136\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1014\n",
      "--- gene=1014, id=13 removed by soft count = 246\n",
      "--- gene=1014, id=61 removed by soft count = 58\n",
      "--- gene=1014, id=64 removed by soft count = 137\n",
      "--- number of probes: 205 , kept by blast: 202 , if remove dups: 202\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1015\n",
      "--- gene=1015, id=0 removed by soft count = 210\n",
      "--- gene=1015, id=2 removed by soft count = 118\n",
      "--- gene=1015, id=6 removed by soft count = 54\n",
      "--- gene=1015, id=8 removed by soft count = 166\n",
      "--- gene=1015, id=44 removed by soft count = 370\n",
      "--- gene=1015, id=47 removed by soft count = 207\n",
      "--- gene=1015, id=58 removed by soft count = 305\n",
      "--- gene=1015, id=59 removed by soft count = 209\n",
      "--- gene=1015, id=61 removed by soft count = 644\n",
      "--- gene=1015, id=62 removed by soft count = 270\n",
      "--- number of probes: 65 , kept by blast: 55 , if remove dups: 55\n",
      "-- number of probes kept for this region: 55\n",
      "-- checking probes in region: 1016\n",
      "--- gene=1016, id=27 removed by soft count = 65\n",
      "--- gene=1016, id=28 removed by soft count = 1009\n",
      "--- gene=1016, id=29 removed by soft count = 168\n",
      "--- gene=1016, id=31 removed by soft count = 357\n",
      "--- gene=1016, id=32 removed by soft count = 595\n",
      "--- gene=1016, id=59 removed by soft count = 107\n",
      "--- gene=1016, id=60 removed by soft count = 128\n",
      "--- gene=1016, id=62 removed by soft count = 341\n",
      "--- gene=1016, id=64 removed by soft count = 94\n",
      "--- gene=1016, id=65 removed by soft count = 97\n",
      "--- gene=1016, id=146 removed by soft count = 3532\n",
      "--- number of probes: 176 , kept by blast: 165 , if remove dups: 165\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1017\n",
      "--- gene=1017, id=0 removed by soft count = 197\n",
      "--- gene=1017, id=1 removed by soft count = 52\n",
      "--- number of probes: 117 , kept by blast: 115 , if remove dups: 115\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 115\n",
      "-- checking probes in region: 1018\n",
      "--- gene=1018, id=0 removed by soft count = 639\n",
      "--- gene=1018, id=1 removed by soft count = 52\n",
      "--- gene=1018, id=2 removed by soft count = 76\n",
      "--- gene=1018, id=5 removed by soft count = 87\n",
      "--- gene=1018, id=7 removed by soft count = 124\n",
      "--- gene=1018, id=54 removed by soft count = 76\n",
      "--- gene=1018, id=56 removed by soft count = 1506\n",
      "--- gene=1018, id=65 removed by soft count = 358\n",
      "--- gene=1018, id=81 removed by soft count = 135\n",
      "--- gene=1018, id=84 removed by soft count = 62\n",
      "--- gene=1018, id=85 removed by soft count = 147\n",
      "--- number of probes: 145 , kept by blast: 134 , if remove dups: 133\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1019\n",
      "--- gene=1019, id=65 removed by soft count = 84\n",
      "--- gene=1019, id=82 removed by soft count = 314\n",
      "--- number of probes: 204 , kept by blast: 202 , if remove dups: 200\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1020\n",
      "--- gene=1020, id=10 removed by soft count = 335\n",
      "--- gene=1020, id=88 removed by soft count = 99\n",
      "--- gene=1020, id=89 removed by soft count = 181\n",
      "--- gene=1020, id=91 removed by soft count = 95\n",
      "--- gene=1020, id=92 removed by soft count = 138\n",
      "--- gene=1020, id=93 removed by soft count = 56\n",
      "--- number of probes: 133 , kept by blast: 127 , if remove dups: 127\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1021\n",
      "--- gene=1021, id=3 removed by soft count = 240\n",
      "--- gene=1021, id=6 removed by soft count = 189\n",
      "--- gene=1021, id=29 removed by soft count = 187\n",
      "--- number of probes: 173 , kept by blast: 170 , if remove dups: 170\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1022\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=1022, id=13 removed by soft count = 68\n",
      "--- gene=1022, id=57 removed by soft count = 1666\n",
      "--- gene=1022, id=74 removed by soft count = 129\n",
      "--- number of probes: 196 , kept by blast: 193 , if remove dups: 193\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1023\n",
      "--- gene=1023, id=103 removed by soft count = 54\n",
      "--- gene=1023, id=106 removed by soft count = 143\n",
      "--- gene=1023, id=109 removed by soft count = 619\n",
      "--- gene=1023, id=111 removed by soft count = 314\n",
      "--- number of probes: 112 , kept by blast: 108 , if remove dups: 106\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 106\n",
      "-- checking probes in region: 1024\n",
      "--- gene=1024, id=0 removed by soft count = 1284\n",
      "--- gene=1024, id=1 removed by soft count = 120\n",
      "--- gene=1024, id=2 removed by soft count = 154\n",
      "--- gene=1024, id=3 removed by soft count = 307\n",
      "--- gene=1024, id=4 removed by soft count = 489\n",
      "--- gene=1024, id=5 removed by soft count = 273\n",
      "--- gene=1024, id=6 removed by soft count = 80\n",
      "--- gene=1024, id=7 removed by soft count = 23939\n",
      "--- gene=1024, id=8 removed by soft count = 317\n",
      "--- gene=1024, id=9 removed by soft count = 345\n",
      "--- gene=1024, id=10 removed by soft count = 569\n",
      "--- gene=1024, id=11 removed by soft count = 256\n",
      "--- gene=1024, id=19 removed by soft count = 94\n",
      "--- gene=1024, id=48 removed by soft count = 103\n",
      "--- gene=1024, id=72 removed by soft count = 133\n",
      "--- gene=1024, id=74 removed by soft count = 374\n",
      "--- gene=1024, id=76 removed by soft count = 140\n",
      "--- number of probes: 87 , kept by blast: 70 , if remove dups: 70\n",
      "-- number of probes kept for this region: 70\n",
      "-- checking probes in region: 1025\n",
      "--- gene=1025, id=0 removed by soft count = 2406\n",
      "--- gene=1025, id=1 removed by soft count = 155\n",
      "--- gene=1025, id=3 removed by soft count = 111\n",
      "--- gene=1025, id=4 removed by soft count = 296\n",
      "--- gene=1025, id=5 removed by soft count = 216\n",
      "--- gene=1025, id=6 removed by soft count = 457\n",
      "--- gene=1025, id=11 removed by soft count = 223\n",
      "--- gene=1025, id=12 removed by soft count = 55\n",
      "--- gene=1025, id=13 removed by soft count = 1090\n",
      "--- gene=1025, id=16 removed by soft count = 65\n",
      "--- gene=1025, id=17 removed by soft count = 111\n",
      "--- gene=1025, id=18 removed by soft count = 80\n",
      "--- gene=1025, id=19 removed by soft count = 264\n",
      "--- gene=1025, id=20 removed by soft count = 369\n",
      "--- gene=1025, id=21 removed by soft count = 92\n",
      "--- gene=1025, id=22 removed by soft count = 139\n",
      "--- gene=1025, id=24 removed by soft count = 304\n",
      "--- gene=1025, id=26 removed by soft count = 219\n",
      "--- gene=1025, id=27 removed by soft count = 161\n",
      "--- gene=1025, id=28 removed by soft count = 87\n",
      "--- gene=1025, id=34 removed by soft count = 120\n",
      "--- gene=1025, id=36 removed by soft count = 55\n",
      "--- gene=1025, id=44 removed by soft count = 228\n",
      "--- gene=1025, id=58 removed by soft count = 335\n",
      "--- gene=1025, id=61 removed by soft count = 60\n",
      "--- gene=1025, id=63 removed by soft count = 173\n",
      "--- number of probes: 65 , kept by blast: 39 , if remove dups: 39\n",
      "-- number of probes kept for this region: 39\n",
      "-- checking probes in region: 1026\n",
      "--- gene=1026, id=15 removed by soft count = 167\n",
      "--- gene=1026, id=55 removed by soft count = 51\n",
      "--- gene=1026, id=62 removed by soft count = 403\n",
      "--- gene=1026, id=63 removed by soft count = 67\n",
      "--- gene=1026, id=66 removed by soft count = 3332\n",
      "--- gene=1026, id=67 removed by soft count = 265\n",
      "--- number of probes: 130 , kept by blast: 124 , if remove dups: 124\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1027\n",
      "--- gene=1027, id=5 removed by soft count = 187\n",
      "--- gene=1027, id=7 removed by soft count = 865\n",
      "--- gene=1027, id=37 removed by soft count = 529\n",
      "--- gene=1027, id=66 removed by hard count = 0\n",
      "--- gene=1027, id=71 removed by soft count = 248\n",
      "--- gene=1027, id=72 removed by soft count = 57\n",
      "--- gene=1027, id=74 removed by soft count = 100\n",
      "--- gene=1027, id=75 removed by soft count = 685\n",
      "--- number of probes: 78 , kept by blast: 70 , if remove dups: 68\n",
      "-- number of probes kept for this region: 70\n",
      "-- checking probes in region: 1028\n",
      "--- gene=1028, id=0 removed by soft count = 333\n",
      "--- gene=1028, id=1 removed by soft count = 55\n",
      "--- gene=1028, id=2 removed by soft count = 54\n",
      "--- gene=1028, id=3 removed by soft count = 186\n",
      "--- gene=1028, id=4 removed by soft count = 564\n",
      "--- gene=1028, id=5 removed by soft count = 1741\n",
      "--- gene=1028, id=6 removed by soft count = 820\n",
      "--- gene=1028, id=8 removed by soft count = 57\n",
      "--- gene=1028, id=16 removed by soft count = 92\n",
      "--- gene=1028, id=21 removed by soft count = 97\n",
      "--- number of probes: 102 , kept by blast: 92 , if remove dups: 92\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 92\n",
      "-- checking probes in region: 1029\n",
      "--- gene=1029, id=61 removed by soft count = 64\n",
      "--- gene=1029, id=63 removed by soft count = 758\n",
      "--- gene=1029, id=67 removed by soft count = 87\n",
      "--- gene=1029, id=69 removed by soft count = 91\n",
      "--- gene=1029, id=70 removed by soft count = 433\n",
      "--- gene=1029, id=71 removed by soft count = 170\n",
      "--- gene=1029, id=72 removed by soft count = 103\n",
      "--- gene=1029, id=78 removed by soft count = 70\n",
      "--- gene=1029, id=106 removed by soft count = 632\n",
      "--- gene=1029, id=107 removed by soft count = 133\n",
      "--- gene=1029, id=108 removed by soft count = 321\n",
      "--- gene=1029, id=112 removed by soft count = 169\n",
      "--- gene=1029, id=113 removed by soft count = 53\n",
      "--- gene=1029, id=116 removed by soft count = 328\n",
      "--- gene=1029, id=121 removed by soft count = 472\n",
      "--- number of probes: 127 , kept by blast: 112 , if remove dups: 112\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 112\n",
      "-- checking probes in region: 1030\n",
      "--- gene=1030, id=168 removed by soft count = 141\n",
      "--- gene=1030, id=184 removed by soft count = 56\n",
      "--- gene=1030, id=194 removed by soft count = 57\n",
      "--- gene=1030, id=195 removed by soft count = 74\n",
      "--- number of probes: 301 , kept by blast: 297 , if remove dups: 297\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1031\n",
      "--- gene=1031, id=141 removed by soft count = 77\n",
      "--- gene=1031, id=144 removed by soft count = 180\n",
      "--- gene=1031, id=145 removed by soft count = 191\n",
      "--- gene=1031, id=170 removed by soft count = 69\n",
      "--- number of probes: 172 , kept by blast: 168 , if remove dups: 168\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1032\n",
      "--- gene=1032, id=63 removed by soft count = 162\n",
      "--- gene=1032, id=106 removed by soft count = 53\n",
      "--- gene=1032, id=161 removed by soft count = 54\n",
      "--- number of probes: 163 , kept by blast: 160 , if remove dups: 160\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1033\n",
      "--- gene=1033, id=39 removed by soft count = 117\n",
      "--- gene=1033, id=40 removed by soft count = 64\n",
      "--- gene=1033, id=43 removed by soft count = 70\n",
      "--- gene=1033, id=84 removed by soft count = 70\n",
      "--- gene=1033, id=129 removed by soft count = 386\n",
      "--- number of probes: 244 , kept by blast: 239 , if remove dups: 239\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1034\n",
      "--- gene=1034, id=162 removed by soft count = 68\n",
      "--- gene=1034, id=163 removed by soft count = 102\n",
      "--- gene=1034, id=164 removed by soft count = 158\n",
      "--- gene=1034, id=188 removed by soft count = 52\n",
      "--- number of probes: 224 , kept by blast: 220 , if remove dups: 220\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1035\n",
      "--- gene=1035, id=13 removed by soft count = 606\n",
      "--- number of probes: 126 , kept by blast: 125 , if remove dups: 124\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1036\n",
      "--- gene=1036, id=27 removed by soft count = 124\n",
      "--- gene=1036, id=86 removed by soft count = 469\n",
      "--- number of probes: 192 , kept by blast: 190 , if remove dups: 190\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1038\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=1038, id=55 removed by soft count = 187\n",
      "--- gene=1038, id=56 removed by soft count = 814\n",
      "--- gene=1038, id=115 removed by soft count = 118\n",
      "--- gene=1038, id=117 removed by soft count = 55\n",
      "--- gene=1038, id=121 removed by soft count = 94\n",
      "--- gene=1038, id=130 removed by soft count = 575\n",
      "--- gene=1038, id=132 removed by soft count = 80\n",
      "--- number of probes: 146 , kept by blast: 139 , if remove dups: 139\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1039\n",
      "--- number of probes: 111 , kept by blast: 111 , if remove dups: 111\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 111\n",
      "-- checking probes in region: 1040\n",
      "--- gene=1040, id=0 removed by soft count = 89\n",
      "--- gene=1040, id=1 removed by soft count = 89\n",
      "--- gene=1040, id=34 removed by soft count = 156\n",
      "--- gene=1040, id=39 removed by soft count = 34840\n",
      "--- gene=1040, id=40 removed by soft count = 7270\n",
      "--- gene=1040, id=42 removed by soft count = 121\n",
      "--- gene=1040, id=44 removed by soft count = 174\n",
      "--- gene=1040, id=46 removed by soft count = 649\n",
      "--- gene=1040, id=47 removed by soft count = 643\n",
      "--- number of probes: 62 , kept by blast: 53 , if remove dups: 53\n",
      "-- number of probes kept for this region: 53\n",
      "-- checking probes in region: 1041\n",
      "--- gene=1041, id=2 removed by soft count = 168\n",
      "--- gene=1041, id=3 removed by soft count = 129\n",
      "--- gene=1041, id=4 removed by soft count = 961\n",
      "--- gene=1041, id=5 removed by soft count = 262\n",
      "--- gene=1041, id=7 removed by soft count = 165\n",
      "--- gene=1041, id=8 removed by soft count = 242\n",
      "--- gene=1041, id=9 removed by soft count = 288\n",
      "--- gene=1041, id=10 removed by soft count = 130\n",
      "--- gene=1041, id=12 removed by soft count = 159\n",
      "--- gene=1041, id=13 removed by soft count = 52\n",
      "--- gene=1041, id=14 removed by soft count = 580\n",
      "--- gene=1041, id=15 removed by soft count = 57\n",
      "--- gene=1041, id=26 removed by soft count = 102\n",
      "--- gene=1041, id=27 removed by soft count = 110\n",
      "--- gene=1041, id=35 removed by soft count = 125\n",
      "--- gene=1041, id=39 removed by soft count = 15874\n",
      "--- gene=1041, id=40 removed by soft count = 1719\n",
      "--- gene=1041, id=44 removed by soft count = 750\n",
      "--- gene=1041, id=48 removed by soft count = 108\n",
      "--- gene=1041, id=59 removed by soft count = 116\n",
      "--- gene=1041, id=60 removed by soft count = 75\n",
      "--- gene=1041, id=61 removed by soft count = 1126\n",
      "--- gene=1041, id=62 removed by soft count = 167\n",
      "--- gene=1041, id=63 removed by soft count = 124\n",
      "--- gene=1041, id=64 removed by soft count = 2408\n",
      "--- gene=1041, id=67 removed by soft count = 83\n",
      "--- gene=1041, id=70 removed by soft count = 334\n",
      "--- gene=1041, id=105 removed by soft count = 97\n",
      "--- number of probes: 106 , kept by blast: 78 , if remove dups: 78\n",
      "-- number of probes kept for this region: 78\n",
      "-- checking probes in region: 1044\n",
      "--- gene=1044, id=38 removed by soft count = 99\n",
      "--- gene=1044, id=130 removed by soft count = 25961\n",
      "--- number of probes: 183 , kept by blast: 181 , if remove dups: 181\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1045\n",
      "--- gene=1045, id=1 removed by soft count = 506\n",
      "--- gene=1045, id=5 removed by soft count = 87\n",
      "--- gene=1045, id=18 removed by soft count = 67\n",
      "--- number of probes: 172 , kept by blast: 169 , if remove dups: 169\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1046\n",
      "--- gene=1046, id=14 removed by soft count = 224\n",
      "--- gene=1046, id=70 removed by soft count = 52\n",
      "--- gene=1046, id=74 removed by soft count = 18174\n",
      "--- gene=1046, id=80 removed by soft count = 83\n",
      "--- number of probes: 221 , kept by blast: 217 , if remove dups: 217\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1047\n",
      "--- gene=1047, id=190 removed by soft count = 142\n",
      "--- gene=1047, id=192 removed by soft count = 81\n",
      "--- gene=1047, id=265 removed by soft count = 150\n",
      "--- number of probes: 268 , kept by blast: 265 , if remove dups: 265\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1048\n",
      "--- gene=1048, id=21 removed by soft count = 360\n",
      "--- gene=1048, id=22 removed by soft count = 252\n",
      "--- gene=1048, id=94 removed by soft count = 61\n",
      "--- gene=1048, id=172 removed by soft count = 80\n",
      "--- gene=1048, id=198 removed by soft count = 111\n",
      "--- gene=1048, id=202 removed by soft count = 62\n",
      "--- gene=1048, id=213 removed by soft count = 570\n",
      "--- number of probes: 219 , kept by blast: 212 , if remove dups: 212\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1049\n",
      "--- gene=1049, id=20 removed by soft count = 67\n",
      "--- gene=1049, id=67 removed by soft count = 645\n",
      "--- gene=1049, id=143 removed by soft count = 141\n",
      "--- number of probes: 181 , kept by blast: 178 , if remove dups: 178\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1050\n",
      "--- number of probes: 190 , kept by blast: 190 , if remove dups: 190\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1051\n",
      "--- gene=1051, id=62 removed by soft count = 2743\n",
      "--- gene=1051, id=63 removed by soft count = 468\n",
      "--- gene=1051, id=64 removed by soft count = 130\n",
      "--- gene=1051, id=88 removed by soft count = 306\n",
      "--- gene=1051, id=106 removed by soft count = 64\n",
      "--- gene=1051, id=111 removed by soft count = 102\n",
      "--- number of probes: 147 , kept by blast: 141 , if remove dups: 141\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1053\n",
      "--- gene=1053, id=18 removed by soft count = 232\n",
      "--- gene=1053, id=37 removed by soft count = 133\n",
      "--- gene=1053, id=66 removed by soft count = 151\n",
      "--- number of probes: 96 , kept by blast: 93 , if remove dups: 93\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 93\n",
      "-- checking probes in region: 1054\n",
      "--- gene=1054, id=76 removed by soft count = 487\n",
      "--- gene=1054, id=134 removed by soft count = 113\n",
      "--- gene=1054, id=137 removed by soft count = 54\n",
      "--- gene=1054, id=138 removed by soft count = 700\n",
      "--- gene=1054, id=145 removed by soft count = 388\n",
      "--- gene=1054, id=146 removed by soft count = 78\n",
      "--- gene=1054, id=148 removed by soft count = 98\n",
      "--- gene=1054, id=165 removed by soft count = 229\n",
      "--- number of probes: 171 , kept by blast: 163 , if remove dups: 163\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1055\n",
      "--- gene=1055, id=15 removed by soft count = 839\n",
      "--- gene=1055, id=37 removed by soft count = 66\n",
      "--- gene=1055, id=38 removed by soft count = 390\n",
      "--- gene=1055, id=39 removed by soft count = 72\n",
      "--- gene=1055, id=41 removed by soft count = 102\n",
      "--- gene=1055, id=84 removed by soft count = 60\n",
      "--- gene=1055, id=94 removed by soft count = 127\n",
      "--- gene=1055, id=114 removed by soft count = 184\n",
      "--- gene=1055, id=117 removed by soft count = 70\n",
      "--- gene=1055, id=131 removed by soft count = 75\n",
      "--- gene=1055, id=137 removed by soft count = 152\n",
      "--- gene=1055, id=146 removed by soft count = 261\n",
      "--- gene=1055, id=147 removed by soft count = 2398\n",
      "--- gene=1055, id=148 removed by soft count = 790\n",
      "--- gene=1055, id=149 removed by soft count = 246\n",
      "--- number of probes: 150 , kept by blast: 135 , if remove dups: 135\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1056\n",
      "--- gene=1056, id=128 removed by soft count = 68\n",
      "--- gene=1056, id=129 removed by soft count = 285\n",
      "--- number of probes: 213 , kept by blast: 211 , if remove dups: 210\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1057\n",
      "--- gene=1057, id=18 removed by soft count = 65\n",
      "--- gene=1057, id=23 removed by soft count = 59\n",
      "--- gene=1057, id=76 removed by soft count = 172\n",
      "--- gene=1057, id=89 removed by soft count = 293\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- gene=1057, id=92 removed by soft count = 1600\n",
      "--- gene=1057, id=93 removed by soft count = 273\n",
      "--- gene=1057, id=97 removed by soft count = 153\n",
      "--- gene=1057, id=100 removed by soft count = 385\n",
      "--- gene=1057, id=101 removed by soft count = 216\n",
      "--- gene=1057, id=105 removed by soft count = 828\n",
      "--- gene=1057, id=106 removed by soft count = 67\n",
      "--- gene=1057, id=108 removed by soft count = 471\n",
      "--- gene=1057, id=114 removed by soft count = 69\n",
      "--- number of probes: 193 , kept by blast: 180 , if remove dups: 180\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1058\n",
      "--- number of probes: 257 , kept by blast: 257 , if remove dups: 257\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1059\n",
      "--- number of probes: 221 , kept by blast: 221 , if remove dups: 221\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "-- checking probes in region: 1060\n",
      "--- gene=1060, id=0 removed by soft count = 177\n",
      "--- gene=1060, id=2 removed by soft count = 438\n",
      "--- gene=1060, id=7 removed by soft count = 749\n",
      "--- gene=1060, id=8 removed by soft count = 192\n",
      "--- gene=1060, id=9 removed by soft count = 777\n",
      "--- gene=1060, id=10 removed by soft count = 783\n",
      "--- gene=1060, id=12 removed by soft count = 675\n",
      "--- gene=1060, id=14 removed by soft count = 867\n",
      "--- gene=1060, id=15 removed by soft count = 732\n",
      "--- gene=1060, id=16 removed by soft count = 575\n",
      "--- gene=1060, id=24 removed by soft count = 85\n",
      "--- gene=1060, id=31 removed by soft count = 789\n",
      "--- gene=1060, id=46 removed by soft count = 104\n",
      "--- gene=1060, id=81 removed by soft count = 248\n",
      "--- gene=1060, id=86 removed by soft count = 122\n",
      "--- gene=1060, id=87 removed by soft count = 62\n",
      "--- gene=1060, id=118 removed by soft count = 2277\n",
      "--- gene=1060, id=119 removed by soft count = 652\n",
      "--- gene=1060, id=120 removed by soft count = 17002\n",
      "--- gene=1060, id=121 removed by soft count = 48320\n",
      "--- number of probes: 122 , kept by blast: 102 , if remove dups: 102\n",
      "--- remove duplicated probes\n",
      "-- number of probes kept for this region: 102\n",
      "-- checking probes in region: 1062\n",
      "--- gene=1062, id=18 removed by soft count = 56\n",
      "--- gene=1062, id=100 removed by soft count = 51\n",
      "--- gene=1062, id=105 removed by soft count = 134\n",
      "--- gene=1062, id=108 removed by soft count = 56\n",
      "--- gene=1062, id=112 removed by soft count = 95\n",
      "--- gene=1062, id=119 removed by soft count = 534\n",
      "--- number of probes: 168 , kept by blast: 162 , if remove dups: 162\n",
      "--- remove duplicated probes\n",
      "--- keep probes in the center\n",
      "-- number of probes kept for this region: 120\n",
      "- summarize\n",
      "-- region: 1 , number of probes: 120\n",
      "-- region: 2 , number of probes: 120\n",
      "-- region: 3 , number of probes: 68\n",
      "-- region: 4 , number of probes: 120\n",
      "-- region: 5 , number of probes: 120\n",
      "-- region: 6 , number of probes: 120\n",
      "-- region: 7 , number of probes: 120\n",
      "-- region: 8 , number of probes: 120\n",
      "-- region: 9 , number of probes: 120\n",
      "-- region: 10 , number of probes: 120\n",
      "-- region: 11 , number of probes: 120\n",
      "-- region: 12 , number of probes: 92\n",
      "-- region: 13 , number of probes: 120\n",
      "-- region: 14 , number of probes: 120\n",
      "-- region: 15 , number of probes: 120\n",
      "-- region: 16 , number of probes: 120\n",
      "-- region: 17 , number of probes: 120\n",
      "-- region: 18 , number of probes: 99\n",
      "-- region: 19 , number of probes: 107\n",
      "-- region: 20 , number of probes: 120\n",
      "-- region: 21 , number of probes: 120\n",
      "-- region: 22 , number of probes: 120\n",
      "-- region: 23 , number of probes: 120\n",
      "-- region: 24 , number of probes: 120\n",
      "-- region: 25 , number of probes: 120\n",
      "-- region: 26 , number of probes: 120\n",
      "-- region: 27 , number of probes: 120\n",
      "-- region: 28 , number of probes: 120\n",
      "-- region: 29 , number of probes: 120\n",
      "-- region: 30 , number of probes: 120\n",
      "-- region: 31 , number of probes: 120\n",
      "-- region: 32 , number of probes: 120\n",
      "-- region: 33 , number of probes: 120\n",
      "-- region: 34 , number of probes: 120\n",
      "-- region: 35 , number of probes: 120\n",
      "-- region: 36 , number of probes: 120\n",
      "-- region: 37 , number of probes: 120\n",
      "-- region: 38 , number of probes: 69\n",
      "-- region: 39 , number of probes: 120\n",
      "-- region: 40 , number of probes: 120\n",
      "-- region: 41 , number of probes: 113\n",
      "-- region: 42 , number of probes: 89\n",
      "-- region: 43 , number of probes: 120\n",
      "-- region: 44 , number of probes: 80\n",
      "-- region: 45 , number of probes: 116\n",
      "-- region: 46 , number of probes: 120\n",
      "-- region: 47 , number of probes: 120\n",
      "-- region: 48 , number of probes: 120\n",
      "-- region: 49 , number of probes: 120\n",
      "-- region: 50 , number of probes: 120\n",
      "-- region: 51 , number of probes: 120\n",
      "-- region: 52 , number of probes: 120\n",
      "-- region: 53 , number of probes: 120\n",
      "-- region: 54 , number of probes: 120\n",
      "-- region: 55 , number of probes: 120\n",
      "-- region: 56 , number of probes: 110\n",
      "-- region: 57 , number of probes: 120\n",
      "-- region: 58 , number of probes: 109\n",
      "-- region: 59 , number of probes: 120\n",
      "-- region: 60 , number of probes: 120\n",
      "-- region: 61 , number of probes: 120\n",
      "-- region: 62 , number of probes: 120\n",
      "-- region: 63 , number of probes: 120\n",
      "-- region: 64 , number of probes: 120\n",
      "-- region: 65 , number of probes: 120\n",
      "-- region: 66 , number of probes: 120\n",
      "-- region: 67 , number of probes: 120\n",
      "-- region: 68 , number of probes: 120\n",
      "-- region: 69 , number of probes: 108\n",
      "-- region: 70 , number of probes: 120\n",
      "-- region: 71 , number of probes: 120\n",
      "-- region: 72 , number of probes: 120\n",
      "-- region: 73 , number of probes: 120\n",
      "-- region: 74 , number of probes: 120\n",
      "-- region: 75 , number of probes: 120\n",
      "-- region: 76 , number of probes: 120\n",
      "-- region: 77 , number of probes: 120\n",
      "-- region: 80 , number of probes: 120\n",
      "-- region: 81 , number of probes: 120\n",
      "-- region: 82 , number of probes: 120\n",
      "-- region: 83 , number of probes: 120\n",
      "-- region: 84 , number of probes: 120\n",
      "-- region: 85 , number of probes: 120\n",
      "-- region: 86 , number of probes: 120\n",
      "-- region: 87 , number of probes: 120\n",
      "-- region: 88 , number of probes: 104\n",
      "-- region: 89 , number of probes: 120\n",
      "-- region: 90 , number of probes: 120\n",
      "-- region: 91 , number of probes: 120\n",
      "-- region: 92 , number of probes: 120\n",
      "-- region: 93 , number of probes: 120\n",
      "-- region: 94 , number of probes: 120\n",
      "-- region: 95 , number of probes: 120\n",
      "-- region: 96 , number of probes: 120\n",
      "-- region: 97 , number of probes: 120\n",
      "-- region: 98 , number of probes: 120\n",
      "-- region: 99 , number of probes: 120\n",
      "-- region: 100 , number of probes: 76\n",
      "-- region: 101 , number of probes: 117\n",
      "-- region: 102 , number of probes: 120\n",
      "-- region: 103 , number of probes: 120\n",
      "-- region: 104 , number of probes: 120\n",
      "-- region: 105 , number of probes: 120\n",
      "-- region: 106 , number of probes: 120\n",
      "-- region: 107 , number of probes: 120\n",
      "-- region: 108 , number of probes: 120\n",
      "-- region: 109 , number of probes: 120\n",
      "-- region: 110 , number of probes: 120\n",
      "-- region: 111 , number of probes: 96\n",
      "-- region: 112 , number of probes: 120\n",
      "-- region: 113 , number of probes: 81\n",
      "-- region: 114 , number of probes: 120\n",
      "-- region: 115 , number of probes: 120\n",
      "-- region: 116 , number of probes: 120\n",
      "-- region: 117 , number of probes: 120\n",
      "-- region: 118 , number of probes: 103\n",
      "-- region: 119 , number of probes: 120\n",
      "-- region: 120 , number of probes: 120\n",
      "-- region: 121 , number of probes: 120\n",
      "-- region: 122 , number of probes: 120\n",
      "-- region: 123 , number of probes: 88\n",
      "-- region: 124 , number of probes: 120\n",
      "-- region: 125 , number of probes: 120\n",
      "-- region: 126 , number of probes: 120\n",
      "-- region: 127 , number of probes: 120\n",
      "-- region: 128 , number of probes: 120\n",
      "-- region: 129 , number of probes: 120\n",
      "-- region: 130 , number of probes: 120\n",
      "-- region: 131 , number of probes: 120\n",
      "-- region: 132 , number of probes: 120\n",
      "-- region: 133 , number of probes: 120\n",
      "-- region: 134 , number of probes: 120\n",
      "-- region: 135 , number of probes: 120\n",
      "-- region: 136 , number of probes: 120\n",
      "-- region: 137 , number of probes: 120\n",
      "-- region: 138 , number of probes: 120\n",
      "-- region: 140 , number of probes: 120\n",
      "-- region: 141 , number of probes: 120\n",
      "-- region: 142 , number of probes: 120\n",
      "-- region: 143 , number of probes: 120\n",
      "-- region: 144 , number of probes: 120\n",
      "-- region: 145 , number of probes: 120\n",
      "-- region: 146 , number of probes: 120\n",
      "-- region: 147 , number of probes: 120\n",
      "-- region: 148 , number of probes: 120\n",
      "-- region: 150 , number of probes: 120\n",
      "-- region: 151 , number of probes: 120\n",
      "-- region: 153 , number of probes: 120\n",
      "-- region: 154 , number of probes: 120\n",
      "-- region: 155 , number of probes: 120\n",
      "-- region: 156 , number of probes: 120\n",
      "-- region: 157 , number of probes: 120\n",
      "-- region: 158 , number of probes: 100\n",
      "-- region: 159 , number of probes: 120\n",
      "-- region: 160 , number of probes: 120\n",
      "-- region: 161 , number of probes: 120\n",
      "-- region: 162 , number of probes: 120\n",
      "-- region: 163 , number of probes: 120\n",
      "-- region: 164 , number of probes: 120\n",
      "-- region: 165 , number of probes: 120\n",
      "-- region: 166 , number of probes: 99\n",
      "-- region: 167 , number of probes: 120\n",
      "-- region: 168 , number of probes: 120\n",
      "-- region: 169 , number of probes: 120\n",
      "-- region: 170 , number of probes: 84\n",
      "-- region: 171 , number of probes: 120\n",
      "-- region: 172 , number of probes: 120\n",
      "-- region: 173 , number of probes: 120\n",
      "-- region: 174 , number of probes: 120\n",
      "-- region: 175 , number of probes: 120\n",
      "-- region: 176 , number of probes: 120\n",
      "-- region: 177 , number of probes: 120\n",
      "-- region: 178 , number of probes: 120\n",
      "-- region: 179 , number of probes: 120\n",
      "-- region: 180 , number of probes: 119\n",
      "-- region: 181 , number of probes: 120\n",
      "-- region: 182 , number of probes: 86\n",
      "-- region: 183 , number of probes: 108\n",
      "-- region: 184 , number of probes: 120\n",
      "-- region: 185 , number of probes: 120\n",
      "-- region: 186 , number of probes: 120\n",
      "-- region: 187 , number of probes: 120\n",
      "-- region: 188 , number of probes: 88\n",
      "-- region: 190 , number of probes: 120\n",
      "-- region: 191 , number of probes: 120\n",
      "-- region: 192 , number of probes: 120\n",
      "-- region: 193 , number of probes: 120\n",
      "-- region: 195 , number of probes: 120\n",
      "-- region: 196 , number of probes: 98\n",
      "-- region: 197 , number of probes: 112\n",
      "-- region: 198 , number of probes: 120\n",
      "-- region: 199 , number of probes: 120\n",
      "-- region: 200 , number of probes: 120\n",
      "-- region: 201 , number of probes: 120\n",
      "-- region: 202 , number of probes: 120\n",
      "-- region: 203 , number of probes: 120\n",
      "-- region: 204 , number of probes: 120\n",
      "-- region: 205 , number of probes: 120\n",
      "-- region: 206 , number of probes: 120\n",
      "-- region: 207 , number of probes: 120\n",
      "-- region: 208 , number of probes: 120\n",
      "-- region: 209 , number of probes: 120\n",
      "-- region: 210 , number of probes: 120\n",
      "-- region: 211 , number of probes: 120\n",
      "-- region: 212 , number of probes: 120\n",
      "-- region: 213 , number of probes: 120\n",
      "-- region: 214 , number of probes: 120\n",
      "-- region: 215 , number of probes: 119\n",
      "-- region: 218 , number of probes: 87\n",
      "-- region: 219 , number of probes: 120\n",
      "-- region: 220 , number of probes: 120\n",
      "-- region: 221 , number of probes: 120\n",
      "-- region: 222 , number of probes: 120\n",
      "-- region: 223 , number of probes: 120\n",
      "-- region: 224 , number of probes: 93\n",
      "-- region: 225 , number of probes: 77\n",
      "-- region: 226 , number of probes: 120\n",
      "-- region: 227 , number of probes: 104\n",
      "-- region: 228 , number of probes: 82\n",
      "-- region: 229 , number of probes: 120\n",
      "-- region: 230 , number of probes: 120\n",
      "-- region: 231 , number of probes: 120\n",
      "-- region: 232 , number of probes: 105\n",
      "-- region: 233 , number of probes: 120\n",
      "-- region: 234 , number of probes: 120\n",
      "-- region: 235 , number of probes: 120\n",
      "-- region: 236 , number of probes: 120\n",
      "-- region: 237 , number of probes: 96\n",
      "-- region: 238 , number of probes: 120\n",
      "-- region: 239 , number of probes: 120\n",
      "-- region: 240 , number of probes: 120\n",
      "-- region: 242 , number of probes: 120\n",
      "-- region: 243 , number of probes: 120\n",
      "-- region: 244 , number of probes: 120\n",
      "-- region: 245 , number of probes: 120\n",
      "-- region: 246 , number of probes: 120\n",
      "-- region: 247 , number of probes: 120\n",
      "-- region: 248 , number of probes: 120\n",
      "-- region: 249 , number of probes: 120\n",
      "-- region: 250 , number of probes: 97\n",
      "-- region: 251 , number of probes: 120\n",
      "-- region: 252 , number of probes: 120\n",
      "-- region: 253 , number of probes: 120\n",
      "-- region: 254 , number of probes: 120\n",
      "-- region: 255 , number of probes: 112\n",
      "-- region: 256 , number of probes: 120\n",
      "-- region: 257 , number of probes: 120\n",
      "-- region: 258 , number of probes: 120\n",
      "-- region: 259 , number of probes: 120\n",
      "-- region: 260 , number of probes: 120\n",
      "-- region: 261 , number of probes: 120\n",
      "-- region: 262 , number of probes: 88\n",
      "-- region: 263 , number of probes: 120\n",
      "-- region: 264 , number of probes: 120\n",
      "-- region: 266 , number of probes: 120\n",
      "-- region: 267 , number of probes: 120\n",
      "-- region: 268 , number of probes: 120\n",
      "-- region: 269 , number of probes: 120\n",
      "-- region: 270 , number of probes: 120\n",
      "-- region: 271 , number of probes: 120\n",
      "-- region: 272 , number of probes: 120\n",
      "-- region: 273 , number of probes: 120\n",
      "-- region: 274 , number of probes: 113\n",
      "-- region: 276 , number of probes: 120\n",
      "-- region: 277 , number of probes: 120\n",
      "-- region: 278 , number of probes: 120\n",
      "-- region: 279 , number of probes: 120\n",
      "-- region: 281 , number of probes: 120\n",
      "-- region: 282 , number of probes: 120\n",
      "-- region: 283 , number of probes: 117\n",
      "-- region: 285 , number of probes: 86\n",
      "-- region: 286 , number of probes: 120\n",
      "-- region: 287 , number of probes: 112\n",
      "-- region: 288 , number of probes: 120\n",
      "-- region: 289 , number of probes: 120\n",
      "-- region: 290 , number of probes: 100\n",
      "-- region: 291 , number of probes: 120\n",
      "-- region: 292 , number of probes: 120\n",
      "-- region: 293 , number of probes: 120\n",
      "-- region: 294 , number of probes: 120\n",
      "-- region: 295 , number of probes: 120\n",
      "-- region: 296 , number of probes: 120\n",
      "-- region: 297 , number of probes: 120\n",
      "-- region: 298 , number of probes: 120\n",
      "-- region: 299 , number of probes: 120\n",
      "-- region: 300 , number of probes: 120\n",
      "-- region: 301 , number of probes: 117\n",
      "-- region: 302 , number of probes: 109\n",
      "-- region: 303 , number of probes: 120\n",
      "-- region: 304 , number of probes: 120\n",
      "-- region: 305 , number of probes: 120\n",
      "-- region: 306 , number of probes: 120\n",
      "-- region: 307 , number of probes: 120\n",
      "-- region: 308 , number of probes: 107\n",
      "-- region: 309 , number of probes: 120\n",
      "-- region: 310 , number of probes: 120\n",
      "-- region: 311 , number of probes: 120\n",
      "-- region: 312 , number of probes: 120\n",
      "-- region: 313 , number of probes: 120\n",
      "-- region: 314 , number of probes: 120\n",
      "-- region: 315 , number of probes: 120\n",
      "-- region: 316 , number of probes: 120\n",
      "-- region: 318 , number of probes: 120\n",
      "-- region: 319 , number of probes: 120\n",
      "-- region: 320 , number of probes: 120\n",
      "-- region: 321 , number of probes: 120\n",
      "-- region: 322 , number of probes: 120\n",
      "-- region: 323 , number of probes: 120\n",
      "-- region: 324 , number of probes: 120\n",
      "-- region: 325 , number of probes: 120\n",
      "-- region: 326 , number of probes: 120\n",
      "-- region: 327 , number of probes: 120\n",
      "-- region: 328 , number of probes: 120\n",
      "-- region: 329 , number of probes: 120\n",
      "-- region: 330 , number of probes: 120\n",
      "-- region: 331 , number of probes: 120\n",
      "-- region: 332 , number of probes: 120\n",
      "-- region: 333 , number of probes: 120\n",
      "-- region: 334 , number of probes: 120\n",
      "-- region: 335 , number of probes: 120\n",
      "-- region: 336 , number of probes: 120\n",
      "-- region: 337 , number of probes: 120\n",
      "-- region: 338 , number of probes: 120\n",
      "-- region: 339 , number of probes: 120\n",
      "-- region: 340 , number of probes: 120\n",
      "-- region: 342 , number of probes: 120\n",
      "-- region: 343 , number of probes: 120\n",
      "-- region: 344 , number of probes: 120\n",
      "-- region: 345 , number of probes: 120\n",
      "-- region: 346 , number of probes: 120\n",
      "-- region: 347 , number of probes: 49\n",
      "-- region: 348 , number of probes: 120\n",
      "-- region: 349 , number of probes: 120\n",
      "-- region: 350 , number of probes: 120\n",
      "-- region: 351 , number of probes: 66\n",
      "-- region: 352 , number of probes: 120\n",
      "-- region: 353 , number of probes: 120\n",
      "-- region: 354 , number of probes: 120\n",
      "-- region: 355 , number of probes: 120\n",
      "-- region: 356 , number of probes: 120\n",
      "-- region: 357 , number of probes: 120\n",
      "-- region: 358 , number of probes: 120\n",
      "-- region: 359 , number of probes: 120\n",
      "-- region: 360 , number of probes: 120\n",
      "-- region: 361 , number of probes: 120\n",
      "-- region: 362 , number of probes: 120\n",
      "-- region: 363 , number of probes: 116\n",
      "-- region: 364 , number of probes: 120\n",
      "-- region: 365 , number of probes: 120\n",
      "-- region: 366 , number of probes: 120\n",
      "-- region: 367 , number of probes: 105\n",
      "-- region: 368 , number of probes: 118\n",
      "-- region: 369 , number of probes: 120\n",
      "-- region: 370 , number of probes: 120\n",
      "-- region: 371 , number of probes: 108\n",
      "-- region: 372 , number of probes: 120\n",
      "-- region: 373 , number of probes: 120\n",
      "-- region: 374 , number of probes: 120\n",
      "-- region: 375 , number of probes: 120\n",
      "-- region: 376 , number of probes: 120\n",
      "-- region: 377 , number of probes: 120\n",
      "-- region: 378 , number of probes: 120\n",
      "-- region: 379 , number of probes: 120\n",
      "-- region: 380 , number of probes: 120\n",
      "-- region: 381 , number of probes: 120\n",
      "-- region: 382 , number of probes: 120\n",
      "-- region: 383 , number of probes: 120\n",
      "-- region: 384 , number of probes: 120\n",
      "-- region: 385 , number of probes: 120\n",
      "-- region: 386 , number of probes: 120\n",
      "-- region: 387 , number of probes: 120\n",
      "-- region: 388 , number of probes: 120\n",
      "-- region: 389 , number of probes: 120\n",
      "-- region: 390 , number of probes: 100\n",
      "-- region: 391 , number of probes: 120\n",
      "-- region: 392 , number of probes: 89\n",
      "-- region: 393 , number of probes: 91\n",
      "-- region: 395 , number of probes: 120\n",
      "-- region: 396 , number of probes: 120\n",
      "-- region: 397 , number of probes: 120\n",
      "-- region: 398 , number of probes: 120\n",
      "-- region: 399 , number of probes: 120\n",
      "-- region: 400 , number of probes: 120\n",
      "-- region: 403 , number of probes: 120\n",
      "-- region: 407 , number of probes: 120\n",
      "-- region: 408 , number of probes: 65\n",
      "-- region: 411 , number of probes: 120\n",
      "-- region: 412 , number of probes: 120\n",
      "-- region: 413 , number of probes: 91\n",
      "-- region: 414 , number of probes: 94\n",
      "-- region: 415 , number of probes: 120\n",
      "-- region: 418 , number of probes: 119\n",
      "-- region: 419 , number of probes: 120\n",
      "-- region: 420 , number of probes: 120\n",
      "-- region: 421 , number of probes: 120\n",
      "-- region: 422 , number of probes: 120\n",
      "-- region: 423 , number of probes: 120\n",
      "-- region: 424 , number of probes: 120\n",
      "-- region: 425 , number of probes: 59\n",
      "-- region: 426 , number of probes: 120\n",
      "-- region: 427 , number of probes: 120\n",
      "-- region: 428 , number of probes: 120\n",
      "-- region: 429 , number of probes: 120\n",
      "-- region: 430 , number of probes: 120\n",
      "-- region: 431 , number of probes: 75\n",
      "-- region: 432 , number of probes: 120\n",
      "-- region: 433 , number of probes: 120\n",
      "-- region: 434 , number of probes: 120\n",
      "-- region: 435 , number of probes: 120\n",
      "-- region: 436 , number of probes: 120\n",
      "-- region: 437 , number of probes: 120\n",
      "-- region: 438 , number of probes: 66\n",
      "-- region: 439 , number of probes: 120\n",
      "-- region: 440 , number of probes: 120\n",
      "-- region: 441 , number of probes: 120\n",
      "-- region: 442 , number of probes: 120\n",
      "-- region: 443 , number of probes: 90\n",
      "-- region: 444 , number of probes: 120\n",
      "-- region: 445 , number of probes: 120\n",
      "-- region: 446 , number of probes: 120\n",
      "-- region: 447 , number of probes: 120\n",
      "-- region: 448 , number of probes: 120\n",
      "-- region: 449 , number of probes: 120\n",
      "-- region: 450 , number of probes: 120\n",
      "-- region: 451 , number of probes: 120\n",
      "-- region: 452 , number of probes: 120\n",
      "-- region: 453 , number of probes: 120\n",
      "-- region: 454 , number of probes: 120\n",
      "-- region: 455 , number of probes: 120\n",
      "-- region: 456 , number of probes: 120\n",
      "-- region: 457 , number of probes: 120\n",
      "-- region: 458 , number of probes: 120\n",
      "-- region: 461 , number of probes: 120\n",
      "-- region: 462 , number of probes: 120\n",
      "-- region: 463 , number of probes: 120\n",
      "-- region: 464 , number of probes: 120\n",
      "-- region: 465 , number of probes: 120\n",
      "-- region: 466 , number of probes: 120\n",
      "-- region: 467 , number of probes: 120\n",
      "-- region: 469 , number of probes: 120\n",
      "-- region: 470 , number of probes: 120\n",
      "-- region: 471 , number of probes: 120\n",
      "-- region: 472 , number of probes: 120\n",
      "-- region: 473 , number of probes: 120\n",
      "-- region: 474 , number of probes: 120\n",
      "-- region: 475 , number of probes: 120\n",
      "-- region: 476 , number of probes: 120\n",
      "-- region: 477 , number of probes: 120\n",
      "-- region: 478 , number of probes: 120\n",
      "-- region: 479 , number of probes: 120\n",
      "-- region: 480 , number of probes: 120\n",
      "-- region: 481 , number of probes: 120\n",
      "-- region: 482 , number of probes: 96\n",
      "-- region: 483 , number of probes: 120\n",
      "-- region: 484 , number of probes: 120\n",
      "-- region: 485 , number of probes: 110\n",
      "-- region: 486 , number of probes: 120\n",
      "-- region: 487 , number of probes: 105\n",
      "-- region: 488 , number of probes: 120\n",
      "-- region: 489 , number of probes: 120\n",
      "-- region: 490 , number of probes: 120\n",
      "-- region: 491 , number of probes: 120\n",
      "-- region: 492 , number of probes: 120\n",
      "-- region: 493 , number of probes: 120\n",
      "-- region: 494 , number of probes: 120\n",
      "-- region: 495 , number of probes: 120\n",
      "-- region: 496 , number of probes: 120\n",
      "-- region: 497 , number of probes: 120\n",
      "-- region: 498 , number of probes: 120\n",
      "-- region: 499 , number of probes: 120\n",
      "-- region: 500 , number of probes: 120\n",
      "-- region: 501 , number of probes: 120\n",
      "-- region: 502 , number of probes: 120\n",
      "-- region: 503 , number of probes: 120\n",
      "-- region: 504 , number of probes: 120\n",
      "-- region: 505 , number of probes: 120\n",
      "-- region: 506 , number of probes: 120\n",
      "-- region: 507 , number of probes: 120\n",
      "-- region: 508 , number of probes: 120\n",
      "-- region: 509 , number of probes: 120\n",
      "-- region: 510 , number of probes: 120\n",
      "-- region: 511 , number of probes: 96\n",
      "-- region: 513 , number of probes: 41\n",
      "-- region: 514 , number of probes: 120\n",
      "-- region: 515 , number of probes: 120\n",
      "-- region: 516 , number of probes: 106\n",
      "-- region: 517 , number of probes: 120\n",
      "-- region: 518 , number of probes: 120\n",
      "-- region: 519 , number of probes: 120\n",
      "-- region: 520 , number of probes: 120\n",
      "-- region: 521 , number of probes: 106\n",
      "-- region: 522 , number of probes: 120\n",
      "-- region: 523 , number of probes: 120\n",
      "-- region: 524 , number of probes: 120\n",
      "-- region: 525 , number of probes: 120\n",
      "-- region: 526 , number of probes: 75\n",
      "-- region: 527 , number of probes: 120\n",
      "-- region: 528 , number of probes: 120\n",
      "-- region: 529 , number of probes: 120\n",
      "-- region: 530 , number of probes: 120\n",
      "-- region: 531 , number of probes: 120\n",
      "-- region: 532 , number of probes: 120\n",
      "-- region: 533 , number of probes: 120\n",
      "-- region: 534 , number of probes: 120\n",
      "-- region: 535 , number of probes: 120\n",
      "-- region: 536 , number of probes: 120\n",
      "-- region: 537 , number of probes: 120\n",
      "-- region: 538 , number of probes: 120\n",
      "-- region: 539 , number of probes: 120\n",
      "-- region: 540 , number of probes: 120\n",
      "-- region: 541 , number of probes: 120\n",
      "-- region: 542 , number of probes: 120\n",
      "-- region: 543 , number of probes: 120\n",
      "-- region: 544 , number of probes: 72\n",
      "-- region: 545 , number of probes: 120\n",
      "-- region: 546 , number of probes: 120\n",
      "-- region: 548 , number of probes: 85\n",
      "-- region: 549 , number of probes: 120\n",
      "-- region: 550 , number of probes: 120\n",
      "-- region: 551 , number of probes: 120\n",
      "-- region: 552 , number of probes: 120\n",
      "-- region: 553 , number of probes: 120\n",
      "-- region: 554 , number of probes: 120\n",
      "-- region: 555 , number of probes: 120\n",
      "-- region: 556 , number of probes: 120\n",
      "-- region: 557 , number of probes: 120\n",
      "-- region: 558 , number of probes: 120\n",
      "-- region: 559 , number of probes: 120\n",
      "-- region: 560 , number of probes: 120\n",
      "-- region: 561 , number of probes: 120\n",
      "-- region: 563 , number of probes: 120\n",
      "-- region: 564 , number of probes: 82\n",
      "-- region: 565 , number of probes: 120\n",
      "-- region: 566 , number of probes: 120\n",
      "-- region: 567 , number of probes: 120\n",
      "-- region: 568 , number of probes: 120\n",
      "-- region: 569 , number of probes: 120\n",
      "-- region: 570 , number of probes: 120\n",
      "-- region: 571 , number of probes: 120\n",
      "-- region: 572 , number of probes: 120\n",
      "-- region: 573 , number of probes: 120\n",
      "-- region: 574 , number of probes: 120\n",
      "-- region: 575 , number of probes: 58\n",
      "-- region: 576 , number of probes: 120\n",
      "-- region: 577 , number of probes: 47\n",
      "-- region: 578 , number of probes: 120\n",
      "-- region: 579 , number of probes: 120\n",
      "-- region: 580 , number of probes: 120\n",
      "-- region: 581 , number of probes: 78\n",
      "-- region: 582 , number of probes: 120\n",
      "-- region: 583 , number of probes: 120\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- region: 584 , number of probes: 117\n",
      "-- region: 585 , number of probes: 120\n",
      "-- region: 586 , number of probes: 120\n",
      "-- region: 587 , number of probes: 77\n",
      "-- region: 588 , number of probes: 120\n",
      "-- region: 589 , number of probes: 120\n",
      "-- region: 590 , number of probes: 120\n",
      "-- region: 591 , number of probes: 120\n",
      "-- region: 592 , number of probes: 120\n",
      "-- region: 593 , number of probes: 120\n",
      "-- region: 594 , number of probes: 120\n",
      "-- region: 595 , number of probes: 120\n",
      "-- region: 596 , number of probes: 120\n",
      "-- region: 597 , number of probes: 120\n",
      "-- region: 598 , number of probes: 120\n",
      "-- region: 599 , number of probes: 120\n",
      "-- region: 600 , number of probes: 120\n",
      "-- region: 601 , number of probes: 120\n",
      "-- region: 602 , number of probes: 120\n",
      "-- region: 603 , number of probes: 78\n",
      "-- region: 604 , number of probes: 120\n",
      "-- region: 605 , number of probes: 120\n",
      "-- region: 606 , number of probes: 120\n",
      "-- region: 607 , number of probes: 120\n",
      "-- region: 608 , number of probes: 120\n",
      "-- region: 609 , number of probes: 120\n",
      "-- region: 610 , number of probes: 120\n",
      "-- region: 611 , number of probes: 120\n",
      "-- region: 612 , number of probes: 120\n",
      "-- region: 613 , number of probes: 120\n",
      "-- region: 616 , number of probes: 120\n",
      "-- region: 617 , number of probes: 120\n",
      "-- region: 618 , number of probes: 119\n",
      "-- region: 619 , number of probes: 120\n",
      "-- region: 620 , number of probes: 102\n",
      "-- region: 621 , number of probes: 120\n",
      "-- region: 622 , number of probes: 120\n",
      "-- region: 623 , number of probes: 120\n",
      "-- region: 624 , number of probes: 120\n",
      "-- region: 625 , number of probes: 95\n",
      "-- region: 626 , number of probes: 120\n",
      "-- region: 627 , number of probes: 61\n",
      "-- region: 628 , number of probes: 120\n",
      "-- region: 629 , number of probes: 120\n",
      "-- region: 630 , number of probes: 120\n",
      "-- region: 631 , number of probes: 120\n",
      "-- region: 632 , number of probes: 120\n",
      "-- region: 633 , number of probes: 120\n",
      "-- region: 634 , number of probes: 120\n",
      "-- region: 635 , number of probes: 120\n",
      "-- region: 636 , number of probes: 120\n",
      "-- region: 637 , number of probes: 120\n",
      "-- region: 638 , number of probes: 120\n",
      "-- region: 639 , number of probes: 120\n",
      "-- region: 640 , number of probes: 120\n",
      "-- region: 641 , number of probes: 120\n",
      "-- region: 642 , number of probes: 120\n",
      "-- region: 643 , number of probes: 120\n",
      "-- region: 644 , number of probes: 120\n",
      "-- region: 645 , number of probes: 120\n",
      "-- region: 646 , number of probes: 120\n",
      "-- region: 647 , number of probes: 120\n",
      "-- region: 648 , number of probes: 120\n",
      "-- region: 649 , number of probes: 120\n",
      "-- region: 650 , number of probes: 120\n",
      "-- region: 651 , number of probes: 120\n",
      "-- region: 652 , number of probes: 120\n",
      "-- region: 653 , number of probes: 120\n",
      "-- region: 654 , number of probes: 120\n",
      "-- region: 655 , number of probes: 120\n",
      "-- region: 656 , number of probes: 120\n",
      "-- region: 657 , number of probes: 120\n",
      "-- region: 658 , number of probes: 120\n",
      "-- region: 659 , number of probes: 120\n",
      "-- region: 660 , number of probes: 120\n",
      "-- region: 661 , number of probes: 120\n",
      "-- region: 662 , number of probes: 120\n",
      "-- region: 663 , number of probes: 120\n",
      "-- region: 665 , number of probes: 120\n",
      "-- region: 666 , number of probes: 120\n",
      "-- region: 667 , number of probes: 120\n",
      "-- region: 668 , number of probes: 120\n",
      "-- region: 669 , number of probes: 120\n",
      "-- region: 670 , number of probes: 120\n",
      "-- region: 672 , number of probes: 120\n",
      "-- region: 674 , number of probes: 120\n",
      "-- region: 675 , number of probes: 120\n",
      "-- region: 676 , number of probes: 120\n",
      "-- region: 677 , number of probes: 102\n",
      "-- region: 678 , number of probes: 120\n",
      "-- region: 679 , number of probes: 120\n",
      "-- region: 680 , number of probes: 120\n",
      "-- region: 681 , number of probes: 53\n",
      "-- region: 682 , number of probes: 120\n",
      "-- region: 683 , number of probes: 120\n",
      "-- region: 684 , number of probes: 120\n",
      "-- region: 685 , number of probes: 120\n",
      "-- region: 686 , number of probes: 120\n",
      "-- region: 687 , number of probes: 120\n",
      "-- region: 688 , number of probes: 120\n",
      "-- region: 689 , number of probes: 110\n",
      "-- region: 690 , number of probes: 119\n",
      "-- region: 691 , number of probes: 120\n",
      "-- region: 692 , number of probes: 120\n",
      "-- region: 693 , number of probes: 120\n",
      "-- region: 694 , number of probes: 120\n",
      "-- region: 695 , number of probes: 120\n",
      "-- region: 696 , number of probes: 120\n",
      "-- region: 697 , number of probes: 120\n",
      "-- region: 698 , number of probes: 120\n",
      "-- region: 699 , number of probes: 120\n",
      "-- region: 700 , number of probes: 120\n",
      "-- region: 701 , number of probes: 120\n",
      "-- region: 702 , number of probes: 119\n",
      "-- region: 703 , number of probes: 120\n",
      "-- region: 706 , number of probes: 120\n",
      "-- region: 707 , number of probes: 120\n",
      "-- region: 708 , number of probes: 120\n",
      "-- region: 709 , number of probes: 100\n",
      "-- region: 710 , number of probes: 120\n",
      "-- region: 711 , number of probes: 120\n",
      "-- region: 713 , number of probes: 120\n",
      "-- region: 714 , number of probes: 120\n",
      "-- region: 715 , number of probes: 120\n",
      "-- region: 716 , number of probes: 90\n",
      "-- region: 717 , number of probes: 120\n",
      "-- region: 718 , number of probes: 120\n",
      "-- region: 719 , number of probes: 120\n",
      "-- region: 720 , number of probes: 120\n",
      "-- region: 721 , number of probes: 120\n",
      "-- region: 722 , number of probes: 120\n",
      "-- region: 723 , number of probes: 120\n",
      "-- region: 724 , number of probes: 120\n",
      "-- region: 726 , number of probes: 120\n",
      "-- region: 727 , number of probes: 120\n",
      "-- region: 728 , number of probes: 120\n",
      "-- region: 729 , number of probes: 120\n",
      "-- region: 730 , number of probes: 120\n",
      "-- region: 731 , number of probes: 120\n",
      "-- region: 732 , number of probes: 120\n",
      "-- region: 733 , number of probes: 120\n",
      "-- region: 734 , number of probes: 120\n",
      "-- region: 735 , number of probes: 120\n",
      "-- region: 736 , number of probes: 120\n",
      "-- region: 737 , number of probes: 120\n",
      "-- region: 739 , number of probes: 120\n",
      "-- region: 740 , number of probes: 120\n",
      "-- region: 741 , number of probes: 120\n",
      "-- region: 742 , number of probes: 120\n",
      "-- region: 743 , number of probes: 81\n",
      "-- region: 744 , number of probes: 120\n",
      "-- region: 745 , number of probes: 120\n",
      "-- region: 746 , number of probes: 120\n",
      "-- region: 747 , number of probes: 105\n",
      "-- region: 748 , number of probes: 120\n",
      "-- region: 749 , number of probes: 120\n",
      "-- region: 750 , number of probes: 120\n",
      "-- region: 751 , number of probes: 120\n",
      "-- region: 752 , number of probes: 120\n",
      "-- region: 753 , number of probes: 120\n",
      "-- region: 754 , number of probes: 120\n",
      "-- region: 755 , number of probes: 120\n",
      "-- region: 756 , number of probes: 120\n",
      "-- region: 757 , number of probes: 120\n",
      "-- region: 758 , number of probes: 120\n",
      "-- region: 759 , number of probes: 120\n",
      "-- region: 765 , number of probes: 120\n",
      "-- region: 766 , number of probes: 85\n",
      "-- region: 767 , number of probes: 120\n",
      "-- region: 768 , number of probes: 114\n",
      "-- region: 769 , number of probes: 120\n",
      "-- region: 770 , number of probes: 120\n",
      "-- region: 772 , number of probes: 120\n",
      "-- region: 773 , number of probes: 120\n",
      "-- region: 774 , number of probes: 120\n",
      "-- region: 775 , number of probes: 120\n",
      "-- region: 776 , number of probes: 120\n",
      "-- region: 779 , number of probes: 120\n",
      "-- region: 780 , number of probes: 120\n",
      "-- region: 781 , number of probes: 120\n",
      "-- region: 783 , number of probes: 63\n",
      "-- region: 784 , number of probes: 120\n",
      "-- region: 785 , number of probes: 120\n",
      "-- region: 786 , number of probes: 120\n",
      "-- region: 787 , number of probes: 120\n",
      "-- region: 788 , number of probes: 120\n",
      "-- region: 789 , number of probes: 120\n",
      "-- region: 790 , number of probes: 120\n",
      "-- region: 791 , number of probes: 120\n",
      "-- region: 792 , number of probes: 120\n",
      "-- region: 793 , number of probes: 120\n",
      "-- region: 794 , number of probes: 119\n",
      "-- region: 795 , number of probes: 120\n",
      "-- region: 796 , number of probes: 120\n",
      "-- region: 797 , number of probes: 120\n",
      "-- region: 798 , number of probes: 120\n",
      "-- region: 799 , number of probes: 119\n",
      "-- region: 800 , number of probes: 120\n",
      "-- region: 801 , number of probes: 120\n",
      "-- region: 802 , number of probes: 120\n",
      "-- region: 803 , number of probes: 120\n",
      "-- region: 804 , number of probes: 120\n",
      "-- region: 805 , number of probes: 120\n",
      "-- region: 806 , number of probes: 120\n",
      "-- region: 807 , number of probes: 120\n",
      "-- region: 808 , number of probes: 120\n",
      "-- region: 809 , number of probes: 120\n",
      "-- region: 810 , number of probes: 120\n",
      "-- region: 812 , number of probes: 120\n",
      "-- region: 813 , number of probes: 120\n",
      "-- region: 814 , number of probes: 120\n",
      "-- region: 815 , number of probes: 120\n",
      "-- region: 816 , number of probes: 120\n",
      "-- region: 817 , number of probes: 120\n",
      "-- region: 818 , number of probes: 120\n",
      "-- region: 819 , number of probes: 120\n",
      "-- region: 820 , number of probes: 94\n",
      "-- region: 821 , number of probes: 75\n",
      "-- region: 822 , number of probes: 120\n",
      "-- region: 823 , number of probes: 120\n",
      "-- region: 824 , number of probes: 93\n",
      "-- region: 825 , number of probes: 120\n",
      "-- region: 827 , number of probes: 120\n",
      "-- region: 828 , number of probes: 103\n",
      "-- region: 829 , number of probes: 116\n",
      "-- region: 830 , number of probes: 120\n",
      "-- region: 831 , number of probes: 120\n",
      "-- region: 832 , number of probes: 120\n",
      "-- region: 833 , number of probes: 120\n",
      "-- region: 834 , number of probes: 120\n",
      "-- region: 835 , number of probes: 120\n",
      "-- region: 836 , number of probes: 120\n",
      "-- region: 837 , number of probes: 120\n",
      "-- region: 838 , number of probes: 120\n",
      "-- region: 839 , number of probes: 120\n",
      "-- region: 840 , number of probes: 120\n",
      "-- region: 841 , number of probes: 120\n",
      "-- region: 842 , number of probes: 120\n",
      "-- region: 843 , number of probes: 120\n",
      "-- region: 844 , number of probes: 120\n",
      "-- region: 845 , number of probes: 120\n",
      "-- region: 846 , number of probes: 120\n",
      "-- region: 847 , number of probes: 120\n",
      "-- region: 848 , number of probes: 120\n",
      "-- region: 849 , number of probes: 120\n",
      "-- region: 850 , number of probes: 120\n",
      "-- region: 851 , number of probes: 120\n",
      "-- region: 852 , number of probes: 120\n",
      "-- region: 854 , number of probes: 120\n",
      "-- region: 855 , number of probes: 120\n",
      "-- region: 856 , number of probes: 120\n",
      "-- region: 857 , number of probes: 120\n",
      "-- region: 858 , number of probes: 120\n",
      "-- region: 859 , number of probes: 120\n",
      "-- region: 860 , number of probes: 120\n",
      "-- region: 861 , number of probes: 120\n",
      "-- region: 862 , number of probes: 98\n",
      "-- region: 863 , number of probes: 120\n",
      "-- region: 864 , number of probes: 120\n",
      "-- region: 865 , number of probes: 120\n",
      "-- region: 866 , number of probes: 120\n",
      "-- region: 867 , number of probes: 120\n",
      "-- region: 868 , number of probes: 120\n",
      "-- region: 869 , number of probes: 120\n",
      "-- region: 870 , number of probes: 120\n",
      "-- region: 871 , number of probes: 120\n",
      "-- region: 872 , number of probes: 120\n",
      "-- region: 873 , number of probes: 84\n",
      "-- region: 874 , number of probes: 111\n",
      "-- region: 875 , number of probes: 120\n",
      "-- region: 876 , number of probes: 120\n",
      "-- region: 877 , number of probes: 69\n",
      "-- region: 878 , number of probes: 120\n",
      "-- region: 879 , number of probes: 120\n",
      "-- region: 880 , number of probes: 84\n",
      "-- region: 881 , number of probes: 120\n",
      "-- region: 882 , number of probes: 120\n",
      "-- region: 883 , number of probes: 120\n",
      "-- region: 884 , number of probes: 120\n",
      "-- region: 885 , number of probes: 120\n",
      "-- region: 886 , number of probes: 120\n",
      "-- region: 887 , number of probes: 120\n",
      "-- region: 888 , number of probes: 120\n",
      "-- region: 889 , number of probes: 120\n",
      "-- region: 890 , number of probes: 120\n",
      "-- region: 891 , number of probes: 120\n",
      "-- region: 894 , number of probes: 120\n",
      "-- region: 896 , number of probes: 120\n",
      "-- region: 897 , number of probes: 120\n",
      "-- region: 898 , number of probes: 120\n",
      "-- region: 899 , number of probes: 120\n",
      "-- region: 900 , number of probes: 59\n",
      "-- region: 901 , number of probes: 88\n",
      "-- region: 902 , number of probes: 120\n",
      "-- region: 903 , number of probes: 120\n",
      "-- region: 904 , number of probes: 120\n",
      "-- region: 905 , number of probes: 120\n",
      "-- region: 906 , number of probes: 120\n",
      "-- region: 907 , number of probes: 120\n",
      "-- region: 908 , number of probes: 82\n",
      "-- region: 909 , number of probes: 120\n",
      "-- region: 910 , number of probes: 120\n",
      "-- region: 911 , number of probes: 120\n",
      "-- region: 912 , number of probes: 120\n",
      "-- region: 913 , number of probes: 120\n",
      "-- region: 914 , number of probes: 67\n",
      "-- region: 915 , number of probes: 120\n",
      "-- region: 916 , number of probes: 120\n",
      "-- region: 917 , number of probes: 120\n",
      "-- region: 918 , number of probes: 120\n",
      "-- region: 919 , number of probes: 120\n",
      "-- region: 920 , number of probes: 120\n",
      "-- region: 921 , number of probes: 120\n",
      "-- region: 922 , number of probes: 120\n",
      "-- region: 923 , number of probes: 120\n",
      "-- region: 924 , number of probes: 120\n",
      "-- region: 925 , number of probes: 70\n",
      "-- region: 926 , number of probes: 120\n",
      "-- region: 927 , number of probes: 120\n",
      "-- region: 928 , number of probes: 120\n",
      "-- region: 929 , number of probes: 106\n",
      "-- region: 930 , number of probes: 73\n",
      "-- region: 932 , number of probes: 52\n",
      "-- region: 933 , number of probes: 120\n",
      "-- region: 934 , number of probes: 120\n",
      "-- region: 935 , number of probes: 120\n",
      "-- region: 936 , number of probes: 120\n",
      "-- region: 937 , number of probes: 120\n",
      "-- region: 938 , number of probes: 120\n",
      "-- region: 939 , number of probes: 120\n",
      "-- region: 940 , number of probes: 120\n",
      "-- region: 941 , number of probes: 120\n",
      "-- region: 942 , number of probes: 120\n",
      "-- region: 943 , number of probes: 120\n",
      "-- region: 944 , number of probes: 120\n",
      "-- region: 945 , number of probes: 120\n",
      "-- region: 946 , number of probes: 120\n",
      "-- region: 947 , number of probes: 120\n",
      "-- region: 948 , number of probes: 120\n",
      "-- region: 949 , number of probes: 120\n",
      "-- region: 950 , number of probes: 120\n",
      "-- region: 951 , number of probes: 120\n",
      "-- region: 952 , number of probes: 120\n",
      "-- region: 953 , number of probes: 120\n",
      "-- region: 954 , number of probes: 120\n",
      "-- region: 955 , number of probes: 120\n",
      "-- region: 956 , number of probes: 120\n",
      "-- region: 957 , number of probes: 120\n",
      "-- region: 958 , number of probes: 120\n",
      "-- region: 959 , number of probes: 120\n",
      "-- region: 960 , number of probes: 120\n",
      "-- region: 961 , number of probes: 120\n",
      "-- region: 962 , number of probes: 120\n",
      "-- region: 963 , number of probes: 120\n",
      "-- region: 964 , number of probes: 120\n",
      "-- region: 965 , number of probes: 120\n",
      "-- region: 966 , number of probes: 120\n",
      "-- region: 969 , number of probes: 120\n",
      "-- region: 970 , number of probes: 120\n",
      "-- region: 971 , number of probes: 120\n",
      "-- region: 972 , number of probes: 120\n",
      "-- region: 973 , number of probes: 114\n",
      "-- region: 974 , number of probes: 120\n",
      "-- region: 975 , number of probes: 120\n",
      "-- region: 976 , number of probes: 120\n",
      "-- region: 977 , number of probes: 120\n",
      "-- region: 978 , number of probes: 120\n",
      "-- region: 979 , number of probes: 120\n",
      "-- region: 980 , number of probes: 120\n",
      "-- region: 982 , number of probes: 120\n",
      "-- region: 983 , number of probes: 120\n",
      "-- region: 984 , number of probes: 120\n",
      "-- region: 985 , number of probes: 120\n",
      "-- region: 986 , number of probes: 120\n",
      "-- region: 987 , number of probes: 120\n",
      "-- region: 988 , number of probes: 120\n",
      "-- region: 989 , number of probes: 120\n",
      "-- region: 990 , number of probes: 120\n",
      "-- region: 991 , number of probes: 120\n",
      "-- region: 992 , number of probes: 96\n",
      "-- region: 995 , number of probes: 120\n",
      "-- region: 997 , number of probes: 120\n",
      "-- region: 998 , number of probes: 120\n",
      "-- region: 999 , number of probes: 113\n",
      "-- region: 1000 , number of probes: 47\n",
      "-- region: 1001 , number of probes: 69\n",
      "-- region: 1002 , number of probes: 114\n",
      "-- region: 1007 , number of probes: 120\n",
      "-- region: 1008 , number of probes: 120\n",
      "-- region: 1009 , number of probes: 120\n",
      "-- region: 1010 , number of probes: 100\n",
      "-- region: 1011 , number of probes: 120\n",
      "-- region: 1013 , number of probes: 120\n",
      "-- region: 1014 , number of probes: 120\n",
      "-- region: 1015 , number of probes: 55\n",
      "-- region: 1016 , number of probes: 120\n",
      "-- region: 1017 , number of probes: 115\n",
      "-- region: 1018 , number of probes: 120\n",
      "-- region: 1019 , number of probes: 120\n",
      "-- region: 1020 , number of probes: 120\n",
      "-- region: 1021 , number of probes: 120\n",
      "-- region: 1022 , number of probes: 120\n",
      "-- region: 1023 , number of probes: 106\n",
      "-- region: 1024 , number of probes: 70\n",
      "-- region: 1025 , number of probes: 39\n",
      "-- region: 1026 , number of probes: 120\n",
      "-- region: 1027 , number of probes: 70\n",
      "-- region: 1028 , number of probes: 92\n",
      "-- region: 1029 , number of probes: 112\n",
      "-- region: 1030 , number of probes: 120\n",
      "-- region: 1031 , number of probes: 120\n",
      "-- region: 1032 , number of probes: 120\n",
      "-- region: 1033 , number of probes: 120\n",
      "-- region: 1034 , number of probes: 120\n",
      "-- region: 1035 , number of probes: 120\n",
      "-- region: 1036 , number of probes: 120\n",
      "-- region: 1038 , number of probes: 120\n",
      "-- region: 1039 , number of probes: 111\n",
      "-- region: 1040 , number of probes: 53\n",
      "-- region: 1041 , number of probes: 78\n",
      "-- region: 1044 , number of probes: 120\n",
      "-- region: 1045 , number of probes: 120\n",
      "-- region: 1046 , number of probes: 120\n",
      "-- region: 1047 , number of probes: 120\n",
      "-- region: 1048 , number of probes: 120\n",
      "-- region: 1049 , number of probes: 120\n",
      "-- region: 1050 , number of probes: 120\n",
      "-- region: 1051 , number of probes: 120\n",
      "-- region: 1053 , number of probes: 93\n",
      "-- region: 1054 , number of probes: 120\n",
      "-- region: 1055 , number of probes: 120\n",
      "-- region: 1056 , number of probes: 120\n",
      "-- region: 1057 , number of probes: 120\n",
      "-- region: 1058 , number of probes: 120\n",
      "-- region: 1059 , number of probes: 120\n",
      "-- region: 1060 , number of probes: 102\n",
      "-- region: 1062 , number of probes: 120\n",
      "- Number of probes kept: 114719\n",
      "- Saving to file: \\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\.\\blast_full_probes.fasta\n"
     ]
    }
   ],
   "source": [
    "# screen\n",
    "import ImageAnalysis3.library_tools.quality_check as check\n",
    "reload(check)\n",
    "num_probes_per_region = 120\n",
    "kept_pbs, blast_keep_dic, hard_count_list, soft_count_list = check.Screening_Probes_by_Blast(library_folder, \n",
    "                                                                                             num_probes_per_region,\n",
    "                                                                                             hard_thres=target_len,\n",
    "                                                                                             soft_count_th=50,\n",
    "                                                                                             )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "c78c3f75",
   "metadata": {},
   "outputs": [],
   "source": [
    "# split probes\n",
    "final_pb_filename = os.path.join(library_folder, 'blast_full_probes.fasta')\n",
    "# load full probes\n",
    "kept_pb_records = []\n",
    "with open(final_pb_filename, 'r') as handle:\n",
    "    for record in SeqIO.parse(handle, \"fasta\"):\n",
    "        kept_pb_records.append(record)\n",
    "\n",
    "kept_pb_dict = library_tools.quality_check.split_probe_by_gene(kept_pb_records)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "9c6e8449",
   "metadata": {},
   "outputs": [],
   "source": [
    "chr_2_region_names = pickle.load(open(os.path.join(library_folder, 'chr_2_region_names.pkl'), 'rb'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "ebaa72df",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31', '32', '33', '34', '35', '36', '37', '38', '39', '40', '41', '42', '43', '44', '45', '46', '47', '48', '49', '50', '51', '52', '53', '54', '55', '56', '57', '58', '59', '60', '61', '62', '63', '64', '65', '66', '67', '68', '69', '70', '71', '72', '73', '74', '75', '76', '77', '80', '81', '82', '83', '84', '85', '86', '87', '88', '89', '90', '91', '92', '93', '94', '95', '96', '97', '98', '99', '100', '101', '102', '103', '104', '105', '106', '107', '108', '109', '110', '111', '112', '113', '114', '115', '116', '117', '118', '119', '120', '121', '122', '123', '124', '125', '126', '127', '128', '129', '130', '131', '132', '133', '134', '135', '136', '137', '138', '140', '141', '142', '143', '144', '145', '146', '147', '148', '150', '151', '153', '154', '155', '156', '157', '158', '159', '160', '161', '162', '163', '164', '165', '166', '167', '168', '169', '170', '171', '172', '173', '174', '175', '176', '177', '178', '179', '180', '181', '182', '183', '184', '185', '186', '187', '188', '190', '191', '192', '193', '195', '196', '197', '198', '199', '200', '201', '202', '203', '204', '205', '206', '207', '208', '209', '210', '211', '212', '213', '214', '215', '218', '219', '220', '221', '222', '223', '224', '225', '226', '227', '228', '229', '230', '231', '232', '233', '234', '235', '236', '237', '238', '239', '240', '242', '243', '244', '245', '246', '247', '248', '249', '250', '251', '252', '253', '254', '255', '256', '257', '258', '259', '260', '261', '262', '263', '264', '266', '267', '268', '269', '270', '271', '272', '273', '274', '276', '277', '278', '279', '281', '282', '283', '285', '286', '287', '288', '289', '290', '291', '292', '293', '294', '295', '296', '297', '298', '299', '300', '301', '302', '303', '304', '305', '306', '307', '308', '309', '310', '311', '312', '313', '314', '315', '316', '318', '319', '320', '321', '322', '323', '324', '325', '326', '327', '328', '329', '330', '331', '332', '333', '334', '335', '336', '337', '338', '339', '340', '342', '343', '344', '345', '346', '347', '348', '349', '350', '351', '352', '353', '354', '355', '356', '357', '358', '359', '360', '361', '362', '363', '364', '365', '366', '367', '368', '369', '370', '371', '372', '373', '374', '375', '376', '377', '378', '379', '380', '381', '382', '383', '384', '385', '386', '387', '388', '389', '390', '391', '392', '393', '395', '396', '397', '398', '399', '400', '403', '407', '408', '411', '412', '413', '414', '415', '418', '419', '420', '421', '422', '423', '424', '425', '426', '427', '428', '429', '430', '431', '432', '433', '434', '435', '436', '437', '438', '439', '440', '441', '442', '443', '444', '445', '446', '447', '448', '449', '450', '451', '452', '453', '454', '455', '456', '457', '458', '461', '462', '463', '464', '465', '466', '467', '469', '470', '471', '472', '473', '474', '475', '476', '477', '478', '479', '480', '481', '482', '483', '484', '485', '486', '487', '488', '489', '490', '491', '492', '493', '494', '495', '496', '497', '498', '499', '500', '501', '502', '503', '504', '505', '506', '507', '508', '509', '510', '511', '513', '514', '515', '516', '517', '518', '519', '520', '521', '522', '523', '524', '525', '526', '527', '528', '529', '530', '531', '532', '533', '534', '535', '536', '537', '538', '539', '540', '541', '542', '543', '544', '545', '546', '548', '549', '550', '551', '552', '553', '554', '555', '556', '557', '558', '559', '560', '561', '563', '564', '565', '566', '567', '568', '569', '570', '571', '572', '573', '574', '575', '576', '577', '578', '579', '580', '581', '582', '583', '584', '585', '586', '587', '588', '589', '590', '591', '592', '593', '594', '595', '596', '597', '598', '599', '600', '601', '602', '603', '604', '605', '606', '607', '608', '609', '610', '611', '612', '613', '616', '617', '618', '619', '620', '621', '622', '623', '624', '625', '626', '627', '628', '629', '630', '631', '632', '633', '634', '635', '636', '637', '638', '639', '640', '641', '642', '643', '644', '645', '646', '647', '648', '649', '650', '651', '652', '653', '654', '655', '656', '657', '658', '659', '660', '661', '662', '663', '665', '666', '667', '668', '669', '670', '672', '674', '675', '676', '677', '678', '679', '680', '681', '682', '683', '684', '685', '686', '687', '688', '689', '690', '691', '692', '693', '694', '695', '696', '697', '698', '699', '700', '701', '702', '703', '706', '707', '708', '709', '710', '711', '713', '714', '715', '716', '717', '718', '719', '720', '721', '722', '723', '724', '726', '727', '728', '729', '730', '731', '732', '733', '734', '735', '736', '737', '739', '740', '741', '742', '743', '744', '745', '746', '747', '748', '749', '750', '751', '752', '753', '754', '755', '756', '757', '758', '759', '765', '766', '767', '768', '769', '770', '772', '773', '774', '775', '776', '779', '780', '781', '783', '784', '785', '786', '787', '788', '789', '790', '791', '792', '793', '794', '795', '796', '797', '798', '799', '800', '801', '802', '803', '804', '805', '806', '807', '808', '809', '810', '812', '813', '814', '815', '816', '817', '818', '819', '820', '821', '822', '823', '824', '825', '827', '828', '829', '830', '831', '832', '833', '834', '835', '836', '837', '838', '839', '840', '841', '842', '843', '844', '845', '846', '847', '848', '849', '850', '851', '852', '854', '855', '856', '857', '858', '859', '860', '861', '862', '863', '864', '865', '866', '867', '868', '869', '870', '871', '872', '873', '874', '875', '876', '877', '878', '879', '880', '881', '882', '883', '884', '885', '886', '887', '888', '889', '890', '891', '894', '896', '897', '898', '899', '900', '901', '902', '903', '904', '905', '906', '907', '908', '909', '910', '911', '912', '913', '914', '915', '916', '917', '918', '919', '920', '921', '922', '923', '924', '925', '926', '927', '928', '929', '930', '932', '933', '934', '935', '936', '937', '938', '939', '940', '941', '942', '943', '944', '945', '946', '947', '948', '949', '950', '951', '952', '953', '954', '955', '956', '957', '958', '959', '960', '961', '962', '963', '964', '965', '966', '969', '970', '971', '972', '973', '974', '975', '976', '977', '978', '979', '980', '982', '983', '984', '985', '986', '987', '988', '989', '990', '991', '992', '995', '997', '998', '999', '1000', '1001', '1002', '1007', '1008', '1009', '1010', '1011', '1013', '1014', '1015', '1016', '1017', '1018', '1019', '1020', '1021', '1022', '1023', '1024', '1025', '1026', '1027', '1028', '1029', '1030', '1031', '1032', '1033', '1034', '1035', '1036', '1038', '1039', '1040', '1041', '1044', '1045', '1046', '1047', '1048', '1049', '1050', '1051', '1053', '1054', '1055', '1056', '1057', '1058', '1059', '1060', '1062'])"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kept_pb_dict.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "a6096960",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[1,\n",
       " 2,\n",
       " 3,\n",
       " 4,\n",
       " 5,\n",
       " 6,\n",
       " 7,\n",
       " 8,\n",
       " 9,\n",
       " 10,\n",
       " 11,\n",
       " 12,\n",
       " 13,\n",
       " 14,\n",
       " 15,\n",
       " 16,\n",
       " 17,\n",
       " 18,\n",
       " 19,\n",
       " 20,\n",
       " 21,\n",
       " 22,\n",
       " 23,\n",
       " 24,\n",
       " 25,\n",
       " 26,\n",
       " 27,\n",
       " 28,\n",
       " 29,\n",
       " 30,\n",
       " 31,\n",
       " 32,\n",
       " 33,\n",
       " 34,\n",
       " 35,\n",
       " 36,\n",
       " 37,\n",
       " 38,\n",
       " 39,\n",
       " 40,\n",
       " 41,\n",
       " 42,\n",
       " 43,\n",
       " 44,\n",
       " 45,\n",
       " 46,\n",
       " 47,\n",
       " 48,\n",
       " 49,\n",
       " 50,\n",
       " 51,\n",
       " 52,\n",
       " 53,\n",
       " 54,\n",
       " 55,\n",
       " 56,\n",
       " 57,\n",
       " 58,\n",
       " 59,\n",
       " 60,\n",
       " 61,\n",
       " 62,\n",
       " 63,\n",
       " 64,\n",
       " 65,\n",
       " 66,\n",
       " 67,\n",
       " 68,\n",
       " 69,\n",
       " 70,\n",
       " 71,\n",
       " 72,\n",
       " 73,\n",
       " 74,\n",
       " 75,\n",
       " 76,\n",
       " 77]"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "_reg_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "a71dada7",
   "metadata": {},
   "outputs": [],
   "source": [
    "sub_library_500 = {}\n",
    "sub_library_250_1 = {}\n",
    "sub_library_250_2 = {}\n",
    "\n",
    "for _chr, _reg_names in chr_2_region_names.items():\n",
    "    for _ireg, _reg in enumerate(_reg_names):\n",
    "        if _ireg % 4 == 0 or _ireg % 4 == 2:\n",
    "            sub_library_500[_reg] = kept_pb_dict[str(_reg)]\n",
    "        elif _ireg % 4 == 1:\n",
    "            sub_library_250_1[_reg] = kept_pb_dict[str(_reg)]\n",
    "        elif _ireg % 4 == 3:\n",
    "            sub_library_250_2[_reg] = kept_pb_dict[str(_reg)]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a305c778",
   "metadata": {},
   "source": [
    "## for 500 gene library, simply remerge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "a186c1b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "57936\n"
     ]
    }
   ],
   "source": [
    "sub_library_500_records = []\n",
    "for _reg, _pbs in sub_library_500.items():\n",
    "    for _record in _pbs:\n",
    "        _record.id += '_library_1000-mouse-genome_500'\n",
    "        _record.name = _record.id\n",
    "        _record.description = ''\n",
    "        # append \n",
    "        sub_library_500_records.append(_record)\n",
    "        \n",
    "print(len(sub_library_500_records))    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "13439828",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SeqRecord(seq=Seq('CCCGCAATGGCTGACAACCGTATACATTAGGATTGCGAGCTGTCATAACTGTGT...TCC'), id='loc_1:3740000-3760000_gene_1_pb_23_pos_2742_strand_-_readouts_[NDB_856_c,NDB_946_c,NDB_935_c]_primers_[W1A03_primer_2,W1A10_primer_9]_library_1000-mouse-genome_500_library_1000-mouse-genome_500', name='loc_1:3740000-3760000_gene_1_pb_23_pos_2742_strand_-_readouts_[NDB_856_c,NDB_946_c,NDB_935_c]_primers_[W1A03_primer_2,W1A10_primer_9]_library_1000-mouse-genome_500_library_1000-mouse-genome_500', description='', dbxrefs=[])"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_library_500_records[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "a3e8e52f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "57936\n"
     ]
    }
   ],
   "source": [
    "# save kept records\n",
    "print(len(sub_library_500_records))\n",
    "with open(os.path.join(library_folder, 'filtered_full_probes_sublibrary_500.fasta'), 'w') as output_handle:\n",
    "    SeqIO.write(sub_library_500_records, output_handle, \"fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "04c97a88",
   "metadata": {},
   "source": [
    "## for 250_1 gene library, change primers from [2,9] into [6,11]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "cacee500",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Picked primer: ID: W1A07_primer_6\n",
      "Name: W1A07_primer_6\n",
      "Description: W1A07_primer_6\n",
      "Number of features: 0\n",
      "Seq('CGCAAACTGGTGCGGAAGGC')\n",
      "- Picked primer: ID: W1A12_primer_11\n",
      "Name: W1A12_primer_11\n",
      "Description: W1A12_primer_11\n",
      "Number of features: 0\n",
      "Seq('TAATACGACTCACTATAGGGCCATTGCCCGCGAGGTCGAG')\n"
     ]
    }
   ],
   "source": [
    "from ImageAnalysis3 import library_tools\n",
    "primer_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Primers'\n",
    "primers_250_1 = library_tools.assemble.load_primers([6,11], _primer_folder=primer_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "e830bb24",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Checking primer usage, total probes: 29037\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "sub_library_250_1_records = []\n",
    "for _reg, _pbs in sub_library_250_1.items():\n",
    "    sub_library_250_1_records.extend(_pbs)\n",
    "\n",
    "# replace primers\n",
    "for _irecord, _record in enumerate(sub_library_250_1_records):\n",
    "    _seq = primers_250_1[0].seq[-primer_len:] + _record.seq[primer_len:-primer_len] + primers_250_1[1].seq[-primer_len:].reverse_complement()\n",
    "    _name = _record.name.split('primers_[')[0] + 'primers_['+primers_250_1[0].id +','+ primers_250_1[1].id+']' \\\n",
    "    + '_library_1000-mouse-genome_250_1'\n",
    "    sub_library_250_1_records[_irecord] = SeqRecord(_seq, name=_name, id=_name, description='')\n",
    "    \n",
    "# quality check primers\n",
    "import ImageAnalysis3.library_tools.quality_check as check\n",
    "reload(check)\n",
    "primer_check = check._check_primer_usage(sub_library_250_1_records, primers_250_1[0], primers_250_1[1], primer_len=primer_len)\n",
    "print(primer_check)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "4b65fbf2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: loc_1:6240000-6260000_gene_2_pb_58_pos_5958_strand_-_readouts_[NDB_965_c,NDB_883_c,NDB_742_c]_primers_[W1A07_primer_6,W1A12_primer_11]_library_1000-mouse-genome_250_1\n",
      "Name: loc_1:6240000-6260000_gene_2_pb_58_pos_5958_strand_-_readouts_[NDB_965_c,NDB_883_c,NDB_742_c]_primers_[W1A07_primer_6,W1A12_primer_11]_library_1000-mouse-genome_250_1\n",
      "Number of features: 0\n",
      "Seq('CGCAAACTGGTGCGGAAGGCCTTAGTCTCGCGCGCCGGATTTTCCCAGGGTGAG...TGG')\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "f010e5b5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: loc_1:6240000-6260000_gene_2_pb_58_pos_5958_strand_-_readouts_[NDB_965_c,NDB_883_c,NDB_742_c]_primers_[W1A07_primer_6,W1A12_primer_11]_library_1000-mouse-genome_250_1\n",
      "Name: loc_1:6240000-6260000_gene_2_pb_58_pos_5958_strand_-_readouts_[NDB_965_c,NDB_883_c,NDB_742_c]_primers_[W1A07_primer_6,W1A12_primer_11]_library_1000-mouse-genome_250_1\n",
      "Number of features: 0\n",
      "Seq('CGCAAACTGGTGCGGAAGGCCTTAGTCTCGCGCGCCGGATTTTCCCAGGGTGAG...TGG')\n",
      "29037\n"
     ]
    }
   ],
   "source": [
    "# save kept records\n",
    "print(sub_library_250_1_records[0])\n",
    "print(len(sub_library_250_1_records))\n",
    "with open(os.path.join(library_folder, 'filtered_full_probes_sublibrary_250_1.fasta'), 'w') as output_handle:\n",
    "    SeqIO.write(sub_library_250_1_records, output_handle, \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "a4948828",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAApQAAAFKCAYAAACjE8/+AAAAAXNSR0IArs4c6QAAIABJREFUeF7sXQd4FFXXPlQp0iG00HsJvZdQQlcEafLRUQRERKUpvQiiIMUCgiKoCCgIiHQIvdfQO6H3XqSX/3nv5mxuhk12Znd2dsJ/z/N8zyfZW9+5u/fMKe+J8+LFixekRCGgEFAIKAQUAgoBhYBCQCHgIQJxlELpIXKqm0JAIaAQUAgoBBQCCgGFgEBAKZTqICgEFAIKAYWAQkAhoBBQCHiFgFIovYJPdVYIKAQUAgoBhYBCQCGgEFAKpToDCgGFgEJAIaAQUAgoBBQCXiGgFEqv4FOdFQIKAYWAQkAhoBBQCCgElEKpzoBCQCGgEFAIKAQUAgoBhYBXCCiF0iv4VGeFgEJAIaAQUAgoBBQCCgGlUKozoBBQCCgEFAIKAYWAQkAh4BUCSqH0Cj7VWSGgEFAIKAQUAgoBhYBCQCmU6gwoBBQCCgGFgEJAIaAQUAh4hYBSKL2CT3VWCCgEFAIKAYWAQkAhoBBQCqU6AwoBhYBCQCGgEFAIKAQUAl4hoBRKr+BTnRUCCgGFgEJAIaAQUAgoBJRCqc6AQkAhoBBQCCgEFAIKAYWAVwgohdIr+FRnhYBCQCGgEFAIKAQUAgoBpVCqM6AQUAgoBBQCCgGFgEJAIeAVAkqh9Ao+1VkhoBBQCCgEFAIKAYWAQkAplOoMKAQUAgoBhYBCQCGgEFAIeIWAUii9gk91VggoBBQCCgGFgEJAIaAQUAqlOgMKAYWAQkAhoBBQCCgEFAJeIaAUSq/gU50VAgoBhYBCQCGgEFAIKASUQqnOgEJAIaAQUAgoBBQCCgGFgFcIKIXSK/hUZ4WAQkAhoBBQCCgEFAIKAaVQqjOgEFAIKAQUAgoBhYBCQCHgFQJKofQKPtVZIaAQUAgoBBQCCgGFgEJAKZTqDCgEFAIKAYWAQkAhoBBQCHiFgFIovYJPdVYIKAQUAgoBhYBCQCGgEFAKpToDCgGFgEJAIaAQUAgoBBQCXiGgFEqv4FOdFQIKAYWAQkAhoBBQCCgElEKpzoBCQCGgEFAIKAQUAgoBhYBXCCiF0iv4VGeFgEJAIaAQUAgoBBQCCgGlUKozoBBQCCgEFAIKAYWAQkAh4BUCSqH0Cj7VWSGgEFAIKAQUAgoBhYBCQCmU6gwoBBQCCgGFgEJAIaAQUAh4hYBSKL2CT3VWCCgEFAIKAYWAQkAhoBBQCqU6AwoBhYBCQCGgEFAIKAQUAl4hoBRKr+BTnRUCCgGFgEJAIaAQUAgoBJRCqc6AQkAhoBBQCCgEFAIKAYWAVwgohdIr+CI7P3z4kPbt20fp0qWj+PHjmzSqGkYhoBBQCCgEFAIKAbsi8PTpU7p69SoFBQVRokSJ7LpMS9alFEqTYN6+fTuVKVPGpNHUMAoBhYBCQCGgEFAIxBYEtm3bRqVLl44ty/XJOpVCaRKsp06dohw5chAOVcaMGU0aVQ2jEFAIKAQUAgoBhYBdEbh48aIwJp08eZKyZ89u12Vasi6lUJoE87lz5yhLlix09uxZCgwMNGlUNYxCQCGgEFAIKAQUAnZFQN39kU9GKZQmnVJ1qEwCUg2jEFAIKAQUAgqBWIKAuvuVQmn6UVWHynRI1YAKAYWAQkAhoBCwNQLq7lcKpekHVB0q0yFVAyoEFAIKAYWAQsDWCKi7XymUph9QdahMh1QNqBBQCCgEFAIKAVsjoO5+pVCafkDVoTIdUjWgQkAhoBBQCCgEbI2AuvuVQmn6AVWHynRI1YAKAYWAQkAhoBCwNQLq7lcKpekHVB0q0yFVAyoEFAIKAYWAQsDWCKi7XymUph9QdahMh1QNqBBQCCgEFAIKAVsjoO5+pVCafkDVoTIdUjWgQkAhoBBQCCgEbI2AuvuVQmn6AVWHynRI1YAKAYWAQkAhoBCwNQLq7lcKpekHVB0q0yFVAyoEFAIKAYWAQsDWCKi7XymUph9QdahMh1QNqBBQCCgEFAIKAVsjoO5+pVCafkDVoTIdUjWgQkAhoBBQCCgEbI2AuvuVQmn6AVWHynRI1YAKAYWAQkAhoBCwNQLq7n9FFcphw4ZRaGgorVmzxrnDBQsW0NChQ+nQoUOUNm1aatq0qfh34sSJRZvnz5/TkCFDaPLkyXTz5k2qVKkSTZgwgXLnzm3oEKtDZQgu1VghoBBQCCgEFAKxHgF197+CCuW4ceOoe/fuFBwc7FQo169fT9WqVSMomk2aNKHjx49Tx44dKSQkhKZOnSpQgDIJBRL/zpw5M/Xu3ZvCw8PpwIEDlDBhQt2HXR0q3VCphgoBhYBCQCGgEHglEFB3/yukUJ4/f546dOhAUB6zZMlC6dOndyqUrVq1oitXrtDy5cudO/7jjz+offv2dO/ePYoTJ46wWo4cOZI6d+4s2ty6dYsyZcpEU6ZMoebNm+s+8OpQ6YZKNVQIKAQUAgoBhcArgYC6+18hhRIu7ZkzZworJFzZp06dciqUYWFhFDduXCpatGgUhbJ169Z09epVYYksW7YsHTlyhPLmzetsA7d3kSJFhOVSr6hDpRcp1U4hoBBQCCgEFAKvBgLq7n+FFEr5SLZr1y6KQqk9ro8fPxYKJJTMnTt30ty5c6lx48Z0//59Z0wl+jRr1kz8beHChbpPvK8O1eFLd2jDsWsu15E6aUJ6o0hGei1+PNp//jZtCb/+Urt4ceNQ7UIZKFNKR8wo5PytB7T8wCV69vyF82+FM6egcjnT0PPnL2jx/ot06fZD3XuXG8aNE4dqFkxPWVInoev3HtGifRfp8dPnlC7Za1QvKCMliBeXws7cpJ2nb3o0vtwpOG86yps+Gd1//JRWHLxMlXKnpTSvv+ZscvzKPVpz5Ir4d8FMyalCrrRif8sOXBIYYC11Cmeg9MkT0dW7j2jJfsdaYxJ5f7fuP6aFey/SwyfPvN6LqwGwXqz7weNntGDvBbrz4IlP5uFBeT7+9+0HT2jR3osC3xSJE9BbxTKJsybLixcvaPnBy3T2xn3DayufKw0VypRC4Id5bt5/7BwDZxvnJVGCeHTgwm3afMJxtktkS0UlsqaiJ8+e0+J9F8Vz80TkZ3/l7kNasu+SGDM6kc8anrn8LHC26xbOSAnjx6U9Z2/R9lM3DC8pV8DrVC1fgOi36vBlCr/630tjJE4Yj94qmomSJUpARy/fpXVHrxqehzvkTJeUqudPL/65+vAVOnH1nsdjxYaOMZ017frxm1I5T1rx55WHrtCp6y8/C22fbGmSUo0CAcLrheeC5+Mv0f6u4V7AbzK+S4cu3qGNx13fJ/5aL+Z9/bX49GbRTOL/w6/eo9VHrhJ+W/RI8kQJ6M2iGSlJwvgU032Jz1uUzapnSENtfHX3G1qETRrHeaH3qdlkwTEtIyaF8unTp9SyZUv6559/aN26dUKxhPsb1spnz54JJZOlTZs2dOHCBZHgE53cuXOH8D+WixcvUpkyZejs2bMUGBhoGlozt52hPnP3RTvesIaFqUnJQCo9LJTuPnrqsl2FXGloxvvlnJ81m7iZtmkuPSieGz+rLhTTDr/v8Gr9+TMkoyUfV6bOf+ykZQcuO8ca904xqlEwPZUatoIePolZcdOzAFzk2/qG0IQ1J2jUsiP0dvHMNPadYs6ulUeuorM3Hoh/x4lDYn+4qFv9stXZpnr+AJrSrjR1mb6TFu+7pGdaKpolJc3/sKJ4Lng+vpJUSRLQ9n41aPKGk/TVksO+msY5Lubb0b8m4SxAhiw4QFM3nnJ+3v+NAtShcs4o69h28gY1m7TZo7VBSd3RvwZN33KaBi84+NIYXzQoRM3LZKWSX6ygOw8dZ/u1+HFFn1WHr9DHf+72aF7uFJI/gH5pV5o6TdsR5Zy6GjQg2Wu0VTpr2jZjmhUVLyclvwilBx6+YCz9pDI9ffaC3vx+Q7T7alM+Gw14syCVH7GKrt3zTJnmwRd1q0RxKA7V+269VzjGhs581mZsPUOD/j3gdsndQvKIl4ZfN0Wef3ed5napQCkTJ6Dqo9e6a+rzz/l37cPpu8RLPc5MuwrZqfTwULrxX+SLm88XYmCCD6vlol6181PI6DV0wsULVUxD9aqdjz6okkvs73o0+0ufHN/hGgZWpK+pUigjcfp/oVDevXtXWB1Xr15Ns2fPpvr16wsE5syZI5J1XFkoHz16RPPnz4/2RA0ePFgk9GjFbIVy4d4LNGb50ZfmuXD7gVDKOlXJSe9WzEFlv1wp2mROmVhcuhBYmPDlypE2Ka3uWdU5RsWvVgkLXdrXEwprx8lrjjfwPzuWo91nbwnlJWG8uBSYKtKqqeer9eT5c6cCN+eDCvS/n7bQY8nq07tOPqpfJBNVHrn6pbXqGZ/b4MK+GGFBPfxFHRr87wH6c/tZKpUtFf39QQXnUDn7LCLJCEv4wT92+S59NidSQc8T8Dqt6F6F6n+/gfadv01QqlIlcZ2MBYUdFjG0CRtYi1r8vIU2nbgurHdpkupP4HK3V2B27qZDEd7wWTUaufQI/bvnAiV7Lb6w9JotMp6HhtYhWMIgWkULVptp75WNMj2/8MSPG4eypk6ia2mwSF6IeH57BtaiH1Yfo5/Xn6RECeJSphSJxdl89PQ5fVA1F3UOzkVFh0bGQGOCNT2riksSLxGenFP+XuRKl5RW9qhKtceuoyOX7xKsolAIZJGxOTqsrlBGsOckCeNRhuSJxHPC88KFhheaCl+tMny2wyO+f1PblRZW0o7TdooxcqZN6lwKnz2cs9HNilK7qdvFZ/huO9R//cLzTWpdUvR1NZ/+0ezdMspZG1SLxq8+Tj+tC3eeNe3q7z16Slc0Vu+MKRJR4gRRLfNyP8ZzfIsS4rehxWTHC6snz8ZbNGHlv3n/CRXImFy81DeasJF2nblF71XKQd1r5qVCg5aJKdztydt1GOl/+c5D+u/xM2pUPDONeacYFRiwVLyU4SUOFsuYBL/H+G7Act//zQJUZvjL9yD3T/N6QprdOfJ+MLLGmNoqhTISnVdeoYTlsG7dunTy5ElhnUTWN8u2bduEpRLZ37ly5XL+HTGUiLscP358tOfIKgtldAtoO2UbrT16lWCxaF8xB1X7xkGVtKpHFcqZ7nXx379uPCksP1AMN3xW3TkULITX7j2mkU2KUNOSgZRvwFLh6v2hRXHafeaWsIiVyZGaZnUqb+g7Bxd6mYg3RFgpD1+6K6xduHihJHxaI69wTYREvMFDMcguXZp6J9t15iY1mrBJNN/Zv4a45OGGLJgxOS3+uLL4+9Nnzyl3vyVRhoTCDIVywPxICwUU8I2fV6fqo9cI62XfevmpY3DkWZAHWLr/krC6Yk/Hh9el+j9soP3n79DHIXno05qRMbh69xFdO7jSiw1dIT7+q2M5GrnsiAgR6Bick/rWK+Dt8C/1l62M+4fUdv6Id/htO4UeukIwWEIxhxK1d1Atih8v0pqPCxqKXb70yWjZp8G61gYrOFvhYPX7YdVxmrblNFXJm45+e7cMNf9pM20Jv0EdKuWgThFWB3ngZZ8E06K9F+i7VcepaGAKmt+1kq55uRErwWlff01YO8t9uZIu3XlIsPa3Kpctylg7Tt2gJhMdFlgov4P+3U//7L5AbxbJSD+0KEFv/bCB9p67Lc5AoxKZqcoox/cwtHsVyh3g+B66k9x9F9PT5y9oYquS9PT5c+o6I0woPIe/qOvsinNbc+w68W/+buH/l36iD3N5DUGDlolL+NvmDms+LL24uPHsXzXZe+4WvfXDRrEtWPtxXmF1dPVyhDZQQN//fQetjwgzqls4A33/v+JRzrwWo6DBy+juw6cED0zKJAmcyv6JL+s5rf1W4frjmhP09dLDlDf967T80yrU4IcNtOfcbWpdLptQKIt/4fhdmfNBeSqZLbVVy4pxno//DKP5uy9Q/aKZBNZ5+i2mJ8/wfShBdQpnjLHvsIUHHfdV9tRCoeRnvaVPCGVIkciS/SmFMhLmV1qhBK9khQoVhGt66dKlFBQUFOWAwQoZEBBAY8aMoffee098FluyvNlF27hEILWvmD3KBY2YQMgfW05T/3/2C4VuS98Q597lC6VBsczEFsshbxUSFsp5YefpjaCMNL5lCcNfyD5z99LMbWed/SrmTiNiAPGWDJfGm0UyUd1vHS62TZ9XjxLbqXcyxCfVirhc1/euJhRKuECzp0lCa3o5XhgQ91dwoONtnAXKCi7mYYsOOf/G1sbyI1YKq+cXDQuLH19XsunENWrxs8P6cHBobbGP09fvC3cSLABmCaJQCg9aJt7a4UqFhRIKD55P2wrZzZrGOc7O0zeo8Y+RSlOKJA4rHb+04PLlCxaufrj8WYYuOEhTNp6kcjlT058d9b2AyMrRul7VhIVy1o5zVKtgevqpTSlq/ctWMR9cdLDAw70ry79dK4qYy0nrwsVFMquzvnl5DMRedpm+S8Q8wurIFhFcZrjUZJGVX1xSUCgRxoEXsVFNizotQDjbbxcPpBpjHO7Otb2qEuLq9EjBgUvp/uNn4jKFQvnpX3soeaL4tHdwVAVP6wr8pEYe+qSG8RcZuAVh2fmqUZAIBYHFnpVrPeuNTW0QU1dnnOP3Bi+OeHnBCwXiHSe3Le1yK/i9glKGmEMoYTgnMUmJL1YIN/KoJkWEdwMhQ8D15Ig3LIfq53XhNHzxIUKM7KoeVemN79bTgQt3qHnpLGIvZSI8WQu6VqKgwBSWr8/VhD1m7aE5u84RlPcJLUtQjj6LRbPJbUqJEKmYZPL6cPF7niV1YhrwRkFhbccLP77XHLrj600qhTIS4VdaoURMJTLAoUwWKBDVspMuXTqKFy8e9evXjyZNmiRogrJnz069evUSiT379u2zNQ9lz9l76O+dji8hLJQcx7ZvcC3hxob8tf1MxGWRUMTGseTtt0S46WARQdwXW1k+qp5bKJS4zGH5HNqgsOHvIqymUERYoKAt3HOBtp68ISxssOzIFgNPXLjnbt6nSl873OZw60ChhJUNY8EKAbl9/8lLrtKf25QSwfKwqLEgPODIsLpUdMhyESIwumlRalzSdQysVrmo++064V6CpbdZqSyGsYqpQ62xa+no5XuEWK7vVx0jxKfr+YH1ZBF45g3HO6w4uwbUFK5fSKvJW2nD8WvihWXBnosiZq9fvQL0fnBkHOWnf+02/AKCBB4Oe1j+abC45OHSh9vqu/8Vp3d/3S5eEBBAD5d38CjHs2aZ3bm8UChjsjTFhAOS3DiOFt+XoMEOlzpeOGAllQUJAhwTh7ARnDUkXbQtn42GNChM70za7DzbcHl78rJUbOhyunXfcfagUEan4H2z7Aj9sPq4c3mw1ObLkMzwIw8euZrO3LhPg+oXFC5veDFwIa/vHenFMDyoTTtonx/OGpSXekFQXkqasuqyX4bS5TuP6Mu3g4TL+wPpZcWUCQwMMmXDSRq68CBlS5OE1vaqRnXGrROeIriT4UXh752nZ8fAUnQ3/XzOXhGyVKNAemGVZM/Sr+1LU9WIRLXoBsPvwIczdonQl35vFBDfT7jzN/eJNKDoXoiHDZVCGQncK6tQogJO0qRJ6eFD19nKcIFDgURCTt++fQWx+YMHDwQxOlzd+MyIWH2oBs3fT79tPk3IPsWF3z4ipkp2s8zZeY56zN4j3DC7B9YS20GWc86+jjfAqe1Li8xSvsD/VyYLhZ25JX6A8DYLZcaowHUOlzqSKPCWvrVPiFgDW5zqF80YaQ0bVEvEHxoV2SUM1w1+ROB6TpowHh0YWkcMh8xdjqfh8RHjBIXy25XHokwJzPIPWCLcLD+2LEF1g1y7Wc5cv+9UbqAI1ft2vdNVCcXcTGk/dZvIdISFd+NxR4YzlGfERpkt+87dFu57yLZ+IRSQzGHhlpWl8zcfiLhF/OhPblvKuYQ2U7YJBQtWXbw86BH52cBSAgslrH7NSgXSyCZFqePvO0TmOP4NlzeHSPDY094rIzK8YQnXrkfP/PJ+F3er7ExK+efDilRMsr5irIu3HzgtpEhiQbzu9lM3heW0T90CUZTuRsUDXeLobk2yxfDJ8xc04J/9lClFItqkuRTlFxrEV67sUUVkFRsVjhlFTDOScmQXqdGx7N5efvmEEgWFfMGeC9SwWCYa17y4KctnD8/QBoUoZZKE1G1mWJTfIlMm0TnI75tP0cD5B0QsPSyysJiD7QIv8vhN55cjOTRK59A+a9b/n330x5YzVC1fOvqxVUnKP2CpmGt6h7JUMbcj2z46kb0rsMJCMcV3GN9lq8Tqu9+qfXkyzyulUHoCgFl9rD5USJyZuPaESESBhRJvaWxt4z3N333+pfgoxAjxF3bG+2UFlU6v2Xto9s5zgloC1iq4w4a/XZhalnXt+nWHGVtP2R353q/baWWExQk/bOw2lhNA3I0pfw6lNW9/R3zk7++WEZc8AuNxt54YXo/ixo1D8kXCfce+U1RY/RBnJEvYgJrO2CJXVipue/O/x852f7xX1mnlmtGhLFVw88NnZH9oC6UCcYWIpeOM+L2DaxEoMsyWgxfuOJUqOfao8Y+bROxml6q5BLUSFHe8AAAvYAx58/v1QpnHywcuLD0CSzAswhBYG2E1gmWblVKcZVgeYPGD4sYuSx4bllootyI0o0hGwouCETl9/T9nrCMSUzpFJMG4umSje3lhdzMr/q3KZRUub2AG2T2wplAu9IiskOCl5ouFB6OEb/AYCIWo+s0aEWbRtVpu6lk7n57hX2rTYPxGQW/UrXpuQX/w3cpjHsWiejS5xZ3cvbyYsRzEryOxESwIcHlrX+LNmEPvGNO3nqZ+8yLDnHhtCCfpXiuv87uEZL/AVPqS6PTO7Wk7/H6ztwEKJcJ9IIgfL5szTYzDIjYf3x8IxxbDa4dxrBKr736r9uXJPEqh9AQ1F32sPlTfrzxGo1ccFRYrWCh7/71XuCrhsmThWDFZ0ZRdwcjELpktlbBQQMlCbBysIEiuwUUL/kpPBPFEv2wIpyYls4hMx87TdtLSA5dE3Bm4xtgljsQWOcHDyFzstodFEYoOZ2YithF8Y7Kri8dFzBgUSsT8QTGCYgNBAgXHvkHBKZ3ddbC6nOjzdeMgZ7b4wo8qEXg8zRS8LMhUQa5i6sya78ilu1R7nCPhQ75oZMUDVlt254LeJn8Gh6VUjr/VG9/56OkzytffYYWAYg4LJZJw3q+cg/q9UZA++TPMmfjSuUqul2h0oEAu2ndB0DwhEWZMs0iqKD2YyEoilACOqUWCl8xjirHktcIyirMmJ28hgQMcqLCOQAF+56ctYglycpO7NckKCZJz8NyZfUDbF99PzAdcOBvf3fjazznpCXjDwoms57I5UtNfBpPwjM7rj/ZRf+8cLy+w/OMFYFjDqDH1nq6v5pi1dOzKPfq8bn7h8vZnTGpkmJMj4Yyp02D9614zX6QFvW8IBUTE2nu6b7P6DV90ULA8IA4bYVickMj3U0zzgBUBxgWEBHHyIGKvB79VyKzluR3H6rvf7YL82EAplCaBb/Wh+mXDSaclAxZKXHTabG4QeMP6guBkuHUhV+48dAZmsyLEY4GWhrks9XyZ9UIHFxBi5BoUyySScnAJ48sf7kXQOsedfdO0qLBQgu4Dgh9RJBjIwfi8Trik4PKGewUxRrD0QJD93TxCEXCnHHICBaw7yDKGGEnA0IsZ3HIfzQxzNmcaEL39jbSDS4wVaiTJZE3jsFxwQD8sj7CIFRmyXOCM5IOmETGjCBWABdVVQkt0a4ClDWEXHBcKNyQs42x1Y4t57ULphcubM/p5PMQa4mUJVu//lclKIxoZUwzwwpQrIuwDlw9zDR4bXleQ3cuCtaItstx/al1SnDVQHoEjs3X57MQ8f1Bs4fLm2Mwjw+q8RAIfHR6yCxo8lGNWHKVCmZLTom4OxgKzha2qsAjDqv/75tNUNV86+rV9GbOn8vt4SLApMNDx8gJPAs4aqL5AtTawfkFT1ofQl4MX71CPmnkpZdKE0YYsmDKZm0E4zImTDSuMWCnOK0Jn8D3m5DsjFnRfr1v2tsGyiBAQiDYBMLp1cMgIf96nbn7xu2GVWH33W7UvT+ZRCqUnqLnoY/Wh+nPbGfp87j6RiAILJTKBtdQtqLjx7q8OkvKTI+oJa4ScEBHaPZhyByQjdo3L2zJTSeIsPgTCQ6FEhq3WPW/0MbBlbHD9gjRk4UGhnEBYIZLpQpIlii9oPWCNgkKJjGJYY+H2g4AuCVQtECRewKoanTDFDBSIubvOi2ZwAacykYcSY8rUSPi3J7GCejGFu84V7RQH9INj8cNquZ3xWJ/VyS84IuVMej3xTvJ6WBEF9rAaIW63Z6281LV6Huo7bx+BgBrE47gYOOEMMbLIfEfyAxRKThgaVN+4NYKZDjAHFFNQIh2MiL/V4gYXHBRp0OxAoUQiFl5kUFCAralIKHq7RGZnLHP4l47QCz3CYQNwo0OhhNLjyziwD/7YSUv2XxLrxwoR7lKnUAaa2No6N6EeXMxoI788IGZ8/KrjtOP0TWHhhUXRDHFa8kPyCAvlkAWRSTFmjG9kDP4tx2/evsG1BY0bvDels6cSFsr//eywoB8YUpuSuuF4NDKvN21HLz9C34MCLEtKkZTDrA6IWUYlLXfCSaXcDt9TsJdYJVbf/Vbty5N5lELpCWou+lh9qGDx4+BvWChxCRXPmpLmdYkMRkayBJImIKBRAP3F8St3qcYYh3sTlDsokYhSXC0jyHh5a2b+4DCVEJQiJOWA9w7W0H1e8N5xFjQy0/FjxMKJKzJ/IBIc8JYOxQgKJTjPkM2LuD0I6Hi4egZ4EZl2ydXRYPcWXITIXIe4smx5e6xkSzLG8qUbx9VLBuZkmhrm5uQkHVAkgSpJm21vJGFIzqoHNyBiYDmDnBPOQFeEi5/PJoiOcTkiOxkxlt4oBvxCAr4+hEFoqbXk51dqWKjIcEfIBM4JSNfhdkf8psy2AJc305awR0DPOWDyaVAPQaEUdEge8MDqmQttuv+1m+ZGxJ9CoQSHq7a6iaEhAAAgAElEQVTKlN6xYkM7WJg5jAdnDbyhRmJ+3e2xyY+bxFnESxYUyi8XHyYmzXfX1+zPUQgDL8f8gsSURlDWYKHkcCNf/GZ5updvQ4/R2FCHVR4ub5kBAmUw3QmHnXA7PbGX7sY08rnVd7+RtVndVimUJiFu9aFaeegyvfebg++sbXmH2w61rP/oEFnJRMubiNhCOVMUZQsRR6N1DyMRBAkznmSQuoJTzkhHUg7iPVHxY6cU72n0MYDmBm5Szuzj/kzYu+n4NWfFClYacImAAxHWGWR5gqAaIiul7hJfOFEF4QWokhKTZcvonuT2yMZH8hRXGtLS9XgztravHNgux0dWGbVahAUMfLMgvVvJkfgFRY4zZGUrMJ8lvetiqhUkf8FqJLuRmay4fM401LFKTqfVD9x6iF+EZQnrQGUjT0nlEQ+KusaogIKqHDERs3McGpRoJMxAprQrJWphc/lN8BoiKQcYaUnJ3WHCijpiGpGU4+q77G4MI5/LFmD8foC83pPQASNz+rOtzPMJhRLWcLa6m7EuVATbHH5dEPHDUwFaMk9J571dz9L9F6nzH5Ecq0UGLxOMG3jZg0uey+qyx8rb+czoLxdH+LFVCcOZ6APn7xdhGyxy2I4Z63M3htV3v7v1+PNzpVCahL7Vh0pWFkHGjJg7ZGmDa5Fl+6kb1JSrfERQ9CBrlzNRUfkDJNawvsAKw6KNxfQWIllBQKUcOQvR07GZIxE1a8FZyIKsb1AprT5yxamIlMiaUhCrw9oFhZIzzmfvOCsucNAlMRm7u0ShdlO30ZojV0VcKqwevqoPi/1UHbWaTkXEeYLwt140dEaeYsj9UPqMS3fKbiY5+7hN+ezC3SsrOzLGRi0esrKKCwUlQpnPU46pgssbFggoPrBgIKMcrmG4vGFZZPe7UQw4MYX7wSUYXVk2jnGUXzxmvl+OyudKQ3yZweKNMAhPrO8ykTuSDKZvPUNci9novvS0h1KMuOkKudI4atwfNzemUM8arGwj83yyNdxVXXpP1yQ/P1C0jQs9RkGZU9CCj4xVcPJ0frkfErbwfeG4+UIDl4owEVRtgoVSJvQ3Yz4zxvhp3Qlh1cULIyyUctEKeNDcCSuk3A7leEFKb5VYffdbtS9P5lEKpSeouehj9aFC/B9idyCw5OANWcutFnbmJr0dUaKQCatlRZS/eFCMUO6K616bHb81culhmrDGQXEECyWIlFH3eV3vyDKYRh8DcxUWzuxQMli4XJeckITsQVyaiDU9dvmeiL2DC3nurnPi7R2KOH6IuXJKTGvhBCNuE102rtH9uGrfcvIWJwel3gB1T+YFTRQHwqMKTZFARyUcmbAZJOPMLMDWF+DXfdYe4ngtI3NzyAK4EGGhxKXHsU9jlh9xlFXMkpI6Bed0XoJFMqdwuhahUMrWUyNzoy2fH+4XU+UUjpEDBojthDBnpaycwW3cywXbgru1Ma0WrIRgEkBMIxKSJrWOfDl0N4aRz5kgHSEycHlzFatetc2JKTSyFivachwhkrcQrwuLPBL08JJkhsi0aHB5j199gvASO1cKPzJjHj1jrD58hdr/6qjzDiskvBwI0cDvbY9aeW1ZZlMmY/+xZUknhdnmPtUpY4rEbrfNv0NoqGU6cdvZhAZW3/0mLNlnQyiF0iRorT5UciwklBrQVuDCQ8ICiyv3tmxVkhMHuL43+sZ0uXoC19gVRwWZeJHAFEKhNCPGiOPAOK6O14VShY1KBAqLLbKk4X4E1yasmC3LZhUK5bZTNwS/4b+7L4hyi2zBlAngo9tnv3n7hAWJxZcXR++/94gEIghnr3uCv7s+oHlCrBVkXpcKVDxrKvHfzprvjYtQs9Kw4p4RLl4u08dlz+SSl+7m4s/lGtiwMIAuh6mqwIuITGfUZsdzgtUPCTnFsqZ0WtNAG4TqJJ7ypXImOa8nJvohtmYicQX0VxCuNDJiySGatNZRAhJJOcAHZ3JbRMUmPXjISTJQKBGKwXWN9fQ32oYtOngxQFgLXP+cEGV0rNjQvtLXq0R4CmKlEWvOZSebl8lqyvI7TdvhJOYHD6WvY2BjWrQcNw+vAcqK4ruF6jGolOOKXs4UELwYZNrmUzQggowdLm+jldRkIwl+MxZ/7Bt2hOi2aPXd7wXUPu+qFEqTILb6UF249YAqRBC6QhFC6Tbm8eMt4aJg7kBUTUD1hKX7L1HnP3aKUlVHh9d17p4zevEHxCV+1biIScgQyRcYLkozYoy4ugJzj/FiuRY3v7WCvxEKJfNgHr1yz0nqvHDfRRGTh7d3lKJzVZ1ECwJzdvLffUm3wsHqZse0avckE43/3bk8lYrg4eTEGVbSlx+4JJJOgPmx4fXom+VHBH+pJ0p104mbRMUZxJ1N3nBSLIlLrWFMrt7SMTiXSHzBGS+RNZWzJCMslDjznG1t9LByGAb3iynpiStJIVEGJT4hnNDGGaqw6kMplauU6F0TW72RKY7Si+DXbFwikEY3K6p3CEPtZMoxKJRMyt2hcmRJTUMD2rxx9dFrxPcccchQKHHe+UybsfSuM3Y5E5twTqdufDme3Yx59Iwhx44jDp4pkxCzDmJzM8KN9KzDSBt+UcWLGGiDjBYHkDmHfRkqohRK909VKZTuMdLVwmqF0lWtam2CgmzFZBogJ62EJstadq8i29RM99fP68Jp+OJDIkYGtEFmVOYYsfiQsARohTnImFYJ1jTEinGtaGR5M0UNlEy4y5FYcz8izggk5zEJKzvchutP6zokBhvxmzeynae9F5lsZXAYt81BieOqOgXHX6G+NvYpUxmBBBxUVX/tQPnDAJrctrTbeeQGHHeG8opshQUfaLmcaYgtn6Bvgssb9Fh4johzREIVlC0olEimAe0QzpRRYSso94spuYe5JhGHBs5OCGrGg7KLlX6EXiApJ7oqNzGtT84UR0xv6KHLIq53RCPzXurk+eG2R2IO4n9RevHSnYceW3qN4u6P9jL9FVze3pwbV+vnevbwvsBCiQpXvnzRjAnDreHXneT6ewbVclakQlgKYihBaWS3uu2zdpx1Wk5RqIKLA+wbXIuS6agMJtOXab10Vpw3q+9+K/bk6RxKofQUOU0/qw+VXH6Ql6IldD117T9Rqg0CRQkXIn95076ekHb0j6yq8/GfYYJOBwJaFlARmSW/bjwp4ibxQ4bLH0oZ4in//qCCx1NoFQIeiOlA5Jq2SJ74O4Jr7+iVu85KJ8huZYsT+sMl/2/XmAPp/9hymvr/s9+5brjRh0thBh5vKJqOqB8ekPw1nwaZRynHKZWRzNd/iYi/4oQguZY5XL6wUCL2lGtwG9l7h9+2i+xixAqijjeE3e2/bTrlJOqHQgl3GKzHKMPG5RaX7LsoYn6RhIYYWKPCc3A/ZHCDDsmVMI8q4rMQHgDhSjgT1hx3csDC5R1TlZvo1siZ4uDERC1vuC3bls9GQxroq41udO/zws7Rp3/tIVjvYaE022JndD2+bt/ghw20J4IqCN4SxIyDpL6Wh5XAtOvl0BScZSiUqCetTZD09R55fLm29bZ+IVRm+ErxEXh/EUNpRriR2Xvh8wilF0k5TBNmJLmGs9mhNOMOsFKsvvut3JvRuZRCaRSxaNr741AhkQYWDRZ29/K/ZZ5ApoPB2zPqRMP9DTe4s29E5if+baTqiR745PqyeIuHixOJRDM7ltPT3WUbtmJpP4QLtf+bBaNYuWCh5MxZWCg5ngrxpMjYZtFTfo75P7kPuOeQaRybRX45QXnBynnSie0wfx9fvloic8Q5gjXAE5JopiCS+TyZQ5TPCyxonYJz0dCFjiQuPEdc1rD+8HPjrH6j+PMlxv1icp1zXXVkRDOBPngmkUkrW98bFc9M3yx3lEPFXvSKXMsYPJRMQYNz7AtxUsugKlAcIjx/WIZQXvNVFA6vQEgQSvxBOLzCjP3KNEyo3z5n1zlCEYcJLa0nigeVGijVIAjLYE5HhKlA2cL59BelUXRYc7w7KLwQQ9luqiOpyB3jhjwefk/wkonyodGVzjXjWbsawx93v6/24u24SqH0FsGI/v44VBzjxlvQxgXJdDBcUpAVsZxpk9KqnlWdu2dLC/4w4/2yIu7QLAE9D2e/QqEEZxhoVn571/NSb+zS1q6R+fRkbrMKudOIuCaUHwPVDILykfG5/thVEa/Goif+Rk5qQj9kKXepmtssqPwyDjgvUQoRgmoi1fIFEEoO5ujj+BtzLuK/mdMPGdmgR0H8HROfG1k8W/1wuSEEAcJVitiKjrivjsE5acQSB1E0eFZ/23xaEPiHnXFUOZrVqbwgATcqchUp9I3JYvXl4kOi3jWLzAYgW9/h8vYknINrGUO5RgIFlPQuVXNRbx+9qKw5csV5afOe+LkbxTE2tOdwHpkezMzfOPmFABZKLjP7bfPilsMjJ2LCK8UlVbEQhDIhA71oYAqa78YTY+XC+QUnQbw4hCxvT7gy8RsGS7vZFcv04OCPu1/PuvzRRimUJqHuj0PFdVp5C3AX1Cmcwbmj6/ceUckIfkl2J8oJMks/CXa25Uscf1jxaTDl0VGhQC90ctwmeCjB+ehtKUGtpZDXwtRJnFkOPjgolMjERQweFEr88KAeNGr6wprAgsonqIASk8g8nmg3rGFhalUum14obNsu++eLxNp+aVuKQgqkJ/Ah5um3RPxNtgLKJN9QnoClJ4kxbNXhKkaYZ9Pn1SlTysQku8Bg/eQkLvCLQrGDcnni6n9ibZ7SKcmuQYwTU3UNWGKxV5YUiRMQ4tMgHAKBLFrQBoEeq2S2VDTHQDjHqGWHnVQzUChRycVTwnY9B0yOs+P2HL+qp39sa8O1y/Eyi6pAEC6AYMZeZJ7dVEkT+DypKqY1y4mY4JR947sNzuZct97bcCMzMJPHCD142alEgvYNxOxQLpH4FxvEH3e/XXFRCqVJT8Yfhwpvn5wkgG3I7kr821X2Ll+O4PjDZcwiW2yYs9IkaEQCBdfvRgylGS4hrYWJ18qxS5yNjcu9Yu60TssRFEoOykf8pFxhoWnJQBrVNObMWhCj1xzrKF0JsbpurFnPRDtO7r6Lo1D3PHj8zJkhKltz3p6wUVgHYTlkq93UdqWpWv4AQ0sbuuAgTdl4UtABgYMSwueOy8fBBYZ5QDmFF4Nq+dIJfkokw8DKDJEr+xhZgPY5srvd1RgT154QsZEsUB439wkR/5y1/Sz1nrOXEJMMhRIuVfCe/tmxvO7ljAs9Kqy9SOyBy9vsSi7ahcgVjvgzTxVz3Zv0Y0Om9ZFDJRZ0rURBge7rROtZNv/WQFGDy9vXSVUxrUk+13O7VKBGETzE6IPa7YglR+jIjPc9DzfSg4mRNrLFHL+noAnDd//QF3WMDOO3tv64+/22WTcTK4XSpCfjj0PFwea8BVhFoECxyDFvXNnDyZunqRV85+ETavDDRmH9QaKDWWUXsRZ+A0UMGhRKxMxoSdiNPoYt4dep+U9bXuoGt/b0DuVEti3oURCrWSlPWqeVC3ydHJS/88xNYblk0ZMIIYcRoN+r4irM239JlFi6uw+fUNDg5Q6lSXIrd/hth7gwodytjog/ZZJvI89QS7+Evlw/nqmt4seNI3go4aaDmxtWbVgr4XJG3B+E3eRG5kZbba10to66GkebwCOHizA9FayWUChRSchoVj6Hm6BEKBRKua650X3paa9VptFn+afBpKdusp7x7daGaZmg8KHmttn7lYn4QWyO+N7W5bIRYtqtFplCB7/5//s58jeybuEMgiXB23Ajs/e08fg1ZyLOyMZFxAuaJ8USzF6X3vH8cffrXZvV7ZRCaRLi/jhU2vJxWivLo6fPKF//pWKHbL2U4320VDSImzNTkWRo1yJrdco28U8oBVBI9FgDY3o0+87dpvo/RLpzuC1X+eFECvx4IvYOtEXZ0iQR1VUgCMrfc/Y2jQ096pxGT4KNrKSjo5muM5OOokfDyPWOwRV687/HVDyC7ByWDnBAQvrM3StCFjIkTyToZiDMyWhkYqbbkftwEL5sfeZECsRJ1iqYnoYtOhRlGr3VNLRrkzPb8RlnbbvagxwOgs9RAnJRN0fSDScUgHoKCqUnZRNlmiSEGnDSWNsK5lRy0e7p7I37zmQN/szq+sdGzoq3bZmWSY7XXdOzKmVPm9TboUV/riAFUm24vP1ZylJ+tnjZbR+R4IJ14kVn/bFrfstAjw5sOQQD5POD/j3gl4o3nh4Gf9z9nq7V1/2UQmkSwv44VFzyK7pLIUqyRYRbkilKvI1hNALb5hPXnW/KXCbSW7od+U1cXgusPMs/rUKf/b1XcCTCBY4fUhBOI8kDNaMheHtHADsUTZYeNfPSR24oJ6B0I7YQsW4Qs+NNjeBqZtugQcvo7qOnThd+dOUYuWwfzw2LDDgZ4yNj2IBo3ciwRh7/0hEzJVf7aFUuK/2x5YxIqKpTOKNgKJDFm/AMtsoiWxvKbHQvU6w08rxyDJqcMQ2FEmcOiu9PbfSXTZQprqBQXrn7SFS8AqeeL0R+tjw+KGYCkiXyxXR+H5PjdcFsgbKLkJgs0kYXLBPxw+WNUBrQXfWpV8DoUF63lwteIKYeRSxYuCLYG0EZaXzLmGPFvV6IgQHkeObP6+YX4SUIa8HvSmwQf9z9dsVFKZQmPRl/HCqUFsRlx+KqPF/OPosEXx9nsXaftZvm7jpPehJQTIKG5B8McD0i6SCmyiR65tW6njkWj+mQmGwYP55QKEGOjepAj585XKUIDzh86Y6oHMHS/40CpKdaSPGhy+nm/Sei25Y+IZQhRey/iIsNXS4qzyBZqXHJQLp4+wGVH7FK7HFxt8pUMFNy8d+c1cyYvVMqC33dxDgBt3ac11+LL6yEEPkFBNVncF4R/1YvKKMgQJaF3eR6zoy2TalhoXTt3iO31hA5aQBjyC7tlYcu03u/7RBDgzZobpjx7xYzFuASRelFnC1PEp30YiAT2XMfvSTSeuewUzv2yiAsAXHlEDNLmcoWZlTKQYxx12q5qWftfJbDIIdycDwiLyJf+mR05PJdYUkf+04xy9cW3YR7zt6iBhFUR0hGQ8y0nqpldtmAP+5+u+xduw6lUJr0ZPxxqD6fs1fw8rGg1FbihPGi7IitMExOzfx/MdUuNgkS5zByEgBn6CLZoq8Xb/DaSxFVVUBhA/JpWK14n/jxhALQfdaeKNsCjdKxK3cFwTMLqIRAO+ROqoxa7XSdHxxam5IkjO+ui+0/L/nFCmG9HdmkCDUrlYVk15lsheWEGd6QpzGkWtonmWh/x6kb1GTiZjEFx33Bol6/aEYRsC+LEa467UMIGb1GZIu7q0Uul7PDGLIFUg7nAPcgaKiMxgfP2XmOeszeQ1B4oFAiScmXyV6YI3dEBj9jgrrPCQxamW1/qCMWyFW15DKtewfXouQ6qrDo2aNMxA/aoH3nb9MnNfLQJzXy6uluahuZ2QPfZfkFjC20nhQiMHWRmsEOXLjtzEaHZRcV0MA7u653NV9Oa9rY/rj7TVu8yQMphdIkQP1xqDhTFltAwkv4l/Vecttx+Ty+oLhCCfM1mrT9GIeBJbDOuPWiDcfeefsGL7vzMW6Z7Klp26kboiLEkWF16f3fd4gqLrCggW4GCqYsUJKgTMguIb2X+JvfrxclG925Sq3A1qw5ygwPFa5WVqqhnFeLqLK0qkcVypnudTGVbD1E4PzO/jVFkoxR0RKLy0T7ssUCMbBQ2qCsvVU0c5TnJbvJjc6P9o0mbKRdZ2655eWTS06iX4NimYg5Brk8Jv7OWcRG44PlOEwk5cCKDvoUuPh9JXn7LXFa62MTRYsneHC9dbmvkSos7uaUS1lCofR1ln5M65FL8iIpSA4RgfUUXgiEkQxrGORuW5Z9jmITtSKYM9qUzyaYN1Cmd1WPSJ5kyxbjwUT+uPs9WKYlXZRCaRLM/jhU8g+l7DKUt8Tk50x6zjWUvXU5G4HtxNV7FDJ6reiS7LX4Ilbv0xp56eMa3pXI4kQSjAsX/qIIjjlYreCGhCKCbEsolFAwZUFt8zM37lPrXxzJQhC95dha/LxFcFgifjBsoIOPMLZL+REr6eLth05eTTkTWE66kevDe+M6Q1WLD6ZHKvnyBSJbLPhFAbXEYVXnKhrAG2EOB4Z6Ti3Sbuo2kZHrLitb5vbDvM1LZ6GvGjvc/NtP3aCmEdZUJA4hfs7oyxpnteMFBQwEEJlM3hdni0vVie9kovi0b7Aj3OBVFE6akfeGl++4MFmaIDIRP5Q2vKh6QvZvwlJI9twghEdOYsOLAyqrvVsxBw2s75sqTJ7sQY6Hx8vY7J3nCO75ZZ9G8iR7Mq5Vffxx91u1N6PzKIXSKGLRtPfHoZKr20QXxFxq2Aq6du8xgY6hWeks1GziZmHJAx1Ln7rWBI3L7lN2O5lRYYZj4PBImLQX/w13Vqffd4oSdqjPDIWSs8z58SE7GAHsjX90uFYh0zuUFZyV7oR57WKTW8bdnip9vSpKdrGsRMkJDIhBQ7wlShBOal2SantYD1nLIyqXK5SVWXAzwhrcuESg4NGTaVA4vMHd3qL7nLN/3SnGp6//R1VGrXEO075idhpUv5D4t1zqjteqh35KXtPqw1eo/a+OcnMsWk5ZT/cYXb+yX4bS5TsOLs+AZK/RtliSAOEJDpPWnhDVlljMtsj+E3aePvlrt1DMYaHEi+qg+gWpfUXXteE92YPePjJ7Qa/a+QTNllas/O3Xs275fgDDBCz2MpOCnjH82cYfd78/9xvT3EqhNOnJ+ONQyfx40cWB8cUx/O3C1LJsNmLuym4heURtVytEm0CDOfUmwMS0PjmWUf7xRKIMXNyoagMqoOA86aIoIhhzZ/8a4kKt953DFQ+R6XFimrfX7D3iLRoKxMKP9NdstgJrT+dgLAe+WZDerZSDZFombQYwkhDgHkcNc1jVPBFtXCJ4Jud1cRDty+52WC7Dr/4nrIJ4IZKJmr0N3IerDTykHSrnEPW3oxMtZ6VcFlEudcdrxUvMAAN1uDccu0atftkaZfqYKvd4gre2T9VRq+lUBIUW6LTW9ood8Wqe7H3qxpM0ZMFBZ1dvLdvaNXBccaIEcYVCCUs/3M3wjlgtcoUr/MbLFZ54Ld2q56butaxPGIoOCzkBEKwcCFXSFt6wGkcj8/nj7jeyPivbKoXSJLT9cai4Rja2AA60xR+/rNxU/GqVoMoAvxd47eqMW2d5jM+N/x5TiQhOQ4Z7aINC1Ka8dzx79b5dTwcv3hFDygHoiPlD8gaC45E1WCVfuiiKCNojo/ja3UdUNSJOEH/TW3WFS+WB3Htqe8/rkZt09EwZpvroNUJx61evAL0fnJPkuMGwATVNr5GrLX0oV5c5d/M+Vfp6tdgXqtLggsbl3LxMliil5JCIBWJzXwtI/4tEkLxjLpleSo7/4rWiXCToT/QK3OTNJkVaytFP78uN3jm07fh3AH8HP6NchtXTMe3ab/rW01HYHMwOVVl24BJ1mraTENML2iAwB3zdOIjeKe0+wc9szOTYclgi5cINPJceejSz1xXTeDKNFXNl2q08ZEzr98fdb+XzMTKXUiiNoBVDW38cKi5piGWhTvXszhVeWiFbItgiiEQLWIDMsBDqhc4VTclXjYKouY6M6pjmaDpxE20/5ah8IZP4IoO7x6w9giIDlkskdrz5fVQS9KPD6tKtB4+pzPCVzin0kjvD4ooqPA2LZXbS6ejFwq7tao5ZS6giBEUICpEcG2hmRizvX7bs4W9IaPk1QjmXLdpM9QI3M2qmcyyulYqQbPXBvPJ3R47/4rV+VD039TBgAQo7c5PelkrkYQ6c4cKZzSkN6OrMcQlNfMbFAOx6Nr1dl/zijbHSJ3+NtvY1j+NQDt9Inig+3Xn41Em/5e3aPenPVHFyGJA8Tp+6+alTlVyeDO2TPnIRBa5mVDZHavqrk/7ypT5ZmM5B/XH361ya5c2UQmkS5P44VHIN1OjKaXG9b1YUKoxYSRcsdsnIFXsYbuY79AZ+TqrAGIu6VXJar+AuBO8kK87ARq6/jfYnR9QT9CyFBy1zLsFMbjpv9uWPvlrLtZy97IqOyts1ogY9ziZLnUIZaGLrkuKfskWbuUNBJ9K6fDan5RLtigamoPldK3m7FF398/RbLBIaIBw+gv+W4794rQglgbtRr8hJSNzH16UQObEM86HYwMyO9qntrBc3ve3+3XOBUH6RxezY5/XHrjqT+/gMfPe/4oREMn8IZ/D/r0wWUdVKKxzW4o+1uZpTLvOK2MkDF+6I6mZ/dChrlyXGuA5/3P12BeaVUiiHDRtGoaGhtGZNZAD97t276eOPP6YdO3ZQmjRpqFu3btSzZ0/n83j+/DkNGTKEJk+eTDdv3qRKlSrRhAkTKHfu3IaemT8OlewqA1/fj60cF7IsrCj0rJWXulbPQ1q+QUOb9LCxluIHw5jxg/vh9F20aN9FsaptfUOozJcOa+PUdqWp/z/7hasfrvWqeQMoeJTDhQphaiFk1Obqu9j591eFU9KTx/TGd+vFDzkrQ/IlCWuuJ9RAMa1DW/5PpuKRLxgeAzRTCNkoPTzUOSyyqlFn3AoJGryM7j58KqZixgT8t6v4YMSWInZXr7iqrW1maUBX65CrbFXPH0BT2pXWu9xY146z6HnhuQNep9DuVUzbx5bw69T8p8ia2RjY17RPMS2+wICl9ODJMyfRvrbtsIaFhbXfLiInEnEcsuyxsMs6o1uHP+5+u2LyyiiU48aNo+7du1NwcLBTobx+/Trlz5+fGjZsKD7bsmULdenSRSiM7du3F88EyiT+PXXqVMqcOTP17t2bwsPD6cCBA5QwYULdz80fh0p2GyILdnSzoi+tt/73G5yxhJ/WzCsscnBB6+Vc1A2Am4ayhcfxg1uS6hTO4NXwvf/eQ7N2nBNjwIpW/Ivl9PDJc/qhRXEa/O9BEcsE13rVfAFUbkSka1umScnXfwk9euqonmMmlbTKDrQAACAASURBVIhXG/NDZ22ylpx5DGuu2TXeteX/5Io72jrbgANE0UgWkmMZ3dH9mAmjnBUtKwsykTTPZzScRJtFjnE2fl6dwM3pK2Hif4xvt1J8Zu9Zm0UfXby5p/Nq44ExzuQ2pahGwfSeDulVPy6jiufKL9zygMz44dUkJnaWifaRaAcPmpWlgb3dij/ufm/X7Kv+sV6hPH/+PHXo0IHWr19PWbJkofTp0zsVyhEjRtD48ePp1KlTFD++o5pJ3759ae7cuXT48GF6/PgxpU2blkaOHEmdO3cWn9+6dYsyZcpEU6ZMoebNm+vG3R+HSs6GBSHs0AaFX1ovx0oxkTgrdmYodLrBQdLQwKV0//EzZxczePa4pBqTuoNGSFR7aVyEhi066IxlqpY/IEpSkFyVBclCcLEmSRiPDnrBaWgECzu21Z6T5QcuUcdpO0UW94mIGttmrlub6CKfX63lGPMiFvb9yjkJlZ9YrLx05KzoX9uXFi8pEO0+8DdOgNOLl1x/mfugjjGowHwlTJmE8aN7GfXV3FaPu/H4NWo5OTKL3uyYUbkSGO/tt3fLiNhtfwiXUQ3JH0ArD195aQnj3ilGDYtn9sfSXM754sULytHH4SlCwhRKj0bncbPNoqWF+OPutyMOWFOsVygXLFhAM2fOJLi7hw4dKpRHdnnXq1ePUqZMSTNmzHDiD5d4zZo16fLly6Jt2bJl6ciRI5Q3bySFDtzeRYoUEZZLveKPQyXTmUSXWcqJK8j46107v9PF62nJPL14aNsxwTr/3QyePc62ZlL3yiNX0dkbD2hw/YL09dIjwu3z/f+KE1x6haRYSbkqC2fBy0qmp3uMzf34nPA5YuJxuLrh8jZbHj99HkU51JbizNFnkeC6ZAFRNBTKnH0XO//+ZpGM9EOLEmYvzeV4db9dT+DmhMDNDnc75MHjZ1Rg4NIofeQYSz2LgyUdL0Oy7BlYi1IkSaCnu0dtUEFl2pbToq/dKqd4tKEYOskJZmhmdqjEwQt3otCPYY4ZHcpSBR2ctmbvFeMx9zDiEDccv/bSFONblBCFIOwkbOhInCCe+N0GHyV+u2OD+OPutysusV6hlIFt165dFIUSSmHdunXp66+/djY7ePAgFSpUiLZv305nzpyhxo0b0/379ylx4kj3UrNmzcTfFi5cGO1zu3PnDuF/LBcvXqQyZcrQ2bNnKTAw0JLnLWdPR0cF8b+ftjgJvnvWyue8/Ga8X5Yq5HJP4m3WRhD7Bjcnixk8e0zszlmbcmLJmBVHRdURkG9Docwj1S6W6WY4uzlL6sS0vnd1s7Yb68Z5Z9Jm2nryBrFix4kMZnP2MTCwSsjKoTYzmmvQc3smis4/YIkIa4BYaVnjMo2Yd0HXShQU6MjAdlUXm+uh6z0EIIvHC5csvkiEkscfvugg/bz+pPhTh0o5qL8B3ky9+7JLO7mUJ9ZkdqiEXD2K9zy7c3kqnd3x0mG1lPtyJV2681AwfzALhryGn9uUIvA92kk47hPeJrxINiqemca8U8xOS4x2LUqhjITmlVYokVjTokULYblkQXxkrly5hIscFsrWrVvTs2fPKG7cyHrEbdq0oQsXLogEn+hk8ODBIv5SK1YqlLJrEETKIFTWCpdaRPUOxFAWG7pCNJnzQQUqmS2VZV9YtgTyhGbw7LFbtmLuNDS9Qzlity2sbBPXnhBTwRJbNW+6KMqLXNarwfiNhAsnNpX68sVDazl5C208ft1Zlo1rbYMGZa+PyvLxJYL9wKX9YbXIRDiuQc97ZaJo2dLdomxW+vJta2oSMz5YT2j3YModkEwsTXbX8VrHvlOU3i6u/6XSVcwoyofGj2e8RrreszFm+RH6btVx0dwozZHeOezSTls6E67gX0xMQjp17b8ofLbY97wuFah4Vut+X2Ws+be2SGAK2nvu9kuPQQ7ZsMszkpPesKZmpQJpZJOXcwLssl55HUqhjETjlVYog4KCCG5vVxbKXbt2ieSbJk2auLRQPnr0iObPnx/t+bWDhRKLY4vNiEZBooawVt79dTutOnyFcPmC5LtsRCa0r3nutOuQY9DwmRnz4zJHNZw8AcmEe7DV5K3CxYOqKn9ud9BlsOtJtmzhh/bfCLoZtuDKlVrs+KPl6zVpa7zL9Yl3Dqjpk+mLD10u4qUg2kQWjgPjiZkoWk6OkUsg+mSB0qAdfttBoYcui79oE2aYpoWbIynszSL6KWO0MaMoPhQ+4g2fbkku26pV5n06sR8GP3H1XhT+UrPj88AmASVOFjN+3zyFiqte4SUZXLxasdo7pWcf8m8B2uMuw50WG0QplP9PFEook6AKmjZtmnPHrmIojx8/LqyWLIihLFq0qEjo0Sv+OlTsiovujbjj7zto+cHL4o2va7U8Tvoc2cqid4/etKs1di0dvXzPOYQvePY6T9tJSw9cEu4clO+C/N25PJXKnpqKDF4mknQgchUGpk9hK6c3e4zNfdtP3Uarj1x1xtPN2HqG+s7bZzoJtIwRu+bwN22pOrlOOz5n3tLgkatFrWQIqHlA0WOFfDQzTNQYhuwaUJNQR5xFm3DmScIb6KugWEJQwu/wF+bHrco4yeUI7cZLaPbzjImiyoy5tKU5MeayT4IpXwaHFdtq4apXKMfL5TXlNcz5oDyVzOYfd3x0WGhDoqJLMrUaSz3z+evu17M2q9u80hbKr776iiZOnEgnTpygePHiCWz79OlD8+bNE1nesEIGBATQmDFj6L333hOfx6Ysb6z3/uOndOXOI8qeNqnLs8NcjYhJwQXMBN/re1ejLKmTWHbemOeQJ0TJPMQymilMIwRrY9iZW2Lof7tWpCKBKUXSA5IfIBVypaEZ7zuInLvO2EUL916MVTQVZmLGY3X4bTuFHrritAxM23yKBsw/IKhrYJHzhchW61FNilDTUlmc0zABP/+BeUs55hV/B5XQJzWsqUevpahKnNDxewLRJpx5wmAgW9BlWitf4I4x/9x2RpD/Q6LzbvhqbqvH1Sp8TUsG0qim5rlT5UovvDeUf82Z7nWrtyrm45d3LgWqXQT/JvplcdFMqv2+v1sxBw2sX9BOS4x2LUqhjITmlVYor1y5Ingo33rrLcEvuW3bNvrggw+Ektm2bVuBQr9+/WjSpEmCJih79uzUq1cvEVu5b98+2/NQ6vm2ffxnGM3ffUFkzaHaCJcgBBF4QPJEeoYwpY1c6g0D+oJnb9jCgzR5w0lCgg2yvSFcn1uO4ZRJc3/deJIGLzhIvevkoy5VjZHZmwKMTQbpNG0HLTvgsGQjdmnKhpM0dOFBypYmCa3tVc0nq6w9dp3TJaclupctkZj8x5YlqG5QRmJeVfzNKIG4N5tgiiqMoeXl5KxaHv/3d8tQsEHKGNmCbgXjwPzd50W9e4jdaGS8eU6u+t6+/4SKDo1MejI7q90VEb/VL+zyvpmRAFZ0UKJphX8TzcbZm/G033fcVX3qFfBmSMv6KoXy/4lCiW0imxvVccLCwihjxozUo0cP6tq1qxMBJOSAmxLE5g8ePBDE6HB1Q7k0InY9VKhpPWfXOaoXlEEk7TT+cbPYlq9pSbTYNZu0mVDZh2VbvxAKSGauQvv9ymM0esVR4TLkTGC2FFT/Zg2FX/tPTF+7UHqa1LqUcy2wYFipXBs5V1a17TJ9Jy3ed8mZOf3zunAavvgQoXLFqh5VfbIMJlPH4D+1Lkm1CkUS3YeMXkMnrjqeF4SJopv8uIl2nHbUb+fMb58sTjPoV0sOi0Qv0Joc+qJOlE/Lj1hJF28/dP5t5vvlqHyuNIaWJVvQYVna3CfEUH+jjZcduESdpu0U3Txx0Rudz5/ttdROZlu/XCVVbekTQhlSmPv7phdDfukCQwPKy2plZY8qlMtP1tPo9sBuev78w2q5qFdta8JZ9OIaXTu73v3e7suT/q+UhdITAMzqY9dD9dnfe+mvHWdFXGH7CtmpRQTB7+Ev6lCiBJFuO7NwiG4cTpjhz3cPrEkpk+ivRKRnfb9vPkUD5x+I0nTDZ9UoMFUSYkohfBibOM707NuMNhwj2LBYJhrXvDhx0oYvs9+bTdxM2045XjK0RNDy88LnnJkqZ1sjwxvJZlbIt6HHaGzoURE7iRhKWZj/lP/GcbtG1iW7/HxpFeY1rTt6ldpM2eYSeyPrjg1ttUlP0XH2eroXV0T8O/rXoLSv+46YPqa1Nhy/kXafvUXx48ahpxFxuXJ7f1pPo1u37K1AGySQgpUkNohd735/YKcUSpNQt+uh6jdvH03feoaq5UtHbcpnp/a/bhc7trrMIGebM9y+qJv9T9h5+uQvhxuPhV37b/2wwUmhYSV/oUnHy+fDfPJnGP0TERoBQuHvVh4jcHkWypScFnWr7JP5ObMcg2t5SWXXNj7nbH35HHGijk8WpxmULbauYkq11pX5H1akollSGlqWHE9qdq1pVwuRyb5lonZDi45FjeWkp24heUTNejMlZ59FJOtuewbVohSJfUdMH9PaZSu+q3Zb+4ZQegvDnfTgrI2x71krL3WtnkdPV7+3sevd7w9glEJpEup2PVQc+wUy35Zls1LnP3ZRwnhx6ehw32aRamHlGD3++7HhdSmByTx72pq9mIstoVwJBn+zkr/QpOPl82Hk0IgJLUsKZRJKZdHAFDQ/gmLJ7EW8//sOZza+VgnTxtyy0sOueazFKD2PN+sHl2GLn7dQs1JZXort0lpXFnWrRIUyOYjP9YqcbGR2rWlXa9h//rYznlomate73tjWLibOUzP2oiXi98ULs951cpGC6NqHDahJqSSWAr3j+rId8wHzHJ/XzU+wJMcGsevd7w/slEJpEup2PVScqFIuZ2qRwYtA/GSvxad9Q2qbtHN9w3A2NbfWJjboGyXmVjtP33DGiHJL/mGXXaXtKmSnwW8VMmPKV2YMzmLm+NKRSw/ThDUnBPk9SPB9IfKZ0CYKaGNumRbr079207yw82I5HFfpi7W5GvP58xcUFySRGnnz+/W0/3xk1awVnwZTnvTGKGNkCw2sm1CwfSnHr9yjGmPWiimsphDz5b6iG1vOxO9XrwC9H5zT1GUUHrSMULmMxRcvzHoXLP/Wueqzf0htQrlaO0njHzcJTmEWLS+tndaqXYtd735/YKYUSpNQt+uh4mQCcC/CutJ7zl6yIotUC2v3Wbtp7i6HIuCr+tCuSqBxxRHmWcT8sSmD0KTj6XaYPnP30cxtZ6hGgQCa3LY0jVh8iCatCze97rG8kJ6z99DfO8+JP63pWTUK9ZX2UmSi6M/n7HWS1ptRD94tMDoacMwaN/WEEkseo0z21DSrc3kdM3veBHF/badso+cvXtC098pSPBeKsuej26+nzHM45K1C1LaCsaRLdzvSEvH74oXZ3Rr4c8TGIkY2Ojk6rK74DbaTaK2qg+sXpHYVX678Zqc181rsevf7AyulUJqEul0P1ejlR+j7VcdFTFeTEpl9zi0YHZyyIoC3Y7wlmy1avjm54giTnmPObtVzU/da+cyePlaP1/+fffTHFkes7dT2ZWjogoM0ZeNJ8iXhO8f3ArjNfapTxhSJnRi2m7qN1hyJvBSZKHrQ/P302+bTop1dYv/k5CKsixPBjBwI2SLrS8yNrOlVaivThvmCd1NWWBPEi0PHhtfzG3zaeHXtQvyp7EYHijZpc1jDwtSqXDa/YWhkYrve/Ub2YFZbpVCahKRdDxVnpyK54u3imWnYokOUM21SWtXTN1Qw0cE5cP5++j1CEXCVKWvGY9DSd8gVR7rNDKN/IyqdxKaAbzNw0TOGHGsLixU/ryp504kMbF/IFwsP0i8bToqhtXFdcqlDfM5UJ2w5xd88SX7xxT4QW7npxHXn0J4kPcgXKiv1vljr/9cxZdowXyRzyVn6rqilrMRdjk3WzuuP+Hk9e9e+QHKpVT19/d3Grne/P3BRCqVJqNv1UI1ffZxGLTtC+TMkE3Q5/N9LPwk2aef6hpGVh/TJX6OtfWvo62iwlRwcnzxRfNo72GEJld2rfevlp47BsSPg2+D2PW6utUii7CLKL4bkD6Bf2pX2eNyYOnKcJtpokxjk5Bt8zlQnY5Yfoe9WHRfD+rO8nbwvuI7XSi5GbWlGPeDJVqVaBdPTT20ieVL19FdtYkZApqECiwF+C80Urp+NMeXfHTPn0DvWB3/spCX7L7lsDm7KA0Oj8qjqHdeX7bQvkL5Q+n21frve/b7ab0zjKoXSJNTteqgmrT1BI5YcplzpktIbQRnFZWxF0L8WVo7lxN8DUyWmDZ/5ppyf7HoCDxz44CCsIOG/Y1N8jknH0+0wXy4+RD+tC6eyOVLTX53KkzZJx+0AHjRgaiJ01dJYyRZlfM4ucX5Bwt+0cZceLMGULly2kgfbO7gWJU9kjDJGZkF4s0hG+qFFCVPWpgZxICDThk1qXZJqSyT6ZmAkE/GnSZqQdmq4Ss2YQ+8Yct15bZ9USRJQ2MBaeoeyrJ1WCf62eTFqUCyzZfN7M5Fd735v9uRpX6VQeoqcpp9dDxVcirAOgiy5TqEMPk+0iA5OpqHB576svoLMVWSwQmTOwCELDtDUjafE330RQ2XSMfLbMKzwl86eimZ3rkCcRIWXkPEtfaPc/LTuBH25+LBLGiumMWJAmCh68vpwEbYhK5l+Ay1iYu1l6EnRAFkJaFQiM41pVszf23ql5pdpw5gk38wNyhbQDMkT0Za+vq10FNPaZSYEbTtfeoe8wVOrBE9oWYLqBWX0ZkjL+tr17rcMAGkipVCahLpdDxVXj4FyhWo5v246ReCkRJyclfLDqmP0zfKjYkq4333lcm80YSPtOnNLzJMjbVJCxq1QIpccoklrw8V/xyZ3ilXP6JtlR+iH1cepeNaUNK9LRWILYYNimejb5sV9sozfNp2iQf8ecEljJSdxYXIuFTpty2ka8M9+sR678OlpL0NmFjACmsyC0Lx0FvqqcREj3VVbNwjIrAEz3i9LFXKlNRUzmYjflx4YPYuWw3u07f29tujW3/2v3TQ3gg4MbbSlWPXs219t7Hr3+wMPpVCahLpdDxXi4ODuDUj2GoUUCKCZ285SjQLpaXJba2O02PUOuIMyp6AFH1UyCfmow8j0QHLZQDn2zhcxVD7ZjIWDaonMtbW9fbGU+bvPC15UV5cclEYojywcYzlrx1nq/fde8Wd/kkfLeMjKYJw4Dvd9HPyHAekzd6/4bkLalM9GQxsUNtBbNXWHgPy7MOeD8lQyW2p3XQx9Lr/I+iPpUV6s9mUMlFCgiYL40jtkCDBN416z99DsCAoxfDSlXSmqnj+9N0Na1teud79lAEgT2UahfPDgAcWLF48SJkxIhw4dokWLFlGFChXE/2KD2PVQ8QWMzOqqedOJt8A3imSk8RbHaE3deJKGLDgoHqUvybI//jOM5u++IOYpnDk5LfzIUTZQjr2LTW+/Vp19banFjr/voOUHL9M7pbLQ1018Yy27//gpjQs9RuVzpaFq+QKibFUOUcAHzJ3HSij+5okl0Bd4fvb3Xvprh0MZ9DSLVmZBeK9SDhrwZkFfLPX/7ZhyjKovKgPJtE95079Oyz+t4jesZTouLAKFLO5GkK770jvkzYaZB5fH+P3dMhScN503Q1rW1653v2UA2E2hXLduHTVs2JD+/vtvKliwIOXNm5fixo1L9+7do5kzZ1LTpk39gY2hOe16qOaFnaNP/9pDyRLFp+A86WjRvovkjxitP7acpv4RrkpU7fmzo2+Im+WLuUTWlDS3i6PiCNdixn/7IobK0GGxYWOZDQDhCJx1jHKdw98OsnzFMj0QJmfuvI3Hr1HLyVspZZIEtNsmyQXM4Yl1eppFyxWtMMYHVXPRZ3XyW475qzyhHJbgC3YAmfYJFG2LujleZP0hMlcr5od36srdR2IpRQJT0L8+KqXqzV7l322M44uwBG/WF1Nfu979vtpvTOPawkJZqVIloUR+++23NGXKFBo1ahQdO3ZM/PfkyZMpLCzMH9gYmtOuhwrci4iHAzcaCJNDD10RJRiRmGKlzNp+VlTpgfgyhpNjATGPrLhyvF5s+7Gy6hlNXHuCkJiTJ+B1WtG9CnG1jbbls9EQP7hf5ecoE0XDdTdlw0nKnzEZVc5jDwuGbE31VNH9eulh+nHNCfG4u4Xkoe4181r16P9fzCMneXlSycgdSLJLvViWlPSPj0tnxrQepgDjNtnTJKFT1++Lf/rSO+QOIyNrnt25PJXObm5YgjfrUwqlPvRsoVAmSZKE9u/fTzlz5qR69epRlixZaNKkSXT69GnKnz8/wR1ud7GrQrlk30X6YPou4YorkyM1bTh+jfxRy/qfsPP0yV+7xWPk8n6+eKayJRIuE7hOIH9uO0Ofz90n/hu1qfHDqiQSAcaNY6yYrNtf7tdxoUeFOxzib6Jod+eEKZfQTqaqctdP/nzsiqP07UrHfnvVzkcfVsttpLtq6wYB2aW68fPqggHCTJHJxJkpwczxjYwln0f0g5v78KW7YojyOdPQzI7ljAxnSVutR2JelwpUPGvs+I22691vyYPTTGILhTJdunS0fv16ypEjB6VJk4amTp0q3Nxbt26lBg0a0KVLrkla/QFYdHPa9VCtOHiZ8GOHHAG8OYeduUVdq+WmnrWtLT24aO9F+nDGLgFf3cIZ6MdWJX3y+P7afoY+m+NQHOXko7m7zlH3WXvE37kutE8WEEsHhdVvaAS91Npe1YjLCXaqkpP61C1g+a7kmFd/E0W72/yoZYdp/GqHdTFTikS0qY9xyhh5v4p43x3ixj/nSlDoub1fDUqX7DXjg8TQ48Ppu0Q4EaRCrjQ0433/KW2ytRvrQegPM1/IL9mmAuDlYPJ3KLb9Rtv17vfykXjU3RYKZbNmzYQVMnXq1DRnzhy6ePEinThxgjp16iSsloijtLvY9VCtPnKF2k/dLuDLHfC64GjsXScfdalqrQWEFVusw5dUNEv3X6TOfzgUV5lDceHeC9R1hiN0YvmnwZQ3fTK7HylL18chAZxxzVmrH1bLRb1qWx/PxxyVAMHfRNHuHoRsXcyaOgmt613NXZeXPpct64p43zB8bjvIVrs9g2pRisTGiOfdTSAnA/pbaRu9/Ah9H1FNCutGqNPG447SoP5g+HCHHT6Xv0P499JPKlP+DMn1dPV7G7ve/f4AxhYK5dWrV6lz584UHh5OgwYNEgk6PXr0EBZKJOpkyJDBH9gYmtOuh2rDsWvU6petYi9wx12794iGNihEbcpnN7Q/bxuvOXKF2kUotk1KBtI3TYt6O6TL/puOX6MWkx37Re3yse84CKJlhXZtr6qULU1Sn8wfWwflpKmMKRLR5j4hzsoiH4fkoU/9EM8nswL4myja3TOVrYuoSLWyh4P71IjIMb7D3y5MLctmM9JdtXWDgByT6wnxvDuAZe5HX4b0uFsHPpfDRfBvKJGhhy6LrvWCMtCElr7xDulZW3RtZJ5itAntXkUYQGKD2PXu9wd2tlAoXW388ePHgkIotohdD9WW8OvU/KctAkbEUT5+9lwoc1DqrBRZ0fNlUtD+87fpze83iK3JlDeotYyayxAu42fl/u0+18xtZwhxZsgI3davBtX7dj0dvHiHetTMSx+F5LF8+TKBuV3JmBkUmWPVU1oW5ovFmKOaFKGmpbJYjvmrPGFMZT7N2LfMI4qKZBNb+09p0ypnqFu+YI+DSq1hsUw0zkeFCrzBEQlpcNWz2KWsqp492fXu17N2s9vYRqGEm/vnn3+mw4cP07hx42jt2rUUFBQkknJig9j1UO08fYMa/7g5CoQ/tixBdS0ua7Xj1A1qMtGxDl8mBZ29cZ8qj1wt5mldLht90dBBEL35xHX6388OxXpn/xqU5nVzY6hiwxmNaY3MV8ru5Vpj19LRy/cEfQ1obKwWORbW30TR7vbO8ado5ylp/987zxGsXJDYVMfYHTZ2+ZyVfpkxwMy1ybQ3UOBQPMFfolXO8GLNPKlNSwbSKB95h7zZr1xSFeNs+KwaBaZK4s2QlvW1691vGQDSRLZQKI8fP05ly5alFClSEB4OlMrPPvuMlixZQsuXL48V5OZ2PVR7zt6iBuM3Rjlb/iCNldfxfuUc1O8N3xA3337whIoOWS72K2coX77zkCp+tYpSJU1Imz6vTgnixfXH9822c87ZeY56zN7j5Hes/s0aCr/2H/V/owB1qJzT8nXzejCxv4mi3W1+2uZTNGD+AdGMS1e666P9XCZs98cLn9H1xrb2HELhKU+ou/3KVD2NimemMRGhNu76+eJzOR4X4+MFHiV3If7ilXW3z183nqTBEYUv0HZr3xBKnzyRu262+Nyud78/wLGFQolMbmR6w0KZPHly2rNnD2XLlo3atWtHZ86cEdZKu4tdD5XsAmYMfVF6zN3zOXjhDtX7br1o1qVqLurtI+Lm589fUK5+i+nFi5cJos9cv09JXosnYkmVREWAFRpU1dg3pDYFj1xNZ27cp0H1C1L7ijksh4v5UzGxv4mi3W1epqQCNdesTsZJ++Vksl/alqKQArGj7Jw7bOzy+fStp6nfvP2UKkkCCvMBIf6IJYdo0tpwsV1fVpfSg6dsMUd7eBiY47R9xew0qH4hPcNY2kYufIGJY5MXya53v6UPMGIyWyiUUCahNKJKTrJkyYRCiexuWCphubx9+7Y/sDE0p10P1ZFLd6n2uHVR9uKPDDpkl9cY43gx+KRGHvqkhu+Im4sMXkZ3Hj5VBNEGTjBirFBNJEnCeHRwaB2qMGIlXbj9UIQMIHTAamH+VMxbNEtKmu9Homh3e5fd1ciond7BOGXMykOX6b3fdoip/OFBcLfH2P45PCQNJ2ykSrnT0rT3ypq+HTnpx99WQNlijo2CJH/MiqNiz52Cc1KfetbTgLkDXA5xQVtfZOK7W4Onn9v17vd0P970s4VCmSpVKsFDWbhw4SgKJbK869atSzdu3PBmj5b0teuhOnH1HoWMjmrhXd+7ly8iFgAAIABJREFUGmVJbW18CqyDwaMcsY2+Jm6uPHIVnb3xwOfzWHKwLJrESYAfP66om11meKgo14aKSkiislpCD16mDr87FCx/E0W727vsrq6aLx392t5Bpm9E1h+7Sq1/cSSN/dmxHJXLmcZId9VWBwIIe0mdNKFPwl2+DT1GY0MdSpsvY8R1bJPkBC+071evAA1ffEh0/ah6bupRy1oOYj1rlkNc0P7g0NqUJGF8PV393saud78/gLGFQtm8eXOx9+nTp1PKlClp7969wvXdqFEjCggIoNmzZ/sDG0Nz2vVQyYocb8gf7oRLtx9SuRErxRJ8HZeHjMvZO84JS0T5XOpi1nOQlx24RJ2m7aT4cePQ8S/rUYkvVtCN/x77LeNYppmya3UPxnXxvovUZbqD+9RTnj+ZjUFVctJzYu3VRqaO8mWMuJ5dy2Vu0R68psMWHaKnz18IayVKe9pN5JcyrA0vtQnjx444d7ve/f54xrZQKC9cuEBVq1al69evC/c2MrtRdhFE5+vWrRPxlHYXux6qC7ceUIWvVkWBzxc8bO6ez/V7j6jksFDRbMhbhahtBd/xYL548YLuPXpKyRKZS17sbo+x+XPZ5XrqqzcoaPAyuvvwKY17pxg1LJ7Z8q3JNFP+Jop2t3mZ49RTnr9dZ25SowmbxFSqkpM7xO33uZwI48sYcT0711r7ELaC0ob3Hz+jz+vmp85VrGdtcLdu+aUMbcO/rEdx48Zx180Wn9v17vcHOLZQKLHx+/fvi4o4YWFh9Pz5c+H+btWqlbBUeiNPnjyhwYMH07Rp0+jmzZtUrFgx+vrrr52Z47t376aPP/6YduzYIco+duvWjXr27Gl4Srseqit3H1KZ4Q7LICQeLFDD61Ic1GK0UO48fEJFBjuyr798O4halLXejWrhdmPdVHJFJfyYFx68TFxAoD8BDYrVsu3kDWo2yUEzFZI/gH5pV9rqJeieT8buraKZ6DsPKGPk5LllnwRTvgyqkpPuB2CDhjIRPyyAsAT6S7TWvq8aBdGIJYcJDBgD3iwo2C/sJuwhwbqgR4aPeMNuS4x2PXa9+/0BoG0USmz+4cOHdPToUXrttddEUk6CBN5bmFB556effqLffvtNjAllctasWXTo0CExD6yhqMzTvXt32rJlC3Xp0oUmTJhA7du3N/Q87Hqobv73mIp/scK5F3/VRX745BnlH7BUrMMfxOqGHub/w8ZyDB/cTYUHLRMk+BNblaA6hTNajkjYmZv0doTFzt9E0e42v/H4NWoZUZ2pcYlAGt3MeBWoY5fvUs2xjuS51T2rUo60qpKTO9zt9LmcpdyzVl7qWt1/bmW5zCwwAlH+qGVHREy0v5Ls3D2rVYcv07u/OmKm4erGb1BsEbve/f7AzxYK5bNnz4RVcOLEiYQKOZDEiRMLLsr+/ft7ZU2DRTIkJIRGjx4txr1z547gu0RJRyiv48ePp1OnTlH8+I4A4L59+9LcuXNFhrkRseuhki2D2A+X1jOyNzPaPgOdT9/FYihF3GwGouaOIbuYERJRcOBSev6C6Oc2pahmQespbGSL3ZtFMtIPLUqYu2ETR9safp3eiahG1bx0FvqqcRHDoz999lxUtIIHYeb75WKNu8/wRl/RDnKWsr/dyjIFFeAe+05RUaQATA7TO5S1ZdnZdUevUpuISma+4gr11dGz693vq/3GNK4tFMp+/frRlClT6IsvvhA0QVAwkfUNV3XXrl1pyJAhHmMDZRIPfNmyZZQlSxYaO3asUFIPHDhAH330kUgCmjFjhnP80NBQqlmzJl2+fFkkBOkVux6qB4+fUYGBDssgxNNaw3pxiKldnn6L6cmzFzSxVUmqU9j+9dnN2HNsGUNWivYOruUMT5javjRVy6f/e2DWfmW6K38TRbvbkxz/KFdnctdPff7qIDB31znqPstR6cjfbmU5phfrQQgGQjHsLJtOXKMWP28VS0yROIGgDYotYte73x/42UKhTJ8+PU2dOpXq1asXBYM5c+YIhRJlGT2V/fv30zvvvEMHDx6kePHiUdy4cYV18q233qIiRYoIWiK4wVnQrlChQrR9+3YqVaqU7mnteqiePHtOefotce6jaGAKmt+1ku59mdnwqyWHaevJ64JWBT8aSuyDgFwac3u/GlR6uCOB6o/3ylKlPGktX6hMd9WsVCCNbGLcjWzVovedu031f3DUj7crcbRVWPx/nUcm4h/aoBC1Ke+7pEN3GK8+fIXa/7rd2WxCyxJUz+JSu+7WqP18+6kb1DSiNG/a1xPSjv41jQ7ht/Z2vfv9AYgtFEpYCTdv3kwFCkQlXIVyB4vl3bt3PcYGlEPffvst9erVizJnziyq8fz1118iexy0RC1atKChQ4c6xw8PD6dcuXIJC2mlStErXnCd438sUHrLlClDZ8+epcDAQI/Xa3ZHZDzn6ONwNUPsTsFi9v7VePoQkGMWUUe30tcOzlC4X/1BvSTXZPc3UbQ7BA9fukN1xjmqQNmVONrdHtTn3iEgu5n9xd3KO5Ddx/jbpNYlqXYhe3uE5N+fDMkT0Za+Id49EAt7K4UyEmxbKJTIsr5y5YpInEmYMKFzde+++66IbURSjScC6qE8efLQypUrqXLlys4h8N9p06Yl1BCHVdSVhXLXrl1UvHjxaKeFO96VK95uCiU2kLvvYsFBBvGUJ88T/FWf2IPA3nO36K0fHDXfV/ao4iTD/7tzeSqVPbXlG7l4+wGVH+Ggu/I3UbS7zctVoD6slot61c7vrov6/BVDQCbiRxJM01JZ/LZDOR4ai4gNpTzlmOnAVIlpw2fV/Yaf0YmVQmkDhbJ69cgD8/TpU9qwYQNlzJiRSpcuLVzToA+CcoY633BReyKwTjZr1owePHhAiRJFFpqHtXLx4sWC3xJUQaAUYtEbQxlbLJTYV/4BS+jhk+diiw2KZaJvm0evKHuCs+oT+xE4cOE2vfGdw227uFtlZ931eV0qUPGsqSzf4LV7j6hUBG+pv4mi3W1eLh7g67Ki7taiPvcPAmuPXqW2EUkl/k46lOOhgcav7UtTVT/EQRt5ErKVHwwHYDqILaIUShsolEZoeRBf6YmABqh8+fKEEo5wR7PUqVNHKJjlypUTmeUnTpwQSiykT58+NG/evFcmyxt7Chq0jO4+eir2B/5H8EAqUQjICMg/6FAimbLHXyTbt+8/oaJDHbylH1TNRZ/Vsa/VTy4e4OuyourU2hMB2So4vkUJeqOI9VRbjMzO0zeo8Y8ODleIv+KgjTyp41fuUo0xDtqs3AGvU2j3Kka6+7WtUihtoFBacQJAkI4KPFevXhXckoht/P3332nEiBHCIgpeSvBQIkGnd+/etG3bNvrggw+Ektm2bVtDS7TzoSo2dDnduv9E7Mfu1h5DoKvGpiEg/6CjljQobCBLP6lM+TN4V1zAk0Xef/yUCg5cJrr6myja3fqv3n3kTGLqWy8/dQy2XyUSd3tQn3uHgJxU8lPrklTLjzGLu8/eoobjHeErEH/FQRtB9NS1/6jqN2tEl/wZktHST4KNdPdrWzvf/VYDY4sYSmwaDwWckKjjDUJzZFp36tSJsmb1rqIKquOAJmjhwoWiUk5QUBB9+eWXVKWK4w0I2dyojgMXO1zuPXr0EJnlRsXOhwquQ7gQIR+H5KFP/VjFwSiuqr01CIRfvUfVR68Vk/32bhmn+y60ezDlDrC+asvjp88pb38HO0GPmnnpIxvWH+Ync+v+Yyr2f+3dB3hUxdoH8D8JHYTQRUBKAtKSSIeAKGICREWaND+k2ICLoDQFpIUOV5DP0K4I3gsCFzT0KyoKGIL5AAOCcENvAoICAaW373kn7pqEQPZkZ3dns//zPD4COWfOnN95d+fNnDMzUcmLB4x4rip6GLgSiXuiyHfPknJQyfxuddCksvun2rLpp3wfUf5tWc8GqOOB96CtRMPPF67YBwIGlyqI1W96ZiYSK3W27Wty25+Z63HmGCMSSpnap3HjxsibN696NC3zUMpSiJcvX0ZcXJxKLk3fTA6qBhO+wemL1xThsMgqeK1xBdM5WT83C6R8D1B6WF5f8IOqwcaBT6GcB1ZtSTk7gacnis7oVly+fgvVRib3ppq6EklG18CfOyeQMonz9CPmlK+vyFXF9A5DTQ+8B21F9Myla6g3PnmJ4BqPBmB574ZWDvfovia3/e6GMSKhlLkg8+XLh08//VQthyibLMMoa3nL/6V30fTN5KB6YvK3OHH+qiIc17o6XqpX1nRO1s/NAil7CGT97jcX71A1iB3cBGUK53VzbZJPZ5sI39MTRWd08Sl7Uye1DUaHOs49VcnofPy5eQL7z/yOiD+Xzvz36/VRr0IRj1Uy5esrUolVfRoipHSAx+rjyInP/XEdtf4chFe3XGEs7dnAkcOM2Mfktt/dQEYklPnz51fraFevXj3V9cvjb+m5TEpKcreL5fOZHFRP/30jDv92WV2Tp0cgWoblAW4R+OXiNdSfkNxDIGutD1yWvOpH/JCmeLjgXzMkuKUyf56kyvB1uHrzNjw9UXRG15yyN/X9F0PRtpY589BmVHf+XI9AyldGPu8Vhlpl3T8zgu1Kjvx2GU3+fB9R/k1mbaj6iPvfg7Yie/HqTYSOTh6EFxZYBIteq2/lcI/ua3Lb724YIxJKeXdReiFr1aqV6vrl/UZZBpEJpXNhETFtk1rLVTZPrc3s3BXwaFcLnP39GuqOS04ox7aqjvdW/KT+vP29Z1A0f/JTA3dvIaO+xKVrt+DpiaIduW7bXK/esMydI9fDfawJpJyIf3WfRgguXdBaARr3TlkXKfartxujUgn3vwdt5ZJSDsJrXKkY/tXjr1lZrJTjiX2ZUP6lbkRCKY+2T506peabLFw4eRLl3377Tc0hKavoxMTEeCJOLJ3T5KCKnB6LvaeTV/VZ9Go9hAW5fyk9S5jc2e0C5y/fQM0xyQNL5BHzmDV71Z93jghHQN6/FhtwZ8Vsvwh5epCDI9fccOK3OJl0FUvfaIC65d0/EbwjdeQ+rhP4/dpN9chWlrr9/l3P9erLFaacxkr+LgsVBBbL77qL11ByytdGmlYujo+71dFQqnuKMLntd4+AYQml3JCwsDCcP38elSpVQrZs2bBv3z4UKlQImzZtUtP7mL6ZHFQtozdj188XFeHKvzVEaBmz36cx/V5nxfqlnPdxcPPHMHndPnWZP41uhvy5snvkkmUFmv+evqTWIfb3y+aROjh60n2//A6pb2Tww+r7i5vvCcjUQVdv3Ib0sHlyO3vpGur+OcBF6rFp0FMoWySfJ6uU4bnv3LmLCkOTlwiOqFoC/3i5dobHmLKDyW2/u42M6KGUd5BkRLesWCMjvuXvMr2PrLNdsKDnHh1YuRkmB1WbmXFIOJ78HqqnpoGxYsl93S/wx/VbqP7nSGWZWmr6NwdUJRLHNEfuHMmT/nOjAAXMF0g5wEVqu/mdJihdyDMD66xoVRiyFrJC8LPBJTHjpZpWDvXovia3/e6GMSKhrF27NubNm4eQkBB3X7+285kcVO3nfI+tR86ra/1+yNMoWTCPtutmQVlDQHpWqoxYpy6m55OBmL3pkPrzwXEtkN3fL2tcJK+CAj4gkPJpg1yuJwfWWeGWeWfl0be3LQ9scttvxV/HvkYklEWLFlXLIwYGeu8KEyYH1Utz4xF38JyKlx9HRqBgnhw6YodlZCGBlO8wdW9YDvPjjqqrOzIhko9ws9B95qVkfYGUTxvkarcNewbFHvLMwDor2tVGrMPlG7fRtmZpvN8+1MqhHt3X5Lbf3TBGJJRTpkzB2rVrMWjQIAQFBSFPntQ9aM6uluMOVJODquu8rdi0/1f2OLkjELz0HLfv3EXgn+8wyXrvi/7vOLL7ZcPB8ZFeekWsNgV8U+DazduoPDz5aYNsO4aHo1A+zwyss3IHZNogmT6oY50ymNjWe55Wmtz2W/HXsa8RCaWf31+P1FK+0C7vUsrfZeUc0zeTg+rVf27D+v+eRa7sftg3toXplKyfBwRSzqUoPQSfJ/yM3Dn8kDiG8eKB28FTUiDTAjLSvOKw5GVLZds1KgIFcpv/VKr22K/x2x838D/1H8XYVsGZvn53H2hy2+9uCyMSShnJ/aDNtu62u3GsnM/koOq54Aes2/MLCufLiYTh4VYui/v6kID0UEpP5XMhJbFm12k1ultGeXOjAAW8RyDliGmp9d6oZsib0zMzNVhRqz/+G/xy6Rq6hZXDqJbmL7dsuzaT234r/jr2NSKhTHkhMnWQv7+/14zu9oag6rMoQSUIZQrnQezgp3XEDcvIggK2l+LDq5bA13vPICBvDuwcEZEFr5SXRIGsLWAbMS1XuW9sc+TKbv5MDY0mfYufL1zFa0+Ux7Bnq3rNDWJC+detMiahlPcop0+fjtOnT6valS9fHu+88w5ee+01rwgsk4PqrSU7sGLnKVR++CGse6uxV3iyku4XsC11+ETFoog98JtaIUdWyuFGAQp4l0ClYV/gxu07qtKHxkcaP4+r1POpKRtw9NwVNcvEuy0qew24yW2/uxGNSCgnTZqEqKgo9O3bFw0aNMCdO3cQFxeHmTNnYtq0aXj99dfd7WL5fCYHlazL/NkPP6v1ZWWdWW4USE8geOSX+P36LbXSi0wz9XCB3Igf2pRYFKCAlwlUHbEOV24kjz3wlpkanpm6SS0O8ObTQRgQ8ZjXiJvc9rsb0YiEskyZMhg/fjy6dOmS6vplbsoJEybgwIHkSZZN3kwOqiExu7F463FIz9OCV+qZzMi6eVDANspSVlL68UQSShfKg83v8BUJD94SnpoCmRIIHvUlfr92y6tmamgxPVatjPX2M5XQ75mKmbpuTxxkctvvbg8jEsr8+fMjISFBLbuYcpNEMjQ0FFeuXHG3i+XzmRxUGxLPYsCyH/Fu88poX6eM5WvjAb4hUGvM1zh3+YZ6NSLxl99RrkhebBzUxDcunldJgSwkUCPqK1y4ctOrZmqYufEg5m0+io+71vaq5YFNbvvdHdJGJJRdu3bFQw89hOjo6FTXP3r0aLUU47Jly9ztYvl8pgeVbQomyxfGA3xGoM649fj19+uoUDQfDv92GUHF82N9/yd95vp5oRTIKgK1x67Hb39c97qZGryxnTK97XdnTBuRUPbv31+9L1mlShXIFEE5cuTAtm3bEBsbixdeeAEBAQF2E3kMbuLGoDLxrrBOVgQaTPgGpy9eQ6mAPDiZdJWDuKzgcV8KGCRgm4JHVkWT1dG4uU6Abf9ftkYklE2aOP5YbcOGDa6LDCdKZlA5gcdDjRBoOPFblUgWzZ9TTTBcvVQBrHnzCSPqxkpQgAKOC9g+y0Xy5cQPnHvYcbhM7Mm237CEMhP30LhDGFTG3RJWyKLAk1M24Ni5K3goV3Y12vvxMgFY8beGFkvh7hSggKcFbJ/l4g/lwtZhnPrLlfeDbT8TSu3xxaDSTsoC3Szw9N83qncnc/hnw83bd1G7bCF8xmmm3HwXeDoKOC/w9PsbcfjXy3ikYG5sGcKpv5wXvX8JbPuZUGqPLwaVdlIW6GaB8KmbcODsH/azPlmpGP7Zo66ba8HTUYACzgpETNuE/Wf+4OpozkI6cDzbfiaUDoSJtV0YVNa8uLd5As0/+E5NF2TbZE3v6M41zasoa0QBCjxQwDanY/mi+bBh4FPUcqEA234DEspVq1YhIiICuXPnduGtdl/RDCr3WfNMrhF49n9jsefUJXvhneqWwYQ2Ia45GUulAAVcJvD8h5ux++RFTv3lMuG/Cmbbb0BCKZOZJyYmonTp0qhQoYKaJqhIkSJuuP2uOQWDyjWuLNV9Ai2jN2PXzxftJ3y9cQUMjazivgrwTBSggBaBVjPisPNEEqf+0qL54ELY9huQUBYvXhxvvPEGnnnmGci0QcuXL0ehQoXSvXONGzd2Q1g4dwoGlXN+PNrzAq1nxmHH8SR7RfqHV0Lfpt6zBJrnBVkDCpgh0G7WFmw/dgHVHimAtX059Zcr7wrbfgMSykmTJmHo0KGwzYwv/09vy5YtG27fTl7k3uSNQWXy3WHdHBGwNUK2fUc+XxXdG5Z35FDuQwEKGCTQYc73+L8j5xFauiBW9mlkUM2yXlXY9huQUEoVLl68iKSkJJQvXx5bt25FsWLF0o22smXLOhWF//rXvzBx4kQcPnwYgYGBGDVqFF588UVV5s6dO9GvXz9s375dPXLv27cvBg4caPl8DCrLZDzAMAFbI2Sr1t9fDEW7WqUNqyWrQwEKZCTw0tx4xB08h5qPBiCmN+eSzcjLmZ+z7TckobRVY9OmTWjYsCGyZ8/uzH1N99iFCxeiW7dueP/99/Hss89i8eLFKqHcvHkzKlWqhMqVK6NVq1aQ5R/j4+PRu3dvtQxk9+7dLdWFQWWJizsbKND5o3hsOXTOXrPZ/1MLzas/bGBNWSUKUOBBAl3nbcWm/b+ibrnCWNqzAbFcKMC237CEUqqzZs0ajBs3Drt27VJreVetWhWDBg1C69atMx0K8hhdBvy0a9cOU6ZMsZfTrFkzPPVU8lQKM2bMwNGjR+3JrDyGj4mJUQOGrGwMKita3NdEgS4f/x9iD/xmr9qiV+shLKioiVVlnShAgQcIvPLJNnyTeBZhgUWw6LX6tHKhANt+wxLKlStXok2bNip5lAE48s7kd999h9WrV6vkrmXLlpkKB0kKq1SpgoSEBNSoUeOeMiIjIxEQEIBFixbZf7Z+/XqEh4fjzJkzkIFDjm4MKkeluJ+pAt3mb8XGfb/aq7eqT0OElA4wtbqsFwUocB+B1/+1HV/tPYMnKhbFglfq0cmFAmz7DUsoa9asqZLJ4cOHp7rtUVFRqudS3q/MzCZzXb7wwguIjY3FmDFjsGPHDvW+5nvvvYfnn38eISEhaNGiBWSAkG3bu3cvqlWrpqYxql279n1Pe+nSJch/tu306dOoW7cuTpw4oaZC4kYBbxN49Z/bsP6/Z+3VlgmRZWJkbhSggHcJ9P70B/xn9y9o8lgxzO/O1a5cefeYUBqWUObJkwe7d+9GUFBQqvt+4MABhIaG4sqVK5mKB3l/skuXLiqJHDlypCrr888/V4/Wv/76azVtUefOnSGJq22zDdyRJLRRo/uPjpP3MEePHn1PvZhQZupW8SADBGy9GraqbH/vGRTNn8uAmrEKFKCAFYE3F+/A6h9P4ZkqJTC36/07RqyUyX3TF2BCaVhCKSOvP/zwQ8gj6JTb2rVr8dprr+HUqVOZiuWlS5eiQ4cOapBNr1697GXI4BzZjh8/rs6ZXg/l/R6T2wphD2WmbgkPMljA1qthq2LimObIncPf4BqzahSgQHoCb/97J5bvOInm1R7G7C61iORCASaUhiWU8qh7wYIFmDVrlhrtLZuMwpYR1/LIevr06ZkKhy1btqjy0vY2Dh48WD1KL1eunJoqSM5t2/gOZaaoeVAWEOizKAFrdp1WV5LT3w/7x7XIAlfFS6CA7wnM2ngIk9Ylou/TQegf8ZjvAbjxiplQGpZQXrt2DR07doS88ygTmcsmI7Sl91B6GfPmzZup8Lh69SpKlCiBESNGpJpbUt6blJHkYWFhmD17Ng4dOgR//+SemCFDhqhVezjKO1PkPMiLBd5asgMrdiY/DSicLycShod78dWw6hTwXQFpPw+c/QNBxfLDzy+5TeXmGgEmlIYllLbqSBIn71LKhyE4OFiN0HZ2Gzt2LCZPnow5c+aoQTNLlixRCeY333yjpiaSeShlFLn0WsrgH3k0Lklm165dLZ2aQWWJizsbKNB/6U7EJJxUNStbJC82DWpiYC1ZJQpQgALmCLDtNzShdFWITJ06FdHR0Th58qRKUmUwjTxKl01Gc8vqODICvGTJkhgwYAD69OljuSoMKstkPMAwgcGf/Yil239WtapeqgDWvMk1gA27RawOBShgmADbfh9LKN0RfwwqdyjzHK4UGBKzC4u3nlCnqF+hMJa8zhU2XOnNsilAAe8XYNvPhFJ7FDOotJOyQDcLvLdiNxbGH1dnDa9aAh+9zOlG3HwLeDoKUMDLBNj2M6HUHrIMKu2kLNDNAiNX/oR/fn9MnbVNjVKY2uFxN9eAp6MABSjgXQJs+5lQao9YBpV2UhboZoGo1XsxL+6IOmvXBmUx+oXqbq4BT0cBClDAuwTY9huWUB45cgTvvPMOfvrpJ8gUQmk3Wb3G9I1BZfodYv0yEhi3di8+ik1OKPs0CcLAZpy/LiMz/pwCFPBtAbb9hiWUssShrIXdvn175M6d+57olGUTTd8YVKbfIdYvI4GJXyRi9qZDard3W1RGzycDMzqEP6cABSjg0wJs+w1LKPPly6dWs6lZs6bXBiaDymtvHSv+p8CULxMxY0NyQjmudXW8VK8sbShAAQpQ4AECbPsNSyglkfzggw/QuHFjrw1cBpXX3jpW/E+BqV/vx/9+c0D9bXrHx/HC46VoQwEKUIACTCgdioFsd2VZGg9v8fHxaoUamWC8fPny8PPzS1Ujb0g0mVB6OIh4eqcFpq8/gGnr96ty5nergyaViztdJgugAAUokJUF2PYb1kP58ccfq4Ty1q1b98SdrO19+/Zt4+ORQWX8LWIFMxCYseEgpny5T+21rGcD1ClXmGYUoAAFKMAeSodiwIgeyjJlyuC5555TPZR58+a9p+Jly5r/LhcTSofijTsZLDBr4yFMWpeoarjurSdQ+eECBteWVaMABSjgeQG2/Yb1UMqgnD179qBcuXKej45M1oBBlUk4HmaMwEffHca4//xX1Sfu3adRKiCPMXVjRShAAQqYKMC237CEsl27doiMjESPHj1MjBeH6sSgcoiJOxks8PHmIxizZq+q4Y8jI1AwTw6Da8uqUYACFPC8ANt+wxLK6OhoDBkyBC1atMBjjz2GHDlSN2QjRozwfNRkUAMGlfG3iBXMQOCfW45i5Ko9aq9D4yPh75eNZhSgAAUo8ADEgPwNAAAavklEQVQBtv2GJZQysvt+mwzK4Uo5/DxTwPUCC+KPYfiKn5Avpz/2RDV3/Ql5BgpQgAJeLsCE0rCE0svjSVWfQZUV7qJvX8O6n06j58IEVCqRH1+9/aRvY/DqKUABCjggwLafCaUDYWJtFwaVNS/ubZ7Azdt38NkPP+PxMgGoUpIjvM27Q6wRBShgmgDbfsMSSpnIXB5t32/jPJSmfYRYHwpQgAIUoAAFmFAallB+8sknqRLKmzdv4sCBA5B/nzp1Kl566SXjo5ZBZfwtYgUpQAEKUIACWgXY9huWUN7v7i5atAgLFizAF198oTUAXFEYg8oVqiyTAhSgAAUoYK4A234vSSiPHTuGqlWr4vLly+ZG0581Y1AZf4tYQQpQgAIUoIBWAbb9XpJQTpw4ETNmzMCJEye0BoArCmNQuUKVZVKAAhSgAAXMFWDbb1hCKfNQphyUc/fuXVy6dAlJSUkYN24c3n33XXOjiT2Uxt8bVpACFKAABSjgCgEmlIYllKNGjbpnlHfOnDnRsGFDNG7c2BUxoL1MBpV2UhZIAQpQgAIUMFqAbb9hCaXR0eJg5RhUDkJxNwpQgAIUoEAWEWDbb2BCGRcXh82bN+PGjRuQR94pN67lnUU+ebwMClCAAhSgQBYSYEJpWEIpg2+GDh2KQoUKoUCB1Ct0cC3vLPTJ46VQgAIUoAAFspAAE0rDEsrSpUujR48eiIqKcmmY7d+/HzVr1kR0dDS6deumzrVz507069cP27dvR5EiRdC3b18MHDjQcj0YVJbJeAAFKEABClDAqwXY9huWUObNmxe7du1CUFCQywJLVt8JCwtTieP8+fNVQnnu3DlUrlwZrVq1Qv/+/REfH4/evXtj5syZ6N69u6W6MKgscXFnClCAAhSggNcLsO03LKGMiIhQCVynTp1cFlzySF0Sxg0bNtgTygkTJqh5Lo8ePYrs2bOrc8t+MTExSExMtFQXBpUlLu5MAQpQgAIU8HoBtv2GJZRz587F4MGDVVIpPYa5cuVKFWQvv/yyU0H33XffoXXr1urx9qOPPmpPKCMjIxEQEABZ4tG2rV+/HuHh4Thz5gyKFy/u8HkZVA5TcUcKUIACFKBAlhBg229YQunn53ffwJJBObdv38504Mnk6KGhoZDeyM6dO6v5Lm2PvENCQtCiRQtMmjTJXv7evXtRrVo1bNu2DbVr13b4vAwqh6m4IwUoQAEKUCBLCLDtNyyhdGVUyWN0mYZoyZIl6jQpE0p5Z1OSzJSDgQ4fPozAwEDExsaiUaNG962arOQj/9m206dPo27dumqZSBlkxI0CFKAABShAgawtwITSRxLKBQsWYMiQIdi9e7eakihtQhkcHAx57J1eD2VCQgJq1Khx30+CrO4zevToe37OhDJrf3nw6ihAAQpQgAI2ASaUPpJQNmnSRE2WnvKdzMuXL6u/Sy9k2bJl1VRBknjaNkffoWQPJb9QKEABClCAAr4twITSRxLKkydP4urVq6mivWLFiqpHskOHDli8eDFmz56NQ4cOwd/fX+0nPZrLly/nKG/f/o7g1VOAAhSgAAUyFGBC6SMJZXqRkPIdyrNnz6pR5S1btlSjzLdu3YpevXqpJLNr164ZBlLKHRhUlri4MwUoQAEKUMDrBdj2M6G0r5Qjo7lldZwdO3agZMmSGDBgAPr06WM5yBlUlsl4AAUoQAEKUMCrBdj2+3BC6arIZVC5SpblUoACFKAABcwUYNvPhFJ7ZDKotJOyQApQgAIUoIDRAmz7mVBqD1AGlXZSFkgBClCAAhQwWoBtPxNK7QHKoNJOygIpQAEKUIACRguw7WdCqT1AGVTaSVkgBShAAQpQwGgBtv1MKLUHKINKOykLpAAFKEABChgtwLafCaX2AGVQaSdlgRSgAAUoQAGjBdj2M6HUHqAMKu2kLJACFKAABShgtADbfiaU2gOUQaWdlAVSgAIUoAAFjBZg28+EUnuAMqi0k7JAClCAAhSggNECbPuZUGoPUAaVdlIWSAEKUIACFDBagG0/E0rtAcqg0k7KAilAAQpQgAJGC7DtZ0KpPUAZVNpJWSAFKEABClDAaAG2/UwotQcog0o7KQukAAUoQAEKGC3Atp8JpfYAZVBpJ2WBFKAABShAAaMF2PYzodQeoAwq7aQskAIUoAAFKGC0ANt+JpTaA5RBpZ2UBVKAAhSgAAWMFmDbz4RSe4AyqLSTskAKUIACFKCA0QJs+5lQag9QBpV2UhZIAQpQgAIUMFqAbT8TSu0ByqDSTsoCKUABClCAAkYLsO1nQqk9QBlU2klZIAUoQAEKUMBoAbb9TCi1ByiDSjspC6QABShAAQoYLcC2nwml9gBlUGknZYEUoAAFKEABowXY9jOh1B6gDCrtpCyQAhSgAAUoYLQA234mlNoDlEGlnZQFUoACFKAABYwWYNvPhFJ7gDKotJOyQApQgAIUoIDRAmz7mVBqD1AGlXZSFkgBClCAAhQwWoBtPxNK7QHKoNJOygIpQAEKUIACRguw7fehhPL8+fMYOnQo1qxZg0uXLiEkJAQTJ05Eo0aNlMLOnTvRr18/bN++HUWKFEHfvn0xcOBAywHMoLJMxgMoQAEKUIACXi3Att+HEsqIiAicPXsW0dHRKFasGGbMmIG5c+ciISFB/b1y5cpo1aoV+vfvj/j4ePTu3RszZ85E9+7dLQU5g8oSF3emAAUoQAEKeL0A234fSSgPHjyIihUrIi4uDmFhYeqq7969i0qVKqFTp07IkyePSjCPHj2K7Nmzq59Lb2ZMTAwSExMtBTqDyhIXd6YABShAAQp4vQDbfh9JKJOSkrBlyxY0bdoUuXLlsl91UFAQmjdvjsOHDyMgIACLFi2y/2z9+vUIDw/HmTNnULx4cYeDnUHlMBV3pAAFKEABCmQJAbb9PpJQphety5YtQ/v27bFq1SoMGzYMLVq0wKRJk+y77t27F9WqVcO2bdtQu3bt+wa8vI8p/9m206dPo27dujhx4gRKly6dJT4ovAgKUIACFKAABe4vwITSRxNKefQtPZPSY7lixQpIT2Xnzp0RFRVlF5Fey8DAQMTGxtoH7qQXSqNGjcLo0aPv+RETSn71UIACFKAABXxDgAmlDyaUK1euVMlj/fr11YhveX8yODgYkZGR6fZQyqCdGjVqsIfSN74TeJUUoAAFKEABywJMKH0soZQR3jI1UJs2bbBw4UL7+5SSTMpUQQsWLLCL8B1Ky58nHkABClCAAhTwSQEmlD6UUM6aNUtNBSTzS06bNg1+fn72q5f5KGfPno1Dhw7B399f/fuQIUOwfPlyjvL2ya8GXjQFKEABClDAcQEmlD6SUO7fvx/Vq1fHc889p+aWTLnJI+/r16+reShbtmyJwYMHY+vWrejVq5dKMrt27ep4RAFgUFni4s4UoAAFKEABrxdg2+8jCeX48ePVSO70NkkYP/nkEzWaW3ovd+zYgZIlS2LAgAHo06eP5SBnUFkm4wEUoAAFKEABrxZg2+8jCaU7o5RB5U5tnosCFKAABSjgeQG2/UwotUchg0o7KQukAAUoQAEKGC3Atp8JpfYAZVBpJ2WBFKAABShAAaMF2PYzodQeoAwq7aQskAIUoAAFKGC0ANt+JpTaA5RBpZ2UBVKAAhSgAAWMFmDbz4RSe4AyqLSTskAKUIACFKCA0QJs+5lQag9QBpV2UhZIAQpQgAIUMFqAbT8TSu0ByqDSTsoCKUABClCAAkYLsO1nQqk9QBlU2klZIAUoQAEKUMBoAbb9TCi1ByiDSjspC6QABShAAQoYLcC2nwml9gBlUGknZYEUoAAFKEABowXY9jOh1B6gDCrtpCyQAhSgAAUoYLQA234mlNoDlEGlnZQFUoACFKAABYwWYNvPhFJ7gDKotJOyQApQgAIUoIDRAmz7mVBqD1AGlXZSFkgBClCAAhQwWoBtPxNK7QHKoNJOygIpQAEKUIACRguw7WdCqT1AGVTaSVkgBShAAQpQwGgBtv1MKLUHKINKOykLpAAFKEABChgtwLafCaX2AGVQaSdlgRSgAAUoQAGjBdj2M6HUHqAMKu2kLJACFKAABShgtADbfiaU2gOUQaWdlAVSgAIUoAAFjBZg28+EUnuAMqi0k7JAClCAAhSggNECbPuZUGoPUAaVdlIWSAEKUIACFDBagG0/E0rtAcqg0k7KAilAAQpQgAJGC7DtZ0KpPUAZVNpJWSAFKEABClDAaAG2/UwotQcog0o7KQukAAUoQAEKGC3Atp8JpfYAZVBpJ2WBFKAABShAAaMF2PYzoUwVoHfu3MHo0aMxd+5cXLhwAY0aNcLMmTMRFBTkcCAzqBym4o4UoAAFKECBLCHAtp8JZapAlmRSEsj58+ejVKlSGDx4MA4fPow9e/YgZ86cDgU9g8ohJu5EAQpQgAIUyDICbPuZUNoFbty4gaJFi2Ly5Mno2bOn+vekpCQ88sgjmDdvHjp27OhQ4DOoHGLiThSgAAUoQIEsI8C2nwmlXWDr1q2oV68e9u3bh0qVKtn/XR57h4SEqJ5LRzYGlSNK3IcCFKAABSiQdQTY9jOhtAvExMSgbdu2uHLlCvLkyWP/9/bt26t/W7NmjUORz6ByiIk7UYACFKAABbKMANt+JpR2gYULF6JLly64ffs2/Pz87P/+8ssv49SpU1i/fn26gX/p0iXIf7btxIkTCAsLg/R4lixZMst8WHghFKAABShAAQqkL3D69GnUrVsXR44cQbly5XyaKdvdu3fv+rLA559/jnbt2qXbQ3n9+nWsXLkyXZ5Ro0apkeHcKEABClCAAhTwbQHpTKpTp45PI/h8Qml7h/LgwYMIDAy0B4O8QxkaGooZM2Y41EN57do1SC9l+fLlkT17dq1BZfsNiL2fWlkfWBjN3WdtOxPN3W8uZ6S7+91p7n5zV8X6rVu38OuvvyI4OBi5c+f2zIUZclafTyilF7J48eKYOnUqXnnlFXVbMjPK25X3k+9ouFI3/bJpTnP3C3jmjIx197vT3P3mcka6u9bd5xNK4R02bBjmzJmjpgmSdyAGDRqEo0ePYvfu3Q7PQ+nK28QPgSt1mVC6X5fmppizkfXMneB3Ot09I+DaszKhBNSAnKFDh6qJza9evYrGjRurR92mvGDLLx/XfgjSK53mNHe/gGfOyFh3vzvN3W/OX55cb86E0vXGTp9BRpPLI/n+/fujQIECTpfHAjIWoHnGRrr3oLluUcfKo7tjTjr3orlOTcfLorvjVpnZkwllZtR4DAUoQAEKUIACFKCAXYAJJYOBAhSgAAUoQAEKUMApASaUTvHxYApQgAIUoAAFKEABJpSMAQpQgAIUoAAFKEABpwSYUDrF59qD79y5o1bjmTt3Li5cuACZbH3mzJkICgpy7Yl9qPRjx46lO5r/o48+wquvvoqdO3eiX79+2L59O4oUKYK+ffti4MCBPiSk91LHjh2rljPduHGjveCMjPk5cP4epOfevXt3fPLJJ6kKL1WqlJqrTza6W3c/f/68mjFkzZo1amnekJAQTJw4UX13y8ZYt26a0REZmTPOMxLU93MmlPostZckyaQkkDKdkXzRDx48GIcPH8aePXuMmB9T+wV7oMDVq1ejffv2yjVbtmz2GhQsWFAtx1m5cmW0atVKjbCPj49H79691T2RLylu1gQ++OAD5SjTctkSynPnzmVozM+BNee0e6fnLvvUqlULzZs3x5tvvmk/xN/fH8WKFVN/p7t194iICJw9exbR0dHKUaafkw6BhIQE9feMvk9ortdcvBnn1k0zewQTyszKufi4GzduoGjRopg8eTJ69uypzmbaCj4uJnBL8ePHj8fSpUtVz0HabcKECapBkEnubctpSu9DTEwMEhMT3VK/rHCSkydPqt7e2NhYlClTBiVKlLAnlBkZ83OQ+Qh4kLvMvZs/f34sXrxY/cKUdqO7dXdZvrdixYqIi4tDWFiYKuDu3buoVKkSOnXqhDx58jzw+4Tm+s1HjhzJOLfOmukjmFBmms61B9rWGN+3b5/6QrJt8uhEHqNILxk35wU6duwIPz8/LFq06J7CIiMjERAQkOpn8rg2PDwcZ86cUUt2cstYQHqBJXGRx65RUVEqQbf1UGZkLPvWq1cP/Bxk7Jx2jwe5yy9EVapUucfVVga/f6x7yy/8W7ZsQdOmTZErVy57AfKKkvQEy1OQB32fMNb1m/fp04dxbp0100cwocw0nWsPlF6wtm3bqseu8putbZPHs/Jv8o4ON+cFqlWrhpIlS0LWdN+/f7/qYRg+fDiaNWumEvcWLVpg0qRJ9hPt3bsXcsy2bdtQu3Zt5yvgYyV069YtVUKZkfHx48f5OdAQI2ndly1bhg4dOqinH1988YX6pUqSe0n65XUPfv9oQAcgzvKdvWrVKrXE74O+Txjr+s2vXbvGONfD6lApTCgdYnL/TgsXLkSXLl3UspDyZW/bXn75ZZw6dUoNbODmnIA8YsqXLx9q1KiBKVOmqEcj4j59+nR8/fXXeOONN9C5c2fVq2bbpJchMDBQPb61vWjvXC186+i0iY303jzIWHpt+DlwPkbSusujwHHjxkFe+ZBE8sCBA2qwmbyS8O2336peebo75y6PvqVnUnosV6xYoQZTMtadM83o6LTmjPOMxPT+nAmlXk9tpX3++edo165duj2U0pu2cuVKbefy5YIuX76s3o9M+YhKGgEZoCOjXaWxTa+HUl6yl0SUmzWBtIlNcHDwA40lgefnwJpxenundZd3+y5evKgewdo2GXTWoEEDNfhMYp/umXeX72dJHuvXr6+eJslTJsZ65j0dOTI9c8a5I3L69mFCqc9Sa0m2d5jkRW/pEbNt0isWGhqqXu7m5hqBQYMGYd26daq3RqYKWrBggf1EfIfSOfO0iY0k7A8ytr1Xxs+BXvf0SpNfrqSX/rPPPlOxL++u0t26u4zwlqnG2rRpo5542H5ZZaxbt3T0iPuZM84dFdSzHxNKPY7aS5FeSBn0MXXqVLzyyiuqfI7y1su8a9cuNRpTkseUj6+bNGmiRiI//vjjmD17Ng4dOgSZTkW2IUOGYPny5RzlnclbkTahlDn6HmTMz0EmodMcltZdes9kyqYvv/zSvqcMlJLYl/eEK1SowO+fTNDPmjVLTS0m89VOmzYt1etKjPVMgDpwyIPMGecOAGrchQmlRkzdRclL3HPmzMG8efPU5NvScyY9Nrt37+Y8lBqwZeJmSSilZ0a+lGSapn/84x9qBL0MupGkUuYxa9mypZoDVHqNe/XqpRKgrl27aqiB7xWRNrGROfsyMubnwPk4Sesuj2ElrseMGQOZ6UAGpP3tb39Tj7w//fRTdUK6W3MXw+rVq+O55567ZxYOeeQtvxwx1q2ZZrR3RubyrjvjPCNFfT9nQqnPUntJMiBH5j2Uic2vXr2qJoSWR92SXHLTI/Drr7/i3XffxX/+8x/VA1yzZk21ssUTTzyhTiCJpfQ27NixQ40GHzBgAGQqCm6ZE0ib2DhizM9B5qxTHpWeu7ynLQNzZAoheZdSenNklHfu3LnVoXS35i4DnCQJT2+TX0BlVaKMvk9ort+ccW7N1Jm9mVA6o8djKUABClCAAhSgAAXAhJJBQAEKUIACFKAABSjglAATSqf4eDAFKEABClCAAhSgABNKxgAFKEABClCAAhSggFMCTCid4uPBFKAABShAAQpQgAJMKBkDFKAABRwUkBkWZMT0qFGjHDzC2m62uSCPHDly39kcXF0HazXm3hSgAAWSBZhQMhIoQAEKOCgg00zJnIKyoowrNllf/vz58yhWrJh9Mv2052FC6Qp5lkkBCjgrwITSWUEeTwEKUMCNAkwo3YjNU1GAAg4LMKF0mIo7UoAC3iKQLVs2Ncm0rMMuK5Rs2rQJ5cuXx/Dhw9X6yhcvXlSrmkRFRSEiIsJ+WV999ZVaXtO2/KBMZC9Ln9oeQadN5tauXatWm/npp59QoEABdOrUSU0WbpscXOohqy8tXboUmzdvRuHChdXE+HKO9La0j7ylnjKx/sqVK9XqWLLQwQcffODSx+7eco9ZTwpQwCwBJpRm3Q/WhgIU0CAgiZwspfnFF1/g1q1bqF+/vloJZs+ePSohK126NFavXq1WSZK12Z999lns3LkTderUwdtvv62SSPm7LLV54cKFdBPKFStWoG3btup9yvbt2+PAgQNq/7p160JW55BN6lGwYEFER0erf5cEV1aj+e677+yrMaW83LQJZbNmzXD8+HGVlD700ENqpaZvv/0WI0eOdNl7nBr4WQQFKOCDAkwoffCm85IpkNUFJJGTxHDq1KnqUg8ePIiKFSuqpe9q165tv3xZEu/YsWOQRE7+LGsDf//99/aff/jhh6qHML0eynr16qFMmTL47LPP7PtLkiprB0sPZ5UqVVRC2a9fP5XE2jbppRw0aFC6vZQpE0rb2s/r169H06ZN1eFnzpxB2bJlVSLsqoFBWT02eH0UoIBrBJhQusaVpVKAAh4UkERu5syZqsdQtmXLlqlexHz58qWq1c2bN9U61pKoBQcHIzw83J6Eyo7SS1mjRo10E8q8efNC1m9+66237GXKI2opb8mSJejQoYNKKGfPno033njDvo88Nu/evbvqZUy7pUwot2/fjhdffFE9npfH6batatWq6lqYUHowwHhqClDgHgEmlAwKClAgywlIIjd//nz1rqFs//73v9GxY0ckJCSoR8cpN39/f/V+pSSOTz75ZKrexAcllDLae+LEiaoH0rbJCO0iRYqoR95t2rRRCWXKesh+DxpUk15CKY/cJUm1baGhoWjdujUTyiwXtbwgCni3ABNK775/rD0FKJCOQNpETt6dlEE4a9asUe9L2jYZuOPn56cG1kjymZiYiPj4ePvP58yZg549e6bbQynvRD766KOpHnnL4JlWrVqleuSd2YTy9u3bCAoKSlXnpKQklCpVSj0yZw8lQ58CFDBJgAmlSXeDdaEABbQIpNcz+Pzzz6tH2DNmzFDJZUxMDAYPHoyPP/5YPYKWkdqPP/44Bg4ciB49eqik8PXXX4fMPXn06FH17mLK3kV5jC6PtUePHq3+L+9fyiN2KUPepZTNmR5KOZc82pb3Pj/66CM8/PDD6r1LGWj03nvvMaHUEikshAIU0CXw/2ZiEA8s51alAAAAAElFTkSuQmCC\" width=\"599.9999869953505\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pb_num_dict = {_k:len(_v) for _k,_v in sub_library_250_1.items()}\n",
    "\n",
    "pb_nums = np.array(list(pb_num_dict.values()))\n",
    "plt.figure(figsize=(6,3), dpi=100)\n",
    "plt.plot(pb_nums)\n",
    "#plt.plot(np.ones(len(pb_nums))*20)\n",
    "plt.ylabel(f\"num of probes\", labelpad=1)\n",
    "plt.xlabel(f\"region id\", labelpad=1)\n",
    "plt.ylim([0, num_probes_per_region*1.1])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a20eed5d",
   "metadata": {},
   "source": [
    "## for 250_2 gene library, change primers from [2,9] into [16,15]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "bea64b8d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Picked primer: ID: W1B05_primer_16\n",
      "Name: W1B05_primer_16\n",
      "Description: W1B05_primer_16\n",
      "Number of features: 0\n",
      "Seq('CGGGTTTCGTTGCGCACACC')\n",
      "- Picked primer: ID: W1B04_primer_15\n",
      "Name: W1B04_primer_15\n",
      "Description: W1B04_primer_15\n",
      "Number of features: 0\n",
      "Seq('TAATACGACTCACTATAGGGCTTGTGCATCGCGCCAAAGA')\n"
     ]
    }
   ],
   "source": [
    "from ImageAnalysis3 import library_tools\n",
    "primer_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Primers'\n",
    "primers_250_2 = library_tools.assemble.load_primers([16,15], _primer_folder=primer_folder)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "2612e9a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Checking primer usage, total probes: 27746\n",
      "True\n"
     ]
    }
   ],
   "source": [
    "sub_library_250_2_records = []\n",
    "for _reg, _pbs in sub_library_250_2.items():\n",
    "    sub_library_250_2_records.extend(_pbs)\n",
    "\n",
    "# replace primers\n",
    "for _irecord, _record in enumerate(sub_library_250_2_records):\n",
    "    _seq = primers_250_2[0].seq[-primer_len:] + _record.seq[primer_len:-primer_len] + primers_250_2[1].seq[-primer_len:].reverse_complement()\n",
    "    _name = _record.name.split('primers_[')[0] + 'primers_['+primers_250_2[0].id +','+ primers_250_2[1].id+']' \\\n",
    "    + '_library_1000-mouse-genome_250_2'\n",
    "    sub_library_250_2_records[_irecord] = SeqRecord(_seq, name=_name, id=_name, description='')\n",
    "    \n",
    "# quality check primers\n",
    "import ImageAnalysis3.library_tools.quality_check as check\n",
    "reload(check)\n",
    "primer_check = check._check_primer_usage(sub_library_250_2_records, primers_250_2[0], primers_250_2[1], primer_len=primer_len)\n",
    "print(primer_check)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "c672e4f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ID: loc_1:11240000-11260000_gene_4_pb_83_pos_7744_strand_+_readouts_[NDB_752_c,NDB_832_c,NDB_940_c]_primers_[W1B05_primer_16,W1B04_primer_15]_library_1000-mouse-genome_250_2\n",
      "Name: loc_1:11240000-11260000_gene_4_pb_83_pos_7744_strand_+_readouts_[NDB_752_c,NDB_832_c,NDB_940_c]_primers_[W1B05_primer_16,W1B04_primer_15]_library_1000-mouse-genome_250_2\n",
      "Number of features: 0\n",
      "Seq('CGGGTTTCGTTGCGCACACCGATACACCCTGGACGCAAATTAAAATTAGCTGCC...AAG')\n",
      "27746\n"
     ]
    }
   ],
   "source": [
    "# save kept records\n",
    "print(sub_library_250_2_records[0])\n",
    "print(len(sub_library_250_2_records))\n",
    "with open(os.path.join(library_folder, 'filtered_full_probes_sublibrary_250_2.fasta'), 'w') as output_handle:\n",
    "    SeqIO.write(sub_library_250_2_records, output_handle, \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "95fffd3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAApQAAAFKCAYAAACjE8/+AAAAAXNSR0IArs4c6QAAIABJREFUeF7sXQeUFEUTLnLOOefMkeHIOfsjSlZAQBABQZSk5AwKklQQEEEFQUFABIlHzvGAI4cj55wz/O/rvdrrG/duZ3dnZnfPrvd479jt6en+pne6usJXMd68efOGlCgEFAIKAYWAQkAhoBBQCCgE3EQghlIo3UROXaYQUAgoBBQCCgGFgEJAISAQUAqlWggKAYWAQkAhoBBQCCgEFAIeIaAUSo/gUxcrBBQCCgGFgEJAIaAQUAgohVKtAYWAQkAhoBBQCCgEFAIKAY8QUAqlR/CpixUCCgGFgEJAIaAQUAgoBJRCqdaAQkAhoBBQCCgEFAIKAYWARwgohdIj+NTFCgGFgEJAIaAQUAgoBBQCSqFUa0AhoBBQCCgEFAIKAYWAQsAjBJRC6RF86mKFgEJAIaAQUAgoBBQCCgGlUKo1oBBQCCgEFAIKAYWAQkAh4BECSqH0CD51sUJAIaAQUAgoBBQCCgGFgFIo1RpQCCgEFAIKAYWAQkAhoBDwCAGlUHoEn7pYIaAQUAgoBBQCCgGFgEJAKZRqDSgEFAIKAYWAQkAhoBBQCHiEgFIoPYJPXawQUAgoBBQCCgGFgEJAIaAUSrUGFAIKAYWAQkAhoBBQCCgEPEJAKZQewacuVggoBBQCCgGFgEJAIaAQUAqlWgMKAYWAQkAhoBBQCCgEFAIeIaAUSo/gUxcrBBQCCgGFgEJAIaAQUAgohVKtAYWAQkAhoBBQCCgEFAIKAY8QUAqlR/CpixUCCgGFgEJAIaAQUAgoBJRCqdaAQkAhoBBQCCgEFAIKAYWARwgohdIj+NTFCgGFgEJAIaAQUAgoBBQCSqFUa0AhoBBQCCgEFAIKAYWAQsAjBJRC6RF86mKFgEJAIaAQUAgoBBQCCgGlUKo1oBBQCCgEFAIKAYWAQkAh4BECSqH0CD51sUJAIaAQUAgoBBQCCgGFgFIo1RpQCCgEFAIKAYWAQkAhoBDwCAGlUHoEn7pYIaAQUAgoBBQCCgGFgEJAKZRqDSgEFAIKAYWAQkAhoBBQCHiEgFIoPYJPXawQUAgoBBQCCgGFgEJAIaAUSrUGFAIKAYWAQkAhoBBQCCgEPEJAKZQewacuVggoBBQCCgGFgEJAIaAQUAqlWgMKAYWAQkAhoBBQCCgEFAIeIaAUSo/gUxcrBBQCCgGFgEJAIaAQUAgohVKtAYWAQkAhoBBQCCgEFAIKAY8QUAqlR/CpixUCCgGFgEJAIaAQUAgoBJRCqdaAQkAhoBBQCCgEFAIKAYWARwgohdIj+MIvfvr0KYWEhFCaNGkoduzYBvWqulEIKAQUAgoBhYBCwFcRePnyJd24cYMCAgIofvz4vjpMS8alFEqDYN69ezeVKVPGoN5UNwoBhYBCQCGgEFAI+AsCu3btotKlS/vLcE0Zp1IoDYL17NmzlCNHDsKiypAhg0G9qm4UAgoBhYBCQCGgEPBVBK5cuSKMSWfOnKHs2bP76jAtGZdSKA2C+eLFi5QlSxa6cOECZc6c2aBeVTcKAYWAQkAhoBBQCPgqAmrvD38ySqE0aJWqRWUQkKobhYBCQCGgEFAI+AkCau9XCqXhS1UtKsMhVR0qBBQCCgGFgELApxFQe79SKA1foGpRGQ6p6lAhoBBQCCgEFAI+jYDa+5VCafgCVYvKcEhVhwoBhYBCQCGgEPBpBNTerxRKwxeoWlSGQ6o6VAgoBBQCCgGFgE8joPZ+pVAavkDVojIcUtWhQkAhoBBQCCgEfBoBtfcrhdLwBaoWleGQqg4VAgoBhYBCQCHg0wiovV8plIYvULWoDIdUdagQUAgoBBQCCgGfRkDt/UqhNHyBqkVlOKSqQ4WAQkAhoBBQCPg0AmrvVwql4QtULSrDIVUdKgQUAgoBhYBCwKcRUHu/UigNX6BqURkOqepQIaAQUAgoBBQCPo2A2vuVQmn4AlWLynBIVYcKAYWAQkAhoBDwaQTU3q8USsMXqFpUhkOqOlQIKAQUAgoBhYBPI6D2fqVQGr5A1aIyHFLVoUJAIaAQUAgoBHwaAbX3R1OFcsSIERQUFEQbNmywz3Dp0qU0bNgwOnr0KKVOnZqaNm0q/p8gQQLR5vXr1zR06FCaMWMG3blzhypWrEhTpkyh3Llzu7SI1aJyCS7VWCGgEFAIKAQUAn6PgNr7o6FCOXHiROrRowdVrlzZrlBu3ryZqlWrRlA0mzRpQqdOnaKOHTtSjRo1aNasWQIFKJNQIPH/TJkyUZ8+fSg0NJQOHz5McePG1b3Y1aLSDZVqqBBQCCgEFAIKgWiBgNr7o5FCeenSJerQoQNBecySJQulS5fOrlC2atWKrl+/TqtXr7bPeM6cOdSuXTt6+PAhxYgRQ1gtx4wZQ506dRJt7t69SxkzZqSZM2dSixYtdC94tah0Q6UaKgQUAgoBhYBCIFogoPb+aKRQwqU9b948YYWEK/vs2bN2hTI4OJhixoxJRYsWjaBQtm7dmm7cuCEskYGBgXT8+HHKmzevvQ3c3kWKFBGWS72iFpVepFQ7hYBCQCGgEFAIRA8E1N4fjRRKeUm2bds2gkKpXa7Pnz8XCiSUzL1799KiRYuocePG9PjxY3tMJa5p1qyZ+GzZsmW6V7xZi+rY1fu05eRNMY5YMWNQ7ULpKVPyBPT0xStadvAK3X38XPcYUyWOS28FZKS4sWPSuVuPaO3R6/T6zRvKnCIB1SmUXlhsD126RztCb+nuU2/DeLFjUr2ADJQ6cTy6//QFLTtwhR4/f0lJ4semt4pkpMTxYtP1+09pxaGr9OLVa73d6m5XMU9qyp8+Kb189ZqCjl6jAhmSUrZUiej5y9e0POQK3Xz4THdfyRLEoQZFM1L8OLHo0t0ntPrwVXr1+g2lTxaf6hXOIJ7TiWsPaNOJG6LPXGkTU7V8acXfe8/doeDzd8TfxbOmoJLZUoi/Nxy/TqeuP7SPAc+iWr40lDNNYoEHxnjjQfgY48SKSfUKp6e0SePTvScvaOmBy2JN+KLg2dcPyECpEscTY/3noO3ZayVloriiHXC9cPsxrTlyTaxPMwTPqGaBdJQlZUKBG/DD2JyJ/Owv3rGNEc/emRTMkJTK504tmu0MvUUhl+45u8T07zH32gXTmfq7104iUbzY4hkDR6znlYeu0LOXr0l+9udvPabDl++Jdx2e08lrD2hj2G/JE1Dwm6qRPy1lT52Inr18Jd5Bd1x4f+q9d/ZUiahmwXSi+YELd2n32dt6L420HX7vdQunp3RJ49vb4D2K39KjZ//+LUXWUcbkCahuofQUM2YMOn71AW0+aXtHsRTKmIzK5Uol/rvt9E06cvm+x2OPrIOk8W3v0QRxY1FUvyW8p2sWSCvWKYsr79Gq+dJQrjSJTZuHWXu/aQM2seMYb96Y9MY2cdCRdR2VQvny5Utq2bIl/fXXX7Rp0yahWML9DWvlq1evhJLJ8sEHH9Dly5dFgk9kcv/+fcI/litXrlCZMmXowoULlDlzZsNmP2/Xeeq7KMTeX5kcKWn+x+VoxuZQGvHPUZfvM6ZxEWpWOgvVn7SZjlwJH//cDoFUMnsKKj0iiO4/1f+CcmUADYtlpEktitPwZUfopy1n7Jd+Wj039aidjz76dY/YoM2QOLFi0LctitNvO8/TllM3hUK5onslmr/7AvVZeNDlWw54qwB1qJSTmk/bTjvPhG8YP35QimoVTEeBo4Lo2v1wBXDVZ5WFwll6ZJBQYiFQ7PcMqElX7z2l2hM2/WsMOdMkonU9q9KS/Zeo++/7//V9pTypaXb7QPpy4UH6ffcFl+dg5QXvFMtIEx08e+0YRrxTmFqVzUbvTN5K+y/cNXWIJbImp0VdKrj8W+Jn3+SHbbTnnO1woEe2fFFNPPNyo9fpUkL19OlpmwWdylHRzMmp1Ig1pv3utWPMkzYxjWoUQN3mBtPV+0/tX5fNmZI+qZabWv+0S3w2rXVJcdAtP3otXb4X3s6TOePea3pUoV+3n6VBSw570lWU1y75pALlSZeYSo0IosfPjTnoQama0aa0/b4j/zlCP24Of4/qncwvH5ahynlSU+mRa/91kI4Zg2jblzXoDb2h8l+tI7O1gz5181GXqrmp8Q/bxGE7Mlncpbw4gEOgSANX+T26u39NYZCo5eA9miN1Ilrfq6peeFxupxTKcMj+EwrlgwcPhNVx/fr1tGDBAmrQoIFAYOHChSJZx5GF8tmzZ7RkyZJIF9eQIUNEQo9WjFYolx28TONXnxA/opsPnwvr5NYvq9OQvw/Tz9vOUoI4sShDsvBTa2QDvnj3ifgBdq+Rhz6vlZeKDVtNdx+HW2SGNChINQqko0pj1osuYLWMGytcyXb5VyZdcPvxc3Ev3sA7/rqHVkuKY+MSmWlcs6J2JTdFwjiUIqH+hChnY7v16LlD69PZr96i8WtO0LdrT4qNPnNyW+Z/VHL53hN6+uI1fVghBw1qUJAqjVlHF24/sV8y6H8FhUKUd8CKCN381KYUZUuVkGqOj6g44kV39uYjavfzbtE+Z+pE9PDZS7r+4BnFjxOTjg2vR5PXn6Kxq46L54HnAosOLKM4sG/qXU3g9uDZS0qbJJ6w9PqSMPalsqWgPzuXJ372SeLFpjRJ4tmHyuuza7Xc1KtOPrtCnjpxXIIlw0hhfDGGkKF1qM+fB2j+nouUKG6sCBYg7T2h/EA5eD8wK416N4BKDl9DmB/mgb4cyas3b+jcrce2903nchQvdiz633dbxP+zpkxIsbGDe0HO3HoklIVv3ytOVfKkoaLDbHHmGZPFFxZis+TsrUckG3Txu8M9z4ZhJN934P8KUtvy2SlXv+Xi4/RJ41PCuO6NjX8z6CdkSG0a+Nch+mv/ZfHc5HXo6bx5fpNaFKNiWZJTlbE2xhFP3qewot55/MJ+COYxfjJ3n7BQ4jeP374zOXf7sTjIfNUogN4tkYnyDVhpxxXP4fxt2zqFMgzPwLtTton/470VS7IOOruPnu9vPHxGD56+FF6Wye+XoAKDVor3Wrqk8ShR3PDfUujNR6K7KS1LCMs25PSNh1Rj3MYIt8F7FF63trNs71EokU+evxIHFvzETo+qH8HCqWeMetsohTIcqWivUMJyWK9ePTpz5oywTiLrm2XXrl3CUons71y5ctk/Rwwl4i4nT54c6ZqyykLJA2BLWqpEcWnvwFp2qxTcdjPalHK69vkE2LlqLvqibn4qOGhlhJMzPoc7qMnU7aKv/YNqUXKDlLrv152kb1afoPzpk9DKzypT65920uYwNz7u9b8iGej790tQjXEb6PSNR9S/fgH6qHJOp3PS2wAWwPd+3EFnwl5OuA5u2OMj6tHoFUdp2sZQKpM9Jc3vVM5pl21m7hLut9Zls9HwdwpT2VFrI1hZetfJR60Cs9k3aO4QG3e2lAmp4eStEe6x8rNKQqHsNGefUBhPjKwnrLSw1uIdHjqqPk0IOimU3iKZk9HfXSsKd3HJ4UH05MUrgpWSsYQFLHOKhE7nYGWDiUEnaGLQSQrIlIyWdqtIWvx4LE2nbqPdZ+9Qpyq56Mt6+YXFDAcotqgbOeatp25Syxk7RZeHh9ahLr/tE8+0ZWBWGvluQKS36vHHfloUfInqB9g2wdz9V4gNemqrksId6Ujgjiw0eJX4at5HZcUhgTfqfQNrCVevN6To0NXikDW+WVGqnDeNsPhA/vqkglCEzBIoQJ/+Hixww2F4ZtvSwsU6beNpGr3iWITbwnoFhbLgIBt+cz8KpPK5bGEDrsqVe0+EZRiyqEt5GrD4kPDQfFItF/Wuk9/V7iJtz+8DWNqBIx8edvarEeVhJaoBTN90mkYtP0a50iSitT3DLW0fz95Dqw5fo6YlM9PYpuF5ApH1VW70Wrpy76l4bzUqninCukRIBh8q/uxUTij9zabZ9oKDQ2obfqjjgzz2hJ/alqYKX9mezd9dK1CRzOHrr9CglfTo+SuCgt6wWCbbb/byPXrrW9uhjGX155XFwU1+b+4IvS3e++KaoXUI4RZmiFIow1GN1goleCXLly8vXNMrV66kgICImwWskGnTpqXx48dT+/btBSq+muXNbk9YUQ4Pq0uf/R4sTtisjDn7obw3fQdtD71FHSrmoAH/K0h5+i+nF6/eCKUFlopGJTJR9fxpqevcYGGtOz68rmEnOri34ebGSXdj72r/cm+wUoyXCixvwxoWog/KZXc2JZe+hzvkq5XH6MrdpwIHyJnR9Wno0iPC0svuY2eddvhlj4jBbFE6C33VuIjdSsXXYYOChZI3L/7868YBlDVlIvsLjj/HBg6F8rM/9gtrCSxmiL38YKbN7XdyZD0at/oETd14mkpnT0ELOpUXn7N1gvspnjU5Le5SwdnwLf8e4/5qxTFiVyOHCHxUKQf1f6ugfTwtpm8nbAC8PlnhGde0KDUuaVwICW546voDu6UYlo3Oc/bSsasPqEetvPRpjTyRYjRs6RGaufUMlc+Viqa2LklFhtisen90LEuBOW1xZ1pBrB5bgma3LyOsf03DDm1mbNR6H7BdYW9ShKrkTUOBo9aKS5d1q0iFMyXT241b7dYfv04L9lyg9hVz2mOI0dGfey8KxX79sevCSg9vChTK4sPXiPss7Fw+QntXbo7IriJDVwurGKzLQ5YeFh4bWVFxpb/I2tYav5FOXn9IUIaLZU5O74cdXI4Mq0MJJcubK/f6eesZGrL0CGVJmYA296luv7T9z7tp7bHr9F6ZrDS6UeQHIb6AvSmDGxSkRsUzR1AgEQIkH3yAF4/96LC6Is7RSFkcfJE+/+OAOFQgTKjVT7YDnvY3wZ60b5oWpSZh7wG4xmEgkQXrFhZWHA7ZWIB8ACMUemfzVgplOELRWqFETCUywKFMFihQIMK6SJMmDcWKFYv69+9P06ZNEzRB2bNnp969e4vEnpCQEJ/ioUTiR8fZe0WQ+qmR9ajznH208vBV3adTKChCUSmXjYY0KEQ5w9xIeEnBZVsxd2qqlj+tUPy0Ly5nPyhn33McKFxLiHWpN2kzHZXiN1mZg5UEyTFQvpqXzuqsW7e+R4ITv7yOj6grQgfm7boggr7l+KTIOu/y215aHnJVKODjmxWjgCGrxCbF0q5CdqFQal0yeIlDof7w5z0RuoYyAjfZFwtDCO7dPQNqiaSN5tNtJ+tDQ+vQuNXHadbWs+IZzekQKD5HMgOsmixwD7avmMMtTMy8SLsZwkKLRIVu1XNTz9r57LdmqzUUiCFvF6LCg1cJpcLoDR83RPgIK4O/dywrFEq4FJ2tO1iJYVnB5jutVUmqPNYWHoL42HzpkziE8fXrN/bf2sy2pSh+7Fj2jfrY8Lqmupejeq4c4wvlqnLe1FTxa9tcYDFH8po3pdnU7bTr7G36uEpOoVDy4eyfTysSkkbcFfbS4OC87th10Q3iqPE8jRLZE4TY1E5z9oqwBhwM5aQSV+43d+d56rc4RLi1d/Wvab+U3+ltymWjoQ0LO+2y+jcbCC5keIDg8pat0rBQcpjOrx+WES5vdh9jv4ltUPgTDxLx0YiThiCG/tt1p+zvP3kivCfATd+ijG1P2Hbqpv03xG0RYwmFErHmfDCHC5xDDoJ6VKHcac1JzFEKZfgTi7YKJSrgJEqUiJ4+dRzMDRc4FEgk5PTr108Qmz958kQQo8PVje9cEbMXlWy1OjGinjDty65XZ2Nly9p7ZbKIDZutJrC2bDt9S1iQ8KKdtimUON7NWZ96v2frKv/Qq32zQbifoRzD9cWJRqycTWxejN4pbnNvGC3IuJQtREOWHBZuzLcCMtDkliWc3o4tw8hO/O694pR/4AoRU8kCfFsGZrOfjJMnjCPiR+EKR8xct3nBYoN5GRZIhpc3FEokCHB8LLLA2S26d0BNGrfmBGFTwfOBixCCzGTE8MEdBNnRt4ZI+vE1+X3XefpyUYiIU8Nhou7ETcIaCDyQgMHSdtYu2nA8PJSAcYVr+a0ittgpowTWF1hjEA85tkkR6v2nLSnr53alqWpYNr6je83efpYGLjksYpaRMPL297YNcVe/GiLbPjJBDCDWOa6BhRJuf4gZG7VejNgbMLxhIeHyDt94K1PutI6VY719e9pOPly0KZ+d8L6ArO1ZxaNs3b6LDorDIxL04J3B+weWQ8S1GiW8jluVzSqSnbC2EBMePKi227eA5bbXggOEd8l+qZ/3f9wh3t04SOJA6UxqT9hIJ67ZrKewUJYdHW6VLpQxKeXoa4tVxcHn9WuiDr/aDr/w5LirDEc2pnuPX9gtpAiHAeuBo31HdtMjzEisg6PXqP0vEQ/mSFSFAgm8EUaCcJJbD59RybBQDjmpxxlOrn5v9t7v6ni82T5aKZTeBNLsRbXrzO0IMS0f/bJHZBdrXYeRYSBb1oa+XYgCwtx1zUplFgkJSePHFgoL3Oh6lSu9eK86fJU+lqyreJEhAxoWOcTJFc2SXASC44QMN9QPLUsIiiEz5ODFu3ZFAAoOLJT/hFwRMUXjmxdzesveCw7Qgr0XBfUG3J6sLMB1g5hGZLLDQslKK+KeEBcKV3i2lIlERjleeKB7gk4Jdw9c3iOXHxUJOet6VRVUHfW/3SzGsr1vdeHyxqaCAPYfWpW0j/HzP/bT4uBLdoXc6eC90OCv4Es2d3782BQypA5VHbteJGBwpjQPqcMvuynoaLj7jkMyoopP9GQ6fKiBdeeX7edEV86sVX8fuEyfzgsWbjo8e1YMYemOSinJN2CFSDiAcowYSt4Mzdio9WJSZex6EXMGyzkUSraob+hVVdDqeFPkwy8UyroTbb8FT2OEZ209I0JcWLQxiUbMGQdGUFDhPYBYQHh8cJDc1Cc8dt/V+6A/9IuEpCPD6tov57hjWHL71ovogXN0D2b2QGjHu8Uz2RMw2cKeu99ycdDFbw6Hrs6/hcd1uzpmPe2LD1stPAMscGnDtS2L7KZvV8HmgUEcLkJ+OFwLn81pH0jnbj+i/osPicSenf1qCmooOdykUp40eoblchuz936XB+TFC5RCaRD4Zi+qkIv3qMH3tkDkXf1r0Ee/7hWuQ6bccTYN2bIGhbJEWFxSz1p5hQUMgiByuCLgth3coJCzLnV/D64zpgLB5gs3BtzErGwhMBubuXxCrp7fxuNmtIB7rc5EW6Y1NigolDZFJguNblTE6e3geoK1EAlM0z8oFSEDFRmFoAyCQsnKBqyvOAwAUyTlIBYKGZ+3Hj4XCuj37xcXCqWctCRnMWKDx/PBpsLUOzxInMAR09e0ZBavKwGRAceueU44kpMWgBOLnGAwpkkR+1qY8UEpO6ef04fjQgOO5UTsafB5Gz2RsyQZeR1jjH3+PPivTd7REAIGrxJZ+LC8Q6GUE7BcGLKhTauP20ChYQlwUCj5N7G5TzXBzelN6Tp3n+DYhdIDhZJdo6DYAo+tuyInY6EP7QHN3X7l6/ovDhHUZOCQxUEZCWmFMyWlZd0qud09H8jh2Tg1qr69H6bWYmYEZzdo+P0WOnDxnogTxgG6qsbyW2DgSvs7CYddHJ44Zt9Z3+58/+6UrfbfHq7Xei3wmeym50TNhXsvUs8FB4QRhCnuZrUtLSyU/H7d8oUt1pSNFHKWuDtjjeoas/d+o8drZn9KoTQIXbMXFch9mWMLVDFweR+/9m/XYWTTYctanULpaOjbhe3ujgnNi4rgaAiCmWFJQZYtsm2Nkr3nblPjH8Kzx6FQ4iSMJBNk9sIyt7x7Jco/0EZj8VuHQKoQRgJt1Bi4Hyhv8osUCiWypDl2z9n9mK4JcZ+wLvKY86VLIp4H4hzh7oLSgA0ALlRO4smaKiGNWXmc8qZLLGiB4ApH0glc3t+tOyUUeiTpgNSb6ZtgPUAMJWiWYE0e08R5NqezOVj5PRIw2oVReYC6o+SINWLecpA9xsNJRtjokLHKVDHO3NDuzgWbJSyOvObhBkUoSVSuPTnIv0vVXDRlw2l7mEJU42BLDNzrcHk7sja5Ow93r6szYZNYr2B8QAwlZ836QuhEz/kHaOE+m0UeCmULKZ7YE1qs6w+eUpmRNjcvBIoVrHVGChLQkIgG9y0USiQklsuZiuZ1LOv2bbS/IbjqIf/7bjMdunSfPquZhz6r6XweHN+JtYsYcKYww36CdxOHHGFPgMsbShsI6A8Mdt9dH9WkmTWB2zgKb+F1Kiubc3acowF/HRI0UrcePRPhCwgnASG+8PSE8feiX6b2MoMtgsdt9t7v9sLxwoVKoTQIdLMXlaxkgCIBCiVcVnqTMfjkDLc2LJSssIAigqmCGAq8UN4tblxmrezCxcuLkxlgzQNFDmIHoVAisxcCvr6S2VIa9GQidgMKIY4dWv5pJWGh5AQAPW4jJhMGCTMUSg4dAO0Q+gHXJixvPeYfEG5e4L1k/2VREQIWyu/X2xRHjAMWTSRFQKGcvimUAnOkpD8+Lkfyxre0a0Uat+Z4hPhCU4AxqdPtp2/ZM9sRrwbLOGJOYZn9X5GM9rt2/z1Y4PR20Yw0tmkRu6sKrixUOTJatKTQHL8a1X3k3yCog5CchWQGrN2opMzIIHGAwLOGhRJrw8yNWg9W7P6EhwIWSqazQhiIkbyMesaibSO/q6BQsrUfiS2oGOOuwI2L9cduVjPic5kzFgdM0HwhPAbViODNcFfkJBQ54zqyeOTI7sNW+Y6VcwrrL5IjIdu+rE6ooGNXvpoUISSTIfYZFmFYhs0QTnLjvh0lXb317WY6fPk+fV4zL3WvaWNg4KIe2VMlFL8rxELjfYL9EHy9TE+Hthxio3efdGeeZu/97ozJW9cohdIg5M1eVLKSAa4uKJSIQxz5bmGRBOJM2LIGCxqScmqOtxHDwqUKdxcskyyomsNl4pz1q+d72SqI4GhOOOH4TcRSYlNm64GZ1CV3Hj2305BgLMCF3UB6rBVjVh4TlimUTIRCyaEDoD7iko6wUHIsDxRKziIHbRDTzoAeiWPYgA/i+LCxI0lHDliHwo8YSlAd6Q3usRzoAAAgAElEQVS+1/NMrGqjTTDiIHmtK7vH/P20aJ8tOQoKJXMPgruRS8EZOWZtpSkm3Y/qHsg6R/Y5BJnBYCqokDsV/dYhauuTTIeFLG/E0XJGv5FzcqWvt7/fQgcv3hPUPFh3TMMSPLAWpfASNyaPnytpIWEQCiXHX8PC7amAWxEhKJA1n1emPOmMTUDixC2QtSOGEkwcXLjB3bHLHp4Dg2pTsoQ2on/m7e1bLz99rMOj1HLGDtp66pYIv0FSjj2EKiypjDP/safA5Q3ydySgbe9bw92hR3kdxyRzI0dckY5YIVhph+IIXk3wqcIIgvepzHmLfl214rozUbP3fnfG5K1rlEJpEPJmL6oHT1/YrWHIaINCycTEjUo4tyaOXn5UZHAjpg8WSvl0CpcSV0kAHEZTLIADskwYzx0qxnBSwseVc4oxwY2FGEq2mprxoufHDFJwWVmBQulK6MCENSdoUhjJOBRK5u/DpgE3HU7NsFCiLCaqNaCGNyuRoA1i5RJ4I+MSLkcolH/suSDiL9EnMrjZlQ7lHjGU4F6Dq6pPXeNImA1a+lF2I1uncXjhcAOt5fGLPw8KDGDNgcubrdVQqEtlN95ard3M9MTTwcKVp/8KEa7BSVjIQIelKyqRrSSwUOKwAXfdjn7mbNR6nmujKVtp3/m7hPg7KJRMYo0qMkkMrkykZzxyGz60QcmHQgkqGKNi+Qb8FUJzdpwXmd5IcPHE4uloXsxogXda0SzJ7AqcJzHp2vj5tElsjAKcWKXX+iYXFQBtUKOwSjgcO1zx63V08c4TsT+AlWCYAQlFUT17OYSEE2m07bm8KRc8wPcIAeIQIYyXqeagUOKwLx8OtfzLrq5FPe3N3vv1jMFX2iiF0qAnYfaievHqtdjMIKjFCoUSGdF6g42/WXXc7m7FC0N2cSEDHLGMLEZvKjLvH2JZuHY2lKmvVx4TL3colNqYHoMeTYRuXr56LSqcQBCfB4XSUdZxZPeWT8dQ/lgJxqkfXJHgioNCCb5CuENrFEgrXn6Ip4LLG4oM3LqgTQJVBuKfoFDKJPUydyGCzeHyRqyU7PYxAxsz+gy98ZCqh5VJQ3woJ1hoFUU52QkKJVt+zaL7kLk+MW+9MbTMi8dYOauug3bwBpy6/lDEJsePHdMhSbUZ2EfVp8z1iNKLZpJYuzo3doWCygYKJZKfmArG1b607VnhM5oaje+z7tg1O9csxg93raexmtpEQq6G5WohiHAmhSwipElbCYeZD8DAAB5KVOeR4xE9xV57vWzx53AfbRttwQN8P2LZEZqx5Qwh7Ahxk6jzjuo/5289ErXN5b6wTyKsqnmpLPR1E+dJl+7M0ey9350xeesapVAahLwVi4opakDrAMJcCDjD9GREyy9pKJQcN4mAa2zmoGKAgJoCrgcjecdkJQ4caEhMgaC6Q99FIeJvuPGZ18+TMmV6HifTYyCQGwollyNjnrOo+pDLoEGhZGUJiiHcLeDabFk2mwjMh1scCiUn4sDlzQk6UDD2nLtDKHkJhXLFoYiusXDanBLC5Y3qG1DA0d6f5PLdJ1Q+rKwaFHgmS9aGNQxacoh+3X5OVG2By5vDHxBDGpDZfTLryLCSwzDQBuuyS9VwXszIrmPlkL/Xk2HLsW69aucVSTmwXjNFlLeeJXMYojIRLJRyZSajrXauzlH+jUHRB/cnXMjbDHC9wvIGTl8oe1Fxh7o6Zm4v89yiTC5qvWspslztG4dPR1yccmwu6ss7k06z9woXPOh5kPymPUQwTyXeM1AotfGIzvp353s+oHHlMW0fzEkqk7ezlblqvjTiYA7L5KD/FRReNm3VM078MZoKTx6nFXu/O9h64xqlUBqEuhWLiquHoKIHKqtA9Na3/WHDaWENBIE5FEr5ZYIXB9yyELhpUY7OaMnbfwU9f/VaxAEi8xEytVUJe7UXZHZzfWUj64g7mgfXh0V9bSiUtx89J9DANCuVxem0mcsO3HJQKJluBXx+4LhDBmarwKwiJhKZ4KAXYioLuLw5hunEtQfi7w8r5BBJOajcgU0BiRsQuVIMrJ380vzQB6vhRAUasGVrI6rewH0JCeoRkUB76NLDwsKLmMSxTYralVAkThXMaFwlEx6rHPqAz/SWeGTuP+5Hj7LA8YqwVMHljQMGkjZWfV7Z6Xozq4FMHo4sb67g5E1uTJ7rL9vO0uC/bUT/UChljlaz8DCqX9mayH16mmGMeGuudS1zpeJ35cq7i+mYQD/2bonM/0p2knkqoVAaQXnkDFdUqcJhGgwITR28fz/8efe/3o0geWdeXhzMcdhGHOm524/tlG6oDw4ZvOSQ/V08u72typjRYsXeb/SYzepPKZQGIWvFouL6u3CdgZ4CsqhLeSqRNYXTWXA9bcT4ISmHLUUIdEeCwuiw/iJzPTi9gZMGXJsZLzO4dyEoe8eUICAzB5EuxIzasfLw+EWMlxgUSlSb0VvijykrEKwOhZJrxY5vVlRk70J4jogHRLIOk5lDCQXPJ0jOj155IF6UcJlCoWRFk2Ot5IzL8atPiIxwvQlYnj4rI6+XFTccZKAoQLR8h6OWH7VnuoNSyIp4Wrlspt5scibdZoy09EeOsON4RcTAwkKJAwIsZP986j43oafPqN2sXbT++A1BcVU5Txp7WVcjEl88Hdsfu8+HlSKNR23LZ4vA0epp32ZfL1vk+V6eFmpAjCCXSUQBCITPQFytLMaFEBD3CwullmCfE2DAbfzqzRuavP40gad1cZcKpsH26NlLEcOOuucxw+iQ5Jt1/HXPvyjTmGIMmepQ4I9cuU+w/uPQLRedQD9jVx0zfR5W7P2mPQCDO1YKpUGAWrGoOGYGAcpwqUL0WnDk7EMolKgLjt9v6Oi3iKuZoD/E98FyZ7RwCS1Y7cD7CI5GxNSxQiZbXbGpMdea0eNAfzyWEe8UFgolV4eoWzi909vN333BnqULKhAObJeTjXiOUCxrFkxHXecGC8sUKuVwAhAC0nEyh/sJ7ld2f8PdJI8RsUHjVx8XVCeRneKdDtqLDeR4UMSATgiykeiDnJ+TC/B/WM9hRUdsG2Io2cW3rmcVypnGnBq8svtabyIY87kypHqI12W6FsRQom4xc45669FwbBkI/aFQiqoosWMKLk5vi1yqFTGUTLWF94Wvi5w8yWPVe1iJbG5yDPqCTuWodFiSmkxELlNwRdaPzEWMGEquM85k6XICDBLQRBJn9pQ0v1M5r8H+yW/7/lXJTI4FPXLlgb3AByyUTNGGsrgQ7JMwvsAzt6ZHFVPmYcXeb8rATehUKZQGgWrFomKaCGwCyBaG6N1w5VM/LEU45UHJOTa8Hm07fZPe/3Gn6E9vKUdXYeOKBxyoDo5GJFxwIk7/+gWEawsJOidHek4PEtX4OPi8X/38IvAcMqtdaZGR7UwWB18URPCo0gALZfMw0mW86LncIs/xvTJZqVbBtHZ3Ilx4cF/BPY4qR7DUgp8SCiUn6DBBsZwZDIUSVlQo+lD4/U043IETlzD+g0NqU1Ipm1jO3PymaRFLErSYRgXjQSwxuCGdiZa/Ug9naqsZO2nLKRt5PiyU2ORA6r+gU3lntzPte3Y14kCDGEoQvWtL+5l2cycdc3UlvAvalMtuT8D4vaP3FBu9WOAAlbv/ckG7w4L4cFAIuSsy64Nc9MHV8qSIV5+36zzVLJBWJOXIewDGJmdEw0LJISjOaLHcnZee65ifFu9JVhLl39Phy/dEQimMLOdvPxLcsCBtH9/MVkZXJkE3i1XBir1fD1a+0EYplAY9BSsWFXNqoY40gqshqPWcIVkCp7NYtO+icMlCERrasJBQiri+slzqT09MmNObOWjABLWgS4H7FjQRf3Yqb3dtdqueW2RDI6klZGgdd26h+xpOkuB74kK9fIfLDl4WFkfQxkChbPWTTREH2XztCbaSjjxHxIvC5f3ejzvE50w1A2ssyv39vttGkwOXN1MIcdINV4hAsghc3rCiIomoTiHnVlTdQFjUkF1zcLMtCr4k7gpLGCxiLBODTthjthBDybRWW7+sLmLpzBAO2MfBCmEWehLROMufx7O2ZxXK5cSCKruXwUOJDFVPq6d4igfXnIbbEDGUfEg6OMTc356eccuVYRBXjFKnSNYCu4U/SJEhq+wlATFexKQjNt1dka38nIQJCyKXqoV3pEYB56Vq5cQ3KF2IZ5bft3ICDBTi2TtsSXLexJ3jJbHnTW1dUkDIllS8K3Ew33baxtELlzcnPX7V2JbRLdM4HTJpX7Fi73d37Vh9nVIoDULcikXFPyTEOe4MI+fVS0QsK0KwUMrkyjJ9g1lWME5mgNUBpbIQywlXCmfzflAum8jytYLwmWva8j2xBPTGosp1daFQtvt5t1hBKFnHFXh4jsgArl0onT17nZcaTtp7zt62k5nDQokMRWQqctJNg++2CKulrPTqtaIatKQN66b0yCC68eCZqBqEuFGEWiCsQVbgZDomxCVyKMSuMNJlwwYjdcRl8pAstbF3NV23gHIDVgSWvQNqUion9aVl6hIorzKJva6bmtCI4+n+VySDsFAaSc3j6XDl6koYH+p6o2TstNbuV5vxdEyuXM+hSXyNpzXI0Y/M+lC3cAbS0shB8XMmw5YeEcmXSHyDhRLKWoqEcSh4kK20Yvufd9PasORAKLE48OJAPKON93Dvu+ignbt3Rhtboo1cPSf4wh1RRQzvciiUG0/Y/h7WsLBoK9M4mRVKZcXe7+zZ+sr3SqE06ElYsajY1I8KAceuPhAj15vAwooQYhMRQ6mtgoC+D12+Ryu7V6b0yWzEuUYKaElA18GCSiNIymEC64bFMor4Fz0l8DwdF3Pw8T3Rn6OyX47us/7YdbsSyVnqUCD3DqxFRYbYSkeyoP4sNkJ26/PnsCjgQMDlFmGh1FY94sMD3KSgwoDozej3FB+jr680Zh1duP1EEA6DTNuRa1WOdUIMJfNV6lHY3B3vjtBbgi4HlCT93yqoq5vlIVeoS1jyGC44NbIexXZSDhA8r+yKg8sbSimy/zkTVdeNDW4kZ8pCoYQ7FByqu/qbU2bPleHvO3/HHpvM8cj4rU5qYXxstyvj0tuWPSDc/viIuhQvdiy9lztsJ7M+NCyWiZ48f0UFBq20vRd0VjaTi1vAW4DSiiiziXKbkI9n76FVh69R05KZhcue66n/0MpmGfSGYJ/SWkrlCkGIPQfPJMLAoFCytRJk7xCZxkkbZmPUfKzY+40aq9n9KIXSIIStWFScYQp3MRQQSOio+g6z47TTkt1IcGuDC0+2zMCFAsuh7IY0CBrRDXOgcZ/gaEQ8EFeEqZYvjcg6tYKfj107fE+MSW91oC0nb9rd3JzZDQXp4ODadsJ0niNiJWsXSm+n/ODPoRjCCsPVHqBQ3n38guSMYY7vg2sKJQkhCzuXF9yW/ia1xm8U1B4gSQ698cghSTWXQsTzBw9l4x+2i2nKpebMmPezl69c2uzleGMOGXE2LjkODEk52kxUZ9eb8T1bflCdCQolNm6juB49Ha9cXQn1sFEiEmVaxzQp6mnXllwvl3eMFzsmHTcg0UlmpgC9jpyog8ppqIDmTOQKRKAN0hoVmFYIYRDg6+QiDGYkaTobK38v04lxLKdM6L4z9LZI2kGlMsRQcjwlmFAgKI+qrVmu995621mx9+sdi7fbKYXSoCdgxaLiHzyUPlTJceVltfXUTTvPI2fb5k6bWChSVgjHq/G9YHlA3WqOA4KihPKCsL6u/Mxcfj7OEuR7YkxaGpvIMJErrCADGy9lJHMgqYOTT/haxEoi5rHYsDURugP1x+aTNwQdCiy1cHk/efFKBJ0j+BzCcXfY8HECh5hZ49zMNcDueyZ6dqS4ML9nlpQJBA8l00kh7gll7HxFZGUHNFCb+jh3lfecf8Bu7YGFcnHwJYIr93snJRvNnDOTQ+NQBYUSHKrAfnOf6mbeVlffcnUlHHpheZLdmLo68WIjfr9gCKkTxyO4vD2VsqPWRqAOk/ld9YbrgK4KBS6gpENp1D5zmVYIBgZtgounc3DneqYTg8IMxRnC9Hng98QBj5MbUSnnwMV7BNqjHrXzibYXbj+2x+mv+qwy5UtvbO123MOKvd8d7LxxjVIoDULdikXFbioeMisyeqYgm/47Vs4p3K0oDbi8uzVceFxaj8eKZBTQ7uQdsEIox6B1gBULHGtQuMwUpqLge+JeWhqbyO4vu+OYD5Q3DW0wPk72cHnnG2BzTbGAogZxP0zYDAslXEyIyYQCCWGLLsqL7Qi9LT7TS21jJnbu9M3xswi3gOXDkRUabi22ksHlzST3x4bXFZnRviJX7j2hcqPXieEUzZyMlnSt6HRo4XFg6QSzAmICsaFPaG7LRPWGgC6Lq4og/s4XqvcwDjKRN95x9568MI19wgzsWTFD30Z5XCqPWR8hzvr6/adUZtRaMXy91aS+W3uSxq2xHWLh8tYSxvf58wDN33NRvLNevaZ/JbiYgZWzPh3xSHJhCnAHw2PEFn/QBsEi2bNWXupWI4/o+t7jF1R0mC0USQ8jg7PxOPreir3fnXF54xqlUBqEuhWLiq0KPGS4vnf203f6BaE2x6Vx5qQVyhuPleuv8v/B0TixRXE7OS+78eWTqEGP5l/dsLVUDh3QSxsD/khOGOGEGba4sRWBbwiOQpReRO1wKFIsW76oJpJTBi05LLLdOXwBFltYiyDsJmUKIny2qXc1ypoqoVmwmNYvhxjwDbChoeKHLKAz4Tg+KJRtZu4SX+uJUTRt4A46lmPX8KzwzJyJHAcGhRJxat524fLvEdnmmAdX0TKLq88ZRvL3tx4+o5IjgsRHMWIQvXlDIjmtZ5jVyZW+vNGWq7Pg3ka9YzlsBIdYUOREVj0nqvnK1dLeLZHJXhJ29ec2L1X/xSH0287zInkO7yskuKAcLTwx3pIJa07QpLUnKSBTMlrazXZ44xLEYL3AGDkmGQolKucwRmgrl/01K6nRir3fW/i7el+lULqKWCTtrVhUWg48ZEpv0JmdKrvqOBnFStJacCmC0JkFHI2o5c21XJlSB65ws0pk8b3ZYsT3xOd6LWFyaTWUTUTWJMeiMtcm34eD5QMGr6IHz17a575vYC1ac+SqqAYij+GPjmUpMGcq0Y5JiOFWRQY4BJnkZiRMGfQTiLQbbXUZR9U35u+5IDKN4RZHDKUvlQLUTizfgBX07OVr0psoIseBgTaIM2m5zKbZ+Dvqf/SKozRto424GrRBHH6hVfS9MTZUTyk0eFWEWyPB7ZNqzmute2O82nt+s+q4IGOHGPU+48zmHrXyEkp4nrv1iKqM3SDuoS1jGhkGHKcMCiNYyOECl71UstUapRe11bu8ge33605GqJQkZ7fjMIeDOVva8Z50VKK24KCV9Ph5xJAiI+dixd5v5HjN7EsplAaha8WiYvJnHrIr8Yanrj+wZxuD2Dbo6HVBH2EVaS2fjnnsUMYGNShIFb9eRxfvPLE/Bbh84fo1U/jFKd9Dbw1jOb6rReksglqDY1H5pc/9cmxT4KgguxUS30F5RdY917Xm9qgEggoqELZGJ08YRyTsQPRSRJmJnTt9c+wvX+uIg1EmjIeF8uPZe0U1Ja7i4c59zbqGnyeyw4eG0ZNEdS97dm32lBQvTkxRKQrZ+2Bb8Jaw0oPMe1goUbdZtgJ5a1y4r2xV4nGYxY9rxjynbTxtL2VbPyA9TWnpeZb0u1O2Cu5aUJH1qpNPWOJQ6QmysXdVypbKOc/lz1vP0JClRyhzigTC5Y0Dvhy2wQYLhNm8fk206+xt+rhyTupbv4AZMOnqk9kf+B2rTUYC7yTCtzDm87ce0+V7T4VFFZZVljIjg+j6g2eEAxy8c0aLFXu/0WM2qz+lUBqErBWLSkuq7Io7BT+2ymPXi9lyXB4C8me1c+6yMwKiX7adtddxRn/8YmQKCL6HFckKbJ3he7qS3CQHeSOBZumByyImCZYdjhXkfld+Vonyp08qygieuflIfMwcjFAoO82x1S5nQR/oC8KccRx3iM8OD61DiXwoQUXvutDG/jpad8goRbWWRHFjidKLoOZx5bnoHYsR7bTWImd9ynFgsFBuD71FHSrmoAFh1CbOrjfje3YlIkEDpRd9rbxh7n7LBZk/i1ZJMAMTo/qUuUqbl8pCXzexkWx7Ily+kyuZHbt6n+pO3Cy61Ev+L1eNgcsbh3wkJoI9AiKXP0WlHCiwn1TLRb3r2DKmvSFsVWVv3PUHT+3cxahAhPcoao7jYAQL5c2HzwnJkM1LhyuOvMegMlrHyrkMn4YVe7/hgzapQ6VQGgSsFYvqpy1naPiyI/YRQzHUW45MTiaAJQKk2ZwYYxAEUXbDNbC5Ebuw6k/aTEeu3LdfC/qHcc3MpQfhbEe+KaoH6a0Qcu3+UwoMC4YHlyDcl3zK18YKcsyjPEdkLCNzWeaz5HHIZTT55S6D6mvxhHrXjTb2t17h9KTltmN+RyiRUChZuTw8rK7e21jWDpvYn3svihKamVM4j2nlKkD43SGGUkttYtnApRuxKxEuT1gofaEcpIwD8y7yZ/5Ux56LSGDsRpWylavYwCoux3LrJf+Xy++Cjox5cP8Iy57m9yLeZ1AoD126T5/VzENcDtYb6/TX7WdFrDnzE8sHelQnw3uDresIA7j/9CWBzq1Ricz24TacvNVe75uzv42cixV7v5HjNbMvpVAahK4Vi+q3neeo/+JD9hFXzZeGftZpYZQD3ZF5GHrzkaXUJbDkodwbCzbjdhVyELty+HO4JMyOLdNaemVyX2fLQabrYEtvqWwp6M/O5e3EwNwHCIPRt2y5ZPLobadu0vszbGUbWWRLAysh/B0slaj04I+iTchylOEsVyBCDKUvlQL0FHO5ClC8OLFM3dz0jpVDUPKmSywslCgH6coBVe993G1XcvgauvXouf1ysyp4uTu+qK5DoggnlXHMo6f34ThkEHiPblSEgs/foXenbBPdIiY7ZaK4Tm+xcO9F6rngACGMBr9B1OqumDs1zekQKK7lLHAcMhBDieIZ3o5dZWsvvzdPXntAtcJK3ILqDZ6NsauOC7o5xE9q6dcwL7n2txlhJlbs/U4fro80UAqlQQ/CikXFLwQeslzf1Nk05NgT/DgRU4I4mvEWUZesPXqN2v+yxz5Mdku0mL7dTouDL9tVyE6DG5gbW8ZuFB4MYoq2fKGPf+/B0xcUEFYRh0mXOSZQpgtB3+yilqsEsetm77nbdvJuHodcom3KhlMiC5MFyTtHh/uetc7ZusP3cpIC/s+bonytXCJtTJMi9gQdVCDyd5m+6TSNWn6McqVJJCiQDl++T0YpGu5i8+OmUBttTJpEQqHkxAazE+L0jldbvnB665KiSIA/iEwtxgdnT8fNVGewLI5vVox2nblNIFCH6K0AI9e1hkIJqi7ZKCFXq4KFEkUIzHIT68WDk/WgMENxPnjxrr2ULaje/t5/2UZ5lSaRiKFEmASyv8H/y9J5zl5acegqNSmZWRSPMFqs2PuNHrNZ/SmF0iBkrVhU/xy8Qp/MDY+7Y+odPVN4+uKVvSoN4tQePX9FSCr5qrHn8T167i9XGEF7tjjgJI8TPQsoMbjKgZ5+3WnDnId8LTb6tT2r6uoKlVWYVxLXnb7xiMDj98uHZUSNZ5yoWbiKkVwliOMtZZcVt5dJvLXhDbAq7A+ruatroD7UiN2rPCRHCSmyVYcJ432lFKCnUM7ccoaGLTtCyNiHy/vEtYf0Rd381Lmq8fFcesfKRPIYE7K85+w4H0G50NuPWe2qj9sgFBoWmVLLrHsa1a+cMDOuaVFqXDLc/eruPeTa6yDElz0cesvvymElUEzn7YpYq1vOAgdtEGISB/2vIH1YMYe7w/b4Ok7W46pUWkV68b5LIjYfLnFQKUFmtS1N1fKntd+bGTPMqgdvxd7vMZAWdaAUSoOAtmJRyVYcDNuRpSey6eAFAf4uWazkGJN5MDEGJvHu+OseWh1WCQafd6+Rhz6vldegp+K4Gz718reuELy/BjF3GI4ZksWnK/eeUs0C6WhGm1Iku3blhJIe8/fbyyeye1x23fA4To6sR3HC6kJzAD1/5wrnqKngudG51iLs6NAgV3Li0qC+UgrQjSlHuGT29rM0cMlhUdoQFkqEm3g7a5nHhI0YCqVWufB0zp5ej3J5IKlmkSm1PO3b7Otl0nGjLKtf/HmQ/thzQRQ+wLtTPoDJ742o5sZhJQifgYUSccCyl0uOV4TLG+82bydDcagUe2g2nbhB8PhAMO8Fey6Kg3yKhHHoThgbxpz2gVQxT2o7FJzgaBariRV7v9lr1qj+lUJpEJJWLCpt3J2r1CNMCMtTbl8xBw20KNP0xLUHVDss9gX3Rx3vCrlTi7hKvDRY+tTNR12qmss3x64fvqcjXsSolgVnoCKZB0HgTA0ic23iBRccZlGUk1KYl07Ouse9OPs7BpiciUir9PpKWTx3fi6svPC1jgL9d4TespdbRNwW4qL8ec4yTr/vOk9fLgoRZfhgoQRN1tC3C1Gb8tndgdOQa5hIHgcVuLx9ob64PDFtbDWqZ4HVwh8EXoxiQ9eIeD69VWyczUsmx4c3RA4h0kt5JhskmItYZtXgeEWsCVTKufnwGX3VKIBalDGeasfZfPn7lYeuCDYMphCTlWLElC/Yc4F6/3mQ4saKSc8xaCLSHj7kZCM9la30jo3bWbH3uzomb7WPVgrliBEjKCgoiDZssBG+Qvbv30/du3enPXv2UKpUqejTTz+lXr162b9//fo1DR06lGbMmEF37tyhihUr0pQpUyh3bteUGisWFWpdN/7BFogNcdU9XGDgSvGSc/d6TxapnJ2HfpijUUspAwUXiq6Zwi8lvoeryQiMI15yiNlhgms57pGzEnEPrkeLv9ntImeL43NtjKRW6bWy7rrR2GuVY7mSBd9LjinlCkRGla0zej6u9gdLENY5whZAG7KGUeoAACAASURBVHT1/lMa+W5hahkYzpXnap+etpeJ5JHljfribxXJQJO9WF9cnpM2ttqsOsye4hjZ9euPX6fLd5/Q+2WyEh8SPbmXXNloXseytPIQaMdc42qVrXt4D6Fik5wgx8oZ4hXh0ULJS6Nc9u7OXas4IwkH/L0I2wIDhPY9Ke8tfE/2kJj1PrFi73cXP6uvizYK5cSJE6lHjx5UuXJlu0J569Ytyp8/P73zzjviux07dlCXLl2EwtiuXTuBNZRJ/H/WrFmUKVMm6tOnD4WGhtLhw4cpblznmXP8wKxYVIcv36O3vt1iXyOuUjpoa02j4gKSA6wQnHZRFYeFORq1lDJWbLQbjl+ntrN228fiSrY8LtLiyMHeTByMNrICyJx/+Jxf4HKNWXyujZHkDYMHiRKM/3xqTd11o9cDc0xyv0MaFKS2FSIeGuSQCK41j1rrvlAK0FM85GQIWCjBlTemcRFqVjqLp127ff2ifRepx/wDhFrZUCjhJXAlJtvtG+u8sO2sXaLePcuGXlUpe2rn5N06u/e7ZmNWHqMpEm8kUxNhPR0bXk/XfOQ4djxzKJhNS2YWNF2Qv4Iv0Wd/7CfEKyK0B3H2qJfdsFgmXf2b0Ujr2kfMZJ+FtopaSNjjuFD53su6VaTCmZLZP2IPgStsHq7MxYq935XxeLOt3yuUly5dog4dOtDmzZspS5YslC5dOrtCOXr0aJo8eTKdPXuWYseOLXDu168fLVq0iI4dO0bPnz+n1KlT05gxY6hTp07i+7t371LGjBlp5syZ1KJFC93PxopFJVdpwcD61stPH1fRH9jPZQ55Ur1q56Wu1fPonqMnDR8/f0kFB4WXU2OORvBqIgGFBVl4UNDMlO2nb9F7P+6w38LVYO1SI9YIpYCFy0jKXJvIAP+7q632LGf54u+WgVlp5LsBJCdJ4XNtjKSWp9JVt7yZ+Lna9+rDV6nj7L32yxy50eQkJVSg+WX7OUEFsvKzyq7ezufa86YXN3ZMih87pgiTmNC8KL1b3Nx1HhUQMpE8lAtkwVrBAav34ciJbLjGX8uO6p2vs3ZMI1Y4U1Ja1q1SuPIXLzaFDK3j7HLx/e6zt6npVFtmOEpuohIOv7vwGSd9QklFpRy4kH9oWYLqBWTQ1b8ZjbTJRwv2XojASxl05Bp1+DWcPQRjAD9l3nRJ7MPBuwVeKVheQVVntFix9xs9ZrP683uFcunSpTRv3jyCu3vYsGFCeWSXd/369Sl58uQ0d+5cO35wideqVYuuXbsm2gYGBtLx48cpb95wSx3c3kWKFBGWS71ixaKCC6X8V+vsQ3I1Dqv86LWiNBWLlZQQcjKLeLmFcTTyyZvH9N17xQkVaMwUmdYD93m7aEaRda5Xyo1eKwLWWbgEn8y1GZgjJTFhsJxVzkTHWjyQbbupTzV7n9p4WVfd8nrnYkU72dWG+01sXozeKR7R6oEEDCRiQFBdBAkIvHlaMUYz77HmyDX66Nc9Ik4WSTlm1hXWOw854xcKJcZoJeuDs3F2/z2YluwPj63eP6gWJU+o32PkrH9/+57DacAbuvrzKvbYQabT0TMfmbuSi1t8UC4bDQsrHyrHJyIp582b8ORJPf2b0Uab1T1v53lR1pKZOWQLJt9/fa+qhHrlVokVe79Vc/H0Pn6vUMoAtG3bNoJCCaWwXr169PXXX9ubHTlyhAoVKkS7d++m8+fPU+PGjenx48eUIEECe5tmzZqJz5YtWxYpvvfv3yf8Y7ly5QqVKVOGLly4QJkzm2N5kEm1cV9tiSlni6Hq2PV09tZjezOjONKc3Ze/l2M4mSJnUtBJmhB0wt6FUVmRUY1JGzogu330zKXymPWCUoOFy+jJ8T6yG13mD5XDDPL2X2EPJOeNgvvU8lQyNZGe8flaG3lTwNimtipBdQtHtHrIteY5YQB1zVHf3N8F8XTtwkIsuJTm1FYlqW5h7/EqstUYccBQKNcdu263nvsC3pzVzGPRS43jC2M3YwzaEoScVOUKtZbsBUA4ycnrD+nDCjloUIOCYsiOqnf93K40Vc0XTsFjxtyi6lM+/O8dUFPQW2G/4BAgLR0d+tryRTVdFayMmotSKMORjNYKJRJr3n//fWG5ZEF8ZK5cuYSLHBbK1q1b06tXryhmzJj2Nh988AFdvnxZJPhEJkOGDBHxl1oxU6F88vwVFRi00n5LV+Nbao3fKF4iLFbEK8r4lBi+hqAUQ5Chh82VyXS5HTIYoTyZKTJPHO7TqmxWGvFOgO5bauuPg08QvIKyVZEzv9EpZyribzlMIWDwKnrw7KW4LywGS7vZXOSQkIv3qMH34fGyTBeie5A+1FAmI8awZrUrTdU0mxRqnaPmOQRUJisPXyWmWPKhqbg1lC0nb1KrnyJWRfqpTSmqUSCdW/0ZcZGc8Qvmgc0nbxJb2o3o39M+Bi85JMIeWJjT1dN+/fV69nIwlZZM+4QKW3pErv8NBoULt5/Qx5VzUt/6BcTlm0/eoNY/2Sh5WJiNQ0//ZrSRleCd/WoIAn5UeeJ3w56zt6lJmBuf76+3FKVR41UKZTiS0VqhDAgIILi9HVko9+3bJ5JvmjRp4tBC+ezZM1qyZEmka84bFkqtm1RbEcDZD+StbzeLKh0sqEjSrJR1iQFc/ULmaGSCZR7T7x3LUtmcqZxNxaPvtRnnrtIn1Z24SZQlY2Gro+xSkuPRIpB2NyxErcvZ6GLkWEyt8qSlWZLpPTyavBcu1s5l3kdlqVyuiM9YfibV8qWh9cdvkBw24IVhG3bLnaG3qPn08JhddGzFwSmqCchrskyOlKLyiqu/A8MActCRzIyA2NMTI/Qlnpg5Jm/2zfHZqRPHpT0DahGT5WdLlZA29g4PlYlqjPJBGhRWSJT8pFou6l0nv7hMpu7ifuZ/XI6wPrwlshIMyyPi7eWSkQcu3CXU6pYleGAtSqGjFKVRc1IK5X9EoYQyCaqg2bNn22fsKIby1KlTwmrJghjKokWLioQevWLVoso7YAU9f2nj23J1U3pn8lZCNi2LqxZOvVhE1o4tpDJHI3Of8TWLu5Sn4llTeHqrKK+//uAplRm51t6mS9Vc1Keu7aWqR97+fgsdvHjP3pSTm45ffUB1Jm4Sn8uk8XIwvJx0JJeX05Lunrv1iKqMDae/8qWECT0YyW20nJuOnrEcH4xSlttDb5FZRMSujt/T9lq6L/Q396NAKp8rnHzZ03u4er1sTS+aORkduHgvgrXK1f6Mbi9zunKVFKPv4U/92TOww5Jw5HKeeqt8ye+UxPFi08NnLyMUknBUDpbp3byFlawEI9N/2qbTYST8aWlGm9KC/J5jr3mMXPLWqjFbtfdbNR9P7hOtLZRfffUVTZ06lU6fPk2xYsUSOPXt25cWL14ssrxhhUybNi2NHz+e2rdvL7735SxvjE+mrHG1egRqv8ISwTKlZQmqb2EGX8Pvt4iNS+Zo1NYnX/5pJSqYMakna9rpteBXKzp0tb3d5zXzUvea+rPdwQUKJYGF3diylU12Jckxm3LWZPVvNoiqKZDq+dPSzLal7X1eufeEyo0OT8CSszGdTtDHGsiVQzC0Fd0rEUpQyiIr+choDz5/117S0sem4/JwtOEL6GBBp3JUOrv3LD+y1TRfuiR0/NqDCNYqlydp8AWT158S5PYQs+heDB6yqd3JTAGw1jI+rjAhXLzzmCp+vV6MEwlir98Q9ayVl7rVsL37tKEp+ExLwWPqJB10LivBQT0q0+T1pyNwpmrDl9AF8IFV2ypRCmU40tFaobx+/brgoXz77bcFv+SuXbuoc+fOQsls06aNQKF///40bdo0QROUPXt26t27t4itDAkJ8TkeSow3cFQQXbv/TIz9764VqEhm/dUjWs3YSVtO3bQ//RkflKKaBa2L42KyYpmjUVuffF3PKpQzTWJT3wVyPW7cyFX6JS3pMte7lbk2ZY5QOT5QDnKXy8vVK5yefmhV0j5vbQKWq1WRTAXQxc61CryjLEx5viiFeeTKfaqRPy39JCnZLt7WZ5o7sqIg2QhJR94S2WqaPVVCkaxnJS+ts3nL5TqjS8UkZ3OO6nstwfektSdpYtBJl5gQtMUUcD+5pryjdeptQnnU54YnB4KD6LdrTwqKK+b+1Xo/0E5v5SBPnod8rVIo/yMKJaaJbG5UxwkODqYMGTJQz549qWvXrnYEkJADbkoQmz958kQQo8PVDeXSFbFqUVUZu57OhWVqa/m2nI33w593i2xOFldd5s76d/Y931/maNTyiCHAHBZMM+XNmzeUo294XXNHRNtR3b/1TztFEgML17uVuTZlSib5RS7HJMkhCFpS6UfPXlKhweG8nR9XyUl969mC5/1NtAr89r7VKUOyiM9YVjpBCXL6xiOqXTAdTf+glL9N91/jdWRF8bblR449S580vqjegyIHUCp9QeRa9v5cJcooLOWEGVjgoFh9v/6UOJToZUK49fAZlZSKS2Bsck15mWmBx722ZxXKZfIBPyqMZM8FDCgoowjCe06k1Hpy4sSKQSdH1jcKdl39WLX36xqMlxtFKwulN7G0alHVmbBJuKcgTA6ud94fz94jym2xOEqO0NuXO+0++W0f/RNyRQR5Q7GCaDML9wyoKWoemy35BqygZ2GxqKMbBQiCX73S/ufdtFZSzJm+SU6aGi4l36CMGRKiQIa+rlcVSho/jrhV82nbaWdYCAK4F79uUsQ+hBevXlOe/ivs//+0em7qUTuf3iH6VDso8Dn7LRe8dhBHQfOyMo4DBSwTbwVkoMktS/jUXNwZjDYeFn142/IjZ8+iStPdxy+oT9181KWqayVn3cFDzzVcrhJtowsfqZ55R9ZGDlEA5dp3607StI2hVDp7ClrQqbyurrWeAlwkH6YdrdPNfapRlpQJdfVvRqM7j55T8eFrRNcLO5ensauO0Y7Q28RUbdoKbAnjxqIjw+qaMZRI+7Rq77d0Um7eTCmUbgKnvcyqRYWMNlgXILv616C0SeLrnkG3ecGixBrLws7lqGQ26+K4uG63zNGo5SgMGVKbkoQpXLon5kZDORZ1fLOi1KiEfu5QbRUP+fr8A1fQ0xevSVvxB0rly9evKV5sWywv5IOZu0T5M4iWsgVKWK5+y0WcE6R3nXz0STXf2OzdgJtkDlJHnIKyFROZrFC+XSWcd2dcVlwju+34flaEdkQ1NzmBDJVRsGatLHTgDHe5SEDJbCmEMvFfFplBAnyMKMOIjGcksKG2tx7RVivDNSPeKUytytpqyjtap468CXruZVSbB09fUMAQW7w7cgZAao7E0m7Vc1PP2vno/tMXVCTse7TRlrA1ahxR9WPV3m/FXDy9h1IoPUUw7HqrFpVs1To4pLbd2qVnGj3m76dF+y7Zmy7tWpECMofXPNXThydtUImh1/wD1P+tAtQizCKopX2wKqC69MgguvHAFos6+f0S9FYR/eXFtIq5XN2n94IDgvLmr0/KOyXXRfUUVCiBoH51vzA+OMaYlVP8v3/9AvRR5ZyewO/Va4sPW013Hr8QY3DEKQiFGwo0JEm82IKfs1HxTDS+eTGvjtuIm2tZBdCnty0/jtzwHAtsxJw97YOrC6Gf6JLt7wkmcmIflLypG04Lnk5wiM5uH6ira7CDgCVEFrk4hqN1apXHKLIJyCVqwYmJUr2gbOMDtraErTcSuKza+3U9ZC83UgqlQQ/AqkXVZuYuAoccxFXlq++ig4JygWXlZ5Uof3pzM6q18MItHBMphmEi84zFiGFTNmLgD5Ol4tfr6OKdJ+IuriYnaRVzbdUT7Rwjm0rXufto2cEr4ms+ccttkYkONxVkWMNC9EEYf6XJ0JjSPZerjIxTUI5rjRsrpqgg5GoFI1MGbkCnstuOu/N2bWpH7k1fWmNyKEx0Sc7yZCnJB4CNvUGfE0qgXHMFG23sOMYzrmlRalzS5p1xtE4PDKpNyRLaQnS8IS9fvabcYaE/SGgc8vdhkUDGsZ/yQRTjkxlErBqvVXu/VfPx5D5KofQEPelaqxYVx0FCJ0O1GVeUr0FLDtGvUvUJb7vdAJ+cAQ3X27Hh1hAYy9VuZrcvQ5Xy6K/O8+XCg/T77nDFfGbbUlQ9v+vZ8j3nH6CF+y6KVcRclvJyLDMyiK6HWVFdLbNp0LI2rBtUwcGzTho/Nh0cUsdhv3n6L6cXr8J8/EQirhXxrf4u4PsrLCVYYT7etvzIFDKMr9WVs6J6rnIojFx1yt/XgrvjlynJ1nxemX7cHErz91ykOoXS0bTW+hPX4AWAEsYicxE7WqdHhtWhhHFjuztsj6+T469//KAUDfzrkEggk131Ofv+Yw8NAmPBBp1E7x4PLqwDq/Z+o8ZrZj9KoTQIXasW1We/B9Nf+y9Torix6LCLwccjlh2hGVvO2GfsbbcbBiITWidLEIcODK5t0BOJupv6kzYLahqIq9Ug8FJDKTQWVxVSvq7f4hBhZYA4cmnLVtQJzYvSu8X1x3laAqILN+HqQlHVHpZd/OhaJod34VY+11TrlsMAsc6x3r0lV+89pbKjw8n9MY4xjYtQs9LWVc6Kau4yJ2J0CX3w5FnLTBFgCEClnEXBl0SoDkJ29IqcjIhrZC5iR+vUVS+Y3nG40i5v/xXCYwEO376LQ0QCmWxZleeEGuVrelRxpXuP21q193s8UAs6UAqlQSBbtajYOpYqUVzaO7CWS6P/euUxUQeVxeqap44GK1NZRKVsuDRRHY3fnbJVkGdDlnxSgYq6wAk4bOkRmrk1XDF3t1zk0KWHRRkxiCN3o2xFtZqEXgeELjVhiqSsKRPSpj6OS8XBigcrCYs/c2/K4GjdcvjO25YfbXYsxiRv0i49XBMay+U6o4ul2hOYtNnOqGmNxCUt3Zize2h/Y9Nbl6TahdKLyxytU6s5HR2Nv+CglfT4+Sv69r3ihBh1sHPIce8Bg1eJmGsIOGyXd6/kDAZDv7dq7zd00CZ1phRKg4C1alEhhgQvE3diRSasOUEgxGWxuuapI6hlvsWolA2DHpO9m/em7xDl/SCuUriMXnFUUHawIAMVmaiuiqzgO3Jpy1bUn9qUohoFXHeruzoms9ozGXzedIlp9eeOLQhyzCjGwdQgZo3Jyn5ltxzu623Lz93Hz6nYMBsdC4vVpVijwl+O8YwuBwtP1pv8ngTd26/bz0Yg+Nbbd7Fhq4WFj2VW29JULX9a+//ldRo7Zgw6NcpaTkdH82BGDhx4ei44IJrI78OSw9fQrUfPxeeu8HLqxcxZO6v2fmfj8IXvlUJp0FOwalF9teIYTd14mkD+rLeGK09RLmeGz6yueeoIajno2kp3RdtZuwRBLgQ1YrOnTqR7JYxbfZy+W3fK3t7dbPmJQSdEtQuIo81cJj53162ue1ImN2S8AzIlo6XdKjq8W6kRawRdEEunKrnoy3r6a6ybPAWPukd2LbJsWbxt+XEUL+cq24FHgDi5WHbxRqd14C5m8nsSBSlmbz9HQUev0XtlstDoRuH8tc761/7Gfv2wDFXOGx4/LruPE8SJRUeHW8vp6Gj8rDDCizNoyWHRBBnfFXKnFn+XHbVWxFVCymRPSfM72TiOrRKr9n6r5uPJfZRC6Ql60rVWLaopG07RmJXH3TqJ/bgplEYuP2oftbetJDwQDhS3ksBYJnl3lWsNVSpQsYHF3Wx5hB/ASgmZ2qoE1S0ckbpIpohyNc7ToGVtWDfM3RkVEbNcVhQ37lotN/Wq459k7lrgZFejL1h+HMXLTWtdkuqEuT8Ne/BudnTv8QsqOszGPyiXMXWzu2hxGVsPkZyCSkJg+3A1zlhWvgDK3I8CqXwum2IGkddpkvixKSSSBDorAeUxgyqI67vLHMqVxqyjC7dtjB3eoJiyau+3EnN376UUSneR01xn1aJCMD2UkHeKZ6Iq0slSzzR+3nqGhiw9Ym/qbSsJD4RjZKwkMP50XjD9HUbyvm9gLUqZKK4eCEUbWRHE/90tT4bA+mHLbM9D63rCZzLxuatxnronY1HD7r8H05L9l6PkzUPNXpArs3SvkYc+r5XXohGaexvZ1egLlh9tJSbM3pfCKmSFV643be5T8u3eOWnt+/eL07xd52nrqVvUrkJ2GtygkO6By4l+uGhBp3JUOnt4cQuZL9adOH3dA3GhIY8ZXL3TN9lCjVDXu0AGG+Vd9XEbKPTGI/G3XDTDhVt41NSqvd+jQVp0sVIoDQLaHxYVMoqRWQwB19+JkdZQ9DiDmF9i5XOlorkf6av64KxPZ98juHvBXhtlj6uu/xmbQ2nEP+GWXnez5X/beY76Lz4kxjC3QyCVD3Ph8Ng7/LJHuLW0L1Bnc/PF7zn2t0HRjAQieEdSecx6On/7sf2rnrXyUjcfqS3tKaYykb4vWH4ccRKC569qvvB4Ok/n7Mn18vgGNyhI7Srk8KS7aHEtxxKiMhdoy0Ct5KggQlSTZfoubrO4S3kqnjU8/ltep+mSxqOd/Wp6HTsec4vSWex0bXLZYWaQwEBrFUxHsOBaKf6w91uFh88olE+ePKFYsWJR3Lhx6ejRo/TPP/9Q+fLlxT9/EH9YVAv2XKDefx4UcCaOF5tQE9YXhF0a1fKloVntylgypAF/hdCcHTbKnlMj61HsWDF13xcB8RzLg4vcJamW6xU7KoP5ydx99E8Y8bkvcIbqBshBQ/DooVTc+4FZKW+6JA67ki0NaBCdLFPlR6+ly/dscV6whsMq7m3RchLKcWneHhvuz/F84CJFpvd/XVjZ+6pRAM3fc4H2nb9LXarmoj519ccZ1xq/kU5ef2iHEhREhTOFV0uTvQSZUySgLV9U9zrsPOZ6hdOLRCSIzOP69vdb6ODFe+JzV2mUjJicP+z9RsxTTx8+oVBu2rSJ3nnnHfrzzz+pYMGClDdvXooZMyY9fPiQ5s2bR02bNtUzF6+28YdFtWT/Jer++36BU4qEcSh4kDWcj84eTJWx6+ncrcdUt1B6mtq6pLPmhnyPEl5QcGLFjCEI4l2R33edpy8X2Sy9EFdd5nydXK9Y+2JHG7kiz9Yvq4vM/ugstSdspBPXwjc7fy83KT8r2frqK5YfLSehu/RXZq1JVm5kahuz7uUP/TIeSE7BYRRK1Kc18lAPF8JC6k3aTEfD+HcxZ238N7+L8V2O1Ilofa+qXoeGxwwP1rbTNmaOo8PqUoK4scTfjX/YRnvP3RF/v1s8E02wuFyrP+z9Vj1En1AoK1asKJTISZMm0cyZM2ns2LF08uRJ8feMGTMoODjYKjzcvo8/LKoVIVeo82/7xBx9ZVPDWFiRaFgsI01q4dgd6vaDieTCMSuP0ZQNpylh3Fh0xEWC+IV7L9rpK9B9yJDalCS+6yTVqw9fpY6z94oRovpFHo3lTi6Vubt/TUKd2ugs2s3Ol2pLe4q7zCnqDuWXp/d3dH2hQSvp0fNX9q8cWcnNuK/ePneE3hJcse0r5iCU7PyvS/VvNlDozUeiCAJIzaEYOqqwFRVOsjUP7YJ6VKHcaRPbL6k5fiOhzCMkKoovK58FjxkckyhGoS3Ry5RkGFOzUplpTJOiVg6P/GHvtwoQn1AoEyZMSIcOHaKcOXNS/fr1KUuWLDRt2jQ6d+4c5c+fn+AO93Xxh0W19ug1av/LHgGlr7gzMBZ+YTQvlYW+bqKfAsOTNcGZ2u64H5HMg6QelmPD61L8OLbTsiuy6cQNkXgDkWOCuA+OO8T/Dw6pTUndUFpdGY+322o3u+ENC1FrP65fLuMpx3n5iuWHY/J4nH99UkGwRyjxTQR4DSHb+a/gS8J1DVot0CrplUZTtgpXOQvqgmdLFU6ZJh/qCmVMSv98ai1JuKN5sAUSBzEk7WmrxLX+aSdtPnlTXNoyMCuNfNfacq3+sPfrXR+etvMJhTJNmjS0efNmypEjB6VKlYpmzZol3Nw7d+6khg0b0tWrtrgJXxZ/WFSyApMzTSJa19P77gw806ZTt9Hus3fog3LZaFjDwpY85mkbT9PoFccoQ7L4tL1vDZfuufLQFeo0x2bphcBlDte5q7Iz9BY1n75DXOaoatHo5UdpWlhW4/ERdSlebNeVVlfH5M32cvUijGPUuwEi5jI6SIPvtlDIJVucl5V8q1FhJxNCo52jsIvogH10mQMfuODmRrjMmZuPaMBbBahDpZy6p9hs2naRzMOiDaWRD3WoHgZ2CW8L06chme3B05fCUwOPDUv7n3fT2mPXxX+9QYLvD3u/Vc/QJxTKZs2aCStkypQpaeHChXTlyhU6ffo0ffzxx8JqiThKXxd/WFRwIbUIU2Dyp09CKz+r7BOwcn1yV0/bngx+2cHL1HVuMJXKloL+7Oxa4pds6XUnBpPHjUSVSmPWiwQpBJlrrZwygbqvUDx5grmza/lgwe18qba0s7E7+15WlkF3AtoTb0uZkUF0/cEz+zDc5VP19jz+K/fn3wcsknh/XbzzhIa+XYjalM+uG4KWM3YIuiEW7UFWjkeMijNW9w0NaNhqxk7acspmgYRkT5WQNvQOL9/KHLf4ztWsdwOGp1zeEog+oVDeuHGDOnXqRKGhoTR48GCRoNOzZ09hoUSiTvr0tlqjviz+oFAicBkvDEiRzMno766OK5ZYjTPqeSPYumaBdPZAa7PHgMoTa45co4DMyShzioQu3U629MaPE5OODXeffmnrqZvClY1xaOX7dSfpm9UnRPwYSOiju8ixUJirL9WW9hR72TJUNHMyWuIDvz0t72dQj8qUO63jDHxP56+u9xwBVgY/rJCDlodcEdVhRr5bmFoGZtPdeZuZuwQhOos2oVAuplAuZyqa19EaGreoJiBXNUM7bb1umVPYG8UQ/GHv171APGzoEwqlozk8f/5cUAj5i/jDogq5eI8afL9FQGolibi/PEO949x++ha996PNVW0mp+D0Tadp1PJjlCRebArxEYonvRi5006OhcL1vlRb2p35yNfIliFf+e1peT9dLUHqKSbqetcQaDdrF60/WQaCUAAAIABJREFUfkPECa46fFWUKXXVii+7h3H3A4NrU7IE4QmF8m8QJRlRmtHbIvPxYixay2nP+Qdo4T4bp/DnNfNS95p5LB2yP+z9VgHiMwol3Nw//vgjHTt2jCZOnEgbN26kgIAAkZTjD+IPi+r41QdUZ+ImAaevnD794dlqx7j33G1q/MN28bGZ1SR+2XaWBv99mFInjkt7Bnift9DsZ6W1RPhSbWlP5y5bhsrmTEm/d7S23rCj8Wt5P90l6PcUG3W9PgS4XGzTkplp9ZFrdO/JCwLJeaMSmfV1QERyyVlcdGRYHUoYN7b9elZa8UGN/Gnpp7aldfdtVsMuv+2l5SHheRSoEId65iwyG0afuvmoS9XcZg3FYb/+sPdbBYhPKJSnTp2iwMBASpYsmYhHgFL5xRdf0IoVK2j16tV+QW7uD4sq9MZDqj5uo1hbvnL6tGqhG3mfgxfv0tvfbxVdpk8an3b0cy2pR+9Ytp2+Se//uJN8RQHRO25323X4ZTcFHbUF10OmtipJdQv7friLnvnKVpZKeVLT7PaBei4ztY2W99PVmvamDk51/i8Eus0LFsk4oFdDuM7j56/o2/eK09tFM+pGSy6WgIsQSiNTMnX8dY9QViFW8gJHNQHZpY12IDj/oVU4X/GgJYfo1+3nRBeuJinpBi6Khv6w9xsxTz19+IRCiUxuZHrDQpk0aVI6cOAAZcuWjdq2bUvnz58X1kpfF39YVBfvPKaKX68XUCJecUYba0tU+foz1Ds+8L+BXgOSJWUC2tzHvGoSsCqD4ilRvHArgt5x+ls7rfVkxgelqGbBdP42DYfj7Txnr73KR/X8aWmmD1h+tLyf/wWuU39eTL0WHBCE5lCo1h69Ts9fvaYfWpagegEZdE+LEyD5Am2y3ye/7aN/Qq6Ir/9XJAN9/34J3X2b1VAu8IB7NC6RmcY1C+ea5CIV+M7VJCUjxuwPe78R89TTh08olFAmoTSiSk6SJEmEQonsblgqYbm8d89Gt+HL4g+L6vr9p1Rm1FoBY/2A9DSlpTVVaXz5ubkzNhD/ggAY4kv0S+7MxZeu0VpPfKm2tKc4yVaW2gXT0XSL6w07Gr+W9zN4YC1Kkch/4tY9fSb+dn2/xSE0d+d54Yped/w6vXlDom416lfrFVZK0d4RQ4WscDYqnonGW1x1xtE8vvjzIP2x54L9Ky293Ncrj9EPG06L771RptMf9n6968PTdj6hUKZIkULwUBYuXDiCQoks73r16tHt2+G8WZ5O2Kzr/WFR3X38nIoNWyMgeKdYRppoUVUaszD3Vr/nbz2mymNtll5fol/yFh5G3bf778G0ZP9le3dz2gdSxTypjereq/3IVpa3AjLQ5Jbet/xoeT/drfjkVWD/QzfnQgeIf98eaqP+mdWuNFXLl1Y3CnK8YbzYMem4hj1CVji9UXXG0UT6Lw6h33aet38F2iRQzLGMX3OCUKgC8k3TotSkpP6YUt3ARdHQH/Z+I+appw+fUChbtGghxvrbb79R8uTJ6eDBg8L13ahRI0qbNi0tWLBAz1y82sYfFtWjZy+p0OBVAicEdo9tam2JKq8+IANvfuXeEyo3ep3oMSBTMlrazTfolwycole60rq25n1UlsrlSuWVsRh90y8XHqTfd9usLL5ymGs2dTvtOht+WJfrIxs9f9Wf5wiMXXWMJq8/LWhzUIIQ4uqha+Bfh2j2Dlu8IfhvD2nYI/ouCqF5u2zKmzeqzjhCSa4Yhu971spL3WqEZ3IzvRq+8wYzhD/s/Z6vPn09+IRCefnyZapatSrdunVLuLeR2Y2yiyA637Rpk4in9HXxh0X14tVrytN/hU+9LHz9uToa382Hz6jUiCDxVYmsyWlRF+9Xk/BHHLVj7vPnAZq/x0b/AVnQqRyVzp4yOkyNBvwVQnN22DZqWFBgSfG2vDd9h93ShbGcHFmP4sRSNbO9/Vwiuz/TiKVNEs9OSP9Hx7IUmFP/oWvo0sM0a+tZcYvkCePQ/kG1I9xOTnDxRtUZR3Mf+c8R+nHzGftX2sQbrnqGBq7GlBrxrP1h7zdinnr68AmFEgN9/PixqIgTHBxMr1+/Fu7vVq1aCUulJ/LixQsaMmQIzZ49m+7cuUPFihWjr7/+2p45vn//furevTvt2bNHlH389NNPqVevXi7f0h8W1Zs3byhH3+Vibr7ysnAZaB+4AHQdRYeuFiMJzJGS/vjY+xQwPgCLx0PgGDHuaHGX8lQ8awqP+/WFDmQry3tlsopYL2+LlvczdFR9iulGCVFvz+O/cv/5uy9Qn4UHKXbMGPTy9Rsx7YWdywtOYb0yavlRmh5WzjV14niiQpcscoLLR5VyUP+3Curt2rR2cowkbqKNk5y55QwNW3ZE3N8biXz+sPeb9nA0HfuMQolxPX36lE6cOEHx4sUTSTlx4oQTrroLCCrvTJ8+nX755RfRJ5TJ+fPn09GjR8V9YA1FZZ4ePXrQjh07qEuXLjRlyhRq166dS7f0l0WVd8AKev7yNX1cOSf1rV/ApTmqxjYEnjx/RQUGrRR/+woFTHR4NrJ1BPNZ2rWiwwpC/jhXeSO3smZ9VFjJnIPQI0NHv+WP0P5nxrzy0FXqNGdvhPn+3bUCFcmcXDcGY1YeoylhCSwZksWn7X0jUp6NXnGUpm0MFf11rpqLvqjrfR7o8auP07frTtnnqHVrw4UPVz4E/JTgqbRS/GXvtwITn1AoX716JayCU6dOJVTIgSRIkEBwUQ4YMIBixIjhNhawSNaoUYPGjRsn+rh//77gu0RJRyivkydPprNnz1Ls2DZaln79+tGiRYtEhrkr4i+LqvDgVfTw2UvyRokqV/D05bYo25g7LHTAVyhgfBkvvWOT3XG4ZvmnlahgRs88FHrvbXY7eSNH6bxBDbxv+ZG5Mf8r5T3Nfs5m9i9X6OL7uPobkZUz0JFt+SIi5dm41cfpuzDl7dMaeahHrbxmTklX30i4QeINi9YKyZZbfO+NuGt/2ft1ge1hI59QKPv3708zZ86k4cOHC5ogKJjI+oarumvXrjR06FC3pwllEg981apVlCVLFpowYYJQUg8fPkzdunUTSUBz58619x8UFES1atWia9euiYQgveIvi6rE8DV0+9Fz8aLAC0OJ6wggdCBnv+WCtqNOoXQ0rbXi83QdxX9foY2VWv15ZcqbLnrUlp6w5gRNCstE/bhKTupbz/veAZkbM2HcWHRkWF0jHqPqwyQEZP5bvsWazytTHhd+I7JyliN1Ilrfq2qE0U4KOkkTgmzKmzb5xaRpOe0WlEBwe7PM/SiQyucKZ39YHHyRPv/jgPh6YedyVDKbtXHX/rL3OwXagAY+oVCmS5eOZs2aRfXr148wpYULFwqFEmUZ3ZVDhw5R8+bN6ciRIxQrViyKGTOmsE6+/fbbVKRIEUFLBDc4C9oVKlSIdu/eTaVK6VcU/GVRBY4Komv3nwlXBlwaStxDgEMHfIX8171Z+NZV2lipdT2rUM40iX1rkG6ORs5E9RXvQNe5+2jZQdu7NWn82HRwSB03Z6cuswKBy3efUPmvbOwSLFAIoRjqlSkbTtGYlcdF8zxpE9OaHlUiXCp/D2oeUPR4W2ZsDqUR/xy1D+OvTypQsSzhbv5lBy9T17nB4ntXQwCMmJu/7P1GzNVZHz6hUMJKuH37dipQIOKpHcodLJYPHjxwNo9Ivwfl0KRJk6h3796UKVMmUY3njz/+ENnjoCV6//33adiwYfbrQ0NDKVeuXMJCWrFi5HQwcJ3jHwuU3jJlytCFCxcoc2ZrebBcAafaNxvozM1HNKRBQWpbIYcrl6q2EgIcOuAr5L/R4eHI7jbMZ1PvapQ1VcLoMDWauvE0fbXCZmX5vGZe6l7T+94BmcQ6ZaK4tG9g9K8X78+LSaZ943m4Wn/9x02hNHK5TTkrkCEpreheKQIksvL2//bOAzrKYn3jD0V6b4p0kmAoSSihYxSRKiDSwUu1UETwUqJBKYl0roDX0BSBK4gICiIglqAixQgYuCD8kV4EbgAhoBBa4H/eSTZswgK7m9ns7O4z53juJfm+95vvN8/uPJnyzqjWVfBCI/f3EQs3H8HY1cmbbqSkH5X9Zs//0G9R8tpSeR95r8wsNJR3aBthKGWX9ZkzZ9TGmRw57pzU0LdvX7W2UTbVOFMk9VBAQADWr1+Pxx+/88GR/1+sWDHIGeIyKmprhDIuLg41atS452NlOt7WVLzphnLZ9hPqPNipHUPwSMFczmDlPQBqRH2LC1duoEtoGUzuGEwmGgjMiNmPGTHJCYqlbH7jKZQqlFtDZPeH+HDTEcgOWikjmj+GVxr7u71S1kmsi+fPCTl6kcVcArLURmZGbiQl7/CWEhvRxKHv8QWbjyAyxZwFly6ILwelHTSxNm9Rz1ZFz/rl3Q5kcewxvJWy6cbW98IP+86gz8Jtqp4xQ5+Af4nMndWgoTTAUD711J3FwDdv3sSmTZtQsmRJ1K5dW01NS/ogMWdyzrdMUTtTZHSyc+fOSExMRK5cd8yTjFZ+9dVXKr+lpAqSlEKWYu8aSk8doXSGI++5m0Cd8TEqF9w/6pXFuHbuTwHjDW0084eDmPpN8nSclK0jm6BEAe/4o+ejn49i9Ko96r1GtgrEy2Hun0q0Trb+aMFc2JJux683aMrb3iF03Hc493fyxlUpkvZH0v/YW6x3RNvKofvxL8fw5srkHdPuOMbQ1nt8uu04Xv98d+qvdo5uikJ57gw8bT54Ds/P+0X93tERW3u53e86Gso7dNw2QulIWh5ZX+lMkTRA9evXhxzhKNPRltKiRQtlMOvVq6d2lh86dEiZWCkRERFYuXKl1+7ydoYj77mbQMNJ3+NkQiL6NCyPMW2qEpEGAtbTwhLu17eeRlEHOksNVXBZCDl9RE4hkTKmTRX0MWC5iXWy9TJFcmNjeNodvy6DwcBOE3jqnR9x+Ozl1Pv/O7oZCuaxP73e0q3H8UaKDutUKIJl6XLoWu+YntoxGJ1CyzhdV103roj7A0OXJW+6kbJ/XEtIVgJL2XrkPDrP/Vn909ERWx11pKG8Q9FthlJHQz4ohiRIlxN4zp49q3JLytrGjz76CBMnTlQjopKXUvJQygad8PBwbN26FQMGDFAms1evXg8Kn+b3FJVDuDz+YstaVObz1NeU6RffO9pZ6quJ/kjLt5/AiM92qcBvt6uGHvXcf/qXdbL1isXy4vt0O371U2DEjBJoP2sz4o4npIbZE9kceXMmp7yzp3z26x+QpQ5SGvoXxccv1ktzm/WOaXccY2jrHb787ykM/iR5040kdT84Ie3mXcvud8mlunNMMxTIZb/BtofZg65h32+goZRGkZyQco63JDSXndb9+vVD2bJlH9Se9/29nI4jaYLWrFmjTsoJCgrChAkT8MQTybvbZDe3nI4jU+wy5T5s2DC1s9zRQlE5Ssyzr282fQP2x//NfJ4am9F6fZeElXOG5bxhbyirdp7EkKU71atM7hCELrUz9r2mg4n1qSi2dvzqeAZj6CVgnYxeIqcfrXvQ06x1KAnAJRG4dZH19a+mmLeZ3WvimeCSDwrp8t+v230aAz6OU8+xlY1A1pZOjzmA4vlyoIcb1nyy7zfMUEpqn7CwMOTJk0dNTUseSjkK8fLly9i8ebMyl6YXisr0FtJbP5likamW9OfK6n2Kb0WzXt8lb77v7RbI9VDyUhRPL2t3ncYrS5I7xXc6haBDLfdngrA+FSXwkfz4+rUwT8fs9fW33pkvL3tkYiuHDv6w1uHTlUtgXq/aaZhZn8bzfo9aaFb1Ebcz/W5vPF76aLuqh63TfdxdQfb9hhlKyQWZN29efPzxx+o4RClyDKOc5S3/K6OLpheKyvQW0lu/305ehKSreLFRRYfWMOmthXdFs15nKG92cHxLZM92Z62UJ7+tdWqTf3ergbYhj7r9daZ+sw8zfzik6hFUqiBWv3rvNGlurywroAhYL1PIljULDqWb/n0QJmsdtqj6COb0qJXmlvX/F48X/pNs3hb0ro3GgfYf7vGgZzv7+x9+P4M+C5J3cfsVz4v1w9ImY3c2rq772PcbZijz5cunztGuVq1amjaW6W8ZuUxIuLNmRJcIdMehqHQTZTxfIyAprcJT1hnKuzs6+mIyL+vUJrOfr4mWQe6fSrQ+vUcSRUvCaBazCcgRhHLajZSc2bPi93EtHaqwtWGU6WyZ1rYuP+0/i57zt6ofLX6hLhoF3DmRxqEHabzYehe3rVRHGh/lVCj2/YYZSlm7KKOQtWql/WtJ1jfKMYg0lE7pnDeRgEcRsN7N6czoi8kvu/HAWfT4MLmj/qBnKJpWedjt1X1v/QG8k3JGcu3yhbG8fwO314kVuD8B63ymsr5Y1hk7UjbsP4teKYaxXfVHMaNr2lzLWw6dQ/cPklPwLH25HupVLOpIeJdc+8vhP9Hl/VgVu26FIvg03c50lzzUgaA0lIYZSpnaPnXqlMo3WaRI8jmc586dUzkk5RSdFStWONC87rmUonIPdz7VewhY7+Z0ZvTFZBKxh/9E15ROcWGf2njyMfdPJVofs1evYhEsfbm+yQhZNzmr+tc/MCxll3ahPA9h5+hmDnHZcvAcuqfkbOxYqzT+1Skkzf3bj55HxznJKXg+H9AAtcoVdii+Ky7+9dgFdJi9RYV+KrAE5vdOu+7TFc90JCb7fsMMpTRIgwYNcP78eVSqVEktMv79999RuHBhbNiwQaX3Mb1QVKa3EOtnOoGvdp/GwJTdnHlzZMOeqBamV9nu+v167Dw6zE7uqD9+sS4a+rt/KtH6GL7HA4ph0Qt17X4fXugeAjF74/FiygYVSWguic0dKdY5G7vVKYOJ7dOe8rXzRALazdysQrrjXGxb77LrjwS0jU6uU+vgkohON03vyPu74lr2/YYZStn2Lzu65cQa2fEt/5b0PnLOdsGCBV2hAe0xKSrtSBnQxwhYbxiwlR7Ek3FYd4qSTFqSSru7zN90BFEpx0E++VhxLOyTNoWMu+vH599NYNvR8+iUMoLozI7nuOMX0H5W8mif5EKVnKjWZc+pi3jm35vUj9xxLratNt976hJa/Xuj+pWJR92y7zfMUIaGhmL+/PkIDvbcM5EpKn79k0DGCFhvGCiSNwfiRjXNWECD7rbuFFcMbICaZd0/lbjo56MYlXIcpK0UMgbhY1VSCByI/wtNp/+k/uXM6Ua7/7iINtHJhtHWKV/74/9Cs5T4MUPD4F8iv9vZW7+ziSeTse83zFAWK1ZMHY/o5+f+822d/fRQVM6S430kkEzAesNAifw5sfVNx6bzTOZo3SmuHtQIQaXdP/Oy5JfjGLky+ThIWylkTObpq3U7c+kq6kxYr17fmdONLKfKyP0vh1XEyFaV06C8dPUG6o5fjyxZgF9GNkH+TD51xla7Hj13GU/+60f1q1ca+2FE80Cjmp99v2GGcurUqVi7di1GjBgBf39/5M6dO41gMnpaTmaoj6LKDMp8hjcTsE4P8mjBXNgS0cRrXte6U/z6tccR+EgBt7+b9bnNtlLIuL2CrMBdBK7eSELgqK/Vzys9nA/f/jP5xDd7y8Ezf+HpackjnAOf9EN4i7vN2YnzV9TvyxTJY29Yl173x4UraDT5B/WMEc0fwyuN/V36PEeDs+83zFBmzXonebFsyLEUWUsp/5aTc0wvFJXpLcT6mU7Aeie0M9N5Jr+fGIH6E9fjZtJtbIl4yoiRH+s0TbZSyJjM05frFjhqHa7euIWqjxbA2sGPO4TC+g+bwU0CMLRpJYfud8fF8Zeuom7KqOzYNlXQu2EFd1Tjns9k32+YoZSd3PcrlnO3jVJRuspQVCa3DuvmCQSsd0I7M51n+jsmXLmOpFu3UTRf8mlg7i7W5zp3qFka73ROm0LG3fXj820TqDshBvGXriGkTCGscjAZvfVo37CmlfBqkwDjMf/59zXUGhej6jmlYzA6h5Yxqs7s+w0zlNbqkNRB2bJl85jd3Za6U1RGfcZZGQ8kYJ2yJKBEPnw31LHpPA98ZbdW2TpNk4m7Z90Kx+CHN5u+Afvj/0ZoucL4bIBjyej/d/Eq6k1MXoMZ3uIxDHzSrOljW9gvJt5ASOS36ldyso8szzCpsO830FDKOsp3330Xp0+fVrWrUKECXn/9dbz00ksmaeeedaGoPKKZWEmDCcj56K3fS96BGvhIfnz9WpjBtfX8qlmnaXq+blmMfy7I81/KB96g85yfsfXoeTiTjP7c39cQmjLa92aryngpzPwcz4nXk1B5dPK6UVPOF7eWGft+wwzl5MmTERUVhcGDB6N+/fq4desWNm/ejFmzZmH69Ol4+eWXjf+aoKiMbyJW0HAC1jtQq5UqgDWvOrY+zPDXM6561mmaetUvh8hn0+YkNK7CrJAi8OJ/tiPm/+LhTDL6i1duICQqebRvTJsq6GPYekRbTSx7Kbp9EIsD8X+rWQtJKWZSYd9vmKEsU6YMJkyYgB49eqTRieSmnDhxIg4cOGCSfmzWhaIyvolYQcMJWO9ArV6mEL5wcH2Y4a9nXPWs0zS90KgCRrWuYlwdWaG7CQxf/l989usfaPxYcSxwMBn95Ws3UXXMNyqoJDWX5OaeUMRU3ki6jRzZ72zgNaXe7PsNM5T58uVDXFycOnbRuoiRDAkJwZUryWkMTC4Ulcmtw7p5AoEj5y6jcUq+OWfWh3nCO5pUR+s0Tf3CKiIiXU5Ck+rKutwhICPLIz7bhYiWgejk4AaVazeT8NhbydPHk9oHoWudskSbQQLs+w0zlL169UL+/PkRHR2dpmkjIyPVUYzLly/PYJO7/naKyvWM+QTvJiD57x6fkpxvrm6FIvi0X33vfmE3v90vh/9El/djVS3ulZPQzVXk4+9BwJJSz1FAt27dRsWRX6nb/tUpBB1rlXY0BK9PR4B9v2GGcujQoWq9ZOXKlSEpgh566CFs27YNGzduxLPPPotChQql1limwU0sFJWJrcI6eRKBUwmJaDDpe1Xlhv5F8fGL9Typ+h5XV+s0TZ6Sk9DjIBtY4YoRa3HrNvBu1+p4tnopA2voWVVi32+YoWzcuLHdCvrhh+QRDNMKRWVai7A+nkbgzF9XUWd8ckqTJyoVx3/61vG0V/Co+lqnaZIE12IqWbyfQOVRXyPxRhLm/KMmWlQzKwWPJ9Jn32+YofREEaWvM0XlDa3Id3AngfOXr6Pm29+pKjQJLIEPe9d2Z3W8/tnWaZo8JSeh1zdKJrxg9PcHEHv4PGb/o6YRJzZlwiu79BHs+2kotQuMotKOlAF9jIB1AuNmVR7G+z1DfYxA5r7uvv9dQosZG9VDR7YKxMthfplbAT6NBLyAAPt+GkrtMqaotCNlQB8jYJ3SpFXQI5j1fC0fI5C5r3vwzN94elrysbejW1dB30ZmnZGcuTT4NBJwjgD7fhpK55Rzn7soKu1IGdDHCFinNGkb8ij+3a2GjxHI3Nc9eu4ynkxJ0xT1bFX0rF8+cyvAp5GAFxBg32+Aofzyyy/RrFkz5MqVywskBVBUXtGMfAk3Eki6dRt+KSlN2tcohWldqruxNt7/aOs0TeOfq4bn63pGkmvvbxm+oScRYN9vgKGUZOb79u1D6dKlUbFiRZUmqGjRop6kozR1pag8tulYcUMISG69ChHJOfI61SqNqZ1CDKmZd1bjfxevot7E5F31UzoEo3PtMt75onwrEnAhAfb9BhjKEiVKoF+/fnj66achaYNWrlyJwoUL22z2sLAwF8pBT2iKSg9HRvFtAv4jv8LNW7fRrU4ZTGwf7NswXPz2Z/+6htrjY9RT3ukUgg5Mcu1i4gzvjQTY9xtgKCdPnoyRI0fCkvFf/tdWyZIlC5KSkozXIUVlfBOxgh5AIHDUOly9cUudMSxnDbO4jkDCleuoHpWcpolJrl3HmZG9mwD7fgMMpVTh4sWLSEhIQIUKFbB161YUL17cpvLKlcvY2p6PPvoIkyZNwuHDh+Hn54exY8eiU6dO6lk7d+7EkCFDsH37djXlPnjwYAwfPtzhTwBF5TAy3kACdxGoNuYb/H3tJno3KI+xbauSkAsJ/HX1BoLGfqueMLN7TTwTzCTXLsTN0F5KgH2/IYbSUo0NGzagYcOGyJ49u3bJLV68GL1798Y777yDZ555Bp988okylJs2bUKlSpUQGBiIdu3aQY5/jI2NxcCBA9UxkH369HGoLhSVQ7h4MQnYJBAS+S0kH+WLjSrgrdZVSMmFBBKvJ6Hy6K/VE+b8oxZaVHvEhU9jaBLwTgLs+w0zlFKdNWvWYPz48di1a5c6y7tKlSoYMWIEnnvuOadVKNPosuGnY8eOmDp1amqc5s2b48knn0z+y3zmTBw9ejTVzMo0/IoVK9SGIUcKReUILV5LArYJhI77Duf+vo7+T/jhjZaBxORCAjeSbiHgzXXqCR/2CkWTyg+78GkMTQLeSYB9v2GGctWqVWjfvr0yj7IBR9ZM/vTTT1i9erUyd23btnVKiWIKK1eujLi4ONSocXdOu1atWqFQoUJYsmRJavyYmBg0bdoU8fHxkI1D9haKyl5SvI4E7k2g7oQYxF+6hkGN/TG8+WNE5UIC8gd33Qnrceava1g35HFULlnAhU9jaBLwTgLs+w0zlDVr1lRmctSoUWkUFxUVpUYuZX2lM0VyXT777LPYuHEj3n77bezYsUOt13zrrbfQpk0bBAcHo2XLlpANQpayd+9eVK1aVaUxCg2999Fvly5dgvxnKadPn0adOnVw4sQJlQqJhQRIwHECDSd9j5MJiRjSJAD/bFrJ8QC8wyECclqOpA9qFFDMoft4MQmQQDIBGkrDDGXu3Lmxe/du+Pv7p9HogQMHEBISgitXrjilXVk/2aNHD2Uix4wZo2J9/vnnamr9u+++U2mLunfvDjGulmLZuCMmtFGjRvd8rqzDjIyMvOv3NJRONRVvIgFFIGzKDzh+/gqGNa2EV5sEkArhok6MAAAeQElEQVQJkAAJGE2AhtIwQyk7r9977z3IFLR1Wbt2LV566SWcOnXKKUEtW7YMXbp0UZtsBgwYkBpDNudIOX78uHqmrRHKe02TW4JwhNKpJuFNJHBfAk+98yMOn72M11sEYsCTfqRFAiRAAkYToKE0zFDKVPeiRYswe/ZstdtbiuzClh3XMmX97rvvOiWoLVu2qHjpRxvDw8PVVHr58uVVqiB5tqVwDaVTqHkTCWgh0H/Rr/h6z/8wr2conq7CTSJaoDIICZCAywjQUBpmKK9evYquXbtC1jxKInMpsmBcRg9llDFPnjxOiSExMREPP/wwRo8enSa3pKyblJ3kDRo0wJw5c3Do0CFky5ZNPSMiIkKd2sNd3k4h500kkCEC124m4cT5RPiXyJehOLyZBEiABDKDAA2lYYbSUh0xcbKWUsxkUFCQ2qGd0TJu3DhMmTIFc+fOVZtmli5dqgzm+vXrVWoiyUMpu8hl1FI2/8jUuJjMXr16OfRoisohXLyYBEiABEiABDyeAPt+Qw2lq5Q1bdo0REdH4+TJk8qkymYamUqXIru55XQc2QFesmRJDBs2DIMGDXK4KhSVw8h4AwmQAAmQAAl4NAH2/T5mKDNDrRRVZlDmM0iABEiABEjAHALs+2kotauRotKOlAFJgARIgARIwGgC7PtpKLULlKLSjpQBSYAESIAESMBoAuz7aSi1C5Si0o6UAUmABEiABEjAaALs+w0zlEeOHMHrr7+O3377DZJCKH2R02tMLxSV6S3E+pEACZAACZCAXgLs+w0zlHLEoZyF3blzZ+TKleuu1pZjE00vFJXpLcT6kQAJkAAJkIBeAuz7DTOUefPmVafZ1KxZU29LZ2I0iioTYfNRJEACJEACJGAAAfb9hhlKMZIzZsxAWFiYAfJwrgoUlXPceBcJkAAJkAAJeCoB9v2GGcrY2Fh1Qo0kGK9QoQKyZs2aRlueYDQpKk/9OmC9SYAESIAESMA5Auz7DTOUH374oTKUN2/evKtF5WzvpKQk51o6E++iqDIRNh9FAiRAAiRAAgYQYN9vmKEsU6YMWrdurUYo8+TJc5dEypUrZ4Bs7l8Fisr4JmIFSYAESIAESEArAfb9hhlK2ZSzZ88elC9fXmtDZ2YwiiozafNZJEACJEACJOB+Auz7DTOUHTt2RKtWrdC3b1/3q8PJGlBUToLjbSRAAiRAAiTgoQTY9xtmKKOjoxEREYGWLVvisccew0MPPZRGWqNHjzZeahSV8U3ECpIACZAACZCAVgLs+w0zlLKz+15FNuXwpByt+mcwEiABEiABEiABDQRoKA0zlBra1O0hKCq3NwErQAIkQAIkQAKZSoB9Pw2ldsFRVNqRMiAJkAAJkAAJGE2Afb9hhlISmcvU9r0K81Aa/Xli5UiABEiABEjAJwnQUBpmKBcuXJjGUN64cQMHDhyA/HzatGl4/vnnjRcqRWV8E7GCJEACJEACJKCVAPt+wwzlvVp3yZIlWLRoEdatW6dVAK4IRlG5gipjkgAJkAAJkIC5BNj3e4ihPHbsGKpUqYLLly+bq6aUmlFUxjcRK0gCJEACJEACWgmw7/cQQzlp0iTMnDkTJ06c0CoAVwSjqFxBlTFJgARIgARIwFwC7PsNM5SSh9J6U87t27dx6dIlJCQkYPz48XjjjTfMVRNHKI1vG1aQBEiABEiABFxBgIbSMEM5duzYu3Z558iRAw0bNkRYWJgrNKA9JkWlHSkDkgAJkAAJkIDRBNj3G2YojVaLnZWjqOwExctIgARIgARIwEsIsO830FBu3rwZmzZtwvXr1yFT3taFZ3l7ySePr0ECJEACJEACXkSAhtIwQymbb0aOHInChQujQIECaaTGs7y96JPHVyEBEiABEiABLyJAQ2mYoSxdujT69u2LqKgol8ps//79qFmzJqKjo9G7d2/1rJ07d2LIkCHYvn07ihYtisGDB2P48OEO14OichgZbyABEiABEiABjybAvt8wQ5knTx7s2rUL/v7+LhOWnL7ToEEDZRwXLFigDOWff/6JwMBAtGvXDkOHDkVsbCwGDhyIWbNmoU+fPg7VhaJyCBcvJgESIAESIAGPJ8C+3zBD2axZM2XgunXr5jJxyZS6GMYffvgh1VBOnDhR5bk8evQosmfPrp4t161YsQL79u1zqC4UlUO4eDEJkAAJkAAJeDwB9v2GGcp58+YhPDxcmUoZMcyZM2cakfXs2TNDovvpp5/w3HPPqentsmXLphrKVq1aoVChQpAjHi0lJiYGTZs2RXx8PEqUKGH3cykqu1HxQhIgARIgARLwCgLs+w0zlFmzZr2nsGRTTlJSktPCk+ToISEhkNHI7t27q3yXlinv4OBgtGzZEpMnT06Nv3fvXlStWhXbtm1DaGio3c+lqOxGxQtJgARIgARIwCsIsO83zFC6UlUyjS5piJYuXaoeY20oZc2mmEzrzUCHDx+Gn58fNm7ciEaNGt2zanKSj/xnKadPn0adOnXUMZGyyYiFBEiABEiABEjAuwnQUPqIoVy0aBEiIiKwe/dulZIovaEMCgqCTHvbGqGMi4tDjRo17vlJkNN9IiMj7/o9DaV3f3nw7UiABEiABEjAQoCG0kcMZePGjVWydOs1mZcvX1b/llHIcuXKqVRBYjwtxd41lByh5BcKCZAACZAACfg2ARpKHzGUJ0+eRGJiYhq1BwQEqBHJLl264JNPPsGcOXNw6NAhZMuWTV0nI5orV67kLm/f/o7g25MACZAACZDAAwnQUPqIobSlBOs1lGfOnFG7ytu2bat2mW/duhUDBgxQJrNXr14PFJL1BRSVQ7h4MQmQAAmQAAl4PAH2/TSUqSflyG5uOR1nx44dKFmyJIYNG4ZBgwY5LHKKymFkvIEESIAESIAEPJoA+34fNpSuUi5F5SqyjEsCJEACJEACZhJg309DqV2ZFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg309DqV2gFJV2pAxIAiRAAiRAAkYTYN9PQ6ldoBSVdqQMSAIkQAIkQAJGE2DfT0OpXaAUlXakDEgCJEACJEACRhNg3+9DhvL8+fMYOXIk1qxZg0uXLiE4OBiTJk1Co0aNFIWdO3diyJAh2L59O4oWLYrBgwdj+PDhDguYonIYGW8gARIgARIgAY8mwL7fhwxls2bNcObMGURHR6N48eKYOXMm5s2bh7i4OPXvwMBAtGvXDkOHDkVsbCwGDhyIWbNmoU+fPg6JnKJyCBcvJgESIAESIAGPJ8C+30cM5cGDBxEQEIDNmzejQYMG6q1v376NSpUqoVu3bsidO7cymEePHkX27NnV72U0c8WKFdi3b59DQqeoHMLFi0mABEiABEjA4wmw7/cRQ5mQkIAtW7agSZMmyJkzZ+pb+/v7o0WLFjh8+DAKFSqEJUuWpP4uJiYGTZs2RXx8PEqUKGG32Ckqu1HxQhIgARIgARLwCgLs+33EUNpS6/Lly9G5c2d8+eWXePPNN9GyZUtMnjw59dK9e/eiatWq2LZtG0JDQ+8peFmPKf9ZyunTp1GnTh2cOHECpUuX9ooPCl+CBEiABEiABEjg3gRoKH3UUMrUt4xMyojlF198ARmp7N69O6KiolKJyKiln58fNm7cmLpxx5aUxo4di8jIyLt+RUPJrx4SIAESIAES8A0CNJQ+aChXrVqlzGO9evXUjm9ZPxkUFIRWrVrZHKGUTTs1atTgCKVvfCfwLUmABEiABEjAYQI0lD5mKGWHt6QGat++PRYvXpy6nlLMpKQKWrRoUSoRrqF0+PPEG0iABEiABEjAJwnQUPqQoZw9e7ZKBST5JadPn46sWbOmvr3ko5wzZw4OHTqEbNmyqZ9HRERg5cqV3OXtk18NfGkSIAESIAESsJ8ADaWPGMr9+/ejWrVqaN26tcotaV1kyvvatWsqD2Xbtm0RHh6OrVu3YsCAAcpk9urVy35FAaCoHMLFi0mABEiABEjA4wmw7/cRQzlhwgS1k9tWEcO4cOFCtZtbRi937NiBkiVLYtiwYRg0aJDDIqeoHEbGG0iABEiABEjAowmw7/cRQ5mZKqWoMpM2n0UCJEACJEAC7ifAvp+GUrsKKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4ZSu0ApKu1IGZAESIAESIAEjCbAvp+GUrtAKSrtSBmQBEiABEiABIwmwL6fhlK7QCkq7UgZkARIgARIgASMJsC+n4YyjUBv3bqFyMhIzJs3DxcuXECjRo0wa9Ys+Pv72y1kispuVLyQBEiABEiABLyCAPt+Gso0QhYzKQZywYIFKFWqFMLDw3H48GHs2bMHOXLksEv0FJVdmHgRCZAACZAACXgNAfb9NJSpBK5fv45ixYphypQp6N+/v/p5QkICHn30UcyfPx9du3a1S/gUlV2YeBEJkAAJkAAJeA0B9v00lKkEtm7dirp16+L3339HpUqVUn8u097BwcFq5NKeQlHZQ4nXkAAJkAAJkID3EGDfT0OZSmDFihXo0KEDrly5gty5c6f+vHPnzupna9assUv5FJVdmHgRCZAACZAACXgNAfb9NJSpBBYvXowePXogKSkJWbNmTf15z549cerUKcTExNgU/qVLlyD/WcqJEyfQoEEDyIhnyZIlvebDwhchARIgARIgARKwTeD06dOoU6cOjhw5gvLly/s0piy3b9++7csEPv/8c3Ts2NHmCOW1a9ewatUqm3jGjh2rdoazkAAJkAAJkAAJ+DYBGUyqXbu2T0PweUNpWUN58OBB+Pn5pYpB1lCGhIRg5syZdo1QXr16FTJKWaFCBWTPnl2rqCx/AXH0UytWh4KxDRzC5bKL2Q4uQ+tQYLaDQ7hcdjHbwWVo7Q588+ZNnD17FkFBQciVK5fd93njhT5vKGUUskSJEpg2bRpeeOEF1cbO7PJ2pTi4RsOVdO2LzTawj5Orr2I7uJqwffHZDvZxcvVVbAdXE2Z8Rwj4vKEUWG+++Sbmzp2r0gTJGogRI0bg6NGj2L17t915KB2B7ui1/NJwlJj+69kG+pk6E5Ht4Aw1/fewHfQzdSYi28EZarzHVQRoKAG1IWfkyJEqsXliYiLCwsLUVLcpC2z5peEq+dsfl21gPytXXsl2cCVd+2OzHexn5cor2Q6upMvYjhKgoXSUmBuul93kMiU/dOhQFChQwA014CPZBmZogO3AdjCDgBm14OfBjHZgLZIJ0FBSCSRAAiRAAiRAAiRAAhkiQEOZIXy8mQRIgARIgARIgARIgIaSGiABEiABEiABEiABEsgQARrKDOHjzSRAAiRAAiRAAiRAAjSUBmvg1q1b6jSeefPm4cKFC5Bk67NmzYK/v7/Btfb8qh07dszmDv8PPvgAL774Inbu3IkhQ4Zg+/btKFq0KAYPHozhw4d7/osb8gbjxo1TR57++OOPqTV6EHN+VvQ3nq126NOnDxYuXJjmYaVKlYLsNpbCdsh4O5w/f15lHVmzZo063jc4OBiTJk1S3/9S+FnIOGNGcA0BGkrXcNUSVcykGEhJZyRf2uHh4Th8+DD27NljRH5MLS9pYJDVq1ejc+fOinWWLFlSa1iwYEF1RGdgYCDatWundt3HxsZi4MCBqp2ks2XJGIEZM2YorpK6y2Io//zzzwcy52clY9zT322rHeSaWrVqoUWLFnj11VdTb8mWLRuKFy+u/s12yHg7NGvWDGfOnEF0dLTiKinsZFAhLi5O/ftB3z9sg4y3ASM4R4CG0jluLr/r+vXrKFasGKZMmYL+/fur55l2go/LIbjpARMmTMCyZcvUSED6MnHiRPUFL4nvLUdsymjCihUrsG/fPjfV2PMfe/LkSTX6u3HjRpQpUwYPP/xwqqF8EHN+VvS1//3aQfL15suXD5988on6gyp9YTtkvB3kCOCAgABs3rwZDRo0UAFv376NSpUqoVu3bsidO/d9v3/YBhlvA0ZwngANpfPsXHqn5Yzx33//XX2ZWIpMe8gUiIyIsbiGQNeuXZE1a1YsWbLkrge0atUKhQoVSvM7mZ5t2rQp4uPj1TGeLI4TkFFhMSoyzRoVFaUMu2WE8kHM5dq6deuCnxXHuae/437tIH8wVa5c+S7Olhj8zso4fxk02LJlC5o0aYKcOXOmBpRlTjIyLLMm9/v+4Wch423ACM4ToKF0np1L75QRrw4dOqgpVvmr1FJkKlZ+JutrWFxDoGrVqihZsiTknPf9+/erEYNRo0ahefPmysy3bNkSkydPTn343r17Ifds27YNoaGhrqmUD0Xt3bt3GkP5IObHjx/nZ8UF+kjfDsuXL0eXLl3UjMm6devUH11i9uWPAFkOwu8sFzQCAOEu3/tffvmlOib4ft8//Cy4pg0Y1T4CNJT2ccr0qxYvXowePXqoYyHli9tSevbsiVOnTqlNCyz6CciUUd68eVGjRg1MnTpVTfFJW7z77rv47rvv0K9fP3Tv3l2NolmKjBr4+fmp6VrLwnn9NfOdiOmNjIzO3I+5jMrws6JfH+nbYcyYMRg/fjxkSYgYyQMHDqjNaLJE4fvvv1ej9mwHve0gU98yMikjll988YXakMnPgl7GjKaPAA2lPpZaI33++efo2LGjzRFKGTlbtWqV1ucx2B0Cly9fVusjraec5EtdNujIblbpTG2NUMqieTGiLBkjkN7IBAUF3Ze5GHp+VjLG3Nbd6dtB1vJdvHhRTblaimxKq1+/vtqcJp8NtoO+dpDveDGP9erVUzNSMlPFz4I+voyknwANpX6mWiJa1iPJIm0Z/bIUGQELCQlRC7NZMo/AiBEj8PXXX6vRGEkVtGjRotSHcw2l3nZIb2TEwN+PuWXdGD8rrm0HW9Hljy8Zxf/ss8/UZ0PWsrIdMt4OssNbUpO1b99ezZBY/rjlZyHjbBnBdQRoKF3HNkORZRRSNnhMmzYNL7zwgorFXd4ZQmrXzbt27VK7K8U8Wk9fN27cWO08rl69OubMmYNDhw5B0qVIiYiIwMqVK7nL2y7CD74ovaGUHHz3Y87PyoOZOnNF+naQ0TJJ4fTNN9+khpONU/LZkHXEFStW5HeWM6DT3TN79myVikzy206fPj3Nkid+FjQAZgiXEaChdBnajAeWBdhz587F/PnzVaJtGSWT0Zjdu3czD2XG8dqMIImZxVDKyIt8sUvqpvfff1/tqpdNN2IqJQ9c27ZtVV5QGUkeMGCAMjy9evVyUa18K2x6IyM5+R7EnJ8V/RpJ3w4y7Sq6f/vttyGZEGTD2iuvvKKmvD/++GNVAbZDxtpBmFarVg2tW7e+K5OHTHnLH0/8LGSMMe92HQEaStexzXBk2ZAjOQ4lsXliYqJK9ixT3WIuWVxH4OzZs3jjjTfw1VdfqVHhmjVrqpMqHn/8cfVQMZYyerBjxw61G3zYsGEYNGiQ6yrkY5HTGxl7mPOzol8kttpB1nbLxhxJISRrKWXUUnZ558qVS1WA7ZCxdpANT2LKbRX5g1VOKXrQ9w/bIGNtwLudJ0BD6Tw73kkCJEACJEACJEACJACAhpIyIAESIAESIAESIAESyBABGsoM4ePNJEACJEACJEACJEACNJTUAAmQAAmQAAmQAAmQQIYI0FBmCB9vJgESIAESIAESIAESoKGkBkiABEjATgKSYUF2P48dO9bOOxy7zJLX8ciRI/fM5uDqOjhWY15NAiRAAskEaCipBBIgARKwk4CklJJ8gHI6jCuKnCV//vx5FC9ePDVxfvrn0FC6gjxjkgAJZJQADWVGCfJ+EiABEshEAjSUmQibjyIBErCbAA2l3ah4IQmQgKcQyJIli0oQLWeuy+kiGzZsQIUKFTBq1Ch1NvLFixfViSRRUVFo1qxZ6mt9++236ihNy1GCkrRejj61TEGnN3Nr165VJ8f89ttvKFCgALp166YSf1sSfUs95KSlZcuWYdOmTShSpIhKgi/PsFXST3lLPSWJ/qpVq9TpWHLQwYwZM1w67e4pbcx6kgAJmEWAhtKs9mBtSIAENBAQIyfHZq5btw43b95EvXr11Kkue/bsUYasdOnSWL16tToRSc5hf+aZZ7Bz507Url0b//znP5WJlH/LsZoXLlywaSi/+OILdOjQQa2n7Ny5Mw4cOKCur1OnDuREGSlSj4IFCyI6Olr9XAyunCzz008/pZ68ZP266Q1l8+bNcfz4cWVK8+fPr05l+v777zFmzBiXrePUgJ8hSIAEfJAADaUPNjpfmQS8nYAYOTGG06ZNU6968OBBBAQEqGPrQkNDU19fjrM7duwYxMjJ/5ezlH/++efU37/33ntqhNDWCGXdunVRpkwZfPbZZ6nXi0mV865lhLNy5crKUA4ZMkSZWEuRUcoRI0bYHKW0NpSWc5tjYmLQpEkTdXt8fDzKlSunjLCrNgZ5uzb4fiRAAq4hQEPpGq6MSgIk4EYCYuRmzZqlRgylLF++XI0i5s2bN02tbty4oc6kFqMWFBSEpk2bpppQuVBGKWvUqGHTUObJkwdy9vJrr72WGlOmqCXe0qVL0aVLF2Uo58yZg379+qVeI9Pmffr0UaOM6Yu1ody+fTs6deqkpudlOt1SqlSpot6FhtKNAuOjSYAE7iJAQ0lRkAAJeB0BMXILFixQaw2lfPrpp+jatSvi4uLU1LF1yZYtm1pfKcbxiSeeSDOaeD9DKbu9J02apEYgLUV2aBctWlRNebdv314ZSut6yHX321Rjy1DKlLuYVEsJCQnBc889R0PpdarlC5GAZxOgofTs9mPtSYAEbBBIb+Rk7aRswlmzZo1aL2kpsnEna9asamONmM99+/YhNjY29fdz585F//79bY5QyprIsmXLppnyls0z7dq1SzPl7ayhTEpKgr+/f5o6JyQkoFSpUmrKnCOUlD4JkIBJBGgoTWoN1oUESEALAVsjg23atFFT2DNnzlTmcsWKFQgPD8eHH36opqBlp3b16tUxfPhw9O3bV5nCl19+GZJ78ujRo2rtovXookyjy7R2ZGSk+l9ZfylT7BJD1lJKycgIpTxLprZl3ecHH3yARx55RK27lI1Gb731Fg2lFqUwCAmQgC4C/w9G4RAtO7AR3AAAAABJRU5ErkJggg==\" width=\"599.9999869953505\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "pb_num_dict = {_k:len(_v) for _k,_v in sub_library_250_2.items()}\n",
    "\n",
    "pb_nums = np.array(list(pb_num_dict.values()))\n",
    "plt.figure(figsize=(6,3), dpi=100)\n",
    "plt.plot(pb_nums)\n",
    "#plt.plot(np.ones(len(pb_nums))*20)\n",
    "plt.ylabel(f\"num of probes\", labelpad=1)\n",
    "plt.xlabel(f\"region id\", labelpad=1)\n",
    "plt.ylim([0, num_probes_per_region*1.1])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b7466a23",
   "metadata": {},
   "source": [
    "# extract smaller set, ~60 probe"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "5020293f",
   "metadata": {},
   "outputs": [],
   "source": [
    "short_probe_num = 60"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "20ae2463",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Picked primer: ID: W1A11_primer_10\n",
      "Name: W1A11_primer_10\n",
      "Description: W1A11_primer_10\n",
      "Number of features: 0\n",
      "Seq('TTGCTAGCGTGGGCCAATCG')\n",
      "- Picked primer: ID: W1B06_primer_17\n",
      "Name: W1B06_primer_17\n",
      "Description: W1B06_primer_17\n",
      "Number of features: 0\n",
      "Seq('TAATACGACTCACTATAGGGCCTGTGCGGAAATCGCGAGA')\n",
      "ID: loc_1:3740000-3760000_gene_1_pb_23_pos_2742_strand_-_readouts_[NDB_856_c,NDB_946_c,NDB_935_c]_primers_[W1A11_primer_10,W1B06_primer_17]_library_1000-mouse-genome-short_500\n",
      "Name: loc_1:3740000-3760000_gene_1_pb_23_pos_2742_strand_-_readouts_[NDB_856_c,NDB_946_c,NDB_935_c]_primers_[W1A11_primer_10,W1B06_primer_17]_library_1000-mouse-genome-short_500\n",
      "Number of features: 0\n",
      "Seq('TTGCTAGCGTGGGCCAATCGTATACATTAGGATTGCGAGCTGTCATAACTGTGT...AGG')\n",
      "29845\n"
     ]
    },
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAApQAAAFKCAYAAACjE8/+AAAAAXNSR0IArs4c6QAAIABJREFUeF7t3Ql8TWf+x/GfJPYt9qbEltBYEksjlKBoLNHaiqJV1cU2/rS2llKkFDWjtLZWq51h1FBRpa0WU3sVg9qqlgiKohStffu/nkdzcxOJ3Cvn3HuO+zmvV18zkXOe89z3+SXPN2d5TpZbt27dEhYEEEAAAQQQQAABBO5RIAuB8h7l2AwBBBBAAAEEEEBACxAoKQQEEEAAAQQQQACBTAkQKDPFx8YIIIAAAggggAACBEpqAAEEEEAAAQQQQCBTAgTKTPGxMQIIIIAAAggggACBkhpAAAEEEEAAAQQQyJQAgTJTfGyMAAIIIIAAAgggQKCkBhBAAAEEEEAAAQQyJUCgzBQfGyOAAAIIIIAAAggQKKkBBBBAAAEEEEAAgUwJECgzxcfGCCCAAAIIIIAAAgRKagABBBBAAAEEEEAgUwIEykzxsTECCCCAAAIIIIAAgZIaQAABBBBAAAEEEMiUAIEyU3xsjAACCCCAAAIIIECgpAYQQAABBBBAAAEEMiVAoMwUHxsjgAACCCCAAAIIECipAQQQQAABBBBAAIFMCRAoM8XHxggggAACCCCAAAIESmoAAQQQQAABBBBAIFMCBMpM8bExAggggAACCCCAAIGSGkAAAQQQQAABBBDIlACBMlN8bIwAAggggAACCCBAoKQGEEAAAQQQQAABBDIlQKDMFB8bI4AAAggggAACCBAoqQEEEEAAAQQQQACBTAkQKDPFx8YIIIAAAggggAACBEpqAAEEEEAAAQQQQCBTAgTKTPGxMQIIIIAAAggggACBkhpAAAEEEEAAAQQQyJQAgTJTfGyMAAIIIIAAAgggQKCkBhBAAAEEEEAAAQQyJUCgzBQfGyOAAAIIIIAAAggQKKkBBBBAAAEEEEAAgUwJECgzxcfGCCCAAAIIIIAAAgRKagABBBBAAAEEEEAgUwIEykzxsTECCCCAAAIIIIAAgZIaQAABBBBAAAEEEMiUAIEyU3zJG1++fFl27NghRYoUkYCAAINapRkEEEAAAQQQsKrA9evX5dSpUxIeHi45cuSwajc90i8CpUHMmzZtkqioKINaoxkEEEAAAQQQsIvAxo0bpUaNGnbprin9JFAaxJqYmChlypQRVVRBQUEGtUozCCCAAAIIIGBVgePHj+uTSQcPHpTSpUtbtZse6ReB0iDmX375RYKDg+XIkSNSokQJg1qlGQQQQAABBBCwqgBjf/KRIVAaVKUUlUGQNIMAAggggIBNBBj7CZSGlypFZTgpDSKAAAIIIGBpAcZ+AqXhBUpRGU5KgwgggAACCFhagLGfQGl4gVJUhpPSIAIIIIAAApYWYOwnUBpeoBSV4aQ0iAACCCCAgKUFGPsJlIYXKEVlOCkNIoAAAgggYGkBxn4CpeEFSlEZTkqDCCCAAAIIWFqAsZ9AaXiBUlSGk9IgAggggAAClhZg7CdQGl6gFJXhpDSIAAIIIICApQUY+wmUhhcoRWU4KQ0igAACCCBgaQHGfgKl4QVKURlOSoMIIIAAAghYWoCxn0BpeIFSVIaT0iACCCCAAAKWFmDsJ1AaXqAUleGkNIgAAggggIClBRj7CZSGFyhFZTgpDSKAAAIIIGBpAcZ+HwuU//rXv2Ts2LGSkJAgISEhMmLECGnXrp1W2LZtm/Tt21c2b94shQoVkj59+siAAQPcLmCKym0yNkAAAQQQQMDWAoz9PhQoZ8+eLc8995z84x//kObNm8unn36qA+XatWulfPnyEhYWJq1atZJ+/frJhg0bpFevXjJ16lTp2rWrW0VOUbnFxcoIIIAAAgjYXoCx30cC5a1bt6Rs2bLStm1bGT9+vONTN2nSRB599FH99ZQpUyQxMVECAgL010OGDJH4+HjZs2ePW4VOUbnFxcoIIIAAAgjYXoCx30cCpQqFFSpUkC1btki1atXuKNzY2FgJDAyUOXPmOL63fPlyiYmJkRMnTkjRokVdLnaKymUqVkQAAQQQQOC+EGDs95FA+cUXX0jLli1lzZo18uabb8rWrVulTJkyMnToUHniiSckIiJCmjVrJuPGjXOI7N69WypVqiSbNm2SyMhIlwverKLa8+t5WbvvN5f7wYoIIGCOgL9fFqlbrrBsOXxWzl+6Zs5OnFr1y5JFYioWk8BcWeXL7cflzyvXTd+nkTsomDubVHowv6zd/5uoq0Us1hYI8Msi0eWKyObEMx6ttZzZ/KVW2UKyfv9vcuX6TY1UokBOCcqfUzYlnnGgPRJSSA7+dkF+PXc5BWS+nFmlWnCgrNn3m9y8S53lyhYgnWqWNPwgmDX2G95RDzSY5dZ9/JOu7p/s3LmzDpHDhw+XKlWqyIIFC2T06NGybNky6d69u3Tq1Eni4uIc1EkP7qgQGh0dne4hOH/+vKj/kpbjx49LVFSUHDlyREqUKGHYoft042EZHL/DsPZoCAEE7CMQUSK/RIcWlqkrD9in0/QUAQsKFMuXXX4Y8pjhPSNQJpPe14Fy3rx58tRTT+mHbHr27On41OrhHLUcPnxY1GXvtM5QpneZPKkR9WDPyJEj7yhOowPlku3HZMK3ew3/IaBBBBBwXeDcpWty+sJVxwY5s/pLUP4crjfg5poXr96QX89flmz+fhIb/oB8vu2Y5M7mL8XymbdPN7t419WPnr3kONukViyaN7vkyX77PnUW6wmcvXRNzjjVtzpW6piZvZz684r8cTn5zHv+nFn12dEbN5PPaBfKnS3Fz566UlCqYC7dtePnLsulazcc3SycJ7vky5F2nRXKk03m96ht+EciUPpIoFy/fr3UqVNHX/J2Pts4aNAgWbJkiZQuXVpPFTRr1iyHiKv3UHrqDKXh1U+DCCDgtsCs7xNl2KJdju0ahhWVmc/VcLsdVzfYevh3aT11vV69ZpmC8sPBM9Kq6oMyscOd94K72qYn1+v4wQb5PuG0Y5cfdYmURhWKebIL7MsNgY/WHpQ3l+x2bNH24RLy93ZV3Gjh3lZ9Y9FO+df3hxwb/61BiL69I/H0Rce/jXsyXF5dkHyVLrhgTlkzqKH+fo9Z/5Olu351rDuhfRVpU924K4SufCoCpY8EykuXLkmxYsXkjTfeSDG3pLpvMmvWrFK7dm2ZPn26HDhwQPz9/bXK4MGDZeHChTzl7cpPEusg4CMCc344LEMWJg9q6t7GGc+6fo+1u0wn/7gsUaNX6M0eyJdDn6301CDvbl/TWr/rxxvlu59POb717xdrSp3QwkY0TRsmCMzecEiGfr7T0fLTNUvK6NbhJuwpZZNxi3fLzHUHHf/Yt1E5Wf7TCdl1LPl2sqlPV5de/97iWKd0oVyycmAD/fXLc7fqs/dJi1o3NjzI9H4774BA6SOBUn3MUaNGydtvvy3vv/++vsdx7ty5OmCuWLFCKlasqOehbNGihaizlhs3btSXxlXI7NKli1tFSVG5xcXKCNhK4D+bDqc4S9K00gMyvfPDpn2GmzdvSdgbS+XqXw8pqB11jAqWMW0iTNunkQ33nP0/+Xpn8pmjBT0fkYdLFTRyF7RloMD8zUdk4GfbHS0+X6eMvPFERQP3kHZTY776Sd5fneD45oDG5WX13t9ko9PDOLNfqCnPfPSDY52yRXLLf/vfnvbvtQXbZe6mI47vzXwuUhqGefZMOGO/DwVK9VEnTJggkydPlqNHj+pphNS9j+rpb7Wop7nV23HUE+BBQUHSv39/6d27t9s/SBSV22RsgIBtBFIPuM3Dg2TK09VN7X+Dv6/UT7UmLZ1rlZI3W1U2dZ9GNf7Kf7bJwq1HHc0t+b9oqVw8v1HN047BAl/8eEz6fLrV0WrPR0Pk1aZhBu/lzubGf7NHpnyX/MDZa83C5IeE046z235ZRD7rWVva/HX7h2qhXNE8sqxffd3YiC92ySfrEx0Ne+NMOGO/jwVK038qRISi8oQy+0DAOwLxW36RfvN+dOz8iSoPynsdzb2fsfNHP+ipUJKW52qXlhEtKnkHwM29pj5ztLxfPQktmtfNVljdUwLf7PpVus/6n2N3Lz9WTl5+rLzpu5+wbK+8u2KfYz9Dm1eQrUfO6vso1ZI3e4DMfrGmtJyyzrFO2AN5ZenL9fTXqc9weuNMOGM/gdLwHxSKynBSGkTAMgKLth2VvnO3OfrTulpxeeepqqb2L3Uoe6luGXm9ufmXIY34UKnPHK0Z1ECC/3oy14j2acNYge9+PildP97kaHRQ04ek16Ohxu4kjdbeW7FP/rEseRaTEU9UlN3Hz8u8zb/otdX9wx92iZTH31vr2LpiUD75qm9d/XXqQOqNM+GM/QRKw39QKCrDSWkQAcsIqOm7es9JviT4ZPUS8o/25j4F+86yvTLJ6eyNpy5DGoE+5uuf5P1VyffGbXr9MSnigWlojOi7L7ax/sBv0mlG8n2Kwx6vKC9ElzGdYtrKAzJuafJrjke1qiwHTv0pH6+7fRlb3S85uWN1iX13jaMv4cXzy+L/uz1H9NSV++XtpT87vre8X30JLZrH9H4774Cxn0BpeMFRVIaT0iAClhH4esdx6en0pGn7yBLydltzA2XqwbZPw1Dp1/ghy5jcrSOpw/COEY0lb46stui7L3byf4d+lyen3Z6mSi0q2D1Tq5TpFB+uSZBRX/7k2I+aIujImUsy+bv9+t/UxP7j21aRJhNXO9apEhwoi/5WR389c+1BiXOa7mjtqw2kRIHbc1R6amHsJ1AaXmsUleGkNIiAZQRS32PmiSeuUw+2rzxWXvo+Vs4yJnfrSOowvG90M8nq72eLvvtiJ3cePZfisvL4thHSLjLYdIqP1x2UkYuT579Uc1+e+uOK46xlrbIFdbh9bEJyoKxeMlDie90OlKmn89o89DFRk5t7cmHsJ1AaXm8UleGkNIiAZQRW/HRCXvjnZkd/nqlVUka1Mneevn+uT5ThXyRPpj6wyUPytwbm39dmBLpzUFBvNjnwVqwRzdKGSQL7TvwhMe8kh7ZJHapKy6rFTdpbcrOzNhySYU7zX6r9qrdSvfHXSwQahRWV15tXkIb/WOXYKLJUAf3kt1pSPyy3c2QTj7+RibGfQGn4DwpFZTgpDSJgGYHUDy10eaSUjGxp7hQ+qc++DIkNk271QixjcreOfLrxsAyOvz0RfK5s/rI7rqkt+u2rnTx8+qLUG/+d4+NPf+ZhaVr5AdM55m48LK/9VSdqZ1M6VdevUhww//aMCi2qPCj9G5eX+uNXOvoSVaagzOv+iP76qx3HU0x67o0z4Yz9BErDf1AoKsNJaRABywis3ntKnp250dGfrnVKy/AnzJ3CJ/Xcl288XlGe98CDEkagL9z6i7zyn9uhoGDubLJlWIwRzdKGSQInzl+Wmm/dfjOTWj7uWkMaPFTUpL0lN/vZ/35xhEf1r+93flhu3bolPWbffjOOurVEPW1e9+3ksPtI2ULyabda+vv/3XNCnv/k9pUDNWdlwpjmpvc59Q4Y+wmUhhcdRWU4KQ0iYBmBdft/k6c/TH4K9sXoMjL0cXOn8Pl861F5+T/JUxW92bKSdH6ktGVM7tYR54eYHsyfQ9YPbmSLfvtqJ89evCpV45Y5Pv6cl2pK7RDzX5WZejou9aabAD8/xx9v6kpAt/ohUmfsfx19iw4trOemVMv6/b9Jp79+Lr11awVjP4HS8N8bFJXhpDSIgGUENiSclg4fbHD0p3v9sjK4WQVT+6cmd/7bnOR3GL/VOlw61Sxp6j6Navy7PSel6ye35zV0fveyUe3TjrECl67ekApvLHU0uqBnbXm4VAFjd5JGa6mn4/rn81GSJ7u/PDnte722CpQ9Hw2VWmOSz57WK19E/vV8lP6+89PpBErTD1eGO8hyS51fZsm0AIEy04Q0gIBlBTYlnpF2028Pcmrp9WiIDDL51XTf7vpVujm9veTtthHS3gNP3hpxEJzPHJUpnFu+G3D73css1hS4cfOWhAz5ytE5T00QvnTnr9JjdvIbeua8WFMK5skmTSfenndSBcq/NQyVqNHJgbLBQ0Xk4663A6Xz0+kESu/XFoHSoGNAoDQIkmYQsKBA6nn6/q9hqPQ3eU5I57N8iuSdp6pI62olLKhzZ5ecvcoWzi3/JVBa/riVfu1LRx+XvVJPyhUz/1WZqWdPUA/bBOXP4bhnUgXKPo3KycOjljv69liFovJhlxr66/0n/5THJtx+AjzAL4vs98JsAoz9yaVNoDTox5yiMgiSZhCwoMC2I2elldP7hPs2KievxJj7ruO1+36TZz5Kvm9TvTtcvUPcDsuuY+ek+bu3X5fHGUo7HDER50C5emADKVnI/AnCV+09JV2cHnaL71VbShfKLdXfvH0/p3p/vXqvuPP9nY0rFpMPno3U3//l94sSPe72AzucofR+nREoDToGBEqDIGkGAQsKpJ74uX9Mefm/RuZOMv5Dwml5yum+zenPVJemlYMsqHNnl9Tr8xr9NXcggdIWhyxFoPxhSCMpli+H6R13vjVC7Wxx72gpVyyPhA27fT9nt3plpXfDUIkY8a2jL80qPyDTnnlYf/3bn1ck8q+zlwRK0w9XhjsgUGZI5NoKBErXnFgLATsK7D52PsX7hD0xyXjqy+wzno2UmIrFbMF39Owlx5O5BEpbHLIUgXLbGzESmCub6R1P/UfTV33qSoWgvNL5o42iznJ/80o9yZ0tQCoN/8bRl+YRQXq+SrX8eeW6VP7rewRK0w9XhjsgUGZI5NoKBErXnFgLATsK/PzrHyneJ/xaszDpUd/cScZ3/HJOnph8+7KxWjw1N6ARx+f0n1cc970RKI0QNb8N50veu+OaSK5sAabvNPUfTUn3bqpnha/duCXZAvzk8rUbjjOWqkNqsvN3O1bTfbt+46aEvv61/v8EStMPV4Y7IFBmSOTaCgRK15xYCwE7Cjjf/K/6P7R5BXmxbllTP8qeX887nnZVO5r9Qk2JLmf+3IBGfKgLV647zioRKI0QNb8N50C5f3QzCfDAu9e3/3JWWkxe5/hwajYAVS/Oy9XrN6X80NuhUS2tqxWXd56q6vg6qd8ESvNrJKM9ECgzEnLx+wRKF6FYDQEbChz87YI0+Hvy69888daa1CF2brdaUqtsIVvoOU9DQ6C0xSFLcck7caxn3jiT+laSNYMaSHDBlA8DpZ7S6MnqJeQf7asQKC1YVgRKgw4KgdIgSJpBwIICqd91PLJFJelS29y31qTe52c9HpHI0gUtqJN2l5LOHDFtkD0OmfMZSk8Fyr0n/pDG76x2AH0/uKEE5c+ZAkxd/i4zOHmOzPaRJeTttncGSqYN8n6dESgNOgYESoMgaQYBCwo4T0+iuvdmq8rSuVYpU3t6/NwleWRM8ivnPv9bHakaHGjqPo1snEBppKb5bXkjUCac+lMa/jUbgPqEm15/TIrkzX7Hhy07+Eu5+dcrWNT7vce0ibjjDCWB0vwayWgPBMqMhFz8PoHSRShWQ8CGAr+eu5zi9W9j2oRLxyhzX4N46o8rUmN08oTOnnp7iVGHJymgcMnbKFFz2/FGoDxy5qJjEnP16dJ7urzc61/ph3TU8nTNkjK6dfgdgZJ7KM2tD1daJ1C6ouTCOgRKF5BYBQGbCpz843KK17+9/WSEtK8RbOqnOXfxmlSJS55/75uX68lDD5j/9hKjPlRSQIkNf0CmPn173kAW6wp4I1CmPgu/c2QTyZP9zqfLHxr6tVy5flPjPftIKYlrWfmOQFm9ZKDE96rjcWDG/mRyAqVB5UdRGQRJMwhYUMB5GhzVvb+3qyJtHzb3NYjOT0qrfa7oX19CiuSxoE7aXVq07ah8veNXfXtAWpcxbfNBfKSj3giUqc/C73mzqeTI6n+HeMU3lsrFqzf0v6u354xoUcmxzrLdJ2Te5iMy/ImKUqKA+W/3Sd05xn4CpeG/Iigqw0lpEAHLCJy9eDXF69888V7t1NOlrBr4qJQqlHJKFcsA0RHbC3gjUP5+4apU++s1iwowvemKwod/I39cua6NX4guI8Mer2gZb8Z+AqXhxUhRGU5KgwhYRuCPy9ck3On1b2piZTXBsplL6qdb173WUIoHpnwC1sz907ZvCXgjUKb+uTo4JlayZMlyB3yVkd/KuUvX9L+/VLeMvN6cQGnF6uSSt0FHhUBpECTNIGBBgYtXr0vFN5Jf/6Ze/aZeAWf24jzIbxzSSIp64P3KZn8m2remQNtp62Xzod915zw1bVDqt+Ckt9/qby6TMxeu6r51r19WBjerYBlExn7OUBpejBSV4aQ0iIBlBFIPfNOfqS5NK3s2UG4ZFiMFc5v/fmXLoNMRjwqcPH9ZPlx7UFpVLS4VH8znkX07vzrxbkE2ctRy+e3PK7pPvR4NkUFNwzzSP1d2wthPoHSlTtxah6Jyi4uVEbCVwLUbN6XcX+8MVh3/oPPD0rjSA6Z/BuczlNtHNJZ8ObKavk92gICnBFLf1pHeGcqaby2XE+dvB8r/axgq/Rs/5KkuZrgfxn4fCpSHDh2S0qXvfKPFjBkz5MUXX5Rt27ZJ3759ZfPmzVKoUCHp06ePDBgwIMMiSr0CReU2GRsgYBuBmzdvSdkhyW/rmPlcpDQMK2Z6/50D5e64JpIr251TqpjeCXaAgIkCrty7WXvMCjl27rLuRd9G5eSVmPIm9si9phn7fShQLl68WNq3by8JCQkpbvbNnz+/XLx4UcLCwqRVq1bSr18/2bBhg/Tq1UumTp0qXbt2dauqKCq3uFgZAdsJOA98n3StIY8+VNT0z+C8z72jmkm2AD/T98kOEPCkgCuBMnrcf+WX3y/pbvWLKS99GpXzZBfvui/Gfh8KlG+99ZbMmzdPn4lMvYwZM0amTJkiiYmJEhBw+y//IUOGSHx8vOzZs8etgqWo3OJiZQRsJ+A88M16IUrqliti+mdw3md6T8Ca3gl2gICJAq4Eyvrjv5NDpy/qXgxs8pD8rUGoiT1yr2nGfh8KlB06dBA/Pz+ZM2fOHVUSGxsrgYGBKb63fPlyiYmJkRMnTkjRoq6fgaCo3PshZG0E7CbgPPDNebGm1A4tbPpHcGWwNb0T7AABEwVcqfGGf18pCb9d0L14tWmY9Hw0xMQeudc0Y78PBcpKlSpJUFCQXLlyRfbu3SvlypWTYcOGSZMmTSQiIkKaNWsm48aNc4js3r1b1DabNm2SyMjIdCvr/Pnzov5LWo4fPy5RUVFy5MgRKVHC3DdouFfurI0AAkYIOA98/+lWS2qWLWREs3dtw5XB1vROsAMETBRwpcYfm7BK9p/8U/diSGyYdKtHoDTxkNxz0/f1PJRXr16V3LlzS7Vq1WT8+PGSJ08emT17tkyaNEmWLVsm3bt3l06dOklcXJwDUN1rGRISImvWrJHo6Oh0YUeMGCEjR4684/sEynuuRTZEwNICzgPfZz0ekcjSBU3vryuDremdYAcImCjgSo03fmeV7D1xO1AObV5BXqxb1sQeudc0ZyiTve7rQKk+5oULF/T9kdmzZ3d86qZNm+oHdFQhqMveaZ2h3LJliw6i6S2coXTvh461EbC7gPPAF9+rtlQvWcD0j+TKYGt6J9gBAiYKuFLjTSeulj2//qF7od7Z3bVOGRN75F7TBEofCpRplcbAgQNl6dKlEhwcrKcKmjVrlmM17qF074eJtRHwFQHnge+L3nUkokSg6R/dlcHW9E6wAwRMFHClxpu/u0Z2Hbt9i1lcy0ry7CN3TgVoYhfv2jSB0kcC5fbt26V27do6PDpfvm7QoIEUK1ZMqlatKtOnT5cDBw6Iv7+/Vhk8eLAsXLiQp7y99dPJfhGwqIDzwLfk/6KlcvH8pvfUlcHW9E6wAwRMFHClxltMXivbfzmnezGqVWV5plYpE3vkXtMESh8JlDdv3tSBUl32njZtmhQuXFg++OADPc+keuhGhUo1D2WLFi1k0KBBsnHjRunZs6cOmV26dHGrqigqt7hYGQHbCTgPfF/3rSsVgsx/PZ0rg63tIOkwAk4CrtR4qynrZNuRs3qrMW3CpWNUScsYMvb7SKBUH/PUqVPy2muvyVdffSVnz56V6tWry9ixY6Vu3bpaQQVL9XacrVu36qfB+/fvL71793a7WCkqt8nYAAFbCTgPfN++Uk/KF8trev9dGWxN7wQ7QMBEAVdq/Mlp6+V/h37XvXj7yQhpXyPYxB651zRjvw8FSvdK497Xpqju3Y4tEbCDgPPAt6J/fQkpksf0brsy2JreCXaAgIkCrtR4++nfy8bEM7oXf29XRdo+bJ2p+Rj7CZSG/3hQVIaT0iAClhJwHvhWDnhUShfObXr/XBlsTe8EO0DARAFXarzDB9/LhoTbgfKdp6pI62oEShMPyT03fd9PG3TPMm5uSKB0E4zVEbCZgPPAt2ZQAwkumMv0T+DKYGt6J9gBAiYKuFLjT3+4QdbtP617MalDVWlZtbiJPXKvacZ+zlC6VzEurE1RuYDEKgjYWMB54Fv3WkMpHpjT9E/jymBreifYAQImCrhS450/+kHW7PtN92Jyp2ryeMSDJvbIvaYZ+wmU7lWMC2tTVC4gsQoCNhZwHvh+GNJIiuXLYfqncWWwNb0T7AABEwVcqfHnPt4oK38+pXsx7enq0iw8yMQeudc0Yz+B0r2KcWFtisoFJFZBwMYCzgPfptcfkyJ5k9++ZdbHqjz8G/nzynXdfOLY5mbthnYR8JqAK4HyhU82yYo9J3Ufpz/zsDSt/IDX+pt6x4z9BErDi5GiMpyUBhGwlIDzwLdlWIwUzJ3N9P7t+OWcDP18h3SvHyKxFjorY/oHZwc+I7Bq7yn5x7c/y+uxFaRm2UJpfu6X/rVZlu0+ob8349lIialYzDI+jP0ESsOLkaIynJQGEbCUgHOg/PGNxpI/V1ZL9Y/OIHC/CnSftVm+2XU7UM58LlIahhEorXisecrboKNCoDQIkmYQsKiAc6DcObKJ5MkeYNGe0i0E7i+BHrP+J0t3/ao/1Cdda8ijDxW1zAdk7OcMpeGYU+G8AAAgAElEQVTFSFEZTkqDCFhKwDlQ/hTXVHJm87dU/+gMAvergHOgnPVClNQtV8QyH5Wxn0BpeDFSVIaT0iAClhJwDpQ/j2oq2QMIlJY6QHTmvhVwDpRzXqwptUMLW+azMvYTKA0vRorKcFIaRMBSAs6Bcv/oZhLg72ep/tEZBO5XAedAObdbLamVzsM73vj8jP02DJSXLl0Sf39/yZYtm/z000/y5ZdfSu3atfV/VlgoKiscBfqAgHkCzoHy4JhYyZIli3k7o2UEEHAIOAfK+T0ekRqlC1pGh7HfZoFy9erV0qpVK/nss8+kYsWKUr58efHz85M///xTPv30U2nXrp3Xi4ui8vohoAMImCrgynx5pnaAxhHwUQHnQLmgZ215uFQBy0gw9tssUEZHR+sQOWnSJJk5c6aMHz9e9u3bp///hx9+KFu3bvV6cVFUXj8EdAABUwUIlKby0jgC6Qo4B8rP/1ZHqgYHWkaLsd9mgTJXrlyyc+dOKVu2rMTGxkpwcLC8//77cujQIQkLCxN1OdzbC0Xl7SPA/hEwV4BAaa4vrSOQnoBzoFzcO1rCS+S3DBZjv80CZZEiRWTNmjVSpkwZKVSokHz88cf6MvcPP/wgLVu2lF9/vT0/lTcXisqb+uwbAfMFCJTmG7MHBNIScA6UX/aJlkoPEiitWCm2mNi8ffv2+ixkwYIFZcGCBXL8+HE5cOCAdO/eXZ+1VPdRenshUHr7CLB/BMwVIFCa60vrCLhyhnLpy3Ul7IF8lsFi7LfZGcpTp05Jjx49JCEhQYYPH64f0Onfv78+Q6ke1HngAe+/KJ6isszPNx1BwBQBAqUprDSKQIYCzmcol71ST8oVy5vhNp5agbHfZoEyrcK4evWqnkLIKgtFZZUjQT8QMEeAQGmOK60ikJGA87u8V/SvLyFF8mS0ice+z9hvw0CpLnPPmDFD9uzZIxMnTpRVq1ZJeHi4fijHCgtFZYWjQB8QME+AQGmeLS0jcDcB50C5csCjUrpwbsuAMfbbLFDu379fatasKfnz5xd18FSofPXVV+Xrr7+Wb7/91hKTm1NUlvn5piMImCJAoDSFlUYRyFDAOVCuGdRAggvmynAbT63A2G+zQKme5FZPeqszlPny5ZMff/xRSpUqJc8995wcPnxYn6309kJRefsIsH8EzBUgUJrrS+sIpCfgHCjXvdZQigfmtAwWY7/NAqUKkyo0qrfk5M2bVwdK9XS3OlOpzlyeO3fO68VFUXn9ENABBEwVIFCaykvjCKQr4PxQzobBjeSB/Dkso8XYb7NAWaBAAT0PZeXKlVMESvWUd7NmzeTMmTNeLy6KyuuHgA4gYKoAgdJUXhpHwKVAuXFIIymaj0BpxXKxxTyUHTp00Hb//ve/JTAwULZv364vfbdp00aKFi0q8+fP97otgdLrh4AOIGCqAIHSVF4aR8ClQLl56GNSOE92y2gx9tvsDOWxY8fk0UcfldOnT+vL2+rJbvXaRTXR+erVq/X9lN5eKCpvHwH2j4C5AgRKc31pHYH0BJwveW8dFiMFcjNloBWrxRZnKBXcxYsX9Rtxtm7dKjdv3tSXv5955hl9ptLVZe/evVK9enWZPHmyfqBHLdu2bZO+ffvK5s2b9Wsd+/TpIwMGDHC1Scd6BEq3ydgAAVsJEChtdbjo7H0k4BwofxzeWPLnzGqZT8fYb7MzlEndvXz5sqhQmD17dv1QTtasrhfVtWvX9PRCKjiqd4GrQKnOeKqznerNO/369ZMNGzZIr169ZOrUqdK1a1e3CpaicouLlRGwnUD1N5fJmQtXdb8Txza3Xf/pMAJ2FXAOlDtHNpE82QMs81EY+20WKG/cuKHPGk6fPl3UG3LUkjNnTj0X5dChQyVLliwZFteQIUN0YPzuu+8cgXLMmDEyZcoUSUxMlICA2wWq1ouPj9dPkLuzUFTuaLEuAvYT+PHIWek5+3/yQt2y8kJ0Gft9AHqMgE0FnAPlT3FNJWc2f8t8EsZ+mwXK119/XWbOnClvvvmmniZIBUz11PeIESOkd+/eMnLkyLsWl7rPsnXr1vrydsmSJR2BMjY2Vj/kM2fOHMf2y5cvl5iYGDlx4oR+4MfVhaJyVYr1EEAAAQQQcF3AOVD+PKqpZA8gULqu57k1bXEPZbFixXQIVAHQeVmwYIEOlOq1jOktZ8+elSpVqog6G9mpUyd9NjPpkndERISedmjcuHGOzXfv3i2VKlWSTZs2SWRkpMtHgkDpMhUrIoAAAggg4LKAc6DcP7qZBPj7ubyt2Ssy9tvsDKU6i/j9999LhQoVUtSGCn/qjOUff/yRbs107NhRbt26JXPnztXrOAfK0NBQHTLj4uIc2yckJEhISIg+AxodHZ1uu+fPnxf1X9KiQm1UVJQcOXJESpQoYXYN0z4CCCCAAAI+IeAcKA+OiXXpNjdPwRAobRYo1VPYJ0+elH/+85+SLVvydAHPP/+8vvfxgw8+SLN2Zs2aJYMHD5YdO3aImhw9daAMDw/XZz3TOkO5ZcsWqVatWro1qS63p3WpnUDpqR9j9oMAAggg4AsCzoHSag/EEShtECgbNmzo6OX169dl7dq1EhQUJDVq1BB/f389fZAKb+o935999lmaP1MNGjTQ26mnwpOWCxcu6K/VWUg1f6WaKkgFz6TF1XsoOUPpC7/G+IwIIIAAAt4WIFB6+wi4tn/L3kPpzrQ96p7ItJajR4/KpUuXUnyrXLly+ozkU089pee1VE+OHzhwQIdUtagzmgsXLuQpb9fqh7UQQAABBBAwVYBAaSqvYY1bNlAa9glTNeR8D6W6jK7moWzRooUMGjRINm7cKD179tQhs0uXLm51gdPebnGxMgIIIIAAAi4JTP7vPvn7t3v1ulzydonMKyvZJlCqwKbmjFTv8VYTmqsnsbt3766nAXJncQ6Uajv1NLd6O466hK4uqffv318/Oe7uQqB0V4z1EUAAAQQQyFjgyvUb8vbSn6V6yQLSPCIo4w08uAZjfzK2LQLlzp07pV69epIrVy79JLWah1K98UbdD7lu3TodLr29UFTePgLsHwEEEEAAAc8KMPbbLFCquSJz584t//73vx0P2KjXMKp3eav/XbJkiWcrKI29UVRePwR0AAEEEEAAAY8KMPbbLFDmyZNHvzaxcuXKKQpFXf5WZy7V5OXeXigqbx8B9o8AAggggIBnBRj7bRYo1b2N6izkww8/nKJS1P2P6jWJBErP/gCxNwQQQAABBBAQIVDaLFCqS9vHjh3T800WLFhQ9/63336T9u3b63dxx8fHe72uKSqvHwI6gAACCCCAgEcFGPttFijVAatdu7acOXNGypcvr1+79PPPP+u336xatUrKli3r0QJKa2cUldcPAR1AAAEEEEDAowKM/TYLlOpd3OqJbvVGG/XEt/pavTZRvYc7f/78Hi2e9HZGUVniMNAJBBBAAAEEPCbA2G+zQBkZGSkzZ86UiIgIjxWJuzuiqNwVY30EEEAAAQTsLcDYb7NAWbhwYfnhhx/0+7etulBUVj0y9AsBBBBAAAFzBBj7bRYox48fL19++aUMHDhQQkNDJWfOnCkqw9235ZhRVhSVGaq0iQACCCCAgHUFGPttFij9/PwcPVYP5CQt6l5K9bV6c463F4rK20eA/SOAAAIIIOBZAcZ+mwVK9ST33Zb69et7toLS2BtF5fVDQAcQQAABBBDwqABjv80CpXN1qKmD/P39LfN0d1LfKCqP/gyzMwQQQAABBLwuwNhvw0Cp7qOcNGmSHD9+XPe+TJky8uqrr8pLL73k9YJSHaCoLHEY6AQCCCCAAAIeE2Dst1mgHDdunMTFxUmfPn3kkUcekZs3b8q6detk6tSp8s4770i3bt08Vjzp7Yii8vohoAMIIIAAAgh4VICx32aBMjg4WN566y3p3LlzikJRc1OOGTNG9u3b59ECSmtnFJXXDwEdQAABBBBAwKMCjP02C5R58uSRLVu26NcuOi8qSFapUkUuXrzo0QIiUHqdmw4ggAACCCDgdQECpc0CZZcuXSRv3rwyefLkFMUzcuRI/SrG+fPnU1ReF6ADCCCAAAII+JYAgdJmgbJfv376fskKFSqImiIoa9assmnTJlmzZo20bNlSAgMDHZ9IXQb3xkJReUOdfSKAAAIIIOA9AcZ+mwXKBg0auFwt3333ncvrGrkiRWWkJm0hgAACCCBgfQHGfpsFSuuXFNMG2eEY0UcEEEAAAQSMFCBQEiiNrCfdFkVlOCkNIoAAAgggYGkBxn4CpeEFSlEZTkqDCCCAAAIIWFqAsZ9AaXiBUlSGk9IgAggggAAClhZg7LdBoPziiy+kcePGkiNHDksXU1LnKCpbHCY6iQACCCCAgGECjP02CJRqMvM9e/ZIiRIlpGzZsnqaoEKFChlWBEY3RFEZLUp7CCCAAAIIWFuAsd8GgbJo0aLSvXt3eeyxx0RNG7Rw4UIpUKBAmpVVr149r1ccReX1Q0AHEEAAAQQQ8KgAY78NAuW4ceNkyJAhcuvWLcmSJYv+37QW9b0bN254tIDS2hlF5fVDQAcQQAABBBDwqABjvw0CperiuXPn5OzZs1KmTBnZuHGjFClSJM1CKVWqVLoFdPLkSenfv78sXbpULl26pN+0M378eKlYsaLeZtu2bdK3b1/ZvHmzvqTep08fGTBggNsFSVG5TcYGCCCAAAII2FqAsd8mgTKpm6tWrZI6depIQECA24VXq1Yt8fPzk3fffVdy584tw4YNk/Xr18v+/ft1wAwLC5NWrVqJer3jhg0bpFevXvo1j127dnVrXxSVW1ysjAACCCCAgO0FGPttFihVd5csWSKjR4+W7du363d5qzOMAwcOlNatW6dbkKdPn5bevXvL0KFDpVKlSnq9H3/8UapWrarPeC5fvlymTJkiiYmJjrCqLrPHx8frB4LcWSgqd7RYFwEEEEAAAfsLMPbbLFAuWrRI2rRpo8OjegBH3TO5evVqWbx4sQ5/LVq0cKkqT5w4oS9nr1y5UgfGdu3aSWBgoMyZM8exvQqZMTExotZVDwa5ulBUrkqxHgIIIIAAAveHAGO/zQJl9erVdZhUl6udl7i4OH3mUp1tzGjp1q2bzJgxQ7Jnzy5Jc1xGRERIs2bNRD0AlLTs3r1bn81U0xRFRkam2+z58+dF/Ze0HD9+XKKiouTIkSN6qiMWBBBAAAEEELi/BQiUNguUOXPmlB07dkhoaGiKyty3b59UqVJFLl68mGHFqqCo7pmcNm2aPiO5du1aad++vXTq1ElUME1aEhISJCQkRNasWSPR0dHptjtixAgZOXLkHd8nUGZ4KFgBAQQQQACB+0KAQGmzQKkC3nvvvSexsbEpCvDLL7+Ul156SY4dO+ZyYd68eVPCw8OlZs2a+iykajOtM5RbtmyRatWqcYbSZVlWRAABBBBAwLcECJQ2C5TqUvesWbP02UX1tLda1BlG9UR2y5YtZdKkSWlWsJoyaMWKFfpMpL+/v2Mdde+kulyt/k1NFaTaTlq4h9K3fhnwaRFAAAEEELhXAQKlzQLl5cuXpUOHDvreRzWRuVrUROfq7OK8efMkV65cadaCmmNSnWVUobJhw4Z6nWvXrumpgtSDPMWKFZPp06fLgQMHHIFz8ODB+q08POV9rz9ebIcAAggggIBvCBAobRYok7qrQp66l1KFSXXZukKFCnetWLWeeuhGTQukHshRr25UUw998803ekLzHDlyOMLloEGD9MM9PXv21CGzS5cubv00UFRucbEyAggggAACthdg7LdpoLyXylNv21FnHT///HP91p26devKhAkTHPNSqvso1dtxtm7dKkFBQfqtOmruSncXispdMdZHAAEEEEDA3gKM/T4UKD1VqhSVp6TZDwIIIIAAAtYQYOwnUBpeiRSV4aQ0iAACCCCAgKUFGPsJlIYXKEVlOCkNIoAAAgggYGkBxn4CpeEFSlEZTkqDCCCAAAIIWFqAsd9mgfLgwYPy6quvys6dO0VNIZR6UW+38fZCUXn7CLB/BBBAAAEEPCvA2G+zQKlegajela0mKFdT/aRehg8f7tkKSmNvFJXXDwEdQAABBBBAwKMCjP02C5S5c+fW79auXr26RwvFnZ1RVO5osS4CCCCAAAL2F2Dst1mgVEFy4sSJUq9ePctWH0Vl2UNDxxBAAAEEEDBFgLHfZoFyw4YN+g02agLyMmXKiJ+fX4rCsELQpKhM+VmlUQQQQAABBCwrwNhvs0D50Ucf6UB5/fr1O4pKvdv7xo0bXi82isrrh4AOIIAAAggg4FEBxn6bBcrg4GB5/PHH9RnKXLly3VEspUqV8mgBpbUzisrrh4AOIIAAAggg4FEBxn6bBUr1UM6uXbukdOnSHi0Ud3ZGUbmjxboIIIAAAgjYX4Cx32aBsm3bthIbGyvPP/+8ZauPorLsoaFjCCCAAAIImCLA2G+zQDl58mQZPHiwNGvWTB566CHJmjVrisJ44403TCkUdxqlqNzRYl0EEEAAAQTsL8DYb7NAqZ7sTm9RD+Xwphz7/1DyCRBAAAEEELCbAIHSZoHSDgVGUdnhKNFHBBBAAAEEjBNg7CdQGldNf7VEURlOSoMIIIAAAghYWoCx32aBUk1kri5tp7cwD6Wlf97oHAIIIIAAAvelAIHSZoHyk08+SREor127Jvv27RP17xMmTJCnn37a64VKUXn9ENABBBBAAAEEPCrA2G+zQJledcyZM0dmzZolX3/9tUcLKK2dUVRePwR0AAEEEEAAAY8KMPbfJ4Hy0KFDUrFiRblw4YJHC4hA6XVuOoAAAggggIDXBQiU90mgHDt2rEyZMkWOHDlCUXldgA4ggAACCCDgWwIESpsFSjUPpfNDObdu3ZLz58/L2bNnZfTo0fLaa695vYIpKq8fAjqAAAIIIICARwUY+20WKEeMGHHHU97ZsmWTOnXqSL169TxaPOntjKKyxGGgEwgggAACCHhMgLHfZoHSY5WRiR1RVJnAY1MEEEAAAQRsKMDYb8NAuW7dOlm7dq1cvXpV1CVv54V3edvwp5AuI4AAAgggYHMBAqXNAqV6+GbIkCFSoEAByZcvX4ry413eNv9ppPsIIIAAAgjYVIBAabNAWaJECXn++eclLi7O7ZI7c+aMDqNLlizRD/JERESICqjR0dG6rW3btknfvn1l8+bNUqhQIenTp48MGDDA7f1QVG6TsQECCCCAAAK2FmDst1mgzJUrl2zfvl1CQ0PdLrzGjRvLyZMnZfLkyVKkSBE9zdCHH34oW7Zs0V+HhYVJq1atpF+/frJhwwbp1auXTJ06Vbp27erWvigqt7hYGQEEEEAAAdsLMPbbLFCqUKgCXseOHd0qvv3790u5cuVE3X9Zu3Ztva26/7J8+fK6rZw5c+qAmZiYKAEBAfr76mxmfHy87Nmzx619UVRucbEyAggggAACthdg7LdZoFRnFAcNGqRDpTqjmD179hRF+Oyzz6ZZlGqeyvXr10ujRo1SbKPOdDZt2lQSEhIkMDBQ1Csck5bly5dLTEyMnDhxQooWLepysVNULlOxIgIIIIAAAveFAGO/zQKln59fuoWnHsq5ceOGy4U5f/58ad++vXzxxRfy+uuvS7NmzWTcuHGO7Xfv3i2VKlWSTZs2SWRkpMvtUlQuU7EiAggggAAC94UAY7/NAqVRVacufaszk+qM5eeff67vyezUqVOKh33UWcuQkBBZs2aN48GdtPavHvBR/yUtx48fl6ioKP0aSPUQEQsCCCCAAAII3N8CBEofDJSLFi3S4bFWrVr6iW91/2R4eLjExsameYZSPbRTrVq1dH8S1Nt7Ro4cecf3CZT39y8PPh0CCCCAAAJJAgRKHwuU6glvNTVQmzZtZPbs2Y77KVWYVFMFzZo1yyHi6j2UnKHkFwoCCCCAAAK+LUCg9KFAOW3aND0VkJpf8p133hHn+zHVfJTTp0+XAwcOiL+/v1YZPHiwLFy4kKe8fft3BJ8eAQQQQACBDAUIlD4SKPfu3SuVK1eWxx9/XM8t6byoS95XrlzRT423aNFCP0W+ceNG6dmzpw6ZXbp0ybCQnFegqNziYmUEEEAAAQRsL8DY7yOB8q233tJPcqe1qMD4ySef6Ke51dnLrVu3SlBQkPTv31969+7tdpFTVG6TsQECCCCAAAK2FmDs95FA6ckqpag8qc2+EEAAAQQQ8L4AYz+B0vAqpKgMJ6VBBBBAAAEELC3A2E+gNLxAKSrDSWkQAQQQQAABSwsw9hMoDS9QispwUhpEAAEEEEDA0gKM/QRKwwuUojKclAYRQAABBBCwtABjP4HS8AKlqAwnpUEEEEAAAQQsLcDYT6A0vEApKsNJaRABBBBAAAFLCzD2EygNL1CKynBSGkQAAQQQQMDSAoz9BErDC5SiMpyUBhFAAAEEELC0AGM/gdLwAqWoDCelQQQQQAABBCwtwNhPoDS8QCkqw0lpEAEEEEAAAUsLMPYTKA0vUIrKcFIaRAABBBBAwNICjP0ESsMLlKIynJQGEUAAAQQQsLQAYz+B0vACpagMJ6VBBBBAAAEELC3A2E+gNLxAKSrDSWkQAQQQQAABSwsw9hMoDS9QispwUhpEAAEEEEDA0gKM/QRKwwuUojKclAYRQAABBBCwtABjP4HS8AKlqAwnpUEEEEAAAQQsLcDYT6A0vEApKsNJaRABBBBAAAFLCzD2EygNL1CKynBSGkQAAQQQQMDSAoz9BErDC5SiMpyUBhFAAAEEELC0AGM/gdLwAqWoDCelQQQQQAABBCwtwNhPoDS8QCkqw0lpEAEEEEAAAUsLMPYTKA0vUIrKcFIaRAABBBBAwNICjP0ESsMLlKIynJQGEUAAAQQQsLQAYz+B0vACpagMJ6VBBBBAAAEELC3A2E+gNLxAKSrDSWkQAQQQQAABSwsw9hMoDS9QispwUhpEAAEEEEDA0gKM/T4aKEeNGiXLly+XlStXOgS2bdsmffv2lc2bN0uhQoWkT58+MmDAALcLmKJym4wNEEAAAQQQsLUAY78PBsqJEydKv379pF69eo5Aefr0aQkLC5NWrVrp723YsEF69eolU6dOla5du7pV5BSVW1ysjAACCCCAgO0FGPt9KFAePXpUXnzxRVmzZo0EBwdLsWLFHIFyzJgxMmXKFElMTJSAgACtMmTIEImPj5c9e/a4VegUlVtcrIwAAggggIDtBRj7fShQLl68WD799FNRl7vj4uJ0eEy65B0bGyuBgYEyZ84ch4i6JB4TEyMnTpyQokWLulzsFJXLVKyIAAIIIIDAfSHA2O9DgdK5Yp977rkUgTIiIkKaNWsm48aNc6y2e/duqVSpkmzatEkiIyPTLfjz58+L+i9pOX78uERFRcmRI0ekRIkS98UPCh8CAQQQQAABBNIXIFASKLVAaGiodOrUSZ+5TFoSEhIkJCREXyKPjo5Ot4pGjBghI0eOvOP7BEp+9SCAAAIIIOAbAgRKAqUWCA8PF3XZO60zlFu2bJFq1apxhtI3fifwKRFAAAEEEHBbgEBJoNQCKkyqqYJmzZrlEOEeSrd/ntgAAQQQQAABnxQgUBIotcDYsWNl+vTpcuDAAfH399f/NnjwYFm4cCFPefvkrwY+NAIIIIAAAq4LECgJlFrg5MmTeh7KFi1ayKBBg2Tjxo3Ss2dPHTK7dOniekWJCEXlFhcrI4AAAgggYHsBxn4CpUNAPc2t3o6zdetWCQoKkv79+0vv3r3dLnKKym0yNkAAAQQQQMDWAoz9PhoozaxaispMXdpGAAEEEEDAegKM/QRKw6uSojKclAYRQAABBBCwtABjP4HS8AKlqAwnpUEEEEAAAQQsLcDYT6A0vEApKsNJaRABBBBAAAFLCzD2EygNL1CKynBSGkQAAQQQQMDSAoz9BErDC5SiMpyUBhFAAAEEELC0AGM/gdLwAqWoDCelQQQQQAABBCwtwNhPoDS8QCkqw0lpEAEEEEAAAUsLMPYTKA0vUIrKcFIaRAABBBBAwNICjP0ESsMLlKIynJQGEUAAAQQQsLQAYz+B0vACpagMJ6VBBBBAAAEELC3A2E+gNLxAKSrDSWkQAQQQQAABSwsw9hMoDS9QispwUhpEAAEEEEDA0gKM/QRKwwuUojKclAYRQAABBBCwtABjP4HS8AKlqAwnpUEEEEAAAQQsLcDYT6A0vEApKsNJaRABBBBAAAFLCzD2EygNL1CKynBSGkQAAQQQQMDSAoz9BErDC5SiMpyUBhFAAAEEELC0AGM/gdLwAqWoDCelQQQQQAABBCwtwNhPoDS8QCkqw0lpEAEEEEAAAUsLMPYTKA0vUIrKcFIaRAABBBBAwNICjP0ESsMLlKIynJQGEUAAAQQQsLQAYz+B0vACpagMJ6VBBBBAAAEELC3A2E+gNLxAKSrDSWkQAQQQQAABSwsw9hMoDS9QispwUhpEAAEEEEDA0gKM/QRKwwuUojKclAYRQAABBBCwtABjP4HS8AKlqAwnpUEEEEAAAQQsLcDYT6BMUaA3b96UkSNHyocffii///67REdHy9SpUyU0NNTlQqaoXKZiRQQQQAABBO4LAcZ+AmWKQlZhUgXIjz/+WIoXLy6DBg2ShIQE2bVrl2TLls2loqeoXGJiJQQQQAABBO4bAcZ+AqVD4OrVq1K4cGF5++23pUePHvrfz549Kw8++KDMnDlTOnTo4FLhU1QuMbESAggggAAC940AYz+B0iGwceNGqVmzpvz8889Svnx5x7+ry94RERH6zKUrC0XlihLrIIAAAgggcP8IMPYTKB0C8fHx8uSTT8rFixclZ86cjn9v3769/rclS5a4VPkUlUtMrIQAAggggMB9I8DYT6B0CMyePVs6d+4sN27cED8/P8e/P/vss3Ls2DFZvnx5moV//vx5Uf8lLUeOHJHatWuLOuMZFBR03/yw8EEQQAABBBBAIG2B48ePS1RUlBw8eHjJpSEAAA15SURBVFBKly7t00xZbt26dcuXBRYsWCBt27ZN8wzllStXZNGiRWnyjBgxQj8ZzoIAAggggAACvi2gTibVqFHDpxF8PlAm3UO5f/9+CQkJcRSDuoeySpUqMmXKFJfOUF6+fFnUWcoyZcpIQECAoUWV9BcQZz8NZb1rY5h7zjppT5h73lztEXfPu2PueXOzav369ety6tQpCQ8Plxw5cnjng1lkrz4fKNVZyKJFi8qECRPkhRde0IflXp7yNvN4co+Gmbppt4055p4X8M4eqXXPu2PueXO1R9zNdff5QKl4X3/9dXn//ff1NEHqHoiBAwdKYmKi7Nixw+V5KM08TPwQmKlLoPS8LuZWMWeQ9c6R4Hc67t4RMHevBEoR/UDOkCFD9MTmly5dknr16ulL3Va5wZZfPub+EKTVOuaYe17AO3uk1j3vjrnnzfnjyXxzAqX5xpneg3qaXF2S79evn+TLly/T7dFAxgKYZ2xk9BqYGy3qWnu4u+Zk5FqYG6npelu4u251L2sSKO9FjW0QQAABBBBAAAEEHAIESooBAQQQQAABBBBAIFMCBMpM8bExAggggAACCCCAAIGSGkAAAQQQQAABBBDIlACBMlN85m588+ZN/TaeDz/8UH7//XdRk61PnTpVQkNDzd2xj7Q+atQo/WrNlStXOj7xtm3bpG/fvrJ582YpVKiQ9OnTRwYMGOD4PsfE/eI4c+aMnkVhyZIl+nWlERERMnbsWF3PasHcfVNXtjh58qT0799fli5dqmevqF+/vowfP14qVqyIuyuAmVxn7969Ur16dZk8ebI899xzmGfS826bHzp0KM1ZWWbMmCEvvvgiv2NMtHdumkDpIeh72Y0KkypAqumMihcvLoMGDZKEhATZtWuXJebHvJfPZJVtJk6cqJ+aV1NEJQXK06dPS1hYmLRq1Up/b8OGDdKrVy99DLp27aq7zjFx/wg2btxYVLhRA2uRIkX0lFzqj6QtW7borzF339SVLWrVqiV+fn7y7rvvSu7cuWXYsGGyfv16UW8FUwETd1cU722da9euSe3atfUfpur3twqU/H65N0tXtlq8eLG0b99ej49ZsmRxbJI/f379WmVq3RXFzK9DoMy8oSktXL16VQoXLixvv/229OjRQ+/Dam/wMeWDm9zo0aNH9V+sa9askeDgYClWrJgjUI4ZM0aHHTWpfdLrM9WZtfj4eNmzZ49wTNw/OCq8lCtXTtatW6cHWLXcunVLypcvLx07dpScOXNi7j5rhluo8NK7d28ZOnSoVKpUSa//448/StWqVUW9wlWdmafWM2S85xXU7w31B+l3333nCJT8frlnzgw3fOutt2TevHn6TGTqBfcM+QxbgUBpGKWxDSW9Y/znn3/Wg2/Soi4TqkuG6qwZi/sC6i/ZTz/9VNTl7ri4OB0ek85QxsbGSmBgoMyZM8fRsBp4Y2Ji5MSJE3rdmjVrCsfEdXf1R5A6K9aoUSPJnj27Y0N120bTpk31GQXMXfe81zVV/apbN1Stqz+O2rVrh/u9Ymaw3erVq6V169Y63JQsWdIRKPn9YhK4iHTo0EGfjXf+3Z20N9zNc0/dMoHSc9Zu7UmdFXvyySf16Xp1FidpUaf11b+p+9FYMiegLkM5B0oV1Js1aybjxo1zNLx79259hmfTpk1y+PBhjknmyPXW8+fP15envvjiC/3aU8wNQL1LE926dRN1L5kK9Mpc3YJArZtjrv6AqlKliqizYp06ddKXX5MueWNujrlqVf2ODgoKkitXroi6d1VdFVG3eDRp0oRaN4/9jpYJlB7EdmdXs2fPls6dO+vXQqq/vJKWZ599Vo4dO6YvWbFkTiB1oFRnzdQgoM5cJi3qDFpISIi+RK7CJ8ckc+bq0rc6M6nOWH7++ef6ATPMM2ea0dbqjyJ1z+S0adP0GZy1a9fqQI97RnLuf1/dxqFu6Zg7d67e2DlQUuvue7qyhboVSd0jXK1aNf3QWZ48eUSNn5MmTZJly5ZJ9+7dqXVXIA1Yh0BpAKIZTSxYsEDatm2b5hlK9VfYokWLzNitT7WZOlCGh4eLujyS1hlK9QCJCpcck3svEVWzKsSoh0XUGXZ15h3ze/d0d0s1Q4HyVrdtqDPu1Lq7gndff9asWTJ48GDZsWOHFChQ4I5ASa0b6+3c2oULF/R978631ag/XFWgV+9Np9bNs3dumUDpGWe395J0D6V6qEGdIUta1D2U6pKKuqGeJXMCqQOl+qWjpgpSA0PSktY9lBwT993VE95qOqY2bdroswdJv/gxd9/SlS3UU/UrVqzQZyL9/f0dm6h7J9XUTerfqHVXJF1fp0GDBvrsr3OoUUFHfa1+h5cqVQpz1zkzvebAgQP1lFnq4UtqPdOcLjVAoHSJyfMrqbOQRYsWlQkTJsgLL7ygO8BT3sYeh9SBUs2NOH36dDlw4IBjEFZnHBYuXKgfZOCY3Ju/utSqpl9Sc3q+8847KW7hwPzeTDPaSj0Qoi4BqlDZsGFDvbqaykZNn9KiRQs9uwG1npGie99XM0ioWwucF3Uvn7ri8dRTT+mHATF3z9SVtbdv365nkFDhMWluW7WdCviqztXMBri7Ipn5dQiUmTc0rQX1wML7778vM2fO1JO2qr+41H186pJKtmzZTNuvrzScOlCqszpJA66a81OdJe7Zs6f+ZdSlSxfNwjFxrzrUDfKVK1eWxx9//I6ZCdQlbxXSMXfP1JW11X186mEn9ftCPZCjLsGOHj1avvnmG/30cY4cOXB3BTKT6zjfQ8nvl0xiprO5upVDBUp1Nlj98aqm2/vggw/07xt1a4cKlfyOMcc+dasESs8439Ne1AM5aj4z9ZSg+stXTcKtLnWrcMmSeYHUgVK1qH4BqTNpW7du1U8NqjeNqPn8khaOiXvuan44FcLTWlRI/+STTzB3j9Tltc+dO6fv6VMPP6mrG3Xr1tVXPJLmpaTWXaa85xWdAyW/X+6ZMcMNT506Ja+99pp89dVXutbVG4rU1Q9V87hnyGfYCgRKwyhpCAEEEEAAAQQQ8E0BAqVvHnc+NQIIIIAAAgggYJgAgdIwShpCAAEEEEAAAQR8U4BA6ZvHnU+NAAIIIIAAAggYJkCgNIyShhBAAAEEEEAAAd8UIFD65nHnUyOAwD0IqBkW1OwAI0aMuIetM95k5cqVev68gwcPpjubg9l9yLiXrIEAAgjcKUCgpCoQQAABFwXU9CRq/kz1vmAzFvVe4jNnzkiRIkVSvOHGeV8ESjPkaRMBBDIrQKDMrCDbI4AAAh4UIFB6EJtdIYCAywIESpepWBEBBOwioCaUVhOqq/eyq7fxrFq1SsqUKSPDhg3T7xJXk36rN/jExcVJ48aNHR/r22+/1ZOB7969W8qWLasntlevPk26BJ06zH355Zfy5ptvys6dOyVfvnzSsWNH/UYa9SYatah+qLd2zJs3T7/nuWDBgnqifLWPtJbUl7xVP9VE+4sWLdJvx1IvOpg4caKpl93tcozpJwIIWEuAQGmt40FvEEDAAAEV5NQr2L7++mu5fv261KpVSzp16iS7du3SgaxEiRKyePFi/XYN9a725s2b61cS1qhRQ1555RUdItXX6tWbv//+e5qBUr2B5sknn9T3U7Zv31727dun14+KipIFCxY4AmX+/Pll8uTJ+t9VwB01apSsXr3a8RYP54+bOlA2adJEDh8+rENp3rx5dcD973//K8OHDzftPk4D+GkCAQR8UIBA6YMHnY+MwP0uoAKlCobqVYNq2b9/v5QrV06/5jEyMtLx8dXrHw8dOiQqyKn/r949/v333zu+/9577+kzhGmdoaxZs6YEBwfLZ5995lhfhdQWLVroM5wVKlTQZyj79u2rQ2zSos5SDhw4MM2zlM6BMuk958uXL5dGjRrpzU+cOCGlSpXSQdisB4Pu99rg8yGAgDkCBEpzXGkVAQS8KKCC3NSpU/UZQ7XMnz9fn0XMnTt3il5du3ZNAgMDdVALDw+XmJgYRwhVK6qzlNWqVUszUObKlUvUu8pffvllR5vqErVqb+7cufLUU0/pQDl9+nTp3r27Yx112bxr1676LGPqxTlQbt68Wdq1a6cvz6vL6UlLxYoV9WchUHqxwNg1AgjcIUCgpCgQQOC+E1BB7uOPP9b3GqrlP//5j3To0EG2bNmiLx07L/7+/vr+ShUc69evn+Js4t0CpXrae+zYsfoMZNKintAuVKiQvuTdpk0bHSid+6HWu9tDNWkFSnXJXYXUpKVKlSrSunVrAuV9V7V8IATsLUCgtPfxo/cIIJCGQOogp+6dVA/hLFmyRN8vmbSoB3f8/Pz0gzUqfO7Zs0c2bNjg+P77778vPXr0SPMMpbonsmTJkikueauHZ1q1apXikve9BsobN25IaGhoij6fPXtWihcvri+Zc4aS0kcAASsJECitdDToCwIIGCKQ1pnBJ554Ql/CnjJlig6X8fHxMmjQIPnoo4/0JWj1pHbVqlVlwIAB8vzzz+tQ2K1bN1FzTyYmJup7F53PLqrL6Oqy9siRI/X/qvsv1SV21Ya6l1ItmTlDqfalLm2r+z5nzJghDzzwgL7vUj1oNHToUAKlIZVCIwggYJTA/wMGui2ZCjL0AQAAAABJRU5ErkJggg==\" width=\"599.9999869953505\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from ImageAnalysis3 import library_tools\n",
    "primer_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Primers'\n",
    "primers_500_short = library_tools.assemble.load_primers([10,17], _primer_folder=primer_folder)\n",
    "\n",
    "sub_library_500_short_records = []\n",
    "for _reg, _pbs in sub_library_500.items():\n",
    "\n",
    "    if len(_pbs) >= short_probe_num:\n",
    "        _sel_num = short_probe_num\n",
    "    else:\n",
    "        _sel_num = len(_pbs)\n",
    "        \n",
    "    sub_library_500_short_records.extend(_pbs[:_sel_num])\n",
    "    \n",
    "# replace primers\n",
    "for _irecord, _record in enumerate(sub_library_500_short_records):\n",
    "    _seq = primers_500_short[0].seq[-primer_len:] \\\n",
    "        + _record.seq[primer_len:-primer_len] \\\n",
    "        + primers_500_short[1].seq[-primer_len:].reverse_complement()\n",
    "    _name = _record.name.split('primers_[')[0] + 'primers_['+primers_500_short[0].id +','+ primers_500_short[1].id+']' \\\n",
    "    + '_library_1000-mouse-genome-short_500'\n",
    "    sub_library_500_short_records[_irecord] = SeqRecord(_seq, name=_name, id=_name, description='')\n",
    "    \n",
    "# save kept records\n",
    "print(sub_library_500_short_records[0])\n",
    "print(len(sub_library_500_short_records))\n",
    "with open(os.path.join(library_folder, 'filtered_full_probes_sublibrary_500_short.fasta'), 'w') as output_handle:\n",
    "    SeqIO.write(sub_library_500_short_records, output_handle, \"fasta\")\n",
    "    \n",
    "    \n",
    "pb_num_dict = {_k:len(_v) for _k,_v in library_tools.quality_check.split_probe_by_gene(sub_library_500_short_records).items()}\n",
    "\n",
    "pb_nums = np.array(list(pb_num_dict.values()))\n",
    "plt.figure(figsize=(6,3), dpi=100)\n",
    "plt.plot(pb_nums)\n",
    "#plt.plot(np.ones(len(pb_nums))*20)\n",
    "plt.ylabel(f\"num of probes\", labelpad=1)\n",
    "plt.xlabel(f\"region id\", labelpad=1)\n",
    "plt.ylim([0, short_probe_num*1.1])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "823dceca",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Picked primer: ID: W1A05_primer_4\n",
      "Name: W1A05_primer_4\n",
      "Description: W1A05_primer_4\n",
      "Number of features: 0\n",
      "Seq('CATTCAGCATTGCGCAACGG')\n",
      "- Picked primer: ID: W1A04_primer_3\n",
      "Name: W1A04_primer_3\n",
      "Description: W1A04_primer_3\n",
      "Number of features: 0\n",
      "Seq('TAATACGACTCACTATAGGGCGTTGTATGCCCTCCACGC')\n",
      "ID: loc_1:6240000-6260000_gene_2_pb_58_pos_5958_strand_-_readouts_[NDB_965_c,NDB_883_c,NDB_742_c]_primers_[W1A05_primer_4,W1A04_primer_3]_library_1000-mouse-genome-short_250_1\n",
      "Name: loc_1:6240000-6260000_gene_2_pb_58_pos_5958_strand_-_readouts_[NDB_965_c,NDB_883_c,NDB_742_c]_primers_[W1A05_primer_4,W1A04_primer_3]_library_1000-mouse-genome-short_250_1\n",
      "Number of features: 0\n",
      "Seq('CATTCAGCATTGCGCAACGGCTTAGTCTCGCGCGCCGGATTTTCCCAGGGTGAG...CGC')\n",
      "14988\n"
     ]
    },
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:,\" width=\"0\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from ImageAnalysis3 import library_tools\n",
    "primer_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Primers'\n",
    "primers_250_1_short = library_tools.assemble.load_primers([4,3], _primer_folder=primer_folder)\n",
    "\n",
    "sub_library_250_1_short_records = []\n",
    "for _reg, _pbs in sub_library_250_1.items():\n",
    "\n",
    "    if len(_pbs) >= short_probe_num:\n",
    "        _sel_num = short_probe_num\n",
    "    else:\n",
    "        _sel_num = len(_pbs)\n",
    "        \n",
    "    sub_library_250_1_short_records.extend(_pbs[:_sel_num])\n",
    "    \n",
    "# replace primers\n",
    "for _irecord, _record in enumerate(sub_library_250_1_short_records):\n",
    "    _seq = primers_250_1_short[0].seq[-primer_len:] \\\n",
    "        + _record.seq[primer_len:-primer_len] \\\n",
    "        + primers_250_1_short[1].seq[-primer_len:].reverse_complement()\n",
    "    _name = _record.name.split('primers_[')[0] + 'primers_['+primers_250_1_short[0].id +','+ primers_250_1_short[1].id+']' \\\n",
    "    + '_library_1000-mouse-genome-short_250_1'\n",
    "    sub_library_250_1_short_records[_irecord] = SeqRecord(_seq, name=_name, id=_name, description='')\n",
    "    \n",
    "# save kept records\n",
    "print(sub_library_250_1_short_records[0])\n",
    "print(len(sub_library_250_1_short_records))\n",
    "with open(os.path.join(library_folder, 'filtered_full_probes_sublibrary_250_1_short.fasta'), 'w') as output_handle:\n",
    "    SeqIO.write(sub_library_250_1_short_records, output_handle, \"fasta\")\n",
    "    \n",
    "pb_num_dict = {_k:len(_v) for _k,_v in library_tools.quality_check.split_probe_by_gene(sub_library_250_1_short_records).items()}\n",
    "\n",
    "pb_nums = np.array(list(pb_num_dict.values()))\n",
    "plt.figure(figsize=(6,3), dpi=100)\n",
    "plt.plot(pb_nums)\n",
    "#plt.plot(np.ones(len(pb_nums))*20)\n",
    "plt.ylabel(f\"num of probes\", labelpad=1)\n",
    "plt.xlabel(f\"region id\", labelpad=1)\n",
    "plt.ylim([0, short_probe_num*1.1])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "7dbdbd9e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "- Picked primer: ID: W1A01_primer_0\n",
      "Name: W1A01_primer_0\n",
      "Description: W1A01_primer_0\n",
      "Number of features: 0\n",
      "Seq('CGGCTCGCAGCGTGTAAACG')\n",
      "- Picked primer: ID: W1A02_primer_1\n",
      "Name: W1A02_primer_1\n",
      "Description: W1A02_primer_1\n",
      "Number of features: 0\n",
      "Seq('TAATACGACTCACTATAGGGCATTTCAGGATCACCGGCGG')\n",
      "ID: loc_1:11240000-11260000_gene_4_pb_83_pos_7744_strand_+_readouts_[NDB_752_c,NDB_832_c,NDB_940_c]_primers_[W1A01_primer_0,W1A02_primer_1]_library_1000-mouse-genome-short_250_2\n",
      "Name: loc_1:11240000-11260000_gene_4_pb_83_pos_7744_strand_+_readouts_[NDB_752_c,NDB_832_c,NDB_940_c]_primers_[W1A01_primer_0,W1A02_primer_1]_library_1000-mouse-genome-short_250_2\n",
      "Number of features: 0\n",
      "Seq('CGGCTCGCAGCGTGTAAACGGATACACCCTGGACGCAAATTAAAATTAGCTGCC...ATG')\n",
      "14339\n"
     ]
    },
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAtYAAAFrCAYAAAD1rM7wAAAAAXNSR0IArs4c6QAAIABJREFUeF7t3Ql4VdW5//GXJMikDCJgrlCZZRAQisEiYtWiQi0gKiC3ilgrQrlwyxAFBwYHQJ6L0JIIYtFbECxIEME6USfQpsA/UFAuyGCQCxSoCFRGGf7Pu/7m/EmI5IV19tknJ9/9PD4t5s3aZ3/WG/yxWXvtUqdOnTolHAgggAACCCCAAAIIIOAlUIpg7eXHNyOAAAIIIIAAAggg4AQI1jQCAggggAACCCCAAAJRECBYRwGRIRBAAAEEEEAAAQQQIFjTAwgggAACCCCAAAIIREGAYB0FRIZAAAEEEEAAAQQQQIBgTQ8ggAACCCCAAAIIIBAFAYJ1FBAZAgEEEEAAAQQQQAABgjU9gAACCCCAAAIIIIBAFAQI1lFAZAgEEEAAAQQQQAABBAjW9AACCCCAAAIIIIAAAlEQIFhHAZEhEEAAAQQQQAABBBAgWNMDCCCAAAIIIIAAAghEQYBgHQVEhkAAAQQQQAABBBBAgGBNDyCAAAIIIIAAAgggEAUBgnUUEBkCAQQQQAABBBBAAAGCNT2AAAIIIIAAAggggEAUBAjWUUBkCAQQQAABBBBAAAEECNb0AAIIIIAAAggggAACURAgWEcBkSEQQAABBBBAAAEEECBY0wMIIIAAAggggAACCERBgGAdBUSGQAABBBBAAAEEEECAYE0PIIAAAggggAACCCAQBQGCdRQQGQIBBBBAAAEEEEAAAYI1PYAAAggggAACCCCAQBQECNZRQGQIBBBAAAEEEEAAAQQI1vQAAggggAACCCCAAAJRECBYRwGRIRBAAAEEEEAAAQQQIFjTAwgggAACCCCAAAIIREGAYB0FRIZAAAEEEEAAAQQQQIBgTQ8ggAACCCCAAAIIIBAFAYJ1FBAZAgEEEEAAAQQQQAABgjU9gAACCCCAAAIIIIBAFAQI1lFAZAgEEEAAAQQQQAABBAjW9AACCCCAAAIIIIAAAlEQIFhHAZEhEEAAAQQQQAABBBAgWNMDCCCAAAIIIIAAAghEQYBgHQVEhkAAAQQQQAABBBBAgGBNDyCAAAIIIIAAAgggEAUBgnUUEBkCAQQQQAABBBBAAAGCNT2AAAIIIIAAAggggEAUBAjWUUBkCAQQQAABBBBAAAEECNb0AAIIIIAAAggggAACURAgWEcB0TLEkSNHZO3atVKtWjVJSUmxfAs1CCCAAAIIIIBAXAgcP35c9uzZI82aNZOyZcvGxWeKxw9BsI7RrKxYsULS0tJidDZOgwACCCCAAAIIRF9g+fLlcvXVV0d/4AQZkWAdo4nMzc2VOnXqiDZkampqjM7KaRBAAAEEEEAAAX+BnTt3uhuEX375pdSuXdt/wAQdgWAdo4n93//9X6lVq5Zs27ZNatasGaOzchoEEEAAAQQQQMBfgBxjMyRY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2eIK1zcm7iob0JmQABBBAAAEEEAhJgBxjgydY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2eIK1zcm7iob0JmQABBBAAAEEEAhJgBxjgydY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1h/7/THP/5Rxo0bJ1u2bJF69erJqFGj5K677nJfXb16tQwaNEhWrlwpVatWlYEDB8rQoUNtwt9X0ZDnxEUxAggggAACCMSRADnGNhkEaxGZNWuW3HffffJf//Vf8vOf/1zmzJnjgvWyZcukYcOG0qhRI+natasMHjxYsrOzpX///pKZmSl9+vSxKYsIDWmmohABBBBAAAEE4kyAHGObkBIfrE+dOiV169aVO++8UyZMmBBRu+WWW+SnP/2p+3VGRobk5uZKSkqK+/WIESMkKytL1q9fb1MmWJudKEQAAQQQQACB+BMgWNvmpMQHaw3HjRs3lpycHGnZsuUZap06dZLKlSvL7NmzI19bsmSJdOjQQXbt2iXVq1c3SdOQJiaKEEAAAQQQQCAOBcgxtkkp8cH6jTfekC5dusjSpUvlySeflFWrVkmdOnXksccek1/84hfSvHlz6dixo4wfPz4ium7dOmnatKmsWLFCWrdubZKORUN++c+DcvT4CdPnoQgBBETKpiTL5VXLS6lSpRzH7n8dkb0Hj0GDAAIIFDuBUlJKrrj0osA+dyxyTGAfPoYDl/hgreur77nnHhemR44cKS1atJD58+fL008/Le+995707dtXevXqJWPGjIlMS94DjhrG27VrV+h0HThwQPSfvGPnzp2SlpYm27Ztk5o1awYyxbdO+ljW/+NfgYzNoAgkqsCgmxrIbzs0lJW5e6XHC9ly4uSpRL1UrgsBBBJY4ILkJPni6Y6BXSHB2kZb4oP13LlzpUePHu5hxH79+kXU9CFGPb766ivR5SCF3bH+oeUj+n368OPo0aPPmAWCta0xqUIgVgJlSyfJ0vQbpf8r/0dW5H4Tq9NyHgQQQCCqAgTrqHKe92AlPlh/+umncu2117qlIKfffU5PT5fFixdL7dq13RZ7M2fOjCBb1liHccd694Ej8h132877h4FvLFkCh48dl64Zn8q3R49Lqx9Vlpyv9jmA53q0kLQ6VUsWBleLAAIJIXBZ5XKBXQd3rG20JT5YHz58WGrUqCFPPPFEvr2pdV116dKlpW3btjJ16lTZvHmzJCcnO9Xhw4fLggUL2BXE1mNUIRC3AhPf3SC/e39T5PM1r1lJFv7m2sia67j94HwwBBBAIMYCBGsbeIkP1sr01FNPybPPPivTpk1z66BfffVVF7T/8pe/SJMmTdw+1p07dxa9i718+XK3ZETDdu/evW3KbLdndqIQgVgK7D/0nbR79n3515Hj7rQv9blabrjCttNPLD8n50IAAQTCFiBY22aAYP2908SJE2XKlCmyfft2t/2ero/W3UL00N0/9G2LumNIamqqDBkyRAYMGGAT/r6KhjwnLooRiJlAxgebZMI7GySt9sXyp77XcLc6ZvKcCAEEipMAOcY2WwRrm5N3FQ3pTcgACAQioC+Jyt6yV5peVlEqli0dyDkYFAEEECjuAuQY2wwSrG1O3lU0pDchAyCAAAIIIIBASALkGBs8wdrm5F1FQ3oTMgACCCCAAAIIhCRAjrHBE6xtTt5VNKQ3IQMggAACCCCAQEgC5BgbPMHa5uRdRUN6EzIAAggggAACCIQkQI6xwROsbU7eVTSkNyEDIIAAAggggEBIAuQYGzzB2ubkXUVDehMyAAIIIIAAAgiEJECOscETrG1O3lU0pDchAyCAAAIIIIBASALkGBs8wdrm5F1FQ3oTMgACCCCAAAIIhCRAjrHBE6xtTt5VNKQ3IQMggAACCCCAQEgC5BgbPMHa5uRdRUN6EzIAAggggAACCIQkQI6xwROsbU7eVTSkNyEDIIAAAggggEBIAuQYGzzB2ubkXUVDehMyAAIIIIAAAgiEJECOscETrG1O3lU0pDchAyCAAAIIIIBASALkGBs8wdrm5F1FQ3oTMgACCCCAAAIIhCRAjrHBE6xtTt5VNKQ3IQMggAACCCCAQEgC5BgbPMHa5uRdRUN6EzIAAggggAACCIQkQI6xwROsbU7eVTSkNyEDIIAAAggggEBIAuQYGzzB2ubkXUVDehMyAAIIIIAAAgiEJECOscETrG1O3lU0pDchAyCAAAIIIIBASALkGBs8wdrm5F1FQ3oTMgACCCCAAAIIhCRAjrHBE6xtTt5VNKQ3IQMggAACCCCAQEgC5BgbPMHa5uRdRUN6EzIAAggggAACCIQkQI6xwROsbU7eVTSkNyEDIIAAAggggEBIAuQYGzzB2ubkXUVDehMyAAIIIIAAAgiEJECOscETrG1O3lU0pDchAyCAAAIIIIBASALkGBs8wdrm5F1FQ3oTMgACCCCAAAIIhCRAjrHBE6xtTt5VNKQ3IQMggAACCCCAQEgC5BgbPMHa5uRdRUN6EzIAAggggAACCIQkQI6xwROsbU7eVTSkNyEDIIAAAggggEBIAuQYGzzB2ubkXUVDehMyAAIIIIAAAgiEJECOscETrG1O3lU0pDchAyCAAAIIIIBASALkGBs8wVpEtm7dKrVr1z5DbPr06fLAAw/I6tWrZdCgQbJy5UqpWrWqDBw4UIYOHWoT/r6KhjwnLooRQAABBBBAII4EyDG2ySBYi8iiRYuke/fusmXLFilVqlRErlKlSnLo0CFp1KiRdO3aVQYPHizZ2dnSv39/yczMlD59+tiURYSGNFNRiAACCCCAAAJxJkCOsU0IwVpEnnnmGZk7d667M13wGDt2rGRkZEhubq6kpKS4L48YMUKysrJk/fr1NmWCtdmJQgQQQAABBBCIPwGCtW1OCNYi0rNnT0lKSpLZs2efodapUyepXLlyvq8tWbJEOnToILt27ZLq1aubpGlIExNFCCCAAAIIIBCHAuQY26QQrEWkadOmkpqaKkePHpUvvvhCGjRoII8//rjccsst0rx5c+nYsaOMHz8+Irpu3Tr3PStWrJDWrVsXKn3gwAHRf/KOnTt3Slpammzbtk1q1qxpmx2qEEAAAQQQQACBOBAgWNsmocQH62PHjkmFChWkZcuWMmHCBLnwwgtl1qxZMnnyZHnvvfekb9++0qtXLxkzZkxEVNdi16tXT5YuXSrt2rUrVHrUqFEyevToM75GsLY1JlUIIIAAAgggED8CBGvbXJT4YK1MBw8edOuny5QpE1G79dZb3YOM2ki6HKSwO9Y5OTkukBd2cMfa1oBUIYAAAggggED8CxCsbXNEsP4Bp2HDhsnbb78ttWrVclvszZw5M1LJGmtbc1GFAAIIIIAAAokhQLC2zWOJD9Zr1qyRtm3buhB9+rKOG264QWrUqCFXXXWVTJ06VTZv3izJyclOdfjw4bJgwQJ2BbH1GFUIIIAAAgggUMwFCNa2CSzxwfrkyZMuWOtykOeff14uueQSeeGFF9w+1fpwooZr3ce6c+fOkp6eLsuXL5d+/fq5sN27d2+bMtvtmZ0oRAABBBBAAIH4EyBY2+akxAdrZdqzZ4888sgj8uc//1n27dsnrVq1knHjxsl1113nFDVg69sWV61a5XYPGTJkiAwYMMAm/H0VDXlOXBQjgAACCCCAQBwJkGNsk0Gwtjl5V9GQ3oQMgAACCCCAAAIhCZBjbPAEa5uTdxUN6U3IAAgggAACCCAQkgA5xgZPsLY5eVfRkN6EDIAAAggggAACIQmQY2zwBGubk3cVDelNyAAIIIAAAgggEJIAOcYGT7C2OXlX0ZDehAyAAAIIIIAAAiEJkGNs8ARrm5N3FQ3pTcgACCCAAAIIIBCSADnGBk+wtjl5V9GQ3oQMgAACCCCAAAIhCZBjbPAEa5uTdxUN6U3IAAgggAACCCAQkgA5xgZPsLY5eVfRkN6EDIAAAggggAACIQmQY2zwBGubk3cVDelNyAAIIIAAAgggEJIAOcYGT7C2OXlX0ZDehAyAAAIIIIAAAiEJkGNs8ARrm5N3FQ3pTcgACCCAAAIIIBCSADnGBk+wtjl5V9GQ3oQMgAACCCCAAAIhCZBjbPAEa5uTdxUN6U3IAAgggAACCCAQkgA5xgZPsLY5eVfRkN6EDIAAAggggAACIQmQY2zwBGubk3cVDelNyAAIIIAAAgggEJIAOcYGT7C2OXlX0ZDehAyAAAIIIIAAAiEJkGNs8ARrm5N3FQ3pTcgACCCAAAIIIBCSADnGBp9Qwfrw4cOSnJwsF1xwgfzP//yPvPnmm9K2bVv3T9gHDRn2DHB+BBBAAAEEEDhfAXKMTS5hgvXHH38sXbt2lddee02aNGkiDRs2lKSkJPn2229lzpw5ctddd9lEAqqiIQOCZVgEEEAAAQQQCFyAHGMjTphg3a5dOxemJ0+eLDNmzJAJEybIxo0b3f9/8cUXZdWqVTaRgKpoyIBgGRYBBBBAAAEEAhcgx9iIEyZYly9fXj777DOpW7eudOrUSWrVqiXTpk2TrVu3SqNGjUSXiYR50JBh6nNuBBBAAAEEEPARIMfY9BImWFerVk2WLl0qderUkapVq8pLL73kln/87W9/ky5dusg//vEPm0hAVTRkQLAMiwACCCCAAAKBC5BjbMQJE6y7d+/u7kpffPHFMn/+fNm5c6ds3rxZ+vbt6+5i6zrrMA8aMkx9zo0AAggggAACPgLkGJtewgTrPXv2yEMPPSRbtmyRkSNHugcZhwwZ4u5Y6wONl156qU0koCoaMiBYhkUAAQQQQACBwAXIMTbihAnWhV3usWPH3NZ78XDQkPEwC3wGBBBAAAEEEDgfAXKMTS2hgrUu/5g+fbqsX79eJk2aJB999JE0a9bMPbwY9kFDhj0DnB8BBBBAAAEEzleAHGOTS5hgvWnTJmnTpo1UqlRJdPI1XD/88MPy1ltvybvvvhv6S2JoSFtDUoUAAggggAAC8SdAjrHNScIEa935Q3cG0TvWFStWlL///e9y+eWXy3333SdfffWVu3sd5kFDhqnPuRFAAAEEEEDAR4AcY9NLmGCtoVrDs7518aKLLnLBWncD0TvXeid7//79NpGAqmjIgGAZFgEEEEAAAQQCFyDH2IgTJlhXqVLF7WN95ZVX5gvWuitIx44dZe/evTaRgKpoyIBgGRYBBBBAAAEEAhcgx9iIEyZY9+zZ013xK6+8IpUrV5Y1a9a4JSHdunWT6tWry7x584oU+eKLL6RVq1YyZcoUt4REj9WrV8ugQYNk5cqV7sUzAwcOlKFDhxY5VsECGvKcyfgGBBBAAAEEEIgTAXKMbSISJljv2LFDfvrTn8rXX3/tln3oTiD6OnN9YczHH3/s1luf7fjuu+/cA44aoPWtjRqsdSwdR/fEHjx4sGRnZ0v//v0lMzNT+vTpYxP+voqGPCcuihFAAAEEEEAgjgTIMbbJSJhgrZd76NAh94bFVatWycmTJ92ykF/+8pfuznVRx4gRI1xw/uCDDyLBeuzYsZKRkSG5ubmSkpLihtC6rKwst3b7XA4a8ly0qEUAAQQQQACBeBIgx9hmI6GCtV7ykSNHRJd0lClTxj28WLp06SIl9I727bff7pZ9/OhHP4oE606dOrllJbNnz46MsWTJEunQoYPs2rXLLTGxHjSkVYo6BBBAAAEEEIg3AXKMbUYSJlifOHHCrX2eOnWq6BsX9ShXrpzby/qxxx6TUqVKFSqyb98+adGihejd6V69erm6vKUgzZs3dw8+jh8/PvK969atk6ZNm8qKFSukdevWNmURt7d2rVq1ZNu2bVKzZk3z91GIAAIIIIAAAgiELUCOsc1AwgTrRx99VGbMmCFPPvmk215Pg7buEjJq1CgZMGCAjB49ulCRu+++W06dOiWvvvqq+/rpwbp+/foubI8ZMybyvVu2bJF69eq5sdu1a/eDygcOHBD9J+/Qt0KmpaURrG19SRUCCCCAAAIIxJEAwdo2GQkTrGvUqOHuNOvyjdOP+fPnu2CtwbbgMXPmTBk+fLisXbtWdLu+gsFaX4eu4xV2xzonJ0datmz5g8oa6AsL89yxtjUmVQgggAACCCAQPwIEa9tcJEyw1rXQf/3rX6Vx48b5rlyXbugd7H/9619niNxwww2ybNkytx477zh48KD7td6V1p1EdIs9DeB5h3WNNXesbQ1IFQIIIIAAAgjEvwDB2jZHCROsda/p3bt3y3//93/LBRdcELn6+++/3+3o8cILL5whsn37djl8+HC+f9+gQQN3h7pHjx5uhxFds71582ZJTk52dXqHe8GCBewKYusvqhBAAAEEEEAgAQQI1rZJLNbB+sYbb4xc5fHjx93d59TUVLn66qtdENZt93TpRZcuXeS1114ziZy+xlqDuu5j3blzZ0lPT5fly5dLv379XNju3bu3aby8IhrynLgoRgABBBBAAIE4EiDH2CajWAfrc3lJi66/thynB2ut190/9G2LGtI1tA8ZMsSt2T7Xg4Y8VzHqEUAAAQQQQCBeBMgxtpko1sHadonxUUVDxsc88CkQQAABBBBA4NwFyDE2s4QK1jrp+qbENWvWuBfD6H7Tffv2dS99CfugIcOeAc6PAAIIIIAAAucrQI6xySVMsP7ss8+kffv2Ur58ebdftO5jvXLlStFdPj755BMXssM8aMgw9Tk3AggggAACCPgIkGNsegkTrPUNiRUqVJBXXnklsn2evt78l7/8pXvN+eLFi20iAVXRkAHBMiwCCCCAAAIIBC5AjrERJ0ywvvDCCyU7O1uuvPLKfFeuy0L0Tra+ujzMg4YMU59zI4AAAggggICPADnGppcwwVp37NC70j/+8Y/zXbnu6tGhQweCta0fqEIAAQQQQAABBM4QIFjbmiJhgrUu+dixY4fbr/riiy92V//Pf/5TunfvLvpWxqysLJtIQFU0ZECwDIsAAggggAACgQuQY2zECROsdcLbtm0re/fulYYNG4ruR71hwwapUqWKfPTRR1K3bl2bSEBVNGRAsAyLAAIIIIAAAoELkGNsxAkTrE+dOuV2AJk5c6boDiH662bNmkmvXr2kUqVKNo0Aq2jIAHEZGgEEEEAAAQQCFSDH2HgTJli3bt1aZsyYIc2bN7ddeYyraMgYg3M6BBBAAAEEEIiaADnGRpkwwfqSSy6Rv/3tb1KvXj3blce4ioaMMTinQwABBBBAAIGoCZBjbJQJE6wnTJggb775pgwbNkzq168v5cqVyycQ9tsXaUhbQ1KFAAIIIIAAAvEnQI6xzUnCBOukpKTIFeuDi3mHrrXWX+ubGMM8aMgw9Tk3AggggAACCPgIkGNsegkTrHXnj7Md119/vU0koCoaMiBYhkUAAQQQQACBwAXIMTbihAnWp1+ubrmXnJwcF7uB5H0uGtLWkFQhgAACCCCAQPwJkGNsc5JQwVrXWU+ePFl27tzprr5OnTry8MMPy69//WubRoBVNGSAuAyNAAIIIIAAAoEKkGNsvAkTrMePHy9jxoyRgQMHyk9+8hM5efKkfPLJJ5KZmSnPPfecPPjggzaRgKpoyIBgGRYBBBBAAAEEAhcgx9iIEyZY16pVS5555hm555578l257m09duxY2bhxo00koCoaMiBYhkUAAQQQQACBwAXIMTbihAnWF154oeTk5LjXmZ9+aKBu0aKFHDp0yCYSUBUNGRAswyKAAAIIIIBA4ALkGBtxwgTr3r17y0UXXSRTpkzJd+WjR492rzifN2+eTSSgKhoyIFiGRQABBBBAAIHABcgxNuKECdaDBw9266kbN24surVe6dKlZcWKFbJ06VLp0qWLVK5cOSKiy0NifdCQsRbnfAgggAACCCAQLQFyjE0yYYL1DTfcYLtiEfnggw/MtdEqpCGjJck4CCCAAAIIIBBrAXKMTTxhgrXtcsOroiHDs+fMCCCAAAIIIOAnQI6x+RGsbU7eVTSkNyEDIIAAAggggEBIAuQYGzzB2ubkXUVDehMyAAIIIIAAAgiEJECOscETrG1O3lU0pDchAyCAAAIIIIBASALkGBt8sQ7Wb7zxhtx8881StmxZ29WGWEVDhojPqRFAAAEEEEDAS4AcY+Mr1sFaXwqzfv16qVmzptStW9dtr1e1alXblce4ioaMMTinQwABBBBAAIGoCZBjbJTFOlhXr15d+vbtKz/72c9Et9tbsGCBVKlSpdArb9++vU0koCoaMiBYhkUAAQQQQACBwAXIMTbiYh2sx48fLyNGjJBTp05JqVKl3P8WdujXTpw4YRMJqIqGDAiWYRFAAAEEEEAgcAFyjI24WAdrvcT9+/fLvn37pE6dOrJ8+XKpVq1aoVd++eWX20QCqqIhA4JlWAQQQAABBBAIXIAcYyMu9sE67zI/+ugjufbaayUlJcV25TGuoiFjDM7pEEAAAQQQQCBqAuQYG2XCBGu93MWLF8vTTz8ta9askdKlS0uTJk1k2LBhcvvtt59VY/fu3TJkyBB5++235fDhw3L99dfLhAkT3PfrsXr1ahk0aJCsXLnSPRw5cOBAGTp0qE34+yoa8py4KEYAAQQQQACBOBIgx9gmI2GC9cKFC6Vbt24uROuDirqm+uOPP5ZFixZJVlaWdO7c+QdFrrnmGklKSpLf/e53UqFCBXn88cfl008/lU2bNrmg3ahRI+natasMHjxYsrOzpX///pKZmSl9+vSxKYsIDWmmohABBBBAAAEE4kyAHGObkIQJ1q1atXKhWkPx6ceYMWPcnWxdf13Y8fXXX8uAAQPksccek6ZNm7qSv//973LVVVe571myZIlkZGRIbm5uZJmJPjCpYV23+rMeNKRVijoEEEAAAQQQiDcBcoxtRhImWJcrV07Wrl0r9evXz3flGzdulBYtWsihQ4dMIrt27XLLPD788EMXnO+66y6pXLmyzJ49O/L9GrY7dOggWqtb/lkOGtKiRA0CCCCAAAIIxKMAOcY2KwkTrOvVqye///3vpVOnTvmu/M0335Rf//rXsmPHjiJFHnzwQZk+fbqUKVNG8t7q2Lx5c+nYsaPo1n55x7p169zdbX0hTevWrQsd98CBA6L/5B07d+6UtLQ02bZtm3uhDQcCCCCAAAIIIFBcBAjWtplKmGCtS0Bmzpwpzz//vNsdRI9ly5a59dBdunSRyZMnFymigVnXVOsYeodE4yXPAAAgAElEQVRav7979+7Sq1cv0SUleceWLVtEg/zSpUulXbt2hY47atQoGT169BlfI1gXOQ0UIIAAAggggECcCRCsbROSMMH6yJEj0rNnT3enWV8Io4e+MEbvYM+dO1fKly9vExGRkydPSrNmzaRNmzburrSOUdgd65ycHGnZsmWh43LH2sxNIQIIIIAAAgjEuQDB2jZBCROs8y5X10XrWmsN1RqOGzdufFYJ3WrvL3/5i7sznZycHKnVtdUajvXf6RZ7ejc872CNta25qEIAAQQQQACBxBAgWNvmMeGCte2y/3+V7lGtd501XN94443uC999953bYk+36KtRo4ZMnTpVNm/eHAnew4cPlwULFrAryLliU48AAggggAACxVKAYG2bthIfrPXOtj6cqNvp6YOLVapUcS+Zeeedd9yLYcqWLRsJ2enp6W4Lvn79+rmw3bt3b5sy+1ibnShEAAEEEEAAgfgTIFjb5qTEB2tl2r9/v+hd6Ndff1327dsn1113nUycODGyr7Wus9a3La5atUpSU1PdWxp17+tzOWjIc9GiFgEEEEAAAQTiSYAcY5sNgrXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2+IQJ1l9++aU8/PDD8tlnn4luvVfw0L2nwzxoyDD1OTcCCCCAAAII+AiQY2x6CROs9UUt+nZD3TZPHzgseIwcOdImElAVDRkQLMMigAACCCCAQOAC5BgbccIE6woVKrg3IbZq1cp25TGuoiFjDM7pEEAAAQQQQCBqAuQYG2XCBGsN1JMmTZL27dvbrjzGVTRkjME5HQIIIIAAAghETYAcY6NMmGCdnZ3t9pfWbfHq1KkjSUlJ+QTCDtw0pK0hqUIAAQQQQACB+BMgx9jmJGGC9R/+8AcXrI8fP37GlZcqVUpOnDhhEwmoioYMCJZhEUAAAQQQQCBwAXKMjThhgnWtWrXktttuc3esy5cvf8bVX3755TaRgKpoyIBgGRYBBBBAAAEEAhcgx9iIEyZY68OLn3/+udSuXdt25TGuoiFjDM7pEEAAAQQQQCBqAuQYG2XCBOs777xTOnXqJPfff7/tymNcRUPGGJzTIYAAAggggEDUBMgxNsqECdZTpkyR4cOHS8eOHeWKK66Q0qVL5xN44oknbCIBVdGQAcEyLAIIIIAAAggELkCOsREnTLDWnUB+6NCHF3nzoq0hqEIAAQQQQAABBAoKEKxtPZEwwdp2ueFV0ZDh2XNmBBBAAAEEEPATIMfY/AjWNifvKhrSm5ABEEAAAQQQQCAkAXKMDT5hgrW+EEaXfPzQwT7WtoagCgEEEEAAAQQQKChAsLb1RMIE65dffjlfsP7uu+9k48aNov9+4sSJ8u///u82kYCqaMiAYBkWAQQQQAABBAIXIMfYiBMmWP/Q5c6ePVtmzpwpb731lk0koCoaMiBYhkUAAQQQQACBwAXIMTbihA/WW7dulSZNmsjBgwdtIgFV0ZABwTIsAggggAACCAQuQI6xESd8sB43bpxkZGTItm3bbCIBVdGQAcEyLAIIIIAAAggELkCOsREnTLDWfaxPf3jx1KlTcuDAAdm3b588/fTT8sgjj9hEAqqiIQOCZVgEEEAAAQQQCFyAHGMjTphgPWrUqDN2Bbngggvk2muvlfbt29s0AqyiIQPEZWgEEEAAAQQQCFSAHGPjTZhgbbvc8KpoyPDsOTMCCCCAAAII+AmQY2x+CRWsP/nkE1m2bJkcO3ZMdCnI6ccTTzxhEwmoioYMCJZhEUAAAQQQQCBwAXKMjThhgrU+pDhixAipUqWKVKxYMd/V69rrLVu22EQCqqIhA4JlWAQQQAABBBAIXIAcYyNOmGBds2ZNuf/++2XMmDG2K49xFQ0ZY3BOhwACCCCAAAJREyDH2CgTJliXL19e1qxZI/Xr17ddeYyraMgYg3M6BBBAAAEEEIiaADnGRpkwwfrmm2+WPn36yN1332278hhX0ZAxBud0CCCAAAIIIBA1AXKMjTJhgvWLL74o6enpLlw3atRIypQpk0/g3nvvtYkEVEVDBgTLsAgggAACCCAQuAA5xkacMME6KSnpB69YH148ceKETSSgKhoyIFiGRQABBBBAAIHABcgxNuKECda2yw2vioYMz54zI4AAAggggICfADnG5lfig/XevXvdNn2LFy92r0Bv3ry56NZ97dq1c4KrV6+WQYMGycqVK6Vq1aoycOBAGTp0qE33tCoa8pzJ+AYEEEAAAQQQiBMBcoxtIkp8sNaHHnfv3i1TpkyRatWqSUZGhuh67ZycHPdrXa/dtWtXGTx4sGRnZ0v//v0lMzPTreU+l4OGPBctahFAAAEEEEAgngTIMbbZKNHBetOmTdKgQQPRNza2bdvWiekbGxs2bOh2FylXrpwL2rm5uZKSkuK+rne3s7KyZP369Tbh76toyHPiohgBBBBAAAEE4kiAHGObjBIdrPft2yeffvqp3HTTTfl2EdG9sG+99Vb3tsbKlSvL7NmzI5pLliyRDh06yK5du6R69eo2ZRGhIc1UFCKAAAIIIIBAnAmQY2wTUqKDdWFE8+bNk+7du8sbb7whjz76qHTs2FHGjx8fKV23bp00bdpUVqxYIa1bt/5BZV2vrf/kHTt37pS0tDTZtm2b6FsiORBAAAEEEEAAgeIiQLC2zRTB+jQnXRKid6r1Dvbrr7/u3uLYq1evfK9J17vY9erVk6VLl0YecCyMetSoUTJ69OgzvkSwtjUmVQgggAACCCAQPwIEa9tcEKy/d1q4cKEL0ddcc43bIUTXVzdr1kw6depU6B1rfbixZcuW3LG29RlVCCCAAAIIIFCMBQjWtskjWIu4HUF0S71u3brJrFmzIuutNVTrFnszZ86MaLLG2tZYVCGAAAIIIIBA4ggQrG1zWeKD9fPPP++20NP9qZ977jk5/Q2Oup/11KlTZfPmzZKcnOxEhw8fLgsWLGBXEFt/UYUAAggggAACCSBAsLZNYokO1l988YVceeWVctttt7m9qU8/dCnI0aNH3T7WnTt3lvT0dFm+fLn069fPhe3evXvbhL+voiHPiYtiBBBAAAEEEIgjAXKMbTJKdLB+5pln3M4fhR0anF9++WW3+4fezV61apWkpqbKkCFDZMCAATbd06poyHMm4xsQQAABBBBAIE4EyDG2iSjRwdpGFJ0qGjI6joyCAAIIIIAAArEXIMfYzAnWNifvKhrSm5ABEEAAAQQQQCAkAXKMDZ5gbXPyrqIhvQkZAAEEEEAAAQRCEiDH2OAJ1jYn7yoa0puQARBAAAEEEEAgJAFyjA2eYG1z8q6iIb0JGQABBBBAAAEEQhIgx9jgCdY2J+8qGtKbkAEQQAABBBBAICQBcowNnmBtc/KuoiG9CRkAAQQQQAABBEISIMfY4AnWNifvKhrSm5ABEEAAAQQQQCAkAXKMDZ5gbXPyrqIhvQkZAAEEEEAAAQRCEiDH2OAJ1jYn7yoa0puQARBAAAEEEEAgJAFyjA2eYG1z8q6iIb0JGQABBBBAAAEEQhIgx9jgCdY2J+8qGtKbkAEQQAABBBBAICQBcowNnmBtc/KuoiG9CRkAAQQQQAABBEISIMfY4AnWNifvKhrSm5ABEEAAAQQQQCAkAXKMDZ5gbXPyrqIhvQkZAAEEEEAAAQRCEiDH2OAJ1jYn7yoa0puQARBAAAEEEEAgJAFyjA2eYG1z8q6iIb0JGQABBBBAAAEEQhIgx9jgCdY2J+8qGtKbkAEQQAABBBBAICQBcowNnmBtc/KuoiG9CRkAAQQQQAABBEISIMfY4AnWNifvKhrSm5ABEEAAAQQQQCAkAXKMDZ5gbXPyrqIhvQkZAAEEEEAAAQRCEiDH2OAJ1jYn7yoa0puQARBAAAEEEEAgJAFyjA2eYG1z8q6iIb0JGQABBBBAAAEEQhIgx9jgCdY2J+8qGtKbkAEQQAABBBBAICQBcowNnmBtc/KuoiG9CRkAAQQQQAABBEISIMfY4AnWNifvKhrSm5ABEEAAAQQQQCAkAXKMDZ5gbXPyrqIhvQkZAAEEEEAAAQRCEiDH2OAJ1jYn7yoa0puQARBAAAEEEEAgJAFyjA2eYG1z8q6iIb0JGQABBBBAAAEEQhIgx9jgCdY2J+8qGtKbkAEQQAABBBBAICQBcowNnmBtc/KuoiG9CRkAAQQQQAABBEISIMfY4AnWBZyeeuopWbJkiXz44YeRr6xevVoGDRokK1eulKpVq8rAgQNl6NChNuHvq2jIc+KiGAEEEEAAAQTiSIAcY5sMgvVpTpMmTZLBgwdL+/btI8H666+/lkaNGknXrl3d17Kzs6V///6SmZkpffr0sSmLCA1ppqIQAQQQQAABBOJMgBxjmxCCtYhs375dHnjgAVm6dKnUqlVLatSoEQnWY8eOlYyMDMnNzZWUlBSnOmLECMnKypL169fblAnWZicKEUAAAQQQQCD+BAjWtjkhWIvIokWLZM6cOaLLQMaMGeNCdN5SkE6dOknlypVl9uzZEVFdKtKhQwfZtWuXVK9e3SRNQ5qYKEIAAQQQQACBOBQgx9gmhWBdwOm+++7LF6ybN28uHTt2lPHjx0cq161bJ02bNpUVK1ZI69atTdI0pImJIgQQQAABBBCIQwFyjG1SCNZFBOv69etLr1693J3svGPLli1Sr149t3SkXbt2hUofOHBA9J+8Y+fOnZKWlibbtm2TmjVr2maHKgQQQAABBBBAIA4ECNa2SSBYFxGsmzVrJrocpLA71jk5OdKyZctCpUeNGiWjR48+42sEa1tjUoUAAggggAAC8SNAsLbNBcG6iGCtoVq32Js5c2ak0rLGmjvWtgakCgEEEEAAAQTiX4BgbZsjgnURwXrcuHEydepU2bx5syQnJ7vq4cOHy4IFC9gVxNZjVCGAAAIIIIBAMRcgWNsmkGBdRLDevXu328e6c+fOkp6eLsuXL5d+/fq5sN27d2+bMtvtmZ0oRAABBBBAAIH4EyBY2+aEYF1EsNYv6+4f+rbFVatWSWpqqgwZMkQGDBhgE/6+ioY8Jy6KEUAAAQQQQCCOBMgxtskgWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2eIK1zcm7iob0JmQABBBAAAEEEAhJgBxjgydY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2eIK1zcm7iob0JmQABBBAAAEEEAhJgBxjgydY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2eIK1zcm7iob0JmQABBBAAAEEEAhJgBxjgydY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2eIK1zcm7iob0JmQABBBAAAEEEAhJgBxjgydY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1jbnLyraEhvQgZAAAEEEEAAgZAEyDE2eIK1zcm7iob0JmQABBBAAAEEEAhJgBxjgydY25y8q2hIb0IGQAABBBBAAIGQBMgxNniCtc3Ju4qG9CZkAAQQQAABBBAISYAcY4MnWNucvKtoSG9CBkAAAQQQQACBkATIMTZ4grXNybuKhvQmZAAEEEAAAQQQCEmAHGODJ1gbnE6ePCmjR4+WF198Ub755htp166dZGZmSv369Q3f/f9KaEgzFYUIIIAAAgggEGcC5BjbhBCsDU4aqjVIv/TSS3LZZZdJenq6bNmyRT7//HO54IILDCMQrE1IFCGAAAIIIIBAXAoQrG3TQrAuwunYsWNyySWXyLPPPisPPfSQq963b5/827/9m8yYMUN69uxpkqYhTUwUIYAAAggggEAcCpBjbJNCsC7Cafny5dKmTRvZsGGDNGzYMFKty0GaN2/u7mRbDhrSokQNAggggAACCMSjADnGNisE6yKcsrKy5I477pBDhw5JuXLlItXdu3d3/27x4sWFjnDgwAHRf/KObdu2Sdu2bUWDempqqm12qEIAAQQQQAABBOJAYOfOnZKWliZffvml1K5dOw4+UXx+BIJ1EfMya9Ysueeee+TEiROSlJQUqb733ntlx44dsmTJkkJHGDVqlHvgkQMBBBBAAAEEEEgUAb1BePXVVyfK5UT9OgjWRZDOnz9f7rzzzkLvWB89elQWLlxoumN95MgR0bvWderUkZSUlKhPpA6Y96dJ7ooHwhv1QZmvqJMGNiBzFRhtIAMzX4GwBjIocxUIayCDHj9+XPbs2SPNmjWTsmXLBnKORBiUYF3ELOatsd60aZPUq1cvUq1rrFu0aCEZGRlx0wesf4qbqTB9EObLxBQXRcxVXEyD+UMwX2aq0AuZq9CngA8QZQGCdRGgele6evXqMnHiRPnVr37lqs9nV5Aoz1uhw/EbVCyUo3cO5it6lkGPxFwFLRzd8Zmv6HoGORpzFaQuY4chQLA2qD/66KMybdo0t72eLtgfNmyY5Obmytq1a837WBtO413Cb1DehDEdgPmKKbfXyZgrL76YfzPzFXPy8z4hc3XedHxjnAoQrA0Tow8ujhgxwr0g5vDhw9K+fXu3BCTenorVXUj0zvrgwYOlYsWKhiujJEwB5itM/XM7N3N1bl5hVzNfYc+A/fzMld2KyuIhQLAuHvPEp0QAAQQQQAABBBCIcwGCdZxPEB8PAQQQQAABBBBAoHgIEKyLxzzxKRFAAAEEEEAAAQTiXIBgHecTxMdDAAEEEEAAAQQQKB4CBOviMU98SgQQQAABBBBAAIE4FyBYx/kEWT/eyZMn3SvUX3zxRfnmm29EX2CTmZkp9evXtw5BXUACW7duLXQHmenTp8sDDzwgq1evlkGDBsnKlSulatWqMnDgQBk6dGhAn4Zhf0jgqaeekiVLlsiHH34YKSlqbvi5C6+fCpuvPn36yMsvv5zvQ1122WWiW7rpwXzFbr727t3rdtNavHix6M4fzZs3l3Hjxrn/NunBz1bs5oIzxVaAYB1b78DOpqFag7RuCaj/IUlPT5ctW7bI559/Hld7bQcGEMcDL1q0SLp37+7mo1SpUpFPWqlSJTl06JA0atRIunbt6rZJzM7Olv79+7u51JDAERuBSZMmOX/dSjMvWH/99ddFzg0/d7GZn4JnKWy+tObHP/6x3HrrrfIf//EfkW9JTk6WatWquV8zX7Gbr5tvvll2794tU6ZMcf66Ra3e+MnJyXG/Lur3PeYqdnPFmaIrQLCOrmcoox07dkwuueQSefbZZ+Whhx5ynyFe3w4ZClDIJ33mmWdk7ty57g5NwWPs2LHuPzj6wqGUlBT3Zb3Lk5WVJevXrw/5kyf+6bdv3+7+1mDp0qVSq1YtqVGjRiRYFzU3/NzFvj/ONl/6voELL7xQ5syZ4/6gWvBgvmI3X5s2bZIGDRrIJ598Im3btnUnPnXqlDRs2FDuvvtuKVeu3Fl/32OuYjdXnCn6AgTr6JvGfMTly5dLmzZtZMOGDe43rrxD/8pN//pN735yhCfQs2dPSUpKktmzZ5/xITp16iSVK1fO9zVdjtChQwfZtWuXVK9ePbwPXgLOrH+boEFMlxWMGTPG/QEn7451UXOjtfzcxbZJzjZf+gfRxo0bn/H7YN4n5PfJ2M2V3tj59NNP5aabbpIyZcpETqxLE/VvFPRv7872+x4/W7GbK84UfQGCdfRNYz6i3t2844473LICvROQd+jyA/13usaNIzyBpk2bSmpqqhw9elS++OILdyfn8ccfl1tuucX9wadjx44yfvz4yAdct26d6PesWLFCWrduHd4HL2Fnvu+++/IF66Lm5quvvuLnLsQeKThf8+bNkx49eri/tXvrrbfcH2b1D0f6hyZddsXvkyFOlojo/Oh/k9544w159NFHz/r7Hj9b4c4VZ/cTIFj7+cXFd8+aNUvuuece0b8K1f+Y5B333nuv7Nixwz2QxRGOgP6VZoUKFaRly5YyYcIE91fVOl+TJ0+W9957T/r27Su9evVyd0vzDr2bU69ePbc8Ie9Bn3A+fck6a8GgpnfXzjY3eleNn7vweqTgfI0cOVKefvpp0aVXGqg3btzoHgLWJT7vv/+++1sh5iuc+dIlIXqnWu9gv/766+6hen62wpkLzhq8AME6eOPAzzB//ny58847C71jrXdJFy5cGPhn4AQ/LHDw4EG3fvr0vxLV/8jog4y6W4GGgMLuWOtDPhrIOWIjUDCoNWvW7Kxzo38A4ucuNnNT2FkKzpeu4d2/f79bYpB36MPAP/nJT9xDwfqzxnzFfr70vz8aoq+55hr3t6f6t6r8bMV+Hjhj7AQI1rGzDuxMeWsH9YERvdOZd+jdzhYtWriHRDjiS2DYsGHy9ttvu7tpusXezJkzIx+QNdbhzFXBoKZ/4Dnb3OStA+XnLj7mq7BPoX+o1b8leu2119zPmq6JZ75iN1+6I4huJdqtWzf3N3V5Nxf42YrdHHCm2AsQrGNvHvUz6l1pfcht4sSJ8qtf/cqNz64gUWc+rwHXrFnjnorXEH36so4bbrjB7UBx1VVXydSpU2Xz5s2i24LpMXz4cFmwYAG7gpyX+Pl/U8FgrXvunm1u+Lk7f+tofGfB+dK7orpF4jvvvBMZXh9E1Z81fW6hbt26/D4ZDXjjGM8//7zbOlT35X/uuefyLVPkZ8uISFmxFCBYF8tpO/ND68Mg06ZNkxkzZriXkegdUb2jtnbtWvaxDnGO9YUUGqz1zpn+h0a3RXzhhRfcTi36cKKGa93PtXPnzm7vcf3bh379+rlA17t37xA/eck7dcGgpnvwFjU3/NyF1ycF50uXGejP0ZNPPim6E48+KPyb3/zGLQV55ZVX3AdlvmIzX2p/5ZVXym233XbGrlS6FET/UMrPVmzmgrPEXoBgHXvzQM6oDy7q/sf6gpjDhw+7F13oEhAN2RzhCuzZs0ceeeQR+fOf/+z+JqFVq1buDWTXXXed+2AasPWuzqpVq9zuIUOGDJEBAwaE+6FL4NkLBjXL3PBzF16jFDZf+ryJPsCoW+/pWmu9i627gpQtW9Z9UOYrNvOlD5DqH2IKO/SGgb4ds6jf95ir2MwVZ4m+AME6+qaMiAACCCCAAAIIIFACBQjWJXDSuWQEEEAAAQQQQACB6AsQrKNvyogIIIAAAggggAACJVCAYF0CJ51LRgABBBBAAAEEEIi+AME6+qaMiAACCAQqoA8l68N7o0aNCuQ8edvUffnllz/4AHTQnyGQC2NQBBBAIGABgnXAwAyPAAIIRFtAd5rRbcv05SdBHMeOHZO9e/dKtWrVIvurFzwPwToIecZEAIHiLkCwLu4zyOdHAAEEQhAgWIeAzikRQCDuBQjWcT9FfEAEECiuAqVKlXL7+eor6/WlGB999JHUqVNHHn/8cfeK5/3797sXaYwZM0ZuvvnmyGW+++677g2ceW8M1L3N9a2qeUszCobaN998070Y5bPPPpOKFSvK3Xff7fZzztu/WT+Hvpho7ty5smzZMrn44ovdXul6jsKOgktB9HPqXusLFy50L5zSPfMnTZoU6HKU4jrnfG4EECjZAgTrkj3/XD0CCAQooIFW37b51ltvyfHjx+Waa65xLy35/PPPXTCtWbOmLFq0yL1ASF9j//Of/1xWr14tV199tfz2t791YVp/rW/j/OabbwoN1q+//rrccccdbr119+7dZePGja4+LS1N9IUpeujnqFSpkkyZMsX9ew36+uKUjz/+OPKiotMZCgbrW265Rb766isXzi+66CL3EqP3339fRo4cGdg67wCnhaERQACBwAQI1oHRMjACCJR0AQ20GpAnTpzoKDZt2iQNGjRwb51r3bp1hEffRrd161bRQKv/X18J/de//jXy9d///vfujnFhd6zbtGkjtWrVktdeey1Sr2FdX++td7wbN27sgvWgQYNcmM879K71sGHDCr1rfXqwznv99JIlS+Smm25y375r1y65/PLL3R8IgnqAsqT3DtePAALFU4BgXTznjU+NAALFQEADbWZmpruDrMe8efPcXeUKFSrk+/TfffedewW3BtZmzZpJhw4dImFcC/WudcuWLQsN1uXLlxd9hfR//ud/RsbUpRs63quvvio9evRwwXrq1KnSt2/fSI0uJ+nTp4+761zwOD1Yr1y5Uu666y63bEWXmeQdTZo0cddCsC4GjchHRACBmAkQrGNGzYkQQKCkCWigfemll9xaZD3+9Kc/Sc+ePSUnJ8ctqTj9SE5OduuvNUBff/31+e4uny1Y6+4g48aNc3ek8w7d0aNq1apuKUi3bt1csD79c2jd2R4+LCxY61IUDet5R4sWLeT2228nWJe0puZ6EUDgrAIEaxoEAQQQCEigYKDVtdX6sOLixYvdeuq8Qx9wTEpKcg8gaghfv369ZGdnR74+bdo0eeihhwq9Y61rpn/0ox/lWwqiDxl27do131KQ8w3WJ06ckPr16+f7zPv27ZPLLrvMLePgP8EAAAGXSURBVCXhjnVAzcOwCCBQLAUI1sVy2vjQCCBQHAQKu1P8i1/8wi3tyMjIcCE7KytL0tPT5Q9/+INbmqE7e1x11VUydOhQuf/++104fvDBB0X3rs7NzXVrm0+/26zLS3S5x+jRo93/6vpsXXqiY+haaz187ljruXTJh64Lnz59ulx66aVuXbY+kPnYY48RrItDI/IZEUAgZgIE65hRcyIEEChpAoUF2kOHDrkt+HRZiC7ZqFu3rgwePFgeeOCBCI8GYg2vusPHFVdcEdn1Y8eOHZKamnrGMo45c+bI2LFjZcOGDe6lLrrziAZtXSYSjWB9+PBhd3daz3Py5Em3VluXi9x6660E65LW1FwvAgicVYBgTYMggAACcSSgd4ZTUlLcWuu8Y/bs2e7u9bfffuu+xoEAAgggEJ8CBOv4nBc+FQIIlFABXW6hd4f/+Mc/uuUcukWf3iG+9tpr5eWXXy6hKlw2AgggUDwECNbFY574lAggUEIETp065R5i1BC9fft2qV69unuT4ulLO0oIBZeJAAIIFDuB/wtxig3NpM9ybgAAAABJRU5ErkJggg==\" width=\"659.9999856948856\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "from ImageAnalysis3 import library_tools\n",
    "primer_folder = r'\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\Primers'\n",
    "primers_250_2_short = library_tools.assemble.load_primers([0,1], _primer_folder=primer_folder)\n",
    "\n",
    "sub_library_250_2_short_records = []\n",
    "for _reg, _pbs in sub_library_250_2.items():\n",
    "\n",
    "    if len(_pbs) >= short_probe_num:\n",
    "        _sel_num = short_probe_num\n",
    "    else:\n",
    "        _sel_num = len(_pbs)\n",
    "        \n",
    "    sub_library_250_2_short_records.extend(_pbs[:_sel_num])\n",
    "    \n",
    "# replace primers\n",
    "for _irecord, _record in enumerate(sub_library_250_2_short_records):\n",
    "    _seq = primers_250_2_short[0].seq[-primer_len:] \\\n",
    "        + _record.seq[primer_len:-primer_len] \\\n",
    "        + primers_250_2_short[1].seq[-primer_len:].reverse_complement()\n",
    "    _name = _record.name.split('primers_[')[0] + 'primers_['+primers_250_2_short[0].id +','+ primers_250_2_short[1].id+']' \\\n",
    "    + '_library_1000-mouse-genome-short_250_2'\n",
    "    sub_library_250_2_short_records[_irecord] = SeqRecord(_seq, name=_name, id=_name, description='')\n",
    "    \n",
    "# save kept records\n",
    "print(sub_library_250_2_short_records[0])\n",
    "print(len(sub_library_250_2_short_records))\n",
    "with open(os.path.join(library_folder, 'filtered_full_probes_sublibrary_250_2_short.fasta'), 'w') as output_handle:\n",
    "    SeqIO.write(sub_library_250_2_short_records, output_handle, \"fasta\")\n",
    "    \n",
    "pb_num_dict = {_k:len(_v) for _k,_v in library_tools.quality_check.split_probe_by_gene(sub_library_250_2_short_records).items()}\n",
    "\n",
    "pb_nums = np.array(list(pb_num_dict.values()))\n",
    "plt.figure(figsize=(6,3), dpi=100)\n",
    "plt.plot(pb_nums)\n",
    "#plt.plot(np.ones(len(pb_nums))*20)\n",
    "plt.ylabel(f\"num of probes\", labelpad=1)\n",
    "plt.xlabel(f\"region id\", labelpad=1)\n",
    "plt.ylim([0, short_probe_num*1.1])\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "16fc2471",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/javascript": "/* Put everything inside the global mpl namespace */\n/* global mpl */\nwindow.mpl = {};\n\nmpl.get_websocket_type = function () {\n    if (typeof WebSocket !== 'undefined') {\n        return WebSocket;\n    } else if (typeof MozWebSocket !== 'undefined') {\n        return MozWebSocket;\n    } else {\n        alert(\n            'Your browser does not have WebSocket support. ' +\n                'Please try Chrome, Safari or Firefox  6. ' +\n                'Firefox 4 and 5 are also supported but you ' +\n                'have to enable WebSockets in about:config.'\n        );\n    }\n};\n\nmpl.figure = function (figure_id, websocket, ondownload, parent_element) {\n    this.id = figure_id;\n\n    this.ws = websocket;\n\n    this.supports_binary = this.ws.binaryType !== undefined;\n\n    if (!this.supports_binary) {\n        var warnings = document.getElementById('mpl-warnings');\n        if (warnings) {\n            warnings.style.display = 'block';\n            warnings.textContent =\n                'This browser does not support binary websocket messages. ' +\n                'Performance may be slow.';\n        }\n    }\n\n    this.imageObj = new Image();\n\n    this.context = undefined;\n    this.message = undefined;\n    this.canvas = undefined;\n    this.rubberband_canvas = undefined;\n    this.rubberband_context = undefined;\n    this.format_dropdown = undefined;\n\n    this.image_mode = 'full';\n\n    this.root = document.createElement('div');\n    this.root.setAttribute('style', 'display: inline-block');\n    this._root_extra_style(this.root);\n\n    parent_element.appendChild(this.root);\n\n    this._init_header(this);\n    this._init_canvas(this);\n    this._init_toolbar(this);\n\n    var fig = this;\n\n    this.waiting = false;\n\n    this.ws.onopen = function () {\n        fig.send_message('supports_binary', { value: fig.supports_binary });\n        fig.send_message('send_image_mode', {});\n        if (mpl.ratio !== 1) {\n            fig.send_message('set_dpi_ratio', { dpi_ratio: mpl.ratio });\n        }\n        fig.send_message('refresh', {});\n    };\n\n    this.imageObj.onload = function () {\n        if (fig.image_mode === 'full') {\n            // Full images could contain transparency (where diff images\n            // almost always do), so we need to clear the canvas so that\n            // there is no ghosting.\n            fig.context.clearRect(0, 0, fig.canvas.width, fig.canvas.height);\n        }\n        fig.context.drawImage(fig.imageObj, 0, 0);\n    };\n\n    this.imageObj.onunload = function () {\n        fig.ws.close();\n    };\n\n    this.ws.onmessage = this._make_on_message_function(this);\n\n    this.ondownload = ondownload;\n};\n\nmpl.figure.prototype._init_header = function () {\n    var titlebar = document.createElement('div');\n    titlebar.classList =\n        'ui-dialog-titlebar ui-widget-header ui-corner-all ui-helper-clearfix';\n    var titletext = document.createElement('div');\n    titletext.classList = 'ui-dialog-title';\n    titletext.setAttribute(\n        'style',\n        'width: 100%; text-align: center; padding: 3px;'\n    );\n    titlebar.appendChild(titletext);\n    this.root.appendChild(titlebar);\n    this.header = titletext;\n};\n\nmpl.figure.prototype._canvas_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._root_extra_style = function (_canvas_div) {};\n\nmpl.figure.prototype._init_canvas = function () {\n    var fig = this;\n\n    var canvas_div = (this.canvas_div = document.createElement('div'));\n    canvas_div.setAttribute(\n        'style',\n        'border: 1px solid #ddd;' +\n            'box-sizing: content-box;' +\n            'clear: both;' +\n            'min-height: 1px;' +\n            'min-width: 1px;' +\n            'outline: 0;' +\n            'overflow: hidden;' +\n            'position: relative;' +\n            'resize: both;'\n    );\n\n    function on_keyboard_event_closure(name) {\n        return function (event) {\n            return fig.key_event(event, name);\n        };\n    }\n\n    canvas_div.addEventListener(\n        'keydown',\n        on_keyboard_event_closure('key_press')\n    );\n    canvas_div.addEventListener(\n        'keyup',\n        on_keyboard_event_closure('key_release')\n    );\n\n    this._canvas_extra_style(canvas_div);\n    this.root.appendChild(canvas_div);\n\n    var canvas = (this.canvas = document.createElement('canvas'));\n    canvas.classList.add('mpl-canvas');\n    canvas.setAttribute('style', 'box-sizing: content-box;');\n\n    this.context = canvas.getContext('2d');\n\n    var backingStore =\n        this.context.backingStorePixelRatio ||\n        this.context.webkitBackingStorePixelRatio ||\n        this.context.mozBackingStorePixelRatio ||\n        this.context.msBackingStorePixelRatio ||\n        this.context.oBackingStorePixelRatio ||\n        this.context.backingStorePixelRatio ||\n        1;\n\n    mpl.ratio = (window.devicePixelRatio || 1) / backingStore;\n\n    var rubberband_canvas = (this.rubberband_canvas = document.createElement(\n        'canvas'\n    ));\n    rubberband_canvas.setAttribute(\n        'style',\n        'box-sizing: content-box; position: absolute; left: 0; top: 0; z-index: 1;'\n    );\n\n    var resizeObserver = new ResizeObserver(function (entries) {\n        var nentries = entries.length;\n        for (var i = 0; i < nentries; i++) {\n            var entry = entries[i];\n            var width, height;\n            if (entry.contentBoxSize) {\n                if (entry.contentBoxSize instanceof Array) {\n                    // Chrome 84 implements new version of spec.\n                    width = entry.contentBoxSize[0].inlineSize;\n                    height = entry.contentBoxSize[0].blockSize;\n                } else {\n                    // Firefox implements old version of spec.\n                    width = entry.contentBoxSize.inlineSize;\n                    height = entry.contentBoxSize.blockSize;\n                }\n            } else {\n                // Chrome <84 implements even older version of spec.\n                width = entry.contentRect.width;\n                height = entry.contentRect.height;\n            }\n\n            // Keep the size of the canvas and rubber band canvas in sync with\n            // the canvas container.\n            if (entry.devicePixelContentBoxSize) {\n                // Chrome 84 implements new version of spec.\n                canvas.setAttribute(\n                    'width',\n                    entry.devicePixelContentBoxSize[0].inlineSize\n                );\n                canvas.setAttribute(\n                    'height',\n                    entry.devicePixelContentBoxSize[0].blockSize\n                );\n            } else {\n                canvas.setAttribute('width', width * mpl.ratio);\n                canvas.setAttribute('height', height * mpl.ratio);\n            }\n            canvas.setAttribute(\n                'style',\n                'width: ' + width + 'px; height: ' + height + 'px;'\n            );\n\n            rubberband_canvas.setAttribute('width', width);\n            rubberband_canvas.setAttribute('height', height);\n\n            // And update the size in Python. We ignore the initial 0/0 size\n            // that occurs as the element is placed into the DOM, which should\n            // otherwise not happen due to the minimum size styling.\n            if (width != 0 && height != 0) {\n                fig.request_resize(width, height);\n            }\n        }\n    });\n    resizeObserver.observe(canvas_div);\n\n    function on_mouse_event_closure(name) {\n        return function (event) {\n            return fig.mouse_event(event, name);\n        };\n    }\n\n    rubberband_canvas.addEventListener(\n        'mousedown',\n        on_mouse_event_closure('button_press')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseup',\n        on_mouse_event_closure('button_release')\n    );\n    // Throttle sequential mouse events to 1 every 20ms.\n    rubberband_canvas.addEventListener(\n        'mousemove',\n        on_mouse_event_closure('motion_notify')\n    );\n\n    rubberband_canvas.addEventListener(\n        'mouseenter',\n        on_mouse_event_closure('figure_enter')\n    );\n    rubberband_canvas.addEventListener(\n        'mouseleave',\n        on_mouse_event_closure('figure_leave')\n    );\n\n    canvas_div.addEventListener('wheel', function (event) {\n        if (event.deltaY < 0) {\n            event.step = 1;\n        } else {\n            event.step = -1;\n        }\n        on_mouse_event_closure('scroll')(event);\n    });\n\n    canvas_div.appendChild(canvas);\n    canvas_div.appendChild(rubberband_canvas);\n\n    this.rubberband_context = rubberband_canvas.getContext('2d');\n    this.rubberband_context.strokeStyle = '#000000';\n\n    this._resize_canvas = function (width, height, forward) {\n        if (forward) {\n            canvas_div.style.width = width + 'px';\n            canvas_div.style.height = height + 'px';\n        }\n    };\n\n    // Disable right mouse context menu.\n    this.rubberband_canvas.addEventListener('contextmenu', function (_e) {\n        event.preventDefault();\n        return false;\n    });\n\n    function set_focus() {\n        canvas.focus();\n        canvas_div.focus();\n    }\n\n    window.setTimeout(set_focus, 100);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'mpl-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'mpl-button-group';\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'mpl-button-group';\n            continue;\n        }\n\n        var button = (fig.buttons[name] = document.createElement('button'));\n        button.classList = 'mpl-widget';\n        button.setAttribute('role', 'button');\n        button.setAttribute('aria-disabled', 'false');\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n\n        var icon_img = document.createElement('img');\n        icon_img.src = '_images/' + image + '.png';\n        icon_img.srcset = '_images/' + image + '_large.png 2x';\n        icon_img.alt = tooltip;\n        button.appendChild(icon_img);\n\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    var fmt_picker = document.createElement('select');\n    fmt_picker.classList = 'mpl-widget';\n    toolbar.appendChild(fmt_picker);\n    this.format_dropdown = fmt_picker;\n\n    for (var ind in mpl.extensions) {\n        var fmt = mpl.extensions[ind];\n        var option = document.createElement('option');\n        option.selected = fmt === mpl.default_extension;\n        option.innerHTML = fmt;\n        fmt_picker.appendChild(option);\n    }\n\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n};\n\nmpl.figure.prototype.request_resize = function (x_pixels, y_pixels) {\n    // Request matplotlib to resize the figure. Matplotlib will then trigger a resize in the client,\n    // which will in turn request a refresh of the image.\n    this.send_message('resize', { width: x_pixels, height: y_pixels });\n};\n\nmpl.figure.prototype.send_message = function (type, properties) {\n    properties['type'] = type;\n    properties['figure_id'] = this.id;\n    this.ws.send(JSON.stringify(properties));\n};\n\nmpl.figure.prototype.send_draw_message = function () {\n    if (!this.waiting) {\n        this.waiting = true;\n        this.ws.send(JSON.stringify({ type: 'draw', figure_id: this.id }));\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    var format_dropdown = fig.format_dropdown;\n    var format = format_dropdown.options[format_dropdown.selectedIndex].value;\n    fig.ondownload(fig, format);\n};\n\nmpl.figure.prototype.handle_resize = function (fig, msg) {\n    var size = msg['size'];\n    if (size[0] !== fig.canvas.width || size[1] !== fig.canvas.height) {\n        fig._resize_canvas(size[0], size[1], msg['forward']);\n        fig.send_message('refresh', {});\n    }\n};\n\nmpl.figure.prototype.handle_rubberband = function (fig, msg) {\n    var x0 = msg['x0'] / mpl.ratio;\n    var y0 = (fig.canvas.height - msg['y0']) / mpl.ratio;\n    var x1 = msg['x1'] / mpl.ratio;\n    var y1 = (fig.canvas.height - msg['y1']) / mpl.ratio;\n    x0 = Math.floor(x0) + 0.5;\n    y0 = Math.floor(y0) + 0.5;\n    x1 = Math.floor(x1) + 0.5;\n    y1 = Math.floor(y1) + 0.5;\n    var min_x = Math.min(x0, x1);\n    var min_y = Math.min(y0, y1);\n    var width = Math.abs(x1 - x0);\n    var height = Math.abs(y1 - y0);\n\n    fig.rubberband_context.clearRect(\n        0,\n        0,\n        fig.canvas.width / mpl.ratio,\n        fig.canvas.height / mpl.ratio\n    );\n\n    fig.rubberband_context.strokeRect(min_x, min_y, width, height);\n};\n\nmpl.figure.prototype.handle_figure_label = function (fig, msg) {\n    // Updates the figure title.\n    fig.header.textContent = msg['label'];\n};\n\nmpl.figure.prototype.handle_cursor = function (fig, msg) {\n    var cursor = msg['cursor'];\n    switch (cursor) {\n        case 0:\n            cursor = 'pointer';\n            break;\n        case 1:\n            cursor = 'default';\n            break;\n        case 2:\n            cursor = 'crosshair';\n            break;\n        case 3:\n            cursor = 'move';\n            break;\n    }\n    fig.rubberband_canvas.style.cursor = cursor;\n};\n\nmpl.figure.prototype.handle_message = function (fig, msg) {\n    fig.message.textContent = msg['message'];\n};\n\nmpl.figure.prototype.handle_draw = function (fig, _msg) {\n    // Request the server to send over a new figure.\n    fig.send_draw_message();\n};\n\nmpl.figure.prototype.handle_image_mode = function (fig, msg) {\n    fig.image_mode = msg['mode'];\n};\n\nmpl.figure.prototype.handle_history_buttons = function (fig, msg) {\n    for (var key in msg) {\n        if (!(key in fig.buttons)) {\n            continue;\n        }\n        fig.buttons[key].disabled = !msg[key];\n        fig.buttons[key].setAttribute('aria-disabled', !msg[key]);\n    }\n};\n\nmpl.figure.prototype.handle_navigate_mode = function (fig, msg) {\n    if (msg['mode'] === 'PAN') {\n        fig.buttons['Pan'].classList.add('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    } else if (msg['mode'] === 'ZOOM') {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.add('active');\n    } else {\n        fig.buttons['Pan'].classList.remove('active');\n        fig.buttons['Zoom'].classList.remove('active');\n    }\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Called whenever the canvas gets updated.\n    this.send_message('ack', {});\n};\n\n// A function to construct a web socket function for onmessage handling.\n// Called in the figure constructor.\nmpl.figure.prototype._make_on_message_function = function (fig) {\n    return function socket_on_message(evt) {\n        if (evt.data instanceof Blob) {\n            /* FIXME: We get \"Resource interpreted as Image but\n             * transferred with MIME type text/plain:\" errors on\n             * Chrome.  But how to set the MIME type?  It doesn't seem\n             * to be part of the websocket stream */\n            evt.data.type = 'image/png';\n\n            /* Free the memory for the previous frames */\n            if (fig.imageObj.src) {\n                (window.URL || window.webkitURL).revokeObjectURL(\n                    fig.imageObj.src\n                );\n            }\n\n            fig.imageObj.src = (window.URL || window.webkitURL).createObjectURL(\n                evt.data\n            );\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        } else if (\n            typeof evt.data === 'string' &&\n            evt.data.slice(0, 21) === 'data:image/png;base64'\n        ) {\n            fig.imageObj.src = evt.data;\n            fig.updated_canvas_event();\n            fig.waiting = false;\n            return;\n        }\n\n        var msg = JSON.parse(evt.data);\n        var msg_type = msg['type'];\n\n        // Call the  \"handle_{type}\" callback, which takes\n        // the figure and JSON message as its only arguments.\n        try {\n            var callback = fig['handle_' + msg_type];\n        } catch (e) {\n            console.log(\n                \"No handler for the '\" + msg_type + \"' message type: \",\n                msg\n            );\n            return;\n        }\n\n        if (callback) {\n            try {\n                // console.log(\"Handling '\" + msg_type + \"' message: \", msg);\n                callback(fig, msg);\n            } catch (e) {\n                console.log(\n                    \"Exception inside the 'handler_\" + msg_type + \"' callback:\",\n                    e,\n                    e.stack,\n                    msg\n                );\n            }\n        }\n    };\n};\n\n// from http://stackoverflow.com/questions/1114465/getting-mouse-location-in-canvas\nmpl.findpos = function (e) {\n    //this section is from http://www.quirksmode.org/js/events_properties.html\n    var targ;\n    if (!e) {\n        e = window.event;\n    }\n    if (e.target) {\n        targ = e.target;\n    } else if (e.srcElement) {\n        targ = e.srcElement;\n    }\n    if (targ.nodeType === 3) {\n        // defeat Safari bug\n        targ = targ.parentNode;\n    }\n\n    // pageX,Y are the mouse positions relative to the document\n    var boundingRect = targ.getBoundingClientRect();\n    var x = e.pageX - (boundingRect.left + document.body.scrollLeft);\n    var y = e.pageY - (boundingRect.top + document.body.scrollTop);\n\n    return { x: x, y: y };\n};\n\n/*\n * return a copy of an object with only non-object keys\n * we need this to avoid circular references\n * http://stackoverflow.com/a/24161582/3208463\n */\nfunction simpleKeys(original) {\n    return Object.keys(original).reduce(function (obj, key) {\n        if (typeof original[key] !== 'object') {\n            obj[key] = original[key];\n        }\n        return obj;\n    }, {});\n}\n\nmpl.figure.prototype.mouse_event = function (event, name) {\n    var canvas_pos = mpl.findpos(event);\n\n    if (name === 'button_press') {\n        this.canvas.focus();\n        this.canvas_div.focus();\n    }\n\n    var x = canvas_pos.x * mpl.ratio;\n    var y = canvas_pos.y * mpl.ratio;\n\n    this.send_message(name, {\n        x: x,\n        y: y,\n        button: event.button,\n        step: event.step,\n        guiEvent: simpleKeys(event),\n    });\n\n    /* This prevents the web browser from automatically changing to\n     * the text insertion cursor when the button is pressed.  We want\n     * to control all of the cursor setting manually through the\n     * 'cursor' event from matplotlib */\n    event.preventDefault();\n    return false;\n};\n\nmpl.figure.prototype._key_event_extra = function (_event, _name) {\n    // Handle any extra behaviour associated with a key event\n};\n\nmpl.figure.prototype.key_event = function (event, name) {\n    // Prevent repeat events\n    if (name === 'key_press') {\n        if (event.which === this._key) {\n            return;\n        } else {\n            this._key = event.which;\n        }\n    }\n    if (name === 'key_release') {\n        this._key = null;\n    }\n\n    var value = '';\n    if (event.ctrlKey && event.which !== 17) {\n        value += 'ctrl+';\n    }\n    if (event.altKey && event.which !== 18) {\n        value += 'alt+';\n    }\n    if (event.shiftKey && event.which !== 16) {\n        value += 'shift+';\n    }\n\n    value += 'k';\n    value += event.which.toString();\n\n    this._key_event_extra(event, name);\n\n    this.send_message(name, { key: value, guiEvent: simpleKeys(event) });\n    return false;\n};\n\nmpl.figure.prototype.toolbar_button_onclick = function (name) {\n    if (name === 'download') {\n        this.handle_save(this, null);\n    } else {\n        this.send_message('toolbar_button', { name: name });\n    }\n};\n\nmpl.figure.prototype.toolbar_button_onmouseover = function (tooltip) {\n    this.message.textContent = tooltip;\n};\nmpl.toolbar_items = [[\"Home\", \"Reset original view\", \"fa fa-home icon-home\", \"home\"], [\"Back\", \"Back to previous view\", \"fa fa-arrow-left icon-arrow-left\", \"back\"], [\"Forward\", \"Forward to next view\", \"fa fa-arrow-right icon-arrow-right\", \"forward\"], [\"\", \"\", \"\", \"\"], [\"Pan\", \"Left button pans, Right button zooms\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-arrows icon-move\", \"pan\"], [\"Zoom\", \"Zoom to rectangle\\nx/y fixes axis, CTRL fixes aspect\", \"fa fa-square-o icon-check-empty\", \"zoom\"], [\"\", \"\", \"\", \"\"], [\"Download\", \"Download plot\", \"fa fa-floppy-o icon-save\", \"download\"]];\n\nmpl.extensions = [\"eps\", \"jpeg\", \"pdf\", \"png\", \"ps\", \"raw\", \"svg\", \"tif\"];\n\nmpl.default_extension = \"png\";/* global mpl */\n\nvar comm_websocket_adapter = function (comm) {\n    // Create a \"websocket\"-like object which calls the given IPython comm\n    // object with the appropriate methods. Currently this is a non binary\n    // socket, so there is still some room for performance tuning.\n    var ws = {};\n\n    ws.close = function () {\n        comm.close();\n    };\n    ws.send = function (m) {\n        //console.log('sending', m);\n        comm.send(m);\n    };\n    // Register the callback with on_msg.\n    comm.on_msg(function (msg) {\n        //console.log('receiving', msg['content']['data'], msg);\n        // Pass the mpl event to the overridden (by mpl) onmessage function.\n        ws.onmessage(msg['content']['data']);\n    });\n    return ws;\n};\n\nmpl.mpl_figure_comm = function (comm, msg) {\n    // This is the function which gets called when the mpl process\n    // starts-up an IPython Comm through the \"matplotlib\" channel.\n\n    var id = msg.content.data.id;\n    // Get hold of the div created by the display call when the Comm\n    // socket was opened in Python.\n    var element = document.getElementById(id);\n    var ws_proxy = comm_websocket_adapter(comm);\n\n    function ondownload(figure, _format) {\n        window.open(figure.canvas.toDataURL());\n    }\n\n    var fig = new mpl.figure(id, ws_proxy, ondownload, element);\n\n    // Call onopen now - mpl needs it, as it is assuming we've passed it a real\n    // web socket which is closed, not our websocket->open comm proxy.\n    ws_proxy.onopen();\n\n    fig.parent_element = element;\n    fig.cell_info = mpl.find_output_cell(\"<div id='\" + id + \"'></div>\");\n    if (!fig.cell_info) {\n        console.error('Failed to find cell for figure', id, fig);\n        return;\n    }\n};\n\nmpl.figure.prototype.handle_close = function (fig, msg) {\n    var width = fig.canvas.width / mpl.ratio;\n    fig.root.removeEventListener('remove', this._remove_fig_handler);\n\n    // Update the output cell to use the data from the current canvas.\n    fig.push_to_output();\n    var dataURL = fig.canvas.toDataURL();\n    // Re-enable the keyboard manager in IPython - without this line, in FF,\n    // the notebook keyboard shortcuts fail.\n    IPython.keyboard_manager.enable();\n    fig.parent_element.innerHTML =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n    fig.close_ws(fig, msg);\n};\n\nmpl.figure.prototype.close_ws = function (fig, msg) {\n    fig.send_message('closing', msg);\n    // fig.ws.close()\n};\n\nmpl.figure.prototype.push_to_output = function (_remove_interactive) {\n    // Turn the data on the canvas into data in the output cell.\n    var width = this.canvas.width / mpl.ratio;\n    var dataURL = this.canvas.toDataURL();\n    this.cell_info[1]['text/html'] =\n        '<img src=\"' + dataURL + '\" width=\"' + width + '\">';\n};\n\nmpl.figure.prototype.updated_canvas_event = function () {\n    // Tell IPython that the notebook contents must change.\n    IPython.notebook.set_dirty(true);\n    this.send_message('ack', {});\n    var fig = this;\n    // Wait a second, then push the new image to the DOM so\n    // that it is saved nicely (might be nice to debounce this).\n    setTimeout(function () {\n        fig.push_to_output();\n    }, 1000);\n};\n\nmpl.figure.prototype._init_toolbar = function () {\n    var fig = this;\n\n    var toolbar = document.createElement('div');\n    toolbar.classList = 'btn-toolbar';\n    this.root.appendChild(toolbar);\n\n    function on_click_closure(name) {\n        return function (_event) {\n            return fig.toolbar_button_onclick(name);\n        };\n    }\n\n    function on_mouseover_closure(tooltip) {\n        return function (event) {\n            if (!event.currentTarget.disabled) {\n                return fig.toolbar_button_onmouseover(tooltip);\n            }\n        };\n    }\n\n    fig.buttons = {};\n    var buttonGroup = document.createElement('div');\n    buttonGroup.classList = 'btn-group';\n    var button;\n    for (var toolbar_ind in mpl.toolbar_items) {\n        var name = mpl.toolbar_items[toolbar_ind][0];\n        var tooltip = mpl.toolbar_items[toolbar_ind][1];\n        var image = mpl.toolbar_items[toolbar_ind][2];\n        var method_name = mpl.toolbar_items[toolbar_ind][3];\n\n        if (!name) {\n            /* Instead of a spacer, we start a new button group. */\n            if (buttonGroup.hasChildNodes()) {\n                toolbar.appendChild(buttonGroup);\n            }\n            buttonGroup = document.createElement('div');\n            buttonGroup.classList = 'btn-group';\n            continue;\n        }\n\n        button = fig.buttons[name] = document.createElement('button');\n        button.classList = 'btn btn-default';\n        button.href = '#';\n        button.title = name;\n        button.innerHTML = '<i class=\"fa ' + image + ' fa-lg\"></i>';\n        button.addEventListener('click', on_click_closure(method_name));\n        button.addEventListener('mouseover', on_mouseover_closure(tooltip));\n        buttonGroup.appendChild(button);\n    }\n\n    if (buttonGroup.hasChildNodes()) {\n        toolbar.appendChild(buttonGroup);\n    }\n\n    // Add the status bar.\n    var status_bar = document.createElement('span');\n    status_bar.classList = 'mpl-message pull-right';\n    toolbar.appendChild(status_bar);\n    this.message = status_bar;\n\n    // Add the close button to the window.\n    var buttongrp = document.createElement('div');\n    buttongrp.classList = 'btn-group inline pull-right';\n    button = document.createElement('button');\n    button.classList = 'btn btn-mini btn-primary';\n    button.href = '#';\n    button.title = 'Stop Interaction';\n    button.innerHTML = '<i class=\"fa fa-power-off icon-remove icon-large\"></i>';\n    button.addEventListener('click', function (_evt) {\n        fig.handle_close(fig, {});\n    });\n    button.addEventListener(\n        'mouseover',\n        on_mouseover_closure('Stop Interaction')\n    );\n    buttongrp.appendChild(button);\n    var titlebar = this.root.querySelector('.ui-dialog-titlebar');\n    titlebar.insertBefore(buttongrp, titlebar.firstChild);\n};\n\nmpl.figure.prototype._remove_fig_handler = function () {\n    this.close_ws(this, {});\n};\n\nmpl.figure.prototype._root_extra_style = function (el) {\n    el.style.boxSizing = 'content-box'; // override notebook setting of border-box.\n    el.addEventListener('remove', this._remove_fig_handler);\n};\n\nmpl.figure.prototype._canvas_extra_style = function (el) {\n    // this is important to make the div 'focusable\n    el.setAttribute('tabindex', 0);\n    // reach out to IPython and tell the keyboard manager to turn it's self\n    // off when our div gets focus\n\n    // location in version 3\n    if (IPython.notebook.keyboard_manager) {\n        IPython.notebook.keyboard_manager.register_events(el);\n    } else {\n        // location in version 2\n        IPython.keyboard_manager.register_events(el);\n    }\n};\n\nmpl.figure.prototype._key_event_extra = function (event, _name) {\n    var manager = IPython.notebook.keyboard_manager;\n    if (!manager) {\n        manager = IPython.keyboard_manager;\n    }\n\n    // Check for shift+enter\n    if (event.shiftKey && event.which === 13) {\n        this.canvas_div.blur();\n        // select the cell after this one\n        var index = IPython.notebook.find_cell_index(this.cell_info[0]);\n        IPython.notebook.select(index + 1);\n    }\n};\n\nmpl.figure.prototype.handle_save = function (fig, _msg) {\n    fig.ondownload(fig, null);\n};\n\nmpl.find_output_cell = function (html_output) {\n    // Return the cell and output element which can be found *uniquely* in the notebook.\n    // Note - this is a bit hacky, but it is done because the \"notebook_saving.Notebook\"\n    // IPython event is triggered only after the cells have been serialised, which for\n    // our purposes (turning an active figure into a static one), is too late.\n    var cells = IPython.notebook.get_cells();\n    var ncells = cells.length;\n    for (var i = 0; i < ncells; i++) {\n        var cell = cells[i];\n        if (cell.cell_type === 'code') {\n            for (var j = 0; j < cell.output_area.outputs.length; j++) {\n                var data = cell.output_area.outputs[j];\n                if (data.data) {\n                    // IPython >= 3 moved mimebundle to data attribute of output\n                    data = data.data;\n                }\n                if (data['text/html'] === html_output) {\n                    return [cell, data, j];\n                }\n            }\n        }\n    }\n};\n\n// Register the function which deals with the matplotlib target/channel.\n// The kernel may be null if the page has been refreshed.\nif (IPython.notebook.kernel !== null) {\n    IPython.notebook.kernel.comm_manager.register_target(\n        'matplotlib',\n        mpl.mpl_figure_comm\n    );\n}\n",
      "text/plain": [
       "<IPython.core.display.Javascript object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<img src=\"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAtYAAAFrCAYAAAD1rM7wAAAAAXNSR0IArs4c6QAAIABJREFUeF7t3Qm8TtX+x/GfeZ4yJWSeZ5lChIwVUiQ3Sdclrj83U0g5RMhNFKJcdSPchGRIOZW5EzpkynjMhAhlnv6vtY7zeM5xOOvYz977efb57Ner173Hs5619nqv9fA9+1l77WQ3bty4IRwIIIAAAggggAACCCBgSSAZwdqSH29GAAEEEEAAAQQQQEALEKyZCAgggAACCCCAAAIIBECAYB0ARKpAAAEEEEAAAQQQQIBgzRxAAAEEEEAAAQQQQCAAAgTrACBSBQIIIIAAAggggAACBGvmAAIIIIAAAggggAACARAgWAcAkSoQQAABBBBAAAEEECBYMwcQQAABBBBAAAEEEAiAAME6AIhUgQACCCCAAAIIIIAAwZo5gAACCCCAAAIIIIBAAAQI1gFApAoEEEAAAQQQQAABBAjWzAEEEEAAAQQQQAABBAIgQLAOACJVIIAAAggggAACCCBAsGYOIIAAAggggAACCCAQAAGCdQAQqQIBBBBAAAEEEEAAAYI1cwABBBBAAAEEEEAAgQAIEKwDgEgVCCCAAAIIIIAAAggQrJkDCCCAAAIIIIAAAggEQIBgHQBEqkAAAQQQQAABBBBAgGDNHEAAAQQQQAABBBBAIAACBOsAIFIFAggggAACCCCAAAIEa+YAAggggAACCCCAAAIBECBYBwCRKhBAAAEEEEAAAQQQIFgzBxBAAAEEEEAAAQQQCIAAwToAiFSBAAIIIIAAAggggADBmjmAAAIIIIAAAggggEAABAjWAUCkCgQQQAABBBBAAAEECNbMAQQQQAABBBBAAAEEAiBAsA4AIlUggAACCCCAAAIIIECwZg4ggAACCCCAAAIIIBAAAYJ1ABCpAgEEEEAAAQQQQAABgjVzAAEEEEAAAQQQQACBAAgQrAOASBUIIIAAAggggAACCBCsmQMIIIAAAggggAACCARAgGAdAESqQAABBBBAAAEEEECAYM0cQAABBBBAAAEEEEAgAAIE6wAgUgUCCCCAAAIIIIAAAgRr5gACCCCAAAIIIIAAAgEQIFgHAJEqEEAAAQQQQAABBBAgWDMHEEAAAQQQQAABBBAIgADBOgCIVIEAAggggAACCCCAAMGaOYAAAggggAACCCCAQAAECNYBQDSp4uLFi7J582bJmTOnpEyZ0uQtlEEAAQQQQAABBIJC4OrVq3LixAkpV66cpE2bNijOKRhPgmDt0KisW7dOqlWr5lBrNIMAAggggAACCAReYO3atVK1atXAV+yRGgnWDg3kvn37pFChQqImZJ48eRxqlWYQQAABBBBAAAHrAkePHtUXCPfu3SsFCxa0XqFHayBYOzSwhw4dkvz588vBgwclX758DrVKMwgggAACCCCAgHUBcoyZIcHazMlyKSakZUIqQAABBBBAAAGXBMgxZvAEazMny6WYkJYJqQABBBBAAAEEXBIgx5jBE6zNnCyXYkJaJqQCBBBAAAEEEHBJgBxjBk+wNnOyXIoJaZmQChBAAAEEEEDAJQFyjBk8wdrMyXIpJqRlQipAAAEEEEAAAZcEyDFm8ARrMyfLpZiQlgmpAAEEEEAAAQRcEiDHmMETrM2cLJdiQlompAIEEEAAAQQQcEmAHGMGT7A2c7JciglpmZAKEEAAAQQQQMAlAXKMGTzB2szJcikmpGVCKkAAAQQQQAABlwTIMWbwBGszJ8ulmJCWCakAAQQQQAABBFwSIMeYwROszZwsl2JCWiakAgQQQAABBBBwSYAcYwZPsDZzslyKCWmZkAoQQAABBBBAwCUBcowZPMHazMlyKSakZUIqQAABBBBAAAGXBMgxZvAEazMny6WYkJYJqQABBBBAAAEEXBIgx5jBE6zNnCyXYkJaJqQCBBBAAAEEEHBJgBxjBk+wNnOyXIoJaZmQChBAAAEEEEDAJQFyjBk8wfqm06effiojR46UqKgoKVKkiISFhUnr1q31qxs3bpSePXvK+vXrJXv27NKjRw/p06ePmfDNUkzIRHFRGAEEEEAAAQSCSIAcYzYYBGsRmT59urz44ovyzjvvyOOPPy4zZ87UwXrVqlVSvHhxKVmypLRs2VJ69eolERER0q1bN5k4caJ07NjRTFlEmJDGVBREAAEEEEAAgSATIMeYDUiSD9Y3btyQwoULyzPPPCOjR4/2qTVu3FgeffRR/fOECRNk3759kjJlSv3zwIEDZe7cubJ9+3YzZYK1sRMFEUAAAQQQQCD4BAjWZmOS5IO1CselSpWSyMhIqVSp0m1qzZo1k6xZs8qMGTN8r4WHh0vDhg3l2LFjkitXLiNpJqQRE4UQQAABBBBAIAgFyDFmg5Lkg/VXX30lLVq0kJUrV8qbb74pGzZskEKFCsmgQYPkySeflPLly0vTpk1l1KhRPtFt27ZJmTJlZN26dVKlShUjaScm5N7fz8mlq9eMzodCCCAQmgKFc2SUy9euy6E/zjvagZTJk4lqO3nyZHLq3GU5/udFR9u3o7G8WdNJ2lQpZM+Jv+yonjpdFMiVKa1kTZdKj+21GzdcOZOY+RV14pzckOhzyJkxjWRLnzrWeamfc2VKI/tOnr/t3/B0qVJIgewZ5OiZC3LmwpW79iOZJJMS92eyra9O5BjbTt7BipN8sFbrq9u3b6/D9ODBg6VChQoyZ84cGT58uCxdulS6dOki7dq1k6FDh/qGJeYGRxXGa9euHe9wnT17VtR/McfRo0elWrVqcvDgQcmXL58tQ9xk7ArZ/tufttRNpQggEBwC+bKlk7MXrsjZi1cdP6EWFR+Qro8WkSffXyVXrrkTVgLZ6UxpUkqW9Knk0B8XAlktdQWBQKoUyaRg9gyy67h7vzRlTJNS7suQWg6cuvVLsPoFtUD29LLnxDmfUrJkIqXuzyzbjt7KDP6EJe/PZPRve+oUyWXn8Ka26ROszWiTfLD+/PPP5dlnn9U3I3bt2tWnpm5iVMeBAwdELQeJ74r1nZaPqPepmx+HDBly2ygQrM0mJqUQQCD4BFRI6N+kpPSbsyn4To4zQiCJCxCsg2MCJPlgvWbNGqlVq5ZeCuJ/9blfv36ycOFCKViwoN5ib9q0ab4RM1lj7cYV6+NnL8qV66F/FSk4PhqcBQLBJbDxwGn554zIWCe15F+PSKa0qWw/0fX7TknPWRt1O4MeLyXDFv0qOTKmlvnd4//GzvYTstjAtWs3pM7oH2LVMun5ylIuX1aLNfP2YBB4euIa+e3sraVKA5uVlMfLP+DYqV25el0e/feyWO2Na1tR3vl2Z6yr14OfLC1zIw/L5sNnfGWfq5Zfutcvpn8e//1umbn2gO+1mkWyy+jWFe7aD7X8xK6DK9Zmskk+WF+4cEFy584tb7zxRqy9qdW66lSpUknNmjVl0qRJsmfPHkmRIoVWHTBggMybN49dQczmGKUQQCAAAircPjPpx1g1bRnSWNTXzXYfap1qg3eW62Y6PFxA/vvjflH/gK/uX9/upm2rv8jAxXLN70LEwv+rLWXzZrGtPSp2TqD+v5dJ1O+3llqoUNuiYl7HTuD69RtSeODiWO3N6lxDhi3aJlsO31ru8WH7h2TKyr2ydt8pX9kudQvLgKal9M/vfLtD3v9+t++1ZuXul4l/e8ixfsRtiGBtRp/kg7ViGjZsmLz99tsyefJkvQ561qxZOmh/9913Urp0ab2PdfPmzUVdxV67dq1eMqLCdocOHcyU2W7P2ImCCCAQv0DkgT+k1cQ1sV78dWgTSZc6+hd+O4+LV65JydeX6CbqFs8py3ee0OtEl/etZ2ezttZd5o0lcu7yrZu9w3vVkaK57Lvxy9bOUHksgbj3G016/iFpUvZ+R5UKDVgk/vdMzun6sLy9ZIf8tPdWiJ7Rqbq89/0uiYi69WfdHi0i/ZqU1Oc64YfdMvqbHb7zblUpr4x5tqKj/fBvjGBtRk+wvuk0ZswYGT9+vBw+fFhvv6fWR6vdQtShdv9QT1tUO4bkyZNHevfuLd27dzcTvlmKCZkoLgojgEAcgV8OnpYWE1bH+tMdw5pImpT2B2vVaJVhS+X3vy5Lwezp9e4FhXNmkO97R+/1H4rHQ28ulZPnLvtOfWW/epL/vvSh2BXOOY6A+pyoz0vM8XHHqlKvhNnWuIHCLP7a13r3npjjy3/Wkve+2yXfbz8e689Gf7NdVu8+6fuz/6tfVHo3KqF/nrIySi+7ijmeq/agjGhVLlCnmOh6yDFmZARrMyfLpZiQlgmpAIEkLbDl8Bl54v1VsQx2D28qKVMkd8SlxfhV8suhW2tBS+TOJN+8UseRtu1opNbI7+Xw6Vu7gax77THJmSmNHU1Rp8MCbSb9GGt5hboyXLNoDkfPovQbS+S83zciaqnR5BVRsuCXI77zUN+SDFmwTVbu+t33Zz0bFJNXGhbXP0+P2C+Dvtzie61jrYIy+MkyjvbDvzFyjBk9wdrMyXIpJqRlQipAIEkL/Hr0rDQdtzKWwd4RzSSZ2qvLgaPbZz/L4s2/+Voq80BmWdTjEQdatqeJ+u8sE7W/cMyxOayRIzeC2tMbavUXaP+fn2KFVbUM46EC9zmKVD7sm1hbYn7zrzry8eq9MmvdQd95rOlfXwbO2yzLdpzw/VmvhsWlR4Pomxe/+PmQ9Jn9i+81tdXlqzeXiTjamZuNkWPM1AnWZk6WSzEhLRNSAQJJWmDnsT+l0bsrfAbJk4lEjYjeFtSJY/iibfLRyr2+pirkzyrz/1nLiaZtaaPZuJWx9g3eNbyppHLo6r8tHaJSn0Cn/66T8F9vLblw48bUuEuNvutdVz6LOCBTV9/6DP0yuJH0+t9G+c5veUjfxiXkn/WK6r4s3HREus/Y4OvXK48Vl56PRYduNw5yjJk6wdrMyXIpJqRlQipAIEkL7D7+lzw2JnpnDnWoB2DsGt7MMZOpq/bK0IXbfO09VCCbzOla07H2A91Qq4mrJfJA9DrcFMmTyZ63nLMMdF+oL7bAPz+LlEWbj/r+cOkrdaRYbmdvTK02PFyO/3nJdw7L+z4qc34+JO/57fKhlnJ1/SxSlm475iunrkirK9PqCN92TDp9ut732oCmJaVL3ejX3DjIMWbqBGszJ8ulmJCWCakAgSQtsO/3c7H2xk2bKrlsf9O+p6zFxZ7x0wH9tXXMUb3QffK/Lg+H7Ji0+yhC1uyJvmksfeoUsm1ok5DtCyceW+CV/22UeRsOxwq16rHgTh5x1/CrrSnV+uqRX2/Xp5EmZXLZMaypvDztZ1my9dYSK7Xnduc60eF51a7f5fn//OQ77bAnS8uLtQo52Y1YbZFjzOgJ1mZOlksxIS0TUgECSVrg4Knz8sjbtx5qkiF1CtnqYBj8fP1B6ffFrScu1i6aQ6Z3qh6yY/LSJ+t8OzSoJ0pGvt4wZPvCiccW6D9nU6y1zBEDGsj9WdI6ylR39A+y/+StR5mvHdhAvtl2TF6/eTNizJyLe3VdPYCp0yOF9bnG3bt+ZKty0rbag472w78xcowZPcHazMlyKSakZUIqQCBJC6gdLNRVsJgjU9qUsjmssWMmX244LP/6X/TTF9XxaImc8knHao61H+iGuk7/Wb7eEn2l8IEsaWXNgAaBboL6XBIYPH+LfohRzLHh9YaSLUNqR8+mwTvLZI/fzbHqF7dlO45Lr8+jb0bMf186WdmvvvzfzA2xdgpRT2PsePOq9OZDZ+TJ8bd2Ahr7bEVpWcm5B93EBSPHmE0hgrWZk+VSTEjLhFSAQJIW+O3MRakx4jufQdb0qWTjG40cM1m06WisR6o/Viq3TOlQxbH2A92Q/3KBwjkyyPd9QndP7kDbhHp9by3+VT5cEeXrxrahjSV9avufUOrv1vjdFbLj2J++P9oU1kjW7D4pL0//Wf9ZzHaV/5q1Qb7ceGsLvqEtysgLDxfUZXYd+1Ma+t2wPOn5ytKkbB7XhoccY0ZPsDZzslyKCWmZkAoQSNICJ/68JFWHh/sMcmRMLesHObd84Zutv0mXadGhQB1Nytwvk9q793hlq5NhwNxNMnNt9NZnpfJklq97hu7WgVYtvPb+f3+zQ8b/cOtR4E7u9x5j+fh7K2XrkVuPL1fhPnL/ad+a6bJ5M8vC/3tEen2+UeZG3loPPqxlWXm+RgFdTdzlX2486MZ/bpBjzD4pBGszJ8ulmJCWCakAgSQtcOrcZan85lKfQa5MaWTta485ZvLD9uPS8ZN1vvaeKJ9Hxrer7Fj7gW4o7Kut8smafbraSg9mlXndQnfrwEDbhHp973+3S95ZulN3w60dX+I+/XHnsKay5cgZaTVxjT6vmGDdd/YvMvvnQz5y9WRF9YRFdRz/86JUG37rW6qZ/6ghDxfJ7trwkGPM6AnWZk6WSzEhLRNSAQJJWuDM+StSYei3PoM8WdLKjw6uC467Q8FTlfLKu89WDNkxUbszTFq+R5//w4Wzy8zONUK2L5x4bIEPV+yRtxZH777h1o4vz3ywRtbv/8N3YuphTruO/+Xbiz4mWMe90fLtp8tLm6r59fvOXrwi5cNufebndasplR7M5tpwk2PM6AnWZk6WSzEhLRNSAQJJWuDPi1eknN8/snmzphO1hZdTR0TUSWn7YYSvudYP5ZPRrSs41XzA2xkbvlPGhu/S9Yb6jZgBxwnxCv+7Zp8M/mqr7kW29Klkg4P3IsTQPTv5R/lp7yn9Y8xV80N/nJfao6J39okJ1moLS7WVZczx79YV5JmH8ukfL1+9LsUHfe17bXGPR6T0A5ldGx1yjBk9wdrMyXIpJqRlQipAIEkLXLh8TUq9scRnUCB7elnet55jJj/v/0Oe/iD6a2x1PFctv4xoVd6x9gPdkLpaHbOncKivFw+0TajXN3PtARkwN3rP9dyZ08hPA51bMhVj9/yUn2TV7t/1j6lTJhe1FOSPc5el0s3lXDHBWm2/Ny3i1g4m7z5bQZ6qFB2sb9y4IYUGLPYNx/e960rhnBldGx5yjBk9wdrMyXIpJqRlQipAIEkLXLp6TUoMuhWsnd7JYtOh09J8/GrfGLSvUUDebFk2ZMfkk9V7JWxB9JMkW1Z8QMa2rRSyfeHEYwvMjTzk29buwfvSy4p+zv0CGnMmL368VpbtOKF/jNlz3v8KdEywjrs14Li2FaVFxVtb6hXsv8jXuTX968sDWdO5NtzkGDN6grWZk+VSTEjLhFSAQJIWuHrtuhR97dbXwkVzZZTwXnUdM/n16FlpOm6lr72OtQrK4CfLONZ+oBuatfaA9L95VbNt1fwy8unQvfoeaJtQr89/a8hiuTLKUgc/JzF2nf67XsJ/jX5UeZZ0qeSXwdFbY8YE5ZhgPWTBVvl4dfRNtOp4/7lK8mSFB3w/+wdrtRe2erCMWwc5xkyeYG3mZLkUE9IyIRUgkKQF4n4tHLMPrlMou4//JY+NWe5rrnOdwjKwWSmnmg94O/4PvHmxZkEJax66vyQEHCfEKwzfdkw6fbpe96LMA5llUQ/nt1L0f1S5/9aYMUG5XN4ssuD/asuwhdtkyqq9PvGJf6sszcrd2qvaP1i7sR+3/1Qgx5h9MAjWZk6WSzEhLRNSAQJJXqDwgEVy/UY0Q+k8mWWxg3sv7z95TuqOXuYbg26PFpF+TUqG7Jh8vfmodP0sUp9/l7qFZUDT0P0lIWQHwaYTX7HzhLwwda2uPSbA2tTUHavtPiNSFm46ql+/P3NaiRgY/WTPRu8ul53H/pKPX6wq9UrmkhGLf5XJfg+zmfT8Q9Kk7P2+ev2DddRbzSR58mROd8XXHjnGjJ5gbeZkuRQT0jIhFSCQ5AWKvbZYrlyLTtZOB4Yjpy9ITb9HqveoX1R6NSoRsmPivy/3/9UvKr1DuC8hOwg2nfhPUSfl2Zs72Dj9OYnpkv+TPf138Dl36aocPn1BiufOpIuOWrJdPlgWve2jOj56oYo0LJ073mC9b+TjNomZVUuOMXMiWJs5WS7FhLRMSAUIJHmBEoO+lktXr2uHCvmzyvx/OvdQk7hPfuzVsLj0aFAsZMdkzZ7fpd1HP+nzD/VfEkJ2EGw68Y0HT0vLCdE32roVrP0f/FIwe3pZdocdfOI+JXLqi1WkfkmCtU1Tw5FqCdaOMIsQrB2CphkEPCxQ+o0lcv7yNd3Dyg9mlbkOPi3w9PnLUnHorSc/9mtSQro9WjRktSMP/OF7Ch7BOmSHMd4T97/Rtny+LPJV99qOd1Bt96e2/VPH3W40HrN0p7z3XfR+6ur4pGNVebRELt/P/ktBuGLt+DDeU4ME63tiS/ybCNaJN+MdCCAQW6Bc2Dfy58Wr+g+rFbxPPn/5YceI1FfYZQZ/42vvtWal5B91CjvWfqAb2nbkrDR7L3qXE4J1oHXdrW/Pib+kwTvRN9q6dcXaf3/qkvdnkiX/qhMvyrjwXfJuePTj19Ux7e/V5JFiOQnW7k4hS60TrC3xmb+ZYG1uRUkEEIhfoOLQb+X0+Sv6xRqF75NZnZ0L1nGfAjf4ydLSsVahkB2qqBN/Sf2b4YtgHbLDGO+J+z/h0K1g7b+N3t3OYfz3u+Tf394K1jM6VZeaRXMQrEN4ShKsHRo8grVD0DSDgIcFHnpzqZw8d1n3sFbR7PJZpxqO9Tbudn9vtigj7R8u6Fj7gW7I/2ZMgnWgdd2tz/9+ALeC9VuLf5UPb+72UTF/VvnyDvdDTFy2W95essMHNqtzDalRODvB2t0pZKl1grUlPvM3E6zNrSiJAALxC1QbHi7H/7ykX6xTPKd8+lI1R6mKDlwsV2/u9zeiVTl5rtqDjrYfyMZO/nVJHhoWrqskWAdS1v26zl68IuXDvtUn4tYa67eXbJeJN3f7qFowm8x+uWa8MJOX75ERX2/3vTb75YelasH7CNbuT6N7PgOC9T3TJe6NBOvEeVEaAQRuF3h4xHdy9MxF/UK9Ejnl447OBuuSr38tF69E70oy+pny0rpK/pAdJv8146Hel5AdBJtO/OKVa1Ly9SW6dreuWI/5doe89/1ufQ4PF84uMzvH/+3SlJVRMmzRrz6JOV1rykMFsvl+rvP2D3Lg1Hn9Mzcv2jRhAlwtwTrAoHeqjmDtEDTNIOBhgVojv9d74KrjsVK5ZEqHqo721v/mybHPVpSWlfI62n6gG1OhZu/v52Roi7KSwsUHbwS6X0m9Pv9lS24Fa7XTh9rxQx2PFMsh0/5ePd5hmbpqrwxduM33mloyopaOxBzqXoB3lu6UDg8XlGqFbl3JdmOMyTFm6gRrMyfLpZiQlgmpAIEkL+B/9apR6dzy4QtVHDXxX+P9/nOV5MkKDzjaPo0hYCoQ99Hhpu8LVDn/tdN3+3bpv2v2yeCvtvqaXdC9tpTLlyVQpxHQesgxZpwEazMny6WYkJYJqQCBJC9Q/9/LJOr3c9qhWbn7ZeLfHnLUpPpb4XLsbPQa70nPV5YmZfM42j6NIWAqEBOs3Vpj/dGKKBm+OHqJh3qSonqiYnzHtIj9orbmizkW9agtZR4gWJuOczCWI1g7NCoEa4egaQYBDws8Nma57D7+l+7hE+XzyPh2lR3trf9SlCkvVJHH/B697OiJ0BgCCQjEBOu7LcOwE9F/icfdfgme8dMBGThvs+9UlvzrESl5f2Y7T+2e6ybHmNERrM2cLJdiQlompAIEkrxA43dXyI5jf2qHFhUfkHFtKzlq4r8UJe4T4hw9ERpDIAGBUUu2y4JfjshnnapLgewZHPea9uM+eX1+9BKP5hUekPeei/+z+r91B+TVObeC9dJX6kix3JkcP1+TBskxJkoiBGszJ8ulmJCWCakAgSQv0HTcSlGPa1ZHq0p5ZcyzFR01qTv6B9l/MnqHAhVYavk9yMLRE6ExBIJcQD3OXD3WXH9WK+eVMW3i/6zOXn9Q+n6xydeb73rXlSI5MwZl78gxZsNCsDZzslyKCWmZkAoQSPICT76/SjYfPqMdWj+UT0a3ruCoyaOjf5B9N4N13AdZOHoiNIZAkAv4B+Y2VfLJ28/E/1mdG3lIen3+i683y/o8KgVzOH+F3YSTHGOixBVrrbR//34pWPD2J4h99NFH0qlTJ9m4caP07NlT1q9fL9mzZ5cePXpInz59zIRvlmJCJoqLwgggEI9Aiwmr5ZeDp/Urbavml5FPl3fUqd6/l+nt6dQxp+vD8lABd7f/crTzNIZAIgTmbTgkr/wvOjC3q/6gvPVUuXjfPX/jYek5a6PvtZX96kn++9InoiXnipJjzKy5Yi0iCxYskDZt2khUVJQkS5bMJ5clSxY5f/68lCxZUlq2bCm9evWSiIgI6datm0ycOFE6duxopiwiTEhjKgoigMAdBFpNXC2RB6KD9d3+sbYL0D9Yz/9nLangt9+uXW1SLwKhKKDWd//fzA361Ds8XECGtCgbbzf8y6kCq/vXl7xZ0wVll8kxZsNCsBaRt956Sz7//HN9ZTruMWLECJkwYYLs27dPUqZMqV8eOHCgzJ07V7Zvv/UY0oS4mZAJCfE6AggkJNB60hpZt+8PXax9jQLyZsv4/7FOqJ57fd1/u79g3hbsXvvH+xAIlMCSLUfl5emRurq/1y4krz9ROt6qF28+Kt0+iy6njogBDeT+LGkDdRoBrYccY8ZJsFZfqbZtK8mTJ5cZM2bcptasWTPJmjVrrNfCw8OlYcOGcuzYMcmVK5eRNBPSiIlCCCBwF4G2H/4oEVGndIkXaxaUsOZlHPWq/84yiToRvRTk21fqSPEg3b3AURQaQyAegfBtx6TTp+v1K13qFJYBzUrF67Rky2/y8vSffa+tfa2B5MpEsA7lSUWwFpEyZcpInjx55NKlS7Jz504pVqyYvP7669K4cWMpX768NG3aVEaNGuUb523btun3rFu3TqpUiX/T97Nnz4r6L+Y4evSoVKtWTQ4ePCj58uUL5TnDuSOAgEsCf5sSIat3n9St3+0qmF2n1+CdZbLnZrAO5t0L7Oo/9SJgKrBsx3F58eN1uvjLdYtI/6Yl433r0m3H5B83A7gq8POgxyTb5NxPAAAgAElEQVR7xjSmzThajguEZtxJPlhfvnxZMmTIIJUqVZLRo0dLxowZZfr06TJu3DhZunSpdOnSRdq1aydDhw71iaq12EWKFJGVK1dK7dq145UOCwuTIUOG3PYawdpsYlIKAQRuF2j/n59k5a7f9Qud6xSWgXe4CmaXnf8Dalb0rScPZg/Om6zs6j/1ImAqsGrX7/L8f37Sxbs+WkRebRJ/sP5++zF56ZPoK9vq2PhGQ8maPrVpM46WI1ibcSf5YK2Yzp07p9dPp0lz67fEJk2a6BsZ1URSy0Hiu2IdGRmpA3l8B1eszSYgpRBAwFyg48dr5YcdJxL8x9q8xsSVbDhmuey6+eTHNf3rywNBepNV4npFaQQCLxARdVLafhiR4GfV/8q2KrwprJFkTpsq8CcUgBoJ1maIBOs7OPXt21eWLFki+fPn11vsTZs2zVeSNdZmk4tSCCAQWIFO/10n4b8e15V2r1dU+jQuEdgGEqjNP1gH81pQR1FoDIF4BH7ef0qe/uBH/crdloKs2HlCXpi61lfDliGNJWOa6I0Sgu0gWJuNSJIP1ps2bZKaNWvqEO2/rKNevXqSO3duqVixokyaNEn27NkjKVKk0KoDBgyQefPmsSuI2RyjFAIIBEig86fr5dttx3RtPRoUk14NiweoZrNq/B+pHvl6Q7kvQ3B+ZW3WG0ohYJ+A2m9e7TufULBevft3+duU6CUj6vh1aBNJlzo6awTbQbA2G5EkH6yvX7+ug7VaDvLBBx9Ijhw55MMPP9T7VKubE1W4VvtYN2/eXPr16ydr166Vrl276rDdoUMHM2X2sTZ2oiACCNxZoOv0n+XrLb/pAq88Vlx6PlbMUa4mY1fI9t/+1G0G81fWjqLQGALxCGw5fEaeeH+VfuVua6x/3HNSnvsoesmIOnYMayJpUhKsQ3lSJflgrQbvxIkT0r9/f1m8eLGcPn1aKleuLCNHjpRHHnlEj60K2Oppixs2bNC7h/Tu3Vu6d++eqHHnN71EcVEYAQTiEeg+I1IWbjqqX+nbuIT8s15RR538g3UwX1lzFIXGEIhHYMdvf0rjsSsSDNZr956SNpOjl4yoY/fwppIyRfKgNCXHmA0LwdrMyXIpJqRlQipAIMkL9Jy1QeZvPKId1C4D6kqYk4d/sN41vKmkCtIA4KQJbSEQn8CeE39Jg3eWJxis/ddiq8J7RzSL9QToYNIlx5iNBsHazMlyKSakZUIqQCDJC/T630aZu+GwdhjYrKR0ruNesA7mAJDkJwoArgscOHle6oz+QZ/H3W5e3HDgD3lq4hrf+e4b+bjr536nEyDHmA0NwdrMyXIpJqRlQipAIMkL9Jn9i3zx8yHtMOjxUtLpkcKOmjQbt1K2HY1+8FUwBwBHUWgMgXgEDp++ILVGfp9gsN506LQ0Hx99k2Owf67IMWZTnWBt5mS5FBPSMiEVIJDkBfrP2SSz1h3UDmFPlpYXaxVy1OTx91bK1iMEa0fRaSwkBY6fvSjV3vpOn/vdbl70v8mRYB2SQ33bSROsHRpHgrVD0DSDgIcFBs7bLDN+OqB7+GaLMtL+4YKO9pZg7Sg3jYWwwKlzl6Xym0sTDNa/Hj0rTcet9PU0mL8JIseYTUiCtZmT5VJMSMuEVIBAkhd4/cstMi1iv3YY/lRZ+Vv1Ao6aPPH+StlymCvWjqLTWEgKnLlwRSoM+Vaf+93WWO889qc0ejd69xB1EKxDcrhjnTTB2qExJFg7BE0zCHhYIOyrrfLJmn26hyNblZO21R50tLdPvr9KNh8+E/QBwFEUGkMgHoGzF69I+bCEg/Xu43/JY2Oidw8hWHtjKhGsHRpHgrVD0DSDgIcF3ly4Tf6zaq/u4ehnykvrKvkd7e3mQ2fkqYmrpUGpXDK5fRVH26YxBEJJ4MaNG/L4e6tk7+/nZHnfRyVX5rTxnr56vd6/lxGsQ2lwEzhXgrVDg0mwdgiaZhDwsMBbi3+VD1dE6R6OaVNBWlXO53hvz5y/IpnSppTkyZM53jYNIhBKAleuXZeLV65JprSp7nja/tvyccU6lEb3zudKsHZoHAnWDkHTDAIeFhj59XaZtHyP7uG4thWlRcW8Hu4tXUPA+wKH/jgvtUdF73dNsPbGeBOsHRpHgrVD0DSDgIcFRn+zXSb8EB2s33+ukjxZ4QEP95auIeB9gaNnLsjDI6L3uyZYe2O8CdYOjSPB2iFomkHAwwJjlu6U977bpXv4wd8qS9NyeTzcW7qGgPcF/Pe7Jlh7Y7wJ1g6NI8HaIWiaQcDDAuPCd8m74Tt1Dye3f0gal7nfw72lawh4X+D3vy5JlWHhXLH20FATrB0aTIK1Q9A0g4CHBcZ/v0v+/W10sJ7yQhV5rHRuD/eWriHgfYE/zl2WSjcfJMMVa2+MN8HaoXEkWDsETTMIeFhgwg+7ZfQ3O3QPP+5YVeqVyOXh3tI1BLwv4P8gGYK1N8abYO3QOBKsHYKmGQQ8LOAfrD99qZrUKZ7Tw72lawh4X+CvS1el7OBvfB3lyYuhP+YEa4fGkGDtEDTNIOBhAf9g/Vmn6lKraA4P95auIeB9gQuXr0mpN5YQrD001ARrhwaTYO0QNM0g4GEB/2A98x815OEi2T3cW7qGgPcFLl29JiUGEay9NNKeCtYXLlyQFClSSOrUqeXXX3+VRYsWSc2aNfV/bh8Ea7dHgPYRCH0B/2A9++WHpWrB+0K/U/QAgSQscPXadSn62tdcsfbQHPBMsF6xYoW0bNlSvvjiCyldurQUL15ckidPLn/99ZfMnDlTWrdu7eqwEaxd5adxBDwh4B+s53StKQ8VyOaJftEJBJKqwI0bN6TQgMUEaw9NAM8E69q1a+swPW7cOJk6daqMHj1adu3apf//lClTZMOGDa4OG8HaVX4aR8ATAv7B+st/1pKK+bN6ol90AoGkLFCw/yKCtYcmgGeCdfr06WXLli1SuHBhadasmeTPn18mT54s+/fvl5IlS4paJuLmQbB2U5+2EfCGgH+wXtC9tpTLl8UbHaMXCCRhAYK1twbfM8E6Z86csnLlSilUqJBkz55dPv74Y73846effpIWLVrIb7/95urIEaxd5adxBDwh4B+sF/d4REo/kNkT/aITCCRlAYK1t0bfM8G6TZs2+qr0fffdJ3PmzJGjR4/Knj17pEuXLvoqtlpn7eZBsHZTn7YR8IaAf7D+5l91pMT9mbzRMXqBQBIWIFh7a/A9E6xPnDghL7/8skRFRcngwYP1jYy9e/fWV6zVDY3333+/qyNHsHaVn8YR8ISAf7AO71VHiuYiWHtiYOlEkhYgWHtr+D0TrOMblsuXL+ut94LhIFgHwyhwDgiEtoB/sP6+d10pnDNjaHeIs0cAASFYe2sSeCpYq+UfH330kWzfvl3Gjh0ry5cvl3LlyumbF90+CNZujwDtIxD6Av7BekXfevJg9vSh3yl6gEASFyBYe2sCeCZY7969W6pXry5ZsmQRFWJVuH711Vfl66+/lm+//db1h8QQrL31waE3CLgh4B+sV71aT/JlI1i7MQ60iUAgBQjWgdR0vy7PBGu184faGURdsc6cObP88ssvUqBAAXnxxRflwIED+uq1mwfB2k192kbAGwL+wXpN//ryQNZ03ugYvUAgCQsQrL01+J4J1ipUq/CsnrqYKVMmHazVbiDqyrW6kn3mzBlXR45g7So/jSPgCQGCtSeGkU4gEEuAYO2tCeGZYJ0tWza9j3XZsmVjBWu1K0jTpk3l1KlTro4cwdpVfhpHwBMCBGtPDCOdQIBg7eE54Jlg3bZtWz1Mn332mWTNmlU2bdqkl4S0atVKcuXKJbNnz05wGHfu3CmVK1eW8ePH6yUk6ti4caP07NlT1q9frx8806NHD+nTp0+CdcUtQLBONBlvQACBOAIEa6YEAt4T4Iq1t8bUM8H6yJEj8uijj8rJkyf1sg+1E4h6nLl6YMyKFSv0euu7HVeuXNE3OKoArZ7aqIK1qkvVo/bE7tWrl0REREi3bt1k4sSJ0rFjx0TNBIJ1orgojAAC8Qis2HlCXpi6Vr+yOayRZEqbCicEEAhxAYJ1iA9gnNP3TLBW/Tp//rx+wuKGDRvk+vXrelnI888/r69cJ3QMHDhQB+cffvjBF6xHjBghEyZMkH379knKlCl1Farc3Llz9drtxBwE68RoURYBBOITuHHjhsxce1ByZUojj5XODRICCHhAgGDtgUH064KngrXq18WLF0Ut6UiTJo2+eTFVqoSv6Kgr2k899ZRe9vHggw/6gnWzZs30spIZM2b4yMLDw6Vhw4Zy7NgxvcTE9CBYm0pRDgEEEEAAgaQjQLD21lh7Jlhfu3ZNr32eNGmSqCcuqiNdunR6L+tBgwZJsmTJ4h2506dPS4UKFURdnW7Xrp0uF7MUpHz58vrGx1GjRvneu23bNilTpoysW7dOqlSpYjwbCNbGVBREAAEEEEAgyQgQrL011J4J1q+99ppMnTpV3nzzTb29ngraapeQsLAw6d69uwwZMiTekXvuuedEfb06a9Ys/bp/sC5atKgO20OHDvW9NyoqSooUKaLrrl279h1nw9mzZ0X9F3Oop0JWq1ZNDh48KPny5fPWLKI3CCCAAAIIIHBPAgTre2IL2jd5Jljnzp1bX2lWyzf8jzlz5uhgrYJt3GPatGkyYMAA2bx5s6jt+uIGa/U4dFVffFesIyMjpVKlSnccWBXo4wvzBOug/SxwYggggAACCDguQLB2nNzWBj0TrNVa6B9//FFKlSoVC0wt3VBXsP/888/bIOvVqyerVq3S67FjjnPnzumf1VVptZOI2mJPBfCYw3SNNVesbZ23VI4AAggggIAnBAjWnhhGXyc8E6zVXtPHjx+X//73v5I6dWpfB1966SW9o8eHH35428gdPnxYLly4EOvPixUrpq9QP/vss3qHEbVme8+ePZIiRQpdTl3hnjdvHruCeOtzQG8QQAABBBBwRYBg7Qq7bY2GdLCuX7++D+bq1av66nOePHmkatWqOgirbffU0osWLVrIF198YYTov8ZaBXW1j3Xz5s2lX79+snbtWunatasO2x06dDCqL6YQNy8miovCCCCAAAIIJAkBgrW3hjmkg3ViHtKi1l+bHP7BWpVXu3+opy2qkK5Ce+/evfWa7cQeBOvEilEeAQQQQAAB7wtMWRkl477bJZOff0hqFs0RtB0mx5gNTUgHa7MuBkcpJmRwjANngQACCCCAQLAJXL9+Q5Inj39b4GA5V3KM2Uh4KlirQVdPSty0aZN+MIzab7pLly76oS9uH0xIt0eA9hFAAAEEEEDgXgXIMWZyngnWW7ZskTp16kj69On1ftFqH+v169eL2uVj9erVOmS7eTAh3dSnbQQQQAABBBCwIkCOMdPzTLBWT0jMkCGDfPbZZ77t89TjzZ9//nn9mPOFCxeaidhUiglpEyzVIoAAAggggIDtAuQYM2LPBOuMGTNKRESElC1bNlbP1bIQdSVbPbrczYMJ6aY+bSOAAAIIIICAFQFyjJmeZ4K12rFDXZV+6KGHYvVc7erRsGFDgrXZfKAUAggggAACCCBwmwDB2mxSeCZYqyUfR44c0ftV33fffbr3v//+u7Rp00bUUxnnzp1rJmJTKSakTbBUiwACCCCAAAK2C5BjzIg9E6zVgNesWVNOnTolxYsXF7Uf9Y4dOyRbtmyyfPlyKVy4sJmITaWYkDbBUi0CCCCAAAII2C5AjjEj9kywvnHjht4BZNq0aaJ2CFE/lytXTtq1aydZsmQx07CxFBPSRlyqRgABBBBAAAFbBcgxZryeCdZVqlSRqVOnSvny5c167nApJqTD4DSHAAIIIIAAAgETIMeYUXomWOfIkUN++uknKVKkiFnPHS7FhHQYnOYQQAABBBBAIGAC5BgzSs8E69GjR8uiRYukb9++UrRoUUmXLl0sAbefvsiENJuQlEIAAQQQQACB4BMgx5iNiWeCdfLkyX09VjcuxhxqrbX6WT2J0c2DCemmPm0jgAACCCCAgBUBcoyZnmeCtdr5425H3bp1zURsKsWEtAmWahFAAAEEEEDAdgFyjBmxZ4K1f3fVlnspUqQIit1AYs6LCWk2ISmFAAIIIIAAAsEnQI4xGxNPBWu1znrcuHFy9OhR3ftChQrJq6++Kv/4xz/MNGwsxYS0EZeqEUAAAQQQQMBWAXKMGa9ngvWoUaNk6NCh0qNHD3n44Yfl+vXrsnr1apk4caK8++670rlzZzMRm0oxIW2CpVoEEEAAAQQQsF2AHGNG7JlgnT9/fnnrrbekffv2sXqu9rYeMWKE7Nq1y0zEplJMSJtgqRYBBBBAAAEEbBcgx5gReyZYZ8yYUSIjI/XjzP0PFagrVKgg58+fNxOxqRQT0iZYqkUAAQQQQAAB2wXIMWbEngnWHTp0kEyZMsn48eNj9XzIkCH6EeezZ882E7GpFBPSJliqRQABBBBAAAHbBcgxZsSeCda9evXS66lLlSolamu9VKlSybp162TlypXSokULyZo1q09ELQ9x+mBCOi1OewgggAACCCAQKAFyjJmkZ4J1vXr1zHosIj/88INx2UAVZEIGSpJ6EEAAAQQQQMBpAXKMmbhngrVZd90rxYR0z56WEUAAAQQQQMCaADnGzI9gbeZkuRQT0jIhFSCAAAIIIICASwLkGDN4grWZk+VSTEjLhFSAAAIIIIAAAi4JkGPM4AnWZk6WSzEhLRNSAQIIIIAAAgi4JECOMYMP6WD91VdfSaNGjSRt2rRmvXWxFBPSRXyaRgABBBBAAAFLAuQYM76QDtbqoTDbt2+XfPnySeHChfX2etmzZzfrucOlmJAOg9McAggggAACCARMgBxjRhnSwTpXrlzSpUsXeeyxx0Rttzdv3jzJli1bvD2vU6eOmYhNpZiQNsFSLQIIIIAAAgjYLkCOMSMO6WA9atQoGThwoNy4cUOSJUum/ze+Q7127do1MxGbSjEhbYKlWgQQQAABBBCwXYAcY0Yc0sFadfHMmTNy+vRpKVSokKxdu1Zy5swZb88LFChgJmJTKSakTbBUiwACCCCAAAK2C5BjzIhDPljHdHP58uVSq1YtSZkypVnPHS7FhHQYnOYQQAABBBBAIGAC5BgzSs8Ea9XdhQsXyvDhw2XTpk2SKlUqKV26tPTt21eeeuqpu2ocP35cevfuLUuWLJELFy5I3bp1ZfTo0fr96ti4caP07NlT1q9fr2+O7NGjh/Tp08dM+GYpJmSiuCiMAAIIIIAAAkEkQI4xGwzPBOv58+dLq1atdIhWNyqqNdUrVqyQBQsWyNy5c6V58+Z3FKlRo4YkT55c3nvvPcmQIYO8/vrrsmbNGtm9e7cO2iVLlpSWLVtKr169JCIiQrp16yYTJ06Ujh07mimLCBPSmIqCCCCAAAIIIBBkAuQYswHxTLCuXLmyDtUqFPsfQ4cO1Vey1frr+I6TJ09K9+7dZdCgQVKmTBld5JdffpGKFSvq94SHh8uECRNk3759vmUm6oZJFdbVVn+mBxPSVIpyCCCAAAIIIBBsAuQYsxHxTLBOly6dbN68WYoWLRqr57t27ZIKFSrI+fPnjUSOHTuml3ksW7ZMB+fWrVtL1qxZZcaMGb73q7DdsGFDUWXVln8mBxPSRIkyCCCAAAIIIBCMAuQYs1HxTLAuUqSIvP/++9KsWbNYPV+0aJH84x//kCNHjiQo0rlzZ/noo48kTZo0EvNUx/Lly0vTpk1Fbe0Xc2zbtk1f3VYPpKlSpUq89Z49e1bUfzHH0aNHpVq1anLw4EH9QBsOBBBAAAEEEEAgVAQI1mYj5ZlgrZaATJs2TT744AO9O4g6Vq1apddDt2jRQsaNG5egiArMak21qkNdoVbvb9OmjbRr107UkpKYIyoqSlSQX7lypdSuXTveesPCwmTIkCG3vUawTnAYKIAAAggggAACQSZAsDYbEM8E64sXL0rbtm31lWb1QBh1qAfGqCvYn3/+uaRPn95MRESuX78u5cqVk+rVq+ur0qqO+K5YR0ZGSqVKleKtlyvWxtwURAABBBBAAIEgFyBYmw2QZ4J1THfVumi11lqFahWOS5UqdVcJtdXed999p69Mp0iRwldWra1W4Vj9mdpiT10NjzlYY202uSiFAAIIIIAAAt4QIFibjaPngrVZt2+VUntUq6vOKlzXr19fv3DlyhW9xZ7aoi937twyadIk2bNnjy94DxgwQObNm8euIInFpjwCCCCAAAIIhKQAwdps2JJ8sFZXttXNiWo7PXXjYrZs2fRDZr755hv9YJi0adP6Qna/fv30Fnxdu3bVYbtDhw5myuxjbexEQQQQQAABBBAIPgGCtdmYJPlgrZjOnDkj6ir0l19+KadPn5ZHHnlExowZ49vXWq2zVk9b3LBhg+TJk0c/pVHtfZ2YgwmZGC3KIoAAAggggEAwCZBjzEaDYG3mZLkUE9IyIRUggAACCCCAgEsC5BgzeIK1mZPlUkxIy4RUgAACCCCAAAIuCZBjzOA9E6z37t0rr776qmzZskXU1ntxD7X3tJsHE9JNfdpGAAEEEEAAASsC5BgzPc8Ea/WgFvV0Q7VtnrrhMO4xePBgMxGbSjEhbYKlWgQQQAABBBCwXYAcY0bsmWCdIUMG/STEypUrm/Xc4VJMSIfBaQ4BBBBAAAEEAiZAjjGj9EywVoF67NixUqdOHbOeO1yKCekwOM0hgAACCCCAQMAEyDFmlJ4J1hEREXp/abUtXqFChSR58uSxBNwO3ExIswlJKQQQQAABBBAIPgFyjNmYeCZY/+c//9HB+urVq7f1PFmyZHLt2jUzEZtKMSFtgqVaBBBAAAEEELBdgBxjRuyZYJ0/f3554okn9BXr9OnT39b7AgUKmInYVIoJaRMs1SKAAAIIIICA7QLkGDNizwRrdfPi1q1bpWDBgmY9d7gUE9JhcJpDAAEEEEAAgYAJkGPMKD0TrJ955hlp1qyZvPTSS2Y9d7gUE9JhcJpDAAEEEEAAgYAJkGPMKD0TrMePHy8DBgyQpk2bSokSJSRVqlSxBN544w0zEZtKMSFtgqVaBBBAAAEEELBdgBxjRuyZYK12ArnToW5e5MmLZhOCUggggAACCCCAQFwBgrXZnPBMsDbrrnulmJDu2dMyAggggAACCFgTIMeY+RGszZwsl2JCWiakAgQQQAABBBBwSYAcYwbvmWCtHgijlnzc6WAfa7MJQSkEEEAAAQQQQCCuAMHabE54Jlh/8sknsYL1lStXZNeuXaL+fMyYMfK3v/3NTMSmUkxIm2CpFgEEEEAAAQRsFyDHmBF7JljfqbszZsyQadOmyddff20mYlMpJqRNsFSLAAIIIIAAArYLkGPMiD0frPfv3y+lS5eWc+fOmYnYVIoJaRMs1SKAAAIIIICA7QLkGDNizwfrkSNHyoQJE+TgwYNmIjaVYkLaBEu1CCCAAAIIIGC7ADnGjNgzwVrtY+1/8+KNGzfk7Nmzcvr0aRk+fLj079/fTMSmUkxIm2CpFgEEEEAAAQRsFyDHmBF7JliHhYXdtitI6tSppVatWlKnTh0zDRtLMSFtxKVqBBBAAAEEELBVgBxjxuuZYG3WXfdKMSHds6dlBBBAAAEEELAmQI4x8/NUsF69erWsWrVKLl++LGopiP/xxhtvmInYVIoJaRMs1SKAAAIIIICA7QLkGDNizwRrdZPiwIEDJVu2bJI5c+ZYvVdrr6OiosxEbCrFhLQJlmoRQAABBBBAwHYBcowZsWeCdb58+eSll16SoUOHmvXc4VJMSIfBaQ4BBBBAAAEEAiZAjjGj9EywTp8+vWzatEmKFi1q1nOHSzEhHQanOQQQQAABBBAImAA5xozSM8G6UaNG0rFjR3nuuefMeu5wKSakw+A0hwACCCCAAAIBEyDHmFF6JlhPmTJF+vXrp8N1yZIlJU2aNLEEXnjhBTMRm0oxIW2CpVoEEEAAAQQQsF2AHGNG7JlgnTx58jv2WN28eO3aNTMRm0oxIW2CpVoEEEAAAQQQsF2AHGNG7JlgbdZd90oxId2zp2UEEEAAAQQQsCZAjjHzS/LB+tSpU3qbvoULF+pHoJcvX17U1n21a9fWghs3bpSePXvK+vXrJXv27NKjRw/p06ePma5fKSZkosl4AwIIIIAAAggEiQA5xmwgknywVjc9Hj9+XMaPHy85c+aUCRMmiFqvHRkZqX9W67VbtmwpvXr1koiICOnWrZtMnDhRr+VOzMGETIwWZRFAAAEEEEAgmATIMWajkaSD9e7du6VYsWKinthYs2ZNLaae2Fi8eHG9u0i6dOl00N63b5+kTJlSv66ubs+dO1e2b99uJnyzFBMyUVwURgABBBBAAIEgEiDHmA1Gkg7Wp0+fljVr1kiDBg1i7SKi9sJu0qSJflpj1qxZZcaMGT7N8PBwadiwoRw7dkxy5cplpiwiTEhjKgoigAACCCCAQJAJkGPMBiRJB+v4iGbPni1t2rSRr776Sl577TVp2rSpjBo1yld027ZtUqZMGVm3bp1UqVLljspqvbb6L+Y4evSoVKtWTQ4ePCjqKZEcCCCAAAIIIIBAqAgQrM1GimDt56SWhKgr1eoK9pdffqmf4tiuXbtYj0lXV7GLFCkiK1eu9N3gGB91WFiYDBky5LaXCNZmE5NSCCCAAAIIIBA8AgRrs7EgWN90mj9/vg7RNWrU0DuEqPXV5cqVk2bNmsV7xVrd3FipUiWuWJvNM0ohgAACCCCAQAgLEKzNBo9gLaJ3BFFb6rVq1UqmT5/uW2+tQrXaYm/atGk+TdZYm00sSiGAAAIIIICAdwQI1mZjmeSD9QcffKC30FP7U7/77rvi/wRHtZ/1pEmTZM+ePZIiRQotOmDAAJk3bx67gpjNL0ohgAACCCCAgAcECNZmg5ikg/XOnTulbNmy8sQTT+i9qf0PtRTk0qVLeh/r5s2bS79+/WTt2rXStWtXHbY7dITxCJcAABlaSURBVOhgJnyzFBMyUVwURgABBBBAAIEgEiDHmA1Gkg7Wb731lt75I75DBedPPvlE7/6hrmZv2LBB8uTJI71795bu3bub6fqVYkImmow3IIAAAggggECQCJBjzAYiSQdrM6LAlGJCBsaRWhBAAAEEEEDAeQFyjJk5wdrMyXIpJqRlQipAAAEEEEAAAZcEyDFm8ARrMyfLpZiQlgmpAAEEEEAAAQRcEiDHmMETrM2cLJdiQlompAIEEEAAAQQQcEmAHGMGT7A2c7JciglpmZAKEEAAAQQQQMAlAXKMGTzB2szJcikmpGVCKkAAAQQQQAABlwTIMWbwBGszJ8ulmJCWCakAAQQQQAABBFwSIMeYwROszZwsl2JCWiakAgQQQAABBBBwSYAcYwZPsDZzslyKCWmZkAoQQAABBBBAwCUBcowZPMHazMlyKSakZUIqQAABBBBAAAGXBMgxZvAEazMny6WYkJYJqQABBBBAAAEEXBIgx5jBE6zNnCyXYkJaJqQCBBBAAAEEEHBJgBxjBk+wNnOyXIoJaZmQChBAAAEEEEDAJQFyjBk8wdrMyXIpJqRlQipAAAEEEEAAAZcEyDFm8ARrMyfLpZiQlgmpAAEEEEAAAQRcEiDHmMETrM2cLJdiQlompAIEEEAAAQQQcEmAHGMGT7A2c7JciglpmZAKEEAAAQQQQMAlAXKMGTzB2szJcikmpGVCKkAAAQQQQAABlwTIMWbwBGszJ8ulmJCWCakAAQQQQAABBFwSIMeYwROszZwsl2JCWiakAgQQQAABBBBwSYAcYwZPsDZzslyKCWmZkAoQQAABBBBAwCUBcowZPMHazMlyKSakZUIqQAABBBBAAAGXBMgxZvAEazMny6WYkJYJqQABBBBAAAEEXBIgx5jBE6zNnCyXYkJaJqQCBBBAAAEEEHBJgBxjBk+wNnOyXIoJaZmQChBAAAEEEEDAJQFyjBk8wdrMyXIpJqRlQipAAAEEEEAAAZcEyDFm8ARrMyfLpZiQlgmpAAEEEEAAAQRcEiDHmMETrM2cLJdiQlompAIEEEAAAQQQcEmAHGMGT7A2c7JciglpmZAKEEAAAQQQQMAlAXKMGTzB2szJcikmpGVCKkAAAQQQQAABlwTIMWbwBGszJ8ulmJCWCakAAQQQQAABBFwSIMeYwROs4zgNGzZMwsPDZdmyZb5XNm7cKD179pT169dL9uzZpUePHtKnTx8z4ZulmJCJ4qIwAggggAACCASRADnGbDAI1n5OY8eOlV69ekmdOnV8wfrkyZNSsmRJadmypX4tIiJCunXrJhMnTpSOHTuaKYsIE9KYioIIIIAAAgggEGQC5BizASFYi8jhw4elU6dOsnLlSsmfP7/kzp3bF6xHjBghEyZMkH379knKlCm16sCBA2Xu3Lmyfft2M2WCtbETBRFAAAEEEEAg+AQI1mZjQrAWkQULFsjMmTNFLQMZOnSoDtExS0GaNWsmWbNmlRkzZvhE1VKRhg0byrFjxyRXrlxG0kxIIyYKIYAAAggggEAQCpBjzAaFYB3H6cUXX4wVrMuXLy9NmzaVUaNG+Upu27ZNypQpI+vWrZMqVaoYSTMhjZgohAACCCCAAAJBKECOMRsUgnUCwbpo0aLSrl07fSU75oiKipIiRYropSO1a9eOV/rs2bOi/os5jh49KtWqVZODBw9Kvnz5zEaHUggggAACCCCAQBAIEKzNBoFgnUCwLleunKjlIPFdsY6MjJRKlSrFKx0WFiZDhgy57TWCtdnEpBQCCCCAAAIIBI8AwdpsLAjWCQRrFarVFnvTpk3zlTRZY80Va7MJSCkEEEAAAQQQCH4BgrXZGBGsEwjWI0eOlEmTJsmePXskRYoUuvSAAQNk3rx57ApiNscohQACCCCAAAIhLkCwNhtAgnUCwfr48eN6H+vmzZtLv379ZO3atdK1a1cdtjt06GCmzHZ7xk4URAABBBBAAIHgEyBYm40JwTqBYK1eVrt/qKctbtiwQfLkySO9e/eW7t27mwnfLMWETBQXhRFAAAEEEEAgiATIMWaDQbA2c7JciglpmZAKEEAAAQQQQMAlAXKMGTzB2szJcikmpGVCKkAAAQQQQAABlwTIMWbwBGszJ8ulmJCWCakAAQQQQAABBFwSIMeYwROszZwsl2JCWiakAgQQQAABBBBwSYAcYwZPsDZzslyKCWmZkAoQQAABBBBAwCUBcowZPMHazMlyKSakZUIqQAABBBBAAAGXBMgxZvAEazMny6WYkJYJqQABBBBAAAEEXBIgx5jBE6zNnCyXYkJaJqQCBBBAAAEEEHBJgBxjBk+wNnOyXIoJaZmQChBAAAEEEEDAJQFyjBk8wdrMyXIpJqRlQipAAAEEEEAAAZcEyDFm8ARrMyfLpZiQlgmpAAEEEEAAAQRcEiDHmMETrM2cLJdiQlompAIEEEAAAQQQcEmAHGMGT7A2c7JciglpmZAKEEAAAQQQQMAlAXKMGTzB2szJcikmpGVCKkAAAQQQQAABlwTIMWbwBGszJ8ulmJCWCakAAQQQQAABBFwSIMeYwROszZwsl2JCWiakAgQQQAABBBBwSYAcYwZPsDZzslyKCWmZkAoQQAABBBBAwCUBcowZPMHazMlyKSakZUIqQAABBBBAAAGXBMgxZvAEazMny6WYkJYJqQABBBBAAAEEXBIgx5jBE6zNnCyXYkJaJqQCBBBAAAEEEHBJgBxjBk+wNnOyXIoJaZmQChBAAAEEEEDAJQFyjBk8wdrMyXIpJqRlQipAAAEEEEAAAZcEyDFm8ARrMyfLpZiQlgmpAAEEEEAAAQRcEiDHmMETrM2cLJdiQlompAIEEEAAAQQQcEmAHGMGT7A2c7JciglpmZAKEEAAAQQQQMAlAXKMGTzB2szJcikmpGVCKkAAAQQQQAABlwTIMWbwBGszJ8ulmJCWCakAAQQQQAABBFwSIMeYwROszZwsl2JCWiakAgQQQAABBBBwSYAcYwZPsDZzslyKCWmZkAoQQAABBBBAwCUBcowZPMHazMlyKSakZUIqQAABBBBAAAGXBMgxZvAEazMny6WYkJYJqQABBBBAAAEEXBIgx5jBE6zNnCyXYkJaJqQCBBBAAAEEEHBJgBxjBk+wNnC6fv26DBkyRKZMmSJ//PGH1K5dWyZOnChFixY1eHd0ESakMRUFEUAAAQQQQCDIBMgxZgNCsDZwUqFaBemPP/5Y8ubNK/369ZOoqCjZunWrpE6d2qAGgrUREoUQQAABBBBAICgFCNZmw0KwTsDp8uXLkiNHDnn77bfl5Zdf1qVPnz4tDzzwgEydOlXatm1rJM2ENGKiEAIIIIAAAggEoQA5xmxQCNYJOK1du1aqV68uO3bskOLFi/tKq+Ug5cuX11eyTQ4mpIkSZRBAAAEEEEAgGAXIMWajQrBOwGnu3Lny9NNPy/nz5yVdunS+0m3atNF/tnDhwnhrOHv2rKj/Yo6DBw9KzZo1RQX1PHnymI0OpRBAAAEEEEAAgSAQOHr0qFSrVk327t0rBQsWDIIzCs5TIFgnMC7Tp0+X9u3by7Vr1yR58uS+0i+88IIcOXJEwsPD460hLCxM3/DIgQACCCCAAAIIeEVAXSCsWrWqV7oT8H4QrBMgnTNnjjzzzDPxXrG+dOmSzJ8/3+iK9cWLF0VdtS5UqJCkTJky4AOpKoz5bZKr4rbwWqqUsbHEZ+ubGRtbee+5csblnulsfyNjYzvxPTdg59hcvXpVTpw4IeXKlZO0adPe8zl6/Y0E6wRGOGaN9e7du6VIkSK+0mqNdYUKFWTChAlBM0dY/xQ0Q3HbiTA2jE3wCgTnmfGZCc5xUWfF2DA2wSvg/pkRrBMYA3VVOleuXDJmzBj5+9//rkvfy64gTgw1f9k5oXxvbTA29+bmxLsYGyeUE98G45J4M6fewdg4JZ34dhibxJsF+h0EawPR1157TSZPnqy311ML9vv27Sv79u2TzZs3G+9jbdCM5SJ8oCwT2lYBY2MbreWKGRvLhLZUwLjYwhqQShmbgDDaUgljYwtroiolWBtwqRsXBw4cqB8Qc+HCBalTp45eAhJsd8WqXUjUlfVevXpJ5syZDXpGEacEGBunpBPfDmOTeDMn3sG4OKF8b20wNvfm5sS7GBsnlO/eBsHa/THgDBBAAAEEEEAAAQQ8IECw9sAg0gUEEEAAAQQQQAAB9wUI1u6PAWeAAAIIIIAAAggg4AEBgrUHBpEuIIAAAggggAACCLgvQLB2fww4AwQQQAABBBBAAAEPCBCsPTCIqgvXr1/Xj1CfMmWK/PHHH6IeYDNx4kQpWrSoR3oY/N0YNmyYfsT9smXLfCe7ceNG6dmzp6xfv16yZ88uPXr0kD59+vheZ9zsGddTp07pnXwWLlwo6i758uXLy8iRI/XnQh2Miz3uJrUeP35cevfuLUuWLNG7LNWtW1dGjx4tpUuXZmxMAB0qs3PnTqlcubKMHz9eXnzxRcbGIfc7NbN///54dyL76KOPpFOnTvyd5vL4+DdPsA6iwbByKipUqyCttgTMmzev9OvXT6KiomTr1q1Btde2lT4G83vHjh2rtzlUWzHGBOuTJ09KyZIlpWXLlvq1iIgI6datmx6njh076u4wbvaMaqNGjUQFOBUKcubMqbfHVL90RkZG6p8ZF3vcTWqtUaOGJE+eXN577z3JkCGDvP7667JmzRpRT7dVQZuxMVG0t8yVK1ekZs2a+oKA+jdFBWv+PrPXPKHaFyxYIG3atNH/ridLlsxXPEuWLHL+/Hk+NwkBOvg6wdpBbLuaunz5suTIkUPefvttefnll3Uzwfp0SLsM3Kr38OHD+mrBypUrJX/+/JI7d25fsB4xYoQOdOphQilTptSnqK6izp07V7Zv3y6Mmz2jpgJasWLFZPXq1TocqOPGjRtSvHhxee655yRdunSMiz30Cdaqwln37t1l0KBBUqZMGV3+l19+kYoVK8ratWv1Nz58ZhJktL2A+ntKXQj44YcffMGav89sZ79rA2+99ZZ8/vnn+sp03IOxcXds4rZOsA6u8bins1H/IFWvXl127Nihw0PMob72Vl+BqyukHPYIqKsIM2fOFLUMZOjQoTpEx1yxbtasmWTNmlVmzJjha1wFh4YNG8qxY8d0WcYt8OOifqlUV0AbNGggadKk8TWglkU1adJEX/FhXALvfi81qs+BWhqlPjPql83WrVszNvcCGcD3rFixQp566ikd4B588EFfsObvswAi30NVbdu21d/0+P97ElMNY3MPoDa+hWBtI65TVasroE8//bT+OkhdjYs51NdG6s/UOlMO+wXU16X+wVr9UtO0aVMZNWqUr/Ft27bpK3Xr1q2TAwcOMG72D4tuYfbs2fpr1K+++kpee+01xsUh97s107lzZ1HrQ9UvP2pc1PIdPjPuDoz6pbRChQqiroC2a9dOLzmIWQrC2Lg7NurfjTx58silS5dErX9X38qpZVSNGzfmc+Pu0NzWOsE6yAbkXk5n+vTp0r59e1GPXle/0cYcL7zwghw5ckR/vcphv0DcYK2ukKp/nNSV7JhDXS0tUqSIXjqiQjjjZv+4qCUh6kq1uoL95Zdf6ht6GRf73RNqQf2SqdZUf/DBB/oq3KpVq/QvP4xNQnL2va6WSqllU7NmzdKN+AdrPjf2uSdUs1o2qO5HqFSpkr7RN2PGjKL+3R83bpwsXbpUunTpwucmIUQHXydYO4htV1Nz5syRZ555Jt4r1uq32/nz59vVNPX6CcQN1uXKlRP1FV18V6zVTXQqZDNu9k4hNfdVUFM3zKlvbtQ3OoyLveaJrV3tjKPGRC2LUt/k8JlJrGBgyk+bNk0GDBggmzdvlmzZst0WrPncBMb5Xms5d+6cvlfHf3mbumCgfvk5dOgQn5t7hbXhfQRrG1CdrjJmjbW6aUtdDY051Bpr9bWeuhmIw36BuMFaBQS1xZ76ByvmiG+NNeNmz9ioHUHUVoetWrXSV3di/kFiXOzxNqlV7dTy3Xff6SvTKVKk8L1Fra1W2yKqP+MzYyIZ+DL16tXT3xr4BzcV5tTP6t+VAgUKMDaBZ7dUY9++ffW2lerGeT43ligD+maCdUA53alMXZXOlSuXjBkzRv7+97/rk2BXEOfHIm6wVvsmT5o0Sfbs2eMLEeqK0Lx58/SNWoybfWOklheorQ3VvuHvvvturCVSjIt97gnVrG6IU19nq3Bdv359XVxt7aa22GvevLneVYfPTEKK9ryudjhSS3P8D7WOV33j9uyzz+qbtBkbe+wTqnXTpk16hyMVomP24lfvUb8Mqc+M2lWHsUlI0bnXCdbOWdvakroha/LkyTJ16lS9ibz6TVat4VVf66VOndrWtqk8WiBusFZX52ICg9pXXH2z0LVrV/0XYIcOHfR7GLfAzx51Y0/ZsmXliSeeuG1HHLUURP1Cw7gE3t2kRrV+V93Qq/5uUjcuqiUHw4cPl2+++UbvQpE2bVrGxgTSoTL+a6z5+8wh9HiaUculVLBW3yCoiwZqe90PP/xQ//2mlk+pcM3fae6NT9yWCdbBMxaWzkTduKj2HlV3cKurDupBJWoJiArZHM4IxA3WqlX1l566arphwwZ9R7d64pzaxzfmYNwCPzZqv1f1C0t8h/qF5pNPPmFcAs9uXOOZM2f0Wl51I6n6Zu2RRx7R37bF7GvNZ8aY0vaC/sGav89s575rAydOnJD+/fvL4sWL9edGPRVTffumPj+MjbtjQ7AOLn/OBgEEEEAAAQQQQMAjAlyx9shA0g0EEEAAAQQQQAABdwUI1u760zoCCCCAAAIIIICARwQI1h4ZSLqBAAIIIIAAAggg4K4Awdpdf1pHAAEEEi2gbkpWN8uGhYUl+r0mb1i2bJneymvv3r13vAHa7nMwOU/KIIAAAsEmQLAOthHhfBBAAIEEBNQOAWrrQPVoYzsO9QjlU6dOSc6cOWM9yMW/LYK1HfLUiQACoS5AsA71EeT8EUAAARcECNYuoNMkAggEvQDBOuiHiBNEAIFQFVD7AKs9tdVj7dWDaZYvXy6FChWS119/XT9mXe3prB5mM3ToUGnUqJGvm99++63e63nbtm1SuHBhvf+5eqpqzNKMuKF20aJF8uabb8qWLVskc+bM8txzz+kHr6gHrqhDnYd6oMTnn3+uH1t933336f3UVRvxHXGXgqjzVPuxz58/Xz9wSu2ZP3bsWFuXo4TqmHPeCCCQtAUI1kl7/Ok9AgjYKKACrXpK2tdffy1Xr16VGjVqSLt27WTr1q06mObLl08WLFigH/ygHnX/+OOP6ycQVq1aVV555RUdptXP6omdf/zxR7zBWj1o5emnn9brrdu0aSO7du3S5atVqyZz5szxBessWbLI+PHj9Z+roD9s2DBZsWKF7wET/gxxg3Xjxo3lwIEDOpxnypRJB/3vv/9eBg8ebNs6bxuHhaoRQAAB2wQI1rbRUjECCCR1ARWsVUBWTxZUx+7du6VYsWL6yY9VqlTx8agnQu7fv19UoFX/Xz2W/ccff/S9/v777+srxvFdsa5evbrkz59fvvjiC195FdabN2+ur3iXKlVKX7Hu2bOnDvMxh7pq3bdv33ivWvsH65hHwIeHh0uDBg30248dOyYFChTQvxDYdQNlUp879B8BBEJTgGAdmuPGWSOAQAgIqEA7ceJEfQVZHbNnz9ZXlTNkyBDr7K9cuSJZs2bVgbVcuXLSsGFDXxhXBdVV60qVKsUbrNOnTy/qMe7/+te/fHWqpRuqvlmzZsmzzz6rg/WkSZOkS5cuvjJqOUnHjh31Vee4h3+wXr9+vbRu3VovW1HLTGKO0qVL674QrENgInKKCCDgmADB2jFqGkIAgaQmoALtxx9/rNciq+N///uftG3bViIjI/WSCv8jRYoUev21CtB169aNdXX5bsFa7Q4ycuRIfUU65lA7emTPnl0vBWnVqpUO1v7nocrd7ebD+IK1WoqiwnrMUaFCBXnqqacI1kltUtNfBBC4qwDBmgmCAAII2CQQN9CqtdXqZsWFCxfq9dQxh7rBMXny5PoGRBXCt2/fLhEREb7XJ0+eLC+//HK8V6zVmukHH3ww1lIQdZNhy5YtYy0Fuddgfe3aNSlatGiscz59+rTkzZtXLyXhirVNk4dqEUAgJAUI1iE5bJw0AgiEgkB8V4qffPJJvbRjwoQJOmTPnTtX+vXrJ//5z3/00gy1s0fFihWlT58+8tJLL+lw3LlzZ1F7V+/bt0+vbfa/2qyWl6jlHkOGDNH/q9Znq6Unqg611lodVq5Yq7bUkg+1Lvyjjz6S+++/X6/LVjdkDho0iGAdChORc0QAAccECNaOUdMQAggkNYH4Au358+f1FnxqWYhasqG20+vVq5d06tTJx6MCsQqvaoePEiVK+Hb9OHLkiOTJk+e2ZRwzZ86UESNGyI4dO/RDXdTOIypoq2UigQjWFy5c0FenVTvXr1/Xa7XVcpEmTZoQrJPapKa/CCBwVwGCNRMEAQQQCCIBdWU4ZcqUeq11zDFjxgx99fqvv/7Sr3EggAACCASnAME6OMeFs0IAgSQqoJZbqKvDn376qV7OobboU1eIa9WqJZ988kkSVaHbCCCAQGgIEKxDY5w4SwQQSCICN27c0DcxqhB9+PBhyZUrl36Sov/SjiRCQTcRQACBkBP4f83MJHJcFFe5AAAAAElFTkSuQmCC\" width=\"659.9999856948856\">"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2f2de78f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "9669ae75",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\filtered_full_probes_sublibrary_500.fasta\n",
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\filtered_full_probes_sublibrary_250_1.fasta\n",
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\filtered_full_probes_sublibrary_250_2.fasta\n",
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\filtered_full_probes_sublibrary_500_short.fasta\n",
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\filtered_full_probes_sublibrary_250_1_short.fasta\n",
      "\\\\10.245.74.212\\Chromatin_NAS_2\\Chromatin_Libraries\\CTP-11_brain\\mouse_genome_1000\\filtered_full_probes_sublibrary_250_2_short.fasta\n"
     ]
    }
   ],
   "source": [
    "# MERGE these 6 libraries into one file:\n",
    "\n",
    "final_probe_files = [os.path.join(library_folder, _fl) for _fl in os.listdir(library_folder)\n",
    "                     if 'filtered_full_probes_sublibrary' in _fl]\n",
    "print('\\n'.join(final_probe_files))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "adaa4f39",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "173891\n"
     ]
    }
   ],
   "source": [
    "pool_records = []\n",
    "\n",
    "for _fl in final_probe_files:\n",
    "    with open(_fl, 'r') as handle:\n",
    "        for record in SeqIO.parse(handle, \"fasta\"):\n",
    "            pool_records.append(record)\n",
    "            \n",
    "print(len(pool_records))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3646516c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "afd08a18",
   "metadata": {},
   "outputs": [],
   "source": [
    "# primer statistics\n",
    "primer_set_count = {}\n",
    "\n",
    "for _record in pool_records:\n",
    "    _primer_info = _record.id.split('primers_[')[1].split(']')[0]\n",
    "    if _primer_info not in primer_set_count:\n",
    "        primer_set_count[_primer_info] = 1\n",
    "    else:\n",
    "        primer_set_count[_primer_info] += 1\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "dd8650e0",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'W1A03_primer_2,W1A10_primer_9': 57936,\n",
       " 'W1A07_primer_6,W1A12_primer_11': 29037,\n",
       " 'W1B05_primer_16,W1B04_primer_15': 27746,\n",
       " 'W1A11_primer_10,W1B06_primer_17': 29845,\n",
       " 'W1A05_primer_4,W1A04_primer_3': 14988,\n",
       " 'W1A01_primer_0,W1A02_primer_1': 14339}"
      ]
     },
     "execution_count": 163,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "primer_set_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "70cf9bf3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "173891\n"
     ]
    }
   ],
   "source": [
    "# save kept records\n",
    "with open(os.path.join(pool_folder, 'final_pool_probes.fasta'), 'w') as output_handle:\n",
    "    print(len(pool_records))\n",
    "    SeqIO.write(pool_records, output_handle, \"fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "11e7b26e",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Generate primer usage information\n",
    "# load full probes\n",
    "final_pool_filename = os.path.join(pool_folder, 'final_pool_probes.fasta')\n",
    "pool_records = []\n",
    "with open(os.path.join(library_folder, final_pool_filename), 'r') as handle:\n",
    "    for record in SeqIO.parse(handle, \"fasta\"):\n",
    "        pool_records.append(record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "bc134515",
   "metadata": {},
   "outputs": [],
   "source": [
    "sublibrary_probe_dict = {}\n",
    "for _record in pool_records:\n",
    "    _lib = _record.name.split('library_')[-1]\n",
    "    if _lib not in sublibrary_probe_dict:\n",
    "        sublibrary_probe_dict[_lib] = [_record]\n",
    "    else:\n",
    "        sublibrary_probe_dict[_lib].append(_record)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "a80565d4",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1000-mouse-genome_500 57936\n",
      "1000-mouse-genome_250_1 29037\n",
      "1000-mouse-genome_250_2 27746\n",
      "1000-mouse-genome-short_500 29845\n",
      "1000-mouse-genome-short_250_1 14988\n",
      "1000-mouse-genome-short_250_2 14339\n"
     ]
    }
   ],
   "source": [
    "for _lib, _pbs in sublibrary_probe_dict.items():\n",
    "    print(_lib, len(_pbs))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c4995d57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'loc_1:3740000-3760000_gene_1_pb_23_pos_2742_strand_-_readouts_[NDB_856_c,NDB_946_c,NDB_935_c]_primers_[W1A03_primer_2,W1A10_primer_9]_library_1000-mouse-genome_500_library_1000-mouse-genome_500'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pool_records[0].id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "49295994",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.00143\n",
      "4.924751179529566e-08\n",
      "1.23539741156229e-12\n",
      "final in ul: 123.53974115622901\n"
     ]
    }
   ],
   "source": [
    "# 1000-mouse-genome_250_1\n",
    "vol = 11 * 1e3 # ul\n",
    "conc = 130 * 1e-9 # g/ul\n",
    "num_pb = 29037\n",
    "pb_len = 142-11\n",
    "final_conc = 10 # nM\n",
    "pb_mw = (pb_len * 303.7 + 79) # g/mol\n",
    "\n",
    "mass_total = vol * conc\n",
    "mass_per_pb = mass_total / num_pb\n",
    "mol_per_pb = mass_per_pb / pb_mw\n",
    "print(mass_total)\n",
    "print(mass_per_pb)\n",
    "print(mol_per_pb)\n",
    "target_conc = final_conc * 1e-9 * 1e-6 # mol/ul\n",
    "final_vol = mol_per_pb / target_conc\n",
    "print(\"final in ul:\", final_vol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "5c7deb9a",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0013750000000000001\n",
      "4.955669285662799e-08\n",
      "1.2431533665120898e-12\n",
      "final in ul: 124.31533665120898\n"
     ]
    }
   ],
   "source": [
    "# 1000-mouse-genome_250_2\n",
    "vol = 11 * 1e3 # ul\n",
    "conc = 125 * 1e-9 # g/ul\n",
    "num_pb = 27746\n",
    "pb_len = 142-11\n",
    "final_conc = 10 # nM\n",
    "pb_mw = (pb_len * 303.7 + 79) # g/mol\n",
    "\n",
    "mass_total = vol * conc\n",
    "mass_per_pb = mass_total / num_pb\n",
    "mol_per_pb = mass_per_pb / pb_mw\n",
    "print(mass_total)\n",
    "print(mass_per_pb)\n",
    "print(mol_per_pb)\n",
    "target_conc = final_conc * 1e-9 * 1e-6 # mol/ul\n",
    "final_vol = mol_per_pb / target_conc\n",
    "print(\"final in ul:\", final_vol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "192fe5bd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.001276\n",
      "2.2024302678818007e-08\n",
      "5.524901772494276e-13\n",
      "final in ul: 55.24901772494276\n"
     ]
    }
   ],
   "source": [
    "# 1000-mouse-genome_500_1\n",
    "vol = 11 * 1e3 # ul\n",
    "conc = 116 * 1e-9 # g/ul\n",
    "num_pb = 57936\n",
    "pb_len = 142-11\n",
    "final_conc = 10 # nM\n",
    "pb_mw = (pb_len * 303.7 + 79) # g/mol\n",
    "\n",
    "mass_total = vol * conc\n",
    "mass_per_pb = mass_total / num_pb\n",
    "mol_per_pb = mass_per_pb / pb_mw\n",
    "print(mass_total)\n",
    "print(mass_per_pb)\n",
    "print(mol_per_pb)\n",
    "target_conc = final_conc * 1e-9 * 1e-6 # mol/ul\n",
    "final_vol = mol_per_pb / target_conc\n",
    "print(\"final in ul:\", final_vol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "8388c033",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.00143\n",
      "2.4682408174537422e-08\n",
      "6.191700262278068e-13\n",
      "final in ul: 61.917002622780686\n"
     ]
    }
   ],
   "source": [
    "# 1000-mouse-genome_500_1\n",
    "vol = 11 * 1e3 # ul\n",
    "conc = 130 * 1e-9 # g/ul\n",
    "num_pb = 57936\n",
    "pb_len = 142-11\n",
    "final_conc = 10 # nM\n",
    "pb_mw = (pb_len * 303.7 + 79) # g/mol\n",
    "\n",
    "mass_total = vol * conc\n",
    "mass_per_pb = mass_total / num_pb\n",
    "mol_per_pb = mass_per_pb / pb_mw\n",
    "print(mass_total)\n",
    "print(mass_per_pb)\n",
    "print(mol_per_pb)\n",
    "target_conc = final_conc * 1e-9 * 1e-6 # mol/ul\n",
    "final_vol = mol_per_pb / target_conc\n",
    "print(\"final in ul:\", final_vol)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "08099069",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
