{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Import module"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The link to get [ImageAnalysis3](https://github.com/zhengpuas47/ImageAnalysis3) \n",
    "\n",
    "Or from the Zhuang lab archived [source_tools](https://github.com/ZhuangLab/Chromatin_Analysis_2020_cell/tree/master/sequential_tracing/source)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ImageAnalysis3 and basic modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12912\n"
     ]
    }
   ],
   "source": [
    "%run \"C:\\Users\\shiwei\\Documents\\ImageAnalysis3\\required_files\\Startup_py3.py\"\n",
    "sys.path.append(r\"C:\\Users\\shiwei\\Documents\")\n",
    "\n",
    "import ImageAnalysis3 as ia\n",
    "from ImageAnalysis3 import *\n",
    "from ImageAnalysis3.classes import _allowed_kwds\n",
    "\n",
    "import h5py\n",
    "import ast\n",
    "import pandas as pd\n",
    "\n",
    "print(os.getpid())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Chromatin_analysis_tools etc"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See **functions** in the repository for [AnalysisTool_Chromatin](../../README.md)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<module 'atac_to_loci' from 'C:\\\\Users\\\\shiwei\\\\Documents\\\\AnalysisTool_Chromatin\\\\atac_to_loci.py'>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Chromatin_analysis_tools (ATC)\n",
    "# Get path for the py containing functions\n",
    "import os\n",
    "import sys\n",
    "import importlib\n",
    "module_path =r'C:\\Users\\shiwei\\Documents\\AnalysisTool_Chromatin'\n",
    "if module_path not in sys.path:\n",
    "    sys.path.append(module_path)\n",
    "    \n",
    "# import relevant modules\n",
    "import gene_selection \n",
    "importlib.reload(gene_selection)\n",
    "import gene_to_loci\n",
    "importlib.reload(gene_to_loci)\n",
    "import gene_activity\n",
    "importlib.reload(gene_activity)\n",
    "import loci_1d_features\n",
    "importlib.reload(loci_1d_features)  \n",
    "\n",
    "import atac_to_loci\n",
    "importlib.reload(atac_to_loci)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Define folders"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Use existing analysis folder: L:\\Shiwei\\postanalysis_2024\\v0\\locus_annotation\\analysis.\n",
      "Use existing figure folder: L:\\Shiwei\\postanalysis_2024\\v0\\locus_annotation\\figures.\n"
     ]
    }
   ],
   "source": [
    "# main folder for postanalysis\n",
    "postanalysis_folder = r'L:\\Shiwei\\postanalysis_2024\\v0'\n",
    "# input files for postanalysis\n",
    "input_folder = os.path.join(postanalysis_folder, 'resources_from_preprocess')\n",
    "\n",
    "# output file to be generated\n",
    "output_main_folder = os.path.join(postanalysis_folder, 'locus_annotation')\n",
    "\n",
    "output_analysis_folder = os.path.join(output_main_folder, 'analysis')\n",
    "output_figure_folder = os.path.join(output_main_folder, 'figures')\n",
    "\n",
    "# make new folder if needed\n",
    "make_output_folder = True\n",
    "\n",
    "if make_output_folder and not os.path.exists(output_analysis_folder):\n",
    "    os.makedirs(output_analysis_folder)\n",
    "    print(f'Generating analysis folder: {output_analysis_folder}.')\n",
    "elif os.path.exists(output_analysis_folder):\n",
    "    print(f'Use existing analysis folder: {output_analysis_folder}.')\n",
    "    \n",
    "if make_output_folder and not os.path.exists(output_figure_folder):\n",
    "    os.makedirs(output_figure_folder)\n",
    "    print(f'Generating figure folder: {output_figure_folder}.')\n",
    "elif os.path.exists(output_figure_folder):\n",
    "    print(f'Use existing figure folder: {output_figure_folder}.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Plotting parameters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline\n",
    "import matplotlib\n",
    "matplotlib.rcParams['pdf.fonttype'] = 42\n",
    "import matplotlib.pyplot as plt\n",
    "plt.rc('font', family='serif')\n",
    "plt.rc('font', serif='Arial')\n",
    "\n",
    "from ImageAnalysis3.figure_tools import _double_col_width, _single_col_width, _font_size, _ticklabel_size,_ticklabel_width\n",
    "\n",
    "import seaborn as sns\n",
    "sns.set_context(\"paper\", rc={\"font.size\":_font_size,\"axes.titlesize\":_font_size+1,\"axes.labelsize\":_font_size})  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load data relevant information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## load codebook"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "[merged codebook](../resources/merged_codebook.csv) as in the repository (merged for all DNA-MERFISH libraries)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>NDB_784</th>\n",
       "      <th>NDB_755</th>\n",
       "      <th>NDB_826</th>\n",
       "      <th>NDB_713</th>\n",
       "      <th>NDB_865</th>\n",
       "      <th>NDB_725</th>\n",
       "      <th>NDB_817</th>\n",
       "      <th>NDB_710</th>\n",
       "      <th>...</th>\n",
       "      <th>NDB_479</th>\n",
       "      <th>NDB_562</th>\n",
       "      <th>NDB_608</th>\n",
       "      <th>NDB_460</th>\n",
       "      <th>NDB_563</th>\n",
       "      <th>NDB_592</th>\n",
       "      <th>NDB_368</th>\n",
       "      <th>NDB_436</th>\n",
       "      <th>NDB_629</th>\n",
       "      <th>NDB_604</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1016</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1017</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 228 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   name id  NDB_784  NDB_755  NDB_826  NDB_713  NDB_865  \\\n",
       "0     1:3742742-3759944  1      0.0      0.0      0.0      0.0      0.0   \n",
       "1     1:6245958-6258969  2      0.0      0.0      0.0      0.0      0.0   \n",
       "2     1:8740008-8759916  3      0.0      0.0      0.0      0.0      0.0   \n",
       "1016  1:9627926-9637875  1      0.0      0.0      0.0      0.0      0.0   \n",
       "1017  1:9799472-9811359  2      0.0      0.0      0.0      0.0      0.0   \n",
       "\n",
       "      NDB_725  NDB_817  NDB_710  ...  NDB_479  NDB_562  NDB_608  NDB_460  \\\n",
       "0         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "2         0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1016      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "1017      0.0      0.0      0.0  ...      0.0      0.0      0.0      0.0   \n",
       "\n",
       "      NDB_563  NDB_592  NDB_368  NDB_436  NDB_629  NDB_604  \n",
       "0         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "2         0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1016      0.0      0.0      0.0      0.0      0.0      0.0  \n",
       "1017      0.0      1.0      0.0      0.0      0.0      0.0  \n",
       "\n",
       "[5 rows x 228 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load codebook \n",
    "codebook_fname = os.path.join(input_folder,'merged_codebook.csv')\n",
    "codebook_df = pd.read_csv (codebook_fname, index_col=0)\n",
    "\n",
    "# sort df by chr and chr_order\n",
    "codebook_df = loci_1d_features.sort_loci_df_by_chr_order (codebook_df)\n",
    "codebook_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## convert to new dataframe to store the transcirption annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\shiwei\\AppData\\Local\\Temp\\3\\ipykernel_12912\\58657782.py:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  loci_df['loci_name'] = list(loci_name_arr[:,0])\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_order</th>\n",
       "      <th>library</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_3742742_3759944</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_6245958_6258969</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_8740008_8759916</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTP11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9627926_9637875</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CTP13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CTP13</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   name id chr  chr_order library\n",
       "loci_name                                                        \n",
       "chr1_3742742_3759944  1:3742742-3759944  1   1        0.0   CTP11\n",
       "chr1_6245958_6258969  1:6245958-6258969  2   1        1.0   CTP11\n",
       "chr1_8740008_8759916  1:8740008-8759916  3   1        2.0   CTP11\n",
       "chr1_9627926_9637875  1:9627926-9637875  1   1        3.0   CTP13\n",
       "chr1_9799472_9811359  1:9799472-9811359  2   1        4.0   CTP13"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Format the chr loci name \n",
    "from gene_to_loci import loci_pos_format\n",
    "loci_name_list = list(map(loci_pos_format, codebook_df['name'].tolist()))\n",
    "loci_name_arr = np.array(loci_name_list)\n",
    "\n",
    "# convert to a new dataframe and set loci name as index\n",
    "loci_df = codebook_df[['name','id','chr','chr_order','library']]\n",
    "loci_df['loci_name'] = list(loci_name_arr[:,0])\n",
    "loci_df = loci_df.set_index ('loci_name')\n",
    "\n",
    "loci_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## load all gene annotation from Ensembl df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "annotation can be generated using the notebook below:\n",
    "[external/scripts/genome_annotation/1_genomic_info_for_mop_genes_using_ensembl_transcriptome](../../../external/scripts/genome_annotation/1_genomic_info_for_mop_genes_using_ensembl_transcriptome.ipynb)\n",
    "\n",
    "\n",
    "Annotation link in this repository:\n",
    "\n",
    "[MOp_annotation_smart_seq](../../../external/resources/MOp_smart_sn_gene_chr_info_NEW_from_transcriptome_FORMAT.csv)\n",
    "\n",
    "[MOp_annotation_10x](../../../external/resources/MOp_10x_snRNA_chr_info_NEW_from_transcriptome_FORMAT.csv)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>gene_biotype</th>\n",
       "      <th>coding_strand</th>\n",
       "      <th>length</th>\n",
       "      <th>genomic_position</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0610005C13Rik</th>\n",
       "      <td>7</td>\n",
       "      <td>45567794</td>\n",
       "      <td>45575327</td>\n",
       "      <td>antisense</td>\n",
       "      <td>-1</td>\n",
       "      <td>7533</td>\n",
       "      <td>chr7_45567794_45575327</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610009B22Rik</th>\n",
       "      <td>11</td>\n",
       "      <td>51685386</td>\n",
       "      <td>51688874</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-1</td>\n",
       "      <td>3488</td>\n",
       "      <td>chr11_51685386_51688874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610009E02Rik</th>\n",
       "      <td>2</td>\n",
       "      <td>26445696</td>\n",
       "      <td>26459390</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>1</td>\n",
       "      <td>13694</td>\n",
       "      <td>chr2_26445696_26459390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610009L18Rik</th>\n",
       "      <td>11</td>\n",
       "      <td>120348678</td>\n",
       "      <td>120351190</td>\n",
       "      <td>bidirectional_promoter_lncRNA</td>\n",
       "      <td>1</td>\n",
       "      <td>2512</td>\n",
       "      <td>chr11_120348678_120351190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0610010F05Rik</th>\n",
       "      <td>11</td>\n",
       "      <td>23564961</td>\n",
       "      <td>23633639</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-1</td>\n",
       "      <td>68678</td>\n",
       "      <td>chr11_23564961_23633639</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              chr      start        end                   gene_biotype  \\\n",
       "gene                                                                     \n",
       "0610005C13Rik   7   45567794   45575327                      antisense   \n",
       "0610009B22Rik  11   51685386   51688874                 protein_coding   \n",
       "0610009E02Rik   2   26445696   26459390           processed_transcript   \n",
       "0610009L18Rik  11  120348678  120351190  bidirectional_promoter_lncRNA   \n",
       "0610010F05Rik  11   23564961   23633639                 protein_coding   \n",
       "\n",
       "               coding_strand  length           genomic_position  \n",
       "gene                                                             \n",
       "0610005C13Rik             -1    7533     chr7_45567794_45575327  \n",
       "0610009B22Rik             -1    3488    chr11_51685386_51688874  \n",
       "0610009E02Rik              1   13694     chr2_26445696_26459390  \n",
       "0610009L18Rik              1    2512  chr11_120348678_120351190  \n",
       "0610010F05Rik             -1   68678    chr11_23564961_23633639  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# load gene annotation (covering all genes from the SMART-seq) for chr locus\n",
    "# this should be enough since we can only analyze the genes with info from SMARTseq\n",
    "# also we used Ensembl for both gene2loci and loci2gene to be consistent \n",
    "external_folder = os.path.join(postanalysis_folder, 'external_resources')\n",
    "gene_annotation_df = pd.read_csv(os.path.join(external_folder, \n",
    "                                              \"MOp_smart_sn_gene_chr_info_NEW_from_transcriptome_FORMAT.csv\"),\n",
    "                                 index_col=0)\n",
    "\n",
    "gene_annotation_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## function to add adjacent genes by defined distance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Function to find genes near a list of loci provided in DataFrame \n",
    "from gene_to_loci import find_genes_near_loci\n",
    "\n",
    "def find_genes_near_loci_for_loci_dataframe (loci_df:pd.core.frame.DataFrame, \n",
    "                                             gene_annotation_df:pd.core.frame.DataFrame, \n",
    "                                             extend_dist = 50*1000,\n",
    "                                             gene_coverage_type = 'tss',\n",
    "                                             key_added = None):\n",
    "\n",
    "\n",
    "    \"\"\"Find genes near loci for a list of loci provided in DataFrame\"\"\"\n",
    "    sel_genes_list = []\n",
    "    for loci_name in loci_df.index.tolist():\n",
    "        sel_genes =  find_genes_near_loci (loci_name, \n",
    "                                           gene_annotation_df,\n",
    "                                           extend_dist = extend_dist, \n",
    "                                           gene_coverage_type = gene_coverage_type\n",
    "                                          )\n",
    "        sel_genes_list.append(sel_genes)\n",
    "    \n",
    "    if isinstance (key_added, type(None)):\n",
    "        extend_dist_name = str(int(extend_dist/1000)) + 'kb'\n",
    "        key_added = f'adjacent_genes_{extend_dist_name}_{gene_coverage_type}'\n",
    "    \n",
    "    loci_df[key_added]=sel_genes_list\n",
    "    \n",
    "    return loci_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## add adjacent genes for different distances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>id</th>\n",
       "      <th>chr</th>\n",
       "      <th>chr_order</th>\n",
       "      <th>library</th>\n",
       "      <th>adjacent_genes_10kb_tss</th>\n",
       "      <th>adjacent_genes_50kb_tss</th>\n",
       "      <th>adjacent_genes_100kb_tss</th>\n",
       "      <th>adjacent_genes_200kb_tss</th>\n",
       "      <th>adjacent_genes_500kb_tss</th>\n",
       "      <th>adjacent_genes_1000kb_tss</th>\n",
       "      <th>adjacent_genes_1500kb_tss</th>\n",
       "      <th>adjacent_genes_2000kb_tss</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>loci_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>chr1_3742742_3759944</th>\n",
       "      <td>1:3742742-3759944</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>Gm18956; Gm7341; Xkr4</td>\n",
       "      <td>Gm18956; Gm2053; Gm6085; Gm6101; Gm6119; Gm734...</td>\n",
       "      <td>Atp6v1h; Gm16041; Gm17101; Gm18956; Gm2053; Gm...</td>\n",
       "      <td>Atp6v1h; Gm16041; Gm17101; Gm18956; Gm2053; Gm...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_6245958_6258969</th>\n",
       "      <td>1:6245958-6258969</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>4732440D04Rik; Rb1cc1</td>\n",
       "      <td>4732440D04Rik; Rb1cc1</td>\n",
       "      <td>4732440D04Rik; Gm19026; Gm2147; Rb1cc1</td>\n",
       "      <td>4732440D04Rik; Gm19026; Gm2147; Npbwr1; Rb1cc1...</td>\n",
       "      <td>4732440D04Rik; Gm19026; Gm2147; Gm5694; Gm7182...</td>\n",
       "      <td>4732440D04Rik; Atp6v1h; Gm16041; Gm17101; Gm19...</td>\n",
       "      <td>4732440D04Rik; Atp6v1h; Gm16041; Gm17101; Gm19...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_8740008_8759916</th>\n",
       "      <td>1:8740008-8759916</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>Gm15452</td>\n",
       "      <td>Gm15452</td>\n",
       "      <td>Gm15452</td>\n",
       "      <td>Gm15452</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Gm15452;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9627926_9637875</th>\n",
       "      <td>1:9627926-9637875</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>CTP13</td>\n",
       "      <td>2610203C22Rik</td>\n",
       "      <td>2610203C22Rik</td>\n",
       "      <td>2610203C22Rik; Adhfe1; Gm6161; Mybl1; Rrs1</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Gm18300;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Cops5; C...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; A830018L16Rik; A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chr1_9799472_9811359</th>\n",
       "      <td>1:9799472-9811359</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>CTP13</td>\n",
       "      <td>Gm6195; Sgk3</td>\n",
       "      <td>Gm6195; Sgk3</td>\n",
       "      <td>1700034P13Rik; Gm6195; Mcmdc2; Mybl1; Sgk3; Vc...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Gm6195; Mcmdc2; ...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...</td>\n",
       "      <td>1700034P13Rik; 2610203C22Rik; A830018L16Rik; A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_166247682_166259932</th>\n",
       "      <td>X:166247682-166259932</td>\n",
       "      <td>1059</td>\n",
       "      <td>X</td>\n",
       "      <td>60.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>Gpm6b</td>\n",
       "      <td>Gpm6b</td>\n",
       "      <td>Gemin8; Gpm6b</td>\n",
       "      <td>Gemin8; Gm15225; Gpm6b; Ofd1; Trappc2</td>\n",
       "      <td>Egfl6; Gemin8; Gm15223; Gm15225; Gm1720; Gpm6b...</td>\n",
       "      <td>Egfl6; Gemin8; Glra2; Gm15223; Gm15225; Gm1523...</td>\n",
       "      <td>Egfl6; Fancb; Gemin8; Glra2; Gm15223; Gm15225;...</td>\n",
       "      <td>Asb11; Asb9; Bmx; Egfl6; Fancb; Gemin8; Glra2;...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_167157164_167167452</th>\n",
       "      <td>X:167157164-167167452</td>\n",
       "      <td>990</td>\n",
       "      <td>X</td>\n",
       "      <td>61.0</td>\n",
       "      <td>CTP13</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>Tmsb4x</td>\n",
       "      <td>Gm15232; Tmsb4x</td>\n",
       "      <td>Gm15232; Gm8814; Tlr7; Tmsb4x</td>\n",
       "      <td>Gm15230; Gm15232; Gm1720; Gm8814; Prps2; Tlr7;...</td>\n",
       "      <td>Egfl6; Gemin8; Gm15223; Gm15230; Gm15232; Gm15...</td>\n",
       "      <td>Egfl6; Frmpd4; Gemin8; Gm15223; Gm15225; Gm152...</td>\n",
       "      <td>Arhgap6; Egfl6; Frmpd4; Gemin8; Glra2; Gm15223...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_168746045_168757590</th>\n",
       "      <td>X:168746045-168757590</td>\n",
       "      <td>1060</td>\n",
       "      <td>X</td>\n",
       "      <td>62.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>intergenic</td>\n",
       "      <td>Arhgap6</td>\n",
       "      <td>Arhgap6; Msl3</td>\n",
       "      <td>Arhgap6; Frmpd4; Msl3</td>\n",
       "      <td>Amelx; Arhgap6; Frmpd4; Msl3</td>\n",
       "      <td>Amelx; Arhgap6; Frmpd4; Gm15238; Gm15240; Hccs...</td>\n",
       "      <td>Amelx; Arhgap6; Erdr1; Frmpd4; G530011O06Rik; ...</td>\n",
       "      <td>Amelx; Arhgap6; Asmt; Erdr1; Frmpd4; G530011O0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrX_169963295_170005197</th>\n",
       "      <td>X:169963295-170005197</td>\n",
       "      <td>991</td>\n",
       "      <td>X</td>\n",
       "      <td>63.0</td>\n",
       "      <td>CTP13</td>\n",
       "      <td>Erdr1; G530011O06Rik</td>\n",
       "      <td>Erdr1; G530011O06Rik</td>\n",
       "      <td>Erdr1; G530011O06Rik</td>\n",
       "      <td>Erdr1; G530011O06Rik</td>\n",
       "      <td>Erdr1; G530011O06Rik; Mid1</td>\n",
       "      <td>Amelx; Asmt; Erdr1; G530011O06Rik; Hccs; Mid1</td>\n",
       "      <td>Amelx; Arhgap6; Asmt; Erdr1; Frmpd4; G530011O0...</td>\n",
       "      <td>Amelx; Arhgap6; Asmt; Erdr1; Frmpd4; G530011O0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chrY_1245775_1259992</th>\n",
       "      <td>Y:1245775-1259992</td>\n",
       "      <td>1062</td>\n",
       "      <td>Y</td>\n",
       "      <td>0.0</td>\n",
       "      <td>CTP11</td>\n",
       "      <td>Gm4017; Uty</td>\n",
       "      <td>Ddx3y; Gm18665; Gm4017; Uty</td>\n",
       "      <td>Ddx3y; Gm18665; Gm4017; Uty</td>\n",
       "      <td>Ddx3y; Gm18665; Gm4017; Usp9y; Uty</td>\n",
       "      <td>Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm4017; Kdm5...</td>\n",
       "      <td>Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm18797; Gm2...</td>\n",
       "      <td>Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm18796; Gm1...</td>\n",
       "      <td>Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm18796; Gm1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1982 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           name    id chr  chr_order library  \\\n",
       "loci_name                                                                      \n",
       "chr1_3742742_3759944          1:3742742-3759944     1   1        0.0   CTP11   \n",
       "chr1_6245958_6258969          1:6245958-6258969     2   1        1.0   CTP11   \n",
       "chr1_8740008_8759916          1:8740008-8759916     3   1        2.0   CTP11   \n",
       "chr1_9627926_9637875          1:9627926-9637875     1   1        3.0   CTP13   \n",
       "chr1_9799472_9811359          1:9799472-9811359     2   1        4.0   CTP13   \n",
       "...                                         ...   ...  ..        ...     ...   \n",
       "chrX_166247682_166259932  X:166247682-166259932  1059   X       60.0   CTP11   \n",
       "chrX_167157164_167167452  X:167157164-167167452   990   X       61.0   CTP13   \n",
       "chrX_168746045_168757590  X:168746045-168757590  1060   X       62.0   CTP11   \n",
       "chrX_169963295_170005197  X:169963295-170005197   991   X       63.0   CTP13   \n",
       "chrY_1245775_1259992          Y:1245775-1259992  1062   Y        0.0   CTP11   \n",
       "\n",
       "                         adjacent_genes_10kb_tss      adjacent_genes_50kb_tss  \\\n",
       "loci_name                                                                       \n",
       "chr1_3742742_3759944                  intergenic                   intergenic   \n",
       "chr1_6245958_6258969                  intergenic        4732440D04Rik; Rb1cc1   \n",
       "chr1_8740008_8759916                  intergenic                      Gm15452   \n",
       "chr1_9627926_9637875               2610203C22Rik                2610203C22Rik   \n",
       "chr1_9799472_9811359                Gm6195; Sgk3                 Gm6195; Sgk3   \n",
       "...                                          ...                          ...   \n",
       "chrX_166247682_166259932                   Gpm6b                        Gpm6b   \n",
       "chrX_167157164_167167452              intergenic                       Tmsb4x   \n",
       "chrX_168746045_168757590              intergenic                      Arhgap6   \n",
       "chrX_169963295_170005197    Erdr1; G530011O06Rik         Erdr1; G530011O06Rik   \n",
       "chrY_1245775_1259992                 Gm4017; Uty  Ddx3y; Gm18665; Gm4017; Uty   \n",
       "\n",
       "                                                   adjacent_genes_100kb_tss  \\\n",
       "loci_name                                                                     \n",
       "chr1_3742742_3759944                                                   Xkr4   \n",
       "chr1_6245958_6258969                                  4732440D04Rik; Rb1cc1   \n",
       "chr1_8740008_8759916                                                Gm15452   \n",
       "chr1_9627926_9637875             2610203C22Rik; Adhfe1; Gm6161; Mybl1; Rrs1   \n",
       "chr1_9799472_9811359      1700034P13Rik; Gm6195; Mcmdc2; Mybl1; Sgk3; Vc...   \n",
       "...                                                                     ...   \n",
       "chrX_166247682_166259932                                      Gemin8; Gpm6b   \n",
       "chrX_167157164_167167452                                    Gm15232; Tmsb4x   \n",
       "chrX_168746045_168757590                                      Arhgap6; Msl3   \n",
       "chrX_169963295_170005197                               Erdr1; G530011O06Rik   \n",
       "chrY_1245775_1259992                            Ddx3y; Gm18665; Gm4017; Uty   \n",
       "\n",
       "                                                   adjacent_genes_200kb_tss  \\\n",
       "loci_name                                                                     \n",
       "chr1_3742742_3759944                                                   Xkr4   \n",
       "chr1_6245958_6258969                 4732440D04Rik; Gm19026; Gm2147; Rb1cc1   \n",
       "chr1_8740008_8759916                                                Gm15452   \n",
       "chr1_9627926_9637875      1700034P13Rik; 2610203C22Rik; Adhfe1; Gm18300;...   \n",
       "chr1_9799472_9811359      1700034P13Rik; 2610203C22Rik; Gm6195; Mcmdc2; ...   \n",
       "...                                                                     ...   \n",
       "chrX_166247682_166259932              Gemin8; Gm15225; Gpm6b; Ofd1; Trappc2   \n",
       "chrX_167157164_167167452                      Gm15232; Gm8814; Tlr7; Tmsb4x   \n",
       "chrX_168746045_168757590                              Arhgap6; Frmpd4; Msl3   \n",
       "chrX_169963295_170005197                               Erdr1; G530011O06Rik   \n",
       "chrY_1245775_1259992                     Ddx3y; Gm18665; Gm4017; Usp9y; Uty   \n",
       "\n",
       "                                                   adjacent_genes_500kb_tss  \\\n",
       "loci_name                                                                     \n",
       "chr1_3742742_3759944                                  Gm18956; Gm7341; Xkr4   \n",
       "chr1_6245958_6258969      4732440D04Rik; Gm19026; Gm2147; Npbwr1; Rb1cc1...   \n",
       "chr1_8740008_8759916                                                Gm15452   \n",
       "chr1_9627926_9637875      1700034P13Rik; 2610203C22Rik; Adhfe1; Cops5; C...   \n",
       "chr1_9799472_9811359      1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...   \n",
       "...                                                                     ...   \n",
       "chrX_166247682_166259932  Egfl6; Gemin8; Gm15223; Gm15225; Gm1720; Gpm6b...   \n",
       "chrX_167157164_167167452  Gm15230; Gm15232; Gm1720; Gm8814; Prps2; Tlr7;...   \n",
       "chrX_168746045_168757590                       Amelx; Arhgap6; Frmpd4; Msl3   \n",
       "chrX_169963295_170005197                         Erdr1; G530011O06Rik; Mid1   \n",
       "chrY_1245775_1259992      Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm4017; Kdm5...   \n",
       "\n",
       "                                                  adjacent_genes_1000kb_tss  \\\n",
       "loci_name                                                                     \n",
       "chr1_3742742_3759944      Gm18956; Gm2053; Gm6085; Gm6101; Gm6119; Gm734...   \n",
       "chr1_6245958_6258969      4732440D04Rik; Gm19026; Gm2147; Gm5694; Gm7182...   \n",
       "chr1_8740008_8759916      1700034P13Rik; 2610203C22Rik; Adhfe1; Gm15452;...   \n",
       "chr1_9627926_9637875      1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...   \n",
       "chr1_9799472_9811359      1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...   \n",
       "...                                                                     ...   \n",
       "chrX_166247682_166259932  Egfl6; Gemin8; Glra2; Gm15223; Gm15225; Gm1523...   \n",
       "chrX_167157164_167167452  Egfl6; Gemin8; Gm15223; Gm15230; Gm15232; Gm15...   \n",
       "chrX_168746045_168757590  Amelx; Arhgap6; Frmpd4; Gm15238; Gm15240; Hccs...   \n",
       "chrX_169963295_170005197      Amelx; Asmt; Erdr1; G530011O06Rik; Hccs; Mid1   \n",
       "chrY_1245775_1259992      Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm18797; Gm2...   \n",
       "\n",
       "                                                  adjacent_genes_1500kb_tss  \\\n",
       "loci_name                                                                     \n",
       "chr1_3742742_3759944      Atp6v1h; Gm16041; Gm17101; Gm18956; Gm2053; Gm...   \n",
       "chr1_6245958_6258969      4732440D04Rik; Atp6v1h; Gm16041; Gm17101; Gm19...   \n",
       "chr1_8740008_8759916      1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...   \n",
       "chr1_9627926_9637875      1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...   \n",
       "chr1_9799472_9811359      1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...   \n",
       "...                                                                     ...   \n",
       "chrX_166247682_166259932  Egfl6; Fancb; Gemin8; Glra2; Gm15223; Gm15225;...   \n",
       "chrX_167157164_167167452  Egfl6; Frmpd4; Gemin8; Gm15223; Gm15225; Gm152...   \n",
       "chrX_168746045_168757590  Amelx; Arhgap6; Erdr1; Frmpd4; G530011O06Rik; ...   \n",
       "chrX_169963295_170005197  Amelx; Arhgap6; Asmt; Erdr1; Frmpd4; G530011O0...   \n",
       "chrY_1245775_1259992      Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm18796; Gm1...   \n",
       "\n",
       "                                                  adjacent_genes_2000kb_tss  \n",
       "loci_name                                                                    \n",
       "chr1_3742742_3759944      Atp6v1h; Gm16041; Gm17101; Gm18956; Gm2053; Gm...  \n",
       "chr1_6245958_6258969      4732440D04Rik; Atp6v1h; Gm16041; Gm17101; Gm19...  \n",
       "chr1_8740008_8759916      1700034P13Rik; 2610203C22Rik; Adhfe1; Arfgef1;...  \n",
       "chr1_9627926_9637875      1700034P13Rik; 2610203C22Rik; A830018L16Rik; A...  \n",
       "chr1_9799472_9811359      1700034P13Rik; 2610203C22Rik; A830018L16Rik; A...  \n",
       "...                                                                     ...  \n",
       "chrX_166247682_166259932  Asb11; Asb9; Bmx; Egfl6; Fancb; Gemin8; Glra2;...  \n",
       "chrX_167157164_167167452  Arhgap6; Egfl6; Frmpd4; Gemin8; Glra2; Gm15223...  \n",
       "chrX_168746045_168757590  Amelx; Arhgap6; Asmt; Erdr1; Frmpd4; G530011O0...  \n",
       "chrX_169963295_170005197  Amelx; Arhgap6; Asmt; Erdr1; Frmpd4; G530011O0...  \n",
       "chrY_1245775_1259992      Ddx3y; Eif2s3y; Gm17790; Gm18665; Gm18796; Gm1...  \n",
       "\n",
       "[1982 rows x 13 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loci_df_copy = loci_df.copy(deep=True)\n",
    "\n",
    "for _dist in [10*1000,50*1000,100*1000,200*1000,500*1000,1000*1000,1500*1000,2000*1000,]:\n",
    "\n",
    "    loci_df_copy = find_genes_near_loci_for_loci_dataframe (loci_df_copy, \n",
    "                                            gene_annotation_df,\n",
    "                                             extend_dist = _dist, # distance to extend both direction\n",
    "                                             gene_coverage_type = 'tss', # distance to gene's coord of interest\n",
    "                                             key_added = None)\n",
    "\n",
    "loci_df_copy"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## save the annotated file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save the file \n",
    "loci_df_copy.to_csv(os.path.join(output_analysis_folder,'MERFISH_loci_adjacent_genes_tss_more_res.csv'))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The scripts above can be used to generate gene annotation as in the repository:\n",
    "\n",
    "[MERFISH_loci_annotation](../resources/MERFISH_loci_adjacent_genes_tss_more_res.csv)\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
